---
title: "Conserved non-coding elements"
author: 
 - "Charles Plessy"
date: "`r format(Sys.time(), '%d %B, %Y')`"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteEncoding{UTF-8}
  %\VignetteIndexEntry{Conserved non-coding elements}
  %\VignetteEngine{knitr::rmarkdown}
editor_options: 
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(cache = TRUE)
```

I am using the [CNEr](https://bioconductor.org/packages/release/bioc/html/CNEr.html)
package to detect the conserved non-coding elements in _Oikopleura_ genomes.

As a data source, I use I use the postmasked alignment files produced by the
pairwise alignment pipeline.  As `last-split` enforces a one-to-one relationship
this data is more stringent than the "nets" expected by _CNEr_.  Also, there
is no whole-genome duplication in the species we study.  Therefore I do not need
to run the realignment step detailed in _CNEr_'s vignette.

```{r prepare_CNEr}
library("CNEr")
library("GenomicBreaks")
library("GenomicFeatures")
genomes     <- OikScrambling:::loadAllGenomes(compat = FALSE)
annots      <- OikScrambling:::loadAllAnnotations()   |> suppressWarnings()
reps        <- OikScrambling:::loadAllRepeats(compat = FALSE)
```

Load the alignments in Axt format.

```{r prepare_axtfiles}
axt <- SimpleList()
readAxt_ <- function(file, genomeT, genomeQ=NULL, basename="/bucket/LuscombeU/common/Breakpoints/Alignments/Oidioi_pairwise_v3/") {
  if(is.null(genomeQ)) {
    axt <- readAxt(paste0(basename, file),
                   system.file("extdata", package=genomeT, "single_sequences.2bit"))
    seqinfo(first(axt)) <- seqinfo(getBSgenome(genomeT))
    second(axt) <- forceSeqLengths(second(axt))
  } else {
    axt <-  readAxt(paste0(basename, file),
                    system.file("extdata", package=genomeT, "single_sequences.2bit"),
                    system.file("extdata", package=genomeQ, "single_sequences.2bit"))
    seqinfo(first(axt))  <- seqinfo(getBSgenome(genomeT))
    seqinfo(second(axt)) <- seqinfo(getBSgenome(genomeQ))
  }
  axt
}

axt$Oki_Osa <- readAxt_("OKI2018_I69_1.0__OSKA2016v1.9.axt.gz", "BSgenome.Oidioi.OIST.OKI2018.I69", "BSgenome.Oidioi.OIST.OSKA2016v1.9")
axt$Oki_Bar <- readAxt_("OKI2018_I69_1.0__Bar2_p4.axt.gz",      "BSgenome.Oidioi.OIST.OKI2018.I69", "BSgenome.Oidioi.OIST.Bar2.p4")
axt$Oki_Kum <- readAxt_("OKI2018_I69_1.0__KUM-M3-7f.axt.gz",    "BSgenome.Oidioi.OIST.OKI2018.I69", "BSgenome.Oidioi.OIST.KUM.M3.7f")
axt$Oki_Aom <- readAxt_("OKI2018_I69_1.0__AOM-5-5f.axt.gz",     "BSgenome.Oidioi.OIST.OKI2018.I69", "BSgenome.Oidioi.OIST.AOM.5.5f")
axt$Oki_Nor <- readAxt_("OKI2018_I69_1.0__OdB3.axt.gz",         "BSgenome.Oidioi.OIST.OKI2018.I69", "BSgenome.Oidioi.genoscope.OdB3")
axt$Oki_Olo <- readAxt_("OKI2018_I69_1.0__O_lon.axt.gz",        "BSgenome.Oidioi.OIST.OKI2018.I69")
axt$Oki_Ova <- readAxt_("OKI2018_I69_1.0__O_van.axt.gz",        "BSgenome.Oidioi.OIST.OKI2018.I69")
axt$Oki_Oal <- readAxt_("OKI2018_I69_1.0__O_alb.axt.gz",        "BSgenome.Oidioi.OIST.OKI2018.I69")

axt$Osa_Oki <- readAxt_("OSKA2016v1.9__OKI2018_I69_1.0.axt.gz", "BSgenome.Oidioi.OIST.OSKA2016v1.9", "BSgenome.Oidioi.OIST.OKI2018.I69")
axt$Osa_Bar <- readAxt_("OSKA2016v1.9__Bar2_p4.axt.gz",         "BSgenome.Oidioi.OIST.OSKA2016v1.9", "BSgenome.Oidioi.OIST.Bar2.p4")
axt$Osa_Aom <- readAxt_("OSKA2016v1.9__AOM-5-5f.axt.gz",        "BSgenome.Oidioi.OIST.OSKA2016v1.9", "BSgenome.Oidioi.OIST.AOM.5.5f")

axt$Bar_Oki <- readAxt_("Bar2_p4__OKI2018_I69_1.0.axt.gz",      "BSgenome.Oidioi.OIST.Bar2.p4",      "BSgenome.Oidioi.OIST.OKI2018.I69")
```

```{r compute_CNEs}
GPairs2GBreaks <- function(GPairs, fix = TRUE) {
  if(isTRUE(fix))
    GPairs <- fixCoordinates(GPairs)
  gb <- GBreaks(target = granges(first(GPairs)), query = granges(second(GPairs)))
  strand(gb) <- strand(gb$query)
  strand(gb$query) <- "*"
  mcols(gb) <- cbind(mcols(gb), mcols(GPairs))
  sort(gb, ignore.strand = TRUE)
}

ceScan_to_GB <- function (axt, tFilter = NULL, qFilter = NULL, window, identity) {
  axt <- fixCoordinates(axt)
  cne <- ceScan(axt, tFilter = tFilter, qFilter = qFilter, window = window, identity = identity)
  GPairs2GBreaks(cne[[1]], fix = FALSE)
}

## CNEs are filtered against coding sequences
cne <- SimpleList()
cne$Oki_Osa <- ceScan_to_GB(axt$Oki_Osa, window = 50L, identity = 50L,
                            tFilter = exons(annots$Oki), qFilter = exons(annots$Osa))

# CE are not filtered against coding sequences
calculateCEs <- function(window, identity) {
  ce <- SimpleList()
  ce$Oki_Osa <- ceScan_to_GB(axt$Oki_Osa, window = window, identity = identity)
  ce$Oki_Bar <- ceScan_to_GB(axt$Oki_Bar, window = window, identity = identity)
  ce$Oki_Kum <- ceScan_to_GB(axt$Oki_Kum, window = window, identity = identity)
  ce$Oki_Aom <- ceScan_to_GB(axt$Oki_Aom, window = window, identity = identity)
  ce$Oki_Nor <- ceScan_to_GB(axt$Oki_Nor, window = window, identity = identity)
  ce$Oki_Olo <- ceScan_to_GB(axt$Oki_Olo, window = window, identity = identity)
  ce$Oki_Ova <- ceScan_to_GB(axt$Oki_Ova, window = window, identity = identity)
  ce$Oki_Oal <- ceScan_to_GB(axt$Oki_Oal, window = window, identity = identity)
  ce$Osa_Oki <- ceScan_to_GB(axt$Osa_Oki, window = window, identity = identity)
  ce$Osa_Bar <- ceScan_to_GB(axt$Osa_Bar, window = window, identity = identity)
  ce$Osa_Aom <- ceScan_to_GB(axt$Osa_Aom, window = window, identity = identity)
  ce$Bar_Oki <- ceScan_to_GB(axt$Bar_Oki, window = window, identity = identity)
  ce
}

ces <- SimpleList()
ces$ce_200_200 <- calculateCEs(200L, 200L)
ces$ce_200_198 <- calculateCEs(200L, 198L)
ces$ce_200_196 <- calculateCEs(200L, 196L)
ces$ce_200_194 <- calculateCEs(200L, 194L)
ces$ce_100_100 <- calculateCEs(100L, 100L)
ces$ce_100__99 <- calculateCEs(100L, 99L)
ces$ce_100__98 <- calculateCEs(100L, 98L)
ces$ce_100__97 <- calculateCEs(100L, 97L)
ces$ce_100__96 <- calculateCEs(100L, 96L)
ces$ce_100__95 <- calculateCEs(100L, 95L)
ces$ce__50__50 <- calculateCEs(50L, 50L)
ces$ce__50__49 <- calculateCEs(50L, 49L)
ces$ce__50__48 <- calculateCEs(50L, 48L)
ces$ce__50__47 <- calculateCEs(50L, 47L)
sapply(ces, sapply, length)

saveRDS(ces, "CEs.Rds")
```

Export the results for supplemental material.

```{r export_to_BED}
dir.create("CEs_BED", showWarnings = FALSE)
export_CEs_to_BED <- function(gr, thresholdname, genomepairname) {
  x <- ces[[thresholdname]][[genomepairname]]
  names(x) <- x$cigar
  rtracklayer::export(x, paste0("CEs_BED/", thresholdname, "_", genomepairname, ".bed"))
}
for (thresholdname in names(ces))
  for (genomepairname in names(ces[[thresholdname]]))
    export_CEs_to_BED(ces, thresholdname, genomepairname)
```
