---
title: "Regions interrupting colinearity"
author: 
 - "Charles Plessy"
date: "`r format(Sys.time(), '%d %B, %Y')`"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteEncoding{UTF-8}
  %\VignetteIndexEntry{Regions interrupting colinearity}
  %\VignetteEngine{knitr::rmarkdown}
editor_options: 
  chunk_output_type: console
---

# Load pacakges and data

See the `vignette("OikScrambling", package = "OikScrambling")` for general
details on package and data load.

See `vignette("LoadGenomicBreaks", package = "OikScrambling")` for how the
different `GBreaks` objects are prepared.

```{r load_pacakges_and_data}
suppressPackageStartupMessages({
  library('GenomicBreaks')
  library('ggplot2')
  library("BreakpointsData")
})
genomes <- OikScrambling:::loadAllGenomes(compat = FALSE)
load("BreakPoints.Rdata")
reps <- OikScrambling:::loadAllRepeats()
transcripts <- OikScrambling:::loadAllTranscriptsGR()
```

# The problem:

The plot below shows high colinearity between Oki and Kum, which are from the
same species.  It also shows that the `GenomicBreaks` objects are still
scattered in small pieces.  What is interrupting colinearity?

```{r parallel_chr_plots_Oki_Kum}
plotApairOfChrs(coa$Oki_Kum, "chr1", main = "Oki – Kum")
```

In this vignette, we look at a couple of example regions in details.  These
regions helped me to decide how to process the data in _Load Genomic Breaks_
(`vignette("LoadGenomicBreaks", package = "OikScrambling")` and to decide
which analysis to prioritise elsewhere.

See `vignette("RegionsOfInterest", package = "OikScrambling")` for other regions
of interest.

# Overview of Kume's chromosome 1.

The best match for chr1 in the Kume assembly is `contig_3_1`.  It maps roughly
in the 8,000,000 – 13,000,000 region.

```{r Kum_chr1_assembly_1}
bestMatch <- function(gb)
  tapply(width(gb$query), seqnames(gb$query), sum) |> sort() |> tail(1) |> names()

coa$Oki_Kum |> plyranges::filter(seqnames == "chr1") |> bestMatch()
```

The next best match, `contig_27_1` is most of chr1's short arm.

```{r Kum_chr1_assembly_2}
coa$Oki_Kum |> plyranges::filter(seqnames == "chr1", seqnames(query)!="contig_3_1") |> bestMatch()
coa$Oki_Kum |> plotApairOfChrs(chrQ = "contig_27_1")
```

The next best match, `contig_90_1` is assembled in reverse direction, and contains
an inversion.

```{r Kum_chr1_assembly_3}
coa$Oki_Kum |> plyranges::filter(seqnames == "chr1", ! seqnames(query) %in% c("contig_3_1", "contig_27_1")) |> bestMatch()
coa$Oki_Kum |> plotApairOfChrs(chrQ = "contig_90_1")
coa$Oki_Kum |> swap() |> reverse () |> swap () |> plotApairOfChrs(chrQ = "contig_90_1")
```

The assembly is dones in details in `vignette("ParallelPlots", package = "OikScrambling")`

# Alignments are real (Region of Interest 1)

When an alignment interrupts colinearity, is it trustable ?

Let's check the second alignment pair, mapping 226 bases of `contig_3_1` to the
short arm.

```{r re_align_a_clolinearity_breaker}
ROI1 <- coa$Oki_Kum |>
  plyranges::filter(seqnames(query) == "contig_3_1", end(query) <= 21079) |>
  plyranges::arrange(start(query))
ROI1
coa$Oki_Kum |> plotApairOfChrs("chr1", "contig_3_1", xlim = gb2xlim(ROI1[-2]), main = "chr1 vs contig_3_1 at ROI 1")
coa$Oki_Kum |> plotApairOfChrs("chr1", "contig_3_1", xlim = list(c(3838658, 3881203), c(1, 21079)), main = "chr1 matching contig_3_1 in the short arm")
```

Is the alignment convincing ?

```{r pairwise_seq_alignments}
# Good match where reported
pairwiseAlignment(getSeq(genomes$KUM.M3.7f.M3.7f, GRanges("contig_3_1:16561-16812:-")), genomes$OKI2018.I692018.I69$chr1[3870099:3870324], type="local")

# 50 bp of flanking regions added to contrast
pairwiseAlignment(getSeq(genomes$KUM.M3.7f, GRanges("contig_3_1:16561-16812:-") + 50), genomes$OKI2018.I69$chr1[(3870099 -50):(3870324 + 50)], type="global") |> writePairwiseAlignments()

# No good match between the flanking pairs.
pairwiseAlignment(getSeq(genomes$KUM.M3.7f, GRanges("contig_3_1:16561-16812:+")), genomes$OKI2018.I69$chr1[8086429:8104950], type="local")
pairwiseAlignment(getSeq(genomes$KUM.M3.7f, GRanges("contig_3_1:16561-16812:-")), genomes$OKI2018.I69$chr1[8086429:8104950], type="local")
```

Yes, the alignment looks real!

# Colinearity interrupted by inversions and translocations (Region of Interest 2)

Let's look at the area `chr1:8080000-12900000` covered by `contig_3_1`.  Can
we coalesce it once we only keep the paired matches?  The answer is no…

```{r ROI2}
coa$Oki_Kum |> flagAll() |> plyranges::filter(seqnames == "chr1", start > 8080000, end < 12900000)

# Even removing all but the best match on the query is not enough to collapse all
coa$Oki_Kum |>
  plyranges::filter(seqnames == "chr1", start > 8080000, end < 12900000) |>
  plyranges::filter(seqnames(query) == "contig_3_1") |>
  coalesce_contigs() |> flagAll() |> head(10)

coa$Oki_Kum |>
  plyranges::filter(seqnames == "chr1", start > 8080000, end < 12900000) |>
  plyranges::filter(seqnames(query) == "contig_3_1")|>
  coalesce_contigs() |>  plotApairOfChrs()
```

Colinearity appears to be interrupted by many inversions and translocations.

# Well-aligned repeats are kept in the one-to-one graph (Region of interest 3)

```{r ROI3}
gbs$Oki_Kum[1] |> as.data.frame()
pairwiseAlignment(genomes$KUM.M3.7f$contig_10_1[52244:60120], genomes$OKI2018.I69$chr1[7785:15657], type="global")
subsetByOverlaps(reps$Oki, gbs$Oki_Kum[1])
```


# What still interrupts colinearity once repeats are ruled out ? (ROI5)

We start with a random region

```{r ROI5_overview}
coa$Oki_Kum |> flagAll() |> plyranges::slice(200:210) |> as.data.frame()
```

Let's focus on alignments between chr1 and contig_27_1:

```{r ROI5}
ROI5.range <- coa$Oki_Kum[200:210] |> dplyr::filter(seqnames == "chr1", seqnames(query) == "contig_27_1") |> range()
ROI5.range

ROI5 <- subsetByOverlaps(coa$Oki_Kum, ROI5.range) |> sort(ignore.strand=T)

# Let's capture all the pairs that overlap that range:
ROI5 |> sort(ignore.strand=T) |> as.data.frame()
ROI5 |> dplyr::arrange(query) |> as.data.frame()
ROI5 |> plotApairOfChrs(xlim = gb2xlim(ROI5.range))
```

Very short gaps between successive ranges in ROI5

```{r ROI5_cleanGaps}
cleanGaps(ROI5) |> as.data.frame()

ROI5 |> flagAll() |> dist2next( ignore.strand = TRUE) |> as.data.frame()
ROI5 |> swap() |> sort(ignore.strand = TRUE) |> flagAll()|> dist2next( ignore.strand = TRUE, step =2)  |> as.data.frame()
```

Can we coalesce better by removing the translocations ?

```{r ROI5_removeTranslocations}
removeTranslocations <- function(gb) {
  gb <- flagTranslocations(gb)
  if(any(gb$tra))
    gb <- gb[-(which(gb$tra) + 1)]
  gb
}

ROI5 |> removeTranslocations() |> coalesce_contigs() |>
  swap() |> sort(i=T) |> removeTranslocations() |> coalesce_contigs() |>
  swap() |> sort(i=T) |> plotApairOfChrs()
```

This is now implemented in the `coa2` objects, with a size treshold of 200.

```{r ROI5_coa2}
subsetByOverlaps(coa2$Oki_Kum, ROI5.range) |> plotApairOfChrs(xlim = gb2xlim(ROI5.range))
```

# What still interrupts colinearity once translocations are removed ? (ROI6)

Inversions …

```{r ROI6}
# We start with a broader random region
(ROI6 <- coa2$Oki_Kum |> flagAll() |> dplyr::slice(160:200))
ROI6_range <- ROI6 |> plyranges::filter(seqnames(query) == "contig_27_1") |> range()
coa2$Oki_Kum |> plotApairOfChrs(chrT = "chr1", chrQ = "contig_27_1", xlim = gb2xlim(ROI6_range), main = "Double-collapsed collinear regions")
coa $Oki_Kum |> plotApairOfChrs(chrT = "chr1", chrQ = "contig_27_1", xlim = gb2xlim(ROI6_range), main = "Collinear regions")
gbs $Oki_Kum |> plotApairOfChrs(chrT = "chr1", chrQ = "contig_27_1", xlim = gb2xlim(ROI6_range), main = "Alignments")
```

Example of a part of ROI6 that can be collapsed if strand is ignored.

```{r ROI6_invs2}
ROI6[1:6] |> plotApairOfChrs()
ROI6[1:6] |> plyranges::mutate(strand = '*') |> coalesce_contigs() |> plotApairOfChrs()

ROI6[13:23] |> plotApairOfChrs()
ROI6[13:23] |> plyranges::mutate(strand = '*') |> coalesce_contigs() |> plotApairOfChrs()
```

# Filling inversions to better see the rest (ROI9)

We will not fill inversions in objects we analyse, because we do not want
to erase their breakpoints.

```{r}
fillInversions <- function(gb) {
  Invs <- which(flagInversions(gb)$inv) + 1
  strand(gb)[Invs] <- ifelse(strand(gb)[Invs] == "+", "-", "+")
  coalesce_contigs(gb)
}

ROI9.all <- fillInversions(coa2$Oki_Kum)

ROI9 <- ROI9.all |>
  plyranges::filter(seqnames(query) == "contig_3_1") |>
  plyranges::arrange(start(query)) |>
  plyranges::slice(10:20)

ROI9.range <- range(ROI9)[1]

plotApairOfChrs(ROI9)
plotApairOfChrs(ROI9.all, xlim = gb2xlim(ROI9.range))

subsetByOverlaps(ROI9.all, granges(ROI9.range)) |> flagAll() |> as.data.frame()

subsetByOverlaps(ROI9.all |> swap(), ROI9.range$query)
```


```{r}

x <- rbind(
  sapply(gbs, \(gb) table(flagPairs(gb)$pairs))
)


```

# ROI 10: small unaligned inversion in Oki-Kum alignment

## The small unaligned inversion

Martin reported a small unaligned inversion at PAR:80769-81032 in the Oki – Kum
alignment (version 3).  Note that it will not be visible on the plot if this
vignette is rebuilt with an alignment updated to include these inversions.

```{r ROI10}
ROI10 <- GRanges("PAR:80769-81032:-")
ROI10_inv <- subsetByOverlaps(gbs$Oki_Kum, ROI10 + 1e2)

ROI10_inv |> plotApairOfChrs(main = "A short unaligned region that is actually an inversion within the inversion.")
ROI10_inv       |> cleanGaps() |> as.character()
ROI10_inv$query |> cleanGaps() |> as.character()
```

Interestingly, this small inversion is flanked by an inverted repeat.

```{r ROI10_pairwise}
pairwiseAlignment(getSeq(genomes$OKI2018.I69, GRanges("PAR:80780-81022"        ) + 200 ),
                  getSeq(genomes$KUM.M3.7f, GRanges("contig_57_1:58108-58350") + 200)) |> writePairwiseAlignments()

pairwiseAlignment(getSeq(genomes$OKI2018.I69, GRanges("PAR:80780-81022")         + 200 ),
                  getSeq(genomes$KUM.M3.7f, GRanges("contig_57_1:58108-58350") + 200) |> reverseComplement()) |> writePairwiseAlignments()

# (Adjusted by hand)
pairwiseAlignment(getSeq(genomes$OKI2018.I69, GRanges("PAR:80523-80834")),
                  getSeq(genomes$OKI2018.I69, GRanges("PAR:80968-81279")) |> reverseComplement()) |> writePairwiseAlignments()

getSeq(genomes$OKI2018.I69, GRanges("PAR:80523-80834")) |> as.character()
```

This sequence matches `MITE_oki2018_i69_4_20065_S1_MITE#DNA/MITE` in Sasha's
repeat annotation.

```{r search_MITE}
vmatchPattern(genomes$OKI2018.I69$PAR[80523:80834], genomes$OKI2018.I69, with.indels = TRUE, max.mismatch = 20) |> sort(ignore.strand = TRUE) |> as.data.frame()
```

## Other interesting things in the area

Expanding the area broader, there is this interesting pattern where there is
only one gap on the _Oki_ side and two gaps on the _Kum_ side.  This is because
the alignments on the _Oki_ side are directly contacting each other.

```{r ROI10_insertion}
ROI10_ins <- subsetByOverlaps(gbs$Oki_Kum, ROI10 + 3e3)
ROI10_ins |> plotApairOfChrs()
ROI10_ins |> as.data.frame()
ROI10_ins |> cleanGaps() |> as.data.frame()
ROI10_ins$query |> cleanGaps() |> as.data.frame()

tail(ROI10_ins, 2) |> plotApairOfChrs(main = "An insertion in the Kume genome")
tail(ROI10_ins, 2) |> swap() |> cleanGaps()

# Left side
pairwiseAlignment(getSeq(genomes$OKI2018.I69, GRanges("PAR:83350-83550")),
                  getSeq(genomes$KUM.M3.7f, GRanges("contig_57_1:55548-55748")) |> reverseComplement(),
                  type = "global") |> writePairwiseAlignments()

# Right side
pairwiseAlignment(getSeq(genomes$OKI2018.I69, GRanges("PAR:83351-83551")),
                  getSeq(genomes$KUM.M3.7f, GRanges("contig_57_1:55236-55436")) |> reverseComplement(),
                  type = "global") |> writePairwiseAlignments()

# Both sides
pairwiseAlignment(getSeq(genomes$OKI2018.I69, GRanges("PAR:83350-83551")),
                  getSeq(genomes$KUM.M3.7f, GRanges("contig_57_1:55236-55748")) |> reverseComplement(),
                  type = "global") |> writePairwiseAlignments()

getSeq(genomes$KUM.M3.7f, GRanges("contig_57_1:55337-55647:+")) |> as.character()
```

Exemplar from the Okinawa genome (found by BLASTing the sequence output by
`getSeq` above).

```
>RepeatInsertion
AAgggggggCGGGCGGCCACGGTACAAAAGTGGAATTTGTACCGCGGGTACAAATTTTGT
ACCGCCGGTACAAAAAATCGCAATTGTTTTGACAATAAAAACCGATTTCTACCCATTATA
ATGCTCACTTTTGAACAGAACTCCACCTCAGCTATACCAGTAAGGAGTTTCCATCAATTC
TAGCGCGAAGCGCACACTTTTTCCTTGATTCTATTTTTGTCTTGTACCGGCGGTACAAAA
ATTTTTGTACCGGCGTTACAAATTTGTACCGCCGTTACAAATTTGTACCGTCGCCGCCCT
ACACCC
```
