---
title: "Distance Measures"
author: 
 - "Charles Plessy"
date: "`r format(Sys.time(), '%d %B, %Y')`"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteEncoding{UTF-8}
  %\VignetteIndexEntry{Distance Measures}
  %\VignetteEngine{knitr::rmarkdown}
editor_options: 
  chunk_output_type: console
---

# Introduction

We want to put numbers on visual impressiosn such as:

 - There is a lot of scrambling.
 - Aligned regions stay on their chromosome arm.
 - Aligned regions are more "off the diagonal" on short arms than on long arms.
 - â€¦

# Load pacakges

## Core packages that provide functions we use a lot.

```{r loadlib, echo=T, results='hide'}
suppressPackageStartupMessages({
  library('GenomicBreaks')
  library('BSgenome')
  library('GenomicFeatures')
  library('ggplot2')
  library("BiocParallel")
})
```

## BSgenome packages

```{r load_BSgenome_packages}
# These will need to be installed by the user
library("BSgenome.Odioica.local.OKI2018.I69")
library("BSgenome.Odioica.local.OSKA2016v1.9")
library("BSgenome.Odioica.local.Bar2.p4")
library("BSgenome.Odioica.local.KUM.M3")
library("BSgenome.Odioica.local.AOM.5")
library("BSgenome.Odioica.local.Odioica.reference.v3.0")
```

For convenience (such as easy looping, and tab-completion to show the available
objects), we create a `genomes` list containing all the genomes.

```{r genomes_object}
genomes <- SimpleList(
  # Chromosome assemblies
  Oki = OKI2018_I69, Osa = OSKA2016v1.9, Bar = Bar2_p4,
  # Less contiguous assemblies
  Kum = KUM_M3, Aom = AOM_5, Nor = OdB3)
```

# Load data

## Genome annotations.

The annotations from `/bucket/LuscombeU/common/Breakpoints/Annotations` are
distributed in the same drat repository as the alignment data.

```
install.packages("BreakpointsData", repos="https://oist.github.io/plessy_oikgenomes_drat/")
```

```{r load_annotations, warning=FALSE}
annots <- SimpleList()

gff2txdb <- function(file, genome) {
  file <- system.file(paste0("extdata/Annotations/", file), package = "BreakpointsData")
  tx <- rtracklayer::import.gff(file)
  tx <- tx[seqnames(tx) %in% seqnames(genome)]
  tx <- GRanges(tx, seqinfo = seqinfo(genome))
  tx <- GenomicFeatures::makeTxDbFromGRanges(tx)
}

gff2txdb_Norway <- function(file, genome) {
  file <- system.file(paste0("extdata/Annotations/", file), package = "BreakpointsData")
  tx <- rtracklayer::import.gff(file)
  tx <- tx[!is.na(tx$mRNA)]
  # Remove "scaffoldA" objects in the OdB3 annotation
  tx <- tx[seqnames(tx) %in% seqnames(genome)]
  tx$ID <- tx$mRNA
  tx <- GRanges(tx, seqinfo = seqinfo(genome))
  tx <- GenomicFeatures::makeTxDbFromGRanges(tx)
}

annots$Oki <- gff2txdb("OKI2018_I69.v2/OKI2018_I69.v2.gm.gff.gz",  OKI2018_I69)
annots$Osa <- gff2txdb("OSKA2016v1.9/OSKA2016v1.9.gm.gff.gz",      OSKA2016v1.9)
annots$Bar <- gff2txdb("Bar2_p4.Flye/Bar2_p4.Flye.gm.gff.gz",      Bar2_p4)
annots$Kum <- gff2txdb("KUM-M3-7f/KUM-M3-7f.gm.gff.gz",            KUM_M3)
annots$Aom <- gff2txdb("AOM-5-5f/AOM-5-5f.gm.gff.gz",              AOM_5)
annots$Nor <- gff2txdb_Norway("OdB3/Oikopleura_annot_v1.0.gff.gz", OdB3)
```

## GenomicBreaks objects

```{r GBreaks_object}
load("BreakPoints.Rdata")
gbs$Oki_Osa
gbs$Oki_Osa$query

longShort
longShort$Oki
```

# Synteny index

Ad-hoc index measuring to what extent a scaffold of the _target_ genome is
mostly aligned to a single scaffold in the _query_ genome.   See
`?GenomicBreaks::synteny_index` for details.  One limitation to the use of this
index is that it requires that at least the _query_ genome is a complete
chromosome assembly.

```{r synteny_index}
pairs2table6x3 <- function (x) {
  x <- as.list(x)
  cbind(
    Oki = c(Oki = NA,        Osa = x$Osa_Oki, Bar = x$Bar_Oki),
    Kum = c(Oki = x$Oki_Kum, Osa = x$Osa_Kum, Bar = x$Bar_Kum),
    Osa = c(Oki = x$Oki_Osa, Osa = NA,        Bar = x$Bar_Osa),
    Aom = c(Oki = x$Oki_Aom, Osa = x$Osa_Aom, Bar = x$Bar_Aom),
    Bar = c(Oki = x$Oki_Bar, Osa = x$Osa_Bar, Bar = NA       ),
    Nor = c(Oki = x$Oki_Nor, Osa = x$Osa_Nor, Bar = x$Bar_Nor)
  )
}

pairs2Kable6x3 <- function (df) df |> pairs2table6x3() |> round(2) |> knitr::kable()

sapply(gbs, synteny_index) |> pairs2Kable6x3()
sapply(gbs, \(g) swap(g) |> synteny_index()) |> pairs2Kable6x3()
```

# Correlation

Ad-hoc index measuring the correlation of the coordinates of the syntenic
alignments in scaffolds of a _target_ genome and their best match in the
_query_ genome.  See `?GenomicBreaks::correlation_index` for details.  It is
intended to be more robust to the presence of uncollapsed haplotypes in the
_query_ genome but probably needs further testing.

```{r correlation_index}
sapply(gbs, correlation_index) |> pairs2Kable6x3()
sapply(gbs, \(g) swap(g) |> correlation_index()) |> pairs2Kable6x3()
```

# Gene Order Conservation

See `?GOC` for details.

```{r goc_index}
sapply(gbs, GOC) |> pairs2Kable6x3()
sapply(gbs, \(g) swap(g) |> GOC()) |> pairs2Kable6x3()
```
