---
title: "Exploring coding-sequence substitutions"
author: 
 - "Michael Mansfield"
date: "`r format(Sys.time(), '%d %B, %Y')`"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{Exploring coding-sequence substitutions}
  %\VignetteEncoding{UTF-8}
  %\VignetteEngine{knitr::rmarkdown}
editor_options: 
  chunk_output_type: console
  markdown: 
    wrap: 72
---

```{r setup}
knitr::opts_knit$set(verbose = TRUE)
```

# Introduction

This vignette loads dN/dS statistics tables and explores the
relationship between substitutions in coding sequences and genomic
breaks.

This vignette assumes that prior vignettes have been explored by the
user previously and that the `GenomicBreaks` and `OikScrambling`
packages have been installed already.

# Packages and data

```{r load_pacakges_and_data}
suppressPackageStartupMessages({
  library("tibble")
  library("dplyr")
  library("GenomicBreaks")
  library("ggplot2")
  library("BreakpointsData")
  library("ggpubr")
})

(genomes <- OikScrambling:::loadAllGenomes())
(transcripts <- OikScrambling:::loadAllTranscriptsGR())

load("BreakPoints.Rdata")
```

# Genomic breaks
The purpose of this vignette is to explore the various relationships of dN/dS - i.e., 
coding sequence substitution rate - to synteny. The input to this vignette is similar
to what is produced in the `vignette("OrthogroupSyntenies", package = "OikScrambling")`.

The two important objects are `orthoPairs` and `orthoBreaks`, containing the orthologue-
based `GBreaks` objects and coalesced `GBreaks` objects, respectively.

The steps to produce `orthoPairs` and `orthoBreaks` are reproduced below.

```{r getting_orthopairs}
orthoPairToGBreaks <- function(name1, name2, annot1, annot2) {
  orthoPairs <- load_one_to_ones(
    system.file("extdata/OrthoFinder/N19.tsv", package = "BreakpointsData"),
    c(name1, name2))

  IDs2GRanges <- function (IDs, annot) {
    prefix <- Biobase::lcPrefix(IDs)                  # Guess prefix in transcript IDs from HOG files
    not_prefix <- Biobase::lcPrefix(annot$tx_name)    # Remove trailing characters that are part of the name
    not_prefix <- paste0(not_prefix,"$")              # Anchor to end of the string
    prefix <- sub(not_prefix, "", prefix)             # Finalise prefix
    IDs <- sub(prefix, "", IDs)                       # Remove prefix from IDS
    names(annot) <- annot$tx_name                     # Store transcript name in names slot
    gr <- annot[IDs]                                  # Sort by ID
    strand(gr) <- "*"                                 # Make strandless
    gr                                                # Return the object
  }
  
  gb       <- IDs2GRanges(orthoPairs[,name1], annot1)
  gb$query <- IDs2GRanges(orthoPairs[,name2], annot2)
  gb <- GenomicBreaks:::GBreaks(gb)
  sort(gb)
}

orthoPairs <- SimpleList()
orthoPairs$Oki_Osa <- orthoPairToGBreaks("OKI2018_I69.v2.prot.longest.fa_1", "OSKA2016v1.9.prot.longest.fa_1"
                                        , transcripts$Oki, transcripts$Osa)
orthoPairs$Oki_Bar <- orthoPairToGBreaks("OKI2018_I69.v2.prot.longest.fa_1", "Bar2_p4.Flye.prot.longest.fa_1"
                                        , transcripts$Oki, transcripts$Bar)
orthoPairs$Oki_Kum <- orthoPairToGBreaks("OKI2018_I69.v2.prot.longest.fa_1", "KUM-M3-7f.prot.longest.fa_1"
                                        , transcripts$Oki, transcripts$Kum)
orthoPairs$Oki_Aom <- orthoPairToGBreaks("OKI2018_I69.v2.prot.longest.fa_1", "AOM-5-5f.prot.longest.fa_1"
                                        , transcripts$Oki, transcripts$Aom)
orthoPairs$Oki_Nor <- orthoPairToGBreaks("OKI2018_I69.v2.prot.longest.fa_1", "OdB3.v1.0.prot.fa_1.nohaplo"
                                        , transcripts$Oki, transcripts$Nor)

orthoPairs$Osa_Bar <- orthoPairToGBreaks("OSKA2016v1.9.prot.longest.fa_1", "Bar2_p4.Flye.prot.longest.fa_1"
                                        , transcripts$Osa, transcripts$Bar)
orthoPairs$Osa_Oki <- orthoPairToGBreaks("OSKA2016v1.9.prot.longest.fa_1", "OKI2018_I69.v2.prot.longest.fa_1"
                                        , transcripts$Osa, transcripts$Oki)
orthoPairs$Osa_Kum <- orthoPairToGBreaks("OSKA2016v1.9.prot.longest.fa_1", "KUM-M3-7f.prot.longest.fa_1"
                                        , transcripts$Osa, transcripts$Kum)
orthoPairs$Osa_Aom <- orthoPairToGBreaks("OSKA2016v1.9.prot.longest.fa_1", "AOM-5-5f.prot.longest.fa_1"
                                        , transcripts$Osa, transcripts$Aom)
orthoPairs$Osa_Nor <- orthoPairToGBreaks("OSKA2016v1.9.prot.longest.fa_1", "OdB3.v1.0.prot.fa_1.nohaplo"
                                        , transcripts$Osa, transcripts$Nor)


orthoPairs$Bar_Osa <- orthoPairToGBreaks("Bar2_p4.Flye.prot.longest.fa_1", "OSKA2016v1.9.prot.longest.fa_1"
                                        , transcripts$Bar, transcripts$Osa)
orthoPairs$Bar_Oki <- orthoPairToGBreaks("Bar2_p4.Flye.prot.longest.fa_1", "OKI2018_I69.v2.prot.longest.fa_1"
                                        , transcripts$Bar, transcripts$Oki)
orthoPairs$Bar_Kum <- orthoPairToGBreaks("Bar2_p4.Flye.prot.longest.fa_1", "KUM-M3-7f.prot.longest.fa_1"
                                        , transcripts$Bar, transcripts$Kum)
orthoPairs$Bar_Aom <- orthoPairToGBreaks("Bar2_p4.Flye.prot.longest.fa_1", "AOM-5-5f.prot.longest.fa_1"
                                        , transcripts$Bar, transcripts$Aom)
orthoPairs$Bar_Nor <- orthoPairToGBreaks("Bar2_p4.Flye.prot.longest.fa_1", "OdB3.v1.0.prot.fa_1.nohaplo"
                                        , transcripts$Bar, transcripts$Nor)

sapply(orthoPairs, length)
orthoPairs[ 1:5 ] <- sapply(orthoPairs[ 1:5 ],  flagLongShort, longShort$Oki)
orthoPairs[ 6:10] <- sapply(orthoPairs[ 6:10],  flagLongShort, longShort$Osa)
orthoPairs[11:15] <- sapply(orthoPairs[11:15],  flagLongShort, longShort$Bar)
```

The `orthoPairs` object contains a `GRanges` object for every species pair,
where each entry has a `GBreaks` object that contains dN/dS metadata columns. 

This object is sufficient to start exploring the relationship of dN/dS to 
synteny. First, plot global patterns of dN/dS between chromosomes and chromosome
arms.

```{r orthopairs_dNdS_oki_1, fig.height=4, fig.width=6, dev=c("svg", "png")}
ggplot(as.data.frame(orthoPairs$Oki_Osa)) +
  aes(dNdS_GUIDANCE2, fill=Arm) +
  geom_boxplot() +
  coord_flip() +
  facet_wrap(~seqnames) +
  ggtitle("Oki/Osa: dN/dS by chromosome and chromosome arm")
```

Notice that an unplaced scaffold contains a gene that was used for the dN/dS analysis. Let's filter those out.

```{r orthopairs_dNdS_oki_2, fig.height=4, fig.width=6, dev=c("svg", "png")}
ggplot(as.data.frame(orthoPairs$Oki_Osa) %>% filter(seqnames %in% c("chr1", "chr2", "PAR", "XSR", "YSR"))) +
  aes(dNdS_GUIDANCE2, fill=Arm) +
  geom_boxplot() +
  coord_flip() +
  facet_wrap(~seqnames) +
  xlab("dN/dS (GUIDANCE2)") + 
  ggtitle("Okinawa: dN/dS by chromosome and chromosome arm")
```

Let's enumerate the rest of the combinations. 
```{r orthopairs_dNdS_osa, fig.height=4, fig.width=6, dev=c("svg", "png")}
ggplot(as.data.frame(orthoPairs$Osa_Oki) %>% filter(seqnames %in% c("Chr1", "Chr2", "PAR", "XSR", "YSR"))) +
  aes(dNdS_GUIDANCE2, fill=Arm) +
  geom_boxplot() +
  coord_flip() +
  facet_wrap(~seqnames) +
  xlab("dN/dS (GUIDANCE2)") + 
  ggtitle("Osaka: dN/dS by chromosome and chromosome arm")
```

```{r orthopairs_dNdS_bar}
ggplot(as.data.frame(orthoPairs$Bar_Oki) %>% filter(seqnames %in% c("Chr1", "Chr2", "PAR", "XSR", "YSR"))) +
  aes(dNdS_GUIDANCE2, fill=Arm) +
  geom_boxplot() +
  coord_flip() +
  facet_wrap(~seqnames) +
  xlab("dN/dS (GUIDANCE2)") + 
  ggtitle("Barcelona: dN/dS by chromosome and chromosome arm")
```

For the rest of the vignette, we will be relating dN/dS values to synteny
in more specific ways. To do that, we'll need the coalesced `GBreaks` object.

```{r getting_orthobreaks, fig.height=4, fig.width=6, dev=c("svg", "png")}
transcripts$Bar |> length()
transcripts$Bar |> reduce(min = 0) |> length()

flagOverlaps <- function(gr) {
  # Find overlaps
  ov <- findOverlaps(gr)
  # Index of self hits in the gr object
  idx <- queryHits(ov[!isSelfHit(ov)])
  # Convert to Boolean indexing the gr object
  seq_along(gr) %in% idx
}

transcripts$Bar[flagOverlaps(transcripts$Bar)]

# Show examples
orthoPairs$Oki_Osa[flagOverlaps(orthoPairs$Oki_Osa)]
orthoPairs$Oki_Osa$query[flagOverlaps(orthoPairs$Oki_Osa$query)]

# Remove and show how much was removed

orthoPairsFiltered <- sapply(orthoPairs, function(gb) {
  gb <- gb[!flagOverlaps(gb)]
  gb <- gb[!flagOverlaps(gb$query)]
  gb
}) |> SimpleList()
sapply(orthoPairs, length) - sapply(orthoPairsFiltered, length)

coalOrtho <- function(gb) {
  # Coalesce
  coal <- coalesce_contigs(gb)
  
  # Map annotations to the coalesced blocks
  ov <- findOverlaps(coal, gb)
  
  # Extract gene names mapped to each blocks
  coal$geneNames <- CharacterList(split(names(gb)[subjectHits(ov)], queryHits(ov))) |> unname()
  # We can use the `ov` object for query ranges as well because of the way `coal` was constructed.
  coal$query$geneNames <- CharacterList(split(names(gb$query)[subjectHits(ov)], queryHits(ov))) |> unname()
  
  # Extract dNdS values
  coal$dNdS_GUIDANCE2  <- NumericList(split(gb$dNdS_GUIDANCE2 [subjectHits(ov)], queryHits(ov)))
  coal$dNdS_HmmCleaner <- NumericList(split(gb$dNdS_HmmCleaner[subjectHits(ov)], queryHits(ov)))
  coal$dNdS_PRANK      <- NumericList(split(gb$dNdS_PRANK     [subjectHits(ov)], queryHits(ov)))
  # We just copy dNdS values to query ranges because they are the same.
  coal$query$dNdS_GUIDANCE2  <- coal$dNdS_GUIDANCE2
  coal$query$dNdS_HmmCleaner <- coal$dNdS_HmmCleaner
  coal$query$dNdS_PRANK      <- coal$dNdS_PRANK

  # Count genes per syntenic block
  coal$geneNumber <- sapply(coal$geneNames, length)
  
  coal
}

orthoBlocks <- sapply(orthoPairsFiltered, coalOrtho) |> SimpleList()
sapply(orthoBlocks, length)
orthoBlocks[ 1:5 ] <- sapply(orthoBlocks[ 1:5 ],  flagLongShort, longShort$Oki)
orthoBlocks[ 6:10] <- sapply(orthoBlocks[ 6:10],  flagLongShort, longShort$Osa)
orthoBlocks[11:15] <- sapply(orthoBlocks[11:15],  flagLongShort, longShort$Bar)
```

## Relating dN/dS values to synteny
The `orthoBlocks` object contains coalesced alignments based on orthologous protein sequences. This is therefore suitable to examine more specific relationships between dN/dS and synteny.

First, let's explore some of the properties of the `orthoBlocks` object, and how they relate to dN/dS.

```{r exploring_dnds_1, fig.height=4, fig.width=6, dev=c("svg", "png")}
# Genes per syntenic block
plot1 <- ggplot(as.data.frame(orthoBlocks$Oki_Osa), aes(x=geneNumber )) +
  geom_histogram() +
  ggtitle("Oki/Osa: genes per syntenic block")

plot2 <- ggplot(as.data.frame(orthoBlocks$Oki_Osa), aes(x=unlist(lapply(dNdS_GUIDANCE2, function(x) median(x, na.rm=T))) )) +
  geom_histogram() +
  ggtitle("Oki/Osa: dN/dS") + xlab("dN/dS")

# Genes per syntenic block, but on a logarithmic scale
plot3 <- ggplot(as.data.frame(orthoBlocks$Oki_Osa), aes(x=geneNumber )) +
  geom_histogram() +
  scale_y_continuous(trans="log10") +
  ggtitle("Oki/Osa: genes per syntenic block (log10 axis)")

plot4 <- ggplot(as.data.frame(orthoBlocks$Oki_Osa), aes(x=unlist(lapply(dNdS_GUIDANCE2, function(x) median(x, na.rm=T))) )) +
  geom_histogram() +
  scale_y_continuous(trans="log10") +
  ggtitle("Oki/Osa: dN/dS (log10 axis)") + xlab("dN/dS")

ggarrange(plot1, plot2, plot3, plot4)

```

How do they relate to one another?

```{r exploring_dnds_2, fig.height=4, fig.width=6, dev=c("svg", "png")}
# Boxplots - distribution of dN/dS, divided by the number of genes per block
plot1 <- ggplot(as.data.frame(orthoBlocks$Oki_Osa), aes(x=geneNumber, y=unlist(lapply(dNdS_GUIDANCE2, function(x) median(x, na.rm=T))), group=geneNumber) ) +
  geom_boxplot(aes(fill=geneNumber)) +
  ylab('dN/dS') +
  ggtitle("Oki/Osa: dN/dS vs. gene count")

# Scatterplot of dN/dS vs. gene number, with linear model
plot2 <- ggplot(as.data.frame(orthoBlocks$Oki_Osa), aes(x=geneNumber, y=unlist(lapply(dNdS_GUIDANCE2, function(x) median(x, na.rm=T)))) ) +
  geom_point() +
  geom_smooth(method='lm', formula=y~x) +
  ylab('dN/dS') +
  ggtitle("Oki/Osa: dN/dS vs. gene count")

ggarrange(plot1, plot2)

```

Are these relationships true for other combinations? Starting with Okinawa/Barcelona:

```{r exploring_dnds_3, fig.height=4, fig.width=6, dev=c("svg", "png")}
plot1 <- ggplot(as.data.frame(orthoBlocks$Oki_Bar), aes(x=geneNumber )) +
  geom_histogram() +
  ggtitle("Oki/Bar: genes per syntenic block")

plot2 <- ggplot(as.data.frame(orthoBlocks$Oki_Bar), aes(x=unlist(lapply(dNdS_GUIDANCE2, function(x) median(x, na.rm=T))) )) +
  geom_histogram() +
  ggtitle("Oki/Bar: dN/dS") + xlab('dN/dS')

plot3 <- ggplot(as.data.frame(orthoBlocks$Oki_Bar), aes(x=geneNumber )) +
  geom_histogram() +
  scale_y_continuous(trans='log10') +
  ggtitle("Oki/Bar: genes per syntenic block (log10 axis)")

plot4 <- ggplot(as.data.frame(orthoBlocks$Oki_Bar), aes(x=unlist(lapply(dNdS_GUIDANCE2, function(x) median(x, na.rm=T))) )) +
  geom_histogram() +
  scale_y_continuous(trans='log10') +
  ggtitle("Oki/Bar: dN/dS (log10 axis)") + xlab('dN/dS')

ggarrange(plot1, plot2, plot3, plot4)

plot5 <- ggplot(as.data.frame(orthoBlocks$Oki_Bar), aes(x=geneNumber, y=unlist(lapply(dNdS_GUIDANCE2, function(x) median(x, na.rm=T))), group=geneNumber) ) +
  geom_boxplot(aes(fill=geneNumber)) +
  ylab('dN/dS') +
  ggtitle("Oki/Bar: dN/dS vs. gene count")

plot6 <- ggplot(as.data.frame(orthoBlocks$Oki_Bar), aes(x=geneNumber, y=unlist(lapply(dNdS_GUIDANCE2, function(x) median(x, na.rm=T)))) ) +
  geom_point() +
  geom_smooth(method='lm', formula=y~x) +
  ylab('dN/dS') +
  ggtitle("Oki/Bar: dN/dS vs. gene count")

ggarrange(plot5, plot6)
```

```{r exploring_dnds_3, fig.height=4, fig.width=6, dev=c("svg", "png")}
plot1 <- ggplot(as.data.frame(orthoBlocks$Osa_Bar), aes(x=geneNumber )) +
  geom_histogram() +
  ggtitle("Osa/Bar: genes per syntenic block")

plot2 <- ggplot(as.data.frame(orthoBlocks$Osa_Bar), aes(x=unlist(lapply(dNdS_GUIDANCE2, function(x) median(x, na.rm=T))) )) +
  geom_histogram() +
  ggtitle("Osa/Bar: dN/dS") + xlab('dN/dS')

plot3 <- ggplot(as.data.frame(orthoBlocks$Osa_Bar), aes(x=geneNumber )) +
  geom_histogram() +
  scale_y_continuous(trans='log10') +
  ggtitle("Osa/Bar: genes per syntenic block (log10 axis)")

plot4 <- ggplot(as.data.frame(orthoBlocks$Osa_Bar), aes(x=unlist(lapply(dNdS_GUIDANCE2, function(x) median(x, na.rm=T))) )) +
  geom_histogram() +
  scale_y_continuous(trans='log10') +
  ggtitle("Osa/Bar: dN/dS (log10 axis)") + xlab('dN/dS')

ggarrange(plot1, plot2, plot3, plot4)

plot5 <- ggplot(as.data.frame(orthoBlocks$Osa_Bar), aes(x=geneNumber, y=unlist(lapply(dNdS_GUIDANCE2, function(x) median(x, na.rm=T))), group=geneNumber) ) +
  geom_boxplot(aes(fill=geneNumber)) +
  ylab('dN/dS') +
  ggtitle("Osa/Bar: dN/dS vs. gene count")

plot6 <- ggplot(as.data.frame(orthoBlocks$Osa_Bar), aes(x=geneNumber, y=unlist(lapply(dNdS_GUIDANCE2, function(x) median(x, na.rm=T)))) ) +
  geom_point() +
  geom_smooth(method='lm', formula=y~x) +
  ylab('dN/dS') +
  ggtitle("Osa/Bar: dN/dS vs. gene count")

ggarrange(plot5, plot6)
```

It seems, therefore, the relationship between dN/dS values (or, the global substitution rates),
is not strongly related to the number of genes in a syntenic block. 

## More targeted dN/dS analyses: grouping genes by organization
For instance, how do dN/dS values compare between individual genes and genes within syntenic blocks?

```{r evaluating_dnds_blocks_oki_osa_1, fig.height=4, fig.width=6, dev=c("svg", "png")}
# Make a function to retrieve dN/dS values based on how many genes are in a block.
get_dnds <- function(orthoblocks, alignment_type="dNdS_GUIDANCE2", block_min_size=0, block_max_size=2){
  gene_numbers <- orthoblocks$geneNumber
  matching_rows <- block_min_size < gene_numbers & gene_numbers < block_max_size
  matching_dnds <- na.omit(unlist(as.data.frame(orthoblocks[matching_rows,])[[alignment_type]], recursive=T))
  matching_dnds
}

dndsBlocks <- SimpleList()
dndsBlocks$Oki_Osa <- SimpleList()
# Subset to syntenic blocks with more than 0 genes and less than 2 - aka, just one gene.
dndsBlocks$Oki_Osa$single_gene_block <- get_dnds(orthoBlocks$Oki_Osa, alignment_type="dNdS_GUIDANCE2", block_min_size=0, block_max_size=2)
# Subset to syntenic blocks with more than 1 gene.
dndsBlocks$Oki_Osa$multi_gene_block <- get_dnds(orthoBlocks$Oki_Osa, alignment_type="dNdS_GUIDANCE2", block_min_size=2, block_max_size=Inf)

# Plot distributions of dN/dS by syntenic block size.
ggplot(as.data.frame(dndsBlocks$Oki_Osa), aes(y=value, fill=group_name)) +
  geom_histogram(position='identity', bins=30) +
  facet_wrap(~group_name, scales='free_x') +
  ggtitle("Oki/Osa: dN/dS of individual genes vs. syntenic block genes")
```

Maybe there is a difference there, maybe not. Let's see.

```{r evaluating_dnds_blocks_oki_osa_2, fig.height=4, fig.width=6, dev=c("svg", "png")}
ggplot(as.data.frame(dndsBlocks$Oki_Osa), aes(y=value, fill=group_name)) +
  geom_boxplot(position='identity') +
  facet_wrap(~group_name, scales='free_x') +
  ggtitle("Oki/Osa: dN/dS of individual genes vs. syntenic block genes")
# Are the differences significant?
(pairwise.t.test(g=as.data.frame(dndsBlocks$Oki_Osa)$group_name,
                 x=as.data.frame(dndsBlocks$Oki_Osa)$value,
                 p.adjust.method='bonferroni'))
```

Indeed, in this case it seems there is a modestly significant difference between individual genes and genes in syntenic blocks. Does this relationship hold for other organisms?

```{r evaluating_dnds_blocks_oki_bar, fig.height=4, fig.width=6, dev=c("svg", "png")}
dndsBlocks$Oki_Bar <- SimpleList()
dndsBlocks$Oki_Bar$single_gene_block <- get_dnds(orthoBlocks$Oki_Bar, alignment_type="dNdS_GUIDANCE2", block_min_size=0, block_max_size=2)
dndsBlocks$Oki_Bar$multi_gene_block <- get_dnds(orthoBlocks$Oki_Bar, alignment_type="dNdS_GUIDANCE2", block_min_size=2, block_max_size=Inf)

ggplot(as.data.frame(dndsBlocks$Oki_Bar), aes(y=value, fill=group_name)) +
  geom_histogram(position='identity', bins=30) +
  facet_wrap(~group_name, scales='free_x') +
  ggtitle("Oki/Bar: dN/dS of individual genes vs. syntenic block genes")

ggplot(as.data.frame(dndsBlocks$Oki_Bar), aes(y=value, fill=group_name)) +
  geom_boxplot(position='identity') +
  facet_wrap(~group_name, scales='free_x') +
  ggtitle("Oki/Bar: dN/dS of individual genes vs. syntenic block genes")

(pairwise.t.test(g=as.data.frame(dndsBlocks$Oki_Bar)$group_name,
                 x=as.data.frame(dndsBlocks$Oki_Bar)$value,
                 p.adjust.method='bonferroni'))
```

For Okinawa-Barcelona, it is also significant.

```{r evaluating_dnds_blocks_osa_bar, fig.height=4, fig.width=6, dev=c("svg", "png")}
dndsBlocks$Osa_Bar <- SimpleList()
dndsBlocks$Osa_Bar$single_gene_block <- get_dnds(orthoBlocks$Osa_Bar, alignment_type="dNdS_GUIDANCE2", block_min_size=0, block_max_size=2)
dndsBlocks$Osa_Bar$multi_gene_block <- get_dnds(orthoBlocks$Osa_Bar, alignment_type="dNdS_GUIDANCE2", block_min_size=2, block_max_size=Inf)

ggplot(as.data.frame(dndsBlocks$Osa_Bar), aes(y=value, fill=group_name)) +
  geom_histogram(position='identity', bins=30) +
  facet_wrap(~group_name, scales='free_x') +
  ggtitle("Osa/Bar: dN/dS of individual genes vs. syntenic block genes")

ggplot(as.data.frame(dndsBlocks$Osa_Bar), aes(y=value, fill=group_name)) +
  geom_boxplot(position='identity') +
  facet_wrap(~group_name, scales='free_x') +
  ggtitle("Osa/Bar: dN/dS of individual genes vs. syntenic block genes")

(pairwise.t.test(g=as.data.frame(dndsBlocks$Osa_Bar)$group_name,
                 x=as.data.frame(dndsBlocks$Osa_Bar)$value,
                 p.adjust.method='bonferroni'))

```

But for Osaka-Barcelona, it is not.

A contributing factor might be that short and long arms differ in the size of their syntenic blocks,
and as we have already observed, there is an arm-specific difference in dN/dS values. Let's modify the
script to account for these differences.

```{r evaluating_dnds_with_arms_1, fig.height=4, fig.width=6, dev=c("svg", "png")}
get_dnds_with_arms <- function(orthoblocks, arm='long', alignment_type="dNdS_GUIDANCE2", block_min_size=0, block_max_size=2){
  gene_numbers <- orthoblocks$geneNumber
  arms <- orthoblocks$Arm
  ob <- subset(orthoblocks, block_min_size < geneNumber & geneNumber < block_max_size & Arm == arm)
  matching_dnds <- na.omit(unlist(as.data.frame(ob)[[alignment_type]], recursive=T))
  matching_dnds
}

dndsBlocks <- SimpleList()
dndsBlocks$Oki_Osa <- SimpleList()
dndsBlocks$Oki_Osa$single_gene_long <- get_dnds_with_arms(orthoBlocks$Oki_Osa, arm="long", alignment_type="dNdS_GUIDANCE2", block_min_size=0, block_max_size=2)
dndsBlocks$Oki_Osa$single_gene_short <- get_dnds_with_arms(orthoBlocks$Oki_Osa, arm="short", alignment_type="dNdS_GUIDANCE2", block_min_size=0, block_max_size=2)

dndsBlocks$Oki_Osa$multi_gene_long <- get_dnds_with_arms(orthoBlocks$Oki_Osa, arm="long", alignment_type="dNdS_GUIDANCE2", block_min_size=2, block_max_size=Inf)
dndsBlocks$Oki_Osa$multi_gene_short <- get_dnds_with_arms(orthoBlocks$Oki_Osa, arm="short", alignment_type="dNdS_GUIDANCE2", block_min_size=2, block_max_size=Inf)

ggplot(as.data.frame(dndsBlocks$Oki_Osa), aes(y=value, fill=group_name)) +
  geom_boxplot(position='identity') +
  facet_grid(~group_name, scales='free_x') +
  ggtitle("Oki/Osa: dN/dS of individual genes vs. syntenic block genes")

(pairwise.t.test(g=as.data.frame(dndsBlocks$Oki_Osa)$group_name,
                 x=as.data.frame(dndsBlocks$Oki_Osa)$value,
                 p.adjust.method='bonferroni'))

```


```{r evaluating_dnds_with_arms_2, fig.height=4, fig.width=6, dev=c("svg", "png")}
get_dnds_with_arms <- function(orthoblocks, arm='long', alignment_type="dNdS_GUIDANCE2", block_min_size=0, block_max_size=2){
  gene_numbers <- orthoblocks$geneNumber
  arms <- orthoblocks$Arm
  ob <- subset(orthoblocks, block_min_size < geneNumber & geneNumber < block_max_size & Arm == arm)
  matching_dnds <- na.omit(unlist(as.data.frame(ob)[[alignment_type]], recursive=T))
  matching_dnds
}

dndsBlocks <- SimpleList()
dndsBlocks$Oki_Osa <- SimpleList()
dndsBlocks$Oki_Osa$single_gene_long <- get_dnds_with_arms(orthoBlocks$Oki_Osa, arm="long", alignment_type="dNdS_GUIDANCE2", block_min_size=0, block_max_size=2)
dndsBlocks$Oki_Osa$single_gene_short <- get_dnds_with_arms(orthoBlocks$Oki_Osa, arm="short", alignment_type="dNdS_GUIDANCE2", block_min_size=0, block_max_size=2)

dndsBlocks$Oki_Osa$btw2_5_genes_long <- get_dnds_with_arms(orthoBlocks$Oki_Osa, arm="long", alignment_type="dNdS_GUIDANCE2", block_min_size=1, block_max_size=6)
dndsBlocks$Oki_Osa$btw2_5_genes_short <- get_dnds_with_arms(orthoBlocks$Oki_Osa, arm="short", alignment_type="dNdS_GUIDANCE2", block_min_size=1, block_max_size=6)


dndsBlocks$Oki_Osa$mt_6_genes_long <- get_dnds_with_arms(orthoBlocks$Oki_Osa, arm="long", alignment_type="dNdS_GUIDANCE2", block_min_size=6, block_max_size=Inf)
dndsBlocks$Oki_Osa$mt_6_genes_short <- get_dnds_with_arms(orthoBlocks$Oki_Osa, arm="short", alignment_type="dNdS_GUIDANCE2", block_min_size=6, block_max_size=Inf)

ggplot(as.data.frame(dndsBlocks$Oki_Osa), aes(y=value, fill=group_name)) +
  geom_boxplot(position='identity') +
  facet_wrap(~group_name, scales='free_x', nrow=3, ncol=2) +
  coord_cartesian(ylim=c(0,0.1)) + 
  ggtitle("Oki/Osa: dN/dS of individual genes vs. syntenic block genes")

(pairwise.t.test(g=as.data.frame(dndsBlocks$Oki_Osa)$group_name,
                 x=as.data.frame(dndsBlocks$Oki_Osa)$value,
                 p.adjust.method='bonferroni'))
```



```{r evaluating_dnds_with_arms_3, fig.height=4, fig.width=6, dev=c("svg", "png")}
get_dnds_with_arms <- function(orthoblocks, arm='long', alignment_type="dNdS_GUIDANCE2", block_min_size=0, block_max_size=2){
  gene_numbers <- orthoblocks$geneNumber
  arms <- orthoblocks$Arm
  ob <- subset(orthoblocks, block_min_size < geneNumber & geneNumber < block_max_size & Arm == arm)
  matching_dnds <- na.omit(unlist(as.data.frame(ob)[[alignment_type]], recursive=T))
  matching_dnds
}

dndsBlocks <- SimpleList()
dndsBlocks$Osa_Bar <- SimpleList()
dndsBlocks$Osa_Bar$single_gene_long <- get_dnds_with_arms(orthoBlocks$Osa_Bar, arm="long", alignment_type="dNdS_GUIDANCE2", block_min_size=0, block_max_size=2)
dndsBlocks$Osa_Bar$single_gene_short <- get_dnds_with_arms(orthoBlocks$Osa_Bar, arm="short", alignment_type="dNdS_GUIDANCE2", block_min_size=0, block_max_size=2)

dndsBlocks$Osa_Bar$btw2_5_genes_long <- get_dnds_with_arms(orthoBlocks$Osa_Bar, arm="long", alignment_type="dNdS_GUIDANCE2", block_min_size=1, block_max_size=6)
dndsBlocks$Osa_Bar$btw2_5_genes_short <- get_dnds_with_arms(orthoBlocks$Osa_Bar, arm="short", alignment_type="dNdS_GUIDANCE2", block_min_size=1, block_max_size=6)


dndsBlocks$Osa_Bar$mt_6_genes_long <- get_dnds_with_arms(orthoBlocks$Osa_Bar, arm="long", alignment_type="dNdS_GUIDANCE2", block_min_size=6, block_max_size=Inf)
dndsBlocks$Osa_Bar$mt_6_genes_short <- get_dnds_with_arms(orthoBlocks$Osa_Bar, arm="short", alignment_type="dNdS_GUIDANCE2", block_min_size=6, block_max_size=Inf)

ggplot(as.data.frame(dndsBlocks$Osa_Bar), aes(y=value, fill=group_name)) +
  geom_boxplot(position='identity') +
  facet_wrap(~group_name, scales='free_x', nrow=3, ncol=2) +
  ggtitle("Osa/Bar: dN/dS of individual genes vs. syntenic block genes")

(pairwise.t.test(g=as.data.frame(dndsBlocks$Osa_Bar)$group_name,
                 x=as.data.frame(dndsBlocks$Osa_Bar)$value,
                 p.adjust.method='bonferroni'))

```


```{r evaluating_dnds_with_arms_4, fig.height=4, fig.width=6, dev=c("svg", "png")}
get_dnds_with_arms <- function(orthoblocks, arm='long', alignment_type="dNdS_GUIDANCE2", block_min_size=0, block_max_size=2){
  gene_numbers <- orthoblocks$geneNumber
  arms <- orthoblocks$Arm
  ob <- subset(orthoblocks, block_min_size < geneNumber & geneNumber < block_max_size & Arm == arm)
  matching_dnds <- na.omit(unlist(as.data.frame(ob)[[alignment_type]], recursive=T))
  matching_dnds
}

dndsBlocks <- SimpleList()
dndsBlocks$Oki_Bar <- SimpleList()
dndsBlocks$Oki_Bar$single_gene_long <- get_dnds_with_arms(orthoBlocks$Oki_Bar, arm="long", alignment_type="dNdS_GUIDANCE2", block_min_size=0, block_max_size=2)
dndsBlocks$Oki_Bar$single_gene_short <- get_dnds_with_arms(orthoBlocks$Oki_Bar, arm="short", alignment_type="dNdS_GUIDANCE2", block_min_size=0, block_max_size=2)

dndsBlocks$Oki_Bar$btw2_5_genes_long <- get_dnds_with_arms(orthoBlocks$Oki_Bar, arm="long", alignment_type="dNdS_GUIDANCE2", block_min_size=1, block_max_size=6)
dndsBlocks$Oki_Bar$btw2_5_genes_short <- get_dnds_with_arms(orthoBlocks$Oki_Bar, arm="short", alignment_type="dNdS_GUIDANCE2", block_min_size=1, block_max_size=6)


dndsBlocks$Oki_Bar$mt_6_genes_long <- get_dnds_with_arms(orthoBlocks$Oki_Bar, arm="long", alignment_type="dNdS_GUIDANCE2", block_min_size=6, block_max_size=Inf)
dndsBlocks$Oki_Bar$mt_6_genes_short <- get_dnds_with_arms(orthoBlocks$Oki_Bar, arm="short", alignment_type="dNdS_GUIDANCE2", block_min_size=6, block_max_size=Inf)

ggplot(as.data.frame(dndsBlocks$Oki_Bar), aes(y=value, fill=group_name)) +
  geom_boxplot(position='identity') +
  facet_wrap(~group_name, scales='free_x', nrow=3, ncol=2) +
  ggtitle("Oki/Bar: dN/dS of individual genes vs. syntenic block genes")

(pairwise.t.test(g=as.data.frame(dndsBlocks$Oki_Bar)$group_name,
                 x=as.data.frame(dndsBlocks$Oki_Bar)$value,
                 p.adjust.method='bonferroni'))

```

  
# Session information

```{r session_info}
sessionInfo()
```
