---
title: "Exploring coding-sequence substitutions"
author: 
 - "Michael Mansfield"
date: "`r format(Sys.time(), '%d %B, %Y')`"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{Exploring coding-sequence substitutions}
  %\VignetteEncoding{UTF-8}
  %\VignetteEngine{knitr::rmarkdown}
editor_options: 
  chunk_output_type: console
  markdown: 
    wrap: 72
---

```{r setup}
knitr::opts_knit$set(verbose = TRUE)
```

# Introduction

This vignette loads dN/dS statistics tables and explores the
relationship between substitutions in coding sequences and genomic
breaks.

This vignette assumes that prior vignettes have been explored by the
user previously and that the `GenomicBreaks` and `OikScrambling`
packages have been installed already.

# Packages and data

```{r load_pacakges_and_data}
suppressPackageStartupMessages({
  library('GenomicBreaks')
  requireNamespace('GenomicFeatures')
  library('ggplot2')
  library("BreakpointsData")
})
(genomes <- OikScrambling:::loadAllGenomes())
```

# Genomic breaks


## dN/dS values

Once you have a fully-populated `GBreaks` object with all of the annotations you want,
adding additional metadata like dN/dS values is simple. First, load the dN/dS tables.
```
dnds <- SimpleList()
dnds$dnds_guidance   <- loaddNdSTable("v2.0.0/genewise_oikopleura.extended.txt", alignment_type = 'GUIDANCE2')
dnds$dnds_hmmcleaner <- loaddNdSTable("v2.0.0/genewise_oikopleura.extended.txt", alignment_type = 'HmmCleaner')
dnds$dnds_prank      <- loaddNdSTable("v2.0.0/genewise_oikopleura.extended.txt", alignment_type = 'Alignment')
transcripts <- sapply(names(transcripts), function(sp){
  sp_dnds = addMetadataToGRanges(gr=transcripts[[sp]], df=dnds$dnds_guidance[[sp]], gr_key='tx_name', df_key='transcript_id_simple', df_col_name = 'dNdS', rename_meta = 'dNdS_GUIDANCE2')
  sp_dnds = addMetadataToGRanges(gr=sp_dnds, df=dnds$dnds_hmmcleaner[[sp]], gr_key='tx_name', df_key='transcript_id_simple', df_col_name = 'dNdS', rename_meta = 'dNdS_HmmCleaner')
  sp_dnds = addMetadataToGRanges(gr=sp_dnds, df=dnds$dnds_prank[[sp]], gr_key='tx_name', df_key='transcript_id_simple', df_col_name = 'dNdS', rename_meta = 'dNdS_PRANK')
  sp_dnds
})
```

This yields a `SimpleList` `dnds`, which contains data frames organized by dN/dS method.
The data frames look like so:

```
> head(dnds$dnds_guidance$Oki)
  gene_id       transcript_id_full transcript_id_simple    dNdS
1  g16808 OKI2018_I69.v2.g16808.t1            g16808.t1 0.00928
2  g13507 OKI2018_I69.v2.g13507.t1            g13507.t1 0.01340
3   g4633  OKI2018_I69.v2.g4633.t1             g4633.t1 0.01920
4  g12080 OKI2018_I69.v2.g12080.t1            g12080.t1 0.02840
5  g12181 OKI2018_I69.v2.g12181.t1            g12181.t1 0.02653
6   g8257  OKI2018_I69.v2.g8257.t1             g8257.t1 0.03106
```

By creating a `GBreaks` object with the metadata column `tx_name`, we can combine
this dN/dS dataframe using that shared key with the `addMetadataToGRanges` function.

The `GBreaks` object from the `OrthogroupSyntenies` vignette is suitable.

```
orthoPairToGBreaks <- function(name1, name2, annot1, annot2) {
  orthoPairs <- load_one_to_ones(
    system.file("extdata/OrthoFinder/N19.tsv", package = "BreakpointsData"),
    c(name1, name2))

  IDs2GRanges <- function (IDs, annot) {
    prefix <- Biobase::lcPrefix(IDs)                  # Guess prefix in transcript IDs from HOG files
    not_prefix <- Biobase::lcPrefix(annot$tx_name)    # Remove trailing characters that are part of the name
    not_prefix <- paste0(not_prefix,"$")              # Anchor to end of the string
    prefix <- sub(not_prefix, "", prefix)             # Finalise prefix
    IDs <- sub(prefix, "", IDs)                       # Remove prefix from IDS
    names(annot) <- annot$tx_name                     # Store transcript name in names slot
    gr <- granges(annot[IDs], use.mcols=T)            # Don't! discard metadata. Sort by ID
    strand(gr) <- "*"                                 # Make strandless
    gr                                                # Return the object
  }
  
  gb       <- IDs2GRanges(orthoPairs[,name1], transcripts(annot1))
  gb$query <- IDs2GRanges(orthoPairs[,name2], transcripts(annot2))
  gb <- GenomicBreaks:::GBreaks(gb)
  sort(gb)
}

orthoPairs <- SimpleList()
orthoPairs$Oki_Osa <- orthoPairToGBreaks("OKI2018_I69.v2.prot.longest.fa_1", "OSKA2016v1.9.prot.longest.fa_1"
                                        , annots$Oki, annots$Osa)
orthoPairs$Oki_Bar <- orthoPairToGBreaks("OKI2018_I69.v2.prot.longest.fa_1", "Bar2_p4.Flye.prot.longest.fa_1"
                                        , annots$Oki, annots$Bar)
orthoPairs$Oki_Kum <- orthoPairToGBreaks("OKI2018_I69.v2.prot.longest.fa_1", "KUM-M3-7f.prot.longest.fa_1"
                                        , annots$Oki, annots$Kum)
orthoPairs$Oki_Aom <- orthoPairToGBreaks("OKI2018_I69.v2.prot.longest.fa_1", "AOM-5-5f.prot.longest.fa_1"
                                        , annots$Oki, annots$Aom)
## Problem: some Norway transcripts come from alternative "scaffoldA" sequences that are not in the genome file.
# orthoPairs$Bar_Nor <- orthoPairToGBreaks("Bar2_p4.Flye.prot.longest.fa_1", "OdB3.v1.0.prot.fa_1"
#                                         , annots$Bar, annots$Nor)

orthoPairs$Osa_Bar <- orthoPairToGBreaks("OSKA2016v1.9.prot.longest.fa_1", "Bar2_p4.Flye.prot.longest.fa_1"
                                        , annots$Osa, annots$Bar)
orthoPairs$Osa_Oki <- orthoPairToGBreaks("OSKA2016v1.9.prot.longest.fa_1", "OKI2018_I69.v2.prot.longest.fa_1"
                                        , annots$Osa, annots$Oki)
orthoPairs$Osa_Kum <- orthoPairToGBreaks("OSKA2016v1.9.prot.longest.fa_1", "KUM-M3-7f.prot.longest.fa_1"
                                        , annots$Osa, annots$Kum)
orthoPairs$Osa_Aom <- orthoPairToGBreaks("OSKA2016v1.9.prot.longest.fa_1", "AOM-5-5f.prot.longest.fa_1"
                                        , annots$Osa, annots$Aom)
## Problem: some Norway transcripts come from alternative "scaffoldA" sequences that are not in the genome file.
# orthoPairs$Bar_Nor <- orthoPairToGBreaks("Bar2_p4.Flye.prot.longest.fa_1", "OdB3.v1.0.prot.fa_1"
#                                         , annots$Bar, annots$Nor)


orthoPairs$Bar_Osa <- orthoPairToGBreaks("Bar2_p4.Flye.prot.longest.fa_1", "OSKA2016v1.9.prot.longest.fa_1"
                                        , annots$Bar, annots$Osa)
orthoPairs$Bar_Oki <- orthoPairToGBreaks("Bar2_p4.Flye.prot.longest.fa_1", "OKI2018_I69.v2.prot.longest.fa_1"
                                        , annots$Bar, annots$Oki)
orthoPairs$Bar_Kum <- orthoPairToGBreaks("Bar2_p4.Flye.prot.longest.fa_1", "KUM-M3-7f.prot.longest.fa_1"
                                        , annots$Bar, annots$Kum)
orthoPairs$Bar_Aom <- orthoPairToGBreaks("Bar2_p4.Flye.prot.longest.fa_1", "AOM-5-5f.prot.longest.fa_1"
                                        , annots$Bar, annots$Aom)
## Problem: some Norway transcripts come from alternative "scaffoldA" sequences that are not in the genome file.
# orthoPairs$Bar_Nor <- orthoPairToGBreaks("Bar2_p4.Flye.prot.longest.fa_1", "OdB3.v1.0.prot.fa_1"
#                                         , annots$Bar, annots$Nor)

load("BreakPoints.Rdata")
orthoPairs[ 1:4] <- sapply(orthoPairs[ 1:4],  flagLongShort, longShort$Oki)
orthoPairs[ 5:8] <- sapply(orthoPairs[ 5:8],  flagLongShort, longShort$Osa)
orthoPairs[9:11] <- sapply(orthoPairs[9:11],  flagLongShort, longShort$Bar)
```

Using these transcript-based `GBreaks` objects, you can annotate those
transcripts with arbitrary metadata. Note that the transcript identifier
is relative to the target.

TODO resolve cases where query metadata is not precisely equal to target metadata

```
dndsPairs <- orthoPairs

dndsPairs$Oki_Osa <- addMetadataToGRanges(gr=dndsPairs$Oki_Osa, gr_key="tx_name", df=dnds$dnds_guidance$Oki, df_key="transcript_id_simple", df_col_name = "dNdS", rename_meta = "dNdS")
dndsPairs$Oki_Bar <- addMetadataToGRanges(gr=dndsPairs$Oki_Bar, gr_key="tx_name", df=dnds$dnds_guidance$Oki, df_key="transcript_id_simple", df_col_name = "dNdS", rename_meta = "dNdS")
dndsPairs$Oki_Kum <- addMetadataToGRanges(gr=dndsPairs$Oki_Kum, gr_key="tx_name", df=dnds$dnds_guidance$Oki, df_key="transcript_id_simple", df_col_name = "dNdS", rename_meta = "dNdS")
dndsPairs$Oki_Aom <- addMetadataToGRanges(gr=dndsPairs$Oki_Aom, gr_key="tx_name", df=dnds$dnds_guidance$Oki, df_key="transcript_id_simple", df_col_name = "dNdS", rename_meta = "dNdS")

dndsPairs$Osa_Bar <- addMetadataToGRanges(gr=dndsPairs$Osa_Bar, gr_key="tx_name", df=dnds$dnds_guidance$Osa, df_key="transcript_id_simple", df_col_name = "dNdS", rename_meta = "dNdS")
dndsPairs$Osa_Oki <- addMetadataToGRanges(gr=dndsPairs$Osa_Oki, gr_key="tx_name", df=dnds$dnds_guidance$Osa, df_key="transcript_id_simple", df_col_name = "dNdS", rename_meta = "dNdS")
dndsPairs$Osa_Kum <- addMetadataToGRanges(gr=dndsPairs$Osa_Kum, gr_key="tx_name", df=dnds$dnds_guidance$Osa, df_key="transcript_id_simple", df_col_name = "dNdS", rename_meta = "dNdS")
dndsPairs$Osa_Aom <- addMetadataToGRanges(gr=dndsPairs$Osa_Aom, gr_key="tx_name", df=dnds$dnds_guidance$Osa, df_key="transcript_id_simple", df_col_name = "dNdS", rename_meta = "dNdS")

dndsPairs$Bar_Osa <- addMetadataToGRanges(gr=dndsPairs$Bar_Osa, gr_key="tx_name", df=dnds$dnds_guidance$Bar, df_key="transcript_id_simple", df_col_name = "dNdS", rename_meta = "dNdS")
dndsPairs$Bar_Oki <- addMetadataToGRanges(gr=dndsPairs$Bar_Oki, gr_key="tx_name", df=dnds$dnds_guidance$Bar, df_key="transcript_id_simple", df_col_name = "dNdS", rename_meta = "dNdS")
dndsPairs$Bar_Kum <- addMetadataToGRanges(gr=dndsPairs$Bar_Kum, gr_key="tx_name", df=dnds$dnds_guidance$Bar, df_key="transcript_id_simple", df_col_name = "dNdS", rename_meta = "dNdS")
dndsPairs$Bar_Aom <- addMetadataToGRanges(gr=dndsPairs$Bar_Aom, gr_key="tx_name", df=dnds$dnds_guidance$Bar, df_key="transcript_id_simple", df_col_name = "dNdS", rename_meta = "dNdS")

```




# Session information

```{r session_info}
sessionInfo()
```
