---
title: "Genomic features of the breakpoints (heatmaps)"
author: 
 - "Charlotte West"
 - "Charles Plessy"
date: "`r format(Sys.time(), '%d %B, %Y')`"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteEncoding{UTF-8}
  %\VignetteIndexEntry{Genomic features of the breakpoints (heatmaps)}
  %\VignetteEngine{knitr::rmarkdown}
editor_options: 
  chunk_output_type: console
---

Load packages and data
======================

See `?OikScrambling:::loadAllGenomes`, `?OikScrambling:::loadAllTranscriptsGR`,
and `vignette("LoadGenomicBreaks", package = "OikScrambling")` for how the
different objects are prepared.

```{r load_packages_and_data}
library('OikScrambling')   |> suppressPackageStartupMessages()
library('GenomicFeatures') |> suppressPackageStartupMessages()
library('heatmaps')        |> suppressPackageStartupMessages()
library('patchwork')       |> suppressPackageStartupMessages()
genomes     <- OikScrambling:::loadAllGenomes(compat = F)
# Cannot use Knitr cache as long as "annot" objects are used.
annots      <- OikScrambling:::loadAllAnnotations()   |> suppressWarnings()
reps        <- OikScrambling:::loadAllRepeats(compat = F)
genes       <- sapply(annots, function(a) sort(genes(a))) |> SimpleList()
load("BreakPoints.Rdata")
requireNamespace("rGADEM")    |> suppressPackageStartupMessages()
requireNamespace("ggseqlogo") |> suppressPackageStartupMessages()
ces <- readRDS("CEs.Rds")
tfbs <- readRDS("pwmMatchesOki_12_95.Rds")
```

# Nucleic acid content heatmaps

The `GenomicBreaks::bp_heatmap()` function plots nucleic acid content in windows
centered on alignment stops.

## GC

Lets look at `GC` content before and after coalescing: 

```{r echo=T, fig.height=6, fig.width=12, message=FALSE, warning=FALSE}
maxHeight <- 2000
sm_GC2 <- sapply(coa[1:15],  bp_heatmap, win = 1000, pat = "GC", direction = "left") |>
          sapply(smoothHeatmap, output.size = c(maxHeight,     500), algorithm = "kernel") |> SimpleList()
plotHeatmapList(sm_GC2[1:5])
```

The alignment stops are ordered in such a way that start (or left breaks) are
centred on the top half of the plot, and end alignment stops are centred on the
bottom. This is why we see different directionality in the plots. At the
alignment stops, the GC content seems to be lower, but higher once within the
aligned region, except for the Oki – Kum pair.

## TATA

Now looking at TATA content:

```{r TATA, fig.height=6, fig.width=12, message=FALSE, warning=FALSE}
maxHeight <- 2000
sm_GC2_TATA <- sapply(coa[1:15], bp_heatmap, window = 1000, pat = "TATA", direction = "left") |> 
               sapply(smoothHeatmap, output.size = c(maxHeight,     500), algorithm = "kernel") |> SimpleList()
plotHeatmapList(sm_GC2_TATA[1:5])
```

The directionaility here is consistent with the `GC` content analysis.
Furthermore; "TATA box sequence can act as a basal promoter element not only for
RNA polymerase II (RNAP II) transcription, but also for transcription by RNA
polymerase III (RNAP III)" - Wang Y, Jensen RC, Stumph WE. Role of TATA box
sequence and orientation in determining RNA polymerase II/III transcription
specificity. Nucleic Acids Res. 1996;24(15):3100–3106.
doi:10.1093/nar/24.15.3100. However, the areas are not particularly enriched
for TATA boxes, so it does not necessarily show that breakpoints are occurring
directly after promoter regions. 


```{r AAGCsGCwwmkCGrCTTyn, fig.height=6, fig.width=12, message=FALSE, warning=FALSE}
maxHeight <- 2000
sm_GC2_mot <- sapply(coa[1:15], bp_heatmap, window = 1000, pat = "AGCNGC", direction = "left") |> 
              sapply(smoothHeatmap, output.size = c(maxHeight,     500), algorithm = "kernel") |> SimpleList()
plotHeatmapList(sm_GC2_mot[1:5])
```


## Cluster analysis

```{r cluster_analysis, fig.height=20, fig.width=20}
# # Accessory functions
# clusterHeatmap <- function(hm, k = 2) kmeans(image(hm), k)
# 
# orderHeatmap <- function(hm, cl) {
#   image(hm) <- image(hm)[order(cl$cluster),]
#   hm
# }
# 
# subsetHeatmap <- function(hm, bool) {
#   image(hm) <- image(hm)[bool,]
#   hm
# }
# 
# # DAta
# hm_GC   <- bp_heatmap(      gr_Oki_O, basep_range  = 1000, pat = "GC",     direction = "left")
# hm_TATA <- bp_heatmap(      gr_Oki_O, basep_range  = 1000, pat = "TATA",   direction = "left")
# hm_tan  <- feature_coverage(gr_Oki_O, tan_Oki, win = 1000, lab = "Tandem", direction = "left")
# 
# # Define 3 K-mean clusters
# clusterHeatmap(hm_GC, 3) -> hm_cl_k4
# 
# # Order and smooth
# shm_GC <- orderHeatmap(hm_GC, hm_cl_k4) %>%
#   smoothHeatmap(output.size = c(2000,500), algorithm = "kernel")
# 
# shm_TATA <- orderHeatmap(hm_TATA, hm_cl_k4) %>%
#   smoothHeatmap(output.size = c(2000,500), algorithm = "kernel")
# 
# shm_tan <- orderHeatmap(hm_tan, hm_cl_k4) %>%
#   smoothHeatmap(output.size = c(2000,500), algorithm = "kernel")
# 
# # Plot the ordered data
# plotHeatmapList(list(shm_GC, shm_TATA, shm_tan))
# par(mfrow = c(1,1)) # plotHeatmapList messes with mrfrow...
# plotHeatmapMeta(list(hm_GC, hm_TATA, hm_tan))
# 
# plotHeatmapMeta(list(subsetHeatmap(hm_GC, hm_cl_k4$cluster == 1),
#                      subsetHeatmap(hm_GC, hm_cl_k4$cluster == 2),
#                      subsetHeatmap(hm_GC, hm_cl_k4$cluster == 3)))
# 
# kmeanOrderedPlot <- function (hm, k=3, seed = 1) {
#   set.seed(seed)
#   
#   cl <- kmeans(image(hm), k)
#   mat <- image(hm)[order(cl$cluster),]
#   
#   kmhm = Heatmap(
#     mat,
#     coords=c(-500,500),
#     label="kmeans",
#     scale=range(mat))
#   
#   plotHeatmapList(kmhm,
#                   cex.label=1.5,
#                   partition=as.vector(table(cl$cluster)),
#                   partition.legend=TRUE,
#                   partition.lines=TRUE,
#                   legend=TRUE,
#                   legend.pos="r",
#                   legend.width=0.3)
# }
# 
# kmeanOrderedPlot(shm_GC)
```

## Evidence for breakpoint hotspots

The function `?GenomicBreaks::bp_pair_analysis` takes two pairwise alignments
with the same target genome, and plots the alignment stops on to the centred
alignment stops of the other.

In the plot below, Osaka is there consistent target genome, and as such acts as
a coordinate system to relate alignment stops across genomes. Lets produce two
of these 3-way analyses; one between Osaka-Okinawa-Norway, and one between
Osaka-Okinawa-Aomori.

```{r breakpoint_hotspot_gbs}
bp_pair_analysis(gbs$Osa_Oki, gbs$Osa_Nor, win = 1000, lab = "Oki~Nor") |> plotHeatmapMeta()
bp_pair_analysis(gbs$Osa_Oki, gbs$Osa_Aom, win = 1000, lab = "Oki~Aom") |> plotHeatmapMeta()
```

The accumulation of alignment breaks of one pairwise alignment onto another
suggests the existence of breakpoint hotspots; regions where breaks are far
more likely to occur. Intuitively, this would lead to the presumption of synteny
blocks, also. The pattern is consistent after coalescing, too;

```{r breakpoint_hotspot_coa}
bp_pair_analysis(coa$Osa_Oki, coa$Osa_Nor, win = 1000, lab = "Oki~Nor") |> plotHeatmapMeta()
bp_pair_analysis(coa$Osa_Oki, coa$Osa_Aom, win = 1000, lab = "Oki~Aom") |> plotHeatmapMeta()
```

# Session information

```{r sessioninfo}
sessionInfo()
```
