---
title: "Introduction to the OikScrambling package"
author: 
 - "Charlotte West"
 - "Charles Plessy"
date: "`r format(Sys.time(), '%d %B, %Y')`"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteEncoding{UTF-8}
  %\VignetteIndexEntry{Introduction to the OikScrambling package}
  %\VignetteEngine{knitr::rmarkdown}
editor_options: 
  chunk_output_type: console
---

```{r setup}
knitr::opts_chunk$set(cache = TRUE)
knitr::opts_knit$set(verbose = TRUE)
```

# Introduction

This package contains a collection of vignettes to perform the analysis of
pairwise genome alignments between _Oikopleura_ genomes.

The core functions used here are maintained in our _GenomicBreaks_ R package,
which is fully documented at: <https://oist.github.io/GenomicBreaks>.

## Load pacakges

### Core packages that provide functions we use a lot.

```{r loadlib, echo=T, results='hide'}
suppressPackageStartupMessages({
  library('GenomicBreaks')
  library('BSgenome')
  library('GenomicFeatures')
  library('ggplot2')
})
```

### BSgenome packages

We need the sequences of the studied genomes.  They are provided in a _drat_
repository: <https://oist.github.io/plessy_oikgenomes_drat/>.

Example command to install one _Oikopleura_ BSgenome package if you do not yet
have it:

```
install.packages("BSgenome.Odioica.local.OKI2018.I69", repos="https://oist.github.io/plessy_oikgenomes_drat/")
```

```{r load_BSgenome_packages}
# These will need to be installed by the user
library("BSgenome.Odioica.local.OKI2018.I69")
library("BSgenome.Odioica.local.OSKA2016v1.9")
library("BSgenome.Odioica.local.Bar2.p4")
library("BSgenome.Odioica.local.KUM.M3")
library("BSgenome.Odioica.local.AOM.5")
library("BSgenome.Odioica.local.Odioica.reference.v3.0")
```

For convenience, we provide a `genomes` object containing all the genomes.

```{r genomes_object}
genomes <- SimpleList(
  # Chromosome assemblies
  Oki = OKI2018_I69, Osa = OSKA2016v1.9, Bar = Bar2_p4,
  # Less contiguous assemblies
  Kum = KUM_M3, Aom = AOM_5, Nor = OdB3)

## Genome lengths
(genome.lengths <- sapply(genomes, \(g) sum(seqlengths(seqinfo(g)))))

## Genome average AT content
(genomes.AT <- sapply(genomes, \(g)
  weighted.mean(letterFrequency(getSeq(g), "AT", as.prob = TRUE), seqlengths(g))))
```

## Load data

### Genome annotations.

The annotations from `/bucket/LuscombeU/common/Breakpoints/Annotations` are
distributed in the same drat repository as the alignment data.

```
install.packages("BreakpointsData", repos="https://oist.github.io/plessy_oikgenomes_drat/")
```

```{r load_annotations}
annots <- SimpleList()

gff2txdb <- function(file, genome) {
  file <- system.file(paste0("extdata/Annotations/", file), package = "BreakpointsData")
  tx <- rtracklayer::import.gff(file)
  tx <- tx[seqnames(tx) %in% seqnames(genome)]
  tx <- GRanges(tx, seqinfo = seqinfo(genome))
  tx <- GenomicFeatures::makeTxDbFromGRanges(tx)
}

gff2txdb_Norway <- function(file, genome) {
  file <- system.file(paste0("extdata/Annotations/", file), package = "BreakpointsData")
  tx <- rtracklayer::import.gff(file)
  # Remove "scaffoldA" objects in the OdB3 annotation
  tx <- tx[seqnames(tx) %in% seqnames(genome)]
  tx$ID <- NA_character_
  mcols(tx)[tx$type == "mRNA","ID"] <- paste0(tx[tx$type == "mRNA"]$mRNA, ".mRNA")
  mcols(tx)[tx$type == "CDS","ID"] <- paste0(tx[tx$type == "CDS"]$CDS, ".CDS")
  mcols(tx)[tx$type == "UTR","ID"] <- paste0(tx[tx$type == "UTR"]$UTR, ".UTR")
  tx <- GRanges(tx, seqinfo = seqinfo(genome))
  tx <- GenomicFeatures::makeTxDbFromGRanges(tx)
}

annots$Oki <- gff2txdb("OKI2018_I69.v2/OKI2018_I69.v2.gm.gff.gz",  OKI2018_I69)
annots$Osa <- gff2txdb("OSKA2016v1.9/OSKA2016v1.9.gm.gff.gz",      OSKA2016v1.9)
annots$Bar <- gff2txdb("Bar2_p4.Flye/Bar2_p4.Flye.gm.gff.gz",      Bar2_p4)
annots$Kum <- gff2txdb("KUM-M3-7f/KUM-M3-7f.gm.gff.gz",            KUM_M3)
annots$Aom <- gff2txdb("AOM-5-5f/AOM-5-5f.gm.gff.gz",              AOM_5)
annots$Nor <- gff2txdb_Norway("OdB3/Oikopleura_annot_v1.0.gff.gz", OdB3)
```

### GenomicBreaks object

The core data is a list of `GenomicBreaks::GBreak()` objects that is built
by a separate vignette (`vignette("Load genomic break data")`).

The _query_ genome is the one that was provided as a FASTA file and aligned to
the _target_ genome (the one that was indexed by the aligner).  The _target_
genome is the main part of the object, and the _query_ genome information is
contained in the metadata columns (`mcols`) of the structure.

```{r GBreaks_object}
load("BreakPoints.Rdata")
gbs$Oki_Osa
gbs$Oki_Osa$query
```

In the example displayed above, the object contains the pairwise alignment
between genomes of two dioceous _Oikopleura_ individuals; one from the Osaka
assembly `OSKA2016` (_target_) and one from the Okinawa assembly `OKI2018_I69`
(_query_).

### Long and short arms

The long and short arms are annotated in the `GBreak` objects that are built
by the vignette (`vignette("Load genomic break data")`), using the function
`GenomicBreaks::flagLongShort()`.  A copy of the arms annotations is provided
in the `longShort` list.

```{r long_and_short_arms}
longShort
longShort$Oki
```

## Breakpoints and alignment stops

We define an **alignment stop** to be a position defined in either the
_target_ or _query_ genome, where an alignment begins or ends, and a
**genomic breakpoint** (or simply breakpoint) to be a genomic structural
mutation, arising from breakage and repair of the chromosome.  Such structural
events include insertion, deletion, inversion and translocation, and often arise
during recombination. 

## Measures of distance

### Synteny index

Ad-hoc index measuring to what extent a scaffold of the _target_ genome is
mostly aligned to a single scaffold in the _query_ genome.   See
`?GenomicBreaks::synteny_index` for details.  One limitation to the use of this
index is that it requires that at least the _query_ genome is a complete
chromosome assembly.

```{r}
sapply(gbs, synteny_index)
sapply(gbs, \(g) swap(g) |> synteny_index())
```

### Correlation

Ad-hoc index measuring the correlation of the coordinates of the syntenic
alignments in scaffolds of a _target_ genome and their best match in the
_query_ genome.  See `?GenomicBreaks::correlation_index` for details.  It is
intended to be more robust to the presence of uncollapsed haplotypes in the
_query_ genome but probably needs further testing.

```{r}
sapply(gbs, correlation_index)
sapply(gbs, \(g) swap(g) |> correlation_index())
```

### Gene Order Conservation

See `?GOC` for details.

```{r goc}
sapply(gbs, GOC)
sapply(gbs, \(g) swap(g) |> GOC())
```

## Phylogenetic cladogram

We assume that the _North Pacific_ and the _Atlantic_ species are more related
to each other than to the _Okinawan_ species.

```{r phylogenetic_tree, fig.height=4, fig.width=6, dev="svg"}
requireNamespace("ade4")
treeLeaf <- function(name, length=NULL) {
  if(!is.null(length)) length <- paste0(':', length)
  paste0(name, length)
}

treeNode <- function(branch1, branch2, length = NULL) {
  if(!is.null(length)) length <- paste0(':', length)
  paste0('(', branch1, ',', branch2, ')', length)
}

addRoot <- function(branch)  paste0(branch, ";")

tree <-
  addRoot(
    treeNode(
      treeNode( length = 2,
        treeLeaf("Okinawa", 1),
        treeLeaf("Kume", 1)
      ),
      treeNode( length = 1,
        treeNode( length = 1,
          treeLeaf("Osaka", 1),
          treeLeaf("Aomori", 1)
        ),
        treeNode( length =1,
          treeLeaf("Norway", 1),
          treeLeaf("Barcelona", 1)
        )
      )
    )
  )

plot(ade4::newick2phylog(tree))
```

## Visualising alignments

We can use the function `scaf_align_plot` to see the alignment of query
scaffolds/chromosomes onto a specific target scaffold/chromosome of interest.

```{r echo=T, message=F, warning=F}
scaf_align_plot(gbs$Osa_Oki, scaf = "Chr1")
scaf_align_plot(gbs$Oki_Osa, scaf = "chr1")
```

Here we have plotted the alignments on the Okinawa genome (y-axis) onto the Osaka genome, scaffold 2 (x-axis). The coverage of the alignment is the bar along the top. 

We can plot alignments for two of the target scaffolds side-by-side, using `align_scaf_plot-2`. This can be used in aid of superscaffolding. 

```{r echo=T, message=F, warning=F}
align_scaf_plot_2(gbs$Osa_Oki, scaf = c("Chr1","Chr2"))
```

As you can see, scaffold 2 of the query genome aligns to the majority of both target scaffolds. This could be used as evidence to support the superscaffolding of these two scaffolds - in an effort toward chromosomal assembly. 

## Coalescing alignments

Large syntenic regions can often appear cluttered with alignment breaks,
spanning just a few basepairs.  They are either an artefact (for instance in
case of incomplete purge of haplotypes) and or true breakpoint.  The algorithm
in `coalesce_contigs` is used to produce a new GRanges object with fewer
alignment breaks by coalescing alignments separated by short (user specified)
distances.  This distance need be agreeable in both the target and query genome
in order for the coalscion to happen. 

For example, coalescing gaps of less than 500 basepairs in the `gbs$Osa_Oki` alignment:

```{r coalescing algorithm}
coa <- sapply(gbs, coalesce_contigs) |> SimpleList()
sapply(gbs, length)
sapply(coa, length)
```

The resulting GRanges object has far fewer alignments and therefore far fewer
alignment stops. The algorithm is an initial step in alignment stop filtering,
with the goal of a reduced number of alignment stops that have a high
probability of being breakpoints.

### Width summary plots

The distribution of widths is not the same on long and short arms.  For this
reason, it would be unwise to filter by width on some operations such as
coalescing.

```{r width_after_coalescing}
width2df <- function(where, what, gbl) {
  gb <- gbl[[where]]
  if (length(gb) == 0) return (NULL)
  data.frame(where = where, what = what, width = width(gb))
}

df <- rbind(
  # do.call allows a list to replace "..." in a function call
  do.call(rbind, lapply(names(gbs), width2df, "aligned", gbs)),
  do.call(rbind, lapply(names(coa), width2df, "collapsed", coa))
)

ggplot(df) +
  aes(width, col = what) +
  geom_freqpoly() +
  scale_x_log10() +
  facet_wrap(~where, scales = "free_y")

gg_freq_poly <- function(DF) {
    ggplot(DF |> as("data.frame")) +
    aes(width) +
    geom_freqpoly() +
    scale_x_log10()
}

DF <- DataFrame(width = width(gbs$Oki_Osa), Type = gbs$Oki_Osa$Type, chr = seqnames(gbs$Oki_Osa))

gg_freq_poly(DF) +
  aes(col=Type) +
  facet_wrap(~chr, scales = "free_y")
```

After coalescing, width distribution becomes more clearly bimodal, with a small
peak under 500 bp, found on all arms except YSR.  The second peak shows two
different distributions for long and short arms.

Therefore, if the first peak represents events that we do not want to consider
breaking synteny (either alignment artefacts or insertion/transpositions of
short mobile elements), we can re-coalesce once after removing the shortest
intervals.

```{r width_of_coalesced_regions}
DF <- DataFrame(
  width = width(coa$Oki_Osa),
  Type = mcols(OKI_longShort)[findOverlaps(coa$Oki_Osa, OKI_longShort) |> subjectHits(), "Type"],
  chr = seqnames(coa$Oki_Osa))

gg_freq_poly(DF) +
  aes(col=Type) +
  facet_wrap(~chr, scales = "free_y")
```

```{r re_coalesce}
coa2 <- sapply(coa, coalesce_contigs, minwidth = 500) |> SimpleList()
sapply(coa2, length)

DF <- DataFrame(
  width = width(coa2$Oki_Osa),
  Type = mcols(OKI_longShort)[findOverlaps(coa2$Oki_Osa, OKI_longShort) |> subjectHits(), "Type"],
  chr = seqnames(coa2$Oki_Osa))

gg_freq_poly(DF) +
  aes(col=Type) +
  facet_wrap(~chr, scales = "free_y")
```

## Genome plots with genoPlotR

```{r test_genoPlotR}
library("genoPlotR")

example("dna_seg")

gr2dna_seg <- function (gr) {
  if (is.null(names(gr))) names(gr) <- as.character(gr)
  strand(gr[strand(gr) == "*"]) <- "+"
  df <- data.frame(
    name   = names(gr),
    start  = start(gr),
    end    = end(gr),
    strand = strand(gr),
    col    = "grey"
  )
  dna_seg(df)
}

gbRef2dna_seq <- function (gb) {
  if (is.null(names(gb))) names(gb) <- as.character(gb)
  strand(gb) <- "+"
  gr2dna_seg(gb)
}

gbQuery2dna_seq <- function (gb) {
  if (is.null(names(gb$query))) names(gb$query) <- as.character(gb)
  strand(gb$query) <- strand(gb)
  gr2dna_seg(gb$query)
}

# Plot whole chromosome 1
ds <- gr2dna_seg(coa$Oki_Osa[seqnames(coa$Oki_Osa) == "chr1"])
head(ds) ; tail(ds)
plot_gene_map(list(ds))

gb2comp <- function(gb) {
  ref <- gr2dna_seg(gb)
  strand(gb$query) <- "+"
  que <- gr2dna_seg(gb$query)
  df <- data.frame(
    start1 = ref$start,
    end1   = ref$end,
    start2 = que$start,
    end2   = que$end
  )
  as.comparison(df)
}

plotApairOfChrs <- function(gb, chr) {
  gb <- gb[seqnames(gb) == chr]
    keepMainMatch <- function(gb) {
    bestMatch <- tapply(width(gb$query), seqnames(gb$query), sum) |> sort() |> tail(1) |> names()
    gb[seqnames(gb$query) == bestMatch]
  }
  roi <- keepMainMatch(gb)
  dsList <-   list(
      roi |> gbQuery2dna_seq(),
      roi |> gbRef2dna_seq()
      )
  
  compList <- list(
      roi |> swap() |> gb2comp()
    )
  plot_gene_map(dsList, compList)
}

plotApairOfChrs(coa$Oki_Bar, "chr1")
plotApairOfChrs(coa$Oki_Osa, "chr1")
plotApairOfChrs(coa$Osa_Oki, "Chr1")
plotApairOfChrs(coa$Osa_Bar, "Chr1")
plotApairOfChrs(coa2$Osa_Bar, "Chr1")
```


## Combine alignments to the same outgroup

Since Okinawa is the outgroup, let's look for regions of it that are aligned
on both the Osaka and the Norway genomes.

```{r combine_GBs}
x <- subsetByOverlaps(coa2$Oki_Osa, coa2$Oki_Bar)
y <- subsetByOverlaps(coa2$Oki_Bar, coa2$Oki_Osa)
findOverlapPairs(x,y)
subsetByOverlaps(coa2$Oki_Osa, gaps(coa2$Oki_Bar))

```

## Extract ranges from the same target region in two GenomicBreaks objects

```{r intersect_GBs}
#roi <- GRanges("chr1:1222000-1240000")
#roi <- GRanges("chr1:1023578-1032430")
#roi <- GRanges("chr1:2668866-2692347")
roi <- GRanges("chr1:6379932-6505436")
#roi <- GRanges("chr1:6683567-6744168")
roi <- GRanges("chr1:7711018-7734330")
#roi <- GRanges("chr1:7711018-7734330") + 10000

roi_Oki_O <- subsetByOverlaps(coa2$Oki_Osa, roi)
roi_Oki_N <- subsetByOverlaps(coa2$Oki_Nor, roi)
roi2 <- range(subsetByOverlaps(coa2$Oki_Osa, roi)$query)
roi_O_N <- subsetByOverlaps(coa2$Osa_Nor, roi2)
```

### Plot a simple region.

This region can be plotted because there is a one-to-one correspondence
between alignments to Osaka and alignments to Norway.

```{r plot_intersect_GBs}
roiNameO <- seqlevelsInUse(roi_Oki_O$query)[1]
roiNameOki <- seqlevelsInUse(roi)[1]
roiNameN <- seqlevelsInUse(roi_Oki_N$query)[1]

dsList <-   list(
    roi_Oki_O[seqnames(roi_Oki_O$query) == roiNameO] |> gbQuery2dna_seq(),
    roi_Oki_O[seqnames(roi_Oki_O$query) == roiNameO] |> gbRef2dna_seq(),
    roi_Oki_N[seqnames(roi_Oki_N$query) == roiNameN] |> gbRef2dna_seq(),
    roi_Oki_N[seqnames(roi_Oki_N$query) == roiNameN] |> gbQuery2dna_seq()
    )
names(dsList) <- c(roiNameO, roiNameOki, roiNameOki, roiNameN)

tmpgb <- roi_Oki_O
tmpgb$query <- granges(roi_Oki_N)

compList <- list(
    roi_Oki_O[seqnames(roi_Oki_O$query) == roiNameO] |> swap() |> gb2comp(),
    tmpgb |> gb2comp(),
    roi_Oki_N[seqnames(roi_Oki_N$query) == roiNameN] |> gb2comp()
  )

plot_gene_map(dsList, compList, dna_seg_scale=TRUE)
```

```{r plot_intersect_GBs_with_tree}
dsList <-   list(
    roi_Oki_N[seqnames(roi_Oki_N$query) == roiNameN] |> gbRef2dna_seq(),
    roi_Oki_O[seqnames(roi_Oki_O$query) == roiNameO] |> gbQuery2dna_seq(),
    roi_Oki_N[seqnames(roi_Oki_N$query) == roiNameN] |> gbQuery2dna_seq()
    )
names(dsList) <- c(roiNameOki, roiNameO, roiNameN)

compList <- list(
    roi_Oki_O[seqnames(roi_Oki_O$query) == roiNameO] |> gb2comp(),
    roi_O_N[seqnames(roi_O_N$query) == roiNameN] |> gb2comp()
  )

plot_gene_map(dsList, compList, dna_seg_scale=TRUE, tree=ade4::newick2phylog(addRoot(treeNode(roiNameOki, treeNode(roiNameO, roiNameN)))))
```

## Trivial inversions

### Representation

In the triplet of alignments below below, the middle block is on a different
strand than its neighbors, and they would be colinear otherwise.  This represents
an inversion.  *Without an outgroup it is not possible to know in which of the
target or query genome the break happened.*

Let's call "gaps" the unaligned regions between the blocks.

```
┌───────────────┬───────────────┬───────────────┐
│ XSR:101-200:+ │ XSR:201-300:- │ XSR:301-400:+ │  (OKI2018_I69)
└───────────────┴───────────────┴───────────────┘
┌───────────────┬───────────────┬───────────────┐
│  S1:101-200:+ │  S1:201-300:+ │  S1:301-400:+ │  (OSKA2016)
└───────────────┴───────────────┴───────────────┘
```

It is represented as:

```{r ideal_breakpoints}
example("flagInversions")

library("Gviz")
options(ucscChromosomeNames=FALSE)
gen <- GenomeAxisTrack(name = "genome")
trk <- AnnotationTrack(inv, name = "inv")
plotTracks(list(gen, trk))
```


### Number of trivial inversions

More inversions are found after coalescing colinear blocks because of situations
where `+ - +` was `+ - - +` before collapsing. 

```{r detect_trivial_invertions}
sapply(gbs, function(gb) sum(flagInversions(gb)$inv))
sapply(coa, function(gb) sum(flagInversions(gb)$inv))
sapply(coa2, function(gb) sum(flagInversions(gb)$inv))
```

### Isolate the left-side gaps in inversions

```{r search_for_inversions}
lgaps <- function(reduced_gr) {
  # Flag inversions.
  #  WHAT FOLLOWS ASSUMES THAT THE FLAG IS ON THE LEFT-SIDE BLOCK OF THE TRIPLE
  lgaps <- flagInversions(reduced_gr)
  # Extract inversions
  invs <- lgaps[lgaps$inv]
  # Extend inversion of 1 bp so that they overlap with their neighbor gap
  invs <- shift(invs, 1)
  # Remove strand information
  strand(lgaps) <- "*"
  # Then extract gap positions ignoring strand
  lgaps <- gaps(lgaps)
  # Remove bogus gaps on + and - strands
  lgaps <- lgaps[strand(lgaps) == "*"] 
  # Return the gaps overlapping with the flagged inversions
  subsetByOverlaps(lgaps, invs)
}

# Using coa and not gbs for reason on better detection explained above.
inv.lgaps <- sapply(coa, lgaps) |> SimpleList()

data.frame(width=width(inv.lgaps$Oki_Osa)) |>
  ggplot(aes(width)) + geom_histogram() +  scale_x_log10()

inv.lgaps.Seq <- SimpleList()

inv.lgaps.Seq$Oki_Osa <- BSgenome::getSeq(OKI2018_I69,  inv.lgaps$Oki_Osa)
inv.lgaps.Seq$Oki_Kum <- BSgenome::getSeq(OKI2018_I69,  inv.lgaps$Oki_Kum)
inv.lgaps.Seq$Osa_Oki <- BSgenome::getSeq(OSKA2016v1.9, inv.lgaps$Osa_Oki)
inv.lgaps.Seq$Osa_Aom <- BSgenome::getSeq(OSKA2016v1.9, inv.lgaps$Osa_Aom)
inv.lgaps.Seq$Oki_Bar <- BSgenome::getSeq(OKI2018_I69,  inv.lgaps$Oki_Bar)
inv.lgaps.Seq$Bar_Oki <- BSgenome::getSeq(Bar2_p4,      inv.lgaps$Bar_Oki)
```

### AT richness

```{r at_richness}
## Inverted regions are AT-rich as usual.
invSeq <- SimpleList()
invSeq$Oki_Osa <- BSgenome::getSeq(OKI2018_I69,  filterInversions(flagInversions(coa$Oki_Osa)))
invSeq$Oki_Bar <- BSgenome::getSeq(OKI2018_I69,  filterInversions(flagInversions(coa$Oki_Bar)))
invSeq$Oki_Kum <- BSgenome::getSeq(OKI2018_I69,  filterInversions(flagInversions(coa$Oki_Kum)))
invSeq$Kum_Oki <- BSgenome::getSeq(KUM_M3,  swap(filterInversions(flagInversions(coa$Oki_Kum))))
invSeq$Osa_Oki <- BSgenome::getSeq(OSKA2016v1.9, filterInversions(flagInversions(coa$Osa_Oki)))
invSeq$Osa_Bar <- BSgenome::getSeq(OSKA2016v1.9, filterInversions(flagInversions(coa$Osa_Bar)))
invSeq$Osa_Kum <- BSgenome::getSeq(OSKA2016v1.9, filterInversions(flagInversions(coa$Osa_Kum)))
invSeq$Osa_Aom <- BSgenome::getSeq(OSKA2016v1.9, filterInversions(flagInversions(coa$Osa_Aom)))
invSeq$Bar_Osa <- BSgenome::getSeq(Bar2_p4,      filterInversions(flagInversions(coa$Bar_Osa)))
invSeq$Bar_Oki <- BSgenome::getSeq(Bar2_p4,      filterInversions(flagInversions(coa$Bar_Oki)))
invSeq$Bar_Kum <- BSgenome::getSeq(Bar2_p4,      filterInversions(flagInversions(coa$Bar_Kum)))

sapply(invSeq, \(seq) letterFrequency(seq, "AT", as.prob = TRUE) |> summary())
genomes.AT

## But gaps are a bit more AT-rich.
sapply(inv.lgaps.Seq, \(seq) letterFrequency(seq, "AT", as.prob = TRUE) |> summary())

## This difference is significant.
t.test(letterFrequency(invSeq$Oki_Osa, "AT", as.prob = TRUE), letterFrequency(inv.lgaps.Seq$Oki_Osa, "AT", as.prob = TRUE))
t.test(letterFrequency(invSeq$Osa_Oki, "AT", as.prob = TRUE), letterFrequency(inv.lgaps.Seq$Osa_Oki, "AT", as.prob = TRUE))
t.test(letterFrequency(invSeq$Oki_Bar, "AT", as.prob = TRUE), letterFrequency(inv.lgaps.Seq$Oki_Bar, "AT", as.prob = TRUE))
t.test(letterFrequency(invSeq$Oki_Kum, "AT", as.prob = TRUE), letterFrequency(inv.lgaps.Seq$Oki_Kum, "AT", as.prob = TRUE))
```

I was tempted to explain this difference with A/T homopolymer stetches.  But
de novo motif prediction with rGADEM finds AT-rich motifs in any set of Oik
sequences that I have tried.

```{r search_for_motifs_in_inversions}
library("rGADEM")
# Slow !
# Note that GADEM crashes on sequences longer than 50 kbp.
gadems <- SimpleList()
gadems$Oki_Osa <- GADEM(inv.lgaps.Seq$Oki_Osa[width(inv.lgaps.Seq$Oki_Osa) < 50000], verbose=1, genome=OKI2018_I69)
gadems$Oki_Kum <- GADEM(inv.lgaps.Seq$Oki_Kum[width(inv.lgaps.Seq$Oki_Kum) < 50000], verbose=1, genome=OKI2018_I69)

gadems$Osa_Oki <- GADEM(inv.lgaps.Seq$Osa_Oki[width(inv.lgaps.Seq$Osa_Oki) < 50000], verbose=1, genome=OSKA2016v1.9)
gadems$Osa_Aom <- GADEM(inv.lgaps.Seq$Osa_Aom[width(inv.lgaps.Seq$Osa_Aom) < 50000], verbose=1, genome=OSKA2016v1.9)


gadems$Kum_Oki <- GADEM(invSeq$Kum_Oki[width(invSeq$Kum_Oki) < 50000], verbose=1, genome=KUM_M3)
gadems$Osa_Oki <- GADEM(invSeq$Osa_Oki[width(invSeq$Osa_Oki) < 50000], verbose=1, genome=OSKA2016v1.9)
# gadem <- GADEM(invSeq_Oki_O[width(invSeq_Oki_O) < 50000], verbose=1, genome=OKI2018_I69)
# consensus(gadem)
# [1] "nAAAAwwnnnrAAAAn"

# gadem <- GADEM(lgapsSeq_Oki_O, verbose=1, genome=OKI2018_I69)
# consensus(gadem)
# [1] "rAAGCsGCwwmkCGrCTTyn" "nAAAAwwnnwnAAAAAn"    "nTTTsAAAAw"           "nCCTGAsTCAGGkTTTn"    "nTTTTTnnnnrAAAn"      "wTAAGGGTrTCm"

#saveRDS(gadem, file = "lgapsSeq_Oki_O_gadem.rds")
gadem <- readRDS("lgapsSeq_Oki_O_gadem.rds")

for (n in seq_along(gadem@motifList)) seqLogo::seqLogo(rGADEM::getPWM(gadem)[[n]])
pwm <- rGADEM::getPWM(gadem)[[1]]

#RGadem positive control: check introns ?
# set.seed(1664)
# intr <- sample(intronicParts(tx_OKI), 1000)
# intr <- BSgenome::getSeq(OKI2018_I69, intr)
# gadem <- GADEM(intr, verbose=1, genome=OKI2018_I69)
# > consensus(gadem)
# [1] "nAAAAwnnwnAAAwn"
# Also found motifs like rryCAATTbwTkCGmAkyT


#RGadem positive control: check promoters ?
# set.seed(1664)
# prom <- sample(promoters(tx_OKI), 1000)
# prom <- BSgenome::getSeq(OKI2018_I69, prom)
# gadem <- GADEM(prom, verbose=1, genome=OKI2018_I69)
# consensus(gadem)
# [1] "nAAAAwwnwnnAAAAwn" "nTTTTCTTyn"       
# Also found motifs like TTTmAAAA
```

A palindromic (rAAGCsGCwwmkCGrCTTyn) motif is found in the left gaps.

Or could it be `GCCGCnnnGCGGC` ?  Its frequency in genomes is something like:
```
  Oki   Osa   Bar   Kum   Aom   Nor 
10827  2273  1967 10880  2210  2247 
```

On the other hand, a motif containing GCGAAGCGAA identified in Osa_Oki shows:

```
  Oki   Osa   Bar   Kum   Aom   Nor 
  483 21360 23340   501 21594 16413 
```
  
```{r screenJaspar}
checkJaspar <- function (pwm) {
  #---- see https://compgenomr.github.io/book/motif-discovery.html
  pwmLib <- getMatrixSet(
    JASPAR2018::JASPAR2018,
    opts = list(
      collection = 'CORE',
   #   species    = 'Homo sapiens',
      matrixtype = 'PWM'))
  
  pwm_sim <- PWMSimilarity( pwmLib, PWMat,  method = 'Pearson')

  pwmLibDf <- data.frame(
    ID        = sapply(pwmLib, ID),
    name      = sapply(pwmLib, name),
    row.names = seq_along(pwmLib)
  )
  pwmLibDf$similarity <- pwm_sim[pwmLibDf$ID]
  pwmLibDf[order(-pwmLibDf$similarity),]
}

PWMat <- PWMatrix(ID="rAAGCsGCwwmkCGrCTTyn", profileMatrix = pwm)
pwm_res <- checkJaspar(PWMat)
head(pwm_res)

pwmhalf <- pwm[,2:8]
pwmhalf_res <- checkJaspar(PWMatrix(ID="AAGCnGC", profileMatrix = pwmhalf))
head(pwmhalf_res)
```

But a search in Jaspar with either the full motif or its halof did not reveal
strong hits.  Visual inspection of PRDM9 sequence logos via Google did not show
similarity either.

### Export rAAGCsGCwwmkCGrCTTyn alignments to a file

```{r pwmHitSequenceExportForGLAM2}
gademHitsToBS <- function(gadem, n) {
  alignList <- gadem[[n]]@alignList
  pwmSeqs <- sapply(alignList, function(align) align@seq)
  Biostrings::DNAStringSet(pwmSeqs)
}

pwmSeqs <- gademHitsToBS(gadem, 1)
writeXStringSet(pwmBS, file="firstGapsGADEMpwm.fasta")

# writeXStringSet(lgapsSeq_Oki_O, file="firstGaps.fasta")
# I used this FASTA export to search for motifs with GLAM2, and it also
# returned A/T polymers.
```

The GLAM2 motif finder was run on this file, and then to screen the genome.
Hits coordinates were encoded in a BED file uploaded to the ZENBU genome browser.
Visual inspection gave me the impression that the motif might be related to
repeats.

GLAM2 search of the gap sequences did not return a motif similar to "AAGCsGCwwmkCGrCTTyn".

### Whole-genome search of the motif

Disappointingly, the motif is found equally in gaps and inversions when searched
with its PWM.

```{r }
# No exact match
vmatchPattern("AAGCsGCwwmkCGrCTTyn", OKI2018_I69)

pwmHits <- sapply(genomes, matchPWM, pwm = pwm)
sapply(pwmHits, length)
hist(pwmHits$Oki$score)
hist(pwmHits$Osa$score)


# Less hits, with lower scores, in the mm10 genome.
pwmHitsmm10 <- matchPWM(pwm, BSgenome.Mmusculus.UCSC.mm10::BSgenome.Mmusculus.UCSC.mm10)

length(subsetByOverlaps(pwmHits$Oki, lgaps_Oki_O))
length(subsetByOverlaps(pwmHits$Oki, invs))

hist(score(subsetByOverlaps(pwmHits$Oki, lgaps_Oki_O)))
hist(score(subsetByOverlaps(pwmHits$Oki, invs)))

# Scores in gaps are higher than average.
t.test(score(pwmHits$Oki), score(subsetByOverlaps(pwmHits$Oki, lgaps_Oki_O)))

# Scores in gaps and inversions are not different 
t.test(score(subsetByOverlaps(pwmHits$Oki, invs)), score(subsetByOverlaps(pwmHits$Oki, lgaps_Oki_O)))

rtracklayer::export.bed(pwmHits, "AAGCsGCwwmkCGrCTTyn.bed")

# PWM hits seem to be more abundant on short arms.
OKI_longShort$pwmHitsPerMb <- countOverlaps(OKI_longShort, pwmHits$Oki) / width(OKI_longShort) * 1e6
# But with a naive approach the difference is not statistically significant.
t.test(c(159, 185, 188), c(77, 142, 160))
```

### Representation of some inversions

```{r}
x <- coa$Oki_Osa |> flagInversions() |> dist2next(ignore.strand = TRUE)

pmpcoord <- showInversions(x)

seqinfo2gieStain <- function(si) {
  data.frame(
    chrom      = seqlevels(si),
    chromStart = 0,
    chromEnd   = seqlengths(si),
    name       = seqlevels(si),
    gieStain   = "gneg"
  )
}

#r <- 63:70
r <- 94:98
#r <- 316:319
options(ucscChromosomeNames=FALSE)
gen <- GenomeAxisTrack(name="OKI2018_I69")
ide <- IdeogramTrack(chromosome = seqlevelsInUse(x[r]), genome = "OKI2018_I69", bands = seqinfo2gieStain(OKI2018_I69))
trk <- AnnotationTrack(x[r], name = "OKI2018_I69", id=LETTERS[seq_along(r)])
p1 <- grid::grid.grabExpr(plotTracks(list(ide, gen, trk), featureAnnotation = "id"))

gen2 <- GenomeAxisTrack(name="OSKA2016v1.9")
ide2 <- IdeogramTrack(chromosome = seqlevelsInUse(x$query[r]), genome = "OSKA2016v1.9", bands = seqinfo2gieStain(OSKA2016v1.9))
trk2 <- AnnotationTrack(x[r]$query, name = "OSKA2016v1.9", id=LETTERS[seq_along(r)])
p2 <- grid::grid.grabExpr(plotTracks(list(ide2,gen2,trk2), featureAnnotation = "id"))

gridExtra::grid.arrange(p1, p2, nrow=2)
```

## Trivial insersions

### Representation

The in<b>s</b>ersion below:

```
┌───────────────┬───────────────┬───────────────┐
│ XSR:101-200:+ │ XSR:201-300:+ │ XSR:301-400:+ │  (OKI2018_I69)
└───────────────┴───────────────┴───────────────┘
┌───────────────┬───────────────┬───────────────┐
│  S1:101-200:+ │  S2:801-900:+ │  S1:201-300:+ │  (OSKA2016)
└───────────────┴───────────────┴───────────────┘
```

Is represented as:

```{r ideal_insertions}
example("flagInsersions")

library("Gviz")
options(ucscChromosomeNames=FALSE)
gen <- GenomeAxisTrack(name = "genome")
trk <- AnnotationTrack(ins, name = "ins")
plotTracks(list(gen, trk))
```

*Without an outgroup it is not possible to know in which genome the break happened.*

### Detection

```{r detect_trivial_insertions}
sapply(gbs, function(gb) sum(flagInsersions(coalesce_contigs(gb))$ins))

y <- flagInsersions(reduced_gr_Oki_O)

#r <- 63:70
r <- 23:30
#r <- 316:319
options(ucscChromosomeNames=FALSE)
gen <- GenomeAxisTrack(name="OKI2018_I69")
ide <- IdeogramTrack(chromosome = seqlevelsInUse(x[r]), genome = "OKI2018_I69", bands = seqinfo2gieStain(OKI2018_I69))
trk <- AnnotationTrack(x[r], name = "OKI2018_I69", id=LETTERS[seq_along(r)])
p1 <- grid::grid.grabExpr(plotTracks(list(ide, gen, trk), featureAnnotation = "id"))

gen2 <- GenomeAxisTrack(name="OSKA2016")
ide2 <- IdeogramTrack(chromosome = seqlevelsInUse(x$query[r])[1], genome = "OSKA2016", bands = seqinfo2gieStain(OSKA2016))
trk2 <- AnnotationTrack(x[r]$query, name = "OSKA2016", id=LETTERS[seq_along(r)])
p2 <- grid::grid.grabExpr(plotTracks(list(ide2,gen2,trk2), featureAnnotation = "id"))


gen3 <- GenomeAxisTrack(name="OSKA2016")
ide3 <- IdeogramTrack(chromosome = seqlevelsInUse(x$query[r])[2], genome = "OSKA2016", bands = seqinfo2gieStain(OSKA2016))
trk3 <- AnnotationTrack(x[r]$query, name = "OSKA2016", id=LETTERS[seq_along(r)])
p3 <- grid::grid.grabExpr(plotTracks(list(ide3,gen3,trk3), featureAnnotation = "id"))

gridExtra::grid.arrange(p1, p2, p3, nrow=3)
```

## Tandem repeats

Repeated regions of genomes are notoriously difficult to assemble and align. Thus, information about an alignment stop's proximity to a tandem repeat is useful information in breakpoint classification. Given a user-defined tolerance, the function `tan_bp` will classify alignment stops as either far from, close to, or within tandem repeats (elements 1,2 and 3 of the produced list, respectively). The function requires the tandem repeat coverage of one of the genomes. By default, the function will expect the target genome tandem coverage. Setting `query_tf = TRUE` will switch the outcome to be query genome oriented.

```{r echo=T, message=F, warning=F}

tan_O <- rtracklayer::import.gff3(system.file("extdata/OSKA2016.tantanf4w100.gff3.gz",  package="GenomicBreaks"))
tan_O <- GRanges(tan_O, seqinfo = seqinfo(OSKA2016))
tan_gr_O <- tan_bp(gr_ob = gr_O_Oki, tan = tan_O, tol = 50)
tan_gr_O

tan_Oki <- rtracklayer::import.gff3(system.file("extdata/I69-5.tantanf4w100.gff3.gz",  package="GenomicBreaks"))
tan_Oki <- GRanges(tan_Oki, seqinfo = seqinfo(OKI2018_I69))
tan_gr_Oki <- tan_bp(gr_ob = gr_O_Oki, tan = tan_Oki, tol = 50, query_tf = TRUE)
tan_gr_Oki

```

We can use another script to plot the density of tandem repeats around alignment stops (and breakpoints). In the following plots, breakpoints are aligned at the centre of the plots. Intuitively, this function can be generalised to see the density of any binary coverage characteristic around breakpoints, given a GRanges object with per-base coverage of a given characteristic. Here we stick with tandem repeat info:

```{r echo=T, message=F, warning=F}
hm_tan_O <- gr_O_Oki %>%
  feature_coverage(tan_O, win = 1000, lab = "Osaka ~ Okinawa", direction = "left") %>%
  smoothHeatmap(output.size = c(2000,500), algorithm = "kernel") %T>%
  plotHeatmap

hm_red_tan_O <- reduced_gr_O_Oki %>%
  feature_coverage(tan_O, win = 1000, lab = "Osaka ~ Okinawa", direction = "left") %>%
  smoothHeatmap(output.size = c(2000,500), algorithm = "kernel") %T>%
  plotHeatmap

hm_tan_Oki <- gr_Oki_O %>%
  feature_coverage(tan_Oki, win = 1000, lab = "Osaka ~ Okinawa", direction = "left") %>%
  smoothHeatmap(output.size = c(2000,500), algorithm = "kernel") %T>%
  plotHeatmapList
```
## Coverage

Another characteristic of alignment stops that we can examine is the coverage depth over said alignment stops. By aligning raw reads to assembled genomes, we can obtain per-base coverage depth for target or query genome. Low coverage directly on and surrounding an alignment stop may suggest unreliability in it being a true breakpoint. The function `bp_coverage` returns a GRanges object of the bps, with associated averaged and point converage in the metadatacolumns. 

```{r echo=T, message=F, warning=F}

# cov_gr_O_Oki <- bp_coverage(gr_ob = q_Oki, cov_gr = Oki_cov_pb, win = 50)
# cov_gr_O_Oki

```


## Breakpoint classification - master script

The function `master_bp_analysis` brings together the information about tandem repeats and coverage, as well as coalescing the alignment. It can perform the analyses for both the target and the query genome, depending on the available input data. The output is split into two GRanges objects; one for the target genome and one for the query, with associated information in the metadata columns. 

```{r echo=T, message=F, warning=F}

# fin_gr_O_Oki <- master_bp_analysis(pair_gr = gr_O_Oki, co_tol = 500, ref_tan = tan_O, ref_tan_tol = 50, q_tan = tan_Oki,
#                                    q_tan_tol = 50, q_cov = Oki_cov_pb, q_cov_tol = 50)
# fin_gr_O_Oki

```


# Examples

## Nucleaic acid content heatmaps

The _GenomicBreaks_ function `bp_heatmap` plots nucleic acid content in windows
centrered on alignment stops.  Lets look at `GC` content before and after
coalescing: 

```{r echo=T, message=F, warning=F}
library(heatmaps)
hm_GC1 <- sapply(gbs, bp_heatmap, basep_range = 1000, pat = "GC", direction = "left") |> SimpleList()
hm_GC2 <- sapply(coa, bp_heatmap, basep_range = 1000, pat = "GC", direction = "left") |> SimpleList()

maxHeight <- 2000
sm_GC1 <- sapply(hm_GC1, smoothHeatmap, output.size = c(maxHeight,     500), algorithm = "kernel") |> SimpleList()
sm_GC2 <- sapply(hm_GC2, smoothHeatmap, output.size = c(maxHeight,     500), algorithm = "kernel") |> SimpleList()
sm_GC3 <- sapply(hm_GC3, smoothHeatmap, output.size = c(maxHeight,     500), algorithm = "kernel") |> SimpleList()

plotHeatmapList(sm_GC1)
plotHeatmapList(sm_GC2)
plotHeatmapList(sm_GC3)
```

The alignment stops are ordered in such a way that start (or left breaks) are
centred on the top half of the plot, and end alignment stops are centred on the
bottom. This is why we see different directionality in the plots. At the
alignment stops, the GC content seems to be lower, but higher once within the
aligned region. 

Now looking at TATA content:

```{r TATA}
hm_GC1_TATA <- sapply(gbs, bp_heatmap, basep_range = 1000, pat = "TATA", direction = "left") |> SimpleList()
hm_GC2_TATA <- sapply(coa, bp_heatmap, basep_range = 1000, pat = "TATA", direction = "left") |> SimpleList()

maxHeight <- 2000
sm_GC1_TATA <- sapply(hm_GC1_TATA, smoothHeatmap, output.size = c(maxHeight,     500), algorithm = "kernel") |> SimpleList()
sm_GC2_TATA <- sapply(hm_GC2_TATA, smoothHeatmap, output.size = c(maxHeight,     500), algorithm = "kernel") |> SimpleList()
sm_GC3_TATA <- sapply(hm_GC3_TATA, smoothHeatmap, output.size = c(maxHeight,     500), algorithm = "kernel") |> SimpleList()

plotHeatmapList(sm_GC1_TATA)
plotHeatmapList(sm_GC2_TATA)
plotHeatmapList(sm_GC3_TATA)
```

The directionaility here is consistent with the `GC` content analysis.
Furthermore; "TATA box sequence can act as a basal promoter element not only for
RNA polymerase II (RNAP II) transcription, but also for transcription by RNA
polymerase III (RNAP III)" - Wang Y, Jensen RC, Stumph WE. Role of TATA box
sequence and orientation in determining RNA polymerase II/III transcription
specificity. Nucleic Acids Res. 1996;24(15):3100–3106.
doi:10.1093/nar/24.15.3100. However, the areas are not particularly enriched
for TATA boxes, so it does not necessarily show that breakpoints are occurring
directly after promoter regions. 


```{r AAGCsGCwwmkCGrCTTyn}
hm_GC1_mot <- sapply(gbs, bp_heatmap, basep_range = 1000, pat = "AGCNGC", direction = "left") |> SimpleList()
hm_GC2_mot <- sapply(coa, bp_heatmap, basep_range = 1000, pat = "AGCNGC", direction = "left") |> SimpleList()

maxHeight <- 2000
sm_GC1_mot <- sapply(hm_GC1_mot, smoothHeatmap, output.size = c(maxHeight,     500), algorithm = "kernel") |> SimpleList()
sm_GC2_mot <- sapply(hm_GC2_mot, smoothHeatmap, output.size = c(maxHeight,     500), algorithm = "kernel") |> SimpleList()
sm_GC3_mot <- sapply(hm_GC3_mot, smoothHeatmap, output.size = c(maxHeight,     500), algorithm = "kernel") |> SimpleList()

plotHeatmapList(sm_GC1_mot)
plotHeatmapList(sm_GC2_mot)
plotHeatmapList(sm_GC3_mot)
```


## Cluster analysis


```{r cluster_analysis, fig.height=20, fig.width=20}
# Accessory functions
clusterHeatmap <- function(hm, k = 2) kmeans(image(hm), k)

orderHeatmap <- function(hm, cl) {
  image(hm) <- image(hm)[order(cl$cluster),]
  hm
}

subsetHeatmap <- function(hm, bool) {
  image(hm) <- image(hm)[bool,]
  hm
}

# DAta
hm_GC   <- bp_heatmap(      gr_Oki_O, basep_range  = 1000, pat = "GC",     direction = "left")
hm_TATA <- bp_heatmap(      gr_Oki_O, basep_range  = 1000, pat = "TATA",   direction = "left")
hm_tan  <- feature_coverage(gr_Oki_O, tan_Oki, win = 1000, lab = "Tandem", direction = "left")

# Define 3 K-mean clusters
clusterHeatmap(hm_GC, 3) -> hm_cl_k4

# Order and smooth
shm_GC <- orderHeatmap(hm_GC, hm_cl_k4) %>%
  smoothHeatmap(output.size = c(2000,500), algorithm = "kernel")

shm_TATA <- orderHeatmap(hm_TATA, hm_cl_k4) %>%
  smoothHeatmap(output.size = c(2000,500), algorithm = "kernel")

shm_tan <- orderHeatmap(hm_tan, hm_cl_k4) %>%
  smoothHeatmap(output.size = c(2000,500), algorithm = "kernel")

# Plot the ordered data
plotHeatmapList(list(shm_GC, shm_TATA, shm_tan))
par(mfrow = c(1,1)) # plotHeatmapList messes with mrfrow...
plotHeatmapMeta(list(hm_GC, hm_TATA, hm_tan))

plotHeatmapMeta(list(subsetHeatmap(hm_GC, hm_cl_k4$cluster == 1),
                     subsetHeatmap(hm_GC, hm_cl_k4$cluster == 2),
                     subsetHeatmap(hm_GC, hm_cl_k4$cluster == 3)))

kmeanOrderedPlot <- function (hm, k=3, seed = 1) {
  set.seed(seed)
  
  cl <- kmeans(image(hm), k)
  mat <- image(hm)[order(cl$cluster),]
  
  kmhm = Heatmap(
    mat,
    coords=c(-500,500),
    label="kmeans",
    scale=range(mat))
  
  plotHeatmapList(kmhm,
                  cex.label=1.5,
                  partition=as.vector(table(cl$cluster)),
                  partition.legend=TRUE,
                  partition.lines=TRUE,
                  legend=TRUE,
                  legend.pos="r",
                  legend.width=0.3)
}

kmeanOrderedPlot(shm_GC)
```

## Evidence for breakpoint hotspots

The function `bp_pair_analysis` takes two pairwise alignments with the same target genome, and plots the alignment stops on to the centred alignment stops of the other. In the plot below, Osaka is there consistent target genome, and as such acts as a coordinate system to relate alignment stops across genomes. Lets produce two of these 3-way analyses; one between Osaka-Okinawa-Norway, and one between Osaka-Okinawa-Aomori

```{r echo=T, message=F, warning=F}

# hm1 <- bp_pair_analysis(gr_ref_q1 = gr_O_Oki, gr_ref_q2 = gr_O_N, win = 1000, lab = "Oki~Nor")
# plotHeatmapMeta(hm1)
# 
# hm2 <- bp_pair_analysis(gr_ref_q1 = gr_O_Oki, gr_ref_q2 = gr_O_A, win = 1000, lab = "Oki~Aom")
# plotHeatmapMeta(hm2)

```

The accumulation of alignment breaks of one pairwise alignment onto another suggests the existence of breakpoint hotspots; regions where breaks are far more likely to occur. Intuitively, this would lead to the presumption of synteny blocks, also. The pattern is consistent after coalescing, too;

```{r echo=T, message=F, warning=F}

# reduced_gr_O_N <- coalesce_contigs(gr_ob = gr_O_N, tol = 500)
# 
# hm3 <- bp_pair_analysis(gr_ref_q1 = reduced_gr_O_Oki, gr_ref_q2 = reduced_gr_O_N, win = 1000, lab = "Oki~Nor")
# plotHeatmapMeta(hm3)

```

## Tandem repeat coverage around alignment stops (breakpoints)

The function `feature_coverage` will plot the coverage of tandem repeats around
centred alignment stops. 

```{r echo=T, message=F, warning=F}

hm_tan1 <- feature_coverage(gr_ob = gr_O_Oki, feat = tan_O, win = 4000, lab = "Osk ~ Oki", direction = "left")
plotHeatmapMeta(hm_tan1)
 
hm_tan2 <- feature_coverage(gr_ob = reduced_gr_O_Oki, feat = tan_O, win = 4000, lab = "OSK~Oki (coalesced)", direction = "left")
plotHeatmapMeta(hm_tan2)

# should I also add comparison of proportions of tandem proximities before and after coalescion?
```

There appears to be less tandem repeats around alignment stops after coalescion,
inferring that perhaps that some are artefacts of difficult alignment near to
tandem repeats. 


## Gene feature coverage around alignment stops (breakpoints)

The function `feature_coverage` will plot the coverage of tandem repeats around
centred alignment stops. 

```{r genomic_features}
#current_GB <- gr_Oki_O # To ease recomputing with another object
current_GB <- reduced_gr_Oki_O
hmList_OKI <- list()

hmList_OKI$genes      <- feature_coverage(current_GB, genes(tx_OKI),                         win = 2000, lab = "Oki genes", direction = "left")
hmList_OKI$promoters  <- feature_coverage(current_GB, promoters(tx_OKI),                     win = 2000, lab = "Oki promoters", direction = "left")
hmList_OKI$exons      <- feature_coverage(current_GB, exonicParts(tx_OKI),                   win = 2000, lab = "Oki exons", direction = "left")
hmList_OKI$introns    <- feature_coverage(current_GB, intronicParts(tx_OKI),                 win = 2000, lab = "Oki introns", direction = "left")
hmList_OKI$cds        <- feature_coverage(current_GB, cds(tx_OKI),                           win = 2000, lab = "Oki cds", direction = "left")
hmList_OKI$utrs5      <- feature_coverage(current_GB, unlist(fiveUTRsByTranscript(tx_OKI)),  win = 2000, lab = "Oki 5′ UTRs", direction = "left")
hmList_OKI$utrs3      <- feature_coverage(current_GB, unlist(threeUTRsByTranscript(tx_OKI)), win = 2000, lab = "Oki 3′ UTRs", direction = "left")
hmList_OKI$pwm        <- feature_coverage(current_GB, pwmHits                              , win = 2000, lab = "PWM", direction = "left")

plotHeatmapMeta(hmList_OKI)
plotHeatmapMeta(hmList_OKI$genes)
plotHeatmapMeta(hmList_OKI[c("exons", "cds")])
plotHeatmapMeta(hmList_OKI[c("exons", "introns", "utrs5", "utrs3", "pwm")])
plotHeatmapMeta(hmList_OKI[c("introns", "utrs5", "utrs3", "pwm")])
plotHeatmapMeta(hmList_OKI[c("utrs5", "utrs3", "pwm")])
```

Aligned regions are enriched in exons and depleted in introns.  Is it because
exons are more alignable or is it because it is deleterous to break genes
in exons ?

## Gene feature coverage around PWM hits

Let's look at coverage around PWM hits of the AAGCsGCwwmkCGrCTTyn motif

```{r genomic_features_pwmHits}
current_GB <- pwmHits

hmList_PWM_OKI <- list()
hmList_PWM_OKI$genes     <- feature_coverage(current_GB, genes(tx_OKI),                         win = 2000, lab = "Oki genes", direction = "left")
hmList_PWM_OKI$promoters <- feature_coverage(current_GB, promoters(tx_OKI),                     win = 2000, lab = "Oki promoters", direction = "left")
hmList_PWM_OKI$exons     <- feature_coverage(current_GB, exonicParts(tx_OKI),                   win = 2000, lab = "Oki exons", direction = "left")
hmList_PWM_OKI$introns   <- feature_coverage(current_GB, intronicParts(tx_OKI),                 win = 2000, lab = "Oki introns", direction = "left")
hmList_PWM_OKI$cds       <- feature_coverage(current_GB, cds(tx_OKI),                           win = 2000, lab = "Oki cds", direction = "left")
hmList_PWM_OKI$utrs5     <- feature_coverage(current_GB, unlist(fiveUTRsByTranscript(tx_OKI)),  win = 2000, lab = "Oki 5′ UTRs", direction = "left")
hmList_PWM_OKI$utrs3     <- feature_coverage(current_GB, unlist(threeUTRsByTranscript(tx_OKI)), win = 2000, lab = "Oki 3′ UTRs", direction = "left")

plotHeatmapMeta(hmList_PWM_OKI)
plotHeatmapMeta(hmList_PWM_OKI[c("exons", "introns")])
plotHeatmapMeta(hmList_PWM_OKI[c("exons", "utrs5", "utrs3")])
plotHeatmapMeta(hmList_PWM_OKI[c("utrs5", "utrs3")])
```

They are enriched in introns...

```{r genomic_features3}

# Sandbox to make sanity checks.

a <- reduced_gr_Oki_O
b <- unlist(fiveUTRsByTranscript(tx_OKI))

hm_a <- feature_coverage(a, b, win = 2000, lab = "All ranges", direction = "left")
hm_b <- feature_coverage(subsetByOverlaps(a,  b), b, win = 2000, lab = "Subsetted ranges", direction = "left")

plotHeatmapMeta(list(hm_a))
plotHeatmapMeta(list(hm_b))
```


## Okinawa genome coverage

As previously discussed, low coverage over an alignment stop could lower the likelihood of it being considered as a breakpoint. We have per-base coverage depth information for the Okinawan genome (`Oki_cov_pb`). However, the coverage of this particular assembly is quite good. In fact, we can investigate the coverge over alignment stops from the information obtained using `master_bp_analysis`. 

```{r echo=T, message=F, warning=F}
# fin_Oki <- fin_gr_O_Oki[[2]]
# min_cov <- min(min(fin_Oki$left_cov_pb), min(fin_Oki$right_cov_pb)) # minimum coverage over an alignment stop
# min_cov
# length(fin_Oki[fin_Oki$left_cov_pb == min_cov]) + length(fin_Oki[fin_Oki$right_cov_pb == min_cov]) # how many of the minimum coverage is observed
# length(fin_Oki[fin_Oki$left_cov_pb <= 50]) + length(fin_Oki[fin_Oki$right_cov_pb <= 50]) # how many alignment stops have a coverage of less than or euqal to 50

```

We may choose to kick out the one alignment stop for which there is no coverage. However, only 15 out of 34572 alignment stops have a coverage of less than or equal to 50, meaning that it would be hard to exclude more than just a few alignment stops using coverage information for breakpoint analysis. 



