---
title: "Inversions"
author: 
 - "Charles Plessy"
date: "`r format(Sys.time(), '%d %B, %Y')`"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteEncoding{UTF-8}
  %\VignetteIndexEntry{Inversions}
  %\VignetteEngine{knitr::rmarkdown}
editor_options: 
  chunk_output_type: console
---

```{r setup}
knitr::opts_chunk$set(cache = TRUE, cache.lazy = FALSE)
knitr::opts_knit$set(verbose = TRUE)
```

# Introduction

After coalescing colinear alignments, removing translocations of
repeat-containing sequences and re-coalescing, colinearity is still broken
hundreds of time.

Here we explore the role of inversions in scrambling _Oikopleura_ genomes.

# Load R pacakges and data

```{r load_packages_and_data}
library('OikScrambling') |> suppressPackageStartupMessages()
(genomes <- OikScrambling:::loadAllGenomes())
(reps <- OikScrambling:::loadAllRepeats())
(load("BreakPoints.Rdata"))
```

See `vignette("LoadGenomicBreaks", package = "OikScrambling")` for how the
different GBreaks objects are prepared.

# Trivial inversions

## Documentation

Details can be found in `vignette("GenomicBreaks", package = "GenomicBreaks")`,
in `vignette("StructuralVariants", package = "GenomicBreaks")`,
and `?GenomicBreaks::flagInversions`

## Number of trivial inversions

More inversions are found after coalescing colinear blocks because of situations
where `+ - +` was `+ - - +` before collapsing.

The number of detected inversions is further increased after double-coalescing,
but not much.

```{r detect_trivial_invertions, fig.width=8, fig.height=6}
(invs_summary <- data.frame(
  align  = sapply(gbs,  function(gb) sum(flagInversions(gb)$inv)),
  mapped = sapply(coa,  function(gb) sum(flagInversions(gb)$inv)),
  map2   = sapply(coa2, function(gb) sum(flagInversions(gb)$inv))
))

sapply(invs_summary, \(x) tapply(x, row.names(invs_summary) |> OikScrambling:::compDistance(), mean))
sapply(invs_summary, \(x) tapply(x, row.names(invs_summary) |> OikScrambling:::compDistance(), median))
sapply(invs_summary, \(x) tapply(x, row.names(invs_summary) |> OikScrambling:::compDistance(), sd))

invs_summary$pairname   <- OikScrambling:::compDistance(rownames(invs_summary))
invs_summary$pairname_s <- OikScrambling:::compDistance(rownames(invs_summary), short = TRUE)
invs_summary$genus      <- OikScrambling:::compGenus(rownames(invs_summary))
invs_summary$class      <- OikScrambling:::compDistClass(rownames(invs_summary))
invs_summary$target     <- sub("_.*", "", rownames(invs_summary))
invs_summary <- invs_summary[invs_summary$class != "Int/Rob â€“ Oki",]

ggplot(invs_summary |> tidyr::pivot_longer(c("align", "mapped", "map2"))) +
  aes(value, pairname_s) + geom_point(aes(col = target)) +
  ggtitle ("Number of inversions found after different post-processings") +
  facet_wrap(~name, ncol= 1)
```

```{r number_of_inversions, fig.width=7, dev='svg', fig.ext='svg'}
ggplot(invs_summary |> tidyr::pivot_longer(c("mapped"))) +
  aes(value, genus) + geom_point(aes(col = class)) +
  ggtitle ("Number of inversions found in mapped regions") 
```

## Motif search

Enrichment for motifs is being searched in
`vignette("PWM", package = "OikScrambling")`.

```{r load_PWMs}
gadems.invLeftGap <- readRDS("gadems.invLeftGap.Rda")
```

