---
title: "Inversions"
author: 
 - "Charles Plessy"
date: "`r format(Sys.time(), '%d %B, %Y')`"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteEncoding{UTF-8}
  %\VignetteIndexEntry{Inversions}
  %\VignetteEngine{knitr::rmarkdown}
editor_options: 
  chunk_output_type: console
---

```{r setup}
knitr::opts_chunk$set(cache = TRUE, cache.lazy = FALSE)
knitr::opts_knit$set(verbose = TRUE)
```

# Introduction

After coalescing colinear alignments, removing translocations of
repeat-containing sequences and re-coalescing, colinearity is still broken
hundreds of time.

Here we explore the role of inversions in scrambling _Oikopleura_ genomes.

# Load R pacakges and data

```{r load_packages_and_data}
library('OikScrambling') |> suppressPackageStartupMessages()
(genomes <- OikScrambling:::loadAllGenomes())
(annots <- OikScrambling:::loadAllAnnotations() |> suppressWarnings())
(reps <- OikScrambling:::loadAllRepeats())
(load("BreakPoints.Rdata"))
```

See `vignette("LoadGenomicBreaks", package = "OikSyntenies")` for how the
different GBreaks objects are prepared.

# Trivial inversions

## Documentation

Details can be found in `vignette("GenomicBreaks", package = "GenomicBreaks")`,
in `vignette("StructuralVariants", package = "GenomicBreaks")`,
and `?GenomicBreaks::flagInversions`

## Number of trivial inversions

More inversions are found after coalescing colinear blocks because of situations
where `+ - +` was `+ - - +` before collapsing.

The number of detected inversions is further increased after double-coalescing.

```{r detect_trivial_invertions}
sapply(gbs,  function(gb) sum(flagInversions(gb)$inv))
sapply(coa,  function(gb) sum(flagInversions(gb)$inv))
sapply(coa2, function(gb) sum(flagInversions(gb)$inv))
```

## Motif search

Enrichment for motifs is being searched in
`vignette("PWM", package = "OikScrambling")`.

```{r load_PWMs}
gadems.invLeftGap <- readRDS("gadems.invLeftGap.Rda")
```

## Representation of some inversions

3 consecutive inversions will cause 5 flags to be set to TRUE.  Think:

```
+ + + + + + + + +  â†’  + - + - + - + + +
                      T T T T T F F F F
```

```{r}
ROI7 <- coa$Oki_Osa |> flagInversions() |> dist2next(ignore.strand = TRUE)

showInversions(ROI7)

ROI7[35:45] |> as.data.frame()
ROI7[37:43] |> plotApairOfChrs()

showInversions(ROI7)
```
