---
title: "Load genomic break data"
author: 
 - "Charles Plessy"
date: "`r format(Sys.time(), '%d %B, %Y')`"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteIndexEntry{Load genomic break data}
  %\VignetteEncoding{UTF-8}
editor_options: 
  chunk_output_type: console
---

```{r setup}
knitr::opts_knit$set(verbose = TRUE)
```

Introduction
============

This vignette loads the genomic breaks data, consolidates colinear regions,
extracts unaligned and unmapped regions, transfers annotation data and saves the
resulting objects in a serialised R object (`RData` file) for reuse in other
vignettes.

Related objects are stored in lists to avoid cluttering the environment.

Packages and data
=================


```{r load_pacakges_and_data}
suppressPackageStartupMessages({
  library('GenomicBreaks')
  library('ggplot2')
  library("BreakpointsData")
})
(genomes <- OikScrambling:::loadAllGenomes())
```

Genomic breaks
==============

Long/short arm information
--------------------------

Short / long arm annotation will be done by the `GenomicBreaks::flagLongShort()`
function.

```{r long_and_short_arms}

loadLongShort <- function(file, genome) {
  file <- system.file(paste0("extdata/Annotations/", file), package = "BreakpointsData")
  gr <- rtracklayer::import.gff3(file)
  gr$source <- gr$type <- gr$score <- gr$phase <- NULL
  GRanges(gr, seqinfo = seqinfo(genome), strand = "*")
}

longShort <- SimpleList()
longShort$Oki <- loadLongShort("OKI2018_I69.v2/OKI2018_I69.arms.gff3", OKI2018_I69)
longShort$Osa <- loadLongShort("OSKA2016v1.9/OSKA2016v1.9.arms.gff3", OSKA2016v1.9)
longShort$Bar <- loadLongShort("Bar2_p4.Flye/Bar2_p4.Flye.arms.gff3", Bar2_p4)

longShort
```

Repeat regions
--------------

Load from `BreakpointsData` package via `OikScrambling:::loadAllRepeats()`

```{r load_repeats}
(reps <- OikScrambling:::loadAllRepeats())

annotateWithReps.GRanges <- function(gr, rep) {
  # Inspired from CAGEr:::ranges2names
  o <- findOverlaps(gr, rep, ignore.strand = TRUE)
  o <- as(o, "List")
  o <- extractList(rep$Class |> as.character(), o)
  o <- endoapply(o, unique)
  o[sapply(o, length) == 0] <- NA
  gr$rep <- o
  gr
}

overlapLength.GRanges <- function(gr, rep) {
  # Inspired from CAGEr:::ranges2names
  redRep <- reduce(rep)
  o <- findOverlaps(gr, redRep, ignore.strand = TRUE)
  o <- as(o, "List")
  repGrl <- extractList(redRep, o)
  gr$repOvlp <- pintersect(gr, repGrl) |> width() |> sum()
  gr
}

annotateWithReps.GBreaks <- function(gb, repT, repQ) {
  gb$query <- annotateWithReps.GRanges(gb$query, repQ)
  gb       <- annotateWithReps.GRanges(gb,       repT)
  gb$query <- overlapLength.GRanges   (gb$query, repQ)
  gb       <- overlapLength.GRanges   (gb,       repT)
}

annotateWithReps.GBreaksSimpleList <- function(gbl, reps) {
  gbl$Oki_Osa <- gbl$Oki_Osa |> annotateWithReps.GBreaks(reps$Oki, reps$Osa)
  gbl$Oki_Bar <- gbl$Oki_Bar |> annotateWithReps.GBreaks(reps$Oki, reps$Bar)
  gbl$Oki_Kum <- gbl$Oki_Kum |> annotateWithReps.GBreaks(reps$Oki, reps$Kum)
  gbl$Oki_Aom <- gbl$Oki_Aom |> annotateWithReps.GBreaks(reps$Oki, reps$Aom)
  gbl$Oki_Nor <- gbl$Oki_Nor |> annotateWithReps.GBreaks(reps$Oki, reps$Nor)
  
  gbl$Osa_Oki <- gbl$Osa_Oki |> annotateWithReps.GBreaks(reps$Osa, reps$Oki)
  gbl$Osa_Bar <- gbl$Osa_Bar |> annotateWithReps.GBreaks(reps$Osa, reps$Bar)
  gbl$Osa_Kum <- gbl$Osa_Kum |> annotateWithReps.GBreaks(reps$Osa, reps$Kum)
  gbl$Osa_Aom <- gbl$Osa_Aom |> annotateWithReps.GBreaks(reps$Osa, reps$Aom)
  gbl$Osa_Nor <- gbl$Osa_Nor |> annotateWithReps.GBreaks(reps$Osa, reps$Nor)
  
  gbl$Bar_Oki <- gbl$Bar_Oki |> annotateWithReps.GBreaks(reps$Bar, reps$Oki)
  gbl$Bar_Osa <- gbl$Bar_Osa |> annotateWithReps.GBreaks(reps$Bar, reps$Osa)
  gbl$Bar_Kum <- gbl$Bar_Kum |> annotateWithReps.GBreaks(reps$Bar, reps$Kum)
  gbl$Bar_Aom <- gbl$Bar_Aom |> annotateWithReps.GBreaks(reps$Bar, reps$Aom)
  gbl$Bar_Nor <- gbl$Bar_Nor |> annotateWithReps.GBreaks(reps$Bar, reps$Nor)
  gbl
}
```

Transcript annotations
----------------------

```{r annotate_transcripts}
(transcripts <- OikScrambling:::loadAllTranscriptsGR())

annotateWithTranscripts <-  function(x, annotT, annotQ) UseMethod("annotateWithTranscripts")

annotateWithTranscripts.GRanges <- function(x, annotT, annotQ = NULL) {
  # Taken from CAGEr (and I am the author)
  ranges2names <- function (rangesA, rangesB) {
    if (is.null(names(rangesB))) 
        stop(sQuote("rangesB"), " must contain have names.")
    names <- findOverlaps(rangesA, rangesB)
    names <- as(names, "List")
    names <- extractList(names(rangesB), names)
    names <- unique(names)
    names <- unstrsplit(names, ";")
    Rle(names)
  }
  x$transcripts <- ranges2names(x, annotT)
  x$transcripts[x$transcripts == ''] <- NA
  x
}

annotateWithTranscripts.GBreaks <- function(x, annotT, annotQ) {
  x$query <- annotateWithTranscripts.GRanges(x$query, annotQ)
  x       <- annotateWithTranscripts.GRanges(x,       annotT)
  x
}

annotateWithTranscripts.GRangesList <- function(x, annotT, annotQ = annotT) {
  x$Oki_Osa <- x$Oki_Osa |> annotateWithTranscripts(annotT$Oki, annotQ$Osa)
  x$Oki_Bar <- x$Oki_Bar |> annotateWithTranscripts(annotT$Oki, annotQ$Bar)
  x$Oki_Kum <- x$Oki_Kum |> annotateWithTranscripts(annotT$Oki, annotQ$Kum)
  x$Oki_Aom <- x$Oki_Aom |> annotateWithTranscripts(annotT$Oki, annotQ$Aom)
  x$Oki_Nor <- x$Oki_Nor |> annotateWithTranscripts(annotT$Oki, annotQ$Nor)
  
  x$Osa_Oki <- x$Osa_Oki |> annotateWithTranscripts(annotT$Osa, annotQ$Oki)
  x$Osa_Bar <- x$Osa_Bar |> annotateWithTranscripts(annotT$Osa, annotQ$Bar)
  x$Osa_Kum <- x$Osa_Kum |> annotateWithTranscripts(annotT$Osa, annotQ$Kum)
  x$Osa_Aom <- x$Osa_Aom |> annotateWithTranscripts(annotT$Osa, annotQ$Aom)
  x$Osa_Nor <- x$Osa_Nor |> annotateWithTranscripts(annotT$Osa, annotQ$Nor)
  
  x$Bar_Oki <- x$Bar_Oki |> annotateWithTranscripts(annotT$Bar, annotQ$Oki)
  x$Bar_Osa <- x$Bar_Osa |> annotateWithTranscripts(annotT$Bar, annotQ$Osa)
  x$Bar_Kum <- x$Bar_Kum |> annotateWithTranscripts(annotT$Bar, annotQ$Kum)
  x$Bar_Aom <- x$Bar_Aom |> annotateWithTranscripts(annotT$Bar, annotQ$Aom)
  x$Bar_Nor <- x$Bar_Nor |> annotateWithTranscripts(annotT$Bar, annotQ$Nor)
  x
}

annotateWithTranscripts.SimpleList <- function(x, annotT, annotQ = annotT)
  annotateWithTranscripts.GRangesList(x, annotT = annotT, annotQ = annotQ)
```

Mapped alignments
-----------------

Let's load data from genome alignments, using the copy distributed in the
`BreakpointsData` package.

```{r GBreaks_object, cache=TRUE}
# Local function to add extra annotation on long/short arms while loading data
load_genomic_breaks_ <- function(file, genomeT, genomeQ, longShort=NULL) {
  file <- system.file(paste0("extdata/Oidioi_pairwise_v3/", file), package = "BreakpointsData")
  gb <- load_genomic_breaks(file, genomeT,  genomeQ)
  if (! is.null(longShort)) {
    gb <-  flagLongShort(gb, longShort)
  }
  gb
}

gbs <- SimpleList()

gbs$Oki_Osa <- load_genomic_breaks_("OKI2018_I69_1.0__OSKA2016v1.9.gff.gz", OKI2018_I69,  OSKA2016v1.9, longShort$Oki) |> annotateWithReps.GBreaks(reps$Oki, reps$Osa)
gbs$Oki_Bar <- load_genomic_breaks_("OKI2018_I69_1.0__Bar2_p4.gff.gz",      OKI2018_I69,  Bar2_p4,      longShort$Oki) |> annotateWithReps.GBreaks(reps$Oki, reps$Bar)
gbs$Oki_Kum <- load_genomic_breaks_("OKI2018_I69_1.0__KUM-M3-7f.gff.gz",    OKI2018_I69,  KUM_M3,       longShort$Oki) |> annotateWithReps.GBreaks(reps$Oki, reps$Kum)
gbs$Oki_Aom <- load_genomic_breaks_("OKI2018_I69_1.0__AOM-5-5f.gff.gz",     OKI2018_I69,  AOM_5,        longShort$Oki) |> annotateWithReps.GBreaks(reps$Oki, reps$Aom)
gbs$Oki_Nor <- load_genomic_breaks_("OKI2018_I69_1.0__OdB3.gff.gz",         OKI2018_I69,  OdB3,         longShort$Oki) |> annotateWithReps.GBreaks(reps$Oki, reps$Nor)

gbs$Osa_Oki <- load_genomic_breaks_("OSKA2016v1.9__OKI2018_I69_1.0.gff.gz", OSKA2016v1.9, OKI2018_I69,  longShort$Osa) |> annotateWithReps.GBreaks(reps$Osa, reps$Oki)
gbs$Osa_Bar <- load_genomic_breaks_("OSKA2016v1.9__Bar2_p4.gff.gz",         OSKA2016v1.9, Bar2_p4,      longShort$Osa) |> annotateWithReps.GBreaks(reps$Osa, reps$Bar)
gbs$Osa_Kum <- load_genomic_breaks_("OSKA2016v1.9__KUM-M3-7f.gff.gz",       OSKA2016v1.9, KUM_M3,       longShort$Osa) |> annotateWithReps.GBreaks(reps$Osa, reps$Kum)
gbs$Osa_Aom <- load_genomic_breaks_("OSKA2016v1.9__AOM-5-5f.gff.gz",        OSKA2016v1.9, AOM_5,        longShort$Osa) |> annotateWithReps.GBreaks(reps$Osa, reps$Aom)
gbs$Osa_Nor <- load_genomic_breaks_("OSKA2016v1.9__OdB3.gff.gz",            OSKA2016v1.9, OdB3,         longShort$Osa) |> annotateWithReps.GBreaks(reps$Osa, reps$Nor)

gbs$Bar_Oki <- load_genomic_breaks_("Bar2_p4__OKI2018_I69_1.0.gff.gz",      Bar2_p4,     OKI2018_I69,   longShort$Bar) |> annotateWithReps.GBreaks(reps$Bar, reps$Oki)
gbs$Bar_Osa <- load_genomic_breaks_("Bar2_p4__OSKA2016v1.9.gff.gz",         Bar2_p4,     OSKA2016v1.9,  longShort$Bar) |> annotateWithReps.GBreaks(reps$Bar, reps$Osa)
gbs$Bar_Kum <- load_genomic_breaks_("Bar2_p4__KUM-M3-7f.gff.gz",            Bar2_p4,     KUM_M3,        longShort$Bar) |> annotateWithReps.GBreaks(reps$Bar, reps$Kum)
gbs$Bar_Aom <- load_genomic_breaks_("Bar2_p4__AOM-5-5f.gff.gz",             Bar2_p4,     AOM_5,         longShort$Bar) |> annotateWithReps.GBreaks(reps$Bar, reps$Aom)
gbs$Bar_Nor <- load_genomic_breaks_("Bar2_p4__OdB3.gff.gz",                 Bar2_p4,     OdB3,          longShort$Bar) |> annotateWithReps.GBreaks(reps$Bar, reps$Nor)

Oik_Oik <- names(gbs)

gbs$Ply_Ros <- load_genomic_breaks(system.file("extdata/Ciona_pairwise_v3/C_int_P__C_int_R.gff.gz",          package = "BreakpointsData"),    NULL,     NULL)
gbs$Ply_Rob <- load_genomic_breaks(system.file("extdata/Ciona_pairwise_v3/C_int_P__C_rob_HT.gff.gz",         package = "BreakpointsData"),    NULL,     NULL)
gbs$Ply_Sav <- load_genomic_breaks(system.file("extdata/Ciona_pairwise_v3/C_int_P__C_sav.gff.gz",            package = "BreakpointsData"),    NULL,     NULL)
gbs$Ply_Oki <- load_genomic_breaks(system.file("extdata/Ciona_pairwise_v3/C_int_P__OKI2018_I69_1.0.gff.gz",  package = "BreakpointsData"),    NULL,     OKI2018_I69)

gbs$Rob_Ros <- load_genomic_breaks(system.file("extdata/Ciona_pairwise_v3/C_rob_HT__C_int_R.gff.gz",         package = "BreakpointsData"),    NULL,     NULL)
gbs$Rob_Ply <- load_genomic_breaks(system.file("extdata/Ciona_pairwise_v3/C_rob_HT__C_int_P.gff.gz",         package = "BreakpointsData"),    NULL,     NULL)
gbs$Rob_Sav <- load_genomic_breaks(system.file("extdata/Ciona_pairwise_v3/C_rob_HT__C_sav.gff.gz",           package = "BreakpointsData"),    NULL,     NULL)
gbs$Rob_Oki <- load_genomic_breaks(system.file("extdata/Ciona_pairwise_v3/C_rob_HT__OKI2018_I69_1.0.gff.gz", package = "BreakpointsData"),    NULL,     OKI2018_I69)

gbs$Dme_Dbu <- load_genomic_breaks(system.file("extdata/Insects_pairwise_v3/Dmel__Dbus_ASM1175060v1.gff.gz",     package = "BreakpointsData"),    NULL,     NULL)
gbs$Dme_Dsu <- load_genomic_breaks(system.file("extdata/Insects_pairwise_v3/Dmel__RU_Dsub_v1.1.gff.gz",          package = "BreakpointsData"),    NULL,     NULL)
gbs$Dme_Dya <- load_genomic_breaks(system.file("extdata/Insects_pairwise_v3/Dmel__Prin_Dyak_Tai18E2_2.0.gff.gz", package = "BreakpointsData"),    NULL,     NULL)
gbs$Dme_Dma <- load_genomic_breaks(system.file("extdata/Insects_pairwise_v3/Dmel__Dmau_ASM438214v1.gff.gz",      package = "BreakpointsData"),    NULL,     NULL)

gbs[Oik_Oik] <- annotateWithTranscripts(gbs[Oik_Oik], transcripts)
gbs <- sapply(gbs, flagAll) |> SimpleList()
```

Unaligned regions
-----------------

Note that the unaligned regions are strandless and unpaired by definition.

```{r unal}
unal  <- BiocParallel::bplapply(gbs, cleanGaps) |> SimpleList()
unal[ 1:5 ] <- BiocParallel::bplapply(unal[ 1:5 ], flagLongShort, longShort$Oki)
unal[ 6:10] <- BiocParallel::bplapply(unal[ 6:10], flagLongShort, longShort$Osa)
unal[11:15] <- BiocParallel::bplapply(unal[11:15], flagLongShort, longShort$Bar)

unal[ 1:5 ] <- BiocParallel::bplapply(unal[ 1:5 ], annotateWithReps.GRanges, reps$Oki)
unal[ 6:10] <- BiocParallel::bplapply(unal[ 6:10], annotateWithReps.GRanges, reps$Osa)
unal[11:15] <- BiocParallel::bplapply(unal[11:15], annotateWithReps.GRanges, reps$Bar)

unal[ 1:5 ] <- BiocParallel::bplapply(unal[ 1:5 ], annotateWithTranscripts, transcripts$Oki)
unal[ 6:10] <- BiocParallel::bplapply(unal[ 6:10], annotateWithTranscripts, transcripts$Osa)
unal[11:15] <- BiocParallel::bplapply(unal[11:15], annotateWithTranscripts, transcripts$Bar)
```

Coalescing alignments
---------------------

```{r coalescing_algorithm, cache=TRUE, dependson="GBreaks_object"}
coa <- BiocParallel::bplapply(gbs, coalesce_contigs) |> SimpleList()
coa[ 1:5]  <- BiocParallel::bplapply(coa[ 1:5],  flagLongShort, longShort$Oki)
coa[ 6:10] <- BiocParallel::bplapply(coa[ 6:10], flagLongShort, longShort$Osa)
coa[11:15] <- BiocParallel::bplapply(coa[11:15], flagLongShort, longShort$Bar)

coa[Oik_Oik] <- annotateWithReps.GBreaksSimpleList(coa[Oik_Oik], reps)
coa[Oik_Oik] <- annotateWithTranscripts(coa[Oik_Oik], transcripts)

coa <- sapply(coa, flagAll) |> SimpleList()
```

Unmapped regions
----------------

We can not call them “uncoalesced” because this would also qualify aligned
regions that were not colinear with any other region.

```{r unmap}
unmap <- BiocParallel::bplapply(coa, cleanGaps) |> SimpleList()

unmap[ 1:5 ] <- BiocParallel::bplapply(unmap[ 1:5 ], flagLongShort, longShort$Oki)
unmap[ 6:10] <- BiocParallel::bplapply(unmap[ 6:10], flagLongShort, longShort$Osa)
unmap[11:15] <- BiocParallel::bplapply(unmap[11:15], flagLongShort, longShort$Bar)

unmap[ 1:5 ] <- BiocParallel::bplapply(unmap[ 1:5 ], annotateWithReps.GRanges, reps$Oki)
unmap[ 6:10] <- BiocParallel::bplapply(unmap[ 6:10], annotateWithReps.GRanges, reps$Osa)
unmap[11:15] <- BiocParallel::bplapply(unmap[11:15], annotateWithReps.GRanges, reps$Bar)

unmap[ 1:5 ] <- BiocParallel::bplapply(unmap[ 1:5 ], annotateWithTranscripts, transcripts$Oki)
unmap[ 6:10] <- BiocParallel::bplapply(unmap[ 6:10], annotateWithTranscripts, transcripts$Osa)
unmap[11:15] <- BiocParallel::bplapply(unmap[11:15], annotateWithTranscripts, transcripts$Bar)
```

Mapped unaligned regions
------------------------

We can pair the unaligned regions that are between two colinearly aligned regions.

```{r mappedUnaligned, cache = TRUE, dependson="GBreaks_object"}
filterMappedUnaligned <- function(gb) {
  colinearRegions <- filterColinearRegions(flagColinearAlignments(gb), rename = FALSE)
  # Turning the sequence of [(TRUE)n, FALSE]n into indices.
  idx <- c(0, head(cumsum(!colinearRegions$colinear), -1))
  gbl <- split(colinearRegions, idx)
  unalMap <- endoapply(gbl, \(gb) {
    # Check strand
    onMinus <- all(strand(gb) == "-")
    # Need to subtract 1 because some ranges are adjacent (no gap)
    GBreaks(target = cleanGaps(gb      - 1),
            query  = cleanGaps(gb$query -1) |> sort(decreasing = onMinus))
  }) |> unlist()
  names(unalMap) <- NULL
  unalMap
}

unalMap <- BiocParallel::bplapply(gbs, filterMappedUnaligned) |> as("SimpleList")

unalMap[ 1:5 ] <- BiocParallel::bplapply(unalMap[ 1:5 ], flagLongShort, longShort$Oki)
unalMap[ 6:10] <- BiocParallel::bplapply(unalMap[ 6:10], flagLongShort, longShort$Osa)
unalMap[11:15] <- BiocParallel::bplapply(unalMap[11:15], flagLongShort, longShort$Bar)

unalMap[Oik_Oik] <- annotateWithReps.GBreaksSimpleList(unalMap[Oik_Oik], reps)
unalMap[Oik_Oik] <- annotateWithTranscripts(unalMap[Oik_Oik], transcripts)
unalMap[Oik_Oik] <- sapply(unalMap[Oik_Oik], flagAll) |> SimpleList()
```

Translocated regions
--------------------

Here we compute new `GBreaks` objects representing the intervals between two
regions that would be colinear if there would not be a translocation separating
them.

```{r translocated_regions, cache=TRUE, dependson="coalescing_algorithm"}
removeTranslocations <- function(gb) {
  if(any(gb$tra))
    gb <- gb[-(which(gb$tra) + 1)]
  gb
}

traGaps <- function(coa) {
  # The input MUST already be coalesced.
  # First, flag translocations on the target genome
  tra.target <- flagTranslocations(coa)$tra
  # Second, do it again on the query genome
  tra <- swap(coa) |> sort(ignore.strand = TRUE) |> flagTranslocations()
  tra.query <- tra$tra
  tra <- swap(tra) |> sort(ignore.strand = TRUE)
  tra$tra <- tra.target | tra.query
  # Then, remove translocated regions
  tra <- removeTranslocations(tra)
  # Then extract the newly "unaligned mapped" regions
  # They contain the gaps left by the removal of the translocated regions.
  filterMappedUnaligned(tra)
}

tra <- BiocParallel::bplapply(coa, traGaps) |> as("SimpleList")

tra[ 1:5 ] <- BiocParallel::bplapply(tra[ 1:5 ], flagLongShort, longShort$Oki)
tra[ 6:10] <- BiocParallel::bplapply(tra[ 6:10], flagLongShort, longShort$Osa)
tra[11:15] <- BiocParallel::bplapply(tra[11:15], flagLongShort, longShort$Bar)

tra[Oik_Oik] <- annotateWithReps.GBreaksSimpleList(tra[Oik_Oik], reps)
tra[Oik_Oik] <- annotateWithTranscripts(tra[Oik_Oik], transcripts)
tra <- sapply(tra, flagAll) |> SimpleList()
```

Coalesce again
--------------

We remove the translocated regions that interrupt colinearity, where one of
the colinear pairs is closer than 200 bp.

```{r re_coalesce, cache=TRUE, dependson="coalescing_algorithm"}
removeTranslocationsOnTargetGenome <- function(gb) {
  gb <- flagTranslocations(gb)
  gb <- dist2next(gb, step = 2, ignore.strand = TRUE)
  idx <- 1 + which( gb$tra & (gb$tdist < 200 | gb$qdist < 200) )
  if(length(idx) != 0)
    gb <- gb[-idx]
  coalesce_contigs(gb)
}
removeTranslocationsOnBothGenomes <- function(gb)
  gb |>
    removeTranslocationsOnTargetGenome() |>
    swap() |> sort(i=T) |>
    removeTranslocationsOnTargetGenome() |>
    swap() |> sort(i=T)

coa2 <- BiocParallel::bplapply(coa, removeTranslocationsOnBothGenomes) |> SimpleList()
coa2[ 1:5]  <- BiocParallel::bplapply(coa2[ 1:5],  flagLongShort, longShort$Oki)
coa2[ 6:10] <- BiocParallel::bplapply(coa2[ 6:10], flagLongShort, longShort$Osa)
coa2[11:15] <- BiocParallel::bplapply(coa2[11:15], flagLongShort, longShort$Bar)

coa2[Oik_Oik] <- annotateWithReps.GBreaksSimpleList(coa2[Oik_Oik], reps)
coa2[Oik_Oik] <- annotateWithTranscripts(coa2[Oik_Oik], transcripts)
coa2 <- sapply(coa2, flagAll) |> SimpleList()
```

Unmapped regions 2
------------------

The intervals between double-coalesced regions.

```{r unmap2}
unmap2 <- BiocParallel::bplapply(coa2, cleanGaps) |> SimpleList()

unmap2[ 1:5 ] <- BiocParallel::bplapply(unmap2[ 1:5 ], flagLongShort, longShort$Oki)
unmap2[ 6:10] <- BiocParallel::bplapply(unmap2[ 6:10], flagLongShort, longShort$Osa)
unmap2[11:15] <- BiocParallel::bplapply(unmap2[11:15], flagLongShort, longShort$Bar)

unmap2[ 1:5 ] <- BiocParallel::bplapply(unmap2[ 1:5 ], annotateWithReps.GRanges, reps$Oki)
unmap2[ 6:10] <- BiocParallel::bplapply(unmap2[ 6:10], annotateWithReps.GRanges, reps$Osa)
unmap2[11:15] <- BiocParallel::bplapply(unmap2[11:15], annotateWithReps.GRanges, reps$Bar)

unmap2[ 1:5 ] <- BiocParallel::bplapply(unmap2[ 1:5 ], annotateWithTranscripts, transcripts$Oki)
unmap2[ 6:10] <- BiocParallel::bplapply(unmap2[ 6:10], annotateWithTranscripts, transcripts$Osa)
unmap2[11:15] <- BiocParallel::bplapply(unmap2[11:15], annotateWithTranscripts, transcripts$Bar)
```

Translocated regions 2
----------------------

To check that we really removed the small ones as intented.

```{r translocated_regions2, cache=TRUE, dependson="re_coalesce"}
tra2 <- BiocParallel::bplapply(coa2, traGaps) |> as("SimpleList")

tra2[ 1:5 ] <- BiocParallel::bplapply(tra2[ 1:5 ], flagLongShort, longShort$Oki)
tra2[ 6:10] <- BiocParallel::bplapply(tra2[ 6:10], flagLongShort, longShort$Osa)
tra2[11:15] <- BiocParallel::bplapply(tra2[11:15], flagLongShort, longShort$Bar)

tra2[Oik_Oik] <- annotateWithReps.GBreaksSimpleList(tra2[Oik_Oik], reps)
tra2[Oik_Oik] <- annotateWithTranscripts(tra2[Oik_Oik], transcripts)

tra2 <- sapply(tra2, flagAll) |> SimpleList()
```


Numbers of blocks
-----------------

```{r lengths}
bp_summary <- 
  rbind(
    gbs     = sapply(gbs, length),
    unal    = sapply(unal, length),
    coa     = sapply(coa, length),
    unmap   = sapply(unmap, length),
    unmap2  = sapply(unmap2, length),
    unalMap = sapply(unalMap, length),
    tra     = sapply(tra, length),
    tra2    = sapply(tra2, length),
    coa2    = sapply(coa2, length)
  ) |> t() |> data.frame()

sapply(bp_summary, \(x) tapply(x, row.names(bp_summary) |> OikScrambling:::compDistance(), mean))
sapply(bp_summary, \(x) tapply(x, row.names(bp_summary) |> OikScrambling:::compDistance(), median))
sapply(bp_summary, \(x) tapply(x, row.names(bp_summary) |> OikScrambling:::compDistance(), sd))
```

Fraction of the genome aligned or mapped
----------------------------------------

```{r fraction_aligned}
  aligned_tot  <- sapply(gbs,   \(x) sum(width(x)))
  mapped_tot   <- sapply(coa,   \(x) sum(width(x)))
  mapped2_tot  <- sapply(coa2,  \(x) sum(width(x)))
unaligned_tot  <- sapply(unal,   \(x) sum(width(x)))
unmapped_tot   <- sapply(unmap,  \(x) sum(width(x)))
unmapped2_tot  <- sapply(unmap2, \(x) sum(width(x)))

(align_summary <- 100 * cbind(
  align_frac = aligned_tot / ( aligned_tot + unaligned_tot ),
  map_frac   = mapped_tot  / ( mapped_tot  + unmapped_tot ),
  map2_frac  = mapped2_tot / ( mapped2_tot + unmapped2_tot)
 )|> as.data.frame())

sapply(align_summary, \(x) tapply(x, row.names(align_summary) |> OikScrambling:::compDistance(), mean))   |> round(1)
sapply(align_summary, \(x) tapply(x, row.names(align_summary) |> OikScrambling:::compDistance(), median)) |> round(1)
sapply(align_summary, \(x) tapply(x, row.names(align_summary) |> OikScrambling:::compDistance(), sd))     |> round(1)
```

Summary
-------

```{r summary}
df <- cbind(
  gbs = sapply(gbs, length),
  coa = sapply(coa, length),
  coa2 = sapply(coa2, length)
) |> as.data.frame() |>
  dplyr::arrange(gbs = gbs - coa - coa2, coa = coa - coa2)

df$pair <- row.names(df)

ggplot(df |> tidyr::pivot_longer(!pair)) +
  aes(pair, value, fill = name) +
  geom_bar( stat = "identity" ) +
  ggplot2::coord_flip() +
  ggtitle("Cumulative plot (gbs ⊃ coa ⊃ coa2) of the computed genomic regions")
```

NOTE: how does that compare with ANI (Average Nucleotide Identity) ?

```{r}
sim <- c(`Bar2_p4__AOM-5-5f` = 87.489, Bar2_p4__Bar2_p4 = 100, `Bar2_p4__KUM-M3-7f` = 78.45822, 
Bar2_p4__O_alb = 69.79983, Bar2_p4__O_lon = 66.11535, Bar2_p4__O_van = 69.29796, 
Bar2_p4__OdB3 = 97.7499, Bar2_p4__OKI2018_I69_1.0 = 78.68358, 
Bar2_p4__OSKA2016v1.9 = 87.46913, C_int_P__C_int_P = 100, C_int_P__C_int_R = 96.16641, 
C_int_P__C_rob_HT = 90.49159, C_int_P__C_sav = 73.27976, C_int_P__OKI2018_I69_1.0 = 65.14351, 
C_int_P__OSKA2016v1.9 = 64.71592, C_int_P__St_cla = 68.16588, 
C_rob_HT__C_int_P = 90.41795, C_rob_HT__C_int_R = 90.4192, C_rob_HT__C_rob_HT = 100, 
C_rob_HT__C_sav = 73.16812, C_rob_HT__OKI2018_I69_1.0 = 65.53052, 
C_rob_HT__OSKA2016v1.9 = 65.09903, C_rob_HT__St_cla = 68.30626, 
Dalb__AaegL5.0 = 65.6799, Dalb__AalbS3_pri_1.0 = 68.17702, Dalb__Cfrigida_v1.1 = 72.42817, 
`Dalb__Dalb_15112-1751.03v1` = 100, Dalb__Dath_EB_1.0 = 80.60732, 
Dalb__Dbif_1.0 = 81.08233, Dalb__Dbus_ASM1175060v1 = 85.11277, 
Dalb__Dinn_ASM435438v2 = 87.86721, Dalb__Dlow_1.0 = 81.1923, 
Dalb__Dmau_ASM438214v1 = 81.65325, Dalb__Dmel_BDGP6 = 81.14, 
Dalb__Dmir_PacBio2.1 = 81.76273, Dalb__Dpse_MV25 = 81.20186, 
Dalb__Eten_idEriTena2.1 = 70.77767, Dalb__Paka_ASM1839793v1 = 66.22446, 
Dalb__Prin_Dyak_Tai18E2_2.0 = 81.21401, Dalb__RU_Dsub_v1.1 = 82.53274, 
Dalb__RU_Dtri_1.1 = 81.93057, Dath__AaegL5.0 = 66.28782, Dath__AalbS3_pri_1.0 = 69.08501, 
Dath__Cfrigida_v1.1 = 72.40521, `Dath__Dalb_15112-1751.03v1` = 81.90288, 
Dath__Dath_EB_1.0 = 100, Dath__Dbif_1.0 = 90.09103, Dath__Dbus_ASM1175060v1 = 81.89253, 
Dath__Dinn_ASM435438v2 = 82.2395, Dath__Dlow_1.0 = 92.30271, 
Dath__Dmau_ASM438214v1 = 85.84368, Dath__Dmel_BDGP6 = 85.72521, 
Dath__Dmir_PacBio2.1 = 91.84388, Dath__Dpse_MV25 = 91.87636, 
Dath__Eten_idEriTena2.1 = 71.32619, Dath__Paka_ASM1839793v1 = 65.05051, 
Dath__Prin_Dyak_Tai18E2_2.0 = 85.03599, Dath__RU_Dsub_v1.1 = 86.28124, 
Dath__RU_Dtri_1.1 = 86.45957, Dinn__AaegL5.0 = 66.17064, Dinn__AalbS3_pri_1.0 = 67.58017, 
DDinn__Cfrigida_v1.1 = 71.83029, `Dinn__Dalb_15112-1751.03v1` = 87.04083, 
Dinn__Dath_EB_1.0 = 79.57953, Dinn__Dbif_1.0 = 80.19957, Dinn__Dbus_ASM1175060v1 = 84.28436, 
Dinn__Dinn_ASM435438v2 = 100, Dinn__Dlow_1.0 = 80.67069, Dinn__Dmau_ASM438214v1 = 81.0131, 
Dinn__Dmel_BDGP6 = 80.77358, Dinn__Dmir_PacBio2.1 = 80.56467, 
Dinn__Dpse_MV25 = 79.94521, Dinn__Eten_idEriTena2.1 = 69.06343, 
Dinn__Paka_ASM1839793v1 = 66.33035, Dinn__Prin_Dyak_Tai18E2_2.0 = 80.57511, 
Dinn__RU_Dsub_v1.1 = 81.50381, Dinn__RU_Dtri_1.1 = 81.46686, 
Dmel__AaegL5.0 = 66.80427, Dmel__AalbS3_pri_1.0 = 68.20054, Dmel__Cfrigida_v1.1 = 72.50621, 
`Dmel__Dalb_15112-1751.03v1` = 80.26274, Dmel__Dath_EB_1.0 = 83.20431, 
Dmel__Dbif_1.0 = 82.25649, Dmel__Dbus_ASM1175060v1 = 80.08149, 
Dmel__Dinn_ASM435438v2 = 80.79603, Dmel__Dlow_1.0 = 83.56816, 
Dmel__Dmau_ASM438214v1 = 95.36722, Dmel__Dmel_BDGP6 = 100, Dmel__Dmir_PacBio2.1 = 84.10367, 
Dmel__Dpse_MV25 = 83.27086, Dmel__Eten_idEriTena2.1 = 70.18941, 
Dmel__Paka_ASM1839793v1 = 65.26859, Dmel__Prin_Dyak_Tai18E2_2.0 = 91.33685, 
Dmel__RU_Dsub_v1.1 = 88.09201, Dmel__RU_Dtri_1.1 = 86.06675, 
Dpse__AaegL5.0 = 67.05445, Dpse__AalbS3_pri_1.0 = 68.72562, Dpse__Cfrigida_v1.1 = 72.71335, 
`Dpse__Dalb_15112-1751.03v1` = 81.69273, Dpse__Dath_EB_1.0 = 91.50727, 
Dpse__Dbif_1.0 = 90.07924, Dpse__Dbus_ASM1175060v1 = 81.28797, 
Dpse__Dinn_ASM435438v2 = 81.96187, Dpse__Dlow_1.0 = 95.02922, 
Dpse__Dmau_ASM438214v1 = 85.43473, Dpse__Dmel_BDGP6 = 85.32083, 
Dpse__Dmir_PacBio2.1 = 97.38482, Dpse__Dpse_MV25 = 100, Dpse__Eten_idEriTena2.1 = 69.74893, 
Dpse__Paka_ASM1839793v1 = 64.9258, Dpse__Prin_Dyak_Tai18E2_2.0 = 84.58586, 
Dpse__RU_Dsub_v1.1 = 86.0418, Dpse__RU_Dtri_1.1 = 85.93798, Dsub__AaegL5.0 = 66.08828, 
Dsub__AalbS3_pri_1.0 = 68.42965, Dsub__Cfrigida_v1.1 = 71.28905, 
`Dsub__Dalb_15112-1751.03v1` = 80.48788, Dsub__Dath_EB_1.0 = 82.75689, 
Dsub__Dbif_1.0 = 83.14488, Dsub__Dbus_ASM1175060v1 = 80.4126, 
Dsub__Dinn_ASM435438v2 = 81.50499, Dsub__Dlow_1.0 = 84.02854, 
Dsub__Dmau_ASM438214v1 = 88.26478, Dsub__Dmel_BDGP6 = 87.81593, 
Dsub__Dmir_PacBio2.1 = 84.50135, Dsub__Dpse_MV25 = 82.88658, 
Dsub__Eten_idEriTena2.1 = 69.00946, Dsub__Paka_ASM1839793v1 = 64.88739, 
Dsub__Prin_Dyak_Tai18E2_2.0 = 87.79101, Dsub__RU_Dsub_v1.1 = 100, 
Dsub__RU_Dtri_1.1 = 87.23365, Dyak__AaegL5.0 = 65.96956, Dyak__AalbS3_pri_1.0 = 68.30608, 
Dyak__Cfrigida_v1.1 = 72.36378, `Dyak__Dalb_15112-1751.03v1` = 80.31928, 
Dyak__Dath_EB_1.0 = 82.77286, Dyak__Dbif_1.0 = 82.36289, Dyak__Dbus_ASM1175060v1 = 79.76272, 
Dyak__Dinn_ASM435438v2 = 80.83953, Dyak__Dlow_1.0 = 83.68068, 
Dyak__Dmau_ASM438214v1 = 91.8662, Dyak__Dmel_BDGP6 = 91.52635, 
Dyak__Dmir_PacBio2.1 = 84.21573, Dyak__Dpse_MV25 = 83.25226, 
Dyak__Eten_idEriTena2.1 = 70.75961, Dyak__Paka_ASM1839793v1 = 65.18495, 
Dyak__Prin_Dyak_Tai18E2_2.0 = 100, Dyak__RU_Dsub_v1.1 = 88.18683, 
Dyak__RU_Dtri_1.1 = 86.18934, `OKI2018_I69_1.0__AOM-5-5f` = 79.19197, 
OKI2018_I69_1.0__Bar2_p4 = 78.97216, `OKI2018_I69_1.0__KUM-M3-7f` = 96.6692, 
OKI2018_I69_1.0__O_alb = 69.58952, OKI2018_I69_1.0__O_lon = 65.9365, 
OKI2018_I69_1.0__O_van = 69.21153, OKI2018_I69_1.0__OdB3 = 78.80925, 
OKI2018_I69_1.0__OKI2018_I69_1.0 = 100, OKI2018_I69_1.0__OSKA2016v1.9 = 78.82942, 
`OSKA2016v1.9__AOM-5-5f` = 97.14968, OSKA2016v1.9__Bar2_p4 = 87.56705, 
`OSKA2016v1.9__KUM-M3-7f` = 78.52209, OSKA2016v1.9__O_alb = 68.88416, 
OSKA2016v1.9__O_lon = 66.94376, OSKA2016v1.9__O_van = 68.66089, 
OSKA2016v1.9__OdB3 = 87.51462, OSKA2016v1.9__OKI2018_I69_1.0 = 78.52812, 
OSKA2016v1.9__OSKA2016v1.9 = 100)

align_summary$sim <- sim[c(
  "OKI2018_I69_1.0__OSKA2016v1.9",
  "OKI2018_I69_1.0__Bar2_p4",
  "OKI2018_I69_1.0__KUM-M3-7f",
  "OKI2018_I69_1.0__AOM-5-5f",
  "OKI2018_I69_1.0__OdB3",
  "OSKA2016v1.9__OKI2018_I69_1.0",
  "OSKA2016v1.9__Bar2_p4",
  "OSKA2016v1.9__KUM-M3-7f",
  "OSKA2016v1.9__AOM-5-5f",
  "OSKA2016v1.9__OdB3",
  "Bar2_p4__OKI2018_I69_1.0",
  "Bar2_p4__OSKA2016v1.9",
  "Bar2_p4__KUM-M3-7f",
  "Bar2_p4__AOM-5-5f",
  "Bar2_p4__OdB3",
  "C_int_P__C_int_R",
  "C_int_P__C_rob_HT",
  "C_int_P__C_sav",
  "C_int_P__OKI2018_I69_1.0",
  "C_rob_HT__C_int_R",
  "C_rob_HT__C_int_P",
  "C_rob_HT__C_sav",
  "C_rob_HT__OKI2018_I69_1.0",
  "Dmel__Dbus_ASM1175060v1",
  "Dmel__RU_Dsub_v1.1",
  "Dmel__Prin_Dyak_Tai18E2_2.0",
  "Dmel__Dmau_ASM438214v1"
)]

align_summary["Hum_Mou",] <- c(10.3264, 54.39799, NA, 75.20007)

align_summary$pairname <- rownames(align_summary)
align_summary$dist <- OikScrambling:::compDistance(align_summary$pairname)
align_summary$genus <- "Oikopleura"
align_summary$genus[grepl("Int", align_summary$dist)] <- "Ciona"
align_summary$genus[grepl("Dme", align_summary$dist)] <- "Drosophila"
align_summary$genus[grepl("Hum", align_summary$dist)] <- "Mammalian"

df <- tidyr::pivot_longer(align_summary, c("align_frac", "map_frac", "map2_frac", "sim"))
df <- df[!grepl("– Oki", df$dist),]

library(ggplot2)

ggplot(df) +
  aes(value, name, col= genus) +
  geom_point() +
  facet_wrap(~name, ncol = 1, scale = "free") +
  theme_bw()

align_summary |>
  dplyr::filter(!grepl("– Oki", dist)) |>
  ggplot() +
  aes(align_frac, sim, col= genus) +
  geom_point(size=10) +
  theme_bw()

align_summary |>
  dplyr::filter(!grepl("– Oki", dist)) |>
  ggplot() +
  aes(map_frac, sim, col= genus) +
  geom_point() +
  theme_bw()
```

```{r breakpoints_per_Mb}
genome_sizes=list(Aom=sum(seqlengths(genomes$Aom)), Bar=sum(seqlengths(genomes$Bar)),
                  Oki=sum(seqlengths(genomes$Oki)), Osa=sum(seqlengths(genomes$Osa)),
                  Kum=sum(seqlengths(genomes$Kum)), Nor=sum(seqlengths(genomes$Nor)),
                  Ply=128639392, Ros=134327879, Sav=177003750, Rob=117529544,
                  Dme=137547960, Dbu=110837441, Dsu=130045071, Dya=136883712
                  ) |> unlist()

genome_sizes[names(coa) |> sub(pat = "_.*", rep = "")]
dens <- tibble::tibble(
  nbreak   = sapply(coa, length),
  nbreak2  = sapply(coa2, length),
  gsize    = genome_sizes[names(coa) |> sub(pat = "_.*", rep = "")],
  density  = nbreak  / gsize * 1e6,
  density2 = nbreak2 / gsize * 1e6,
  pairname = names(coa),
  dist     = OikScrambling:::compDistance(pairname),
  genus    = dist |>
              sub(pat = "Dme_.*", rep="Drosophila") |>
              sub(pat = "Int.*",  rep="Ciona") |>
              sub(pat = ".*[_–].*",  rep="Oikopleura")
    
)

set.seed(1664)
ggplot(dens) +
  aes(density, genus, col = dist, pch = genus) +
  geom_point(size = 5) +
  scale_x_continuous("Density (breakpoints / megabase)") +
  ggtitle("Density of breakpoints in paired comparisons, arranged per genus") +
  scale_colour_manual(values=rainbow(11, s=.6, v=.9)[sample(1:11,11)])

set.seed(1664)
ggplot(dens) +
  aes(density2, genus, col = dist, pch = genus) +
  geom_point(size = 5) +
  scale_x_continuous("Density (breakpoints / megabase)") +
  ggtitle("Density of breakpoints (double-coalesced) in paired comparisons, arranged per genus") +
  scale_colour_manual(values=rainbow(11, s=.6, v=.9)[sample(1:11,11)])
```

Save data
=========

```{r save_data}
save(gbs, unal, coa, unmap, unalMap, tra, tra2, coa2, unmap2, longShort, file = "BreakPoints.Rdata")
system("echo '9f273f099ecbfa44899b6c343d1382f1  BreakPoints.Rdata' | md5sum -c", intern = TRUE)
```

Session information
===================

```{r session_info}
sessionInfo()
```
