---
title: "Load genomic break data"
author: 
 - "Charles Plessy"
date: "`r format(Sys.time(), '%d %B, %Y')`"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteIndexEntry{Load genomic break data}
  %\VignetteEncoding{UTF-8}
editor_options: 
  chunk_output_type: console
---

```{r setup}
knitr::opts_knit$set(verbose = TRUE)
```

Introduction
============

This vignette loads the genomic breaks data, processes it and saves the
resulting objects in a serialised R object (`RData` file) for reuse in other
vignettes.

Similar objects are stored in lists to avoid cluttering the environment.

Pacakges and data
=================


```{r load_pacakges_and_data}
suppressPackageStartupMessages({
  library('GenomicBreaks')
  library('GenomicFeatures')
  library('ggplot2')
  library("BreakpointsData")
})
(genomes <- OikScrambling:::loadAllGenomes())
```

Genomic breaks
==============

Long/short arm information
--------------------------

Short / long arm annotation will be done by the `GenomicBreaks::flagLongShort()`
function.

```{r long_and_short_arms}

loadLongShort <- function(file, genome) {
  file <- system.file(paste0("extdata/Annotations/", file), package = "BreakpointsData")
  gr <- rtracklayer::import.gff3(file)
  gr$source <- gr$type <- gr$score <- gr$phase <- NULL
  GRanges(gr, seqinfo = seqinfo(genome), strand = "*")
}

longShort <- SimpleList()
longShort$Oki <- loadLongShort("OKI2018_I69.v2/OKI2018_I69.arms.gff3", OKI2018_I69)
longShort$Osa <- loadLongShort("OSKA2016v1.9/OSKA2016v1.9.arms.gff3", OSKA2016v1.9)
longShort$Bar <- loadLongShort("Bar2_p4.Flye/Bar2_p4.Flye.arms.gff3", Bar2_p4)

longShort
```

Mapped alignments
-----------------

Let's load data from genome alignments, using the copy distributed in the
`BreakpointsData` package.

```{r GBreaks_object, cache=TRUE}
# Local function to add extra annotation on long/short arms while loading data
load_genomic_breaks_ <- function(file, genomeT, genomeQ, longShort=NULL) {
  file <- system.file(paste0("extdata/Oidioi_pairwise_v3/", file), package = "BreakpointsData")
  gb <- load_genomic_breaks(file, genomeT,  genomeQ)
  if (! is.null(longShort)) {
    gb <-  flagLongShort(gb, longShort)
  }
  gb
}

gbs <- SimpleList()

gbs$Oki_Osa <- load_genomic_breaks_("OKI2018_I69_1.0__OSKA2016v1.9.gff.gz", OKI2018_I69,  OSKA2016v1.9, longShort$Oki)
gbs$Oki_Bar <- load_genomic_breaks_("OKI2018_I69_1.0__Bar2_p4.gff.gz",      OKI2018_I69,  Bar2_p4,      longShort$Oki)
gbs$Oki_Kum <- load_genomic_breaks_("OKI2018_I69_1.0__KUM-M3-7f.gff.gz",    OKI2018_I69,  KUM_M3,       longShort$Oki)
gbs$Oki_Aom <- load_genomic_breaks_("OKI2018_I69_1.0__AOM-5-5f.gff.gz",     OKI2018_I69,  AOM_5,        longShort$Oki)
gbs$Oki_Nor <- load_genomic_breaks_("OKI2018_I69_1.0__OdB3.gff.gz",         OKI2018_I69,  OdB3,         longShort$Oki)

gbs$Osa_Oki <- load_genomic_breaks_("OSKA2016v1.9__OKI2018_I69_1.0.gff.gz", OSKA2016v1.9, OKI2018_I69,  longShort$Osa)
gbs$Osa_Bar <- load_genomic_breaks_("OSKA2016v1.9__Bar2_p4.gff.gz",         OSKA2016v1.9, Bar2_p4,      longShort$Osa)
gbs$Osa_Kum <- load_genomic_breaks_("OSKA2016v1.9__KUM-M3-7f.gff.gz",       OSKA2016v1.9, KUM_M3,       longShort$Osa)
gbs$Osa_Aom <- load_genomic_breaks_("OSKA2016v1.9__AOM-5-5f.gff.gz",        OSKA2016v1.9, AOM_5,        longShort$Osa)
gbs$Osa_Nor <- load_genomic_breaks_("OSKA2016v1.9__OdB3.gff.gz",            OSKA2016v1.9, OdB3,         longShort$Osa)

gbs$Bar_Oki <- load_genomic_breaks_("Bar2_p4__OKI2018_I69_1.0.gff.gz",      Bar2_p4,     OKI2018_I69,   longShort$Bar)
gbs$Bar_Osa <- load_genomic_breaks_("Bar2_p4__OSKA2016v1.9.gff.gz",         Bar2_p4,     OSKA2016v1.9,  longShort$Bar)
gbs$Bar_Kum <- load_genomic_breaks_("Bar2_p4__KUM-M3-7f.gff.gz",            Bar2_p4,     KUM_M3,        longShort$Bar)
gbs$Bar_Aom <- load_genomic_breaks_("Bar2_p4__AOM-5-5f.gff.gz",             Bar2_p4,     AOM_5,         longShort$Bar)
gbs$Bar_Nor <- load_genomic_breaks_("Bar2_p4__OdB3.gff.gz",                 Bar2_p4,     OdB3,          longShort$Bar)
```

Unaligned regions
-----------------

Note that the unaligned regions are strandless by definition.

```{r unal}
unal  <- sapply(gbs, cleanGaps) |> SimpleList()
unal[ 1:5 ] <- sapply(unal[ 1:5 ], flagLongShort, longShort$Oki)
unal[ 6:10] <- sapply(unal[ 6:10], flagLongShort, longShort$Osa)
unal[11:15] <- sapply(unal[11:15], flagLongShort, longShort$Bar)
```

Coalescing alignments
---------------------

```{r coalescing_algorithm, cache=TRUE, dependson="GBreaks_object"}
coa <- sapply(gbs, coalesce_contigs) |> SimpleList()
coa[ 1:5]  <- sapply(coa[ 1:5],  flagLongShort, longShort$Oki)
coa[ 6:10] <- sapply(coa[ 6:10], flagLongShort, longShort$Osa)
coa[11:15] <- sapply(coa[11:15], flagLongShort, longShort$Bar)
```

Unmapped regions
----------------

We can not call them “uncoalesced” because this would also qualify aligned
regions that were not colinear with any other region.

```{r unmap}
unmap <- sapply(coa, cleanGaps) |> SimpleList()
unmap[ 1:5 ] <- sapply(unmap[ 1:5 ], flagLongShort, longShort$Oki)
unmap[ 6:10] <- sapply(unmap[ 6:10], flagLongShort, longShort$Osa)
unmap[11:15] <- sapply(unmap[11:15], flagLongShort, longShort$Bar)
```

Coalesce again
--------------

See the Genomic Breaks vignette for justification.

```{r re_coalesce, cache=TRUE, dependson="coalescing_algorithm"}
coa2 <- sapply(coa, coalesce_contigs, minwidth = 500) |> SimpleList()
coa2[ 1:5]  <- sapply(coa2[ 1:5],  flagLongShort, longShort$Oki)
coa2[ 6:10] <- sapply(coa2[ 6:10], flagLongShort, longShort$Osa)
coa2[11:15] <- sapply(coa2[11:15], flagLongShort, longShort$Bar)
```

Coalesce within chromosomes
---------------------------

The idea is that if colinearity is only lost because of a translocation to
another chromosome, then it is fine.

```{r coalesce_by_chr, cache=TRUE, dependson="GBreaks_object"}
gbs$Oki_Kum -> x
gbl <- split(x, seqnames(x), drop=T)
gbl <- endoapply(gbl, coalesce_contigs)
unlist(gbl)
```

Numbers of blocks
-----------------

```{r lengths}
sapply(gbs,  length)
sapply(coa,  length)
sapply(coa2, length)
```

Summary
-------

```{r summary}
df <- rbind(
  gbs = sapply(gbs, length),
  coa = sapply(coa, length),
  coa2 = sapply(coa2, length)
) |> as.data.frame()

df$what <- row.names(df)
df <- reshape2::melt(df)

ggplot(df) +
  aes(variable, value, fill = what) +
  geom_bar( stat = "identity" ) +
  ggplot2::coord_flip()
```

Save data
=========

```{r save_data}
save(gbs, unal, coa, unmap, coa2, longShort, file = "BreakPoints.Rdata")
system("echo 'd7345eb72bd7166f0f7e716c5817b14c  BreakPoints.Rdata' | md5sum -c", intern = TRUE)
```

Session information
===================

```{r session_info}
sessionInfo()
```
