---
title: "Load genomic break data"
author: 
 - "Charles Plessy"
date: "`r format(Sys.time(), '%d %B, %Y')`"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteIndexEntry{Load genomic break data}
  %\VignetteEncoding{UTF-8}
editor_options: 
  chunk_output_type: console
---

```{r setup}
knitr::opts_knit$set(verbose = TRUE)
```

Introduction
============

This vignette loads the genomic breaks data, consolidates colinear regions,
extracts unaligned and unmapped regions, transfers annotation data and saves the
resulting objects in a serialised R object (`RData` file) for reuse in other
vignettes.

Related objects are stored in lists to avoid cluttering the environment.

Packages and data
=================


```{r load_pacakges_and_data}
suppressPackageStartupMessages({
  library('GenomicBreaks')
  requireNamespace('GenomicFeatures')
  library('ggplot2')
  library("BreakpointsData")
})
(genomes <- OikScrambling:::loadAllGenomes())
```

Genomic breaks
==============

Long/short arm information
--------------------------

Short / long arm annotation will be done by the `GenomicBreaks::flagLongShort()`
function.

```{r long_and_short_arms}

loadLongShort <- function(file, genome) {
  file <- system.file(paste0("extdata/Annotations/", file), package = "BreakpointsData")
  gr <- rtracklayer::import.gff3(file)
  gr$source <- gr$type <- gr$score <- gr$phase <- NULL
  GRanges(gr, seqinfo = seqinfo(genome), strand = "*")
}

longShort <- SimpleList()
longShort$Oki <- loadLongShort("OKI2018_I69.v2/OKI2018_I69.arms.gff3", OKI2018_I69)
longShort$Osa <- loadLongShort("OSKA2016v1.9/OSKA2016v1.9.arms.gff3", OSKA2016v1.9)
longShort$Bar <- loadLongShort("Bar2_p4.Flye/Bar2_p4.Flye.arms.gff3", Bar2_p4)

longShort
```

Repeat regions
--------------

Load from `BreakpointsData` package via `OikScrambling:::loadAllRepeats()`

```{r load_repeats}
(reps <- OikScrambling:::loadAllRepeats())

annotateWithReps.GRanges <- function(gr, rep) {
  # Inspired from CAGEr:::ranges2names
  o <- findOverlaps(gr, rep)
  o <- as(o, "List")
  o <- extractList(rep$Class |> as.character(), o)
  o <- endoapply(o, unique)
  o[sapply(o, length) == 0] <- NA
  gr$rep <- o
  gr
}

annotateWithReps.GBreaks <- function(gb, repT, repQ) {
  gb$query <- annotateWithReps.GRanges(gb$query, repQ)
  gb       <- annotateWithReps.GRanges(gb,       repT)
}

annotateWithReps.GBreaksSimpleList <- function(gbl, reps) {
  gbl$Oki_Osa <- gbl$Oki_Osa |> annotateWithReps.GBreaks(reps$Oki, reps$Osa)
  gbl$Oki_Bar <- gbl$Oki_Bar |> annotateWithReps.GBreaks(reps$Oki, reps$Bar)
  gbl$Oki_Kum <- gbl$Oki_Kum |> annotateWithReps.GBreaks(reps$Oki, reps$Kum)
  gbl$Oki_Aom <- gbl$Oki_Aom |> annotateWithReps.GBreaks(reps$Oki, reps$Aom)
  gbl$Oki_Nor <- gbl$Oki_Nor |> annotateWithReps.GBreaks(reps$Oki, reps$Nor)
  
  gbl$Osa_Oki <- gbl$Osa_Oki |> annotateWithReps.GBreaks(reps$Osa, reps$Oki)
  gbl$Osa_Bar <- gbl$Osa_Bar |> annotateWithReps.GBreaks(reps$Osa, reps$Bar)
  gbl$Osa_Kum <- gbl$Osa_Kum |> annotateWithReps.GBreaks(reps$Osa, reps$Kum)
  gbl$Osa_Aom <- gbl$Osa_Aom |> annotateWithReps.GBreaks(reps$Osa, reps$Aom)
  gbl$Osa_Nor <- gbl$Osa_Nor |> annotateWithReps.GBreaks(reps$Osa, reps$Nor)
  
  gbl$Bar_Oki <- gbl$Bar_Oki |> annotateWithReps.GBreaks(reps$Bar, reps$Oki)
  gbl$Bar_Osa <- gbl$Bar_Osa |> annotateWithReps.GBreaks(reps$Bar, reps$Osa)
  gbl$Bar_Kum <- gbl$Bar_Kum |> annotateWithReps.GBreaks(reps$Bar, reps$Kum)
  gbl$Bar_Aom <- gbl$Bar_Aom |> annotateWithReps.GBreaks(reps$Bar, reps$Aom)
  gbl$Bar_Nor <- gbl$Bar_Nor |> annotateWithReps.GBreaks(reps$Bar, reps$Nor)
  gbl
}
```

Gene annotations
----------------

```{r load_genes}
(annots <- OikScrambling:::loadAllAnnotations())

annotateWithGenes <-  function(x, annotT, annotQ) UseMethod("annotateWithGenes")

annotateWithGenes.GRanges <- function(x, annotT, annotQ = NULL) {
  # Taken from CAGEr (and I am the author)
  ranges2names <- function (rangesA, rangesB) {
    if (is.null(names(rangesB))) 
        stop(sQuote("rangesB"), " must contain have names.")
    names <- findOverlaps(rangesA, rangesB)
    names <- as(names, "List")
    names <- extractList(names(rangesB), names)
    names <- unique(names)
    names <- unstrsplit(names, ";")
    Rle(names)
  }
  x$genes <- ranges2names(x, GenomicFeatures::genes(annotT))
  x$genes[x$genes == ''] <- NA
  x
}

annotateWithGenes.GBreaks <- function(x, annotT, annotQ) {
  x$query <- annotateWithGenes.GRanges(x$query, annotQ)
  x       <- annotateWithGenes.GRanges(x,       annotT)
  x
}

annotateWithGenes.GRangesList <- function(x, annotT, annotQ = annotT) {
  x$Oki_Osa <- x$Oki_Osa |> annotateWithGenes(annotT$Oki, annotQ$Osa)
  x$Oki_Bar <- x$Oki_Bar |> annotateWithGenes(annotT$Oki, annotQ$Bar)
  x$Oki_Kum <- x$Oki_Kum |> annotateWithGenes(annotT$Oki, annotQ$Kum)
  x$Oki_Aom <- x$Oki_Aom |> annotateWithGenes(annotT$Oki, annotQ$Aom)
  x$Oki_Nor <- x$Oki_Nor |> annotateWithGenes(annotT$Oki, annotQ$Nor)
  
  x$Osa_Oki <- x$Osa_Oki |> annotateWithGenes(annotT$Osa, annotQ$Oki)
  x$Osa_Bar <- x$Osa_Bar |> annotateWithGenes(annotT$Osa, annotQ$Bar)
  x$Osa_Kum <- x$Osa_Kum |> annotateWithGenes(annotT$Osa, annotQ$Kum)
  x$Osa_Aom <- x$Osa_Aom |> annotateWithGenes(annotT$Osa, annotQ$Aom)
  x$Osa_Nor <- x$Osa_Nor |> annotateWithGenes(annotT$Osa, annotQ$Nor)
  
  x$Bar_Oki <- x$Bar_Oki |> annotateWithGenes(annotT$Bar, annotQ$Oki)
  x$Bar_Osa <- x$Bar_Osa |> annotateWithGenes(annotT$Bar, annotQ$Osa)
  x$Bar_Kum <- x$Bar_Kum |> annotateWithGenes(annotT$Bar, annotQ$Kum)
  x$Bar_Aom <- x$Bar_Aom |> annotateWithGenes(annotT$Bar, annotQ$Aom)
  x$Bar_Nor <- x$Bar_Nor |> annotateWithGenes(annotT$Bar, annotQ$Nor)
  x
}

annotateWithGenes.SimpleList <- function(x, annotT, annotQ = annotT)
  annotateWithGenes.GRangesList(x, annotT = annotT, annotQ = annotQ)
```

Mapped alignments
-----------------

Let's load data from genome alignments, using the copy distributed in the
`BreakpointsData` package.

```{r GBreaks_object, cache=TRUE}
# Local function to add extra annotation on long/short arms while loading data
load_genomic_breaks_ <- function(file, genomeT, genomeQ, longShort=NULL) {
  file <- system.file(paste0("extdata/Oidioi_pairwise_v3/", file), package = "BreakpointsData")
  gb <- load_genomic_breaks(file, genomeT,  genomeQ)
  if (! is.null(longShort)) {
    gb <-  flagLongShort(gb, longShort)
  }
  gb
}

gbs <- SimpleList()

gbs$Oki_Osa <- load_genomic_breaks_("OKI2018_I69_1.0__OSKA2016v1.9.gff.gz", OKI2018_I69,  OSKA2016v1.9, longShort$Oki) |> annotateWithReps.GBreaks(reps$Oki, reps$Osa)
gbs$Oki_Bar <- load_genomic_breaks_("OKI2018_I69_1.0__Bar2_p4.gff.gz",      OKI2018_I69,  Bar2_p4,      longShort$Oki) |> annotateWithReps.GBreaks(reps$Oki, reps$Bar)
gbs$Oki_Kum <- load_genomic_breaks_("OKI2018_I69_1.0__KUM-M3-7f.gff.gz",    OKI2018_I69,  KUM_M3,       longShort$Oki) |> annotateWithReps.GBreaks(reps$Oki, reps$Kum)
gbs$Oki_Aom <- load_genomic_breaks_("OKI2018_I69_1.0__AOM-5-5f.gff.gz",     OKI2018_I69,  AOM_5,        longShort$Oki) |> annotateWithReps.GBreaks(reps$Oki, reps$Aom)
gbs$Oki_Nor <- load_genomic_breaks_("OKI2018_I69_1.0__OdB3.gff.gz",         OKI2018_I69,  OdB3,         longShort$Oki) |> annotateWithReps.GBreaks(reps$Oki, reps$Nor)

gbs$Osa_Oki <- load_genomic_breaks_("OSKA2016v1.9__OKI2018_I69_1.0.gff.gz", OSKA2016v1.9, OKI2018_I69,  longShort$Osa) |> annotateWithReps.GBreaks(reps$Osa, reps$Oki)
gbs$Osa_Bar <- load_genomic_breaks_("OSKA2016v1.9__Bar2_p4.gff.gz",         OSKA2016v1.9, Bar2_p4,      longShort$Osa) |> annotateWithReps.GBreaks(reps$Osa, reps$Bar)
gbs$Osa_Kum <- load_genomic_breaks_("OSKA2016v1.9__KUM-M3-7f.gff.gz",       OSKA2016v1.9, KUM_M3,       longShort$Osa) |> annotateWithReps.GBreaks(reps$Osa, reps$Kum)
gbs$Osa_Aom <- load_genomic_breaks_("OSKA2016v1.9__AOM-5-5f.gff.gz",        OSKA2016v1.9, AOM_5,        longShort$Osa) |> annotateWithReps.GBreaks(reps$Osa, reps$Aom)
gbs$Osa_Nor <- load_genomic_breaks_("OSKA2016v1.9__OdB3.gff.gz",            OSKA2016v1.9, OdB3,         longShort$Osa) |> annotateWithReps.GBreaks(reps$Osa, reps$Nor)

gbs$Bar_Oki <- load_genomic_breaks_("Bar2_p4__OKI2018_I69_1.0.gff.gz",      Bar2_p4,     OKI2018_I69,   longShort$Bar) |> annotateWithReps.GBreaks(reps$Bar, reps$Oki)
gbs$Bar_Osa <- load_genomic_breaks_("Bar2_p4__OSKA2016v1.9.gff.gz",         Bar2_p4,     OSKA2016v1.9,  longShort$Bar) |> annotateWithReps.GBreaks(reps$Bar, reps$Osa)
gbs$Bar_Kum <- load_genomic_breaks_("Bar2_p4__KUM-M3-7f.gff.gz",            Bar2_p4,     KUM_M3,        longShort$Bar) |> annotateWithReps.GBreaks(reps$Bar, reps$Kum)
gbs$Bar_Aom <- load_genomic_breaks_("Bar2_p4__AOM-5-5f.gff.gz",             Bar2_p4,     AOM_5,         longShort$Bar) |> annotateWithReps.GBreaks(reps$Bar, reps$Aom)
gbs$Bar_Nor <- load_genomic_breaks_("Bar2_p4__OdB3.gff.gz",                 Bar2_p4,     OdB3,          longShort$Bar) |> annotateWithReps.GBreaks(reps$Bar, reps$Nor)

gbs <- annotateWithGenes(gbs, annots)
```

Unaligned regions
-----------------

Note that the unaligned regions are strandless and unpaired by definition.

```{r unal}
unal  <- BiocParallel::bplapply(gbs, cleanGaps) |> SimpleList()
unal[ 1:5 ] <- BiocParallel::bplapply(unal[ 1:5 ], flagLongShort, longShort$Oki)
unal[ 6:10] <- BiocParallel::bplapply(unal[ 6:10], flagLongShort, longShort$Osa)
unal[11:15] <- BiocParallel::bplapply(unal[11:15], flagLongShort, longShort$Bar)

unal[ 1:5 ] <- BiocParallel::bplapply(unal[ 1:5 ], annotateWithReps.GRanges, reps$Oki)
unal[ 6:10] <- BiocParallel::bplapply(unal[ 6:10], annotateWithReps.GRanges, reps$Osa)
unal[11:15] <- BiocParallel::bplapply(unal[11:15], annotateWithReps.GRanges, reps$Bar)

unal[ 1:5 ] <- BiocParallel::bplapply(unal[ 1:5 ], annotateWithGenes, annots$Oki)
unal[ 6:10] <- BiocParallel::bplapply(unal[ 6:10], annotateWithGenes, annots$Osa)
unal[11:15] <- BiocParallel::bplapply(unal[11:15], annotateWithGenes, annots$Bar)
```

Coalescing alignments
---------------------

```{r coalescing_algorithm, cache=TRUE, dependson="GBreaks_object"}
coa <- BiocParallel::bplapply(gbs, coalesce_contigs) |> SimpleList()
coa[ 1:5]  <- BiocParallel::bplapply(coa[ 1:5],  flagLongShort, longShort$Oki)
coa[ 6:10] <- BiocParallel::bplapply(coa[ 6:10], flagLongShort, longShort$Osa)
coa[11:15] <- BiocParallel::bplapply(coa[11:15], flagLongShort, longShort$Bar)

coa <- annotateWithReps.GBreaksSimpleList(coa, reps)
coa <- annotateWithGenes(coa, annots)
```

Unmapped regions
----------------

We can not call them “uncoalesced” because this would also qualify aligned
regions that were not colinear with any other region.

```{r unmap}
unmap <- BiocParallel::bplapply(coa, cleanGaps) |> SimpleList()

unmap[ 1:5 ] <- BiocParallel::bplapply(unmap[ 1:5 ], flagLongShort, longShort$Oki)
unmap[ 6:10] <- BiocParallel::bplapply(unmap[ 6:10], flagLongShort, longShort$Osa)
unmap[11:15] <- BiocParallel::bplapply(unmap[11:15], flagLongShort, longShort$Bar)

unmap[ 1:5 ] <- BiocParallel::bplapply(unmap[ 1:5 ], annotateWithReps.GRanges, reps$Oki)
unmap[ 6:10] <- BiocParallel::bplapply(unmap[ 6:10], annotateWithReps.GRanges, reps$Osa)
unmap[11:15] <- BiocParallel::bplapply(unmap[11:15], annotateWithReps.GRanges, reps$Bar)

unmap[ 1:5 ] <- BiocParallel::bplapply(unmap[ 1:5 ], annotateWithGenes, annots$Oki)
unmap[ 6:10] <- BiocParallel::bplapply(unmap[ 6:10], annotateWithGenes, annots$Osa)
unmap[11:15] <- BiocParallel::bplapply(unmap[11:15], annotateWithGenes, annots$Bar)
```

Mapped unaligned regions
------------------------

We can pair the unaligned regions that are between two colinearly aligned regions.

```{r mappedUnaligned, cache = TRUE, dependson="GBreaks_object"}
filterMappedUnaligned <- function(gb) {
  colinearRegions <- filterColinearRegions(flagColinearAlignments(gb), rename = FALSE)
  # Turning the sequence of [(TRUE)n, FALSE]n into indices.
  idx <- c(0, head(cumsum(!colinearRegions$colinear), -1))
  gbl <- split(colinearRegions, idx)
  unalMap <- endoapply(gbl, \(gb) {
    # Check strand
    onMinus <- all(strand(gb) == "-")
    # Need to subtract 1 because some ranges are adjacent (no gap)
    GBreaks(target = cleanGaps(gb      - 1),
            query  = cleanGaps(gb$query -1) |> sort(decreasing = onMinus))
  }) |> unlist()
  names(unalMap) <- NULL
  unalMap
}

unalMap <- BiocParallel::bplapply(gbs, filterMappedUnaligned) |> as("SimpleList")

unalMap[ 1:5 ] <- BiocParallel::bplapply(unalMap[ 1:5 ], flagLongShort, longShort$Oki)
unalMap[ 6:10] <- BiocParallel::bplapply(unalMap[ 6:10], flagLongShort, longShort$Osa)
unalMap[11:15] <- BiocParallel::bplapply(unalMap[11:15], flagLongShort, longShort$Bar)

unalMap <- annotateWithReps.GBreaksSimpleList(unalMap, reps)
unalMap <- annotateWithGenes(unalMap, annots)
```

Translocated regions
--------------------

Here we compute new `GBreaks` objects representing the intervals between two
regions that would be colinear if there would not be a translocation separating
them.

```{r translocated_regions, cache=TRUE, dependson="coalescing_algorithm"}
removeTranslocations <- function(gb) {
  if(any(gb$tra))
    gb <- gb[-(which(gb$tra) + 1)]
  gb
}

traGaps <- function(coa) {
  # The input MUST already be coalesced.
  # First, flag translocations on the target genome
  tra.target <- flagTranslocations(coa)$tra
  # Second, do it again on the query genome
  tra <- swap(coa) |> sort(ignore.strand = TRUE) |> flagTranslocations()
  tra.query <- tra$tra
  tra <- swap(tra) |> sort(ignore.strand = TRUE)
  tra$tra <- tra.target | tra.query
  # Then, remove translocated regions
  tra <- removeTranslocations(tra)
  # Then extract the newly "unaligned mapped" regions
  # They contain the gaps left by the removal of the translocated regions.
  filterMappedUnaligned(tra)
}

tra <- BiocParallel::bplapply(coa, traGaps) |> as("SimpleList")

tra[ 1:5 ] <- BiocParallel::bplapply(tra[ 1:5 ], flagLongShort, longShort$Oki)
tra[ 6:10] <- BiocParallel::bplapply(tra[ 6:10], flagLongShort, longShort$Osa)
tra[11:15] <- BiocParallel::bplapply(tra[11:15], flagLongShort, longShort$Bar)

tra <- annotateWithReps.GBreaksSimpleList(tra, reps)
tra <- annotateWithGenes(tra, annots)
```

Coalesce again
--------------

We remove the translocated regions that interrupt colinearity, where one of
the colinear pairs is closer than 200 bp.

```{r re_coalesce, cache=TRUE, dependson="coalescing_algorithm"}
removeTranslocationsOnTargetGenome <- function(gb) {
  gb <- flagTranslocations(gb)
  gb <- dist2next(gb, step = 2, ignore.strand = TRUE)
  idx <- 1 + which( gb$tra & (gb$tdist < 200 | gb$qdist < 200) )
  gb <- gb[-idx]
  coalesce_contigs(gb)
}
removeTranslocationsOnBothGenomes <- function(gb)
  gb |>
    removeTranslocationsOnTargetGenome() |>
    swap() |> sort(i=T) |>
    removeTranslocationsOnTargetGenome() |>
    swap() |> sort(i=T)

coa2 <- BiocParallel::bplapply(coa, removeTranslocationsOnBothGenomes) |> SimpleList()
coa2[ 1:5]  <- BiocParallel::bplapply(coa2[ 1:5],  flagLongShort, longShort$Oki)
coa2[ 6:10] <- BiocParallel::bplapply(coa2[ 6:10], flagLongShort, longShort$Osa)
coa2[11:15] <- BiocParallel::bplapply(coa2[11:15], flagLongShort, longShort$Bar)

coa2 <- annotateWithReps.GBreaksSimpleList(coa2, reps)
coa2 <- annotateWithGenes(coa2, annots)
```

Unmapped regions 2
------------------

The intervals between double-coalesced regions.

```{r unmap2}
unmap2 <- BiocParallel::bplapply(coa2, cleanGaps) |> SimpleList()

unmap2[ 1:5 ] <- BiocParallel::bplapply(unmap2[ 1:5 ], flagLongShort, longShort$Oki)
unmap2[ 6:10] <- BiocParallel::bplapply(unmap2[ 6:10], flagLongShort, longShort$Osa)
unmap2[11:15] <- BiocParallel::bplapply(unmap2[11:15], flagLongShort, longShort$Bar)

unmap2[ 1:5 ] <- BiocParallel::bplapply(unmap2[ 1:5 ], annotateWithReps.GRanges, reps$Oki)
unmap2[ 6:10] <- BiocParallel::bplapply(unmap2[ 6:10], annotateWithReps.GRanges, reps$Osa)
unmap2[11:15] <- BiocParallel::bplapply(unmap2[11:15], annotateWithReps.GRanges, reps$Bar)

unmap2[ 1:5 ] <- BiocParallel::bplapply(unmap2[ 1:5 ], annotateWithGenes, annots$Oki)
unmap2[ 6:10] <- BiocParallel::bplapply(unmap2[ 6:10], annotateWithGenes, annots$Osa)
unmap2[11:15] <- BiocParallel::bplapply(unmap2[11:15], annotateWithGenes, annots$Bar)
```

Numbers of blocks
-----------------

```{r lengths}
rbind(
  gbs     = sapply(gbs, length),
  unal    = sapply(unal, length),
  coa     = sapply(coa, length),
  unmap   = sapply(unmap, length),
  unmap2  = sapply(unmap2, length),
  unalMap = sapply(unalMap, length),
  tra     = sapply(tra, length),
  coa2    = sapply(coa2, length)
)
```

Summary
-------

```{r summary}
df <- cbind(
  gbs = sapply(gbs, length),
  coa = sapply(coa, length),
  coa2 = sapply(coa2, length)
) |> as.data.frame() |>
  dplyr::arrange(gbs = gbs - coa - coa2, coa = coa - coa2)

df$pair <- row.names(df)

ggplot(df |> tidyr::pivot_longer(!pair)) +
  aes(pair, value, fill = name) +
  geom_bar( stat = "identity" ) +
  ggplot2::coord_flip() +
  ggtitle("Cumulative plot (gbs ⊃ coa ⊃ coa2) of the computed genomic regions")
```

Save data
=========

```{r save_data}
save(gbs, unal, coa, unmap, unalMap, tra, coa2, unmap2, longShort, file = "BreakPoints.Rdata")
system("echo '9f273f099ecbfa44899b6c343d1382f1  BreakPoints.Rdata' | md5sum -c", intern = TRUE)
```

Session information
===================

```{r session_info}
sessionInfo()
```
