---
title: "Load genomic break data"
author: 
 - "Charles Plessy"
date: "`r format(Sys.time(), '%d %B, %Y')`"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteIndexEntry{Load genomic break data}
  %\VignetteEncoding{UTF-8}
editor_options: 
  chunk_output_type: console
---

```{r setup}
knitr::opts_knit$set(verbose = TRUE)
```

Introduction
============

This vignette loads the genomic breaks data, processes it and saves the
resulting objects in a serialised R object (`RData` file) for reuse in other
vignettes.

Similar objects are stored in lists to avoid cluttering the environment.

Pacakges and data
=================


```{r load_pacakges_and_data}
suppressPackageStartupMessages({
  library('GenomicBreaks')
  library('GenomicFeatures')
  library('ggplot2')
  library("BreakpointsData")
})
(genomes <- OikScrambling:::loadAllGenomes())
```

Genomic breaks
==============

Long/short arm information
--------------------------

Short / long arm annotation will be done by the `GenomicBreaks::flagLongShort()`
function.

```{r long_and_short_arms}

loadLongShort <- function(file, genome) {
  file <- system.file(paste0("extdata/Annotations/", file), package = "BreakpointsData")
  gr <- rtracklayer::import.gff3(file)
  gr$source <- gr$type <- gr$score <- gr$phase <- NULL
  GRanges(gr, seqinfo = seqinfo(genome), strand = "*")
}

longShort <- SimpleList()
longShort$Oki <- loadLongShort("OKI2018_I69.v2/OKI2018_I69.arms.gff3", OKI2018_I69)
longShort$Osa <- loadLongShort("OSKA2016v1.9/OSKA2016v1.9.arms.gff3", OSKA2016v1.9)
longShort$Bar <- loadLongShort("Bar2_p4.Flye/Bar2_p4.Flye.arms.gff3", Bar2_p4)

longShort
```

Repeat regions
--------------

Load from `BreakpointsData` package via `OikScrambling:::loadAllRepeats()`

```{r load_repeats}
(reps <- OikScrambling:::loadAllRepeats())
```

Mapped alignments
-----------------

Let's load data from genome alignments, using the copy distributed in the
`BreakpointsData` package.

```{r GBreaks_object, cache=TRUE}
# Local function to add extra annotation on long/short arms while loading data
load_genomic_breaks_ <- function(file, genomeT, genomeQ, longShort=NULL) {
  file <- system.file(paste0("extdata/Oidioi_pairwise_v3/", file), package = "BreakpointsData")
  gb <- load_genomic_breaks(file, genomeT,  genomeQ)
  if (! is.null(longShort)) {
    gb <-  flagLongShort(gb, longShort)
  }
  gb
}

annotateWithReps.GRanges <- function(gr, rep) {
  o         <- findOverlaps(gr, rep)
  o.grouped <- tapply(subjectHits(o), queryHits(o), c)
  o.names   <- sapply(o.grouped, \(x) rep[x]$Class |> unique() |> as.character())
  gr$rep <- NA
  gr$rep[as.integer(names(o.names))] <- o.names
  gr
}

annotateWithReps.GBreaks <- function(gb, repT, repQ) {
  gb$query <- annotateWithReps.GRanges(gb$query, repQ)
  gb       <- annotateWithReps.GRanges(gb,       repT)
}

gbs <- SimpleList()

gbs$Oki_Osa <- load_genomic_breaks_("OKI2018_I69_1.0__OSKA2016v1.9.gff.gz", OKI2018_I69,  OSKA2016v1.9, longShort$Oki) |> annotateWithReps.GBreaks(reps$Oki, reps$Osa)
gbs$Oki_Bar <- load_genomic_breaks_("OKI2018_I69_1.0__Bar2_p4.gff.gz",      OKI2018_I69,  Bar2_p4,      longShort$Oki) |> annotateWithReps.GBreaks(reps$Oki, reps$Bar)
gbs$Oki_Kum <- load_genomic_breaks_("OKI2018_I69_1.0__KUM-M3-7f.gff.gz",    OKI2018_I69,  KUM_M3,       longShort$Oki) |> annotateWithReps.GBreaks(reps$Oki, reps$Kum)
gbs$Oki_Aom <- load_genomic_breaks_("OKI2018_I69_1.0__AOM-5-5f.gff.gz",     OKI2018_I69,  AOM_5,        longShort$Oki) |> annotateWithReps.GBreaks(reps$Oki, reps$Aom)
gbs$Oki_Nor <- load_genomic_breaks_("OKI2018_I69_1.0__OdB3.gff.gz",         OKI2018_I69,  OdB3,         longShort$Oki)

gbs$Osa_Oki <- load_genomic_breaks_("OSKA2016v1.9__OKI2018_I69_1.0.gff.gz", OSKA2016v1.9, OKI2018_I69,  longShort$Osa) |> annotateWithReps.GBreaks(reps$Osa, reps$Oki)
gbs$Osa_Bar <- load_genomic_breaks_("OSKA2016v1.9__Bar2_p4.gff.gz",         OSKA2016v1.9, Bar2_p4,      longShort$Osa) |> annotateWithReps.GBreaks(reps$Osa, reps$Bar)
gbs$Osa_Kum <- load_genomic_breaks_("OSKA2016v1.9__KUM-M3-7f.gff.gz",       OSKA2016v1.9, KUM_M3,       longShort$Osa) |> annotateWithReps.GBreaks(reps$Osa, reps$Kum)
gbs$Osa_Aom <- load_genomic_breaks_("OSKA2016v1.9__AOM-5-5f.gff.gz",        OSKA2016v1.9, AOM_5,        longShort$Osa) |> annotateWithReps.GBreaks(reps$Osa, reps$Aom)
gbs$Osa_Nor <- load_genomic_breaks_("OSKA2016v1.9__OdB3.gff.gz",            OSKA2016v1.9, OdB3,         longShort$Osa)

gbs$Bar_Oki <- load_genomic_breaks_("Bar2_p4__OKI2018_I69_1.0.gff.gz",      Bar2_p4,     OKI2018_I69,   longShort$Bar) |> annotateWithReps.GBreaks(reps$Bar, reps$Oki)
gbs$Bar_Osa <- load_genomic_breaks_("Bar2_p4__OSKA2016v1.9.gff.gz",         Bar2_p4,     OSKA2016v1.9,  longShort$Bar) |> annotateWithReps.GBreaks(reps$Bar, reps$Osa)
gbs$Bar_Kum <- load_genomic_breaks_("Bar2_p4__KUM-M3-7f.gff.gz",            Bar2_p4,     KUM_M3,        longShort$Bar) |> annotateWithReps.GBreaks(reps$Bar, reps$Kum)
gbs$Bar_Aom <- load_genomic_breaks_("Bar2_p4__AOM-5-5f.gff.gz",             Bar2_p4,     AOM_5,         longShort$Bar) |> annotateWithReps.GBreaks(reps$Bar, reps$Aom)
gbs$Bar_Nor <- load_genomic_breaks_("Bar2_p4__OdB3.gff.gz",                 Bar2_p4,     OdB3,          longShort$Bar)
```

Unaligned regions
-----------------

Note that the unaligned regions are strandless and unpaired by definition.

```{r unal}
unal  <- sapply(gbs, cleanGaps) |> SimpleList()
unal[ 1:5 ] <- sapply(unal[ 1:5 ], flagLongShort, longShort$Oki)
unal[ 6:10] <- sapply(unal[ 6:10], flagLongShort, longShort$Osa)
unal[11:15] <- sapply(unal[11:15], flagLongShort, longShort$Bar)

unal[ 1:5 ] <- sapply(unal[ 1:5 ], annotateWithReps.GRanges, reps$Oki)
unal[ 6:10] <- sapply(unal[ 6:10], annotateWithReps.GRanges, reps$Osa)
unal[11:15] <- sapply(unal[11:15], annotateWithReps.GRanges, reps$Bar)
```

Coalescing alignments
---------------------

```{r coalescing_algorithm, cache=TRUE, dependson="GBreaks_object"}
coa <- sapply(gbs, coalesce_contigs) |> SimpleList()
coa[ 1:5]  <- sapply(coa[ 1:5],  flagLongShort, longShort$Oki)
coa[ 6:10] <- sapply(coa[ 6:10], flagLongShort, longShort$Osa)
coa[11:15] <- sapply(coa[11:15], flagLongShort, longShort$Bar)

coa$Oki_Osa <- coa$Oki_Osa |> annotateWithReps.GBreaks(reps$Oki, reps$Osa)
coa$Oki_Bar <- coa$Oki_Bar |> annotateWithReps.GBreaks(reps$Oki, reps$Bar)
coa$Oki_Kum <- coa$Oki_Kum |> annotateWithReps.GBreaks(reps$Oki, reps$Kum)
coa$Oki_Aom <- coa$Oki_Aom |> annotateWithReps.GBreaks(reps$Oki, reps$Aom)

coa$Osa_Oki <- coa$Osa_Oki |> annotateWithReps.GBreaks(reps$Osa, reps$Oki)
coa$Osa_Bar <- coa$Osa_Bar |> annotateWithReps.GBreaks(reps$Osa, reps$Bar)
coa$Osa_Kum <- coa$Osa_Kum |> annotateWithReps.GBreaks(reps$Osa, reps$Kum)
coa$Osa_Aom <- coa$Osa_Aom |> annotateWithReps.GBreaks(reps$Osa, reps$Aom)

coa$Bar_Oki <- coa$Bar_Oki |> annotateWithReps.GBreaks(reps$Bar, reps$Oki)
coa$Bar_Osa <- coa$Bar_Osa |> annotateWithReps.GBreaks(reps$Bar, reps$Osa)
coa$Bar_Kum <- coa$Bar_Kum |> annotateWithReps.GBreaks(reps$Bar, reps$Kum)
coa$Bar_Aom <- coa$Bar_Aom |> annotateWithReps.GBreaks(reps$Bar, reps$Aom)
```

Unmapped regions
----------------

We can not call them “uncoalesced” because this would also qualify aligned
regions that were not colinear with any other region.

```{r unmap}
unmap <- sapply(coa, cleanGaps) |> SimpleList()

unmap[ 1:5 ] <- sapply(unmap[ 1:5 ], flagLongShort, longShort$Oki)
unmap[ 6:10] <- sapply(unmap[ 6:10], flagLongShort, longShort$Osa)
unmap[11:15] <- sapply(unmap[11:15], flagLongShort, longShort$Bar)

unmap[ 1:5 ] <- sapply(unmap[ 1:5 ], annotateWithReps.GRanges, reps$Oki)
unmap[ 6:10] <- sapply(unmap[ 6:10], annotateWithReps.GRanges, reps$Osa)
unmap[11:15] <- sapply(unmap[11:15], annotateWithReps.GRanges, reps$Bar)
```

Mapped unaligned regions
------------------------

We can pair the unaligned regions that are between two colinearly aligned regions.

```{r mappedUnaligned}
filterMappedUnaligned <- function(gb) {
  colinearRegions <- filterColinearRegions(flagColinearAlignments(gb), rename = FALSE)
  # Turning the sequence of [(TRUE)n, FALSE]n into indices.
  idx <- c(0, head(cumsum(!colinearRegions$colinear), -1))
  gbl <- split(colinearRegions, idx)
  unalMap <- endoapply(gbl, \(gb)
    # Need to subtract 1 because some ranges are adjacent (no gap)
    GBreaks(target = cleanGaps(gb      - 1),
            query  = cleanGaps(gb$query -1))
  ) |> unlist()
  names(unalMap) <- NULL
  unalMap
}

unalMap <- sapply(gbs, filterMappedUnaligned) |> as("SimpleList")

unalMap[ 1:5 ] <- sapply(unalMap[ 1:5 ], flagLongShort, longShort$Oki)
unalMap[ 6:10] <- sapply(unalMap[ 6:10], flagLongShort, longShort$Osa)
unalMap[11:15] <- sapply(unalMap[11:15], flagLongShort, longShort$Bar)

unalMap$Oki_Osa <- unalMap$Oki_Osa |> annotateWithReps.GBreaks(reps$Oki, reps$Osa)
unalMap$Oki_Bar <- unalMap$Oki_Bar |> annotateWithReps.GBreaks(reps$Oki, reps$Bar)
unalMap$Oki_Kum <- unalMap$Oki_Kum |> annotateWithReps.GBreaks(reps$Oki, reps$Kum)
unalMap$Oki_Aom <- unalMap$Oki_Aom |> annotateWithReps.GBreaks(reps$Oki, reps$Aom)

unalMap$Osa_Oki <- unalMap$Osa_Oki |> annotateWithReps.GBreaks(reps$Osa, reps$Oki)
unalMap$Osa_Bar <- unalMap$Osa_Bar |> annotateWithReps.GBreaks(reps$Osa, reps$Bar)
unalMap$Osa_Kum <- unalMap$Osa_Kum |> annotateWithReps.GBreaks(reps$Osa, reps$Kum)
unalMap$Osa_Aom <- unalMap$Osa_Aom |> annotateWithReps.GBreaks(reps$Osa, reps$Aom)

unalMap$Bar_Oki <- unalMap$Bar_Oki |> annotateWithReps.GBreaks(reps$Bar, reps$Oki)
unalMap$Bar_Osa <- unalMap$Bar_Osa |> annotateWithReps.GBreaks(reps$Bar, reps$Osa)
unalMap$Bar_Kum <- unalMap$Bar_Kum |> annotateWithReps.GBreaks(reps$Bar, reps$Kum)
unalMap$Bar_Aom <- unalMap$Bar_Aom |> annotateWithReps.GBreaks(reps$Bar, reps$Aom)
```

Coalesce again
--------------

See the Genomic Breaks vignette for justification.

```{r re_coalesce, cache=TRUE, dependson="coalescing_algorithm"}
coa2 <- sapply(coa, coalesce_contigs, minwidth = 500) |> SimpleList()
coa2[ 1:5]  <- sapply(coa2[ 1:5],  flagLongShort, longShort$Oki)
coa2[ 6:10] <- sapply(coa2[ 6:10], flagLongShort, longShort$Osa)
coa2[11:15] <- sapply(coa2[11:15], flagLongShort, longShort$Bar)
```

Coalesce within chromosomes
---------------------------

The idea is that if colinearity is only lost because of a translocation to
another chromosome, then it is fine.

```{r coalesce_by_chr, cache=TRUE, dependson="GBreaks_object"}
gbs$Oki_Kum -> x
gbl <- split(x, seqnames(x), drop=T)
gbl <- endoapply(gbl, coalesce_contigs)
unlist(gbl)
```

Numbers of blocks
-----------------

```{r lengths}
rbind(
  gbs     = sapply(gbs, length),
  unal    = sapply(unal, length),
  coa     = sapply(coa, length),
  unmap   = sapply(unmap, length),
  unalMap = sapply(unalMap, length),
  coa2    = sapply(coa2, length)
)
```

Summary
-------

```{r summary}
df <- cbind(
  gbs = sapply(gbs, length),
  coa = sapply(coa, length),
  coa2 = sapply(coa2, length)
) |> as.data.frame() |>
  dplyr::arrange(gbs = gbs - coa - coa2, coa = coa - coa2)

df$pair <- row.names(df)

ggplot(df |> tidyr::pivot_longer(!pair)) +
  aes(pair, value, fill = name) +
  geom_bar( stat = "identity" ) +
  ggplot2::coord_flip() +
  ggtitle("Cumulative plot (gbs ⊃ coa ⊃ coa2) of the computed genomic regions")
```

Save data
=========

```{r save_data}
save(gbs, unal, coa, unmap, unalMap, coa2, longShort, file = "BreakPoints.Rdata")
system("echo 'c042554ce0eaed4e2d38235c075edf92  BreakPoints.Rdata' | md5sum -c", intern = TRUE)
```

Session information
===================

```{r session_info}
sessionInfo()
```
