---
title: "Load genomic break data"
author: 
 - "Charles Plessy"
date: "`r format(Sys.time(), '%d %B, %Y')`"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteIndexEntry{Load genomic break data}
  %\VignetteEncoding{UTF-8}
editor_options: 
  chunk_output_type: console
---

```{r setup}
knitr::opts_chunk$set(cache = TRUE)
knitr::opts_knit$set(verbose = TRUE)
```

Introduction
============

This vignette loads the genomic breaks data, processes it and saves the
resulting objects in a serialised R object (`RData` file) for reuse in other
vignettes.

Similar objects are stored in lists to avoid cluttering the environment.

Pacakges
========

Core packages that provide functions we use a lot.

```{r loadlib, echo=T, results='hide'}
library('GenomicBreaks')
suppressPackageStartupMessages({
  library('BSgenome')
  library('GenomicFeatures')
  library('GenomicRanges')
  library('ggplot2')
})
```

BSgenome packages
-----------------

Example command to install one _Oikopleura_ BSgenome package if you do not yet
have it:

```
install.packages("BSgenome.Odioica.local.OKI2018.I69", repos="https://oist.github.io/plessy_oikgenomes_drat/")
```

```{r load_BSgenome_packages}
# These will need to be installed by the user
library("BSgenome.Odioica.local.OKI2018.I69")
library("BSgenome.Odioica.local.OSKA2016v1.9")
library("BSgenome.Odioica.local.Bar2.p4")
library("BSgenome.Odioica.local.KUM.M3")
library("BSgenome.Odioica.local.AOM.5")
library("BSgenome.Odioica.local.Odioica.reference.v3.0")

genomes <- SimpleList(
  # Chromosome assemblies
  Oki = OKI2018_I69, Osa = OSKA2016v1.9, Bar = Bar2_p4,
  # Less contiguous assemblies
  Kum = KUM_M3, Aom = AOM_5, Nor = OdB3)
```

Annotations
===========

The annotations from `/bucket/LuscombeU/common/Breakpoints/Annotations` are
distributed in the same drat repository as the alignment data.

```
install.packages("BreakpointsData", repos="https://oist.github.io/plessy_oikgenomes_drat/")
```

```{r load_annotations}
annots <- SimpleList()

gff2txdb <- function(file, genome) {
  tx <- rtracklayer::import.gff(file)
  tx <- tx[seqnames(tx) %in% seqnames(genome)]
  tx <- GRanges(tx, seqinfo = seqinfo(genome))
  tx <- GenomicFeatures::makeTxDbFromGRanges(tx)
}

gff2txdb_Norway <- function(file, genome) {
  tx <- rtracklayer::import.gff(file)
  # Remove "scaffoldA" objects in the OdB3 annotation
  tx <- tx[seqnames(tx) %in% seqnames(genome)]
  tx$ID <- NA_character_
  mcols(tx)[tx$type == "mRNA","ID"] <- paste0(tx[tx$type == "mRNA"]$mRNA, ".mRNA")
  mcols(tx)[tx$type == "CDS","ID"] <- paste0(tx[tx$type == "CDS"]$CDS, ".CDS")
  mcols(tx)[tx$type == "UTR","ID"] <- paste0(tx[tx$type == "UTR"]$UTR, ".UTR")
  tx <- GRanges(tx, seqinfo = seqinfo(genome))
  tx <- GenomicFeatures::makeTxDbFromGRanges(tx)
}

annots$Oki <- gff2txdb("../Annotations/OKI2018_I69.v2/OKI2018_I69.v2.gm.gff", OKI2018_I69)
annots$Osa <- gff2txdb("../Annotations/OSKA2016v1.9/OSKA2016v1.9.gm.gff",     OSKA2016v1.9)
annots$Bar <- gff2txdb("../Annotations/Bar2_p4.Flye/Bar2_p4.Flye.gm.gff", Bar2_p4)
annots$Kum <- gff2txdb("../Annotations/KUM-M3-7f/KUM-M3-7f.gm.gff", KUM_M3)
annots$Aom <- gff2txdb("../Annotations/AOM-5-5f/AOM-5-5f.gm.gff", AOM_5)
annots$Nor <- gff2txdb_Norway("../Annotations/OdB3/Oikopleura_annot_v1.0.gff", OdB3)
```

Long and short arms
-------------------

(in progress)

```{r long_and_short_arms}
# OKI_longShort <- rtracklayer::import.gff3(system.file("extdata/OKI2018_I69.arms.gff3", package = "GenomicBreaks"))
# OKI_longShort <- GRanges(OKI_longShort, seqinfo = seqinfo(OKI2018_I69), strand = "*")
```

Genomic breaks
==============

Mapped alignments
-----------------

Let's load data from genome alignments, copied in the `Oidioi_pairwise_v3`
directory.

```{r GRanges object}
gbs <- SimpleList()
load_genomic_breaks_ <- function(file, genomeT, genomeQ, longShort=NULL) {
  file <- system.file(paste0("extdata/", file), package = "BreakpointsData")
  gb <- load_genomic_breaks(file, genomeT,  genomeQ)
  if (! is.null(longShort))
    gb$Type <- mcols(longShort)[findOverlaps(gb, longShort) |> subjectHits(), "Type"]
  gb
}

gbs$Oki_Osa <- load_genomic_breaks_("Oidioi_pairwise_v3/OKI2018_I69_1.0__OSKA2016v1.9.gff.gz", OKI2018_I69,  OSKA2016v1.9)
gbs$Oki_Bar <- load_genomic_breaks_("Oidioi_pairwise_v3/OKI2018_I69_1.0__Bar2_p4.gff.gz",      OKI2018_I69,  Bar2_p4)
gbs$Oki_Kum <- load_genomic_breaks_("Oidioi_pairwise_v3/OKI2018_I69_1.0__KUM-M3-7f.gff.gz",    OKI2018_I69,  KUM_M3)
gbs$Oki_Aom <- load_genomic_breaks_("Oidioi_pairwise_v3/OKI2018_I69_1.0__AOM-5-5f.gff.gz",     OKI2018_I69,  AOM_5)
gbs$Oki_Nor <- load_genomic_breaks_("Oidioi_pairwise_v3/OKI2018_I69_1.0__OdB3.gff.gz",         OKI2018_I69,  OdB3)

gbs$Osa_Oki <- load_genomic_breaks_("Oidioi_pairwise_v3/OSKA2016v1.9__OKI2018_I69_1.0.gff.gz", OSKA2016v1.9, OKI2018_I69)
gbs$Osa_Bar <- load_genomic_breaks_("Oidioi_pairwise_v3/OSKA2016v1.9__Bar2_p4.gff.gz",         OSKA2016v1.9, Bar2_p4)
gbs$Osa_Kum <- load_genomic_breaks_("Oidioi_pairwise_v3/OSKA2016v1.9__KUM-M3-7f.gff.gz",       OSKA2016v1.9, KUM_M3)
gbs$Osa_Aom <- load_genomic_breaks_("Oidioi_pairwise_v3/OSKA2016v1.9__AOM-5-5f.gff.gz",        OSKA2016v1.9, AOM_5)
gbs$Osa_Nor <- load_genomic_breaks_("Oidioi_pairwise_v3/OSKA2016v1.9__OdB3.gff.gz",            OSKA2016v1.9, OdB3)

gbs$Bar_Oki <- load_genomic_breaks_("Oidioi_pairwise_v3/Bar2_p4__OKI2018_I69_1.0.gff.gz",      Bar2_p4,     OKI2018_I69)
gbs$Bar_Osa <- load_genomic_breaks_("Oidioi_pairwise_v3/Bar2_p4__OSKA2016v1.9.gff.gz",         Bar2_p4,     OSKA2016v1.9)
gbs$Bar_Kum <- load_genomic_breaks_("Oidioi_pairwise_v3/Bar2_p4__KUM-M3-7f.gff.gz",            Bar2_p4,     KUM_M3)
gbs$Bar_Aom <- load_genomic_breaks_("Oidioi_pairwise_v3/Bar2_p4__AOM-5-5f.gff.gz",             Bar2_p4,     AOM_5)
gbs$Bar_Nor <- load_genomic_breaks_("Oidioi_pairwise_v3/Bar2_p4__OdB3.gff.gz",                 Bar2_p4,     OdB3)
```

Coalescing alignments
---------------------

```{r coalescing algorithm}
coa <- sapply(gbs, coalesce_contigs) |> SimpleList()
sapply(gbs, length)
sapply(coa, length)
```

Coalesce again
--------------

See the Genomic Breaks vignette for justification.

```{r re_coalesce}
coa2 <- sapply(coa, coalesce_contigs, minwidth = 500) |> SimpleList()
```

Summary
-------

```{r summary}
df <- rbind(
  gbs = sapply(gbs, length),
  coa = sapply(coa, length),
  coa2 = sapply(coa2, length)
) |> as.data.frame()

df$what <- row.names(df)
df <- reshape2::melt(df)

ggplot(df) +
  aes(variable, value, fill = what) +
  geom_bar( stat = "identity" ) +
  ggplot2::coord_flip()
```

Save data
=========

```{r save_data}
save(annots, gbs, coa, coa2, file = "BreakPoints.Rdata")
system("echo 'f314eefba7f0a8124468cd88b0ea93cf  BreakPoints.Rdata' | md5sum -c", intern = TRUE)
```

Session information
===================

```{r session_info}
sessionInfo()
```
