---
title: "Load genomic break data"
author: 
 - "Charles Plessy"
date: "`r format(Sys.time(), '%d %B, %Y')`"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{Load genomic break data}
  %\VignetteEncoding{UTF-8}
  %\VignetteEngine{knitr::rmarkdown}
editor_options: 
  chunk_output_type: console
---

```{r setup}
knitr::opts_knit$set(verbose = TRUE, cache = TRUE)
```

Introduction
============

This vignette loads the genomic breaks data, consolidates colinear regions,
extracts unaligned and unmapped regions, transfers annotation data and saves the
resulting objects in a serialised R object (`RData` file) for reuse in other
vignettes.

Related objects are stored in lists to avoid cluttering the environment.

Packages and data
=================


```{r load_pacakges_and_data}
suppressPackageStartupMessages({
  library('GenomicBreaks')
  library('ggplot2')
  library("BreakpointsData")
})
genomes <- OikScrambling:::loadAllGenomes(compat = FALSE)
```

Genomic breaks
==============

Long/short arm information
--------------------------

Short / long arm annotation will be done by the `GenomicBreaks::flagLongShort()`
function.

```{r long_and_short_arms}
loadLongShort <- function(file, genome) {
  file <- system.file(paste0("extdata/Annotations/", file), package = "BreakpointsData")
  gr <- rtracklayer::import.gff3(file)
  gr$source <- gr$type <- gr$score <- gr$phase <- NULL
  GRanges(gr, seqinfo = seqinfo(getBSgenome(genome)), strand = "*")
}

flagLongShort_ <- function(gr, longShort) {
  genomeT <- gr |> genome() |> unique()
  if(genomeT %in% names(longShort))
    gr <- flagLongShort(gr, longShort[[genomeT]])
  if("GBreaks" %in% class(gr)) {
      genomeQ <- gr$query |> genome() |> unique()
      if(genomeQ %in% names(longShort)) {
        # sort, flag, unsort, extract flag
        grq <- gr$query
        grq$idx <- seq_along(grq)
        grq <- sort(grq, ignore.strand = TRUE)
        grq <- flagLongShort(grq, longShort[[genomeQ]])
        gr$query$Arm <- grq$Arm[order(grq$idx)]
      }
  }
  gr
}

longShort <- SimpleList()
longShort$OKI2018.I69   <- loadLongShort("OKI2018_I69.v2/OKI2018_I69.arms.gff3", "OKI2018.I69")
longShort$OSKA2016v1.9  <- loadLongShort("OSKA2016v1.9/OSKA2016v1.9.arms.gff3",  "OSKA2016v1.9")
longShort$Bar2.p4       <- loadLongShort("Bar2_p4.Flye/Bar2_p4.Flye.arms.gff3",  "Bar2.p4")

longShort
```

Repeat regions
--------------

Load from `BreakpointsData` package via `OikScrambling:::loadAllRepeats()`

```{r load_repeats}
(reps <- OikScrambling:::loadAllRepeats(compat = FALSE))

annotateWithReps <-  function(x, rep) UseMethod("annotateWithReps")

annotateWithReps.GRanges <- function(x, rep) {
  genome <- x |> genome() |> unique()
  # Inspired from CAGEr:::ranges2names
  o <- findOverlaps(x, rep[[genome]], ignore.strand = TRUE)
  o <- as(o, "List")
  o <- extractList(rep[[genome]]$Class |> as.character(), o)
  o <- endoapply(o, unique)
  o[sapply(o, length) == 0] <- NA
  x$rep <- o
  x
}

overlapLength.GRanges <- function(x, rep) {
  genome <- x |> genome() |> unique()
  # Inspired from CAGEr:::ranges2names
  redRep <- reduce(rep[[genome]])
  o <- findOverlaps(x, redRep, ignore.strand = TRUE)
  o <- as(o, "List")
  repGrl <- extractList(redRep, o)
  x$repOvlp <- pintersect(x, repGrl) |> width() |> sum()
  x
}

annotateWithReps.GBreaks <-  function(x,       rep) {
  x$query <- annotateWithReps.GRanges(x$query, rep)
  x       <- annotateWithReps.GRanges(x,       rep)
  x$query <- overlapLength.GRanges   (x$query, rep)
  x       <- overlapLength.GRanges   (x,       rep)
  x
}

annotateWithReps.SimpleList <- function(x, rep)
  sapply(x, annotateWithReps, rep) |> SimpleList()
```

Transcript annotations
----------------------

```{r annotate_transcripts}
(transcripts <- OikScrambling:::loadAllTranscriptsGR(compat = FALSE))

annotateWithTranscripts <-  function(x, transcripts) UseMethod("annotateWithTranscripts")

annotateWithTranscripts.GRanges <- function(x, transcripts) {
  # Taken from CAGEr (and I am the author)
  ranges2names <- function (rangesA, rangesB) {
    if (is.null(names(rangesB))) 
        stop(sQuote("rangesB"), " must contain have names.")
    names <- findOverlaps(rangesA, rangesB)
    names <- as(names, "List")
    names <- extractList(names(rangesB), names)
    names <- unique(names)
    names <- unstrsplit(names, ";")
    Rle(names)
  }
  genomeT <- x |> genome() |> unique()
  x$transcripts <- ranges2names(x, transcripts[[genomeT]])
  x$transcripts[x$transcripts == ''] <- NA
  x
}

annotateWithTranscripts.GBreaks <- function(x, transcripts) {
  x$query <- annotateWithTranscripts.GRanges(x$query, transcripts)
  x       <- annotateWithTranscripts.GRanges(x,       transcripts)
  x
}

annotateWithTranscripts.SimpleList <- function(x, transcripts)
  sapply(x, annotateWithTranscripts, transcripts) |> SimpleList()
```

Mapped alignments
-----------------

Let's load data from genome alignments, using the copy distributed in the
`BreakpointsData` package.

```{r GBreaks_object, cache=TRUE}
# Local function to add extra annotation on long/short arms while loading data
load_genomic_breaks_ <- function(file, genomeT, genomeQ, longShort=NULL) {
  file <- system.file(paste0("extdata/Oidioi_pairwise_v3/", file), package = "BreakpointsData")
  gb <- load_genomic_breaks(file, getBSgenome(genomeT),  getBSgenome(genomeQ))
  flagLongShort_(gb, longShort)
}

gbs <- SimpleList()

gbs$Oki_Osa <- load_genomic_breaks_("OKI2018_I69_1.0__OSKA2016v1.9.gff.gz", "OKI2018.I69",  "OSKA2016v1.9", longShort)
gbs$Oki_Bar <- load_genomic_breaks_("OKI2018_I69_1.0__Bar2_p4.gff.gz",      "OKI2018.I69",  "Bar2.p4",      longShort)
gbs$Oki_Kum <- load_genomic_breaks_("OKI2018_I69_1.0__KUM-M3-7f.gff.gz",    "OKI2018.I69",  "KUM.M3.7f",    longShort)
gbs$Oki_Aom <- load_genomic_breaks_("OKI2018_I69_1.0__AOM-5-5f.gff.gz",     "OKI2018.I69",  "AOM.5.5f",     longShort)
gbs$Oki_Nor <- load_genomic_breaks_("OKI2018_I69_1.0__OdB3.gff.gz",         "OKI2018.I69",  "OdB3",         longShort)

gbs$Osa_Oki <- load_genomic_breaks_("OSKA2016v1.9__OKI2018_I69_1.0.gff.gz", "OSKA2016v1.9", "OKI2018.I69",  longShort)
gbs$Osa_Bar <- load_genomic_breaks_("OSKA2016v1.9__Bar2_p4.gff.gz",         "OSKA2016v1.9", "Bar2.p4",      longShort)
gbs$Osa_Kum <- load_genomic_breaks_("OSKA2016v1.9__KUM-M3-7f.gff.gz",       "OSKA2016v1.9", "KUM.M3.7f",    longShort)
gbs$Osa_Aom <- load_genomic_breaks_("OSKA2016v1.9__AOM-5-5f.gff.gz",        "OSKA2016v1.9", "AOM.5.5f",     longShort)
gbs$Osa_Nor <- load_genomic_breaks_("OSKA2016v1.9__OdB3.gff.gz",            "OSKA2016v1.9", "OdB3",         longShort)

gbs$Bar_Oki <- load_genomic_breaks_("Bar2_p4__OKI2018_I69_1.0.gff.gz",      "Bar2.p4",     "OKI2018.I69",   longShort)
gbs$Bar_Osa <- load_genomic_breaks_("Bar2_p4__OSKA2016v1.9.gff.gz",         "Bar2.p4",     "OSKA2016v1.9",  longShort)
gbs$Bar_Kum <- load_genomic_breaks_("Bar2_p4__KUM-M3-7f.gff.gz",            "Bar2.p4",     "KUM.M3.7f",     longShort)
gbs$Bar_Aom <- load_genomic_breaks_("Bar2_p4__AOM-5-5f.gff.gz",             "Bar2.p4",     "AOM.5.5f",      longShort)
gbs$Bar_Nor <- load_genomic_breaks_("Bar2_p4__OdB3.gff.gz",                 "Bar2.p4",     "OdB3",          longShort)

Oik_Oik <- names(gbs)

gbs$Ply_Ros <- load_genomic_breaks(system.file("extdata/Ciona_pairwise_v3/C_int_P__C_int_R.gff.gz",          package = "BreakpointsData"),    NULL,     NULL)
gbs$Ply_Rob <- load_genomic_breaks(system.file("extdata/Ciona_pairwise_v3/C_int_P__C_rob_HT.gff.gz",         package = "BreakpointsData"),    NULL,     NULL)
gbs$Ply_Sav <- load_genomic_breaks(system.file("extdata/Ciona_pairwise_v3/C_int_P__C_sav.gff.gz",            package = "BreakpointsData"),    NULL,     NULL)
gbs$Ply_Oki <- load_genomic_breaks(system.file("extdata/Ciona_pairwise_v3/C_int_P__OKI2018_I69_1.0.gff.gz",  package = "BreakpointsData"),    NULL, genomes$OKI2018.I69)

gbs$Rob_Ros <- load_genomic_breaks(system.file("extdata/Ciona_pairwise_v3/C_rob_HT__C_int_R.gff.gz",         package = "BreakpointsData"),    NULL,     NULL)
gbs$Rob_Ply <- load_genomic_breaks(system.file("extdata/Ciona_pairwise_v3/C_rob_HT__C_int_P.gff.gz",         package = "BreakpointsData"),    NULL,     NULL)
gbs$Rob_Sav <- load_genomic_breaks(system.file("extdata/Ciona_pairwise_v3/C_rob_HT__C_sav.gff.gz",           package = "BreakpointsData"),    NULL,     NULL)
gbs$Rob_Oki <- load_genomic_breaks(system.file("extdata/Ciona_pairwise_v3/C_rob_HT__OKI2018_I69_1.0.gff.gz", package = "BreakpointsData"),    NULL,  genomes$OKI2018.I69)

gbs$Dme_Dbu <- load_genomic_breaks(system.file("extdata/Insects_pairwise_v3/Dmel__Dbus_ASM1175060v1.gff.gz",     package = "BreakpointsData"),    NULL,     NULL)
gbs$Dme_Dsu <- load_genomic_breaks(system.file("extdata/Insects_pairwise_v3/Dmel__RU_Dsub_v1.1.gff.gz",          package = "BreakpointsData"),    NULL,     NULL)
gbs$Dme_Dya <- load_genomic_breaks(system.file("extdata/Insects_pairwise_v3/Dmel__Prin_Dyak_Tai18E2_2.0.gff.gz", package = "BreakpointsData"),    NULL,     NULL)
gbs$Dme_Dma <- load_genomic_breaks(system.file("extdata/Insects_pairwise_v3/Dmel__Dmau_ASM438214v1.gff.gz",      package = "BreakpointsData"),    NULL,     NULL)

gbs[Oik_Oik] <-        annotateWithReps(gbs[Oik_Oik], reps)
gbs[Oik_Oik] <- annotateWithTranscripts(gbs[Oik_Oik], transcripts)
gbs <- sapply(gbs, flagAll) |> SimpleList()
```

Unaligned regions
-----------------

Note that the unaligned regions are strandless and unpaired by definition.

```{r unal}
unal  <- BiocParallel::bplapply(gbs, cleanGaps) |> SimpleList()
unal[Oik_Oik] <- BiocParallel::bplapply(unal[Oik_Oik], flagLongShort_, longShort)
unal[Oik_Oik] <- BiocParallel::bplapply(unal[Oik_Oik], annotateWithReps, reps)
unal[Oik_Oik] <- BiocParallel::bplapply(unal[Oik_Oik], annotateWithTranscripts, transcripts)
```

Coalescing alignments
---------------------

### Coalesce

We call these coalesced regions “mapped regions”.  Some of the bases in the
region are aligned and some not (see the “mapped unaligned regions” later).

```{r coalescing_algorithm, cache=TRUE, dependson="GBreaks_object"}
coa <- BiocParallel::bplapply(gbs, coalesce_contigs) |> SimpleList()
coa[Oik_Oik] <- BiocParallel::bplapply(coa[Oik_Oik],  flagLongShort_, longShort)
coa[Oik_Oik] <- annotateWithReps(coa[Oik_Oik], reps)
coa[Oik_Oik] <- annotateWithTranscripts(coa[Oik_Oik], transcripts)
coa <- sapply(coa, flagAll) |> SimpleList()
```

### Flag non-coalesced

Mark the regions that are identical in the aligned and the mapped objects with
a `nonCoa` flag meaning _non coalesced_.

```{r mark_non_coalesced_regions}
for(name in names(coa)) {
  coa[[name]]$nonCoa <- coa[[name]] %in% gbs [[name]]
}

for(name in names(gbs)) {
  gbs[[name]]$nonCoa <- gbs[[name]] %in% coa [[name]]
}
```

Unmapped regions
----------------

### Generate

We can not call them “uncoalesced” because this would also qualify aligned
regions that were not colinear with any other region.

```{r unmap}
unmap  <- BiocParallel::bplapply(coa, cleanGaps) |> SimpleList()
unmap[Oik_Oik] <- BiocParallel::bplapply(unmap[Oik_Oik], flagLongShort_, longShort)
unmap[Oik_Oik] <- BiocParallel::bplapply(unmap[Oik_Oik], annotateWithReps, reps)
unmap[Oik_Oik] <- BiocParallel::bplapply(unmap[Oik_Oik], annotateWithTranscripts, transcripts)
```

### Flag non-coalesced

Mark the regions that are identical in the unaligned and the unmapped objects
the same `nonCoa` flag.  The flag is `TRUE` by definition in all unmapped
objects, but we keep it so that objects have similar structures.

```{r mark_non_coalesced_regions2}
for(name in names(unal)) {
  unal[[name]]$nonCoa  <- unal[[name]]  %in% unmap [[name]]
}

for(name in names(unmap)) {
  unmap[[name]]$nonCoa <- unmap[[name]] %in% unal  [[name]]
}
```

Whole-genome objects
--------------------

Let's have objects that cover the whole genomes (with the possible exception
of the telomeric regions) and contain _isolated alignments_, _breakpoint regions_,
_colinear alignments_ and _bridge regions_.  These are `GRanges` objects, not
`GBreaks`, since they contain unaligned regions.

```{r}
makeWholeGenomeObject <- function(aligned, unaligned) {
  # Isolated alignments are aligned regions not in collinear regions
  iso <- granges(aligned[aligned$nonCoa]) |> plyranges::mutate(type = "isolated alignment")
  # Breakpoint regions are unaligned regions not in collinear regions
  brk <- granges(unaligned[unaligned$nonCoa])|> plyranges::mutate(type = "breakpoint region")
  # Collinear alignments are aligned regions in collinear regions
  col <- granges(aligned[!aligned$nonCoa]) |> plyranges::mutate(type = "collinear alignment")
  # Bridge regions are unaligned regions in collinear regions
  bri <- granges(unaligned[!unaligned$nonCoa]) |> plyranges::mutate(type = "bridge region")
  
  wholeGenome <- c(iso, brk, col, bri) |> sort(ignore.strand = TRUE)
  wholeGenome$type <- factor(wholeGenome$type)
  wholeGenome
}

wgo <- mapply(makeWholeGenomeObject, gbs, unal) |> SimpleList()
```

```{r calc_proportion_of_genome_region_classes}
for_wgo_barplots <- sapply(wgo, \(gr) tapply(width(gr), gr$type, sum)) |>
  as.data.frame() |>
  tibble::as_tibble(rownames = "type") |>
  tidyr::pivot_longer(!type) |>
  dplyr::mutate(distClass = OikScrambling:::compDistClass(name)) |>
  dplyr::mutate(genus = OikScrambling:::compGenus(name))

p1 <- ggplot(for_wgo_barplots |> dplyr::filter(genus == "Oikopleura", distClass == "distant")) +
  aes(name, value, fill = type) +
  ggplot2::coord_flip() +
  geom_bar( stat = "identity") +
  ggtitle("Oki or Kume vs other populations")

p2 <- ggplot(for_wgo_barplots |> dplyr::filter(genus == "Oikopleura", distClass == "intermediate")) +
  aes(name, value, fill = type) +
  ggplot2::coord_flip() +
  geom_bar( stat = "identity")  +
  ggtitle("North Atlantic vs North Pacific")

p3 <- ggplot(for_wgo_barplots |> dplyr::filter(genus == "Oikopleura", distClass == "same_or_sister")) +
  aes(name, value, fill = type) +
  ggplot2::coord_flip() +
  geom_bar( stat = "identity") +
  ggtitle("same population")

p4 <- ggplot(for_wgo_barplots |> dplyr::filter(genus == "Drosophila")) +
  aes(name, value, fill = type) +
  ggplot2::coord_flip() +
  geom_bar( stat = "identity") +
  ggtitle("Drosophila (various evol. distances)")

p5 <- ggplot(for_wgo_barplots |> dplyr::filter(genus == "Ciona")) +
  aes(name, value, fill = type) +
  ggplot2::coord_flip() +
  geom_bar( stat = "identity") +
  ggtitle("Ciona (various evol. distances)")
```

We can see that:

 - The larger size of Okinawa and Kume genomes is explained by sequences that
   are not aligned.  (Repeats ?)
 - The fraction of isolated alignments is always small, but it is comparatively
   larger in Oiks compared to Drosophila and Ciona.
 - The fraction of bridge regions increases with evolutionary distance, until
   becoming larger than the fraction of colinear alignments (_D. melanogaster_ /
   _D. buskii_, _C. savignyi_ / _C intestinalis or robusta_).

```{r proportion_of_genome_region_classes, fig.width=6, fig.height=18, dev=c("svg", "pdf", "png"), fig.ext=c("svg", "pdf", "png")}
library("patchwork")
p1 / p2 / p3 / p4 / p5
```

```{r proportion_of_genome_region_classes_oiks, fig.width=8, fig.height=10, dev=c("svg", "pdf", "png"), fig.ext=c("svg", "pdf", "png")}
for_wgo_barplots2 <- sapply(wgo, \(gr) tapply(width(gr), gr$type, sum)) |>
  prop.table(2) |>
  as.data.frame() |>
  tibble::as_tibble(rownames = "type") |>
  tidyr::pivot_longer(!type) |>
  dplyr::mutate(distClass = OikScrambling:::compDistClass(name)) |>
  dplyr::mutate(genus = OikScrambling:::compGenus(name))

for_wgo_barplots2$type <- for_wgo_barplots2$type |> factor(c("isolated alignment", "bridge region", "collinear alignment", "breakpoint region"))

ggplot(for_wgo_barplots2 |> dplyr::filter(genus == "Oikopleura")) +
  aes(name, value, fill = type) +
  ggplot2::coord_flip() +
  geom_bar( stat = "identity") +
  ggtitle("Proportion of the genome in aligned, bridge and breakpoint regions") +
  facet_grid(distClass ~ ., scales = "free", space = "free") +
  theme_bw() +
  scale_y_continuous(labels = scales::percent_format(accuracy = 5L))
```

Bridge regions
--------------

We can pair the unaligned regions that are between two colinearly aligned regions.

```{r mappedUnaligned, cache = TRUE, dependson="GBreaks_object"}
bri <- BiocParallel::bplapply(gbs, bridgeRegions) |> as("SimpleList")

bri[Oik_Oik] <- BiocParallel::bplapply(bri[Oik_Oik], flagLongShort_, longShort)
bri[Oik_Oik] <- annotateWithReps(bri[Oik_Oik], reps)
bri[Oik_Oik] <- annotateWithTranscripts(bri[Oik_Oik], transcripts)
```

Translocated regions
--------------------

Here we compute new `GBreaks` objects representing the intervals between two
regions that would be colinear if there would not be a translocation separating
them.

```{r translocated_regions, cache=TRUE, dependson="coalescing_algorithm"}
traGaps <- function(coa) {
  # The input MUST already be coalesced.
  tra <- flagTranslocations(coa)
  # Consider both sides
  tra$tra <- tra$tra | tra$query$tra
  # Then, remove translocated regions
  tra <- filterTranslocations(tra, remove = TRUE)
  # Finally, extract the newly "unaligned mapped" regions.
  # They contain the gaps left by the removal of the translocated regions.
  bridgeRegions(tra)
}

tra <- BiocParallel::bplapply(coa, traGaps) |> as("SimpleList")

tra[Oik_Oik] <- BiocParallel::bplapply(tra[Oik_Oik], flagLongShort_, longShort)
tra[Oik_Oik] <- annotateWithReps(tra[Oik_Oik], reps)
tra[Oik_Oik] <- annotateWithTranscripts(tra[Oik_Oik], transcripts)
```

Coalesce again
--------------

We remove the translocated regions that interrupt colinearity, where one of
the colinear pairs is closer than 200 bp.

```{r re_coalesce, cache=TRUE, dependson="coalescing_algorithm"}
removeTranslocationsOnTargetGenome <- function(gb) {
  gb <- flagTranslocations(gb)
  gb <- dist2next(gb, step = 2, ignore.strand = TRUE)
  idx <- 1 + which( gb$tra & (gb$tdist < 200 | gb$qdist < 200) )
  if(length(idx) != 0)
    gb <- gb[-idx]
  coalesce_contigs(gb)
}
removeTranslocationsOnBothGenomes <- function(gb)
  gb |>
    removeTranslocationsOnTargetGenome() |>
    swap() |> sort(i=T) |>
    removeTranslocationsOnTargetGenome() |>
    swap() |> sort(i=T)

coa2 <- BiocParallel::bplapply(coa, removeTranslocationsOnBothGenomes) |> SimpleList()

coa2[Oik_Oik] <- BiocParallel::bplapply(coa2[Oik_Oik], flagLongShort_, longShort)
coa2[Oik_Oik] <- annotateWithReps(coa2[Oik_Oik], reps)
coa2[Oik_Oik] <- annotateWithTranscripts(coa2[Oik_Oik], transcripts)
coa2 <- sapply(coa2, flagAll) |> SimpleList()
```

Unmapped regions 2
------------------

The intervals between double-coalesced regions.

```{r unmap2}
unmap2  <- BiocParallel::bplapply(coa2, cleanGaps) |> SimpleList()
unmap2[Oik_Oik] <- BiocParallel::bplapply(unmap2[Oik_Oik], flagLongShort_, longShort)
unmap2[Oik_Oik] <- BiocParallel::bplapply(unmap2[Oik_Oik], annotateWithReps, reps)
unmap2[Oik_Oik] <- BiocParallel::bplapply(unmap2[Oik_Oik], annotateWithTranscripts, transcripts)
```

Translocated regions 2
----------------------

To check that we really removed the small ones as intented.

```{r translocated_regions2, cache=TRUE, dependson="re_coalesce"}
tra2 <- BiocParallel::bplapply(coa2, traGaps) |> as("SimpleList")

tra2[Oik_Oik] <- BiocParallel::bplapply(tra2[Oik_Oik], flagLongShort_, longShort)
tra2[Oik_Oik] <- annotateWithReps(tra2[Oik_Oik], reps)
tra2[Oik_Oik] <- annotateWithTranscripts(tra2[Oik_Oik], transcripts)
```


Numbers of blocks
-----------------

```{r lengths}
bp_summary <- 
  rbind(
    gbs     = sapply(gbs, length),
    unal    = sapply(unal, length),
    coa     = sapply(coa, length),
    unmap   = sapply(unmap, length),
    unmap2  = sapply(unmap2, length),
    bri = sapply(bri, length),
    tra     = sapply(tra, length),
    tra2    = sapply(tra2, length),
    coa2    = sapply(coa2, length)
  ) |> t() |> data.frame()

sapply(bp_summary, \(x) tapply(x, row.names(bp_summary) |> OikScrambling:::compDistance(), mean))
sapply(bp_summary, \(x) tapply(x, row.names(bp_summary) |> OikScrambling:::compDistance(), median))
sapply(bp_summary, \(x) tapply(x, row.names(bp_summary) |> OikScrambling:::compDistance(), sd))
```

Fraction of the genome aligned or mapped
----------------------------------------

```{r fraction_aligned}
  aligned_tot  <- sapply(gbs,   \(x) sum(width(x)))
  mapped_tot   <- sapply(coa,   \(x) sum(width(x)))
  mapped2_tot  <- sapply(coa2,  \(x) sum(width(x)))
unaligned_tot  <- sapply(unal,   \(x) sum(width(x)))
unmapped_tot   <- sapply(unmap,  \(x) sum(width(x)))
unmapped2_tot  <- sapply(unmap2, \(x) sum(width(x)))

(align_summary <- 100 * cbind(
  align_frac = aligned_tot / ( aligned_tot + unaligned_tot ),
  map_frac   = mapped_tot  / ( mapped_tot  + unmapped_tot ),
  map2_frac  = mapped2_tot / ( mapped2_tot + unmapped2_tot)
 )|> as.data.frame())

sapply(align_summary, \(x) tapply(x, row.names(align_summary) |> OikScrambling:::compDistance(), mean))   |> round(1)
sapply(align_summary, \(x) tapply(x, row.names(align_summary) |> OikScrambling:::compDistance(), median)) |> round(1)
sapply(align_summary, \(x) tapply(x, row.names(align_summary) |> OikScrambling:::compDistance(), sd))     |> round(1)
```

Summary
-------

```{r summary}
df <- cbind(
  gbs = sapply(gbs, length),
  coa = sapply(coa, length),
  coa2 = sapply(coa2, length)
) |> as.data.frame() |>
  dplyr::arrange(gbs = gbs - coa - coa2, coa = coa - coa2)

df$pair <- row.names(df)

ggplot(df |> tidyr::pivot_longer(!pair)) +
  aes(pair, value, fill = name) +
  geom_bar( stat = "identity" ) +
  ggplot2::coord_flip() +
  ggtitle("Cumulative plot (gbs ⊃ coa ⊃ coa2) of the computed genomic regions")
```

NOTE: how does that compare with ANI (Average Nucleotide Identity) ?

```{r}
sim <- c(`Bar2_p4__AOM-5-5f` = 87.489, Bar2_p4__Bar2_p4 = 100, `Bar2_p4__KUM-M3-7f` = 78.45822, 
Bar2_p4__O_alb = 69.79983, Bar2_p4__O_lon = 66.11535, Bar2_p4__O_van = 69.29796, 
Bar2_p4__OdB3 = 97.7499, Bar2_p4__OKI2018_I69_1.0 = 78.68358, 
Bar2_p4__OSKA2016v1.9 = 87.46913, C_int_P__C_int_P = 100, C_int_P__C_int_R = 96.16641, 
C_int_P__C_rob_HT = 90.49159, C_int_P__C_sav = 73.27976, C_int_P__OKI2018_I69_1.0 = 65.14351, 
C_int_P__OSKA2016v1.9 = 64.71592, C_int_P__St_cla = 68.16588, 
C_rob_HT__C_int_P = 90.41795, C_rob_HT__C_int_R = 90.4192, C_rob_HT__C_rob_HT = 100, 
C_rob_HT__C_sav = 73.16812, C_rob_HT__OKI2018_I69_1.0 = 65.53052, 
C_rob_HT__OSKA2016v1.9 = 65.09903, C_rob_HT__St_cla = 68.30626, 
Dalb__AaegL5.0 = 65.6799, Dalb__AalbS3_pri_1.0 = 68.17702, Dalb__Cfrigida_v1.1 = 72.42817, 
`Dalb__Dalb_15112-1751.03v1` = 100, Dalb__Dath_EB_1.0 = 80.60732, 
Dalb__Dbif_1.0 = 81.08233, Dalb__Dbus_ASM1175060v1 = 85.11277, 
Dalb__Dinn_ASM435438v2 = 87.86721, Dalb__Dlow_1.0 = 81.1923, 
Dalb__Dmau_ASM438214v1 = 81.65325, Dalb__Dmel_BDGP6 = 81.14, 
Dalb__Dmir_PacBio2.1 = 81.76273, Dalb__Dpse_MV25 = 81.20186, 
Dalb__Eten_idEriTena2.1 = 70.77767, Dalb__Paka_ASM1839793v1 = 66.22446, 
Dalb__Prin_Dyak_Tai18E2_2.0 = 81.21401, Dalb__RU_Dsub_v1.1 = 82.53274, 
Dalb__RU_Dtri_1.1 = 81.93057, Dath__AaegL5.0 = 66.28782, Dath__AalbS3_pri_1.0 = 69.08501, 
Dath__Cfrigida_v1.1 = 72.40521, `Dath__Dalb_15112-1751.03v1` = 81.90288, 
Dath__Dath_EB_1.0 = 100, Dath__Dbif_1.0 = 90.09103, Dath__Dbus_ASM1175060v1 = 81.89253, 
Dath__Dinn_ASM435438v2 = 82.2395, Dath__Dlow_1.0 = 92.30271, 
Dath__Dmau_ASM438214v1 = 85.84368, Dath__Dmel_BDGP6 = 85.72521, 
Dath__Dmir_PacBio2.1 = 91.84388, Dath__Dpse_MV25 = 91.87636, 
Dath__Eten_idEriTena2.1 = 71.32619, Dath__Paka_ASM1839793v1 = 65.05051, 
Dath__Prin_Dyak_Tai18E2_2.0 = 85.03599, Dath__RU_Dsub_v1.1 = 86.28124, 
Dath__RU_Dtri_1.1 = 86.45957, Dinn__AaegL5.0 = 66.17064, Dinn__AalbS3_pri_1.0 = 67.58017, 
DDinn__Cfrigida_v1.1 = 71.83029, `Dinn__Dalb_15112-1751.03v1` = 87.04083, 
Dinn__Dath_EB_1.0 = 79.57953, Dinn__Dbif_1.0 = 80.19957, Dinn__Dbus_ASM1175060v1 = 84.28436, 
Dinn__Dinn_ASM435438v2 = 100, Dinn__Dlow_1.0 = 80.67069, Dinn__Dmau_ASM438214v1 = 81.0131, 
Dinn__Dmel_BDGP6 = 80.77358, Dinn__Dmir_PacBio2.1 = 80.56467, 
Dinn__Dpse_MV25 = 79.94521, Dinn__Eten_idEriTena2.1 = 69.06343, 
Dinn__Paka_ASM1839793v1 = 66.33035, Dinn__Prin_Dyak_Tai18E2_2.0 = 80.57511, 
Dinn__RU_Dsub_v1.1 = 81.50381, Dinn__RU_Dtri_1.1 = 81.46686, 
Dmel__AaegL5.0 = 66.80427, Dmel__AalbS3_pri_1.0 = 68.20054, Dmel__Cfrigida_v1.1 = 72.50621, 
`Dmel__Dalb_15112-1751.03v1` = 80.26274, Dmel__Dath_EB_1.0 = 83.20431, 
Dmel__Dbif_1.0 = 82.25649, Dmel__Dbus_ASM1175060v1 = 80.08149, 
Dmel__Dinn_ASM435438v2 = 80.79603, Dmel__Dlow_1.0 = 83.56816, 
Dmel__Dmau_ASM438214v1 = 95.36722, Dmel__Dmel_BDGP6 = 100, Dmel__Dmir_PacBio2.1 = 84.10367, 
Dmel__Dpse_MV25 = 83.27086, Dmel__Eten_idEriTena2.1 = 70.18941, 
Dmel__Paka_ASM1839793v1 = 65.26859, Dmel__Prin_Dyak_Tai18E2_2.0 = 91.33685, 
Dmel__RU_Dsub_v1.1 = 88.09201, Dmel__RU_Dtri_1.1 = 86.06675, 
Dpse__AaegL5.0 = 67.05445, Dpse__AalbS3_pri_1.0 = 68.72562, Dpse__Cfrigida_v1.1 = 72.71335, 
`Dpse__Dalb_15112-1751.03v1` = 81.69273, Dpse__Dath_EB_1.0 = 91.50727, 
Dpse__Dbif_1.0 = 90.07924, Dpse__Dbus_ASM1175060v1 = 81.28797, 
Dpse__Dinn_ASM435438v2 = 81.96187, Dpse__Dlow_1.0 = 95.02922, 
Dpse__Dmau_ASM438214v1 = 85.43473, Dpse__Dmel_BDGP6 = 85.32083, 
Dpse__Dmir_PacBio2.1 = 97.38482, Dpse__Dpse_MV25 = 100, Dpse__Eten_idEriTena2.1 = 69.74893, 
Dpse__Paka_ASM1839793v1 = 64.9258, Dpse__Prin_Dyak_Tai18E2_2.0 = 84.58586, 
Dpse__RU_Dsub_v1.1 = 86.0418, Dpse__RU_Dtri_1.1 = 85.93798, Dsub__AaegL5.0 = 66.08828, 
Dsub__AalbS3_pri_1.0 = 68.42965, Dsub__Cfrigida_v1.1 = 71.28905, 
`Dsub__Dalb_15112-1751.03v1` = 80.48788, Dsub__Dath_EB_1.0 = 82.75689, 
Dsub__Dbif_1.0 = 83.14488, Dsub__Dbus_ASM1175060v1 = 80.4126, 
Dsub__Dinn_ASM435438v2 = 81.50499, Dsub__Dlow_1.0 = 84.02854, 
Dsub__Dmau_ASM438214v1 = 88.26478, Dsub__Dmel_BDGP6 = 87.81593, 
Dsub__Dmir_PacBio2.1 = 84.50135, Dsub__Dpse_MV25 = 82.88658, 
Dsub__Eten_idEriTena2.1 = 69.00946, Dsub__Paka_ASM1839793v1 = 64.88739, 
Dsub__Prin_Dyak_Tai18E2_2.0 = 87.79101, Dsub__RU_Dsub_v1.1 = 100, 
Dsub__RU_Dtri_1.1 = 87.23365, Dyak__AaegL5.0 = 65.96956, Dyak__AalbS3_pri_1.0 = 68.30608, 
Dyak__Cfrigida_v1.1 = 72.36378, `Dyak__Dalb_15112-1751.03v1` = 80.31928, 
Dyak__Dath_EB_1.0 = 82.77286, Dyak__Dbif_1.0 = 82.36289, Dyak__Dbus_ASM1175060v1 = 79.76272, 
Dyak__Dinn_ASM435438v2 = 80.83953, Dyak__Dlow_1.0 = 83.68068, 
Dyak__Dmau_ASM438214v1 = 91.8662, Dyak__Dmel_BDGP6 = 91.52635, 
Dyak__Dmir_PacBio2.1 = 84.21573, Dyak__Dpse_MV25 = 83.25226, 
Dyak__Eten_idEriTena2.1 = 70.75961, Dyak__Paka_ASM1839793v1 = 65.18495, 
Dyak__Prin_Dyak_Tai18E2_2.0 = 100, Dyak__RU_Dsub_v1.1 = 88.18683, 
Dyak__RU_Dtri_1.1 = 86.18934, `OKI2018_I69_1.0__AOM-5-5f` = 79.19197, 
OKI2018_I69_1.0__Bar2_p4 = 78.97216, `OKI2018_I69_1.0__KUM-M3-7f` = 96.6692, 
OKI2018_I69_1.0__O_alb = 69.58952, OKI2018_I69_1.0__O_lon = 65.9365, 
OKI2018_I69_1.0__O_van = 69.21153, OKI2018_I69_1.0__OdB3 = 78.80925, 
OKI2018_I69_1.0__OKI2018_I69_1.0 = 100, OKI2018_I69_1.0__OSKA2016v1.9 = 78.82942, 
`OSKA2016v1.9__AOM-5-5f` = 97.14968, OSKA2016v1.9__Bar2_p4 = 87.56705, 
`OSKA2016v1.9__KUM-M3-7f` = 78.52209, OSKA2016v1.9__O_alb = 68.88416, 
OSKA2016v1.9__O_lon = 66.94376, OSKA2016v1.9__O_van = 68.66089, 
OSKA2016v1.9__OdB3 = 87.51462, OSKA2016v1.9__OKI2018_I69_1.0 = 78.52812, 
OSKA2016v1.9__OSKA2016v1.9 = 100)

align_summary$sim <- sim[c(
  "OKI2018_I69_1.0__OSKA2016v1.9",
  "OKI2018_I69_1.0__Bar2_p4",
  "OKI2018_I69_1.0__KUM-M3-7f",
  "OKI2018_I69_1.0__AOM-5-5f",
  "OKI2018_I69_1.0__OdB3",
  "OSKA2016v1.9__OKI2018_I69_1.0",
  "OSKA2016v1.9__Bar2_p4",
  "OSKA2016v1.9__KUM-M3-7f",
  "OSKA2016v1.9__AOM-5-5f",
  "OSKA2016v1.9__OdB3",
  "Bar2_p4__OKI2018_I69_1.0",
  "Bar2_p4__OSKA2016v1.9",
  "Bar2_p4__KUM-M3-7f",
  "Bar2_p4__AOM-5-5f",
  "Bar2_p4__OdB3",
  "C_int_P__C_int_R",
  "C_int_P__C_rob_HT",
  "C_int_P__C_sav",
  "C_int_P__OKI2018_I69_1.0",
  "C_rob_HT__C_int_R",
  "C_rob_HT__C_int_P",
  "C_rob_HT__C_sav",
  "C_rob_HT__OKI2018_I69_1.0",
  "Dmel__Dbus_ASM1175060v1",
  "Dmel__RU_Dsub_v1.1",
  "Dmel__Prin_Dyak_Tai18E2_2.0",
  "Dmel__Dmau_ASM438214v1"
)]

align_summary["Hum_Mou",] <- c(10.3264, 54.39799, NA, 75.20007)

align_summary$pairname <- rownames(align_summary)
align_summary$dist <- OikScrambling:::compDistance(align_summary$pairname)
align_summary$genus <- "Oikopleura"
align_summary$genus[grepl("Int", align_summary$dist)] <- "Ciona"
align_summary$genus[grepl("Dme", align_summary$dist)] <- "Drosophila"
align_summary$genus[grepl("Hum", align_summary$dist)] <- "Mammalian"

df <- tidyr::pivot_longer(align_summary, c("align_frac", "map_frac", "map2_frac", "sim"))
df <- df[!grepl("– Oki", df$dist),]

library(ggplot2)

ggplot(df) +
  aes(value, name, col= genus) +
  geom_point() +
  facet_wrap(~name, ncol = 1, scale = "free") +
  theme_bw()

align_summary |>
  dplyr::filter(!grepl("– Oki", dist)) |>
  ggplot() +
  aes(align_frac, sim, col= genus) +
  geom_point(size=10) +
  theme_bw()

align_summary |>
  dplyr::filter(!grepl("– Oki", dist)) |>
  ggplot() +
  aes(map_frac, sim, col= genus) +
  geom_point() +
  theme_bw()
```

```{r breakpoints_per_Mb}
genome_sizes=list(Aom=sum(seqlengths(genomes$AOM.5.5f)), Bar=sum(seqlengths(genomes$Bar2.p4)),
                  Oki=sum(seqlengths(genomes$OKI2018.I69)), Osa=sum(seqlengths(genomes$OSKA2016v1.9)),
                  Kum=sum(seqlengths(genomes$KUM.M3.7f)), Nor=sum(seqlengths(genomes$OdB3)),
                  Ply=128639392, Ros=134327879, Sav=177003750, Rob=117529544,
                  Dme=137547960, Dbu=110837441, Dsu=130045071, Dya=136883712
                  ) |> unlist()

genome_sizes[names(coa) |> sub(pat = "_.*", rep = "")]
dens <- tibble::tibble(
  nbreak   = sapply(coa, length),
  nbreak2  = sapply(coa2, length),
  nbreak3  = sapply(wgo, \(gr) sum(gr$type == "breakpoint region")),
  gsize    = genome_sizes[names(coa) |> sub(pat = "_.*", rep = "")],
  density  = nbreak  / gsize * 1e6,
  density2 = nbreak2 / gsize * 1e6,
  density3 = nbreak3 / gsize * 1e6,
  pairname = names(coa),
  dist     = OikScrambling:::compDistance(pairname),
  genus    = dist |>
              sub(pat = "Dme_.*", rep="Drosophila") |>
              sub(pat = "Int.*",  rep="Ciona") |>
              sub(pat = ".*[_–].*",  rep="Oikopleura")
    
)

set.seed(1664)
ggplot(dens) +
  aes(density, genus, col = dist, pch = genus) +
  geom_point(size = 5) +
  scale_x_continuous("Density (breakpoints / megabase)") +
  ggtitle("Density of breakpoints in paired comparisons, arranged per genus") +
  scale_colour_manual(values=rainbow(11, s=.6, v=.9)[sample(1:11,11)])

set.seed(1664)
ggplot(dens) +
  aes(density2, genus, col = dist, pch = genus) +
  geom_point(size = 5) +
  scale_x_continuous("Density (breakpoints / megabase)") +
  ggtitle("Density of breakpoints (double-coalesced) in paired comparisons, arranged per genus") +
  scale_colour_manual(values=rainbow(11, s=.6, v=.9)[sample(1:11,11)])

set.seed(1664)
ggplot(dens) +
  aes(density3, genus, col = dist, pch = genus) +
  geom_point(size = 5) +
  scale_x_continuous("Frequency (breakpoints / megabase)") +
  ggtitle("Frequency of breakpoints regions in paired comparisons, arranged per genus") +
  scale_colour_manual(values=rainbow(11, s=.6, v=.9)[sample(1:11,11)])
```

Save data
=========

```{r save_data}
save(gbs, unal, coa, unmap, bri, tra, tra2, coa2, unmap2, wgo, longShort, file = "BreakPoints.Rdata")
system("echo '9f273f099ecbfa44899b6c343d1382f1  BreakPoints.Rdata' | md5sum -c", intern = TRUE)
```

Session information
===================

```{r session_info}
sessionInfo()
```
