---
title: "Orthogroup syntenies"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteEncoding{UTF-8}
  %\VignetteIndexEntry{Orthogroup syntenies}
  %\VignetteEngine{knitr::rmarkdown}
editor_options: 
  chunk_output_type: console
---

Load packages and data
======================

R packages
----------

Install the _GenomicBreaks_ package with:

```
devtools::install_github("oist/GenomicBreaks")
```

```{r load_packages}
suppressPackageStartupMessages({
  library('BreakpointsData')
  library('GenomicBreaks')
  library('BSgenome')
  library('GenomicFeatures')
})
```

BSgenome packages
-----------------

Example command to install one _Oikopleura_ BSgenome package if you do not yet
have it:

```
install.packages("BSgenome.Odioica.local.OKI2018.I69", repos="https://oist.github.io/plessy_oikgenomes_drat/")
```

```{r load_BSgenome_packages}
library("BSgenome.Odioica.local.OKI2018.I69")
library("BSgenome.Odioica.local.OSKA2016v1.9")
library("BSgenome.Odioica.local.Bar2.p4")
library("BSgenome.Odioica.local.KUM.M3")
library("BSgenome.Odioica.local.AOM.5")
library("BSgenome.Odioica.local.Odioica.reference.v3.0")

genomes <- SimpleList(
  # Chromosome assemblies
  Oki = OKI2018_I69, Osa = OSKA2016v1.9, Bar = Bar2_p4,
  # Less contiguous assemblies
  Kum = KUM_M3, Aom = AOM_5, Nor = OdB3)

## Genome lengths
sapply(genomes, \(g) sum(seqlengths(seqinfo(g))))
```

Annotations
-----------

The annotations from `/bucket/LuscombeU/common/Breakpoints/Annotations` are
distributed in the same drat repository as the alignment data.

```
install.packages("BreakpointsData", repos="https://oist.github.io/plessy_oikgenomes_drat/")
```

```{r load_annotations, warning=FALSE}
annots <- SimpleList()

gff2txdb <- function(file, genome) {
  file <- system.file(paste0("extdata/Annotations/", file), package = "BreakpointsData")
  tx <- rtracklayer::import.gff(file)
  tx <- tx[seqnames(tx) %in% seqnames(genome)]
  tx <- GRanges(tx, seqinfo = seqinfo(genome))
  tx <- GenomicFeatures::makeTxDbFromGRanges(tx)
}

gff2txdb_Norway <- function(file, genome) {
  file <- system.file(paste0("extdata/Annotations/", file), package = "BreakpointsData")
  tx <- rtracklayer::import.gff(file)
  tx <- tx[!is.na(tx$mRNA)]
  # Remove "scaffoldA" objects in the OdB3 annotation
  tx <- tx[seqnames(tx) %in% seqnames(genome)]
  tx$ID <- tx$mRNA
  tx <- GRanges(tx, seqinfo = seqinfo(genome))
  tx <- GenomicFeatures::makeTxDbFromGRanges(tx)
}

annots$Oki <- gff2txdb("OKI2018_I69.v2/OKI2018_I69.v2.gm.gff.gz",  OKI2018_I69)
annots$Osa <- gff2txdb("OSKA2016v1.9/OSKA2016v1.9.gm.gff.gz",      OSKA2016v1.9)
annots$Bar <- gff2txdb("Bar2_p4.Flye/Bar2_p4.Flye.gm.gff.gz",      Bar2_p4)
annots$Kum <- gff2txdb("KUM-M3-7f/KUM-M3-7f.gm.gff.gz",            KUM_M3)
annots$Aom <- gff2txdb("AOM-5-5f/AOM-5-5f.gm.gff.gz",              AOM_5)
annots$Nor <- gff2txdb_Norway("OdB3/Oikopleura_annot_v1.0.gff.gz", OdB3)
```

Orthogroups
-----------

Load Orthogroups for _Oikopleura_ from the `BreakpointsData` package with the
`GenomicBreaks::load_one_to_ones()` function.  Original file is in
`/bucket/LuscombeU/common/Breakpoints/Orthologues/selectedTunicates+Amph+vertebrates_lp_100clstr_blast/OrthoFinder/Results_Sep07/Phylogenetic_Hierarchical_Orthogroups/N6.tsv`.

```{r load_orthogroups}
# Example for loading all orthogroups:
load_one_to_ones(system.file("extdata/OrthoFinder/N6.tsv.gz", package = "BreakpointsData"))

# Example for loading a pair
load_one_to_ones( system.file("extdata/OrthoFinder/N6.tsv.gz", package = "BreakpointsData")
                , c("Bar2_p4.Flye.prot.longest.fa_1", "OSKA2016v1.9.prot.longest.fa_1"))
```


Prepare GBreak objects
======================

```{r Bar_Osk}
orthoPairToGBreaks <- function(name1, name2, annot1, annot2) {
  orthoPairs <- load_one_to_ones(
    system.file("extdata/OrthoFinder/N6.tsv.gz", package = "BreakpointsData"),
    c(name1, name2))

  IDs2GRanges <- function (IDs, annot) {
    prefix <- Biobase::lcPrefix(IDs)                  # Guess prefix in transcript IDs from HOG files
    not_prefix <- Biobase::lcPrefix(annot$tx_name)    # Remove trailing characters that are part of the name
    not_prefix <- paste0(not_prefix,"$")              # Anchor to end of the string
    prefix <- sub(not_prefix, "", prefix)             # Finalise prefix
    IDs <- sub(prefix, "", IDs)                       # Remove prefix from IDS
    names(annot) <- annot$tx_name                     # Store transcript name in names slot
    gr <- granges(annot[IDs])                         # Discard extra metadata and sort by ID
    strand(gr) <- "*"                                 # Make strandless
    gr                                                # Return the object
  }
  
  gb       <- IDs2GRanges(orthoPairs[,name1], transcripts(annot1))
  gb$query <- IDs2GRanges(orthoPairs[,name2], transcripts(annot2))
  gb <- GenomicBreaks:::GBreaks(gb)
  sort(gb)
}

orthoPairs <- SimpleList()
orthoPairs$Oki_Osa <- orthoPairToGBreaks("OKI2018_I69.v2.prot.longest.fa_1", "OSKA2016v1.9.prot.longest.fa_1"
                                        , annots$Oki, annots$Osa)
orthoPairs$Oki_Bar <- orthoPairToGBreaks("OKI2018_I69.v2.prot.longest.fa_1", "Bar2_p4.Flye.prot.longest.fa_1"
                                        , annots$Oki, annots$Bar)
orthoPairs$Oki_Kum <- orthoPairToGBreaks("OKI2018_I69.v2.prot.longest.fa_1", "KUM-M3-7f.prot.longest.fa_1"
                                        , annots$Oki, annots$Kum)
orthoPairs$Oki_Aom <- orthoPairToGBreaks("OKI2018_I69.v2.prot.longest.fa_1", "AOM-5-5f.prot.longest.fa_1"
                                        , annots$Oki, annots$Aom)
## Problem: some Norway transcripts come from alternative "scaffoldA" sequences that are not in the genome file.
# orthoPairs$Bar_Nor <- orthoPairToGBreaks("Bar2_p4.Flye.prot.longest.fa_1", "OdB3.v1.0.prot.fa_1"
#                                         , annots$Bar, annots$Nor)

orthoPairs$Osa_Bar <- orthoPairToGBreaks("OSKA2016v1.9.prot.longest.fa_1", "Bar2_p4.Flye.prot.longest.fa_1"
                                        , annots$Osa, annots$Bar)
orthoPairs$Osa_Oki <- orthoPairToGBreaks("OSKA2016v1.9.prot.longest.fa_1", "OKI2018_I69.v2.prot.longest.fa_1"
                                        , annots$Osa, annots$Oki)
orthoPairs$Osa_Kum <- orthoPairToGBreaks("OSKA2016v1.9.prot.longest.fa_1", "KUM-M3-7f.prot.longest.fa_1"
                                        , annots$Osa, annots$Kum)
orthoPairs$Osa_Aom <- orthoPairToGBreaks("OSKA2016v1.9.prot.longest.fa_1", "AOM-5-5f.prot.longest.fa_1"
                                        , annots$Osa, annots$Aom)
## Problem: some Norway transcripts come from alternative "scaffoldA" sequences that are not in the genome file.
# orthoPairs$Bar_Nor <- orthoPairToGBreaks("Bar2_p4.Flye.prot.longest.fa_1", "OdB3.v1.0.prot.fa_1"
#                                         , annots$Bar, annots$Nor)


orthoPairs$Bar_Osa <- orthoPairToGBreaks("Bar2_p4.Flye.prot.longest.fa_1", "OSKA2016v1.9.prot.longest.fa_1"
                                        , annots$Bar, annots$Osa)
orthoPairs$Bar_Oki <- orthoPairToGBreaks("Bar2_p4.Flye.prot.longest.fa_1", "OKI2018_I69.v2.prot.longest.fa_1"
                                        , annots$Bar, annots$Oki)
orthoPairs$Bar_Kum <- orthoPairToGBreaks("Bar2_p4.Flye.prot.longest.fa_1", "KUM-M3-7f.prot.longest.fa_1"
                                        , annots$Bar, annots$Kum)
orthoPairs$Bar_Aom <- orthoPairToGBreaks("Bar2_p4.Flye.prot.longest.fa_1", "AOM-5-5f.prot.longest.fa_1"
                                        , annots$Bar, annots$Aom)
## Problem: some Norway transcripts come from alternative "scaffoldA" sequences that are not in the genome file.
# orthoPairs$Bar_Nor <- orthoPairToGBreaks("Bar2_p4.Flye.prot.longest.fa_1", "OdB3.v1.0.prot.fa_1"
#                                         , annots$Bar, annots$Nor)

sapply(orthoPairs, length)
```

Problem: some annotations overlap
---------------------------------

```{r fix_Bar_Osk}
transcripts(annots$Bar) |> length()
transcripts(annots$Bar) |> reduce(min = 0) |> length()

flagOverlaps <- function(gr) {
  # Find overlaps
  ov <- findOverlaps(gr)
  # Index of self hits in the gr object
  idx <- queryHits(ov[!isSelfHit(ov)])
  # Convert to Boolean indexing the gr object
  seq_along(gr) %in% idx
}

transcripts(annots$Bar)[flagOverlaps(transcripts(annots$Bar))]

# Show examples
gb[flagOverlaps(gb)]
gb$query[flagOverlaps(gb$query)]

# Remove
gb <- gb[!flagOverlaps(gb)]
gb <- gb[!flagOverlaps(gb$query)]
```

Coalesce syntenic blocks and annotate them
------------------------------------------

```{r coalesce}
# Coalesce
coal <- coalesce_contigs(gb)

# Map annotations to the coalesced blocks
ov <- findOverlaps(coal, gb)

# Extract gene names mapped to each blocks
geneNames <- split(names(gb)[subjectHits(ov)], queryHits(ov))

# Concatenate them as single strings and graft them to the coal object
coal$geneNames <- sapply(geneNames, paste, collapse = ", ")

# Count genes per syntenic block
coal$geneNumber <- sapply(geneNames, length)
```

Quick overview of the results
-----------------------------

```{r quick_overview}
stem(coal$geneNumber)
table(coal$geneNumber)
coal[order(coal$geneNumber)] |> tail(11)
```

Gene Order Conservation is smaller in short arms
------------------------------------------------

```{r GOC}
load("BreakPoints.Rdata")
gb <- flagLongShort(gb, longShort$Bar)
gb <- gb[!is.na(gb$Arm)]
gbl <- split(gb, paste(seqnames(gb), gb$Arm))
sapply(gbl, GOC)
t.test(c(0.9165414, 0.9278024, 0.9370166), c(0.5928854, 0.7259159, 0.7618384))
```

Agreement with nucleotide alignments
------------------------------------

```{r DNA_prot_comparison}
(gb_P <- coal)
sum(width(gb_P))
(gb_N <- coa$Bar_Osa)
sum(width(gb_N))
(ov <- subsetByOverlaps(gb_P, gb_N, type = "within"))
(jaccard <- sum(width(ov)) / sum(width(reduce(c(gb_P, gb_N)))))

gb_P <- gb_P[gb_P$geneNumber > 1]
gb_P
sum(gb_P$geneNumber)
(valid <- subsetByOverlaps(gb_P, gb_N, type = "within"))
sum(valid$geneNumber)
sum(width(valid))
```
