---
title: "Orthogroup syntenies"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteEncoding{UTF-8}
  %\VignetteIndexEntry{Orthogroup syntenies}
  %\VignetteEngine{knitr::rmarkdown}
editor_options: 
  chunk_output_type: console
---

Load packages and data
======================

See `?OikScrambling:::loadAllGenomes`, `?OikScrambling:::loadAllTranscriptsGR`,
and `vignette("LoadGenomicBreaks", package = "OikScrambling")` for how the
different objects are prepared.

```{r load_packages_and_data}
library('OikScrambling')   |> suppressPackageStartupMessages()
genomes <- OikScrambling:::loadAllGenomes(compat = F)
transcripts <- OikScrambling:::loadAllTranscriptsGR(compat = F) |> suppressWarnings()
load("BreakPoints.Rdata")
```

Orthogroups
-----------

Load Orthogroups for _Oikopleura_ from the `BreakpointsData` package with the
`OikScrambling::load_one_to_ones()` function.  Original file is in
`/bucket/LuscombeU/common/Breakpoints/Orthologues/selTun+Apps+Amph+Vert_lp_100clstr_blast/OrthoFinder/Results_Oct07/Phylogenetic_Hierarchical_Orthogroups/N19.tsv`.
See `news(package = "BreakpointsData")` for inspecting changes in more details.

```{r load_orthogroups}
# Example for loading all orthogroups:
OikScrambling:::load_one_to_ones(system.file("extdata/OrthoFinder/N19.tsv", package = "BreakpointsData"))

# Example for loading a pair
OikScrambling:::load_one_to_ones( system.file("extdata/OrthoFinder/N19.tsv", package = "BreakpointsData")
                , c("Bar2_p4.Flye.prot.longest.fa_1", "OSKA2016v1.9.prot.longest.fa_1"))
```


Prepare GBreak objects
======================

```{r orthoPairs_object}
orthoPairToGBreaks <- function(name1, name2, annot1, annot2, treeName="N19", HOGs=NULL) {
  if(is.null(HOGs))
    HOGs <- OikScrambling:::load_one_to_ones(
      system.file(paste0("extdata/OrthoFinder/",treeName,".tsv"), package = "BreakpointsData"),
      c(name1, name2))

  IDs2GRanges <- function (IDs, annot) {
    prefix <- Biobase::lcPrefix(IDs)                  # Guess prefix in transcript IDs from HOG files
    not_prefix <- Biobase::lcPrefix(annot$tx_name)    # Remove trailing characters that are part of the name
    not_prefix <- paste0(not_prefix,"$")              # Anchor to end of the string
    prefix <- sub(not_prefix, "", prefix)             # Finalise prefix
    IDs <- sub(prefix, "", IDs)                       # Remove prefix from IDS
    names(annot) <- annot$tx_name                     # Store transcript name in names slot
    gr <- annot[IDs]                                  # Sort by ID
    strand(gr) <- "*"                                 # Make strandless
    gr                                                # Return the object
  }
  
  gb       <- IDs2GRanges(HOGs[,name1], annot1)
  gb$query <- IDs2GRanges(HOGs[,name2], annot2)
  gb <- GenomicBreaks:::GBreaks(gb)
  gb$HOG <- HOGs$HOG
  gb$OG  <- HOGs$OG
  sort(gb)
}

flagLongShort_ <- function(gr, annot) {
  genome <- gr |> genome() |> unique()
  flagLongShort(gr, annot[[genome]])
}

orthoPairToGBreaks_all_Oiks <- function(treeName=NULL, HOGs=NULL) {
  orthoPairs <- SimpleList()
  orthoPairs$Oki_Osa <- orthoPairToGBreaks("OKI2018_I69.v2.prot.longest.fa_1", "OSKA2016v1.9.prot.longest.fa_1",   transcripts$OKI2018.I69, transcripts$OSKA2016v1.9, treeName, HOGs)
  orthoPairs$Oki_Bar <- orthoPairToGBreaks("OKI2018_I69.v2.prot.longest.fa_1", "Bar2_p4.Flye.prot.longest.fa_1",   transcripts$OKI2018.I69, transcripts$Bar2.p4, treeName, HOGs)
  orthoPairs$Oki_Kum <- orthoPairToGBreaks("OKI2018_I69.v2.prot.longest.fa_1", "KUM-M3-7f.prot.longest.fa_1",      transcripts$OKI2018.I69, transcripts$KUM.M3.7f, treeName, HOGs)
  orthoPairs$Oki_Aom <- orthoPairToGBreaks("OKI2018_I69.v2.prot.longest.fa_1", "AOM-5-5f.prot.longest.fa_1",       transcripts$OKI2018.I69, transcripts$AOM.5.5f, treeName, HOGs)
  orthoPairs$Oki_Nor <- orthoPairToGBreaks("OKI2018_I69.v2.prot.longest.fa_1", "OdB3.v1.0.prot.fa_1.nohaplo",      transcripts$OKI2018.I69, transcripts$OdB3, treeName, HOGs)
  
  orthoPairs$Osa_Bar <- orthoPairToGBreaks("OSKA2016v1.9.prot.longest.fa_1",   "Bar2_p4.Flye.prot.longest.fa_1",   transcripts$OSKA2016v1.9, transcripts$Bar2.p4, treeName, HOGs)
  orthoPairs$Osa_Oki <- orthoPairToGBreaks("OSKA2016v1.9.prot.longest.fa_1",   "OKI2018_I69.v2.prot.longest.fa_1", transcripts$OSKA2016v1.9, transcripts$OKI2018.I69, treeName, HOGs)
  orthoPairs$Osa_Kum <- orthoPairToGBreaks("OSKA2016v1.9.prot.longest.fa_1",   "KUM-M3-7f.prot.longest.fa_1",      transcripts$OSKA2016v1.9, transcripts$KUM.M3.7f, treeName, HOGs)
  orthoPairs$Osa_Aom <- orthoPairToGBreaks("OSKA2016v1.9.prot.longest.fa_1",   "AOM-5-5f.prot.longest.fa_1",       transcripts$OSKA2016v1.9, transcripts$AOM.5.5f, treeName, HOGs)
  orthoPairs$Osa_Nor <- orthoPairToGBreaks("OSKA2016v1.9.prot.longest.fa_1",   "OdB3.v1.0.prot.fa_1.nohaplo",      transcripts$OSKA2016v1.9, transcripts$OdB3, treeName, HOGs)
  
  orthoPairs$Bar_Osa <- orthoPairToGBreaks("Bar2_p4.Flye.prot.longest.fa_1",   "OSKA2016v1.9.prot.longest.fa_1",   transcripts$Bar2.p4, transcripts$OSKA2016v1.9, treeName, HOGs)
  orthoPairs$Bar_Oki <- orthoPairToGBreaks("Bar2_p4.Flye.prot.longest.fa_1",   "OKI2018_I69.v2.prot.longest.fa_1", transcripts$Bar2.p4, transcripts$OKI2018.I69, treeName, HOGs)
  orthoPairs$Bar_Kum <- orthoPairToGBreaks("Bar2_p4.Flye.prot.longest.fa_1",   "KUM-M3-7f.prot.longest.fa_1",      transcripts$Bar2.p4, transcripts$KUM.M3.7f, treeName, HOGs)
  orthoPairs$Bar_Aom <- orthoPairToGBreaks("Bar2_p4.Flye.prot.longest.fa_1",   "AOM-5-5f.prot.longest.fa_1",       transcripts$Bar2.p4, transcripts$AOM.5.5f, treeName, HOGs)
  orthoPairs$Bar_Nor <- orthoPairToGBreaks("Bar2_p4.Flye.prot.longest.fa_1",   "OdB3.v1.0.prot.fa_1.nohaplo",      transcripts$Bar2.p4, transcripts$OdB3, treeName, HOGs)
  
  orthoPairs <- sapply(orthoPairs,  flagLongShort_, longShort)
  
  orthoPairs
}

orthoPairs         <- orthoPairToGBreaks_all_Oiks("N19")
orthoPairs$Ply_Ros <- orthoPairToGBreaks("C_int_P.prot.longest.fa_1", "C_int_R.prot.longest.fa_1"
                                        , transcripts$Ply, transcripts$Ros, treeName="N20")

# Same as above but using the same core orthogroups for all species.
orthoPairs_core         <- orthoPairToGBreaks_all_Oiks("N3")
orthoPairs_core$Ply_Ros <- orthoPairToGBreaks("C_int_P.prot.longest.fa_1", "C_int_R.prot.longest.fa_1"
                                             , transcripts$Ply, transcripts$Ros, treeName="N3")

orthoPairs_one2oneInOiks <-
  orthoPairToGBreaks_all_Oiks(treeName = NULL,
                              HOGs = OikScrambling:::load_one_to_ones(
                                       system.file("extdata/OrthoFinder/N19.tsv", package = "BreakpointsData")))


sapply(orthoPairs, length)
sapply(orthoPairs_core, length)
sapply(orthoPairs_one2oneInOiks, length)
```

Problem: some annotations overlap
---------------------------------

```{r fix_Bar_Osk}
transcripts$Bar2.p4 |> length()
transcripts$Bar2.p4 |> reduce(min = 0) |> length()

flagOverlaps <- function(gr) {
  # Find overlaps
  ov <- findOverlaps(gr)
  # Index of self hits in the gr object
  idx <- queryHits(ov[!isSelfHit(ov)])
  # Convert to Boolean indexing the gr object
  seq_along(gr) %in% idx
}

transcripts$Bar2.p4[flagOverlaps(transcripts$Bar2.p4)]

# Show examples
orthoPairs$Oki_Osa[flagOverlaps(orthoPairs$Oki_Osa)]
orthoPairs$Oki_Osa$query[flagOverlaps(orthoPairs$Oki_Osa$query)]

# Remove and show how much was removed

orthoPairsFiltered <- sapply(orthoPairs, function(gb) {
  gb <- gb[!flagOverlaps(gb)]
  gb <- gb[!flagOverlaps(gb$query)]
  gb
}) |> SimpleList()

orthoPairsFiltered_core <- sapply(orthoPairs_core, function(gb) {
  gb <- gb[!flagOverlaps(gb)]
  gb <- gb[!flagOverlaps(gb$query)]
  gb
}) |> SimpleList()

sapply(orthoPairs, length) - sapply(orthoPairsFiltered, length)
sapply(orthoPairs_core, length) - sapply(orthoPairsFiltered_core, length)
```


Coalesce syntenic blocks and annotate them
------------------------------------------

```{r coalesce}
coalOrtho <- function(gb) {
  # Coalesce
  coal <- coalesce_contigs(gb)
  
  # Map annotations to the coalesced blocks
  ov <- findOverlaps(coal, gb)
  
  # Extract gene names mapped to each blocks
  coal$geneNames <- CharacterList(split(names(gb)[subjectHits(ov)], queryHits(ov))) |> unname()
  # We can use the `ov` object for query ranges as well because of the way `coal` was constructed.
  coal$query$geneNames <- CharacterList(split(names(gb$query)[subjectHits(ov)], queryHits(ov))) |> unname()
  
  # Extract dNdS values
  coal$dNdS_GUIDANCE2  <- NumericList(split(gb$dNdS_GUIDANCE2 [subjectHits(ov)], queryHits(ov)))
  coal$dNdS_HmmCleaner <- NumericList(split(gb$dNdS_HmmCleaner[subjectHits(ov)], queryHits(ov)))
  coal$dNdS_PRANK      <- NumericList(split(gb$dNdS_PRANK     [subjectHits(ov)], queryHits(ov)))
  # We just copy dNdS values to query ranges because they are the same.
  coal$query$dNdS_GUIDANCE2  <- coal$dNdS_GUIDANCE2
  coal$query$dNdS_HmmCleaner <- coal$dNdS_HmmCleaner
  coal$query$dNdS_PRANK      <- coal$dNdS_PRANK

  # Count genes per syntenic block
  coal$geneNumber <- sapply(coal$geneNames, length)
  
  coal
}

orthoBlocks <- sapply(orthoPairsFiltered[1:15], coalOrtho) |> SimpleList()
orthoBlocks$Ply_Ros <- coalesce_contigs(orthoPairsFiltered$Ply_Ros)
orthoBlocks[ 1:15 ] <- sapply(orthoBlocks[ 1:15 ],  flagLongShort_, longShort)

orthoBlocks_core <- sapply(orthoPairsFiltered_core[1:15], coalOrtho) |> SimpleList()
orthoBlocks_core$Ply_Ros <- coalesce_contigs(orthoPairsFiltered_core$Ply_Ros)
orthoBlocks_core[ 1:15 ] <- sapply(orthoBlocks_core[ 1:15 ],  flagLongShort_, longShort)

sapply(orthoBlocks, length)
sapply(orthoBlocks_core, length)

```


Quick overview of the results
=============================

Size distribution of the blocks
-------------------------------

```{r quick_overview}
stem(orthoBlocks$Oki_Osa$geneNumber)
table(orthoBlocks$Oki_Osa$geneNumber)
orthoBlocks$Oki_Osa[order(orthoBlocks$Oki_Osa$geneNumber)] |> tail(11)

stem(orthoBlocks$Oki_Kum$geneNumber)
table(orthoBlocks$Oki_Kum$geneNumber)
orthoBlocks$Oki_Kum[order(orthoBlocks$Oki_Kum$geneNumber)] |> tail(11)
```

Gene Order Conservation is smaller in short arms
------------------------------------------------

```{r GOC}
orthoBlocks_LSnoNA <- sapply(orthoBlocks[1:15], function(gb) gb[!is.na(gb$Arm)]) |> SimpleList()
sapply(orthoBlocks_LSnoNA, length)

df <- sapply(orthoBlocks_LSnoNA,
             function(gb) tapply(gb, paste(seqnames(gb), gb$Arm), GOC))

apply(df, 2, \(x) {
  tst <- t.test(x[c(1,3,5)], x[c(2,4,6)])
  c(pval=tst$p.value, mx=tst$estimate[1], my=tst$estimate[2])
}) 
```

Fraction of the genome covered or syntenic
------------------------------------------

```{r fraction_aligned}
uncov  <- BiocParallel::bplapply(orthoPairsFiltered, cleanGaps) |> SimpleList()
unsyn  <- BiocParallel::bplapply(orthoBlocks,        cleanGaps) |> SimpleList()

  covered_tot    <- sapply(orthoPairsFiltered,   \(x) sum(width(x)))
  syntenic_tot   <- sapply(orthoBlocks,          \(x) sum(width(x)))
uncovered_tot    <- sapply(uncov,                \(x) sum(width(x)))
unsyntenic_tot   <- sapply(unsyn,                \(x) sum(width(x)))

(syn_summary <- cbind(
  covered_frac   = covered_tot   / ( covered_tot   + uncovered_tot ) * 100,
  orthogenes_n   = sapply(orthoPairsFiltered, length),
  syntenic_frac  = syntenic_tot  / ( syntenic_tot  + unsyntenic_tot ) * 100,
  syntenic_reg   = sapply(orthoBlocks, length)
 )|> as.data.frame())

sapply(syn_summary, \(x) tapply(x, row.names(syn_summary) |> OikScrambling:::compDistance(), mean))   |> round(1)
sapply(syn_summary, \(x) tapply(x, row.names(syn_summary) |> OikScrambling:::compDistance(), median)) |> round(1)
sapply(syn_summary, \(x) tapply(x, row.names(syn_summary) |> OikScrambling:::compDistance(), sd))     |> round(1)
```

Unfortunately, there are not enough transcripts from Ciona that are part of the
pan-tunicate orthogroup set, which makes the comparison difficult between
Oikopleura and Ciona…

```{r fraction_aligned_core}
uncov  <- BiocParallel::bplapply(orthoPairsFiltered_core, cleanGaps) |> SimpleList()
unsyn  <- BiocParallel::bplapply(orthoBlocks_core,        cleanGaps) |> SimpleList()

  covered_tot    <- sapply(orthoPairsFiltered_core,   \(x) sum(width(x)))
  syntenic_tot   <- sapply(orthoBlocks_core,          \(x) sum(width(x)))
uncovered_tot    <- sapply(uncov,                     \(x) sum(width(x)))
unsyntenic_tot   <- sapply(unsyn,                     \(x) sum(width(x)))

(syn_summary_core <- cbind(
  covered_frac   = covered_tot   / ( covered_tot   + uncovered_tot ) * 100,
  orthogenes_n   = sapply(orthoPairsFiltered_core, length),
  syntenic_frac  = syntenic_tot  / ( syntenic_tot  + unsyntenic_tot ) * 100,
  syntenic_reg   = sapply(orthoBlocks_core, length)
 )|> as.data.frame())

sapply(syn_summary_core, \(x) tapply(x, row.names(syn_summary_core) |> OikScrambling:::compDistance(), mean))   |> round(1)
sapply(syn_summary_core, \(x) tapply(x, row.names(syn_summary_core) |> OikScrambling:::compDistance(), median)) |> round(1)
sapply(syn_summary_core, \(x) tapply(x, row.names(syn_summary_core) |> OikScrambling:::compDistance(), sd))     |> round(1)
```

Agreement with nucleotide alignments
------------------------------------

```{r DNA_prot_comparison}
(gb_P <- orthoBlocks$Bar_Osa)
sum(width(gb_P))
(gb_N <- coa$Bar_Osa)
sum(width(gb_N))
(ov <- subsetByOverlaps(gb_P, gb_N, type = "within"))
(jaccard <- sum(width(ov)) / sum(width(reduce(c(gb_P, gb_N)))))

gb_P <- gb_P[gb_P$geneNumber > 1]
gb_P
sum(gb_P$geneNumber)
(valid <- subsetByOverlaps(gb_P, gb_N, type = "within"))
sum(valid$geneNumber)
sum(width(valid))
```

Question: what does interrupt the colinearity of nucleotide regions in gene-centric
colinear regions ?

XSR plot
--------

Ugly...

```{r XSR_plot}
XSR_HOGs <- sapply(orthoPairs_one2oneInOiks, \(gb) {
  gb <- gb[seqnames(gb) == "XSR"]
  # Keep biggest match only
  chrQ <- tapply(width(gb$query), seqnames(gb$query), sum) |> sort() |> tail(1) |> names()
  gb <- gb[seqnames(gb$query) == chrQ]
  seqlevels(gb) <- seqlevelsInUse(gb)
  seqlevels(gb$query) <- seqlevelsInUse(gb$query)
  gb
})

removeSeqInfoGR <- function(gr) 
  GRanges(seqnames = seqnames(gr), ranges = ranges(gr), strand = strand(gr))

removeSeqInfoGB <- function(gb)
  GBreaks(target = removeSeqInfoGR(gb), query = removeSeqInfoGR(gb$query))

XSR_toPlot <- sapply(XSR_HOGs, removeSeqInfoGB)
XSR_toPlot <- as(XSR_toPlot, "GRangesList")
seqlevels(XSR_toPlot$Oki_Osa)       <- "XSR_01_Oki"
seqlevels(XSR_toPlot$Oki_Osa$query) <- "XSR_03_Osa"
seqlevels(XSR_toPlot$Oki_Bar)       <- "XSR_01_Oki"
seqlevels(XSR_toPlot$Oki_Bar$query) <- "XSR_06_Bar"
seqlevels(XSR_toPlot$Oki_Kum)       <- "XSR_01_Oki"
seqlevels(XSR_toPlot$Oki_Kum$query) <- "XSR_02_Kum"
seqlevels(XSR_toPlot$Oki_Aom)       <- "XSR_01_Oki"
seqlevels(XSR_toPlot$Oki_Aom$query) <- "XSR_04_Aom"
seqlevels(XSR_toPlot$Oki_Nor)       <- "XSR_01_Oki"
seqlevels(XSR_toPlot$Oki_Nor$query) <- "XSR_06_Nor"

seqlevels(XSR_toPlot$Osa_Oki)       <- "XSR_03_Osa"
seqlevels(XSR_toPlot$Osa_Oki$query) <- "XSR_01_Oki"
seqlevels(XSR_toPlot$Osa_Bar)       <- "XSR_03_Osa"
seqlevels(XSR_toPlot$Osa_Bar$query) <- "XSR_06_Bar"
seqlevels(XSR_toPlot$Osa_Kum)       <- "XSR_03_Osa"
seqlevels(XSR_toPlot$Osa_Kum$query) <- "XSR_02_Kum"
seqlevels(XSR_toPlot$Osa_Aom)       <- "XSR_03_Osa"
seqlevels(XSR_toPlot$Osa_Aom$query) <- "XSR_04_Aom"
seqlevels(XSR_toPlot$Osa_Nor)       <- "XSR_03_Osa"
seqlevels(XSR_toPlot$Osa_Nor$query) <- "XSR_06_Nor"

seqlevels(XSR_toPlot$Bar_Osa)       <- "XSR_06_Bar"
seqlevels(XSR_toPlot$Bar_Osa$query) <- "XSR_03_Osa"
seqlevels(XSR_toPlot$Bar_Oki)       <- "XSR_06_Bar"
seqlevels(XSR_toPlot$Bar_Oki$query) <- "XSR_01_Oki"
seqlevels(XSR_toPlot$Bar_Kum)       <- "XSR_06_Bar"
seqlevels(XSR_toPlot$Bar_Kum$query) <- "XSR_02_Kum"
seqlevels(XSR_toPlot$Bar_Aom)       <- "XSR_06_Bar"
seqlevels(XSR_toPlot$Bar_Aom$query) <- "XSR_04_Aom"
seqlevels(XSR_toPlot$Bar_Nor)       <- "XSR_06_Bar"
seqlevels(XSR_toPlot$Bar_Nor$query) <- "XSR_06_Nor"

makeOxfordPlots(unlist(XSR_toPlot))
```

# HOX genes

## Hox 4

Appears to be duplicated in Bar so missing from HOG `N19.HOG0000192` in that
genome.

```{r Hox4}
sapply(orthoPairs[1:15], \(gb) {
  x <- gb |> plyranges::filter(HOG == "N19.HOG0000192")
  if (length(x) == 0) return(NULL)
  x})

Hox4 <- orthoPairs$Oki_Osa |> plyranges::filter(HOG == "N19.HOG0000192")
Hox4_1e4 <- GBreaks(target = granges(Hox4) + 1e4, query = Hox4$query + 1e4)
Hox4_4e4 <- GBreaks(target = granges(Hox4) + 4e4, query = Hox4$query + 4e4)
Hox4_1e5 <- GBreaks(target = granges(Hox4) + 1e5, query = Hox4$query + 1e5)

subsetByOverlaps(coa$Oki_Osa, Hox4_1e4) |> plotApairOfChrs("chr2", xlim = gb2xlim(Hox4_1e4))
subsetByOverlaps(coa$Oki_Osa, Hox4_4e4) |> plotApairOfChrs("chr2", xlim = gb2xlim(Hox4_4e4))
subsetByOverlaps(coa$Oki_Osa, Hox4_1e5) |> plotApairOfChrs("chr2", xlim = gb2xlim(Hox4_1e5))

## Genes
Hox4_4e4.genes <- transcripts$OKI2018.I69 |> subsetByOverlaps(Hox4_4e4 |> range())
mids <- Hox4_4e4.genes |> gr2dna_seg() |> dplyr::mutate(mid = (start + end) / 2) |> dplyr::pull(mid)
text <- Hox4_4e4.genes$tx_name |> as.character()
annot <- genoPlotR::annotation(x1=start(Hox4_4e4.genes), x2=end(Hox4_4e4.genes), text=text, rot=30)
gbs$Oki_Osa |> plotApairOfChrs("chr2", xlim = gb2xlim(Hox4_4e4), annotations = annot)
```

## Hox 1

HOG `N19.HOG0000193`.

```{r Hox1}
sapply(orthoPairs[1:15], \(gb) {
  x <- gb |> plyranges::filter(HOG == "N19.HOG0000193")
  if (length(x) == 0) return(NULL)
  x})

Hox1 <- orthoPairs$Oki_Osa |> plyranges::filter(HOG == "N19.HOG0000193")
Hox1_1e4 <- GBreaks(target = granges(Hox1) + 1e4, query = Hox1$query + 1e4)
Hox1_4e4 <- GBreaks(target = granges(Hox1) + 4e4, query = Hox1$query + 4e4)
Hox1_1e5 <- GBreaks(target = granges(Hox1) + 1e5, query = Hox1$query + 1e5)

subsetByOverlaps(coa$Oki_Osa, Hox1_1e4) |> plotApairOfChrs("chr2", xlim = gb2xlim(Hox1_1e4))
subsetByOverlaps(coa$Oki_Osa, Hox1_4e4) |> plotApairOfChrs("chr2", xlim = gb2xlim(Hox1_4e4))
subsetByOverlaps(coa$Oki_Osa, Hox1_1e5) |> plotApairOfChrs("chr2", xlim = gb2xlim(Hox1_1e5))

## Genes
Hox1_4e4.genes <- transcripts$OKI2018.I69 |> subsetByOverlaps(Hox1_4e4 |> range())
mids <- Hox1_4e4.genes |> gr2dna_seg() |> dplyr::mutate(mid = (start + end) / 2) |> dplyr::pull(mid)
text <- Hox1_4e4.genes$tx_name |> as.character()
annot <- genoPlotR::annotation(x1=start(Hox1_4e4.genes), x2=end(Hox1_4e4.genes), text=text, rot=30)
gbs$Oki_Osa |> plotApairOfChrs("chr2", xlim = gb2xlim(Hox1_4e4), annotations = annot)
```

# Session information

```{r session_information}
sessionInfo()
```
