---
title: "OSKA2016 superscaffolding"
output: html_notebook
editor_options: 
  chunk_output_type: console
---

# First, let's superscaffold O9 with O3

```{r}
library("magrittr")
library("BSgenome.Odioica.local.OSKA2016")
library(GenomicBreaks)
O95f <- load_genomic_breaks("~/FromSango/OSKA2016__O9-5f/OSKA2016__O9-5f.gff3", target_bsgenome = OSKA2016, type = "match")
genome(O95f$query) <- "O95f"
O96f <- load_genomic_breaks("~/FromSango/OSKA2016__O9-6f/OSKA2016__O9-6f.gff3", target_bsgenome = OSKA2016, type = "match")
genome(O96f$query) <- "O96f"
O97f <- load_genomic_breaks("~/FromSango/O9-7f/pair_align/align/O9__ref_genome.filtered.gff3", query_bsgenome = OSKA2016, type = "match")
genome(O97f) <- "O97f"
O97f <- sort(ignore.strand = TRUE, swap(O97f))

O33f <- load_genomic_breaks("~/FromSango/OSKA2016__O3-3f/OSKA2016__O3-3f.gff3", target_bsgenome = OSKA2016, type = "match")
genome(O33f$query) <- "O33f"
O34f <- load_genomic_breaks("~/FromSango/OSKA2016__O3-4f/OSKA2016__O3-4f.gff3", target_bsgenome = OSKA2016, type = "match")
genome(O34f$query) <- "O34f"
O35f <- load_genomic_breaks("~/FromSango/O3-5f/pair_align/align/O3__ref_genome.filtered.gff3", query_bsgenome = OSKA2016, type = "match")
genome(O35f) <- "O35f"
O35f <- sort(ignore.strand = TRUE, swap(O35f))

OSK421f <- load_genomic_breaks("~/FromSango/OSKA2016__OSK42-1f/OSKA2016__OSK42-1f.gff3", target_bsgenome = OSKA2016, type = "match")
genome(OSK421f$query) <- "OSK421f"

AOM5.4f <- load_genomic_breaks("~/FromSango/AOM-5-4f/pair_align/align/AOM5__ref_genome.gff3", query_bsgenome = OSKA2016, type = "match")
genome(AOM5.4f) <- "AOM5.4f"
AOM5.4f <- sort(ignore.strand = TRUE, swap(AOM5.4f))
```

# Telomeres

```{r}
possibleTelomeres <- function(file) {
  tantan <- rtracklayer::import.gff3(file)
  tantan$motifLength <- nchar(tantan$Name)
  possibleTelomereFwd <- grepl(pattern = "TTAGGG", paste0(tantan$Name, tantan$Name))
  possibleTelomereRev <- grepl(pattern = "CCCTAA", paste0(tantan$Name, tantan$Name))
  tantan[tantan$motifLength == 6 & (possibleTelomereFwd | possibleTelomereRev) & score(tantan) > 100]
}
```

# Long tandems

```{r}
longTandems <- function(file) {
  tantan <- rtracklayer::import.gff3(file)
  tantan$motifLength <- nchar(tantan$Name)
  tantan[tantan$motifLength > 50 & tantan$motifLength < 60 & score(tantan) > 1000]
}
```

# Prior information

We inspected by eye the dotplots of different genome alignments to OSKA2016
and came to the conclusion that the following scaffolds belong to the following
chromosomes (or chromosome regions).

```{r}
XSR <- c( # High confidence
          "S10+", "S9+", "S32+", "S84-", "S59+", "S8-", "S1+", "S66-")

YSR <- c( # High confidence
          "S63-", "S71-", "S68-", "S48-", "S55+", "S42-", "S65-", "S87+", "S86+", "S78+", "S110+", "S62+", "S39+", "S26+", "S47+", "S64-")

PAR <- c( # Short arm – low confidence
  "S29+", "S49-", "S33+", "S80-", "S119+", "S93+", "S43+", "S128+", "S83+", "S57-", "S56+", "S45+", "S117-", "S88+", "S92-", "S21+", "S12+", "S79+", "S89+", "S53+", "S54-", "S11-", "S74+", "S25-",
          # Long arm – higher confidence
          "S2-", "S18+", "S73-", "S23+", "S52-", "S70-", "S13-")

chr1 <- c( # Short arm – high confidence until S37
           "S31-", "S51-", "S38-", "S85+", "S61-", "S44-", "S109+", "S37-", "S28+", "S36+", "S76+", "S58+",
           # Long arm – low confidence
           "S6+", "S3+", "S5+", "S7-")

chr2 <- c( # Short arm – good confidence
           "S82-", "S30+", "S15-", "S50-", "S41+", "S17+", "S16-", "S20+",
           # Long arm – high confidence
          "S60-", "S14-", "S90-", "S22-", "S35+", "S118+", "S46+", "S102-", "S72-", "S104+", "S40+", "S27+", "S75-", "S24-", "S19+", "S101+", "S4-")
```

```{r old versions}

chr1.old <- c("S6+", "S36+", "S28+", "S51+", "S31+", "S38+", "S37+",
          "S3+", "S5+", "S7-")

chr2.old <- c("S16+", "S17-", "S41-", "S50+", "S15+", "S20+",
          "S30+", "S4+", "S19-", "S24+", "S75+", "S27-",
          "S40-", "S72+", "S46-", "S35-", "S22+", "S14+")

PAR.old <- c("S25+", "S11+", "S54+", "S53-", "S21+", "S12+", "S29+", "S49-",
             "S33+", "S80-", "S93+", "S13+", "S70+", "S52+", "S23-", "S73+",
             "S18-", "S2+")

YSR.old <- c("S68-", "S48-", "S42-",  "S65-", "S55-", "S87+", "S63-", "S86+", "S62+", "S39+", "S47+")
```

The purpose of this computational analysis is to validate and extend our findings
in a systematic manner.

We rely on the largest scaffolds of the multiple other assemblies to support the
superscaffolding of OSKA2016.  In this anlysis, all support assemblies have
been mapped to OSKA2016, which means that they are in the _query_ slot of the
`GenomicBreaks` objects.  We do not prepare _BSgenome_ packages for these
assemblies, so we need to identify the longest scaffolds by ourselves.


# Chains

Example:

```
> O95f.cc
$contig_113
[1] "S82-"  "S30+"  "S121+" "S15-"  "S50-" 

$contig_123
[1] "S20-" "S16+"

$contig_136
[1] "S42+" "S55-" "S68+" "S71+"
…
```

## Filter data

```{r}
# getScafInChain <- function(chain, scafName) {
#   n <-  grep(paste0(scafName, "[+-]"), chain)
#   chain[n]
# }
```


```{r}
# bigScaff <- function(gb, n = 10) {
#   gb <- reduce(gb$query, min = 1e7) # Why does it crash with min = Inf or 1e10?
#   w <- tapply(end(gb), seqnames(gb), max)
#   head(sort(w, decreasing = TRUE), n)
# }
```

```{r}
# minwidth adjusted to avoid having two strands in the same scaffold in the output.
# superScaf <- function(gb, n = 1, minwidth = 8e4) {
#   bs <- bigScaff(gb, n)
#   q <- swap(gb)
#   scaf <- sort(q[seqnames(q) == names(bs[n])], ignore.strand = TRUE) 
#   coalesce_contigs(scaf, min = minwidth)
# }
# 
# superScafByName <- function(gb, n , minwidth = 8e4) {
#   q <- swap(gb)
#   scaf <- sort(q[seqnames(q) == n], ignore.strand = TRUE) 
#   coalesce_contigs(scaf, min = minwidth)
# }
```

```{r}
# validate <- function(scaf, chr, flip = FALSE) {
#   names <- shortNames(scaf, flip)
#   match(names, chr)
# }
# 
# shortNamesList <- function(gb, n, names = FALSE, minwidth = 8e4) {
#   l <- sapply(1:n, function(n) superScaf(gb, n, minwidth) %>% shortNames())
#   if (isTRUE(names)) {
#     names(l) <- names(bigScaff(gb, n))
#   }
#   l
# }
```

```{r}
sortQ <- function(gb) {
  gb[order(gb$query),]
}
```

## O9

```{r}
O95f.c  <- longestMatchesInTarget(O95f)
O95f.cc <- queryStrandNames(O95f.c)

O96f.c <- longestMatchesInTarget(O96f)
O96f.cc <- queryStrandNames(O96f.c)

O97f.c <- longestMatchesInTarget(O97f)
O97f.cc <- queryStrandNames(O97f.c)
names(O97f.cc) <- paste("O97f", names(O97f.cc), sep = "_")
O97f.all <-  longestMatchesInTarget(O97f, min.width = 0, min.matches = 1) %>% queryStrandNames()
```


## O3

```{r}
O35f.c  <- longestMatchesInTarget(O35f)
O35f.cc <- queryStrandNames(O35f.c)
names(O35f.cc) <- paste("O35f", names(O35f.cc), sep = "_")
O35f.all  <- longestMatchesInTarget(O35f, min.width = 0, min.matches = 0) %>% queryStrandNames()

O33f.pc <- O33f
O33f.pc %<>% split(seqnames(O33f.pc)) 
normRanges <- function(x, s) {
  newStart <- round(start(x) * 100 / s)
  newEnd <- round(end(x) * 100 / s)
  ranges(x) <- IRanges(newStart, newEnd)
  x
}
O33f.pc <- mendoapply(normRanges, O33f.pc, seqlengths(OSKA2016)[names(O33f.pc)])
O33f.pc %<>% unlist %>% sort

O33f.1000 <- O33f[width(O33f) > 1000]
O33f.c  <- longestMatchesInTarget(O33f)
O33f.cc <- queryStrandNames(O33f.c)
names(O33f.cc) <- paste("O33f", names(O33f.cc), sep = "_")
O33f.all  <- longestMatchesInTarget(O33f, min.width = 0, min.matches = 0) %>% queryStrandNames()



O33f.tel <- possibleTelomeres("~/FromSango/O3-3f.tantan.gff3")
O33f.tel %>% as.data.frame()
O33f.telnames  <- O35f.tel %>% seqnames() %>% unique() %>% as.character()
O33f.telnames2 <- O35f.tel %>% seqnames() %>% unique() %>% paste0("O35f_", .)

O33f.all[O33f.telnames %>% Filter (function(X) X %in% names(O33f.all), .)]
O33f.cc[O33f.telnames2 %>% Filter (function(X) X %in% names(O33f.cc), .)]

O35f.tel <- possibleTelomeres("~/projects/GenomicBreaks/inst/extdata/O3-5f.tantan.gff3")
O35f.tel %>% as.data.frame()
O35f.telnames  <- O35f.tel %>% seqnames() %>% unique() %>% as.character()
O35f.telnames2 <- O35f.tel %>% seqnames() %>% unique() %>% paste0("O35f_", .)

O35f.all[O35f.telnames %>% Filter (function(X) X %in% names(O35f.all), .)]
O35f.cc[O35f.telnames2 %>% Filter (function(X) X %in% names(O35f.cc), .)]
```

## OSK42

```{r}
OSK421f.c  <- longestMatchesInTarget(OSK421f)
OSK421f.cc <- queryStrandNames(OSK421f.c)
names(OSK421f.cc) <- paste("OSK421f", names(OSK421f.cc), sep = "_")
OSK421f.all  <- longestMatchesInTarget(OSK421f, min.width = 0, min.matches = 1) %>% queryStrandNames()
```

## AOM5

```{r}
AOM5.4f.c  <- longestMatchesInTarget(AOM5.4f)
AOM5.4f.cc <- queryStrandNames(AOM5.4f.c)
names(AOM5.4f.cc) <- paste("AOM5.4f", names(AOM5.4f.cc), sep = "_")
AOM5.4f.all  <- longestMatchesInTarget(AOM5.4f, min.width = 0, min.matches = 1) %>% queryStrandNames()


shortNamesList(AOM5.4f, 20, min = 1e4)


# Contains telomere ?
superScafByName(AOM5.4f,"contig_178", 1e4) %>% shortNames()
#[1] "S25+" "S11-" "S29-"
```

## XSR

```{r}
grepByContigName <- function (l, name) {
  name <- sub(pat="\\+", rep="\\\\+", name)
  idx <- grep(name, l)
  if (length(idx) != 1) {
    name <- paste0(name, "[+-]")
      idx <- grep(name, l)
  }
  if (length(idx) != 1) return(paste("found", length(idx) ,"matches."))
  l[idx]
}

XSR
OSK421f.cc %>% flipStrandNames() %>% grepByContigName(name = "S10+")
AOM5.4f.cc %>% grepByContigName(name = "S10+")
O97f.cc %>% flipStrandNames %>% grepByContigName("S9+")
O35f.cc %>% grepByContigName("S9+")
O35f.cc %>% grepByContigName("S1+")

XSR
OSK421f.all %>% flipStrandNames() %>% grepByContigName(name = "S10+")
AOM5.4f.all %>% grepByContigName(name = "S10+")
O97f.all %>% flipStrandNames %>% grepByContigName("S9+")
O35f.all %>% grepByContigName("S9+")
O35f.all %>% flipStrandNames %>% grepByContigName("S10+")
O35f.all %>% grepByContigName("S1+")


#S66
O33f %>% subset(seqnames == "S66" & width > 100)%>% as.data.frame()
O33f %>% subset(seqnames(query) == "scaffold_16" & width > 1000) %>% sortQ %>% as.data.frame()   %>% tail
```

```
XSR         "S10+" "S9+"  "S32+" "S84-" "S59+" "S8-"  "S1+"          "S66-"
OSK421f.cc  "S10+" "S9+"  "S32+" "S84-" "S59+" "S8+"  "S1+" 
AOM5.4f.cc  "S10+" "S9+"  "S32+" "S84-" "S59+" "S8-"  "S1+" 
O97f.cc            "S9+"  "S32+" "S84-" "S59+" "S8-"  "S1+"  "S18+"
O35f.cc            "S9+"  "S32+" "S84-" "S59+"
```

Are we missing important pieces matches between the scaffolds above ?  No, the
assemblies do not agree with each other, and the length of the scafoolds
are short.

The table below is made with "longest matches" only.

```
XSR                          "S10+"          "S9+"           "S32+"          "S84-"         "S59+"          "S8-"                           "S1+" 
OSK421f.all                  "S10+"  "S246+" "S9+"           "S32+"  "S195+" "S84-"         "S59+"  "S494+" "S8+"           "S477+"         "S1+"   "S462-" "S510+" "S564-" "S391+"
AOM5.4f.all  "S307-" "S273-" "S10+"  "S176-" "S9+"           "S32+"          "S84-"         "S59+"          "S8-"   "S366-" "S477+" "S478-" "S1+"   "S517-" "S510+" "S491+" "S66-"  "S481+" "S154+"
O97f.all                                     "S9+"   "S116-" "S32+"  "S494-" "S84-"         "S59+"  "S477+" "S8-"                           "S1+"           "S510+" "S395+" "S346-" "S542-" "S528-" "S456+" "S509-" "S507-" "S236-" "S417-" "S18+" 
O35f.all                             "S246+" "S9+"           "S32+"          "S84-" "S494-" "S59+" 
O35f.all                                                                            "S564-" "S565+" "S477+" "S8-"                   "S443+" "S1+"   "S278+" "S510+"
```


## chr1
 
```{r}
grepAllContigsFromScaff <- function(scaf, l) {
  sapply(chr1, grepByContigName, l = l)
}

# Example command for exploration
# grepAllContigsFromScaff(chr1, O35f.cc %>% flipStrandNames())

chr1
O35f.cc %>% grepByContigName("S37")
O97f.cc %>% flipStrandNames() %>% grepByContigName("S37")
O97f.cc %>% grepByContigName("S38")

AOM5.4f.cc %>% grepByContigName("S31")
O97f.cc %>% grepByContigName("S31")
O33f.cc %>% grepByContigName("S31")
O35f.cc %>% grepByContigName("S31")
OSK421f.cc %>% grepByContigName("S31")

O35f.cc %>% grepByContigName("S31")
O97f.cc %>% grepByContigName("S31")
O97f.cc %>% flipStrandNames %>% grepByContigName("S44")
AOM5.4f.cc %>% grepByContigName("S37")

# On dotplots, S28 does not align neatly.  Maybe there is residual heterozygocity in the Nanopore assemblies.
O35f %>% subset(seqnames == "S28" & width > 2000)
O35f %>% subset(seqnames == "S36" & width > 1000) # contig_164 - ... - contig_68
O35f %>% subset(seqnames(query) == "contig_164" & width > 1000) %>% sortQ %>% as.data.frame()  # S6- ... S36 +
O35f %>% subset(seqnames(query) == "contig_68" & width > 1000) %>% sortQ %>% as.data.frame()  # S6+ S36 + S76+ S58+ S6+

O97f %>% subset(seqnames == "S28" & width > 2000) %>% as.data.frame()
O97f %>% subset(seqnames == "S36" & width > 1000) # contig_187 - ... - contig_189
O97f %>% subset(seqnames(query) == "contig_187" & width > 1000) %>% sortQ %>% as.data.frame()  # S6+ ... S36 - ... S6 +
O97f %>% subset(seqnames(query) == "contig_189" & width > 1000) %>% sortQ %>% as.data.frame()  # S6+ S36 + S76+ S58+
O97f %>% subset(seqnames == "S58" & width > 2000) # contig_189 - ... - contig_102
O97f %>% subset(seqnames(query) == "contig_102" & width > 1000) %>% sortQ %>% as.data.frame()  # S6+ S36 + S76+ S58+



O35f.all %>% grepByContigName("S3")
O97f.cc %>% grepByContigName("S7")
O97f.cc %>% flipStrandNames %>% grepByContigName("S7")
AOM5.4f.cc %>% grepByContigName("S7")

# S7+ -- S5-
O33f.10000[seqnames(O33f.10000$query) == "contig_353"] %>% swap() %>% sort %>% as.data.frame() 
O33f.10000[seqnames(O33f.10000$query) == "contig_56"] %>% swap() %>% sort %>% as.data.frame()
```

```
chr1     "S31-" "S51-" "S38-" "S85+" "S61-" "S44-" "S109+" "S37-"  "S6+"   "S3+"   "S5+"   "S7-"
O35f.cc  "S31-" "S51-" "S38-" "S85+" "S61-" "S44-" "S109+" "S37-"
O97f.cc  "S31-" "S51-" "S38-"
O97f.cc                                     "S44-"         "S37-"

O9tmp, 8, 1e4                                                   "S4-" "S5+" "S7-"
O9tmp,13, 1e4                                                   "S3-" "S5+"
O9tmp,14, 1e4  "S6+"         "S28-"
O97f,7, 1e4                                                           "S5+" "S7-"
O97f,13, 1e4                                                    "S3-" "S5+"
```

## chr2

### Short arm

```{r}
chr2

AOM5.4f.cc %>% grepByContigName("S82")
O97f.cc %>% grepByContigName("S82")
O35f.cc %>% flipStrandNames %>% grepByContigName("S82")
O33f.cc %>% grepByContigName("S82")

# O35f contig2 ends with a telomere.  That area matches S82
O35f %>% subset(seqnames == "S82" & width > 10000)
# S30 follows
O35f %>% subset(seqnames(query) == "contig_2" & width > 10000) %>% sortQ
O35f %>% subset(seqnames == "S30" & width > 10000) 
# Then S15, S50, S41, S17 and a piece of S12
O35f %>% subset(seqnames(query) == "contig_37" & width > 10000) %>% sortQ %>% as.data.frame()

# But unfortunately it is at the centre of S12...

# S16 and S20 are also obviously part of chr2 given synteny with Oki.
# Fortunately, they are together.
O33f %>% subset(seqnames == "S20" & width > 50000)
O33f %>% subset(seqnames(query) == "contig_263" & width > 50000) %>% sortQ
O33f %>% subset(seqnames == "S16" & width > 50000) %>% as.data.frame()
O33f %>% subset(seqnames(query) == "contig_456" & width > 50000) %>% sortQ 
# And that leads to S17 !
O33f %>% subset(seqnames == "S17" & width > 10000) %>% as.data.frame()
O33f %>% subset(seqnames(query) == "contig_30" & width > 1000) %>% sortQ %>% as.data.frame()


O35f.cc %>% grepByContigName("S15")

O35f %>% subset(seqnames == "S12" & width > 10000)%>% as.data.frame()
O35f %>% subset(seqnames(query) == "contig_127" & width > 10000) %>% sortQ
O35f %>% subset(seqnames == "S21" & width > 10000)

O35f %>% subset(seqnames == "S54" & width > 10000)

O35f.cc %>% flipStrandNames %>% grepByContigName("S82")
O35f.cc %>% grepByContigName("S15")
O35f.cc %>% flipStrandNames %>% grepByContigName("S12")


```


```
O97f_contig_275 "S82-" "S30+" "S15-" "S50-" "S41+" "S17+"
O33f_contig_12  "S82-" "S30+"
O35f_contig_37                "S15-" "S50-" "S41+" "S17+"
```

### Long arm

```{r chr2_long}
#Luckily, O33f's scaffold 2 is the whole long arm of Chr2.

flipStrandNames(O33f.cc)$O33f_scaffold_2
O33f %>% subset(seqnames(query) == "scaffold_2" & width > 100000) %>% sortQ %>% as.data.frame()

# And S4's tail aligns to a contig ending with a telomere !
O33f %>% subset(seqnames == "S4" & width > 100000)%>% as.data.frame()

# Dead end on the other side.
O33f %>% subset(seqnames == "S14" & width > 100000)%>% as.data.frame()

# But it allows to place S60, although with low confidence
O33f %>% subset(seqnames == "S60" & width > 100)%>% as.data.frame() %>% head
O33f %>% subset(seqnames(query) == "scaffold_2" & width > 1000) %>% sortQ %>% as.data.frame()   %>% tail


OSK421f.cc %>% grepByContigName("S82")

AOM5.4f.cc %>% grepByContigName("S17")
O97f.cc %>% grepByContigName("S17")
O35f.cc %>% grepByContigName("S17")
OSK421f.cc %>% grepByContigName("S17")


AOM5.4f.cc %>% grepByContigName("S4+")
O97f.cc %>% grepByContigName("S4+")
O35f.cc %>% grepByContigName("S4+")
```

```


O97f.cc$O97f_contig_275  TELOMERE-with-TTTTAATTTTAA [1] "S82-" "S30+" "S15-" "S50-" "S41+" "S17+"
O35f.cc                                                               "S15-" "S50-" "S41+" "S17+"
$O35f_contig_2
[1] "S82-" "S30+"
                                                                     
chr2                  "S16+" "S17-" "S41-" "S50+" "S15+" "S20+" "S30+" "S4+"          "S19-" "S24+" "S75+" "S27-" "S40-"         "S72+"          "S46-"          "S35-"  "S22+"          "S14+"
AOM5.4f.cc                                                             "S4+"  "S101-" "S19-" 
O97f.cc                                                                "S4+"  "S101-" "S19-" "S24+" "S75+" "S27-" "S40-" "S114-" "S72+"  "S102+" "S46-"  "S118-" "S35-"  "S22+"  "S90+"  "S14+" 
O35f.cc                                                                "S4+"  "S101-" "S19-" "S24+" "S75+" "S27-" "S40-" "S104-" "S72+"          "S46-"  "S118-" "S35-"  "S22+"  "S90+"  "S14+" 


O97f,4, 1e4)                 "S17-" "S41-" "S50+" "S15+"        "S30-" "S82+"
AOM5.4f,4, 1e4)                                                        "S4+"   "S101-" "S19-" 
O97f,2, 1e4                                                            "S4+"   "S101-" "S19-" "S24+"  "S75+"  "S27-"  "S40-"  "S114-" "S72+"  "S102+" "S46-"  "S118-" "S35-"  "S22+"  "S90+"  "S22+"  "S14+"
O97f,14, 1e4  "S20-" "S16+"
O35f,2, 5e4                                                            "S4+"           "S19-" "S24+"        "S27-" "S40-"             "S35-" "S22+" "S14+"  (S101+  "S72+"  "S46-" are there at lower thresholds)

```

## PAR

### Short arm

```{r}
AOM5.4f.cc %>% grepByContigName("S11")
OSK421f.cc %>% grepByContigName("S33")
O97f.cc %>% grepByContigName("S33")
O35f.cc %>% grepByContigName("S33")
O33f.cc %>% flipStrandNames() %>% grepByContigName("S57")

O97f.cc %>% grepByContigName("S29")
O35f.cc %>% grepByContigName("S29")
```

This suggests S11 comes before S33

```
AOM5.4f_contig_866]  "S11+"                  "S33+"
O97f_contig_40                               "S33+"          "S119+" "S93+"  "S43+"  "S128+" "S83+"  "S57-"  "S56+"  "S45+"          "S88+"  "S92-" 
O35f_contig_35                       "S49-"  "S33+"  "S80-"  "S119+" "S93+"  "S43+"  "S128+" "S83+"  "S57-"  "S56+"  "S45+"          "S88+"  "S92-" 
$O33f_contig_26                                      "S80-"  "S119+" "S93+"  "S43+"  "S128+" "S83+"  "S57-"  "S56+"  "S45+"  "S117-" "S88+" 
O97f_scaffold_105             "S29+" "S49-"
```


```{r}
# S11 may be misassembled...
O33f %>% subset(seqnames == "S11" & width > 10000)
O33f %>% subset(seqnames(query) == "contig_70" & width > 5000) %>% sortQ %>% as.data.frame()
O33f %>% subset(seqnames(query) == "contig_13" & width > 5000) %>% sortQ %>% as.data.frame()

O33f %>% subset(seqnames == "S25" & width > 10000)
O33f %>% subset(seqnames(query) == "contig_13" & width > 1000) %>% sortQ

AOM5.4f.cc %>% grepByContigName("S25")
OSK421f.cc %>% grepByContigName("S25")
O97f.cc %>% grepByContigName("S25")
O35f.cc %>% grepByContigName("S25")
O33f.cc %>% grepByContigName("S25")

# O33f does not tell us where to place S25
O33f %>% subset(seqnames == "S25" & width > 1000)
# contig_13 aligns with an internal part of S11 ; let's ignore that.
# contig_13 aligns with an internal part of S19 ; let's ignore that.
# But then, contig_13 is a dead end.
O33f %>% subset(seqnames(query) == "contig_13" & width > 2000) %>% sortQ %>% as.data.frame()

# What does O9 tell us about S25 ?
O97f %>% subset(seqnames == "S25" & width > 1000) %>% as.data.frame()
O97f %>% subset(seqnames(query) == "contig_46" & width > 1000) %>% sortQ %>% as.data.frame()
# contig_546 suggests S25+ S11+
O97f %>% subset(seqnames(query) == "contig_546" & width > 2000) %>% sortQ %>% as.data.frame()

# So O97f suggests S25 comes before S11
```


```{r}
#But we wtill miss 21, 12

O97f.cc %>% grepByContigName("S12")
O35f.cc %>% grepByContigName("S12")

# S21 is followed by S12...

# S21 ends with contig_72 and a piece of contig_4. Is it near rRNA and sex-specific region ?
O33f %>% subset(seqnames == "S21" & width > 10000) %>% as.data.frame()

# contig_26 says that S92 is followed by S21
O33f %>% subset(seqnames(query) == "contig_26" & width > 1000) %>% sortQ %>% as.data.frame() %>% head(10)
O35f %>% subset(seqnames(query) == "contig_35" & width > 1000) %>% sortQ %>% as.data.frame() %>% tail(10)

O33f %>% subset(seqnames == "S92" & width > 100) %>% as.data.frame()
O35f %>% subset(seqnames == "S92" & width > 100) %>% as.data.frame()


O33f %>% subset(seqnames == "S54" & width > 100) %>% as.data.frame()
O35f %>% subset(seqnames == "S54" & width > 100) %>% as.data.frame()
```

```
AOM5.4f_contig_866]  "S11+"                  "S33+"
O97f_contig_40                               "S33+"          "S119+" "S93+"  ...  "S45+"          "S88+"  "S92-" 
O35f_contig_35                       "S49-"  "S33+"  "S80-"  "S119+" "S93+"  ...  "S45+"          "S88+"  "S92-" 
$O33f_contig_26                                      "S80-"  "S119+" "S93+"  ...  "S45+"  "S117-" "S88+" 
O97f_scaffold_105             "S29+" "S49-"
                                                                                                                 "S21+" "S12+"
```

```{r}
O33f %>% subset(seqnames(query) == "contig_72" & width > 1000)
# A piece of S12 alignes on a different contig, that also contains S11 and S74
O33f %>% subset(seqnames(query) == "contig_70" & width > 3000) %>% sortQ %>% seqnames() %>% Rle
O33f %>% subset(seqnames(query) == "contig_70" & width > 3000) %>% sortQ %>% as.data.frame() 

# S12+, S79+, S89+, S53+,  S54-, S11-, S74+

# S11 and S74 seems mixed ?
O33f %>% subset(seqnames == "S11" & width > 1000) %>% as.data.frame()
O33f %>% subset(seqnames == "S74" & width > 100)

#Let's put S25- after S11-/S74+...

PAR.short <- c("S29+", "S49-",  "S33+",  "S80-",  "S119+", "S93+",  "S43+",  "S128+", "S83+",  "S57-",  "S56+",  "S45+", "S117-", "S88+", "S92-", "S21+", "S12+", "S79+", "S89+", "S53+",  "S54-", "S11-", "S74+", "S25-")
```

```
O97f_contig_40              "S33+"          "S119+" "S93+"  "S43+"  "S128+" "S83+"  "S57-"  "S56+"  "S45+"  "S88+"  "S92-" 
O35f_contig_35      "S49-"  "S33+"  "S80-"  "S119+" "S93+"  "S43+"  "S128+" "S83+"  "S57-"  "S56+"  "S45+"  "S88+"  "S92-" 

              "S29"                                                                                                        "S21+" "S12+" 

```


### Long arm

```{r}
PAR

# Telomeric region

#  "S52-" "S70-" "S13-" supported by O9 and O3
O33f.pc %>% subset(seqnames == "S13" & width > 10)
O33f.pc %>% subset(seqnames(query) == "contig_67" & width > 5) %>% sortQ
O33f.pc %>% subset(seqnames == "S52" & width > 1)
O97f.cc %>% flipStrandNames %>% grepByContigName("S13")
AOM5.4f.cc %>% grepByContigName("S13")
OSK421f.cc %>% grepByContigName("S13")

# Search connections with S52

# Dead end in O3-3f
O33f %>% subset(seqnames == "S52" & width > 10)
O33f %>% subset(seqnames(query) == "contig_67" & width > 5) %>% sortQ

# Connects with S23+ in O3-5f
O35f %>% subset(seqnames == "S52" & width > 10)
O35f %>% subset(seqnames(query) == "scaffold_181" & width > 5) %>% sortQ %>% as.data.frame()

# S2 etc.

# In O3-3f, S2 is fully in contig_227
O33f %>% subset(seqnames == "S2" & width > 10000)
# S2 S18 S73 S23
O33f %>% subset(seqnames(query) == "contig_227" & width > 10000) %>% sortQ %>% strandNames() %>% Rle()
O33f %>% subset(seqnames == "S23" & width > 10000) %>% queryStrandNames() %>% Rle()
O33f %>% subset(seqnames(query) == "contig_62" & width > 1000) %>% sortQ %>% strandNames() %>% Rle()
# Check S34 because of O9


O97f %>% subset(seqnames == "S2" & width > 100000)
O97f %>% subset(seqnames(query) == "contig_48" & width > 10000) %>% sortQ %>% strandNames() %>% Rle()
# The above suggests S2 between S1 and S18.  Or is there a second rRNA locus on PAR ?

O97f %>% subset(seqnames(query) == "contig_58" & width > 10000) %>% sortQ %>% strandNames() %>% Rle()
# S2+ Connects to S94+ and S34+ on the other end.

#Is S2 on the short arm ???  Not according to the alignment to OKI
#Or could there be a misassembly in S2 ?

AOM5.4f.cc %>% grepByContigName("S18")
OSK421f.cc %>% grepByContigName("S23")
O97f.cc %>% grepByContigName("S18")
O35f.cc %>% grepByContigName("S18")
O35f.cc %>% grepByContigName("S23")

O33f.pc %>% subset(seqnames == "S52" & width > 1)
O97f.cc %>% flipStrandNames %>% grepByContigName("S13")
AOM5.4f.cc %>% grepByContigName("S13")
OSK421f.cc %>% grepByContigName("S13")

```



## YSR

YSR has changed too much outside Osaka lab strain…

```{r}
YSR
O33f.cc$O35f_contig_4  # Published in OSKA2016 paper.
OSK421f.cc %>%  flipStrandNames %>% grepByContigName("S26")
AOM5.4f.cc %>% flipStrandNames %>% grepByContigName("S26")
O97f.cc %>% flipStrandNames %>% grepByContigName("S26")
O35f.cc %>% grepByContigName("S26")

```

```
YSR.old                                       "S68-"  "S48-"          "S42-" "S65-" "S55-" "S87+" "S63-" "S86+"                  "S62+"          "S39+"          "S47+"
O33f.cc$O35f_contig_4         "S63-"  "S71-"  "S68-"  "S48-"  "S55+"  "S42-" "S65-"        "S87+"        "S86+"  "S78+"  "S110+" "S62+"          "S39+"  "S26+"  "S47+"  "S64-"
O35f.cc                       "S63-"  "S71-"  "S68-"  "S48-"  "S55+"  "S42-" "S65-"        "S87+"        "S86+"  "S78+"  "S110+" "S62+"          "S39+"  "S26+"  "S47+"  "S64-" 
O97f.cc                                                                                    "S87+"        "S86+"  "S78+"  "S110+" "S62+"  "S65-"  "S39+"  "S26+"  "S47+"  "S64-" 
OSKA42                                                        "S55+"         "S65-"                                                                      "S26+"                  "S71-"
AOM5.4f.cc                                             "S48-" "S55+"                                                                       "S71-" "S47+" "S26+"          "S64-"
```

### Where is S51 ?

```{r}

O33f %>% subset(seqnames == "S51" & width > 1000)
O33f %>% subset(seqnames(query) == "contig_3" & width > 1000) %>% sortQ %>% as.data.frame()

O97f %>% subset(seqnames == "S51" & width > 1000)
O97f %>% subset(seqnames(query) == "contig_54" & width > 1000) %>% sortQ %>% as.data.frame()

OSK421f %>% subset(seqnames == "S51" & width > 1000)
OSK421f %>% subset(seqnames(query) == "contig_306" & width > 2000) %>% sortQ %>% as.data.frame()
```


### Where is S67 ?

```{r}

O33f %>% subset(seqnames == "S67" & width > 1000)
O33f %>% subset(seqnames(query) == "contig_586" & width > 1000) %>% sortQ %>% as.data.frame()
O33f %>% subset(seqnames(query) == "contig_70" & width > 5000) %>% sortQ %>% as.data.frame()

O97f %>% subset(seqnames == "S67" & width > 1000)
O97f %>% subset(seqnames(query) == "contig_280" & width > 1000) %>% sortQ %>% as.data.frame()

OSK421f %>% subset(seqnames == "S67" & width > 1000)
OSK421f %>% subset(seqnames(query) == "contig_5" & width > 2000) %>% sortQ %>% as.data.frame()
```


### Where is S99 ?

```{r}

O33f %>% subset(seqnames == "S99" & width > 1000)
O33f %>% subset(seqnames(query) == "scaffold_82" & width > 1000) %>% sortQ %>% as.data.frame()
O33f %>% subset(seqnames(query) == "scaffold_82" & width > 1000) %>% sortQ %>% strandNames()
# (edited) -> S38-, S85+, S38-, S77-, (S38-, S61-) a couple times, S61-, S44-, S37-, S99-, S37- again, S109+, S3-
O33f %>% subset(seqnames == "S37" & width > 1000)  %>% as.data.frame()


O97f %>% subset(seqnames == "S99" & width > 100)
O97f %>% subset(seqnames(query) == "contig_524" & width > 1000) %>% sortQ %>% as.data.frame()

OSK421f %>% subset(seqnames == "S99" & width > 100)
OSK421f %>% subset(seqnames(query) == "contig_98" & width > 1000) %>% sortQ %>% as.data.frame()
OSK421f %>% subset(seqnames(query) == "contig_3" & width > 1000) %>% sortQ %>% as.data.frame()
```
