---
title: "Alignment line plots, whole genomes"
author: 
 - "Charles Plessy"
date: "`r format(Sys.time(), '%d %B, %Y')`"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteEncoding{UTF-8}
  %\VignetteIndexEntry{Alignment line plots, whole genomes}
  %\VignetteEngine{knitr::rmarkdown}
editor_options: 
  chunk_output_type: console
---

```{r setup}
knitr::opts_knit$set(cache = TRUE,
                     dev=c('png', 'svg', 'pdf'),
                     fig.ext=c('png', 'svg', 'pdf'))
```

# Introduction

This vignette produces alignment line plots of pairs of genomes to illustrate
differences in the extent of scrambling in _Oikopleura_ compared to equivalent
pairs in other genera.

Figure 1 panel A was generated in this vignette.

# Load R pacakges and data

```{r load_packages_and_data}
library('OikScrambling') |> suppressPackageStartupMessages()
ggplot2::theme_set(theme_bw())
genomes <- OikScrambling:::loadAllGenomes()
load("BreakPoints.Rdata")
```

# Oikopleura

## Fig 1A

Line plot used for panel A of Figure 1.  Is based on aligned regions
(before coalescing), stored in the `gbs` object.  The manually set colors are
used throughout the manuscript to represent chromosomes 1 and 2, the PAR, the
XSR and the YSR.

```{r Oki_Osa_gbs}
makeOxfordPlots(gbs$Oki_Osa, diag = FALSE) +
  scale_color_manual(values = c("#E78AC3", "#A6D854", "#FFD92F", "#8DA0CB", "#66C2A5")) +
  theme_bw() +
  theme(legend.position="none")
```

## Okinawa vs others

### Oki vs Osa (distant)

Same as above, but colored by strand.

```{r Oki_Osa_strand, fig.height=15, fig.width=15}
makeOxfordPlots(gbs$Oki_Osa, diag = FALSE, col = "strand") + theme(legend.position="none")
```

### Oki vs Bar (distant)

```{r Oki_Bar_all, fig.height=15, fig.width=15}
makeOxfordPlots(gbs$Oki_Bar, diag = FALSE, col = "strand") + theme(legend.position="none")
```

### Oki vs Kum (near)

```{r Oki_Kum, fig.height=15, fig.width=15}
makeOxfordPlots(gbs$Oki_Kum, col = "strand") + theme(legend.position="none")
```

## Osaka vs others

### Osa vs Oki (distant)

```{r Osa_Oki, fig.height=15, fig.width=15}
makeOxfordPlots(gbs$Osa_Oki, diag = FALSE, col = "strand") + theme(legend.position="none")
```

### Osa vs Bar (intermediate)

```{r Osa_bar, fig.height=15, fig.width=15}
makeOxfordPlots(gbs$Osa_Bar, diag = FALSE, col = "strand") + theme(legend.position="none")
```

### Osa vs Aom (near)

```{r Osa_Aom, fig.height=15, fig.width=15}
makeOxfordPlots(gbs$Osa_Aom, col = "strand") + theme(legend.position="none")
```

## Barcelona vs others

### Bar vs Oki (distant)

```{r Bar_Oki, fig.height=15, fig.width=15}
makeOxfordPlots(gbs$Bar_Oki, diag = FALSE, col = "strand") + theme(legend.position="none")
```

### Bar vs Osa (intermediate)

```{r Bar_Osa, fig.height=15, fig.width=15}
makeOxfordPlots(gbs$Bar_Osa, diag = FALSE, col = "strand") + theme(legend.position="none")
```

### Bar vs Nor (near)

In these vignettes, the manuscript's "Bergen" genome is called "Norway".

**Note this plot does not demonstrate, nor rule out, collinearity between the
two genomes, because most contigs of the Bergen assembly are too short.
Alignment on the diagonal only demonstrates collinearity _within_ a contig,
not _between_.**

```{r Bar_Nor, fig.height=15, fig.width=15}
makeOxfordPlots(gbs$Bar_Nor, col = "strand") + theme(legend.position="none")
```

# Ciona

## _Ciona intestinalis_ Plymouth vs Roscoff (near)

```{r Ply_Ros, fig.height=15, fig.width=15}
makeOxfordPlots(gbs$Ply_Ros, col = "strand") + theme(legend.position="none") + ggtitle("Ciona intestinalis Plymouth vs Roscoff")
```

## _Ciona intestinalis_ Plymouth vs _Ciona robusta_ (intermediate)

```{r Ply_Rob, fig.height=15, fig.width=15}
makeOxfordPlots(gbs$Ply_Rob, col = "strand") + theme(legend.position="none") + ggtitle("Ciona intestinalis Plymouth vs Ciona robusta")
```

## _Ciona intestinalis_ Plymouth vs _Ciona savignyi_ (distant)

```{r Ply_Sav, fig.height=15, fig.width=15}
makeOxfordPlots(gbs$Ply_Sav, col = "strand") + theme(legend.position="none") + ggtitle("Ciona intestinalis Plymouth vs Ciona savignyi")
```

# Drosophila

## vs Drosophila mauritania (near)

```{r Dmel_Dmau, fig.height=15, fig.width=15}
makeOxfordPlots(gbs$Dme_Dma, col = "strand") + theme(legend.position="none") + ggtitle("Dmel vs Dmau")
```

## vs Drosophila yakuba (intermediate)

```{r Dmel_Dyak, fig.height=15, fig.width=15}
makeOxfordPlots(gbs$Dme_Dya, col = "strand") + theme(legend.position="none") + ggtitle("Dmel vs Dyak")
```

## vs Drosophila subpulchrella (intermediate)

```{r Dmel_Dsub, fig.height=15, fig.width=15}
makeOxfordPlots(gbs$Dme_Dsu, col = "strand") + theme(legend.position="none") + ggtitle("Dmel vs Dsub")
```

## vs Drosophila buskii (distant)

```{r Dmel_Dbus, fig.height=15, fig.width=15}
makeOxfordPlots(gbs$Dme_Dbu, col = "strand") + theme(legend.position="none") + ggtitle("Dmel vs Dbus")
```

# Nematodes

## C. briggsae vs C. nigoni (near)

```{r Cbr_Cni, fig.height=15, fig.width=15}
makeOxfordPlots(gbs$Cbr_Cni, col = "strand") + theme(legend.position="none") + ggtitle("C. briggsae vs C. nigoni")
```

## C. briggsae vs C. remanei (intermediate)

```{r Cbr_Cre, fig.height=15, fig.width=15}
makeOxfordPlots(gbs$Cbr_Cre, col = "strand") + theme(legend.position="none") + ggtitle("C. briggsae vs C. remanei")
```

## C. nigoni vs C. remanei (intermediate)

```{r Cni_Cre, fig.height=15, fig.width=15}
makeOxfordPlots(gbs$Cni_Cre, col = "strand") + theme(legend.position="none") + ggtitle("C. nigoni vs C. remanei")
```

## C. briggsae vs C. elegans (distant)

```{r Cbr_Cel, fig.height=15, fig.width=15}
makeOxfordPlots(gbs$Cbr_Cel, col = "strand") + theme(legend.position="none") + ggtitle("C. briggsae vs C. elegans")
```

## C. nigoni vs C. inopinata (distant)

```{r Cni_Cin, fig.height=15, fig.width=15}
makeOxfordPlots(gbs$Cni_Cin, col = "strand") + theme(legend.position="none") + ggtitle("C. nigoni vs C. inopinata")
```

# Mammals

## Human versus mouse chromosome 18

Chromosome 18 was chosen because it is small and it has the same name in both
species.  Region was centered on the alignment with the longest width, as
otherwise the window out of target.

Note that each of these chromosomes is longer than any _Oikopleura dioica_
genome.

```{r hg38_mm10, fig.height=15, fig.width=15}
hgmm <- load_genomic_breaks(system.file(("extdata/hg38_mm10/hg38_chr18__mm10_chr18.gff3"), package = "BreakpointsData"))
makeOxfordPlots(hgmm, col = "strand") + ggtitle("Homo sapiens vs Mus musculus")
```

## Muntjaks

```{r Mmun_Mree, fig.height=15, fig.width=15}
Mmun_Mree <- load_genomic_breaks(system.file("extdata/Muntjaks/Rrev_Rmun.07.postmasked.gff.gz", package = "BreakpointsData"), type = "match")
makeOxfordPlots(Mmun_Mree |> swap(), col = "strand") + ggtitle("Muntiacus reevesi vs Muntiacus muntjak")
```
