---
title: "Alignment line plots, whole genomes"
author: 
 - "Charles Plessy"
date: "`r format(Sys.time(), '%d %B, %Y')`"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteEncoding{UTF-8}
  %\VignetteIndexEntry{Alignment line plots, whole genomes}
  %\VignetteEngine{knitr::rmarkdown}
editor_options: 
  chunk_output_type: console
---

```{r setup}
knitr::opts_knit$set(cache = TRUE)
```

# Introduction

This vignette produces alignment line plots of pairs of genomes to illustrate
differences in the extent of scrambling in _Oikopleura_ compared to equivalent
pairs in other genera.

Figure 1 panel A was generated in this vignette.

# Load R pacakges and data

```{r load_packages_and_data}
library('OikScrambling') |> suppressPackageStartupMessages()
ggplot2::theme_set(theme_bw())
genomes <- OikScrambling:::loadAllGenomes()
load("BreakPoints.Rdata")
```

# Oikopleura

## Oki vs Osa

Line plot used for panel A of Figure 1.  Is based on aligned regions
(before coalescing), stored in the `gbs` object.  The manually set colors are
used throughout the manuscript to represent chromosomes 1 and 2, the PAR, the
XSR and the YSR.

```{r Oki_Osa_gbs, dev=c('png', 'svg', 'pdf'), fig.ext=c('png', 'svg', 'pdf')}
makeOxfordPlots(gbs$Oki_Osa, diag = FALSE) +
  scale_color_manual(values = c("#E78AC3", "#A6D854", "#FFD92F", "#8DA0CB", "#66C2A5")) +
  theme_bw() +
  theme(legend.position="none")
```

Same, colored by strand.

```{r Oki_Osa_strand, dev=c('png', 'svg', 'pdf'), fig.ext=c('png', 'svg', 'pdf')}
makeOxfordPlots(gbs$Oki_Osa, diag = FALSE, col = "strand") + theme_bw() + theme(legend.position="none")
```

## Oki vs Bar

```{r Oki_Bar_all}
makeOxfordPlots(gbs$Oki_Bar, diag = FALSE)
```

## Osa vs Bar

```{r Osa_bar}
makeOxfordPlots(gbs$Osa_Bar, diag = FALSE)
```

## Osa vs Oki

```{r Osa_Oki}
makeOxfordPlots(gbs$Osa_Oki, diag = FALSE)
```

## Osa vs Aom

```{r Osa_Aom}
makeOxfordPlots(gbs$Osa_Aom, diag = FALSE)
```

## Oki vs Kum

```{r Oki_Kum}
makeOxfordPlots(gbs$Oki_Kum)
```


# Ciona

## _Ciona intestinalis_ (Plymouth vs Roscoff)

```{r Ply_Ros}
makeOxfordPlots(gbs$Ply_Ros) + ggtitle("Ciona intestinalis Plymouth vs Roscoff")
```

## _Ciona intestinalis_ (Plymouth) vs _Ciona robusta_

```{r Ply_Rob}
makeOxfordPlots(gbs$Ply_Rob) + ggtitle("Ciona intestinalis Plymouth vs Ciona robusta")
```

## Troubleshoot of grid pattern

We see horizontal patterns what are they?  I think they may be pericentromeric
regions.

```{r Ply_Rob_point_plot}
makeOxfordPlots(gbs$Ply_Rob) + ggtitle("Ciona intestinalis Plymouth vs Ciona robusta")
```

```{r Ply_Rob_subset}
coa$Ply_Rob |>
  plyranges::filter(seqnames(query) %in% c("BJTB01000010.1", "BJTB01000011.1", "BJTB01000012.1", "BJTB01000013.1", "BJTB01000014.1")) |>
  plyranges::filter(seqnames        %in% c("BNJZ01000005.1", "BNJZ01000006.1", "BNJZ01000007.1", "BNJZ01000008.1", "BNJZ01000009.1")) |>
  makeOxfordPlots() + ggtitle("C. int. P vs C. rob., subset near BJTB01000012.1 and BNJZ01000007.1")
```

```{r Ply_Rob_subset2}
coa$Ply_Rob |>
  plyranges::filter(seqnames(query) %in% c("BJTB01000011.1", "BJTB01000012.1", "BJTB01000013.1")) |>
  plyranges::filter(seqnames        %in% c("BNJZ01000006.1", "BNJZ01000007.1", "BNJZ01000008.1")) |>
  makeOxfordPlots() + ggtitle("C. int. P vs C. rob., subset near (closer) BJTB01000012.1 and BNJZ01000007.1")
```

```{r Ply_Rob_subset3}
coa$Ply_Rob |>
  plyranges::filter(seqnames(query) %in% c("BJTB01000002.1", "BJTB01000003.1", "BJTB01000004.1", "BJTB01000005.1", "BJTB01000006.1")) |>
  plyranges::filter(seqnames        %in% c("BNJZ01000010.1", "BNJZ01000011.1", "BNJZ01000012.1", "BNJZ01000013.1", "BNJZ01000014.1")) |>
  makeOxfordPlots() + ggtitle("C. int. P vs C. rob., subset near BJTB01000004.1 and BNJZ01000012.1")
```

```{r Ply_Rob_subset4}
coa$Ply_Rob |>
  plyranges::filter(seqnames(query) %in% c("BJTB01000003.1", "BJTB01000004.1", "BJTB01000005.1")) |>
  plyranges::filter(seqnames        %in% c("BNJZ01000011.1", "BNJZ01000012.1", "BNJZ01000013.1")) |>
  makeOxfordPlots() + ggtitle("C. int. P vs C. rob., subset near (closer) BJTB01000004.1 and BNJZ01000012.1")
```

## _Ciona intestinalis_ (Plymouth) vs _Ciona savignyi_

```{r Ply_Sav}
makeOxfordPlots(gbs$Ply_Sav) + ggtitle("Ciona intestinalis Plymouth vs Ciona savignyi")
```

# Drosophila

## vs Drosophila mauritania

```{r Dmel_Dmau}
makeOxfordPlots(gbs$Dme_Dma) + ggtitle("Dmel vs Dmau")
```

## vs Drosophila yakuba

```{r Dmel_Dyak}
makeOxfordPlots(gbs$Dme_Dya) + ggtitle("Dmel vs Dyak")
```

## vs Drosophila subpulchrella

```{r Dmel_Dsub}
makeOxfordPlots(gbs$Dme_Dsu) + ggtitle("Dmel vs Dsub")
```

## vs Drosophila buskii

```{r Dmel_Dbus}
makeOxfordPlots(gbs$Dme_Dbu) + ggtitle("Dmel vs Dbus")
```

# Mammals

## Human versus mouse

Chromosome 18 was chosen because it is small and it has the same name in both
species.  Region was centered on the alignment with the longest width, as
otherwise the window out of target.

Note that each of these chromosomes is longer than any _Oikopleura dioica_
genome.

```{r hg38_mm10, dev='svg', fig.ext='svg'}
hgmm <- load_genomic_breaks(system.file(("extdata/hg38_mm10/hg38_chr18__mm10_chr18.gff3"), package = "BreakpointsData"))
makeOxfordPlots(hgmm, col = "strand") + ggtitle("Homo sapiens vs Mus musculus")
```
