---
title: "Region of interest: PAC3"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteEncoding{UTF-8}
  %\VignetteIndexEntry{Region of interest: PAC3}
  %\VignetteEngine{knitr::rmarkdown}
editor_options: 
  chunk_output_type: console
---

```{css, echo=FALSE}
pre, code {white-space:pre !important; overflow-x:auto}
```

```{r setup}
knitr::opts_knit$set(cache = TRUE)
options(width = 200)
```

Load packages and data
======================

See `?OikScrambling:::loadAllGenomes()` and `vignette("LoadGenomicBreaks",
package = "OikScrambling")` for how the different objects are prepared.

```{r load_packages_and_data}
library('OikScrambling')      |> suppressPackageStartupMessages()
library('GenomicFeatures')    |> suppressPackageStartupMessages()
genomes     <- OikScrambling:::loadAllGenomes()
transcripts <- OikScrambling:::loadAllTranscriptsGR(compat=F) |> suppressWarnings()
annots      <- OikScrambling:::loadAllAnnotations()   |> suppressWarnings()
genes       <- sapply(annots, function(x) sort(genes(x))) |> SimpleList()
operons     <- lapply(annots, \(a) {a |> genes() |> OikScrambling:::calcOperons(window = 500) }) |> SimpleList() |> suppressWarnings()
#operons$Nor <- OikScrambling:::calcOperons(window = 500, transcripts(annots$Nor) |> reduce()) |> suppressWarnings()

load("BreakPoints.Rdata")
library("purrr")              |> suppressPackageStartupMessages()
library("dplyr")              |> suppressPackageStartupMessages()
library("stringr")            |> suppressPackageStartupMessages()
#library("Repitools")
library("genoPlotR")          |> suppressPackageStartupMessages()
```


# PAC3 moved between chr1 and XSR; its ancestral state is unknown

Annotated versions of the alignments can be found in the `PAC3.svg` file that
should be in the same directory as the `PAC3.Rmd` source file of this document.

## General information.

The gene g3146 (`Oidioi.mRNA.OKI2018_I69.chr1.g3144.t1.cds` or `CAG5107096` in
OKI2018, `CBY24606` in OdB3) is on `chr1` in Oki and Kum but on `XSR` in Osa and
Bar.  The OdB3 protein contains a non-specific hit for the pfam10178 (Proteasome
assembly chaperone 3) PAC3 domain, and Psi-BLAST also discovers it.

<details>
<summary>Click here to see a provisional alignment with similar sequences in
other larvacean genomes.  Proper attention to splicing would be needed to make
it more accurate.</summary>
```
O_dio       MHPTTILREIELENEPVALSIMKFAEETLVTISDNGTFGAFHDI-DIA-EMRPGKEPLIT
O_dio_Oska  MHPETILREVEIDNSPVALAIVKFADQTMVTISDNGTFGAFHDI-EIA-EMRPGTDPVVT
O_van       ----SIVESTQIEGVDVSLAIMGFADCTMIAVSDTGTFGSFYKV-QVAKGGHTNQEAVVT
O_alb       ---------LDLENADIHLAAMSFSDCLMLAVSDTGTFGSFYQV-EKSETSHQNIEPVIS
O_lon       ---------IYLGEEKVHIAVMGFGDATMITVSDGALFGPIYKV-TKS--QKGANEPVIT
O_lon_N     --------EIEIDNEVCYISVTQFGDTNFVTISDTGTFGSFYQV-EKK--EKMGNEPIFN
B_sty       ---------FEIDDLVIYFSFMEFDDYVFLALSNTGTFGSFYKV-QKN--FRMDSNPIIS
M_ery       ------IRIIVIDNTEIYLSVMKFGDCNFIVISDTGTFGSFFKI-EKN--FKMDNNPVVN
O_ruf_Y     ------LKRLRLMSQKVHLSSMKFDDHIMLGVSDSGTFWLILSCFFKSTNQHKDKEPVFE


O_dio       IKPF-FGYSDDFTKV------------VCRNLATSLQVKNK-LMISIGLKQEKVNKQNLD
O_dio_Oska  VKPF-FGYSDDFAKVV-----------VCRNLATGLEVKNK-LMISIGLKQERINKNN--
O_van       IDPI-FGINDHYFQVLIR------------------------------------------
O_alb       IRPI-FGVDDQYFQVFLQVLPDSYFKVVCRYLGVHLAESRR-LLVSITLQKDKINKDN--
O_lon       VDGV-FGEQHDYAQVV------------CRHLRLHCRVETEQLLASVLLKRDLVTREN--
O_lon_N     IQPI-FGQAEDYLQVCKFQIETS--IVVCRYLCQNCTLQGPM-MVSVILSREKVNKKN--
B_sty       TEQI-FGNKKDLLEVINK------------------------------------------
M_ery       IEPI-FGENKDYLQV---------------------------------------------
O_ruf_Y     IVPIFFGCDDQYFQV---------------------------------------------
```
</details>
<br></br>

Private ZENBU links:

 - [chr1 on Oki](https://fantom.gsc.riken.jp/zenbu/gLyphs/#config=nBuUjj3vMBkuTVQnE57TC;loc=OKI2018_I69_1.0::chr1:11497331..11498497+)
 - [XSR on Oki](https://fantom.gsc.riken.jp/zenbu/gLyphs/#config=nBuUjj3vMBkuTVQnE57TC;loc=OKI2018_I69_1.0::XSR:6230961..6232916+)
 - [Chr1 on Osa](https://fantom.gsc.riken.jp/zenbu/gLyphs/#config=lc9rquqRE5PnyllQ68DSiC;loc=OSKA2016v1.9::Chr1:8150980..8160990+)
 - [XSR on Osa](https://fantom.gsc.riken.jp/zenbu/gLyphs/index_3.0.html#config=lc9rquqRE5PnyllQ68DSiC;loc=OSKA2016v1.9::XSR:3167071..3168763+)

## PAC3 in five genome pairs

PAC3 is on range number 2 in all pairs except for Oki_Kum since these two are
syntenic.

```{r PAC3}
PAC3 <- GRanges("chr1:11497331-11498497")
PAC3_in_Kum <- GRanges("contig_3_1:3247998-3248575")

coa$Oki_Osa |> subsetByOverlaps(PAC3)
coa$Oki_Aom |> subsetByOverlaps(PAC3)
coa$Oki_Bar |> subsetByOverlaps(PAC3)
coa$Oki_Nor |> subsetByOverlaps(PAC3)
coa$Oki_Kum |> subsetByOverlaps(PAC3)
```

# PAC3 on chr1

## The bridge region in the "Northern" genomes.

On chromosome 1, PAC3 is flanked by _UBXN6_ and a gene ressembling _mediator
of RNA polymerase II transcription subunit 30_.  In genomes where PAC3 is on the
XSR, these flanking genes are separated by a short bridge region, where
little similarity remains between the Atlantic and Pacific branches of the "North"
species.  The conserved `AG` residues are trans-splicing sites according to
CAGE data.

```{r PAC3_bridge}
PAC3_bridge_Osa <- coa$Oki_Osa |> subsetByOverlaps(PAC3) |> swap() |> cleanGaps() |> plyranges::mutate(strand = '-')
PAC3_bridge_Aom <- coa$Oki_Aom |> subsetByOverlaps(PAC3) |> swap() |> cleanGaps() |> plyranges::mutate(strand = '-')
PAC3_bridge_Bar <- coa$Oki_Bar |> subsetByOverlaps(PAC3) |> swap() |> cleanGaps()
PAC3_bridge_Nor <- coa$Oki_Nor |> subsetByOverlaps(PAC3) |> swap() |> cleanGaps()

list( Osa = PAC3_bridge_Osa
    , Aom = PAC3_bridge_Aom
    , Bar = PAC3_bridge_Bar
    , Nor = PAC3_bridge_Nor) |> lapply(\(gb) gb + 0)  |> lapply(getSeq) |> lapply(unlist) |> as("DNAStringSet") |> msa::msaClustalW() |> as("DNAMultipleAlignment")
```

## The left region of PAC3 in all genomes

Alignment of 70 nucleotides flanking the alignment breakpoint on the left
side of PAC3 illustrates the conservation in all 6 genomes (upstream)
and the lack of similarity between the "Oki" sequences and the "North" bridge
region.

```{r PAC3_left}
PAC3_left_Oki_Osa  <- coa$Oki_Osa |> subsetByOverlaps(PAC3) |> plyranges::slice(1) |> get_bps("right")
PAC3_left_Kum_Osa  <- coa$Osa_Kum |> swap () |> subsetByOverlaps(PAC3_in_Kum + 1) |> sort(i=T) |> plyranges::slice(1) |> get_bps("right")
PAC3_left_Osa      <- coa$Oki_Osa |> subsetByOverlaps(PAC3) |> swap() |> plyranges::slice(1) |> get_bps("left") |> plyranges::mutate(strand = '-')
PAC3_left_Aom      <- coa$Oki_Aom |> subsetByOverlaps(PAC3) |> swap() |> plyranges::slice(1) |> get_bps("left") |> plyranges::mutate(strand = '-')
PAC3_left_Oki_Bar  <- coa$Oki_Bar |> subsetByOverlaps(PAC3) |> plyranges::slice(1) |> get_bps("right")
PAC3_left_Bar      <- coa$Oki_Bar |> subsetByOverlaps(PAC3) |> swap()  |> plyranges::slice(1) |> get_bps("right")
PAC3_left_Nor      <- coa$Oki_Nor |> subsetByOverlaps(PAC3) |> swap()  |> plyranges::slice(1) |> get_bps("right")

# Shifting Bar and Nor so that they all align well

PAC3_left <- 
list( Aom = PAC3_left_Aom
    , Osa = PAC3_left_Osa
    , Oki = PAC3_left_Oki_Osa
    , Kum = PAC3_left_Kum_Osa
    , Bar = PAC3_left_Bar |> shift(20)
    , Nor = PAC3_left_Nor |> shift(20))

PAC3_left |> lapply(\(gb) gb + 70) |> lapply(getSeq) |> lapply(unlist) |> as("DNAStringSet") |> msa::msaClustalW() |> as("DNAMultipleAlignment")
```

## The right region of PAC3 on in all genomes

Same story on the right side.  Very near the alignment break (at the center),
there is the `ATG` and upstream of it there are two conserved `AG` that may be
trans-splicing sites (with CAGE support at least in Oki).

```{r PAC3_right}
PAC3_right_Oki_Osa  <- coa$Oki_Osa |> subsetByOverlaps(PAC3) |> plyranges::slice(3) |> get_bps("left")
PAC3_right_Kum_Osa  <- coa$Osa_Kum |> swap () |> subsetByOverlaps(PAC3_in_Kum + 1) |> sort(i=T) |> plyranges::slice(3) |> get_bps("left")
PAC3_right_Osa      <- coa$Oki_Osa |> subsetByOverlaps(PAC3) |> swap() |> plyranges::slice(3) |> get_bps("right") |> plyranges::mutate(strand = '-')
PAC3_right_Aom      <- coa$Oki_Aom |> subsetByOverlaps(PAC3) |> swap() |> plyranges::slice(3) |> get_bps("right") |> plyranges::mutate(strand = '-')
PAC3_right_Oki_Bar  <- coa$Oki_Bar |> subsetByOverlaps(PAC3)           |> plyranges::slice(3) |> get_bps("left")
PAC3_right_Bar      <- coa$Oki_Bar |> subsetByOverlaps(PAC3) |> swap() |> plyranges::slice(3) |> get_bps("left")
PAC3_right_Nor      <- coa$Oki_Nor |> subsetByOverlaps(PAC3) |> swap() |> plyranges::slice(3) |> get_bps("left")

PAC3_right <-
list( Aom = PAC3_right_Aom
    , Osa = PAC3_right_Osa
    , Oki = PAC3_right_Oki_Osa
    , Kum = PAC3_right_Kum_Osa
    , Bar = PAC3_right_Bar
    , Nor = PAC3_right_Nor)
PAC3_right |> lapply(\(gb) gb + 70) |> lapply(getSeq) |> lapply(unlist) |> as("DNAStringSet") |> msa::msaClustalW() |> as("DNAMultipleAlignment") 
```

## Combined view of the left and right regions

The combined view shows that the sequence on the left and right of PAC3 do not
have visible similarity with each other or with the bridge region.

```{r PAC3_left_right}
PAC3_left_right <- list(
  Aom  = range(PAC3_left_Aom, PAC3_right_Aom),
  Osa  = range(PAC3_left_Osa, PAC3_right_Osa),
  OkiL = PAC3_left_Oki_Osa,
  OkiR = PAC3_right_Oki_Osa,
  KumL = PAC3_left_Kum_Osa,
  KumR = PAC3_right_Kum_Osa,
  Bar  = range(PAC3_left_Bar |> shift(20), PAC3_right_Bar),
  Nor  = range(PAC3_left_Nor |> shift(20), PAC3_right_Nor)
)
PAC3_left_right.aln <- PAC3_left_right |>
  lapply(\(gb) gb + 50) |>
  lapply(getSeq) |>
  lapply(unlist) |>
  as("DNAStringSet") |>
  msa::msaClustalOmega() |>
  as("DNAMultipleAlignment")

# Re-order based on my preference
PAC3_left_right.aln@unmasked[c(1,2,5,6,7,8,3,4)]
```

# PAC3 on the X-specific region of chrX (XSR)

## The PAC3 (North) or bridge region (Oki) on the XSR.

In Okinawa, the bridge region contains a bidirectional promoter.

Flanking genes might be a _leucine-rich repeat domain-containing protein_ on the
left (same operon) and a _glutamine-tRNA ligase_ on the right (different strand).

```{r PAC3_XSR}
PAC3_XSR_Osa_range <- coa$Oki_Osa |> subsetByOverlaps(PAC3) |> range()
PAC3_XSR_Aom_range <- coa$Oki_Aom |> subsetByOverlaps(PAC3) |> range()
PAC3_XSR_Bar_range <- coa$Oki_Bar |> subsetByOverlaps(PAC3) |> range()
PAC3_XSR_Nor_range <- coa$Oki_Nor |> subsetByOverlaps(PAC3) |> range()

PAC3_XSR_Osa    <- coa$Oki_Osa |> subsetByOverlaps(PAC3) |> plyranges::slice(2) |> swap()
PAC3_XSR_Aom    <- coa$Oki_Aom |> subsetByOverlaps(PAC3) |> plyranges::slice(2) |> swap()
PAC3_XSR_Bar    <- coa$Oki_Bar |> subsetByOverlaps(PAC3) |> plyranges::slice(2) |> swap()
PAC3_XSR_Nor    <- coa$Oki_Nor |> subsetByOverlaps(PAC3) |> plyranges::slice(2) |> swap()


(PAC3_XSR_Osa_Oki_triple <- coa$Osa_Oki |> subsetByOverlaps(PAC3_XSR_Osa +1000, ignore.strand = TRUE) |> sort(i=T))
(PAC3_XSR_Osa_Kum_triple <- coa$Osa_Kum |> subsetByOverlaps(PAC3_XSR_Osa +1000, ignore.strand = TRUE) |> sort(i=T))
(PAC3_XSR_Aom_Oki_triple <- coa$Oki_Aom |> swap () |> subsetByOverlaps(PAC3_XSR_Aom + 1000, ignore.strand=T) |> sort(i=T))

# Bar-Oki alignment is in the opposite orientation
(PAC3_XSR_Bar_Oki_triple <- coa$Bar_Oki |> subsetByOverlaps(PAC3_XSR_Bar +1000, ignore.strand = TRUE) |> sort(i=T))

(PAC3_XSR_Nor_Oki_triple <- coa$Oki_Nor |> swap () |> subsetByOverlaps(PAC3_XSR_Nor + 1000, ignore.strand=T) |> sort(i=T))

coa$Osa_Nor |> subsetByOverlaps(PAC3_XSR_Osa)

PAC3_XSR_Osa_left  <- PAC3_XSR_Osa_Oki_triple[1] |> get_bps(dir = 'r')
PAC3_XSR_Osa_right <- PAC3_XSR_Osa_Oki_triple[3] |> get_bps(dir = 'l')
PAC3_XSR_Oki_left  <- PAC3_XSR_Osa_Oki_triple[1] |> swap() |> get_bps(dir = 'r')
PAC3_XSR_Oki_right <- PAC3_XSR_Osa_Oki_triple[3] |> swap() |> get_bps(dir = 'l')

PAC3_XSR_Kum_left  <- PAC3_XSR_Osa_Kum_triple[1] |> swap() |> get_bps(dir = 'r')
PAC3_XSR_Kum_right <- PAC3_XSR_Osa_Kum_triple[3] |> swap() |> get_bps(dir = 'l')

PAC3_XSR_Nor_left  <- PAC3_XSR_Nor_Oki_triple[1] |> get_bps(dir = 'r')
PAC3_XSR_Nor_right <- PAC3_XSR_Nor_Oki_triple[3] |> get_bps(dir = 'l')

# Remember that Bar-Oki alignment is in the opposite orientation
PAC3_XSR_Bar_left  <- PAC3_XSR_Bar_Oki_triple[3] |> get_bps(dir = 'l') |> plyranges::mutate(strand = '-')
PAC3_XSR_Bar_right <- PAC3_XSR_Bar_Oki_triple[1] |> get_bps(dir = 'r') |> plyranges::mutate(strand = '-')

# Remember that Oki-Aom alignment is in the opposite orientation
PAC3_XSR_Aom_left  <- PAC3_XSR_Aom_Oki_triple[3] |> get_bps(dir = 'l') |> plyranges::mutate(strand = '-')
PAC3_XSR_Aom_right <- PAC3_XSR_Aom_Oki_triple[1] |> get_bps(dir = 'r') |> plyranges::mutate(strand = '-')

PAC3_XSR_Aom    <- coa$Oki_Aom |> subsetByOverlaps(PAC3) |> plyranges::slice(2) |> swap()
PAC3_XSR_Bar    <- coa$Oki_Bar |> subsetByOverlaps(PAC3) |> plyranges::slice(2) |> swap()
PAC3_XSR_Nor    <- coa$Oki_Nor |> subsetByOverlaps(PAC3) |> plyranges::slice(2) |> swap()
```

## Left side

In Okinawa and Kume, the bridge region is ~400 bp-long.  It will not be possible
to make a combined left-right alignment like on chr1.

```{r XSR_left}
n <- -40
PAC3_XSR_Osa.aln.left <- list(
  Osa_left = PAC3_XSR_Osa_left |> shift(n),
  Aom_left = PAC3_XSR_Aom_left |> shift(-n),
  Oki_left = PAC3_XSR_Oki_left |> shift(n),
  Kum_left = PAC3_XSR_Kum_left |> shift(n),
  Nor_left = PAC3_XSR_Nor_left |> shift(n),
  Bar_left = PAC3_XSR_Bar_left |> shift(-n))

PAC3_XSR_Osa.aln.left |> lapply(\(gb) gb + 90) |> lapply(getSeq) |> lapply(unlist) |> as("DNAStringSet") |> msa::msaClustalW() |> as("DNAMultipleAlignment") 
```

## Right side

The right-side region is near a bidirectional promoter in both the Oki and Osa
genomes.

```{r XSR_right}
PAC3_XSR_Osa.aln.right <- list(
  Osa_right = PAC3_XSR_Osa_right,
  Aom_right = PAC3_XSR_Aom_right,
  Oki_right = PAC3_XSR_Oki_right,
  Kum_right = PAC3_XSR_Kum_right,
  Nor_right = PAC3_XSR_Nor_right |> shift(-306),
  Bar_right = PAC3_XSR_Bar_right |> shift(306))

PAC3_XSR_Osa.aln.right |> lapply(\(gb) gb + 90) |> lapply(getSeq) |> lapply(unlist) |> as("DNAStringSet") |> msa::msaClustalW() |> as("DNAMultipleAlignment") 
```

# PAC3 context diagram
```{r PAC3Figure, fig.height=6, fig.width=12, message=FALSE, warning=FALSE, dev=c("svg", "png")}
# Section of the genome that is shared the Okinawa's PAC3
coa$Oki_Osa |> subsetByOverlaps(PAC3)
coa$Oki_Bar |> subsetByOverlaps(PAC3)
coa$Oki_Kum |> subsetByOverlaps(PAC3)

extract_gene_coords <- function(gr, gene_name, upstream=0, downstream=0) {
  matching_gene_index <- which(mcols(gr)$gene_id == gene_name)
  sgr <- gr[(matching_gene_index-upstream):(matching_gene_index+downstream),]

  sgr
}

# Get the operon of a gene of interest, with an option buffer (i.e., a number of
# operons that are encoded up or downstream).
operon_of_gene <- function(operon, gene_of_interest, buffer=0) {
  index <- unlist(lapply(operon$gene_id, function(x) gene_of_interest %in% x)) |> which()
  operon_start <- index-buffer
  operon_end <- index+buffer
  if(buffer == 0){
    matching_operon <- operon[index]
  } else {
    matching_operon <- operon[operon_start:operon_end]
  }
  matching_operon
}
(operon_of_gene(operons$Oki, 'g3146'))

genes_of_operon <- function(operon, gene, gene_of_interest=NULL, buffer=0) {
  if(!is.null(gene_of_interest)) {
    gene_ids <- operon$gene_id[which(unlist(lapply(operon$gene_id, function(x) gene_of_interest %in% x)))] |> unlist()
    if(buffer==0){
      matching_genes <- gene[gene$gene_id %in% gene_ids]
    } else {
      m1 <- min(which(gene$gene_id %in% gene_ids))-buffer
      m2 <- max(which(gene$gene_id %in% gene_ids))+buffer
      matching_genes <- gene[m1:m2]
    }
  } else {
    gene_ids <- unlist(operon$gene_id)
    matching_genes <- gene[gene$gene_id %in% gene_ids]
  }
  matching_genes
}
(genes_of_operon(operons$Oki, genes$Oki, 'g3146'))
(genes_of_operon(operons$Oki, genes$Oki, 'g3146', buffer=10))


gr2dna_seg <- function (gr, gene_of_interest=NULL, gene_of_interest_colour='red', ...) {
  # DNA segments need unique names
  if (length(gr) == 0) return(NULL)
  if (is.null(names(gr))) names(gr) <- as.character(gr)
  # Unstranded ranges are not recognized
  strand(gr[strand(gr) == "*"]) <- "+"
  df <- data.frame(
    name   = names(gr),
    start  = start(gr),
    end    = end(gr),
    strand = strand(gr)
  )

  dna_seg <- genoPlotR::dna_seg(df, ...)
  dna_seg$col <- 'black'
  dna_seg$fill <- '#595959'
  if(!is.null(gene_of_interest)){
    dna_seg$fill[which(dna_seg$name==gene_of_interest)] <- gene_of_interest_colour
  }
  dna_seg
}
gr2dna_seg(genes_of_operon(operons$Oki, genes$Oki, 'g3146', buffer=10))

op2dna_seg <- function (gene, operon, gene_of_interest=NULL, buffer=0, gene_fill="#d9d9d9", operon_fill="#595959", gene_of_interest_colour='red', ...) {
  gr <- genes_of_operon(operon, gene, gene_of_interest, buffer=buffer)

  # DNA segments need unique names
  if (length(gr) == 0) return(NULL)
  if (is.null(names(gr))) names(gr) <- as.character(gr)
  # Unstranded ranges are not recognized
  strand(gr[strand(gr) == "*"]) <- "+"
  df <- data.frame(
    name   = names(gr),
    start  = start(gr),
    end    = end(gr),
    strand = strand(gr)
  )

  dna_seg <- genoPlotR::dna_seg(df, ...)
  dna_seg$col <- 'black'
  dna_seg$fill <- gene_fill
  if(!is.null(gene_of_interest)){
    op_gr <- genes_of_operon(operon, gene, gene_of_interest, buffer=0)
    op_gene_ids <- op_gr$gene_id
    dna_seg$fill[which(dna_seg$name %in% op_gene_ids)] <- operon_fill
    dna_seg$fill[which(dna_seg$name==gene_of_interest)] <- gene_of_interest_colour
  }
  dna_seg
}
op2dna_seg(genes$Oki, operons$Oki, gene_of_interest='g3146', buffer=3)

dnaseg_to_annot <- function(ds, ...) {
  annot <- annotation(x1=middle(ds), text=ds$name, ...)
  annot
}

```

## PAC3 context diagram (original annotations)
```{r pac3_operon_diagram_1, dependson="chromosome_plots_operons", fig.height=6, fig.width=12, message=FALSE, warning=FALSE, dev=c("svg", "png")}
pac3_operon_oki     <- operon_of_gene(operons$Oki, 'g3146')
pac3_operon_osa     <- operon_of_gene(operons$Osa, 'g12442')
not_pac3_operon_osa <- operon_of_gene(operons$Osa, 'g2061')

#pac3_operon_oki_genes     <- genes_of_operon(pac3_operon_oki, genes$Oki, gene_of_interest = 'g3146', buffer = 4)
pac3_operon_oki_genes     <- extract_gene_coords(genes$Oki, 'g3146', upstream=5, downstream=8)
#pac3_operon_osa_genes     <- genes_of_operon(pac3_operon_osa, genes$Osa, gene_of_interest = 'g12442', buffer = 1)
pac3_operon_osa_genes     <- extract_gene_coords(genes$Osa, 'g12442', upstream=1, downstream=1)
#not_pac3_operon_osa_genes <- genes_of_operon(not_pac3_operon_osa, genes$Osa, gene_of_interest = 'g2061', buffer = 5)
not_pac3_operon_osa_genes <- extract_gene_coords(genes$Osa, 'g2061', upstream=8, downstream=7)

#pac3_operon_oki_genes_seg <- op2dna_seg(genes$Oki, operons$Oki, 'g3146')
pac3_operon_oki_genes_seg <- gr2dna_seg(pac3_operon_oki_genes)
pac3_operon_oki_genes_seg$fill[which(pac3_operon_oki_genes_seg$name == 'g3146')] <- "red"
pac3_operon_oki_genes_seg$name[which(pac3_operon_oki_genes_seg$name == 'g3146')] <- "PAC3\ng3146"
pac3_operon_oki_genes_seg_annot <- pac3_operon_oki_genes_seg |> dnaseg_to_annot(rot=0, col='grey80')
pac3_operon_oki_genes_seg_annot$color[which(pac3_operon_oki_genes_seg_annot$text == "PAC3\ng3146")] <- '#595959'

not_pac3_operon_osa_genes_seg <- not_pac3_operon_osa_genes |> gr2dna_seg() |> reverse()
not_pac3_operon_osa_genes_seg_annot <- dnaseg_to_annot(not_pac3_operon_osa_genes_seg, rot=0, col='grey80')

pac3_operon_osa_genes_seg <- pac3_operon_osa_genes |> gr2dna_seg() |> reverse()
pac3_operon_osa_genes_seg$fill[pac3_operon_osa_genes_seg$name == "g12442"] <- "red"
pac3_operon_osa_genes_annot <- dnaseg_to_annot(pac3_operon_osa_genes_seg, rot=0, col='grey80')
pac3_operon_osa_genes_annot$color[pac3_operon_osa_genes_annot$text== "g12442"] <- "#595959"
pac3_operon_osa_genes_annot$text[pac3_operon_osa_genes_annot$text == "g12442"] <- "PAC3\ng12442"

plot_gene_map(
  dna_segs = list(
    `Osaka\nChr1: 8145427 - 8174791 (-)`=not_pac3_operon_osa_genes_seg,
    `Okinawa\nchr1: 11487473 - 11503891 (+)`=pac3_operon_oki_genes_seg,
    `Osaka\nXSR: 11487473 - 11503891 (-)`=pac3_operon_osa_genes_seg
  ),
  dna_seg_scale = T, 
  annotations = list(
    not_pac3_operon_osa_genes_seg_annot,
    pac3_operon_oki_genes_seg_annot,
    pac3_operon_osa_genes_annot
  ),
  comparisons = list(
     gb2comp(coa$Oki_Osa |> subsetByOverlaps(PAC3) |> swap()) |> reverse(side=1),
     gb2comp(coa$Oki_Osa |> subsetByOverlaps(PAC3)) |> reverse(side = 2)
   ),
  main = "The PAC3 gene in operon is conserved between Osaka and Barcelona in XSR chromosome but on Chr1 in Okinawa"
)

```


## Adding Norway to the PAC3 context diagram
It is necessary to establish some baseline truths about what genes are present or absent in these operons, in which it seems that Okinawa has translocated the PAC3 gene. To do so, add Norway to this diagram, showing where PAC3 might be and where the not-PAC3 might be.

Note that the comparisons below are based on `coa` and therefore the boundaries of the coalesced alignments do not necessarily correspond nicely to gene boundaries.

```{r pac3_operon_diagram_2, dependson="chromosome_plots_operons", fig.height=6, fig.width=12, message=FALSE, warning=FALSE, dev=c("svg", "png")}
# Find the corresponding PAC3 gene in Norway.
pac3_nor_locus <- coa$Oki_Nor |> subsetByOverlaps(pac3_operon_oki, ignore.strand=T) |> swap()
(pac3_nor_locus)
# The first and third ranges are the not-PAC3 of Osaka, and the second is PAC3.
# The first selection yields just the operons that overlap with the Osa-Nor PAC3-containing operon.
# The second yields a larger area.
# pac3_nor_operon_notpac3 <- operons$Nor |> subsetByOverlaps(c(pac3_nor_locus[1], pac3_nor_locus[3]), ignore.strand=T) |> genes_of_operon(genes$Nor)
pac3_nor_operon_notpac3 <- extract_gene_coords(genes$Nor, 'GSOIDT00009294001', upstream=7, downstream=8)
(pac3_nor_operon_notpac3)
#pac3_nor_operon_pac3 <- operons$Nor |> subsetByOverlaps(pac3_nor_locus[2], ignore.strand=T) |> genes_of_operon(genes$Nor)
pac3_nor_operon_pac3 <- extract_gene_coords(genes$Nor, 'GSOIDT00012497001', upstream=5, downstream=1)
(pac3_nor_operon_pac3)
# Thus GSOIDT00012497001 is the likely orthologue of PAC3 in Norway.

# Make Norway's DNAseg and annotation objects.
pac3_nor_operon_notpac3_seg <- pac3_nor_operon_notpac3 |> gr2dna_seg()
pac3_nor_operon_notpac3_annot <- pac3_nor_operon_notpac3_seg |> dnaseg_to_annot(col='grey80')
#pac3_nor_operon_notpac3_annot$rot <- 45

pac3_nor_operon_pac3_seg <- pac3_nor_operon_pac3 |> gr2dna_seg()
pac3_nor_operon_pac3_seg <- pac3_nor_operon_pac3_seg |> reverse()
pac3_nor_operon_pac3_seg$fill[which(pac3_nor_operon_pac3_seg$name=='GSOIDT00012497001')] <- 'red'
pac3_nor_operon_pac3_seg_annot <- pac3_nor_operon_pac3_seg |> dnaseg_to_annot(col='grey80')
pac3_nor_operon_pac3_seg_annot$rot <- 45
pac3_nor_operon_pac3_seg_annot$col[which(pac3_nor_operon_pac3_seg_annot$text=='GSOIDT00012497001')] <- '#595959'
pac3_nor_operon_pac3_seg_annot$text[which(pac3_nor_operon_pac3_seg_annot$text=='GSOIDT00012497001')] <- 'PAC3 - GSOIDT00012497001'

# Simple function to overwrite the colours of a comparison object. Needed for hacky things I do.
comp_recolor <- function(comp, col='darksalmon'){
  comp$color <- col
  comp$col <- col
  comp
}

# Note that the comparisons here become very complicated. 
plot_gene_map(
  dna_segs = list(
    `Norway\nscaffold_3: 1859411 - 1879701 (+)`=pac3_nor_operon_notpac3_seg,
    `Osaka\nChr1: 8145427 - 8174791 (-)`=not_pac3_operon_osa_genes_seg,
    `Okinawa\nchr1: 11487473 - 11503891 (+)`=pac3_operon_oki_genes_seg,
    `Osaka\nXSR: 11487473 - 11503891 (-)`=pac3_operon_osa_genes_seg,
    `Norway\nscaffold_42: 104419 - 106562 (-)`=pac3_nor_operon_pac3_seg
  ),
  annotations = list(
    pac3_nor_operon_notpac3_annot,
    not_pac3_operon_osa_genes_seg_annot,
    pac3_operon_oki_genes_seg_annot,
    pac3_operon_osa_genes_annot,
    pac3_nor_operon_pac3_seg_annot
  ),
  comparisons = list(
    gb2comp(coa$Osa_Nor |> swap() |> subsetByOverlaps(pac3_nor_operon_notpac3, ignore.strand = T)) |> reverse(side=2),
    gb2comp(coa$Oki_Osa |> subsetByOverlaps(PAC3) |> swap()) |> reverse(side=1),
    gb2comp(coa$Oki_Osa |> subsetByOverlaps(PAC3)) |> reverse(side = 2),
    gb2comp(coa$Osa_Nor |> swap() |> subsetByOverlaps(pac3_nor_operon_pac3, ignore.strand = T) |> swap()) |> reverse(side=1) |> reverse(side=2) |> comp_recolor(col='cornflowerblue')
  ),
  dna_seg_scale = T
)
```

## Adding orthologous pair annotations to PAC3 context diagram
Just to make sure that genes are associated with other genes properly, ensure that the gene IDs computed with orthology finding lines up with the comparison here.

Note that this section actually requires the `orthoPairs` object from the `OrthogroupSyntenies`. Much of this code is copied from that vignette. Please see that vignette for a proper description of what these steps are attempting to do.

```{r pac3_operon_diagram_op, cache=TRUE, message=FALSE, warning=FALSE}
orthoGroupFileNames <- SimpleList(
  AOM.5.5f      = "AOM-5-5f.prot.longest.fa_1", 
  Bar2.p4       = "Bar2_p4.Flye.prot.longest.fa_1", 
  Ply           = "C_int_P.prot.longest.fa_1",
  Ros           = "C_int_R.prot.longest.fa_1",
  Int           = "Ciona_intestinalis.Uniprot.rn.fa_1",
  Sav           = "Ciona_savignyi.Uniprot.rn.fa_1",
  KUM.M3.7f     = "KUM-M3-7f.prot.longest.fa_1",
  OKI2018.I69   = "OKI2018_I69.v2.prot.longest.fa_1",
  OSKA2016v1.9  = "OSKA2016v1.9.prot.longest.fa_1",
  OdB3          = "OdB3.v1.0.prot.fa_1.nohaplo"
)

orthoPairToGBreaks <- function(genome1, genome2, transcripts, treeName="N19", HOGs=NULL) {
  if(is.null(HOGs))
    HOGs <- OikScrambling:::load_one_to_ones(
      system.file(paste0("extdata/OrthoFinder/",treeName,".tsv"), package = "BreakpointsData"),
      c(orthoGroupFileNames[[genome1]], orthoGroupFileNames[[genome2]]))

  IDs2GRanges <- function (IDs, annot) {
    prefix <- Biobase::lcPrefix(IDs)                  # Guess prefix in transcript IDs from HOG files
    not_prefix <- Biobase::lcPrefix(annot$tx_name)    # Remove trailing characters that are part of the name
    not_prefix <- paste0(not_prefix,"$")              # Anchor to end of the string
    prefix <- sub(not_prefix, "", prefix)             # Finalise prefix
    IDs <- sub(prefix, "", IDs)                       # Remove prefix from IDS
    names(annot) <- annot$tx_name                     # Store transcript name in names slot
    gr <- annot[IDs]                                  # Sort by ID
    strand(gr) <- "*"                                 # Make strandless
    gr                                                # Return the object
  }
  
  gb       <- IDs2GRanges(HOGs[,orthoGroupFileNames[[genome1]]], transcripts[[genome1]])
  if (!is.null(genomes[[genome1]]))
    seqinfo(gb) <- seqinfo(genomes[[genome1]])
  gb$query <- IDs2GRanges(HOGs[,orthoGroupFileNames[[genome2]]], transcripts[[genome2]])
  if (!is.null(genomes[[genome2]]))
   seqinfo(gb$query) <- seqinfo(genomes[[genome2]])
  gb <- GenomicBreaks:::GBreaks(gb)
  gb$HOG <- HOGs$HOG
  gb$OG  <- HOGs$OG
  sort(gb)
}

flagLongShort_ <- function(gr, transcripts) {
  genome <- unique(genome(gr))
  flagLongShort(gr, transcripts[[genome]])
}

orthoPairToGBreaks_all_Oiks <- function(treeName=NULL, HOGs=NULL) {
  orthoPairs <- SimpleList()
  orthoPairs$Oki_Osa <- orthoPairToGBreaks("OKI2018.I69", "OSKA2016v1.9",   transcripts, treeName, HOGs)
  orthoPairs$Oki_Bar <- orthoPairToGBreaks("OKI2018.I69", "Bar2.p4",        transcripts, treeName, HOGs)
  orthoPairs$Oki_Kum <- orthoPairToGBreaks("OKI2018.I69", "KUM.M3.7f",      transcripts, treeName, HOGs)
  orthoPairs$Oki_Aom <- orthoPairToGBreaks("OKI2018.I69", "AOM.5.5f",       transcripts, treeName, HOGs)
  orthoPairs$Oki_Nor <- orthoPairToGBreaks("OKI2018.I69", "OdB3",           transcripts, treeName, HOGs)
  
  orthoPairs$Osa_Bar <- orthoPairToGBreaks("OSKA2016v1.9",   "Bar2.p4",     transcripts, treeName, HOGs)
  orthoPairs$Osa_Oki <- orthoPairToGBreaks("OSKA2016v1.9",   "OKI2018.I69", transcripts, treeName, HOGs)
  orthoPairs$Osa_Kum <- orthoPairToGBreaks("OSKA2016v1.9",   "KUM.M3.7f",   transcripts, treeName, HOGs)
  orthoPairs$Osa_Aom <- orthoPairToGBreaks("OSKA2016v1.9",   "AOM.5.5f",    transcripts, treeName, HOGs)
  orthoPairs$Osa_Nor <- orthoPairToGBreaks("OSKA2016v1.9",   "OdB3",        transcripts, treeName, HOGs)
  
  orthoPairs$Bar_Osa <- orthoPairToGBreaks("Bar2.p4",   "OSKA2016v1.9",     transcripts, treeName, HOGs)
  orthoPairs$Bar_Oki <- orthoPairToGBreaks("Bar2.p4",   "OKI2018.I69",      transcripts, treeName, HOGs)
  orthoPairs$Bar_Kum <- orthoPairToGBreaks("Bar2.p4",   "KUM.M3.7f",        transcripts, treeName, HOGs)
  orthoPairs$Bar_Aom <- orthoPairToGBreaks("Bar2.p4",   "AOM.5.5f",         transcripts, treeName, HOGs)
  orthoPairs$Bar_Nor <- orthoPairToGBreaks("Bar2.p4",   "OdB3",             transcripts, treeName, HOGs)
  
  orthoPairs <- sapply(orthoPairs,  flagLongShort_, longShort)
  
  SimpleList(orthoPairs)
}

orthoPairs         <- orthoPairToGBreaks_all_Oiks("N19")
orthoPairs$Ply_Ros <- orthoPairToGBreaks("Ply", "Ros", transcripts, treeName="N20")

orthoPairs_core         <- orthoPairToGBreaks_all_Oiks("N3")
orthoPairs_core$Ply_Ros <- orthoPairToGBreaks("Ply", "Ros", transcripts, treeName="N3")

orthoPairs_one2oneInOiks <-
  orthoPairToGBreaks_all_Oiks(treeName = NULL,
                              HOGs = OikScrambling:::load_one_to_ones(
                                       system.file("extdata/OrthoFinder/N19.tsv", package = "BreakpointsData")))

flagOverlaps <- function(gr) {
  # Find overlaps
  ov <- findOverlaps(gr)
  # Index of self hits in the gr object
  idx <- queryHits(ov[!isSelfHit(ov)])
  # Convert to Boolean indexing the gr object
  seq_along(gr) %in% idx
}

orthoPairsFiltered <- sapply(orthoPairs, function(gb) {
  gb <- gb[!flagOverlaps(gb)]
  gb <- gb[!flagOverlaps(gb$query)]
  gb
}) |> SimpleList()

orthoPairsFiltered_core <- sapply(orthoPairs_core, function(gb) {
  gb <- gb[!flagOverlaps(gb)]
  gb <- gb[!flagOverlaps(gb$query)]
  gb
}) |> SimpleList()

orthoPairs_one2oneInOiks_Filtered <- sapply(orthoPairs_one2oneInOiks, function(gb) {
  gb <- gb[!flagOverlaps(gb)]
  gb <- gb[!flagOverlaps(gb$query)]
  gb
}) |> SimpleList()

coalOrtho <- function(gb) {
  # Coalesce
  coal <- coalesce_contigs(gb)
  
  # Map annotations to the coalesced blocks
  ov <- findOverlaps(coal, gb)
  
  # Extract gene names mapped to each blocks
  coal$geneNames <- CharacterList(split(names(gb)[subjectHits(ov)], queryHits(ov))) |> unname()
  # We can use the `ov` object for query ranges as well because of the way `coal` was constructed.
  coal$query$geneNames <- CharacterList(split(names(gb$query)[subjectHits(ov)], queryHits(ov))) |> unname()
  
  # Extract dNdS values
  coal$dNdS_GUIDANCE2  <- NumericList(split(gb$dNdS_GUIDANCE2 [subjectHits(ov)], queryHits(ov)))
  coal$dNdS_HmmCleaner <- NumericList(split(gb$dNdS_HmmCleaner[subjectHits(ov)], queryHits(ov)))
  coal$dNdS_PRANK      <- NumericList(split(gb$dNdS_PRANK     [subjectHits(ov)], queryHits(ov)))
  # We just copy dNdS values to query ranges because they are the same.
  coal$query$dNdS_GUIDANCE2  <- coal$dNdS_GUIDANCE2
  coal$query$dNdS_HmmCleaner <- coal$dNdS_HmmCleaner
  coal$query$dNdS_PRANK      <- coal$dNdS_PRANK

  # Count genes per syntenic block
  coal$geneNumber <- sapply(coal$geneNames, length)
  
  coal
}

orthoBlocks <- sapply(orthoPairsFiltered[1:15], coalOrtho) |> SimpleList()
orthoBlocks$Ply_Ros <- coalesce_contigs(orthoPairsFiltered$Ply_Ros)
orthoBlocks[ 1:15 ] <- sapply(orthoBlocks[ 1:15 ],  flagLongShort_, longShort)

orthoBlocks_core <- sapply(orthoPairsFiltered_core[1:15], coalOrtho) |> SimpleList()
orthoBlocks_core$Ply_Ros <- coalesce_contigs(orthoPairsFiltered_core$Ply_Ros)
orthoBlocks_core[ 1:15 ] <- sapply(orthoBlocks_core[ 1:15 ],  flagLongShort_, longShort)

orthoBlocks_one2oneInOiks <- sapply(orthoPairs_one2oneInOiks_Filtered, coalOrtho) |> SimpleList()
orthoBlocks_one2oneInOiks <- sapply(orthoBlocks_one2oneInOiks,  flagLongShort_, longShort) |> SimpleList()
```

```{r pac3_operon_diagram_3, dependson="pac3_operon_diagram_op", fig.height=6, fig.width=12, message=FALSE, warning=FALSE, dev=c("svg", "png")}
# Create comparisons based on orthoPairs. Need 4 comparisons:
#  - Norway not-PAC3 to Osaka not-PAC3
#  - Osaka not-PAC3 to Okinawa PAC3
#  - Okinawa PAC3 to Osaka PAC3
#  - Osaka PAC3 to Norway PAC3
op_norNP3_osaNP3      <- orthoPairs_one2oneInOiks$Osa_Nor |> swap() |> subsetByOverlaps( pac3_nor_operon_notpac3 )
(op_norNP3_osaNP3)
# Make a modified gb2comp that adds orthopair names.
gb2comp_op <- function(gb, color = NULL, ignore.strand = FALSE) {
    dna_segs <- gb2dna_seg(gb)
    df <- data.frame(start1 = dna_segs$target$start, end1 = dna_segs$target$end, 
        start2 = dna_segs$query$start, end2 = dna_segs$query$end)
    cmp <- genoPlotR::as.comparison(df)
    cmp$direction <- 1L
    cmp$color <- "darksalmon"
    if (isFALSE(ignore.strand)) {
        onMinus <- decode(strand(gb) == "-")
        cmp$direction[onMinus] <- -1
        cmp$start2[onMinus] <- dna_segs$query$end[onMinus]
        cmp$end2[onMinus] <- dna_segs$query$start[onMinus]
        cmp$col[onMinus] <- "cornflowerblue"
    }
    if (!is.null(color)) {
        cmp$col <- color
    }
    rownames(cmp) <- paste0(names(gb), "__", names(gb$query))
    cmp
}

# Norway not-PAC3 to Osaka not-PAC3
op_norNP3_osaNP3_comp        <- gb2comp_op(op_norNP3_osaNP3) |> reverse(side=2)
op_norNP3_osaNP3_comp$etc    <- op_norNP3_osaNP3_comp$end1
op_norNP3_osaNP3_comp$end1   <- op_norNP3_osaNP3_comp$start1
op_norNP3_osaNP3_comp$start1 <- op_norNP3_osaNP3_comp$etc
op_norNP3_osaNP3_comp$color  <- "cornflowerblue"
op_norNP3_osaNP3_comp$col    <- "cornflowerblue"

# Osaka not-PAC3 to Okinawa PAC3
op_osaNP3_okiP3             <- orthoPairs_one2oneInOiks$Oki_Osa |> subsetByOverlaps( pac3_operon_oki_genes )  |> swap()
op_osaNP3_okiP3_comp        <- gb2comp_op(op_osaNP3_okiP3) |> reverse(side=1)
op_osaNP3_okiP3_comp$etc    <- op_osaNP3_okiP3_comp$end1
op_osaNP3_okiP3_comp$end1   <- op_osaNP3_okiP3_comp$start1
op_osaNP3_okiP3_comp$start1 <- op_osaNP3_okiP3_comp$etc
op_osaNP3_okiP3_comp$color  <- "cornflowerblue"
op_osaNP3_okiP3_comp$col    <- "cornflowerblue"

# Okinawa PAC3 to Osaka PAC3
op_okiP3_osaP3              <- orthoPairs_one2oneInOiks$Oki_Osa |> subsetByOverlaps( pac3_operon_oki_genes )
op_okiP3_osaP3_comp         <- gb2comp_op(op_okiP3_osaP3) |> reverse(side=2)
op_okiP3_osaP3_comp$etc     <- op_okiP3_osaP3_comp$end1
op_okiP3_osaP3_comp$end1    <- op_okiP3_osaP3_comp$start1
op_okiP3_osaP3_comp$start1  <- op_okiP3_osaP3_comp$etc
op_okiP3_osaP3_comp$color   <- "cornflowerblue"
op_okiP3_osaP3_comp$col     <- "cornflowerblue"

# Osaka PAC3 to Norway PAC3
op_osaP3_norP3              <- orthoPairs_one2oneInOiks$Osa_Nor |> subsetByOverlaps( pac3_operon_osa_genes )
op_osaP3_norP3_comp         <- gb2comp_op(op_osaP3_norP3) |> reverse(side=1) |> reverse(side=2)
#op_osaP3_norP3_comp$etc     <- op_osaP3_norP3_comp$end1
#op_osaP3_norP3_comp$end1    <- op_osaP3_norP3_comp$start1
#op_osaP3_norP3_comp$start1  <- op_osaP3_norP3_comp$etc
op_osaP3_norP3_comp$color   <- "cornflowerblue"
op_osaP3_norP3_comp$col     <- "cornflowerblue"

# Note that the comparisons here become very complicated.
plot_gene_map(
  dna_segs = list(
    `Norway\nscaffold_3: 1859411 - 1879701 (+)`=pac3_nor_operon_notpac3_seg,
    `Osaka\nChr1: 8145427 - 8174791 (-)`=not_pac3_operon_osa_genes_seg,
    `Okinawa\nchr1: 11487473 - 11503891 (+)`=pac3_operon_oki_genes_seg,
    `Osaka\nXSR: 11487473 - 11503891 (-)`=pac3_operon_osa_genes_seg,
    `Norway\nscaffold_42: 104419 - 106562 (-)`=pac3_nor_operon_pac3_seg
  ),
  annotations = list(
    pac3_nor_operon_notpac3_annot,
    not_pac3_operon_osa_genes_seg_annot,
    pac3_operon_oki_genes_seg_annot,
    pac3_operon_osa_genes_annot,
    pac3_nor_operon_pac3_seg_annot
  ),
  comparisons = list(
    op_norNP3_osaNP3_comp,
    op_osaNP3_okiP3_comp,
    op_okiP3_osaP3_comp,
    op_osaP3_norP3_comp
  ),
  dna_seg_scale = T
)
```

## Final, manual PAC3 annotations
```{r pac3_operon_diagram_4, dependson="chromosome_plots_operons", fig.height=6, fig.width=12, message=FALSE, warning=FALSE, dev=c("svg", "png")}
# To be done

```

# Ganot operon
Some details about the Ganot operon are available in the `GenomicFeatures` vignette. I use that vignette here to extract the Ganot operon and visualize it.

```{r r ganot_2, fig.height=6, fig.width=12, message=FALSE, warning=FALSE, dev=c("svg", "png")}
# Create a factor to make GRanges bigger - makes it easier to select a region and resize arbitrarily
expansion_factor_oki <- 1.0
ganot_oki_loc    <- GRanges("PAR:15956430-15967745")
ganot_oki_operon <- operons$Oki  |> subsetByOverlaps(ganot_oki_loc |> resize(width=expansion_factor_oki*width(ganot_oki_loc), fix='center'))
# Get genes of the "Ganot operon" in Okinawa from the operon list.
ganot_oki_genes  <- operons$Oki  |> subsetByOverlaps(ganot_oki_loc) |> genes_of_operon(genes$Oki)
ganot_oki_genes <- extract_gene_coords(genes$Oki, gene_name='g13003', upstream=7, downstream=1) # More than one way to skin a cat, as they say...
(ganot_oki_genes)

# Osaka
# Check the corresponding regions in Osaka and Barcelona, using some trickery.
# Some room is added to flank the corresponding genes to allow for missed genes or incorrect
# annotations.
expansion_factor_osa <- 1.0
ganot_osa_chunk    <- coa$Oki_Osa |> subsetByOverlaps(ganot_oki_genes) |> swap()
ganot_osa_operon   <- operons$Osa |> subsetByOverlaps( ganot_osa_chunk, ignore.strand=TRUE)
(ganot_osa_operon) # No matching operons
ganot_osa_genes    <- genes$Osa |> subsetByOverlaps(ganot_osa_chunk, ignore.strand=T)
(ganot_osa_genes)
# Note that many genes are collinear on the negative strand. Why are they not operons? Check the intergenic distances.
ganot_osa_loc      <- GRanges(unique(seqnames(ganot_osa_genes)), ranges=IRanges(start=min(start(ganot_osa_genes)), end=max(end(ganot_osa_genes))))
(ganot_osa_loc)

# Barcelona
ganot_bar        <- coa$Oki_Bar |> subsetByOverlaps(ganot_oki_genes) |> swap()
ganot_bar_operon <- operons$Bar |> subsetByOverlaps( coa$Oki_Bar |> subsetByOverlaps(ganot_oki_genes)  |> swap() )
(ganot_bar_operon) # No matching operons
ganot_bar_genes  <- genes$Bar |> subsetByOverlaps( coa$Oki_Bar |> subsetByOverlaps(ganot_oki_genes, ignore.strand=T)  |> swap(), ignore.strand=T )
(ganot_bar_genes)
# As above, verify the intergenic distances.
ganot_bar_loc   <- GRanges(unique(seqnames(ganot_bar_genes)), ranges=IRanges(start=min(start(ganot_bar_genes)), end=max(end(ganot_bar_genes))))
(ganot_bar_loc)

# Some manual annotations to make things nicer to look at.
oki_seg <- gr2dna_seg(ganot_oki_genes |> GenomicBreaks::reverse())
oki_seg$name <- c('g12994', 'cycDc\ng12997', 'MBF1\ng12998', 'Dtu\ng12999', 'DynL\ng13000', 'RBP16\ng13001', 'IQCK\ng13002', 'POC5*\ng13003*', 'PDCL3\ng13004')
oki_seg$fill <- '#d9d9d9'
oki_seg$fill[2:8] <- palette("Set2")[1:7] |> rev()
#oki_seg$fill[98]   <- "#d9d9d9"
(oki_seg)

# Norway
ganot_nor_genes <- extract_gene_coords(genes$Nor, 'GSOIDT00004452001', upstream=1, downstream=7)
nor_seg <- gr2dna_seg(ganot_nor_genes)
nor_seg$name <- c("PDCL3\nGSOIDT00004451001","POC5\nGSOIDT00004452001", "IQCK\nGSOIDT00004453001", "RBP16\nGSOIDT00004454001", "DynL\nGSOIDT00004455001", "Dtu\nGSOIDT00004456001", "MBF1\nGSOIDT00004457001", "cycDc\nGSOIDT00004458001", "GSOIDT00004460001")
nor_seg$fill <- '#d9d9d9'
nor_seg$fill[2:8] <- palette("Set2")[1:7]
(nor_seg)

# The Osaka operon is the region before g10134 to g10138 in reality. Let's subset what we've got from above.
osa_seg <- gr2dna_seg(ganot_osa_genes)[11:18,]
osa_seg$name <- c("PDCL3\ng10132", "POC5*\ng10133*", "IQCK\ng10134", "RBP16*\ng10135*", "DynL/Dtu*\ng10136*", "MBF1\ng10137", "cycDc\ng10138", "g10140")
osa_seg$fill <- '#d9d9d9'
osa_seg$fill[2:4] <- palette("Set2")[1:4]
osa_seg$fill[6:7] <- palette("Set2")[6:7]
#osa_seg$fill[2] <- palette("Set2")[3] # Osaka does not have a gene model for POC5
#osa_seg$fill[3] <- palette("Set2")[3] # Osaka does not have a gene model for POC5
#osa_seg$fill[5] <- palette("Set2")[6]
#osa_seg$fill[6] <- palette("Set2")[7]

bar_seg <- gr2dna_seg(ganot_bar_genes[11:18])
bar_seg$name <- c("PDCL3\ng8790", "POC5*\ng8791*", "POC5*\ng8792*", "IQCK/RBP16*\ng8793*", "DynL\ng8794", "MBF1\ng8795", "cycDc\ng8796", "g8798")
bar_seg$fill <- "#d9d9d9"
bar_seg$fill[2:3] <- palette("Set2")[1]
bar_seg$fill[5] <- palette("Set2")[4]
bar_seg$fill[6] <- palette("Set2")[6]
bar_seg$fill[7] <- palette("Set2")[7]

plot_gene_map(
  dna_segs = list(
    `Norway\nscaffold_16: 458958-475478 (+)`=nor_seg,
    `Barcelona\nPAR: 9520509-9540505 (+)`=bar_seg,
    `Osaka\nPAR: 11014236-11034479 (+)`=osa_seg,
    `Okinawa\nPAR: 15956430-15967745 (-)`=oki_seg
  ),
  dna_seg_scale = T, 
  annotations = list(
    dnaseg_to_annot(nor_seg, rot=0, col='#595959'),
    dnaseg_to_annot(bar_seg, rot=0, col='#595959'),
    dnaseg_to_annot(osa_seg, rot=0, col='#595959'),
    dnaseg_to_annot(oki_seg, rot=0, col='#595959')
  ),
  main = "Ganot operon in Norway, Barcelona, Osaka, and Okinawa (original annotations)"
)

```

## Ganot - manual annotations
Note that there are several errors in the above annotations. Manually correcting these creates a slightly nicer picture of conservation in the Ganot operon.

The list of changes are tabulated elsewhere, but here they are:

 - Barcelona, POC5: the gene models `g8791` and `g8792` should be merged. An `N` needs to be added at position 761 to recover the Norway sequence.
 - Barcelona, IQCK/RBP16: `g8793` needs to be split into two gene models
 - Barcelona, Dtu: No gene model is found between DynL and cycDc
 - Okinawa, POC5: One exon is missing but can be recovered by adding one `N` at position 1278
 - Osaka, POC5: `g10133` is truncated at the 5' end, requiring two `N`s to be added at positions 39 and 722 to recover. Although two mutations could indicate pseudogenization, a transcript with 99.9% identity to the corrected gene sequence is found in the transcriptome, suggesting it is an error in the assembly (and thus gene model).
 - Osaka, RBP16: `g10135` is truncated at the 5' end and missing a central exon. Recovered by adding an `N` at position 4934
 - Osaka, Dynl/Dtu: `g10136` contains both genes, and aligning the splice boundaries to Norway corrects both.

Incorporating these changes manually results in the following diagram.

```{r ganot_3, fig.height=6, fig.width=12, message=FALSE, warning=FALSE, dev=c("svg", "png")}
# Correct Barcelona
corr_bar_seg <- bar_seg
corr_bar_seg$name[2]  <- 'POC5*\ng8791, g8792*'
corr_bar_seg$start[2] <- 9491186	
corr_bar_seg$end[2]   <- 9493059
corr_bar_seg[3,] <- data.frame(name='IQCK*,\ng8793*', start=9493245, end=9493900, strand=1, col='black', fill=palette("Set2")[2], lty=1, lwd=1, pch=8, cex=1, gene_type="arrows")
corr_bar_seg[4,] <- data.frame(name='RBP16*,\ng8793*', start=9494034, end=9498057, strand=1, col='black', fill=palette("Set2")[3], lty=1, lwd=1, pch=8, cex=1, gene_type="arrows")
#corr_bar_seg <- rbind(corr_bar_seg, data.frame(name='Dtu*', start=9498946, end=9500281, strand=1, col='black', fill=palette("Set2")[5], lty=1, lwd=1, pch=8, cex=1, gene_type="arrows"))
corr_bar_seg[9,] <- data.frame(name='Dtu*', start=9498946, end=9500281, strand=1, col='black', fill=palette("Set2")[5], lty=1, lwd=1, pch=8, cex=1, gene_type="arrows")

(corr_bar_seg)

# Correct Osaka
corr_osa_seg <- osa_seg
corr_osa_seg[2,] <- data.frame(name='POC5*,\ng10133*', start=10983992, end=10985909, strand=1, col='black', fill=palette("Set2")[1], lty=1, lwd=1, pch=8, cex=1, gene_type="arrows")
corr_osa_seg[4,] <- data.frame(name='RBP16*,\ng10135*', start=10986800, end=10990906, strand=1, col='black', fill=palette("Set2")[3], lty=1, lwd=1, pch=8, cex=1, gene_type="arrows")
corr_osa_seg[5,] <- data.frame(name='DynL*,\ng10136*', start=10991263, end=10991569, strand=1, col='black', fill=palette("Set2")[4], lty=1, lwd=1, pch=8, cex=1, gene_type="arrows")
corr_osa_seg[9,] <- data.frame(name='Dtu*,\ng10136*', start=10991720, end=10993085, strand=1, col='black', fill=palette("Set2")[5], lty=1, lwd=1, pch=8, cex=1, gene_type="arrows")
(corr_osa_seg)

# Correct Okinawa
corr_oki_seg <- oki_seg
corr_oki_seg$end[8] <- 17092476-15965688
corr_oki_seg$start[8] <- 17092476-15967583


plot_gene_map(
  dna_segs = list(
    `Norway\nscaffold_16: 458958-475478 (+)`=nor_seg,
    `Barcelona\nPAR: 9520509-9540505 (+)`=corr_bar_seg,
    `Osaka\nPAR: 11014236-11034479 (+)`=corr_osa_seg,
    `Okinawa\nPAR: 15956430-15967745 (-)`=corr_oki_seg
  ),
  dna_seg_scale = T, 
  annotations = list(
    dnaseg_to_annot(nor_seg, rot=0, col='#595959'),
    dnaseg_to_annot(corr_bar_seg, rot=0, col='#595959'),
    dnaseg_to_annot(corr_osa_seg, rot=0, col='#595959'),
    dnaseg_to_annot(corr_oki_seg, rot=0, col='#595959')
  ),
  main = "Ganot operon in Norway, Barcelona, Osaka, and Okinawa (corrected annotations)"
)
```

# Session information

```{r session_information}
sessionInfo()
```




