---
title: "Region of interest: PAC3"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteEncoding{UTF-8}
  %\VignetteIndexEntry{Region of interest: PAC3}
  %\VignetteEngine{knitr::rmarkdown}
editor_options: 
  chunk_output_type: console
---

```{css, echo=FALSE}
pre, code {white-space:pre !important; overflow-x:auto}
```

```{r setup}
knitr::opts_knit$set(cache = TRUE)
options(width = 200)
```

Load packages and data
======================

See `?OikScrambling:::loadAllGenomes()` and `vignette("LoadGenomicBreaks",
package = "OikScrambling")` for how the different objects are prepared.

```{r load_packages_and_data}
library("purrr")              |> suppressPackageStartupMessages()
library('OikScrambling')      |> suppressPackageStartupMessages()
library('GenomicFeatures')    |> suppressPackageStartupMessages()
genomes <- OikScrambling:::loadAllGenomes()
transcripts <- OikScrambling:::loadAllTranscriptsGR()
annots      <- OikScrambling:::loadAllAnnotations()   |> suppressWarnings()
Genes <- sapply(annots, function(x) sort(genes(x))) |> SimpleList()
load("BreakPoints.Rdata")
#Loading this too early breaks the function calkOperons
library("dplyr")              |> suppressPackageStartupMessages()
library("stringr")            |> suppressPackageStartupMessages()
#library("Repitools")
library("genoPlotR")          |> suppressPackageStartupMessages()
```


# PAC3 moved between chr1 and XSR; its ancestral state is unknown

Annotated versions of the alignments can be found in the `PAC3.svg` file that
should be in the same directory as the `PAC3.Rmd` source file of this document.

## General information.

The gene g3146 (`Oidioi.mRNA.OKI2018_I69.chr1.g3144.t1.cds` or `CAG5107096` in
OKI2018, `CBY24606` in OdB3) is on `chr1` in Oki and Kum but on `XSR` in Osa and
Bar.  The OdB3 protein contains a non-specific hit for the pfam10178 (Proteasome
assembly chaperone 3) PAC3 domain, and Psi-BLAST also discovers it.

<details>
<summary>Click here to see a provisional alignment with similar sequences in
other larvacean genomes.  Proper attention to splicing would be needed to make
it more accurate.</summary>
```
O_dio       MHPTTILREIELENEPVALSIMKFAEETLVTISDNGTFGAFHDI-DIA-EMRPGKEPLIT
O_dio_Oska  MHPETILREVEIDNSPVALAIVKFADQTMVTISDNGTFGAFHDI-EIA-EMRPGTDPVVT
O_van       ----SIVESTQIEGVDVSLAIMGFADCTMIAVSDTGTFGSFYKV-QVAKGGHTNQEAVVT
O_alb       ---------LDLENADIHLAAMSFSDCLMLAVSDTGTFGSFYQV-EKSETSHQNIEPVIS
O_lon       ---------IYLGEEKVHIAVMGFGDATMITVSDGALFGPIYKV-TKS--QKGANEPVIT
O_lon_N     --------EIEIDNEVCYISVTQFGDTNFVTISDTGTFGSFYQV-EKK--EKMGNEPIFN
B_sty       ---------FEIDDLVIYFSFMEFDDYVFLALSNTGTFGSFYKV-QKN--FRMDSNPIIS
M_ery       ------IRIIVIDNTEIYLSVMKFGDCNFIVISDTGTFGSFFKI-EKN--FKMDNNPVVN
O_ruf_Y     ------LKRLRLMSQKVHLSSMKFDDHIMLGVSDSGTFWLILSCFFKSTNQHKDKEPVFE


O_dio       IKPF-FGYSDDFTKV------------VCRNLATSLQVKNK-LMISIGLKQEKVNKQNLD
O_dio_Oska  VKPF-FGYSDDFAKVV-----------VCRNLATGLEVKNK-LMISIGLKQERINKNN--
O_van       IDPI-FGINDHYFQVLIR------------------------------------------
O_alb       IRPI-FGVDDQYFQVFLQVLPDSYFKVVCRYLGVHLAESRR-LLVSITLQKDKINKDN--
O_lon       VDGV-FGEQHDYAQVV------------CRHLRLHCRVETEQLLASVLLKRDLVTREN--
O_lon_N     IQPI-FGQAEDYLQVCKFQIETS--IVVCRYLCQNCTLQGPM-MVSVILSREKVNKKN--
B_sty       TEQI-FGNKKDLLEVINK------------------------------------------
M_ery       IEPI-FGENKDYLQV---------------------------------------------
O_ruf_Y     IVPIFFGCDDQYFQV---------------------------------------------
```
</details>
<br></br>

Private ZENBU links:

 - [chr1 on Oki](https://fantom.gsc.riken.jp/zenbu/gLyphs/#config=nBuUjj3vMBkuTVQnE57TC;loc=OKI2018_I69_1.0::chr1:11497331..11498497+)
 - [XSR on Oki](https://fantom.gsc.riken.jp/zenbu/gLyphs/#config=nBuUjj3vMBkuTVQnE57TC;loc=OKI2018_I69_1.0::XSR:6230961..6232916+)
 - [Chr1 on Osa](https://fantom.gsc.riken.jp/zenbu/gLyphs/#config=lc9rquqRE5PnyllQ68DSiC;loc=OSKA2016v1.9::Chr1:8150980..8160990+)
 - [XSR on Osa](https://fantom.gsc.riken.jp/zenbu/gLyphs/index_3.0.html#config=lc9rquqRE5PnyllQ68DSiC;loc=OSKA2016v1.9::XSR:3167071..3168763+)

## PAC3 in five genome pairs

PAC3 is on range number 2 in all pairs except for Oki_Kum since these two are
syntenic.

```{r PAC3}
PAC3 <- GRanges("chr1:11497331-11498497")
PAC3_in_Kum <- GRanges("contig_3_1:3247998-3248575")

coa$Oki_Osa |> subsetByOverlaps(PAC3)
coa$Oki_Aom |> subsetByOverlaps(PAC3)
coa$Oki_Bar |> subsetByOverlaps(PAC3)
coa$Oki_Nor |> subsetByOverlaps(PAC3)
coa$Oki_Kum |> subsetByOverlaps(PAC3)
```

# PAC3 on chr1

## The bridge region in the "Northern" genomes.

On chromosome 1, PAC3 is flanked by _UBXN6_ and a gene ressembling _mediator
of RNA polymerase II transcription subunit 30_.  In genomes where PAC3 is on the
XSR, these flanking genes are separated by a short bridge region, where
little similarity remains between the Atlantic and Pacific branches of the "North"
species.  The conserved `AG` residues are trans-splicing sites according to
CAGE data.

```{r PAC3_bridge}
PAC3_bridge_Osa <- coa$Oki_Osa |> subsetByOverlaps(PAC3) |> swap() |> cleanGaps() |> plyranges::mutate(strand = '-')
PAC3_bridge_Aom <- coa$Oki_Aom |> subsetByOverlaps(PAC3) |> swap() |> cleanGaps() |> plyranges::mutate(strand = '-')
PAC3_bridge_Bar <- coa$Oki_Bar |> subsetByOverlaps(PAC3) |> swap() |> cleanGaps()
PAC3_bridge_Nor <- coa$Oki_Nor |> subsetByOverlaps(PAC3) |> swap() |> cleanGaps()

list( Osa = PAC3_bridge_Osa
    , Aom = PAC3_bridge_Aom
    , Bar = PAC3_bridge_Bar
    , Nor = PAC3_bridge_Nor) |> lapply(\(gb) gb + 0)  |> lapply(getSeq) |> lapply(unlist) |> as("DNAStringSet") |> msa::msaClustalW() |> as("DNAMultipleAlignment")
```

## The left region of PAC3 in all genomes

Alignment of 70 nucleotides flanking the alignment breakpoint on the left
side of PAC3 illustrates the conservation in all 6 genomes (upstream)
and the lack of similarity between the "Oki" sequences and the "North" bridge
region.

```{r PAC3_left}
PAC3_left_Oki_Osa  <- coa$Oki_Osa |> subsetByOverlaps(PAC3) |> plyranges::slice(1) |> get_bps("right")
PAC3_left_Kum_Osa  <- coa$Osa_Kum |> swap () |> subsetByOverlaps(PAC3_in_Kum + 1) |> sort(i=T) |> plyranges::slice(1) |> get_bps("right")
PAC3_left_Osa      <- coa$Oki_Osa |> subsetByOverlaps(PAC3) |> swap() |> plyranges::slice(1) |> get_bps("left") |> plyranges::mutate(strand = '-')
PAC3_left_Aom      <- coa$Oki_Aom |> subsetByOverlaps(PAC3) |> swap() |> plyranges::slice(1) |> get_bps("left") |> plyranges::mutate(strand = '-')
PAC3_left_Oki_Bar  <- coa$Oki_Bar |> subsetByOverlaps(PAC3) |> plyranges::slice(1) |> get_bps("right")
PAC3_left_Bar      <- coa$Oki_Bar |> subsetByOverlaps(PAC3) |> swap()  |> plyranges::slice(1) |> get_bps("right")
PAC3_left_Nor      <- coa$Oki_Nor |> subsetByOverlaps(PAC3) |> swap()  |> plyranges::slice(1) |> get_bps("right")

# Shifting Bar and Nor so that they all align well

PAC3_left <- 
list( Aom = PAC3_left_Aom
    , Osa = PAC3_left_Osa
    , Oki = PAC3_left_Oki_Osa
    , Kum = PAC3_left_Kum_Osa
    , Bar = PAC3_left_Bar |> shift(20)
    , Nor = PAC3_left_Nor |> shift(20))

PAC3_left |> lapply(\(gb) gb + 70) |> lapply(getSeq) |> lapply(unlist) |> as("DNAStringSet") |> msa::msaClustalW() |> as("DNAMultipleAlignment")
```

## The right region of PAC3 on in all genomes

Same story on the right side.  Very near the alignment break (at the center),
there is the `ATG` and upstream of it there are two conserved `AG` that may be
trans-splicing sites (with CAGE support at least in Oki).

```{r PAC3_right}
PAC3_right_Oki_Osa  <- coa$Oki_Osa |> subsetByOverlaps(PAC3) |> plyranges::slice(3) |> get_bps("left")
PAC3_right_Kum_Osa  <- coa$Osa_Kum |> swap () |> subsetByOverlaps(PAC3_in_Kum + 1) |> sort(i=T) |> plyranges::slice(3) |> get_bps("left")
PAC3_right_Osa      <- coa$Oki_Osa |> subsetByOverlaps(PAC3) |> swap() |> plyranges::slice(3) |> get_bps("right") |> plyranges::mutate(strand = '-')
PAC3_right_Aom      <- coa$Oki_Aom |> subsetByOverlaps(PAC3) |> swap() |> plyranges::slice(3) |> get_bps("right") |> plyranges::mutate(strand = '-')
PAC3_right_Oki_Bar  <- coa$Oki_Bar |> subsetByOverlaps(PAC3)           |> plyranges::slice(3) |> get_bps("left")
PAC3_right_Bar      <- coa$Oki_Bar |> subsetByOverlaps(PAC3) |> swap() |> plyranges::slice(3) |> get_bps("left")
PAC3_right_Nor      <- coa$Oki_Nor |> subsetByOverlaps(PAC3) |> swap() |> plyranges::slice(3) |> get_bps("left")

PAC3_right <-
list( Aom = PAC3_right_Aom
    , Osa = PAC3_right_Osa
    , Oki = PAC3_right_Oki_Osa
    , Kum = PAC3_right_Kum_Osa
    , Bar = PAC3_right_Bar
    , Nor = PAC3_right_Nor)
PAC3_right |> lapply(\(gb) gb + 70) |> lapply(getSeq) |> lapply(unlist) |> as("DNAStringSet") |> msa::msaClustalW() |> as("DNAMultipleAlignment") 
```

## Combined view of the left and right regions

The combined view shows that the sequence on the left and right of PAC3 do not
have visible similarity with each other or with the bridge region.

```{r PAC3_left_right}
PAC3_left_right <- list(
  Aom  = range(PAC3_left_Aom, PAC3_right_Aom),
  Osa  = range(PAC3_left_Osa, PAC3_right_Osa),
  OkiL = PAC3_left_Oki_Osa,
  OkiR = PAC3_right_Oki_Osa,
  KumL = PAC3_left_Kum_Osa,
  KumR = PAC3_right_Kum_Osa,
  Bar  = range(PAC3_left_Bar |> shift(20), PAC3_right_Bar),
  Nor  = range(PAC3_left_Nor |> shift(20), PAC3_right_Nor)
)
PAC3_left_right.aln <- PAC3_left_right |>
  lapply(\(gb) gb + 50) |>
  lapply(getSeq) |>
  lapply(unlist) |>
  as("DNAStringSet") |>
  msa::msaClustalOmega() |>
  as("DNAMultipleAlignment")

# Re-order based on my preference
PAC3_left_right.aln@unmasked[c(1,2,5,6,7,8,3,4)]
```

# PAC3 on the X-specific region of chrX (XSR)

## The PAC3 (North) or bridge region (Oki) on the XSR.

In Okinawa, the bridge region contains a bidirectional promoter.

Flanking genes might be a _leucine-rich repeat domain-containing protein_ on the
left (same operon) and a _glutamine-tRNA ligase_ on the right (different strand).

```{r PAC3_XSR}
PAC3_XSR_Osa_range <- coa$Oki_Osa |> subsetByOverlaps(PAC3) |> range()
PAC3_XSR_Aom_range <- coa$Oki_Aom |> subsetByOverlaps(PAC3) |> range()
PAC3_XSR_Bar_range <- coa$Oki_Bar |> subsetByOverlaps(PAC3) |> range()
PAC3_XSR_Nor_range <- coa$Oki_Nor |> subsetByOverlaps(PAC3) |> range()

PAC3_XSR_Osa    <- coa$Oki_Osa |> subsetByOverlaps(PAC3) |> plyranges::slice(2) |> swap()
PAC3_XSR_Aom    <- coa$Oki_Aom |> subsetByOverlaps(PAC3) |> plyranges::slice(2) |> swap()
PAC3_XSR_Bar    <- coa$Oki_Bar |> subsetByOverlaps(PAC3) |> plyranges::slice(2) |> swap()
PAC3_XSR_Nor    <- coa$Oki_Nor |> subsetByOverlaps(PAC3) |> plyranges::slice(2) |> swap()


(PAC3_XSR_Osa_Oki_triple <- coa$Osa_Oki |> subsetByOverlaps(PAC3_XSR_Osa +1000, ignore.strand = TRUE) |> sort(i=T))
(PAC3_XSR_Osa_Kum_triple <- coa$Osa_Kum |> subsetByOverlaps(PAC3_XSR_Osa +1000, ignore.strand = TRUE) |> sort(i=T))
(PAC3_XSR_Aom_Oki_triple <- coa$Oki_Aom |> swap () |> subsetByOverlaps(PAC3_XSR_Aom + 1000, ignore.strand=T) |> sort(i=T))

# Bar-Oki alignment is in the opposite orientation
(PAC3_XSR_Bar_Oki_triple <- coa$Bar_Oki |> subsetByOverlaps(PAC3_XSR_Bar +1000, ignore.strand = TRUE) |> sort(i=T))

(PAC3_XSR_Nor_Oki_triple <- coa$Oki_Nor |> swap () |> subsetByOverlaps(PAC3_XSR_Nor + 1000, ignore.strand=T) |> sort(i=T))

coa$Osa_Nor |> subsetByOverlaps(PAC3_XSR_Osa)

PAC3_XSR_Osa_left  <- PAC3_XSR_Osa_Oki_triple[1] |> get_bps(dir = 'r')
PAC3_XSR_Osa_right <- PAC3_XSR_Osa_Oki_triple[3] |> get_bps(dir = 'l')
PAC3_XSR_Oki_left  <- PAC3_XSR_Osa_Oki_triple[1] |> swap() |> get_bps(dir = 'r')
PAC3_XSR_Oki_right <- PAC3_XSR_Osa_Oki_triple[3] |> swap() |> get_bps(dir = 'l')

PAC3_XSR_Kum_left  <- PAC3_XSR_Osa_Kum_triple[1] |> swap() |> get_bps(dir = 'r')
PAC3_XSR_Kum_right <- PAC3_XSR_Osa_Kum_triple[3] |> swap() |> get_bps(dir = 'l')

PAC3_XSR_Nor_left  <- PAC3_XSR_Nor_Oki_triple[1] |> get_bps(dir = 'r')
PAC3_XSR_Nor_right <- PAC3_XSR_Nor_Oki_triple[3] |> get_bps(dir = 'l')

# Remember that Bar-Oki alignment is in the opposite orientation
PAC3_XSR_Bar_left  <- PAC3_XSR_Bar_Oki_triple[3] |> get_bps(dir = 'l') |> plyranges::mutate(strand = '-')
PAC3_XSR_Bar_right <- PAC3_XSR_Bar_Oki_triple[1] |> get_bps(dir = 'r') |> plyranges::mutate(strand = '-')

# Remember that Oki-Aom alignment is in the opposite orientation
PAC3_XSR_Aom_left  <- PAC3_XSR_Aom_Oki_triple[3] |> get_bps(dir = 'l') |> plyranges::mutate(strand = '-')
PAC3_XSR_Aom_right <- PAC3_XSR_Aom_Oki_triple[1] |> get_bps(dir = 'r') |> plyranges::mutate(strand = '-')

PAC3_XSR_Aom    <- coa$Oki_Aom |> subsetByOverlaps(PAC3) |> plyranges::slice(2) |> swap()
PAC3_XSR_Bar    <- coa$Oki_Bar |> subsetByOverlaps(PAC3) |> plyranges::slice(2) |> swap()
PAC3_XSR_Nor    <- coa$Oki_Nor |> subsetByOverlaps(PAC3) |> plyranges::slice(2) |> swap()
```

## Left side

In Okinawa and Kume, the bridge region is ~400 bp-long.  It will not be possible
to make a combined left-right alignment like on chr1.

```{r XSR_left}
n <- -40
PAC3_XSR_Osa.aln.left <- list(
  Osa_left = PAC3_XSR_Osa_left |> shift(n),
  Aom_left = PAC3_XSR_Aom_left |> shift(-n),
  Oki_left = PAC3_XSR_Oki_left |> shift(n),
  Kum_left = PAC3_XSR_Kum_left |> shift(n),
  Nor_left = PAC3_XSR_Nor_left |> shift(n),
  Bar_left = PAC3_XSR_Bar_left |> shift(-n))

PAC3_XSR_Osa.aln.left |> lapply(\(gb) gb + 90) |> lapply(getSeq) |> lapply(unlist) |> as("DNAStringSet") |> msa::msaClustalW() |> as("DNAMultipleAlignment") 
```

## Right side

The right-side region is near a bidirectional promoter in both the Oki and Osa
genomes.

```{r XSR_right}
PAC3_XSR_Osa.aln.right <- list(
  Osa_right = PAC3_XSR_Osa_right,
  Aom_right = PAC3_XSR_Aom_right,
  Oki_right = PAC3_XSR_Oki_right,
  Kum_right = PAC3_XSR_Kum_right,
  Nor_right = PAC3_XSR_Nor_right |> shift(-306),
  Bar_right = PAC3_XSR_Bar_right |> shift(306))

PAC3_XSR_Osa.aln.right |> lapply(\(gb) gb + 90) |> lapply(getSeq) |> lapply(unlist) |> as("DNAStringSet") |> msa::msaClustalW() |> as("DNAMultipleAlignment") 
```




Conserved and non-conserved operons documentation 

# PAC 3 context diagram


Identifying operons  

```{r Identifying operons}
calkOperons <- function(genes, window = 100) {
  # Divide the window by two because we will expand both sides of the range.
  halfwin <- round(window / 2)
  # Sort the genes ignoring strand.
  g <- sort(genes, ignore.strand = TRUE)
  # Expand by `halfwin` nucleotides
  # We suppress warnings because coordinates at the edges of seqfeatures can
  # become transiently negative.
  g <- (g + halfwin) |> suppressWarnings()
  # Give a unique ID to all runs of successive genes on the same strand of the
  # same seqname, see also https://stackoverflow.com/questions/21421047 and
  # data.table::rleid
  g$namedStrand <- Rle(paste0(seqnames(g), strand(g)))
  g$idx <- Rle(rep(seq_along(runValue(g$namedStrand))), runLength(g$namedStrand))
  # Split the genes by run ID
  opL <- GenomicRanges::split(g, g$idx)
  # Remove all runs of length one
  opL <- opL[runLength(strand(g)) > 1]
  # Merge into operons
  operons   <- unlist(endoapply(opL, GenomicRanges::reduce, min.gapwidth = 0L)) |> suppressWarnings()
  # Removed un-merged genes
  operons$n <- countOverlaps(operons, genes)
  ov <- findOverlaps(operons, genes)
  # This function is the same as calcOperons but with the addition of a gene_id metadata column
  operons$gene_id <- CharacterList(split(genes$gene_id[subjectHits(ov)], queryHits(ov))) |> unname()
  operons <- operons[operons$n > 1]
  # Remove the flanking sequences
  operons - halfwin
}

operons <- SimpleList()
operons$Oki <- calkOperons(Genes$Oki, window = 500) |> suppressWarnings()
operons$Osa <- calkOperons(Genes$Osa, window=500)   |> suppressWarnings()
operons$Bar <- calkOperons(Genes$Bar, window=500)   |> suppressWarnings()


#Function to find find upstream and downstream genes in specific operon 
Find_UpAndDown <- function(op, gene_ofi) {
  ss <- lapply(op$gene_id, function(x) gene_ofi %in% x) 
  ss1 <- unlist(ss)
  xx <- which(ss1)
  #yy <- as.data.frame(op[xx])
  yy <- op[xx]
  yy
}

pac3_operon_oki <- Find_UpAndDown(operons$Oki, 'g3146')
pac3_operon_osa <- Find_UpAndDown(operons$Osa, 'g12442')
pac3_operon_bar <- Find_UpAndDown(operons$Bar, 'g12642')

hhhh <- Find_UpAndDown(operons$Osa, 'g4968')
```


PAC3 Figure
Identifying regions of interest

```{r PAC3Figure, fig.height=6, fig.width=12, message=FALSE, warning=FALSE, dev=c("svg", "png")}
coa$Oki_Osa |> subsetByOverlaps(PAC3)
coa$Oki_Bar |> subsetByOverlaps(PAC3)
coa$Oki_Kum |> subsetByOverlaps(PAC3)

extract_coords <- function(gr, gene_name, upstream=0, downstream=0) {
  matching_gene_index <- which(mcols(gr)$gene_id == gene_name)
  sgr <- gr[(matching_gene_index-upstream):(matching_gene_index+downstream),]

  sgr
}

not_pac3_operon_osa <- extract_coords(Genes$Osa, 'g2061', upstream=5, downstream = 4)

#Okinawa region g3143.t1 - g3145.t1
First3GenesOki <- GRanges('chr1:11487473-11497595')
RegionokiOsaNp3 <- subsetByOverlaps(coa$Oki_Osa, First3GenesOki)
First3GenesOsa <- subsetByOverlaps(operons0$Osa, RegionokiOsaNp3$query)

Last5GenesOki <- GRanges('chr1:11498263-11503891 ')
RegionokiOsaNp3_5 <- subsetByOverlaps(coa$Oki_Osa, Last5GenesOki)
Last5GenesOsa <- subsetByOverlaps(operons0$Osa, RegionokiOsaNp3_5$query)

NotPAC3OpInOsa <- c(First3GenesOsa, Last5GenesOsa)

#Not pac3
notpac3 <- GRanges('Chr1:8150909-8166679')
regionnotpac3 <- subsetByOverlaps(coa$Osa_Oki, notpac3)
#8098120-8165043
np3gr <- subsetByOverlaps(Genes$Osa, GRanges('Chr1:8150909-8166679'))
```

## Plotting PAC3 Figure
```{r pac3_operon_diagram, dependson="chromosome_plots_operons", fig.height=6, fig.width=12, message=FALSE, warning=FALSE, dev=c("svg", "png")}
#Gene Ranges to dna_segs
gr2dna_seg <- function (gr, gene_of_interest=NULL, gene_of_interest_colour='red', ...) {
  # DNA segments need unique names
  if (length(gr) == 0) return(NULL)
  if (is.null(names(gr))) names(gr) <- as.character(gr)
  # Unstranded ranges are not recognized
  strand(gr[strand(gr) == "*"]) <- "+"
  df <- data.frame(
    name   = names(gr),
    start  = start(gr),
    end    = end(gr),
    strand = strand(gr)
  )
  dna_seg <- genoPlotR::dna_seg(df, ...)
  dna_seg$col <- 'black'
  dna_seg$fill <- '#595959'
  if(!is.null(gene_of_interest)){
    dna_seg$fill[which(dna_seg$name==gene_of_interest)] <- gene_of_interest_colour
  }
  dna_seg
}

tmp_dnaseg <- gr2dna_seg(pac3_operon_oki, gene_of_interest='g3146')

dnaseg_to_annot <- function(ds, ...) {
  annot <- annotation(x1=middle(ds), text=ds$name, ...)
  annot
}
dnaseg_to_annot(tmp_dnaseg)

PAC3_in_Osa <- GRanges("XSR:3162950-3168025")

operons_to_transcripts <- function(operons, genes){
  operons$idx <- 1:length(operons)
  operon_list <- endoapply(split(operons, operons$idx), function(o) {
    gr <- genes[match(unlist(o$gene_id), genes$gene_id)]
    gr$operon_idx <- o$idx
    gr
  })
  operon_list <- unlist(operon_list)
  # Avoid prepending the operon index to the names of the unlisted GRangesList.
  names(operon_list) <- lapply(names(operon_list), function(x) unlist(strsplit(x, '\\.'))[2]) |> unname()
  operon_list
}
operons_to_transcripts(pac3_operon_oki, Genes$Oki)

pac3_operon_oki_genes <- operons_to_transcripts(pac3_operon_oki, Genes$Oki)
pac3_operon_osa_genes <- operons_to_transcripts(pac3_operon_osa, Genes$Osa)


plot_gene_map(
  dna_segs = list(
    `Osaka\nChr1: 8145427 - 8174791`=reverse(gr2dna_seg(not_pac3_operon_osa)),
    `Okinawa\nchr1: 11487473 - 11503891`=gr2dna_seg(pac3_operon_oki_genes, gene_of_interest='g3146'),
    `Osaka\nXSR: 11487473 - 11503891`=reverse(gr2dna_seg(pac3_operon_osa_genes, gene_of_interest='g12442'))
  ),
  dna_seg_scale = T, 
  annotations = list(
    dnaseg_to_annot(reverse(gr2dna_seg(not_pac3_operon_osa)), rot=0, col='grey80'),
    dnaseg_to_annot(gr2dna_seg(pac3_operon_oki_genes, gene_of_interest='g3146'), rot=0, col='grey80'),
    dnaseg_to_annot(reverse(gr2dna_seg(pac3_operon_osa_genes, gene_of_interest='g12442')), rot=0, col='grey80')
  ),
  comparisons = list(
     gb2comp(coa$Oki_Osa |> subsetByOverlaps(PAC3) |> swap()) |> reverse(side=1),
     gb2comp(coa$Oki_Osa |> subsetByOverlaps(PAC3)) |> reverse(side = 2)
   ),
#  annotation_height = 2.9, annotation_cex = 0.9,
  main = "The PAC3 gene in operon is conserved between Osaka and Barcelona in XSR chromosome but on CHR1 in Okinawa"
)

```



Identifying conserved and non-conserved operons
```{r}
#Operon conservation algorithm
N19 <- OikScrambling:::load_one_to_ones(system.file("extdata/OrthoFinder/N19.tsv", package="BreakpointsData"))
#Subset BAR, OKI, OSA
N19ss <- as.data.frame(N19[, c("OKI2018_I69.v2.prot.longest.fa_1", "OSKA2016v1.9.prot.longest.fa_1", "Bar2_p4.Flye.prot.longest.fa_1") ])
#Remove species identifier before gene symbol
#Okinawa
rmSIOki <- N19ss %>%
  mutate_at("OKI2018_I69.v2.prot.longest.fa_1", str_replace, "OKI2018_I69.v2.", "")
#Osaka
rmSIOsa <- rmSIOki %>%
  mutate_at("OSKA2016v1.9.prot.longest.fa_1", str_replace, "OSKA2016v1.9.", "")
#Barcelona

#Gene equivalencies table. Transcripts version
GeneEquivTable <- rmSIOsa %>%
  mutate_at("Bar2_p4.Flye.prot.longest.fa_1", str_replace, "Bar2_p4_Flye.", "")


#Gene equivalencies table. Genes version.
fufu <- as.data.frame(unlist(lapply(GeneEquivTable$OKI2018_I69.v2.prot.longest.fa_1, function(x) unlist(strsplit(x, "\\."))[1])))
fufu1 <- as.data.frame(unlist(lapply(GeneEquivTable$OSKA2016v1.9.prot.longest.fa_1, function(x) unlist(strsplit(x, "\\."))[1])))
ActualGeneEquiv <- data.frame(fufu, fufu1)
colnames(ActualGeneEquiv) <- c('Oki', 'Osa')


#Function to find out if a gene has an equivalent in the other species
GeneMatch <- function(gene_ID, SP1, SP2, G_E) {
  GMatch <- match(gene_ID, G_E[[SP1]])
  sp2_gene <- NA
  if(!is.na(GMatch)){
    sp2_gene <- G_E[[SP2]][GMatch]
  }
  return(sp2_gene)
}

#Transcripts
GeneMatch('g3151.t1', 'OKI2018_I69.v2.prot.longest.fa_1', 'OSKA2016v1.9.prot.longest.fa_1', GeneEquivTable)
#Genes
GeneMatch('g3151', 'Oki', 'Osa', ActualGeneEquiv)

#Funtion to find if the whole operon has an equivalent in other species
OpGeneMatch <- function(operon, SP1, SP2, G_E){
  Matchh <- sapply(operon, function(G) GeneMatch(G, SP1, SP2, G_E))
  return(Matchh)
}


operons0$Oki$tx_names[2103]
operons0$Osa$tx_names[853]





#Find if genes match in candidate operons
OpGeneMatch(operons0$Oki$tx_names[[1264]], 'OKI2018_I69.v2.prot.longest.fa_1', 'OSKA2016v1.9.prot.longest.fa_1', GeneEquivTable)
OpGeneMatch(operons$Osa$tx_names[[853]], 'OKI2018_I69.v2.prot.longest.fa_1', 'OSKA2016v1.9.prot.longest.fa_1', GeneEquivTable) 

hyhy <- OpGeneMatch(operons0$Oki$gene_id, 'OKI2018_I69.v2.prot.longest.fa_1', 'OSKA2016v1.9.prot.longest.fa_1', GeneEquivTable) 
huhu <- OpGeneMatch(operons0$Osa$gene_id, 'OKI2018_I69.v2.prot.longest.fa_1', 'OSKA2016v1.9.prot.longest.fa_1', GeneEquivTable) 


OkiGenes <- annots$Oki |> genes()


jajaje <- OpGeneMatch(OkiGenes$gene_id,   )



hyhy1 <- unlist(hyhy)


hyhy1 <- hyhy %>% 
  map(discard, is.na) %>%
  compact()

gygyg <- sapply(hyhy1, length)
hyhy1[order(gygyg)]

operons0$Oki[2103]



write.csv(operons0$Oki, "OperonsInOki.csv", row.names = T)
system("scp OperonsInOki.csv ")


#Subsets of nucleotide alignments that match between species
library("GenomicFeatures")
annots      <- OikScrambling:::loadAllAnnotations()   |> suppressWarnings()
operons <- lapply(annots, \(a) {a |> genes() |> OikScrambling:::calcOperons(window = 500) }) |> SimpleList() |> suppressWarnings()
operons$Nor <- OikScrambling:::calcOperons(window = 500, transcripts(annots$Nor) |> reduce()) |> suppressWarnings()
sapply(operons, length)

wgo$Oki_Osa$type |> levels() 


## Find operon candidates using number of genes in operon 
#Oki Operons
OkiOpsAl <- subsetByOverlaps(operons$Oki, wgo$Oki_Osa[ !wgo$Oki_Osa$type=="aligned"])
c2 <- subset(OkiOpsAl, n > 9)[3]
c2$tx_names |> unlist()

#Osa Operons
OsaOpsAl <- subsetByOverlaps(operons$Osa, wgo$Osa_Oki[ !wgo$Osa_Oki$type=="breakpoint region"])
c3 <- subset(OsaOpsAl, n > 9)[4]
c3$tx_names |> unlist()


gygyg <- OpGeneMatch(c2$tx_names[[1264]], 'OKI2018_I69.v2.prot.longest.fa_1', 'OSKA2016v1.9.prot.longest.fa_1', GeneEquivTable) 





```




# Session information

```{r session_information}
sessionInfo()
```




