---
title: "Whole-chromosome parallel plots"
author: 
 - "Charles Plessy"
date: "`r format(Sys.time(), '%d %B, %Y')`"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteEncoding{UTF-8}
  %\VignetteIndexEntry{Whole-chromosome parallel plots}
  %\VignetteEngine{knitr::rmarkdown}
editor_options: 
  chunk_output_type: console
---

```{r setup}
knitr::opts_chunk$set(cache = TRUE, cache.lazy = FALSE)
knitr::opts_knit$set(verbose = TRUE)
```

# Introduction

In this vignette I plot whole chromosomes in parallel to reflect synteny and
the differences between short and long arms.

The core functions used here are maintained in our _GenomicBreaks_ R package,
which is fully documented at: <https://oist.github.io/GenomicBreaks>.

## Load pacakges

### Core packages that provide functions we use a lot.

```{r loadlib, echo=T, results='hide'}
suppressPackageStartupMessages({
  library('GenomicBreaks')
  library('BSgenome')
  library('GenomicFeatures')
  library('ggplot2')
})
```

### BSgenome packages

BSgenome packages will be automatically attached together when the `GBreaks`
objects will be loaded.

## Load and prepare data

### Genome annotations.

The annotations from `/bucket/LuscombeU/common/Breakpoints/Annotations` are
distributed in the same drat repository as the alignment data.

```
install.packages("BreakpointsData", repos="https://oist.github.io/plessy_oikgenomes_drat/")
```

```{r load_annotations, warning=FALSE}
# annots <- SimpleList()
# 
# gff2txdb <- function(file, genome) {
#   file <- system.file(paste0("extdata/Annotations/", file), package = "BreakpointsData")
#   tx <- rtracklayer::import.gff(file)
#   tx <- tx[seqnames(tx) %in% seqnames(genome)]
#   tx <- GRanges(tx, seqinfo = seqinfo(genome))
#   tx <- GenomicFeatures::makeTxDbFromGRanges(tx)
# }
# 
# gff2txdb_Norway <- function(file, genome) {
#   file <- system.file(paste0("extdata/Annotations/", file), package = "BreakpointsData")
#   tx <- rtracklayer::import.gff(file)
#   tx <- tx[!is.na(tx$mRNA)]
#   # Remove "scaffoldA" objects in the OdB3 annotation
#   tx <- tx[seqnames(tx) %in% seqnames(genome)]
#   tx$ID <- tx$mRNA
#   tx <- GRanges(tx, seqinfo = seqinfo(genome))
#   tx <- GenomicFeatures::makeTxDbFromGRanges(tx)
# }
# 
# annots$Oki <- gff2txdb("OKI2018_I69.v2/OKI2018_I69.v2.gm.gff.gz",  OKI2018_I69)
# annots$Osa <- gff2txdb("OSKA2016v1.9/OSKA2016v1.9.gm.gff.gz",      OSKA2016v1.9)
# annots$Bar <- gff2txdb("Bar2_p4.Flye/Bar2_p4.Flye.gm.gff.gz",      Bar2_p4)
# annots$Kum <- gff2txdb("KUM-M3-7f/KUM-M3-7f.gm.gff.gz",            KUM_M3)
# annots$Aom <- gff2txdb("AOM-5-5f/AOM-5-5f.gm.gff.gz",              AOM_5)
# annots$Nor <- gff2txdb_Norway("OdB3/Oikopleura_annot_v1.0.gff.gz", OdB3)
```

### GenomicBreaks objects

The core data is a list of `GenomicBreaks::GBreak()` objects that is built
by a separate vignette (`vignette("Load genomic break data")`).

```{r GBreaks_object}
load("BreakPoints.Rdata")
```

### Long and short arms

The long and short arms are annotated in the `GBreak` objects that are built
by the vignette (`vignette("Load genomic break data")`), using the function
`GenomicBreaks::flagLongShort()`.  A copy of the arms annotations is provided
in the `longShort` list.  The loaded `GBreaks` objects are already annotated.

```{r long_and_short_arms}
longShort
longShort$Oki
gbs$Oki_Osa
```

### Phylogenetic cladogram

We assume that the _North Pacific_ and the _Atlantic_ species are more related
to each other than to the _Okinawan_ species.

```{r phylogenetic_tree, fig.height=4, fig.width=6, dev="svg"}
requireNamespace("ade4")
treeLeaf <- function(name, length=NULL) {
  if(!is.null(length)) length <- paste0(':', length)
  paste0(name, length)
}

treeNode <- function(branch1, branch2, length = NULL) {
  if(!is.null(length)) length <- paste0(':', length)
  paste0('(', branch1, ',', branch2, ')', length)
}

addRoot <- function(branch)  paste0(branch, ";")

tree <-
  addRoot(
    treeNode(
      treeNode( length = 2,
        treeLeaf("Okinawa", 1),
        treeLeaf("Kume", 1)
      ),
      treeNode( length = 1,
        treeNode( length = 1,
          treeLeaf("Osaka", 1),
          treeLeaf("Aomori", 1)
        ),
        treeNode( length =1,
          treeLeaf("Norway", 1),
          treeLeaf("Barcelona", 1)
        )
      )
    )
  )

plot(ade4::newick2phylog(tree))
```

## Genome plots with genoPlotR

```{r Oki_Bar_chr1, cache=TRUE, fig.width=15}
plotApairOfChrs(coa$Oki_Bar, "chr1")
```
```{r Oki_Osa_chr1, cache=TRUE, fig.width=15}
plotApairOfChrs(coa$Oki_Osa, "chr1")
```
```{r Osa_Oki_chr1, cache=TRUE, fig.width=15}
plotApairOfChrs(coa$Osa_Oki, "Chr1")
```
```{r Osa_Bar_chr1, cache=TRUE, fig.width=15}
plotApairOfChrs(coa$Osa_Bar, "Chr1")
```
```{r Oki_Bar_chr1_coa2, cache=TRUE, fig.width=15}
plotApairOfChrs(coa2$Osa_Bar, "Chr1")
```


```{r coa_based, fig.width=15}
x <- coa$Oki_Kum
x[seqnames(x$query) == "contig_90_1"] <- reverse(query = TRUE, x[seqnames(x$query) == "contig_90_1"])
x[seqnames(x$query) == "contig_88_1"] <- reverse(query = TRUE, x[seqnames(x$query) == "contig_88_1"])

x$query <- mergeSeqLevels(x$query, c("contig_27_1", "contig_90_1", "contig_3_1", "contig_88_1"), "new_scaffold")
plotApairOfChrs(x, "chr1", dna_seg_scale=TRUE)
```

```{r coa2_based fig.width=15}
x <- coa2$Oki_Kum
x[seqnames(x$query) == "contig_90_1"] <- reverse(query = TRUE, x[seqnames(x$query) == "contig_90_1"])
x[seqnames(x$query) == "contig_88_1"] <- reverse(query = TRUE, x[seqnames(x$query) == "contig_88_1"])

x$query <- mergeSeqLevels(x$query, c("contig_27_1", "contig_90_1", "contig_3_1", "contig_88_1"), "new_scaffold")
plotApairOfChrs(x, "chr1", dna_seg_scale=TRUE)
```

## Plot chr1 in three species

```{r plot_intersect_GBs_with_tree, fig.width=15}
doubleSubset <- function(gb, a, b) {
  gb <- gb[seqnames(gb) == a]
  gb <- gb[seqnames(gb$query) == b]
  gb
}

dsList <-   list(
    OKI = doubleSubset(coa$Oki_Osa, "chr1", "Chr1") |> gr2dna_seg(),
    OSA = doubleSubset(coa$Osa_Bar, "Chr1", "Chr1") |> gr2dna_seg(),
    BAR = doubleSubset(coa$Bar_Osa, "Chr1", "Chr1") |> gr2dna_seg()
)

compList <- list(
    doubleSubset(coa$Oki_Osa, "chr1", "Chr1") |> gb2comp(),
    Osa_Bar = doubleSubset(coa$Osa_Bar, "Chr1", "Chr1") |> gb2comp()
  )

genoPlotR::plot_gene_map(dsList, compList, dna_seg_scale=TRUE, tree=ade4::newick2phylog(addRoot(treeNode("OKI", treeNode("OSA", "BAR")))))
```


## Region that is conserved between Osaka and Barcelona

Since Okinawa is the outgroup, let's look for regions of it that are aligned
on both the Osaka and the Barcelona genomes.

```{r combine_GBs}
x <- subsetByOverlaps(coa2$Oki_Osa, granges(coa2$Oki_Bar)) |> subset(width > 1e5)
y <- subsetByOverlaps(coa2$Oki_Bar, granges(coa2$Oki_Osa)) |> subset(width > 1e5)
findOverlapPairs(x,y)
subsetByOverlaps(coa2$Oki_Osa, gaps(coa2$Oki_Bar)) |> subset(width > 1e5)

a <- coa$Oki_Osa |> subsetByOverlaps(GRanges("chr2:10000000-11000000")) |> doubleSubset("chr2", "Chr2")
b <- coa$Oki_Bar |> subsetByOverlaps(GRanges("chr2:10000000-11000000")) |> doubleSubset("chr2", "Chr2")
c <- coa$Bar_Osa |> subsetByOverlaps(range(b$query))                    |> doubleSubset("Chr2", "Chr2")

dsList <- list(
  OSA = swap(a) |> gr2dna_seg(),
  OKI = a |> gr2dna_seg(),
  BAR = swap(b) |> gr2dna_seg(),
  Osa = swap(c) |> gr2dna_seg()
)

compList <- list(
  a |> swap() |> gb2comp(),
  b |> gb2comp(),
  c |> gb2comp()
)

genoPlotR::plot_gene_map(dsList, compList, dna_seg_scale=TRUE)

```


# Session information

```{r sessionInfo}
sessionInfo()
```

