---
title: "Whole-chromosome parallel plots"
author: 
 - "Charles Plessy"
date: "`r format(Sys.time(), '%d %B, %Y')`"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteEncoding{UTF-8}
  %\VignetteIndexEntry{Whole-chromosome parallel plots}
  %\VignetteEngine{knitr::rmarkdown}
editor_options: 
  chunk_output_type: console
---

```{r setup}
knitr::opts_chunk$set(cache = TRUE, cache.lazy = FALSE)
knitr::opts_knit$set(verbose = TRUE)
```

# Introduction

Pairwise alignments between whole chromosomes are plotted in parallel in order
to reflect synteny and the difference of scrambling between short and long arms.

Figure 1 panel B was generated in this vignette.

# Load R pacakges and data

## Load pacakges

```{r load_packages_and_data}
library('OikScrambling') |> suppressPackageStartupMessages()
genomes <- OikScrambling:::loadAllGenomes()
load("BreakPoints.Rdata")
```

## Phylogenetic cladogram

We assume that the _North Pacific_ and the _Atlantic_ species are more related
to each other than to the _Okinawan_ species.

```{r phylogenetic_tree, fig.height=4, fig.width=6, dev="svg", fig.ext="svg"}
requireNamespace("ade4")
treeLeaf <- function(name, length=NULL) {
  if(!is.null(length)) length <- paste0(':', length)
  paste0(name, length)
}

treeNode <- function(branch1, branch2, length = NULL) {
  if(!is.null(length)) length <- paste0(':', length)
  paste0('(', branch1, ',', branch2, ')', length)
}

addRoot <- function(branch)  paste0(branch, ";")

tree <-
  addRoot(
    treeNode(
      treeNode( length = 2,
        treeLeaf("Okinawa", 1),
        treeLeaf("Kume", 1)
      ),
      treeNode( length = 1,
        treeNode( length = 1,
          treeLeaf("Osaka", 1),
          treeLeaf("Aomori", 1)
        ),
        treeNode( length =1,
          treeLeaf("Norway", 1),
          treeLeaf("Barcelona", 1)
        )
      )
    )
  )

plot(ade4::newick2phylog(tree))
```

# Scaffolding of Kume chromosomes

The Kume draft assembly contains long contigs that can be scaffolded into
chromosomes.  Some sequences may be missing between these contigs, but the
absence of large gaps in the plots at the junction between contigs suggest that
this is not a problem for the use we make of this scaffolding.

## Chromosome 1

Construction of a `GRanges` object representing the contigs to merge and whether
they have to be reverse-complemented or not.

```{r Kum_chr1_contigs_to_merge}
contigsToMerge        <- c("contig_27_1", "contig_90_1", "contig_3_1", "contig_88_1", "contig_87_1", "contig_63_1")
contigsToMerge_start  <- c(1, cumsum(seqlengths(coa$Oki_Kum$query)[contigsToMerge]))    |> unname() |> head(-1)
contigsToMerge_end    <-     (cumsum(seqlengths(coa$Oki_Kum$query)[contigsToMerge]) -1) |> unname()
contigsToMerge_oldPos <- GRanges("chr1", IRanges(contigsToMerge_start, contigsToMerge_end))
names(contigsToMerge_oldPos)  <- contigsToMerge
strand(contigsToMerge_oldPos) <- c('+', '-', '+', '-', '+', '+')
contigsToMerge_oldPos
```

Construction of the scaffold and control plot.

```{r Kum_chr1_coa, fig.width=15}
KumChr1 <- coa$Oki_Kum
KumChr1[seqnames(KumChr1$query) == "contig_90_1"] <- reverse(query = TRUE, KumChr1[seqnames(KumChr1$query) == "contig_90_1"])
KumChr1[seqnames(KumChr1$query) == "contig_88_1"] <- reverse(query = TRUE, KumChr1[seqnames(KumChr1$query) == "contig_88_1"])

KumChr1$query <- mergeSeqLevels(KumChr1$query, contigsToMerge, "chr1")
annot <- genoPlotR::annotation(x1=start(contigsToMerge_oldPos), x2=end(contigsToMerge_oldPos), text=names(contigsToMerge_oldPos), rot=30)

plotApairOfChrs(KumChr1 |> swap(), "chr1", dna_seg_scale=TRUE, main = "Kum vs Oki", annotations = annot)
```

## Chromosome 2

Construction of a `GRanges` object representing the contigs to merge and whether
they have to be reverse-complemented or not.

```{r Kum_chr2_contigs_to_merge}
contigsToMerge        <- c("contig_36_1", "contig_16_1", "contig_24_1", "contig_79_1", "contig_37_1", "contig_81_1")
contigsToMerge_start  <- c(1, cumsum(seqlengths(coa$Oki_Kum$query)[contigsToMerge]))    |> unname() |> head(-1)
contigsToMerge_end    <-     (cumsum(seqlengths(coa$Oki_Kum$query)[contigsToMerge]) -1) |> unname()
contigsToMerge_oldPos <- GRanges("chr2", IRanges(contigsToMerge_start, contigsToMerge_end))
names(contigsToMerge_oldPos)  <- contigsToMerge
strand(contigsToMerge_oldPos) <- c('-', '+', '+', '-', '-', '-')
contigsToMerge_oldPos
```

Construction of the scaffold and control plot.

```{r Kum_chr2_coa, fig.width=15}
KumChr2 <- coa$Oki_Kum
KumChr2[seqnames(KumChr2$query) == "contig_36_1"] <- reverse(query = TRUE, KumChr2[seqnames(KumChr2$query) == "contig_36_1"])
KumChr2[seqnames(KumChr2$query) == "contig_81_1"] <- reverse(query = TRUE, KumChr2[seqnames(KumChr2$query) == "contig_81_1"])
KumChr2[seqnames(KumChr2$query) == "contig_79_1"] <- reverse(query = TRUE, KumChr2[seqnames(KumChr2$query) == "contig_79_1"])
KumChr2[seqnames(KumChr2$query) == "contig_37_1"] <- reverse(query = TRUE, KumChr2[seqnames(KumChr2$query) == "contig_37_1"])

KumChr2$query <- mergeSeqLevels(KumChr2$query, contigsToMerge, "chr2")
annot <- genoPlotR::annotation(x1=start(contigsToMerge_oldPos), x2=end(contigsToMerge_oldPos), text=names(contigsToMerge_oldPos), rot=30)

plotApairOfChrs(KumChr2 |> swap(), "chr2", dna_seg_scale=TRUE, main = "Kum vs Oki", annotations = annot)
```

# Plot chr2 in four species

We chose chromosome 2 because it has _Hox4_.

## Parallel plot

```{r plot_intersect_GBs_with_tree_with_Kume, fig.width=15, fig.height=7, dev=c("svg", "pdf", "png"), fig.ext=c("svg", "pdf", "png")}
dsList4 <- list(
    KUM = KumChr2 |> swap() |> plyranges::filter(seqnames == 'chr2', seqnames(query) == 'chr2') |> gr2dna_seg(),
    OKI = KumChr2           |> plyranges::filter(seqnames == 'chr2', seqnames(query) == 'chr2') |> gr2dna_seg(),
    OSA = coa$Osa_Bar |> plyranges::filter(seqnames == 'Chr2', seqnames(query) == 'Chr2') |> gr2dna_seg(),
    BAR = coa$Bar_Osa |> plyranges::filter(seqnames == 'Chr2', seqnames(query) == 'Chr2') |> gr2dna_seg()
)

compList4 <- list(
    Oki_Kum = KumChr2 |> swap() |> plyranges::filter(seqnames == 'chr2', seqnames(query) == 'chr2') |> gb2comp(),
    Oki_Osa = coa$Oki_Osa |> plyranges::filter(seqnames == 'chr2', seqnames(query) == 'Chr2') |> gb2comp(),
    Osa_Bar = coa$Osa_Bar |> plyranges::filter(seqnames == 'Chr2', seqnames(query) == 'Chr2') |> gb2comp()
  )

dsList4$KUM$col <- dsList4$KUM$fill <- dsList4$OKI$col <- dsList4$OKI$fill <-
  dsList4$OSA$col <- dsList4$OSA$fill <- dsList4$BAR$col <- dsList4$BAR$fill <-
  "#A6D854"

compList4$Oki_Kum$color <- compList4$Oki_Osa$color <- compList4$Osa_Bar$color <- "#D4EACC"

compList4$Oki_Kum$col <- ifelse(compList4$Oki_Kum$col == "darksalmon", "#D4EACC", "#00A08E")
compList4$Osa_Bar$col <- ifelse(compList4$Osa_Bar$col == "darksalmon", "#D4EACC", "#00A08E")
compList4$Oki_Osa$col <- ifelse(compList4$Oki_Osa$col == "darksalmon", "#D4EACC", "#00A08E")


genoPlotR::plot_gene_map(dsList4, compList4, dna_seg_scale = FALSE,
                         tree = ade4::newick2phylog(addRoot(treeNode(treeNode("OKI", "KUM"), treeNode("OSA", "BAR")))),
                         main = "Chromosome 2 in four Oikopleura genome assemblies")
```

## Strand randomisation indexes

These numbers were added to the plot in Figure 1B.

```{r SR_index_for_tree_with_Kume}
chr2Long <-  \(gb) plyranges::filter(gb, seqnames %in% c("chr2", "Chr2"), seqnames(query) %in% c("chr2", "Chr2"), Arm == "long",  query$Arm == "long")
chr2short <- \(gb) plyranges::filter(gb, seqnames %in% c("chr2", "Chr2"), seqnames(query) %in% c("chr2", "Chr2"), Arm == "short", query$Arm == "short")
chr2both <- \(gb) plyranges::filter(gb, seqnames %in% c("chr2", "Chr2"), seqnames(query) %in% c("chr2", "Chr2"))

KumChr2$query$Arm <- Rle("long")
KumChr2$query$Arm[end(KumChr2) <= end(contigsToMerge_oldPos["contig_16_1"]) & seqnames(KumChr2) == "chr2" & seqnames(KumChr2$query) == "chr2"] <- "short"

SRI_Both_Strands <- \(gb) mean(c(strand_randomisation_index(gb)
                                ,strand_randomisation_index(swap(gb))))

(SRIs <- c(
  Kum_Chr2_long  = KumChr2 |> chr2Long()  |> SRI_Both_Strands(),
  Kum_Chr2_short = KumChr2 |> chr2short() |> SRI_Both_Strands(),
  Kum_Chr2_both  = KumChr2 |> chr2both()  |> SRI_Both_Strands(),

  Oki_Osa_long   = coa$Oki_Osa |> chr2Long() |> SRI_Both_Strands(),
  Osa_Oki_long   = coa$Osa_Oki |> chr2Long() |> SRI_Both_Strands(),

  Oki_Osa_short  = coa$Oki_Osa |> chr2short() |> SRI_Both_Strands(),
  Osa_Oki_short  = coa$Osa_Oki |> chr2short() |> SRI_Both_Strands(),

  Oki_Osa_both   = coa$Oki_Osa |> chr2both() |> SRI_Both_Strands(),
  Osa_Oki_both   = coa$Osa_Oki |> chr2both() |> SRI_Both_Strands(),

  Osa_Bar_long   = coa$Osa_Bar |> chr2Long() |> SRI_Both_Strands(),
  Bar_Osa_long   = coa$Bar_Osa |> chr2Long() |> SRI_Both_Strands(),

  Osa_Bar_short  = coa$Osa_Bar |> chr2short() |> SRI_Both_Strands(),
  Bar_Osa_short  = coa$Bar_Osa |> chr2short() |> SRI_Both_Strands(),

  Osa_Bar_both   = coa$Osa_Bar |> chr2both() |> SRI_Both_Strands(),
  Bar_Osa_both   = coa$Bar_Osa |> chr2both() |> SRI_Both_Strands()
))

roundMean <- \(x) round(mean(SRIs[x]), 2)
c(Kum_Chr2_long  = roundMean(c("Kum_Chr2_long")),
  Kum_Chr2_short = roundMean(c("Kum_Chr2_short")),
  Oki_Osa_long   = roundMean(c("Oki_Osa_long",  "Osa_Oki_long")),
  Oki_Osa_short  = roundMean(c("Oki_Osa_short", "Osa_Oki_short")),
  Osa_Bar_long   = roundMean(c("Osa_Bar_long",  "Bar_Osa_long")),
  Osa_Bar_short  = roundMean(c("Osa_Bar_short", "Bar_Osa_short"))
)
```

# Plot pairs of chromosome 1s.

```{r Oki_Bar_chr1, cache=TRUE, fig.width=15}
plotApairOfChrs(coa$Oki_Bar, "chr1", main = "Oki vs Bar")
```

```{r Oki_Osa_chr1, cache=TRUE, fig.width=15}
plotApairOfChrs(coa$Oki_Osa, "chr1", main = "Oki vs Osa")
```

```{r Osa_Oki_chr1, cache=TRUE, fig.width=15}
plotApairOfChrs(coa$Osa_Oki, "Chr1", main = "Osa vs Oki")
```

```{r Osa_Bar_chr1, cache=TRUE, fig.width=15}
plotApairOfChrs(coa$Osa_Bar, "Chr1", main = "Osa vs Bar")
```

```{r Oki_Bar_chr1_coa2, cache=TRUE, fig.width=15}
plotApairOfChrs(coa2$Osa_Bar, "Chr1", main = "Osa vs Bar (double-collapsed")
```

# Scrambling of a 1-megabase region.

```{r combine_GBs}
x <- subsetByOverlaps(coa2$Oki_Osa, granges(coa2$Oki_Bar)) |> subset(width > 1e5)
y <- subsetByOverlaps(coa2$Oki_Bar, granges(coa2$Oki_Osa)) |> subset(width > 1e5)
findOverlapPairs(x,y)
subsetByOverlaps(coa2$Oki_Osa, gaps(coa2$Oki_Bar)) |> subset(width > 1e5)

ROI <- GRanges("chr2:10000000-11000000")

a <- coa$Oki_Osa |> subsetByOverlaps(ROI) |> plyranges::filter(seqnames == 'chr2', seqnames(query) == 'Chr2')
b <- coa$Oki_Bar |> subsetByOverlaps(ROI) |> plyranges::filter(seqnames == 'chr2', seqnames(query) == 'Chr2')
c <- coa$Bar_Osa |> subsetByOverlaps(range(b$query))                    |> plyranges::filter(seqnames == 'Chr2', seqnames(query) == 'Chr2')

dsList <- list(
  OSA = swap(a) |> gr2dna_seg(),
  OKI = a |> gr2dna_seg(),
  BAR = swap(b) |> gr2dna_seg(),
  Osa = swap(c) |> gr2dna_seg()
)

compList <- list(
  a |> swap() |> gb2comp(),
  b |> gb2comp(),
  c |> gb2comp()
)

genoPlotR::plot_gene_map(dsList, compList, dna_seg_scale=TRUE)

```

# _Ciona_ species

```{r Ciona_Ply_Ros_chr1}
plotApairOfChrs(coa2$Ply_Ros, "BNJZ01000001.1", main = "Ciona intestinalis Plymouth vs Roscoff")
```

```{r Ciona_Ply_Rob_chr1}
plotApairOfChrs(coa2$Ply_Rob, "BNJZ01000001.1", main = "Ciona intestinalis Plymouth vs Ciona robusta")
```

Unfortunately, _C. savignyi_'s genome assembly is too fragmented.  Here is a plot with the largest fragment.

```{r Ciona_Ply_Sav_chr1}
plotApairOfChrs(coa2$Ply_Sav, "BNJZ01000001.1", main = "Ciona intestinalis Plymouth vs Ciona savigny")
```

# Session information

```{r sessionInfo}
sessionInfo()
```
