---
title: "Whole-chromosome parallel plots"
author: 
 - "Charles Plessy"
date: "`r format(Sys.time(), '%d %B, %Y')`"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteEncoding{UTF-8}
  %\VignetteIndexEntry{Whole-chromosome parallel plots}
  %\VignetteEngine{knitr::rmarkdown}
editor_options: 
  chunk_output_type: console
---

```{r setup}
knitr::opts_chunk$set(cache = TRUE, cache.lazy = FALSE)
knitr::opts_knit$set(verbose = TRUE)
```

# Introduction

Pairwise alignments between whole chromosomes are plotted in parallel in order
to reflect synteny and the difference of scrambling between short and long arms.

Figure 1 panel B was generated in this vignette.

# Load R pacakges and data

## Load pacakges

```{r load_packages_and_data}
library('OikScrambling') |> suppressPackageStartupMessages()
genomes <- OikScrambling:::loadAllGenomes()
load("BreakPoints.Rdata")
```

## Phylogenetic cladogram

We assume that the _North Pacific_ and the _Atlantic_ species are more related
to each other than to the _Okinawan_ species.

```{r phylogenetic_tree, fig.height=4, fig.width=6, dev="svg", fig.ext="svg"}
requireNamespace("ade4")
treeLeaf <- function(name, length=NULL) {
  if(!is.null(length)) length <- paste0(':', length)
  paste0(name, length)
}

treeNode <- function(branch1, branch2, length = NULL) {
  if(!is.null(length)) length <- paste0(':', length)
  paste0('(', branch1, ',', branch2, ')', length)
}

addRoot <- function(branch)  paste0(branch, ";")

tree <-
  addRoot(
    treeNode(
      treeNode( length = 2,
        treeLeaf("Okinawa", 1),
        treeLeaf("Kume", 1)
      ),
      treeNode( length = 1,
        treeNode( length = 1,
          treeLeaf("Osaka", 1),
          treeLeaf("Aomori", 1)
        ),
        treeNode( length =1,
          treeLeaf("Norway", 1),
          treeLeaf("Barcelona", 1)
        )
      )
    )
  )

plot(ade4::newick2phylog(tree))
```

# Plot chr2 in six species

We chose chromosome 2 because it has _Hox4_.

## Parallel plot

```{r plot_intersect_GBs_with_tree_with_valid, fig.width=15, fig.height=7, dev=c("svg", "pdf", "png"), fig.ext=c("svg", "pdf", "png")}
KumChr2 <- scaffoldByFlipAndMerge(gbs$Oki_Kum |> swap(), scafs$Kum_Oki, drop = TRUE) |> swap() |> coalesce_contigs()
Aom_Osa <- scaffoldByFlipAndMerge(gbs$Osa_Aom |> swap(), scafs$Aom_Osa, drop = TRUE) |> swap() |> coalesce_contigs()
Aom_Oki <- scaffoldByFlipAndMerge(gbs$Oki_Aom |> swap(), scafs$Aom_Osa, drop = TRUE) |> swap() |> coalesce_contigs()
Bar_Nor <- gbs$Bar_Nor |> swap() |>
  splitSeqLevel("scaffold_3", 700924) |>
  splitSeqLevel("scaffold_8", 1110680) |>
  scaffoldByFlipAndMerge(scafs$Nor_Bar) |>
  coalesce_contigs() |> swap()

dsList6 <- list(
    KUM = KumChr2 |> swap() |> plyranges::filter(seqnames == 'Chr2', seqnames(query) == 'chr2') |> gr2dna_seg(),
    OKI = KumChr2           |> plyranges::filter(seqnames == 'chr2', seqnames(query) == 'Chr2') |> gr2dna_seg(),
    AOM = Aom_Oki |> swap() |> plyranges::filter(seqnames == 'Chr2', seqnames(query) == 'chr2') |> gr2dna_seg(),
    OSA = coa$Osa_Bar       |> plyranges::filter(seqnames == 'Chr2', seqnames(query) == 'Chr2') |> gr2dna_seg(),
    BAR = coa$Bar_Osa       |> plyranges::filter(seqnames == 'Chr2', seqnames(query) == 'Chr2') |> gr2dna_seg(),
    NOR = Bar_Nor |> swap() |> plyranges::filter(seqnames == 'Chr2', seqnames(query) == 'Chr2') |> gr2dna_seg()
)

compList6 <- list(
    Oki_Kum = KumChr2 |> swap() |> plyranges::filter(seqnames == 'Chr2', seqnames(query) == 'chr2') |> gb2comp(),
    Aom_Oki = Aom_Oki           |> plyranges::filter(seqnames == 'chr2', seqnames(query) == 'Chr2') |> gb2comp(),
    Osa_Aom = Aom_Osa |> swap() |> plyranges::filter(seqnames == 'Chr2', seqnames(query) == 'Chr2') |> gb2comp(),
    Osa_Bar = coa$Osa_Bar       |> plyranges::filter(seqnames == 'Chr2', seqnames(query) == 'Chr2') |> gb2comp(),
    Bar_Nor = Bar_Nor           |> plyranges::filter(seqnames == 'Chr2', seqnames(query) == 'Chr2') |> gb2comp()
)

dsList6$KUM$col <- dsList6$KUM$fill <- dsList6$OKI$col <- dsList6$OKI$fill <-
  dsList6$AOM$col <- dsList6$AOM$fill <-
  dsList6$OSA$col <- dsList6$OSA$fill <- dsList6$BAR$col <- dsList6$BAR$fill <-
  dsList6$NOR$col <- dsList6$NOR$fill <- "#A6D854"

compList6$Oki_Kum$color <- compList6$Aom_Oki$color <- compList6$Osa_Aom$color <-
  compList6$Osa_Bar$color <- compList6$Bar_Nor$color <- "#D4EACC"

compList6$Oki_Kum$col <- ifelse(compList6$Oki_Kum$col == "darksalmon", "#D4EACC", "#00A08E")
compList6$Aom_Oki$col <- ifelse(compList6$Aom_Oki$col == "darksalmon", "#D4EACC", "#00A08E")
compList6$Osa_Aom$col <- ifelse(compList6$Osa_Aom$col == "darksalmon", "#D4EACC", "#00A08E")
compList6$Osa_Bar$col <- ifelse(compList6$Osa_Bar$col == "darksalmon", "#D4EACC", "#00A08E")
compList6$Bar_Nor$col <- ifelse(compList6$Bar_Nor$col == "darksalmon", "#D4EACC", "#00A08E")

genoPlotR::plot_gene_map(dsList6, compList6, dna_seg_scale = FALSE,
                         tree = ade4::newick2phylog("((KUM:1,OKI:1):2,((AOM:1,OSA:1):1,(BAR:1,NOR:1):1):1);"),
                         main = "Chromosome 2 in six Oikopleura genome assemblies")
```

# Plot chr2 in four species

We chose chromosome 2 because it has _Hox4_.

## Parallel plot

```{r plot_intersect_GBs_with_tree_with_Kume, fig.width=15, fig.height=7, dev=c("svg", "pdf", "png"), fig.ext=c("svg", "pdf", "png")}
dsList4 <- list(
    KUM = KumChr2 |> swap() |> plyranges::filter(seqnames == 'Chr2', seqnames(query) == 'chr2') |> gr2dna_seg(),
    OKI = KumChr2           |> plyranges::filter(seqnames == 'chr2', seqnames(query) == 'Chr2') |> gr2dna_seg(),
    OSA = coa$Osa_Bar |> plyranges::filter(seqnames == 'Chr2', seqnames(query) == 'Chr2') |> gr2dna_seg(),
    BAR = coa$Bar_Osa |> plyranges::filter(seqnames == 'Chr2', seqnames(query) == 'Chr2') |> gr2dna_seg()
)

compList4 <- list(
    Oki_Kum = KumChr2 |> swap() |> plyranges::filter(seqnames == 'Chr2', seqnames(query) == 'chr2') |> gb2comp(),
    Oki_Osa = coa$Oki_Osa |> plyranges::filter(seqnames == 'chr2', seqnames(query) == 'Chr2') |> gb2comp(),
    Osa_Bar = coa$Osa_Bar |> plyranges::filter(seqnames == 'Chr2', seqnames(query) == 'Chr2') |> gb2comp()
  )

dsList4$KUM$col <- dsList4$KUM$fill <- dsList4$OKI$col <- dsList4$OKI$fill <-
  dsList4$OSA$col <- dsList4$OSA$fill <- dsList4$BAR$col <- dsList4$BAR$fill <-
  "#A6D854"

compList4$Oki_Kum$color <- compList4$Oki_Osa$color <- compList4$Osa_Bar$color <- "#D4EACC"

compList4$Oki_Kum$col <- ifelse(compList4$Oki_Kum$col == "darksalmon", "#D4EACC", "#00A08E")
compList4$Osa_Bar$col <- ifelse(compList4$Osa_Bar$col == "darksalmon", "#D4EACC", "#00A08E")
compList4$Oki_Osa$col <- ifelse(compList4$Oki_Osa$col == "darksalmon", "#D4EACC", "#00A08E")


genoPlotR::plot_gene_map(dsList4, compList4, dna_seg_scale = FALSE,
                         tree = ade4::newick2phylog(addRoot(treeNode(treeNode("OKI", "KUM"), treeNode("OSA", "BAR")))),
                         main = "Chromosome 2 in four Oikopleura genome assemblies")
```


# Plot chr2 in five species

We chose chromosome 2 because it has _Hox4_.

## Parallel plot

```{r plot_intersect_GBs_with_tree_with_Kume_and_aom, fig.width=15, fig.height=7, dev=c("svg", "pdf", "png"), fig.ext=c("svg", "pdf", "png")}

Aom_Osa <- scaffoldByFlipAndMerge(gbs$Osa_Aom |> swap(), scafs$Aom_Osa, drop = TRUE) |> swap() |> coalesce_contigs()
Aom_Oki <- scaffoldByFlipAndMerge(gbs$Oki_Aom |> swap(), scafs$Aom_Osa, drop = TRUE) |> swap() |> coalesce_contigs()

dsList5 <- list(
    KUM = KumChr2 |> swap() |> plyranges::filter(seqnames == 'Chr2', seqnames(query) == 'chr2') |> gr2dna_seg(),
    OKI = KumChr2           |> plyranges::filter(seqnames == 'chr2', seqnames(query) == 'Chr2') |> gr2dna_seg(),
    AOM = Aom_Osa |> swap() |> plyranges::filter(seqnames == 'Chr2', seqnames(query) == 'Chr2') |> gr2dna_seg(),
    OSA = coa$Osa_Bar |> plyranges::filter(seqnames == 'Chr2', seqnames(query) == 'Chr2') |> gr2dna_seg(),
    BAR = coa$Bar_Osa |> plyranges::filter(seqnames == 'Chr2', seqnames(query) == 'Chr2') |> gr2dna_seg()
)

compList5 <- list(
    Oki_Kum = KumChr2 |> swap() |> plyranges::filter(seqnames == 'Chr2', seqnames(query) == 'chr2') |> gb2comp(),
    Aom_Oki = Aom_Oki |> swap() |> plyranges::filter(seqnames == 'Chr2', seqnames(query) == 'chr2') |> gb2comp(),
    Osa_Aom = Aom_Osa |> plyranges::filter(seqnames == 'Chr2', seqnames(query) == 'Chr2') |> gb2comp(),
    Osa_Bar = coa$Osa_Bar |> plyranges::filter(seqnames == 'Chr2', seqnames(query) == 'Chr2') |> gb2comp()
  )

dsList5$KUM$col <- dsList5$KUM$fill <- dsList5$OKI$col <- dsList5$OKI$fill <-
  dsList5$AOM$col <- dsList5$AOM$fill <-
  dsList5$OSA$col <- dsList5$OSA$fill <- dsList5$BAR$col <- dsList5$BAR$fill <-
  "#A6D854"

compList5$Oki_Kum$color <- compList5$Aom_Oki$color <- compList5$Osa_Aom$color <-  compList5$Osa_Bar$color <- "#D4EACC"

compList5$Oki_Kum$col <- ifelse(compList5$Oki_Kum$col == "darksalmon", "#D4EACC", "#00A08E")
compList5$Aom_Oki$col <- ifelse(compList5$Aom_Oki$col == "darksalmon", "#D4EACC", "#00A08E")
compList5$Osa_Aom$col <- ifelse(compList5$Osa_Aom$col == "darksalmon", "#D4EACC", "#00A08E")
compList5$Osa_Bar$col <- ifelse(compList5$Osa_Bar$col == "darksalmon", "#D4EACC", "#00A08E")


genoPlotR::plot_gene_map(dsList5, compList5, dna_seg_scale = FALSE,
                         tree = ade4::newick2phylog(addRoot(treeNode(treeNode("OKI", "KUM"), treeNode(treeNode("OSA", "AOM"), "BAR")))),
                         main = "Chromosome 2 in five Oikopleura genome assemblies")
```

## Strand randomisation indexes

These numbers were added to the plot in Figure 1B.

```{r SR_index_for_tree_with_Kume}
chr2Long <-  \(gb) plyranges::filter(gb, seqnames %in% c("chr2", "Chr2"), seqnames(query) %in% c("chr2", "Chr2"), Arm == "long",  query$Arm == "long")
chr2short <- \(gb) plyranges::filter(gb, seqnames %in% c("chr2", "Chr2"), seqnames(query) %in% c("chr2", "Chr2"), Arm == "short", query$Arm == "short")
chr2both <- \(gb) plyranges::filter(gb, seqnames %in% c("chr2", "Chr2"), seqnames(query) %in% c("chr2", "Chr2"))

KumChr2 <- GenomicBreaks::flagLongShort(KumChr2, longShort$OKI2018.I69)
KumChr2$query$Arm <- Rle("long")
KumChr2$query$Arm[
  end(KumChr2$query) <= (cumsum(seqlengths(coa$Oki_Kum$query)[scafs$Kum_Oki$Chr2$contig]) -1) ["contig_16_1"] &
    seqnames(KumChr2) == "chr2" &
    seqnames(KumChr2$query) == "Chr2"
  ] <- "short"

SRI_Both_Strands <- \(gb) mean(c(strand_randomisation_index(gb)
                                ,strand_randomisation_index(swap(gb))))

(SRIs <- c(
  Kum_Chr2_long  = KumChr2 |> chr2Long()  |> SRI_Both_Strands(),
  Kum_Chr2_short = KumChr2 |> chr2short() |> SRI_Both_Strands(),
  Kum_Chr2_both  = KumChr2 |> chr2both()  |> SRI_Both_Strands(),

  Oki_Osa_long   = coa$Oki_Osa |> chr2Long() |> SRI_Both_Strands(),
  Osa_Oki_long   = coa$Osa_Oki |> chr2Long() |> SRI_Both_Strands(),

  Oki_Osa_short  = coa$Oki_Osa |> chr2short() |> SRI_Both_Strands(),
  Osa_Oki_short  = coa$Osa_Oki |> chr2short() |> SRI_Both_Strands(),

  Oki_Osa_both   = coa$Oki_Osa |> chr2both() |> SRI_Both_Strands(),
  Osa_Oki_both   = coa$Osa_Oki |> chr2both() |> SRI_Both_Strands(),

  Osa_Bar_long   = coa$Osa_Bar |> chr2Long() |> SRI_Both_Strands(),
  Bar_Osa_long   = coa$Bar_Osa |> chr2Long() |> SRI_Both_Strands(),

  Osa_Bar_short  = coa$Osa_Bar |> chr2short() |> SRI_Both_Strands(),
  Bar_Osa_short  = coa$Bar_Osa |> chr2short() |> SRI_Both_Strands(),

  Osa_Bar_both   = coa$Osa_Bar |> chr2both() |> SRI_Both_Strands(),
  Bar_Osa_both   = coa$Bar_Osa |> chr2both() |> SRI_Both_Strands()
))

roundMean <- \(x) round(mean(SRIs[x]), 2)
c(Kum_Chr2_long  = roundMean(c("Kum_Chr2_long")),
  Kum_Chr2_short = roundMean(c("Kum_Chr2_short")),
  Oki_Osa_long   = roundMean(c("Oki_Osa_long",  "Osa_Oki_long")),
  Oki_Osa_short  = roundMean(c("Oki_Osa_short", "Osa_Oki_short")),
  Osa_Bar_long   = roundMean(c("Osa_Bar_long",  "Bar_Osa_long")),
  Osa_Bar_short  = roundMean(c("Osa_Bar_short", "Bar_Osa_short"))
)
```

# Plot pairs of chromosome 1s.

```{r Oki_Bar_chr1, cache=TRUE, fig.width=15}
plotApairOfChrs(coa$Oki_Bar, "chr1", main = "Oki vs Bar")
```

```{r Oki_Osa_chr1, cache=TRUE, fig.width=15}
plotApairOfChrs(coa$Oki_Osa, "chr1", main = "Oki vs Osa")
```

```{r Osa_Oki_chr1, cache=TRUE, fig.width=15}
plotApairOfChrs(coa$Osa_Oki, "Chr1", main = "Osa vs Oki")
```

```{r Osa_Bar_chr1, cache=TRUE, fig.width=15}
plotApairOfChrs(coa$Osa_Bar, "Chr1", main = "Osa vs Bar")
```

```{r Oki_Bar_chr1_coa2, cache=TRUE, fig.width=15}
plotApairOfChrs(coa2$Osa_Bar, "Chr1", main = "Osa vs Bar (double-collapsed")
```

# Scrambling of a 1-megabase region.

```{r combine_GBs}
x <- subsetByOverlaps(coa2$Oki_Osa, granges(coa2$Oki_Bar)) |> subset(width > 1e5)
y <- subsetByOverlaps(coa2$Oki_Bar, granges(coa2$Oki_Osa)) |> subset(width > 1e5)
findOverlapPairs(x,y)
subsetByOverlaps(coa2$Oki_Osa, gaps(coa2$Oki_Bar)) |> subset(width > 1e5)

ROI <- GRanges("chr2:10000000-11000000")

a <- coa$Oki_Osa |> subsetByOverlaps(ROI) |> plyranges::filter(seqnames == 'chr2', seqnames(query) == 'Chr2')
b <- coa$Oki_Bar |> subsetByOverlaps(ROI) |> plyranges::filter(seqnames == 'chr2', seqnames(query) == 'Chr2')
c <- coa$Bar_Osa |> subsetByOverlaps(range(b$query))                    |> plyranges::filter(seqnames == 'Chr2', seqnames(query) == 'Chr2')

dsList <- list(
  OSA = swap(a) |> gr2dna_seg(),
  OKI = a |> gr2dna_seg(),
  BAR = swap(b) |> gr2dna_seg(),
  Osa = swap(c) |> gr2dna_seg()
)

compList <- list(
  a |> swap() |> gb2comp(),
  b |> gb2comp(),
  c |> gb2comp()
)

genoPlotR::plot_gene_map(dsList, compList, dna_seg_scale=TRUE)

```

# _Ciona_ species

```{r Ciona_Ply_Ros_chr1}
plotApairOfChrs(coa2$Ply_Ros, "BNJZ01000001.1", main = "Ciona intestinalis Plymouth vs Roscoff")
```

```{r Ciona_Ply_Rob_chr1}
plotApairOfChrs(coa2$Ply_Rob, "BNJZ01000001.1", main = "Ciona intestinalis Plymouth vs Ciona robusta")
```

Unfortunately, _C. savignyi_'s genome assembly is too fragmented.  Here is a plot with the largest fragment.

```{r Ciona_Ply_Sav_chr1}
plotApairOfChrs(coa2$Ply_Sav, "BNJZ01000001.1", main = "Ciona intestinalis Plymouth vs Ciona savigny")
```

# Session information

```{r sessionInfo}
sessionInfo()
```
