---
title: "Motif search and analysis"
author: 
 - "Charles Plessy"
date: "`r format(Sys.time(), '%d %B, %Y')`"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteIndexEntry{Motif search and analysis}
  %\VignetteEncoding{UTF-8}
editor_options: 
  chunk_output_type: console
---

```{r setup}
knitr::opts_chunk$set(cache = TRUE)
```

# Introduction

Search for motifs in regions near genomic breaks.  By default the computations
that take time are loaded from R objects.  The source code to generate them
is only executed when the object files can not be found.

# Short summary

Motifs found in _isolated regions_ are made of one or two stretches of `A`s (or
`T`s when found in reverse-complement).  Typical example: `nrAAAwnnnnrAAAn`.
The same motifs are found in _breakpoint regions_, with the addition of
species-specific motifs such as `GsCsCkwmGsGsC` (Okinawa) or `GCT..GCT` (North).
In addition to this, other species-specific palindromic motifs are enriched in
breakpoint regions flanking inversions, such as `CCTGA.TCAGG` in the Okinawan
genome and `nGAGGCmACGmsCGTTCn` or `nACGCyCryTCATTCTwTy` in Northern genomes.

This search is intuition-driven and not exhaustive.

# Load pacakges

Core packages that provide functions we use a lot.

```{r loadlib}
library('OikScrambling') |> suppressPackageStartupMessages()
genomes <- OikScrambling:::loadAllGenomes(compat = F)
annots  <- OikScrambling:::loadAllAnnotations() |> suppressWarnings()
load("BreakPoints.Rdata")

library('ggplot2') |> suppressPackageStartupMessages()

requireNamespace("ggseqlogo")
requireNamespace("JASPAR2020") |> suppressPackageStartupMessages()
library("rGADEM") |> suppressPackageStartupMessages()
```

# Motif searches

[GADEM](https://www.niehs.nih.gov/research/resources/software/biostatistics/gadem/index.cfm)'s
documentation states:

> _The background sequences are simulated using the [a,c,g,t] frequencies in
> the input sequences, with length matched between the two sets. The background
> sequences are used as the random sequences for assessing motif enrichment in
> the input data._

```{r functions_for_motif_search}
# Pass genome information through metadata
getSeq_ <- function(x) {
  seqs <- getSeq(x)
  genome(seqs) <- unique(genome(x))
  seqs
}

# Note that GADEM crashes on sequences longer than ~48 kbp.
# Slow !
doGadem <- function(seqs) {
  seqs <- seqs[width(seqs) < 48000]
  bsgenome_name <- genome(seqs@metadata$seqinfo) |> unique()
  bsgenome_obj <- BSgenome::getBSgenome(bsgenome_name)
  GADEM(seqs, verbose=1, genome=bsgenome_obj)
}

gadems <- SimpleList()
```


## Isolated alignments

Aligned regions that do not coalesce into collinear regions are relatively
depleted in genes, and show no enrichment of exons versus introns.  They are
also depleted in repeats, and are mildly enriched in CNEs (see
`vignette("GenomicFeatures", package = "OikScrambling")` for details).
Altogether, they can not be summarised in one feature.  So let's see if there
are motifs enriched there.

```{r search_in_aln_nonCoa}
isolAln_seqs <- sapply(gbs[1:15], \(gb) gb[gb$nonCoa] |> getSeq_()) |> SimpleList()
sapply(isolAln_seqs, length)
if(!file.exists("gadems.alnNonCoa.Rds")) {
  gadems$isolAln <- BiocParallel::bplapply(isolAln_seqs, doGadem) |> SimpleList()
  saveRDS(gadems$isolAln, file = "gadems.alnNonCoa.Rds")
} else {
  # File names still follow old nomenclature: isolated ~ aligned non-coalesced
  gadems$isolAln <- readRDS("gadems.alnNonCoa.Rds")
}
```

## Breakpoint regions

There regions are not aligned and not within a collinear region.

```{r search_in_brkReg}
brkReg_seqs <- sapply(unmap[1:15], getSeq_) |> SimpleList()
sapply(brkReg_seqs, length)
if(!file.exists("gadems.unmap.Rds")) {
  gadems$brkReg <- BiocParallel::bplapply(brkReg_seqs, doGadem) |> SimpleList()
  saveRDS(gadems$brkReg, file = "gadems.unmap.Rds")
} else {
  gadems$brkReg <- readRDS("gadems.unmap.Rds")
}
```

## Unaligned regions near inversions

I wanted to search search for motifs near inversions to see if they can
lead to an explanation.  Here I do not discriminate between breakpoint region or
bridge region among unaligned regions.  Maybe I should...

See `vignette("Inversions", package = "OikScrambling")` for details on inversion.

### Isolate the left-side gaps in inversions

```{r search_in_gaps_near_inversions}
inv.lgaps <- sapply(coa, leftInversionGaps) |> SimpleList()
inv.lgaps_seqs <- sapply(inv.lgaps[1:15], getSeq_) |> SimpleList()
sapply(inv.lgaps_seqs, length)
if(!file.exists("gadems.inv.lgaps.Rds")) {
  gadems$inv.lgaps <- BiocParallel::bplapply(inv.lgaps_seqs, doGadem) |> SimpleList()
  saveRDS(gadems$inv.lgaps, file = "gadems.inv.lgaps.Rds")
} else {
  gadems$inv.lgaps <- readRDS("gadems.inv.lgaps.Rds")
}
```

## Inversions

```{r search_in_inversions}
invRegions <- sapply(coa[1:15], \(gr) filterInversions(flagInversions(gr))) |> SimpleList()
inv_seqs <-  sapply(invRegions, getSeq) |> SimpleList()
sapply(inv_seqs, length)
# sapply(inv_seqs, length)
# if (!file.exists("gadems.invs.Rds")) {
#   gadems$invs <- BiocParallel::bplapply(inv_seqs, doGadem) |> SimpleList()
#   saveRDS(gadems$inv, file = "gadems.invs.Rds")
# } else {
#   gadems$invs <- readRDS("gadems.invs.Rds")
# }
```

## Promoters

```{r search_in_promoters}
#prom_seqs <- sapply(annots[1:3], \(gb) suppressWarnings(trim(promoters(gb)))) |>
#  sapply(getSeq_) |> SimpleList()
#sapply(prom_seqs, length)
#if (!file.exists("gadems.proms.Rds")) {
#  gadems$proms <- BiocParallel::bplapply(prom_seqs, doGadem) |> SimpleList()
#  saveRDS(gadems$proms, file = "gadems.proms.Rds")
#} else {
#  gadems$proms <- readRDS("gadems.proms.Rds")
#}
```

## Introns

```{r search_in_introns}
#intron_seqs <- sapply(annots[1:3], GenomicFeatures::intronicParts) |>
#  sapply(getSeq_) |> SimpleList()
#sapply(intron_seqs, length)
#if (!file.exists("gadems.introns.Rds")) {
#  gadems$introns <- BiocParallel::bplapply(intron_seqs, doGadem) |> SimpleList()
#  saveRDS(gadems$introns, file = "gadems.introns.Rds")
#} else {
#  gadems$introns <- readRDS("gadems.introns.Rds")
#}
```

## Summary

Number of motifs found

```{r summarise_gadems}
sapply(gadems, sapply, \(x) length(x@motifList))
```

# Display

Service functions.

```{r plot_all_motifs, fig.width=6}
showAllMotifs <- function(gadems) {
  sapply(names(gadems), \(name) {
    sapply(seq_along(gadems[[name]]@motifList), \(n) {
      gadems[[name]][[n]]@consensus
    })
  })
}

plotAllMotifs <- function(gadems) {
  for (curObjName in names(gadems)) {
    for (n in seq_along(gadems[[curObjName]]@motifList)) {
      p <- plotOneMotif(gadems[[curObjName]][[n]], curObjName)
      print(p)
    }
  }
}

plotOneMotif <- function(gadem, pairName = NULL, range = NULL) {
  if(is.null(range)) {
    pwm <- rGADEM::getPWM(gadem)
    consensus <- gadem@consensus
  } else {
    pwm <- rGADEM::getPWM(gadem)[,range]
    consensus <- gadem@consensus |> subseq(min(range), max(range))
  }
  title = paste0("Motif consensus: ", consensus)
  if(!is.null(pairName))
  title <- paste0("Genome pair: ", pairName, ", ", title)
  # Suppressed warning message: `guides(<scale> = FALSE)` is deprecated. Please use `guides(<scale> = "none")` instead. 
  p <- ggseqlogo::ggseqlogo(pwm) |> suppressWarnings()
  p  + ggtitle(title)
}
```

## Isolated alignments

Many comparison return a motif that looks like `AAAA.....AAAA`, and some also
return `AAGAA` (or their reverse-complement).

The Oki - Kume comparison shows T-rich motifs (note that they could have been
A-rich reverse-complement).

Sometimes a weaker A-rich / T-rich motif is also found.

```{r show_isolAln_motifs}
showAllMotifs(gadems$isolAln)
```

Here are a few graphical examples.

```{r plot_some_isolAln_motifs}
plotOneMotif(gadems$isolAln$Oki_Osa[[1]], "Okinawa - Osaka")
plotOneMotif(gadems$isolAln$Oki_Osa[[2]], "Okinawa - Osaka")
plotOneMotif(gadems$isolAln$Oki_Kum[[1]], "Okinawa - Kume")
plotOneMotif(gadems$isolAln$Oki_Kum[[2]], "Okinawa - Kume")
plotOneMotif(gadems$isolAln$Osa_Oki[[2]], "Osaka - Okinawa")
plotOneMotif(gadems$isolAln$Osa_Bar[[2]], "Osaka - Barcelona")
```

## Breakpoint regions

We still find `AAAA.....AAAA`,`AAGAA`, `A`-rich (or their reverse-complement),
but not as systematicaly.

We start to see species-specific motifs:

  - Variations over `GsCsCkwmGsGsC` in some Oki -> other comparisons.
  - Variations over `GCT..GCT` in some Nor -> other comparisons (pay attention
    the reverse complements).


```{r show_brkReg_motifs}
showAllMotifs(gadems$brkReg)
```

Here are a few graphical examples.

```{r plot_some_brkReg_motifs}
plotOneMotif(gadems$brkReg$Oki_Osa[[2]], "Okinawa - Osaka")
plotOneMotif(gadems$brkReg$Oki_Kum[[1]], "Okinawa - Kume")
plotOneMotif(gadems$brkReg$Osa_Oki[[1]], "Osaka - Okinawa")
plotOneMotif(gadems$brkReg$Bar_Osa[[1]], "Barcelona - Okinawa")
```

## Unaligned regions left to inversion

  - We still find `AAAA.....AAAA`,`AAGAA`, `A`-rich (or their reverse-complement).
  - We see palindromic motifs (`CCTGA.TCAGG`, `aaGCCGc...gCGGCtt`) in the Okinawan genome.
  - Two other motifs are found when comparing Oki to Aom (`nGAGGCmACGmsCGTTCn`)
    and to Nor `nACGCyCryTCATTCTwTy`).
  - In the Osaka genome, we see a the `GCT..GCT` motif again.


```{r show_invLeft_motifs}
showAllMotifs(gadems$inv.lgaps)
```


Here are a few graphical examples.

### `CCTGA.TCAGG`

```{r plot_some_CCTGA.TCAGG_motifs_from_invs}
plotOneMotif(gadems$inv.lgaps$Oki_Osa[[1]], "Okinawa - Osaka")
plotOneMotif(gadems$inv.lgaps$Oki_Kum[[2]], "Okinawa - Kume")
```

### `aaGCCGc...gCGGCtt`

```{r plot_some_aaGCCGc...gCGGCtt_motifs_from_invs}
plotOneMotif(gadems$inv.lgaps$Oki_Osa[[3]], "Okinawa - Osaka")
plotOneMotif(gadems$inv.lgaps$Oki_Kum[[1]], "Okinawa - Kume")
```

### Other from Okinawan genome

```{r plot_some_more_motifs_from_invs}
plotOneMotif(gadems$inv.lgaps$Oki_Aom[[6]], "Okinawa - Aomori")
plotOneMotif(gadems$inv.lgaps$Oki_Nor[[4]], "Okinawa - Norway")
```

### `GCT..GCT`

```{r plot_GCT..GCT_motifs_from_invs}
plotOneMotif(gadems$inv.lgaps$Osa_Oki[[1]], "Osaka - Okinawa")
plotOneMotif(gadems$inv.lgaps$Osa_Bar[[1]], "Osaka - Barcelona")
```

### `GCTTCGCTTC`


```{r plot_GCTTCGCTTC_from_invs}
plotOneMotif(gadems$inv.lgaps$Osa_Kum[[1]], "Osaka - Kume")
plotOneMotif(gadems$inv.lgaps$Bar_Osa[[1]], "Barcelona - Okinawa")
```

### `CAA....TTGGTCA`

```{r plot_CAA....TTGGTCA_from_invs}
plotOneMotif(gadems$inv.lgaps$Osa_Kum[[5]], "Osaka - Kume")
plotOneMotif(gadems$inv.lgaps$Osa_Oki[[4]], "Osaka - Okinawa")
plotOneMotif(gadems$inv.lgaps$Osa_Nor[[2]], "Osaka - Norway")
```

# AT richness

Breakpoint regions have a much higher AT-richness than aligned ones.

```{r at_richness2}
rbind(
  isolAln = sapply(isolAln_seqs, \(seq) letterFrequency(seq, "AT", as.prob = TRUE) |> mean()),
  brkReg     = sapply(brkReg_seqs, \(seq) letterFrequency(seq, "AT", as.prob = TRUE) |> mean()),
  inv.lgaps = sapply(inv.lgaps_seqs, \(seq) letterFrequency(seq, "AT", as.prob = TRUE) |> mean()),
  inv       = sapply(inv_seqs, \(seq) letterFrequency(seq, "AT", as.prob = TRUE) |> mean())
)

(genomes.AT <- sapply(genomes, \(g)
  weighted.mean(letterFrequency(getSeq_(g), "AT", as.prob = TRUE), seqlengths(g))))

## This difference is significant.
t.test(letterFrequency(inv_seqs$Oki_Osa, "AT", as.prob = TRUE), letterFrequency(inv.lgaps_seqs$Oki_Osa, "AT", as.prob = TRUE))
t.test(letterFrequency(inv_seqs$Osa_Oki, "AT", as.prob = TRUE), letterFrequency(inv.lgaps_seqs$Osa_Oki, "AT", as.prob = TRUE))
t.test(letterFrequency(inv_seqs$Oki_Bar, "AT", as.prob = TRUE), letterFrequency(inv.lgaps_seqs$Oki_Bar, "AT", as.prob = TRUE))
t.test(letterFrequency(inv_seqs$Oki_Kum, "AT", as.prob = TRUE), letterFrequency(inv.lgaps_seqs$Oki_Kum, "AT", as.prob = TRUE))
```


A palindromic (rAAGCsGCwwmkCGrCTTyn) motif is found in the left gaps.

Or could it be `GCCGCnnnGCGGC` ?  Its frequency in genomes is something like:
```
  Oki   Osa   Bar   Kum   Aom   Nor 
10827  2273  1967 10880  2210  2247 
```

On the other hand, a motif containing GCGAAGCGAA identified in Osa_Oki shows:

```
  Oki   Osa   Bar   Kum   Aom   Nor 
  483 21360 23340   501 21594 16413 
```

# Search for the motifs in JASPAR

```{r screenJaspar}
checkJaspar <- function (pwm, range=NULL) {
  if(is.null(range))
    PWMat <- TFBSTools::PWMatrix(ID=pwm@consensus, profileMatrix = pwm@pwm)
  else
    PWMat <- TFBSTools::PWMatrix(ID = subseq(pwm@consensus, min(range), max(range)),
                                 profileMatrix = pwm@pwm[,range])
  #---- see https://compgenomr.github.io/book/motif-discovery.html
  pwmLib <- TFBSTools::getMatrixSet(
    JASPAR2020::JASPAR2020,
    opts = list(
      collection = 'CORE',
   #   species    = 'Homo sapiens',
      matrixtype = 'PWM'))
  
  pwm_sim <- TFBSTools::PWMSimilarity( pwmLib, PWMat,  method = 'Pearson')

  pwmLibDf <- data.frame(
    ID        = sapply(pwmLib, TFBSTools::ID),
    name      = sapply(pwmLib, TFBSTools::name),
    row.names = seq_along(pwmLib)
  )
  pwmLibDf$similarity <- pwm_sim[pwmLibDf$ID]
  pwmLibDf[order(-pwmLibDf$similarity),]
}

gadems$inv.lgaps$Oki_Osa[[1]] |> checkJaspar() |> head()
gadems$inv.lgaps$Oki_Kum[[1]] |> checkJaspar() |> head()

# Let's trim the motif.
gadems$inv.lgaps$Oki_Kum[[1]] |> plotOneMotif(range = 2:9)
gadems$inv.lgaps$Oki_Osa[[1]] |> checkJaspar(range = 2:9) |> head()
```

But a search in Jaspar with either the full motif or its halof did not reveal
strong hits.  Visual inspection of PRDM9 sequence logos via Google did not show
similarity either.


```
# writeXStringSet(lgapsSeq_Oki_O, file="firstGaps.fasta")
# I used this FASTA export to search for motifs with GLAM2, and it also
# returned A/T polymers.
```

The GLAM2 motif finder was run on this file, and then to screen the genome.
Hits coordinates were encoded in a BED file uploaded to the ZENBU genome browser.
Visual inspection gave me the impression that the motif might be related to
repeats.

GLAM2 search of the gap sequences did not return a motif similar to "AAGCsGCwwmkCGrCTTyn".

### Whole-genome search of the motif

Disappointingly, the motif is found equally in gaps and inversions when searched
with its PWM.

```{r PWM_collection_Oki}
gadems$isolAln |> showAllMotifs()

gadems_to_export <- SimpleList()
# gadems_to_export$nrAAAwnnnnrAAAn            <- gadems $ isolAln   $ Oki_Osa [1] # Too many hits
gadems_to_export$GsCsCkwmGsGsCkTyn          <- gadems $ brkReg    $ Oki_Osa [2]
gadems_to_export$mCGsCCCkwmGGGCCGkCwmrsCCCy <- gadems $ brkReg    $ Oki_Kum [1]
gadems_to_export$nCCTGAsTCAGGkTyTn          <- gadems $ inv.lgaps $ Oki_Osa [1]
gadems_to_export$nGCCGCwwmkCGGCTTyTn        <- gadems $ inv.lgaps $ Oki_Osa [3]
gadems_to_export$mGCTymGCTymGCnmwTTTn       <- gadems $ brkReg    $ Osa_Oki [1]
gadems_to_export$wGCTTCGCTTCGCk             <- gadems $ brkReg    $ Bar_Osa [1]
gadems_to_export$mCAAyTwATTGGTCAyTTTn       <- gadems $ inv.lgaps $ Osa_Oki [4]

# Search for all selected motifs in all genomes
gadems_to_export_pwms <- sapply(gadems_to_export, rGADEM::getPWM) |> SimpleList()
gadems_to_export_hits <- sapply(gadems_to_export_pwms, \(pwm) {
  sapply(genomes, matchPWM, pwm = pwm) |> SimpleList()
}) |> suppressWarnings() |> SimpleList()

# Just fix names
names(gadems_to_export_hits) <- names(gadems_to_export_pwms) <- names(gadems_to_export)

sapply(gadems_to_export_hits, sapply, length) |> t()
```

Overlap between detected motifs in Oki

```{r Oki_pwm_found_overlap}
sapply(names(gadems_to_export_hits), \(motif) {
  sapply(names(gadems_to_export_hits), \(motif2) {
    sum(width(intersect(gadems_to_export_hits[[motif]][["OKI2018.I69"]],gadems_to_export_hits[[motif2]][["OKI2018.I69"]])))
  })
})
```

Overlap between detected motifs in Osa

```{r Osa_pwm_found_overlap}
sapply(names(gadems_to_export_hits), \(motif) {
  sapply(names(gadems_to_export_hits), \(motif2) {
    sum(width(intersect(gadems_to_export_hits[[motif]][["OSKA2016v1.9"]],gadems_to_export_hits[[motif2]][["OSKA2016v1.9"]])))
  })
})
```

Export to ZENBU

```{r export_for_ZENBU}
toZenbuOki <- c(
  gadems_to_export_hits$GsCsCkwmGsGsCkTyn$OKI2018.I69          |> plyranges::mutate(Name = "GsCsCkwmGsGsCkTyn"),
  gadems_to_export_hits$mCGsCCCkwmGGGCCGkCwmrsCCCy$OKI2018.I69 |> plyranges::mutate(Name = "mCGsCCCkwmGGGCCGkCwmrsCCCy"),
  gadems_to_export_hits$nCCTGAsTCAGGkTyTn$OKI2018.I69          |> plyranges::mutate(Name = "nCCTGAsTCAGGkTyTn"),
  gadems_to_export_hits$nGCCGCwwmkCGGCTTyTn$OKI2018.I69        |> plyranges::mutate(Name = "nGCCGCwwmkCGGCTTyTn"),
  gadems_to_export_hits$mGCTymGCTymGCnmwTTTn$OKI2018.I69       |> plyranges::mutate(Name = "mGCTymGCTymGCnmwTTTn"),
  gadems_to_export_hits$wGCTTCGCTTCGCk$OKI2018.I69             |> plyranges::mutate(Name = "wGCTTCGCTTCGCk"),
  gadems_to_export_hits$mCAAyTwATTGGTCAyTTTn$OKI2018.I69       |> plyranges::mutate(Name = "mCAAyTwATTGGTCAyTTTn")
)
toZenbuOki$sequence <- as.character(toZenbuOki$string)
toZenbuOki$string <- NULL

# Export to GFF3
rtracklayer::export.gff3(toZenbuOki |> sort(i=T), "PWMs_of_interest_Oki.gff")

# And also to BED for expression tracks
names(toZenbuOki) <- toZenbuOki$Name
rtracklayer::export.bed(toZenbuOki |> sort(i=T), "PWMs_of_interest_Oki.bed")
```

```{r PWM_score_on_chrs, warning=FALSE}
# No exact match or can not the pattern acept Ns ?
#vmatchPattern("nCCTGAsTCAGGkTyTn", BSgenome.Oidioi.OIST.OKI2018.I69)

sapply(gadems_to_export_hits$nCCTGAsTCAGGkTyTn, length)
hist(gadems_to_export_hits$nCCTGAsTCAGGkTyTn$OKI2018.I69$score)
hist(gadems_to_export_hits$nCCTGAsTCAGGkTyTn$OSKA2016v1.9$score)

pwmHit <- gadems_to_export_hits$nCCTGAsTCAGGkTyTn$OKI2018.I69

library("Gviz")
options(ucscChromosomeNames=FALSE)
gen <- GenomeAxisTrack(name = "genome")
a <- AnnotationTrack(longShort$Oki, chromosome = "chr1")
trk <- DataTrack(pwmHit |> (\(.){strand(.)<-"*";.})(), name = "pwmHits", chromosome = "chr1")
plotTracks(list(gen, a, trk))
a <- AnnotationTrack(longShort$Oki, chromosome = "chr2")
trk <- DataTrack(pwmHit |> (\(.){strand(.)<-"*";.})(), name = "pwmHits", chromosome = "chr2")
plotTracks(list(gen, a, trk))
a <- AnnotationTrack(longShort$Oki, chromosome = "PAR")
trk <- DataTrack(pwmHit |> (\(.){strand(.)<-"*";.})(), name = "pwmHits", chromosome = "PAR")
plotTracks(list(gen, a, trk))
a <- AnnotationTrack(longShort$Oki, chromosome = "XSR")
trk <- DataTrack(pwmHit |> (\(.){strand(.)<-"*";.})(), name = "pwmHits", chromosome = "XSR")
plotTracks(list(gen, a, trk))
```

```{r}

length(subsetByOverlaps(gadems_to_export_hits$nCCTGAsTCAGGkTyTn$OKI2018.I69, inv.lgaps$Oki_Osa))
length(subsetByOverlaps(gadems_to_export_hits$nCCTGAsTCAGGkTyTn$OKI2018.I69, invRegions$Oki_Osa))

hist(score(subsetByOverlaps(gadems_to_export_hits$nCCTGAsTCAGGkTyTn$OKI2018.I69, inv.lgaps$Oki_Osa)))
hist(score(subsetByOverlaps(gadems_to_export_hits$nCCTGAsTCAGGkTyTn$OKI2018.I69, invRegions$Oki_Osa)))

# Scores in gaps are higher than average.
t.test(score(gadems_to_export_hits$nCCTGAsTCAGGkTyTn$OKI2018.I69), score(subsetByOverlaps(gadems_to_export_hits$nCCTGAsTCAGGkTyTn$OKI2018.I69, inv.lgaps$Oki_Osa)))

# Scores in gaps and inversions are not different 
t.test(score(subsetByOverlaps(gadems_to_export_hits$nCCTGAsTCAGGkTyTn$OKI2018.I69, invRegions$Oki_Osa)), score(subsetByOverlaps(gadems_to_export_hits$nCCTGAsTCAGGkTyTn$OKI2018.I69, inv.lgaps$Oki_Osa)))

# PWM hits seem to be more abundant on short arms.
#OKI_longShort$pwmHitsPerMb <- countOverlaps(OKI_longShort, pwmHits$Oki) / width(OKI_longShort) * 1e6
# But with a naive approach the difference is not statistically significant.
t.test(c(159, 185, 188), c(77, 142, 160))
```
