---
title: "Scaling breakpoints"
author: 
 - "Michael Mansfield"
 - "Charles Plessy"
date: "`r format(Sys.time(), '%d %B, %Y')`"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteEncoding{UTF-8}
  %\VignetteIndexEntry{Scaling breakpoints}
  %\VignetteEngine{knitr::rmarkdown}
editor_options: 
  chunk_output_type: console
---

```{r setup}
knitr::opts_chunk$set(cache = TRUE)
```

# Introduction

In this vignette computes the observed number of breakpoints for a given species
pair by a few key metrics: 

* Millions of years since divergence
* Branch length (i.e., a kind of substitution rate)
* dN/dS

**Panel E of Figure 7 was generated with this vignette (colors were edited by
hand).**

**The vignette also produce trees, which were edited for panels 7Aâ€“D.**

The core functions used here are maintained in our _GenomicBreaks_ R package,
which is fully documented at: <https://oist.github.io/GenomicBreaks>.

# Setup

## Load R packages and data

```{r load_packages_and_data}
library('OikScrambling') |> suppressPackageStartupMessages()
library('dplyr') |> suppressPackageStartupMessages()
genomes <- OikScrambling:::loadAllGenomes()
library('ggtree') |> suppressPackageStartupMessages()
library('treeio') |> suppressPackageStartupMessages()
library('phangorn') |> suppressPackageStartupMessages()
load("BreakPoints.Rdata")
```

## Service functions

```{r local_functions}
toTreedata <- function(tree) as.treedata(midpoint(read.tree(text=tree)))
plotTree <- function(sp) {
  ggtree(sp, ladderize=T) +
    geom_tiplab(align=T) + hexpand(0.5) +
    geom_text(aes(x=branch, label=round(branch.length,2)), nudge_y=0.1) +
    geom_treescale() + geom_rootedge()
}
```

# Inputting distance metrics

The resulting trees are displayed to help proofchecking.

## _Oikopleura_

```{r species_trees_Oiks, fig.height=6, fig.width=6, message=FALSE, warning=FALSE, dev=c("svg", "png")}
species <- SimpleList()
species$Oikopleura <- "(Kum:0.01136111178249175555,((Osa:0.00925951950869623869,Aom:0.00920798155370218530)100:0.05729289951873162146,(Bar:0.00721724152898447612,Nor:0.00688754343189192098)100:0.05951986243208125943)100:0.31437914450027298896,Oki:0.01142149265410641150);" |> toTreedata()
plotTree(species$Oikopleura)
```

## _Ciona_

```{r species_trees_Ciona, fig.height=6, fig.width=6, message=FALSE, warning=FALSE, dev=c("svg", "png")}
species$Ciona <- "(Rob:0.03128841990749874763,(Ros:0.01304087257513942937,Ply:0.01288782021364278871)100:0.02128682526879402057,Sav:0.55033545948338902232);" |> toTreedata()
plotTree(species$Ciona)
```

## _Drosophila_

The tree originates from <https://github.com/SchriderLab/Drosophila_phylogeny>,
and the relevant clades were extracted with:

```
tree <- read.tree('Drosophila_phylogeny/tree/iqtree.tre')
tree <- drop.tip(tree, which(!tmp$tip.label %in% c('D_melanogaster', 'D_busckii', 'D_yakuba', 'D_subpulchrella', 'D_mauritiana')))
write.tree(tree, 'drosophila_4spp.newick')
```

```{r species_trees_Droso, fig.height=6, fig.width=6, message=FALSE, warning=FALSE, dev=c("svg", "png")}
species$Drosophila <- "(Dbu:0.3372973403,(Dsu:0.0947791254,(Dya:0.0703231461,(Dma:0.0280994984,Dme:0.0264284139)100_1_100:0.0267889071)100_1_100:0.0834632289)100_1_100:0.3712228549)100_1_100;" |> toTreedata()
plotTree(species$Drosophila)
```

## _Caenorhabditis_

Tree topologies taken from <https://zenodo.org/record/1402254/files/RAxML_species_tree.nwk>

The Zenodo link also has trees calculated using ASTRAL and PhyloBayes. However,
since the other trees here are ML trees, the RAxML tree was elected for maximum
comparability.

```{r species_trees_Caeno, fig.height=6, fig.width=6, message=FALSE, warning=FALSE, dev=c("svg", "png")}
species$Caenorhabditis <- ape::read.tree(text="(CGUAD:0.10367478589587060533,(((CSP43:0.12383810371576783882,CVIRI:0.15037624074929095697)100:0.01497242793907414826,((CPLIC:0.26149908669946614337,((DCORO:0.59783158562631621979,CMONO:0.24005751591713977988)100:0.04780411675113813846,CSP21:0.36368990218252150726)100:0.10208494374306537056)100:0.03524817284922288768,((CCAST:0.01140212554998711741,CANGA:0.01414003367272922800)100:0.05852891527717005854,CSP38:0.05923127343611131468)100:0.11896541306229711787)98:0.01073075219687256478)100:0.02583935505413810507,((CKAMA:0.05811607292620620674,((CELEG:0.04500887872372311138,CSP34:0.07438201522663016874)100:0.00705149839556874856,(((CLATE:0.00465806789922303816,CREMA:0.00437786777324163708)100:0.02881157060040358958,((CNIGO:0.00418426739023885358,CBRIG:0.00712154940320354219)100:0.02811281384583694601,(CSINI:0.01686763323477479329,(CSP26:0.01529017647605383778,CSP40:0.02038338447129417474)100:0.00329577341321305025)100:0.01374371022227172112)100:0.01161394824780575923)100:0.00600824892975430989,((CDOUG:0.02983523068133404899,CBREN:0.03506566315430653669)80:0.00382524869406435401,(CWALL:0.01854144710471255370,CTROP:0.02398428150833109909)100:0.01471009970302557951)100:0.00927549430957550110)100:0.00763851852828213371)100:0.00805135107229970441)100:0.01328077288786689257,((((CNOUR:0.00956145701022028879,CSP29:0.00805739385901662990)100:0.01930453335852322372,(CSP39:0.03253018121281636005,CSP28:0.02417332848702777315)100:0.00890124689939411802)100:0.00884911431899864341,CMACR:0.03525690105425406673)100:0.02033464491626665177,(CJAPO:0.08234400443891473631,(CAFRA:0.02943481705945342336,CSP32:0.02434121084253646969)100:0.04055371130568757865)79:0.00548050967800332942)100:0.00698694826746024816)100:0.06205429593773054836)100:0.02963813048003952413,CSP31:0.12329044085858320567);") |>
  root("DCORO") |>
  reorder.phylo(order='cladewise')

species$Caenorhabditis <- drop.tip(species$Caenorhabditis, which(!species$Caenorhabditis$tip.label %in% c('CELEG', 'CSP34', 'CREMA', 'CNIGO', 'CBRIG')))

species$Caenorhabditis$tip.label <- c('Cel', 'Cin', 'Cre', 'Cni', 'Cbr')
write.tree(species$Caenorhabditis)
species$Caenorhabditis <- as.treedata(species$Caenorhabditis)
plotTree(species$Caenorhabditis)
```

```{r species_trees_Caeno_4, fig.height=6, fig.width=6, message=FALSE, warning=FALSE, dev=c("svg", "png", "pdf"), fig.ext=c('svg', 'png', 'pdf')}


```

## Distances

With the species trees loaded, we can calculate per-pair branch length distances using `ape::cophenetic()`.

```{r species_distances}
species_dist <- SimpleList()
species_dist$cophenetic <- lapply(species, function(sp) cophenetic.phylo(sp@phylo)) |> SimpleList()
```

### Tunicates

For tunicates, the distances expressed in million years are taken from the
molecular clock analysis **except** for the same-species pairs where they are
arbitrarily set to 3, as an upper-bound estimate.  The rationale is that as we
are testing for speed, using the maximally conceivable distance prevents us from
overestimating that speed.

```{r clock_distances}
# These are taken from the molecular clock analysis.
species_dist$mya <- SimpleList()
species_dist$mya$Oikopleura <- data.frame(
  'Oki'=c(0,       3,       24.5347, 24.5347, 24.5347),
  'Kum'=c(3,       0,       24.5347, 24.5347, 24.5347),
  'Osa'=c(24.5347, 24.5347, 0,       7.2383,  3),
  'Bar'=c(24.5347, 24.5347, 7.2383,  0,       7.2383),
  'Aom'=c(24.5347, 24.5347, 3,       7.2383,  0)
)

species_dist$mya$Ciona <- data.frame(
  'Sav'=c(0,        101.4633, 101.4633, 101.4633),
  'Rob'=c(101.4633, 0,        11.8551, 11.8551),
  'Ply'=c(101.4633, 11.8551,  0, 3),
  'Ros'=c(101.4633, 11.8551,  3, 0)
)
```

### Drosophila

For Drosophila, the numbers are taken from https://www.sciencedirect.com/science/article/pii/S0960982221014962
which links to <https://github.com/SchriderLab/Drosophila_phylogeny>.  Note that
the smallest distance here is 3.6 million years, close to our upper-bound
estimate for same-species tunicate populations.

```{r published_distances_Droso}
species_dist$mya$Drosophila <- data.frame(
  'Dbu'=c(0,      46.836,  46.836,  46.836,  46.836),
  'Dsu'=c(46.836, 0,       13.1873, 13.1873, 13.1873),
  'Dya'=c(46.836, 13.1873, 0,       6.7256,  6.7256),
  'Dme'=c(46.836, 13.1873, 6.7256,  0,       3.6217),
  'Dma'=c(46.836, 13.1873, 6.7256,  3.6217,  0)
)
```

### Caenorhabditis

See <https://academic.oup.com/nar/article/46/3/1295/4793371> Figure S2 for all
distances.


```{r published_distances_Caeno}
species_dist$mya$Caenorhabditis <- data.frame(
  'Cin'=c(0,        30, 30, 30, NA),
  'Cre'=c(30, 0,        25, 25, 30),
  'Cbr'=c(30, 25,  0, 3.7, 30),
  'Cni'=c(30, 25,  3.7, 0, 30),
  'Cel'=c(NA, 30,  30, 30, 0)
)
```

```{r distances_colnames}
species_dist$mya <- lapply(species_dist$mya, function(x) {
  rownames(x) <- colnames(x)
  x
}) |> SimpleList()
```

# Normalised breakpoint accumulation

```{r breakpoints_mya}
bp_vs_dist <- function(query, target, wgo, dist_type="mya", species, extra = FALSE) {
  dist_df <- species_dist[[dist_type]][[species]]
  relevant_dist <- dist_df[rownames(dist_df)==query, colnames(dist_df)==target]
  breakpoints <- wgo[wgo$type == "breakpoint region"]
  wgo_size <- wgo |> width() |> sum()
  wgo_size_mbp <- wgo_size / 1000000
  aln_length <- wgo[wgo$type %in% c("collinear alignment", "isolated alignment")] |> width() |> sum()
  breakpoints_per_scale = length(breakpoints)/relevant_dist
  breakpoints_per_mbp = length(breakpoints)/wgo_size_mbp
  breakpoints_per_aln_mbp <- length(breakpoints) / aln_length * 1e6
  data.frame(
    species                    = species,
    query                      = query,
    target                     = target,
    pair                       = paste0(query, "_", target),
    dist_type                  = dist_type,
    dist                       = relevant_dist,
    breakpoints                = length(breakpoints),
    breakpoints_per_scale      = breakpoints_per_scale,
    breakpoints_per_mbp        = breakpoints_per_mbp,
    breakpoints_per_mbp_scaled = breakpoints_per_mbp / relevant_dist,
    breakpoints_per_aln_mbp        = breakpoints_per_aln_mbp,
    breakpoints_per_aln_mbp_scaled = breakpoints_per_aln_mbp / relevant_dist,
    extra                      = extra)
}

mya_df <- rbind(
  bp_vs_dist(query="Osa", target="Oki", wgo$Osa_Oki, dist_type="mya", "Oikopleura"),
  bp_vs_dist(query="Oki", target="Bar", wgo$Oki_Bar, dist_type="mya", "Oikopleura", extra = TRUE),
  bp_vs_dist(query="Osa", target="Bar", wgo$Osa_Bar, dist_type="mya", "Oikopleura"),
  bp_vs_dist(query="Bar", target="Aom", wgo$Bar_Aom, dist_type="mya", "Oikopleura", extra = TRUE),
  bp_vs_dist(query="Osa", target="Aom", wgo$Osa_Aom, dist_type="mya", "Oikopleura"),
  bp_vs_dist(query="Oki", target="Kum", wgo$Oki_Kum, dist_type="mya", "Oikopleura", extra = TRUE),

  bp_vs_dist(query="Ply", target="Sav", wgo$Ply_Sav, dist_type="mya", "Ciona"),
  bp_vs_dist(query="Rob", target="Sav", wgo$Rob_Sav, dist_type="mya", "Ciona", extra = TRUE),
  bp_vs_dist(query="Ply", target="Rob", wgo$Ply_Rob, dist_type="mya", "Ciona"),
  bp_vs_dist(query="Rob", target="Ros", wgo$Rob_Ros, dist_type="mya", "Ciona", extra = TRUE),
  bp_vs_dist(query="Ply", target="Ros", wgo$Ply_Ros, dist_type="mya", "Ciona"),

  bp_vs_dist(query="Dme", target="Dbu", wgo$Dme_Dbu, dist_type="mya", "Drosophila"),
  bp_vs_dist(query="Dme", target="Dsu", wgo$Dme_Dsu, dist_type="mya", "Drosophila"),
  bp_vs_dist(query="Dme", target="Dya", wgo$Dme_Dya, dist_type="mya", "Drosophila"),
  
  bp_vs_dist(query="Cni", target="Cbr", wgo$Cni_Cbr, dist_type="mya", "Caenorhabditis", extra = TRUE),
  bp_vs_dist(query="Cni", target="Cre", wgo$Cni_Cre, dist_type="mya", "Caenorhabditis", extra = TRUE),
  bp_vs_dist(query="Cni", target="Cin", wgo$Cni_Cin, dist_type="mya", "Caenorhabditis", extra = TRUE),
  bp_vs_dist(query="Cbr", target="Cni", wgo$Cbr_Cni, dist_type="mya", "Caenorhabditis"),
  bp_vs_dist(query="Cbr", target="Cre", wgo$Cbr_Cre, dist_type="mya", "Caenorhabditis"),
  bp_vs_dist(query="Cbr", target="Cel", wgo$Cbr_Cel, dist_type="mya", "Caenorhabditis")
  
)
mya_df$species <- factor(mya_df$species, levels=c("Caenorhabditis","Drosophila", "Ciona", "Oikopleura"))
mya_df$pair <- factor(mya_df$pair,
                      levels=c("Cni_Cin", "Cbr_Cel", "Cni_Cre", "Cbr_Cre", "Cni_Cbr", "Cbr_Cni",
                               "Dme_Dbu", "Dme_Dsu", "Dme_Dya",
                               "Ply_Sav", "Rob_Sav", "Ply_Rob", "Rob_Ros", "Ply_Ros",
                               "Osa_Oki", "Oki_Bar", "Osa_Bar", "Bar_Aom", "Osa_Aom", "Oki_Kum"))
rownames(mya_df) <- NULL
mya_df$comp <- factor(c("far", "far", "mid", "mid", "close", "close",
                        "far", "far", "mid", "mid", "close",
                        "far", "mid", "close",
                        "close", "mid", "far", "close", "mid", "far"),
                      levels=c("close", "mid", "far"))
(mya_df)
```

Breakpoint computation is tricky as expected number increases with time but the
sensitivity of the counting decreases with distance.  For instance we might need
to mask repeat movements, which count as breakpoints but are better detected in
closely related genomes.  Also the Drosophila centromeres may need to be masked.
But at this level of manual curation it may be better to find a more robust
metric instead.

```{r breakpoints_sum}
sub_df <- rbind(
  bp_vs_dist(query="Osa", target="Oki", wgo$Osa_Oki, dist_type="cophenetic", "Oikopleura"),
  bp_vs_dist(query="Osa", target="Bar", wgo$Osa_Bar, dist_type="cophenetic", "Oikopleura"),
  bp_vs_dist(query="Osa", target="Aom", wgo$Osa_Aom, dist_type="cophenetic", "Oikopleura"),

  bp_vs_dist(query="Ply", target="Ros", wgo$Ply_Ros, dist_type="cophenetic", "Ciona"),
  bp_vs_dist(query="Ply", target="Rob", wgo$Ply_Rob, dist_type="cophenetic", "Ciona"),
  bp_vs_dist(query="Ply", target="Sav", wgo$Ply_Sav, dist_type="cophenetic", "Ciona"),

  bp_vs_dist(query="Dme", target="Dbu", wgo$Dme_Dbu, dist_type="cophenetic", "Drosophila"),
  bp_vs_dist(query="Dme", target="Dsu", wgo$Dme_Dsu, dist_type="cophenetic", "Drosophila"),
  bp_vs_dist(query="Dme", target="Dya", wgo$Dme_Dya, dist_type="cophenetic", "Drosophila"),
  
  bp_vs_dist(query="Cni", target="Cin", wgo$Cni_Cin, dist_type="cophenetic", "Caenorhabditis"),
  bp_vs_dist(query="Cni", target="Cre", wgo$Cni_Cre, dist_type="cophenetic", "Caenorhabditis"),
  bp_vs_dist(query="Cni", target="Cbr", wgo$Cni_Cbr, dist_type="cophenetic", "Caenorhabditis")
)
sub_df$species <- factor(sub_df$species, levels=c("Caenorhabditis", "Drosophila", "Ciona", "Oikopleura"))
sub_df <- sub_df[order(sub_df$species, sub_df$breakpoints_per_mbp_scaled),]
rownames(sub_df) <- NULL
sub_df$pair <- factor(sub_df$pair, levels=sub_df$pair)
sub_df$comp <- factor(c("far", "mid", "close"), levels=c("close", "mid", "far"))
(sub_df)
```

This said, O. dioica is always ahead of other species pairs after normalising by
sequence length and evolutionary distance.

```{r breakpoints_2, dependson="breakpoints_1", fig.height=3, fig.width=10, message=FALSE, warning=FALSE, dev=c("svg", "png", "pdf"), fig.ext=c("svg", "png", "pdf")}
customPlot <- function(df, what, title, extra = "drop") {
  if (extra == "drop" & ! is.null(df$extra)) df <- df[! df$extra,]
  ggplot(df) +
    aes(x=.data[[what]], y = pair, fill = species) +
    geom_bar(stat='identity') +
    facet_wrap(~comp, scales="free") +
    theme_bw() +
    ggtitle(title)
}

customPlot(mya_df, 'breakpoints', 'Number of breakpoints')
customPlot(mya_df, 'breakpoints_per_scale', 'Number of breakpoints per million years')
customPlot(mya_df, 'breakpoints_per_mbp', 'Number of breakpoints per megabase')
customPlot(mya_df, 'breakpoints_per_aln_mbp', 'Number of breakpoints per megabase aligned')
customPlot(mya_df, 'breakpoints_per_mbp_scaled', 'Number of breakpoints per megabase per million years diverged')
customPlot(mya_df, 'breakpoints_per_mbp_scaled', 'Number of breakpoints per megabase per million years diverged', extra = TRUE)
customPlot(mya_df, 'breakpoints_per_aln_mbp_scaled', 'Number of breakpoints per megabase aligned per million years diverged')
customPlot(mya_df, 'breakpoints_per_aln_mbp_scaled', 'Number of breakpoints per megabase aligned per million years diverged', extra = TRUE)
customPlot(sub_df, 'breakpoints_per_mbp_scaled', 'Number of breakpoints per megabase per substitution distance')
customPlot(sub_df, 'breakpoints_per_aln_mbp_scaled', 'Number of breakpoints per megabase aligned per substitution distance')
```

# Session information

```{r sessionInfo}
sessionInfo()
```
