---
title: "Segment movement distances"
author: 
 - "Michael Mansfield"
date: "`r format(Sys.time(), '%d %B, %Y')`"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteEncoding{UTF-8}
  %\VignetteIndexEntry{Regions interrupting colinearity}
  %\VignetteEngine{knitr::rmarkdown}
editor_options: 
  chunk_output_type: console
---

```{r setup, echo=FALSE}
knitr::opts_knit$set(verbose = TRUE)
```

# Load packages and data

See the `vignette("OikScrambling", package = "OikScrambling")` for general
details on package and data load.

See `vignette("LoadGenomicBreaks", package = "OikScrambling")` for how the
different `GBreaks` objects are prepared.

```{r load_packages_and_data}
suppressPackageStartupMessages({
  library('GenomicBreaks')
  library('ggplot2')
  library("BreakpointsData")
})
load("BreakPoints.Rdata")
```

# Segment movement distances
Whenever a collinear segment moves between species, if the homologous segment is located on an equivalent chromosome, a distance can be calculated.

```{r calculate_movement_distances}
# Find midpoints of a GRanges.
grMidpoint <- function(gr) {
  start(resize(gr, 1, fix = "center"))
}

# Find distance between two equivalent GRanges.
# Input: two GRanges (probably the target and query of a GBreaks object)
# Output: Absolute distance between the two GRanges based on some metric ("start", "end", or "midpoint")
GRangesDist <- function(gr1, gr2, distMetric="midpoint") {
  if(distMetric == "start"){
    segDistance <- abs(gr2$start - gr1$start)
  } else if(distMetric=="end") {
    segDistance <- abs(gr2$end - gr1$end)
  } else if(distMetric=="midpoint") {
    # Add midpoints
    gr1$midpoint <- grMidpoint(gr1)
    gr2$midpoint <- grMidpoint(gr2)
    segDistance <- abs(gr2$midpoint - gr1$midpoint)
  }
  segDistance
}

chromDist <- function(gb, chromosomePairing=NULL, distMetric="midpoint"){
  chrom_sp1 <- chromosomePairing[["sp1"]]
  chrom_sp2 <- chromosomePairing[["sp2"]]

  # Subset to include only chromosomes of interest
  chrom_gb <- gb[seqnames(gb) %in% chrom_sp1]
  chrom_gb <- chrom_gb[seqnames(chrom_gb$query) %in% chrom_sp2]
  
  # Subset to include only ranges in which all chrom in sp1 are paired to their equivalent in sp2
  # Note that a similar function could be used to compare the identity of chromosome arms
  pairedChromosomes <- apply(as.data.frame(chrom_gb), 1, function(GRange) {
    t_seqname_index = which(chromosomePairing[["sp1"]] == GRange$seqnames)
    q_seqname_index = which(chromosomePairing[["sp2"]] == GRange$query.seqnames)
    if(t_seqname_index == q_seqname_index) {
      paired = TRUE
    } else {
      paired = FALSE
    }
    paired
  })
  chrom_gb <- chrom_gb[pairedChromosomes]
  
  # Now that the GRanges are only on equivalent chromosomes, calculate the distances between them.
  # This can be "start", "end", or "midpoint" distance.
  chrom_gb$distance <- GRangesDist(chrom_gb, chrom_gb$query, distMetric=distMetric)
  chrom_gb
}

chromosomePairing <- list("sp1"=c("chr1", "chr2", "PAR", "XSR", "YSR"), "sp2"=c("Chr1", "Chr2", "PAR", "XSR", "YSR"))
segDists <- SimpleList()
segDists$Oki_Osa <- chromDist(gbs$Oki_Osa, chromosomePairing = chromosomePairing)
segDists$Oki_Bar <- chromDist(gbs$Oki_Osa, chromosomePairing = chromosomePairing)
chromosomePairing <- list("sp1"=c("Chr1", "Chr2", "PAR", "XSR", "YSR"), "sp2"=c("Chr1", "Chr2", "PAR", "XSR", "YSR"))
segDists$Osa_Bar <- chromDist(gbs$Osa_Bar, chromosomePairing = chromosomePairing)
```

## Plotting segment movement distances

```{r plot_movement_distances_oki_osa, fig.width=15, fig.height=8, dev="svg", fig.ext="svg"}
plotSegDists <- function(segDist, facet.by='seqnames~Arm', binWidth=100000) {
  ggplot(as.data.frame(segDist)) +
    aes(x=distance) +
    geom_histogram(binwidth = binWidth) +
    facet_wrap(facet.by, scales="free_x", ncol=2)
}
plotSegDists(segDists$Oki_Osa, facet.by='seqnames~Arm', binWidth=100000) + ggtitle("Oki-Osa: segment movement distances (uncoalesced)")
```

```{r plot_movement_distances_oki_bar, fig.width=15, fig.height=8, dev="svg", fig.ext="svg"}
plotSegDists(segDists$Oki_Bar, facet.by='seqnames~Arm', binWidth=100000) + ggtitle("Oki-Bar: segment movement distances (uncoalesced)")
```

```{r plot_movement_distances_osa_bar, fig.width=15, fig.height=8, dev="svg", fig.ext="svg"}
plotSegDists(segDists$Osa_Bar, facet.by='seqnames~Arm', binWidth=100000) + ggtitle("Osa-Bar: segment movement distances (uncoalesced)")
```

## Segment movement distances (coalesced)
```{r calculate_movement_distances_coa}
segDistsCoa <- SimpleList()
chromosomePairing <- list("sp1"=c("chr1", "chr2", "PAR", "XSR", "YSR"), "sp2"=c("Chr1", "Chr2", "PAR", "XSR", "YSR"))
segDistsCoa$Oki_Osa <- chromDist(coa$Oki_Osa, chromosomePairing = chromosomePairing)
segDistsCoa$Oki_Bar <- chromDist(coa$Oki_Osa, chromosomePairing = chromosomePairing)
chromosomePairing <- list("sp1"=c("Chr1", "Chr2", "PAR", "XSR", "YSR"), "sp2"=c("Chr1", "Chr2", "PAR", "XSR", "YSR"))
segDistsCoa$Osa_Bar <- chromDist(coa$Osa_Bar, chromosomePairing = chromosomePairing)
```

```{r plot_movement_distances_oki_osa_coa, fig.width=15, fig.height=8, dev="svg", fig.ext="svg"}
plotSegDists(segDistsCoa$Oki_Osa, facet.by='seqnames~Arm', binWidth=100000) + ggtitle("Oki-Osa: segment movement distances (coalesced)")
```

```{r plot_movement_distances_oki_bar_coa, fig.width=15, fig.height=8, dev="svg", fig.ext="svg"}
plotSegDists(segDistsCoa$Oki_Bar, facet.by='seqnames~Arm', binWidth=100000) + ggtitle("Oki-Bar: segment movement distances (coalesced)")
```

```{r plot_movement_distances_osa_bar_coa, fig.width=15, fig.height=8, dev="svg", fig.ext="svg"}
plotSegDists(segDistsCoa$Osa_Bar, facet.by='seqnames~Arm', binWidth=100000) + ggtitle("Osa-Bar: segment movement distances (coalesced)")
```

# Session information

```{r session_information}
sessionInfo()
```
