---
title: "Tandem site duplications"
author: 
 - "Charles Plessy"
date: "`r format(Sys.time(), '%d %B, %Y')`"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteEncoding{UTF-8}
  %\VignetteIndexEntry{Tandem site duplications}
  %\VignetteEngine{knitr::rmarkdown}
editor_options: 
  chunk_output_type: console
---

# Load pacakges and data

See the `vignette("OikScrambling", package = "OikScrambling")` for general
details on package and data load.

See `vignette("LoadGenomicBreaks", package = "OikScrambling")` for how the
different `GBreaks` objects are prepared.

```{r load_pacakges_and_data}
suppressPackageStartupMessages({
  library('GenomicBreaks')
  library('ggplot2')
  library("BreakpointsData")
})
genomes <- OikScrambling:::loadAllGenomes(compat = F)
load("BreakPoints.Rdata")
reps <- OikScrambling:::loadAllRepeats(compat = F)
transcripts <- OikScrambling:::loadAllTranscriptsGR(compat = F)
```

# Purpose

Here we explore regions of interest that feature an insertion-deletion site
flanked by a _tandem site duplication_.

In our _genomic breaks_ objects, representing _one-to-one_ alignments, one copy
of the tandem-duplicated site is part of the alignment, and the other is not.

```
                   TSD
query       …──────────━━━━━─────────━━━━━─────────────…
alignment    ...............              .............
target      …──────────━━━━━              ─────────────…

```

# ROI1: composite transposon and tandem site duplication.

## Details on the region

_ROI1_ identified and analysed by Martin Frith.  Here I am just transposing the
finding in Bioconductor's context.

In `contig_27_1` (which is most of Kume genome's chr1's short arm), there is an
indel region where the aligned regions in the Okinawa genome are directly in
contact, but on the Kume genome the aligned regions are separated by a 5146-bp
gap.

(The regions align on opposite strands, so the plot looks strange if you are
not familiar with it.)

```{r ROI1}
(ROI1 <- gbs$Oki_Kum |> subsetByOverlaps(GRanges("chr1:3720291-3734643")))
plotApairOfChrs(ROI1)
cleanGaps(ROI1$query) |> as.data.frame()
```

Interestingly, the unaligned region in Kume contains two repeat elements in
inverted tandem.  You can see them in the alignment below, of the left-side
boundary of the unaligned region to the reverse-complement of its right side.

```{r ROI_breaks_pair_align}
pairwiseAlignment(getSeq(cleanGaps(ROI1$query) |> flank(20, both = T, start = T)),
                  getSeq(cleanGaps(ROI1$query) |> flank(20, both = T, start = F)) |> reverseComplement(),
                  type="global") |> writePairwiseAlignments()
```

<details>
<summary>Click here to see the full pairwise alignment between the Kume insert
and its own reverse-complement.</summary>
```{r ROI_pair_align}
pairwiseAlignment(getSeq(cleanGaps(ROI1$query)),
                  getSeq(cleanGaps(ROI1$query)) |> reverseComplement(),
                  type="global") |> writePairwiseAlignments()
```
</details>

Immediately upstream these repeats, there a tandem octamer `AAGATACG`.  In
the Okinawan genome, it is present as a single copy.  You can see it in the
alignment below, where the right-side boundary was not reverse-complemented.

```{r ROI_breaks_pair_align2}
pairwiseAlignment(getSeq(cleanGaps(ROI1$query) |> flank(20, both = T, start = T)),
                  getSeq(cleanGaps(ROI1$query) |> flank(20, both = T, start = F)),
                  type="global") |> writePairwiseAlignments()
```

The same can be seen in the multiple alignment below comparising the left and
right arms of the Kume alignments and their match to the Okinawa genome.

```{r ROI_breaks_pair_align3}
# Note that genome alignment is +/- therefore the order below is counter-intuitive.
list(
  kum_right = getSeq(ROI1$query[1] |> flank(20, both = T, start = T))[[1]],
  kum_left  = getSeq(ROI1$query[2] |> flank(20, both = T, start = F))[[1]],
  oki       = getSeq(ROI1[2]       |> flank(20, both = T, start = F))[[1]]
) |> as("DNAStringSet") |> msa::msaClustalW()
```

The microhomology does not appear to be more frequent in the Okinawan genome
compared to the others.

```{r search_for_AAGATACG}
sapply(genomes, vmatchPattern, pat = "AAGATACG") |> sapply(length)
```
 
## How often do we see tandem site duplication in similar indel configurations ?

The `unalMap` objects contain unaligned regions flanked by colinear regions.
Let's search for other situations where one of the regions has a width of zero.

Note: _in order to represent zero-width intervals in this object, which would
be impossible with `GRanges` objects, the unaligned regions were arbitrarly
added one base of each flanking aligned regions_.

These regions are rare in between-species alignments (at most a few percents),
and common in within-population alignments (~10 to 20 percents).

Note2: _the numbers would double if I would also inspect the query genome widths_.

```{r search_for_indels_with_TSDs}
unalMap |> sapply(\(gb) length(gb[width(gb) <= 2]))  # See note above
round(unalMap |> sapply(\(gb) length(gb[width(gb) <= 2])) / sapply(unalMap, length) * 100 )
```

Let's inspect within-population alignments first.  Here is an example with
a `CATCTTTG` TSD.

```{r search_for_indels_with_TSDs2}
gb <- unalMap$Oki_Kum |> plyranges::filter(width == 2)
gb$query |> width() |> summary()

alignFlank <- function(gb)
  pairwiseAlignment(getSeq(gb$query |> flank(20, both = T, start = T)),
                    getSeq(gb$query |> flank(20, both = T, start = F)),
                    type="global") |> writePairwiseAlignments()

alignFlank(gb[4])

alignFlank2 <- function(gb, n = 8) {
  # Remove 1-bp padding
  gb$query <- gb$query -1
  pairwiseAlignment(getSeq(gb$query |> resize(n, fix = "start")),
                    getSeq(gb$query |> flank(n, start = F)),
                    type="global")
}
alignFlank2(gb[4])

msAlignFlank <- function(gb)
list(
  kum_left  = getSeq(gb$query |> flank(20, both = T, start = T))[[1]],
  oki_left  = getSeq(gb       |> flank(20, both = T, start = F))[[1]],
  kum_right = getSeq(gb$query |> flank(20, both = T, start = F))[[1]]
) |> as("DNAStringSet") |> msa::msaClustalW(order = "input")

msAlignFlank(gb[4])
```

Let's count the number of perfect 8-nt matches.  See the variety in their
sequences: none of them is found more than twice!

```{r search_for_indels_with_TSDs3}
alignFlank2(gb) |> nedit() |> table()

gb$nedit <- alignFlank2(gb) |> nedit() 
gb$consensus <- alignFlank2(gb) |> compareStrings()

gb[gb$nedit==0]$consensus |> table() |> sort() |> tail()
gb[gb$nedit==0]$consensus |> table() |> length()

```

## Remaining questions

Questions:

 - How frequent is the inverted pair of repeats that we see?
 - Is there any hint of functionality for the sequence in the gap?
 - are the indels fixed in their respective populations?
 - Am I missing half of the microhomologies because of alignments to negative strands ?


# Session information
  
```{r sessioninfo}
sessionInfo()
```
