<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Changelog • OikScrambling</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Changelog"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-news">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">OikScrambling</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">3.0.5</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../articles/OikScrambling.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/CNEs.html">Conserved non-coding elements</a>
    </li>
    <li>
      <a href="../articles/ChromosomePlots.html">Chromosome-scale feature density plots</a>
    </li>
    <li>
      <a href="../articles/CoalescingCutoffs.html">Modifying coalescent cutoffs</a>
    </li>
    <li>
      <a href="../articles/ColinearityInterruptors.html">Regions interrupting colinearity</a>
    </li>
    <li>
      <a href="../articles/DistanceMeasures.html">Distance Measures</a>
    </li>
    <li>
      <a href="../articles/GenomicFeatures.html">Genomic features of the breakpoints</a>
    </li>
    <li>
      <a href="../articles/Inversions.html">Inversions</a>
    </li>
    <li>
      <a href="../articles/LoadGenomicBreaks.html">Load genomic break data</a>
    </li>
    <li>
      <a href="../articles/OrthogroupSyntenies.html">Orthogroup syntenies</a>
    </li>
    <li>
      <a href="../articles/OxfordPlots.html">Whole-genome dot-plots</a>
    </li>
    <li>
      <a href="../articles/PAC3.html">Region of interest: PAC3</a>
    </li>
    <li>
      <a href="../articles/PWM.html">Motif search and analysis</a>
    </li>
    <li>
      <a href="../articles/ParallelPlots.html">Whole-chromosome parallel plots</a>
    </li>
    <li>
      <a href="../articles/RegionWidths.html">Width of the aligned and mapped regions.</a>
    </li>
    <li>
      <a href="../articles/RegionsOfInterest.html">Regions of interest</a>
    </li>
    <li>
      <a href="../articles/RepeatRegions.html">Repeat regions</a>
    </li>
    <li>
      <a href="../articles/SegmentMovements.html">Segment movement distances</a>
    </li>
    <li>
      <a href="../articles/TFBS.html">Transcription factor binding sites</a>
    </li>
    <li>
      <a href="../articles/TandemSiteDuplications.html">Tandem site duplications</a>
    </li>
    <li>
      <a href="../articles/Translocations.html">Translocations</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/oist/LuscombeU_OikScrambling/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
      <h1 data-toc-skip>Changelog <small></small></h1>
      <small>Source: <a href="https://github.com/oist/LuscombeU_OikScrambling/blob/HEAD/NEWS.md" class="external-link"><code>NEWS.md</code></a></small>
    </div>

    <div class="section level2">
<h2 class="page-header" data-toc-text="3.0" id="oikscrambling-30x">OikScrambling 3.0.x<a class="anchor" aria-label="anchor" href="#oikscrambling-30x"></a></h2>
<ul><li>New chromosome plots for Figure 4.</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="3.0.5" id="oikscrambling-305">OikScrambling 3.0.5<a class="anchor" aria-label="anchor" href="#oikscrambling-305"></a></h2>
<ul><li>Explore strand proportion indices.</li>
<li>New region width plots for Figure 2.</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="3.0.4" id="oikscrambling-304">OikScrambling 3.0.4<a class="anchor" aria-label="anchor" href="#oikscrambling-304"></a></h2>
<ul><li>Draft chromosome plots for Figure 4 (ex-3, ex-2b).</li>
<li>Natural scale width plots for Figure 2</li>
<li>5-panel feature plots for Figure 2</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="3.0.3" id="oikscrambling-303">OikScrambling 3.0.3<a class="anchor" aria-label="anchor" href="#oikscrambling-303"></a></h2>
<ul><li>Mid-centered feature plots for Figure 2.</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="3.0.2" id="oikscrambling-302">OikScrambling 3.0.2<a class="anchor" aria-label="anchor" href="#oikscrambling-302"></a></h2>
<ul><li>Plot genomic features on white background for Figure 2.</li>
<li>Compute strand randomisation indexes.</li>
<li>Provide alignment information for manual brush-up of Hox panels in Figure 1.</li>
<li>Add Osaka–Aomori Oxford plots.</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="3.0.1" id="oikscrambling-301">OikScrambling 3.0.1<a class="anchor" aria-label="anchor" href="#oikscrambling-301"></a></h2>
<ul><li>Plot operons density near alignment boundaries.</li>
<li>Compute confidence intervals on width distributions.</li>
<li>Plot width distributions in PDF format.</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="3.0.0" id="oikscrambling-300">OikScrambling 3.0.0<a class="anchor" aria-label="anchor" href="#oikscrambling-300"></a></h2>
<ul><li>Corrected arm names on query genomes.</li>
<li>New functions <code><a href="../reference/isSyntenic.html">isSyntenic()</a></code>, <code><a href="../reference/isSynbrachial.html">isSynbrachial()</a></code>, <code><a href="../reference/nameSyntenic.html">nameSyntenic()</a></code> and <code><a href="../reference/nameSynbrachial.html">nameSynbrachial()</a></code>.</li>
<li>Renamed <em>mapped unaligned</em> regions to <em>bridge</em> regions and renamed <code>unalMap</code> object to <code>bri</code>.</li>
<li>Added an ‘isolated’ category in the <code>RegionWidths</code> vignette.</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="2.2.2" id="oikscrambling-222">OikScrambling 2.2.2<a class="anchor" aria-label="anchor" href="#oikscrambling-222"></a></h2>
<ul><li>Parallel plot of chr2</li>
<li>Export aggregated numbers about segment widths.</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="2.2.1" id="oikscrambling-221">OikScrambling 2.2.1<a class="anchor" aria-label="anchor" href="#oikscrambling-221"></a></h2>
<ul><li>New Oki-Osa Oxford plots for Figure 1.</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="2.2.0" id="oikscrambling-220">OikScrambling 2.2.0<a class="anchor" aria-label="anchor" href="#oikscrambling-220"></a></h2>
<ul><li>Strand-colored version of the 10-Mb window Oxford plots. Needs <code>GenomicBreaks</code> version <code>0.13.1</code> or superior.</li>
<li>Human-mouse 10-Mb window Oxford plot. Needs <code>BreakpointsData</code> version <code>3.9.0</code> or superior.</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="2.1.0" id="oikscrambling-210">OikScrambling 2.1.0<a class="anchor" aria-label="anchor" href="#oikscrambling-210"></a></h2>
<ul><li>Add a vignette about the movement of the PAC3 gene.</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="2.0.0" id="oikscrambling-200">OikScrambling 2.0.0<a class="anchor" aria-label="anchor" href="#oikscrambling-200"></a></h2>
<ul><li><p>Adopt semantic versionning: any backwards-incompatible change increases the major version number, no matter how (un)important it is.</p></li>
<li><p>Use the new <code><a href="https://oist.github.io/GenomicBreaks/reference/bridgeRegions.html" class="external-link">GenomicBreaks::bridgeRegions()</a></code> function that produces zero-width ranges instead of the local function that artificially added 1nt to the ends.</p></li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.0.0" id="oikscrambling-100">OikScrambling 1.0.0<a class="anchor" aria-label="anchor" href="#oikscrambling-100"></a></h2>
<ul><li>Major change in the BSgenome packages, needing the removal of the old ones and the recomputation of the data. The new packages allow to much easier link the GBreaks objects to their genome sequence or annotations.</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="0.8.3" id="oikscrambling-083">OikScrambling 0.8.3<a class="anchor" aria-label="anchor" href="#oikscrambling-083"></a></h2>
<ul><li>Move some regions of interest in a separate vignette.</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="0.8.2" id="oikscrambling-082">OikScrambling 0.8.2<a class="anchor" aria-label="anchor" href="#oikscrambling-082"></a></h2>
<ul><li>Added a new <code>SegmentMovements</code> vignette.</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="0.8.1" id="oikscrambling-081">OikScrambling 0.8.1<a class="anchor" aria-label="anchor" href="#oikscrambling-081"></a></h2>
<ul><li>Added a new <code>wgo</code> whole-genome object indicating if a base is part of a syntenic block, or just aligned, or unaligned.</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="0.8.0" id="oikscrambling-080">OikScrambling 0.8.0<a class="anchor" aria-label="anchor" href="#oikscrambling-080"></a></h2>
<ul><li>Added <code>nonCoa</code> flags to the <code>gbs</code>, <code>unal</code> and <code>coa</code> objects.</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="0.7.1" id="oikscrambling-071">OikScrambling 0.7.1<a class="anchor" aria-label="anchor" href="#oikscrambling-071"></a></h2>
<ul><li>Created compound plots of width distribution for Figure 2.</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="0.7.0" id="oikscrambling-070">OikScrambling 0.7.0<a class="anchor" aria-label="anchor" href="#oikscrambling-070"></a></h2>
<ul><li>Overlap with repeats on both strands. The difference is not so massive, considering that the unaligned and unmapped regions are already strandless anyway.</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="0.6.6" id="oikscrambling-066">OikScrambling 0.6.6<a class="anchor" aria-label="anchor" href="#oikscrambling-066"></a></h2>
<ul><li>New functions <code>OikScrambling:::compDistClass()</code> and <code>OikScrambling:::compDistClass()</code>, and new option <code>short</code> for <code>OikScrambling:::compDistance()</code>, to produce more compact summary graphs.</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="0.6.5" id="oikscrambling-065">OikScrambling 0.6.5<a class="anchor" aria-label="anchor" href="#oikscrambling-065"></a></h2>
<ul><li>Identify the flipped repeat as MITE and search for other copies.</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="0.6.4" id="oikscrambling-064">OikScrambling 0.6.4<a class="anchor" aria-label="anchor" href="#oikscrambling-064"></a></h2>
<ul><li>Add regions of interest with repeat flipping and insertion.</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="0.6.3" id="oikscrambling-063">OikScrambling 0.6.3<a class="anchor" aria-label="anchor" href="#oikscrambling-063"></a></h2>
<ul><li>Add Hox regions of interest for Figure 1</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="0.6.2" id="oikscrambling-062">OikScrambling 0.6.2<a class="anchor" aria-label="anchor" href="#oikscrambling-062"></a></h2>
<ul><li>Add 10-Mbp dot plots for Figure 1</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="0.6.1" id="oikscrambling-061">OikScrambling 0.6.1<a class="anchor" aria-label="anchor" href="#oikscrambling-061"></a></h2>
<ul><li>Add Mike’s vignette on coalescing cutoffs.</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="0.6.0" id="oikscrambling-060">OikScrambling 0.6.0<a class="anchor" aria-label="anchor" href="#oikscrambling-060"></a></h2>
<ul><li>Adding <em>Drosophila</em> data for a plot on CP’s MBSJ2021 poster.</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="0.5.0" id="oikscrambling-050">OikScrambling 0.5.0<a class="anchor" aria-label="anchor" href="#oikscrambling-050"></a></h2>
<ul><li>
<code><a href="../reference/load_one_to_ones.html">load_one_to_ones()</a></code> function moved from <em>GenomicBreaks</em> to this package.</li>
<li>Internal functions updated to load <em>C. intestinalis</em> annotations.</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="0.4.0" id="oikscrambling-040">OikScrambling 0.4.0<a class="anchor" aria-label="anchor" href="#oikscrambling-040"></a></h2>
<ul><li>From now on the <code>main</code> branch must remain buildable.</li>
<li>Added a <code>NEWS.md</code> file to track changes to the package.</li>
</ul></div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>

</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Aleksandra Bliznina, Michael Mansfield, Charles Plessy.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.2.</p>
</div>

      </footer></div>

  


  

  </body></html>

