---
title: "Aggregate Results"
author: "Charles Plessy"
date: "21/06/2021"
output: html_document
params:
  resultsDir: "~/projects/GenomicBreaks_results/results-v2/genomicbreaks/"
  alnDir:     "~/projects/GenomicBreaks_results/results-v2/last/"
  prefix: "test"
editor_options: 
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

How to use this Rmarkdown template:

It is essential to set `R_LIBS_USER` to an empty value, otherwise it will
attempt to load packages from the user's library, which were compiled in a
different environment than the GenomicBreakds Singularity image.

Example command

```
R_LIBS_USER='' R -e 'rmarkdown::render("inst/rmarkdown/templates/aggregateResults/skeleton/skeleton.Rmd", output_file = "./toto.html", params = list(resultsDir = "/path/do/dir/with/yaml/files"))'
```

```{r load_data}
library("yaml")
yamlFiles <- list.files(params$resultsDir, pattern = "*.yaml", full.names = TRUE)
names(yamlFiles) <- yamlFiles |> basename() |> sub(pat = ".yaml", rep="")

getStats <- function(file) {
  y <- yaml::read_yaml(file) |> yaml::yaml.load()
  res <- list()
  res$unaligned_mean_width <- y$unaligned_width_summary$Mean
  res$aligned_mean_width   <- y$width_summary_aligned$Mean
  res$collapsed_mean_width <- y$width_summary_collapsed$Mean
  res$similarity           <- y$percent_similarity * 100
  res$aligned_regions      <- y$number_of_ranges_part
  res$collapsed_regions    <- y$number_of_ranges_col
  res$collapsed_regions2   <- y$number_of_ranges_match
  res$aligned_total        <- y$aligned_total
  res$unaligned_total      <- y$unal_total
  res$synteny_index        <- y$synteny_index
  res$correlation_index    <- y$correlation_index
  res$N50_part             <- y$N50_part
  res$N50_match            <- y$N50_match
  unlist(res)
}

# getStats_long <- function(file) {
#   pairName <- file |> basename() |> sub(pat = ".yaml", rep="")
#   y <- yaml::read_yaml(file) |> yaml::yaml.load()
#   rbind(
#     list(name = pairName, stat = "unaligned_mean_width", value = y$unaligned_width_summary$Mean),
#     list(name = pairName, stat = "aligned_mean_width", value = y$aligned_width_summary$Mean),
#     list(name = pairName, stat = "similarity", value = 100 - y$alignment_summary$percent_mismatch)
#   )
# }

df <- do.call(rbind, lapply(yamlFiles, getStats)) |> as.data.frame()
df$similarity[is.na(df$similarity)] <- 100
df$species1 <- strsplit(rownames(df), "__") |> lapply(\(.) .[1]) |> unlist()
df$species2 <- strsplit(rownames(df), "__") |> lapply(\(.) .[2]) |> unlist()
df$speciesGroup <- "Drosophila"
df$speciesGroup[df$species1 %in% c("Bar2_p4", "OSKA2016v1.9", "OKI2018_I69_1.0")] <- "Oikopleura"
df$speciesGroup[df$species1 %in% c("C_rob_HT", "C_int_P")] <- "Ciona"
```

```{r plot_data}
library("ggplot2")
p <- ggplot(df) +
  aes( x = similarity, y = aligned_regions) +
  aes(color = speciesGroup) +
  xlab("Similarity between aligned regions") +
  ylab("Number of aligned regions") +
  geom_point()
p
# p + geom_text(aes(label=species2))

p <- ggplot(df) +
  aes( x = similarity, y = collapsed_regions) +
  aes(color = speciesGroup) +
  xlab("Similarity between aligned regions") +
  ylab("Number of collapsed regions") +
  geom_point()
p

ggplot(df) +
  aes( x = similarity, y = aligned_mean_width) +
  aes(color = speciesGroup) +
  xlab("Similarity between aligned regions") +
  ylab("Mean width of aligned regions") +
  scale_y_log10() +
  geom_point()

ggplot(df) +
  aes( x = similarity, y = N50_match) +
  aes(color = speciesGroup) +
  xlab("Similarity between aligned regions") +
  ylab("N50 of width of aligned regions") +
  scale_y_log10() +
  geom_point()

ggplot(df) +
  aes( x = similarity, y = synteny_index) +
  aes(color = speciesGroup) +
  xlab("Similarity between aligned regions") +
  ylab("Synteny index") +
  scale_y_log10() +
  geom_point()

ggplot(df) +
  aes( x = similarity, y = correlation_index) +
  aes(color = species1) +
  xlab("Similarity between aligned regions") +
  ylab("Correlation index") +
  scale_y_log10() +
  geom_point()
```

```{r plot_similarity}
sim <- df[c("similarity", "species1", "species2")]
sim <- sim[order(sim$similarity, decreasing = FALSE), ]
ggplot(sim) +
  aes(similarity, species2) +
  geom_point() +
  facet_grid(species1 ~ ., scales = "free", space = "free")

# subsetAndLevel <- function (x, species) {
#   x <- subset(x, species1 == species)
#   x$species2 <- factor(x$species2, levels = x$species2)
#   x
# }

plotSortedSubset <- function (x, species) {
  x <- subset(x, species1 == species)
  x$species2 <- factor(x$species2, levels = x$species2)
  x
  ggplot(x) +
    aes(similarity, species2) +
    geom_point() +
    facet_wrap(~species1) +
    scale_y_discrete(position="right") +
    ylab(NULL)
}

p_Bar <- plotSortedSubset(sim, "Bar2_p4")
p_Oki <- plotSortedSubset(sim, "OKI2018_I69_1.0")
p_Osa <- plotSortedSubset(sim, "OSKA2016v1.9")
p_Cio <- plotSortedSubset(sim, "C_rob_HT")


requireNamespace("ggpubr")
ggpubr::ggarrange(p_Oki, p_Bar, p_Cio, p_Osa)

```


```{r plot_tot_align}
sim <- df[c("aligned_total", "unaligned_total", "species1", "species2")]
sim$aligned_prop <- sim$aligned_total / (sim$aligned_total + sim$unaligned_total)
sim
sim <- sim[order(sim$aligned_prop, decreasing = FALSE), ]
ggplot(sim) +
  aes(aligned_prop, species2) +
  geom_point() +
  facet_grid(species1 ~ ., scales = "free", space = "free")
```

A closer look at the Ciona alignments
-------------------------------------

```{r}
C_int_P__C_int_R  <- load_genomic_breaks(paste0(params$alnDir, "C_int_P__C_int_R.gff.gz"))
C_int_P__C_rob_HT <- load_genomic_breaks(paste0(params$alnDir, "C_int_P__C_rob_HT.gff.gz"))
C_int_P__C_sav    <- load_genomic_breaks(paste0(params$alnDir, "C_int_P__C_sav.gff.gz"))
C_int_P__St_cla   <- load_genomic_breaks(paste0(params$alnDir, "C_int_P__St_cla.gff.gz"))

C_int_P__list <- list(C_int_P__C_int_R  = C_int_P__C_int_R,
                      C_int_P__C_rob_HT = C_int_P__C_rob_HT,
                      C_int_P__C_sav    = C_int_P__C_sav,
                      C_int_P__St_cla   = C_int_P__St_cla)

sapply(C_int_P__list, length)
sapply(C_int_P__list, synteny_index)
sapply(C_int_P__list, correlation_index)
```

Our question is: why are there so many alignments in int/int and int/rob
comparisons despite these species are highly syntenic ?  What does interrupt the
alignments ?

```{r}
flagColinearAlignments(C_int_P__C_int_R, details = TRUE) |> head(11)
C_int_P__C_int_R_col <- coalesce_contigs(C_int_P__C_int_R)
flagColinearAlignments(C_int_P__C_int_R_col, details = TRUE) |> head(11)
```

Colinearity seem to be interrupted by short-width alignments to non-homologous
contigs.  This needs to be checked again once the alignment pipeline has been
updated (update LAST version, posmask later, replace maf-swap and pass -fMAF+
to last-split).  A different seeding scheme can also be tested.
