<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Modifying coalescent cutoffs • OikScrambling</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Modifying coalescent cutoffs">
<meta property="og:description" content="OikScrambling">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">OikScrambling</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">5.0.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/OikScrambling.html">Get started</a>
</li>
<li>
  <a href="../articles/LoadGenomicBreaks.html">Data load</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Supplemental material
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/OxfordPlots.html">Figure 1A DNA line plots</a>
    </li>
    <li>
      <a href="../articles/ParallelPlots.html">Figure 1B chromosome parallel plots</a>
    </li>
    <li>
      <a href="../articles/DistanceMeasures.html">Table 1 scrambling index</a>
    </li>
    <li>
      <a href="../articles/ChromosomePlots.html">Figure 5 genome-wide feature patterns</a>
    </li>
    <li>
      <a href="../articles/SquareWindowPlots.html">Figure 7 DNA line plots</a>
    </li>
    <li>
      <a href="../articles/ScalingBreakpoints.html">Figure 7 breakpoint number barplots</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/CNEs.html">Conserved non-coding elements</a>
    </li>
    <li>
      <a href="../articles/ChromosomePlots.html">Chromosome-scale feature density plots</a>
    </li>
    <li>
      <a href="../articles/CoalescingCutoffs.html">Modifying coalescent cutoffs</a>
    </li>
    <li>
      <a href="../articles/ColinearityInterruptors.html">Regions interrupting colinearity</a>
    </li>
    <li>
      <a href="../articles/DistanceMeasures.html">Distance Measures</a>
    </li>
    <li>
      <a href="../articles/GenomicFeatures.html">Genomic features of the breakpoints</a>
    </li>
    <li>
      <a href="../articles/GenomicHeatmaps.html">Genomic features of the breakpoints (heatmaps)</a>
    </li>
    <li>
      <a href="../articles/Inversions.html">Inversions</a>
    </li>
    <li>
      <a href="../articles/LoadGenomicBreaks.html">Load genomic break data</a>
    </li>
    <li>
      <a href="../articles/OperonConservation.html">Operon conservation analysis</a>
    </li>
    <li>
      <a href="../articles/OrthogroupSyntenies.html">Orthogroup syntenies</a>
    </li>
    <li>
      <a href="../articles/OxfordPlots.html">Alignment line plots, whole genomes</a>
    </li>
    <li>
      <a href="../articles/PAC3.html">Region of interest: PAC3</a>
    </li>
    <li>
      <a href="../articles/PWM.html">Motif search and analysis</a>
    </li>
    <li>
      <a href="../articles/ParallelPlots.html">Whole-chromosome parallel plots</a>
    </li>
    <li>
      <a href="../articles/RegionWidths.html">Width of the aligned and mapped regions.</a>
    </li>
    <li>
      <a href="../articles/RegionsOfInterest.html">Regions of interest</a>
    </li>
    <li>
      <a href="../articles/RepeatRegions.html">Repeat regions</a>
    </li>
    <li>
      <a href="../articles/ScalingBreakpoints.html">Scaling breakpoints</a>
    </li>
    <li>
      <a href="../articles/SegmentMovements.html">Segment movement distances</a>
    </li>
    <li>
      <a href="../articles/SquareWindowPlots.html">Alignment line plots, 10×10 Mbp windows</a>
    </li>
    <li>
      <a href="../articles/TFBS.html">Transcription factor binding sites</a>
    </li>
    <li>
      <a href="../articles/TandemSiteDuplications.html">Tandem site duplications</a>
    </li>
    <li>
      <a href="../articles/Translocations.html">Translocations</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/oist/LuscombeU_OikScrambling/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Modifying coalescent cutoffs</h1>
                        <h4 data-toc-skip class="author">Michael
Mansfield</h4>
            
            <h4 data-toc-skip class="date">13 October, 2023</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/oist/LuscombeU_OikScrambling/blob/HEAD/vignettes/CoalescingCutoffs.Rmd" class="external-link"><code>vignettes/CoalescingCutoffs.Rmd</code></a></small>
      <div class="hidden name"><code>CoalescingCutoffs.Rmd</code></div>

    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">knitr</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/knitr/man/opts_chunk.html" class="external-link">opts_chunk</a></span><span class="op">$</span><span class="fu">set</span><span class="op">(</span>cache <span class="op">=</span> <span class="cn">TRUE</span>, cache.lazy <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="fu">knitr</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/knitr/man/opts_knit.html" class="external-link">opts_knit</a></span><span class="op">$</span><span class="fu">set</span><span class="op">(</span>verbose <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>In this vignette, we examine the behaviour of the coalescing
algorithm under varying thresholds of minimum alignment width. In doing
so we attempt to fully describe the contiguiuty of the pairwise
alignments.</p>
<p>The brunt of this vignette is the product of Charles’ hard work.</p>
<p>The core functions used here are maintained in our
<em>GenomicBreaks</em> R package, which is fully documented at: <a href="https://oist.github.io/GenomicBreaks" class="external-link uri">https://oist.github.io/GenomicBreaks</a>.</p>
<div class="section level3">
<h3 id="load-packages">Load packages<a class="anchor" aria-label="anchor" href="#load-packages"></a>
</h3>
</div>
</div>
<div class="section level2">
<h2 id="load-r-packages-and-data">Load R packages and data<a class="anchor" aria-label="anchor" href="#load-r-packages-and-data"></a>
</h2>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/oist/LuscombeU_OikScrambling" class="external-link">'OikScrambling'</a></span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://dplyr.tidyverse.org" class="external-link">'dplyr'</a></span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Attaching package: 'dplyr'</span></span></code></pre>
<pre><code><span><span class="co">## The following objects are masked from 'package:Biostrings':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     collapse, intersect, setdiff, setequal, union</span></span></code></pre>
<pre><code><span><span class="co">## The following object is masked from 'package:XVector':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     slice</span></span></code></pre>
<pre><code><span><span class="co">## The following objects are masked from 'package:GenomicRanges':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     intersect, setdiff, union</span></span></code></pre>
<pre><code><span><span class="co">## The following object is masked from 'package:GenomeInfoDb':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     intersect</span></span></code></pre>
<pre><code><span><span class="co">## The following objects are masked from 'package:IRanges':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     collapse, desc, intersect, setdiff, slice, union</span></span></code></pre>
<pre><code><span><span class="co">## The following objects are masked from 'package:S4Vectors':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     first, intersect, rename, setdiff, setequal, union</span></span></code></pre>
<pre><code><span><span class="co">## The following objects are masked from 'package:BiocGenerics':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     combine, intersect, setdiff, union</span></span></code></pre>
<pre><code><span><span class="co">## The following objects are masked from 'package:stats':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     filter, lag</span></span></code></pre>
<pre><code><span><span class="co">## The following objects are masked from 'package:base':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     intersect, setdiff, setequal, union</span></span></code></pre>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://ggplot2.tidyverse.org" class="external-link">'ggplot2'</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st">'pheatmap'</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://purrr.tidyverse.org/" class="external-link">'purrr'</a></span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Attaching package: 'purrr'</span></span></code></pre>
<pre><code><span><span class="co">## The following object is masked from 'package:XVector':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     compact</span></span></code></pre>
<pre><code><span><span class="co">## The following object is masked from 'package:GenomicRanges':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     reduce</span></span></code></pre>
<pre><code><span><span class="co">## The following object is masked from 'package:IRanges':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     reduce</span></span></code></pre>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://sjmgarnier.github.io/viridis/" class="external-link">'viridis'</a></span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="op">(</span><span class="va">genomes</span> <span class="op">&lt;-</span> <span class="fu">OikScrambling</span><span class="fu">:::</span><span class="fu"><a href="../reference/loadAllGenomes.html">loadAllGenomes</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Warning in runHook(".onLoad", env, package.lib, package): input string</span></span>
<span><span class="co">## 'Génoscope' cannot be translated from 'ANSI_X3.4-1968' to UTF-8, but is valid</span></span>
<span><span class="co">## UTF-8</span></span>
<span></span>
<span><span class="co">## Warning in runHook(".onLoad", env, package.lib, package): input string</span></span>
<span><span class="co">## 'Génoscope' cannot be translated from 'ANSI_X3.4-1968' to UTF-8, but is valid</span></span>
<span><span class="co">## UTF-8</span></span></code></pre>
<pre><code><span><span class="co">## List of length 6</span></span>
<span><span class="co">## names(6): Oki Osa Bar Kum Aom Nor</span></span></code></pre>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/load.html" class="external-link">load</a></span><span class="op">(</span><span class="st">"BreakPoints.Rdata"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##  [1] "gbs"       "unal"      "coa"       "unmap"     "bri"       "tra"      </span></span>
<span><span class="co">##  [7] "tra2"      "coa2"      "unmap2"    "wgo"       "longShort"</span></span></code></pre>
<div class="section level4">
<h4 id="generating-coalesced-alignments">Generating coalesced alignments<a class="anchor" aria-label="anchor" href="#generating-coalesced-alignments"></a>
</h4>
<p>The function <code><a href="https://oist.github.io/GenomicBreaks/reference/coalesce_contigs.html" class="external-link">coalesce_contigs()</a></code> is the primary driver of
our synteny analysis. It accepts an argument, <code>minwidth</code>,
which controls the minimum width of segments to be considered for
synteny analyses.</p>
<p>In the next session, I calculate these alignments at a large number
of minimum width cutoffs.</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># use coalesce_contigs at different cutoffs</span></span>
<span><span class="va">minwidthCutoffs</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">25</span>, <span class="fl">50</span>, <span class="fl">100</span>, <span class="fl">250</span>, <span class="fl">500</span>, <span class="fl">750</span>, <span class="fl">1000</span>, <span class="fl">1250</span>, <span class="fl">1500</span>, <span class="fl">2000</span>, <span class="fl">2500</span>, <span class="fl">5000</span>, <span class="fl">7500</span>, <span class="fl">10000</span>, <span class="fl">12500</span>, <span class="fl">25000</span>, <span class="fl">50000</span>, <span class="fl">100000</span><span class="op">)</span></span>
<span><span class="va">cutoff_coas</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">minwidthCutoffs</span>, <span class="kw">function</span><span class="op">(</span><span class="va">MinWidth</span><span class="op">)</span> <span class="fu">BiocParallel</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocParallel/man/bplapply.html" class="external-link">bplapply</a></span><span class="op">(</span><span class="va">gbs</span>, <span class="kw">function</span><span class="op">(</span><span class="va">gb</span><span class="op">)</span> <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/coalesce_contigs.html" class="external-link">coalesce_contigs</a></span><span class="op">(</span>gb<span class="op">=</span><span class="va">gb</span>, minwidth<span class="op">=</span><span class="va">MinWidth</span><span class="op">)</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/SimpleList-class.html" class="external-link">SimpleList</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/SimpleList-class.html" class="external-link">SimpleList</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">cutoff_coas</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">minwidthCutoffs</span>, <span class="kw">function</span><span class="op">(</span><span class="va">s</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="st">'coa_minWidth_'</span>, <span class="fu"><a href="https://rdrr.io/r/base/character.html" class="external-link">as.character</a></span><span class="op">(</span><span class="va">s</span><span class="op">)</span>, sep<span class="op">=</span><span class="st">''</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co"># Append uncoalesced alignment</span></span>
<span><span class="va">cutoff_coas</span><span class="op">$</span><span class="va">uncoalesced</span> <span class="op">&lt;-</span> <span class="va">gbs</span></span>
<span><span class="co"># Reorder to put uncoalesced first</span></span>
<span><span class="va">cutoff_coas</span> <span class="op">&lt;-</span> <span class="va">cutoff_coas</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">20</span>,<span class="fl">1</span><span class="op">:</span><span class="fl">19</span><span class="op">)</span><span class="op">]</span></span></code></pre></div>
<p>Let’s make some tables to help understand what’s going on, in terms
of how much is being removed, how big the syntenic regions are, and so
on.</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># You can then make a simple table that tabulates the number of segments at a number of cutoffs.</span></span>
<span><span class="va">cutoff_coas_tables</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/SimpleList-class.html" class="external-link">SimpleList</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">cutoff_coas_tables_map</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">f</span><span class="op">)</span></span>
<span>  <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map_dfr.html" class="external-link">map_dfr</a></span><span class="op">(</span><span class="va">cutoff_coas</span>,</span>
<span>    \<span class="op">(</span><span class="va">cutoff</span><span class="op">)</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html" class="external-link">map</a></span><span class="op">(</span><span class="va">cutoff</span>, <span class="va">f</span><span class="op">)</span></span>
<span>  <span class="op">)</span></span>
<span><span class="co"># Count number of segments. We make 0-count cells to NA to make them more obvious in the heatmap later.</span></span>
<span><span class="va">cutoff_coas_tables</span><span class="op">$</span><span class="va">NumSegments</span> <span class="op">&lt;-</span> <span class="fu">cutoff_coas_tables_map</span><span class="op">(</span><span class="va">length</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Show how many total bases are left after coalescing.</span></span>
<span><span class="co"># Divide by 1,000,000 to convert it to Mbp. 0-counts are converted to NA.</span></span>
<span><span class="va">cutoff_coas_tables</span><span class="op">$</span><span class="va">TotalAligned</span> <span class="op">&lt;-</span> <span class="fu">cutoff_coas_tables_map</span><span class="op">(</span>\<span class="op">(</span><span class="va">gr</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/start.html" class="external-link">width</a></span><span class="op">(</span><span class="va">gr</span><span class="op">)</span><span class="op">/</span><span class="fl">1000000</span><span class="op">)</span><span class="op">)</span></span>
<span>  </span>
<span><span class="co"># Show largest segment. Combine it with the query name to see which chromosome segment is biggest.</span></span>
<span><span class="va">cutoff_coas_tables</span><span class="op">$</span><span class="va">LargestSegment</span> <span class="op">&lt;-</span> <span class="fu">cutoff_coas_tables_map</span><span class="op">(</span>\<span class="op">(</span><span class="va">gr</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="kw">if</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">gr</span><span class="op">)</span> <span class="op">==</span> <span class="fl">0</span><span class="op">)</span> <span class="kw"><a href="https://rdrr.io/r/base/function.html" class="external-link">return</a></span><span class="op">(</span><span class="cn">NA</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html" class="external-link">max</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/start.html" class="external-link">width</a></span><span class="op">(</span><span class="va">gr</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="op">}</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Get name of largest segment, which can change based on the cutoffs.</span></span>
<span><span class="va">cutoff_coas_tables</span><span class="op">$</span><span class="va">LargestSegmentName</span> <span class="op">&lt;-</span> <span class="fu">cutoff_coas_tables_map</span><span class="op">(</span>\<span class="op">(</span><span class="va">gb</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="kw">if</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">gb</span><span class="op">)</span> <span class="op">==</span> <span class="fl">0</span><span class="op">)</span> <span class="kw"><a href="https://rdrr.io/r/base/function.html" class="external-link">return</a></span><span class="op">(</span><span class="cn">NA</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/character.html" class="external-link">as.character</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqnames</a></span><span class="op">(</span><span class="va">gb</span><span class="op">)</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/which.min.html" class="external-link">which.max</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/start.html" class="external-link">width</a></span><span class="op">(</span><span class="va">gb</span><span class="op">)</span><span class="op">)</span><span class="op">]</span><span class="op">)</span></span>
<span>  <span class="op">}</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Write a little function to relate a pair of species (A_B) to their genome size.</span></span>
<span><span class="co"># named list of genome sizes (name=species, content=genome size).</span></span>
<span><span class="va">genome_sizes</span><span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>Aom<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqlengths</a></span><span class="op">(</span><span class="va">genomes</span><span class="op">$</span><span class="va">Aom</span><span class="op">)</span><span class="op">)</span>, Bar<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqlengths</a></span><span class="op">(</span><span class="va">genomes</span><span class="op">$</span><span class="va">Bar</span><span class="op">)</span><span class="op">)</span>,</span>
<span>                  Oki<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqlengths</a></span><span class="op">(</span><span class="va">genomes</span><span class="op">$</span><span class="va">Oki</span><span class="op">)</span><span class="op">)</span>, Osa<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqlengths</a></span><span class="op">(</span><span class="va">genomes</span><span class="op">$</span><span class="va">Osa</span><span class="op">)</span><span class="op">)</span>,</span>
<span>                  Kum<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqlengths</a></span><span class="op">(</span><span class="va">genomes</span><span class="op">$</span><span class="va">Kum</span><span class="op">)</span><span class="op">)</span>, Nor<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqlengths</a></span><span class="op">(</span><span class="va">genomes</span><span class="op">$</span><span class="va">Nor</span><span class="op">)</span><span class="op">)</span>,</span>
<span>                  Ply<span class="op">=</span><span class="fl">128639392</span>, Ros<span class="op">=</span><span class="fl">134327879</span>, Sav<span class="op">=</span><span class="fl">177003750</span>, Rob<span class="op">=</span><span class="fl">117529544</span>,</span>
<span>                  Dme<span class="op">=</span><span class="fl">137547960</span>, Dbu<span class="op">=</span><span class="fl">110837441</span>, Dsu<span class="op">=</span><span class="fl">130045071</span>, Dya<span class="op">=</span><span class="fl">136883712</span>,</span>
<span>                  Cin<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/guessSeqLengths.html" class="external-link">guessSeqLengths</a></span><span class="op">(</span><span class="va">gbs</span><span class="op">$</span><span class="va">Cni_Cin</span><span class="op">)</span><span class="op">)</span>,</span>
<span>                  Cel<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/guessSeqLengths.html" class="external-link">guessSeqLengths</a></span><span class="op">(</span><span class="va">gbs</span><span class="op">$</span><span class="va">Cbr_Cel</span><span class="op">$</span><span class="va">query</span><span class="op">)</span><span class="op">)</span>,</span>
<span>                  Cre<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/guessSeqLengths.html" class="external-link">guessSeqLengths</a></span><span class="op">(</span><span class="va">gbs</span><span class="op">$</span><span class="va">Cni_Cre</span><span class="op">$</span><span class="va">query</span><span class="op">)</span><span class="op">)</span>,</span>
<span>                  Cbr<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/guessSeqLengths.html" class="external-link">guessSeqLengths</a></span><span class="op">(</span><span class="va">gbs</span><span class="op">$</span><span class="va">Cbr_Cni</span><span class="op">)</span><span class="op">)</span>,</span>
<span>                  Cni<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/guessSeqLengths.html" class="external-link">guessSeqLengths</a></span><span class="op">(</span><span class="va">gbs</span><span class="op">$</span><span class="va">Cni_Cbr</span><span class="op">)</span><span class="op">)</span></span>
<span>                  <span class="op">)</span></span>
<span><span class="va">match_pair_with_sizes</span> <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">pair</span>, <span class="va">genome_sizes</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="va">spp</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/unlist.html" class="external-link">unlist</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/strsplit.html" class="external-link">strsplit</a></span><span class="op">(</span><span class="va">pair</span>, <span class="st">"_"</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="va">g1</span> <span class="op">=</span> <span class="va">spp</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span></span>
<span>  <span class="va">g2</span> <span class="op">=</span> <span class="va">spp</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span></span>
<span>  <span class="va">gs1</span> <span class="op">=</span> <span class="va">genome_sizes</span><span class="op">[[</span><span class="va">g1</span><span class="op">]</span><span class="op">]</span></span>
<span>  <span class="va">gs2</span> <span class="op">=</span> <span class="va">genome_sizes</span><span class="op">[[</span><span class="va">g2</span><span class="op">]</span><span class="op">]</span></span>
<span>  <span class="va">result</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="op">)</span></span>
<span>  <span class="va">result</span><span class="op">[[</span><span class="va">g1</span><span class="op">]</span><span class="op">]</span> <span class="op">=</span> <span class="va">gs1</span></span>
<span>  <span class="va">result</span><span class="op">[[</span><span class="va">g2</span><span class="op">]</span><span class="op">]</span> <span class="op">=</span> <span class="va">gs2</span></span>
<span>  <span class="kw"><a href="https://rdrr.io/r/base/function.html" class="external-link">return</a></span><span class="op">(</span><span class="va">result</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span><span class="co"># Use the genome sizes to calculate how much of the genomes are covered.</span></span>
<span><span class="co"># This coverage value is just the total alignment length divided by the mean of the two genome sizes.</span></span>
<span><span class="va">cutoff_coas_tables</span><span class="op">$</span><span class="va">PercentGenomeCovered</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/do.call.html" class="external-link">do.call</a></span><span class="op">(</span><span class="va">rbind</span>, <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">cutoff_coas</span>, <span class="kw">function</span><span class="op">(</span><span class="va">cutoff</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="va">cut_gb</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">cutoff</span><span class="op">)</span>, <span class="kw">function</span><span class="op">(</span><span class="va">pair</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="va">gb</span> <span class="op">&lt;-</span> <span class="va">cutoff</span><span class="op">[[</span><span class="va">pair</span><span class="op">]</span><span class="op">]</span></span>
<span>    <span class="va">pair_sizes</span> <span class="op">&lt;-</span> <span class="fu">match_pair_with_sizes</span><span class="op">(</span><span class="va">pair</span>, genome_sizes<span class="op">=</span><span class="va">genome_sizes</span><span class="op">)</span></span>
<span>    <span class="va">mean_genome_size</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/mean.html" class="external-link">mean</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/unlist.html" class="external-link">unlist</a></span><span class="op">(</span><span class="va">pair_sizes</span><span class="op">)</span><span class="op">)</span></span>
<span>    <span class="va">covered</span> <span class="op">&lt;-</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/start.html" class="external-link">width</a></span><span class="op">(</span><span class="va">gb</span><span class="op">)</span><span class="op">)</span><span class="op">/</span><span class="va">mean_genome_size</span><span class="op">)</span><span class="op">*</span><span class="fl">100</span></span>
<span>    <span class="va">covered</span></span>
<span>  <span class="op">}</span><span class="op">)</span></span>
<span>  <span class="va">cut_gb</span></span>
<span><span class="op">}</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co"># Show how many bases of the original alignment are covered.</span></span>
<span><span class="va">uncoalescedAliTot</span> <span class="op">&lt;-</span> <span class="va">cutoff_coas_tables</span><span class="op">$</span><span class="va">TotalAligned</span><span class="op">[</span><span class="fl">1</span>,<span class="op">]</span></span>
<span><span class="va">cutoff_coas_tables</span><span class="op">$</span><span class="va">PercentOriginalAlignment</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/do.call.html" class="external-link">do.call</a></span><span class="op">(</span><span class="va">rbind</span>, <span class="fu"><a href="https://rdrr.io/r/base/apply.html" class="external-link">apply</a></span><span class="op">(</span><span class="va">cutoff_coas_tables</span><span class="op">$</span><span class="va">TotalAligned</span>, <span class="fl">1</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="va">x</span><span class="op">/</span><span class="va">uncoalescedAliTot</span><span class="op">)</span><span class="op">)</span> <span class="op">*</span><span class="fl">100</span></span></code></pre></div>
<p>Note the names of the largest segments.</p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="op">(</span><span class="va">cutoff_coas_tables</span><span class="op">$</span><span class="va">LargestSegmentName</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## <span style="color: #949494;"># A tibble: 20 × 33</span></span></span>
<span><span class="co">##    Oki_Osa Oki_Bar Oki_Kum Oki_Aom Oki_Nor Osa_Oki Osa_Bar Osa_Kum Osa_Aom</span></span>
<span><span class="co">##    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 1</span> PAR     PAR     XSR     PAR     PAR     PAR     PAR     PAR     XSR    </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 2</span> PAR     chr2    XSR     XSR     XSR     XSR     Chr2    XSR     XSR    </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 3</span> PAR     chr2    XSR     XSR     XSR     XSR     Chr2    XSR     XSR    </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 4</span> PAR     chr2    XSR     XSR     XSR     XSR     Chr2    XSR     XSR    </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 5</span> PAR     chr2    XSR     XSR     XSR     XSR     Chr2    XSR     XSR    </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 6</span> PAR     chr2    XSR     XSR     XSR     XSR     XSR     XSR     XSR    </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 7</span> chr2    chr2    XSR     chr2    XSR     XSR     Chr2    XSR     XSR    </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 8</span> chr2    chr2    XSR     chr2    XSR     XSR     XSR     XSR     XSR    </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 9</span> chr2    XSR     XSR     XSR     XSR     XSR     XSR     XSR     XSR    </span></span>
<span><span class="co">## <span style="color: #BCBCBC;">10</span> chr2    chr2    XSR     XSR     XSR     XSR     XSR     XSR     XSR    </span></span>
<span><span class="co">## <span style="color: #BCBCBC;">11</span> XSR     chr2    XSR     XSR     XSR     XSR     XSR     XSR     XSR    </span></span>
<span><span class="co">## <span style="color: #BCBCBC;">12</span> XSR     XSR     XSR     XSR     XSR     XSR     XSR     XSR     XSR    </span></span>
<span><span class="co">## <span style="color: #BCBCBC;">13</span> XSR     XSR     XSR     YSR     XSR     XSR     XSR     XSR     XSR    </span></span>
<span><span class="co">## <span style="color: #BCBCBC;">14</span> PAR     chr2    XSR     PAR     XSR     PAR     XSR     PAR     XSR    </span></span>
<span><span class="co">## <span style="color: #BCBCBC;">15</span> XSR     chr1    XSR     XSR     XSR     XSR     XSR     XSR     XSR    </span></span>
<span><span class="co">## <span style="color: #BCBCBC;">16</span> XSR     chr1    XSR     XSR     PAR     XSR     XSR     XSR     XSR    </span></span>
<span><span class="co">## <span style="color: #BCBCBC;">17</span> chr2    chr1    XSR     chr2    chr2    Chr2    XSR     XSR     XSR    </span></span>
<span><span class="co">## <span style="color: #BCBCBC;">18</span> PAR     PAR     XSR     PAR     PAR     PAR     XSR     PAR     XSR    </span></span>
<span><span class="co">## <span style="color: #BCBCBC;">19</span> <span style="color: #BB0000;">NA</span>      <span style="color: #BB0000;">NA</span>      XSR     <span style="color: #BB0000;">NA</span>      <span style="color: #BB0000;">NA</span>      <span style="color: #BB0000;">NA</span>      <span style="color: #BB0000;">NA</span>      <span style="color: #BB0000;">NA</span>      XSR    </span></span>
<span><span class="co">## <span style="color: #BCBCBC;">20</span> <span style="color: #BB0000;">NA</span>      <span style="color: #BB0000;">NA</span>      <span style="color: #BB0000;">NA</span>      <span style="color: #BB0000;">NA</span>      <span style="color: #BB0000;">NA</span>      <span style="color: #BB0000;">NA</span>      <span style="color: #BB0000;">NA</span>      <span style="color: #BB0000;">NA</span>      XSR    </span></span>
<span><span class="co">## <span style="color: #949494;"># ℹ 24 more variables: Osa_Nor &lt;chr&gt;, Bar_Oki &lt;chr&gt;, Bar_Osa &lt;chr&gt;,</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   Bar_Kum &lt;chr&gt;, Bar_Aom &lt;chr&gt;, Bar_Nor &lt;chr&gt;, Ply_Ros &lt;chr&gt;, Ply_Rob &lt;chr&gt;,</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   Ply_Sav &lt;chr&gt;, Ply_Oki &lt;chr&gt;, Rob_Ros &lt;chr&gt;, Rob_Ply &lt;chr&gt;, Rob_Sav &lt;chr&gt;,</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   Rob_Oki &lt;chr&gt;, Dme_Dbu &lt;chr&gt;, Dme_Dsu &lt;chr&gt;, Dme_Dya &lt;chr&gt;, Dme_Dma &lt;chr&gt;,</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   Cni_Cbr &lt;chr&gt;, Cni_Cre &lt;chr&gt;, Cni_Cin &lt;chr&gt;, Cbr_Cni &lt;chr&gt;, Cbr_Cre &lt;chr&gt;,</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   Cbr_Cel &lt;chr&gt;</span></span></span></code></pre>
</div>
<div class="section level4">
<h4 id="plotting-features-at-different-cutoffs">Plotting features at different cutoffs<a class="anchor" aria-label="anchor" href="#plotting-features-at-different-cutoffs"></a>
</h4>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pheatmap_</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">x</span>, <span class="va">main</span><span class="op">)</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/pheatmap/man/pheatmap.html" class="external-link">pheatmap</a></span><span class="op">(</span><span class="va">x</span>, col<span class="op">=</span><span class="fu"><a href="https://sjmgarnier.github.io/viridisLite/reference/viridis.html" class="external-link">magma</a></span><span class="op">(</span><span class="fl">100</span><span class="op">)</span>, cluster_cols<span class="op">=</span><span class="cn">F</span>, cluster_row<span class="op">=</span><span class="cn">F</span>, gaps_col<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">15</span>,<span class="fl">23</span>,<span class="fl">27</span><span class="op">)</span>, angle_col <span class="op">=</span> <span class="fl">90</span>, main<span class="op">=</span><span class="va">main</span>, display_numbers<span class="op">=</span><span class="cn">T</span>, number_color<span class="op">=</span><span class="st">'#d9d9d9'</span>, number_format<span class="op">=</span><span class="st">"%.0f"</span>, fontsize_number<span class="op">=</span><span class="fl">6</span><span class="op">)</span></span></code></pre></div>
<div class="section level6">
<h6 id="number-of-segments">Number of segments<a class="anchor" aria-label="anchor" href="#number-of-segments"></a>
</h6>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Number of segments</span></span>
<span><span class="fu">pheatmap_</span><span class="op">(</span><span class="va">cutoff_coas_tables</span><span class="op">$</span><span class="va">NumSegments</span>, main<span class="op">=</span><span class="st">'Number of alignments (uncoalesced) or syntenic regions (coalesced)'</span><span class="op">)</span></span></code></pre></div>
<p><img src="CoalescingCutoffs_files/figure-html/coa_cutoff_numSegs-1.svg" width="768"></p>
</div>
<div class="section level6">
<h6 id="total-aligned">Total aligned<a class="anchor" aria-label="anchor" href="#total-aligned"></a>
</h6>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">pheatmap_</span><span class="op">(</span><span class="va">cutoff_coas_tables</span><span class="op">$</span><span class="va">TotalAligned</span>, main<span class="op">=</span><span class="st">'Sum of all alignment widths (Mbp)'</span><span class="op">)</span></span></code></pre></div>
<p><img src="CoalescingCutoffs_files/figure-html/coa_cutoff_totAli-1.svg" width="768"></p>
</div>
<div class="section level6">
<h6 id="largest-segment">Largest segment<a class="anchor" aria-label="anchor" href="#largest-segment"></a>
</h6>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">pheatmap_</span><span class="op">(</span><span class="va">cutoff_coas_tables</span><span class="op">$</span><span class="va">LargestSegment</span>, main<span class="op">=</span><span class="st">'Size of largest segment (Mbp)'</span><span class="op">)</span></span></code></pre></div>
<p><img src="CoalescingCutoffs_files/figure-html/coa_cutoff_lrgstSeg-1.svg" width="768"></p>
</div>
<div class="section level6">
<h6 id="percent-of-genome-covered-by-alignments">Percent of genome covered by alignments<a class="anchor" aria-label="anchor" href="#percent-of-genome-covered-by-alignments"></a>
</h6>
<p>Note the denominator here is the mean of the query and target genome
sizes.</p>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">pheatmap_</span><span class="op">(</span><span class="va">cutoff_coas_tables</span><span class="op">$</span><span class="va">PercentGenomeCovered</span>, main<span class="op">=</span><span class="st">'Percentage of query/target genomes covered (%)'</span><span class="op">)</span></span></code></pre></div>
<p><img src="CoalescingCutoffs_files/figure-html/coa_cutoff_pctGen-1.svg" width="768"></p>
</div>
<div class="section level6">
<h6 id="percent-of-total-aligned-positions-represented-in-alignments">Percent of total aligned positions represented in alignments<a class="anchor" aria-label="anchor" href="#percent-of-total-aligned-positions-represented-in-alignments"></a>
</h6>
<p>Note the denominator here is the mean of the query and target genome
sizes.</p>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">pheatmap_</span><span class="op">(</span><span class="va">cutoff_coas_tables</span><span class="op">$</span><span class="va">PercentOriginalAlignment</span>, main<span class="op">=</span><span class="st">'Percentage of originally-aligned positions covered (%)'</span><span class="op">)</span></span></code></pre></div>
<p><img src="CoalescingCutoffs_files/figure-html/coa_cutoff_pctOrig-1.svg" width="768"></p>
</div>
</div>
<div class="section level4">
<h4 id="the-effect-of-coalescing-cutoffs-on-synteny">The effect of coalescing cutoffs on synteny<a class="anchor" aria-label="anchor" href="#the-effect-of-coalescing-cutoffs-on-synteny"></a>
</h4>
<p>Now let’s make some chromosome plots. Here are the original
Okinawa-Osaka alignments, with no coalescing.</p>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/plotApairOfChrs.html" class="external-link">plotApairOfChrs</a></span><span class="op">(</span><span class="va">cutoff_coas</span><span class="op">$</span><span class="va">uncoalesced</span><span class="op">$</span><span class="va">Oki_Osa</span>, <span class="st">'chr1'</span><span class="op">)</span></span></code></pre></div>
<p><img src="CoalescingCutoffs_files/figure-html/coa_cutoff_chrPlot_1-1.svg" width="576"></p>
<p>You can see many details here, including the huge number of small
alignments. Many of these are removed by coalescing:</p>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/plotApairOfChrs.html" class="external-link">plotApairOfChrs</a></span><span class="op">(</span><span class="va">cutoff_coas</span><span class="op">$</span><span class="va">coa_minWidth_0</span><span class="op">$</span><span class="va">Oki_Osa</span>, <span class="st">'chr1'</span><span class="op">)</span></span></code></pre></div>
<p><img src="CoalescingCutoffs_files/figure-html/coa_cutoff_chrPlot_2-1.svg" width="576"></p>
<p>However, many small alignments persist after coalescing. As seen
above, applying length filters to remove small alignments can
drastically reduce the number of segments - or in other words, greatly
increase the contiguity of the coalesced alignments.</p>
<p>Here is the result of removing alignments below 250bp - a fairly
conservative threshold:</p>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/plotApairOfChrs.html" class="external-link">plotApairOfChrs</a></span><span class="op">(</span><span class="va">cutoff_coas</span><span class="op">$</span><span class="va">coa_minWidth_250</span><span class="op">$</span><span class="va">Oki_Osa</span>, <span class="st">'chr1'</span><span class="op">)</span></span></code></pre></div>
<p><img src="CoalescingCutoffs_files/figure-html/coa_cutoff_chrPlot_3-1.svg" width="576"></p>
<p>And if we go too far, we lose too much information for it to be
useful:</p>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/plotApairOfChrs.html" class="external-link">plotApairOfChrs</a></span><span class="op">(</span><span class="va">cutoff_coas</span><span class="op">$</span><span class="va">coa_minWidth_10000</span><span class="op">$</span><span class="va">Oki_Osa</span>, <span class="st">'chr1'</span><span class="op">)</span></span></code></pre></div>
<p><img src="CoalescingCutoffs_files/figure-html/coa_cutoff_chrPlot_4-1.svg" width="576"></p>
<div class="section level5">
<h5 id="using-maximal-coverage-of-querytarget-genomes-as-the-selector">Using “maximal coverage of query/target genomes” as the
selector<a class="anchor" aria-label="anchor" href="#using-maximal-coverage-of-querytarget-genomes-as-the-selector"></a>
</h5>
<p>For the following plots, I use my own arbitrary choices to select a
cutoff for each pair that balances retaining as much information as
possible with the largest possible synteny. I select the cutoff that
maximizes the total coverage of the pair - the maximum “Sum of all
alignment widths” cell from the heatmap above.</p>
<div class="section level7">
<h6 id="okinawa-osaka">Okinawa-Osaka<a class="anchor" aria-label="anchor" href="#okinawa-osaka"></a>
</h6>
<div class="sourceCode" id="cb37"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/plotApairOfChrs.html" class="external-link">plotApairOfChrs</a></span><span class="op">(</span><span class="va">cutoff_coas</span><span class="op">$</span><span class="va">coa_minWidth_250</span><span class="op">$</span><span class="va">Oki_Osa</span>, <span class="st">'chr1'</span><span class="op">)</span></span></code></pre></div>
<p><img src="CoalescingCutoffs_files/figure-html/coa_cutoff_chrPlot_5-1.svg" width="576"></p>
<div class="sourceCode" id="cb38"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/plotApairOfChrs.html" class="external-link">plotApairOfChrs</a></span><span class="op">(</span><span class="va">cutoff_coas</span><span class="op">$</span><span class="va">coa_minWidth_250</span><span class="op">$</span><span class="va">Oki_Osa</span>, <span class="st">'chr2'</span><span class="op">)</span></span></code></pre></div>
<p><img src="CoalescingCutoffs_files/figure-html/coa_cutoff_chrPlot_6-1.svg" width="576"></p>
<div class="sourceCode" id="cb39"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/plotApairOfChrs.html" class="external-link">plotApairOfChrs</a></span><span class="op">(</span><span class="va">cutoff_coas</span><span class="op">$</span><span class="va">coa_minWidth_250</span><span class="op">$</span><span class="va">Oki_Osa</span>, <span class="st">'PAR'</span><span class="op">)</span></span></code></pre></div>
<p><img src="CoalescingCutoffs_files/figure-html/coa_cutoff_chrPlot_7-1.svg" width="576"></p>
</div>
<div class="section level7">
<h6 id="okinawa-barcelona">Okinawa-Barcelona<a class="anchor" aria-label="anchor" href="#okinawa-barcelona"></a>
</h6>
<div class="sourceCode" id="cb40"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/plotApairOfChrs.html" class="external-link">plotApairOfChrs</a></span><span class="op">(</span><span class="va">cutoff_coas</span><span class="op">$</span><span class="va">coa_minWidth_100</span><span class="op">$</span><span class="va">Oki_Bar</span>, <span class="st">'chr1'</span><span class="op">)</span></span></code></pre></div>
<p><img src="CoalescingCutoffs_files/figure-html/coa_cutoff_chrPlot_8-1.svg" width="576"></p>
<div class="sourceCode" id="cb41"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/plotApairOfChrs.html" class="external-link">plotApairOfChrs</a></span><span class="op">(</span><span class="va">cutoff_coas</span><span class="op">$</span><span class="va">coa_minWidth_100</span><span class="op">$</span><span class="va">Oki_Bar</span>, <span class="st">'chr2'</span><span class="op">)</span></span></code></pre></div>
<p><img src="CoalescingCutoffs_files/figure-html/coa_cutoff_chrPlot_9-1.svg" width="576"></p>
<div class="sourceCode" id="cb42"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/plotApairOfChrs.html" class="external-link">plotApairOfChrs</a></span><span class="op">(</span><span class="va">cutoff_coas</span><span class="op">$</span><span class="va">coa_minWidth_100</span><span class="op">$</span><span class="va">Oki_Bar</span>, <span class="st">'PAR'</span><span class="op">)</span></span></code></pre></div>
<p><img src="CoalescingCutoffs_files/figure-html/coa_cutoff_chrPlot_10-1.svg" width="576"></p>
</div>
<div class="section level7">
<h6 id="osaka-barcelona">Osaka-Barcelona<a class="anchor" aria-label="anchor" href="#osaka-barcelona"></a>
</h6>
<div class="sourceCode" id="cb43"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/plotApairOfChrs.html" class="external-link">plotApairOfChrs</a></span><span class="op">(</span><span class="va">cutoff_coas</span><span class="op">$</span><span class="va">coa_minWidth_250</span><span class="op">$</span><span class="va">Osa_Bar</span>, <span class="st">'Chr1'</span><span class="op">)</span></span></code></pre></div>
<p><img src="CoalescingCutoffs_files/figure-html/coa_cutoff_chrPlot_11-1.svg" width="576"></p>
<div class="sourceCode" id="cb44"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/plotApairOfChrs.html" class="external-link">plotApairOfChrs</a></span><span class="op">(</span><span class="va">cutoff_coas</span><span class="op">$</span><span class="va">coa_minWidth_250</span><span class="op">$</span><span class="va">Osa_Bar</span>, <span class="st">'Chr2'</span><span class="op">)</span></span></code></pre></div>
<p><img src="CoalescingCutoffs_files/figure-html/coa_cutoff_chrPlot_12-1.svg" width="576"></p>
<div class="sourceCode" id="cb45"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/plotApairOfChrs.html" class="external-link">plotApairOfChrs</a></span><span class="op">(</span><span class="va">cutoff_coas</span><span class="op">$</span><span class="va">coa_minWidth_250</span><span class="op">$</span><span class="va">Osa_Bar</span>, <span class="st">'PAR'</span><span class="op">)</span></span></code></pre></div>
<p><img src="CoalescingCutoffs_files/figure-html/coa_cutoff_chrPlot_13-1.svg" width="576"></p>
</div>
<div class="section level7">
<h6 id="okinawa-kume">Okinawa-Kume<a class="anchor" aria-label="anchor" href="#okinawa-kume"></a>
</h6>
<p>The following is taken from the
<code><a href="../articles/ParallelPlots.html">vignette("ParallelPlots", package = "OikScrambling")</a></code>
vignette. It is invoked here to allow later plots between Okinawa and
Kume.</p>
<div class="sourceCode" id="cb46"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"ade4"</span><span class="op">)</span></span>
<span><span class="va">treeLeaf</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">name</span>, <span class="va">length</span><span class="op">=</span><span class="cn">NULL</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="kw">if</span><span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NULL.html" class="external-link">is.null</a></span><span class="op">(</span><span class="va">length</span><span class="op">)</span><span class="op">)</span> <span class="va">length</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">':'</span>, <span class="va">length</span><span class="op">)</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">name</span>, <span class="va">length</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="va">treeNode</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">branch1</span>, <span class="va">branch2</span>, <span class="va">length</span> <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="kw">if</span><span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NULL.html" class="external-link">is.null</a></span><span class="op">(</span><span class="va">length</span><span class="op">)</span><span class="op">)</span> <span class="va">length</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">':'</span>, <span class="va">length</span><span class="op">)</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">'('</span>, <span class="va">branch1</span>, <span class="st">','</span>, <span class="va">branch2</span>, <span class="st">')'</span>, <span class="va">length</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="va">addRoot</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">branch</span><span class="op">)</span>  <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">branch</span>, <span class="st">";"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">tree</span> <span class="op">&lt;-</span></span>
<span>  <span class="fu">addRoot</span><span class="op">(</span></span>
<span>    <span class="fu">treeNode</span><span class="op">(</span></span>
<span>      <span class="fu">treeNode</span><span class="op">(</span> length <span class="op">=</span> <span class="fl">2</span>,</span>
<span>        <span class="fu">treeLeaf</span><span class="op">(</span><span class="st">"Okinawa"</span>, <span class="fl">1</span><span class="op">)</span>,</span>
<span>        <span class="fu">treeLeaf</span><span class="op">(</span><span class="st">"Kume"</span>, <span class="fl">1</span><span class="op">)</span></span>
<span>      <span class="op">)</span>,</span>
<span>      <span class="fu">treeNode</span><span class="op">(</span> length <span class="op">=</span> <span class="fl">1</span>,</span>
<span>        <span class="fu">treeNode</span><span class="op">(</span> length <span class="op">=</span> <span class="fl">1</span>,</span>
<span>          <span class="fu">treeLeaf</span><span class="op">(</span><span class="st">"Osaka"</span>, <span class="fl">1</span><span class="op">)</span>,</span>
<span>          <span class="fu">treeLeaf</span><span class="op">(</span><span class="st">"Aomori"</span>, <span class="fl">1</span><span class="op">)</span></span>
<span>        <span class="op">)</span>,</span>
<span>        <span class="fu">treeNode</span><span class="op">(</span> length <span class="op">=</span><span class="fl">1</span>,</span>
<span>          <span class="fu">treeLeaf</span><span class="op">(</span><span class="st">"Norway"</span>, <span class="fl">1</span><span class="op">)</span>,</span>
<span>          <span class="fu">treeLeaf</span><span class="op">(</span><span class="st">"Barcelona"</span>, <span class="fl">1</span><span class="op">)</span></span>
<span>        <span class="op">)</span></span>
<span>      <span class="op">)</span></span>
<span>    <span class="op">)</span></span>
<span>  <span class="op">)</span></span></code></pre></div>
<p>The optimal Okinawa-Kume plot:</p>
<div class="sourceCode" id="cb47"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Note that this uses the cutoff coalesced object and not the translocation-removed coa2 object.</span></span>
<span><span class="va">x</span> <span class="op">&lt;-</span> <span class="va">cutoff_coas</span><span class="op">$</span><span class="va">coa_minWidth_2000</span><span class="op">$</span><span class="va">Oki_Kum</span></span>
<span><span class="va">x</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqnames</a></span><span class="op">(</span><span class="va">x</span><span class="op">$</span><span class="va">query</span><span class="op">)</span> <span class="op">==</span> <span class="st">"contig_90_1"</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/reverse-methods.html" class="external-link">reverse</a></span><span class="op">(</span>query <span class="op">=</span> <span class="cn">TRUE</span>, <span class="va">x</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqnames</a></span><span class="op">(</span><span class="va">x</span><span class="op">$</span><span class="va">query</span><span class="op">)</span> <span class="op">==</span> <span class="st">"contig_90_1"</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="va">x</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqnames</a></span><span class="op">(</span><span class="va">x</span><span class="op">$</span><span class="va">query</span><span class="op">)</span> <span class="op">==</span> <span class="st">"contig_88_1"</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/reverse-methods.html" class="external-link">reverse</a></span><span class="op">(</span>query <span class="op">=</span> <span class="cn">TRUE</span>, <span class="va">x</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqnames</a></span><span class="op">(</span><span class="va">x</span><span class="op">$</span><span class="va">query</span><span class="op">)</span> <span class="op">==</span> <span class="st">"contig_88_1"</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="va">x</span><span class="op">$</span><span class="va">query</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/mergeSeqLevels.html" class="external-link">mergeSeqLevels</a></span><span class="op">(</span><span class="va">x</span><span class="op">$</span><span class="va">query</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"contig_27_1"</span>, <span class="st">"contig_90_1"</span>, <span class="st">"contig_3_1"</span>, <span class="st">"contig_88_1"</span><span class="op">)</span>, <span class="st">"new_scaffold"</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/plotApairOfChrs.html" class="external-link">plotApairOfChrs</a></span><span class="op">(</span><span class="va">x</span>, <span class="st">'chr1'</span>, dna_seg_scale <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p><img src="CoalescingCutoffs_files/figure-html/coa_cutoff_chrPlot_14-1.svg" width="576"></p>
<p>Note that when you try to increase this further, you begin losing
many regions:</p>
<div class="sourceCode" id="cb48"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">x</span> <span class="op">&lt;-</span> <span class="va">cutoff_coas</span><span class="op">$</span><span class="va">coa_minWidth_10000</span><span class="op">$</span><span class="va">Oki_Kum</span></span>
<span><span class="va">x</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqnames</a></span><span class="op">(</span><span class="va">x</span><span class="op">$</span><span class="va">query</span><span class="op">)</span> <span class="op">==</span> <span class="st">"contig_90_1"</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/reverse-methods.html" class="external-link">reverse</a></span><span class="op">(</span>query <span class="op">=</span> <span class="cn">TRUE</span>, <span class="va">x</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqnames</a></span><span class="op">(</span><span class="va">x</span><span class="op">$</span><span class="va">query</span><span class="op">)</span> <span class="op">==</span> <span class="st">"contig_90_1"</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="va">x</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqnames</a></span><span class="op">(</span><span class="va">x</span><span class="op">$</span><span class="va">query</span><span class="op">)</span> <span class="op">==</span> <span class="st">"contig_88_1"</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/reverse-methods.html" class="external-link">reverse</a></span><span class="op">(</span>query <span class="op">=</span> <span class="cn">TRUE</span>, <span class="va">x</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqnames</a></span><span class="op">(</span><span class="va">x</span><span class="op">$</span><span class="va">query</span><span class="op">)</span> <span class="op">==</span> <span class="st">"contig_88_1"</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="va">x</span><span class="op">$</span><span class="va">query</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/mergeSeqLevels.html" class="external-link">mergeSeqLevels</a></span><span class="op">(</span><span class="va">x</span><span class="op">$</span><span class="va">query</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"contig_27_1"</span>, <span class="st">"contig_90_1"</span>, <span class="st">"contig_3_1"</span>, <span class="st">"contig_88_1"</span><span class="op">)</span>, <span class="st">"new_scaffold"</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/plotApairOfChrs.html" class="external-link">plotApairOfChrs</a></span><span class="op">(</span><span class="va">x</span>, <span class="st">'chr1'</span>, dna_seg_scale <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p><img src="CoalescingCutoffs_files/figure-html/coa_cutoff_chrPlot_15-1.svg" width="576"></p>
</div>
</div>
</div>
</div>
<div class="section level2">
<h2 id="session-information">Session information<a class="anchor" aria-label="anchor" href="#session-information"></a>
</h2>
<div class="sourceCode" id="cb49"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## R version 4.3.1 (2023-06-16)</span></span>
<span><span class="co">## Platform: x86_64-pc-linux-gnu (64-bit)</span></span>
<span><span class="co">## Running under: Debian GNU/Linux 12 (bookworm)</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Matrix products: default</span></span>
<span><span class="co">## BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.11.0 </span></span>
<span><span class="co">## LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.11.0</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## locale:</span></span>
<span><span class="co">##  [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       </span></span>
<span><span class="co">##  [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   </span></span>
<span><span class="co">##  [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          </span></span>
<span><span class="co">## [10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## time zone: Etc/UTC</span></span>
<span><span class="co">## tzcode source: system (glibc)</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## attached base packages:</span></span>
<span><span class="co">## [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span></span>
<span><span class="co">## [8] base     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## other attached packages:</span></span>
<span><span class="co">##  [1] BSgenome.Oidioi.genoscope.OdB3_1.0.0   </span></span>
<span><span class="co">##  [2] BSgenome.Oidioi.OIST.AOM.5.5f_1.0.1    </span></span>
<span><span class="co">##  [3] BSgenome.Oidioi.OIST.KUM.M3.7f_1.0.1   </span></span>
<span><span class="co">##  [4] BSgenome.Oidioi.OIST.Bar2.p4_1.0.1     </span></span>
<span><span class="co">##  [5] BSgenome.Oidioi.OIST.OSKA2016v1.9_1.0.0</span></span>
<span><span class="co">##  [6] BSgenome.Oidioi.OIST.OKI2018.I69_1.0.1 </span></span>
<span><span class="co">##  [7] viridis_0.6.4                          </span></span>
<span><span class="co">##  [8] viridisLite_0.4.2                      </span></span>
<span><span class="co">##  [9] purrr_1.0.2                            </span></span>
<span><span class="co">## [10] pheatmap_1.0.12                        </span></span>
<span><span class="co">## [11] dplyr_1.1.3                            </span></span>
<span><span class="co">## [12] OikScrambling_5.0.0                    </span></span>
<span><span class="co">## [13] ggplot2_3.4.3                          </span></span>
<span><span class="co">## [14] GenomicBreaks_0.14.2                   </span></span>
<span><span class="co">## [15] BSgenome_1.68.0                        </span></span>
<span><span class="co">## [16] rtracklayer_1.60.0                     </span></span>
<span><span class="co">## [17] Biostrings_2.68.1                      </span></span>
<span><span class="co">## [18] XVector_0.40.0                         </span></span>
<span><span class="co">## [19] GenomicRanges_1.52.0                   </span></span>
<span><span class="co">## [20] GenomeInfoDb_1.36.1                    </span></span>
<span><span class="co">## [21] IRanges_2.34.1                         </span></span>
<span><span class="co">## [22] S4Vectors_0.38.1                       </span></span>
<span><span class="co">## [23] BiocGenerics_0.46.0                    </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## loaded via a namespace (and not attached):</span></span>
<span><span class="co">##   [1] splines_4.3.1               BiocIO_1.10.0              </span></span>
<span><span class="co">##   [3] bitops_1.0-7                tibble_3.2.1               </span></span>
<span><span class="co">##   [5] R.oo_1.25.0                 XML_3.99-0.14              </span></span>
<span><span class="co">##   [7] rpart_4.1.19                lifecycle_1.0.3            </span></span>
<span><span class="co">##   [9] rprojroot_2.0.3             lattice_0.20-45            </span></span>
<span><span class="co">##  [11] MASS_7.3-58.2               backports_1.4.1            </span></span>
<span><span class="co">##  [13] magrittr_2.0.3              Hmisc_5.1-0                </span></span>
<span><span class="co">##  [15] sass_0.4.7                  rmarkdown_2.23             </span></span>
<span><span class="co">##  [17] jquerylib_0.1.4             yaml_2.3.7                 </span></span>
<span><span class="co">##  [19] plotrix_3.8-2               DBI_1.1.3                  </span></span>
<span><span class="co">##  [21] CNEr_1.36.0                 minqa_1.2.5                </span></span>
<span><span class="co">##  [23] RColorBrewer_1.1-3          ade4_1.7-22                </span></span>
<span><span class="co">##  [25] abind_1.4-5                 zlibbioc_1.46.0            </span></span>
<span><span class="co">##  [27] R.utils_2.12.2              RCurl_1.98-1.12            </span></span>
<span><span class="co">##  [29] nnet_7.3-18                 pracma_2.4.2               </span></span>
<span><span class="co">##  [31] GenomeInfoDbData_1.2.10     gdata_2.19.0               </span></span>
<span><span class="co">##  [33] annotate_1.78.0             pkgdown_2.0.7              </span></span>
<span><span class="co">##  [35] codetools_0.2-19            DelayedArray_0.26.7        </span></span>
<span><span class="co">##  [37] tidyselect_1.2.0            shape_1.4.6                </span></span>
<span><span class="co">##  [39] lme4_1.1-34                 matrixStats_1.0.0          </span></span>
<span><span class="co">##  [41] base64enc_0.1-3             GenomicAlignments_1.36.0   </span></span>
<span><span class="co">##  [43] jsonlite_1.8.7              mitml_0.4-5                </span></span>
<span><span class="co">##  [45] Formula_1.2-5               survival_3.5-3             </span></span>
<span><span class="co">##  [47] iterators_1.0.14            systemfonts_1.0.5          </span></span>
<span><span class="co">##  [49] foreach_1.5.2               tools_4.3.1                </span></span>
<span><span class="co">##  [51] ragg_1.2.5                  Rcpp_1.0.11                </span></span>
<span><span class="co">##  [53] glue_1.6.2                  gridExtra_2.3              </span></span>
<span><span class="co">##  [55] pan_1.8                     xfun_0.40                  </span></span>
<span><span class="co">##  [57] MatrixGenerics_1.12.2       EBImage_4.42.0             </span></span>
<span><span class="co">##  [59] withr_2.5.1                 fastmap_1.1.1              </span></span>
<span><span class="co">##  [61] boot_1.3-28.1               fansi_1.0.5                </span></span>
<span><span class="co">##  [63] digest_0.6.33               R6_2.5.1                   </span></span>
<span><span class="co">##  [65] mice_3.16.0                 textshaping_0.3.7          </span></span>
<span><span class="co">##  [67] colorspace_2.1-0            GO.db_3.17.0               </span></span>
<span><span class="co">##  [69] gtools_3.9.4                poweRlaw_0.70.6            </span></span>
<span><span class="co">##  [71] jpeg_0.1-10                 RSQLite_2.3.1              </span></span>
<span><span class="co">##  [73] weights_1.0.4               R.methodsS3_1.8.2          </span></span>
<span><span class="co">##  [75] utf8_1.2.3                  tidyr_1.3.0                </span></span>
<span><span class="co">##  [77] generics_0.1.3              data.table_1.14.8          </span></span>
<span><span class="co">##  [79] httr_1.4.7                  htmlwidgets_1.6.2          </span></span>
<span><span class="co">##  [81] S4Arrays_1.0.5              pkgconfig_2.0.3            </span></span>
<span><span class="co">##  [83] gtable_0.3.4                blob_1.2.4                 </span></span>
<span><span class="co">##  [85] htmltools_0.5.6.1           fftwtools_0.9-11           </span></span>
<span><span class="co">##  [87] scales_1.2.1                Biobase_2.60.0             </span></span>
<span><span class="co">##  [89] png_0.1-8                   knitr_1.44                 </span></span>
<span><span class="co">##  [91] heatmaps_1.24.0             rstudioapi_0.15.0          </span></span>
<span><span class="co">##  [93] tzdb_0.4.0                  reshape2_1.4.4             </span></span>
<span><span class="co">##  [95] rjson_0.2.21                checkmate_2.2.0            </span></span>
<span><span class="co">##  [97] nlme_3.1-162                nloptr_2.0.3               </span></span>
<span><span class="co">##  [99] cachem_1.0.8                stringr_1.5.0              </span></span>
<span><span class="co">## [101] KernSmooth_2.23-20          parallel_4.3.1             </span></span>
<span><span class="co">## [103] genoPlotR_0.8.11            foreign_0.8-84             </span></span>
<span><span class="co">## [105] AnnotationDbi_1.62.2        restfulr_0.0.15            </span></span>
<span><span class="co">## [107] desc_1.4.2                  pillar_1.9.0               </span></span>
<span><span class="co">## [109] grid_4.3.1                  vctrs_0.6.3                </span></span>
<span><span class="co">## [111] jomo_2.7-6                  xtable_1.8-4               </span></span>
<span><span class="co">## [113] cluster_2.1.4               htmlTable_2.4.1            </span></span>
<span><span class="co">## [115] evaluate_0.22               readr_2.1.4                </span></span>
<span><span class="co">## [117] cli_3.6.1                   locfit_1.5-9.8             </span></span>
<span><span class="co">## [119] compiler_4.3.1              Rsamtools_2.16.0           </span></span>
<span><span class="co">## [121] rlang_1.1.1                 crayon_1.5.2               </span></span>
<span><span class="co">## [123] plyr_1.8.8                  fs_1.6.3                   </span></span>
<span><span class="co">## [125] stringi_1.7.12              BiocParallel_1.34.2        </span></span>
<span><span class="co">## [127] munsell_0.5.0               tiff_0.1-11                </span></span>
<span><span class="co">## [129] glmnet_4.1-7                Matrix_1.5-3               </span></span>
<span><span class="co">## [131] hms_1.1.3                   bit64_4.0.5                </span></span>
<span><span class="co">## [133] KEGGREST_1.40.0             SummarizedExperiment_1.30.2</span></span>
<span><span class="co">## [135] broom_1.0.5                 memoise_2.0.1              </span></span>
<span><span class="co">## [137] bslib_0.5.1                 bit_4.0.5</span></span></code></pre>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Aleksandra Bliznina, Michael Mansfield, Charles Plessy.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
