<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Distance Measures • OikScrambling</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Distance Measures">
<meta property="og:description" content="OikScrambling">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">OikScrambling</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">4.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/OikScrambling.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/CNEs.html">Conserved non-coding elements</a>
    </li>
    <li>
      <a href="../articles/ChromosomePlots.html">Chromosome-scale feature density plots</a>
    </li>
    <li>
      <a href="../articles/CoalescingCutoffs.html">Modifying coalescent cutoffs</a>
    </li>
    <li>
      <a href="../articles/ColinearityInterruptors.html">Regions interrupting colinearity</a>
    </li>
    <li>
      <a href="../articles/DistanceMeasures.html">Distance Measures</a>
    </li>
    <li>
      <a href="../articles/GenomicFeatures.html">Genomic features of the breakpoints</a>
    </li>
    <li>
      <a href="../articles/Inversions.html">Inversions</a>
    </li>
    <li>
      <a href="../articles/LoadGenomicBreaks.html">Load genomic break data</a>
    </li>
    <li>
      <a href="../articles/OrthogroupSyntenies.html">Orthogroup syntenies</a>
    </li>
    <li>
      <a href="../articles/OxfordPlots.html">Whole-genome dot-plots</a>
    </li>
    <li>
      <a href="../articles/PAC3.html">Region of interest: PAC3</a>
    </li>
    <li>
      <a href="../articles/PWM.html">Motif search and analysis</a>
    </li>
    <li>
      <a href="../articles/ParallelPlots.html">Whole-chromosome parallel plots</a>
    </li>
    <li>
      <a href="../articles/RegionWidths.html">Width of the aligned and mapped regions.</a>
    </li>
    <li>
      <a href="../articles/RegionsOfInterest.html">Regions of interest</a>
    </li>
    <li>
      <a href="../articles/RepeatRegions.html">Repeat regions</a>
    </li>
    <li>
      <a href="../articles/ScalingBreakpoints.html">Scaling breakpoints</a>
    </li>
    <li>
      <a href="../articles/SegmentMovements.html">Segment movement distances</a>
    </li>
    <li>
      <a href="../articles/TFBS.html">Transcription factor binding sites</a>
    </li>
    <li>
      <a href="../articles/TandemSiteDuplications.html">Tandem site duplications</a>
    </li>
    <li>
      <a href="../articles/Translocations.html">Translocations</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/oist/LuscombeU_OikScrambling/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="DistanceMeasures_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Distance Measures</h1>
                        <h4 data-toc-skip class="author">Charles Plessy</h4>
            
            <h4 data-toc-skip class="date">06 January, 2023</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/oist/LuscombeU_OikScrambling/blob/HEAD/vignettes/DistanceMeasures.Rmd" class="external-link"><code>vignettes/DistanceMeasures.Rmd</code></a></small>
      <div class="hidden name"><code>DistanceMeasures.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="purpose">Purpose<a class="anchor" aria-label="anchor" href="#purpose"></a>
</h2>
<p>We want to put numbers on visual impressions such as:</p>
<ul>
<li>There is a lot of scrambling.</li>
<li>Aligned regions stay on their chromosome arm.</li>
<li>Aligned regions are more “off the diagonal” on short arms than on long arms.</li>
<li>…</li>
</ul>
</div>
<div class="section level2">
<h2 id="load-r-pacakges-and-data">Load R pacakges and data<a class="anchor" aria-label="anchor" href="#load-r-pacakges-and-data"></a>
</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/oist/LuscombeU_OikScrambling" class="external-link">'OikScrambling'</a></span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/load.html" class="external-link">load</a></span><span class="op">(</span><span class="st">"BreakPoints.Rdata"</span><span class="op">)</span></code></pre></div>
<p>See <code><a href="../articles/LoadGenomicBreaks.html">vignette("LoadGenomicBreaks", package = "OikScrambling")</a></code> for how the different GBreaks objects are prepared.</p>
</div>
<div class="section level2">
<h2 id="declare-service-functions">Declare service functions<a class="anchor" aria-label="anchor" href="#declare-service-functions"></a>
</h2>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">pairs2table6x3</span> <span class="op">&lt;-</span> <span class="kw">function</span> <span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="op">{</span>
  <span class="va">x</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/as.list.html" class="external-link">as.list</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span>
  <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/cbind.html" class="external-link">cbind</a></span><span class="op">(</span>
    Oki <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span>Oki <span class="op">=</span> <span class="cn">NA</span>,        Osa <span class="op">=</span> <span class="va">x</span><span class="op">$</span><span class="va">Osa_Oki</span>, Bar <span class="op">=</span> <span class="va">x</span><span class="op">$</span><span class="va">Bar_Oki</span><span class="op">)</span>,
    Kum <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span>Oki <span class="op">=</span> <span class="va">x</span><span class="op">$</span><span class="va">Oki_Kum</span>, Osa <span class="op">=</span> <span class="va">x</span><span class="op">$</span><span class="va">Osa_Kum</span>, Bar <span class="op">=</span> <span class="va">x</span><span class="op">$</span><span class="va">Bar_Kum</span><span class="op">)</span>,
    Osa <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span>Oki <span class="op">=</span> <span class="va">x</span><span class="op">$</span><span class="va">Oki_Osa</span>, Osa <span class="op">=</span> <span class="cn">NA</span>,        Bar <span class="op">=</span> <span class="va">x</span><span class="op">$</span><span class="va">Bar_Osa</span><span class="op">)</span>,
    Aom <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span>Oki <span class="op">=</span> <span class="va">x</span><span class="op">$</span><span class="va">Oki_Aom</span>, Osa <span class="op">=</span> <span class="va">x</span><span class="op">$</span><span class="va">Osa_Aom</span>, Bar <span class="op">=</span> <span class="va">x</span><span class="op">$</span><span class="va">Bar_Aom</span><span class="op">)</span>,
    Bar <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span>Oki <span class="op">=</span> <span class="va">x</span><span class="op">$</span><span class="va">Oki_Bar</span>, Osa <span class="op">=</span> <span class="va">x</span><span class="op">$</span><span class="va">Osa_Bar</span>, Bar <span class="op">=</span> <span class="cn">NA</span>       <span class="op">)</span>,
    Nor <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span>Oki <span class="op">=</span> <span class="va">x</span><span class="op">$</span><span class="va">Oki_Nor</span>, Osa <span class="op">=</span> <span class="va">x</span><span class="op">$</span><span class="va">Osa_Nor</span>, Bar <span class="op">=</span> <span class="va">x</span><span class="op">$</span><span class="va">Bar_Nor</span><span class="op">)</span>
  <span class="op">)</span>
<span class="op">}</span>

<span class="va">pairs2Kable6x3</span> <span class="op">&lt;-</span> <span class="kw">function</span> <span class="op">(</span><span class="va">df</span><span class="op">)</span> <span class="va">df</span> |&gt; <span class="fu">pairs2table6x3</span><span class="op">(</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/r/base/Round.html" class="external-link">round</a></span><span class="op">(</span><span class="fl">2</span><span class="op">)</span> |&gt; <span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="synteny-index">Synteny index<a class="anchor" aria-label="anchor" href="#synteny-index"></a>
</h2>
<p>Ad-hoc index measuring to what extent a scaffold of the <em>target</em> genome is mostly aligned to a single scaffold in the <em>query</em> genome. See <code><a href="https://oist.github.io/GenomicBreaks/reference/synteny_index.html" class="external-link">?GenomicBreaks::synteny_index</a></code> for details. One limitation to the use of this index is that it requires that at least the <em>query</em> genome is a complete chromosome assembly.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">synteny_index</span></code></pre></div>
<pre><code><span class="co">## function (gb) </span>
<span class="co">## {</span>
<span class="co">##     gbl &lt;- split(gb, droplevels(seqnames(gb)))</span>
<span class="co">##     synIdx &lt;- sapply(gbl, function(x) {</span>
<span class="co">##         alnOnMainHit &lt;- tail(sort(sum(width(split(x, seqnames(x$query))))), </span>
<span class="co">##             1)</span>
<span class="co">##         alnOnMainHit/sum(width(x))</span>
<span class="co">##     })</span>
<span class="co">##     weighted.mean(synIdx, sum(width(gbl)))</span>
<span class="co">## }</span>
<span class="co">## &lt;bytecode: 0x562feb8b6730&gt;</span>
<span class="co">## &lt;environment: namespace:GenomicBreaks&gt;</span></code></pre>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">BiocParallel</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocParallel/man/bplapply.html" class="external-link">bplapply</a></span><span class="op">(</span><span class="va">gbs</span>,                 <span class="va">synteny_index</span>  <span class="op">)</span> |&gt; <span class="fu">pairs2Kable6x3</span><span class="op">(</span><span class="op">)</span></code></pre></div>
<table class="table">
<thead><tr class="header">
<th align="left"></th>
<th align="right">Oki</th>
<th align="right">Kum</th>
<th align="right">Osa</th>
<th align="right">Aom</th>
<th align="right">Bar</th>
<th align="right">Nor</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">Oki</td>
<td align="right">NA</td>
<td align="right">0.51</td>
<td align="right">0.94</td>
<td align="right">0.65</td>
<td align="right">0.94</td>
<td align="right">0.17</td>
</tr>
<tr class="even">
<td align="left">Osa</td>
<td align="right">0.94</td>
<td align="right">0.52</td>
<td align="right">NA</td>
<td align="right">0.61</td>
<td align="right">0.97</td>
<td align="right">0.17</td>
</tr>
<tr class="odd">
<td align="left">Bar</td>
<td align="right">0.94</td>
<td align="right">0.53</td>
<td align="right">0.97</td>
<td align="right">0.66</td>
<td align="right">NA</td>
<td align="right">0.16</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">BiocParallel</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocParallel/man/bplapply.html" class="external-link">bplapply</a></span><span class="op">(</span><span class="va">gbs</span>, \<span class="op">(</span><span class="va">g</span><span class="op">)</span> <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/swap.html" class="external-link">swap</a></span><span class="op">(</span><span class="va">g</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/synteny_index.html" class="external-link">synteny_index</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span> |&gt; <span class="fu">pairs2Kable6x3</span><span class="op">(</span><span class="op">)</span></code></pre></div>
<table class="table">
<thead><tr class="header">
<th align="left"></th>
<th align="right">Oki</th>
<th align="right">Kum</th>
<th align="right">Osa</th>
<th align="right">Aom</th>
<th align="right">Bar</th>
<th align="right">Nor</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">Oki</td>
<td align="right">NA</td>
<td align="right">0.97</td>
<td align="right">0.94</td>
<td align="right">0.94</td>
<td align="right">0.94</td>
<td align="right">0.93</td>
</tr>
<tr class="even">
<td align="left">Osa</td>
<td align="right">0.94</td>
<td align="right">0.94</td>
<td align="right">NA</td>
<td align="right">0.97</td>
<td align="right">0.97</td>
<td align="right">0.96</td>
</tr>
<tr class="odd">
<td align="left">Bar</td>
<td align="right">0.94</td>
<td align="right">0.94</td>
<td align="right">0.97</td>
<td align="right">0.97</td>
<td align="right">NA</td>
<td align="right">0.95</td>
</tr>
</tbody>
</table>
</div>
<div class="section level2">
<h2 id="correlation">Correlation<a class="anchor" aria-label="anchor" href="#correlation"></a>
</h2>
<p>Ad-hoc index measuring the correlation of the coordinates of the syntenic alignments in scaffolds of a <em>target</em> genome and their best match in the <em>query</em> genome. See <code><a href="https://oist.github.io/GenomicBreaks/reference/correlation_index.html" class="external-link">?GenomicBreaks::correlation_index</a></code> for details. It is intended to be more robust to the presence of uncollapsed haplotypes in the <em>query</em> genome but probably needs further testing.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">correlation_index</span></code></pre></div>
<pre><code><span class="co">## function (gb) </span>
<span class="co">## {</span>
<span class="co">##     gbl &lt;- split(gb, droplevels(seqnames(gb)))</span>
<span class="co">##     corIdx &lt;- sapply(gbl, function(x) {</span>
<span class="co">##         bestQuery &lt;- names(tail(sort(table(seqnames(x$query))), </span>
<span class="co">##             1))</span>
<span class="co">##         x &lt;- x[seqnames(x$query) == bestQuery]</span>
<span class="co">##         if (length(x) == 1) </span>
<span class="co">##             return(1)</span>
<span class="co">##         weights &lt;- width(x)</span>
<span class="co">##         x &lt;- resize(x, fix = "center", width = 1)</span>
<span class="co">##         x$query &lt;- resize(x$query, fix = "center", width = 1)</span>
<span class="co">##         weights::wtd.cor(start(x), start(x$query), weights)[1]</span>
<span class="co">##     })</span>
<span class="co">##     weighted.mean(corIdx, sum(width(gbl)))</span>
<span class="co">## }</span>
<span class="co">## &lt;bytecode: 0x562fe83efa78&gt;</span>
<span class="co">## &lt;environment: namespace:GenomicBreaks&gt;</span></code></pre>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">BiocParallel</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocParallel/man/bplapply.html" class="external-link">bplapply</a></span><span class="op">(</span><span class="va">gbs</span>,                 <span class="va">correlation_index</span>  <span class="op">)</span> |&gt; <span class="fu">pairs2Kable6x3</span><span class="op">(</span><span class="op">)</span></code></pre></div>
<table class="table">
<thead><tr class="header">
<th align="left"></th>
<th align="right">Oki</th>
<th align="right">Kum</th>
<th align="right">Osa</th>
<th align="right">Aom</th>
<th align="right">Bar</th>
<th align="right">Nor</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">Oki</td>
<td align="right">NA</td>
<td align="right">0.89</td>
<td align="right">0.48</td>
<td align="right">-0.03</td>
<td align="right">0.44</td>
<td align="right">-0.19</td>
</tr>
<tr class="even">
<td align="left">Osa</td>
<td align="right">0.48</td>
<td align="right">-0.06</td>
<td align="right">NA</td>
<td align="right">-0.43</td>
<td align="right">0.37</td>
<td align="right">0.30</td>
</tr>
<tr class="odd">
<td align="left">Bar</td>
<td align="right">0.44</td>
<td align="right">-0.01</td>
<td align="right">0.36</td>
<td align="right">0.07</td>
<td align="right">NA</td>
<td align="right">0.36</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">BiocParallel</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocParallel/man/bplapply.html" class="external-link">bplapply</a></span><span class="op">(</span><span class="va">gbs</span>, \<span class="op">(</span><span class="va">g</span><span class="op">)</span> <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/swap.html" class="external-link">swap</a></span><span class="op">(</span><span class="va">g</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/correlation_index.html" class="external-link">correlation_index</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span> |&gt; <span class="fu">pairs2Kable6x3</span><span class="op">(</span><span class="op">)</span></code></pre></div>
<table class="table">
<thead><tr class="header">
<th align="left"></th>
<th align="right">Oki</th>
<th align="right">Kum</th>
<th align="right">Osa</th>
<th align="right">Aom</th>
<th align="right">Bar</th>
<th align="right">Nor</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">Oki</td>
<td align="right">NA</td>
<td align="right">0.51</td>
<td align="right">0.48</td>
<td align="right">-0.01</td>
<td align="right">0.44</td>
<td align="right">0.01</td>
</tr>
<tr class="even">
<td align="left">Osa</td>
<td align="right">0.48</td>
<td align="right">-0.01</td>
<td align="right">NA</td>
<td align="right">-0.22</td>
<td align="right">0.36</td>
<td align="right">0.01</td>
</tr>
<tr class="odd">
<td align="left">Bar</td>
<td align="right">0.44</td>
<td align="right">0.03</td>
<td align="right">0.36</td>
<td align="right">0.11</td>
<td align="right">NA</td>
<td align="right">0.20</td>
</tr>
</tbody>
</table>
</div>
<div class="section level2">
<h2 id="gene-order-conservation">Gene Order Conservation<a class="anchor" aria-label="anchor" href="#gene-order-conservation"></a>
</h2>
<p>See <code><a href="https://oist.github.io/GenomicBreaks/reference/GOC.html" class="external-link">?GenomicBreaks::GOC</a></code> for details.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">GOC</span></code></pre></div>
<pre><code><span class="co">## function (gb, vicinity = 4, debug = FALSE) </span>
<span class="co">## {</span>
<span class="co">##     if (length(gb) == 0) </span>
<span class="co">##         stop("Can not compute GOC on zero-length objects.")</span>
<span class="co">##     gb &lt;- sort(gb, ignore.strand = TRUE)</span>
<span class="co">##     gb &lt;- flagColinearAlignments(gb, details = TRUE)</span>
<span class="co">##     gb$GOCi &lt;- abs(gb$qfoll) &lt;= vicinity</span>
<span class="co">##     if (debug) </span>
<span class="co">##         return(gb)</span>
<span class="co">##     gb[is.na(gb$tfoll)] &lt;- NULL</span>
<span class="co">##     sum(gb$GOCi %in% TRUE)/length(gb)</span>
<span class="co">## }</span>
<span class="co">## &lt;bytecode: 0x562fe5b65578&gt;</span>
<span class="co">## &lt;environment: namespace:GenomicBreaks&gt;</span></code></pre>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">BiocParallel</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocParallel/man/bplapply.html" class="external-link">bplapply</a></span><span class="op">(</span><span class="va">gbs</span>,                 <span class="va">GOC</span>  <span class="op">)</span> |&gt; <span class="fu">pairs2Kable6x3</span><span class="op">(</span><span class="op">)</span></code></pre></div>
<table class="table">
<thead><tr class="header">
<th align="left"></th>
<th align="right">Oki</th>
<th align="right">Kum</th>
<th align="right">Osa</th>
<th align="right">Aom</th>
<th align="right">Bar</th>
<th align="right">Nor</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">Oki</td>
<td align="right">NA</td>
<td align="right">0.76</td>
<td align="right">0.84</td>
<td align="right">0.85</td>
<td align="right">0.85</td>
<td align="right">0.82</td>
</tr>
<tr class="even">
<td align="left">Osa</td>
<td align="right">0.84</td>
<td align="right">0.84</td>
<td align="right">NA</td>
<td align="right">0.78</td>
<td align="right">0.89</td>
<td align="right">0.85</td>
</tr>
<tr class="odd">
<td align="left">Bar</td>
<td align="right">0.85</td>
<td align="right">0.85</td>
<td align="right">0.89</td>
<td align="right">0.89</td>
<td align="right">NA</td>
<td align="right">0.70</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">BiocParallel</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocParallel/man/bplapply.html" class="external-link">bplapply</a></span><span class="op">(</span><span class="va">gbs</span>, \<span class="op">(</span><span class="va">g</span><span class="op">)</span> <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/swap.html" class="external-link">swap</a></span><span class="op">(</span><span class="va">g</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/GOC.html" class="external-link">GOC</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span> |&gt; <span class="fu">pairs2Kable6x3</span><span class="op">(</span><span class="op">)</span></code></pre></div>
<table class="table">
<thead><tr class="header">
<th align="left"></th>
<th align="right">Oki</th>
<th align="right">Kum</th>
<th align="right">Osa</th>
<th align="right">Aom</th>
<th align="right">Bar</th>
<th align="right">Nor</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">Oki</td>
<td align="right">NA</td>
<td align="right">0.76</td>
<td align="right">0.84</td>
<td align="right">0.84</td>
<td align="right">0.84</td>
<td align="right">0.84</td>
</tr>
<tr class="even">
<td align="left">Osa</td>
<td align="right">0.84</td>
<td align="right">0.84</td>
<td align="right">NA</td>
<td align="right">0.77</td>
<td align="right">0.89</td>
<td align="right">0.88</td>
</tr>
<tr class="odd">
<td align="left">Bar</td>
<td align="right">0.85</td>
<td align="right">0.85</td>
<td align="right">0.89</td>
<td align="right">0.89</td>
<td align="right">NA</td>
<td align="right">0.73</td>
</tr>
</tbody>
</table>
</div>
<div class="section level2">
<h2 id="strand-randomisation-index">Strand randomisation index<a class="anchor" aria-label="anchor" href="#strand-randomisation-index"></a>
</h2>
<p>See <code><a href="https://oist.github.io/GenomicBreaks/reference/strand_randomisation_index.html" class="external-link">?GenomicBreaks::strand_randomisation_index</a></code> for details.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">strand_randomisation_index</span></code></pre></div>
<pre><code><span class="co">## function (gb) </span>
<span class="co">## {</span>
<span class="co">##     gbl &lt;- split(gb, droplevels(seqnames(gb)))</span>
<span class="co">##     idx &lt;- sapply(gbl, function(x) {</span>
<span class="co">##         onPlus &lt;- sum(width(x[strand(x) == "+"]))</span>
<span class="co">##         onMinus &lt;- sum(width(x[strand(x) == "-"]))</span>
<span class="co">##         abs((onPlus - onMinus)/(onPlus + onMinus))</span>
<span class="co">##     })</span>
<span class="co">##     weighted.mean(idx, sum(width(gbl)))</span>
<span class="co">## }</span>
<span class="co">## &lt;bytecode: 0x562fe0b3a1e8&gt;</span>
<span class="co">## &lt;environment: namespace:GenomicBreaks&gt;</span></code></pre>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">BiocParallel</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocParallel/man/bplapply.html" class="external-link">bplapply</a></span><span class="op">(</span><span class="va">gbs</span>,                 <span class="va">strand_randomisation_index</span>  <span class="op">)</span> |&gt; <span class="fu">pairs2Kable6x3</span><span class="op">(</span><span class="op">)</span></code></pre></div>
<table class="table">
<thead><tr class="header">
<th align="left"></th>
<th align="right">Oki</th>
<th align="right">Kum</th>
<th align="right">Osa</th>
<th align="right">Aom</th>
<th align="right">Bar</th>
<th align="right">Nor</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">Oki</td>
<td align="right">NA</td>
<td align="right">0.52</td>
<td align="right">0.04</td>
<td align="right">0.06</td>
<td align="right">0.10</td>
<td align="right">0.05</td>
</tr>
<tr class="even">
<td align="left">Osa</td>
<td align="right">0.05</td>
<td align="right">0.06</td>
<td align="right">NA</td>
<td align="right">0.43</td>
<td align="right">0.23</td>
<td align="right">0.10</td>
</tr>
<tr class="odd">
<td align="left">Bar</td>
<td align="right">0.10</td>
<td align="right">0.09</td>
<td align="right">0.23</td>
<td align="right">0.19</td>
<td align="right">NA</td>
<td align="right">0.18</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">BiocParallel</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocParallel/man/bplapply.html" class="external-link">bplapply</a></span><span class="op">(</span><span class="va">gbs</span>, \<span class="op">(</span><span class="va">g</span><span class="op">)</span> <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/swap.html" class="external-link">swap</a></span><span class="op">(</span><span class="va">g</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/strand_randomisation_index.html" class="external-link">strand_randomisation_index</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span> |&gt; <span class="fu">pairs2Kable6x3</span><span class="op">(</span><span class="op">)</span></code></pre></div>
<table class="table">
<thead><tr class="header">
<th align="left"></th>
<th align="right">Oki</th>
<th align="right">Kum</th>
<th align="right">Osa</th>
<th align="right">Aom</th>
<th align="right">Bar</th>
<th align="right">Nor</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">Oki</td>
<td align="right">NA</td>
<td align="right">0.83</td>
<td align="right">0.05</td>
<td align="right">0.10</td>
<td align="right">0.10</td>
<td align="right">0.31</td>
</tr>
<tr class="even">
<td align="left">Osa</td>
<td align="right">0.04</td>
<td align="right">0.06</td>
<td align="right">NA</td>
<td align="right">0.66</td>
<td align="right">0.23</td>
<td align="right">0.47</td>
</tr>
<tr class="odd">
<td align="left">Bar</td>
<td align="right">0.10</td>
<td align="right">0.14</td>
<td align="right">0.23</td>
<td align="right">0.25</td>
<td align="right">NA</td>
<td align="right">0.82</td>
</tr>
</tbody>
</table>
</div>
<div class="section level2">
<h2 id="strand-proportion-index">Strand proportion index<a class="anchor" aria-label="anchor" href="#strand-proportion-index"></a>
</h2>
<p>Attempts to use pure proportions lead to scores drifting towards 0.5 when comparing fragmented assemblies, as the fragments are not oriented.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">strand_proportion_index</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">gb</span><span class="op">)</span> <span class="op">{</span>
  <span class="va">gbl</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/splitAsList.html" class="external-link">split</a></span><span class="op">(</span><span class="va">gb</span>, <span class="fu"><a href="https://rdrr.io/r/base/droplevels.html" class="external-link">droplevels</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqnames</a></span><span class="op">(</span><span class="va">gb</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
  <span class="co"># Calculate an index for each sequence feature</span>
  <span class="va">idx</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">gbl</span>, \<span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="op">{</span>
    <span class="va">onPlus</span>  <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/start.html" class="external-link">width</a></span><span class="op">(</span><span class="va">x</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/strand.html" class="external-link">strand</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="op">==</span> <span class="st">'+'</span><span class="op">]</span><span class="op">)</span><span class="op">)</span>
    <span class="va">onMinus</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/start.html" class="external-link">width</a></span><span class="op">(</span><span class="va">x</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/strand.html" class="external-link">strand</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="op">==</span> <span class="st">'-'</span><span class="op">]</span><span class="op">)</span><span class="op">)</span>
    <span class="va">onPlus</span> <span class="op">/</span> <span class="op">(</span><span class="va">onPlus</span> <span class="op">+</span> <span class="va">onMinus</span><span class="op">)</span>
  <span class="op">}</span><span class="op">)</span>
  <span class="co"># Average by the sum of all widths</span>
  <span class="fu"><a href="https://rdrr.io/r/stats/weighted.mean.html" class="external-link">weighted.mean</a></span><span class="op">(</span><span class="va">idx</span>, <span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/start.html" class="external-link">width</a></span><span class="op">(</span><span class="va">gbl</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
<span class="op">}</span></code></pre></div>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">BiocParallel</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocParallel/man/bplapply.html" class="external-link">bplapply</a></span><span class="op">(</span><span class="va">gbs</span>,                 <span class="va">strand_proportion_index</span>  <span class="op">)</span> |&gt; <span class="fu">pairs2Kable6x3</span><span class="op">(</span><span class="op">)</span></code></pre></div>
<table class="table">
<thead><tr class="header">
<th align="left"></th>
<th align="right">Oki</th>
<th align="right">Kum</th>
<th align="right">Osa</th>
<th align="right">Aom</th>
<th align="right">Bar</th>
<th align="right">Nor</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">Oki</td>
<td align="right">NA</td>
<td align="right">0.73</td>
<td align="right">0.49</td>
<td align="right">0.47</td>
<td align="right">0.47</td>
<td align="right">0.49</td>
</tr>
<tr class="even">
<td align="left">Osa</td>
<td align="right">0.49</td>
<td align="right">0.50</td>
<td align="right">NA</td>
<td align="right">0.41</td>
<td align="right">0.38</td>
<td align="right">0.46</td>
</tr>
<tr class="odd">
<td align="left">Bar</td>
<td align="right">0.47</td>
<td align="right">0.46</td>
<td align="right">0.38</td>
<td align="right">0.54</td>
<td align="right">NA</td>
<td align="right">0.59</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">BiocParallel</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocParallel/man/bplapply.html" class="external-link">bplapply</a></span><span class="op">(</span><span class="va">gbs</span>, \<span class="op">(</span><span class="va">g</span><span class="op">)</span> <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/swap.html" class="external-link">swap</a></span><span class="op">(</span><span class="va">g</span><span class="op">)</span> |&gt; <span class="fu">strand_proportion_index</span><span class="op">(</span><span class="op">)</span><span class="op">)</span> |&gt; <span class="fu">pairs2Kable6x3</span><span class="op">(</span><span class="op">)</span></code></pre></div>
<table class="table">
<thead><tr class="header">
<th align="left"></th>
<th align="right">Oki</th>
<th align="right">Kum</th>
<th align="right">Osa</th>
<th align="right">Aom</th>
<th align="right">Bar</th>
<th align="right">Nor</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">Oki</td>
<td align="right">NA</td>
<td align="right">0.73</td>
<td align="right">0.49</td>
<td align="right">0.47</td>
<td align="right">0.47</td>
<td align="right">0.49</td>
</tr>
<tr class="even">
<td align="left">Osa</td>
<td align="right">0.49</td>
<td align="right">0.50</td>
<td align="right">NA</td>
<td align="right">0.41</td>
<td align="right">0.38</td>
<td align="right">0.46</td>
</tr>
<tr class="odd">
<td align="left">Bar</td>
<td align="right">0.47</td>
<td align="right">0.46</td>
<td align="right">0.38</td>
<td align="right">0.54</td>
<td align="right">NA</td>
<td align="right">0.59</td>
</tr>
</tbody>
</table>
</div>
<div class="section level2">
<h2 id="strand-proportion-majority-index">Strand proportion majority index<a class="anchor" aria-label="anchor" href="#strand-proportion-majority-index"></a>
</h2>
<p>One possible solution to the problem above could be to report the highest proportion, so that the value returned is not sensitive to orientation. But it scales the index between 0.5 and 1, which may be counter-intuitive.</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">strand_proportion_majority_index</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">gb</span><span class="op">)</span> <span class="op">{</span>
  <span class="va">gbl</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/splitAsList.html" class="external-link">split</a></span><span class="op">(</span><span class="va">gb</span>, <span class="fu"><a href="https://rdrr.io/r/base/droplevels.html" class="external-link">droplevels</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqnames</a></span><span class="op">(</span><span class="va">gb</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
  <span class="co"># Calculate an index for each sequence feature</span>
  <span class="va">idx</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">gbl</span>, \<span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="op">{</span>
    <span class="va">onPlus</span>  <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/start.html" class="external-link">width</a></span><span class="op">(</span><span class="va">x</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/strand.html" class="external-link">strand</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="op">==</span> <span class="st">'+'</span><span class="op">]</span><span class="op">)</span><span class="op">)</span>
    <span class="va">onMinus</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/start.html" class="external-link">width</a></span><span class="op">(</span><span class="va">x</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/strand.html" class="external-link">strand</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="op">==</span> <span class="st">'-'</span><span class="op">]</span><span class="op">)</span><span class="op">)</span>
    <span class="va">p</span> <span class="op">&lt;-</span> <span class="va">onPlus</span> <span class="op">/</span> <span class="op">(</span><span class="va">onPlus</span> <span class="op">+</span> <span class="va">onMinus</span><span class="op">)</span>
    <span class="kw">if</span><span class="op">(</span><span class="va">p</span> <span class="op">&lt;</span> <span class="fl">0.5</span><span class="op">)</span> <span class="va">p</span> <span class="op">&lt;-</span> <span class="fl">1</span> <span class="op">-</span> <span class="va">p</span>
    <span class="va">p</span>
  <span class="op">}</span><span class="op">)</span>
  <span class="co"># Average by the sum of all widths</span>
  <span class="fu"><a href="https://rdrr.io/r/stats/weighted.mean.html" class="external-link">weighted.mean</a></span><span class="op">(</span><span class="va">idx</span>, <span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/start.html" class="external-link">width</a></span><span class="op">(</span><span class="va">gbl</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
<span class="op">}</span></code></pre></div>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">BiocParallel</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocParallel/man/bplapply.html" class="external-link">bplapply</a></span><span class="op">(</span><span class="va">gbs</span>,                 <span class="va">strand_proportion_majority_index</span>  <span class="op">)</span> |&gt; <span class="fu">pairs2Kable6x3</span><span class="op">(</span><span class="op">)</span></code></pre></div>
<table class="table">
<thead><tr class="header">
<th align="left"></th>
<th align="right">Oki</th>
<th align="right">Kum</th>
<th align="right">Osa</th>
<th align="right">Aom</th>
<th align="right">Bar</th>
<th align="right">Nor</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">Oki</td>
<td align="right">NA</td>
<td align="right">0.76</td>
<td align="right">0.52</td>
<td align="right">0.53</td>
<td align="right">0.55</td>
<td align="right">0.53</td>
</tr>
<tr class="even">
<td align="left">Osa</td>
<td align="right">0.52</td>
<td align="right">0.53</td>
<td align="right">NA</td>
<td align="right">0.72</td>
<td align="right">0.62</td>
<td align="right">0.55</td>
</tr>
<tr class="odd">
<td align="left">Bar</td>
<td align="right">0.55</td>
<td align="right">0.54</td>
<td align="right">0.62</td>
<td align="right">0.59</td>
<td align="right">NA</td>
<td align="right">0.59</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">BiocParallel</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocParallel/man/bplapply.html" class="external-link">bplapply</a></span><span class="op">(</span><span class="va">gbs</span>, \<span class="op">(</span><span class="va">g</span><span class="op">)</span> <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/swap.html" class="external-link">swap</a></span><span class="op">(</span><span class="va">g</span><span class="op">)</span> |&gt; <span class="fu">strand_proportion_majority_index</span><span class="op">(</span><span class="op">)</span><span class="op">)</span> |&gt; <span class="fu">pairs2Kable6x3</span><span class="op">(</span><span class="op">)</span></code></pre></div>
<table class="table">
<thead><tr class="header">
<th align="left"></th>
<th align="right">Oki</th>
<th align="right">Kum</th>
<th align="right">Osa</th>
<th align="right">Aom</th>
<th align="right">Bar</th>
<th align="right">Nor</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">Oki</td>
<td align="right">NA</td>
<td align="right">0.91</td>
<td align="right">0.52</td>
<td align="right">0.55</td>
<td align="right">0.55</td>
<td align="right">0.66</td>
</tr>
<tr class="even">
<td align="left">Osa</td>
<td align="right">0.52</td>
<td align="right">0.53</td>
<td align="right">NA</td>
<td align="right">0.83</td>
<td align="right">0.62</td>
<td align="right">0.74</td>
</tr>
<tr class="odd">
<td align="left">Bar</td>
<td align="right">0.55</td>
<td align="right">0.57</td>
<td align="right">0.62</td>
<td align="right">0.63</td>
<td align="right">NA</td>
<td align="right">0.91</td>
</tr>
</tbody>
</table>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Aleksandra Bliznina, Michael Mansfield, Charles Plessy.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.2.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
