<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Genomic features of the breakpoints • OikScrambling</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Genomic features of the breakpoints">
<meta property="og:description" content="OikScrambling">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">OikScrambling</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">4.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/OikScrambling.html">Get started</a>
</li>
<li>
  <a href="../articles/LoadGenomicBreaks.html">Data load</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Supplemental material
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/OxfordPlots.html">Figure 1A DNA line plots</a>
    </li>
    <li>
      <a href="../articles/ParallelPlots.html">Figure 1B chromosome parallel plots</a>
    </li>
    <li>
      <a href="../articles/DistanceMeasures.html">Table 1 scrambling index</a>
    </li>
    <li>
      <a href="../articles/ChromosomePlots.html">Figure 5 genome-wide feature patterns</a>
    </li>
    <li>
      <a href="../articles/SquareWindowPlots.html">Figure 7 DNA line plots</a>
    </li>
    <li>
      <a href="../articles/ScalingBreakpoints.html">Figure 7 breakpoint number barplots</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/CNEs.html">Conserved non-coding elements</a>
    </li>
    <li>
      <a href="../articles/ChromosomePlots.html">Chromosome-scale feature density plots</a>
    </li>
    <li>
      <a href="../articles/CoalescingCutoffs.html">Modifying coalescent cutoffs</a>
    </li>
    <li>
      <a href="../articles/ColinearityInterruptors.html">Regions interrupting colinearity</a>
    </li>
    <li>
      <a href="../articles/DistanceMeasures.html">Distance Measures</a>
    </li>
    <li>
      <a href="../articles/GenomicFeatures.html">Genomic features of the breakpoints</a>
    </li>
    <li>
      <a href="../articles/GenomicHeatmaps.html">Genomic features of the breakpoints (heatmaps)</a>
    </li>
    <li>
      <a href="../articles/Inversions.html">Inversions</a>
    </li>
    <li>
      <a href="../articles/LoadGenomicBreaks.html">Load genomic break data</a>
    </li>
    <li>
      <a href="../articles/OperonConservation.html">Operon conservation analysis</a>
    </li>
    <li>
      <a href="../articles/OrthogroupSyntenies.html">Orthogroup syntenies</a>
    </li>
    <li>
      <a href="../articles/OxfordPlots.html">Alignment line plots, whole genomes</a>
    </li>
    <li>
      <a href="../articles/PAC3.html">Region of interest: PAC3</a>
    </li>
    <li>
      <a href="../articles/PWM.html">Motif search and analysis</a>
    </li>
    <li>
      <a href="../articles/ParallelPlots.html">Whole-chromosome parallel plots</a>
    </li>
    <li>
      <a href="../articles/RegionWidths.html">Width of the aligned and mapped regions.</a>
    </li>
    <li>
      <a href="../articles/RegionsOfInterest.html">Regions of interest</a>
    </li>
    <li>
      <a href="../articles/RepeatRegions.html">Repeat regions</a>
    </li>
    <li>
      <a href="../articles/ScalingBreakpoints.html">Scaling breakpoints</a>
    </li>
    <li>
      <a href="../articles/SegmentMovements.html">Segment movement distances</a>
    </li>
    <li>
      <a href="../articles/SquareWindowPlots.html">Alignment line plots, 10×10 Mbp windows</a>
    </li>
    <li>
      <a href="../articles/TFBS.html">Transcription factor binding sites</a>
    </li>
    <li>
      <a href="../articles/TandemSiteDuplications.html">Tandem site duplications</a>
    </li>
    <li>
      <a href="../articles/Translocations.html">Translocations</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/oist/LuscombeU_OikScrambling/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="GenomicFeatures_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Genomic features of the breakpoints</h1>
                        <h4 data-toc-skip class="author">Charlotte West</h4>
                        <h4 data-toc-skip class="author">Charles Plessy</h4>
            
            <h4 data-toc-skip class="date">21 July, 2023</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/oist/LuscombeU_OikScrambling/blob/HEAD/vignettes/GenomicFeatures.Rmd" class="external-link"><code>vignettes/GenomicFeatures.Rmd</code></a></small>
      <div class="hidden name"><code>GenomicFeatures.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="load-packages-and-data">Load packages and data<a class="anchor" aria-label="anchor" href="#load-packages-and-data"></a>
</h2>
<p>See <code>?OikScrambling:::loadAllGenomes</code>, <code>?OikScrambling:::loadAllTranscriptsGR</code>, and <code><a href="../articles/LoadGenomicBreaks.html">vignette("LoadGenomicBreaks", package = "OikScrambling")</a></code> for how the different objects are prepared.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/oist/LuscombeU_OikScrambling" class="external-link">'OikScrambling'</a></span><span class="op">)</span>   |&gt; <span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://bioconductor.org/packages/GenomicFeatures" class="external-link">'GenomicFeatures'</a></span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st">'heatmaps'</span><span class="op">)</span>        |&gt; <span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://patchwork.data-imaginist.com" class="external-link">'patchwork'</a></span><span class="op">)</span>       |&gt; <span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="op">)</span>
<span class="va">genomes</span>     <span class="op">&lt;-</span> <span class="fu">OikScrambling</span><span class="fu">:::</span><span class="fu"><a href="../reference/loadAllGenomes.html">loadAllGenomes</a></span><span class="op">(</span>compat <span class="op">=</span> <span class="cn">F</span><span class="op">)</span>
<span class="co"># Cannot use Knitr cache as long as "annot" objects are used.</span>
<span class="va">annots</span>      <span class="op">&lt;-</span> <span class="fu">OikScrambling</span><span class="fu">:::</span><span class="fu"><a href="../reference/loadAllAnnotations.html">loadAllAnnotations</a></span><span class="op">(</span><span class="op">)</span>   |&gt; <span class="fu"><a href="https://rdrr.io/r/base/warning.html" class="external-link">suppressWarnings</a></span><span class="op">(</span><span class="op">)</span>
<span class="va">reps</span>        <span class="op">&lt;-</span> <span class="fu">OikScrambling</span><span class="fu">:::</span><span class="fu"><a href="../reference/loadAllRepeats.html">loadAllRepeats</a></span><span class="op">(</span>compat <span class="op">=</span> <span class="cn">F</span><span class="op">)</span>
<span class="va">genes</span>       <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">annots</span>, <span class="kw">function</span><span class="op">(</span><span class="va">a</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/sort.html" class="external-link">sort</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/GenomicFeatures/man/transcripts.html" class="external-link">genes</a></span><span class="op">(</span><span class="va">a</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/SimpleList-class.html" class="external-link">SimpleList</a></span><span class="op">(</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/load.html" class="external-link">load</a></span><span class="op">(</span><span class="st">"BreakPoints.Rdata"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"rGADEM"</span><span class="op">)</span>    |&gt; <span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"ggseqlogo"</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="op">)</span>
<span class="va">ces</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="st">"CEs.Rds"</span><span class="op">)</span>
<span class="va">tfbs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="st">"pwmMatchesOki_12_95.Rds"</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="gene-feature-coverage-around-aligned-and-unaligned-genomic-regions-">Gene feature coverage around aligned and unaligned genomic regions.<a class="anchor" aria-label="anchor" href="#gene-feature-coverage-around-aligned-and-unaligned-genomic-regions-"></a>
</h2>
<div class="section level3">
<h3 id="model-operon-areas">Model operon areas<a class="anchor" aria-label="anchor" href="#model-operon-areas"></a>
</h3>
<div class="section level4">
<h4 id="notes-about-operons">Notes about operons<a class="anchor" aria-label="anchor" href="#notes-about-operons"></a>
</h4>
<p>The operon of <a href="https://journals.asm.org/doi/10.1128/MCB.24.17.7795-7805.2004" class="external-link">Ganot et al., 2004</a> appears to be conserved in OKI and OSA (near <code>OSKA2016v1.9::PAR:10984831-10995276</code>). In the annotation visible in ZENBU and NCBI:</p>
<ul>
<li>
<code>Oidioi.mRNA.OKI2018_I69.PAR.g12978</code> is the most likely start of the Ganot operon as it starts downstream of a CAGE TSS peak and has no trans-splicing site. It is homologous to <code>g10133 (PAR:10985271..10985910+)</code> in OSKA2016v1.9 which also has a transcription start site and no trans-splicing site. Best Norwegian hit is <code>CBY19033</code>. Gene name is unclear but weaker blast hits suggest <em>centrosomal POC5 isoform X1</em>.</li>
<li>IQ domain-containing protein K isoform X2 is <code>Oidioi.mRNA.OKI2018_I69.PAR.g12977</code> and is <code>g10134</code> in OSKA2016. It has a bona fide trans-splicing site in both genomes. It is <code>CBY19034</code> in Norway. (Not reported in Ganot et al., 2004).</li>
<li>Exportin 7 is <code>Oidioi.mRNA.OKI2018_I69.PAR.g12976</code>. (First member of the operon of Ganot et al., 2004, which calls it Ran-binding protein 16). In the current annotation of the OSKA2016v1.9 genome, <code>g10135</code> matches well but misses the N-terminal part. Nevertheless a blastx search in region <code>PAR:10986742..10988032+</code> finds it. In line with this, there is a trans-splicing site upstream the unannotated area.</li>
<li>Dynein light chain is <code>Oidioi.mRNA.OKI2018_I69.PAR.g12975</code>.</li>
<li>Delta tubulin is <code>Oidioi.mRNA.OKI2018_I69.PAR.g12974</code>.</li>
<li>
<em>MBF</em> is <code>Oidioi.mRNA.OKI2018_I69.PAR.g12973</code>. Also called <em>enthelial differenciation-related factor 1</em> in other species.</li>
<li>CycD is <code>Oidioi.mRNA.OKI2018_I69.PAR.g12972</code> and <code>g10138</code> in OSKA. Polyadenylation signal conserved.</li>
</ul>
<p>(Note that the transcripts IDs here differ from the public ones.)</p>
<p>Ganot et al., 2004 reported very small (&lt;30) intercistronic regions and Denoeud et al., 2010 stated: “<em>The operons were predicted as co-oriented genes separated by 60 nucleotides at most : 1761 operons containing 4997 genes were predicted on the reference assembly</em>”. However, in our annotation, we do not have the UTRs, but only the distances between translation stops and starts, so we need a broader window. The example below shows that a window of 400 would break the Ganot 2004 operon, so let’s use 500 instead.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/GenomicFeatures/man/transcripts.html" class="external-link">transcripts</a></span><span class="op">(</span><span class="va">annots</span><span class="op">$</span><span class="va">Oki</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/findOverlaps-methods.html" class="external-link">subsetByOverlaps</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a></span><span class="op">(</span><span class="st">"PAR:15956430-15967745"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## GRanges object with 8 ranges and 2 metadata columns:</span>
<span class="co">##       seqnames            ranges strand |     tx_id     tx_name</span>
<span class="co">##          &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt; &lt;character&gt;</span>
<span class="co">##   [1]      PAR 15956464-15957432      - |     14303   g12997.t1</span>
<span class="co">##   [2]      PAR 15957795-15958315      - |     14304   g12998.t1</span>
<span class="co">##   [3]      PAR 15958381-15959726      - |     14305   g12999.t1</span>
<span class="co">##   [4]      PAR 15959875-15960185      - |     14306   g13000.t1</span>
<span class="co">##   [5]      PAR 15960589-15964701      - |     14307   g13001.t1</span>
<span class="co">##   [6]      PAR 15960653-15964701      - |     14308   g13001.t2</span>
<span class="co">##   [7]      PAR 15964824-15965481      - |     14309   g13002.t1</span>
<span class="co">##   [8]      PAR 15965608-15967583      - |     14310   g13003.t1</span>
<span class="co">##   -------</span>
<span class="co">##   seqinfo: 19 sequences from OKI2018.I69 genome</span></code></pre>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/GenomicFeatures/man/transcripts.html" class="external-link">transcripts</a></span><span class="op">(</span><span class="va">annots</span><span class="op">$</span><span class="va">Oki</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/findOverlaps-methods.html" class="external-link">subsetByOverlaps</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a></span><span class="op">(</span><span class="st">"PAR:15956430-15967745"</span><span class="op">)</span><span class="op">)</span> |&gt; <span class="fu">cleanGaps</span><span class="op">(</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/start.html" class="external-link">width</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## [1] 362  65 148 403 122 126</span></code></pre>
</div>
<div class="section level4">
<h4 id="compute-operons">Compute operons<a class="anchor" aria-label="anchor" href="#compute-operons"></a>
</h4>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">operons</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">annots</span>, \<span class="op">(</span><span class="va">a</span><span class="op">)</span> <span class="op">{</span><span class="va">a</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/GenomicFeatures/man/transcripts.html" class="external-link">genes</a></span><span class="op">(</span><span class="op">)</span> |&gt; <span class="fu">OikScrambling</span><span class="fu">:::</span><span class="fu"><a href="../reference/calcOperons.html">calcOperons</a></span><span class="op">(</span>window <span class="op">=</span> <span class="fl">500</span><span class="op">)</span> <span class="op">}</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/SimpleList-class.html" class="external-link">SimpleList</a></span><span class="op">(</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/r/base/warning.html" class="external-link">suppressWarnings</a></span><span class="op">(</span><span class="op">)</span>
<span class="va">operons</span><span class="op">$</span><span class="va">Nor</span> <span class="op">&lt;-</span> <span class="fu">OikScrambling</span><span class="fu">:::</span><span class="fu"><a href="../reference/calcOperons.html">calcOperons</a></span><span class="op">(</span>window <span class="op">=</span> <span class="fl">500</span>, <span class="fu"><a href="https://rdrr.io/pkg/GenomicFeatures/man/transcripts.html" class="external-link">transcripts</a></span><span class="op">(</span><span class="va">annots</span><span class="op">$</span><span class="va">Nor</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/inter-range-methods.html" class="external-link">reduce</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/r/base/warning.html" class="external-link">suppressWarnings</a></span><span class="op">(</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">operons</span>, <span class="va">length</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">##  Oki  Osa  Bar  Kum  Aom  Nor  Ply  Ros </span>
<span class="co">## 3124 2741 2379 2974 2648 2996 1088 1472</span></code></pre>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">operons</span>, \<span class="op">(</span><span class="va">o</span><span class="op">)</span> <span class="op">{</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">o</span><span class="op">$</span><span class="va">n</span><span class="op">)</span><span class="op">}</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## $Oki</span>
<span class="co">## </span>
<span class="co">##    2    3    4    5    6    7    8    9   10   11   14 </span>
<span class="co">## 1943  647  259  134   69   33   19   10    6    3    1 </span>
<span class="co">## </span>
<span class="co">## $Osa</span>
<span class="co">## </span>
<span class="co">##    2    3    4    5    6    7    8    9   10   11   12   13 </span>
<span class="co">## 1714  557  248   98   63   32   13   10    1    3    1    1 </span>
<span class="co">## </span>
<span class="co">## $Bar</span>
<span class="co">## </span>
<span class="co">##    2    3    4    5    6    7    8    9   10   11   12   13   14   17 </span>
<span class="co">## 1450  475  221  106   57   30   23    9    3    1    1    1    1    1 </span>
<span class="co">## </span>
<span class="co">## $Kum</span>
<span class="co">## </span>
<span class="co">##    2    3    4    5    6    7    8    9   10   11   12   13   14   17 </span>
<span class="co">## 1855  648  233  114   56   28   18    4    7    6    1    1    2    1 </span>
<span class="co">## </span>
<span class="co">## $Aom</span>
<span class="co">## </span>
<span class="co">##    2    3    4    5    6    7    8    9   10   11   12   14 </span>
<span class="co">## 1689  492  239  107   52   36   14   10    3    4    1    1 </span>
<span class="co">## </span>
<span class="co">## $Nor</span>
<span class="co">## </span>
<span class="co">##    2    3    4    5    6    7    8    9   10   11   12   13   14   15   16   17 </span>
<span class="co">## 1724  562  264  138   93   65   38   40   16   21   11   10    3    4    3    1 </span>
<span class="co">##   20   22 </span>
<span class="co">##    1    2 </span>
<span class="co">## </span>
<span class="co">## $Ply</span>
<span class="co">## </span>
<span class="co">##   2   3   4   5   6   7   8   9  13  14  15 </span>
<span class="co">## 937 114  22   6   2   1   1   1   1   1   2 </span>
<span class="co">## </span>
<span class="co">## $Ros</span>
<span class="co">## </span>
<span class="co">##    2    3    4    5    6    7    9   10   11   21   26   29 </span>
<span class="co">## 1262  163   33    5    1    1    1    1    1    1    1    2</span></code></pre>
<p>With the OdB3 genome, the results are apparently close to those of Denoeud et al., 2010 (see above for why the difference). We have a few more (1844 instead of 1761), they contain more genes (5814 instead of 4997), mostly because our computation outputs 50 operons longer than 9 genes (617 genes in total). Nevertheless, the proportion of operons of length 2, 3 and 4 are visually similar in both cases.</p>
</div>
</div>
<div class="section level3">
<h3 id="accessory-functions">Accessory functions<a class="anchor" aria-label="anchor" href="#accessory-functions"></a>
</h3>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">prepareFreqPlot</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">gb</span>, <span class="va">annot</span> <span class="op">=</span> <span class="cn">NULL</span>, <span class="va">rep</span> <span class="op">=</span> <span class="cn">NULL</span>, <span class="va">cne</span> <span class="op">=</span> <span class="cn">NULL</span>, <span class="va">tfbs</span> <span class="op">=</span> <span class="cn">NULL</span>, <span class="va">operons</span> <span class="op">=</span> <span class="cn">NULL</span>, <span class="va">win</span> <span class="op">=</span> <span class="fl">1000</span>, <span class="va">direction</span> <span class="op">=</span> <span class="st">"left"</span>, <span class="va">withGenes</span><span class="op">=</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"yes"</span>, <span class="st">"no"</span>, <span class="st">"only"</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span>
  <span class="va">withGenes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/match.arg.html" class="external-link">match.arg</a></span><span class="op">(</span><span class="va">withGenes</span><span class="op">)</span>
  <span class="va">l</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="op">)</span>
  <span class="co"># Note that get_bps(direction="mid") is probably not what you think.</span>
  <span class="kw">if</span><span class="op">(</span><span class="va">direction</span> <span class="op">==</span> <span class="st">"mid"</span><span class="op">)</span> <span class="op">{</span>
    <span class="va">gb</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/intra-range-methods.html" class="external-link">resize</a></span><span class="op">(</span><span class="va">gb</span>, <span class="fl">1</span>, <span class="st">"center"</span><span class="op">)</span>
    <span class="va">direction</span> <span class="op">&lt;-</span> <span class="st">"left"</span>
  <span class="op">}</span>
  <span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NULL.html" class="external-link">is.null</a></span><span class="op">(</span><span class="va">annot</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span>
    <span class="kw">if</span> <span class="op">(</span><span class="va">withGenes</span> <span class="op"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"yes"</span>, <span class="st">"only"</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span>
      <span class="va">l</span><span class="op">$</span><span class="va">genes</span>  <span class="op">&lt;-</span> <span class="fu">feature_coverage</span><span class="op">(</span><span class="va">gb</span>,     <span class="fu"><a href="https://rdrr.io/pkg/GenomicFeatures/man/transcripts.html" class="external-link">genes</a></span><span class="op">(</span><span class="va">annot</span><span class="op">)</span>,         win <span class="op">=</span> <span class="va">win</span>, lab <span class="op">=</span> <span class="st">"genes"</span>,  direction <span class="op">=</span> <span class="va">direction</span><span class="op">)</span>
    <span class="op">}</span>
    <span class="kw">if</span> <span class="op">(</span><span class="va">withGenes</span> <span class="op"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"yes"</span>, <span class="st">"no"</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span>
      <span class="va">l</span><span class="op">$</span><span class="va">intron</span> <span class="op">&lt;-</span> <span class="fu">feature_coverage</span><span class="op">(</span><span class="va">gb</span>,     <span class="fu"><a href="https://rdrr.io/pkg/GenomicFeatures/man/exonicParts.html" class="external-link">intronicParts</a></span><span class="op">(</span><span class="va">annot</span><span class="op">)</span>, win <span class="op">=</span> <span class="va">win</span>, lab <span class="op">=</span> <span class="st">"intron"</span>, direction <span class="op">=</span> <span class="va">direction</span><span class="op">)</span>
      <span class="va">l</span><span class="op">$</span><span class="va">exon</span>   <span class="op">&lt;-</span> <span class="fu">feature_coverage</span><span class="op">(</span><span class="va">gb</span>,     <span class="fu"><a href="https://rdrr.io/pkg/GenomicFeatures/man/exonicParts.html" class="external-link">exonicParts</a></span><span class="op">(</span><span class="va">annot</span><span class="op">)</span>,   win <span class="op">=</span> <span class="va">win</span>, lab <span class="op">=</span> <span class="st">"exon"</span>,   direction <span class="op">=</span> <span class="va">direction</span><span class="op">)</span>
      <span class="co">#reorder</span>
      <span class="va">newOrder</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/order.html" class="external-link">order</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/colSums.html" class="external-link">rowSums</a></span><span class="op">(</span><span class="va">l</span><span class="op">$</span><span class="va">intron</span><span class="op">@</span><span class="va">image</span><span class="op">)</span><span class="op">)</span>
      <span class="va">l</span><span class="op">$</span><span class="va">intron</span><span class="op">@</span><span class="va">image</span> <span class="op">&lt;-</span> <span class="va">l</span><span class="op">$</span><span class="va">intron</span><span class="op">@</span><span class="va">image</span><span class="op">[</span><span class="va">newOrder</span>,<span class="op">]</span>
      <span class="va">l</span><span class="op">$</span><span class="va">exon</span><span class="op">@</span><span class="va">image</span>   <span class="op">&lt;-</span> <span class="va">l</span><span class="op">$</span><span class="va">exon</span><span class="op">@</span><span class="va">image</span><span class="op">[</span><span class="va">newOrder</span>,<span class="op">]</span>
    <span class="op">}</span>
  <span class="op">}</span>
  <span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NULL.html" class="external-link">is.null</a></span><span class="op">(</span><span class="va">operons</span><span class="op">)</span><span class="op">)</span>
    <span class="va">l</span><span class="op">$</span><span class="va">operons</span><span class="op">&lt;-</span><span class="fu">feature_coverage</span><span class="op">(</span><span class="va">gb</span>,    <span class="va">operons</span>,              win <span class="op">=</span> <span class="va">win</span>, lab <span class="op">=</span> <span class="st">"operons"</span>,direction <span class="op">=</span> <span class="va">direction</span><span class="op">)</span>
  <span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NULL.html" class="external-link">is.null</a></span><span class="op">(</span><span class="va">rep</span><span class="op">)</span><span class="op">)</span>
    <span class="va">l</span><span class="op">$</span><span class="va">reps</span>   <span class="op">&lt;-</span> <span class="fu">feature_coverage</span><span class="op">(</span><span class="va">gb</span>,   <span class="va">rep</span>,                  win <span class="op">=</span> <span class="va">win</span>, lab <span class="op">=</span> <span class="st">"repeats"</span>,direction <span class="op">=</span> <span class="va">direction</span><span class="op">)</span>
  <span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NULL.html" class="external-link">is.null</a></span><span class="op">(</span><span class="va">cne</span><span class="op">)</span><span class="op">)</span>
    <span class="va">l</span><span class="op">$</span><span class="va">CNE</span>   <span class="op">&lt;-</span> <span class="fu">feature_coverage</span><span class="op">(</span><span class="va">gb</span>,    <span class="va">cne</span>,                  win <span class="op">=</span> <span class="va">win</span>, lab <span class="op">=</span> <span class="st">"CNE"</span>,    direction <span class="op">=</span> <span class="va">direction</span><span class="op">)</span>
  <span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NULL.html" class="external-link">is.null</a></span><span class="op">(</span><span class="va">tfbs</span><span class="op">)</span><span class="op">)</span>
    <span class="va">l</span><span class="op">$</span><span class="va">tfbs</span>  <span class="op">&lt;-</span> <span class="fu">feature_coverage</span><span class="op">(</span><span class="va">gb</span>,    <span class="va">tfbs</span>,                 win <span class="op">=</span> <span class="va">win</span>, lab <span class="op">=</span> <span class="st">"TFBS"</span>,   direction <span class="op">=</span> <span class="va">direction</span><span class="op">)</span>
  <span class="va">l</span>
<span class="op">}</span>

<span class="co">#' Prepare a long tibble for frequency plotting</span>
<span class="co">#' </span>
<span class="co">#' Loop over a list of `Heatmap` objects to produce a long tibble that can be</span>
<span class="co">#' used with ggplot to plot frequency of each feature included in the list.</span>
<span class="co">#'</span>
<span class="co">#' @param x A list of `Heatmap` objects</span>
<span class="co">#' @param desc An additional description (character string)</span>
<span class="co">#'</span>
<span class="co">#' @return A long `tibble` that is the vertical concatenation of all tibbles</span>
<span class="co">#' produced for each element of the list `x`.  Its `what` column is made from</span>
<span class="co">#' the names of the heatmaps in the list, and its `desc` column is made from the</span>
<span class="co">#' `desc` arguments.  The other columns (`val`, `freq` and `pos`) originate from</span>
<span class="co">#' the `Heatmap_to_freq_tibble` function call.</span>
<span class="va">prepareGGFreqPlot</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">x</span>, <span class="va">desc</span> <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span> <span class="op">{</span>
  <span class="va">l</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span>, \<span class="op">(</span><span class="va">name</span><span class="op">)</span><span class="op">{</span>
    <span class="va">tib</span> <span class="op">&lt;-</span> <span class="fu">Heatmap_to_freq_tibble</span><span class="op">(</span><span class="va">x</span><span class="op">[[</span><span class="va">name</span><span class="op">]</span><span class="op">]</span>, desc <span class="op">=</span> <span class="va">desc</span><span class="op">)</span>
    <span class="va">tib</span><span class="op">$</span><span class="va">what</span> <span class="op">&lt;-</span> <span class="va">name</span>
    <span class="va">tib</span>
  <span class="op">}</span><span class="op">)</span>
   <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/do.call.html" class="external-link">do.call</a></span><span class="op">(</span><span class="va">rbind</span>, <span class="va">l</span><span class="op">)</span>
<span class="op">}</span>

<span class="co">#' Heatmap to long tibble for ggplot</span>
<span class="co">#' </span>
<span class="co">#' Coerce a Heatmap object into a \dQuote{long} tibble with one value per</span>
<span class="co">#' position, either as raw counts or as a frequency.</span>
<span class="co">#' </span>
<span class="co">#' The tibble can then be passed to ggplot.  The `image` matrix of the `Heatmap`</span>
<span class="co">#' object has one column per nucleotide position and it is assumed that its</span>
<span class="co">#' central column is for position zero.</span>
<span class="co">#'</span>
<span class="co">#' @param h a `Heatmap` object</span>
<span class="co">#'</span>
<span class="co">#' @return a `tibble` with `val`, `freq` and `pos` columns</span>
<span class="va">Heatmap_to_freq_tibble</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">h</span>, <span class="va">desc</span> <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span> <span class="op">{</span>
  <span class="va">val</span>  <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/colSums.html" class="external-link">colSums</a></span><span class="op">(</span><span class="va">h</span><span class="op">@</span><span class="va">image</span><span class="op">)</span>
  <span class="fu">tibble</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/tibble.html" class="external-link">tibble</a></span><span class="op">(</span> val  <span class="op">=</span> <span class="va">val</span>
                , freq <span class="op">=</span> <span class="va">val</span> <span class="op">/</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">h</span><span class="op">@</span><span class="va">image</span><span class="op">)</span>
                , pos  <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/nrow.html" class="external-link">ncol</a></span><span class="op">(</span><span class="va">h</span><span class="op">@</span><span class="va">image</span><span class="op">)</span> <span class="op">-</span> <span class="fu"><a href="https://rdrr.io/r/base/Round.html" class="external-link">round</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/nrow.html" class="external-link">ncol</a></span><span class="op">(</span><span class="va">h</span><span class="op">@</span><span class="va">image</span><span class="op">)</span><span class="op">)</span> <span class="op">/</span> <span class="fl">2</span>
                , desc <span class="op">=</span> <span class="va">desc</span><span class="op">)</span>
<span class="op">}</span></code></pre></div>
</div>
<div class="section level3">
<h3 id="centering-on-left-aligned-boundaries-">Centering on left-aligned boundaries.<a class="anchor" aria-label="anchor" href="#centering-on-left-aligned-boundaries-"></a>
</h3>
<p>Breakpoints are between mapped regions. If the unmapped region is broad we have little information where the breakpoint originally was. We set arbitrarly the breakpoints to the left-side (end-side) coordinate of genomic regions.</p>
<p>Mapped regions can consist of multiple colinear aligned regions, or be a single uncoalesced aligned region. We will see that they have different properties.</p>
<p>Our annotations is mostly made of coding exons, so the features <em>exons</em> and <em>cds</em> have very similar profiles. Therefore we keep only <em>exons</em> in the rest of the analysis.</p>
<p>Positive coordinates (and zero) start in selected regions and negative coordinates start outside these regions. As we move further from 0, we can not guarantee that the statement is still true because some regions have a width narrower than the plot window. This is why the values tend to equilibrate away from the boundaries.</p>
<p>Aligned regions are enriched in exons and depleted in introns.</p>
<p>Close to the breakpoint of uncoalesced mapped regions, exon annotations are depleted, which makes sense, as exons are usually mapable. Within the mapped regions, frequency of introns and exons are comparable. In contrary, within coalesced mapped regions, the exon annotations are over-represented.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#' Pre-compute feature frequencies in fixed windows</span>
<span class="co">#'</span>
<span class="co">#' @param target 3-letter name of one Oik genome</span>
<span class="co">#' @param query 3-letter name of another Oik genome</span>
<span class="co">#' @param perpareFreqPlot a `perpareFreqPlot` function</span>
<span class="co">#' @param reps `SimpleList` of `GRanges` of repeat regions</span>
<span class="co">#' @param CE `SimpleList` of `GRanges` with conserved non-coding elements</span>
<span class="co">#' @param tfbs `SimpleList` of `GRanges` of putative transcription factor binding sites</span>
<span class="co">#' @param operons `SimpleList` of `GRanges` of operon regions.</span>
<span class="co">#' @param direction Where to center the windows (`left`, `right` or `mid`)</span>
<span class="co">#'</span>
<span class="co">#' @return A `SimpleList` of Heatmap objects</span>
<span class="va">computePlotData</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">target</span>, <span class="va">query</span>, <span class="va">annots</span><span class="op">=</span><span class="cn">NULL</span>, <span class="va">reps</span><span class="op">=</span><span class="cn">NULL</span>, <span class="va">CE</span><span class="op">=</span><span class="cn">NULL</span>, <span class="va">tfbs</span><span class="op">=</span><span class="cn">NULL</span>, <span class="va">operons</span><span class="op">=</span><span class="cn">NULL</span>, <span class="va">direction</span><span class="op">=</span><span class="st">"left"</span><span class="op">)</span> <span class="op">{</span>
  <span class="va">pair</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="va">target</span>, <span class="va">query</span>, sep <span class="op">=</span> <span class="st">"_"</span><span class="op">)</span>
  <span class="va">current_GB_aln_nonCoa</span>     <span class="op">&lt;-</span> <span class="va">gbs</span><span class="op">[[</span><span class="va">pair</span><span class="op">]</span><span class="op">]</span><span class="op">[</span>   <span class="va">gbs</span><span class="op">[[</span><span class="va">pair</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">nonCoa</span><span class="op">]</span>
  <span class="va">current_GB_aln_coa</span>        <span class="op">&lt;-</span> <span class="va">gbs</span><span class="op">[[</span><span class="va">pair</span><span class="op">]</span><span class="op">]</span><span class="op">[</span> <span class="op">!</span> <span class="va">gbs</span><span class="op">[[</span><span class="va">pair</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">nonCoa</span><span class="op">]</span>
  <span class="va">current_GB_syn_block</span>      <span class="op">&lt;-</span> <span class="va">coa</span><span class="op">[[</span><span class="va">pair</span><span class="op">]</span><span class="op">]</span><span class="op">[</span> <span class="op">!</span> <span class="va">coa</span><span class="op">[[</span><span class="va">pair</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">nonCoa</span><span class="op">]</span>
  <span class="co"># We do not use unal, because it overlaps with bri</span>
  <span class="va">current_GB_unmap</span>          <span class="op">&lt;-</span> <span class="va">unmap</span><span class="op">[[</span><span class="va">pair</span><span class="op">]</span><span class="op">]</span>
  <span class="va">current_GB_bri</span>        <span class="op">&lt;-</span> <span class="va">bri</span><span class="op">[[</span><span class="va">pair</span><span class="op">]</span><span class="op">]</span>
  
  <span class="va">targetGenome</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">genome</a></span><span class="op">(</span><span class="va">gbs</span><span class="op">[[</span><span class="va">pair</span><span class="op">]</span><span class="op">]</span><span class="op">)</span><span class="op">)</span>
  
  <span class="va">prepareFreqPlot_default</span> <span class="op">&lt;-</span> \<span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="fu">prepareFreqPlot</span><span class="op">(</span><span class="va">x</span>, annot <span class="op">=</span> <span class="va">annots</span><span class="op">[[</span><span class="va">target</span><span class="op">]</span><span class="op">]</span>, rep <span class="op">=</span> <span class="va">reps</span><span class="op">[[</span><span class="va">targetGenome</span><span class="op">]</span><span class="op">]</span>, cne <span class="op">=</span> <span class="va">CE</span><span class="op">[[</span><span class="va">pair</span><span class="op">]</span><span class="op">]</span>, tfbs <span class="op">=</span> <span class="va">tfbs</span>, operons <span class="op">=</span> <span class="va">operons</span><span class="op">[[</span><span class="va">target</span><span class="op">]</span><span class="op">]</span>, direction <span class="op">=</span> <span class="va">direction</span><span class="op">)</span>
  
  <span class="va">hmList</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/SimpleList-class.html" class="external-link">SimpleList</a></span><span class="op">(</span><span class="op">)</span>
  <span class="va">hmList</span><span class="op">$</span><span class="va">aln_nonCoa</span> <span class="op">&lt;-</span> <span class="fu">prepareFreqPlot_default</span><span class="op">(</span><span class="va">current_GB_aln_nonCoa</span><span class="op">)</span>
  <span class="va">hmList</span><span class="op">$</span><span class="va">aln_coa</span>    <span class="op">&lt;-</span> <span class="fu">prepareFreqPlot_default</span><span class="op">(</span><span class="va">current_GB_aln_coa</span><span class="op">)</span>
  <span class="va">hmList</span><span class="op">$</span><span class="va">syn_block</span>  <span class="op">&lt;-</span> <span class="fu">prepareFreqPlot_default</span><span class="op">(</span><span class="va">current_GB_syn_block</span><span class="op">)</span>
  <span class="va">hmList</span><span class="op">$</span><span class="va">unmap</span>      <span class="op">&lt;-</span> <span class="fu">prepareFreqPlot_default</span><span class="op">(</span><span class="va">current_GB_unmap</span><span class="op">)</span>
  <span class="va">hmList</span><span class="op">$</span><span class="va">bri</span>        <span class="op">&lt;-</span> <span class="fu">prepareFreqPlot_default</span><span class="op">(</span><span class="va">current_GB_bri</span><span class="op">)</span>
  <span class="va">hmList</span>
<span class="op">}</span>

<span class="va">hmList_to_plot_noncoding</span> <span class="op">&lt;-</span> <span class="fu">computePlotData</span><span class="op">(</span><span class="st">"Oki"</span>, <span class="st">"Osa"</span>, reps <span class="op">=</span> <span class="va">reps</span>, CE <span class="op">=</span> <span class="va">ces</span><span class="op">$</span><span class="va">ce__50__48</span><span class="op">)</span>
<span class="va">hmList_to_plot_coding</span>    <span class="op">&lt;-</span> <span class="fu">computePlotData</span><span class="op">(</span><span class="st">"Oki"</span>, <span class="st">"Osa"</span>, annots <span class="op">=</span> <span class="va">annots</span>, operons <span class="op">=</span> <span class="va">operons</span><span class="op">)</span>

<span class="va">list_of_heatmaps_to_long_tibble</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">hmList_to_plot</span>, <span class="va">desc</span> <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span> <span class="op">{</span>
  <span class="va">tib</span> <span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/cbind.html" class="external-link">rbind</a></span><span class="op">(</span>
    <span class="va">hmList_to_plot</span><span class="op">$</span><span class="va">aln_nonCoa</span>  |&gt; <span class="fu">prepareGGFreqPlot</span><span class="op">(</span><span class="st">"01_isolated_alns"</span><span class="op">)</span>,
    <span class="va">hmList_to_plot</span><span class="op">$</span><span class="va">unmap</span>       |&gt; <span class="fu">prepareGGFreqPlot</span><span class="op">(</span><span class="st">"02_unmapped"</span><span class="op">)</span>,
    <span class="va">hmList_to_plot</span><span class="op">$</span><span class="va">syn_block</span>   |&gt; <span class="fu">prepareGGFreqPlot</span><span class="op">(</span><span class="st">"03_colinear_reg"</span><span class="op">)</span>,
    <span class="va">hmList_to_plot</span><span class="op">$</span><span class="va">bri</span>         |&gt; <span class="fu">prepareGGFreqPlot</span><span class="op">(</span><span class="st">"04_bridge"</span><span class="op">)</span>,
    <span class="va">hmList_to_plot</span><span class="op">$</span><span class="va">aln_coa</span>     |&gt; <span class="fu">prepareGGFreqPlot</span><span class="op">(</span><span class="st">"05_colinear_alns"</span><span class="op">)</span>
  <span class="op">)</span>
  <span class="va">tib</span><span class="op">$</span><span class="va">desc2</span> <span class="op">&lt;-</span> <span class="va">desc</span>
  <span class="va">tib</span>
<span class="op">}</span>

<span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/cbind.html" class="external-link">rbind</a></span><span class="op">(</span>
  <span class="fu">list_of_heatmaps_to_long_tibble</span><span class="op">(</span><span class="va">hmList_to_plot_noncoding</span>, <span class="st">"noncoding"</span><span class="op">)</span>,
  <span class="fu">list_of_heatmaps_to_long_tibble</span><span class="op">(</span><span class="va">hmList_to_plot_coding</span>, <span class="st">"coding"</span><span class="op">)</span>
<span class="op">)</span> |&gt; <span class="fu">ggplot</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">aes</span><span class="op">(</span><span class="va">pos</span>, <span class="va">freq</span>, col <span class="op">=</span> <span class="va">what</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">geom_line</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">facet_wrap</span><span class="op">(</span><span class="op">~</span><span class="va">desc2</span> <span class="op">+</span> <span class="va">desc</span>, nrow <span class="op">=</span> <span class="fl">2</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">theme_bw</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">ggtitle</span><span class="op">(</span><span class="st">"Aligned on regions left boundary. CNE is 50/48.  Okinawa - Osaka"</span><span class="op">)</span></code></pre></div>
<p><img src="GenomicFeatures_files/figure-html/genomic_features_byType-1.svg" width="1440"></p>
<p>Most of the narrow-width aligned regions did not coalesce. We use this as a classifier instead of an arbitrary width cutoff. See <code><a href="../articles/RegionWidths.html">vignette("RegionWidths", package = "OikScrambling")</a></code> for details.</p>
<div class="section level5">
<h5 id="simplified-version">Simplified version<a class="anchor" aria-label="anchor" href="#simplified-version"></a>
</h5>
<p>With only 2 panels, for unmapped–aligned and bridge–aligned, we display almost all the information present in the 5-panel version above.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">computePlotData_simplified</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">target</span>, <span class="va">query</span>, <span class="va">annots</span><span class="op">=</span><span class="cn">NULL</span>, <span class="va">reps</span><span class="op">=</span><span class="cn">NULL</span>, <span class="va">CE</span><span class="op">=</span><span class="cn">NULL</span>, <span class="va">tfbs</span><span class="op">=</span><span class="cn">NULL</span>, <span class="va">operons</span><span class="op">=</span><span class="cn">NULL</span>, <span class="va">direction</span><span class="op">=</span><span class="st">"left"</span>, <span class="va">withGenes</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"yes"</span>, <span class="st">"no"</span>, <span class="st">"only"</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span>
  <span class="va">withGenes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/match.arg.html" class="external-link">match.arg</a></span><span class="op">(</span><span class="va">withGenes</span><span class="op">)</span>
  <span class="va">pair</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="va">target</span>, <span class="va">query</span>, sep <span class="op">=</span> <span class="st">"_"</span><span class="op">)</span>
  <span class="va">current_GB_brk</span> <span class="op">&lt;-</span> <span class="va">unmap</span><span class="op">[[</span><span class="va">pair</span><span class="op">]</span><span class="op">]</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/findOverlaps-methods.html" class="external-link">subsetByOverlaps</a></span><span class="op">(</span><span class="va">coa</span><span class="op">[[</span><span class="va">pair</span><span class="op">]</span><span class="op">]</span><span class="op">[</span><span class="op">!</span><span class="va">coa</span><span class="op">[[</span><span class="va">pair</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">nonCoa</span><span class="op">]</span> <span class="op">+</span> <span class="fl">1</span><span class="op">)</span> <span class="co"># The unmapped region flanking colinear regions</span>
  <span class="va">current_GB_bri</span> <span class="op">&lt;-</span> <span class="va">bri</span><span class="op">[[</span><span class="va">pair</span><span class="op">]</span><span class="op">]</span>
  
  <span class="va">targetGenome</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">genome</a></span><span class="op">(</span><span class="va">gbs</span><span class="op">[[</span><span class="va">pair</span><span class="op">]</span><span class="op">]</span><span class="op">)</span><span class="op">)</span>
  
  <span class="va">prepareFreqPlot_default</span> <span class="op">&lt;-</span> \<span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="fu">prepareFreqPlot</span><span class="op">(</span><span class="va">x</span>, annot <span class="op">=</span> <span class="va">annots</span><span class="op">[[</span><span class="va">target</span><span class="op">]</span><span class="op">]</span>, rep <span class="op">=</span> <span class="va">reps</span><span class="op">[[</span><span class="va">targetGenome</span><span class="op">]</span><span class="op">]</span>, cne <span class="op">=</span> <span class="va">CE</span><span class="op">[[</span><span class="va">pair</span><span class="op">]</span><span class="op">]</span>, tfbs <span class="op">=</span> <span class="va">tfbs</span>, operons <span class="op">=</span> <span class="va">operons</span><span class="op">[[</span><span class="va">target</span><span class="op">]</span><span class="op">]</span>, direction <span class="op">=</span> <span class="va">direction</span>, withGenes <span class="op">=</span> <span class="va">withGenes</span><span class="op">)</span>
  
  <span class="va">hmList</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/SimpleList-class.html" class="external-link">SimpleList</a></span><span class="op">(</span><span class="op">)</span>
  <span class="va">hmList</span><span class="op">$</span><span class="va">brk</span> <span class="op">&lt;-</span> <span class="fu">prepareFreqPlot_default</span><span class="op">(</span><span class="va">current_GB_brk</span><span class="op">)</span>
  <span class="va">hmList</span><span class="op">$</span><span class="va">bri</span> <span class="op">&lt;-</span> <span class="fu">prepareFreqPlot_default</span><span class="op">(</span><span class="va">current_GB_bri</span><span class="op">)</span>
  <span class="va">hmList</span>
<span class="op">}</span>

<span class="va">hmList_to_plot_noncoding</span>    <span class="op">&lt;-</span> <span class="fu">computePlotData_simplified</span><span class="op">(</span><span class="st">"Oki"</span>, <span class="st">"Osa"</span>, reps <span class="op">=</span> <span class="va">reps</span>, CE <span class="op">=</span> <span class="va">ces</span><span class="op">$</span><span class="va">ce__50__48</span>, direction <span class="op">=</span> <span class="st">"right"</span><span class="op">)</span>
<span class="va">hmList_to_plot_genesoperons</span> <span class="op">&lt;-</span> <span class="fu">computePlotData_simplified</span><span class="op">(</span><span class="st">"Oki"</span>, <span class="st">"Osa"</span>, annots <span class="op">=</span> <span class="va">annots</span>, operons <span class="op">=</span> <span class="va">operons</span>, direction <span class="op">=</span> <span class="st">"right"</span>, withGenes <span class="op">=</span> <span class="st">"only"</span><span class="op">)</span>
<span class="va">hmList_to_plot_exonintron</span>   <span class="op">&lt;-</span> <span class="fu">computePlotData_simplified</span><span class="op">(</span><span class="st">"Oki"</span>, <span class="st">"Osa"</span>, annots <span class="op">=</span> <span class="va">annots</span>, direction <span class="op">=</span> <span class="st">"right"</span>, withGenes <span class="op">=</span> <span class="st">"no"</span><span class="op">)</span>

<span class="va">list_of_heatmaps_to_long_tibble</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">hmList_to_plot</span>, <span class="va">desc</span> <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span> <span class="op">{</span>
  <span class="va">tib</span> <span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/cbind.html" class="external-link">rbind</a></span><span class="op">(</span>
    <span class="va">hmList_to_plot</span><span class="op">$</span><span class="va">brk</span>         |&gt; <span class="fu">prepareGGFreqPlot</span><span class="op">(</span><span class="st">"01_break"</span><span class="op">)</span>,
    <span class="va">hmList_to_plot</span><span class="op">$</span><span class="va">bri</span>         |&gt; <span class="fu">prepareGGFreqPlot</span><span class="op">(</span><span class="st">"02_bridge"</span><span class="op">)</span>
  <span class="op">)</span>
  <span class="va">tib</span><span class="op">$</span><span class="va">desc2</span> <span class="op">&lt;-</span> <span class="va">desc</span>
  <span class="va">tib</span>
<span class="op">}</span>

<span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/cbind.html" class="external-link">rbind</a></span><span class="op">(</span>
  <span class="fu">list_of_heatmaps_to_long_tibble</span><span class="op">(</span><span class="va">hmList_to_plot_noncoding</span>,    <span class="st">"noncoding"</span><span class="op">)</span>,
  <span class="fu">list_of_heatmaps_to_long_tibble</span><span class="op">(</span><span class="va">hmList_to_plot_genesoperons</span>, <span class="st">"genes and operons"</span><span class="op">)</span>,
  <span class="fu">list_of_heatmaps_to_long_tibble</span><span class="op">(</span><span class="va">hmList_to_plot_exonintron</span>,   <span class="st">"exons and introns"</span><span class="op">)</span>
<span class="op">)</span> |&gt; <span class="fu">ggplot</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">aes</span><span class="op">(</span><span class="va">pos</span>, <span class="va">freq</span>, col <span class="op">=</span> <span class="va">what</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">geom_line</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">facet_wrap</span><span class="op">(</span><span class="op">~</span><span class="va">desc2</span> <span class="op">+</span> <span class="va">desc</span>, nrow <span class="op">=</span> <span class="fl">3</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">theme_bw</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">ggtitle</span><span class="op">(</span><span class="st">"Aligned on regions left boundary. CNE is 50/48.  Okinawa - Osaka"</span><span class="op">)</span></code></pre></div>
<p><img src="GenomicFeatures_files/figure-html/genomic_features_byType_2-1.svg" width="576"></p>
</div>
<div class="section level4">
<h4 id="bridge-regions-classified-by-width-">Bridge regions classified by width.<a class="anchor" aria-label="anchor" href="#bridge-regions-classified-by-width-"></a>
</h4>
<p>Short bridge regions contain small introns. Long bridge regions contain repeats, which may be in introns.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">computePlotData_for_bridge</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">target</span>, <span class="va">query</span>, <span class="va">annots</span><span class="op">=</span><span class="cn">NULL</span>, <span class="va">reps</span><span class="op">=</span><span class="cn">NULL</span>, <span class="va">CE</span><span class="op">=</span><span class="cn">NULL</span>, <span class="va">tfbs</span><span class="op">=</span><span class="cn">NULL</span>, <span class="va">operons</span><span class="op">=</span><span class="cn">NULL</span>, <span class="va">direction</span><span class="op">=</span><span class="st">"left"</span><span class="op">)</span> <span class="op">{</span>
  <span class="va">pair</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="va">target</span>, <span class="va">query</span>, sep <span class="op">=</span> <span class="st">"_"</span><span class="op">)</span>

  <span class="va">current_GB_bri</span>       <span class="op">&lt;-</span> <span class="va">bri</span><span class="op">[[</span><span class="va">pair</span><span class="op">]</span><span class="op">]</span>
  <span class="va">current_GB_bri_small</span> <span class="op">&lt;-</span> <span class="va">current_GB_bri</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/start.html" class="external-link">width</a></span><span class="op">(</span><span class="va">current_GB_bri</span><span class="op">)</span> <span class="op">&lt;</span>  <span class="fl">150</span><span class="op">]</span>
  <span class="va">current_GB_bri_long</span>  <span class="op">&lt;-</span> <span class="va">current_GB_bri</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/start.html" class="external-link">width</a></span><span class="op">(</span><span class="va">current_GB_bri</span><span class="op">)</span> <span class="op">&gt;=</span> <span class="fl">150</span><span class="op">]</span>

  <span class="va">targetGenome</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">genome</a></span><span class="op">(</span><span class="va">gbs</span><span class="op">[[</span><span class="va">pair</span><span class="op">]</span><span class="op">]</span><span class="op">)</span><span class="op">)</span>
  
  <span class="va">prepareFreqPlot_default</span> <span class="op">&lt;-</span> \<span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="fu">prepareFreqPlot</span><span class="op">(</span><span class="va">x</span>, annot <span class="op">=</span> <span class="va">annots</span><span class="op">[[</span><span class="va">target</span><span class="op">]</span><span class="op">]</span>, rep <span class="op">=</span> <span class="va">reps</span><span class="op">[[</span><span class="va">targetGenome</span><span class="op">]</span><span class="op">]</span>, cne <span class="op">=</span> <span class="va">CE</span><span class="op">[[</span><span class="va">pair</span><span class="op">]</span><span class="op">]</span>, tfbs <span class="op">=</span> <span class="va">tfbs</span>, operons <span class="op">=</span> <span class="va">operons</span><span class="op">[[</span><span class="va">target</span><span class="op">]</span><span class="op">]</span>, direction <span class="op">=</span> <span class="va">direction</span><span class="op">)</span>
  
  <span class="va">hmList</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/SimpleList-class.html" class="external-link">SimpleList</a></span><span class="op">(</span><span class="op">)</span>
  <span class="va">hmList</span><span class="op">$</span><span class="va">bri</span>       <span class="op">&lt;-</span> <span class="fu">prepareFreqPlot_default</span><span class="op">(</span><span class="va">current_GB_bri</span><span class="op">)</span>
  <span class="va">hmList</span><span class="op">$</span><span class="va">bri_small</span> <span class="op">&lt;-</span> <span class="fu">prepareFreqPlot_default</span><span class="op">(</span><span class="va">current_GB_bri_small</span><span class="op">)</span>
  <span class="va">hmList</span><span class="op">$</span><span class="va">bri_long</span>  <span class="op">&lt;-</span> <span class="fu">prepareFreqPlot_default</span><span class="op">(</span><span class="va">current_GB_bri_long</span><span class="op">)</span>
  <span class="va">hmList</span>
<span class="op">}</span>

<span class="va">hmList_to_plot_noncoding</span> <span class="op">&lt;-</span> <span class="fu">computePlotData_for_bridge</span><span class="op">(</span><span class="st">"Oki"</span>, <span class="st">"Osa"</span>, reps <span class="op">=</span> <span class="va">reps</span>, CE <span class="op">=</span> <span class="va">ces</span><span class="op">$</span><span class="va">ce__50__48</span><span class="op">)</span>
<span class="va">hmList_to_plot_coding</span>    <span class="op">&lt;-</span> <span class="fu">computePlotData_for_bridge</span><span class="op">(</span><span class="st">"Oki"</span>, <span class="st">"Osa"</span>, annots <span class="op">=</span> <span class="va">annots</span>, operons <span class="op">=</span> <span class="va">operons</span><span class="op">)</span>

<span class="va">list_of_heatmaps_to_long_tibble</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">hmList_to_plot</span>, <span class="va">desc</span> <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span> <span class="op">{</span>
  <span class="va">tib</span> <span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/cbind.html" class="external-link">rbind</a></span><span class="op">(</span>
    <span class="va">hmList_to_plot</span><span class="op">$</span><span class="va">bri</span>         |&gt; <span class="fu">prepareGGFreqPlot</span><span class="op">(</span><span class="st">"01_bridge"</span><span class="op">)</span>,
    <span class="va">hmList_to_plot</span><span class="op">$</span><span class="va">bri_small</span>   |&gt; <span class="fu">prepareGGFreqPlot</span><span class="op">(</span><span class="st">"02_bridge_small"</span><span class="op">)</span>,
    <span class="va">hmList_to_plot</span><span class="op">$</span><span class="va">bri_long</span>    |&gt; <span class="fu">prepareGGFreqPlot</span><span class="op">(</span><span class="st">"02_bridge_long"</span><span class="op">)</span>
  <span class="op">)</span>
  <span class="va">tib</span><span class="op">$</span><span class="va">desc2</span> <span class="op">&lt;-</span> <span class="va">desc</span>
  <span class="va">tib</span>
<span class="op">}</span>

<span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/cbind.html" class="external-link">rbind</a></span><span class="op">(</span>
  <span class="fu">list_of_heatmaps_to_long_tibble</span><span class="op">(</span><span class="va">hmList_to_plot_noncoding</span>, <span class="st">"noncoding"</span><span class="op">)</span>,
  <span class="fu">list_of_heatmaps_to_long_tibble</span><span class="op">(</span><span class="va">hmList_to_plot_coding</span>, <span class="st">"coding"</span><span class="op">)</span>
<span class="op">)</span> |&gt; <span class="fu">ggplot</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">aes</span><span class="op">(</span><span class="va">pos</span>, <span class="va">freq</span>, col <span class="op">=</span> <span class="va">what</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">geom_line</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">facet_wrap</span><span class="op">(</span><span class="op">~</span><span class="va">desc2</span> <span class="op">+</span> <span class="va">desc</span>, nrow <span class="op">=</span> <span class="fl">2</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">theme_bw</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">ggtitle</span><span class="op">(</span><span class="st">"Aligned on regions left boundary. CNE is 50/48.  Okinawa - Osaka"</span><span class="op">)</span></code></pre></div>
<p><img src="GenomicFeatures_files/figure-html/bridge_width_features-1.png" width="700"></p>
</div>
<div class="section level4">
<h4 id="aligning-on-center-instead-of-border-">Aligning on center instead of border.<a class="anchor" aria-label="anchor" href="#aligning-on-center-instead-of-border-"></a>
</h4>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">hmList_to_plot.centered</span> <span class="op">&lt;-</span> <span class="fu">computePlotData</span><span class="op">(</span><span class="st">"Oki"</span>, <span class="st">"Osa"</span>, annots <span class="op">=</span> <span class="va">annots</span>, CE <span class="op">=</span> <span class="va">ces</span><span class="op">$</span><span class="va">ce__50__48</span>, tfbs <span class="op">=</span>  <span class="va">tfbs</span>, operons <span class="op">=</span> <span class="va">operons</span>, direction <span class="op">=</span> <span class="st">"mid"</span><span class="op">)</span>

<span class="op">(</span><span class="va">p_boundary</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/cbind.html" class="external-link">rbind</a></span><span class="op">(</span>
  <span class="va">hmList_to_plot.centered</span><span class="op">$</span><span class="va">aln_nonCoa</span>  |&gt; <span class="fu">prepareGGFreqPlot</span><span class="op">(</span><span class="st">"01_isolated_alns"</span><span class="op">)</span>,
  <span class="va">hmList_to_plot.centered</span><span class="op">$</span><span class="va">unmap</span>       |&gt; <span class="fu">prepareGGFreqPlot</span><span class="op">(</span><span class="st">"02_unmapped"</span><span class="op">)</span>,
  <span class="va">hmList_to_plot.centered</span><span class="op">$</span><span class="va">syn_block</span>   |&gt; <span class="fu">prepareGGFreqPlot</span><span class="op">(</span><span class="st">"03_colinear_reg"</span><span class="op">)</span>,
  <span class="va">hmList_to_plot.centered</span><span class="op">$</span><span class="va">bri</span>         |&gt; <span class="fu">prepareGGFreqPlot</span><span class="op">(</span><span class="st">"04_bridge"</span><span class="op">)</span>,
  <span class="va">hmList_to_plot.centered</span><span class="op">$</span><span class="va">aln_coa</span>     |&gt; <span class="fu">prepareGGFreqPlot</span><span class="op">(</span><span class="st">"05_colinear_alns"</span><span class="op">)</span>

<span class="op">)</span>|&gt; <span class="fu">ggplot</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">aes</span><span class="op">(</span><span class="va">pos</span>, <span class="va">freq</span>, col <span class="op">=</span> <span class="va">what</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">geom_line</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">facet_wrap</span><span class="op">(</span><span class="op">~</span><span class="va">desc</span>, nrow <span class="op">=</span> <span class="fl">1</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">theme_bw</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">ggtitle</span><span class="op">(</span><span class="st">"Aligned on center of regions. CNE is 50/48.  Okinawa - Osaka"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="GenomicFeatures_files/figure-html/genomic_features_byType.centered-1.svg" width="1440"></p>
</div>
<div class="section level4">
<h4 id="splice-junctions">Splice junctions<a class="anchor" aria-label="anchor" href="#splice-junctions"></a>
</h4>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/cbind.html" class="external-link">rbind</a></span><span class="op">(</span>
  <span class="fu">feature_coverage</span><span class="op">(</span><span class="va">gbs</span><span class="op">$</span><span class="va">Oki_Osa</span><span class="op">[</span><span class="va">gbs</span><span class="op">$</span><span class="va">Oki_Osa</span><span class="op">$</span><span class="va">nonCoa</span><span class="op">]</span>, <span class="fu"><a href="https://rdrr.io/pkg/GenomicFeatures/man/exonicParts.html" class="external-link">intronicParts</a></span><span class="op">(</span><span class="va">annots</span><span class="op">$</span><span class="va">Oki</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/intra-range-methods.html" class="external-link">resize</a></span><span class="op">(</span><span class="fl">1</span><span class="op">)</span>, window <span class="op">=</span> <span class="fl">1000</span>, lab <span class="op">=</span> <span class="st">"splicejunctions"</span>, dir <span class="op">=</span> <span class="st">"left"</span><span class="op">)</span> |&gt;
    <span class="fu">Heatmap_to_freq_tibble</span><span class="op">(</span><span class="st">"01_isolated"</span><span class="op">)</span>,
  <span class="fu">feature_coverage</span><span class="op">(</span><span class="va">unal</span><span class="op">$</span><span class="va">Oki_Osa</span><span class="op">[</span><span class="va">unal</span><span class="op">$</span><span class="va">Oki_Osa</span><span class="op">$</span><span class="va">nonCoa</span><span class="op">]</span>, <span class="fu"><a href="https://rdrr.io/pkg/GenomicFeatures/man/exonicParts.html" class="external-link">intronicParts</a></span><span class="op">(</span><span class="va">annots</span><span class="op">$</span><span class="va">Oki</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/intra-range-methods.html" class="external-link">resize</a></span><span class="op">(</span><span class="fl">1</span><span class="op">)</span>, window <span class="op">=</span> <span class="fl">1000</span>, lab <span class="op">=</span> <span class="st">"splicejunctions"</span>, dir <span class="op">=</span> <span class="st">"left"</span><span class="op">)</span> |&gt;
    <span class="fu">Heatmap_to_freq_tibble</span><span class="op">(</span><span class="st">"02_unaligned"</span><span class="op">)</span>,
  <span class="fu">feature_coverage</span><span class="op">(</span><span class="va">coa</span><span class="op">$</span><span class="va">Oki_Osa</span><span class="op">[</span><span class="op">!</span><span class="va">coa</span><span class="op">$</span><span class="va">Oki_Osa</span><span class="op">$</span><span class="va">nonCoa</span><span class="op">]</span>, <span class="fu"><a href="https://rdrr.io/pkg/GenomicFeatures/man/exonicParts.html" class="external-link">intronicParts</a></span><span class="op">(</span><span class="va">annots</span><span class="op">$</span><span class="va">Oki</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/intra-range-methods.html" class="external-link">resize</a></span><span class="op">(</span><span class="fl">1</span><span class="op">)</span>, window <span class="op">=</span> <span class="fl">1000</span>, lab <span class="op">=</span> <span class="st">"splicejunctions"</span>, dir <span class="op">=</span> <span class="st">"left"</span><span class="op">)</span> |&gt;
    <span class="fu">Heatmap_to_freq_tibble</span><span class="op">(</span><span class="st">"03_colinear_reg"</span><span class="op">)</span>,
  <span class="fu">feature_coverage</span><span class="op">(</span><span class="va">bri</span><span class="op">$</span><span class="va">Oki_Osa</span>, <span class="fu"><a href="https://rdrr.io/pkg/GenomicFeatures/man/exonicParts.html" class="external-link">intronicParts</a></span><span class="op">(</span><span class="va">annots</span><span class="op">$</span><span class="va">Oki</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/intra-range-methods.html" class="external-link">resize</a></span><span class="op">(</span><span class="fl">1</span><span class="op">)</span>, window <span class="op">=</span> <span class="fl">1000</span>, lab <span class="op">=</span> <span class="st">"splicejunctions"</span>, dir <span class="op">=</span> <span class="st">"left"</span><span class="op">)</span> |&gt;
    <span class="fu">Heatmap_to_freq_tibble</span><span class="op">(</span><span class="st">"04_bridge"</span><span class="op">)</span> 
<span class="op">)</span> |&gt; <span class="fu">ggplot</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu">aes</span><span class="op">(</span><span class="va">pos</span>, <span class="va">freq</span>, col <span class="op">=</span> <span class="va">desc</span><span class="op">)</span> <span class="op">+</span> <span class="fu">geom_line</span><span class="op">(</span><span class="op">)</span></code></pre></div>
<p><img src="GenomicFeatures_files/figure-html/splice_junctions-1.png" width="700"></p>
</div>
<div class="section level4">
<h4 id="pwm">PWM<a class="anchor" aria-label="anchor" href="#pwm"></a>
</h4>
<p>The PWM motif, which is rare in comparison to the other features, follows a profile that bears resemblance to the repeat profile.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">gadems.invLeftGap</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="st">"gadems.invLeftGap.Rda"</span><span class="op">)</span>
<span class="va">pwm</span> <span class="op">&lt;-</span> <span class="va">gadems.invLeftGap</span><span class="op">$</span><span class="va">Oki_Osa</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span>
<span class="va">pwm</span><span class="op">@</span><span class="va">consensus</span></code></pre></div>
<pre><code><span class="co">## [1] "rArAAGCCGCdwAGCsGCw"</span></code></pre>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># pwmHits &lt;- matchPWM(pwm = sapply(pwm@alignList, \(x) x@seq) |&gt; as("DNAStringSet") |&gt; PWM(), genomes$Oki) # Gives different scores...</span>
<span class="va">pwmHits</span> <span class="op">&lt;-</span> <span class="fu">matchPWM</span><span class="op">(</span>pwm <span class="op">=</span> <span class="fu">rGADEM</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/rGADEM/man/gadem-class.html" class="external-link">getPWM</a></span><span class="op">(</span><span class="va">pwm</span><span class="op">)</span>, <span class="va">genomes</span><span class="op">$</span><span class="va">OKI2018.I69</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/r/base/warning.html" class="external-link">suppressWarnings</a></span><span class="op">(</span><span class="op">)</span>
<span class="co"># Suppressing warnings like:</span>
<span class="co"># Warning messages:</span>
<span class="co"># 1: In .Call2("XString_match_PWM", pwm, subject, min.score, count.only,  :</span>
<span class="co">#   'subject' contains letters not in [ACGT] ==&gt; assigned weight 0 to them</span>
<span class="co"># </span>
<span class="co"># plotHeatmapMeta(list(feature_coverage(current_GB_wide,   pwmHits, win = 1e3, lab = "wide",    direction = "left")))</span>
<span class="co"># plotHeatmapMeta(list(feature_coverage(current_GB_narrow, pwmHits, win = 1e3, lab = "narrow",  direction = "left")))</span></code></pre></div>
</div>
<div class="section level4">
<h4 id="question">Question<a class="anchor" aria-label="anchor" href="#question"></a>
</h4>
<p>in <code><a href="../articles/ColinearityInterruptors.html">vignette("ColinearityInterruptors", package = "OikScrambling")</a></code>, we saw that some uncoalesced regions map uniquely and are well conserved across genomes. What are they… enhancers ?</p>
</div>
</div>
<div class="section level3">
<h3 id="gene-feature-coverage-around-pwm-hits">Gene feature coverage around PWM hits<a class="anchor" aria-label="anchor" href="#gene-feature-coverage-around-pwm-hits"></a>
</h3>
<p>Let’s look at coverage around PWM hits of the AAGCsGCwwmkCGrCTTyn motif</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">current_GB</span> <span class="op">&lt;-</span> <span class="va">pwmHits</span>

<span class="va">hmList_PWM_OKI</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="op">)</span>
<span class="va">hmList_PWM_OKI</span><span class="op">$</span><span class="va">genes</span>     <span class="op">&lt;-</span> <span class="fu">feature_coverage</span><span class="op">(</span><span class="va">current_GB</span>, <span class="fu"><a href="https://rdrr.io/pkg/GenomicFeatures/man/transcripts.html" class="external-link">genes</a></span><span class="op">(</span><span class="va">annots</span><span class="op">$</span><span class="va">Oki</span><span class="op">)</span>,                         win <span class="op">=</span> <span class="fl">2000</span>, lab <span class="op">=</span> <span class="st">"Oki genes"</span>, direction <span class="op">=</span> <span class="st">"left"</span><span class="op">)</span>
<span class="va">hmList_PWM_OKI</span><span class="op">$</span><span class="va">promoters</span> <span class="op">&lt;-</span> <span class="fu">feature_coverage</span><span class="op">(</span><span class="va">current_GB</span>, <span class="fu"><a href="https://rdrr.io/pkg/GenomicFeatures/man/transcripts.html" class="external-link">promoters</a></span><span class="op">(</span><span class="va">annots</span><span class="op">$</span><span class="va">Oki</span><span class="op">)</span>,                     win <span class="op">=</span> <span class="fl">2000</span>, lab <span class="op">=</span> <span class="st">"Oki promoters"</span>, direction <span class="op">=</span> <span class="st">"left"</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Warning in valid.GenomicRanges.seqinfo(x, suggest.trim = TRUE): GRanges object contains 4 out-of-bound ranges located on sequences</span>
<span class="co">##   chr1, chrUn_1, and chrUn_10. Note that ranges located on a sequence</span>
<span class="co">##   whose length is unknown (NA) or on a circular sequence are not</span>
<span class="co">##   considered out-of-bound (use seqlengths() and isCircular() to get the</span>
<span class="co">##   lengths and circularity flags of the underlying sequences). You can use</span>
<span class="co">##   trim() to trim these ranges. See ?`trim,GenomicRanges-method` for more</span>
<span class="co">##   information.</span></code></pre>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">hmList_PWM_OKI</span><span class="op">$</span><span class="va">exons</span>     <span class="op">&lt;-</span> <span class="fu">feature_coverage</span><span class="op">(</span><span class="va">current_GB</span>, <span class="fu"><a href="https://rdrr.io/pkg/GenomicFeatures/man/exonicParts.html" class="external-link">exonicParts</a></span><span class="op">(</span><span class="va">annots</span><span class="op">$</span><span class="va">Oki</span><span class="op">)</span>,                   win <span class="op">=</span> <span class="fl">2000</span>, lab <span class="op">=</span> <span class="st">"Oki exons"</span>, direction <span class="op">=</span> <span class="st">"left"</span><span class="op">)</span>
<span class="va">hmList_PWM_OKI</span><span class="op">$</span><span class="va">introns</span>   <span class="op">&lt;-</span> <span class="fu">feature_coverage</span><span class="op">(</span><span class="va">current_GB</span>, <span class="fu"><a href="https://rdrr.io/pkg/GenomicFeatures/man/exonicParts.html" class="external-link">intronicParts</a></span><span class="op">(</span><span class="va">annots</span><span class="op">$</span><span class="va">Oki</span><span class="op">)</span>,                 win <span class="op">=</span> <span class="fl">2000</span>, lab <span class="op">=</span> <span class="st">"Oki introns"</span>, direction <span class="op">=</span> <span class="st">"left"</span><span class="op">)</span>
<span class="va">hmList_PWM_OKI</span><span class="op">$</span><span class="va">cds</span>       <span class="op">&lt;-</span> <span class="fu">feature_coverage</span><span class="op">(</span><span class="va">current_GB</span>, <span class="fu"><a href="https://rdrr.io/pkg/GenomicFeatures/man/transcripts.html" class="external-link">cds</a></span><span class="op">(</span><span class="va">annots</span><span class="op">$</span><span class="va">Oki</span><span class="op">)</span>,                           win <span class="op">=</span> <span class="fl">2000</span>, lab <span class="op">=</span> <span class="st">"Oki cds"</span>, direction <span class="op">=</span> <span class="st">"left"</span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/pkg/heatmaps/man/plotHeatmapMeta.html" class="external-link">plotHeatmapMeta</a></span><span class="op">(</span><span class="va">hmList_PWM_OKI</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"genes"</span>, <span class="st">"exons"</span>, <span class="st">"cds"</span><span class="op">)</span><span class="op">]</span><span class="op">)</span> <span class="co"># Show they look alike.  We keep "exons"</span></code></pre></div>
<p><img src="GenomicFeatures_files/figure-html/genomic_features_pwmHits-1.png" width="700"></p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/heatmaps/man/plotHeatmapMeta.html" class="external-link">plotHeatmapMeta</a></span><span class="op">(</span><span class="va">hmList_PWM_OKI</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"promoters"</span>, <span class="st">"exons"</span>, <span class="st">"introns"</span><span class="op">)</span><span class="op">]</span><span class="op">)</span></code></pre></div>
<p><img src="GenomicFeatures_files/figure-html/genomic_features_pwmHits-2.png" width="700"></p>
<p>They are enriched in introns…</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># </span>
<span class="co"># # Sandbox to make sanity checks.</span>
<span class="co"># </span>
<span class="co"># a &lt;- coa$Oki_Osa</span>
<span class="co"># b &lt;- unlist(fiveUTRsByTranscript(annots$Oki))</span>
<span class="co"># </span>
<span class="co"># hm_a &lt;- feature_coverage(a, b, win = 2000, lab = "All ranges", direction = "left")</span>
<span class="co"># hm_b &lt;- feature_coverage(subsetByOverlaps(a,  b), b, win = 2000, lab = "Subsetted ranges", direction = "left")</span>
<span class="co"># </span>
<span class="co"># plotHeatmapMeta(list(hm_a))</span>
<span class="co"># plotHeatmapMeta(list(hm_b))</span></code></pre></div>
</div>
<div class="section level3">
<h3 id="breakpoint-coverage-around-pwm-hits">Breakpoint coverage around PWM hits<a class="anchor" aria-label="anchor" href="#breakpoint-coverage-around-pwm-hits"></a>
</h3>
<p>TODO: triplecheck <code>feature_coverage</code>, it is old…</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">current_GB</span> <span class="op">&lt;-</span> <span class="va">pwmHits</span>

<span class="va">hmList_PWM_OKI</span><span class="op">$</span><span class="va">gbOsa</span>     <span class="op">&lt;-</span> <span class="fu">feature_coverage</span><span class="op">(</span><span class="va">current_GB</span>, <span class="va">gbs</span><span class="op">$</span><span class="va">Oki_Osa</span>,  win <span class="op">=</span> <span class="fl">5000</span>, lab <span class="op">=</span> <span class="st">"Oki-Osa"</span>, direction <span class="op">=</span> <span class="st">"left"</span><span class="op">)</span>
<span class="va">hmList_PWM_OKI</span><span class="op">$</span><span class="va">gbBar</span>     <span class="op">&lt;-</span> <span class="fu">feature_coverage</span><span class="op">(</span><span class="va">current_GB</span>, <span class="va">gbs</span><span class="op">$</span><span class="va">Oki_Bar</span>,  win <span class="op">=</span> <span class="fl">5000</span>, lab <span class="op">=</span> <span class="st">"Oki-Bar"</span>, direction <span class="op">=</span> <span class="st">"left"</span><span class="op">)</span>
<span class="va">hmList_PWM_OKI</span><span class="op">$</span><span class="va">gbKum</span>     <span class="op">&lt;-</span> <span class="fu">feature_coverage</span><span class="op">(</span><span class="va">current_GB</span>, <span class="va">gbs</span><span class="op">$</span><span class="va">Oki_Kum</span>,  win <span class="op">=</span> <span class="fl">5000</span>, lab <span class="op">=</span> <span class="st">"Oki-Kum"</span>, direction <span class="op">=</span> <span class="st">"left"</span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/pkg/heatmaps/man/plotHeatmapMeta.html" class="external-link">plotHeatmapMeta</a></span><span class="op">(</span><span class="va">hmList_PWM_OKI</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"gbOsa"</span>, <span class="st">"gbBar"</span>, <span class="st">"gbKum"</span><span class="op">)</span><span class="op">]</span><span class="op">)</span></code></pre></div>
<p><img src="GenomicFeatures_files/figure-html/genomic_features_pwmHits2-1.png" width="700"></p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">hmList_PWM_OKI</span><span class="op">$</span><span class="va">coOsa</span>     <span class="op">&lt;-</span> <span class="fu">feature_coverage</span><span class="op">(</span><span class="va">current_GB</span>, <span class="va">coa</span><span class="op">$</span><span class="va">Oki_Osa</span>,  win <span class="op">=</span> <span class="fl">5000</span>, lab <span class="op">=</span> <span class="st">"Oki-Osa"</span>, direction <span class="op">=</span> <span class="st">"left"</span><span class="op">)</span>
<span class="va">hmList_PWM_OKI</span><span class="op">$</span><span class="va">coBar</span>     <span class="op">&lt;-</span> <span class="fu">feature_coverage</span><span class="op">(</span><span class="va">current_GB</span>, <span class="va">coa</span><span class="op">$</span><span class="va">Oki_Bar</span>,  win <span class="op">=</span> <span class="fl">5000</span>, lab <span class="op">=</span> <span class="st">"Oki-Bar"</span>, direction <span class="op">=</span> <span class="st">"left"</span><span class="op">)</span>
<span class="va">hmList_PWM_OKI</span><span class="op">$</span><span class="va">coKum</span>     <span class="op">&lt;-</span> <span class="fu">feature_coverage</span><span class="op">(</span><span class="va">current_GB</span>, <span class="va">coa</span><span class="op">$</span><span class="va">Oki_Kum</span>,  win <span class="op">=</span> <span class="fl">5000</span>, lab <span class="op">=</span> <span class="st">"Oki-Kum"</span>, direction <span class="op">=</span> <span class="st">"left"</span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/pkg/heatmaps/man/plotHeatmapMeta.html" class="external-link">plotHeatmapMeta</a></span><span class="op">(</span><span class="va">hmList_PWM_OKI</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"coOsa"</span>, <span class="st">"coBar"</span>, <span class="st">"coKum"</span><span class="op">)</span><span class="op">]</span><span class="op">)</span></code></pre></div>
<p><img src="GenomicFeatures_files/figure-html/genomic_features_pwmHits2-2.png" width="700"></p>
</div>
<div class="section level3">
<h3 id="okinawa-genome-coverage">Okinawa genome coverage<a class="anchor" aria-label="anchor" href="#okinawa-genome-coverage"></a>
</h3>
<p>As previously discussed, low coverage over an alignment stop could lower the likelihood of it being considered as a breakpoint. We have per-base coverage depth information for the Okinawan genome (<code>Oki_cov_pb</code>). However, the coverage of this particular assembly is quite good. In fact, we can investigate the coverge over alignment stops from the information obtained using <code>master_bp_analysis</code>.</p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># fin_Oki &lt;- fin_gr_O_Oki[[2]]</span>
<span class="co"># min_cov &lt;- min(min(fin_Oki$left_cov_pb), min(fin_Oki$right_cov_pb)) # minimum coverage over an alignment stop</span>
<span class="co"># min_cov</span>
<span class="co"># length(fin_Oki[fin_Oki$left_cov_pb == min_cov]) + length(fin_Oki[fin_Oki$right_cov_pb == min_cov]) # how many of the minimum coverage is observed</span>
<span class="co"># length(fin_Oki[fin_Oki$left_cov_pb &lt;= 50]) + length(fin_Oki[fin_Oki$right_cov_pb &lt;= 50]) # how many alignment stops have a coverage of less than or euqal to 50</span></code></pre></div>
<p>We may choose to kick out the one alignment stop for which there is no coverage. However, only 15 out of 34572 alignment stops have a coverage of less than or equal to 50, meaning that it would be hard to exclude more than just a few alignment stops using coverage information for breakpoint analysis.</p>
</div>
</div>
<div class="section level2">
<h2 id="operons-and-genes">Operons and genes<a class="anchor" aria-label="anchor" href="#operons-and-genes"></a>
</h2>
<p>Do alignments often cross gene or operon boundaries ?</p>
<div class="section level3">
<h3 id="operons-boundaries">Operons boundaries<a class="anchor" aria-label="anchor" href="#operons-boundaries"></a>
</h3>
<p>Operon boundaries tend to coincide with alignment boundaries and the largest shift in frequency is for colinear alignments. Baseline frequency of colinear regions is higher than 0.6 on both side of operon boundaries.</p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">feature_coverage_on_operons</span> <span class="op">&lt;-</span> <span class="op">{</span>
  <span class="kw">function</span><span class="op">(</span><span class="va">feat</span>, <span class="va">desc</span><span class="op">)</span>
    <span class="fu">feature_coverage</span><span class="op">(</span><span class="va">operons</span><span class="op">$</span><span class="va">Oki</span>, <span class="va">feat</span>, window <span class="op">=</span> <span class="fl">1000</span>, lab <span class="op">=</span> <span class="st">""</span>, dir <span class="op">=</span> <span class="st">"left"</span><span class="op">)</span> |&gt; <span class="fu">Heatmap_to_freq_tibble</span><span class="op">(</span><span class="va">desc</span><span class="op">)</span>
<span class="op">}</span>

<span class="va">pairname</span> <span class="op">&lt;-</span> <span class="st">"Oki_Bar"</span>
<span class="va">pairname</span> <span class="op">&lt;-</span> <span class="st">"Oki_Osa"</span>

<span class="va">p_op_scrambling</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/cbind.html" class="external-link">rbind</a></span><span class="op">(</span>
  <span class="fu">feature_coverage_on_operons</span><span class="op">(</span><span class="va">gbs</span><span class="op">[[</span><span class="va">pairname</span><span class="op">]</span><span class="op">]</span><span class="op">[</span><span class="va">gbs</span><span class="op">[[</span><span class="va">pairname</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">nonCoa</span><span class="op">]</span>, <span class="st">"01_isolated"</span><span class="op">)</span>,
  <span class="fu">feature_coverage_on_operons</span><span class="op">(</span><span class="va">unal</span><span class="op">[[</span><span class="va">pairname</span><span class="op">]</span><span class="op">]</span><span class="op">[</span><span class="va">unal</span><span class="op">[[</span><span class="va">pairname</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">nonCoa</span><span class="op">]</span>, <span class="st">"02_unaligned"</span><span class="op">)</span>,
  <span class="fu">feature_coverage_on_operons</span><span class="op">(</span><span class="va">gbs</span><span class="op">[[</span><span class="va">pairname</span><span class="op">]</span><span class="op">]</span><span class="op">[</span><span class="op">!</span><span class="va">gbs</span><span class="op">[[</span><span class="va">pairname</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">nonCoa</span><span class="op">]</span>, <span class="st">"03_colinear_aln"</span><span class="op">)</span>,
  <span class="co">#feature_coverage_on_operons(subsetByOverlaps(gbs$Oki_Osa[!gbs$Oki_Osa$nonCoa], granges(coa$Oki_Osa[!coa$Oki_Osa$nonCoa] - 1),type = "within"), "03b_colinear_aln_intern"),</span>
  <span class="co">#feature_coverage_on_operons(subsetByOverlaps(gbs$Oki_Osa[!gbs$Oki_Osa$nonCoa], unal$Oki_Osa[unal$Oki_Osa$nonCoa] + 1), "03c_colinear_aln_extern"),</span>

  <span class="fu">feature_coverage_on_operons</span><span class="op">(</span><span class="va">bri</span><span class="op">[[</span><span class="va">pairname</span><span class="op">]</span><span class="op">]</span>, <span class="st">"04_bridge"</span><span class="op">)</span>,
  <span class="fu">feature_coverage_on_operons</span><span class="op">(</span><span class="va">coa</span><span class="op">[[</span><span class="va">pairname</span><span class="op">]</span><span class="op">]</span><span class="op">[</span><span class="op">!</span><span class="va">coa</span><span class="op">[[</span><span class="va">pairname</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">nonCoa</span><span class="op">]</span>, <span class="st">"05_colinear_reg"</span><span class="op">)</span>
<span class="op">)</span> |&gt; <span class="fu">ggplot</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu">aes</span><span class="op">(</span><span class="va">pos</span>, <span class="va">freq</span>, col <span class="op">=</span> <span class="va">desc</span><span class="op">)</span> <span class="op">+</span> <span class="fu">geom_line</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu">ggtitle</span><span class="op">(</span><span class="st">"operons"</span>, subtitle <span class="op">=</span> <span class="st">"n = 2017"</span><span class="op">)</span> <span class="op">+</span> <span class="fu">theme_bw</span><span class="op">(</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level3">
<h3 id="gene-boundaries-inside-operons">Gene boundaries inside operons<a class="anchor" aria-label="anchor" href="#gene-boundaries-inside-operons"></a>
</h3>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Shrinking operons of 1 base and requesting full overlap</span>
<span class="co"># in order to exclude the genes at the boundaries.</span>
<span class="va">genes_inside_operons</span>  <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/findOverlaps-methods.html" class="external-link">subsetByOverlaps</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/GenomicFeatures/man/transcripts.html" class="external-link">genes</a></span><span class="op">(</span><span class="va">annots</span><span class="op">$</span><span class="va">Oki</span><span class="op">)</span>,           <span class="va">operons</span><span class="op">$</span><span class="va">Oki</span> <span class="op">-</span><span class="fl">1</span>, type <span class="op">=</span> <span class="st">"within"</span><span class="op">)</span>
<span class="va">genes_outside_operons</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/findOverlaps-methods.html" class="external-link">subsetByOverlaps</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/GenomicFeatures/man/transcripts.html" class="external-link">genes</a></span><span class="op">(</span><span class="va">annots</span><span class="op">$</span><span class="va">Oki</span><span class="op">)</span>, <span class="fu">cleanGaps</span><span class="op">(</span><span class="va">operons</span><span class="op">$</span><span class="va">Oki</span><span class="op">)</span>,   type <span class="op">=</span> <span class="st">"within"</span><span class="op">)</span>

<span class="va">feature_coverage_on_genes</span> <span class="op">&lt;-</span> <span class="op">{</span>
  <span class="kw">function</span><span class="op">(</span><span class="va">feat</span>, <span class="va">desc</span><span class="op">)</span>
    <span class="fu">feature_coverage</span><span class="op">(</span><span class="va">genes_inside_operons</span>, <span class="va">feat</span>, window <span class="op">=</span> <span class="fl">1000</span>, lab <span class="op">=</span> <span class="st">""</span>, dir <span class="op">=</span> <span class="st">"left"</span><span class="op">)</span> |&gt; <span class="fu">Heatmap_to_freq_tibble</span><span class="op">(</span><span class="va">desc</span><span class="op">)</span>
<span class="op">}</span>

<span class="va">p_gene_int_scrambling</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/cbind.html" class="external-link">rbind</a></span><span class="op">(</span>
  <span class="fu">feature_coverage_on_genes</span><span class="op">(</span><span class="va">gbs</span><span class="op">$</span><span class="va">Oki_Osa</span><span class="op">[</span><span class="va">gbs</span><span class="op">$</span><span class="va">Oki_Osa</span><span class="op">$</span><span class="va">nonCoa</span><span class="op">]</span>, <span class="st">"01_isolated"</span><span class="op">)</span>,
  <span class="fu">feature_coverage_on_genes</span><span class="op">(</span><span class="va">unal</span><span class="op">$</span><span class="va">Oki_Osa</span><span class="op">[</span><span class="va">unal</span><span class="op">$</span><span class="va">Oki_Osa</span><span class="op">$</span><span class="va">nonCoa</span><span class="op">]</span>, <span class="st">"02_unmapped"</span><span class="op">)</span>,
  <span class="fu">feature_coverage_on_genes</span><span class="op">(</span><span class="va">gbs</span><span class="op">$</span><span class="va">Oki_Osa</span><span class="op">[</span><span class="op">!</span><span class="va">gbs</span><span class="op">$</span><span class="va">Oki_Osa</span><span class="op">$</span><span class="va">nonCoa</span><span class="op">]</span>, <span class="st">"03_colinear_aln"</span><span class="op">)</span>,
  <span class="fu">feature_coverage_on_genes</span><span class="op">(</span><span class="va">bri</span><span class="op">$</span><span class="va">Oki_Osa</span>, <span class="st">"04_bridge"</span><span class="op">)</span>,
  <span class="fu">feature_coverage_on_genes</span><span class="op">(</span><span class="va">coa</span><span class="op">$</span><span class="va">Oki_Osa</span><span class="op">[</span><span class="op">!</span><span class="va">coa</span><span class="op">$</span><span class="va">Oki_Osa</span><span class="op">$</span><span class="va">nonCoa</span><span class="op">]</span>, <span class="st">"05_colinear_reg"</span><span class="op">)</span>
<span class="op">)</span> |&gt; <span class="fu">ggplot</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu">aes</span><span class="op">(</span><span class="va">pos</span>, <span class="va">freq</span>, col <span class="op">=</span> <span class="va">desc</span><span class="op">)</span> <span class="op">+</span> <span class="fu">geom_line</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu">ggtitle</span><span class="op">(</span><span class="st">"genes_internal_to_operons"</span>, subtitle <span class="op">=</span> <span class="st">"n = 1348"</span><span class="op">)</span> <span class="op">+</span> <span class="fu">theme_bw</span><span class="op">(</span><span class="op">)</span>

<span class="va">feature_coverage_on_genes_outside</span> <span class="op">&lt;-</span> <span class="op">{</span>
  <span class="kw">function</span><span class="op">(</span><span class="va">feat</span>, <span class="va">desc</span><span class="op">)</span>
    <span class="fu">feature_coverage</span><span class="op">(</span><span class="va">genes_outside_operons</span>, <span class="va">feat</span>, window <span class="op">=</span> <span class="fl">1000</span>, lab <span class="op">=</span> <span class="st">""</span>, dir <span class="op">=</span> <span class="st">"left"</span><span class="op">)</span> |&gt; <span class="fu">Heatmap_to_freq_tibble</span><span class="op">(</span><span class="va">desc</span><span class="op">)</span>
<span class="op">}</span>

<span class="va">p_gene_ext_scrambling</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/cbind.html" class="external-link">rbind</a></span><span class="op">(</span>
  <span class="fu">feature_coverage_on_genes_outside</span><span class="op">(</span><span class="va">gbs</span><span class="op">$</span><span class="va">Oki_Osa</span><span class="op">[</span><span class="va">gbs</span><span class="op">$</span><span class="va">Oki_Osa</span><span class="op">$</span><span class="va">nonCoa</span><span class="op">]</span>, <span class="st">"01_isolated"</span><span class="op">)</span>,
  <span class="fu">feature_coverage_on_genes_outside</span><span class="op">(</span><span class="va">unal</span><span class="op">$</span><span class="va">Oki_Osa</span><span class="op">[</span><span class="va">unal</span><span class="op">$</span><span class="va">Oki_Osa</span><span class="op">$</span><span class="va">nonCoa</span><span class="op">]</span>, <span class="st">"02_unmapped"</span><span class="op">)</span>,
  <span class="fu">feature_coverage_on_genes_outside</span><span class="op">(</span><span class="va">gbs</span><span class="op">$</span><span class="va">Oki_Osa</span><span class="op">[</span><span class="op">!</span><span class="va">gbs</span><span class="op">$</span><span class="va">Oki_Osa</span><span class="op">$</span><span class="va">nonCoa</span><span class="op">]</span>, <span class="st">"03_colinear_aln"</span><span class="op">)</span>,
  <span class="fu">feature_coverage_on_genes_outside</span><span class="op">(</span><span class="va">bri</span><span class="op">$</span><span class="va">Oki_Osa</span>, <span class="st">"04_bridge"</span><span class="op">)</span>,
  <span class="fu">feature_coverage_on_genes_outside</span><span class="op">(</span><span class="va">coa</span><span class="op">$</span><span class="va">Oki_Osa</span><span class="op">[</span><span class="op">!</span><span class="va">coa</span><span class="op">$</span><span class="va">Oki_Osa</span><span class="op">$</span><span class="va">nonCoa</span><span class="op">]</span>, <span class="st">"05_colinear_reg"</span><span class="op">)</span>
<span class="op">)</span> |&gt; <span class="fu">ggplot</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu">aes</span><span class="op">(</span><span class="va">pos</span>, <span class="va">freq</span>, col <span class="op">=</span> <span class="va">desc</span><span class="op">)</span> <span class="op">+</span> <span class="fu">geom_line</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu">ggtitle</span><span class="op">(</span><span class="st">"genes_outside_operons"</span>, subtitle <span class="op">=</span> <span class="st">"n = 11688"</span><span class="op">)</span> <span class="op">+</span> <span class="fu">theme_bw</span><span class="op">(</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level3">
<h3 id="summary-plot">Summary plot<a class="anchor" aria-label="anchor" href="#summary-plot"></a>
</h3>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">p_gene_ext_scrambling</span> <span class="op">+</span> <span class="va">p_op_scrambling</span> <span class="op">+</span> <span class="va">p_gene_int_scrambling</span> <span class="op">+</span> <span class="fu"><a href="https://patchwork.data-imaginist.com/reference/plot_layout.html" class="external-link">plot_layout</a></span><span class="op">(</span>nrow <span class="op">=</span> <span class="fl">1</span>, guides <span class="op">=</span> <span class="st">'collect'</span><span class="op">)</span></code></pre></div>
<p><img src="GenomicFeatures_files/figure-html/operon_scrambling-1.svg" width="1440"></p>
</div>
<div class="section level3">
<h3 id="operon-conservation">Operon conservation<a class="anchor" aria-label="anchor" href="#operon-conservation"></a>
</h3>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Distribution of operon lengths</span>
<span class="va">operons</span><span class="op">$</span><span class="va">Oki</span><span class="op">$</span><span class="va">n</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/table.html" class="external-link">table</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## </span>
<span class="co">##    2    3    4    5    6    7    8    9   10   11   14 </span>
<span class="co">## 1943  647  259  134   69   33   19   10    6    3    1</span></code></pre>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># An operon is scrambled if it overlaps with an unaligned noncoalesced (= unmapped) region</span>
<span class="va">scrambled_operons</span>     <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/findOverlaps-methods.html" class="external-link">subsetByOverlaps</a></span><span class="op">(</span><span class="va">operons</span><span class="op">$</span><span class="va">Oki</span>, <span class="va">unal</span><span class="op">$</span><span class="va">Oki_Osa</span><span class="op">[</span><span class="va">unal</span><span class="op">$</span><span class="va">Oki_Osa</span><span class="op">$</span><span class="va">nonCoa</span><span class="op">]</span><span class="op">)</span>
<span class="va">not_scrambled_operons</span> <span class="op">&lt;-</span> <span class="va">operons</span><span class="op">$</span><span class="va">Oki</span><span class="op">[</span><span class="op">!</span> <span class="va">operons</span><span class="op">$</span><span class="va">Oki</span> <span class="op"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html" class="external-link">%in%</a></span> <span class="va">scrambled_operons</span> <span class="op">]</span>

<span class="co"># Distribution of operon lengths (scrambled)</span>
<span class="va">scrambled_operons</span><span class="op">$</span><span class="va">n</span>     |&gt; <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/table.html" class="external-link">table</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## </span>
<span class="co">##   2   3   4   5   6   7   8   9  10  11 </span>
<span class="co">## 905 298 106  44  31   9   7   6   4   3</span></code></pre>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">not_scrambled_operons</span><span class="op">$</span><span class="va">n</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/table.html" class="external-link">table</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## </span>
<span class="co">##    2    3    4    5    6    7    8    9   10   14 </span>
<span class="co">## 1038  349  153   90   38   24   12    4    2    1</span></code></pre>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Example of long scrambled operons</span>
<span class="va">scrambled_operons</span><span class="op">[</span><span class="va">scrambled_operons</span><span class="op">$</span><span class="va">n</span> <span class="op">&gt;</span> <span class="fl">5</span><span class="op">]</span></code></pre></div>
<pre><code><span class="co">## GRanges object with 60 ranges and 2 metadata columns:</span>
<span class="co">##        seqnames            ranges strand |         n                  gene_id</span>
<span class="co">##           &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt;          &lt;CharacterList&gt;</span>
<span class="co">##    [1]     chr1   5715727-5730794      + |         7    g1521,g1522,g1523,...</span>
<span class="co">##    [2]     chr1   6147428-6157837      + |         6    g1613,g1614,g1615,...</span>
<span class="co">##    [3]     chr1   9809914-9836205      + |        10    g2620,g2621,g2622,...</span>
<span class="co">##    [4]     chr1 10022216-10035952      + |         7    g2668,g2669,g2670,...</span>
<span class="co">##    [5]     chr1 10547772-10569235      - |         9    g2832,g2833,g2834,...</span>
<span class="co">##    ...      ...               ...    ... .       ...                      ...</span>
<span class="co">##   [56]      XSR   6611822-6622862      + |         7 g15207,g15208,g15209,...</span>
<span class="co">##   [57]      XSR   6843015-6861410      + |         8 g15282,g15283,g15284,...</span>
<span class="co">##   [58]      XSR   7875989-7897781      - |         7 g15606,g15607,g15608,...</span>
<span class="co">##   [59]      XSR 10693937-10705412      + |         6 g16499,g16500,g16501,...</span>
<span class="co">##   [60]      XSR 12736159-12757464      - |         8 g17016,g17017,g17018,...</span>
<span class="co">##   -------</span>
<span class="co">##   seqinfo: 19 sequences from OKI2018.I69 genome</span></code></pre>
<p>In comparison: genes</p>
<div class="sourceCode" id="cb38"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">all_genes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicFeatures/man/transcripts.html" class="external-link">genes</a></span><span class="op">(</span><span class="va">annots</span><span class="op">$</span><span class="va">Oki</span><span class="op">)</span>
<span class="va">scrambled_genes</span> <span class="op">&lt;-</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/findOverlaps-methods.html" class="external-link">subsetByOverlaps</a></span><span class="op">(</span><span class="va">all_genes</span>, <span class="va">unal</span><span class="op">$</span><span class="va">Oki_Osa</span><span class="op">[</span><span class="va">unal</span><span class="op">$</span><span class="va">Oki_Osa</span><span class="op">$</span><span class="va">nonCoa</span><span class="op">]</span><span class="op">)</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/sort.html" class="external-link">sort</a></span><span class="op">(</span>i<span class="op">=</span><span class="cn">F</span><span class="op">)</span> |&gt; <span class="fu">flagLongShort</span><span class="op">(</span><span class="va">longShort</span><span class="op">$</span><span class="va">OKI2018.I69</span><span class="op">)</span>
<span class="va">not_scrambled_genes</span> <span class="op">&lt;-</span> <span class="va">all_genes</span><span class="op">[</span><span class="op">!</span> <span class="va">all_genes</span> <span class="op"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html" class="external-link">%in%</a></span> <span class="va">scrambled_genes</span><span class="op">]</span>                  |&gt; <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/sort.html" class="external-link">sort</a></span><span class="op">(</span>i<span class="op">=</span><span class="cn">F</span><span class="op">)</span> |&gt; <span class="fu">flagLongShort</span><span class="op">(</span><span class="va">longShort</span><span class="op">$</span><span class="va">OKI2018.I69</span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">scrambled_genes</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## [1] 5294</span></code></pre>
<div class="sourceCode" id="cb40"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">all_genes</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## [1] 17291</span></code></pre>
<div class="sourceCode" id="cb42"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">scr_g_tbl</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/table.html" class="external-link">table</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqnames</a></span><span class="op">(</span>    <span class="va">scrambled_genes</span><span class="op">)</span>, <span class="va">scrambled_genes</span><span class="op">$</span><span class="va">Arm</span><span class="op">)</span><span class="op">)</span>
<span class="va">nsc_g_tbl</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/table.html" class="external-link">table</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqnames</a></span><span class="op">(</span><span class="va">not_scrambled_genes</span><span class="op">)</span>, <span class="va">not_scrambled_genes</span><span class="op">$</span><span class="va">Arm</span><span class="op">)</span><span class="op">)</span>

<span class="va">Oki_Osa_scr_ratio_long_arm</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span>
  <span class="va">scr_g_tbl</span><span class="op">[[</span><span class="st">"chr1 long"</span><span class="op">]</span><span class="op">]</span> <span class="op">/</span> <span class="op">(</span> <span class="va">scr_g_tbl</span><span class="op">[[</span><span class="st">"chr1 long"</span><span class="op">]</span><span class="op">]</span> <span class="op">+</span> <span class="va">nsc_g_tbl</span><span class="op">[[</span><span class="st">"chr1 long"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span>,
  <span class="va">scr_g_tbl</span><span class="op">[[</span><span class="st">"chr2 long"</span><span class="op">]</span><span class="op">]</span> <span class="op">/</span> <span class="op">(</span> <span class="va">scr_g_tbl</span><span class="op">[[</span><span class="st">"chr2 long"</span><span class="op">]</span><span class="op">]</span> <span class="op">+</span> <span class="va">nsc_g_tbl</span><span class="op">[[</span><span class="st">"chr2 long"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span>,
  <span class="va">scr_g_tbl</span><span class="op">[[</span><span class="st">"PAR long"</span><span class="op">]</span><span class="op">]</span>  <span class="op">/</span> <span class="op">(</span> <span class="va">scr_g_tbl</span><span class="op">[[</span><span class="st">"PAR long"</span><span class="op">]</span><span class="op">]</span>  <span class="op">+</span> <span class="va">nsc_g_tbl</span><span class="op">[[</span><span class="st">"PAR long"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span>
<span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/mean.html" class="external-link">mean</a></span><span class="op">(</span><span class="va">Oki_Osa_scr_ratio_long_arm</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## [1] 0.2314521</span></code></pre>
<div class="sourceCode" id="cb44"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">Oki_Osa_scr_ratio_short_arm</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span>
  <span class="va">scr_g_tbl</span><span class="op">[[</span><span class="st">"chr1 short"</span><span class="op">]</span><span class="op">]</span> <span class="op">/</span> <span class="op">(</span> <span class="va">scr_g_tbl</span><span class="op">[[</span><span class="st">"chr1 short"</span><span class="op">]</span><span class="op">]</span> <span class="op">+</span> <span class="va">nsc_g_tbl</span><span class="op">[[</span><span class="st">"chr1 short"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span>,
  <span class="va">scr_g_tbl</span><span class="op">[[</span><span class="st">"chr2 short"</span><span class="op">]</span><span class="op">]</span> <span class="op">/</span> <span class="op">(</span> <span class="va">scr_g_tbl</span><span class="op">[[</span><span class="st">"chr2 short"</span><span class="op">]</span><span class="op">]</span> <span class="op">+</span> <span class="va">nsc_g_tbl</span><span class="op">[[</span><span class="st">"chr2 short"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span>,
  <span class="va">scr_g_tbl</span><span class="op">[[</span><span class="st">"PAR short"</span><span class="op">]</span><span class="op">]</span>  <span class="op">/</span> <span class="op">(</span> <span class="va">scr_g_tbl</span><span class="op">[[</span><span class="st">"PAR short"</span><span class="op">]</span><span class="op">]</span>  <span class="op">+</span> <span class="va">nsc_g_tbl</span><span class="op">[[</span><span class="st">"PAR short"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span>
<span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/mean.html" class="external-link">mean</a></span><span class="op">(</span><span class="va">Oki_Osa_scr_ratio_short_arm</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## [1] 0.5374672</span></code></pre>
<div class="sourceCode" id="cb46"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/stats/t.test.html" class="external-link">t.test</a></span><span class="op">(</span><span class="va">Oki_Osa_scr_ratio_long_arm</span>, <span class="va">Oki_Osa_scr_ratio_short_arm</span>, paired <span class="op">=</span> <span class="cn">T</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## </span>
<span class="co">##  Paired t-test</span>
<span class="co">## </span>
<span class="co">## data:  Oki_Osa_scr_ratio_long_arm and Oki_Osa_scr_ratio_short_arm</span>
<span class="co">## t = -9.5775, df = 2, p-value = 0.01073</span>
<span class="co">## alternative hypothesis: true difference in means is not equal to 0</span>
<span class="co">## 95 percent confidence interval:</span>
<span class="co">##  -0.4434904 -0.1685397</span>
<span class="co">## sample estimates:</span>
<span class="co">## mean of the differences </span>
<span class="co">##              -0.3060151</span></code></pre>
<p>Replicate on an independent pair of genomes.</p>
<div class="sourceCode" id="cb48"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">all_genes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicFeatures/man/transcripts.html" class="external-link">genes</a></span><span class="op">(</span><span class="va">annots</span><span class="op">$</span><span class="va">Bar</span><span class="op">)</span>
<span class="va">scrambled_genes</span> <span class="op">&lt;-</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/findOverlaps-methods.html" class="external-link">subsetByOverlaps</a></span><span class="op">(</span><span class="va">all_genes</span>, <span class="va">unal</span><span class="op">$</span><span class="va">Bar_Kum</span><span class="op">[</span><span class="va">unal</span><span class="op">$</span><span class="va">Bar_Kum</span><span class="op">$</span><span class="va">nonCoa</span><span class="op">]</span><span class="op">)</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/sort.html" class="external-link">sort</a></span><span class="op">(</span>i<span class="op">=</span><span class="cn">F</span><span class="op">)</span> |&gt; <span class="fu">flagLongShort</span><span class="op">(</span><span class="va">longShort</span><span class="op">$</span><span class="va">Bar2.p4</span><span class="op">)</span>
<span class="va">not_scrambled_genes</span> <span class="op">&lt;-</span> <span class="va">all_genes</span><span class="op">[</span><span class="op">!</span> <span class="va">all_genes</span> <span class="op"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html" class="external-link">%in%</a></span> <span class="va">scrambled_genes</span><span class="op">]</span>                  |&gt; <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/sort.html" class="external-link">sort</a></span><span class="op">(</span>i<span class="op">=</span><span class="cn">F</span><span class="op">)</span> |&gt; <span class="fu">flagLongShort</span><span class="op">(</span><span class="va">longShort</span><span class="op">$</span><span class="va">Bar2.p4</span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">scrambled_genes</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## [1] 4002</span></code></pre>
<div class="sourceCode" id="cb50"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">all_genes</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## [1] 14272</span></code></pre>
<div class="sourceCode" id="cb52"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">scr_g_tbl</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/table.html" class="external-link">table</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqnames</a></span><span class="op">(</span>    <span class="va">scrambled_genes</span><span class="op">)</span>, <span class="va">scrambled_genes</span><span class="op">$</span><span class="va">Arm</span><span class="op">)</span><span class="op">)</span>
<span class="va">nsc_g_tbl</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/table.html" class="external-link">table</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqnames</a></span><span class="op">(</span><span class="va">not_scrambled_genes</span><span class="op">)</span>, <span class="va">not_scrambled_genes</span><span class="op">$</span><span class="va">Arm</span><span class="op">)</span><span class="op">)</span>

<span class="va">Bar_Kum_scr_ratio_long_arm</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span>
  <span class="va">scr_g_tbl</span><span class="op">[[</span><span class="st">"Chr1 long"</span><span class="op">]</span><span class="op">]</span> <span class="op">/</span> <span class="op">(</span> <span class="va">scr_g_tbl</span><span class="op">[[</span><span class="st">"Chr1 long"</span><span class="op">]</span><span class="op">]</span> <span class="op">+</span> <span class="va">nsc_g_tbl</span><span class="op">[[</span><span class="st">"Chr1 long"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span>,
  <span class="va">scr_g_tbl</span><span class="op">[[</span><span class="st">"Chr2 long"</span><span class="op">]</span><span class="op">]</span> <span class="op">/</span> <span class="op">(</span> <span class="va">scr_g_tbl</span><span class="op">[[</span><span class="st">"Chr2 long"</span><span class="op">]</span><span class="op">]</span> <span class="op">+</span> <span class="va">nsc_g_tbl</span><span class="op">[[</span><span class="st">"Chr2 long"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span>,
  <span class="va">scr_g_tbl</span><span class="op">[[</span><span class="st">"PAR long"</span><span class="op">]</span><span class="op">]</span>  <span class="op">/</span> <span class="op">(</span> <span class="va">scr_g_tbl</span><span class="op">[[</span><span class="st">"PAR long"</span><span class="op">]</span><span class="op">]</span>  <span class="op">+</span> <span class="va">nsc_g_tbl</span><span class="op">[[</span><span class="st">"PAR long"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span>
<span class="op">)</span>
<span class="va">Bar_Kum_scr_ratio_short_arm</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span>
  <span class="va">scr_g_tbl</span><span class="op">[[</span><span class="st">"Chr1 short"</span><span class="op">]</span><span class="op">]</span> <span class="op">/</span> <span class="op">(</span> <span class="va">scr_g_tbl</span><span class="op">[[</span><span class="st">"Chr1 short"</span><span class="op">]</span><span class="op">]</span> <span class="op">+</span> <span class="va">nsc_g_tbl</span><span class="op">[[</span><span class="st">"Chr1 short"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span>,
  <span class="va">scr_g_tbl</span><span class="op">[[</span><span class="st">"Chr2 short"</span><span class="op">]</span><span class="op">]</span> <span class="op">/</span> <span class="op">(</span> <span class="va">scr_g_tbl</span><span class="op">[[</span><span class="st">"Chr2 short"</span><span class="op">]</span><span class="op">]</span> <span class="op">+</span> <span class="va">nsc_g_tbl</span><span class="op">[[</span><span class="st">"Chr2 short"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span>,
  <span class="va">scr_g_tbl</span><span class="op">[[</span><span class="st">"PAR short"</span><span class="op">]</span><span class="op">]</span>  <span class="op">/</span> <span class="op">(</span> <span class="va">scr_g_tbl</span><span class="op">[[</span><span class="st">"PAR short"</span><span class="op">]</span><span class="op">]</span>  <span class="op">+</span> <span class="va">nsc_g_tbl</span><span class="op">[[</span><span class="st">"PAR short"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span>
<span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/r/stats/t.test.html" class="external-link">t.test</a></span><span class="op">(</span><span class="va">Bar_Kum_scr_ratio_long_arm</span>, <span class="va">Bar_Kum_scr_ratio_short_arm</span>, paired <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## </span>
<span class="co">##  Paired t-test</span>
<span class="co">## </span>
<span class="co">## data:  Bar_Kum_scr_ratio_long_arm and Bar_Kum_scr_ratio_short_arm</span>
<span class="co">## t = -16.994, df = 2, p-value = 0.003445</span>
<span class="co">## alternative hypothesis: true difference in means is not equal to 0</span>
<span class="co">## 95 percent confidence interval:</span>
<span class="co">##  -0.3825454 -0.2279699</span>
<span class="co">## sample estimates:</span>
<span class="co">## mean of the differences </span>
<span class="co">##              -0.3052576</span></code></pre>
<div class="sourceCode" id="cb54"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/mean.html" class="external-link">mean</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="va">Oki_Osa_scr_ratio_long_arm</span>, <span class="va">Bar_Kum_scr_ratio_long_arm</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## [1] 0.2055334</span></code></pre>
<div class="sourceCode" id="cb56"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/mean.html" class="external-link">mean</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="va">Oki_Osa_scr_ratio_short_arm</span>, <span class="va">Bar_Kum_scr_ratio_short_arm</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## [1] 0.5111697</span></code></pre>
<div class="sourceCode" id="cb58"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/stats/t.test.html" class="external-link">t.test</a></span><span class="op">(</span>
  <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="va">Oki_Osa_scr_ratio_long_arm</span>, <span class="va">Bar_Kum_scr_ratio_long_arm</span><span class="op">)</span>,
  <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="va">Oki_Osa_scr_ratio_short_arm</span>, <span class="va">Bar_Kum_scr_ratio_short_arm</span><span class="op">)</span>,
  paired <span class="op">=</span> <span class="cn">TRUE</span>
<span class="op">)</span></code></pre></div>
<pre><code><span class="co">## </span>
<span class="co">##  Paired t-test</span>
<span class="co">## </span>
<span class="co">## data:  c(Oki_Osa_scr_ratio_long_arm, Bar_Kum_scr_ratio_long_arm) and c(Oki_Osa_scr_ratio_short_arm, Bar_Kum_scr_ratio_short_arm)</span>
<span class="co">## t = -18.644, df = 5, p-value = 8.172e-06</span>
<span class="co">## alternative hypothesis: true difference in means is not equal to 0</span>
<span class="co">## 95 percent confidence interval:</span>
<span class="co">##  -0.3477765 -0.2634962</span>
<span class="co">## sample estimates:</span>
<span class="co">## mean of the differences </span>
<span class="co">##              -0.3056363</span></code></pre>
<p>How about exons ?</p>
<div class="sourceCode" id="cb60"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">all_exons</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicFeatures/man/exonicParts.html" class="external-link">exonicParts</a></span><span class="op">(</span><span class="va">annots</span><span class="op">$</span><span class="va">Oki</span><span class="op">)</span>
<span class="va">scrambled_exons</span> <span class="op">&lt;-</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/findOverlaps-methods.html" class="external-link">subsetByOverlaps</a></span><span class="op">(</span><span class="va">all_exons</span>, <span class="va">unal</span><span class="op">$</span><span class="va">Oki_Osa</span><span class="op">[</span><span class="va">unal</span><span class="op">$</span><span class="va">Oki_Osa</span><span class="op">$</span><span class="va">nonCoa</span><span class="op">]</span><span class="op">)</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/sort.html" class="external-link">sort</a></span><span class="op">(</span>i<span class="op">=</span><span class="cn">F</span><span class="op">)</span> |&gt; <span class="fu">flagLongShort</span><span class="op">(</span><span class="va">longShort</span><span class="op">$</span><span class="va">OKI2018.I69</span><span class="op">)</span>
<span class="va">not_scrambled_exons</span> <span class="op">&lt;-</span> <span class="va">all_exons</span><span class="op">[</span><span class="op">!</span> <span class="va">all_exons</span> <span class="op"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html" class="external-link">%in%</a></span> <span class="va">scrambled_exons</span><span class="op">]</span>                  |&gt; <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/sort.html" class="external-link">sort</a></span><span class="op">(</span>i<span class="op">=</span><span class="cn">F</span><span class="op">)</span> |&gt; <span class="fu">flagLongShort</span><span class="op">(</span><span class="va">longShort</span><span class="op">$</span><span class="va">OKI2018.I69</span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">scrambled_exons</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## [1] 16787</span></code></pre>
<div class="sourceCode" id="cb62"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">all_exons</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## [1] 106811</span></code></pre>
<div class="sourceCode" id="cb64"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">scr_g_tbl</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/table.html" class="external-link">table</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqnames</a></span><span class="op">(</span>    <span class="va">scrambled_exons</span><span class="op">)</span>, <span class="va">scrambled_exons</span><span class="op">$</span><span class="va">Arm</span><span class="op">)</span><span class="op">)</span>
<span class="va">nsc_g_tbl</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/table.html" class="external-link">table</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqnames</a></span><span class="op">(</span><span class="va">not_scrambled_exons</span><span class="op">)</span>, <span class="va">not_scrambled_exons</span><span class="op">$</span><span class="va">Arm</span><span class="op">)</span><span class="op">)</span>

<span class="va">Oki_Osa_scr_ratio_long_arm</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span>
  <span class="va">scr_g_tbl</span><span class="op">[[</span><span class="st">"chr1 long"</span><span class="op">]</span><span class="op">]</span> <span class="op">/</span> <span class="op">(</span> <span class="va">scr_g_tbl</span><span class="op">[[</span><span class="st">"chr1 long"</span><span class="op">]</span><span class="op">]</span> <span class="op">+</span> <span class="va">nsc_g_tbl</span><span class="op">[[</span><span class="st">"chr1 long"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span>,
  <span class="va">scr_g_tbl</span><span class="op">[[</span><span class="st">"chr2 long"</span><span class="op">]</span><span class="op">]</span> <span class="op">/</span> <span class="op">(</span> <span class="va">scr_g_tbl</span><span class="op">[[</span><span class="st">"chr2 long"</span><span class="op">]</span><span class="op">]</span> <span class="op">+</span> <span class="va">nsc_g_tbl</span><span class="op">[[</span><span class="st">"chr2 long"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span>,
  <span class="va">scr_g_tbl</span><span class="op">[[</span><span class="st">"PAR long"</span><span class="op">]</span><span class="op">]</span>  <span class="op">/</span> <span class="op">(</span> <span class="va">scr_g_tbl</span><span class="op">[[</span><span class="st">"PAR long"</span><span class="op">]</span><span class="op">]</span>  <span class="op">+</span> <span class="va">nsc_g_tbl</span><span class="op">[[</span><span class="st">"PAR long"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span>
<span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/mean.html" class="external-link">mean</a></span><span class="op">(</span><span class="va">Oki_Osa_scr_ratio_long_arm</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## [1] 0.1172023</span></code></pre>
<div class="sourceCode" id="cb66"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">Oki_Osa_scr_ratio_short_arm</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span>
  <span class="va">scr_g_tbl</span><span class="op">[[</span><span class="st">"chr1 short"</span><span class="op">]</span><span class="op">]</span> <span class="op">/</span> <span class="op">(</span> <span class="va">scr_g_tbl</span><span class="op">[[</span><span class="st">"chr1 short"</span><span class="op">]</span><span class="op">]</span> <span class="op">+</span> <span class="va">nsc_g_tbl</span><span class="op">[[</span><span class="st">"chr1 short"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span>,
  <span class="va">scr_g_tbl</span><span class="op">[[</span><span class="st">"chr2 short"</span><span class="op">]</span><span class="op">]</span> <span class="op">/</span> <span class="op">(</span> <span class="va">scr_g_tbl</span><span class="op">[[</span><span class="st">"chr2 short"</span><span class="op">]</span><span class="op">]</span> <span class="op">+</span> <span class="va">nsc_g_tbl</span><span class="op">[[</span><span class="st">"chr2 short"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span>,
  <span class="va">scr_g_tbl</span><span class="op">[[</span><span class="st">"PAR short"</span><span class="op">]</span><span class="op">]</span>  <span class="op">/</span> <span class="op">(</span> <span class="va">scr_g_tbl</span><span class="op">[[</span><span class="st">"PAR short"</span><span class="op">]</span><span class="op">]</span>  <span class="op">+</span> <span class="va">nsc_g_tbl</span><span class="op">[[</span><span class="st">"PAR short"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span>
<span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/mean.html" class="external-link">mean</a></span><span class="op">(</span><span class="va">Oki_Osa_scr_ratio_short_arm</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## [1] 0.3003286</span></code></pre>
<div class="sourceCode" id="cb68"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/stats/t.test.html" class="external-link">t.test</a></span><span class="op">(</span><span class="va">Oki_Osa_scr_ratio_long_arm</span>, <span class="va">Oki_Osa_scr_ratio_short_arm</span>, paired <span class="op">=</span> <span class="cn">T</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## </span>
<span class="co">##  Paired t-test</span>
<span class="co">## </span>
<span class="co">## data:  Oki_Osa_scr_ratio_long_arm and Oki_Osa_scr_ratio_short_arm</span>
<span class="co">## t = -5.9254, df = 2, p-value = 0.02732</span>
<span class="co">## alternative hypothesis: true difference in means is not equal to 0</span>
<span class="co">## 95 percent confidence interval:</span>
<span class="co">##  -0.31610038 -0.05015222</span>
<span class="co">## sample estimates:</span>
<span class="co">## mean of the differences </span>
<span class="co">##              -0.1831263</span></code></pre>
</div>
</div>
<div class="section level2">
<h2 id="operon-summary-plots-for-figure-3">Operon summary plots for Figure 3<a class="anchor" aria-label="anchor" href="#operon-summary-plots-for-figure-3"></a>
</h2>
<div class="sourceCode" id="cb70"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Add an index for matching to scrambled set</span>
<span class="va">operons</span><span class="op">$</span><span class="va">Oki</span><span class="op">$</span><span class="va">index</span> <span class="op">&lt;-</span> <span class="fl">1</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">operons</span><span class="op">$</span><span class="va">Oki</span><span class="op">)</span>
<span class="va">operons</span><span class="op">$</span><span class="va">Oki</span><span class="op">$</span><span class="va">scrambled</span> <span class="op">&lt;-</span> <span class="st">'no_breakpoint_region'</span>
<span class="va">operons</span><span class="op">$</span><span class="va">Oki</span><span class="op">$</span><span class="va">scrambled</span><span class="op">[</span><span class="va">operons</span><span class="op">$</span><span class="va">Oki</span> <span class="op"><a href="https://rdrr.io/pkg/IRanges/man/findOverlaps-methods.html" class="external-link">%over%</a></span> <span class="va">unal</span><span class="op">$</span><span class="va">Oki_Osa</span><span class="op">[</span><span class="va">unal</span><span class="op">$</span><span class="va">Oki_Osa</span><span class="op">$</span><span class="va">nonCoa</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">'overlaps_breakpoint_region'</span>

<span class="va">operons</span><span class="op">$</span><span class="va">Osa</span><span class="op">$</span><span class="va">index</span> <span class="op">&lt;-</span> <span class="fl">1</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">operons</span><span class="op">$</span><span class="va">Osa</span><span class="op">)</span>
<span class="va">operons</span><span class="op">$</span><span class="va">Osa</span><span class="op">$</span><span class="va">scrambled</span> <span class="op">&lt;-</span> <span class="st">'no_breakpoint_region'</span>
<span class="va">operons</span><span class="op">$</span><span class="va">Osa</span><span class="op">$</span><span class="va">scrambled</span><span class="op">[</span><span class="va">operons</span><span class="op">$</span><span class="va">Osa</span> <span class="op"><a href="https://rdrr.io/pkg/IRanges/man/findOverlaps-methods.html" class="external-link">%over%</a></span> <span class="va">unal</span><span class="op">$</span><span class="va">Osa_Oki</span><span class="op">[</span><span class="va">unal</span><span class="op">$</span><span class="va">Osa_Oki</span><span class="op">$</span><span class="va">nonCoa</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">'overlaps_breakpoint_region'</span>

<span class="va">operons</span><span class="op">$</span><span class="va">Bar</span><span class="op">$</span><span class="va">index</span> <span class="op">&lt;-</span> <span class="fl">1</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">operons</span><span class="op">$</span><span class="va">Bar</span><span class="op">)</span>
<span class="va">operons</span><span class="op">$</span><span class="va">Bar</span><span class="op">$</span><span class="va">scrambled</span> <span class="op">&lt;-</span> <span class="st">'no_breakpoint_region'</span>
<span class="va">operons</span><span class="op">$</span><span class="va">Bar</span><span class="op">$</span><span class="va">scrambled</span><span class="op">[</span><span class="va">operons</span><span class="op">$</span><span class="va">Bar</span> <span class="op"><a href="https://rdrr.io/pkg/IRanges/man/findOverlaps-methods.html" class="external-link">%over%</a></span> <span class="va">unal</span><span class="op">$</span><span class="va">Bar_Oki</span><span class="op">[</span><span class="va">unal</span><span class="op">$</span><span class="va">Bar_Oki</span><span class="op">$</span><span class="va">nonCoa</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">'overlaps_breakpoint_region'</span>

<span class="va">tb1</span> <span class="op">&lt;-</span> <span class="va">operons</span><span class="op">$</span><span class="va">Oki</span>
<span class="co"># Note that the operon objects do not have unique names, which makes as.data.frame() complain.</span>
<span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">tb1</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="cn">NULL</span>
<span class="va">p1</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="va">tb1</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu">aes</span><span class="op">(</span>y<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="va">n</span>, levels <span class="op">=</span> <span class="fl">2</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/Extremes.html" class="external-link">max</a></span><span class="op">(</span><span class="va">operons</span><span class="op">$</span><span class="va">Oki</span><span class="op">$</span><span class="va">n</span>, <span class="va">operons</span><span class="op">$</span><span class="va">Osa</span><span class="op">$</span><span class="va">n</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu">geom_bar</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu">geom_text</span><span class="op">(</span>stat<span class="op">=</span><span class="st">'count'</span>, <span class="fu">aes</span><span class="op">(</span>label<span class="op">=</span><span class="va">..count..</span>, hjust<span class="op">=</span><span class="op">-</span><span class="fl">0.2</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu">expand_limits</span><span class="op">(</span>x<span class="op">=</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">750</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu">ylab</span><span class="op">(</span><span class="st">"number of genes in operon"</span><span class="op">)</span> <span class="op">+</span> <span class="fu">xlab</span><span class="op">(</span><span class="st">"Count"</span><span class="op">)</span> <span class="op">+</span> <span class="fu">facet_wrap</span><span class="op">(</span><span class="op">~</span><span class="va">scrambled</span>, nrow<span class="op">=</span><span class="fl">2</span><span class="op">)</span> <span class="op">+</span> <span class="fu">ggtitle</span><span class="op">(</span><span class="st">"Okinawa operons"</span><span class="op">)</span> <span class="op">+</span> <span class="fu">scale_y_discrete</span><span class="op">(</span>drop<span class="op">=</span><span class="cn">F</span><span class="op">)</span>
<span class="va">p1</span></code></pre></div>
<p><img src="GenomicFeatures_files/figure-html/operon_summary_stats_plots-1.svg" width="1152"></p>
<div class="sourceCode" id="cb71"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">tb2</span> <span class="op">&lt;-</span> <span class="va">operons</span><span class="op">$</span><span class="va">Osa</span>
<span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">tb2</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="cn">NULL</span>
<span class="va">p2</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="va">tb2</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu">aes</span><span class="op">(</span>y<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="va">n</span>, levels <span class="op">=</span> <span class="fl">2</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/Extremes.html" class="external-link">max</a></span><span class="op">(</span><span class="va">operons</span><span class="op">$</span><span class="va">Oki</span><span class="op">$</span><span class="va">n</span>, <span class="va">operons</span><span class="op">$</span><span class="va">Osa</span><span class="op">$</span><span class="va">n</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu">geom_bar</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu">geom_text</span><span class="op">(</span>stat<span class="op">=</span><span class="st">'count'</span>, <span class="fu">aes</span><span class="op">(</span>label<span class="op">=</span><span class="va">..count..</span>, hjust<span class="op">=</span><span class="op">-</span><span class="fl">0.2</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu">expand_limits</span><span class="op">(</span>x<span class="op">=</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">750</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu">ylab</span><span class="op">(</span><span class="st">"number of genes in operon"</span><span class="op">)</span> <span class="op">+</span> <span class="fu">xlab</span><span class="op">(</span><span class="st">"Count"</span><span class="op">)</span> <span class="op">+</span> <span class="fu">facet_wrap</span><span class="op">(</span><span class="op">~</span><span class="va">scrambled</span>, nrow<span class="op">=</span><span class="fl">2</span><span class="op">)</span> <span class="op">+</span> <span class="fu">ggtitle</span><span class="op">(</span><span class="st">"Osaka operons"</span><span class="op">)</span> <span class="op">+</span> <span class="fu">scale_y_discrete</span><span class="op">(</span>drop<span class="op">=</span><span class="cn">F</span><span class="op">)</span>
<span class="va">p2</span></code></pre></div>
<p><img src="GenomicFeatures_files/figure-html/operon_summary_stats_plots-2.svg" width="1152"></p>
<div class="sourceCode" id="cb72"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/cbind.html" class="external-link">rbind</a></span><span class="op">(</span>
  <span class="va">operons</span><span class="op">$</span><span class="va">Oki</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="op">)</span> |&gt; <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>species<span class="op">=</span><span class="st">"Okinawa"</span><span class="op">)</span>,
  <span class="va">operons</span><span class="op">$</span><span class="va">Osa</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="op">)</span> |&gt; <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>species<span class="op">=</span><span class="st">"Osaka"</span><span class="op">)</span>,
  <span class="va">operons</span><span class="op">$</span><span class="va">Bar</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="op">)</span> |&gt; <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>species<span class="op">=</span><span class="st">"Barcelona"</span><span class="op">)</span>
<span class="op">)</span> |&gt; <span class="fu">ggplot</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">aes</span><span class="op">(</span>y<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="va">n</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">geom_bar</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">facet_wrap</span><span class="op">(</span><span class="op">~</span><span class="va">scrambled</span> <span class="op">+</span> <span class="va">species</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">geom_text</span><span class="op">(</span>stat<span class="op">=</span><span class="st">'count'</span>, <span class="fu">aes</span><span class="op">(</span>label<span class="op">=</span><span class="va">..count..</span>, hjust<span class="op">=</span><span class="op">-</span><span class="fl">0.2</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">theme_bw</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">scale_x_continuous</span><span class="op">(</span><span class="st">"Number operons"</span>, limits <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1200</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">scale_y_discrete</span><span class="op">(</span><span class="st">"Number of genes in operon"</span><span class="op">)</span></code></pre></div>
<p><img src="GenomicFeatures_files/figure-html/operon_summary_stats_plots-3.svg" width="1152"></p>
<div class="sourceCode" id="cb73"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">p3</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="va">tb1</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu">aes</span><span class="op">(</span>y<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="va">n</span>, levels <span class="op">=</span> <span class="fl">2</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/Extremes.html" class="external-link">max</a></span><span class="op">(</span><span class="va">operons</span><span class="op">$</span><span class="va">Oki</span><span class="op">$</span><span class="va">n</span>, <span class="va">operons</span><span class="op">$</span><span class="va">Osa</span><span class="op">$</span><span class="va">n</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu">geom_bar</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu">geom_text</span><span class="op">(</span>stat<span class="op">=</span><span class="st">'count'</span>, <span class="fu">aes</span><span class="op">(</span>label<span class="op">=</span><span class="va">..count..</span>, hjust<span class="op">=</span><span class="op">-</span><span class="fl">0.2</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu">expand_limits</span><span class="op">(</span>x<span class="op">=</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">750</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu">ylab</span><span class="op">(</span><span class="st">"number of genes in operon"</span><span class="op">)</span> <span class="op">+</span> <span class="fu">xlab</span><span class="op">(</span><span class="st">"Count"</span><span class="op">)</span> <span class="op">+</span> <span class="fu">facet_wrap</span><span class="op">(</span><span class="op">~</span><span class="va">scrambled</span>, ncol<span class="op">=</span><span class="fl">2</span><span class="op">)</span> <span class="op">+</span> <span class="fu">ggtitle</span><span class="op">(</span><span class="st">"Okinawa operons"</span><span class="op">)</span> <span class="op">+</span> <span class="fu">scale_y_discrete</span><span class="op">(</span>drop<span class="op">=</span><span class="cn">F</span><span class="op">)</span>
<span class="va">p3</span></code></pre></div>
<p><img src="GenomicFeatures_files/figure-html/operon_summary_stats_plots-4.svg" width="1152"></p>
<div class="sourceCode" id="cb74"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">p4</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="va">tb2</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu">aes</span><span class="op">(</span>y<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="va">n</span>, levels <span class="op">=</span> <span class="fl">2</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/Extremes.html" class="external-link">max</a></span><span class="op">(</span><span class="va">operons</span><span class="op">$</span><span class="va">Oki</span><span class="op">$</span><span class="va">n</span>, <span class="va">operons</span><span class="op">$</span><span class="va">Osa</span><span class="op">$</span><span class="va">n</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu">geom_bar</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu">geom_text</span><span class="op">(</span>stat<span class="op">=</span><span class="st">'count'</span>, <span class="fu">aes</span><span class="op">(</span>label<span class="op">=</span><span class="va">..count..</span>, hjust<span class="op">=</span><span class="op">-</span><span class="fl">0.2</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu">expand_limits</span><span class="op">(</span>x<span class="op">=</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">750</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu">ylab</span><span class="op">(</span><span class="st">"number of genes in operon"</span><span class="op">)</span> <span class="op">+</span> <span class="fu">xlab</span><span class="op">(</span><span class="st">"Count"</span><span class="op">)</span> <span class="op">+</span> <span class="fu">facet_wrap</span><span class="op">(</span><span class="op">~</span><span class="va">scrambled</span>, ncol<span class="op">=</span><span class="fl">2</span><span class="op">)</span> <span class="op">+</span> <span class="fu">ggtitle</span><span class="op">(</span><span class="st">"Osaka operons"</span><span class="op">)</span> <span class="op">+</span> <span class="fu">scale_y_discrete</span><span class="op">(</span>drop<span class="op">=</span><span class="cn">F</span><span class="op">)</span>
<span class="va">p4</span></code></pre></div>
<p><img src="GenomicFeatures_files/figure-html/operon_summary_stats_plots-5.svg" width="1152"></p>
<div class="sourceCode" id="cb75"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="op">(</span> <span class="va">p3</span> <span class="op">/</span> <span class="va">p4</span> <span class="op">)</span></code></pre></div>
<p><img src="GenomicFeatures_files/figure-html/operon_summary_stats_plots-6.svg" width="1152"></p>
<div class="sourceCode" id="cb76"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">tb1</span><span class="op">$</span><span class="va">sp</span> <span class="op">&lt;-</span> <span class="st">"Oki"</span>
<span class="va">tb2</span><span class="op">$</span><span class="va">sp</span> <span class="op">&lt;-</span> <span class="st">"Osa"</span>
<span class="va">tb3</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/cbind.html" class="external-link">rbind</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="va">tb1</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="va">tb2</span><span class="op">)</span><span class="op">)</span>
<span class="fu">ggplot</span><span class="op">(</span><span class="va">tb3</span><span class="op">)</span> <span class="op">+</span> <span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="va">n</span>, levels <span class="op">=</span> <span class="fl">2</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/Extremes.html" class="external-link">max</a></span><span class="op">(</span><span class="va">operons</span><span class="op">$</span><span class="va">Oki</span><span class="op">$</span><span class="va">n</span>, <span class="va">operons</span><span class="op">$</span><span class="va">Osa</span><span class="op">$</span><span class="va">n</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu">geom_bar</span><span class="op">(</span>position<span class="op">=</span><span class="st">'dodge'</span>, <span class="fu">aes</span><span class="op">(</span>fill<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">sp</span>, <span class="st">"_"</span>, <span class="va">scrambled</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu">scale_x_discrete</span><span class="op">(</span>drop<span class="op">=</span><span class="cn">F</span><span class="op">)</span></code></pre></div>
<p><img src="GenomicFeatures_files/figure-html/operon_summary_stats_plots-7.svg" width="1152"></p>
<div class="sourceCode" id="cb77"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">genes</span><span class="op">$</span><span class="va">Oki</span><span class="op">$</span><span class="va">scrambled</span> <span class="op">&lt;-</span> <span class="st">'no_breakpoint_region'</span>
<span class="va">genes</span><span class="op">$</span><span class="va">Oki</span><span class="op">$</span><span class="va">scrambled</span><span class="op">[</span><span class="va">genes</span><span class="op">$</span><span class="va">Oki</span> <span class="op"><a href="https://rdrr.io/pkg/IRanges/man/findOverlaps-methods.html" class="external-link">%over%</a></span> <span class="va">unal</span><span class="op">$</span><span class="va">Oki_Osa</span><span class="op">[</span><span class="va">unal</span><span class="op">$</span><span class="va">Oki_Osa</span><span class="op">$</span><span class="va">nonCoa</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">'overlaps_breakpoint_region'</span>
<span class="va">genes</span><span class="op">$</span><span class="va">Oki</span><span class="op">$</span><span class="va">scrambled</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/table.html" class="external-link">table</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## </span>
<span class="co">##       no_breakpoint_region overlaps_breakpoint_region </span>
<span class="co">##                      11997                       5294</span></code></pre>
<div class="sourceCode" id="cb79"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">genes</span><span class="op">$</span><span class="va">Osa</span><span class="op">$</span><span class="va">scrambled</span> <span class="op">&lt;-</span> <span class="st">'no_breakpoint_region'</span>
<span class="va">genes</span><span class="op">$</span><span class="va">Osa</span><span class="op">$</span><span class="va">scrambled</span><span class="op">[</span><span class="va">genes</span><span class="op">$</span><span class="va">Osa</span> <span class="op"><a href="https://rdrr.io/pkg/IRanges/man/findOverlaps-methods.html" class="external-link">%over%</a></span> <span class="va">unal</span><span class="op">$</span><span class="va">Osa_Oki</span><span class="op">[</span><span class="va">unal</span><span class="op">$</span><span class="va">Osa_Oki</span><span class="op">$</span><span class="va">nonCoa</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">'overlaps_breakpoint_region'</span>
<span class="va">genes</span><span class="op">$</span><span class="va">Osa</span><span class="op">$</span><span class="va">scrambled</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/table.html" class="external-link">table</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## </span>
<span class="co">##       no_breakpoint_region overlaps_breakpoint_region </span>
<span class="co">##                      11617                       4103</span></code></pre>
<div class="sourceCode" id="cb81"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">genes</span><span class="op">$</span><span class="va">Bar</span><span class="op">$</span><span class="va">scrambled</span> <span class="op">&lt;-</span> <span class="st">'no_breakpoint_region'</span>
<span class="va">genes</span><span class="op">$</span><span class="va">Bar</span><span class="op">$</span><span class="va">scrambled</span><span class="op">[</span><span class="va">genes</span><span class="op">$</span><span class="va">Bar</span> <span class="op"><a href="https://rdrr.io/pkg/IRanges/man/findOverlaps-methods.html" class="external-link">%over%</a></span> <span class="va">unal</span><span class="op">$</span><span class="va">Bar_Oki</span><span class="op">[</span><span class="va">unal</span><span class="op">$</span><span class="va">Bar_Oki</span><span class="op">$</span><span class="va">nonCoa</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">'overlaps_breakpoint_region'</span>
<span class="va">genes</span><span class="op">$</span><span class="va">Bar</span><span class="op">$</span><span class="va">scrambled</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/table.html" class="external-link">table</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## </span>
<span class="co">##       no_breakpoint_region overlaps_breakpoint_region </span>
<span class="co">##                      10265                       4007</span></code></pre>
<div class="sourceCode" id="cb83"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">exons</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">annots</span>, <span class="kw">function</span><span class="op">(</span><span class="va">a</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/sort.html" class="external-link">sort</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/GenomicFeatures/man/transcripts.html" class="external-link">exons</a></span><span class="op">(</span><span class="va">a</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/SimpleList-class.html" class="external-link">SimpleList</a></span><span class="op">(</span><span class="op">)</span>
<span class="va">exons</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">exons</span>, <span class="kw">function</span><span class="op">(</span><span class="va">e</span><span class="op">)</span> <span class="op">{</span>
  <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">e</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">e</span><span class="op">$</span><span class="va">exon_id</span>
  <span class="va">e</span>
<span class="op">}</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/SimpleList-class.html" class="external-link">SimpleList</a></span><span class="op">(</span><span class="op">)</span>

<span class="va">exons</span><span class="op">$</span><span class="va">Oki</span><span class="op">$</span><span class="va">scrambled</span> <span class="op">&lt;-</span> <span class="st">'no_breakpoint_region'</span>
<span class="va">exons</span><span class="op">$</span><span class="va">Oki</span><span class="op">$</span><span class="va">scrambled</span><span class="op">[</span><span class="va">exons</span><span class="op">$</span><span class="va">Oki</span> <span class="op"><a href="https://rdrr.io/pkg/IRanges/man/findOverlaps-methods.html" class="external-link">%over%</a></span> <span class="va">unal</span><span class="op">$</span><span class="va">Oki_Osa</span><span class="op">[</span><span class="va">unal</span><span class="op">$</span><span class="va">Oki_Osa</span><span class="op">$</span><span class="va">nonCoa</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">'overlaps_breakpoint_region'</span>
<span class="va">exons</span><span class="op">$</span><span class="va">Oki</span><span class="op">$</span><span class="va">scrambled</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/table.html" class="external-link">table</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## </span>
<span class="co">##       no_breakpoint_region overlaps_breakpoint_region </span>
<span class="co">##                     102165                      17383</span></code></pre>
<div class="sourceCode" id="cb85"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">exons</span><span class="op">$</span><span class="va">Osa</span><span class="op">$</span><span class="va">scrambled</span> <span class="op">&lt;-</span> <span class="st">'no_breakpoint_region'</span>
<span class="va">exons</span><span class="op">$</span><span class="va">Osa</span><span class="op">$</span><span class="va">scrambled</span><span class="op">[</span><span class="va">exons</span><span class="op">$</span><span class="va">Osa</span> <span class="op"><a href="https://rdrr.io/pkg/IRanges/man/findOverlaps-methods.html" class="external-link">%over%</a></span> <span class="va">unal</span><span class="op">$</span><span class="va">Osa_Oki</span><span class="op">[</span><span class="va">unal</span><span class="op">$</span><span class="va">Osa_Oki</span><span class="op">$</span><span class="va">nonCoa</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">'overlaps_breakpoint_region'</span>
<span class="va">exons</span><span class="op">$</span><span class="va">Osa</span><span class="op">$</span><span class="va">scrambled</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/table.html" class="external-link">table</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## </span>
<span class="co">##       no_breakpoint_region overlaps_breakpoint_region </span>
<span class="co">##                      96290                      10615</span></code></pre>
<div class="sourceCode" id="cb87"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">exons</span><span class="op">$</span><span class="va">Bar</span><span class="op">$</span><span class="va">scrambled</span> <span class="op">&lt;-</span> <span class="st">'no_breakpoint_region'</span>
<span class="va">exons</span><span class="op">$</span><span class="va">Bar</span><span class="op">$</span><span class="va">scrambled</span><span class="op">[</span><span class="va">exons</span><span class="op">$</span><span class="va">Bar</span> <span class="op"><a href="https://rdrr.io/pkg/IRanges/man/findOverlaps-methods.html" class="external-link">%over%</a></span> <span class="va">unal</span><span class="op">$</span><span class="va">Bar_Oki</span><span class="op">[</span><span class="va">unal</span><span class="op">$</span><span class="va">Bar_Oki</span><span class="op">$</span><span class="va">nonCoa</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">'overlaps_breakpoint_region'</span>
<span class="va">exons</span><span class="op">$</span><span class="va">Bar</span><span class="op">$</span><span class="va">scrambled</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/table.html" class="external-link">table</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## </span>
<span class="co">##       no_breakpoint_region overlaps_breakpoint_region </span>
<span class="co">##                      89412                      10498</span></code></pre>
<div class="sourceCode" id="cb89"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">oki_feat_bp_ovl</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/cbind.html" class="external-link">rbind</a></span><span class="op">(</span>
  <span class="va">operons</span><span class="op">$</span><span class="va">Oki</span><span class="op">$</span><span class="va">scrambled</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/table.html" class="external-link">table</a></span><span class="op">(</span><span class="op">)</span>,
  <span class="va">genes</span><span class="op">$</span><span class="va">Oki</span><span class="op">$</span><span class="va">scrambled</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/table.html" class="external-link">table</a></span><span class="op">(</span><span class="op">)</span>,
  <span class="va">exons</span><span class="op">$</span><span class="va">Oki</span><span class="op">$</span><span class="va">scrambled</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/table.html" class="external-link">table</a></span><span class="op">(</span><span class="op">)</span>
<span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span>row.names <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'operons'</span>, <span class="st">'genes'</span>, <span class="st">'exons'</span><span class="op">)</span><span class="op">)</span>
<span class="va">oki_feat_bp_ovl</span><span class="op">$</span><span class="va">pct</span> <span class="op">&lt;-</span> <span class="va">oki_feat_bp_ovl</span><span class="op">[</span>,<span class="fl">2</span><span class="op">]</span><span class="op">/</span><span class="op">(</span><span class="va">oki_feat_bp_ovl</span><span class="op">[</span>,<span class="fl">1</span><span class="op">]</span><span class="op">+</span><span class="va">oki_feat_bp_ovl</span><span class="op">[</span>,<span class="fl">2</span><span class="op">]</span><span class="op">)</span><span class="op">*</span><span class="fl">100</span>
<span class="va">oki_feat_bp_ovl</span><span class="op">$</span><span class="va">type</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">oki_feat_bp_ovl</span><span class="op">)</span>
<span class="va">oki_feat_bp_ovl</span><span class="op">$</span><span class="va">species</span> <span class="op">&lt;-</span> <span class="st">"Okinawa"</span>

<span class="va">osa_feat_bp_ovl</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/cbind.html" class="external-link">rbind</a></span><span class="op">(</span>
  <span class="va">operons</span><span class="op">$</span><span class="va">Osa</span><span class="op">$</span><span class="va">scrambled</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/table.html" class="external-link">table</a></span><span class="op">(</span><span class="op">)</span>,
  <span class="va">genes</span><span class="op">$</span><span class="va">Osa</span><span class="op">$</span><span class="va">scrambled</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/table.html" class="external-link">table</a></span><span class="op">(</span><span class="op">)</span>,
  <span class="va">exons</span><span class="op">$</span><span class="va">Osa</span><span class="op">$</span><span class="va">scrambled</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/table.html" class="external-link">table</a></span><span class="op">(</span><span class="op">)</span>
<span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span>row.names <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'operons'</span>, <span class="st">'genes'</span>, <span class="st">'exons'</span><span class="op">)</span><span class="op">)</span>
<span class="va">osa_feat_bp_ovl</span><span class="op">$</span><span class="va">pct</span> <span class="op">&lt;-</span> <span class="va">osa_feat_bp_ovl</span><span class="op">[</span>,<span class="fl">2</span><span class="op">]</span><span class="op">/</span><span class="op">(</span><span class="va">osa_feat_bp_ovl</span><span class="op">[</span>,<span class="fl">1</span><span class="op">]</span><span class="op">+</span><span class="va">osa_feat_bp_ovl</span><span class="op">[</span>,<span class="fl">2</span><span class="op">]</span><span class="op">)</span><span class="op">*</span><span class="fl">100</span>
<span class="va">osa_feat_bp_ovl</span><span class="op">$</span><span class="va">type</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">osa_feat_bp_ovl</span><span class="op">)</span>
<span class="va">osa_feat_bp_ovl</span><span class="op">$</span><span class="va">species</span> <span class="op">&lt;-</span> <span class="st">"Osaka"</span>

<span class="va">bar_feat_bp_ovl</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/cbind.html" class="external-link">rbind</a></span><span class="op">(</span>
  <span class="va">operons</span><span class="op">$</span><span class="va">Bar</span><span class="op">$</span><span class="va">scrambled</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/table.html" class="external-link">table</a></span><span class="op">(</span><span class="op">)</span>,
  <span class="va">genes</span><span class="op">$</span><span class="va">Bar</span><span class="op">$</span><span class="va">scrambled</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/table.html" class="external-link">table</a></span><span class="op">(</span><span class="op">)</span>,
  <span class="va">exons</span><span class="op">$</span><span class="va">Bar</span><span class="op">$</span><span class="va">scrambled</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/table.html" class="external-link">table</a></span><span class="op">(</span><span class="op">)</span>
<span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span>row.names <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'operons'</span>, <span class="st">'genes'</span>, <span class="st">'exons'</span><span class="op">)</span><span class="op">)</span>
<span class="va">bar_feat_bp_ovl</span><span class="op">$</span><span class="va">pct</span> <span class="op">&lt;-</span> <span class="va">bar_feat_bp_ovl</span><span class="op">[</span>,<span class="fl">2</span><span class="op">]</span><span class="op">/</span><span class="op">(</span><span class="va">bar_feat_bp_ovl</span><span class="op">[</span>,<span class="fl">1</span><span class="op">]</span><span class="op">+</span><span class="va">bar_feat_bp_ovl</span><span class="op">[</span>,<span class="fl">2</span><span class="op">]</span><span class="op">)</span><span class="op">*</span><span class="fl">100</span>
<span class="va">bar_feat_bp_ovl</span><span class="op">$</span><span class="va">type</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">bar_feat_bp_ovl</span><span class="op">)</span>
<span class="va">bar_feat_bp_ovl</span><span class="op">$</span><span class="va">species</span> <span class="op">&lt;-</span> <span class="st">"Barcelona"</span>

<span class="va">df</span> <span class="op">&lt;-</span> <span class="va">operons</span><span class="op">$</span><span class="va">Oki</span>
<span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">df</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="cn">NULL</span>
<span class="va">oki_feat_bp_ovl_join</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/cbind.html" class="external-link">rbind</a></span><span class="op">(</span>
  <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="va">df</span><span class="op">)</span>        |&gt; <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>type<span class="op">=</span><span class="st">'operon'</span><span class="op">)</span> |&gt; <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">seqnames</span>, <span class="va">start</span>, <span class="va">end</span>, <span class="va">scrambled</span>, <span class="va">type</span><span class="op">)</span>,
  <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="va">genes</span><span class="op">$</span><span class="va">Oki</span><span class="op">)</span> |&gt; <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>type<span class="op">=</span><span class="st">'gene'</span><span class="op">)</span>   |&gt; <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">seqnames</span>, <span class="va">start</span>, <span class="va">end</span>, <span class="va">scrambled</span>, <span class="va">type</span><span class="op">)</span>,
  <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="va">exons</span><span class="op">$</span><span class="va">Oki</span><span class="op">)</span> |&gt; <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>type<span class="op">=</span><span class="st">'exon'</span><span class="op">)</span>   |&gt; <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">seqnames</span>, <span class="va">start</span>, <span class="va">end</span>, <span class="va">scrambled</span>, <span class="va">type</span><span class="op">)</span>
<span class="op">)</span> |&gt; <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>species<span class="op">=</span><span class="st">'Oki'</span><span class="op">)</span>

<span class="va">df</span> <span class="op">&lt;-</span> <span class="va">operons</span><span class="op">$</span><span class="va">Osa</span>
<span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">df</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="cn">NULL</span>
<span class="va">osa_feat_bp_ovl_join</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/cbind.html" class="external-link">rbind</a></span><span class="op">(</span>
  <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="va">df</span><span class="op">)</span>        |&gt; <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>type<span class="op">=</span><span class="st">'operon'</span><span class="op">)</span> |&gt; <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">seqnames</span>, <span class="va">start</span>, <span class="va">end</span>, <span class="va">scrambled</span>, <span class="va">type</span><span class="op">)</span>,
  <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="va">genes</span><span class="op">$</span><span class="va">Osa</span><span class="op">)</span> |&gt; <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>type<span class="op">=</span><span class="st">'gene'</span><span class="op">)</span>   |&gt; <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">seqnames</span>, <span class="va">start</span>, <span class="va">end</span>, <span class="va">scrambled</span>, <span class="va">type</span><span class="op">)</span>,
  <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="va">exons</span><span class="op">$</span><span class="va">Osa</span><span class="op">)</span> |&gt; <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>type<span class="op">=</span><span class="st">'exon'</span><span class="op">)</span>   |&gt; <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">seqnames</span>, <span class="va">start</span>, <span class="va">end</span>, <span class="va">scrambled</span>, <span class="va">type</span><span class="op">)</span>
<span class="op">)</span> |&gt; <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>species<span class="op">=</span><span class="st">'Osa'</span><span class="op">)</span>

<span class="va">p1</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">oki_feat_bp_ovl</span><span class="op">)</span> <span class="op">+</span> <span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">type</span>, y<span class="op">=</span><span class="va">pct</span>, fill<span class="op">=</span><span class="va">type</span><span class="op">)</span> <span class="op">+</span> <span class="fu">geom_bar</span><span class="op">(</span>stat<span class="op">=</span><span class="st">'identity'</span><span class="op">)</span> <span class="op">+</span> <span class="fu">ylab</span><span class="op">(</span><span class="st">'percent overlapping breakpoint region'</span><span class="op">)</span> <span class="op">+</span> <span class="fu">ggtitle</span><span class="op">(</span><span class="st">'Okinawa feature plot'</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://rdrr.io/r/graphics/plot.window.html" class="external-link">ylim</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">50</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu">theme_bw</span><span class="op">(</span><span class="op">)</span>

<span class="va">p2</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">osa_feat_bp_ovl</span><span class="op">)</span> <span class="op">+</span> <span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">type</span>, y<span class="op">=</span><span class="va">pct</span>, fill<span class="op">=</span><span class="va">type</span><span class="op">)</span> <span class="op">+</span> <span class="fu">geom_bar</span><span class="op">(</span>stat<span class="op">=</span><span class="st">'identity'</span><span class="op">)</span> <span class="op">+</span> <span class="fu">ylab</span><span class="op">(</span><span class="st">'percent overlapping breakpoint region'</span><span class="op">)</span> <span class="op">+</span> <span class="fu">ggtitle</span><span class="op">(</span><span class="st">'Osaka feature plot'</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://rdrr.io/r/graphics/plot.window.html" class="external-link">ylim</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">50</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu">theme_bw</span><span class="op">(</span><span class="op">)</span>

<span class="va">p3</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">bar_feat_bp_ovl</span><span class="op">)</span> <span class="op">+</span> <span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">type</span>, y<span class="op">=</span><span class="va">pct</span>, fill<span class="op">=</span><span class="va">type</span><span class="op">)</span> <span class="op">+</span> <span class="fu">geom_bar</span><span class="op">(</span>stat<span class="op">=</span><span class="st">'identity'</span><span class="op">)</span> <span class="op">+</span> <span class="fu">ylab</span><span class="op">(</span><span class="st">'percent overlapping breakpoint region'</span><span class="op">)</span> <span class="op">+</span> <span class="fu">ggtitle</span><span class="op">(</span><span class="st">'Barcelona feature plot'</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://rdrr.io/r/graphics/plot.window.html" class="external-link">ylim</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">50</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu">theme_bw</span><span class="op">(</span><span class="op">)</span>

<span class="op">(</span> <span class="va">p1</span> <span class="op">|</span> <span class="va">p2</span> <span class="op">|</span> <span class="va">p3</span><span class="op">)</span></code></pre></div>
<p><img src="GenomicFeatures_files/figure-html/operon_summary_stats_plots_2-1.png" width="1152"></p>
<div class="sourceCode" id="cb90"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/cbind.html" class="external-link">rbind</a></span><span class="op">(</span><span class="va">oki_feat_bp_ovl</span>, <span class="va">osa_feat_bp_ovl</span>, <span class="va">bar_feat_bp_ovl</span><span class="op">)</span> |&gt;
  <span class="fu">ggplot</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">aes</span><span class="op">(</span><span class="va">type</span>, <span class="va">pct</span>, fill <span class="op">=</span> <span class="va">species</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">geom_bar</span><span class="op">(</span>stat <span class="op">=</span> <span class="st">"identity"</span>, position<span class="op">=</span><span class="st">"dodge"</span><span class="op">)</span>  <span class="op">+</span>
  <span class="fu">scale_y_continuous</span><span class="op">(</span><span class="st">"Percent features overlaping breakpoint regions"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">scale_x_discrete</span><span class="op">(</span><span class="st">"Feature type"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">theme_bw</span><span class="op">(</span><span class="op">)</span></code></pre></div>
<p><img src="GenomicFeatures_files/figure-html/operon_overlap_stats_plot-1.svg" width="700"></p>
</div>
<div class="section level2">
<h2 id="coverage">Coverage<a class="anchor" aria-label="anchor" href="#coverage"></a>
</h2>
<p><em>Needs coverage information to be recomputed</em></p>
<p>Another characteristic of alignment stops that we can examine is the coverage depth over said alignment stops. By aligning raw reads to assembled genomes, we can obtain per-base coverage depth for target or query genome. Low coverage directly on and surrounding an alignment stop may suggest unreliability in it being a true breakpoint. The function <code>bp_coverage</code> returns a GRanges object of the bps, with associated averaged and point converage in the metadatacolumns.</p>
<div class="sourceCode" id="cb91"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># cov_gr_O_Oki &lt;- bp_coverage(gr_ob = q_Oki, cov_gr = Oki_cov_pb, win = 50)</span>
<span class="co"># cov_gr_O_Oki</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="session-information">Session information<a class="anchor" aria-label="anchor" href="#session-information"></a>
</h2>
<div class="sourceCode" id="cb92"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## R version 4.1.2 (2021-11-01)</span>
<span class="co">## Platform: x86_64-pc-linux-gnu (64-bit)</span>
<span class="co">## Running under: Debian GNU/Linux bookworm/sid</span>
<span class="co">## </span>
<span class="co">## Matrix products: default</span>
<span class="co">## BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.10.0</span>
<span class="co">## LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.10.0</span>
<span class="co">## </span>
<span class="co">## locale:</span>
<span class="co">##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span>
<span class="co">##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span>
<span class="co">##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span>
<span class="co">##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span>
<span class="co">##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span>
<span class="co">## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span>
<span class="co">## </span>
<span class="co">## attached base packages:</span>
<span class="co">## [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span>
<span class="co">## [8] base     </span>
<span class="co">## </span>
<span class="co">## other attached packages:</span>
<span class="co">##  [1] BSgenome.Oidioi.genoscope.OdB3_1.0.0   </span>
<span class="co">##  [2] BSgenome.Oidioi.OIST.AOM.5.5f_1.0.1    </span>
<span class="co">##  [3] BSgenome.Oidioi.OIST.KUM.M3.7f_1.0.1   </span>
<span class="co">##  [4] BSgenome.Oidioi.OIST.Bar2.p4_1.0.1     </span>
<span class="co">##  [5] BSgenome.Oidioi.OIST.OSKA2016v1.9_1.0.0</span>
<span class="co">##  [6] BSgenome.Oidioi.OIST.OKI2018.I69_1.0.1 </span>
<span class="co">##  [7] patchwork_1.1.1                        </span>
<span class="co">##  [8] heatmaps_1.18.0                        </span>
<span class="co">##  [9] GenomicFeatures_1.46.5                 </span>
<span class="co">## [10] AnnotationDbi_1.56.2                   </span>
<span class="co">## [11] Biobase_2.54.0                         </span>
<span class="co">## [12] OikScrambling_4.1.0                    </span>
<span class="co">## [13] ggplot2_3.3.5                          </span>
<span class="co">## [14] GenomicBreaks_0.13.1                   </span>
<span class="co">## [15] BSgenome_1.62.0                        </span>
<span class="co">## [16] rtracklayer_1.54.0                     </span>
<span class="co">## [17] Biostrings_2.62.0                      </span>
<span class="co">## [18] XVector_0.34.0                         </span>
<span class="co">## [19] GenomicRanges_1.46.1                   </span>
<span class="co">## [20] GenomeInfoDb_1.30.1                    </span>
<span class="co">## [21] IRanges_2.28.0                         </span>
<span class="co">## [22] S4Vectors_0.32.3                       </span>
<span class="co">## [23] BiocGenerics_0.40.0                    </span>
<span class="co">## </span>
<span class="co">## loaded via a namespace (and not attached):</span>
<span class="co">##   [1] utf8_1.2.2                  R.utils_2.11.0             </span>
<span class="co">##   [3] tidyselect_1.1.2            lme4_1.1-28                </span>
<span class="co">##   [5] poweRlaw_0.70.6             RSQLite_2.2.10             </span>
<span class="co">##   [7] htmlwidgets_1.5.4           grid_4.1.2                 </span>
<span class="co">##   [9] BiocParallel_1.28.3         munsell_0.5.0              </span>
<span class="co">##  [11] ragg_1.2.1                  withr_2.5.0                </span>
<span class="co">##  [13] colorspace_2.0-3            filelock_1.0.2             </span>
<span class="co">##  [15] highr_0.9                   knitr_1.37                 </span>
<span class="co">##  [17] rstudioapi_0.13             genoPlotR_0.8.11           </span>
<span class="co">##  [19] MatrixGenerics_1.6.0        labeling_0.4.2             </span>
<span class="co">##  [21] GenomeInfoDbData_1.2.7      bit64_4.0.5                </span>
<span class="co">##  [23] farver_2.1.0                rprojroot_2.0.2            </span>
<span class="co">##  [25] vctrs_0.3.8                 generics_0.1.2             </span>
<span class="co">##  [27] xfun_0.30                   BiocFileCache_2.2.1        </span>
<span class="co">##  [29] ggseqlogo_0.1               R6_2.5.1                   </span>
<span class="co">##  [31] locfit_1.5-9.5              bitops_1.0-7               </span>
<span class="co">##  [33] cachem_1.0.6                DelayedArray_0.20.0        </span>
<span class="co">##  [35] assertthat_0.2.1            BiocIO_1.4.0               </span>
<span class="co">##  [37] scales_1.1.1                nnet_7.3-16                </span>
<span class="co">##  [39] gtable_0.3.0                weights_1.0.4              </span>
<span class="co">##  [41] seqLogo_1.60.0              rlang_1.0.2                </span>
<span class="co">##  [43] systemfonts_1.0.3           splines_4.1.2              </span>
<span class="co">##  [45] broom_0.7.12                checkmate_2.0.0            </span>
<span class="co">##  [47] yaml_2.3.5                  reshape2_1.4.4             </span>
<span class="co">##  [49] abind_1.4-5                 backports_1.4.1            </span>
<span class="co">##  [51] Hmisc_4.6-0                 tools_4.1.2                </span>
<span class="co">##  [53] ellipsis_0.3.2              jquerylib_0.1.4            </span>
<span class="co">##  [55] RColorBrewer_1.1-2          Rcpp_1.0.8.3               </span>
<span class="co">##  [57] plyr_1.8.6                  base64enc_0.1-3            </span>
<span class="co">##  [59] progress_1.2.2              zlibbioc_1.40.0            </span>
<span class="co">##  [61] purrr_0.3.4                 RCurl_1.98-1.6             </span>
<span class="co">##  [63] prettyunits_1.1.1           rpart_4.1-15               </span>
<span class="co">##  [65] SummarizedExperiment_1.24.0 cluster_2.1.2              </span>
<span class="co">##  [67] fs_1.5.2                    magrittr_2.0.2             </span>
<span class="co">##  [69] data.table_1.14.2           matrixStats_0.61.0         </span>
<span class="co">##  [71] hms_1.1.1                   fftwtools_0.9-11           </span>
<span class="co">##  [73] evaluate_0.15               xtable_1.8-4               </span>
<span class="co">##  [75] XML_3.99-0.9                jpeg_0.1-9                 </span>
<span class="co">##  [77] gridExtra_2.3               compiler_4.1.2             </span>
<span class="co">##  [79] biomaRt_2.50.3              tibble_3.1.6               </span>
<span class="co">##  [81] mice_3.14.0                 KernSmooth_2.23-20         </span>
<span class="co">##  [83] crayon_1.5.0                minqa_1.2.4                </span>
<span class="co">##  [85] R.oo_1.24.0                 htmltools_0.5.2            </span>
<span class="co">##  [87] tzdb_0.2.0                  tiff_0.1-11                </span>
<span class="co">##  [89] Formula_1.2-4               tidyr_1.2.0                </span>
<span class="co">##  [91] DBI_1.1.2                   dbplyr_2.1.1               </span>
<span class="co">##  [93] MASS_7.3-54                 rappdirs_0.3.3             </span>
<span class="co">##  [95] boot_1.3-28                 rGADEM_2.42.0              </span>
<span class="co">##  [97] Matrix_1.4-0                ade4_1.7-18                </span>
<span class="co">##  [99] readr_2.1.2                 cli_3.2.0                  </span>
<span class="co">## [101] R.methodsS3_1.8.1           gdata_2.18.0               </span>
<span class="co">## [103] parallel_4.1.2              pkgconfig_2.0.3            </span>
<span class="co">## [105] pkgdown_2.0.2               GenomicAlignments_1.30.0   </span>
<span class="co">## [107] foreign_0.8-81              xml2_1.3.3                 </span>
<span class="co">## [109] annotate_1.72.0             bslib_0.3.1                </span>
<span class="co">## [111] stringr_1.4.0               digest_0.6.29              </span>
<span class="co">## [113] pracma_2.3.8                CNEr_1.30.0                </span>
<span class="co">## [115] rmarkdown_2.11              htmlTable_2.4.0            </span>
<span class="co">## [117] restfulr_0.0.13             curl_4.3.2                 </span>
<span class="co">## [119] EBImage_4.36.0              Rsamtools_2.10.0           </span>
<span class="co">## [121] gtools_3.9.2                rjson_0.2.21               </span>
<span class="co">## [123] nloptr_1.2.2.3              lifecycle_1.0.1            </span>
<span class="co">## [125] nlme_3.1-153                jsonlite_1.8.0             </span>
<span class="co">## [127] desc_1.4.1                  fansi_1.0.2                </span>
<span class="co">## [129] pillar_1.7.0                lattice_0.20-45            </span>
<span class="co">## [131] KEGGREST_1.34.0             fastmap_1.1.0              </span>
<span class="co">## [133] httr_1.4.2                  plotrix_3.8-2              </span>
<span class="co">## [135] survival_3.2-13             GO.db_3.14.0               </span>
<span class="co">## [137] glue_1.6.2                  png_0.1-7                  </span>
<span class="co">## [139] bit_4.0.4                   stringi_1.7.6              </span>
<span class="co">## [141] sass_0.4.0                  blob_1.2.2                 </span>
<span class="co">## [143] textshaping_0.3.6           latticeExtra_0.6-29        </span>
<span class="co">## [145] memoise_2.0.1               dplyr_1.0.8</span></code></pre>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Aleksandra Bliznina, Michael Mansfield, Charles Plessy.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.2.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
