<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Genomic features of the breakpoints • OikScrambling</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Genomic features of the breakpoints">
<meta property="og:description" content="OikScrambling">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">OikScrambling</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">3.0.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/OikScrambling.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/CNEs.html">Conserved non-coding elements</a>
    </li>
    <li>
      <a href="../articles/CoalescingCutoffs.html">Modifying coalescent cutoffs</a>
    </li>
    <li>
      <a href="../articles/ColinearityInterruptors.html">Regions interrupting colinearity</a>
    </li>
    <li>
      <a href="../articles/DistanceMeasures.html">Distance Measures</a>
    </li>
    <li>
      <a href="../articles/GenomicFeatures.html">Genomic features of the breakpoints</a>
    </li>
    <li>
      <a href="../articles/Inversions.html">Inversions</a>
    </li>
    <li>
      <a href="../articles/LoadGenomicBreaks.html">Load genomic break data</a>
    </li>
    <li>
      <a href="../articles/OrthogroupSyntenies.html">Orthogroup syntenies</a>
    </li>
    <li>
      <a href="../articles/OxfordPlots.html">Whole-genome dot-plots</a>
    </li>
    <li>
      <a href="../articles/PAC3.html">Region of interest: PAC3</a>
    </li>
    <li>
      <a href="../articles/PWM.html">Motif search and analysis</a>
    </li>
    <li>
      <a href="../articles/ParallelPlots.html">Whole-chromosome parallel plots</a>
    </li>
    <li>
      <a href="../articles/RegionWidths.html">Width of the aligned and mapped regions.</a>
    </li>
    <li>
      <a href="../articles/RegionsOfInterest.html">Regions of interest</a>
    </li>
    <li>
      <a href="../articles/RepeatRegions.html">Repeat regions</a>
    </li>
    <li>
      <a href="../articles/SegmentMovements.html">Segment movement distances</a>
    </li>
    <li>
      <a href="../articles/TFBS.html">Transcription factor binding sites</a>
    </li>
    <li>
      <a href="../articles/TandemSiteDuplications.html">Tandem site duplications</a>
    </li>
    <li>
      <a href="../articles/Translocations.html">Translocations</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/oist/LuscombeU_OikScrambling/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="GenomicFeatures_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Genomic features of the breakpoints</h1>
                        <h4 data-toc-skip class="author">Charlotte West</h4>
                        <h4 data-toc-skip class="author">Charles Plessy</h4>
            
            <h4 data-toc-skip class="date">12 April, 2022</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/oist/LuscombeU_OikScrambling/blob/HEAD/vignettes/GenomicFeatures.Rmd" class="external-link"><code>vignettes/GenomicFeatures.Rmd</code></a></small>
      <div class="hidden name"><code>GenomicFeatures.Rmd</code></div>

    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">knitr</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/knitr/man/opts_chunk.html" class="external-link">opts_chunk</a></span><span class="op">$</span><span class="fu">set</span><span class="op">(</span>cache <span class="op">=</span> <span class="cn">TRUE</span>, cache.lazy <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
<span class="fu">knitr</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/knitr/man/opts_knit.html" class="external-link">opts_knit</a></span><span class="op">$</span><span class="fu">set</span><span class="op">(</span>verbose <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<div class="section level2">
<h2 id="load-packages-and-data">Load packages and data<a class="anchor" aria-label="anchor" href="#load-packages-and-data"></a>
</h2>
<p>See <code>?OikScrambling:::loadAllGenomes</code>, <code>?OikScrambling:::loadAllTranscriptsGR</code>, and <code><a href="../articles/LoadGenomicBreaks.html">vignette("LoadGenomicBreaks", package = "OikScrambling")</a></code> for how the different objects are prepared.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/oist/LuscombeU_OikScrambling" class="external-link">'OikScrambling'</a></span><span class="op">)</span>   |&gt; <span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://bioconductor.org/packages/GenomicFeatures" class="external-link">'GenomicFeatures'</a></span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st">'heatmaps'</span><span class="op">)</span>        |&gt; <span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="op">)</span>
<span class="va">genomes</span>     <span class="op">&lt;-</span> <span class="fu">OikScrambling</span><span class="fu">:::</span><span class="fu"><a href="../reference/loadAllGenomes.html">loadAllGenomes</a></span><span class="op">(</span>compat <span class="op">=</span> <span class="cn">F</span><span class="op">)</span>
<span class="va">annots</span>      <span class="op">&lt;-</span> <span class="fu">OikScrambling</span><span class="fu">:::</span><span class="fu"><a href="../reference/loadAllAnnotations.html">loadAllAnnotations</a></span><span class="op">(</span><span class="op">)</span>   |&gt; <span class="fu"><a href="https://rdrr.io/r/base/warning.html" class="external-link">suppressWarnings</a></span><span class="op">(</span><span class="op">)</span>
<span class="va">reps</span>        <span class="op">&lt;-</span> <span class="fu">OikScrambling</span><span class="fu">:::</span><span class="fu"><a href="../reference/loadAllRepeats.html">loadAllRepeats</a></span><span class="op">(</span>compat <span class="op">=</span> <span class="cn">F</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/load.html" class="external-link">load</a></span><span class="op">(</span><span class="st">"BreakPoints.Rdata"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"rGADEM"</span><span class="op">)</span>    |&gt; <span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"ggseqlogo"</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="op">)</span>
<span class="va">ces</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="st">"CEs.Rds"</span><span class="op">)</span>
<span class="va">tfbs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="st">"pwmMatchesOki_12_95.Rds"</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="nucleic-acid-content-heatmaps">Nucleic acid content heatmaps<a class="anchor" aria-label="anchor" href="#nucleic-acid-content-heatmaps"></a>
</h2>
<p>The <code><a href="https://oist.github.io/GenomicBreaks/reference/bp_heatmap.html" class="external-link">GenomicBreaks::bp_heatmap()</a></code> function plots nucleic acid content in windows centered on alignment stops.</p>
<div class="section level3">
<h3 id="gc">GC<a class="anchor" aria-label="anchor" href="#gc"></a>
</h3>
<p>Lets look at <code>GC</code> content before and after coalescing:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#hm_GC1 &lt;- sapply(gbs,  bp_heatmap, basep_range = 1000, pat = "GC", direction = "left") |&gt; SimpleList()</span>
<span class="va">hm_GC2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">coa</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">15</span><span class="op">]</span>,  <span class="va">bp_heatmap</span>, win <span class="op">=</span> <span class="fl">1000</span>, pat <span class="op">=</span> <span class="st">"GC"</span>, direction <span class="op">=</span> <span class="st">"left"</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/SimpleList-class.html" class="external-link">SimpleList</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">#hm_GC3 &lt;- sapply(coa2, bp_heatmap, basep_range = 1000, pat = "GC", direction = "left") |&gt; SimpleList()</span>

<span class="va">maxHeight</span> <span class="op">&lt;-</span> <span class="fl">2000</span>
<span class="co">#sm_GC1 &lt;- sapply(hm_GC1, smoothHeatmap, output.size = c(maxHeight,     500), algorithm = "kernel") |&gt; SimpleList()</span>
<span class="va">sm_GC2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">hm_GC2</span>, <span class="va">smoothHeatmap</span>, output.size <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="va">maxHeight</span>,     <span class="fl">500</span><span class="op">)</span>, algorithm <span class="op">=</span> <span class="st">"kernel"</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/SimpleList-class.html" class="external-link">SimpleList</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">#sm_GC3 &lt;- sapply(hm_GC3, smoothHeatmap, output.size = c(maxHeight,     500), algorithm = "kernel") |&gt; SimpleList()</span>

<span class="co">#plotHeatmapList(sm_GC1[1:5])</span>

<span class="fu"><a href="https://rdrr.io/pkg/heatmaps/man/plotHeatmapList.html" class="external-link">plotHeatmapList</a></span><span class="op">(</span><span class="va">sm_GC2</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span><span class="op">)</span></code></pre></div>
<p><img src="GenomicFeatures_files/figure-html/unnamed-chunk-1-1.png" width="1152"></p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#plotHeatmapList(sm_GC2[6:10])</span>
<span class="co">#plotHeatmapList(sm_GC2[11:15])</span>

<span class="co">#plotHeatmapList(sm_GC3[1:5])</span></code></pre></div>
<p>The alignment stops are ordered in such a way that start (or left breaks) are centred on the top half of the plot, and end alignment stops are centred on the bottom. This is why we see different directionality in the plots. At the alignment stops, the GC content seems to be lower, but higher once within the aligned region, except for the Oki – Kum pair.</p>
</div>
<div class="section level3">
<h3 id="tata">TATA<a class="anchor" aria-label="anchor" href="#tata"></a>
</h3>
<p>Now looking at TATA content:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#hm_GC1_TATA &lt;- sapply(gbs, bp_heatmap, basep_range = 1000, pat = "TATA", direction = "left") |&gt; SimpleList()</span>
<span class="va">hm_GC2_TATA</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">coa</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">15</span><span class="op">]</span>, <span class="va">bp_heatmap</span>, window <span class="op">=</span> <span class="fl">1000</span>, pat <span class="op">=</span> <span class="st">"TATA"</span>, direction <span class="op">=</span> <span class="st">"left"</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/SimpleList-class.html" class="external-link">SimpleList</a></span><span class="op">(</span><span class="op">)</span>

<span class="va">maxHeight</span> <span class="op">&lt;-</span> <span class="fl">2000</span>
<span class="co">#sm_GC1_TATA &lt;- sapply(hm_GC1_TATA, smoothHeatmap, output.size = c(maxHeight,     500), algorithm = "kernel") |&gt; SimpleList()</span>
<span class="va">sm_GC2_TATA</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">hm_GC2_TATA</span>, <span class="va">smoothHeatmap</span>, output.size <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="va">maxHeight</span>,     <span class="fl">500</span><span class="op">)</span>, algorithm <span class="op">=</span> <span class="st">"kernel"</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/SimpleList-class.html" class="external-link">SimpleList</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">#sm_GC3_TATA &lt;- sapply(hm_GC3_TATA, smoothHeatmap, output.size = c(maxHeight,     500), algorithm = "kernel") |&gt; SimpleList()</span>

<span class="co">#plotHeatmapList(sm_GC1_TATA)</span>
<span class="fu"><a href="https://rdrr.io/pkg/heatmaps/man/plotHeatmapList.html" class="external-link">plotHeatmapList</a></span><span class="op">(</span><span class="va">sm_GC2_TATA</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span><span class="op">)</span></code></pre></div>
<p><img src="GenomicFeatures_files/figure-html/TATA-1.png" width="1152"></p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#plotHeatmapList(sm_GC3_TATA)</span></code></pre></div>
<p>The directionaility here is consistent with the <code>GC</code> content analysis. Furthermore; “TATA box sequence can act as a basal promoter element not only for RNA polymerase II (RNAP II) transcription, but also for transcription by RNA polymerase III (RNAP III)” - Wang Y, Jensen RC, Stumph WE. Role of TATA box sequence and orientation in determining RNA polymerase II/III transcription specificity. Nucleic Acids Res. 1996;24(15):3100–3106. <a href="doi:10.1093/nar/24.15.3100" class="uri">doi:10.1093/nar/24.15.3100</a>. However, the areas are not particularly enriched for TATA boxes, so it does not necessarily show that breakpoints are occurring directly after promoter regions.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#hm_GC1_mot &lt;- sapply(gbs, bp_heatmap, basep_range = 1000, pat = "AGCNGC", direction = "left") |&gt; SimpleList()</span>
<span class="va">hm_GC2_mot</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">coa</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">15</span><span class="op">]</span>, <span class="va">bp_heatmap</span>, window <span class="op">=</span> <span class="fl">1000</span>, pat <span class="op">=</span> <span class="st">"AGCNGC"</span>, direction <span class="op">=</span> <span class="st">"left"</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/SimpleList-class.html" class="external-link">SimpleList</a></span><span class="op">(</span><span class="op">)</span>

<span class="va">maxHeight</span> <span class="op">&lt;-</span> <span class="fl">2000</span>
<span class="co">#sm_GC1_mot &lt;- sapply(hm_GC1_mot, smoothHeatmap, output.size = c(maxHeight,     500), algorithm = "kernel") |&gt; SimpleList()</span>
<span class="va">sm_GC2_mot</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">hm_GC2_mot</span>, <span class="va">smoothHeatmap</span>, output.size <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="va">maxHeight</span>,     <span class="fl">500</span><span class="op">)</span>, algorithm <span class="op">=</span> <span class="st">"kernel"</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/SimpleList-class.html" class="external-link">SimpleList</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">#sm_GC3_mot &lt;- sapply(hm_GC3_mot, smoothHeatmap, output.size = c(maxHeight,     500), algorithm = "kernel") |&gt; SimpleList()</span>

<span class="co">#plotHeatmapList(sm_GC1_mot)</span>
<span class="fu"><a href="https://rdrr.io/pkg/heatmaps/man/plotHeatmapList.html" class="external-link">plotHeatmapList</a></span><span class="op">(</span><span class="va">sm_GC2_mot</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span><span class="op">)</span></code></pre></div>
<p><img src="GenomicFeatures_files/figure-html/AAGCsGCwwmkCGrCTTyn-1.png" width="1152"></p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#plotHeatmapList(sm_GC3_mot)</span></code></pre></div>
</div>
<div class="section level3">
<h3 id="cluster-analysis">Cluster analysis<a class="anchor" aria-label="anchor" href="#cluster-analysis"></a>
</h3>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># # Accessory functions</span>
<span class="co"># clusterHeatmap &lt;- function(hm, k = 2) kmeans(image(hm), k)</span>
<span class="co"># </span>
<span class="co"># orderHeatmap &lt;- function(hm, cl) {</span>
<span class="co">#   image(hm) &lt;- image(hm)[order(cl$cluster),]</span>
<span class="co">#   hm</span>
<span class="co"># }</span>
<span class="co"># </span>
<span class="co"># subsetHeatmap &lt;- function(hm, bool) {</span>
<span class="co">#   image(hm) &lt;- image(hm)[bool,]</span>
<span class="co">#   hm</span>
<span class="co"># }</span>
<span class="co"># </span>
<span class="co"># # DAta</span>
<span class="co"># hm_GC   &lt;- bp_heatmap(      gr_Oki_O, basep_range  = 1000, pat = "GC",     direction = "left")</span>
<span class="co"># hm_TATA &lt;- bp_heatmap(      gr_Oki_O, basep_range  = 1000, pat = "TATA",   direction = "left")</span>
<span class="co"># hm_tan  &lt;- feature_coverage(gr_Oki_O, tan_Oki, win = 1000, lab = "Tandem", direction = "left")</span>
<span class="co"># </span>
<span class="co"># # Define 3 K-mean clusters</span>
<span class="co"># clusterHeatmap(hm_GC, 3) -&gt; hm_cl_k4</span>
<span class="co"># </span>
<span class="co"># # Order and smooth</span>
<span class="co"># shm_GC &lt;- orderHeatmap(hm_GC, hm_cl_k4) %&gt;%</span>
<span class="co">#   smoothHeatmap(output.size = c(2000,500), algorithm = "kernel")</span>
<span class="co"># </span>
<span class="co"># shm_TATA &lt;- orderHeatmap(hm_TATA, hm_cl_k4) %&gt;%</span>
<span class="co">#   smoothHeatmap(output.size = c(2000,500), algorithm = "kernel")</span>
<span class="co"># </span>
<span class="co"># shm_tan &lt;- orderHeatmap(hm_tan, hm_cl_k4) %&gt;%</span>
<span class="co">#   smoothHeatmap(output.size = c(2000,500), algorithm = "kernel")</span>
<span class="co"># </span>
<span class="co"># # Plot the ordered data</span>
<span class="co"># plotHeatmapList(list(shm_GC, shm_TATA, shm_tan))</span>
<span class="co"># par(mfrow = c(1,1)) # plotHeatmapList messes with mrfrow...</span>
<span class="co"># plotHeatmapMeta(list(hm_GC, hm_TATA, hm_tan))</span>
<span class="co"># </span>
<span class="co"># plotHeatmapMeta(list(subsetHeatmap(hm_GC, hm_cl_k4$cluster == 1),</span>
<span class="co">#                      subsetHeatmap(hm_GC, hm_cl_k4$cluster == 2),</span>
<span class="co">#                      subsetHeatmap(hm_GC, hm_cl_k4$cluster == 3)))</span>
<span class="co"># </span>
<span class="co"># kmeanOrderedPlot &lt;- function (hm, k=3, seed = 1) {</span>
<span class="co">#   set.seed(seed)</span>
<span class="co">#   </span>
<span class="co">#   cl &lt;- kmeans(image(hm), k)</span>
<span class="co">#   mat &lt;- image(hm)[order(cl$cluster),]</span>
<span class="co">#   </span>
<span class="co">#   kmhm = Heatmap(</span>
<span class="co">#     mat,</span>
<span class="co">#     coords=c(-500,500),</span>
<span class="co">#     label="kmeans",</span>
<span class="co">#     scale=range(mat))</span>
<span class="co">#   </span>
<span class="co">#   plotHeatmapList(kmhm,</span>
<span class="co">#                   cex.label=1.5,</span>
<span class="co">#                   partition=as.vector(table(cl$cluster)),</span>
<span class="co">#                   partition.legend=TRUE,</span>
<span class="co">#                   partition.lines=TRUE,</span>
<span class="co">#                   legend=TRUE,</span>
<span class="co">#                   legend.pos="r",</span>
<span class="co">#                   legend.width=0.3)</span>
<span class="co"># }</span>
<span class="co"># </span>
<span class="co"># kmeanOrderedPlot(shm_GC)</span></code></pre></div>
</div>
<div class="section level3">
<h3 id="evidence-for-breakpoint-hotspots">Evidence for breakpoint hotspots<a class="anchor" aria-label="anchor" href="#evidence-for-breakpoint-hotspots"></a>
</h3>
<p>The function <code><a href="https://oist.github.io/GenomicBreaks/reference/bp_pair_analysis.html" class="external-link">?GenomicBreaks::bp_pair_analysis</a></code> takes two pairwise alignments with the same target genome, and plots the alignment stops on to the centred alignment stops of the other.</p>
<p>In the plot below, Osaka is there consistent target genome, and as such acts as a coordinate system to relate alignment stops across genomes. Lets produce two of these 3-way analyses; one between Osaka-Okinawa-Norway, and one between Osaka-Okinawa-Aomori.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">bp_pair_analysis</span><span class="op">(</span><span class="va">gbs</span><span class="op">$</span><span class="va">Osa_Oki</span>, <span class="va">gbs</span><span class="op">$</span><span class="va">Osa_Nor</span>, win <span class="op">=</span> <span class="fl">1000</span>, lab <span class="op">=</span> <span class="st">"Oki~Nor"</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/heatmaps/man/plotHeatmapMeta.html" class="external-link">plotHeatmapMeta</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<p><img src="GenomicFeatures_files/figure-html/breakpoint_hotspot_gbs-1.png" width="700"></p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">bp_pair_analysis</span><span class="op">(</span><span class="va">gbs</span><span class="op">$</span><span class="va">Osa_Oki</span>, <span class="va">gbs</span><span class="op">$</span><span class="va">Osa_Aom</span>, win <span class="op">=</span> <span class="fl">1000</span>, lab <span class="op">=</span> <span class="st">"Oki~Aom"</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/heatmaps/man/plotHeatmapMeta.html" class="external-link">plotHeatmapMeta</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<p><img src="GenomicFeatures_files/figure-html/breakpoint_hotspot_gbs-2.png" width="700"></p>
<p>The accumulation of alignment breaks of one pairwise alignment onto another suggests the existence of breakpoint hotspots; regions where breaks are far more likely to occur. Intuitively, this would lead to the presumption of synteny blocks, also. The pattern is consistent after coalescing, too;</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">bp_pair_analysis</span><span class="op">(</span><span class="va">coa</span><span class="op">$</span><span class="va">Osa_Oki</span>, <span class="va">coa</span><span class="op">$</span><span class="va">Osa_Nor</span>, win <span class="op">=</span> <span class="fl">1000</span>, lab <span class="op">=</span> <span class="st">"Oki~Nor"</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/heatmaps/man/plotHeatmapMeta.html" class="external-link">plotHeatmapMeta</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<p><img src="GenomicFeatures_files/figure-html/breakpoint_hotspot_coa-1.png" width="700"></p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">bp_pair_analysis</span><span class="op">(</span><span class="va">coa</span><span class="op">$</span><span class="va">Osa_Oki</span>, <span class="va">coa</span><span class="op">$</span><span class="va">Osa_Aom</span>, win <span class="op">=</span> <span class="fl">1000</span>, lab <span class="op">=</span> <span class="st">"Oki~Aom"</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/heatmaps/man/plotHeatmapMeta.html" class="external-link">plotHeatmapMeta</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<p><img src="GenomicFeatures_files/figure-html/breakpoint_hotspot_coa-2.png" width="700"></p>
</div>
</div>
<div class="section level2">
<h2 id="gene-feature-coverage-around-aligned-and-unaligned-genomic-regions-">Gene feature coverage around aligned and unaligned genomic regions.<a class="anchor" aria-label="anchor" href="#gene-feature-coverage-around-aligned-and-unaligned-genomic-regions-"></a>
</h2>
<div class="section level3">
<h3 id="model-operon-areas">Model operon areas<a class="anchor" aria-label="anchor" href="#model-operon-areas"></a>
</h3>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">calcOperons</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">genes</span>, <span class="va">window</span> <span class="op">=</span> <span class="fl">200</span><span class="op">)</span> <span class="op">{</span>
  <span class="co"># Sort the genes ignoring strand.</span>
  <span class="va">g</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/sort.html" class="external-link">sort</a></span><span class="op">(</span><span class="va">genes</span>, ignore.strand<span class="op">=</span><span class="cn">T</span><span class="op">)</span>
  <span class="co"># Expand by `window` nucleotides (default 200)</span>
  <span class="co"># We suppress warnings because coordinates at the edges of seqfeatures can</span>
  <span class="co"># become transiently negative.</span>
  <span class="va">g</span> <span class="op">&lt;-</span> <span class="op">(</span><span class="va">g</span> <span class="op">+</span> <span class="va">window</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/r/base/warning.html" class="external-link">suppressWarnings</a></span><span class="op">(</span><span class="op">)</span>
  <span class="co"># Give a unique ID to all runs of successive genes on the same strand</span>
  <span class="co"># https://stackoverflow.com/questions/21421047</span>
  <span class="co"># op$idx &lt;- data.table::rleid(as.character(strand(g)))</span>
  <span class="va">g</span><span class="op">$</span><span class="va">idx</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Rle-class.html" class="external-link">Rle</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">rep</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq_along</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Rle-class.html" class="external-link">runValue</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/strand.html" class="external-link">strand</a></span><span class="op">(</span><span class="va">g</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Rle-class.html" class="external-link">runLength</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/strand.html" class="external-link">strand</a></span><span class="op">(</span><span class="va">g</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
  <span class="co"># Split the genes by run ID</span>
  <span class="va">opL</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/splitAsList.html" class="external-link">split</a></span><span class="op">(</span><span class="va">g</span>, <span class="va">g</span><span class="op">$</span><span class="va">idx</span><span class="op">)</span>
  <span class="co"># Remove all runs of length one</span>
  <span class="va">opL</span> <span class="op">&lt;-</span> <span class="va">opL</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Rle-class.html" class="external-link">runLength</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/strand.html" class="external-link">strand</a></span><span class="op">(</span><span class="va">g</span><span class="op">)</span><span class="op">)</span> <span class="op">&gt;</span> <span class="fl">1</span><span class="op">]</span>
  <span class="co"># Merge into operons</span>
  <span class="va">operons</span>   <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/unlist.html" class="external-link">unlist</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/List-utils.html" class="external-link">endoapply</a></span><span class="op">(</span><span class="va">opL</span>, <span class="va">reduce</span><span class="op">)</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/r/base/warning.html" class="external-link">suppressWarnings</a></span><span class="op">(</span><span class="op">)</span>
  <span class="co"># Removed un-merged genes</span>
  <span class="va">operons</span><span class="op">$</span><span class="va">n</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/findOverlaps-methods.html" class="external-link">countOverlaps</a></span><span class="op">(</span><span class="va">operons</span>, <span class="va">genes</span><span class="op">)</span>
  <span class="va">operons</span>   <span class="op">&lt;-</span> <span class="va">operons</span><span class="op">[</span><span class="va">operons</span><span class="op">$</span><span class="va">n</span> <span class="op">&gt;</span> <span class="fl">1</span><span class="op">]</span>
  <span class="co"># Remove the flanking sequences</span>
  <span class="va">operons</span> <span class="op">-</span> <span class="va">window</span>
<span class="op">}</span>

<span class="va">operons</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">annots</span>, \<span class="op">(</span><span class="va">a</span><span class="op">)</span> <span class="op">{</span><span class="va">a</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/GenomicFeatures/man/transcripts.html" class="external-link">genes</a></span><span class="op">(</span><span class="op">)</span> |&gt; <span class="fu">calcOperons</span><span class="op">(</span><span class="op">)</span> <span class="op">}</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/SimpleList-class.html" class="external-link">SimpleList</a></span><span class="op">(</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/r/base/warning.html" class="external-link">suppressWarnings</a></span><span class="op">(</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">operons</span>, <span class="va">length</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">##  Oki  Osa  Bar  Kum  Aom  Nor  Ply  Ros </span>
<span class="co">## 2892 2498 2209 2738 2403    0  902 1192</span></code></pre>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">operons</span>, \<span class="op">(</span><span class="va">o</span><span class="op">)</span> <span class="op">{</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">o</span><span class="op">$</span><span class="va">n</span><span class="op">)</span><span class="op">}</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## $Oki</span>
<span class="co">## </span>
<span class="co">##    2    3    4    5    6    7    8    9   10   13 </span>
<span class="co">## 1899  574  238   87   49   24   13    3    4    1 </span>
<span class="co">## </span>
<span class="co">## $Osa</span>
<span class="co">## </span>
<span class="co">##    2    3    4    5    6    7    8    9   11 </span>
<span class="co">## 1625  508  199   91   42   17    7    7    2 </span>
<span class="co">## </span>
<span class="co">## $Bar</span>
<span class="co">## </span>
<span class="co">##    2    3    4    5    6    7    8    9   10   14 </span>
<span class="co">## 1404  437  190   89   47   21   11    6    3    1 </span>
<span class="co">## </span>
<span class="co">## $Kum</span>
<span class="co">## </span>
<span class="co">##    2    3    4    5    6    7    8    9   10   11   12   14 </span>
<span class="co">## 1794  577  191   90   44   19    9    2    4    4    2    2 </span>
<span class="co">## </span>
<span class="co">## $Aom</span>
<span class="co">## </span>
<span class="co">##    2    3    4    5    6    7    8    9   10   11   14 </span>
<span class="co">## 1592  449  204   86   37   15    8    7    2    2    1 </span>
<span class="co">## </span>
<span class="co">## $Nor</span>
<span class="co">## &lt; table of extent 0 &gt;</span>
<span class="co">## </span>
<span class="co">## $Ply</span>
<span class="co">## </span>
<span class="co">##   2   3   4   5   6   7   8   9  13  14  15 </span>
<span class="co">## 790  82  16   6   1   1   1   1   1   1   2 </span>
<span class="co">## </span>
<span class="co">## $Ros</span>
<span class="co">## </span>
<span class="co">##    2    3    4    5    6    7    9   10   11   19   20   23   28 </span>
<span class="co">## 1035  123   22    3    1    1    1    1    1    1    1    1    1</span></code></pre>
</div>
<div class="section level3">
<h3 id="accessory-functions">Accessory functions<a class="anchor" aria-label="anchor" href="#accessory-functions"></a>
</h3>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">prepareFreqPlot</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">gb</span>, <span class="va">annot</span>, <span class="va">rep</span>, <span class="va">cne</span> <span class="op">=</span> <span class="cn">NULL</span>, <span class="va">tfbs</span> <span class="op">=</span> <span class="cn">NULL</span>, <span class="va">operons</span> <span class="op">=</span> <span class="cn">NULL</span>, <span class="va">win</span> <span class="op">=</span> <span class="fl">1000</span><span class="op">)</span> <span class="op">{</span>
  <span class="va">l</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="op">)</span>
  <span class="va">l</span><span class="op">$</span><span class="va">genes</span>  <span class="op">&lt;-</span> <span class="fu">feature_coverage</span><span class="op">(</span><span class="va">gb</span>,     <span class="fu"><a href="https://rdrr.io/pkg/GenomicFeatures/man/transcripts.html" class="external-link">genes</a></span><span class="op">(</span><span class="va">annot</span><span class="op">)</span>,         win <span class="op">=</span> <span class="va">win</span>, lab <span class="op">=</span> <span class="st">"genes"</span>,  direction <span class="op">=</span> <span class="st">"left"</span><span class="op">)</span>
  <span class="co">#l$cds    &lt;- feature_coverage(gb,     cds(annot),           win = win, lab = "CDS",    direction = "left")</span>
  <span class="va">l</span><span class="op">$</span><span class="va">intron</span> <span class="op">&lt;-</span> <span class="fu">feature_coverage</span><span class="op">(</span><span class="va">gb</span>,     <span class="fu"><a href="https://rdrr.io/pkg/GenomicFeatures/man/exonicParts.html" class="external-link">intronicParts</a></span><span class="op">(</span><span class="va">annot</span><span class="op">)</span>, win <span class="op">=</span> <span class="va">win</span>, lab <span class="op">=</span> <span class="st">"intron"</span>, direction <span class="op">=</span> <span class="st">"left"</span><span class="op">)</span>
  <span class="va">l</span><span class="op">$</span><span class="va">exon</span>   <span class="op">&lt;-</span> <span class="fu">feature_coverage</span><span class="op">(</span><span class="va">gb</span>,     <span class="fu"><a href="https://rdrr.io/pkg/GenomicFeatures/man/exonicParts.html" class="external-link">exonicParts</a></span><span class="op">(</span><span class="va">annot</span><span class="op">)</span>,   win <span class="op">=</span> <span class="va">win</span>, lab <span class="op">=</span> <span class="st">"exon"</span>,   direction <span class="op">=</span> <span class="st">"left"</span><span class="op">)</span>
  <span class="va">l</span><span class="op">$</span><span class="va">reps</span>   <span class="op">&lt;-</span> <span class="fu">feature_coverage</span><span class="op">(</span><span class="va">gb</span>,     <span class="va">rep</span>,                  win <span class="op">=</span> <span class="va">win</span>, lab <span class="op">=</span> <span class="st">"repeats"</span>,direction <span class="op">=</span> <span class="st">"left"</span><span class="op">)</span>
  <span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NULL.html" class="external-link">is.null</a></span><span class="op">(</span><span class="va">cne</span><span class="op">)</span><span class="op">)</span>
    <span class="va">l</span><span class="op">$</span><span class="va">CNE</span>   <span class="op">&lt;-</span> <span class="fu">feature_coverage</span><span class="op">(</span><span class="va">gb</span>,    <span class="va">cne</span>,                  win <span class="op">=</span> <span class="va">win</span>, lab <span class="op">=</span> <span class="st">"CNE"</span>,    direction <span class="op">=</span> <span class="st">"left"</span><span class="op">)</span>
  <span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NULL.html" class="external-link">is.null</a></span><span class="op">(</span><span class="va">tfbs</span><span class="op">)</span><span class="op">)</span>
    <span class="va">l</span><span class="op">$</span><span class="va">tfbs</span>  <span class="op">&lt;-</span> <span class="fu">feature_coverage</span><span class="op">(</span><span class="va">gb</span>,    <span class="va">tfbs</span>,                 win <span class="op">=</span> <span class="va">win</span>, lab <span class="op">=</span> <span class="st">"TFBS"</span>,   direction <span class="op">=</span> <span class="st">"left"</span><span class="op">)</span>
  <span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NULL.html" class="external-link">is.null</a></span><span class="op">(</span><span class="va">operons</span><span class="op">)</span><span class="op">)</span>
    <span class="va">l</span><span class="op">$</span><span class="va">operons</span><span class="op">&lt;-</span><span class="fu">feature_coverage</span><span class="op">(</span><span class="va">gb</span>,    <span class="va">operons</span>,              win <span class="op">=</span> <span class="va">win</span>, lab <span class="op">=</span> <span class="st">"operons"</span>,direction <span class="op">=</span> <span class="st">"left"</span><span class="op">)</span>
  <span class="co">#reorder</span>
  <span class="va">newOrder</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/order.html" class="external-link">order</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/colSums.html" class="external-link">rowSums</a></span><span class="op">(</span><span class="va">l</span><span class="op">$</span><span class="va">intron</span><span class="op">@</span><span class="va">image</span><span class="op">)</span><span class="op">)</span>
  <span class="va">l</span><span class="op">$</span><span class="va">intron</span><span class="op">@</span><span class="va">image</span> <span class="op">&lt;-</span> <span class="va">l</span><span class="op">$</span><span class="va">intron</span><span class="op">@</span><span class="va">image</span><span class="op">[</span><span class="va">newOrder</span>,<span class="op">]</span>
  <span class="va">l</span><span class="op">$</span><span class="va">exon</span><span class="op">@</span><span class="va">image</span>   <span class="op">&lt;-</span> <span class="va">l</span><span class="op">$</span><span class="va">exon</span><span class="op">@</span><span class="va">image</span><span class="op">[</span><span class="va">newOrder</span>,<span class="op">]</span>
  <span class="va">l</span>
<span class="op">}</span>

<span class="va">prepareGGFreqPlot</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">x</span>, <span class="va">desc</span> <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span> <span class="op">{</span>
  <span class="va">l</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span>, \<span class="op">(</span><span class="va">name</span><span class="op">)</span><span class="op">{</span>
    <span class="va">val</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/colSums.html" class="external-link">colSums</a></span><span class="op">(</span><span class="va">x</span><span class="op">[[</span><span class="va">name</span><span class="op">]</span><span class="op">]</span><span class="op">@</span><span class="va">image</span><span class="op">)</span>
    <span class="va">what</span> <span class="op">&lt;-</span> <span class="va">name</span>
    <span class="fu">tibble</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/tibble.html" class="external-link">tibble</a></span><span class="op">(</span><span class="va">val</span>, freq <span class="op">=</span> <span class="va">val</span> <span class="op">/</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">x</span><span class="op">[[</span><span class="va">name</span><span class="op">]</span><span class="op">]</span><span class="op">@</span><span class="va">image</span><span class="op">)</span>, <span class="va">what</span>, pos <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">val</span><span class="op">)</span> <span class="op">-</span> <span class="fu"><a href="https://rdrr.io/r/base/Round.html" class="external-link">round</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">val</span><span class="op">)</span><span class="op">)</span> <span class="op">/</span> <span class="fl">2</span>, <span class="va">desc</span><span class="op">)</span>
  <span class="op">}</span><span class="op">)</span>
   <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/do.call.html" class="external-link">do.call</a></span><span class="op">(</span><span class="va">rbind</span>, <span class="va">l</span><span class="op">)</span>
<span class="op">}</span></code></pre></div>
</div>
<div class="section level3">
<h3 id="centering-on-left-aligned-boundaries-">Centering on left-aligned boundaries.<a class="anchor" aria-label="anchor" href="#centering-on-left-aligned-boundaries-"></a>
</h3>
<p>Breakpoints are between mapped regions. If the unmapped region is broad we have little information where the breakpoint originally was. We set arbitrarly the breakpoints to the left-side (end-side) coordinate of genomic regions.</p>
<p>Mapped regions can consist of multiple colinear aligned regions, or be a single uncoalesced aligned region. We will see that they have different properties.</p>
<p>Our annotations is mostly made of coding exons, so the features <em>exons</em> and <em>cds</em> have very similar profiles. Therefore we keep only <em>exons</em> in the rest of the analysis.</p>
<p>Positive coordinates (and zero) start in selected regions and negative coordinates start outside these regions. As we move further from 0, we can not guarantee that the statement is still true because some regions have a width narrower than the plot window. This is why the values tend to equilibrate away from the boundaries.</p>
<p>Aligned regions are enriched in exons and depleted in introns.</p>
<p>Close to the breakpoint of uncoalesced mapped regions, exon annotations are depleted, which makes sense, as exons are usually mapable. Within the mapped regions, frequency of introns and exons are comparable. In contrary, within coalesced mapped regions, the exon annotations are over-represented.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">computePlotData</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">target</span>, <span class="va">query</span>, <span class="va">CE</span><span class="op">=</span><span class="cn">NULL</span>, <span class="va">tfbs</span><span class="op">=</span><span class="cn">NULL</span>, <span class="va">operons</span> <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span> <span class="op">{</span>
  <span class="va">pair</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="va">target</span>, <span class="va">query</span>, sep <span class="op">=</span> <span class="st">"_"</span><span class="op">)</span>
  <span class="va">current_GB_aln_nonCoa</span>     <span class="op">&lt;-</span> <span class="va">gbs</span><span class="op">[[</span><span class="va">pair</span><span class="op">]</span><span class="op">]</span><span class="op">[</span>   <span class="va">gbs</span><span class="op">[[</span><span class="va">pair</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">nonCoa</span><span class="op">]</span>
  <span class="va">current_GB_aln_coa</span>        <span class="op">&lt;-</span> <span class="va">gbs</span><span class="op">[[</span><span class="va">pair</span><span class="op">]</span><span class="op">]</span><span class="op">[</span> <span class="op">!</span> <span class="va">gbs</span><span class="op">[[</span><span class="va">pair</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">nonCoa</span><span class="op">]</span>
  <span class="va">current_GB_syn_block</span>      <span class="op">&lt;-</span> <span class="va">coa</span><span class="op">[[</span><span class="va">pair</span><span class="op">]</span><span class="op">]</span><span class="op">[</span> <span class="op">!</span> <span class="va">coa</span><span class="op">[[</span><span class="va">pair</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">nonCoa</span><span class="op">]</span>
  <span class="co"># We do not use unal, because it overlaps with bri</span>
  <span class="va">current_GB_unmap</span>          <span class="op">&lt;-</span> <span class="va">unmap</span><span class="op">[[</span><span class="va">pair</span><span class="op">]</span><span class="op">]</span>
  <span class="va">current_GB_bri</span>        <span class="op">&lt;-</span> <span class="va">bri</span><span class="op">[[</span><span class="va">pair</span><span class="op">]</span><span class="op">]</span>
  
  <span class="va">targetGenome</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">genome</a></span><span class="op">(</span><span class="va">gbs</span><span class="op">[[</span><span class="va">pair</span><span class="op">]</span><span class="op">]</span><span class="op">)</span><span class="op">)</span>
  
  <span class="va">hmList</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/SimpleList-class.html" class="external-link">SimpleList</a></span><span class="op">(</span><span class="op">)</span>
  <span class="va">hmList</span><span class="op">$</span><span class="va">aln_nonCoa</span> <span class="op">&lt;-</span> <span class="fu">prepareFreqPlot</span><span class="op">(</span><span class="va">current_GB_aln_nonCoa</span>, <span class="va">annots</span><span class="op">[[</span><span class="va">target</span><span class="op">]</span><span class="op">]</span>, <span class="va">reps</span><span class="op">[[</span><span class="va">targetGenome</span><span class="op">]</span><span class="op">]</span>, <span class="va">CE</span>, <span class="va">tfbs</span>, <span class="va">operons</span><span class="op">[[</span><span class="va">target</span><span class="op">]</span><span class="op">]</span><span class="op">)</span>
  <span class="va">hmList</span><span class="op">$</span><span class="va">aln_coa</span>    <span class="op">&lt;-</span> <span class="fu">prepareFreqPlot</span><span class="op">(</span><span class="va">current_GB_aln_coa</span>,    <span class="va">annots</span><span class="op">[[</span><span class="va">target</span><span class="op">]</span><span class="op">]</span>, <span class="va">reps</span><span class="op">[[</span><span class="va">targetGenome</span><span class="op">]</span><span class="op">]</span>, <span class="va">CE</span>, <span class="va">tfbs</span>, <span class="va">operons</span><span class="op">[[</span><span class="va">target</span><span class="op">]</span><span class="op">]</span><span class="op">)</span>
  <span class="va">hmList</span><span class="op">$</span><span class="va">syn_block</span>  <span class="op">&lt;-</span> <span class="fu">prepareFreqPlot</span><span class="op">(</span><span class="va">current_GB_syn_block</span>,  <span class="va">annots</span><span class="op">[[</span><span class="va">target</span><span class="op">]</span><span class="op">]</span>, <span class="va">reps</span><span class="op">[[</span><span class="va">targetGenome</span><span class="op">]</span><span class="op">]</span>, <span class="va">CE</span>, <span class="va">tfbs</span>, <span class="va">operons</span><span class="op">[[</span><span class="va">target</span><span class="op">]</span><span class="op">]</span><span class="op">)</span>
  <span class="va">hmList</span><span class="op">$</span><span class="va">unmap</span>      <span class="op">&lt;-</span> <span class="fu">prepareFreqPlot</span><span class="op">(</span><span class="va">current_GB_unmap</span>,      <span class="va">annots</span><span class="op">[[</span><span class="va">target</span><span class="op">]</span><span class="op">]</span>, <span class="va">reps</span><span class="op">[[</span><span class="va">targetGenome</span><span class="op">]</span><span class="op">]</span>, <span class="va">CE</span>, <span class="va">tfbs</span>, <span class="va">operons</span><span class="op">[[</span><span class="va">target</span><span class="op">]</span><span class="op">]</span><span class="op">)</span>
  <span class="va">hmList</span><span class="op">$</span><span class="va">bri</span>        <span class="op">&lt;-</span> <span class="fu">prepareFreqPlot</span><span class="op">(</span><span class="va">current_GB_bri</span>,        <span class="va">annots</span><span class="op">[[</span><span class="va">target</span><span class="op">]</span><span class="op">]</span>, <span class="va">reps</span><span class="op">[[</span><span class="va">targetGenome</span><span class="op">]</span><span class="op">]</span>, <span class="va">CE</span>, <span class="va">tfbs</span>, <span class="va">operons</span><span class="op">[[</span><span class="va">target</span><span class="op">]</span><span class="op">]</span><span class="op">)</span>
  <span class="va">hmList</span>
<span class="op">}</span>

<span class="va">hmList_to_plot</span> <span class="op">&lt;-</span> <span class="fu">computePlotData</span><span class="op">(</span><span class="st">"Oki"</span>, <span class="st">"Osa"</span>, <span class="va">ces</span><span class="op">$</span><span class="va">ce__50__48</span><span class="op">$</span><span class="va">Oki_Osa</span>, <span class="va">tfbs</span>, <span class="va">operons</span><span class="op">)</span>

<span class="op">(</span><span class="va">p_boundary</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/cbind.html" class="external-link">rbind</a></span><span class="op">(</span>
  <span class="va">hmList_to_plot</span><span class="op">$</span><span class="va">aln_nonCoa</span>  |&gt; <span class="fu">prepareGGFreqPlot</span><span class="op">(</span><span class="st">"01_isolated_alns"</span><span class="op">)</span>,
  <span class="va">hmList_to_plot</span><span class="op">$</span><span class="va">unmap</span>       |&gt; <span class="fu">prepareGGFreqPlot</span><span class="op">(</span><span class="st">"02_unmapped"</span><span class="op">)</span>,
  <span class="va">hmList_to_plot</span><span class="op">$</span><span class="va">syn_block</span>   |&gt; <span class="fu">prepareGGFreqPlot</span><span class="op">(</span><span class="st">"03_colinear_reg"</span><span class="op">)</span>,
  <span class="va">hmList_to_plot</span><span class="op">$</span><span class="va">bri</span>         |&gt; <span class="fu">prepareGGFreqPlot</span><span class="op">(</span><span class="st">"04_bridge"</span><span class="op">)</span><span class="co">#,</span>
<span class="co">#  hmList_to_plot$aln_coa     |&gt; prepareGGFreqPlot("05_colinear_alns")</span>

<span class="op">)</span>|&gt; <span class="fu">ggplot</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">aes</span><span class="op">(</span><span class="va">pos</span>, <span class="va">freq</span>, col <span class="op">=</span> <span class="va">what</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">geom_line</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">facet_wrap</span><span class="op">(</span><span class="op">~</span><span class="va">desc</span>, nrow <span class="op">=</span> <span class="fl">1</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">theme_bw</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">ggtitle</span><span class="op">(</span><span class="st">"Aligned on regions left boundary. CNE is 50/48.  Okinawa - Osaka"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="GenomicFeatures_files/figure-html/genomic_features_byType-1.svg" width="1440"></p>
<p>Most of the narrow-width aligned regions did not coalesce. We use this as a classifier instead of an arbitrary width cutoff. See <code><a href="../articles/RegionWidths.html">vignette("RegionWidths", package = "OikScrambling")</a></code> for details.</p>
<div class="section level5">
<h5 id="linked-regions">Linked regions<a class="anchor" aria-label="anchor" href="#linked-regions"></a>
</h5>
<p>Same approach, but enforcing a succession of just-aligned, unaligned and syntenic regions.</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">flag_aln_unal_syn</span> <span class="op">&lt;-</span> <span class="kw">function</span> <span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="op">{</span>
  <span class="va">p0</span> <span class="op">&lt;-</span> <span class="va">x</span><span class="op">$</span><span class="va">type</span>
  <span class="va">p1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">tail</a></span><span class="op">(</span><span class="va">p0</span>, <span class="op">-</span><span class="fl">1</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="st">'aligned'</span><span class="op">)</span><span class="op">)</span>
  <span class="va">p2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">tail</a></span><span class="op">(</span><span class="va">p0</span>, <span class="op">-</span><span class="fl">2</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="st">'aligned'</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="st">'aligned'</span><span class="op">)</span><span class="op">)</span>
  <span class="va">p0</span> <span class="op">==</span> <span class="st">'aligned'</span> <span class="op">&amp;</span> <span class="va">p1</span> <span class="op">==</span> <span class="st">'unaligned'</span> <span class="op">&amp;</span> <span class="va">p2</span> <span class="op">==</span> <span class="st">'syntenic'</span>
<span class="op">}</span>

<span class="va">current_GB</span> <span class="op">&lt;-</span> <span class="va">wgo</span><span class="op">$</span><span class="va">Oki_Osa</span>

<span class="va">current_GB</span><span class="op">$</span><span class="va">flag_a_u_s</span> <span class="op">&lt;-</span> <span class="fu">flag_aln_unal_syn</span><span class="op">(</span><span class="va">current_GB</span><span class="op">)</span>

<span class="va">current_GB_aln</span>     <span class="op">&lt;-</span> <span class="va">current_GB</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/which.html" class="external-link">which</a></span><span class="op">(</span><span class="va">current_GB</span><span class="op">$</span><span class="va">flag_a_u_s</span><span class="op">)</span> <span class="op">+</span> <span class="fl">0</span><span class="op">]</span>
<span class="va">current_GB_unal</span>    <span class="op">&lt;-</span> <span class="va">current_GB</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/which.html" class="external-link">which</a></span><span class="op">(</span><span class="va">current_GB</span><span class="op">$</span><span class="va">flag_a_u_s</span><span class="op">)</span> <span class="op">+</span> <span class="fl">1</span><span class="op">]</span>
<span class="va">current_GB_syn</span>     <span class="op">&lt;-</span> <span class="va">current_GB</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/which.html" class="external-link">which</a></span><span class="op">(</span><span class="va">current_GB</span><span class="op">$</span><span class="va">flag_a_u_s</span><span class="op">)</span> <span class="op">+</span> <span class="fl">2</span><span class="op">]</span>

<span class="va">current_GB_aln</span>   |&gt; <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/start.html" class="external-link">width</a></span><span class="op">(</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">##    Min. 1st Qu.  Median    Mean 3rd Qu.    Max. </span>
<span class="co">##      69     218     440    1029    1183   24353</span></code></pre>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">current_GB_unal</span>  |&gt; <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/start.html" class="external-link">width</a></span><span class="op">(</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">##    Min. 1st Qu.  Median    Mean 3rd Qu.    Max. </span>
<span class="co">##       3     167     508    1980    1659   95225</span></code></pre>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">current_GB_syn</span>   |&gt; <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/start.html" class="external-link">width</a></span><span class="op">(</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">##    Min. 1st Qu.  Median    Mean 3rd Qu.    Max. </span>
<span class="co">##     224    2004    4410    9189   10524  120337</span></code></pre>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">hmListAln</span>     <span class="op">&lt;-</span> <span class="fu">prepareFreqPlot</span><span class="op">(</span><span class="va">current_GB_aln</span>,     <span class="va">annots</span><span class="op">$</span><span class="va">Oki</span>, <span class="va">reps</span><span class="op">$</span><span class="va">OKI2018.I69</span><span class="op">)</span>
<span class="va">hmListUnal</span>    <span class="op">&lt;-</span> <span class="fu">prepareFreqPlot</span><span class="op">(</span><span class="va">current_GB_unal</span>,    <span class="va">annots</span><span class="op">$</span><span class="va">Oki</span>, <span class="va">reps</span><span class="op">$</span><span class="va">OKI2018.I69</span><span class="op">)</span>
<span class="va">hmListSyn</span>     <span class="op">&lt;-</span> <span class="fu">prepareFreqPlot</span><span class="op">(</span><span class="va">current_GB_syn</span>,     <span class="va">annots</span><span class="op">$</span><span class="va">Oki</span>, <span class="va">reps</span><span class="op">$</span><span class="va">OKI2018.I69</span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/cbind.html" class="external-link">rbind</a></span><span class="op">(</span>
  <span class="va">hmListAln</span>  |&gt; <span class="fu">prepareGGFreqPlot</span><span class="op">(</span><span class="st">"01_narrow"</span><span class="op">)</span>,
  <span class="va">hmListUnal</span> |&gt; <span class="fu">prepareGGFreqPlot</span><span class="op">(</span><span class="st">"02_unal"</span><span class="op">)</span>,
  <span class="va">hmListSyn</span>  |&gt; <span class="fu">prepareGGFreqPlot</span><span class="op">(</span><span class="st">"03_syn"</span><span class="op">)</span>

<span class="op">)</span>|&gt; <span class="fu">ggplot</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">aes</span><span class="op">(</span><span class="va">pos</span>, <span class="va">freq</span>, col <span class="op">=</span> <span class="va">what</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">geom_line</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">theme_bw</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">facet_wrap</span><span class="op">(</span><span class="op">~</span><span class="va">desc</span>, nrow <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></code></pre></div>
<p><img src="GenomicFeatures_files/figure-html/genomic_features_byType2-1.png" width="700"></p>
<p>This mildens the differences observed in the earlier section, probably because it requires the aligned region to be nearby a syntenic region.</p>
</div>
<div class="section level5">
<h5 id="aligning-on-center-instead-of-border-">Aligning on center instead of border.<a class="anchor" aria-label="anchor" href="#aligning-on-center-instead-of-border-"></a>
</h5>
<p>As the plots are hardcoded to use the left border, let’s artificially make regions of 1 bp on centers, rather than changing the accessory function.</p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># current_GB &lt;- coa$Oki_Osa</span>
<span class="co"># </span>
<span class="co"># current_GB_wide_c    &lt;- current_GB_wide    |&gt; plyranges::anchor_center() |&gt; plyranges::mutate(width=1)</span>
<span class="co"># current_GB_narrow_c  &lt;- current_GB_narrow  |&gt; plyranges::anchor_center() |&gt; plyranges::mutate(width=1)</span>
<span class="co"># current_GB_unmap_c   &lt;- current_GB_unmap   |&gt; plyranges::anchor_center() |&gt; plyranges::mutate(width=1)</span>
<span class="co"># current_GB_unal_c    &lt;- current_GB_unal    |&gt; plyranges::anchor_center() |&gt; plyranges::mutate(width=1)</span>
<span class="co"># current_GB_bri_c &lt;- current_GB_bri |&gt; plyranges::anchor_center() |&gt; plyranges::mutate(width=1)</span>
<span class="co"># </span>
<span class="co"># currentCNE &lt;- ces$ce__50__47$Oki_Osa</span>
<span class="co"># </span>
<span class="co"># hmListWide_c    &lt;- prepareFreqPlot(current_GB_wide_c,    annots$Oki, reps$Oki, currentCNE)</span>
<span class="co"># hmListNarrow_c  &lt;- prepareFreqPlot(current_GB_narrow_c,  annots$Oki, reps$Oki, currentCNE)</span>
<span class="co"># hmListUnmap_c   &lt;- prepareFreqPlot(current_GB_unmap_c,   annots$Oki, reps$Oki, currentCNE)</span>
<span class="co"># hmListUnal_c    &lt;- prepareFreqPlot(current_GB_unal_c,    annots$Oki, reps$Oki, currentCNE)</span>
<span class="co"># hmListUnalMap_c &lt;- prepareFreqPlot(current_GB_bri_c, annots$Oki, reps$Oki, currentCNE)</span>
<span class="co"># </span>
<span class="co"># p_center &lt;- rbind(</span>
<span class="co">#   hmListNarrow_c  |&gt; prepareGGFreqPlot("01_narrow"),</span>
<span class="co">#   hmListUnmap_c   |&gt; prepareGGFreqPlot("02_unmap"),</span>
<span class="co"># #  hmListUnal_c    |&gt; prepareGGFreqPlot("02b_unmap"),</span>
<span class="co">#   hmListWide_c    |&gt; prepareGGFreqPlot("03_wide"),</span>
<span class="co">#   hmListUnalMap_c |&gt; prepareGGFreqPlot("04_bri")</span>
<span class="co"># )|&gt; ggplot() +</span>
<span class="co">#   aes(pos, freq, col = what) +</span>
<span class="co">#   geom_line() +</span>
<span class="co">#   facet_wrap(~desc, nrow = 1) +</span>
<span class="co">#   ggtitle("Aligned on region midpoints.")</span>
<span class="co"># </span>
<span class="co"># p_center$data$what  &lt;- p_center$data$what |&gt; sub(pat = "CNE", rep = "CE_50_47")</span>
<span class="co"># p_center</span></code></pre></div>
</div>
<div class="section level5">
<h5 id="comparison">Comparison<a class="anchor" aria-label="anchor" href="#comparison"></a>
</h5>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># library(patchwork)</span>
<span class="co"># p_boundary / p_center</span></code></pre></div>
</div>
<div class="section level4">
<h4 id="pwm">PWM<a class="anchor" aria-label="anchor" href="#pwm"></a>
</h4>
<p>The PWM motif, which is rare in comparison to the other features, follows a profile that bears resemblance to the repeat profile.</p>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">gadems.invLeftGap</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="st">"gadems.invLeftGap.Rda"</span><span class="op">)</span>
<span class="va">pwm</span> <span class="op">&lt;-</span> <span class="va">gadems.invLeftGap</span><span class="op">$</span><span class="va">Oki_Osa</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span>
<span class="va">pwm</span><span class="op">@</span><span class="va">consensus</span></code></pre></div>
<pre><code><span class="co">## [1] "rArAAGCCGCdwAGCsGCw"</span></code></pre>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># pwmHits &lt;- matchPWM(pwm = sapply(pwm@alignList, \(x) x@seq) |&gt; as("DNAStringSet") |&gt; PWM(), genomes$Oki) # Gives different scores...</span>
<span class="va">pwmHits</span> <span class="op">&lt;-</span> <span class="fu">matchPWM</span><span class="op">(</span>pwm <span class="op">=</span> <span class="fu">rGADEM</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/rGADEM/man/gadem-class.html" class="external-link">getPWM</a></span><span class="op">(</span><span class="va">pwm</span><span class="op">)</span>, <span class="va">genomes</span><span class="op">$</span><span class="va">OKI2018.I69</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/r/base/warning.html" class="external-link">suppressWarnings</a></span><span class="op">(</span><span class="op">)</span>
<span class="co"># Suppressing warnings like:</span>
<span class="co"># Warning messages:</span>
<span class="co"># 1: In .Call2("XString_match_PWM", pwm, subject, min.score, count.only,  :</span>
<span class="co">#   'subject' contains letters not in [ACGT] ==&gt; assigned weight 0 to them</span>
<span class="co"># </span>
<span class="co"># plotHeatmapMeta(list(feature_coverage(current_GB_wide,   pwmHits, win = 1e3, lab = "wide",    direction = "left")))</span>
<span class="co"># plotHeatmapMeta(list(feature_coverage(current_GB_narrow, pwmHits, win = 1e3, lab = "narrow",  direction = "left")))</span></code></pre></div>
</div>
<div class="section level4">
<h4 id="question">Question<a class="anchor" aria-label="anchor" href="#question"></a>
</h4>
<p>in <code><a href="../articles/ColinearityInterruptors.html">vignette("ColinearityInterruptors", package = "OikScrambling")</a></code>, we saw that some uncoalesced regions map uniquely and are well conserved across genomes. What are they… enhancers ?</p>
</div>
</div>
<div class="section level3">
<h3 id="gene-feature-coverage-around-pwm-hits">Gene feature coverage around PWM hits<a class="anchor" aria-label="anchor" href="#gene-feature-coverage-around-pwm-hits"></a>
</h3>
<p>Let’s look at coverage around PWM hits of the AAGCsGCwwmkCGrCTTyn motif</p>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">current_GB</span> <span class="op">&lt;-</span> <span class="va">pwmHits</span>

<span class="va">hmList_PWM_OKI</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="op">)</span>
<span class="va">hmList_PWM_OKI</span><span class="op">$</span><span class="va">genes</span>     <span class="op">&lt;-</span> <span class="fu">feature_coverage</span><span class="op">(</span><span class="va">current_GB</span>, <span class="fu"><a href="https://rdrr.io/pkg/GenomicFeatures/man/transcripts.html" class="external-link">genes</a></span><span class="op">(</span><span class="va">annots</span><span class="op">$</span><span class="va">Oki</span><span class="op">)</span>,                         win <span class="op">=</span> <span class="fl">2000</span>, lab <span class="op">=</span> <span class="st">"Oki genes"</span>, direction <span class="op">=</span> <span class="st">"left"</span><span class="op">)</span>
<span class="va">hmList_PWM_OKI</span><span class="op">$</span><span class="va">promoters</span> <span class="op">&lt;-</span> <span class="fu">feature_coverage</span><span class="op">(</span><span class="va">current_GB</span>, <span class="fu"><a href="https://rdrr.io/pkg/GenomicFeatures/man/transcripts.html" class="external-link">promoters</a></span><span class="op">(</span><span class="va">annots</span><span class="op">$</span><span class="va">Oki</span><span class="op">)</span>,                     win <span class="op">=</span> <span class="fl">2000</span>, lab <span class="op">=</span> <span class="st">"Oki promoters"</span>, direction <span class="op">=</span> <span class="st">"left"</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Warning in valid.GenomicRanges.seqinfo(x, suggest.trim = TRUE): GRanges object contains 4 out-of-bound ranges located on sequences</span>
<span class="co">##   chr1, chrUn_1, and chrUn_10. Note that ranges located on a sequence</span>
<span class="co">##   whose length is unknown (NA) or on a circular sequence are not</span>
<span class="co">##   considered out-of-bound (use seqlengths() and isCircular() to get the</span>
<span class="co">##   lengths and circularity flags of the underlying sequences). You can use</span>
<span class="co">##   trim() to trim these ranges. See ?`trim,GenomicRanges-method` for more</span>
<span class="co">##   information.</span></code></pre>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">hmList_PWM_OKI</span><span class="op">$</span><span class="va">exons</span>     <span class="op">&lt;-</span> <span class="fu">feature_coverage</span><span class="op">(</span><span class="va">current_GB</span>, <span class="fu"><a href="https://rdrr.io/pkg/GenomicFeatures/man/exonicParts.html" class="external-link">exonicParts</a></span><span class="op">(</span><span class="va">annots</span><span class="op">$</span><span class="va">Oki</span><span class="op">)</span>,                   win <span class="op">=</span> <span class="fl">2000</span>, lab <span class="op">=</span> <span class="st">"Oki exons"</span>, direction <span class="op">=</span> <span class="st">"left"</span><span class="op">)</span>
<span class="va">hmList_PWM_OKI</span><span class="op">$</span><span class="va">introns</span>   <span class="op">&lt;-</span> <span class="fu">feature_coverage</span><span class="op">(</span><span class="va">current_GB</span>, <span class="fu"><a href="https://rdrr.io/pkg/GenomicFeatures/man/exonicParts.html" class="external-link">intronicParts</a></span><span class="op">(</span><span class="va">annots</span><span class="op">$</span><span class="va">Oki</span><span class="op">)</span>,                 win <span class="op">=</span> <span class="fl">2000</span>, lab <span class="op">=</span> <span class="st">"Oki introns"</span>, direction <span class="op">=</span> <span class="st">"left"</span><span class="op">)</span>
<span class="va">hmList_PWM_OKI</span><span class="op">$</span><span class="va">cds</span>       <span class="op">&lt;-</span> <span class="fu">feature_coverage</span><span class="op">(</span><span class="va">current_GB</span>, <span class="fu"><a href="https://rdrr.io/pkg/GenomicFeatures/man/transcripts.html" class="external-link">cds</a></span><span class="op">(</span><span class="va">annots</span><span class="op">$</span><span class="va">Oki</span><span class="op">)</span>,                           win <span class="op">=</span> <span class="fl">2000</span>, lab <span class="op">=</span> <span class="st">"Oki cds"</span>, direction <span class="op">=</span> <span class="st">"left"</span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/pkg/heatmaps/man/plotHeatmapMeta.html" class="external-link">plotHeatmapMeta</a></span><span class="op">(</span><span class="va">hmList_PWM_OKI</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"genes"</span>, <span class="st">"exons"</span>, <span class="st">"cds"</span><span class="op">)</span><span class="op">]</span><span class="op">)</span> <span class="co"># Show they look alike.  We keep "exons"</span></code></pre></div>
<p><img src="GenomicFeatures_files/figure-html/genomic_features_pwmHits-1.png" width="700"></p>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/heatmaps/man/plotHeatmapMeta.html" class="external-link">plotHeatmapMeta</a></span><span class="op">(</span><span class="va">hmList_PWM_OKI</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"promoters"</span>, <span class="st">"exons"</span>, <span class="st">"introns"</span><span class="op">)</span><span class="op">]</span><span class="op">)</span></code></pre></div>
<p><img src="GenomicFeatures_files/figure-html/genomic_features_pwmHits-2.png" width="700"></p>
<p>They are enriched in introns…</p>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># </span>
<span class="co"># # Sandbox to make sanity checks.</span>
<span class="co"># </span>
<span class="co"># a &lt;- coa$Oki_Osa</span>
<span class="co"># b &lt;- unlist(fiveUTRsByTranscript(annots$Oki))</span>
<span class="co"># </span>
<span class="co"># hm_a &lt;- feature_coverage(a, b, win = 2000, lab = "All ranges", direction = "left")</span>
<span class="co"># hm_b &lt;- feature_coverage(subsetByOverlaps(a,  b), b, win = 2000, lab = "Subsetted ranges", direction = "left")</span>
<span class="co"># </span>
<span class="co"># plotHeatmapMeta(list(hm_a))</span>
<span class="co"># plotHeatmapMeta(list(hm_b))</span></code></pre></div>
</div>
<div class="section level3">
<h3 id="breakpoint-coverage-around-pwm-hits">Breakpoint coverage around PWM hits<a class="anchor" aria-label="anchor" href="#breakpoint-coverage-around-pwm-hits"></a>
</h3>
<p>TODO: triplecheck <code>feature_coverage</code>, it is old…</p>
<div class="sourceCode" id="cb37"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">current_GB</span> <span class="op">&lt;-</span> <span class="va">pwmHits</span>

<span class="va">hmList_PWM_OKI</span><span class="op">$</span><span class="va">gbOsa</span>     <span class="op">&lt;-</span> <span class="fu">feature_coverage</span><span class="op">(</span><span class="va">current_GB</span>, <span class="va">gbs</span><span class="op">$</span><span class="va">Oki_Osa</span>,  win <span class="op">=</span> <span class="fl">5000</span>, lab <span class="op">=</span> <span class="st">"Oki-Osa"</span>, direction <span class="op">=</span> <span class="st">"left"</span><span class="op">)</span>
<span class="va">hmList_PWM_OKI</span><span class="op">$</span><span class="va">gbBar</span>     <span class="op">&lt;-</span> <span class="fu">feature_coverage</span><span class="op">(</span><span class="va">current_GB</span>, <span class="va">gbs</span><span class="op">$</span><span class="va">Oki_Bar</span>,  win <span class="op">=</span> <span class="fl">5000</span>, lab <span class="op">=</span> <span class="st">"Oki-Bar"</span>, direction <span class="op">=</span> <span class="st">"left"</span><span class="op">)</span>
<span class="va">hmList_PWM_OKI</span><span class="op">$</span><span class="va">gbKum</span>     <span class="op">&lt;-</span> <span class="fu">feature_coverage</span><span class="op">(</span><span class="va">current_GB</span>, <span class="va">gbs</span><span class="op">$</span><span class="va">Oki_Kum</span>,  win <span class="op">=</span> <span class="fl">5000</span>, lab <span class="op">=</span> <span class="st">"Oki-Kum"</span>, direction <span class="op">=</span> <span class="st">"left"</span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/pkg/heatmaps/man/plotHeatmapMeta.html" class="external-link">plotHeatmapMeta</a></span><span class="op">(</span><span class="va">hmList_PWM_OKI</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"gbOsa"</span>, <span class="st">"gbBar"</span>, <span class="st">"gbKum"</span><span class="op">)</span><span class="op">]</span><span class="op">)</span></code></pre></div>
<p><img src="GenomicFeatures_files/figure-html/genomic_features_pwmHits2-1.png" width="700"></p>
<div class="sourceCode" id="cb38"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">hmList_PWM_OKI</span><span class="op">$</span><span class="va">coOsa</span>     <span class="op">&lt;-</span> <span class="fu">feature_coverage</span><span class="op">(</span><span class="va">current_GB</span>, <span class="va">coa</span><span class="op">$</span><span class="va">Oki_Osa</span>,  win <span class="op">=</span> <span class="fl">5000</span>, lab <span class="op">=</span> <span class="st">"Oki-Osa"</span>, direction <span class="op">=</span> <span class="st">"left"</span><span class="op">)</span>
<span class="va">hmList_PWM_OKI</span><span class="op">$</span><span class="va">coBar</span>     <span class="op">&lt;-</span> <span class="fu">feature_coverage</span><span class="op">(</span><span class="va">current_GB</span>, <span class="va">coa</span><span class="op">$</span><span class="va">Oki_Bar</span>,  win <span class="op">=</span> <span class="fl">5000</span>, lab <span class="op">=</span> <span class="st">"Oki-Bar"</span>, direction <span class="op">=</span> <span class="st">"left"</span><span class="op">)</span>
<span class="va">hmList_PWM_OKI</span><span class="op">$</span><span class="va">coKum</span>     <span class="op">&lt;-</span> <span class="fu">feature_coverage</span><span class="op">(</span><span class="va">current_GB</span>, <span class="va">coa</span><span class="op">$</span><span class="va">Oki_Kum</span>,  win <span class="op">=</span> <span class="fl">5000</span>, lab <span class="op">=</span> <span class="st">"Oki-Kum"</span>, direction <span class="op">=</span> <span class="st">"left"</span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/pkg/heatmaps/man/plotHeatmapMeta.html" class="external-link">plotHeatmapMeta</a></span><span class="op">(</span><span class="va">hmList_PWM_OKI</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"coOsa"</span>, <span class="st">"coBar"</span>, <span class="st">"coKum"</span><span class="op">)</span><span class="op">]</span><span class="op">)</span></code></pre></div>
<p><img src="GenomicFeatures_files/figure-html/genomic_features_pwmHits2-2.png" width="700"></p>
</div>
<div class="section level3">
<h3 id="okinawa-genome-coverage">Okinawa genome coverage<a class="anchor" aria-label="anchor" href="#okinawa-genome-coverage"></a>
</h3>
<p>As previously discussed, low coverage over an alignment stop could lower the likelihood of it being considered as a breakpoint. We have per-base coverage depth information for the Okinawan genome (<code>Oki_cov_pb</code>). However, the coverage of this particular assembly is quite good. In fact, we can investigate the coverge over alignment stops from the information obtained using <code>master_bp_analysis</code>.</p>
<div class="sourceCode" id="cb39"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># fin_Oki &lt;- fin_gr_O_Oki[[2]]</span>
<span class="co"># min_cov &lt;- min(min(fin_Oki$left_cov_pb), min(fin_Oki$right_cov_pb)) # minimum coverage over an alignment stop</span>
<span class="co"># min_cov</span>
<span class="co"># length(fin_Oki[fin_Oki$left_cov_pb == min_cov]) + length(fin_Oki[fin_Oki$right_cov_pb == min_cov]) # how many of the minimum coverage is observed</span>
<span class="co"># length(fin_Oki[fin_Oki$left_cov_pb &lt;= 50]) + length(fin_Oki[fin_Oki$right_cov_pb &lt;= 50]) # how many alignment stops have a coverage of less than or euqal to 50</span></code></pre></div>
<p>We may choose to kick out the one alignment stop for which there is no coverage. However, only 15 out of 34572 alignment stops have a coverage of less than or equal to 50, meaning that it would be hard to exclude more than just a few alignment stops using coverage information for breakpoint analysis.</p>
</div>
</div>
<div class="section level2">
<h2 id="coverage">Coverage<a class="anchor" aria-label="anchor" href="#coverage"></a>
</h2>
<p><em>Needs coverage information to be recomputed</em></p>
<p>Another characteristic of alignment stops that we can examine is the coverage depth over said alignment stops. By aligning raw reads to assembled genomes, we can obtain per-base coverage depth for target or query genome. Low coverage directly on and surrounding an alignment stop may suggest unreliability in it being a true breakpoint. The function <code>bp_coverage</code> returns a GRanges object of the bps, with associated averaged and point converage in the metadatacolumns.</p>
<div class="sourceCode" id="cb40"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># cov_gr_O_Oki &lt;- bp_coverage(gr_ob = q_Oki, cov_gr = Oki_cov_pb, win = 50)</span>
<span class="co"># cov_gr_O_Oki</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="session-information">Session information<a class="anchor" aria-label="anchor" href="#session-information"></a>
</h2>
<div class="sourceCode" id="cb41"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## R version 4.1.2 (2021-11-01)</span>
<span class="co">## Platform: x86_64-pc-linux-gnu (64-bit)</span>
<span class="co">## Running under: Debian GNU/Linux bookworm/sid</span>
<span class="co">## </span>
<span class="co">## Matrix products: default</span>
<span class="co">## BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.10.0</span>
<span class="co">## LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.10.0</span>
<span class="co">## </span>
<span class="co">## locale:</span>
<span class="co">##  [1] LC_CTYPE=en_GB.UTF-8       LC_NUMERIC=C              </span>
<span class="co">##  [3] LC_TIME=en_GB.UTF-8        LC_COLLATE=en_GB.UTF-8    </span>
<span class="co">##  [5] LC_MONETARY=en_GB.UTF-8    LC_MESSAGES=en_GB.UTF-8   </span>
<span class="co">##  [7] LC_PAPER=en_GB.UTF-8       LC_NAME=C                 </span>
<span class="co">##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span>
<span class="co">## [11] LC_MEASUREMENT=en_GB.UTF-8 LC_IDENTIFICATION=C       </span>
<span class="co">## </span>
<span class="co">## attached base packages:</span>
<span class="co">## [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span>
<span class="co">## [8] base     </span>
<span class="co">## </span>
<span class="co">## other attached packages:</span>
<span class="co">##  [1] BSgenome.Oidioi.genoscope.OdB3_1.0.0   </span>
<span class="co">##  [2] BSgenome.Oidioi.OIST.AOM.5.5f_1.0.1    </span>
<span class="co">##  [3] BSgenome.Oidioi.OIST.KUM.M3.7f_1.0.1   </span>
<span class="co">##  [4] BSgenome.Oidioi.OIST.Bar2.p4_1.0.1     </span>
<span class="co">##  [5] BSgenome.Oidioi.OIST.OSKA2016v1.9_1.0.0</span>
<span class="co">##  [6] BSgenome.Oidioi.OIST.OKI2018.I69_1.0.1 </span>
<span class="co">##  [7] heatmaps_1.18.0                        </span>
<span class="co">##  [8] GenomicFeatures_1.46.5                 </span>
<span class="co">##  [9] AnnotationDbi_1.56.2                   </span>
<span class="co">## [10] Biobase_2.54.0                         </span>
<span class="co">## [11] OikScrambling_3.0.0                    </span>
<span class="co">## [12] ggplot2_3.3.5                          </span>
<span class="co">## [13] GenomicBreaks_0.13.1                   </span>
<span class="co">## [14] BSgenome_1.62.0                        </span>
<span class="co">## [15] rtracklayer_1.54.0                     </span>
<span class="co">## [16] Biostrings_2.62.0                      </span>
<span class="co">## [17] XVector_0.34.0                         </span>
<span class="co">## [18] GenomicRanges_1.46.1                   </span>
<span class="co">## [19] GenomeInfoDb_1.30.1                    </span>
<span class="co">## [20] IRanges_2.28.0                         </span>
<span class="co">## [21] S4Vectors_0.32.3                       </span>
<span class="co">## [22] BiocGenerics_0.40.0                    </span>
<span class="co">## </span>
<span class="co">## loaded via a namespace (and not attached):</span>
<span class="co">##   [1] utf8_1.2.2                  R.utils_2.11.0             </span>
<span class="co">##   [3] tidyselect_1.1.2            lme4_1.1-28                </span>
<span class="co">##   [5] poweRlaw_0.70.6             RSQLite_2.2.10             </span>
<span class="co">##   [7] htmlwidgets_1.5.4           grid_4.1.2                 </span>
<span class="co">##   [9] BiocParallel_1.28.3         munsell_0.5.0              </span>
<span class="co">##  [11] codetools_0.2-18            ragg_1.2.1                 </span>
<span class="co">##  [13] withr_2.5.0                 colorspace_2.0-3           </span>
<span class="co">##  [15] filelock_1.0.2              highr_0.9                  </span>
<span class="co">##  [17] knitr_1.37                  rstudioapi_0.13            </span>
<span class="co">##  [19] genoPlotR_0.8.11            MatrixGenerics_1.6.0       </span>
<span class="co">##  [21] labeling_0.4.2              GenomeInfoDbData_1.2.7     </span>
<span class="co">##  [23] bit64_4.0.5                 farver_2.1.0               </span>
<span class="co">##  [25] rprojroot_2.0.2             vctrs_0.3.8                </span>
<span class="co">##  [27] generics_0.1.2              xfun_0.30                  </span>
<span class="co">##  [29] BiocFileCache_2.2.1         ggseqlogo_0.1              </span>
<span class="co">##  [31] R6_2.5.1                    locfit_1.5-9.5             </span>
<span class="co">##  [33] bitops_1.0-7                cachem_1.0.6               </span>
<span class="co">##  [35] DelayedArray_0.20.0         assertthat_0.2.1           </span>
<span class="co">##  [37] BiocIO_1.4.0                scales_1.1.1               </span>
<span class="co">##  [39] nnet_7.3-16                 gtable_0.3.0               </span>
<span class="co">##  [41] weights_1.0.4               seqLogo_1.60.0             </span>
<span class="co">##  [43] rlang_1.0.2                 systemfonts_1.0.3          </span>
<span class="co">##  [45] splines_4.1.2               broom_0.7.12               </span>
<span class="co">##  [47] checkmate_2.0.0             yaml_2.3.5                 </span>
<span class="co">##  [49] reshape2_1.4.4              abind_1.4-5                </span>
<span class="co">##  [51] backports_1.4.1             Hmisc_4.6-0                </span>
<span class="co">##  [53] tools_4.1.2                 ellipsis_0.3.2             </span>
<span class="co">##  [55] jquerylib_0.1.4             RColorBrewer_1.1-2         </span>
<span class="co">##  [57] Rcpp_1.0.8.3                plyr_1.8.6                 </span>
<span class="co">##  [59] base64enc_0.1-3             progress_1.2.2             </span>
<span class="co">##  [61] zlibbioc_1.40.0             purrr_0.3.4                </span>
<span class="co">##  [63] RCurl_1.98-1.6              prettyunits_1.1.1          </span>
<span class="co">##  [65] rpart_4.1-15                SummarizedExperiment_1.24.0</span>
<span class="co">##  [67] cluster_2.1.2               fs_1.5.2                   </span>
<span class="co">##  [69] magrittr_2.0.2              data.table_1.14.2          </span>
<span class="co">##  [71] matrixStats_0.61.0          hms_1.1.1                  </span>
<span class="co">##  [73] fftwtools_0.9-11            evaluate_0.15              </span>
<span class="co">##  [75] xtable_1.8-4                XML_3.99-0.9               </span>
<span class="co">##  [77] jpeg_0.1-9                  gridExtra_2.3              </span>
<span class="co">##  [79] compiler_4.1.2              biomaRt_2.50.3             </span>
<span class="co">##  [81] tibble_3.1.6                mice_3.14.0                </span>
<span class="co">##  [83] KernSmooth_2.23-20          crayon_1.5.0               </span>
<span class="co">##  [85] minqa_1.2.4                 R.oo_1.24.0                </span>
<span class="co">##  [87] htmltools_0.5.2             tzdb_0.2.0                 </span>
<span class="co">##  [89] tiff_0.1-11                 Formula_1.2-4              </span>
<span class="co">##  [91] tidyr_1.2.0                 DBI_1.1.2                  </span>
<span class="co">##  [93] dbplyr_2.1.1                MASS_7.3-54                </span>
<span class="co">##  [95] rappdirs_0.3.3              boot_1.3-28                </span>
<span class="co">##  [97] rGADEM_2.42.0               Matrix_1.4-0               </span>
<span class="co">##  [99] ade4_1.7-18                 readr_2.1.2                </span>
<span class="co">## [101] cli_3.2.0                   R.methodsS3_1.8.1          </span>
<span class="co">## [103] gdata_2.18.0                parallel_4.1.2             </span>
<span class="co">## [105] pkgconfig_2.0.3             pkgdown_2.0.2              </span>
<span class="co">## [107] GenomicAlignments_1.30.0    foreign_0.8-81             </span>
<span class="co">## [109] xml2_1.3.3                  annotate_1.72.0            </span>
<span class="co">## [111] bslib_0.3.1                 stringr_1.4.0              </span>
<span class="co">## [113] digest_0.6.29               pracma_2.3.8               </span>
<span class="co">## [115] CNEr_1.30.0                 rmarkdown_2.11             </span>
<span class="co">## [117] htmlTable_2.4.0             restfulr_0.0.13            </span>
<span class="co">## [119] curl_4.3.2                  EBImage_4.36.0             </span>
<span class="co">## [121] Rsamtools_2.10.0            gtools_3.9.2               </span>
<span class="co">## [123] rjson_0.2.21                nloptr_1.2.2.3             </span>
<span class="co">## [125] lifecycle_1.0.1             nlme_3.1-153               </span>
<span class="co">## [127] jsonlite_1.8.0              desc_1.4.1                 </span>
<span class="co">## [129] fansi_1.0.2                 pillar_1.7.0               </span>
<span class="co">## [131] lattice_0.20-45             KEGGREST_1.34.0            </span>
<span class="co">## [133] fastmap_1.1.0               httr_1.4.2                 </span>
<span class="co">## [135] plotrix_3.8-2               survival_3.2-13            </span>
<span class="co">## [137] GO.db_3.14.0                glue_1.6.2                 </span>
<span class="co">## [139] png_0.1-7                   bit_4.0.4                  </span>
<span class="co">## [141] stringi_1.7.6               sass_0.4.0                 </span>
<span class="co">## [143] blob_1.2.2                  textshaping_0.3.6          </span>
<span class="co">## [145] latticeExtra_0.6-29         memoise_2.0.1              </span>
<span class="co">## [147] dplyr_1.0.8</span></code></pre>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Aleksandra Bliznina, Michael Mansfield, Charles Plessy.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.2.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
