<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Introduction to the OikScrambling package • OikScrambling</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Introduction to the OikScrambling package">
<meta property="og:description" content="OikScrambling">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">OikScrambling</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.5.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/OikScrambling.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/ColinearityInterruptors.html">Regions interrupting colinearity</a>
    </li>
    <li>
      <a href="../articles/DistanceMeasures.html">Distance Measures</a>
    </li>
    <li>
      <a href="../articles/GenomicFeatures.html">Introduction to the OikScrambling package</a>
    </li>
    <li>
      <a href="../articles/Inversions.html">Inversions</a>
    </li>
    <li>
      <a href="../articles/LoadGenomicBreaks.html">Load genomic break data</a>
    </li>
    <li>
      <a href="../articles/OrthogroupSyntenies.html">Orthogroup syntenies</a>
    </li>
    <li>
      <a href="../articles/PWM.html">Motif finding and analysis</a>
    </li>
    <li>
      <a href="../articles/ParallelPlots.html">Whole-chromosome parallel plots</a>
    </li>
    <li>
      <a href="../articles/RegionWidths.html">Width of the aligned and mapped regions.</a>
    </li>
    <li>
      <a href="../articles/RepeatRegions.html">Repeat regions</a>
    </li>
    <li>
      <a href="../articles/Translocations.html">Translocations</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/oist/LuscombeU_OikScrambling/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="GenomicFeatures_files/header-attrs-2.11/header-attrs.js"></script><script src="GenomicFeatures_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Introduction to the OikScrambling package</h1>
                        <h4 class="author">Charlotte West</h4>
                        <h4 class="author">Charles Plessy</h4>
            
            <h4 class="date">24 November, 2021</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/oist/LuscombeU_OikScrambling/blob/master/vignettes/GenomicFeatures.Rmd"><code>vignettes/GenomicFeatures.Rmd</code></a></small>
      <div class="hidden name"><code>GenomicFeatures.Rmd</code></div>

    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">knitr</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/knitr/man/opts_chunk.html">opts_chunk</a></span><span class="op">$</span><span class="fu">set</span><span class="op">(</span>cache <span class="op">=</span> <span class="cn">TRUE</span>, cache.lazy <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
<span class="fu">knitr</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/knitr/man/opts_knit.html">opts_knit</a></span><span class="op">$</span><span class="fu">set</span><span class="op">(</span>verbose <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<div id="load-packages-and-data" class="section level1">
<h1 class="hasAnchor">
<a href="#load-packages-and-data" class="anchor"></a>Load packages and data</h1>
<p>See <code>?OikScrambling:::loadAllGenomes</code>, <code>?OikScrambling:::loadAllTranscriptsGR</code>, and <code><a href="../articles/LoadGenomicBreaks.html">vignette("LoadGenomicBreaks", package = "OikScrambling")</a></code> for how the different objects are prepared.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/oist/LuscombeU_OikScrambling">'OikScrambling'</a></span><span class="op">)</span>   |&gt; <span class="fu"><a href="https://rdrr.io/r/base/message.html">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="st"><a href="https://bioconductor.org/packages/GenomicFeatures">'GenomicFeatures'</a></span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/r/base/message.html">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="op">)</span>
<span class="op">(</span><span class="va">genomes</span>     <span class="op">&lt;-</span> <span class="fu">OikScrambling</span><span class="fu">:::</span><span class="fu"><a href="../reference/loadAllGenomes.html">loadAllGenomes</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code>## List of length 6
## names(6): Oki Osa Bar Kum Aom Nor</code></pre>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="op">(</span><span class="va">transcripts</span> <span class="op">&lt;-</span> <span class="fu">OikScrambling</span><span class="fu">:::</span><span class="fu"><a href="../reference/loadAllTranscriptsGR.html">loadAllTranscriptsGR</a></span><span class="op">(</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/r/base/warning.html">suppressWarnings</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code>## List of length 8
## names(8): Oki Osa Bar Kum Aom Nor Ply Ros</code></pre>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="op">(</span><span class="va">annots</span>      <span class="op">&lt;-</span> <span class="fu">OikScrambling</span><span class="fu">:::</span><span class="fu"><a href="../reference/loadAllAnnotations.html">loadAllAnnotations</a></span><span class="op">(</span><span class="op">)</span>   |&gt; <span class="fu"><a href="https://rdrr.io/r/base/warning.html">suppressWarnings</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code>## List of length 8
## names(8): Oki Osa Bar Kum Aom Nor Ply Ros</code></pre>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/load.html">load</a></span><span class="op">(</span><span class="st">"BreakPoints.Rdata"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code>##  [1] "gbs"       "unal"      "coa"       "unmap"     "unalMap"   "tra"      
##  [7] "tra2"      "coa2"      "unmap2"    "longShort"</code></pre>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html">requireNamespace</a></span><span class="op">(</span><span class="st">"rGADEM"</span><span class="op">)</span></code></pre></div>
<pre><code>## Loading required namespace: rGADEM</code></pre>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html">requireNamespace</a></span><span class="op">(</span><span class="st">"ggseqlogo"</span><span class="op">)</span></code></pre></div>
<pre><code>## Loading required namespace: ggseqlogo</code></pre>
</div>
<div id="tandem-repeats" class="section level1">
<h1 class="hasAnchor">
<a href="#tandem-repeats" class="anchor"></a>Tandem repeats</h1>
<p>Repeated regions of genomes are notoriously difficult to assemble and align. Thus, information about an alignment stop’s proximity to a tandem repeat is useful information in breakpoint classification. Given a user-defined tolerance, the function <code>tan_bp</code> will classify alignment stops as either far from, close to, or within tandem repeats (elements 1,2 and 3 of the produced list, respectively). The function requires the tandem repeat coverage of one of the genomes. By default, the function will expect the target genome tandem coverage. Setting <code>query_tf = TRUE</code> will switch the outcome to be query genome oriented.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># </span>
<span class="co"># tan_O &lt;- rtracklayer::import.gff3(system.file("extdata/OSKA2016.tantanf4w100.gff3.gz",  package="GenomicBreaks"))</span>
<span class="co"># tan_O &lt;- GRanges(tan_O, seqinfo = seqinfo(OSKA2016))</span>
<span class="co"># tan_gr_O &lt;- tan_bp(gr_ob = gr_O_Oki, tan = tan_O, tol = 50)</span>
<span class="co"># tan_gr_O</span>
<span class="co"># </span>
<span class="co"># tan_Oki &lt;- rtracklayer::import.gff3(system.file("extdata/I69-5.tantanf4w100.gff3.gz",  package="GenomicBreaks"))</span>
<span class="co"># tan_Oki &lt;- GRanges(tan_Oki, seqinfo = seqinfo(OKI2018_I69))</span>
<span class="co"># tan_gr_Oki &lt;- tan_bp(gr_ob = gr_O_Oki, tan = tan_Oki, tol = 50, query_tf = TRUE)</span>
<span class="co"># tan_gr_Oki</span></code></pre></div>
<p>We can use another script to plot the density of tandem repeats around alignment stops (and breakpoints). In the following plots, breakpoints are aligned at the centre of the plots. Intuitively, this function can be generalised to see the density of any binary coverage characteristic around breakpoints, given a GRanges object with per-base coverage of a given characteristic. Here we stick with tandem repeat info:</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># hm_tan_O &lt;- gr_O_Oki %&gt;%</span>
<span class="co">#   feature_coverage(tan_O, win = 1000, lab = "Osaka ~ Okinawa", direction = "left") %&gt;%</span>
<span class="co">#   smoothHeatmap(output.size = c(2000,500), algorithm = "kernel") %T&gt;%</span>
<span class="co">#   plotHeatmap</span>
<span class="co"># </span>
<span class="co"># hm_red_tan_O &lt;- reduced_gr_O_Oki %&gt;%</span>
<span class="co">#   feature_coverage(tan_O, win = 1000, lab = "Osaka ~ Okinawa", direction = "left") %&gt;%</span>
<span class="co">#   smoothHeatmap(output.size = c(2000,500), algorithm = "kernel") %T&gt;%</span>
<span class="co">#   plotHeatmap</span>
<span class="co"># </span>
<span class="co"># hm_tan_Oki &lt;- gr_Oki_O %&gt;%</span>
<span class="co">#   feature_coverage(tan_Oki, win = 1000, lab = "Osaka ~ Okinawa", direction = "left") %&gt;%</span>
<span class="co">#   smoothHeatmap(output.size = c(2000,500), algorithm = "kernel") %T&gt;%</span>
<span class="co">#   plotHeatmapList</span></code></pre></div>
</div>
<div id="coverage" class="section level1">
<h1 class="hasAnchor">
<a href="#coverage" class="anchor"></a>Coverage</h1>
<p>Another characteristic of alignment stops that we can examine is the coverage depth over said alignment stops. By aligning raw reads to assembled genomes, we can obtain per-base coverage depth for target or query genome. Low coverage directly on and surrounding an alignment stop may suggest unreliability in it being a true breakpoint. The function <code>bp_coverage</code> returns a GRanges object of the bps, with associated averaged and point converage in the metadatacolumns.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># cov_gr_O_Oki &lt;- bp_coverage(gr_ob = q_Oki, cov_gr = Oki_cov_pb, win = 50)</span>
<span class="co"># cov_gr_O_Oki</span></code></pre></div>
</div>
<div id="nucleic-acid-content-heatmaps" class="section level1">
<h1 class="hasAnchor">
<a href="#nucleic-acid-content-heatmaps" class="anchor"></a>Nucleic acid content heatmaps</h1>
<p>The <code><a href="https://oist.github.io/GenomicBreaks/reference/bp_heatmap.html">GenomicBreaks::bp_heatmap()</a></code> function plots nucleic acid content in windows centered on alignment stops. Lets look at <code>GC</code> content before and after coalescing:</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">heatmaps</span><span class="op">)</span>
<span class="co">#hm_GC1 &lt;- sapply(gbs,  bp_heatmap, basep_range = 1000, pat = "GC", direction = "left") |&gt; SimpleList()</span>
<span class="va">hm_GC2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="va">coa</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">15</span><span class="op">]</span>,  <span class="va">bp_heatmap</span>, basep_range <span class="op">=</span> <span class="fl">1000</span>, pat <span class="op">=</span> <span class="st">"GC"</span>, direction <span class="op">=</span> <span class="st">"left"</span><span class="op">)</span> |&gt; <span class="fu">SimpleList</span><span class="op">(</span><span class="op">)</span>
<span class="co">#hm_GC3 &lt;- sapply(coa2, bp_heatmap, basep_range = 1000, pat = "GC", direction = "left") |&gt; SimpleList()</span>

<span class="va">maxHeight</span> <span class="op">&lt;-</span> <span class="fl">2000</span>
<span class="co">#sm_GC1 &lt;- sapply(hm_GC1, smoothHeatmap, output.size = c(maxHeight,     500), algorithm = "kernel") |&gt; SimpleList()</span>
<span class="va">sm_GC2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="va">hm_GC2</span>, <span class="va">smoothHeatmap</span>, output.size <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">maxHeight</span>,     <span class="fl">500</span><span class="op">)</span>, algorithm <span class="op">=</span> <span class="st">"kernel"</span><span class="op">)</span> |&gt; <span class="fu">SimpleList</span><span class="op">(</span><span class="op">)</span>
<span class="co">#sm_GC3 &lt;- sapply(hm_GC3, smoothHeatmap, output.size = c(maxHeight,     500), algorithm = "kernel") |&gt; SimpleList()</span>

<span class="co">#plotHeatmapList(sm_GC1[1:5])</span>

<span class="fu"><a href="https://rdrr.io/pkg/heatmaps/man/plotHeatmapList.html">plotHeatmapList</a></span><span class="op">(</span><span class="va">sm_GC2</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span><span class="op">)</span></code></pre></div>
<p><img src="GenomicFeatures_files/figure-html/unnamed-chunk-4-1.png" width="1152"></p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#plotHeatmapList(sm_GC2[6:10])</span>
<span class="co">#plotHeatmapList(sm_GC2[11:15])</span>

<span class="co">#plotHeatmapList(sm_GC3[1:5])</span></code></pre></div>
<p>The alignment stops are ordered in such a way that start (or left breaks) are centred on the top half of the plot, and end alignment stops are centred on the bottom. This is why we see different directionality in the plots. At the alignment stops, the GC content seems to be lower, but higher once within the aligned region.</p>
<p>Now looking at TATA content:</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#hm_GC1_TATA &lt;- sapply(gbs, bp_heatmap, basep_range = 1000, pat = "TATA", direction = "left") |&gt; SimpleList()</span>
<span class="va">hm_GC2_TATA</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="va">coa</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">15</span><span class="op">]</span>, <span class="va">bp_heatmap</span>, basep_range <span class="op">=</span> <span class="fl">1000</span>, pat <span class="op">=</span> <span class="st">"TATA"</span>, direction <span class="op">=</span> <span class="st">"left"</span><span class="op">)</span> |&gt; <span class="fu">SimpleList</span><span class="op">(</span><span class="op">)</span>

<span class="va">maxHeight</span> <span class="op">&lt;-</span> <span class="fl">2000</span>
<span class="co">#sm_GC1_TATA &lt;- sapply(hm_GC1_TATA, smoothHeatmap, output.size = c(maxHeight,     500), algorithm = "kernel") |&gt; SimpleList()</span>
<span class="va">sm_GC2_TATA</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="va">hm_GC2_TATA</span>, <span class="va">smoothHeatmap</span>, output.size <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">maxHeight</span>,     <span class="fl">500</span><span class="op">)</span>, algorithm <span class="op">=</span> <span class="st">"kernel"</span><span class="op">)</span> |&gt; <span class="fu">SimpleList</span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code>## 
## Calculating kernel density...
## 
## Calculating kernel density...
## 
## Calculating kernel density...
## 
## Calculating kernel density...
## 
## Calculating kernel density...
## 
## Calculating kernel density...
## 
## Calculating kernel density...
## 
## Calculating kernel density...
## 
## Calculating kernel density...
## 
## Calculating kernel density...
## 
## Calculating kernel density...
## 
## Calculating kernel density...
## 
## Calculating kernel density...
## 
## Calculating kernel density...
## 
## Calculating kernel density...</code></pre>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#sm_GC3_TATA &lt;- sapply(hm_GC3_TATA, smoothHeatmap, output.size = c(maxHeight,     500), algorithm = "kernel") |&gt; SimpleList()</span>

<span class="co">#plotHeatmapList(sm_GC1_TATA)</span>
<span class="fu"><a href="https://rdrr.io/pkg/heatmaps/man/plotHeatmapList.html">plotHeatmapList</a></span><span class="op">(</span><span class="va">sm_GC2_TATA</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span><span class="op">)</span></code></pre></div>
<pre><code>## plotting heatmap TATA</code></pre>
<pre><code>## plotting heatmap TATA
## plotting heatmap TATA
## plotting heatmap TATA
## plotting heatmap TATA</code></pre>
<p><img src="GenomicFeatures_files/figure-html/TATA-1.png" width="1152"></p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#plotHeatmapList(sm_GC3_TATA)</span></code></pre></div>
<p>The directionaility here is consistent with the <code>GC</code> content analysis. Furthermore; “TATA box sequence can act as a basal promoter element not only for RNA polymerase II (RNAP II) transcription, but also for transcription by RNA polymerase III (RNAP III)” - Wang Y, Jensen RC, Stumph WE. Role of TATA box sequence and orientation in determining RNA polymerase II/III transcription specificity. Nucleic Acids Res. 1996;24(15):3100–3106. <a href="doi:10.1093/nar/24.15.3100" class="uri">doi:10.1093/nar/24.15.3100</a>. However, the areas are not particularly enriched for TATA boxes, so it does not necessarily show that breakpoints are occurring directly after promoter regions.</p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#hm_GC1_mot &lt;- sapply(gbs, bp_heatmap, basep_range = 1000, pat = "AGCNGC", direction = "left") |&gt; SimpleList()</span>
<span class="va">hm_GC2_mot</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="va">coa</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">15</span><span class="op">]</span>, <span class="va">bp_heatmap</span>, basep_range <span class="op">=</span> <span class="fl">1000</span>, pat <span class="op">=</span> <span class="st">"AGCNGC"</span>, direction <span class="op">=</span> <span class="st">"left"</span><span class="op">)</span> |&gt; <span class="fu">SimpleList</span><span class="op">(</span><span class="op">)</span>

<span class="va">maxHeight</span> <span class="op">&lt;-</span> <span class="fl">2000</span>
<span class="co">#sm_GC1_mot &lt;- sapply(hm_GC1_mot, smoothHeatmap, output.size = c(maxHeight,     500), algorithm = "kernel") |&gt; SimpleList()</span>
<span class="va">sm_GC2_mot</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="va">hm_GC2_mot</span>, <span class="va">smoothHeatmap</span>, output.size <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">maxHeight</span>,     <span class="fl">500</span><span class="op">)</span>, algorithm <span class="op">=</span> <span class="st">"kernel"</span><span class="op">)</span> |&gt; <span class="fu">SimpleList</span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code>## 
## Calculating kernel density...
## 
## Calculating kernel density...
## 
## Calculating kernel density...
## 
## Calculating kernel density...
## 
## Calculating kernel density...
## 
## Calculating kernel density...
## 
## Calculating kernel density...
## 
## Calculating kernel density...
## 
## Calculating kernel density...
## 
## Calculating kernel density...
## 
## Calculating kernel density...
## 
## Calculating kernel density...
## 
## Calculating kernel density...
## 
## Calculating kernel density...
## 
## Calculating kernel density...</code></pre>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#sm_GC3_mot &lt;- sapply(hm_GC3_mot, smoothHeatmap, output.size = c(maxHeight,     500), algorithm = "kernel") |&gt; SimpleList()</span>

<span class="co">#plotHeatmapList(sm_GC1_mot)</span>
<span class="fu"><a href="https://rdrr.io/pkg/heatmaps/man/plotHeatmapList.html">plotHeatmapList</a></span><span class="op">(</span><span class="va">sm_GC2_mot</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span><span class="op">)</span></code></pre></div>
<pre><code>## plotting heatmap AGCNGC</code></pre>
<pre><code>## plotting heatmap AGCNGC
## plotting heatmap AGCNGC
## plotting heatmap AGCNGC
## plotting heatmap AGCNGC</code></pre>
<p><img src="GenomicFeatures_files/figure-html/AAGCsGCwwmkCGrCTTyn-1.png" width="1152"></p>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#plotHeatmapList(sm_GC3_mot)</span></code></pre></div>
<div id="cluster-analysis" class="section level2">
<h2 class="hasAnchor">
<a href="#cluster-analysis" class="anchor"></a>Cluster analysis</h2>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># # Accessory functions</span>
<span class="co"># clusterHeatmap &lt;- function(hm, k = 2) kmeans(image(hm), k)</span>
<span class="co"># </span>
<span class="co"># orderHeatmap &lt;- function(hm, cl) {</span>
<span class="co">#   image(hm) &lt;- image(hm)[order(cl$cluster),]</span>
<span class="co">#   hm</span>
<span class="co"># }</span>
<span class="co"># </span>
<span class="co"># subsetHeatmap &lt;- function(hm, bool) {</span>
<span class="co">#   image(hm) &lt;- image(hm)[bool,]</span>
<span class="co">#   hm</span>
<span class="co"># }</span>
<span class="co"># </span>
<span class="co"># # DAta</span>
<span class="co"># hm_GC   &lt;- bp_heatmap(      gr_Oki_O, basep_range  = 1000, pat = "GC",     direction = "left")</span>
<span class="co"># hm_TATA &lt;- bp_heatmap(      gr_Oki_O, basep_range  = 1000, pat = "TATA",   direction = "left")</span>
<span class="co"># hm_tan  &lt;- feature_coverage(gr_Oki_O, tan_Oki, win = 1000, lab = "Tandem", direction = "left")</span>
<span class="co"># </span>
<span class="co"># # Define 3 K-mean clusters</span>
<span class="co"># clusterHeatmap(hm_GC, 3) -&gt; hm_cl_k4</span>
<span class="co"># </span>
<span class="co"># # Order and smooth</span>
<span class="co"># shm_GC &lt;- orderHeatmap(hm_GC, hm_cl_k4) %&gt;%</span>
<span class="co">#   smoothHeatmap(output.size = c(2000,500), algorithm = "kernel")</span>
<span class="co"># </span>
<span class="co"># shm_TATA &lt;- orderHeatmap(hm_TATA, hm_cl_k4) %&gt;%</span>
<span class="co">#   smoothHeatmap(output.size = c(2000,500), algorithm = "kernel")</span>
<span class="co"># </span>
<span class="co"># shm_tan &lt;- orderHeatmap(hm_tan, hm_cl_k4) %&gt;%</span>
<span class="co">#   smoothHeatmap(output.size = c(2000,500), algorithm = "kernel")</span>
<span class="co"># </span>
<span class="co"># # Plot the ordered data</span>
<span class="co"># plotHeatmapList(list(shm_GC, shm_TATA, shm_tan))</span>
<span class="co"># par(mfrow = c(1,1)) # plotHeatmapList messes with mrfrow...</span>
<span class="co"># plotHeatmapMeta(list(hm_GC, hm_TATA, hm_tan))</span>
<span class="co"># </span>
<span class="co"># plotHeatmapMeta(list(subsetHeatmap(hm_GC, hm_cl_k4$cluster == 1),</span>
<span class="co">#                      subsetHeatmap(hm_GC, hm_cl_k4$cluster == 2),</span>
<span class="co">#                      subsetHeatmap(hm_GC, hm_cl_k4$cluster == 3)))</span>
<span class="co"># </span>
<span class="co"># kmeanOrderedPlot &lt;- function (hm, k=3, seed = 1) {</span>
<span class="co">#   set.seed(seed)</span>
<span class="co">#   </span>
<span class="co">#   cl &lt;- kmeans(image(hm), k)</span>
<span class="co">#   mat &lt;- image(hm)[order(cl$cluster),]</span>
<span class="co">#   </span>
<span class="co">#   kmhm = Heatmap(</span>
<span class="co">#     mat,</span>
<span class="co">#     coords=c(-500,500),</span>
<span class="co">#     label="kmeans",</span>
<span class="co">#     scale=range(mat))</span>
<span class="co">#   </span>
<span class="co">#   plotHeatmapList(kmhm,</span>
<span class="co">#                   cex.label=1.5,</span>
<span class="co">#                   partition=as.vector(table(cl$cluster)),</span>
<span class="co">#                   partition.legend=TRUE,</span>
<span class="co">#                   partition.lines=TRUE,</span>
<span class="co">#                   legend=TRUE,</span>
<span class="co">#                   legend.pos="r",</span>
<span class="co">#                   legend.width=0.3)</span>
<span class="co"># }</span>
<span class="co"># </span>
<span class="co"># kmeanOrderedPlot(shm_GC)</span></code></pre></div>
</div>
<div id="evidence-for-breakpoint-hotspots" class="section level2">
<h2 class="hasAnchor">
<a href="#evidence-for-breakpoint-hotspots" class="anchor"></a>Evidence for breakpoint hotspots</h2>
<p>The function <code>bp_pair_analysis</code> takes two pairwise alignments with the same target genome, and plots the alignment stops on to the centred alignment stops of the other. In the plot below, Osaka is there consistent target genome, and as such acts as a coordinate system to relate alignment stops across genomes. Lets produce two of these 3-way analyses; one between Osaka-Okinawa-Norway, and one between Osaka-Okinawa-Aomori</p>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># hm1 &lt;- bp_pair_analysis(gr_ref_q1 = gr_O_Oki, gr_ref_q2 = gr_O_N, win = 1000, lab = "Oki~Nor")</span>
<span class="co"># plotHeatmapMeta(hm1)</span>
<span class="co"># </span>
<span class="co"># hm2 &lt;- bp_pair_analysis(gr_ref_q1 = gr_O_Oki, gr_ref_q2 = gr_O_A, win = 1000, lab = "Oki~Aom")</span>
<span class="co"># plotHeatmapMeta(hm2)</span></code></pre></div>
<p>The accumulation of alignment breaks of one pairwise alignment onto another suggests the existence of breakpoint hotspots; regions where breaks are far more likely to occur. Intuitively, this would lead to the presumption of synteny blocks, also. The pattern is consistent after coalescing, too;</p>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># reduced_gr_O_N &lt;- coalesce_contigs(gr_ob = gr_O_N, tol = 500)</span>
<span class="co"># </span>
<span class="co"># hm3 &lt;- bp_pair_analysis(gr_ref_q1 = reduced_gr_O_Oki, gr_ref_q2 = reduced_gr_O_N, win = 1000, lab = "Oki~Nor")</span>
<span class="co"># plotHeatmapMeta(hm3)</span></code></pre></div>
</div>
<div id="tandem-repeat-coverage-around-alignment-stops-breakpoints" class="section level2">
<h2 class="hasAnchor">
<a href="#tandem-repeat-coverage-around-alignment-stops-breakpoints" class="anchor"></a>Tandem repeat coverage around alignment stops (breakpoints)</h2>
<p>The function <code><a href="https://oist.github.io/GenomicBreaks/reference/feature_coverage.html">GenomicBreaks::feature_coverage()</a></code> will plot the coverage of tandem repeats around centred alignment stops.</p>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># </span>
<span class="co"># hm_tan1 &lt;- feature_coverage(gr_ob = gr_O_Oki, feat = tan_O, win = 4000, lab = "Osk ~ Oki", direction = "left")</span>
<span class="co"># plotHeatmapMeta(hm_tan1)</span>
<span class="co">#  </span>
<span class="co"># hm_tan2 &lt;- feature_coverage(gr_ob = reduced_gr_O_Oki, feat = tan_O, win = 4000, lab = "OSK~Oki (coalesced)", direction = "left")</span>
<span class="co"># plotHeatmapMeta(hm_tan2)</span>
<span class="co"># </span>
<span class="co"># # should I also add comparison of proportions of tandem proximities before and after coalescion?</span></code></pre></div>
<p>There appears to be less tandem repeats around alignment stops after coalescion, inferring that perhaps that some are artefacts of difficult alignment near to tandem repeats.</p>
</div>
<div id="gene-feature-coverage-around-alignment-stops-breakpoints" class="section level2">
<h2 class="hasAnchor">
<a href="#gene-feature-coverage-around-alignment-stops-breakpoints" class="anchor"></a>Gene feature coverage around alignment stops (breakpoints)</h2>
<p>TODO: analyse separately the breakpoint regions that are in colinear regions?</p>
<p>The function <code>feature_coverage</code> will plot the coverage of tandem repeats around centred alignment stops.</p>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#current_GB &lt;- gr_Oki_O # To ease recomputing with another object</span>
<span class="va">current_GB</span> <span class="op">&lt;-</span> <span class="va">gbs</span><span class="op">$</span><span class="va">Oki_Osa</span>
<span class="co">#current_GB &lt;- coa$Oki_Osa</span>
<span class="va">gadems.invLeftGap</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html">readRDS</a></span><span class="op">(</span><span class="st">"gadems.invLeftGap.Rda"</span><span class="op">)</span>
<span class="va">pwm</span> <span class="op">&lt;-</span> <span class="va">gadems.invLeftGap</span><span class="op">$</span><span class="va">Oki_Osa</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span>
<span class="va">pwm</span><span class="op">@</span><span class="va">consensus</span></code></pre></div>
<pre><code>## [1] "rArAAGCCGCdwAGCsGCw"</code></pre>
<div class="sourceCode" id="cb37"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">pwmHits</span> <span class="op">&lt;-</span> <span class="fu">matchPWM</span><span class="op">(</span>pwm <span class="op">=</span> <span class="fu">rGADEM</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/rGADEM/man/gadem-class.html">getPWM</a></span><span class="op">(</span><span class="va">pwm</span><span class="op">)</span>, <span class="va">genomes</span><span class="op">$</span><span class="va">Oki</span><span class="op">)</span></code></pre></div>
<pre><code>## Warning in .Call2("XString_match_PWM", pwm, subject, min.score, count.only, :
## 'subject' contains letters not in [ACGT] ==&gt; assigned weight 0 to them

## Warning in .Call2("XString_match_PWM", pwm, subject, min.score, count.only, :
## 'subject' contains letters not in [ACGT] ==&gt; assigned weight 0 to them

## Warning in .Call2("XString_match_PWM", pwm, subject, min.score, count.only, :
## 'subject' contains letters not in [ACGT] ==&gt; assigned weight 0 to them

## Warning in .Call2("XString_match_PWM", pwm, subject, min.score, count.only, :
## 'subject' contains letters not in [ACGT] ==&gt; assigned weight 0 to them

## Warning in .Call2("XString_match_PWM", pwm, subject, min.score, count.only, :
## 'subject' contains letters not in [ACGT] ==&gt; assigned weight 0 to them

## Warning in .Call2("XString_match_PWM", pwm, subject, min.score, count.only, :
## 'subject' contains letters not in [ACGT] ==&gt; assigned weight 0 to them

## Warning in .Call2("XString_match_PWM", pwm, subject, min.score, count.only, :
## 'subject' contains letters not in [ACGT] ==&gt; assigned weight 0 to them

## Warning in .Call2("XString_match_PWM", pwm, subject, min.score, count.only, :
## 'subject' contains letters not in [ACGT] ==&gt; assigned weight 0 to them

## Warning in .Call2("XString_match_PWM", pwm, subject, min.score, count.only, :
## 'subject' contains letters not in [ACGT] ==&gt; assigned weight 0 to them

## Warning in .Call2("XString_match_PWM", pwm, subject, min.score, count.only, :
## 'subject' contains letters not in [ACGT] ==&gt; assigned weight 0 to them</code></pre>
<div class="sourceCode" id="cb39"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">hmList_OKI</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span><span class="op">)</span>

<span class="va">hmList_OKI</span><span class="op">$</span><span class="va">genes</span>      <span class="op">&lt;-</span> <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/feature_coverage.html">feature_coverage</a></span><span class="op">(</span><span class="va">current_GB</span>, <span class="fu"><a href="https://rdrr.io/pkg/GenomicFeatures/man/transcripts.html">genes</a></span><span class="op">(</span><span class="va">annots</span><span class="op">$</span><span class="va">Oki</span><span class="op">)</span>,                         win <span class="op">=</span> <span class="fl">2000</span>, lab <span class="op">=</span> <span class="st">"Oki genes"</span>, direction <span class="op">=</span> <span class="st">"left"</span><span class="op">)</span>
<span class="va">hmList_OKI</span><span class="op">$</span><span class="va">promoters</span>  <span class="op">&lt;-</span> <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/feature_coverage.html">feature_coverage</a></span><span class="op">(</span><span class="va">current_GB</span>, <span class="fu"><a href="https://rdrr.io/pkg/GenomicFeatures/man/transcripts.html">promoters</a></span><span class="op">(</span><span class="va">annots</span><span class="op">$</span><span class="va">Oki</span><span class="op">)</span>,                     win <span class="op">=</span> <span class="fl">2000</span>, lab <span class="op">=</span> <span class="st">"Oki promoters"</span>, direction <span class="op">=</span> <span class="st">"left"</span><span class="op">)</span></code></pre></div>
<pre><code>## Warning in valid.GenomicRanges.seqinfo(x, suggest.trim = TRUE): GRanges object contains 4 out-of-bound ranges located on sequences
##   chr1, chrUn_1, and chrUn_10. Note that ranges located on a sequence
##   whose length is unknown (NA) or on a circular sequence are not
##   considered out-of-bound (use seqlengths() and isCircular() to get the
##   lengths and circularity flags of the underlying sequences). You can use
##   trim() to trim these ranges. See ?`trim,GenomicRanges-method` for more
##   information.</code></pre>
<div class="sourceCode" id="cb41"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">hmList_OKI</span><span class="op">$</span><span class="va">exons</span>      <span class="op">&lt;-</span> <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/feature_coverage.html">feature_coverage</a></span><span class="op">(</span><span class="va">current_GB</span>, <span class="fu"><a href="https://rdrr.io/pkg/GenomicFeatures/man/exonicParts.html">exonicParts</a></span><span class="op">(</span><span class="va">annots</span><span class="op">$</span><span class="va">Oki</span><span class="op">)</span>,                   win <span class="op">=</span> <span class="fl">2000</span>, lab <span class="op">=</span> <span class="st">"Oki exons"</span>, direction <span class="op">=</span> <span class="st">"left"</span><span class="op">)</span>
<span class="va">hmList_OKI</span><span class="op">$</span><span class="va">introns</span>    <span class="op">&lt;-</span> <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/feature_coverage.html">feature_coverage</a></span><span class="op">(</span><span class="va">current_GB</span>, <span class="fu"><a href="https://rdrr.io/pkg/GenomicFeatures/man/exonicParts.html">intronicParts</a></span><span class="op">(</span><span class="va">annots</span><span class="op">$</span><span class="va">Oki</span><span class="op">)</span>,                 win <span class="op">=</span> <span class="fl">2000</span>, lab <span class="op">=</span> <span class="st">"Oki introns"</span>, direction <span class="op">=</span> <span class="st">"left"</span><span class="op">)</span>
<span class="va">hmList_OKI</span><span class="op">$</span><span class="va">cds</span>        <span class="op">&lt;-</span> <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/feature_coverage.html">feature_coverage</a></span><span class="op">(</span><span class="va">current_GB</span>, <span class="fu"><a href="https://rdrr.io/pkg/GenomicFeatures/man/transcripts.html">cds</a></span><span class="op">(</span><span class="va">annots</span><span class="op">$</span><span class="va">Oki</span><span class="op">)</span>,                           win <span class="op">=</span> <span class="fl">2000</span>, lab <span class="op">=</span> <span class="st">"Oki cds"</span>, direction <span class="op">=</span> <span class="st">"left"</span><span class="op">)</span>
<span class="va">hmList_OKI</span><span class="op">$</span><span class="va">utrs5</span>      <span class="op">&lt;-</span> <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/feature_coverage.html">feature_coverage</a></span><span class="op">(</span><span class="va">current_GB</span>, <span class="fu"><a href="https://rdrr.io/r/base/unlist.html">unlist</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/GenomicFeatures/man/transcriptsBy.html">fiveUTRsByTranscript</a></span><span class="op">(</span><span class="va">annots</span><span class="op">$</span><span class="va">Oki</span><span class="op">)</span><span class="op">)</span>,  win <span class="op">=</span> <span class="fl">2000</span>, lab <span class="op">=</span> <span class="st">"Oki 5′ UTRs"</span>, direction <span class="op">=</span> <span class="st">"left"</span><span class="op">)</span>
<span class="va">hmList_OKI</span><span class="op">$</span><span class="va">utrs3</span>      <span class="op">&lt;-</span> <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/feature_coverage.html">feature_coverage</a></span><span class="op">(</span><span class="va">current_GB</span>, <span class="fu"><a href="https://rdrr.io/r/base/unlist.html">unlist</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/GenomicFeatures/man/transcriptsBy.html">threeUTRsByTranscript</a></span><span class="op">(</span><span class="va">annots</span><span class="op">$</span><span class="va">Oki</span><span class="op">)</span><span class="op">)</span>, win <span class="op">=</span> <span class="fl">2000</span>, lab <span class="op">=</span> <span class="st">"Oki 3′ UTRs"</span>, direction <span class="op">=</span> <span class="st">"left"</span><span class="op">)</span>
<span class="va">hmList_OKI</span><span class="op">$</span><span class="va">pwm</span>        <span class="op">&lt;-</span> <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/feature_coverage.html">feature_coverage</a></span><span class="op">(</span><span class="va">current_GB</span>, <span class="va">pwmHits</span>                                  , win <span class="op">=</span> <span class="fl">2000</span>, lab <span class="op">=</span> <span class="st">"PWM"</span>, direction <span class="op">=</span> <span class="st">"left"</span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/pkg/heatmaps/man/plotHeatmapMeta.html">plotHeatmapMeta</a></span><span class="op">(</span><span class="va">hmList_OKI</span><span class="op">)</span></code></pre></div>
<p><img src="GenomicFeatures_files/figure-html/genomic_features-1.png" width="700"></p>
<div class="sourceCode" id="cb42"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/heatmaps/man/plotHeatmapMeta.html">plotHeatmapMeta</a></span><span class="op">(</span><span class="va">hmList_OKI</span><span class="op">$</span><span class="va">genes</span><span class="op">)</span></code></pre></div>
<p><img src="GenomicFeatures_files/figure-html/genomic_features-2.png" width="700"></p>
<div class="sourceCode" id="cb43"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/heatmaps/man/plotHeatmapMeta.html">plotHeatmapMeta</a></span><span class="op">(</span><span class="va">hmList_OKI</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"exons"</span>, <span class="st">"cds"</span><span class="op">)</span><span class="op">]</span><span class="op">)</span></code></pre></div>
<p><img src="GenomicFeatures_files/figure-html/genomic_features-3.png" width="700"></p>
<div class="sourceCode" id="cb44"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/heatmaps/man/plotHeatmapMeta.html">plotHeatmapMeta</a></span><span class="op">(</span><span class="va">hmList_OKI</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"exons"</span>, <span class="st">"introns"</span>, <span class="st">"utrs5"</span>, <span class="st">"utrs3"</span>, <span class="st">"pwm"</span><span class="op">)</span><span class="op">]</span><span class="op">)</span></code></pre></div>
<p><img src="GenomicFeatures_files/figure-html/genomic_features-4.png" width="700"></p>
<div class="sourceCode" id="cb45"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/heatmaps/man/plotHeatmapMeta.html">plotHeatmapMeta</a></span><span class="op">(</span><span class="va">hmList_OKI</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"introns"</span>, <span class="st">"utrs5"</span>, <span class="st">"utrs3"</span>, <span class="st">"pwm"</span><span class="op">)</span><span class="op">]</span><span class="op">)</span></code></pre></div>
<p><img src="GenomicFeatures_files/figure-html/genomic_features-5.png" width="700"></p>
<div class="sourceCode" id="cb46"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/heatmaps/man/plotHeatmapMeta.html">plotHeatmapMeta</a></span><span class="op">(</span><span class="va">hmList_OKI</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"utrs5"</span>, <span class="st">"utrs3"</span>, <span class="st">"pwm"</span><span class="op">)</span><span class="op">]</span><span class="op">)</span></code></pre></div>
<p><img src="GenomicFeatures_files/figure-html/genomic_features-6.png" width="700"></p>
<p>Aligned regions are enriched in exons and depleted in introns. Is it because exons are more alignable or is it because it is deleterous to break genes in exons ?</p>
</div>
<div id="gene-feature-coverage-around-pwm-hits" class="section level2">
<h2 class="hasAnchor">
<a href="#gene-feature-coverage-around-pwm-hits" class="anchor"></a>Gene feature coverage around PWM hits</h2>
<p>Let’s look at coverage around PWM hits of the AAGCsGCwwmkCGrCTTyn motif</p>
<div class="sourceCode" id="cb47"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">current_GB</span> <span class="op">&lt;-</span> <span class="va">pwmHits</span>

<span class="va">hmList_PWM_OKI</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span><span class="op">)</span>
<span class="va">hmList_PWM_OKI</span><span class="op">$</span><span class="va">genes</span>     <span class="op">&lt;-</span> <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/feature_coverage.html">feature_coverage</a></span><span class="op">(</span><span class="va">current_GB</span>, <span class="fu"><a href="https://rdrr.io/pkg/GenomicFeatures/man/transcripts.html">genes</a></span><span class="op">(</span><span class="va">annots</span><span class="op">$</span><span class="va">Oki</span><span class="op">)</span>,                         win <span class="op">=</span> <span class="fl">2000</span>, lab <span class="op">=</span> <span class="st">"Oki genes"</span>, direction <span class="op">=</span> <span class="st">"left"</span><span class="op">)</span>
<span class="va">hmList_PWM_OKI</span><span class="op">$</span><span class="va">promoters</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/feature_coverage.html">feature_coverage</a></span><span class="op">(</span><span class="va">current_GB</span>, <span class="fu"><a href="https://rdrr.io/pkg/GenomicFeatures/man/transcripts.html">promoters</a></span><span class="op">(</span><span class="va">annots</span><span class="op">$</span><span class="va">Oki</span><span class="op">)</span>,                     win <span class="op">=</span> <span class="fl">2000</span>, lab <span class="op">=</span> <span class="st">"Oki promoters"</span>, direction <span class="op">=</span> <span class="st">"left"</span><span class="op">)</span></code></pre></div>
<pre><code>## Warning in valid.GenomicRanges.seqinfo(x, suggest.trim = TRUE): GRanges object contains 4 out-of-bound ranges located on sequences
##   chr1, chrUn_1, and chrUn_10. Note that ranges located on a sequence
##   whose length is unknown (NA) or on a circular sequence are not
##   considered out-of-bound (use seqlengths() and isCircular() to get the
##   lengths and circularity flags of the underlying sequences). You can use
##   trim() to trim these ranges. See ?`trim,GenomicRanges-method` for more
##   information.</code></pre>
<div class="sourceCode" id="cb49"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">hmList_PWM_OKI</span><span class="op">$</span><span class="va">exons</span>     <span class="op">&lt;-</span> <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/feature_coverage.html">feature_coverage</a></span><span class="op">(</span><span class="va">current_GB</span>, <span class="fu"><a href="https://rdrr.io/pkg/GenomicFeatures/man/exonicParts.html">exonicParts</a></span><span class="op">(</span><span class="va">annots</span><span class="op">$</span><span class="va">Oki</span><span class="op">)</span>,                   win <span class="op">=</span> <span class="fl">2000</span>, lab <span class="op">=</span> <span class="st">"Oki exons"</span>, direction <span class="op">=</span> <span class="st">"left"</span><span class="op">)</span>
<span class="va">hmList_PWM_OKI</span><span class="op">$</span><span class="va">introns</span>   <span class="op">&lt;-</span> <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/feature_coverage.html">feature_coverage</a></span><span class="op">(</span><span class="va">current_GB</span>, <span class="fu"><a href="https://rdrr.io/pkg/GenomicFeatures/man/exonicParts.html">intronicParts</a></span><span class="op">(</span><span class="va">annots</span><span class="op">$</span><span class="va">Oki</span><span class="op">)</span>,                 win <span class="op">=</span> <span class="fl">2000</span>, lab <span class="op">=</span> <span class="st">"Oki introns"</span>, direction <span class="op">=</span> <span class="st">"left"</span><span class="op">)</span>
<span class="va">hmList_PWM_OKI</span><span class="op">$</span><span class="va">cds</span>       <span class="op">&lt;-</span> <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/feature_coverage.html">feature_coverage</a></span><span class="op">(</span><span class="va">current_GB</span>, <span class="fu"><a href="https://rdrr.io/pkg/GenomicFeatures/man/transcripts.html">cds</a></span><span class="op">(</span><span class="va">annots</span><span class="op">$</span><span class="va">Oki</span><span class="op">)</span>,                           win <span class="op">=</span> <span class="fl">2000</span>, lab <span class="op">=</span> <span class="st">"Oki cds"</span>, direction <span class="op">=</span> <span class="st">"left"</span><span class="op">)</span>
<span class="va">hmList_PWM_OKI</span><span class="op">$</span><span class="va">utrs5</span>     <span class="op">&lt;-</span> <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/feature_coverage.html">feature_coverage</a></span><span class="op">(</span><span class="va">current_GB</span>, <span class="fu"><a href="https://rdrr.io/r/base/unlist.html">unlist</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/GenomicFeatures/man/transcriptsBy.html">fiveUTRsByTranscript</a></span><span class="op">(</span><span class="va">annots</span><span class="op">$</span><span class="va">Oki</span><span class="op">)</span><span class="op">)</span>,  win <span class="op">=</span> <span class="fl">2000</span>, lab <span class="op">=</span> <span class="st">"Oki 5′ UTRs"</span>, direction <span class="op">=</span> <span class="st">"left"</span><span class="op">)</span>
<span class="va">hmList_PWM_OKI</span><span class="op">$</span><span class="va">utrs3</span>     <span class="op">&lt;-</span> <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/feature_coverage.html">feature_coverage</a></span><span class="op">(</span><span class="va">current_GB</span>, <span class="fu"><a href="https://rdrr.io/r/base/unlist.html">unlist</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/GenomicFeatures/man/transcriptsBy.html">threeUTRsByTranscript</a></span><span class="op">(</span><span class="va">annots</span><span class="op">$</span><span class="va">Oki</span><span class="op">)</span><span class="op">)</span>, win <span class="op">=</span> <span class="fl">2000</span>, lab <span class="op">=</span> <span class="st">"Oki 3′ UTRs"</span>, direction <span class="op">=</span> <span class="st">"left"</span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/pkg/heatmaps/man/plotHeatmapMeta.html">plotHeatmapMeta</a></span><span class="op">(</span><span class="va">hmList_PWM_OKI</span><span class="op">)</span></code></pre></div>
<p><img src="GenomicFeatures_files/figure-html/genomic_features_pwmHits-1.png" width="700"></p>
<div class="sourceCode" id="cb50"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/heatmaps/man/plotHeatmapMeta.html">plotHeatmapMeta</a></span><span class="op">(</span><span class="va">hmList_PWM_OKI</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"exons"</span>, <span class="st">"introns"</span><span class="op">)</span><span class="op">]</span><span class="op">)</span></code></pre></div>
<p><img src="GenomicFeatures_files/figure-html/genomic_features_pwmHits-2.png" width="700"></p>
<div class="sourceCode" id="cb51"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/heatmaps/man/plotHeatmapMeta.html">plotHeatmapMeta</a></span><span class="op">(</span><span class="va">hmList_PWM_OKI</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"exons"</span>, <span class="st">"utrs5"</span>, <span class="st">"utrs3"</span><span class="op">)</span><span class="op">]</span><span class="op">)</span></code></pre></div>
<p><img src="GenomicFeatures_files/figure-html/genomic_features_pwmHits-3.png" width="700"></p>
<div class="sourceCode" id="cb52"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/heatmaps/man/plotHeatmapMeta.html">plotHeatmapMeta</a></span><span class="op">(</span><span class="va">hmList_PWM_OKI</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"utrs5"</span>, <span class="st">"utrs3"</span><span class="op">)</span><span class="op">]</span><span class="op">)</span></code></pre></div>
<p><img src="GenomicFeatures_files/figure-html/genomic_features_pwmHits-4.png" width="700"></p>
<p>They are enriched in introns…</p>
<div class="sourceCode" id="cb53"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># </span>
<span class="co"># # Sandbox to make sanity checks.</span>
<span class="co"># </span>
<span class="co"># a &lt;- coa$Oki_Osa</span>
<span class="co"># b &lt;- unlist(fiveUTRsByTranscript(annots$Oki))</span>
<span class="co"># </span>
<span class="co"># hm_a &lt;- feature_coverage(a, b, win = 2000, lab = "All ranges", direction = "left")</span>
<span class="co"># hm_b &lt;- feature_coverage(subsetByOverlaps(a,  b), b, win = 2000, lab = "Subsetted ranges", direction = "left")</span>
<span class="co"># </span>
<span class="co"># plotHeatmapMeta(list(hm_a))</span>
<span class="co"># plotHeatmapMeta(list(hm_b))</span></code></pre></div>
</div>
<div id="okinawa-genome-coverage" class="section level2">
<h2 class="hasAnchor">
<a href="#okinawa-genome-coverage" class="anchor"></a>Okinawa genome coverage</h2>
<p>As previously discussed, low coverage over an alignment stop could lower the likelihood of it being considered as a breakpoint. We have per-base coverage depth information for the Okinawan genome (<code>Oki_cov_pb</code>). However, the coverage of this particular assembly is quite good. In fact, we can investigate the coverge over alignment stops from the information obtained using <code>master_bp_analysis</code>.</p>
<div class="sourceCode" id="cb54"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># fin_Oki &lt;- fin_gr_O_Oki[[2]]</span>
<span class="co"># min_cov &lt;- min(min(fin_Oki$left_cov_pb), min(fin_Oki$right_cov_pb)) # minimum coverage over an alignment stop</span>
<span class="co"># min_cov</span>
<span class="co"># length(fin_Oki[fin_Oki$left_cov_pb == min_cov]) + length(fin_Oki[fin_Oki$right_cov_pb == min_cov]) # how many of the minimum coverage is observed</span>
<span class="co"># length(fin_Oki[fin_Oki$left_cov_pb &lt;= 50]) + length(fin_Oki[fin_Oki$right_cov_pb &lt;= 50]) # how many alignment stops have a coverage of less than or euqal to 50</span></code></pre></div>
<p>We may choose to kick out the one alignment stop for which there is no coverage. However, only 15 out of 34572 alignment stops have a coverage of less than or equal to 50, meaning that it would be hard to exclude more than just a few alignment stops using coverage information for breakpoint analysis.</p>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Charles Plessy.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
