<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Genomic features of the breakpoints (heatmaps) • OikScrambling</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Genomic features of the breakpoints (heatmaps)">
<meta property="og:description" content="OikScrambling">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">OikScrambling</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">5.0.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/OikScrambling.html">Get started</a>
</li>
<li>
  <a href="../articles/LoadGenomicBreaks.html">Data load</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Supplemental material
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/OxfordPlots.html">Figure 1A DNA line plots</a>
    </li>
    <li>
      <a href="../articles/ParallelPlots.html">Figure 1B chromosome parallel plots</a>
    </li>
    <li>
      <a href="../articles/DistanceMeasures.html">Table 1 scrambling index</a>
    </li>
    <li>
      <a href="../articles/ChromosomePlots.html">Figure 5 genome-wide feature patterns</a>
    </li>
    <li>
      <a href="../articles/SquareWindowPlots.html">Figure 7 DNA line plots</a>
    </li>
    <li>
      <a href="../articles/ScalingBreakpoints.html">Figure 7 breakpoint number barplots</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/CNEs.html">Conserved non-coding elements</a>
    </li>
    <li>
      <a href="../articles/ChromosomePlots.html">Chromosome-scale feature density plots</a>
    </li>
    <li>
      <a href="../articles/CoalescingCutoffs.html">Modifying coalescent cutoffs</a>
    </li>
    <li>
      <a href="../articles/ColinearityInterruptors.html">Regions interrupting colinearity</a>
    </li>
    <li>
      <a href="../articles/DistanceMeasures.html">Distance Measures</a>
    </li>
    <li>
      <a href="../articles/GenomicFeatures.html">Genomic features of the breakpoints</a>
    </li>
    <li>
      <a href="../articles/GenomicHeatmaps.html">Genomic features of the breakpoints (heatmaps)</a>
    </li>
    <li>
      <a href="../articles/Inversions.html">Inversions</a>
    </li>
    <li>
      <a href="../articles/LoadGenomicBreaks.html">Load genomic break data</a>
    </li>
    <li>
      <a href="../articles/OperonConservation.html">Operon conservation analysis</a>
    </li>
    <li>
      <a href="../articles/OrthogroupSyntenies.html">Orthogroup syntenies</a>
    </li>
    <li>
      <a href="../articles/OxfordPlots.html">Alignment line plots, whole genomes</a>
    </li>
    <li>
      <a href="../articles/PAC3.html">Region of interest: PAC3</a>
    </li>
    <li>
      <a href="../articles/PWM.html">Motif search and analysis</a>
    </li>
    <li>
      <a href="../articles/ParallelPlots.html">Whole-chromosome parallel plots</a>
    </li>
    <li>
      <a href="../articles/RegionWidths.html">Width of the aligned and mapped regions.</a>
    </li>
    <li>
      <a href="../articles/RegionsOfInterest.html">Regions of interest</a>
    </li>
    <li>
      <a href="../articles/RepeatRegions.html">Repeat regions</a>
    </li>
    <li>
      <a href="../articles/ScalingBreakpoints.html">Scaling breakpoints</a>
    </li>
    <li>
      <a href="../articles/SegmentMovements.html">Segment movement distances</a>
    </li>
    <li>
      <a href="../articles/SquareWindowPlots.html">Alignment line plots, 10×10 Mbp windows</a>
    </li>
    <li>
      <a href="../articles/TFBS.html">Transcription factor binding sites</a>
    </li>
    <li>
      <a href="../articles/TandemSiteDuplications.html">Tandem site duplications</a>
    </li>
    <li>
      <a href="../articles/Translocations.html">Translocations</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/oist/LuscombeU_OikScrambling/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Genomic features of the breakpoints
(heatmaps)</h1>
                        <h4 data-toc-skip class="author">Charlotte
West</h4>
                        <h4 data-toc-skip class="author">Charles
Plessy</h4>
            
            <h4 data-toc-skip class="date">13 October, 2023</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/oist/LuscombeU_OikScrambling/blob/HEAD/vignettes/GenomicHeatmaps.Rmd" class="external-link"><code>vignettes/GenomicHeatmaps.Rmd</code></a></small>
      <div class="hidden name"><code>GenomicHeatmaps.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="load-packages-and-data">Load packages and data<a class="anchor" aria-label="anchor" href="#load-packages-and-data"></a>
</h2>
<p>See <code>?OikScrambling:::loadAllGenomes</code>,
<code>?OikScrambling:::loadAllTranscriptsGR</code>, and
<code><a href="../articles/LoadGenomicBreaks.html">vignette("LoadGenomicBreaks", package = "OikScrambling")</a></code>
for how the different objects are prepared.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/oist/LuscombeU_OikScrambling" class="external-link">'OikScrambling'</a></span><span class="op">)</span>   <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://bioconductor.org/packages/GenomicFeatures" class="external-link">'GenomicFeatures'</a></span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st">'heatmaps'</span><span class="op">)</span>        <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://patchwork.data-imaginist.com" class="external-link">'patchwork'</a></span><span class="op">)</span>       <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">genomes</span>     <span class="op">&lt;-</span> <span class="fu">OikScrambling</span><span class="fu">:::</span><span class="fu"><a href="../reference/loadAllGenomes.html">loadAllGenomes</a></span><span class="op">(</span>compat <span class="op">=</span> <span class="cn">F</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Warning in runHook(".onLoad", env, package.lib, package): input string</span></span>
<span><span class="co">## 'Génoscope' cannot be translated from 'ANSI_X3.4-1968' to UTF-8, but is valid</span></span>
<span><span class="co">## UTF-8</span></span>
<span></span>
<span><span class="co">## Warning in runHook(".onLoad", env, package.lib, package): input string</span></span>
<span><span class="co">## 'Génoscope' cannot be translated from 'ANSI_X3.4-1968' to UTF-8, but is valid</span></span>
<span><span class="co">## UTF-8</span></span></code></pre>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Cannot use Knitr cache as long as "annot" objects are used.</span></span>
<span><span class="va">annots</span>      <span class="op">&lt;-</span> <span class="fu">OikScrambling</span><span class="fu">:::</span><span class="fu"><a href="../reference/loadAllAnnotations.html">loadAllAnnotations</a></span><span class="op">(</span><span class="op">)</span>   <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/base/warning.html" class="external-link">suppressWarnings</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">reps</span>        <span class="op">&lt;-</span> <span class="fu">OikScrambling</span><span class="fu">:::</span><span class="fu"><a href="../reference/loadAllRepeats.html">loadAllRepeats</a></span><span class="op">(</span>compat <span class="op">=</span> <span class="cn">F</span><span class="op">)</span></span>
<span><span class="va">genes</span>       <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">annots</span>, <span class="kw">function</span><span class="op">(</span><span class="va">a</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/sort.html" class="external-link">sort</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/GenomicFeatures/man/transcripts.html" class="external-link">genes</a></span><span class="op">(</span><span class="va">a</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/SimpleList-class.html" class="external-link">SimpleList</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/load.html" class="external-link">load</a></span><span class="op">(</span><span class="st">"BreakPoints.Rdata"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"rGADEM"</span><span class="op">)</span>    <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"ggseqlogo"</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">ces</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="st">"CEs.Rds"</span><span class="op">)</span></span>
<span><span class="va">tfbs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="st">"pwmMatchesOki_12_95.Rds"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="nucleic-acid-content-heatmaps">Nucleic acid content heatmaps<a class="anchor" aria-label="anchor" href="#nucleic-acid-content-heatmaps"></a>
</h2>
<p>The <code><a href="https://oist.github.io/GenomicBreaks/reference/bp_heatmap.html" class="external-link">GenomicBreaks::bp_heatmap()</a></code> function plots nucleic
acid content in windows centered on alignment stops.</p>
<div class="section level3">
<h3 id="gc">GC<a class="anchor" aria-label="anchor" href="#gc"></a>
</h3>
<p>Lets look at <code>GC</code> content before and after coalescing:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">maxHeight</span> <span class="op">&lt;-</span> <span class="fl">2000</span></span>
<span><span class="va">sm_GC2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">coa</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">15</span><span class="op">]</span>,  <span class="va">bp_heatmap</span>, win <span class="op">=</span> <span class="fl">1000</span>, pat <span class="op">=</span> <span class="st">"GC"</span>, direction <span class="op">=</span> <span class="st">"left"</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>          <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">smoothHeatmap</span>, output.size <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">maxHeight</span>,     <span class="fl">500</span><span class="op">)</span>, algorithm <span class="op">=</span> <span class="st">"kernel"</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/SimpleList-class.html" class="external-link">SimpleList</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/heatmaps/man/plotHeatmapList.html" class="external-link">plotHeatmapList</a></span><span class="op">(</span><span class="va">sm_GC2</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span><span class="op">)</span></span></code></pre></div>
<p><img src="GenomicHeatmaps_files/figure-html/unnamed-chunk-1-1.png" width="1152"></p>
<p>The alignment stops are ordered in such a way that start (or left
breaks) are centred on the top half of the plot, and end alignment stops
are centred on the bottom. This is why we see different directionality
in the plots. At the alignment stops, the GC content seems to be lower,
but higher once within the aligned region, except for the Oki – Kum
pair.</p>
</div>
<div class="section level3">
<h3 id="tata">TATA<a class="anchor" aria-label="anchor" href="#tata"></a>
</h3>
<p>Now looking at TATA content:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">maxHeight</span> <span class="op">&lt;-</span> <span class="fl">2000</span></span>
<span><span class="va">sm_GC2_TATA</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">coa</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">15</span><span class="op">]</span>, <span class="va">bp_heatmap</span>, window <span class="op">=</span> <span class="fl">1000</span>, pat <span class="op">=</span> <span class="st">"TATA"</span>, direction <span class="op">=</span> <span class="st">"left"</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>               <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">smoothHeatmap</span>, output.size <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">maxHeight</span>,     <span class="fl">500</span><span class="op">)</span>, algorithm <span class="op">=</span> <span class="st">"kernel"</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/SimpleList-class.html" class="external-link">SimpleList</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/heatmaps/man/plotHeatmapList.html" class="external-link">plotHeatmapList</a></span><span class="op">(</span><span class="va">sm_GC2_TATA</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span><span class="op">)</span></span></code></pre></div>
<p><img src="GenomicHeatmaps_files/figure-html/TATA-1.png" width="1152"></p>
<p>The directionaility here is consistent with the <code>GC</code>
content analysis. Furthermore; “TATA box sequence can act as a basal
promoter element not only for RNA polymerase II (RNAP II) transcription,
but also for transcription by RNA polymerase III (RNAP III)” - Wang Y,
Jensen RC, Stumph WE. Role of TATA box sequence and orientation in
determining RNA polymerase II/III transcription specificity. Nucleic
Acids Res. 1996;24(15):3100–3106. <a href="doi:10.1093/nar/24.15.3100" class="uri">doi:10.1093/nar/24.15.3100</a>. However, the areas are not
particularly enriched for TATA boxes, so it does not necessarily show
that breakpoints are occurring directly after promoter regions.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">maxHeight</span> <span class="op">&lt;-</span> <span class="fl">2000</span></span>
<span><span class="va">sm_GC2_mot</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">coa</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">15</span><span class="op">]</span>, <span class="va">bp_heatmap</span>, window <span class="op">=</span> <span class="fl">1000</span>, pat <span class="op">=</span> <span class="st">"AGCNGC"</span>, direction <span class="op">=</span> <span class="st">"left"</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>              <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">smoothHeatmap</span>, output.size <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">maxHeight</span>,     <span class="fl">500</span><span class="op">)</span>, algorithm <span class="op">=</span> <span class="st">"kernel"</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/SimpleList-class.html" class="external-link">SimpleList</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/heatmaps/man/plotHeatmapList.html" class="external-link">plotHeatmapList</a></span><span class="op">(</span><span class="va">sm_GC2_mot</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span><span class="op">)</span></span></code></pre></div>
<p><img src="GenomicHeatmaps_files/figure-html/AAGCsGCwwmkCGrCTTyn-1.png" width="1152"></p>
</div>
<div class="section level3">
<h3 id="cluster-analysis">Cluster analysis<a class="anchor" aria-label="anchor" href="#cluster-analysis"></a>
</h3>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># # Accessory functions</span></span>
<span><span class="co"># clusterHeatmap &lt;- function(hm, k = 2) kmeans(image(hm), k)</span></span>
<span><span class="co"># </span></span>
<span><span class="co"># orderHeatmap &lt;- function(hm, cl) {</span></span>
<span><span class="co">#   image(hm) &lt;- image(hm)[order(cl$cluster),]</span></span>
<span><span class="co">#   hm</span></span>
<span><span class="co"># }</span></span>
<span><span class="co"># </span></span>
<span><span class="co"># subsetHeatmap &lt;- function(hm, bool) {</span></span>
<span><span class="co">#   image(hm) &lt;- image(hm)[bool,]</span></span>
<span><span class="co">#   hm</span></span>
<span><span class="co"># }</span></span>
<span><span class="co"># </span></span>
<span><span class="co"># # DAta</span></span>
<span><span class="co"># hm_GC   &lt;- bp_heatmap(      gr_Oki_O, basep_range  = 1000, pat = "GC",     direction = "left")</span></span>
<span><span class="co"># hm_TATA &lt;- bp_heatmap(      gr_Oki_O, basep_range  = 1000, pat = "TATA",   direction = "left")</span></span>
<span><span class="co"># hm_tan  &lt;- feature_coverage(gr_Oki_O, tan_Oki, win = 1000, lab = "Tandem", direction = "left")</span></span>
<span><span class="co"># </span></span>
<span><span class="co"># # Define 3 K-mean clusters</span></span>
<span><span class="co"># clusterHeatmap(hm_GC, 3) -&gt; hm_cl_k4</span></span>
<span><span class="co"># </span></span>
<span><span class="co"># # Order and smooth</span></span>
<span><span class="co"># shm_GC &lt;- orderHeatmap(hm_GC, hm_cl_k4) %&gt;%</span></span>
<span><span class="co">#   smoothHeatmap(output.size = c(2000,500), algorithm = "kernel")</span></span>
<span><span class="co"># </span></span>
<span><span class="co"># shm_TATA &lt;- orderHeatmap(hm_TATA, hm_cl_k4) %&gt;%</span></span>
<span><span class="co">#   smoothHeatmap(output.size = c(2000,500), algorithm = "kernel")</span></span>
<span><span class="co"># </span></span>
<span><span class="co"># shm_tan &lt;- orderHeatmap(hm_tan, hm_cl_k4) %&gt;%</span></span>
<span><span class="co">#   smoothHeatmap(output.size = c(2000,500), algorithm = "kernel")</span></span>
<span><span class="co"># </span></span>
<span><span class="co"># # Plot the ordered data</span></span>
<span><span class="co"># plotHeatmapList(list(shm_GC, shm_TATA, shm_tan))</span></span>
<span><span class="co"># par(mfrow = c(1,1)) # plotHeatmapList messes with mrfrow...</span></span>
<span><span class="co"># plotHeatmapMeta(list(hm_GC, hm_TATA, hm_tan))</span></span>
<span><span class="co"># </span></span>
<span><span class="co"># plotHeatmapMeta(list(subsetHeatmap(hm_GC, hm_cl_k4$cluster == 1),</span></span>
<span><span class="co">#                      subsetHeatmap(hm_GC, hm_cl_k4$cluster == 2),</span></span>
<span><span class="co">#                      subsetHeatmap(hm_GC, hm_cl_k4$cluster == 3)))</span></span>
<span><span class="co"># </span></span>
<span><span class="co"># kmeanOrderedPlot &lt;- function (hm, k=3, seed = 1) {</span></span>
<span><span class="co">#   set.seed(seed)</span></span>
<span><span class="co">#   </span></span>
<span><span class="co">#   cl &lt;- kmeans(image(hm), k)</span></span>
<span><span class="co">#   mat &lt;- image(hm)[order(cl$cluster),]</span></span>
<span><span class="co">#   </span></span>
<span><span class="co">#   kmhm = Heatmap(</span></span>
<span><span class="co">#     mat,</span></span>
<span><span class="co">#     coords=c(-500,500),</span></span>
<span><span class="co">#     label="kmeans",</span></span>
<span><span class="co">#     scale=range(mat))</span></span>
<span><span class="co">#   </span></span>
<span><span class="co">#   plotHeatmapList(kmhm,</span></span>
<span><span class="co">#                   cex.label=1.5,</span></span>
<span><span class="co">#                   partition=as.vector(table(cl$cluster)),</span></span>
<span><span class="co">#                   partition.legend=TRUE,</span></span>
<span><span class="co">#                   partition.lines=TRUE,</span></span>
<span><span class="co">#                   legend=TRUE,</span></span>
<span><span class="co">#                   legend.pos="r",</span></span>
<span><span class="co">#                   legend.width=0.3)</span></span>
<span><span class="co"># }</span></span>
<span><span class="co"># </span></span>
<span><span class="co"># kmeanOrderedPlot(shm_GC)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="evidence-for-breakpoint-hotspots">Evidence for breakpoint hotspots<a class="anchor" aria-label="anchor" href="#evidence-for-breakpoint-hotspots"></a>
</h3>
<p>The function <code><a href="https://oist.github.io/GenomicBreaks/reference/bp_pair_analysis.html" class="external-link">?GenomicBreaks::bp_pair_analysis</a></code> takes two
pairwise alignments with the same target genome, and plots the alignment
stops on to the centred alignment stops of the other.</p>
<p>In the plot below, Osaka is there consistent target genome, and as
such acts as a coordinate system to relate alignment stops across
genomes. Lets produce two of these 3-way analyses; one between
Osaka-Okinawa-Norway, and one between Osaka-Okinawa-Aomori.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/bp_pair_analysis.html" class="external-link">bp_pair_analysis</a></span><span class="op">(</span><span class="va">gbs</span><span class="op">$</span><span class="va">Osa_Oki</span>, <span class="va">gbs</span><span class="op">$</span><span class="va">Osa_Nor</span>, win <span class="op">=</span> <span class="fl">1000</span>, lab <span class="op">=</span> <span class="st">"Oki~Nor"</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/pkg/heatmaps/man/plotHeatmapMeta.html" class="external-link">plotHeatmapMeta</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p><img src="GenomicHeatmaps_files/figure-html/breakpoint_hotspot_gbs-1.png" width="700"></p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/bp_pair_analysis.html" class="external-link">bp_pair_analysis</a></span><span class="op">(</span><span class="va">gbs</span><span class="op">$</span><span class="va">Osa_Oki</span>, <span class="va">gbs</span><span class="op">$</span><span class="va">Osa_Aom</span>, win <span class="op">=</span> <span class="fl">1000</span>, lab <span class="op">=</span> <span class="st">"Oki~Aom"</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/pkg/heatmaps/man/plotHeatmapMeta.html" class="external-link">plotHeatmapMeta</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p><img src="GenomicHeatmaps_files/figure-html/breakpoint_hotspot_gbs-2.png" width="700"></p>
<p>The accumulation of alignment breaks of one pairwise alignment onto
another suggests the existence of breakpoint hotspots; regions where
breaks are far more likely to occur. Intuitively, this would lead to the
presumption of synteny blocks, also. The pattern is consistent after
coalescing, too;</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/bp_pair_analysis.html" class="external-link">bp_pair_analysis</a></span><span class="op">(</span><span class="va">coa</span><span class="op">$</span><span class="va">Osa_Oki</span>, <span class="va">coa</span><span class="op">$</span><span class="va">Osa_Nor</span>, win <span class="op">=</span> <span class="fl">1000</span>, lab <span class="op">=</span> <span class="st">"Oki~Nor"</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/pkg/heatmaps/man/plotHeatmapMeta.html" class="external-link">plotHeatmapMeta</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p><img src="GenomicHeatmaps_files/figure-html/breakpoint_hotspot_coa-1.png" width="700"></p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/bp_pair_analysis.html" class="external-link">bp_pair_analysis</a></span><span class="op">(</span><span class="va">coa</span><span class="op">$</span><span class="va">Osa_Oki</span>, <span class="va">coa</span><span class="op">$</span><span class="va">Osa_Aom</span>, win <span class="op">=</span> <span class="fl">1000</span>, lab <span class="op">=</span> <span class="st">"Oki~Aom"</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/pkg/heatmaps/man/plotHeatmapMeta.html" class="external-link">plotHeatmapMeta</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p><img src="GenomicHeatmaps_files/figure-html/breakpoint_hotspot_coa-2.png" width="700"></p>
</div>
</div>
<div class="section level2">
<h2 id="session-information">Session information<a class="anchor" aria-label="anchor" href="#session-information"></a>
</h2>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## R version 4.3.1 (2023-06-16)</span></span>
<span><span class="co">## Platform: x86_64-pc-linux-gnu (64-bit)</span></span>
<span><span class="co">## Running under: Debian GNU/Linux 12 (bookworm)</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Matrix products: default</span></span>
<span><span class="co">## BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.11.0 </span></span>
<span><span class="co">## LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.11.0</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## locale:</span></span>
<span><span class="co">##  [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       </span></span>
<span><span class="co">##  [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   </span></span>
<span><span class="co">##  [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          </span></span>
<span><span class="co">## [10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## time zone: Etc/UTC</span></span>
<span><span class="co">## tzcode source: system (glibc)</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## attached base packages:</span></span>
<span><span class="co">## [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span></span>
<span><span class="co">## [8] base     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## other attached packages:</span></span>
<span><span class="co">##  [1] BSgenome.Oidioi.genoscope.OdB3_1.0.0   </span></span>
<span><span class="co">##  [2] BSgenome.Oidioi.OIST.AOM.5.5f_1.0.1    </span></span>
<span><span class="co">##  [3] BSgenome.Oidioi.OIST.KUM.M3.7f_1.0.1   </span></span>
<span><span class="co">##  [4] BSgenome.Oidioi.OIST.Bar2.p4_1.0.1     </span></span>
<span><span class="co">##  [5] BSgenome.Oidioi.OIST.OSKA2016v1.9_1.0.0</span></span>
<span><span class="co">##  [6] BSgenome.Oidioi.OIST.OKI2018.I69_1.0.1 </span></span>
<span><span class="co">##  [7] patchwork_1.1.2                        </span></span>
<span><span class="co">##  [8] heatmaps_1.24.0                        </span></span>
<span><span class="co">##  [9] GenomicFeatures_1.52.1                 </span></span>
<span><span class="co">## [10] AnnotationDbi_1.62.2                   </span></span>
<span><span class="co">## [11] Biobase_2.60.0                         </span></span>
<span><span class="co">## [12] OikScrambling_5.0.0                    </span></span>
<span><span class="co">## [13] ggplot2_3.4.3                          </span></span>
<span><span class="co">## [14] GenomicBreaks_0.14.2                   </span></span>
<span><span class="co">## [15] BSgenome_1.68.0                        </span></span>
<span><span class="co">## [16] rtracklayer_1.60.0                     </span></span>
<span><span class="co">## [17] Biostrings_2.68.1                      </span></span>
<span><span class="co">## [18] XVector_0.40.0                         </span></span>
<span><span class="co">## [19] GenomicRanges_1.52.0                   </span></span>
<span><span class="co">## [20] GenomeInfoDb_1.36.1                    </span></span>
<span><span class="co">## [21] IRanges_2.34.1                         </span></span>
<span><span class="co">## [22] S4Vectors_0.38.1                       </span></span>
<span><span class="co">## [23] BiocGenerics_0.46.0                    </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## loaded via a namespace (and not attached):</span></span>
<span><span class="co">##   [1] splines_4.3.1               BiocIO_1.10.0              </span></span>
<span><span class="co">##   [3] filelock_1.0.2              bitops_1.0-7               </span></span>
<span><span class="co">##   [5] tibble_3.2.1                R.oo_1.25.0                </span></span>
<span><span class="co">##   [7] XML_3.99-0.14               rpart_4.1.19               </span></span>
<span><span class="co">##   [9] rGADEM_2.48.0               lifecycle_1.0.3            </span></span>
<span><span class="co">##  [11] rprojroot_2.0.3             lattice_0.20-45            </span></span>
<span><span class="co">##  [13] MASS_7.3-58.2               backports_1.4.1            </span></span>
<span><span class="co">##  [15] magrittr_2.0.3              Hmisc_5.1-0                </span></span>
<span><span class="co">##  [17] sass_0.4.7                  rmarkdown_2.23             </span></span>
<span><span class="co">##  [19] jquerylib_0.1.4             yaml_2.3.7                 </span></span>
<span><span class="co">##  [21] plotrix_3.8-2               DBI_1.1.3                  </span></span>
<span><span class="co">##  [23] CNEr_1.36.0                 minqa_1.2.5                </span></span>
<span><span class="co">##  [25] RColorBrewer_1.1-3          ade4_1.7-22                </span></span>
<span><span class="co">##  [27] abind_1.4-5                 zlibbioc_1.46.0            </span></span>
<span><span class="co">##  [29] purrr_1.0.2                 R.utils_2.12.2             </span></span>
<span><span class="co">##  [31] RCurl_1.98-1.12             nnet_7.3-18                </span></span>
<span><span class="co">##  [33] pracma_2.4.2                rappdirs_0.3.3             </span></span>
<span><span class="co">##  [35] GenomeInfoDbData_1.2.10     seqLogo_1.66.0             </span></span>
<span><span class="co">##  [37] gdata_2.19.0                annotate_1.78.0            </span></span>
<span><span class="co">##  [39] pkgdown_2.0.7               codetools_0.2-19           </span></span>
<span><span class="co">##  [41] DelayedArray_0.26.7         xml2_1.3.5                 </span></span>
<span><span class="co">##  [43] tidyselect_1.2.0            shape_1.4.6                </span></span>
<span><span class="co">##  [45] ggseqlogo_0.1               lme4_1.1-34                </span></span>
<span><span class="co">##  [47] BiocFileCache_2.8.0         matrixStats_1.0.0          </span></span>
<span><span class="co">##  [49] base64enc_0.1-3             GenomicAlignments_1.36.0   </span></span>
<span><span class="co">##  [51] jsonlite_1.8.7              mitml_0.4-5                </span></span>
<span><span class="co">##  [53] Formula_1.2-5               survival_3.5-3             </span></span>
<span><span class="co">##  [55] iterators_1.0.14            systemfonts_1.0.5          </span></span>
<span><span class="co">##  [57] foreach_1.5.2               progress_1.2.2             </span></span>
<span><span class="co">##  [59] tools_4.3.1                 ragg_1.2.5                 </span></span>
<span><span class="co">##  [61] Rcpp_1.0.11                 glue_1.6.2                 </span></span>
<span><span class="co">##  [63] gridExtra_2.3               pan_1.8                    </span></span>
<span><span class="co">##  [65] xfun_0.40                   MatrixGenerics_1.12.2      </span></span>
<span><span class="co">##  [67] EBImage_4.42.0              dplyr_1.1.3                </span></span>
<span><span class="co">##  [69] withr_2.5.1                 fastmap_1.1.1              </span></span>
<span><span class="co">##  [71] boot_1.3-28.1               fansi_1.0.5                </span></span>
<span><span class="co">##  [73] digest_0.6.33               R6_2.5.1                   </span></span>
<span><span class="co">##  [75] mice_3.16.0                 textshaping_0.3.7          </span></span>
<span><span class="co">##  [77] colorspace_2.1-0            GO.db_3.17.0               </span></span>
<span><span class="co">##  [79] gtools_3.9.4                poweRlaw_0.70.6            </span></span>
<span><span class="co">##  [81] jpeg_0.1-10                 biomaRt_2.56.1             </span></span>
<span><span class="co">##  [83] RSQLite_2.3.1               weights_1.0.4              </span></span>
<span><span class="co">##  [85] R.methodsS3_1.8.2           utf8_1.2.3                 </span></span>
<span><span class="co">##  [87] tidyr_1.3.0                 generics_0.1.3             </span></span>
<span><span class="co">##  [89] data.table_1.14.8           prettyunits_1.2.0          </span></span>
<span><span class="co">##  [91] httr_1.4.7                  htmlwidgets_1.6.2          </span></span>
<span><span class="co">##  [93] S4Arrays_1.0.5              pkgconfig_2.0.3            </span></span>
<span><span class="co">##  [95] gtable_0.3.4                blob_1.2.4                 </span></span>
<span><span class="co">##  [97] htmltools_0.5.6.1           fftwtools_0.9-11           </span></span>
<span><span class="co">##  [99] scales_1.2.1                png_0.1-8                  </span></span>
<span><span class="co">## [101] knitr_1.44                  rstudioapi_0.15.0          </span></span>
<span><span class="co">## [103] tzdb_0.4.0                  reshape2_1.4.4             </span></span>
<span><span class="co">## [105] rjson_0.2.21                curl_5.1.0                 </span></span>
<span><span class="co">## [107] checkmate_2.2.0             nlme_3.1-162               </span></span>
<span><span class="co">## [109] nloptr_2.0.3                cachem_1.0.8               </span></span>
<span><span class="co">## [111] stringr_1.5.0               KernSmooth_2.23-20         </span></span>
<span><span class="co">## [113] parallel_4.3.1              genoPlotR_0.8.11           </span></span>
<span><span class="co">## [115] foreign_0.8-84              restfulr_0.0.15            </span></span>
<span><span class="co">## [117] desc_1.4.2                  pillar_1.9.0               </span></span>
<span><span class="co">## [119] grid_4.3.1                  vctrs_0.6.3                </span></span>
<span><span class="co">## [121] jomo_2.7-6                  dbplyr_2.3.3               </span></span>
<span><span class="co">## [123] xtable_1.8-4                cluster_2.1.4              </span></span>
<span><span class="co">## [125] htmlTable_2.4.1             evaluate_0.22              </span></span>
<span><span class="co">## [127] readr_2.1.4                 cli_3.6.1                  </span></span>
<span><span class="co">## [129] locfit_1.5-9.8              compiler_4.3.1             </span></span>
<span><span class="co">## [131] Rsamtools_2.16.0            rlang_1.1.1                </span></span>
<span><span class="co">## [133] crayon_1.5.2                plyr_1.8.8                 </span></span>
<span><span class="co">## [135] fs_1.6.3                    stringi_1.7.12             </span></span>
<span><span class="co">## [137] BiocParallel_1.34.2         munsell_0.5.0              </span></span>
<span><span class="co">## [139] tiff_0.1-11                 glmnet_4.1-7               </span></span>
<span><span class="co">## [141] Matrix_1.5-3                hms_1.1.3                  </span></span>
<span><span class="co">## [143] bit64_4.0.5                 KEGGREST_1.40.0            </span></span>
<span><span class="co">## [145] SummarizedExperiment_1.30.2 broom_1.0.5                </span></span>
<span><span class="co">## [147] memoise_2.0.1               bslib_0.5.1                </span></span>
<span><span class="co">## [149] bit_4.0.5</span></span></code></pre>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Aleksandra Bliznina, Michael Mansfield, Charles Plessy.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
