<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Inversions • OikScrambling</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Inversions">
<meta property="og:description" content="OikScrambling">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">OikScrambling</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">4.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/OikScrambling.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/CNEs.html">Conserved non-coding elements</a>
    </li>
    <li>
      <a href="../articles/ChromosomePlots.html">Chromosome-scale feature density plots</a>
    </li>
    <li>
      <a href="../articles/CoalescingCutoffs.html">Modifying coalescent cutoffs</a>
    </li>
    <li>
      <a href="../articles/ColinearityInterruptors.html">Regions interrupting colinearity</a>
    </li>
    <li>
      <a href="../articles/DistanceMeasures.html">Distance Measures</a>
    </li>
    <li>
      <a href="../articles/GenomicFeatures.html">Genomic features of the breakpoints</a>
    </li>
    <li>
      <a href="../articles/Inversions.html">Inversions</a>
    </li>
    <li>
      <a href="../articles/LoadGenomicBreaks.html">Load genomic break data</a>
    </li>
    <li>
      <a href="../articles/OrthogroupSyntenies.html">Orthogroup syntenies</a>
    </li>
    <li>
      <a href="../articles/OxfordPlots.html">Whole-genome dot-plots</a>
    </li>
    <li>
      <a href="../articles/PAC3.html">Region of interest: PAC3</a>
    </li>
    <li>
      <a href="../articles/PWM.html">Motif search and analysis</a>
    </li>
    <li>
      <a href="../articles/ParallelPlots.html">Whole-chromosome parallel plots</a>
    </li>
    <li>
      <a href="../articles/RegionWidths.html">Width of the aligned and mapped regions.</a>
    </li>
    <li>
      <a href="../articles/RegionsOfInterest.html">Regions of interest</a>
    </li>
    <li>
      <a href="../articles/RepeatRegions.html">Repeat regions</a>
    </li>
    <li>
      <a href="../articles/SegmentMovements.html">Segment movement distances</a>
    </li>
    <li>
      <a href="../articles/TFBS.html">Transcription factor binding sites</a>
    </li>
    <li>
      <a href="../articles/TandemSiteDuplications.html">Tandem site duplications</a>
    </li>
    <li>
      <a href="../articles/Translocations.html">Translocations</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/oist/LuscombeU_OikScrambling/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="Inversions_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Inversions</h1>
                        <h4 data-toc-skip class="author">Charles Plessy</h4>
            
            <h4 data-toc-skip class="date">25 July, 2022</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/oist/LuscombeU_OikScrambling/blob/HEAD/vignettes/Inversions.Rmd" class="external-link"><code>vignettes/Inversions.Rmd</code></a></small>
      <div class="hidden name"><code>Inversions.Rmd</code></div>

    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">knitr</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/knitr/man/opts_chunk.html" class="external-link">opts_chunk</a></span><span class="op">$</span><span class="fu">set</span><span class="op">(</span>cache <span class="op">=</span> <span class="cn">TRUE</span>, cache.lazy <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></code></pre></div>
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>After coalescing colinear alignments, removing translocations of repeat-containing sequences and re-coalescing, colinearity is still broken hundreds of time.</p>
<p>Here we explore the role of inversions in scrambling <em>Oikopleura</em> genomes.</p>
</div>
<div class="section level2">
<h2 id="load-r-pacakges-and-data">Load R pacakges and data<a class="anchor" aria-label="anchor" href="#load-r-pacakges-and-data"></a>
</h2>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/oist/LuscombeU_OikScrambling" class="external-link">'OikScrambling'</a></span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/load.html" class="external-link">load</a></span><span class="op">(</span><span class="st">"BreakPoints.Rdata"</span><span class="op">)</span></code></pre></div>
<p>See <code><a href="../articles/LoadGenomicBreaks.html">vignette("LoadGenomicBreaks", package = "OikScrambling")</a></code> for how the different GBreaks objects are prepared.</p>
</div>
<div class="section level2">
<h2 id="trivial-inversions">Trivial inversions<a class="anchor" aria-label="anchor" href="#trivial-inversions"></a>
</h2>
<div class="section level3">
<h3 id="documentation">Documentation<a class="anchor" aria-label="anchor" href="#documentation"></a>
</h3>
<p>Details can be found in <code><a href="https://oist.github.io/GenomicBreaks/articles/GenomicBreaks.html" class="external-link">vignette("GenomicBreaks", package = "GenomicBreaks")</a></code>, in <code><a href="https://oist.github.io/GenomicBreaks/articles/StructuralVariants.html" class="external-link">vignette("StructuralVariants", package = "GenomicBreaks")</a></code>, and <code><a href="https://oist.github.io/GenomicBreaks/reference/flagInversions.html" class="external-link">?GenomicBreaks::flagInversions</a></code></p>
</div>
<div class="section level3">
<h3 id="number-of-trivial-inversions">Number of trivial inversions<a class="anchor" aria-label="anchor" href="#number-of-trivial-inversions"></a>
</h3>
<p>More inversions are found after coalescing colinear blocks because of situations where <code>+ - +</code> was <code>+ - - +</code> before collapsing.</p>
<p>The number of detected inversions is further increased after double-coalescing, but not much.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="op">(</span><span class="va">invs_summary</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>
  align  <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">gbs</span>,  <span class="kw">function</span><span class="op">(</span><span class="va">gb</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/flagInversions.html" class="external-link">flagInversions</a></span><span class="op">(</span><span class="va">gb</span><span class="op">)</span><span class="op">$</span><span class="va">inv</span><span class="op">)</span><span class="op">)</span>,
  mapped <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">coa</span>,  <span class="kw">function</span><span class="op">(</span><span class="va">gb</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/flagInversions.html" class="external-link">flagInversions</a></span><span class="op">(</span><span class="va">gb</span><span class="op">)</span><span class="op">$</span><span class="va">inv</span><span class="op">)</span><span class="op">)</span>,
  map2   <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">coa2</span>, <span class="kw">function</span><span class="op">(</span><span class="va">gb</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/flagInversions.html" class="external-link">flagInversions</a></span><span class="op">(</span><span class="va">gb</span><span class="op">)</span><span class="op">$</span><span class="va">inv</span><span class="op">)</span><span class="op">)</span>
<span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">##         align mapped map2</span>
<span class="co">## Oki_Osa   614    670  704</span>
<span class="co">## Oki_Bar   604    675  705</span>
<span class="co">## Oki_Kum    89    105  122</span>
<span class="co">## Oki_Aom   608    665  695</span>
<span class="co">## Oki_Nor   541    591  611</span>
<span class="co">## Osa_Oki   613    671  706</span>
<span class="co">## Osa_Bar   241    321  330</span>
<span class="co">## Osa_Kum   620    678  708</span>
<span class="co">## Osa_Aom    56     73   79</span>
<span class="co">## Osa_Nor   212    275  279</span>
<span class="co">## Bar_Oki   597    667  695</span>
<span class="co">## Bar_Osa   237    319  327</span>
<span class="co">## Bar_Kum   608    686  711</span>
<span class="co">## Bar_Aom   237    319  335</span>
<span class="co">## Bar_Nor    19     24   27</span>
<span class="co">## Ply_Ros    42     51   55</span>
<span class="co">## Ply_Rob    58     71   76</span>
<span class="co">## Ply_Sav    16     92   92</span>
<span class="co">## Ply_Oki     0      0    0</span>
<span class="co">## Rob_Ros    50     65   71</span>
<span class="co">## Rob_Ply    58     70   77</span>
<span class="co">## Rob_Sav    14     92   92</span>
<span class="co">## Rob_Oki     0      0    0</span>
<span class="co">## Dme_Dbu   697   1003 1003</span>
<span class="co">## Dme_Dsu    20     44   45</span>
<span class="co">## Dme_Dya    13     21   23</span>
<span class="co">## Dme_Dma    12     16   16</span></code></pre>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">invs_summary</span>, \<span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/base/tapply.html" class="external-link">tapply</a></span><span class="op">(</span><span class="va">x</span>, <span class="fu"><a href="https://rdrr.io/r/base/row.names.html" class="external-link">row.names</a></span><span class="op">(</span><span class="va">invs_summary</span><span class="op">)</span> |&gt; <span class="fu">OikScrambling</span><span class="fu">:::</span><span class="fu"><a href="../reference/compDistance.html">compDistance</a></span><span class="op">(</span><span class="op">)</span>, <span class="va">mean</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">##                   align     mapped       map2</span>
<span class="co">## Dme_Dbu       697.00000 1003.00000 1003.00000</span>
<span class="co">## Dme_Dma        12.00000   16.00000   16.00000</span>
<span class="co">## Dme_Dsu        20.00000   44.00000   45.00000</span>
<span class="co">## Dme_Dya        13.00000   21.00000   23.00000</span>
<span class="co">## In same pop    54.66667   67.33333   76.00000</span>
<span class="co">## Int – Int      42.00000   51.00000   55.00000</span>
<span class="co">## Int – Rob      55.33333   68.66667   74.66667</span>
<span class="co">## Int/Rob – Oki   0.00000    0.00000    0.00000</span>
<span class="co">## Int/Rob – Sav  15.00000   92.00000   92.00000</span>
<span class="co">## North – North 231.75000  308.50000  317.75000</span>
<span class="co">## Oki – North   600.62500  662.87500  691.87500</span></code></pre>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">invs_summary</span>, \<span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/base/tapply.html" class="external-link">tapply</a></span><span class="op">(</span><span class="va">x</span>, <span class="fu"><a href="https://rdrr.io/r/base/row.names.html" class="external-link">row.names</a></span><span class="op">(</span><span class="va">invs_summary</span><span class="op">)</span> |&gt; <span class="fu">OikScrambling</span><span class="fu">:::</span><span class="fu"><a href="../reference/compDistance.html">compDistance</a></span><span class="op">(</span><span class="op">)</span>, <span class="va">median</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">##               align mapped   map2</span>
<span class="co">## Dme_Dbu         697 1003.0 1003.0</span>
<span class="co">## Dme_Dma          12   16.0   16.0</span>
<span class="co">## Dme_Dsu          20   44.0   45.0</span>
<span class="co">## Dme_Dya          13   21.0   23.0</span>
<span class="co">## In same pop      56   73.0   79.0</span>
<span class="co">## Int – Int        42   51.0   55.0</span>
<span class="co">## Int – Rob        58   70.0   76.0</span>
<span class="co">## Int/Rob – Oki     0    0.0    0.0</span>
<span class="co">## Int/Rob – Sav    15   92.0   92.0</span>
<span class="co">## North – North   237  319.0  328.5</span>
<span class="co">## Oki – North     608  670.5  704.5</span></code></pre>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">invs_summary</span>, \<span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/base/tapply.html" class="external-link">tapply</a></span><span class="op">(</span><span class="va">x</span>, <span class="fu"><a href="https://rdrr.io/r/base/row.names.html" class="external-link">row.names</a></span><span class="op">(</span><span class="va">invs_summary</span><span class="op">)</span> |&gt; <span class="fu">OikScrambling</span><span class="fu">:::</span><span class="fu"><a href="../reference/compDistance.html">compDistance</a></span><span class="op">(</span><span class="op">)</span>, <span class="va">sd</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">##                   align   mapped     map2</span>
<span class="co">## Dme_Dbu              NA       NA       NA</span>
<span class="co">## Dme_Dma              NA       NA       NA</span>
<span class="co">## Dme_Dsu              NA       NA       NA</span>
<span class="co">## Dme_Dya              NA       NA       NA</span>
<span class="co">## In same pop   35.019042 40.79624 47.57100</span>
<span class="co">## Int – Int            NA       NA       NA</span>
<span class="co">## Int – Rob      4.618802  3.21455  3.21455</span>
<span class="co">## Int/Rob – Oki  0.000000  0.00000  0.00000</span>
<span class="co">## Int/Rob – Sav  1.414214  0.00000  0.00000</span>
<span class="co">## North – North 13.301002 22.35322 26.04323</span>
<span class="co">## Oki – North   25.059572 29.79663 33.17675</span></code></pre>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">invs_summary</span><span class="op">$</span><span class="va">pairname</span>   <span class="op">&lt;-</span> <span class="fu">OikScrambling</span><span class="fu">:::</span><span class="fu"><a href="../reference/compDistance.html">compDistance</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">invs_summary</span><span class="op">)</span><span class="op">)</span>
<span class="va">invs_summary</span><span class="op">$</span><span class="va">pairname_s</span> <span class="op">&lt;-</span> <span class="fu">OikScrambling</span><span class="fu">:::</span><span class="fu"><a href="../reference/compDistance.html">compDistance</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">invs_summary</span><span class="op">)</span>, short <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="va">invs_summary</span><span class="op">$</span><span class="va">genus</span>      <span class="op">&lt;-</span> <span class="fu">OikScrambling</span><span class="fu">:::</span><span class="fu"><a href="../reference/compGenus.html">compGenus</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">invs_summary</span><span class="op">)</span><span class="op">)</span>
<span class="va">invs_summary</span><span class="op">$</span><span class="va">class</span>      <span class="op">&lt;-</span> <span class="fu">OikScrambling</span><span class="fu">:::</span><span class="fu"><a href="../reference/compDistClass.html">compDistClass</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">invs_summary</span><span class="op">)</span><span class="op">)</span>
<span class="va">invs_summary</span><span class="op">$</span><span class="va">target</span>     <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">sub</a></span><span class="op">(</span><span class="st">"_.*"</span>, <span class="st">""</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">invs_summary</span><span class="op">)</span><span class="op">)</span>
<span class="va">invs_summary</span> <span class="op">&lt;-</span> <span class="va">invs_summary</span><span class="op">[</span><span class="va">invs_summary</span><span class="op">$</span><span class="va">class</span> <span class="op">!=</span> <span class="st">"Int/Rob – Oki"</span>,<span class="op">]</span>

<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">invs_summary</span> |&gt; <span class="fu">tidyr</span><span class="fu">::</span><span class="fu"><a href="https://tidyr.tidyverse.org/reference/pivot_longer.html" class="external-link">pivot_longer</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"align"</span>, <span class="st">"mapped"</span>, <span class="st">"map2"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span><span class="va">value</span>, <span class="va">pairname_s</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>col <span class="op">=</span> <span class="va">target</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span> <span class="op">(</span><span class="st">"Number of inversions found after different post-processings"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html" class="external-link">facet_wrap</a></span><span class="op">(</span><span class="op">~</span><span class="va">name</span>, ncol<span class="op">=</span> <span class="fl">1</span><span class="op">)</span></code></pre></div>
<p><img src="Inversions_files/figure-html/detect_trivial_invertions-1.png" width="768"></p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">invs_summary</span> |&gt; <span class="fu">tidyr</span><span class="fu">::</span><span class="fu"><a href="https://tidyr.tidyverse.org/reference/pivot_longer.html" class="external-link">pivot_longer</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"mapped"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span><span class="va">value</span>, <span class="va">genus</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>col <span class="op">=</span> <span class="va">class</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span> <span class="op">(</span><span class="st">"Number of inversions found in mapped regions"</span><span class="op">)</span> </code></pre></div>
<p><img src="Inversions_files/figure-html/number_of_inversions-1.svg" width="672"></p>
</div>
<div class="section level3">
<h3 id="motif-search">Motif search<a class="anchor" aria-label="anchor" href="#motif-search"></a>
</h3>
<p>Enrichment for motifs is being searched in <code><a href="../articles/PWM.html">vignette("PWM", package = "OikScrambling")</a></code>.</p>
</div>
<div class="section level3">
<h3 id="breakpoint-regions-flanking-inversions-in-the-same-population-alignments-">Breakpoint regions flanking inversions in the same-population alignments.<a class="anchor" aria-label="anchor" href="#breakpoint-regions-flanking-inversions-in-the-same-population-alignments-"></a>
</h3>
<div class="section level4">
<h4 id="large-inversions">Large inversions<a class="anchor" aria-label="anchor" href="#large-inversions"></a>
</h4>
<p>The comparison of <code>contig_90_1</code> to Okinawa’s chromosome 1 shows a large inversion. Note that <code>contig_90_1</code> appears to be assembled in reverse orientation compared with the <em>short to long</em> arm orientation that we chose in our chromosomal assemblies.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">coa2</span><span class="op">$</span><span class="va">Oki_Kum</span> |&gt; <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/swap.html" class="external-link">swap</a></span><span class="op">(</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/reverse.html" class="external-link">reverse</a></span> <span class="op">(</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/swap.html" class="external-link">swap</a></span> <span class="op">(</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/plotApairOfChrs.html" class="external-link">plotApairOfChrs</a></span><span class="op">(</span>chrQ <span class="op">=</span> <span class="st">"contig_90_1"</span><span class="op">)</span></code></pre></div>
<p><img src="Inversions_files/figure-html/Kum_chr1_large_inversion-1.png" width="700"></p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">coa2</span><span class="op">$</span><span class="va">Oki_Kum</span> |&gt; <span class="fu">plyranges</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/plyranges/man/tidyverse-reexports.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">seqnames</span> <span class="op">==</span> <span class="st">"chr1"</span>, <span class="fu">seqnames</span><span class="op">(</span><span class="va">query</span><span class="op">)</span> <span class="op">==</span> <span class="st">"contig_90_1"</span>, <span class="va">width</span> <span class="op">&gt;</span> <span class="fl">1e3</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/coalesce_contigs.html" class="external-link">coalesce_contigs</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## GBreaks object with 15 ranges and 2 metadata columns:</span>
<span class="co">##        seqnames          ranges strand |                       query     score</span>
<span class="co">##           &lt;Rle&gt;       &lt;IRanges&gt;  &lt;Rle&gt; |                   &lt;GRanges&gt; &lt;integer&gt;</span>
<span class="co">##    [1]     chr1 5197932-5218400      + | contig_90_1:2733373-2753783     20469</span>
<span class="co">##    [2]     chr1 5225466-5354754      - | contig_90_1:2610517-2732825    129289</span>
<span class="co">##    [3]     chr1 5355072-5521877      + | contig_90_1:1710995-1839819    166806</span>
<span class="co">##    [4]     chr1 5522689-5557995      - | contig_90_1:1840801-1875801     35307</span>
<span class="co">##    [5]     chr1 5557996-5663066      + | contig_90_1:1875862-1969463    105071</span>
<span class="co">##    ...      ...             ...    ... .                         ...       ...</span>
<span class="co">##   [11]     chr1 6163602-6171132      - | contig_90_1:2412844-2420381      7531</span>
<span class="co">##   [12]     chr1 6175547-6297768      + | contig_90_1:2420606-2511021    122222</span>
<span class="co">##   [13]     chr1 6297954-6308357      - | contig_90_1:2517620-2523307     10404</span>
<span class="co">##   [14]     chr1 6311181-6375852      + | contig_90_1:2532049-2610516     64672</span>
<span class="co">##   [15]     chr1 6378758-8086428      - |      contig_90_1:27-1710992   1707671</span>
<span class="co">##   -------</span>
<span class="co">##   seqinfo: 19 sequences from OKI2018.I69 genome</span></code></pre>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">coa2</span><span class="op">$</span><span class="va">Oki_Kum</span> |&gt; <span class="fu">plyranges</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/plyranges/man/tidyverse-reexports.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">seqnames</span> <span class="op">==</span> <span class="st">"chr1"</span>, <span class="fu">seqnames</span><span class="op">(</span><span class="va">query</span><span class="op">)</span> <span class="op">==</span> <span class="st">"contig_90_1"</span>, <span class="va">width</span> <span class="op">&gt;</span> <span class="fl">1e3</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/inter-range-methods.html" class="external-link">range</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## GBreaks object with 1 range and 1 metadata column:</span>
<span class="co">##       seqnames          ranges strand |                  query</span>
<span class="co">##          &lt;Rle&gt;       &lt;IRanges&gt;  &lt;Rle&gt; |              &lt;GRanges&gt;</span>
<span class="co">##   [1]     chr1 5197932-8086428      * | contig_90_1:27-2753783</span>
<span class="co">##   -------</span>
<span class="co">##   seqinfo: 19 sequences from OKI2018.I69 genome</span></code></pre>
</div>
<div class="section level4">
<h4 id="small-inversions-">Small inversions.<a class="anchor" aria-label="anchor" href="#small-inversions-"></a>
</h4>
<p>Check <code><a href="../articles/ColinearityInterruptors.html">vignette("ColinearityInterruptors", package = "OikScrambling")</a></code>.</p>
</div>
</div>
<div class="section level3">
<h3 id="breakpoint-regions-flanking-large-inversions-in-the-northnorth-alignments-">Breakpoint regions flanking large inversions in the North–North alignments.<a class="anchor" aria-label="anchor" href="#breakpoint-regions-flanking-large-inversions-in-the-northnorth-alignments-"></a>
</h3>
<p>Let’s try to isolate some inversions that are easy to spot on a whole-chromosome alignment.</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/plotApairOfChrs.html" class="external-link">plotApairOfChrs</a></span><span class="op">(</span><span class="va">coa2</span><span class="op">$</span><span class="va">Osa_Bar</span>, <span class="st">"PAR"</span><span class="op">)</span></code></pre></div>
<p><img src="Inversions_files/figure-html/unnamed-chunk-1-1.png" width="700"></p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="op">(</span><span class="va">ROI1</span> <span class="op">&lt;-</span> <span class="va">coa2</span><span class="op">$</span><span class="va">Osa_Bar</span> |&gt; <span class="fu">plyranges</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/plyranges/man/tidyverse-reexports.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">seqnames</span> <span class="op">==</span> <span class="st">"PAR"</span>, <span class="va">start</span> <span class="op">&gt;</span> <span class="fl">9e6</span>, <span class="va">end</span> <span class="op">&lt;</span> <span class="fl">14e6</span>, <span class="fu"><a href="https://rdrr.io/r/stats/start.html" class="external-link">start</a></span><span class="op">(</span><span class="va">query</span><span class="op">)</span> <span class="op">&gt;</span> <span class="fl">7e6</span>, <span class="fu"><a href="https://rdrr.io/r/stats/start.html" class="external-link">end</a></span><span class="op">(</span><span class="va">query</span><span class="op">)</span> <span class="op">&lt;</span> <span class="fl">12e6</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## GBreaks object with 35 ranges and 6 metadata columns:</span>
<span class="co">##        seqnames            ranges strand |                 query      Arm</span>
<span class="co">##           &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; |             &lt;GRanges&gt; &lt;factor&gt;</span>
<span class="co">##    [1]      PAR   9112106-9113024      + |   PAR:7914713-7915633     long</span>
<span class="co">##    [2]      PAR   9122674-9169015      + |   PAR:7647348-7694603     long</span>
<span class="co">##    [3]      PAR   9169027-9383526      + |   PAR:7700204-7911308     long</span>
<span class="co">##    [4]      PAR   9393845-9625372      - |   PAR:8648572-8868624     long</span>
<span class="co">##    [5]      PAR   9625425-9626193      + |   PAR:8647713-8648486     long</span>
<span class="co">##    ...      ...               ...    ... .                   ...      ...</span>
<span class="co">##   [31]      PAR 12882838-12884857      + | PAR:11615009-11617029     long</span>
<span class="co">##   [32]      PAR 12884875-13019548      - | PAR:10442170-10570866     long</span>
<span class="co">##   [33]      PAR 13027569-13077236      - | PAR:10392248-10442013     long</span>
<span class="co">##   [34]      PAR 13077254-13079729      + | PAR:11592137-11594612     long</span>
<span class="co">##   [35]      PAR 13079758-13079924      + | PAR:10378144-10378306     long</span>
<span class="co">##                                 rep   repOvlp            transcripts</span>
<span class="co">##                     &lt;CharacterList&gt; &lt;integer&gt;                  &lt;Rle&gt;</span>
<span class="co">##    [1]                         &lt;NA&gt;         0                   &lt;NA&gt;</span>
<span class="co">##    [2]                   rnd,tandem      1671 g9531.t1;g9532.t1;g9..</span>
<span class="co">##    [3]       rnd,unknown,tandem,...      4642 g9552.t1;g9553.t1;g9..</span>
<span class="co">##    [4] rnd,tandem,LowComplexity,...      5474 g9629.t1;g9630.t1;g9..</span>
<span class="co">##    [5]                         &lt;NA&gt;         0                   &lt;NA&gt;</span>
<span class="co">##    ...                          ...       ...                    ...</span>
<span class="co">##   [31]                          rnd      1167              g10776.t1</span>
<span class="co">##   [32] tandem,rnd,LowComplexity,...      3102 g10777.t1;g10779.t1;..</span>
<span class="co">##   [33]           unknown,rnd,tandem      1637 g10840.t1;g10841.t1;..</span>
<span class="co">##   [34]                          rnd         0                   &lt;NA&gt;</span>
<span class="co">##   [35]                         &lt;NA&gt;         0                   &lt;NA&gt;</span>
<span class="co">##               flag</span>
<span class="co">##        &lt;character&gt;</span>
<span class="co">##    [1]        &lt;NA&gt;</span>
<span class="co">##    [2]        &lt;NA&gt;</span>
<span class="co">##    [3]        &lt;NA&gt;</span>
<span class="co">##    [4]         Inv</span>
<span class="co">##    [5]        &lt;NA&gt;</span>
<span class="co">##    ...         ...</span>
<span class="co">##   [31]        &lt;NA&gt;</span>
<span class="co">##   [32]        &lt;NA&gt;</span>
<span class="co">##   [33]        &lt;NA&gt;</span>
<span class="co">##   [34]        &lt;NA&gt;</span>
<span class="co">##   [35]        &lt;NA&gt;</span>
<span class="co">##   -------</span>
<span class="co">##   seqinfo: 483 sequences from OSKA2016v1.9 genome</span></code></pre>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/plotApairOfChrs.html" class="external-link">plotApairOfChrs</a></span><span class="op">(</span><span class="va">ROI1</span><span class="op">)</span></code></pre></div>
<p><img src="Inversions_files/figure-html/unnamed-chunk-1-2.png" width="700"></p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># The contours of the main 4 blocks can be coalesced by applying a size threshold.</span>
<span class="va">ROI1</span> |&gt; <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/coalesce_contigs.html" class="external-link">coalesce_contigs</a></span><span class="op">(</span>min <span class="op">=</span> <span class="fl">1e4</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## GBreaks object with 4 ranges and 2 metadata columns:</span>
<span class="co">##       seqnames            ranges strand |                 query     score</span>
<span class="co">##          &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; |             &lt;GRanges&gt; &lt;integer&gt;</span>
<span class="co">##   [1]      PAR   9122674-9383526      + |   PAR:7647348-7911308    260853</span>
<span class="co">##   [2]      PAR  9393845-10352419      - |   PAR:7917928-8868624    958575</span>
<span class="co">##   [3]      PAR 10363392-11843885      + |  PAR:8872985-10375361   1480494</span>
<span class="co">##   [4]      PAR 11852584-13077236      - | PAR:10392248-11592118   1224653</span>
<span class="co">##   -------</span>
<span class="co">##   seqinfo: 483 sequences from OSKA2016v1.9 genome</span></code></pre>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># The gaps flanking the first inverted region are quite large!</span>
<span class="op">(</span><span class="va">ROI1_gaps</span> <span class="op">&lt;-</span> <span class="va">ROI1</span> |&gt; <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/coalesce_contigs.html" class="external-link">coalesce_contigs</a></span><span class="op">(</span>min <span class="op">=</span> <span class="fl">1e4</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/cleanGaps.html" class="external-link">cleanGaps</a></span><span class="op">(</span><span class="op">)</span> |&gt; <span class="fu">plyranges</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/plyranges/man/tidyverse-reexports.html" class="external-link">mutate</a></span><span class="op">(</span>w <span class="op">=</span> <span class="va">width</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## GRanges object with 3 ranges and 1 metadata column:</span>
<span class="co">##       seqnames            ranges strand |         w</span>
<span class="co">##          &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt;</span>
<span class="co">##   [1]      PAR   9383527-9393844      * |     10318</span>
<span class="co">##   [2]      PAR 10352420-10363391      * |     10972</span>
<span class="co">##   [3]      PAR 11843886-11852583      * |      8698</span>
<span class="co">##   -------</span>
<span class="co">##   seqinfo: 483 sequences from OSKA2016v1.9 genome</span></code></pre>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># The left-side gap contains no alignment at all even before discarding the short ones.</span>
<span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/findOverlaps-methods.html" class="external-link">subsetByOverlaps</a></span><span class="op">(</span><span class="va">gbs</span><span class="op">$</span><span class="va">Osa_Bar</span>, <span class="va">ROI1_gaps</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## GBreaks object with 4 ranges and 8 metadata columns:</span>
<span class="co">##       seqnames            ranges strand |     score                 query</span>
<span class="co">##          &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; | &lt;numeric&gt;             &lt;GRanges&gt;</span>
<span class="co">##   [1]      PAR 10355613-10355774      - |       500     XSR:671520-671674</span>
<span class="co">##   [2]      PAR 10355951-10356573      - |      1610     XSR:670597-671182</span>
<span class="co">##   [3]      PAR 10356752-10361833      - |     13551     XSR:665505-670596</span>
<span class="co">##   [4]      PAR 11847299-11849269      + |      5701 PAR:10385920-10388171</span>
<span class="co">##            Arm             rep   repOvlp transcripts        flag    nonCoa</span>
<span class="co">##       &lt;factor&gt; &lt;CharacterList&gt; &lt;integer&gt;       &lt;Rle&gt; &lt;character&gt; &lt;logical&gt;</span>
<span class="co">##   [1]     long            &lt;NA&gt;         0        &lt;NA&gt;         Col     FALSE</span>
<span class="co">##   [2]     long            &lt;NA&gt;         0        &lt;NA&gt;         Col     FALSE</span>
<span class="co">##   [3]     long            &lt;NA&gt;         0        &lt;NA&gt;        &lt;NA&gt;     FALSE</span>
<span class="co">##   [4]     long            &lt;NA&gt;         0   g10434.t1        &lt;NA&gt;      TRUE</span>
<span class="co">##   -------</span>
<span class="co">##   seqinfo: 483 sequences from OSKA2016v1.9 genome</span></code></pre>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># The same region aligns well to Aomori, suggesting that it is not missassembled</span>
<span class="va">coa2</span><span class="op">$</span><span class="va">Osa_Aom</span> |&gt; <span class="fu">plyranges</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/plyranges/man/tidyverse-reexports.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">seqnames</span> <span class="op">==</span> <span class="st">"PAR"</span>, <span class="va">start</span> <span class="op">&gt;</span> <span class="fl">9e6</span>, <span class="va">end</span> <span class="op">&lt;</span> <span class="fl">14e6</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/plotApairOfChrs.html" class="external-link">plotApairOfChrs</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<p><img src="Inversions_files/figure-html/unnamed-chunk-1-3.png" width="700"></p>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># The Osa–Bar unaligned regions appear to align well to Aom, but the first one is split</span>
<span class="co"># and the second one contains a small inversion. Are there a hotspots ?</span>
<span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/findOverlaps-methods.html" class="external-link">subsetByOverlaps</a></span><span class="op">(</span><span class="va">gbs</span><span class="op">$</span><span class="va">Osa_Aom</span>, <span class="va">ROI1_gaps</span>, type <span class="op">=</span> <span class="st">"any"</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/plotApairOfChrs.html" class="external-link">plotApairOfChrs</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<p><img src="Inversions_files/figure-html/unnamed-chunk-1-4.png" width="700"></p>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/findOverlaps-methods.html" class="external-link">subsetByOverlaps</a></span><span class="op">(</span><span class="va">gbs</span><span class="op">$</span><span class="va">Osa_Aom</span>, <span class="va">ROI1_gaps</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>, type <span class="op">=</span> <span class="st">"within"</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/plotApairOfChrs.html" class="external-link">plotApairOfChrs</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<p><img src="Inversions_files/figure-html/unnamed-chunk-1-5.png" width="700"></p>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/findOverlaps-methods.html" class="external-link">subsetByOverlaps</a></span><span class="op">(</span><span class="va">gbs</span><span class="op">$</span><span class="va">Osa_Aom</span>, <span class="va">ROI1_gaps</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span>, type <span class="op">=</span> <span class="st">"within"</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/plotApairOfChrs.html" class="external-link">plotApairOfChrs</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<p><img src="Inversions_files/figure-html/unnamed-chunk-1-6.png" width="700"></p>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/findOverlaps-methods.html" class="external-link">subsetByOverlaps</a></span><span class="op">(</span><span class="va">gbs</span><span class="op">$</span><span class="va">Osa_Aom</span>, <span class="va">ROI1_gaps</span><span class="op">[</span><span class="fl">3</span><span class="op">]</span>, type <span class="op">=</span> <span class="st">"within"</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/plotApairOfChrs.html" class="external-link">plotApairOfChrs</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<p><img src="Inversions_files/figure-html/unnamed-chunk-1-7.png" width="700"></p>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># In the alignment to Okinawa, it is scrambled.</span>
<span class="va">coa2</span><span class="op">$</span><span class="va">Osa_Oki</span> |&gt; <span class="fu">plyranges</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/plyranges/man/tidyverse-reexports.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">seqnames</span> <span class="op">==</span> <span class="st">"PAR"</span>, <span class="va">start</span> <span class="op">&gt;</span> <span class="fl">9e6</span>, <span class="va">end</span> <span class="op">&lt;</span> <span class="fl">14e6</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/plotApairOfChrs.html" class="external-link">plotApairOfChrs</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<p><img src="Inversions_files/figure-html/unnamed-chunk-1-8.png" width="700"></p>
</div>
</div>
<div class="section level2">
<h2 id="export-inversions-to-zenbu">Export inversions to ZENBU<a class="anchor" aria-label="anchor" href="#export-inversions-to-zenbu"></a>
</h2>
<div class="section level3">
<h3 id="okinawa">Okinawa<a class="anchor" aria-label="anchor" href="#okinawa"></a>
</h3>
<pre><code><span class="va">invs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">coa</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span>, <span class="va">inv2UCSCData</span><span class="op">)</span> |&gt; <span class="fu">SimpleList</span><span class="op">(</span><span class="op">)</span>
<span class="fu">rtracklayer</span><span class="fu">::</span><span class="fu">export</span><span class="op">(</span><span class="va">invs</span><span class="op">$</span><span class="va">Oki_Osa</span>, <span class="st">"inversions_v4.0.0_Oki_Osa.bed"</span><span class="op">)</span>
<span class="fu">rtracklayer</span><span class="fu">::</span><span class="fu">export</span><span class="op">(</span><span class="va">invs</span><span class="op">$</span><span class="va">Oki_Bar</span>, <span class="st">"inversions_v4.0.0_Oki_Bar.bed"</span><span class="op">)</span>
<span class="fu">rtracklayer</span><span class="fu">::</span><span class="fu">export</span><span class="op">(</span><span class="va">invs</span><span class="op">$</span><span class="va">Oki_Kum</span>, <span class="st">"inversions_v4.0.0_Oki_Kum.bed"</span><span class="op">)</span>
<span class="fu">rtracklayer</span><span class="fu">::</span><span class="fu">export</span><span class="op">(</span><span class="va">invs</span><span class="op">$</span><span class="va">Oki_Aom</span>, <span class="st">"inversions_v4.0.0_Oki_Aom.bed"</span><span class="op">)</span>
<span class="fu">rtracklayer</span><span class="fu">::</span><span class="fu">export</span><span class="op">(</span><span class="va">invs</span><span class="op">$</span><span class="va">Oki_Nor</span>, <span class="st">"inversions_v4.0.0_Oki_Nor.bed"</span><span class="op">)</span></code></pre>
</div>
<div class="section level3">
<h3 id="ōsaka">Ōsaka<a class="anchor" aria-label="anchor" href="#%C5%8Dsaka"></a>
</h3>
<pre><code><span class="va">invsOska</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">coa</span><span class="op">[</span><span class="fl">6</span><span class="op">:</span><span class="fl">10</span><span class="op">]</span>, <span class="va">inv2UCSCData</span><span class="op">)</span> |&gt; <span class="fu">SimpleList</span><span class="op">(</span><span class="op">)</span>
<span class="fu">rtracklayer</span><span class="fu">::</span><span class="fu">export</span><span class="op">(</span><span class="va">invsOska</span><span class="op">$</span><span class="va">Osa_Oki</span>, <span class="st">"inversions_v4.0.0_Osa_Oki.bed"</span><span class="op">)</span>
<span class="fu">rtracklayer</span><span class="fu">::</span><span class="fu">export</span><span class="op">(</span><span class="va">invsOska</span><span class="op">$</span><span class="va">Osa_Bar</span>, <span class="st">"inversions_v4.0.0_Osa_Bar.bed"</span><span class="op">)</span>
<span class="fu">rtracklayer</span><span class="fu">::</span><span class="fu">export</span><span class="op">(</span><span class="va">invsOska</span><span class="op">$</span><span class="va">Osa_Kum</span>, <span class="st">"inversions_v4.0.0_Osa_Kum.bed"</span><span class="op">)</span>
<span class="fu">rtracklayer</span><span class="fu">::</span><span class="fu">export</span><span class="op">(</span><span class="va">invsOska</span><span class="op">$</span><span class="va">Osa_Aom</span>, <span class="st">"inversions_v4.0.0_Osa_Aom.bed"</span><span class="op">)</span>
<span class="fu">rtracklayer</span><span class="fu">::</span><span class="fu">export</span><span class="op">(</span><span class="va">invsOska</span><span class="op">$</span><span class="va">Osa_Nor</span>, <span class="st">"inversions_v4.0.0_Osa_Nor.bed"</span><span class="op">)</span></code></pre>
</div>
</div>
<div class="section level2">
<h2 id="breakpoint-regions-flanking-inversions">Breakpoint regions flanking inversions<a class="anchor" aria-label="anchor" href="#breakpoint-regions-flanking-inversions"></a>
</h2>
<p>Inversions break synteny. Therefore, the unaligned regions flanking them are classified as <em>breakpoint regions</em>. This means that they are not paired with the other genome. Here, we will do the pairing by 1) flipping the inversions and re-coalescing the alignments, and 2) isolating the new bridge regions produced in that process.</p>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">flipInversions</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">gb</span><span class="op">)</span> <span class="op">{</span>
  <span class="va">Invs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/which.html" class="external-link">which</a></span><span class="op">(</span><span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/flagInversions.html" class="external-link">flagInversions</a></span><span class="op">(</span><span class="va">gb</span><span class="op">)</span><span class="op">$</span><span class="va">inv</span><span class="op">)</span> <span class="op">+</span> <span class="fl">1</span>
  <span class="fu">strand</span><span class="op">(</span><span class="va">gb</span><span class="op">)</span><span class="op">[</span><span class="va">Invs</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html" class="external-link">ifelse</a></span><span class="op">(</span><span class="fu">strand</span><span class="op">(</span><span class="va">gb</span><span class="op">)</span><span class="op">[</span><span class="va">Invs</span><span class="op">]</span> <span class="op">==</span> <span class="st">"+"</span>, <span class="st">"-"</span>, <span class="st">"+"</span><span class="op">)</span>
  <span class="va">gb</span>
<span class="op">}</span>

<span class="va">flippedInvs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">coa</span>, <span class="va">flipInversions</span><span class="op">)</span> |&gt; <span class="fu">SimpleList</span><span class="op">(</span><span class="op">)</span>

<span class="co"># See https://github.com/oist/GenomicBreaks/issues/20</span>
<span class="va">bridgeRegions_</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">gb</span><span class="op">)</span> <span class="op">{</span>
  <span class="kw">if</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/flagColinearAlignments.html" class="external-link">flagColinearAlignments</a></span><span class="op">(</span><span class="va">gb</span><span class="op">)</span><span class="op">$</span><span class="va">colinear</span><span class="op">)</span> <span class="op">==</span> <span class="fl">0</span><span class="op">)</span> <span class="kw"><a href="https://rdrr.io/r/base/function.html" class="external-link">return</a></span><span class="op">(</span><span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/GBreaks-class.html" class="external-link">GBreaks</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span>
  <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/bridgeRegions.html" class="external-link">bridgeRegions</a></span><span class="op">(</span><span class="va">gb</span><span class="op">)</span>
<span class="op">}</span>

<span class="va">flippedInvs_bri</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">flippedInvs</span>, <span class="va">bridgeRegions_</span><span class="op">)</span> |&gt; <span class="fu">SimpleList</span><span class="op">(</span><span class="op">)</span>

<span class="co"># Copied from the LoadGenomicBreaks vignette</span>
<span class="va">flagLongShort_</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">gr</span>, <span class="va">longShort</span><span class="op">)</span> <span class="op">{</span>
  <span class="va">genomeT</span> <span class="op">&lt;-</span> <span class="va">gr</span> |&gt; <span class="fu">genome</span><span class="op">(</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="op">)</span>
  <span class="kw">if</span><span class="op">(</span><span class="va">genomeT</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">longShort</span><span class="op">)</span><span class="op">)</span>
    <span class="va">gr</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/flagLongShort.html" class="external-link">flagLongShort</a></span><span class="op">(</span><span class="va">gr</span>, <span class="va">longShort</span><span class="op">[[</span><span class="va">genomeT</span><span class="op">]</span><span class="op">]</span><span class="op">)</span>
  <span class="kw">if</span><span class="op">(</span><span class="st">"GBreaks"</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/class.html" class="external-link">class</a></span><span class="op">(</span><span class="va">gr</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span>
      <span class="va">genomeQ</span> <span class="op">&lt;-</span> <span class="va">gr</span><span class="op">$</span><span class="va">query</span> |&gt; <span class="fu">genome</span><span class="op">(</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="op">)</span>
      <span class="kw">if</span><span class="op">(</span><span class="va">genomeQ</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">longShort</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span>
        <span class="co"># sort, flag, unsort, extract flag</span>
        <span class="va">grq</span> <span class="op">&lt;-</span> <span class="va">gr</span><span class="op">$</span><span class="va">query</span>
        <span class="va">grq</span><span class="op">$</span><span class="va">idx</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq_along</a></span><span class="op">(</span><span class="va">grq</span><span class="op">)</span>
        <span class="va">grq</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sort.html" class="external-link">sort</a></span><span class="op">(</span><span class="va">grq</span>, ignore.strand <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
        <span class="va">grq</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/flagLongShort.html" class="external-link">flagLongShort</a></span><span class="op">(</span><span class="va">grq</span>, <span class="va">longShort</span><span class="op">[[</span><span class="va">genomeQ</span><span class="op">]</span><span class="op">]</span><span class="op">)</span>
        <span class="va">gr</span><span class="op">$</span><span class="va">query</span><span class="op">$</span><span class="va">Arm</span> <span class="op">&lt;-</span> <span class="va">grq</span><span class="op">$</span><span class="va">Arm</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/order.html" class="external-link">order</a></span><span class="op">(</span><span class="va">grq</span><span class="op">$</span><span class="va">idx</span><span class="op">)</span><span class="op">]</span>
      <span class="op">}</span>
  <span class="op">}</span>
  <span class="va">gr</span>
<span class="op">}</span>

<span class="va">flippedInvs_bri</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">15</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu">BiocParallel</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocParallel/man/bplapply.html" class="external-link">bplapply</a></span><span class="op">(</span><span class="va">flippedInvs_bri</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">15</span><span class="op">]</span>, <span class="va">flagLongShort_</span>, <span class="va">longShort</span><span class="op">)</span>


<span class="co"># Copied from the RegionWidths vignette</span>
<span class="va">GBreaks2widthTibble</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">gb</span>, <span class="va">pair</span> <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span> <span class="op">{</span>
  <span class="va">df</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="va">gb</span><span class="op">)</span>
  <span class="kw">if</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/NULL.html" class="external-link">is.null</a></span><span class="op">(</span><span class="va">df</span><span class="op">$</span><span class="va">Arm</span><span class="op">)</span><span class="op">)</span>       <span class="va">df</span><span class="op">$</span><span class="va">Arm</span>        <span class="op">&lt;-</span> <span class="cn">NA</span>
  <span class="kw">if</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/NULL.html" class="external-link">is.null</a></span><span class="op">(</span><span class="va">df</span><span class="op">$</span><span class="va">rep</span><span class="op">)</span><span class="op">)</span>       <span class="va">df</span><span class="op">$</span><span class="va">rep</span>        <span class="op">&lt;-</span> <span class="cn">NA</span>
  <span class="kw">if</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/NULL.html" class="external-link">is.null</a></span><span class="op">(</span><span class="va">df</span><span class="op">$</span><span class="va">query.rep</span><span class="op">)</span><span class="op">)</span> <span class="va">df</span><span class="op">$</span><span class="va">query.rep</span>  <span class="op">&lt;-</span> <span class="cn">NA</span>
  <span class="fu">tibble</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/tibble.html" class="external-link">tibble</a></span><span class="op">(</span>target     <span class="op">=</span> <span class="va">df</span><span class="op">$</span><span class="va">width</span>,        query     <span class="op">=</span> <span class="va">df</span><span class="op">$</span><span class="va">query.width</span>,
                 target.rep <span class="op">=</span> <span class="op">!</span> <span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">df</span><span class="op">$</span><span class="va">rep</span><span class="op">)</span>, query.rep <span class="op">=</span> <span class="op">!</span> <span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">df</span><span class="op">$</span><span class="va">query.rep</span><span class="op">)</span>,  <span class="co"># TRUE if overlaps a repeat</span>
                 pair <span class="op">=</span> <span class="va">pair</span>, Arm <span class="op">=</span> <span class="va">df</span><span class="op">$</span><span class="va">Arm</span>,
                 dist <span class="op">=</span> <span class="fu">OikScrambling</span><span class="fu">:::</span><span class="fu"><a href="../reference/compDistance.html">compDistance</a></span><span class="op">(</span><span class="va">pair</span><span class="op">)</span><span class="op">)</span>
<span class="op">}</span>

<span class="va">ggPlotW2W</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">tibble</span>, <span class="va">plus1</span> <span class="op">=</span> <span class="cn">FALSE</span>, <span class="va">alpha</span> <span class="op">=</span> <span class="fl">0.05</span><span class="op">)</span> <span class="op">{</span>
  <span class="kw">if</span> <span class="op">(</span><span class="va">plus1</span><span class="op">)</span> <span class="op">{</span>
    <span class="va">tibble</span><span class="op">$</span><span class="va">target</span> <span class="op">&lt;-</span> <span class="va">tibble</span><span class="op">$</span><span class="va">target</span> <span class="op">+</span> <span class="fl">1</span>
    <span class="va">tibble</span><span class="op">$</span><span class="va">query</span>  <span class="op">&lt;-</span> <span class="va">tibble</span><span class="op">$</span><span class="va">query</span>  <span class="op">+</span> <span class="fl">1</span>
  <span class="op">}</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">tibble</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span><span class="va">target</span>, <span class="va">query</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span>alpha <span class="op">=</span> <span class="va">alpha</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_density_2d.html" class="external-link">geom_density_2d</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_continuous.html" class="external-link">scale_x_log10</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_continuous.html" class="external-link">scale_y_log10</a></span><span class="op">(</span><span class="op">)</span>
<span class="op">}</span>

<span class="co"># Remove empty objects.</span>
<span class="va">flippedInvs_bri</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">flippedInvs_bri</span>, <span class="va">length</span><span class="op">)</span> <span class="op">==</span> <span class="fl">0</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="cn">NULL</span>

<span class="va">w2w</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/do.call.html" class="external-link">do.call</a></span><span class="op">(</span><span class="va">rbind</span>, <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">flippedInvs_bri</span><span class="op">)</span>,  \<span class="op">(</span><span class="va">name</span><span class="op">)</span> <span class="fu">GBreaks2widthTibble</span><span class="op">(</span><span class="va">flippedInvs_bri</span><span class="op">[[</span><span class="va">name</span><span class="op">]</span><span class="op">]</span>,  pair <span class="op">=</span> <span class="va">name</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>

<span class="fu">ggPlotW2W</span><span class="op">(</span><span class="va">w2w</span> |&gt; <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">Arm</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html" class="external-link">facet_wrap</a></span><span class="op">(</span><span class="op">~</span><span class="va">Arm</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Width of the target and query regions flanking inversions"</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Warning: Transformation introduced infinite values in continuous x-axis</span></code></pre>
<pre><code><span class="co">## Warning: Transformation introduced infinite values in continuous y-axis</span></code></pre>
<pre><code><span class="co">## Warning: Transformation introduced infinite values in continuous x-axis</span></code></pre>
<pre><code><span class="co">## Warning: Transformation introduced infinite values in continuous y-axis</span></code></pre>
<pre><code><span class="co">## Warning: Removed 115 rows containing non-finite values (stat_density2d).</span></code></pre>
<p><img src="Inversions_files/figure-html/bridges_of_inversions-1.png" width="700"></p>
<div class="sourceCode" id="cb42"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">ggPlotW2W</span><span class="op">(</span><span class="va">w2w</span><span class="op">)</span>                              <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html" class="external-link">facet_wrap</a></span><span class="op">(</span><span class="op">~</span><span class="va">dist</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Width of the target and query regions flanking inversions"</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Warning: Transformation introduced infinite values in continuous x-axis</span></code></pre>
<pre><code><span class="co">## Warning: Transformation introduced infinite values in continuous y-axis</span></code></pre>
<pre><code><span class="co">## Warning: Transformation introduced infinite values in continuous x-axis</span></code></pre>
<pre><code><span class="co">## Warning: Transformation introduced infinite values in continuous y-axis</span></code></pre>
<pre><code><span class="co">## Warning: Removed 183 rows containing non-finite values (stat_density2d).</span></code></pre>
<p><img src="Inversions_files/figure-html/bridges_of_inversions-2.png" width="700"></p>
</div>
<div class="section level2">
<h2 id="same-culture-inversions">Same-culture inversions<a class="anchor" aria-label="anchor" href="#same-culture-inversions"></a>
</h2>
<div class="sourceCode" id="cb48"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># i28.gbs &lt;- load_genomic_breaks("~/FromSango/I28-4.07.postmasked.gff.gz", type = "match",</span>
<span class="co">#                                target = genomes$OKI2018.I69, query = BSgenome.Oidioi.OIST.I28) |&gt; flagLongShort(longShort$OKI2018.I69)</span>
<span class="co"># i28.coa &lt;- coalesce_contigs(i28.gbs) |&gt; flagLongShort(longShort$OKI2018.I69)</span>
<span class="co"># i28.briInv &lt;- i28.coa |&gt; flipInversions() |&gt; bridgeRegions()</span>
<span class="co"># i28.briInv$wT &lt;- width(i28.briInv)</span>
<span class="co"># i28.briInv$wQ &lt;- width(i28.briInv$query)</span>
<span class="co"># i28.briInv &lt;- dist2next(i28.briInv)</span>
<span class="co"># i28.w2w &lt;- GBreaks2widthTibble(i28.briInv, pair = "Oki_Oki")</span>
<span class="co"># ggPlotW2W(i28.w2w)</span>
<span class="co"># ggPlotW2W(i28.w2w, plus1 = TRUE, alpha = .5)</span></code></pre></div>
<p>It looks like some aligned inversions with zero-width bridges on both sides are flanked by tandem inverted unmapped regions.</p>
<p><a href="https://fantom.gsc.riken.jp/zenbu/gLyphs/#config=f863qz3iKBFqUgA75J9BBC;loc=OKI2018_I69_1.0" class="external-link uri">https://fantom.gsc.riken.jp/zenbu/gLyphs/#config=f863qz3iKBFqUgA75J9BBC;loc=OKI2018_I69_1.0</a>::chr1:7668403..7668663+</p>
<p>A region in Oki that inverted but not in the other genomes, and maybe in only one haplotype. The unaligned regions are shorter in Oki, suggesting that the lesion happened there. They still contain repeats.</p>
<p><a href="https://fantom.gsc.riken.jp/zenbu/gLyphs/#config=Px8xzKj9BdRgDGPSmcNT_;loc=OKI2018_I69_1.0" class="external-link uri">https://fantom.gsc.riken.jp/zenbu/gLyphs/#config=Px8xzKj9BdRgDGPSmcNT_;loc=OKI2018_I69_1.0</a>::PAR:16104430..16107799+</p>
<pre><code><span class="va">coa</span><span class="op">$</span><span class="va">Oki_Kum</span> |&gt; <span class="fu">plyranges</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/plyranges/man/tidyverse-reexports.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">start</span> <span class="op">&gt;=</span> <span class="fl">16098309</span>, <span class="va">end</span> <span class="op">&lt;=</span> <span class="fl">16113733</span>, <span class="va">seqnames</span> <span class="op">==</span> <span class="st">'PAR'</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/dist2next.html" class="external-link">dist2next</a></span><span class="op">(</span>ignore.strand <span class="op">=</span> <span class="cn">T</span><span class="op">)</span></code></pre>
<p>A region in Oki inverted compared to I28 and Kum, and maybe also Osaka, flanked by a pair of repeats of the same family that is inserted only in Oki (because gap length is near-zero in I28 and Kum).</p>
<div class="sourceCode" id="cb50"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">Add</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">gb</span>, <span class="va">n</span><span class="op">)</span> <span class="op">{</span>
  <span class="fu"><a href="https://rdrr.io/r/stats/start.html" class="external-link">start</a></span><span class="op">(</span><span class="va">gb</span><span class="op">)</span>       <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/start.html" class="external-link">start</a></span><span class="op">(</span><span class="va">gb</span>      <span class="op">)</span> <span class="op">-</span> <span class="va">n</span>
  <span class="fu"><a href="https://rdrr.io/r/stats/start.html" class="external-link">end</a></span>  <span class="op">(</span><span class="va">gb</span><span class="op">)</span>       <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/start.html" class="external-link">end</a></span>  <span class="op">(</span><span class="va">gb</span>      <span class="op">)</span> <span class="op">+</span> <span class="va">n</span>
  <span class="fu"><a href="https://rdrr.io/r/stats/start.html" class="external-link">start</a></span><span class="op">(</span><span class="va">gb</span><span class="op">$</span><span class="va">query</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/start.html" class="external-link">start</a></span><span class="op">(</span><span class="va">gb</span><span class="op">$</span><span class="va">query</span><span class="op">)</span> <span class="op">-</span> <span class="va">n</span>
  <span class="fu"><a href="https://rdrr.io/r/stats/start.html" class="external-link">end</a></span>  <span class="op">(</span><span class="va">gb</span><span class="op">$</span><span class="va">query</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/start.html" class="external-link">end</a></span>  <span class="op">(</span><span class="va">gb</span><span class="op">$</span><span class="va">query</span><span class="op">)</span> <span class="op">+</span> <span class="va">n</span>
  <span class="va">gb</span>
<span class="op">}</span>

<span class="co"># (ROI3_I28 &lt;- i28.gbs     |&gt; plyranges::filter(start &gt;= 10262805, end &lt;= 10277383, seqnames == 'PAR') |&gt; dist2next(ignore.strand = T))</span>
<span class="op">(</span><span class="va">ROI3_Kum</span> <span class="op">&lt;-</span> <span class="va">coa</span><span class="op">$</span><span class="va">Oki_Kum</span> |&gt; <span class="fu">plyranges</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/plyranges/man/tidyverse-reexports.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">start</span> <span class="op">&gt;=</span> <span class="fl">10255040</span>, <span class="va">end</span> <span class="op">&lt;=</span> <span class="fl">10280531</span>, <span class="va">seqnames</span> <span class="op">==</span> <span class="st">'PAR'</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/dist2next.html" class="external-link">dist2next</a></span><span class="op">(</span>ignore.strand <span class="op">=</span> <span class="cn">T</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## GBreaks object with 3 ranges and 10 metadata columns:</span>
<span class="co">##       seqnames            ranges strand |                     query     score</span>
<span class="co">##          &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; |                 &lt;GRanges&gt; &lt;integer&gt;</span>
<span class="co">##   [1]      PAR 10255449-10267754      + | contig_28_1:275929-288260     12306</span>
<span class="co">##   [2]      PAR 10268555-10269727      - | contig_28_1:288261-289408      1173</span>
<span class="co">##   [3]      PAR 10270601-10280100      + | contig_28_1:289410-297588      9500</span>
<span class="co">##            Arm              rep   repOvlp         transcripts        flag</span>
<span class="co">##       &lt;factor&gt;  &lt;CharacterList&gt; &lt;integer&gt;               &lt;Rle&gt; &lt;character&gt;</span>
<span class="co">##   [1]     long      rnd,unknown       108           g11347.t1         Inv</span>
<span class="co">##   [2]     long             &lt;NA&gt;         0 g11350.t1;g11350.t3        &lt;NA&gt;</span>
<span class="co">##   [3]     long rnd,tandem,ltr-1      1180           g11351.t1        &lt;NA&gt;</span>
<span class="co">##          nonCoa     tdist     qdist</span>
<span class="co">##       &lt;logical&gt; &lt;numeric&gt; &lt;numeric&gt;</span>
<span class="co">##   [1]     FALSE       801         1</span>
<span class="co">##   [2]      TRUE       874         2</span>
<span class="co">##   [3]     FALSE       Inf       Inf</span>
<span class="co">##   -------</span>
<span class="co">##   seqinfo: 19 sequences from OKI2018.I69 genome</span></code></pre>
<div class="sourceCode" id="cb52"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># </span>
<span class="co"># # Sequence near the </span>
<span class="co"># # Need to shrink to avoid zero-length regions not yet supported by cleanGaps.</span>
<span class="co"># # Add 10 to the gaps to see sequence</span>
<span class="op">(</span><span class="op">(</span><span class="va">ROI3_Kum</span><span class="op">$</span><span class="va">query</span> <span class="op">-</span><span class="fl">1</span><span class="op">)</span>|&gt; <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/cleanGaps.html" class="external-link">cleanGaps</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span><span class="fl">10</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/BSgenome/man/getSeq-methods.html" class="external-link">getSeq</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## DNAStringSet object of length 2:</span>
<span class="co">##     width seq</span>
<span class="co">## [1]    22 <span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span></span>
<span class="co">## [2]    23 <span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">CC</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">TTT</span><span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span></span></code></pre>
<div class="sourceCode" id="cb54"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># ((ROI3_I28$query -1)|&gt; cleanGaps() +10) |&gt; getSeq()</span>
<span class="co"># </span>
<span class="co"># ROI3_I28</span>
<span class="co"># pairwiseAlignment(ROI3_I28[1] |&gt; Add(6) )</span>
<span class="co"># pairwiseAlignment(ROI3_I28[2] |&gt; Add(8) )</span>
<span class="co"># getSeq(ROI3_I28[2] |&gt; Add(8) |&gt; plyranges::mutate(strand = "+"))</span>
<span class="co"># pairwiseAlignment(ROI3_I28[3] |&gt; Add(6) )</span>

<span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/pairwiseAlignment.html" class="external-link">pairwiseAlignment</a></span><span class="op">(</span><span class="va">ROI3_Kum</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span> |&gt; <span class="fu">Add</span><span class="op">(</span><span class="fl">6</span><span class="op">)</span> <span class="op">)</span></code></pre></div>
<pre><code><span class="co">## </span>
<span class="co">## Attaching package: 'BSgenome.Oidioi.OIST.OKI2018.I69'</span></code></pre>
<pre><code><span class="co">## The following object is masked from 'package:BSgenome.Oidioi.OIST.KUM.M3.7f':</span>
<span class="co">## </span>
<span class="co">##     Oidioi</span></code></pre>
<pre><code><span class="co">## Global PairwiseAlignmentsSingleSubject (1 of 1)</span>
<span class="co">## pattern: <span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">CCCCC</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #FFAFAF;">AAA</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">GG</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">CC</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span>...<span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">TTT</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">TTT</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span></span>
<span class="co">## subject: <span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">TTT</span><span style="color: #000000; background-color: #FFAFAF;">AAA</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">GG</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">CC</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">AAA</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span>...<span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">TTT</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">TTT</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span></span>
<span class="co">## score: 19147.97</span></code></pre>
<div class="sourceCode" id="cb58"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/pairwiseAlignment.html" class="external-link">pairwiseAlignment</a></span><span class="op">(</span><span class="va">ROI3_Kum</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span> |&gt; <span class="fu">Add</span><span class="op">(</span><span class="fl">8</span><span class="op">)</span> <span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Global PairwiseAlignmentsSingleSubject (1 of 1)</span>
<span class="co">## pattern: <span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">CC</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">GG</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">GG</span><span style="color: #000000; background-color: #FFD7AF;">TT</span>...<span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">TTT</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">TTT</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">CC</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">TTT</span><span style="color: #000000; background-color: #FFAFAF;">AAAA</span></span>
<span class="co">## subject: <span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">CC</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">GG</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">GG</span><span style="color: #000000; background-color: #FFD7AF;">TT</span>...<span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">TTT</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">TTT</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">CC</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">TTT</span><span style="color: #000000; background-color: #FFAFAF;">A</span></span>
<span class="co">## score: 1722.804</span></code></pre>
<div class="sourceCode" id="cb60"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/BSgenome/man/getSeq-methods.html" class="external-link">getSeq</a></span><span class="op">(</span><span class="va">ROI3_Kum</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span> |&gt; <span class="fu">Add</span><span class="op">(</span><span class="fl">8</span><span class="op">)</span> |&gt; <span class="fu">plyranges</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/plyranges/man/tidyverse-reexports.html" class="external-link">mutate</a></span><span class="op">(</span>strand <span class="op">=</span> <span class="st">"+"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## DNAStringSet object of length 1:</span>
<span class="co">##     width seq</span>
<span class="co">## [1]  1189 <span style="color: #000000; background-color: #FFD7AF;">TTTT</span><span style="color: #000000; background-color: #FFAFAF;">AAA</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">GG</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">AAA</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">AAA</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">TT</span>...<span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">CC</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">CC</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #AFFFFF;">GG</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span></span></code></pre>
<div class="sourceCode" id="cb62"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#pairwiseAlignment(ROI3_I28[3] |&gt; Add(6) )</span></code></pre></div>
<p>My hypothesis:</p>
<ol style="list-style-type: lower-alpha">
<li>Ancestral version with no repeats.</li>
<li>One <code>i69_juicer_1_5_16917</code> repeat inserts next to <code>g11343.t3</code> gene</li>
<li>A second copy of <code>i69_juicer_1_5_16917</code> inserts on the other side of the gene, in the opposite orientation.</li>
<li>Later, because of the tandem inverted repeat, the gene was inverted.</li>
</ol>
<p>The repeats have inverted tandem sites <code>cagtcgactctggcaattacgag</code></p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Aleksandra Bliznina, Michael Mansfield, Charles Plessy.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.2.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
