<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Inversions • OikScrambling</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Inversions">
<meta property="og:description" content="OikScrambling">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">OikScrambling</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">5.0.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/OikScrambling.html">Get started</a>
</li>
<li>
  <a href="../articles/LoadGenomicBreaks.html">Data load</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Supplemental material
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/OxfordPlots.html">Figure 1A DNA line plots</a>
    </li>
    <li>
      <a href="../articles/ParallelPlots.html">Figure 1B chromosome parallel plots</a>
    </li>
    <li>
      <a href="../articles/DistanceMeasures.html">Table 1 scrambling index</a>
    </li>
    <li>
      <a href="../articles/ChromosomePlots.html">Figure 5 genome-wide feature patterns</a>
    </li>
    <li>
      <a href="../articles/SquareWindowPlots.html">Figure 7 DNA line plots</a>
    </li>
    <li>
      <a href="../articles/ScalingBreakpoints.html">Figure 7 breakpoint number barplots</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/CNEs.html">Conserved non-coding elements</a>
    </li>
    <li>
      <a href="../articles/ChromosomePlots.html">Chromosome-scale feature density plots</a>
    </li>
    <li>
      <a href="../articles/CoalescingCutoffs.html">Modifying coalescent cutoffs</a>
    </li>
    <li>
      <a href="../articles/ColinearityInterruptors.html">Regions interrupting colinearity</a>
    </li>
    <li>
      <a href="../articles/DistanceMeasures.html">Distance Measures</a>
    </li>
    <li>
      <a href="../articles/GenomicFeatures.html">Genomic features of the breakpoints</a>
    </li>
    <li>
      <a href="../articles/GenomicHeatmaps.html">Genomic features of the breakpoints (heatmaps)</a>
    </li>
    <li>
      <a href="../articles/Inversions.html">Inversions</a>
    </li>
    <li>
      <a href="../articles/LoadGenomicBreaks.html">Load genomic break data</a>
    </li>
    <li>
      <a href="../articles/OperonConservation.html">Operon conservation analysis</a>
    </li>
    <li>
      <a href="../articles/OrthogroupSyntenies.html">Orthogroup syntenies</a>
    </li>
    <li>
      <a href="../articles/OxfordPlots.html">Alignment line plots, whole genomes</a>
    </li>
    <li>
      <a href="../articles/PAC3.html">Region of interest: PAC3</a>
    </li>
    <li>
      <a href="../articles/PWM.html">Motif search and analysis</a>
    </li>
    <li>
      <a href="../articles/ParallelPlots.html">Whole-chromosome parallel plots</a>
    </li>
    <li>
      <a href="../articles/RegionWidths.html">Width of the aligned and mapped regions.</a>
    </li>
    <li>
      <a href="../articles/RegionsOfInterest.html">Regions of interest</a>
    </li>
    <li>
      <a href="../articles/RepeatRegions.html">Repeat regions</a>
    </li>
    <li>
      <a href="../articles/ScalingBreakpoints.html">Scaling breakpoints</a>
    </li>
    <li>
      <a href="../articles/SegmentMovements.html">Segment movement distances</a>
    </li>
    <li>
      <a href="../articles/SquareWindowPlots.html">Alignment line plots, 10×10 Mbp windows</a>
    </li>
    <li>
      <a href="../articles/TFBS.html">Transcription factor binding sites</a>
    </li>
    <li>
      <a href="../articles/TandemSiteDuplications.html">Tandem site duplications</a>
    </li>
    <li>
      <a href="../articles/Translocations.html">Translocations</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/oist/LuscombeU_OikScrambling/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Inversions</h1>
                        <h4 data-toc-skip class="author">Charles
Plessy</h4>
            
            <h4 data-toc-skip class="date">13 October, 2023</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/oist/LuscombeU_OikScrambling/blob/HEAD/vignettes/Inversions.Rmd" class="external-link"><code>vignettes/Inversions.Rmd</code></a></small>
      <div class="hidden name"><code>Inversions.Rmd</code></div>

    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">knitr</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/knitr/man/opts_chunk.html" class="external-link">opts_chunk</a></span><span class="op">$</span><span class="fu">set</span><span class="op">(</span>cache <span class="op">=</span> <span class="cn">TRUE</span>, cache.lazy <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>After coalescing colinear alignments, removing translocations of
repeat-containing sequences and re-coalescing, colinearity is still
broken hundreds of time.</p>
<p>Here we explore the role of inversions in scrambling
<em>Oikopleura</em> genomes.</p>
</div>
<div class="section level2">
<h2 id="load-r-pacakges-and-data">Load R pacakges and data<a class="anchor" aria-label="anchor" href="#load-r-pacakges-and-data"></a>
</h2>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/oist/LuscombeU_OikScrambling" class="external-link">'OikScrambling'</a></span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/load.html" class="external-link">load</a></span><span class="op">(</span><span class="st">"BreakPoints.Rdata"</span><span class="op">)</span></span></code></pre></div>
<p>See
<code><a href="../articles/LoadGenomicBreaks.html">vignette("LoadGenomicBreaks", package = "OikScrambling")</a></code>
for how the different GBreaks objects are prepared.</p>
</div>
<div class="section level2">
<h2 id="trivial-inversions">Trivial inversions<a class="anchor" aria-label="anchor" href="#trivial-inversions"></a>
</h2>
<div class="section level3">
<h3 id="documentation">Documentation<a class="anchor" aria-label="anchor" href="#documentation"></a>
</h3>
<p>Details can be found in
<code><a href="https://oist.github.io/GenomicBreaks/articles/GenomicBreaks.html" class="external-link">vignette("GenomicBreaks", package = "GenomicBreaks")</a></code>, in
<code><a href="https://oist.github.io/GenomicBreaks/articles/StructuralVariants.html" class="external-link">vignette("StructuralVariants", package = "GenomicBreaks")</a></code>,
and <code><a href="https://oist.github.io/GenomicBreaks/reference/flagInversions.html" class="external-link">?GenomicBreaks::flagInversions</a></code></p>
</div>
<div class="section level3">
<h3 id="number-of-trivial-inversions">Number of trivial inversions<a class="anchor" aria-label="anchor" href="#number-of-trivial-inversions"></a>
</h3>
<p>More inversions are found after coalescing colinear blocks because of
situations where <code>+ - +</code> was <code>+ - - +</code> before
collapsing.</p>
<p>The number of detected inversions is further increased after
double-coalescing, but not much.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="op">(</span><span class="va">invs_summary</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span></span>
<span>  align  <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">gbs</span>,  <span class="kw">function</span><span class="op">(</span><span class="va">gb</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/flagInversions.html" class="external-link">flagInversions</a></span><span class="op">(</span><span class="va">gb</span><span class="op">)</span><span class="op">$</span><span class="va">inv</span><span class="op">)</span><span class="op">)</span>,</span>
<span>  mapped <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">coa</span>,  <span class="kw">function</span><span class="op">(</span><span class="va">gb</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/flagInversions.html" class="external-link">flagInversions</a></span><span class="op">(</span><span class="va">gb</span><span class="op">)</span><span class="op">$</span><span class="va">inv</span><span class="op">)</span><span class="op">)</span>,</span>
<span>  map2   <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">coa2</span>, <span class="kw">function</span><span class="op">(</span><span class="va">gb</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/flagInversions.html" class="external-link">flagInversions</a></span><span class="op">(</span><span class="va">gb</span><span class="op">)</span><span class="op">$</span><span class="va">inv</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##         align mapped map2</span></span>
<span><span class="co">## Oki_Osa   614    670  704</span></span>
<span><span class="co">## Oki_Bar   604    675  705</span></span>
<span><span class="co">## Oki_Kum    89    105  122</span></span>
<span><span class="co">## Oki_Aom   608    665  695</span></span>
<span><span class="co">## Oki_Nor   541    591  611</span></span>
<span><span class="co">## Osa_Oki   613    671  706</span></span>
<span><span class="co">## Osa_Bar   241    321  330</span></span>
<span><span class="co">## Osa_Kum   620    678  708</span></span>
<span><span class="co">## Osa_Aom    56     73   79</span></span>
<span><span class="co">## Osa_Nor   212    275  279</span></span>
<span><span class="co">## Bar_Oki   597    667  695</span></span>
<span><span class="co">## Bar_Osa   237    319  327</span></span>
<span><span class="co">## Bar_Kum   608    686  711</span></span>
<span><span class="co">## Bar_Aom   237    319  335</span></span>
<span><span class="co">## Bar_Nor    19     24   27</span></span>
<span><span class="co">## Ply_Ros    42     51   55</span></span>
<span><span class="co">## Ply_Rob    58     71   76</span></span>
<span><span class="co">## Ply_Sav    16     92   92</span></span>
<span><span class="co">## Ply_Oki     0      0    0</span></span>
<span><span class="co">## Rob_Ros    50     65   71</span></span>
<span><span class="co">## Rob_Ply    58     70   77</span></span>
<span><span class="co">## Rob_Sav    14     92   92</span></span>
<span><span class="co">## Rob_Oki     0      0    0</span></span>
<span><span class="co">## Dme_Dbu   697   1003 1003</span></span>
<span><span class="co">## Dme_Dsu    20     44   45</span></span>
<span><span class="co">## Dme_Dya    13     21   23</span></span>
<span><span class="co">## Dme_Dma    12     16   16</span></span>
<span><span class="co">## Cni_Cbr   221    300  331</span></span>
<span><span class="co">## Cni_Cre    91    212  216</span></span>
<span><span class="co">## Cni_Cin    40    132  134</span></span>
<span><span class="co">## Cbr_Cni   236    318  358</span></span>
<span><span class="co">## Cbr_Cre    92    200  207</span></span>
<span><span class="co">## Cbr_Cel    64    166  166</span></span></code></pre>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">invs_summary</span>, \<span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/tapply.html" class="external-link">tapply</a></span><span class="op">(</span><span class="va">x</span>, <span class="fu"><a href="https://rdrr.io/r/base/row.names.html" class="external-link">row.names</a></span><span class="op">(</span><span class="va">invs_summary</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu">OikScrambling</span><span class="fu">:::</span><span class="fu"><a href="../reference/compDistance.html">compDistance</a></span><span class="op">(</span><span class="op">)</span>, <span class="va">mean</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##                   align     mapped       map2</span></span>
<span><span class="co">## Cbr_Cel        64.00000  166.00000  166.00000</span></span>
<span><span class="co">## Cbr_Cni       236.00000  318.00000  358.00000</span></span>
<span><span class="co">## Cbr_Cre        92.00000  200.00000  207.00000</span></span>
<span><span class="co">## Cni_Cbr       221.00000  300.00000  331.00000</span></span>
<span><span class="co">## Cni_Cin        40.00000  132.00000  134.00000</span></span>
<span><span class="co">## Cni_Cre        91.00000  212.00000  216.00000</span></span>
<span><span class="co">## Dme_Dbu       697.00000 1003.00000 1003.00000</span></span>
<span><span class="co">## Dme_Dma        12.00000   16.00000   16.00000</span></span>
<span><span class="co">## Dme_Dsu        20.00000   44.00000   45.00000</span></span>
<span><span class="co">## Dme_Dya        13.00000   21.00000   23.00000</span></span>
<span><span class="co">## In same pop    54.66667   67.33333   76.00000</span></span>
<span><span class="co">## Int – Int      42.00000   51.00000   55.00000</span></span>
<span><span class="co">## Int – Rob      55.33333   68.66667   74.66667</span></span>
<span><span class="co">## Int/Rob – Oki   0.00000    0.00000    0.00000</span></span>
<span><span class="co">## Int/Rob – Sav  15.00000   92.00000   92.00000</span></span>
<span><span class="co">## North – North 231.75000  308.50000  317.75000</span></span>
<span><span class="co">## Oki – North   600.62500  662.87500  691.87500</span></span></code></pre>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">invs_summary</span>, \<span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/tapply.html" class="external-link">tapply</a></span><span class="op">(</span><span class="va">x</span>, <span class="fu"><a href="https://rdrr.io/r/base/row.names.html" class="external-link">row.names</a></span><span class="op">(</span><span class="va">invs_summary</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu">OikScrambling</span><span class="fu">:::</span><span class="fu"><a href="../reference/compDistance.html">compDistance</a></span><span class="op">(</span><span class="op">)</span>, <span class="va">median</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##               align mapped   map2</span></span>
<span><span class="co">## Cbr_Cel          64  166.0  166.0</span></span>
<span><span class="co">## Cbr_Cni         236  318.0  358.0</span></span>
<span><span class="co">## Cbr_Cre          92  200.0  207.0</span></span>
<span><span class="co">## Cni_Cbr         221  300.0  331.0</span></span>
<span><span class="co">## Cni_Cin          40  132.0  134.0</span></span>
<span><span class="co">## Cni_Cre          91  212.0  216.0</span></span>
<span><span class="co">## Dme_Dbu         697 1003.0 1003.0</span></span>
<span><span class="co">## Dme_Dma          12   16.0   16.0</span></span>
<span><span class="co">## Dme_Dsu          20   44.0   45.0</span></span>
<span><span class="co">## Dme_Dya          13   21.0   23.0</span></span>
<span><span class="co">## In same pop      56   73.0   79.0</span></span>
<span><span class="co">## Int – Int        42   51.0   55.0</span></span>
<span><span class="co">## Int – Rob        58   70.0   76.0</span></span>
<span><span class="co">## Int/Rob – Oki     0    0.0    0.0</span></span>
<span><span class="co">## Int/Rob – Sav    15   92.0   92.0</span></span>
<span><span class="co">## North – North   237  319.0  328.5</span></span>
<span><span class="co">## Oki – North     608  670.5  704.5</span></span></code></pre>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">invs_summary</span>, \<span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/tapply.html" class="external-link">tapply</a></span><span class="op">(</span><span class="va">x</span>, <span class="fu"><a href="https://rdrr.io/r/base/row.names.html" class="external-link">row.names</a></span><span class="op">(</span><span class="va">invs_summary</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu">OikScrambling</span><span class="fu">:::</span><span class="fu"><a href="../reference/compDistance.html">compDistance</a></span><span class="op">(</span><span class="op">)</span>, <span class="va">sd</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##                   align   mapped     map2</span></span>
<span><span class="co">## Cbr_Cel              NA       NA       NA</span></span>
<span><span class="co">## Cbr_Cni              NA       NA       NA</span></span>
<span><span class="co">## Cbr_Cre              NA       NA       NA</span></span>
<span><span class="co">## Cni_Cbr              NA       NA       NA</span></span>
<span><span class="co">## Cni_Cin              NA       NA       NA</span></span>
<span><span class="co">## Cni_Cre              NA       NA       NA</span></span>
<span><span class="co">## Dme_Dbu              NA       NA       NA</span></span>
<span><span class="co">## Dme_Dma              NA       NA       NA</span></span>
<span><span class="co">## Dme_Dsu              NA       NA       NA</span></span>
<span><span class="co">## Dme_Dya              NA       NA       NA</span></span>
<span><span class="co">## In same pop   35.019042 40.79624 47.57100</span></span>
<span><span class="co">## Int – Int            NA       NA       NA</span></span>
<span><span class="co">## Int – Rob      4.618802  3.21455  3.21455</span></span>
<span><span class="co">## Int/Rob – Oki  0.000000  0.00000  0.00000</span></span>
<span><span class="co">## Int/Rob – Sav  1.414214  0.00000  0.00000</span></span>
<span><span class="co">## North – North 13.301002 22.35322 26.04323</span></span>
<span><span class="co">## Oki – North   25.059572 29.79663 33.17675</span></span></code></pre>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">invs_summary</span><span class="op">$</span><span class="va">pairname</span>   <span class="op">&lt;-</span> <span class="fu">OikScrambling</span><span class="fu">:::</span><span class="fu"><a href="../reference/compDistance.html">compDistance</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">invs_summary</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">invs_summary</span><span class="op">$</span><span class="va">pairname_s</span> <span class="op">&lt;-</span> <span class="fu">OikScrambling</span><span class="fu">:::</span><span class="fu"><a href="../reference/compDistance.html">compDistance</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">invs_summary</span><span class="op">)</span>, short <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">invs_summary</span><span class="op">$</span><span class="va">genus</span>      <span class="op">&lt;-</span> <span class="fu">OikScrambling</span><span class="fu">:::</span><span class="fu"><a href="../reference/compGenus.html">compGenus</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">invs_summary</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">invs_summary</span><span class="op">$</span><span class="va">class</span>      <span class="op">&lt;-</span> <span class="fu">OikScrambling</span><span class="fu">:::</span><span class="fu"><a href="../reference/compDistClass.html">compDistClass</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">invs_summary</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">invs_summary</span><span class="op">$</span><span class="va">target</span>     <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">sub</a></span><span class="op">(</span><span class="st">"_.*"</span>, <span class="st">""</span>, <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">invs_summary</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">invs_summary</span> <span class="op">&lt;-</span> <span class="va">invs_summary</span><span class="op">[</span><span class="va">invs_summary</span><span class="op">$</span><span class="va">class</span> <span class="op">!=</span> <span class="st">"Int/Rob – Oki"</span>,<span class="op">]</span></span>
<span></span>
<span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">invs_summary</span> <span class="op">|&gt;</span> <span class="fu">tidyr</span><span class="fu">::</span><span class="fu"><a href="https://tidyr.tidyverse.org/reference/pivot_longer.html" class="external-link">pivot_longer</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"align"</span>, <span class="st">"mapped"</span>, <span class="st">"map2"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span><span class="va">value</span>, <span class="va">pairname_s</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>col <span class="op">=</span> <span class="va">target</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span> <span class="op">(</span><span class="st">"Number of inversions found after different post-processings"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html" class="external-link">facet_wrap</a></span><span class="op">(</span><span class="op">~</span><span class="va">name</span>, ncol<span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span></code></pre></div>
<p><img src="Inversions_files/figure-html/detect_trivial_invertions-1.png" width="768"></p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">invs_summary</span> <span class="op">|&gt;</span> <span class="fu">tidyr</span><span class="fu">::</span><span class="fu"><a href="https://tidyr.tidyverse.org/reference/pivot_longer.html" class="external-link">pivot_longer</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"mapped"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span><span class="va">value</span>, <span class="va">genus</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>col <span class="op">=</span> <span class="va">class</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span> <span class="op">(</span><span class="st">"Number of inversions found in mapped regions"</span><span class="op">)</span> </span></code></pre></div>
<p><img src="Inversions_files/figure-html/number_of_inversions-1.svg" width="672"></p>
</div>
<div class="section level3">
<h3 id="motif-search">Motif search<a class="anchor" aria-label="anchor" href="#motif-search"></a>
</h3>
<p>Enrichment for motifs is being searched in
<code><a href="../articles/PWM.html">vignette("PWM", package = "OikScrambling")</a></code>.</p>
</div>
<div class="section level3">
<h3 id="breakpoint-regions-flanking-inversions-in-the-same-population-alignments-">Breakpoint regions flanking inversions in the same-population
alignments.<a class="anchor" aria-label="anchor" href="#breakpoint-regions-flanking-inversions-in-the-same-population-alignments-"></a>
</h3>
<p>Check
<code><a href="../articles/ColinearityInterruptors.html">vignette("ColinearityInterruptors", package = "OikScrambling")</a></code>.</p>
</div>
<div class="section level3">
<h3 id="breakpoint-regions-flanking-large-inversions-in-the-northnorth-alignments-">Breakpoint regions flanking large inversions in the North–North
alignments.<a class="anchor" aria-label="anchor" href="#breakpoint-regions-flanking-large-inversions-in-the-northnorth-alignments-"></a>
</h3>
<p>Let’s try to isolate some inversions that are easy to spot on a
whole-chromosome alignment.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/plotApairOfChrs.html" class="external-link">plotApairOfChrs</a></span><span class="op">(</span><span class="va">coa2</span><span class="op">$</span><span class="va">Osa_Bar</span>, <span class="st">"PAR"</span><span class="op">)</span></span></code></pre></div>
<p><img src="Inversions_files/figure-html/unnamed-chunk-1-1.png" width="700"></p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="op">(</span><span class="va">ROI1</span> <span class="op">&lt;-</span> <span class="va">coa2</span><span class="op">$</span><span class="va">Osa_Bar</span> <span class="op">|&gt;</span> <span class="fu">plyranges</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">seqnames</span> <span class="op">==</span> <span class="st">"PAR"</span>, <span class="va">start</span> <span class="op">&gt;</span> <span class="fl">9e6</span>, <span class="va">end</span> <span class="op">&lt;</span> <span class="fl">14e6</span>, <span class="fu"><a href="https://rdrr.io/r/stats/start.html" class="external-link">start</a></span><span class="op">(</span><span class="va">query</span><span class="op">)</span> <span class="op">&gt;</span> <span class="fl">7e6</span>, <span class="fu"><a href="https://rdrr.io/r/stats/start.html" class="external-link">end</a></span><span class="op">(</span><span class="va">query</span><span class="op">)</span> <span class="op">&lt;</span> <span class="fl">12e6</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## GBreaks object with 35 ranges and 6 metadata columns:</span></span>
<span><span class="co">##        seqnames            ranges strand |                 query      Arm</span></span>
<span><span class="co">##           &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; |             &lt;GRanges&gt; &lt;factor&gt;</span></span>
<span><span class="co">##    [1]      PAR   9112106-9113024      + |   PAR:7914713-7915633     long</span></span>
<span><span class="co">##    [2]      PAR   9122674-9169015      + |   PAR:7647348-7694603     long</span></span>
<span><span class="co">##    [3]      PAR   9169027-9383526      + |   PAR:7700204-7911308     long</span></span>
<span><span class="co">##    [4]      PAR   9393845-9625372      - |   PAR:8648572-8868624     long</span></span>
<span><span class="co">##    [5]      PAR   9625425-9626193      + |   PAR:8647713-8648486     long</span></span>
<span><span class="co">##    ...      ...               ...    ... .                   ...      ...</span></span>
<span><span class="co">##   [31]      PAR 12882838-12884857      + | PAR:11615009-11617029     long</span></span>
<span><span class="co">##   [32]      PAR 12884875-13019548      - | PAR:10442170-10570866     long</span></span>
<span><span class="co">##   [33]      PAR 13027569-13077236      - | PAR:10392248-10442013     long</span></span>
<span><span class="co">##   [34]      PAR 13077254-13079729      + | PAR:11592137-11594612     long</span></span>
<span><span class="co">##   [35]      PAR 13079758-13079924      + | PAR:10378144-10378306     long</span></span>
<span><span class="co">##                                 rep   repOvlp            transcripts</span></span>
<span><span class="co">##                     &lt;CharacterList&gt; &lt;integer&gt;                  &lt;Rle&gt;</span></span>
<span><span class="co">##    [1]                         &lt;NA&gt;         0                   &lt;NA&gt;</span></span>
<span><span class="co">##    [2]                   rnd,tandem      1671 g9531.t1;g9532.t1;g9..</span></span>
<span><span class="co">##    [3]       rnd,unknown,tandem,...      4642 g9552.t1;g9553.t1;g9..</span></span>
<span><span class="co">##    [4] rnd,tandem,LowComplexity,...      5474 g9629.t1;g9630.t1;g9..</span></span>
<span><span class="co">##    [5]                         &lt;NA&gt;         0                   &lt;NA&gt;</span></span>
<span><span class="co">##    ...                          ...       ...                    ...</span></span>
<span><span class="co">##   [31]                          rnd      1167              g10776.t1</span></span>
<span><span class="co">##   [32] tandem,rnd,LowComplexity,...      3102 g10777.t1;g10779.t1;..</span></span>
<span><span class="co">##   [33]           unknown,rnd,tandem      1637 g10840.t1;g10841.t1;..</span></span>
<span><span class="co">##   [34]                          rnd         0                   &lt;NA&gt;</span></span>
<span><span class="co">##   [35]                         &lt;NA&gt;         0                   &lt;NA&gt;</span></span>
<span><span class="co">##               flag</span></span>
<span><span class="co">##        &lt;character&gt;</span></span>
<span><span class="co">##    [1]        &lt;NA&gt;</span></span>
<span><span class="co">##    [2]        &lt;NA&gt;</span></span>
<span><span class="co">##    [3]        &lt;NA&gt;</span></span>
<span><span class="co">##    [4]         Inv</span></span>
<span><span class="co">##    [5]        &lt;NA&gt;</span></span>
<span><span class="co">##    ...         ...</span></span>
<span><span class="co">##   [31]        &lt;NA&gt;</span></span>
<span><span class="co">##   [32]        &lt;NA&gt;</span></span>
<span><span class="co">##   [33]        &lt;NA&gt;</span></span>
<span><span class="co">##   [34]        &lt;NA&gt;</span></span>
<span><span class="co">##   [35]        &lt;NA&gt;</span></span>
<span><span class="co">##   -------</span></span>
<span><span class="co">##   seqinfo: 483 sequences from OSKA2016v1.9 genome</span></span></code></pre>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/plotApairOfChrs.html" class="external-link">plotApairOfChrs</a></span><span class="op">(</span><span class="va">ROI1</span><span class="op">)</span></span></code></pre></div>
<p><img src="Inversions_files/figure-html/unnamed-chunk-1-2.png" width="700"></p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># The contours of the main 4 blocks can be coalesced by applying a size threshold.</span></span>
<span><span class="va">ROI1</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/coalesce_contigs.html" class="external-link">coalesce_contigs</a></span><span class="op">(</span>min <span class="op">=</span> <span class="fl">1e4</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## GBreaks object with 4 ranges and 2 metadata columns:</span></span>
<span><span class="co">##       seqnames            ranges strand |                 query     score</span></span>
<span><span class="co">##          &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; |             &lt;GRanges&gt; &lt;integer&gt;</span></span>
<span><span class="co">##   [1]      PAR   9122674-9383526      + |   PAR:7647348-7911308    260853</span></span>
<span><span class="co">##   [2]      PAR  9393845-10352419      - |   PAR:7917928-8868624    958575</span></span>
<span><span class="co">##   [3]      PAR 10363392-11843885      + |  PAR:8872985-10375361   1480494</span></span>
<span><span class="co">##   [4]      PAR 11852584-13077236      - | PAR:10392248-11592118   1224653</span></span>
<span><span class="co">##   -------</span></span>
<span><span class="co">##   seqinfo: 483 sequences from OSKA2016v1.9 genome</span></span></code></pre>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># The gaps flanking the first inverted region are quite large!</span></span>
<span><span class="op">(</span><span class="va">ROI1_gaps</span> <span class="op">&lt;-</span> <span class="va">ROI1</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/coalesce_contigs.html" class="external-link">coalesce_contigs</a></span><span class="op">(</span>min <span class="op">=</span> <span class="fl">1e4</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/cleanGaps.html" class="external-link">cleanGaps</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu">plyranges</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>w <span class="op">=</span> <span class="va">width</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## GRanges object with 3 ranges and 1 metadata column:</span></span>
<span><span class="co">##       seqnames            ranges strand |         w</span></span>
<span><span class="co">##          &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt;</span></span>
<span><span class="co">##   [1]      PAR   9383527-9393844      * |     10318</span></span>
<span><span class="co">##   [2]      PAR 10352420-10363391      * |     10972</span></span>
<span><span class="co">##   [3]      PAR 11843886-11852583      * |      8698</span></span>
<span><span class="co">##   -------</span></span>
<span><span class="co">##   seqinfo: 483 sequences from OSKA2016v1.9 genome</span></span></code></pre>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># The left-side gap contains no alignment at all even before discarding the short ones.</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/findOverlaps-methods.html" class="external-link">subsetByOverlaps</a></span><span class="op">(</span><span class="va">gbs</span><span class="op">$</span><span class="va">Osa_Bar</span>, <span class="va">ROI1_gaps</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## GBreaks object with 4 ranges and 8 metadata columns:</span></span>
<span><span class="co">##       seqnames            ranges strand |     score                 query</span></span>
<span><span class="co">##          &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; | &lt;numeric&gt;             &lt;GRanges&gt;</span></span>
<span><span class="co">##   [1]      PAR 10355613-10355774      - |       500     XSR:671520-671674</span></span>
<span><span class="co">##   [2]      PAR 10355951-10356573      - |      1610     XSR:670597-671182</span></span>
<span><span class="co">##   [3]      PAR 10356752-10361833      - |     13551     XSR:665505-670596</span></span>
<span><span class="co">##   [4]      PAR 11847299-11849269      + |      5701 PAR:10385920-10388171</span></span>
<span><span class="co">##            Arm             rep   repOvlp transcripts        flag    nonCoa</span></span>
<span><span class="co">##       &lt;factor&gt; &lt;CharacterList&gt; &lt;integer&gt;       &lt;Rle&gt; &lt;character&gt; &lt;logical&gt;</span></span>
<span><span class="co">##   [1]     long            &lt;NA&gt;         0        &lt;NA&gt;         Col     FALSE</span></span>
<span><span class="co">##   [2]     long            &lt;NA&gt;         0        &lt;NA&gt;         Col     FALSE</span></span>
<span><span class="co">##   [3]     long            &lt;NA&gt;         0        &lt;NA&gt;        &lt;NA&gt;     FALSE</span></span>
<span><span class="co">##   [4]     long            &lt;NA&gt;         0   g10434.t1        &lt;NA&gt;      TRUE</span></span>
<span><span class="co">##   -------</span></span>
<span><span class="co">##   seqinfo: 483 sequences from OSKA2016v1.9 genome</span></span></code></pre>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># The same region aligns well to Aomori, suggesting that it is not missassembled</span></span>
<span><span class="va">coa2</span><span class="op">$</span><span class="va">Osa_Aom</span> <span class="op">|&gt;</span> <span class="fu">plyranges</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">seqnames</span> <span class="op">==</span> <span class="st">"PAR"</span>, <span class="va">start</span> <span class="op">&gt;</span> <span class="fl">9e6</span>, <span class="va">end</span> <span class="op">&lt;</span> <span class="fl">14e6</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/plotApairOfChrs.html" class="external-link">plotApairOfChrs</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p><img src="Inversions_files/figure-html/unnamed-chunk-1-3.png" width="700"></p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># The Osa–Bar unaligned regions appear to align well to Aom, but the first one is split</span></span>
<span><span class="co"># and the second one contains a small inversion. Are there a hotspots ?</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/findOverlaps-methods.html" class="external-link">subsetByOverlaps</a></span><span class="op">(</span><span class="va">gbs</span><span class="op">$</span><span class="va">Osa_Aom</span>, <span class="va">ROI1_gaps</span>, type <span class="op">=</span> <span class="st">"any"</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/plotApairOfChrs.html" class="external-link">plotApairOfChrs</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p><img src="Inversions_files/figure-html/unnamed-chunk-1-4.png" width="700"></p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/findOverlaps-methods.html" class="external-link">subsetByOverlaps</a></span><span class="op">(</span><span class="va">gbs</span><span class="op">$</span><span class="va">Osa_Aom</span>, <span class="va">ROI1_gaps</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>, type <span class="op">=</span> <span class="st">"within"</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/plotApairOfChrs.html" class="external-link">plotApairOfChrs</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p><img src="Inversions_files/figure-html/unnamed-chunk-1-5.png" width="700"></p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/findOverlaps-methods.html" class="external-link">subsetByOverlaps</a></span><span class="op">(</span><span class="va">gbs</span><span class="op">$</span><span class="va">Osa_Aom</span>, <span class="va">ROI1_gaps</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span>, type <span class="op">=</span> <span class="st">"within"</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/plotApairOfChrs.html" class="external-link">plotApairOfChrs</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p><img src="Inversions_files/figure-html/unnamed-chunk-1-6.png" width="700"></p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/findOverlaps-methods.html" class="external-link">subsetByOverlaps</a></span><span class="op">(</span><span class="va">gbs</span><span class="op">$</span><span class="va">Osa_Aom</span>, <span class="va">ROI1_gaps</span><span class="op">[</span><span class="fl">3</span><span class="op">]</span>, type <span class="op">=</span> <span class="st">"within"</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/plotApairOfChrs.html" class="external-link">plotApairOfChrs</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p><img src="Inversions_files/figure-html/unnamed-chunk-1-7.png" width="700"></p>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># In the alignment to Okinawa, it is scrambled.</span></span>
<span><span class="va">coa2</span><span class="op">$</span><span class="va">Osa_Oki</span> <span class="op">|&gt;</span> <span class="fu">plyranges</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">seqnames</span> <span class="op">==</span> <span class="st">"PAR"</span>, <span class="va">start</span> <span class="op">&gt;</span> <span class="fl">9e6</span>, <span class="va">end</span> <span class="op">&lt;</span> <span class="fl">14e6</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/plotApairOfChrs.html" class="external-link">plotApairOfChrs</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p><img src="Inversions_files/figure-html/unnamed-chunk-1-8.png" width="700"></p>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Aleksandra Bliznina, Michael Mansfield, Charles Plessy.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
