<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Introduction to the OikScrambling package • OikScrambling</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Introduction to the OikScrambling package">
<meta property="og:description" content="OikScrambling">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">OikScrambling</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.3.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/OikScrambling.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/ColinearityInterruptors.html">Regions interrupting colinearity</a>
    </li>
    <li>
      <a href="../articles/DistanceMeasures.html">Distance Measures</a>
    </li>
    <li>
      <a href="../articles/GenomicFeatures.html">Introduction to the OikScrambling package</a>
    </li>
    <li>
      <a href="../articles/Inversions.html">Inversions</a>
    </li>
    <li>
      <a href="../articles/LoadGenomicBreaks.html">Load genomic break data</a>
    </li>
    <li>
      <a href="../articles/OrthogroupSyntenies.html">Orthogroup syntenies</a>
    </li>
    <li>
      <a href="../articles/PWM.html">Motif finding and analysis</a>
    </li>
    <li>
      <a href="../articles/ParallelPlots.html">Whole-chromosome parallel plots</a>
    </li>
    <li>
      <a href="../articles/RegionWidths.html">Width of the aligned and mapped regions.</a>
    </li>
    <li>
      <a href="../articles/RepeatRegions.html">Repeat regions</a>
    </li>
    <li>
      <a href="../articles/Translocations.html">Translocations</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/oist/LuscombeU_OikScrambling/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="OikScrambling_files/header-attrs-2.11/header-attrs.js"></script><script src="OikScrambling_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Introduction to the OikScrambling package</h1>
                        <h4 class="author">Charlotte West</h4>
                        <h4 class="author">Charles Plessy</h4>
            
            <h4 class="date">18 November, 2021</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/oist/LuscombeU_OikScrambling/blob/master/vignettes/OikScrambling.Rmd"><code>vignettes/OikScrambling.Rmd</code></a></small>
      <div class="hidden name"><code>OikScrambling.Rmd</code></div>

    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">knitr</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/knitr/man/opts_chunk.html">opts_chunk</a></span><span class="op">$</span><span class="fu">set</span><span class="op">(</span>cache <span class="op">=</span> <span class="cn">TRUE</span>, cache.lazy <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
<span class="fu">knitr</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/knitr/man/opts_knit.html">opts_knit</a></span><span class="op">$</span><span class="fu">set</span><span class="op">(</span>verbose <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<div id="introduction" class="section level1">
<h1 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h1>
<p>This package contains a collection of vignettes to perform the analysis of pairwise genome alignments between <em>Oikopleura</em> genomes.</p>
<p>The core functions used here are maintained in our <em>GenomicBreaks</em> R package, which is fully documented at: <a href="https://oist.github.io/GenomicBreaks" class="uri">https://oist.github.io/GenomicBreaks</a>.</p>
<div id="how-to-build-this-site" class="section level2">
<h2 class="hasAnchor">
<a href="#how-to-build-this-site" class="anchor"></a>How to build this site</h2>
<p>As some of the computations take time, I build the vignettes on <em>Deigo</em> using the <code><a href="https://pkgdown.r-lib.org/reference/build_site.html">?pkgdown::build_site</a></code> function, in a Git clone on my subdirectory on <code>/flash</code>.</p>
<pre><code>/bucket/LuscombeU/common/Singularity/GenomicBreaks-0.9.1.sif Rscript -e 'pkgdown::build_site(devel=TRUE, lazy=TRUE)'</code></pre>
<p>Sometimes it is necessary to upgrade the OikScrambling package; this is done with:</p>
<pre><code>/bucket/LuscombeU/common/Singularity/GenomicBreaks-0.9.1.sif R CMD INSTALL .</code></pre>
<p>And to update <code>GenomicBreaks</code>:</p>
<pre><code>/bucket/LuscombeU/common/Singularity/GenomicBreaks-0.9.1.sif Rscript -e "remotes::install_github('oist/GenomicBreaks', repos=BiocManager::repositories())"</code></pre>
<p>The trick is that the <code>GenomicBreaks</code> singularity images are hardcoded to install new packages in <code>R/library/GBreaks_singularity</code>, which bypasses the the problem that the images are read-only.</p>
</div>
<div id="load-pacakges" class="section level2">
<h2 class="hasAnchor">
<a href="#load-pacakges" class="anchor"></a>Load pacakges</h2>
<div id="core-packages-that-provide-functions-we-use-a-lot-" class="section level3">
<h3 class="hasAnchor">
<a href="#core-packages-that-provide-functions-we-use-a-lot-" class="anchor"></a>Core packages that provide functions we use a lot.</h3>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/message.html">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="op">{</span>
  <span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="st"><a href="https://oist.github.io/GenomicBreaks">'GenomicBreaks'</a></span><span class="op">)</span>
  <span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="st"><a href="https://bioconductor.org/packages/BSgenome">'BSgenome'</a></span><span class="op">)</span>
  <span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="st"><a href="https://bioconductor.org/packages/GenomicFeatures">'GenomicFeatures'</a></span><span class="op">)</span>
  <span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="st"><a href="https://ggplot2.tidyverse.org">'ggplot2'</a></span><span class="op">)</span>
  <span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/Bioconductor/BiocParallel">"BiocParallel"</a></span><span class="op">)</span>
<span class="op">}</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/ns-load.html">requireNamespace</a></span><span class="op">(</span><span class="st">"rGADEM"</span><span class="op">)</span></code></pre></div>
<pre><code>## Loading required namespace: rGADEM</code></pre>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html">requireNamespace</a></span><span class="op">(</span><span class="st">"ggseqlogo"</span><span class="op">)</span></code></pre></div>
<pre><code>## Loading required namespace: ggseqlogo</code></pre>
</div>
<div id="bsgenome-packages" class="section level3">
<h3 class="hasAnchor">
<a href="#bsgenome-packages" class="anchor"></a>BSgenome packages</h3>
<p>We need the sequences of the studied genomes. They are provided in a <em>drat</em> repository: <a href="https://oist.github.io/plessy_oikgenomes_drat/" class="uri">https://oist.github.io/plessy_oikgenomes_drat/</a>.</p>
<p>Example command to install one <em>Oikopleura</em> BSgenome package if you do not yet have it:</p>
<pre><code><a href="https://rdrr.io/r/utils/install.packages.html">install.packages("BSgenome.Odioica.local.OKI2018.I69", repos="https://oist.github.io/plessy_oikgenomes_drat/")</a></code></pre>
<p>The accessory function <code>OikScrambling:::loadAllGenomes()</code> attaches <code>BSgenome</code> packages in the session and outputs a <code>SimpleList</code> containing references to the <code>BSgenome</code> objects for convenience, such as easy looping, and tab-completion to show the available objects.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">OikScrambling</span><span class="fu">:::</span><span class="va"><a href="../reference/loadAllGenomes.html">loadAllGenomes</a></span></code></pre></div>
<pre><code>## function () 
## {
##     library("BSgenome.Odioica.local.OKI2018.I69")
##     library("BSgenome.Odioica.local.OSKA2016v1.9")
##     library("BSgenome.Odioica.local.Bar2.p4")
##     library("BSgenome.Odioica.local.KUM.M3")
##     library("BSgenome.Odioica.local.AOM.5")
##     library("BSgenome.Odioica.local.Odioica.reference.v3.0")
##     S4Vectors::SimpleList(Oki = OKI2018_I69, Osa = OSKA2016v1.9, 
##         Bar = Bar2_p4, Kum = KUM_M3, Aom = AOM_5, Nor = OdB3)
## }
## &lt;bytecode: 0x5589758a6d20&gt;
## &lt;environment: namespace:OikScrambling&gt;</code></pre>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">genomes</span> <span class="op">&lt;-</span> <span class="op">(</span><span class="va">genomes</span> <span class="op">&lt;-</span> <span class="fu">OikScrambling</span><span class="fu">:::</span><span class="fu"><a href="../reference/loadAllGenomes.html">loadAllGenomes</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span>

<span class="co">## Genome lengths</span>
<span class="op">(</span><span class="va">genome.lengths</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="va">genomes</span>, \<span class="op">(</span><span class="va">g</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="fu">seqlengths</span><span class="op">(</span><span class="fu">seqinfo</span><span class="op">(</span><span class="va">g</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code>##      Oki      Osa      Bar      Kum      Aom      Nor 
## 64281565 56625162 55793437 64653574 56753784 70471451</code></pre>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## Genome average AT content (multicore computation)</span>
<span class="op">(</span><span class="va">genomes.AT</span> <span class="op">&lt;-</span> <span class="fu">BiocParallel</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocParallel/man/bplapply.html">bplapply</a></span><span class="op">(</span><span class="va">genomes</span>, \<span class="op">(</span><span class="va">g</span><span class="op">)</span>
  <span class="fu"><a href="https://rdrr.io/r/stats/weighted.mean.html">weighted.mean</a></span><span class="op">(</span><span class="fu">letterFrequency</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BSgenome/man/getSeq-methods.html">getSeq</a></span><span class="op">(</span><span class="va">g</span><span class="op">)</span>, <span class="st">"AT"</span>, as.prob <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>, <span class="fu">seqlengths</span><span class="op">(</span><span class="va">g</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/r/base/unlist.html">unlist</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code>##       Oki       Osa       Bar       Kum       Aom       Nor 
## 0.5893477 0.5831737 0.6004930 0.5891578 0.5852745 0.5686432</code></pre>
</div>
</div>
<div id="load-data" class="section level2">
<h2 class="hasAnchor">
<a href="#load-data" class="anchor"></a>Load data</h2>
<div id="genome-annotations" class="section level3">
<h3 class="hasAnchor">
<a href="#genome-annotations" class="anchor"></a>Genome annotations</h3>
<p>The annotations from <code>/bucket/LuscombeU/common/Breakpoints/Annotations</code> are distributed in the same drat repository as the alignment data.</p>
<pre><code><a href="https://rdrr.io/r/utils/install.packages.html">install.packages("BreakpointsData", repos="https://oist.github.io/plessy_oikgenomes_drat/")</a></code></pre>
<p>The <code>OikScrambling:::loadAllAnnotations()</code> function returns <code>TxDB</code> objects. Be careful that they are broken each time the R session restarts, because they rely on local file storage.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="op">(</span><span class="va">annots</span> <span class="op">&lt;-</span> <span class="fu">OikScrambling</span><span class="fu">:::</span><span class="fu"><a href="../reference/loadAllAnnotations.html">loadAllAnnotations</a></span><span class="op">(</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/r/base/warning.html">suppressWarnings</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code>## List of length 6
## names(6): Oki Osa Bar Kum Aom Nor</code></pre>
<p>For vignettes that do not use information on the intron/exon structure of genes, the <code>OikScrambling:::loadAllTranscriptsGR()</code> returns a <code>SimpleList</code> of <code>GRanges</code>, which is more convenient. Bonus point: the dNdS value for conserved genes is included.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="op">(</span><span class="va">transcripts</span> <span class="op">&lt;-</span> <span class="fu">OikScrambling</span><span class="fu">:::</span><span class="fu">loadAllTranscriptsGR</span><span class="op">(</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/r/base/warning.html">suppressWarnings</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code>## List of length 6
## names(6): Oki Osa Bar Kum Aom Nor</code></pre>
</div>
<div id="genomicbreaks-object" class="section level3">
<h3 class="hasAnchor">
<a href="#genomicbreaks-object" class="anchor"></a>GenomicBreaks object</h3>
<p>The core data is a list of <code><a href="https://oist.github.io/GenomicBreaks/reference/GBreaks-class.html">GenomicBreaks::GBreaks()</a></code> objects that is built by a separate vignette (<code>vignette("Load genomic break data"), package="OikScrambling"</code>).</p>
<p>The <em>query</em> genome is the one that was provided as a FASTA file and aligned to the <em>target</em> genome (the one that was indexed by the aligner). The <em>target</em> genome is the main part of the object, and the <em>query</em> genome information is contained in the metadata columns (<code>mcols</code>) of the structure.</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/load.html">load</a></span><span class="op">(</span><span class="st">"BreakPoints.Rdata"</span><span class="op">)</span>
<span class="va">gbs</span><span class="op">$</span><span class="va">Oki_Osa</span></code></pre></div>
<pre><code>## GBreaks object with 30852 ranges and 6 metadata columns:
##           seqnames          ranges strand |     score                 query
##              &lt;Rle&gt;       &lt;IRanges&gt;  &lt;Rle&gt; | &lt;numeric&gt;             &lt;GRanges&gt;
##       [1]     chr1   152288-154766      - |      2142 PAR:15511940-15514448
##       [2]     chr1   217240-217647      + |       899   PAR:2352378-2352781
##       [3]     chr1   300272-301113      + |      1163  Chr2:1044232-1044999
##       [4]     chr1   476328-477468      - |       915  Chr2:3294310-3295456
##       [5]     chr1   511233-512037      - |       295   YSR:1698314-1699115
##       ...      ...             ...    ... .       ...                   ...
##   [30848]      YSR 2759988-2760289      - |       332  Chr2:9474939-9475240
##   [30849]      YSR 2760320-2761360      - |       754  Chr2:9473557-9474600
##   [30850]      YSR 2783954-2785297      - |      2677  Chr2:9464587-9465931
##   [30851]      YSR 2790986-2795531      - |      4996  Chr2:9459905-9464402
##   [30852]      YSR 2832349-2836860      - |      7232   YSR:1771962-1776463
##                Arm             rep genes        flag
##           &lt;factor&gt; &lt;CharacterList&gt; &lt;Rle&gt; &lt;character&gt;
##       [1]    short             rnd  &lt;NA&gt;        &lt;NA&gt;
##       [2]    short            &lt;NA&gt;    g4        &lt;NA&gt;
##       [3]    short            &lt;NA&gt;  &lt;NA&gt;        &lt;NA&gt;
##       [4]    short            &lt;NA&gt;  &lt;NA&gt;        &lt;NA&gt;
##       [5]    short            &lt;NA&gt;  &lt;NA&gt;        &lt;NA&gt;
##       ...      ...             ...   ...         ...
##   [30848]      YSR            &lt;NA&gt;  &lt;NA&gt;         Col
##   [30849]      YSR            &lt;NA&gt;  &lt;NA&gt;         Col
##   [30850]      YSR            &lt;NA&gt;  &lt;NA&gt;         Col
##   [30851]      YSR           ltr-1  &lt;NA&gt;        &lt;NA&gt;
##   [30852]      YSR           ltr-1  &lt;NA&gt;        &lt;NA&gt;
##   -------
##   seqinfo: 19 sequences from OKI2018_I69 genome</code></pre>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">gbs</span><span class="op">$</span><span class="va">Oki_Osa</span><span class="op">$</span><span class="va">query</span></code></pre></div>
<pre><code>## GRanges object with 30852 ranges and 2 metadata columns:
##           seqnames            ranges strand |             rep       genes
##              &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; | &lt;CharacterList&gt;       &lt;Rle&gt;
##       [1]      PAR 15511940-15514448      * |            &lt;NA&gt;        &lt;NA&gt;
##       [2]      PAR   2352378-2352781      * |            &lt;NA&gt;       g7612
##       [3]     Chr2   1044232-1044999      * |            &lt;NA&gt;       g3324
##       [4]     Chr2   3294310-3295456      * |             rnd        &lt;NA&gt;
##       [5]      YSR   1698314-1699115      * |            &lt;NA&gt;      g15233
##       ...      ...               ...    ... .             ...         ...
##   [30848]     Chr2   9474939-9475240      * |            &lt;NA&gt;       g5781
##   [30849]     Chr2   9473557-9474600      * |            &lt;NA&gt;       g5781
##   [30850]     Chr2   9464587-9465931      * |            &lt;NA&gt;       g5779
##   [30851]     Chr2   9459905-9464402      * |            &lt;NA&gt; g5777;g5778
##   [30852]      YSR   1771962-1776463      * |            &lt;NA&gt;      g15241
##   -------
##   seqinfo: 483 sequences from OSKA2016v1.9 genome</code></pre>
<p>In the example displayed above, the object contains the pairwise alignment between genomes of two dioceous <em>Oikopleura</em> individuals; one from the Osaka assembly <code>OSKA2016</code> (<em>target</em>) and one from the Okinawa assembly <code>OKI2018_I69</code> (<em>query</em>).</p>
</div>
<div id="long-and-short-arms" class="section level3">
<h3 class="hasAnchor">
<a href="#long-and-short-arms" class="anchor"></a>Long and short arms</h3>
<p>The long and short arms are annotated in the <code>GBreak</code> objects that are built by the vignette (<code>vignette("Load genomic break data")</code>), using the function <code><a href="https://oist.github.io/GenomicBreaks/reference/flagLongShort.html">GenomicBreaks::flagLongShort()</a></code>. A copy of the arms annotations is provided in the <code>longShort</code> list.</p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">longShort</span></code></pre></div>
<pre><code>## List of length 3
## names(3): Oki Osa Bar</code></pre>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">longShort</span><span class="op">$</span><span class="va">Oki</span></code></pre></div>
<pre><code>## GRanges object with 8 ranges and 2 metadata columns:
##       seqnames           ranges strand |              Name        Type
##          &lt;Rle&gt;        &lt;IRanges&gt;  &lt;Rle&gt; |       &lt;character&gt; &lt;character&gt;
##   [1]     chr1        1-5191657      * |    chr1 short arm       short
##   [2]     chr1 5192156-14533022      * |     chr1 long arm        long
##   [3]     chr2        1-5707009      * |    chr2 short arm       short
##   [4]     chr2 5707508-16158756      * |     chr2 long arm        long
##   [5]      PAR        1-6029625      * |     PAR short arm       short
##   [6]      PAR 6030124-17092476      * |      PAR long arm        long
##   [7]      XSR       1-12959145      * | X-specific region         XSR
##   [8]      YSR        1-2916375      * | Y-specific region         YSR
##   -------
##   seqinfo: 19 sequences from OKI2018_I69 genome</code></pre>
</div>
<div id="repeats" class="section level3">
<h3 class="hasAnchor">
<a href="#repeats" class="anchor"></a>Repeats</h3>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="op">(</span><span class="va">reps</span> <span class="op">&lt;-</span> <span class="fu">OikScrambling</span><span class="fu">:::</span><span class="fu"><a href="../reference/loadAllRepeats.html">loadAllRepeats</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code>## List of length 6
## names(6): Oki Osa Bar Kum Aom Nor</code></pre>
</div>
</div>
<div id="breakpoints-and-alignment-stops" class="section level2">
<h2 class="hasAnchor">
<a href="#breakpoints-and-alignment-stops" class="anchor"></a>Breakpoints and alignment stops</h2>
<p>We define an <strong>alignment stop</strong> to be a position defined in either the <em>target</em> or <em>query</em> genome, where an alignment begins or ends, and a <strong>genomic breakpoint</strong> (or simply breakpoint) to be a genomic structural mutation, arising from breakage and repair of the chromosome. Such structural events include insertion, deletion, inversion and translocation, and often arise during recombination.</p>
</div>
</div>
<div id="measures-of-distance" class="section level1">
<h1 class="hasAnchor">
<a href="#measures-of-distance" class="anchor"></a>Measures of distance</h1>
<p>See <code><a href="../articles/DistanceMeasures.html">vignette("DistanceMeasures", package="OikScrambling")</a></code>.</p>
</div>
<div id="phylogenetic-cladogram" class="section level1">
<h1 class="hasAnchor">
<a href="#phylogenetic-cladogram" class="anchor"></a>Phylogenetic cladogram</h1>
<p>We assume that the <em>North Pacific</em> and the <em>Atlantic</em> species are more related to each other than to the <em>Okinawan</em> species.</p>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html">requireNamespace</a></span><span class="op">(</span><span class="st">"ade4"</span><span class="op">)</span>
<span class="va">treeLeaf</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">name</span>, <span class="va">length</span><span class="op">=</span><span class="cn">NULL</span><span class="op">)</span> <span class="op">{</span>
  <span class="kw">if</span><span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NULL.html">is.null</a></span><span class="op">(</span><span class="va">length</span><span class="op">)</span><span class="op">)</span> <span class="va">length</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="st">':'</span>, <span class="va">length</span><span class="op">)</span>
  <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">name</span>, <span class="va">length</span><span class="op">)</span>
<span class="op">}</span>

<span class="va">treeNode</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">branch1</span>, <span class="va">branch2</span>, <span class="va">length</span> <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span> <span class="op">{</span>
  <span class="kw">if</span><span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NULL.html">is.null</a></span><span class="op">(</span><span class="va">length</span><span class="op">)</span><span class="op">)</span> <span class="va">length</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="st">':'</span>, <span class="va">length</span><span class="op">)</span>
  <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="st">'('</span>, <span class="va">branch1</span>, <span class="st">','</span>, <span class="va">branch2</span>, <span class="st">')'</span>, <span class="va">length</span><span class="op">)</span>
<span class="op">}</span>

<span class="va">addRoot</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">branch</span><span class="op">)</span>  <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">branch</span>, <span class="st">";"</span><span class="op">)</span>

<span class="va">tree</span> <span class="op">&lt;-</span>
  <span class="fu">addRoot</span><span class="op">(</span>
    <span class="fu">treeNode</span><span class="op">(</span>
      <span class="fu">treeNode</span><span class="op">(</span> length <span class="op">=</span> <span class="fl">2</span>,
        <span class="fu">treeLeaf</span><span class="op">(</span><span class="st">"Okinawa"</span>, <span class="fl">1</span><span class="op">)</span>,
        <span class="fu">treeLeaf</span><span class="op">(</span><span class="st">"Kume"</span>, <span class="fl">1</span><span class="op">)</span>
      <span class="op">)</span>,
      <span class="fu">treeNode</span><span class="op">(</span> length <span class="op">=</span> <span class="fl">1</span>,
        <span class="fu">treeNode</span><span class="op">(</span> length <span class="op">=</span> <span class="fl">1</span>,
          <span class="fu">treeLeaf</span><span class="op">(</span><span class="st">"Osaka"</span>, <span class="fl">1</span><span class="op">)</span>,
          <span class="fu">treeLeaf</span><span class="op">(</span><span class="st">"Aomori"</span>, <span class="fl">1</span><span class="op">)</span>
        <span class="op">)</span>,
        <span class="fu">treeNode</span><span class="op">(</span> length <span class="op">=</span><span class="fl">1</span>,
          <span class="fu">treeLeaf</span><span class="op">(</span><span class="st">"Norway"</span>, <span class="fl">1</span><span class="op">)</span>,
          <span class="fu">treeLeaf</span><span class="op">(</span><span class="st">"Barcelona"</span>, <span class="fl">1</span><span class="op">)</span>
        <span class="op">)</span>
      <span class="op">)</span>
    <span class="op">)</span>
  <span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html">plot</a></span><span class="op">(</span><span class="fu">ade4</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ade4/man/newick2phylog.html">newick2phylog</a></span><span class="op">(</span><span class="va">tree</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="OikScrambling_files/figure-html/phylogenetic_tree-1.png" width="576"></p>
</div>
<div id="coalescing-alignments" class="section level1">
<h1 class="hasAnchor">
<a href="#coalescing-alignments" class="anchor"></a>Coalescing alignments</h1>
<p>Large syntenic regions can often appear cluttered with alignment breaks. They are either an artefact (for instance in case of incomplete purge of haplotypes) or true breakpoint. Even in that case, it may be needed to coalese them in order to facilitate other computations such as the detection of inversions.</p>
<p>The algorithm in <code><a href="https://oist.github.io/GenomicBreaks/reference/coalesce_contigs.html">GenomicBreaks::coalesce_contigs()</a></code> is used to produce a new <code>GBreaks</code> object with fewer alignment breaks by coalescing alignments separated by short (user specified) distances.</p>
<p>The computation is done in <code>vignette("Load genomic break data")</code></p>
<p>The resulting GRanges object has far fewer alignments and therefore far fewer alignment stops. The algorithm is an initial step in alignment stop filtering, with the goal of a reduced number of alignment stops that have a high probability of being breakpoints.</p>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="va">gbs</span>, <span class="va">length</span><span class="op">)</span></code></pre></div>
<pre><code>## Oki_Osa Oki_Bar Oki_Kum Oki_Aom Oki_Nor Osa_Oki Osa_Bar Osa_Kum Osa_Aom Osa_Nor 
##   30852   30841   15086   32398   30055   31322   24430   31182   15261   24497 
## Bar_Oki Bar_Osa Bar_Kum Bar_Aom Bar_Nor 
##   32265   24851   31557   24717   15184</code></pre>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="va">coa</span>, <span class="va">length</span><span class="op">)</span></code></pre></div>
<pre><code>## Oki_Osa Oki_Bar Oki_Kum Oki_Aom Oki_Nor Osa_Oki Osa_Bar Osa_Kum Osa_Aom Osa_Nor 
##    8843    8533    5424    8864    8875    8836    4081    8776    4950    5227 
## Bar_Oki Bar_Osa Bar_Kum Bar_Aom Bar_Nor 
##    8493    4073    8529    4197    6188</code></pre>
</div>
<div id="width-summary-plots" class="section level1">
<h1 class="hasAnchor">
<a href="#width-summary-plots" class="anchor"></a>Width summary plots</h1>
<p>See <code>vignette("RegionWidths"), package="OikScrambling"</code>.</p>
</div>
<div id="repeats-1" class="section level1">
<h1 class="hasAnchor">
<a href="#repeats-1" class="anchor"></a>Repeats</h1>
<p>See <code>vignette("RepeatRegions"), package="OikScrambling"</code>.</p>
</div>
<div id="pairwise-chromosome-plots" class="section level1">
<h1 class="hasAnchor">
<a href="#pairwise-chromosome-plots" class="anchor"></a>Pairwise chromosome plots</h1>
<div id="oki-bar" class="section level2">
<h2 class="hasAnchor">
<a href="#oki-bar" class="anchor"></a>Oki – Bar</h2>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/plotApairOfChrs.html">plotApairOfChrs</a></span><span class="op">(</span><span class="va">coa</span><span class="op">$</span><span class="va">Oki_Bar</span>, <span class="st">"chr1"</span>, main <span class="op">=</span> <span class="st">"Oki – Bar"</span><span class="op">)</span></code></pre></div>
<p><img src="OikScrambling_files/figure-html/parallel_chr_plots_Oki_Bar-1.png" width="700"></p>
</div>
<div id="osa-bar" class="section level2">
<h2 class="hasAnchor">
<a href="#osa-bar" class="anchor"></a>Osa – Bar</h2>
<div class="sourceCode" id="cb37"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/plotApairOfChrs.html">plotApairOfChrs</a></span><span class="op">(</span><span class="va">coa</span><span class="op">$</span><span class="va">Osa_Bar</span>, <span class="st">"Chr1"</span>, main <span class="op">=</span> <span class="st">"Osa – Bar"</span><span class="op">)</span></code></pre></div>
<p><img src="OikScrambling_files/figure-html/parallel_chr_plots_Osa_Bar-1.png" width="700"></p>
</div>
<div id="oki-kum" class="section level2">
<h2 class="hasAnchor">
<a href="#oki-kum" class="anchor"></a>Oki – Kum</h2>
<p>The plot is not complete because the Kume assembly is not chromosome-scale.</p>
<div class="sourceCode" id="cb38"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/plotApairOfChrs.html">plotApairOfChrs</a></span><span class="op">(</span><span class="va">coa</span><span class="op">$</span><span class="va">Oki_Kum</span>, <span class="st">"chr1"</span>, main <span class="op">=</span> <span class="st">"Oki – Kum"</span><span class="op">)</span></code></pre></div>
<p><img src="OikScrambling_files/figure-html/parallel_chr_plots_Oki_Kum-1.png" width="700"></p>
<p>Same after double-coalescing.</p>
<div class="sourceCode" id="cb39"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/plotApairOfChrs.html">plotApairOfChrs</a></span><span class="op">(</span><span class="va">coa2</span><span class="op">$</span><span class="va">Oki_Kum</span>, <span class="st">"chr1"</span>, main <span class="op">=</span> <span class="st">"Oki – Kum"</span><span class="op">)</span></code></pre></div>
<p><img src="OikScrambling_files/figure-html/parallel_chr_plots_Oki_Kum_coa2-1.png" width="700"></p>
</div>
<div id="more-in-the-parallel-plots-vignette" class="section level2">
<h2 class="hasAnchor">
<a href="#more-in-the-parallel-plots-vignette" class="anchor"></a>More in the <em>parallel plots</em> vignette!</h2>
<p>In the <em>parallel plots</em> vignette (<code><a href="../articles/ParallelPlots.html">vignette("ParallelPlots", package = "OikScrambling")</a></code>), we build a more more comprehensive collection of plots and explore arrangements for a figure panel in paper.</p>
</div>
</div>
<div id="structural-variants" class="section level1">
<h1 class="hasAnchor">
<a href="#structural-variants" class="anchor"></a>Structural variants</h1>
<div id="trivial-inversions" class="section level2">
<h2 class="hasAnchor">
<a href="#trivial-inversions" class="anchor"></a>Trivial inversions</h2>
<div id="documentation" class="section level3">
<h3 class="hasAnchor">
<a href="#documentation" class="anchor"></a>Documentation</h3>
<p>Details can be found in <code><a href="../articles/Inversions.html">vignette("Inversions", package = "OikScrambling")</a></code>.</p>
</div>
</div>
<div id="trivial-translocations" class="section level2">
<h2 class="hasAnchor">
<a href="#trivial-translocations" class="anchor"></a>Trivial translocations</h2>
<div id="representation" class="section level3">
<h3 class="hasAnchor">
<a href="#representation" class="anchor"></a>Representation</h3>
<p>The translocation below:</p>
<pre><code>┌──────────────┬──────────────┬──────────────┐
│ chrA:101-200 │ chrA:201-300 │ chrA:301-400 │ (Target genome)
└──────────────┴──────────────┴──────────────┘
      +               +             +         (Alignment direction)
┌──────────────┬──────────────┬──────────────┐
│ chrB:101-200 │ chrC:201-300 │ chrB:301-400 │ (Query genome)
└──────────────┴──────────────┴──────────────┘</code></pre>
<p>Is represented as:</p>
<div class="sourceCode" id="cb41"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">exampleTranslocation</span></code></pre></div>
<pre><code>## GBreaks object with 3 ranges and 1 metadata column:
##       seqnames    ranges strand |        query
##          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; |    &lt;GRanges&gt;
##   [1]     chrA   100-200      + | chrB:100-200
##   [2]     chrA   201-300      + | chrC:201-300
##   [3]     chrA   301-400      + | chrB:301-400
##   -------
##   seqinfo: 1 sequence from an unspecified genome</code></pre>
<p>See in in <code><a href="../articles/LoadGenomicBreaks.html">vignette("LoadGenomicBreaks", package = "OikScrambling")</a></code> how trivial translocations are removed, and in <code><a href="../articles/RepeatRegions.html">vignette("RepeatRegions", package = "OikScrambling")</a></code> why they are removed.</p>
</div>
</div>
</div>
<div id="genomic-features" class="section level1">
<h1 class="hasAnchor">
<a href="#genomic-features" class="anchor"></a>Genomic Features</h1>
<p>Moved to <code>vignette("GenomicFeatuers", package = "OikScrambling")</code>:</p>
<ul>
<li>Tandem repeats</li>
<li>Coverage</li>
<li>Nucleic acid content heatmaps</li>
<li>Cluster analysis</li>
<li>Evidence for breakpoint hotspots</li>
<li>Tandem repeat coverage around alignment stops (breakpoints)</li>
<li>Gene feature coverage around alignment stops (breakpoints)</li>
<li>Gene feature coverage around PWM hits</li>
<li>Okinawa genome coverage</li>
</ul>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Charles Plessy.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
