<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Introduction to the OikScrambling package • OikScrambling</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Introduction to the OikScrambling package">
<meta property="og:description" content="OikScrambling">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">OikScrambling</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">3.0.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/OikScrambling.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/CNEs.html">Conserved non-coding elements</a>
    </li>
    <li>
      <a href="../articles/CoalescingCutoffs.html">Modifying coalescent cutoffs</a>
    </li>
    <li>
      <a href="../articles/ColinearityInterruptors.html">Regions interrupting colinearity</a>
    </li>
    <li>
      <a href="../articles/DistanceMeasures.html">Distance Measures</a>
    </li>
    <li>
      <a href="../articles/GenomicFeatures.html">Genomic features of the breakpoints</a>
    </li>
    <li>
      <a href="../articles/Inversions.html">Inversions</a>
    </li>
    <li>
      <a href="../articles/LoadGenomicBreaks.html">Load genomic break data</a>
    </li>
    <li>
      <a href="../articles/OrthogroupSyntenies.html">Orthogroup syntenies</a>
    </li>
    <li>
      <a href="../articles/OxfordPlots.html">Whole-genome dot-plots</a>
    </li>
    <li>
      <a href="../articles/PAC3.html">Region of interest: PAC3</a>
    </li>
    <li>
      <a href="../articles/PWM.html">Motif search and analysis</a>
    </li>
    <li>
      <a href="../articles/ParallelPlots.html">Whole-chromosome parallel plots</a>
    </li>
    <li>
      <a href="../articles/RegionWidths.html">Width of the aligned and mapped regions.</a>
    </li>
    <li>
      <a href="../articles/RegionsOfInterest.html">Regions of interest</a>
    </li>
    <li>
      <a href="../articles/RepeatRegions.html">Repeat regions</a>
    </li>
    <li>
      <a href="../articles/SegmentMovements.html">Segment movement distances</a>
    </li>
    <li>
      <a href="../articles/TFBS.html">Transcription factor binding sites</a>
    </li>
    <li>
      <a href="../articles/TandemSiteDuplications.html">Tandem site duplications</a>
    </li>
    <li>
      <a href="../articles/Translocations.html">Translocations</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/oist/LuscombeU_OikScrambling/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="OikScrambling_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Introduction to the OikScrambling package</h1>
                        <h4 data-toc-skip class="author">Charlotte West</h4>
                        <h4 data-toc-skip class="author">Charles Plessy</h4>
            
            <h4 data-toc-skip class="date">30 March, 2022</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/oist/LuscombeU_OikScrambling/blob/HEAD/vignettes/OikScrambling.Rmd" class="external-link"><code>vignettes/OikScrambling.Rmd</code></a></small>
      <div class="hidden name"><code>OikScrambling.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>This package contains a collection of vignettes to perform the analysis of pairwise genome alignments between <em>Oikopleura</em> genomes.</p>
<p>The core functions used here are maintained in our <em>GenomicBreaks</em> R package, which is fully documented at: <a href="https://oist.github.io/GenomicBreaks" class="external-link uri">https://oist.github.io/GenomicBreaks</a>.</p>
<div class="section level3">
<h3 id="how-to-build-this-site">How to build this site<a class="anchor" aria-label="anchor" href="#how-to-build-this-site"></a>
</h3>
<p>As some of the computations take time, we build the vignettes on OIST’s HPC cluster <em>Deigo</em> using the <code><a href="https://pkgdown.r-lib.org/reference/build_site.html" class="external-link">?pkgdown::build_site</a></code> function, in a Git clone in a subdirectory of the <code>/flash</code> filesytem.</p>
<pre><code>/bucket/LuscombeU/common/Singularity/GenomicBreaks-0.10.0.sif Rscript -e "pkgdown::build_site(devel=TRUE, lazy=TRUE)"</code></pre>
</div>
<div class="section level3">
<h3 id="how-to-install-or-update-the-necessary-packages">How to install or update the necessary packages<a class="anchor" aria-label="anchor" href="#how-to-install-or-update-the-necessary-packages"></a>
</h3>
<div class="section level4">
<h4 id="oikscrambling-and-genomicbreaks">OikScrambling and GenomicBreaks<a class="anchor" aria-label="anchor" href="#oikscrambling-and-genomicbreaks"></a>
</h4>
<p>Sometimes it is necessary to upgrade the OikScrambling package; this is done with:</p>
<pre><code>/bucket/LuscombeU/common/Singularity/GenomicBreaks-0.10.0.sif R CMD INSTALL .</code></pre>
<p>And before, it may be needed to update <code>GenomicBreaks</code> or <code>BreakpointsData</code>:</p>
<pre><code>/bucket/LuscombeU/common/Singularity/GenomicBreaks-0.10.0.sif Rscript -e "remotes::install_github('oist/GenomicBreaks', repos=BiocManager::repositories())"
/bucket/LuscombeU/common/Singularity/GenomicBreaks-0.10.0.sif Rscript -e "install.packages('BreakpointsData', repos='https://oist.github.io/plessy_oikgenomes_drat/')"</code></pre>
<p>The trick is that the <code>GenomicBreaks</code> singularity images are hardcoded to install new packages in <code>R/library/GBreaks_singularity</code>, which bypasses the the problem that the images are read-only.</p>
<p>Command to recompute the <code>BreakPoints.Rdata</code> file</p>
<pre><code>srun -pcompute -c8 --mem 300G --pty /bucket/LuscombeU/common/Singularity/GenomicBreaks-0.10.0.sif Rscript -e "pkgdown::build_article('LoadGenomicBreaks')"</code></pre>
</div>
<div class="section level4">
<h4 id="data-packages">Data packages<a class="anchor" aria-label="anchor" href="#data-packages"></a>
</h4>
<div class="section level5">
<h5 id="bsgenome-packages">BSgenome packages<a class="anchor" aria-label="anchor" href="#bsgenome-packages"></a>
</h5>
<p>We need the sequences of the studied genomes. They are provided in a <em>drat</em> repository: <a href="https://oist.github.io/plessy_oikgenomes_drat/" class="external-link uri">https://oist.github.io/plessy_oikgenomes_drat/</a>.</p>
<p>Example command to install one <em>Oikopleura</em> BSgenome package if you do not yet have it:</p>
<pre><code>/bucket/LuscombeU/common/Singularity/GenomicBreaks-0.10.0.sif Rscript -e "install.packages('BSgenome.Odioica.local.OKI2018.I69', repos='https://oist.github.io/plessy_oikgenomes_drat/')"</code></pre>
</div>
<div class="section level5">
<h5 id="genome-annotations">Genome annotations<a class="anchor" aria-label="anchor" href="#genome-annotations"></a>
</h5>
<p>The annotations from <code>/bucket/LuscombeU/common/Breakpoints/Annotations</code> are distributed in the same drat repository as the alignment data.</p>
<pre><code><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">'BreakpointsData'</span>, repos<span class="op">=</span><span class="st">'https://oist.github.io/plessy_oikgenomes_drat/'</span><span class="op">)</span></code></pre>
</div>
</div>
</div>
<div class="section level3">
<h3 id="load-pacakges">Load pacakges<a class="anchor" aria-label="anchor" href="#load-pacakges"></a>
</h3>
<div class="section level4">
<h4 id="core-packages-that-provide-functions-we-use-a-lot-">Core packages that provide functions we use a lot.<a class="anchor" aria-label="anchor" href="#core-packages-that-provide-functions-we-use-a-lot-"></a>
</h4>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="op">{</span>
  <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://oist.github.io/GenomicBreaks" class="external-link">'GenomicBreaks'</a></span><span class="op">)</span>
  <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://bioconductor.org/packages/BSgenome" class="external-link">'BSgenome'</a></span><span class="op">)</span>
  <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://bioconductor.org/packages/GenomicFeatures" class="external-link">'GenomicFeatures'</a></span><span class="op">)</span>
  <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://ggplot2.tidyverse.org" class="external-link">'ggplot2'</a></span><span class="op">)</span>
  <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/Bioconductor/BiocParallel" class="external-link">"BiocParallel"</a></span><span class="op">)</span>
<span class="op">}</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"rGADEM"</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Loading required namespace: rGADEM</span></code></pre>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"ggseqlogo"</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Loading required namespace: ggseqlogo</span></code></pre>
<p>The accessory function <code>OikScrambling:::loadAllGenomes()</code> attaches <code>BSgenome</code> packages in the session and outputs a <code>SimpleList</code> containing references to the <code>BSgenome</code> objects for convenience, such as easy looping, and tab-completion to show the available objects.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">OikScrambling</span><span class="fu">:::</span><span class="va"><a href="../reference/loadAllGenomes.html">loadAllGenomes</a></span></code></pre></div>
<pre><code><span class="co">## function (compat = TRUE) </span>
<span class="co">## {</span>
<span class="co">##     suppressPackageStartupMessages({</span>
<span class="co">##         library("BSgenome.Oidioi.OIST.OKI2018.I69")</span>
<span class="co">##         library("BSgenome.Oidioi.OIST.OSKA2016v1.9")</span>
<span class="co">##         library("BSgenome.Oidioi.OIST.Bar2.p4")</span>
<span class="co">##         library("BSgenome.Oidioi.OIST.KUM.M3.7f")</span>
<span class="co">##         library("BSgenome.Oidioi.OIST.AOM.5.5f")</span>
<span class="co">##         library("BSgenome.Oidioi.genoscope.OdB3")</span>
<span class="co">##     })</span>
<span class="co">##     genomes &lt;- SimpleList(sapply(c("OKI2018.I69", "OSKA2016v1.9", </span>
<span class="co">##         "Bar2.p4", "KUM.M3.7f", "AOM.5.5f", "OdB3"), BSgenome::getBSgenome))</span>
<span class="co">##     if (isTRUE(compat)) </span>
<span class="co">##         names(genomes) &lt;- c("Oki", "Osa", "Bar", "Kum", "Aom", </span>
<span class="co">##             "Nor")</span>
<span class="co">##     genomes</span>
<span class="co">## }</span>
<span class="co">## &lt;bytecode: 0x55ac458eec90&gt;</span>
<span class="co">## &lt;environment: namespace:OikScrambling&gt;</span></code></pre>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="op">(</span><span class="va">genomes</span> <span class="op">&lt;-</span> <span class="fu">OikScrambling</span><span class="fu">:::</span><span class="fu"><a href="../reference/loadAllGenomes.html">loadAllGenomes</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## List of length 6</span>
<span class="co">## names(6): Oki Osa Bar Kum Aom Nor</span></code></pre>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## Genome lengths</span>
<span class="op">(</span><span class="va">genome.lengths</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">genomes</span>, \<span class="op">(</span><span class="va">g</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqlengths</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqinfo</a></span><span class="op">(</span><span class="va">g</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">##      Oki      Osa      Bar      Kum      Aom      Nor </span>
<span class="co">## 64281565 56625162 55793437 64653574 56753784 70471451</span></code></pre>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## Genome average AT content (multicore computation)</span>
<span class="op">(</span><span class="va">genomes.AT</span> <span class="op">&lt;-</span> <span class="fu">BiocParallel</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocParallel/man/bplapply.html" class="external-link">bplapply</a></span><span class="op">(</span><span class="va">genomes</span>, \<span class="op">(</span><span class="va">g</span><span class="op">)</span>
  <span class="fu"><a href="https://rdrr.io/r/stats/weighted.mean.html" class="external-link">weighted.mean</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/letterFrequency.html" class="external-link">letterFrequency</a></span><span class="op">(</span><span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/getSeq.html" class="external-link">getSeq</a></span><span class="op">(</span><span class="va">g</span><span class="op">)</span>, <span class="st">"AT"</span>, as.prob <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqlengths</a></span><span class="op">(</span><span class="va">g</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/unlist.html" class="external-link">unlist</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">##       Oki       Osa       Bar       Kum       Aom       Nor </span>
<span class="co">## 0.5893477 0.5831737 0.6004930 0.5891578 0.5852745 0.5686432</span></code></pre>
</div>
</div>
<div class="section level3">
<h3 id="load-data">Load data<a class="anchor" aria-label="anchor" href="#load-data"></a>
</h3>
<p>The <code>OikScrambling:::loadAllAnnotations()</code> function returns <code>TxDB</code> objects. Be careful that they are broken each time the R session restarts, because they rely on local file storage.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="op">(</span><span class="va">annots</span> <span class="op">&lt;-</span> <span class="fu">OikScrambling</span><span class="fu">:::</span><span class="fu"><a href="../reference/loadAllAnnotations.html">loadAllAnnotations</a></span><span class="op">(</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/r/base/warning.html" class="external-link">suppressWarnings</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## List of length 8</span>
<span class="co">## names(8): Oki Osa Bar Kum Aom Nor Ply Ros</span></code></pre>
<p>For vignettes that do not use information on the intron/exon structure of genes, the <code>OikScrambling:::loadAllTranscriptsGR()</code> returns a <code>SimpleList</code> of <code>GRanges</code>, which is more convenient. Bonus point: the dNdS value for conserved genes is included.</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="op">(</span><span class="va">transcripts</span> <span class="op">&lt;-</span> <span class="fu">OikScrambling</span><span class="fu">:::</span><span class="fu"><a href="../reference/loadAllTranscriptsGR.html">loadAllTranscriptsGR</a></span><span class="op">(</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/r/base/warning.html" class="external-link">suppressWarnings</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## List of length 8</span>
<span class="co">## names(8): Oki Osa Bar Kum Aom Nor Ply Ros</span></code></pre>
<div class="section level4">
<h4 id="genomicbreaks-object">GenomicBreaks object<a class="anchor" aria-label="anchor" href="#genomicbreaks-object"></a>
</h4>
<p>The core data is a list of <code><a href="https://oist.github.io/GenomicBreaks/reference/GBreaks-class.html" class="external-link">GenomicBreaks::GBreaks()</a></code> objects that is built by a separate vignette (<code>vignette("Load genomic break data"), package="OikScrambling"</code>).</p>
<p>The <em>query</em> genome is the one that was provided as a FASTA file and aligned to the <em>target</em> genome (the one that was indexed by the aligner). The <em>target</em> genome is the main part of the object, and the <em>query</em> genome information is contained in the metadata columns (<code>mcols</code>) of the structure.</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/load.html" class="external-link">load</a></span><span class="op">(</span><span class="st">"BreakPoints.Rdata"</span><span class="op">)</span>
<span class="va">gbs</span><span class="op">$</span><span class="va">Oki_Osa</span></code></pre></div>
<pre><code><span class="co">## GBreaks object with 30852 ranges and 8 metadata columns:</span>
<span class="co">##           seqnames          ranges strand |     score                 query</span>
<span class="co">##              &lt;Rle&gt;       &lt;IRanges&gt;  &lt;Rle&gt; | &lt;numeric&gt;             &lt;GRanges&gt;</span>
<span class="co">##       [1]     chr1   152288-154766      - |      2142 PAR:15511940-15514448</span>
<span class="co">##       [2]     chr1   217240-217647      + |       899   PAR:2352378-2352781</span>
<span class="co">##       [3]     chr1   300272-301113      + |      1163  Chr2:1044232-1044999</span>
<span class="co">##       [4]     chr1   476328-477468      - |       915  Chr2:3294310-3295456</span>
<span class="co">##       [5]     chr1   511233-512037      - |       295   YSR:1698314-1699115</span>
<span class="co">##       ...      ...             ...    ... .       ...                   ...</span>
<span class="co">##   [30848]      YSR 2759988-2760289      - |       332  Chr2:9474939-9475240</span>
<span class="co">##   [30849]      YSR 2760320-2761360      - |       754  Chr2:9473557-9474600</span>
<span class="co">##   [30850]      YSR 2783954-2785297      - |      2677  Chr2:9464587-9465931</span>
<span class="co">##   [30851]      YSR 2790986-2795531      - |      4996  Chr2:9459905-9464402</span>
<span class="co">##   [30852]      YSR 2832349-2836860      - |      7232   YSR:1771962-1776463</span>
<span class="co">##                Arm             rep   repOvlp transcripts        flag    nonCoa</span>
<span class="co">##           &lt;factor&gt; &lt;CharacterList&gt; &lt;integer&gt;       &lt;Rle&gt; &lt;character&gt; &lt;logical&gt;</span>
<span class="co">##       [1]    short             rnd      1820        &lt;NA&gt;        &lt;NA&gt;      TRUE</span>
<span class="co">##       [2]    short            &lt;NA&gt;         0       g4.t1        &lt;NA&gt;      TRUE</span>
<span class="co">##       [3]    short            &lt;NA&gt;         0        &lt;NA&gt;        &lt;NA&gt;      TRUE</span>
<span class="co">##       [4]    short            &lt;NA&gt;         0        &lt;NA&gt;        &lt;NA&gt;      TRUE</span>
<span class="co">##       [5]    short           ltr-1         0        &lt;NA&gt;        &lt;NA&gt;      TRUE</span>
<span class="co">##       ...      ...             ...       ...         ...         ...       ...</span>
<span class="co">##   [30848]      YSR            &lt;NA&gt;         0        &lt;NA&gt;         Col     FALSE</span>
<span class="co">##   [30849]      YSR            &lt;NA&gt;         0        &lt;NA&gt;         Col     FALSE</span>
<span class="co">##   [30850]      YSR            &lt;NA&gt;         0        &lt;NA&gt;         Col     FALSE</span>
<span class="co">##   [30851]      YSR    ltr-1,tandem      4396        &lt;NA&gt;        &lt;NA&gt;     FALSE</span>
<span class="co">##   [30852]      YSR           ltr-1      4512        &lt;NA&gt;        &lt;NA&gt;      TRUE</span>
<span class="co">##   -------</span>
<span class="co">##   seqinfo: 19 sequences from OKI2018.I69 genome</span></code></pre>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">gbs</span><span class="op">$</span><span class="va">Oki_Osa</span><span class="op">$</span><span class="va">query</span></code></pre></div>
<pre><code><span class="co">## GRanges object with 30852 ranges and 4 metadata columns:</span>
<span class="co">##           seqnames            ranges strand |      Arm             rep</span>
<span class="co">##              &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; | &lt;factor&gt; &lt;CharacterList&gt;</span>
<span class="co">##       [1]      PAR 15511940-15514448      * |    long             &lt;NA&gt;</span>
<span class="co">##       [2]      PAR   2352378-2352781      * |    short            &lt;NA&gt;</span>
<span class="co">##       [3]     Chr2   1044232-1044999      * |    short            &lt;NA&gt;</span>
<span class="co">##       [4]     Chr2   3294310-3295456      * |    short             rnd</span>
<span class="co">##       [5]      YSR   1698314-1699115      * |    YSR              &lt;NA&gt;</span>
<span class="co">##       ...      ...               ...    ... .      ...             ...</span>
<span class="co">##   [30848]     Chr2   9474939-9475240      * |     long            &lt;NA&gt;</span>
<span class="co">##   [30849]     Chr2   9473557-9474600      * |     long            &lt;NA&gt;</span>
<span class="co">##   [30850]     Chr2   9464587-9465931      * |     long            &lt;NA&gt;</span>
<span class="co">##   [30851]     Chr2   9459905-9464402      * |     long            &lt;NA&gt;</span>
<span class="co">##   [30852]      YSR   1771962-1776463      * |     YSR             &lt;NA&gt;</span>
<span class="co">##             repOvlp       transcripts</span>
<span class="co">##           &lt;integer&gt;             &lt;Rle&gt;</span>
<span class="co">##       [1]         0              &lt;NA&gt;</span>
<span class="co">##       [2]         0          g7612.t1</span>
<span class="co">##       [3]         0          g3324.t1</span>
<span class="co">##       [4]        63              &lt;NA&gt;</span>
<span class="co">##       [5]         0         g15233.t1</span>
<span class="co">##       ...       ...               ...</span>
<span class="co">##   [30848]         0          g5781.t1</span>
<span class="co">##   [30849]         0          g5781.t1</span>
<span class="co">##   [30850]         0          g5779.t1</span>
<span class="co">##   [30851]         0 g5777.t1;g5778.t1</span>
<span class="co">##   [30852]         0         g15241.t1</span>
<span class="co">##   -------</span>
<span class="co">##   seqinfo: 483 sequences from OSKA2016v1.9 genome</span></code></pre>
<p>In the example displayed above, the object contains the pairwise alignment between genomes of two dioceous <em>Oikopleura</em> individuals; one from the Osaka assembly <code>OSKA2016</code> (<em>target</em>) and one from the Okinawa assembly <code>OKI2018_I69</code> (<em>query</em>).</p>
</div>
<div class="section level4">
<h4 id="long-and-short-arms">Long and short arms<a class="anchor" aria-label="anchor" href="#long-and-short-arms"></a>
</h4>
<p>The long and short arms are annotated in the <code>GBreak</code> objects that are built by the vignette (<code>vignette("Load genomic break data")</code>), using the function <code><a href="https://oist.github.io/GenomicBreaks/reference/flagLongShort.html" class="external-link">GenomicBreaks::flagLongShort()</a></code>. A copy of the arms annotations is provided in the <code>longShort</code> list.</p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">longShort</span></code></pre></div>
<pre><code><span class="co">## List of length 3</span>
<span class="co">## names(3): OKI2018.I69 OSKA2016v1.9 Bar2.p4</span></code></pre>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">longShort</span><span class="op">$</span><span class="va">Oki</span></code></pre></div>
<pre><code><span class="co">## NULL</span></code></pre>
</div>
<div class="section level4">
<h4 id="repeats">Repeats<a class="anchor" aria-label="anchor" href="#repeats"></a>
</h4>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="op">(</span><span class="va">reps</span> <span class="op">&lt;-</span> <span class="fu">OikScrambling</span><span class="fu">:::</span><span class="fu"><a href="../reference/loadAllRepeats.html">loadAllRepeats</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## List of length 6</span>
<span class="co">## names(6): Oki Osa Bar Kum Aom Nor</span></code></pre>
</div>
</div>
<div class="section level3">
<h3 id="breakpoints-and-alignment-stops">Breakpoints and alignment stops<a class="anchor" aria-label="anchor" href="#breakpoints-and-alignment-stops"></a>
</h3>
<p>We define an <strong>alignment stop</strong> to be a position defined in either the <em>target</em> or <em>query</em> genome, where an alignment begins or ends, and a <strong>genomic breakpoint</strong> (or simply breakpoint) to be a genomic structural mutation, arising from breakage and repair of the chromosome. Such structural events include insertion, deletion, inversion and translocation, and often arise during recombination.</p>
</div>
</div>
<div class="section level2">
<h2 id="measures-of-distance">Measures of distance<a class="anchor" aria-label="anchor" href="#measures-of-distance"></a>
</h2>
<p>See <code><a href="../articles/DistanceMeasures.html">vignette("DistanceMeasures", package="OikScrambling")</a></code>.</p>
</div>
<div class="section level2">
<h2 id="phylogenetic-cladogram">Phylogenetic cladogram<a class="anchor" aria-label="anchor" href="#phylogenetic-cladogram"></a>
</h2>
<p>We assume that the <em>North Pacific</em> and the <em>Atlantic</em> species are more related to each other than to the <em>Okinawan</em> species.</p>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"ade4"</span><span class="op">)</span>
<span class="va">treeLeaf</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">name</span>, <span class="va">length</span><span class="op">=</span><span class="cn">NULL</span><span class="op">)</span> <span class="op">{</span>
  <span class="kw">if</span><span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NULL.html" class="external-link">is.null</a></span><span class="op">(</span><span class="va">length</span><span class="op">)</span><span class="op">)</span> <span class="va">length</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">':'</span>, <span class="va">length</span><span class="op">)</span>
  <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">name</span>, <span class="va">length</span><span class="op">)</span>
<span class="op">}</span>

<span class="va">treeNode</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">branch1</span>, <span class="va">branch2</span>, <span class="va">length</span> <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span> <span class="op">{</span>
  <span class="kw">if</span><span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NULL.html" class="external-link">is.null</a></span><span class="op">(</span><span class="va">length</span><span class="op">)</span><span class="op">)</span> <span class="va">length</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">':'</span>, <span class="va">length</span><span class="op">)</span>
  <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">'('</span>, <span class="va">branch1</span>, <span class="st">','</span>, <span class="va">branch2</span>, <span class="st">')'</span>, <span class="va">length</span><span class="op">)</span>
<span class="op">}</span>

<span class="va">addRoot</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">branch</span><span class="op">)</span>  <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">branch</span>, <span class="st">";"</span><span class="op">)</span>

<span class="va">tree</span> <span class="op">&lt;-</span>
  <span class="fu">addRoot</span><span class="op">(</span>
    <span class="fu">treeNode</span><span class="op">(</span>
      <span class="fu">treeNode</span><span class="op">(</span> length <span class="op">=</span> <span class="fl">2</span>,
        <span class="fu">treeLeaf</span><span class="op">(</span><span class="st">"Okinawa"</span>, <span class="fl">1</span><span class="op">)</span>,
        <span class="fu">treeLeaf</span><span class="op">(</span><span class="st">"Kume"</span>, <span class="fl">1</span><span class="op">)</span>
      <span class="op">)</span>,
      <span class="fu">treeNode</span><span class="op">(</span> length <span class="op">=</span> <span class="fl">1</span>,
        <span class="fu">treeNode</span><span class="op">(</span> length <span class="op">=</span> <span class="fl">1</span>,
          <span class="fu">treeLeaf</span><span class="op">(</span><span class="st">"Osaka"</span>, <span class="fl">1</span><span class="op">)</span>,
          <span class="fu">treeLeaf</span><span class="op">(</span><span class="st">"Aomori"</span>, <span class="fl">1</span><span class="op">)</span>
        <span class="op">)</span>,
        <span class="fu">treeNode</span><span class="op">(</span> length <span class="op">=</span><span class="fl">1</span>,
          <span class="fu">treeLeaf</span><span class="op">(</span><span class="st">"Norway"</span>, <span class="fl">1</span><span class="op">)</span>,
          <span class="fu">treeLeaf</span><span class="op">(</span><span class="st">"Barcelona"</span>, <span class="fl">1</span><span class="op">)</span>
        <span class="op">)</span>
      <span class="op">)</span>
    <span class="op">)</span>
  <span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="fu">ade4</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ade4/man/newick2phylog.html" class="external-link">newick2phylog</a></span><span class="op">(</span><span class="va">tree</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="OikScrambling_files/figure-html/phylogenetic_tree-1.png" width="576"></p>
</div>
<div class="section level2">
<h2 id="coalescing-alignments">Coalescing alignments<a class="anchor" aria-label="anchor" href="#coalescing-alignments"></a>
</h2>
<p>Large syntenic regions can often appear cluttered with alignment breaks. They are either an artefact (for instance in case of incomplete purge of haplotypes) or true breakpoint. Even in that case, it may be needed to coalese them in order to facilitate other computations such as the detection of inversions.</p>
<p>The algorithm in <code><a href="https://oist.github.io/GenomicBreaks/reference/coalesce_contigs.html" class="external-link">GenomicBreaks::coalesce_contigs()</a></code> is used to produce a new <code>GBreaks</code> object with fewer alignment breaks by coalescing alignments separated by short (user specified) distances.</p>
<p>The computation is done in <code>vignette("Load genomic break data")</code></p>
<p>The resulting GRanges object has far fewer alignments and therefore far fewer alignment stops. The algorithm is an initial step in alignment stop filtering, with the goal of a reduced number of alignment stops that have a high probability of being breakpoints.</p>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">gbs</span>, <span class="va">length</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Oki_Osa Oki_Bar Oki_Kum Oki_Aom Oki_Nor Osa_Oki Osa_Bar Osa_Kum Osa_Aom Osa_Nor </span>
<span class="co">##   30852   30841   15086   32398   30055   31322   24430   31182   15261   24497 </span>
<span class="co">## Bar_Oki Bar_Osa Bar_Kum Bar_Aom Bar_Nor Ply_Ros Ply_Rob Ply_Sav Ply_Oki Rob_Ros </span>
<span class="co">##   32265   24851   31557   24717   15184   40659   56000   68463    2127   60384 </span>
<span class="co">## Rob_Ply Rob_Sav Rob_Oki Dme_Dbu Dme_Dsu Dme_Dya Dme_Dma </span>
<span class="co">##   60205   70201    2347   47893   68750   23059    9732</span></code></pre>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">coa</span>, <span class="va">length</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Oki_Osa Oki_Bar Oki_Kum Oki_Aom Oki_Nor Osa_Oki Osa_Bar Osa_Kum Osa_Aom Osa_Nor </span>
<span class="co">##    8843    8533    5424    8864    8875    8836    4081    8776    4950    5227 </span>
<span class="co">## Bar_Oki Bar_Osa Bar_Kum Bar_Aom Bar_Nor Ply_Ros Ply_Rob Ply_Sav Ply_Oki Rob_Ros </span>
<span class="co">##    8493    4073    8529    4197    6188    7701    4969    4041    1306    4566 </span>
<span class="co">## Rob_Ply Rob_Sav Rob_Oki Dme_Dbu Dme_Dsu Dme_Dya Dme_Dma </span>
<span class="co">##    4812    4148    1402    5959     992    1816    2706</span></code></pre>
</div>
<div class="section level2">
<h2 id="width-summary-plots">Width summary plots<a class="anchor" aria-label="anchor" href="#width-summary-plots"></a>
</h2>
<p>See <code>vignette("RegionWidths"), package="OikScrambling"</code>.</p>
</div>
<div class="section level2">
<h2 id="repeats-1">Repeats<a class="anchor" aria-label="anchor" href="#repeats-1"></a>
</h2>
<p>See <code>vignette("RepeatRegions"), package="OikScrambling"</code>.</p>
</div>
<div class="section level2">
<h2 id="pairwise-chromosome-plots">Pairwise chromosome plots<a class="anchor" aria-label="anchor" href="#pairwise-chromosome-plots"></a>
</h2>
<div class="section level3">
<h3 id="oki-bar">Oki – Bar<a class="anchor" aria-label="anchor" href="#oki-bar"></a>
</h3>
<div class="sourceCode" id="cb38"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/plotApairOfChrs.html" class="external-link">plotApairOfChrs</a></span><span class="op">(</span><span class="va">coa</span><span class="op">$</span><span class="va">Oki_Bar</span>, <span class="st">"chr1"</span>, main <span class="op">=</span> <span class="st">"Oki – Bar"</span><span class="op">)</span></code></pre></div>
<p><img src="OikScrambling_files/figure-html/parallel_chr_plots_Oki_Bar-1.png" width="700"></p>
</div>
<div class="section level3">
<h3 id="osa-bar">Osa – Bar<a class="anchor" aria-label="anchor" href="#osa-bar"></a>
</h3>
<div class="sourceCode" id="cb39"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/plotApairOfChrs.html" class="external-link">plotApairOfChrs</a></span><span class="op">(</span><span class="va">coa</span><span class="op">$</span><span class="va">Osa_Bar</span>, <span class="st">"Chr1"</span>, main <span class="op">=</span> <span class="st">"Osa – Bar"</span><span class="op">)</span></code></pre></div>
<p><img src="OikScrambling_files/figure-html/parallel_chr_plots_Osa_Bar-1.png" width="700"></p>
</div>
<div class="section level3">
<h3 id="oki-kum">Oki – Kum<a class="anchor" aria-label="anchor" href="#oki-kum"></a>
</h3>
<p>The plot is not complete because the Kume assembly is not chromosome-scale.</p>
<div class="sourceCode" id="cb40"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/plotApairOfChrs.html" class="external-link">plotApairOfChrs</a></span><span class="op">(</span><span class="va">coa</span><span class="op">$</span><span class="va">Oki_Kum</span>, <span class="st">"chr1"</span>, main <span class="op">=</span> <span class="st">"Oki – Kum"</span><span class="op">)</span></code></pre></div>
<p><img src="OikScrambling_files/figure-html/parallel_chr_plots_Oki_Kum-1.png" width="700"></p>
<p>Same after double-coalescing.</p>
<div class="sourceCode" id="cb41"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/plotApairOfChrs.html" class="external-link">plotApairOfChrs</a></span><span class="op">(</span><span class="va">coa2</span><span class="op">$</span><span class="va">Oki_Kum</span>, <span class="st">"chr1"</span>, main <span class="op">=</span> <span class="st">"Oki – Kum"</span><span class="op">)</span></code></pre></div>
<p><img src="OikScrambling_files/figure-html/parallel_chr_plots_Oki_Kum_coa2-1.png" width="700"></p>
</div>
<div class="section level3">
<h3 id="more-in-the-parallel-plots-vignette">More in the <em>parallel plots</em> vignette!<a class="anchor" aria-label="anchor" href="#more-in-the-parallel-plots-vignette"></a>
</h3>
<p>In the <em>parallel plots</em> vignette (<code><a href="../articles/ParallelPlots.html">vignette("ParallelPlots", package = "OikScrambling")</a></code>), we build a more more comprehensive collection of plots and explore arrangements for a figure panel in paper.</p>
</div>
</div>
<div class="section level2">
<h2 id="structural-variants">Structural variants<a class="anchor" aria-label="anchor" href="#structural-variants"></a>
</h2>
<div class="section level3">
<h3 id="trivial-inversions">Trivial inversions<a class="anchor" aria-label="anchor" href="#trivial-inversions"></a>
</h3>
<div class="section level4">
<h4 id="documentation">Documentation<a class="anchor" aria-label="anchor" href="#documentation"></a>
</h4>
<p>Details can be found in <code><a href="../articles/Inversions.html">vignette("Inversions", package = "OikScrambling")</a></code>.</p>
</div>
</div>
<div class="section level3">
<h3 id="trivial-translocations">Trivial translocations<a class="anchor" aria-label="anchor" href="#trivial-translocations"></a>
</h3>
<div class="section level4">
<h4 id="representation">Representation<a class="anchor" aria-label="anchor" href="#representation"></a>
</h4>
<p>The translocation below:</p>
<pre><code>┌──────────────┬──────────────┬──────────────┐
│ chrA:101-200 │ chrA:201-300 │ chrA:301-400 │ (Target genome)
└──────────────┴──────────────┴──────────────┘
      +               +             +         (Alignment direction)
┌──────────────┬──────────────┬──────────────┐
│ chrB:101-200 │ chrC:201-300 │ chrB:301-400 │ (Query genome)
└──────────────┴──────────────┴──────────────┘</code></pre>
<p>Is represented as:</p>
<div class="sourceCode" id="cb43"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">exampleTranslocation</span></code></pre></div>
<pre><code><span class="co">## GBreaks object with 3 ranges and 1 metadata column:</span>
<span class="co">##       seqnames    ranges strand |        query</span>
<span class="co">##          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; |    &lt;GRanges&gt;</span>
<span class="co">##   [1]     chrA   100-200      + | chrB:100-200</span>
<span class="co">##   [2]     chrA   201-300      + | chrC:201-300</span>
<span class="co">##   [3]     chrA   301-400      + | chrB:301-400</span>
<span class="co">##   -------</span>
<span class="co">##   seqinfo: 1 sequence from an unspecified genome</span></code></pre>
<p>See in in <code><a href="../articles/LoadGenomicBreaks.html">vignette("LoadGenomicBreaks", package = "OikScrambling")</a></code> how trivial translocations are removed, and in <code><a href="../articles/RepeatRegions.html">vignette("RepeatRegions", package = "OikScrambling")</a></code> why they are removed.</p>
</div>
</div>
</div>
<div class="section level2">
<h2 id="genomic-features">Genomic Features<a class="anchor" aria-label="anchor" href="#genomic-features"></a>
</h2>
<p>Moved to <code>vignette("GenomicFeatuers", package = "OikScrambling")</code>:</p>
<ul>
<li>Tandem repeats</li>
<li>Coverage</li>
<li>Nucleic acid content heatmaps</li>
<li>Cluster analysis</li>
<li>Evidence for breakpoint hotspots</li>
<li>Tandem repeat coverage around alignment stops (breakpoints)</li>
<li>Gene feature coverage around alignment stops (breakpoints)</li>
<li>Gene feature coverage around PWM hits</li>
<li>Okinawa genome coverage</li>
</ul>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Aleksandra Bliznina, Michael Mansfield, Charles Plessy.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.2.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
