<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Orthogroup syntenies • OikScrambling</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Orthogroup syntenies">
<meta property="og:description" content="OikScrambling">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">OikScrambling</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">3.0.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/OikScrambling.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/CNEs.html">Conserved non-coding elements</a>
    </li>
    <li>
      <a href="../articles/CoalescingCutoffs.html">Modifying coalescent cutoffs</a>
    </li>
    <li>
      <a href="../articles/ColinearityInterruptors.html">Regions interrupting colinearity</a>
    </li>
    <li>
      <a href="../articles/DistanceMeasures.html">Distance Measures</a>
    </li>
    <li>
      <a href="../articles/GenomicFeatures.html">Genomic features of the breakpoints</a>
    </li>
    <li>
      <a href="../articles/Inversions.html">Inversions</a>
    </li>
    <li>
      <a href="../articles/LoadGenomicBreaks.html">Load genomic break data</a>
    </li>
    <li>
      <a href="../articles/OrthogroupSyntenies.html">Orthogroup syntenies</a>
    </li>
    <li>
      <a href="../articles/OxfordPlots.html">Whole-genome dot-plots</a>
    </li>
    <li>
      <a href="../articles/PAC3.html">Region of interest: PAC3</a>
    </li>
    <li>
      <a href="../articles/PWM.html">Motif search and analysis</a>
    </li>
    <li>
      <a href="../articles/ParallelPlots.html">Whole-chromosome parallel plots</a>
    </li>
    <li>
      <a href="../articles/RegionWidths.html">Width of the aligned and mapped regions.</a>
    </li>
    <li>
      <a href="../articles/RegionsOfInterest.html">Regions of interest</a>
    </li>
    <li>
      <a href="../articles/RepeatRegions.html">Repeat regions</a>
    </li>
    <li>
      <a href="../articles/SegmentMovements.html">Segment movement distances</a>
    </li>
    <li>
      <a href="../articles/TFBS.html">Transcription factor binding sites</a>
    </li>
    <li>
      <a href="../articles/TandemSiteDuplications.html">Tandem site duplications</a>
    </li>
    <li>
      <a href="../articles/Translocations.html">Translocations</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/oist/LuscombeU_OikScrambling/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="OrthogroupSyntenies_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Orthogroup syntenies</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/oist/LuscombeU_OikScrambling/blob/HEAD/vignettes/OrthogroupSyntenies.Rmd" class="external-link"><code>vignettes/OrthogroupSyntenies.Rmd</code></a></small>
      <div class="hidden name"><code>OrthogroupSyntenies.Rmd</code></div>

    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">knitr</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/knitr/man/opts_knit.html" class="external-link">opts_knit</a></span><span class="op">$</span><span class="fu">set</span><span class="op">(</span>cache <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<div class="section level2">
<h2 id="load-packages-and-data">Load packages and data<a class="anchor" aria-label="anchor" href="#load-packages-and-data"></a>
</h2>
<p>See <code>?OikScrambling:::loadAllGenomes()</code>, <code>?OikScrambling:::loadAllTranscriptsGR()</code>, and <code><a href="../articles/LoadGenomicBreaks.html">vignette("LoadGenomicBreaks", package = "OikScrambling")</a></code> for how the different objects are prepared.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/oist/LuscombeU_OikScrambling" class="external-link">'OikScrambling'</a></span><span class="op">)</span>   |&gt; <span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="op">)</span>
<span class="va">genomes</span> <span class="op">&lt;-</span> <span class="fu">OikScrambling</span><span class="fu">:::</span><span class="fu"><a href="../reference/loadAllGenomes.html">loadAllGenomes</a></span><span class="op">(</span>compat <span class="op">=</span> <span class="cn">F</span><span class="op">)</span>
<span class="va">transcripts</span> <span class="op">&lt;-</span> <span class="fu">OikScrambling</span><span class="fu">:::</span><span class="fu"><a href="../reference/loadAllTranscriptsGR.html">loadAllTranscriptsGR</a></span><span class="op">(</span>compat <span class="op">=</span> <span class="cn">F</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/r/base/warning.html" class="external-link">suppressWarnings</a></span><span class="op">(</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/load.html" class="external-link">load</a></span><span class="op">(</span><span class="st">"BreakPoints.Rdata"</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="prepare-gbreaks-objects">Prepare GBreaks objects<a class="anchor" aria-label="anchor" href="#prepare-gbreaks-objects"></a>
</h2>
<p>Orthogroups for <em>Oikopleura</em> are loaded from the <code>BreakpointsData</code> package with the <code><a href="../reference/load_one_to_ones.html">OikScrambling::load_one_to_ones()</a></code> function. Original file is in <code>/bucket/LuscombeU/common/Breakpoints/Orthologues/selTun+Apps+Amph+Vert_lp_100clstr_blast/OrthoFinder/Results_Oct07/Phylogenetic_Hierarchical_Orthogroups/N19.tsv</code>. See <code>news(package = "BreakpointsData")</code> for inspecting changes in more details.</p>
<div class="section level3">
<h3 id="load-orthogroup-data">Load Orthogroup data<a class="anchor" aria-label="anchor" href="#load-orthogroup-data"></a>
</h3>
<p>This produces <code>GBreaks</code> objects that map orthologs from between two genomes. In these objects, the <em>target</em> and <em>query</em> roles are equivalent. The genomic coordinates are taken from the annotation files (see <code>?OikScrambling:::loadAllTranscriptsGR</code>.</p>
<div class="section level4">
<h4 id="orthogroup-pairs">Orthogroup pairs<a class="anchor" aria-label="anchor" href="#orthogroup-pairs"></a>
</h4>
<p>To maximise number of entries, we load data from the larvacean and the ascidian subtrees respectively. This produces the <code>orthoPairs</code> collection of objects.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">orthoGroupFileNames</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/SimpleList-class.html" class="external-link">SimpleList</a></span><span class="op">(</span>
  AOM.5.5f      <span class="op">=</span> <span class="st">"AOM-5-5f.prot.longest.fa_1"</span>, 
  Bar2.p4       <span class="op">=</span> <span class="st">"Bar2_p4.Flye.prot.longest.fa_1"</span>, 
  Ply           <span class="op">=</span> <span class="st">"C_int_P.prot.longest.fa_1"</span>,
  Ros           <span class="op">=</span> <span class="st">"C_int_R.prot.longest.fa_1"</span>,
  Int           <span class="op">=</span> <span class="st">"Ciona_intestinalis.Uniprot.rn.fa_1"</span>,
  Sav           <span class="op">=</span> <span class="st">"Ciona_savignyi.Uniprot.rn.fa_1"</span>,
  KUM.M3.7f     <span class="op">=</span> <span class="st">"KUM-M3-7f.prot.longest.fa_1"</span>,
  OKI2018.I69   <span class="op">=</span> <span class="st">"OKI2018_I69.v2.prot.longest.fa_1"</span>,
  OSKA2016v1.9  <span class="op">=</span> <span class="st">"OSKA2016v1.9.prot.longest.fa_1"</span>,
  OdB3          <span class="op">=</span> <span class="st">"OdB3.v1.0.prot.fa_1.nohaplo"</span>
<span class="op">)</span>

<span class="va">orthoPairToGBreaks</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">genome1</span>, <span class="va">genome2</span>, <span class="va">transcripts</span>, <span class="va">treeName</span><span class="op">=</span><span class="st">"N19"</span>, <span class="va">HOGs</span><span class="op">=</span><span class="cn">NULL</span><span class="op">)</span> <span class="op">{</span>
  <span class="kw">if</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/NULL.html" class="external-link">is.null</a></span><span class="op">(</span><span class="va">HOGs</span><span class="op">)</span><span class="op">)</span>
    <span class="va">HOGs</span> <span class="op">&lt;-</span> <span class="fu">OikScrambling</span><span class="fu">:::</span><span class="fu"><a href="../reference/load_one_to_ones.html">load_one_to_ones</a></span><span class="op">(</span>
      <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"extdata/OrthoFinder/"</span>,<span class="va">treeName</span>,<span class="st">".tsv"</span><span class="op">)</span>, package <span class="op">=</span> <span class="st">"BreakpointsData"</span><span class="op">)</span>,
      <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="va">orthoGroupFileNames</span><span class="op">[[</span><span class="va">genome1</span><span class="op">]</span><span class="op">]</span>, <span class="va">orthoGroupFileNames</span><span class="op">[[</span><span class="va">genome2</span><span class="op">]</span><span class="op">]</span><span class="op">)</span><span class="op">)</span>

  <span class="va">IDs2GRanges</span> <span class="op">&lt;-</span> <span class="kw">function</span> <span class="op">(</span><span class="va">IDs</span>, <span class="va">annot</span><span class="op">)</span> <span class="op">{</span>
    <span class="va">prefix</span> <span class="op">&lt;-</span> <span class="fu">Biobase</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/lcSuffix.html" class="external-link">lcPrefix</a></span><span class="op">(</span><span class="va">IDs</span><span class="op">)</span>                  <span class="co"># Guess prefix in transcript IDs from HOG files</span>
    <span class="va">not_prefix</span> <span class="op">&lt;-</span> <span class="fu">Biobase</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/lcSuffix.html" class="external-link">lcPrefix</a></span><span class="op">(</span><span class="va">annot</span><span class="op">$</span><span class="va">tx_name</span><span class="op">)</span>    <span class="co"># Remove trailing characters that are part of the name</span>
    <span class="va">not_prefix</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">not_prefix</span>,<span class="st">"$"</span><span class="op">)</span>              <span class="co"># Anchor to end of the string</span>
    <span class="va">prefix</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">sub</a></span><span class="op">(</span><span class="va">not_prefix</span>, <span class="st">""</span>, <span class="va">prefix</span><span class="op">)</span>             <span class="co"># Finalise prefix</span>
    <span class="va">IDs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">sub</a></span><span class="op">(</span><span class="va">prefix</span>, <span class="st">""</span>, <span class="va">IDs</span><span class="op">)</span>                       <span class="co"># Remove prefix from IDS</span>
    <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">annot</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">annot</span><span class="op">$</span><span class="va">tx_name</span>                     <span class="co"># Store transcript name in names slot</span>
    <span class="va">gr</span> <span class="op">&lt;-</span> <span class="va">annot</span><span class="op">[</span><span class="va">IDs</span><span class="op">]</span>                                  <span class="co"># Sort by ID</span>
    <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/strand.html" class="external-link">strand</a></span><span class="op">(</span><span class="va">gr</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="st">"*"</span>                                 <span class="co"># Make strandless</span>
    <span class="va">gr</span>                                                <span class="co"># Return the object</span>
  <span class="op">}</span>
  
  <span class="va">gb</span>       <span class="op">&lt;-</span> <span class="fu">IDs2GRanges</span><span class="op">(</span><span class="va">HOGs</span><span class="op">[</span>,<span class="va">orthoGroupFileNames</span><span class="op">[[</span><span class="va">genome1</span><span class="op">]</span><span class="op">]</span><span class="op">]</span>, <span class="va">transcripts</span><span class="op">[[</span><span class="va">genome1</span><span class="op">]</span><span class="op">]</span><span class="op">)</span>
  <span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NULL.html" class="external-link">is.null</a></span><span class="op">(</span><span class="va">genomes</span><span class="op">[[</span><span class="va">genome1</span><span class="op">]</span><span class="op">]</span><span class="op">)</span><span class="op">)</span>
    <span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqinfo</a></span><span class="op">(</span><span class="va">gb</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqinfo</a></span><span class="op">(</span><span class="va">genomes</span><span class="op">[[</span><span class="va">genome1</span><span class="op">]</span><span class="op">]</span><span class="op">)</span>
  <span class="va">gb</span><span class="op">$</span><span class="va">query</span> <span class="op">&lt;-</span> <span class="fu">IDs2GRanges</span><span class="op">(</span><span class="va">HOGs</span><span class="op">[</span>,<span class="va">orthoGroupFileNames</span><span class="op">[[</span><span class="va">genome2</span><span class="op">]</span><span class="op">]</span><span class="op">]</span>, <span class="va">transcripts</span><span class="op">[[</span><span class="va">genome2</span><span class="op">]</span><span class="op">]</span><span class="op">)</span>
  <span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NULL.html" class="external-link">is.null</a></span><span class="op">(</span><span class="va">genomes</span><span class="op">[[</span><span class="va">genome2</span><span class="op">]</span><span class="op">]</span><span class="op">)</span><span class="op">)</span>
   <span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqinfo</a></span><span class="op">(</span><span class="va">gb</span><span class="op">$</span><span class="va">query</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqinfo</a></span><span class="op">(</span><span class="va">genomes</span><span class="op">[[</span><span class="va">genome2</span><span class="op">]</span><span class="op">]</span><span class="op">)</span>
  <span class="va">gb</span> <span class="op">&lt;-</span> <span class="fu">GenomicBreaks</span><span class="fu">:::</span><span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/GBreaks-class.html" class="external-link">GBreaks</a></span><span class="op">(</span><span class="va">gb</span><span class="op">)</span>
  <span class="va">gb</span><span class="op">$</span><span class="va">HOG</span> <span class="op">&lt;-</span> <span class="va">HOGs</span><span class="op">$</span><span class="va">HOG</span>
  <span class="va">gb</span><span class="op">$</span><span class="va">OG</span>  <span class="op">&lt;-</span> <span class="va">HOGs</span><span class="op">$</span><span class="va">OG</span>
  <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/sort.html" class="external-link">sort</a></span><span class="op">(</span><span class="va">gb</span><span class="op">)</span>
<span class="op">}</span>

<span class="va">flagLongShort_</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">gr</span>, <span class="va">transcripts</span><span class="op">)</span> <span class="op">{</span>
  <span class="va">genome</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">genome</a></span><span class="op">(</span><span class="va">gr</span><span class="op">)</span><span class="op">)</span>
  <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/flagLongShort.html" class="external-link">flagLongShort</a></span><span class="op">(</span><span class="va">gr</span>, <span class="va">transcripts</span><span class="op">[[</span><span class="va">genome</span><span class="op">]</span><span class="op">]</span><span class="op">)</span>
<span class="op">}</span>

<span class="va">orthoPairToGBreaks_all_Oiks</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">treeName</span><span class="op">=</span><span class="cn">NULL</span>, <span class="va">HOGs</span><span class="op">=</span><span class="cn">NULL</span><span class="op">)</span> <span class="op">{</span>
  <span class="va">orthoPairs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/SimpleList-class.html" class="external-link">SimpleList</a></span><span class="op">(</span><span class="op">)</span>
  <span class="va">orthoPairs</span><span class="op">$</span><span class="va">Oki_Osa</span> <span class="op">&lt;-</span> <span class="fu">orthoPairToGBreaks</span><span class="op">(</span><span class="st">"OKI2018.I69"</span>, <span class="st">"OSKA2016v1.9"</span>,   <span class="va">transcripts</span>, <span class="va">treeName</span>, <span class="va">HOGs</span><span class="op">)</span>
  <span class="va">orthoPairs</span><span class="op">$</span><span class="va">Oki_Bar</span> <span class="op">&lt;-</span> <span class="fu">orthoPairToGBreaks</span><span class="op">(</span><span class="st">"OKI2018.I69"</span>, <span class="st">"Bar2.p4"</span>,        <span class="va">transcripts</span>, <span class="va">treeName</span>, <span class="va">HOGs</span><span class="op">)</span>
  <span class="va">orthoPairs</span><span class="op">$</span><span class="va">Oki_Kum</span> <span class="op">&lt;-</span> <span class="fu">orthoPairToGBreaks</span><span class="op">(</span><span class="st">"OKI2018.I69"</span>, <span class="st">"KUM.M3.7f"</span>,      <span class="va">transcripts</span>, <span class="va">treeName</span>, <span class="va">HOGs</span><span class="op">)</span>
  <span class="va">orthoPairs</span><span class="op">$</span><span class="va">Oki_Aom</span> <span class="op">&lt;-</span> <span class="fu">orthoPairToGBreaks</span><span class="op">(</span><span class="st">"OKI2018.I69"</span>, <span class="st">"AOM.5.5f"</span>,       <span class="va">transcripts</span>, <span class="va">treeName</span>, <span class="va">HOGs</span><span class="op">)</span>
  <span class="va">orthoPairs</span><span class="op">$</span><span class="va">Oki_Nor</span> <span class="op">&lt;-</span> <span class="fu">orthoPairToGBreaks</span><span class="op">(</span><span class="st">"OKI2018.I69"</span>, <span class="st">"OdB3"</span>,           <span class="va">transcripts</span>, <span class="va">treeName</span>, <span class="va">HOGs</span><span class="op">)</span>
  
  <span class="va">orthoPairs</span><span class="op">$</span><span class="va">Osa_Bar</span> <span class="op">&lt;-</span> <span class="fu">orthoPairToGBreaks</span><span class="op">(</span><span class="st">"OSKA2016v1.9"</span>,   <span class="st">"Bar2.p4"</span>,     <span class="va">transcripts</span>, <span class="va">treeName</span>, <span class="va">HOGs</span><span class="op">)</span>
  <span class="va">orthoPairs</span><span class="op">$</span><span class="va">Osa_Oki</span> <span class="op">&lt;-</span> <span class="fu">orthoPairToGBreaks</span><span class="op">(</span><span class="st">"OSKA2016v1.9"</span>,   <span class="st">"OKI2018.I69"</span>, <span class="va">transcripts</span>, <span class="va">treeName</span>, <span class="va">HOGs</span><span class="op">)</span>
  <span class="va">orthoPairs</span><span class="op">$</span><span class="va">Osa_Kum</span> <span class="op">&lt;-</span> <span class="fu">orthoPairToGBreaks</span><span class="op">(</span><span class="st">"OSKA2016v1.9"</span>,   <span class="st">"KUM.M3.7f"</span>,   <span class="va">transcripts</span>, <span class="va">treeName</span>, <span class="va">HOGs</span><span class="op">)</span>
  <span class="va">orthoPairs</span><span class="op">$</span><span class="va">Osa_Aom</span> <span class="op">&lt;-</span> <span class="fu">orthoPairToGBreaks</span><span class="op">(</span><span class="st">"OSKA2016v1.9"</span>,   <span class="st">"AOM.5.5f"</span>,    <span class="va">transcripts</span>, <span class="va">treeName</span>, <span class="va">HOGs</span><span class="op">)</span>
  <span class="va">orthoPairs</span><span class="op">$</span><span class="va">Osa_Nor</span> <span class="op">&lt;-</span> <span class="fu">orthoPairToGBreaks</span><span class="op">(</span><span class="st">"OSKA2016v1.9"</span>,   <span class="st">"OdB3"</span>,        <span class="va">transcripts</span>, <span class="va">treeName</span>, <span class="va">HOGs</span><span class="op">)</span>
  
  <span class="va">orthoPairs</span><span class="op">$</span><span class="va">Bar_Osa</span> <span class="op">&lt;-</span> <span class="fu">orthoPairToGBreaks</span><span class="op">(</span><span class="st">"Bar2.p4"</span>,   <span class="st">"OSKA2016v1.9"</span>,     <span class="va">transcripts</span>, <span class="va">treeName</span>, <span class="va">HOGs</span><span class="op">)</span>
  <span class="va">orthoPairs</span><span class="op">$</span><span class="va">Bar_Oki</span> <span class="op">&lt;-</span> <span class="fu">orthoPairToGBreaks</span><span class="op">(</span><span class="st">"Bar2.p4"</span>,   <span class="st">"OKI2018.I69"</span>,      <span class="va">transcripts</span>, <span class="va">treeName</span>, <span class="va">HOGs</span><span class="op">)</span>
  <span class="va">orthoPairs</span><span class="op">$</span><span class="va">Bar_Kum</span> <span class="op">&lt;-</span> <span class="fu">orthoPairToGBreaks</span><span class="op">(</span><span class="st">"Bar2.p4"</span>,   <span class="st">"KUM.M3.7f"</span>,        <span class="va">transcripts</span>, <span class="va">treeName</span>, <span class="va">HOGs</span><span class="op">)</span>
  <span class="va">orthoPairs</span><span class="op">$</span><span class="va">Bar_Aom</span> <span class="op">&lt;-</span> <span class="fu">orthoPairToGBreaks</span><span class="op">(</span><span class="st">"Bar2.p4"</span>,   <span class="st">"AOM.5.5f"</span>,         <span class="va">transcripts</span>, <span class="va">treeName</span>, <span class="va">HOGs</span><span class="op">)</span>
  <span class="va">orthoPairs</span><span class="op">$</span><span class="va">Bar_Nor</span> <span class="op">&lt;-</span> <span class="fu">orthoPairToGBreaks</span><span class="op">(</span><span class="st">"Bar2.p4"</span>,   <span class="st">"OdB3"</span>,             <span class="va">transcripts</span>, <span class="va">treeName</span>, <span class="va">HOGs</span><span class="op">)</span>
  
  <span class="va">orthoPairs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">orthoPairs</span>,  <span class="va">flagLongShort_</span>, <span class="va">longShort</span><span class="op">)</span>
  
  <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/SimpleList-class.html" class="external-link">SimpleList</a></span><span class="op">(</span><span class="va">orthoPairs</span><span class="op">)</span>
<span class="op">}</span>

<span class="va">orthoPairs</span>         <span class="op">&lt;-</span> <span class="fu">orthoPairToGBreaks_all_Oiks</span><span class="op">(</span><span class="st">"N19"</span><span class="op">)</span>
<span class="va">orthoPairs</span><span class="op">$</span><span class="va">Ply_Ros</span> <span class="op">&lt;-</span> <span class="fu">orthoPairToGBreaks</span><span class="op">(</span><span class="st">"Ply"</span>, <span class="st">"Ros"</span>, <span class="va">transcripts</span>, treeName<span class="op">=</span><span class="st">"N20"</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level4">
<h4 id="orthogroup-core-pairs">Orthogroup core pairs<a class="anchor" aria-label="anchor" href="#orthogroup-core-pairs"></a>
</h4>
<p>The <code>orthoPairs_core</code> collection is built similarly, but from “core” orthogroups, that have a member in each species of the tunicate clade.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">orthoPairs_core</span>         <span class="op">&lt;-</span> <span class="fu">orthoPairToGBreaks_all_Oiks</span><span class="op">(</span><span class="st">"N3"</span><span class="op">)</span>
<span class="va">orthoPairs_core</span><span class="op">$</span><span class="va">Ply_Ros</span> <span class="op">&lt;-</span> <span class="fu">orthoPairToGBreaks</span><span class="op">(</span><span class="st">"Ply"</span>, <span class="st">"Ros"</span>, <span class="va">transcripts</span>, treeName<span class="op">=</span><span class="st">"N3"</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level4">
<h4 id="orthogroups-one-to-one-in-oiks-">Orthogroups one-to-one in oiks.<a class="anchor" aria-label="anchor" href="#orthogroups-one-to-one-in-oiks-"></a>
</h4>
<p>Lastly we build an object for Oiks only where we require every orthologue to be present as a single copy in every genome. This is the <code>orthoPairs_one2oneInOiks</code> collection.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">orthoPairs_one2oneInOiks</span> <span class="op">&lt;-</span>
  <span class="fu">orthoPairToGBreaks_all_Oiks</span><span class="op">(</span>treeName <span class="op">=</span> <span class="cn">NULL</span>,
                              HOGs <span class="op">=</span> <span class="fu">OikScrambling</span><span class="fu">:::</span><span class="fu"><a href="../reference/load_one_to_ones.html">load_one_to_ones</a></span><span class="op">(</span>
                                       <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata/OrthoFinder/N19.tsv"</span>, package <span class="op">=</span> <span class="st">"BreakpointsData"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level4">
<h4 id="summary">Summary<a class="anchor" aria-label="anchor" href="#summary"></a>
</h4>
<p>Here is the number of entries in each object.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">orthoPairs</span>, <span class="va">length</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Oki_Osa Oki_Bar Oki_Kum Oki_Aom Oki_Nor Osa_Bar Osa_Oki Osa_Kum Osa_Aom Osa_Nor </span>
<span class="co">##    9538    9273   11525    9604    8771    9172    9538    9465   10379    8415 </span>
<span class="co">## Bar_Osa Bar_Oki Bar_Kum Bar_Aom Bar_Nor Ply_Ros </span>
<span class="co">##    9172    9273    9223    9250    8508    5438</span></code></pre>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">orthoPairs_core</span>, <span class="va">length</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Oki_Osa Oki_Bar Oki_Kum Oki_Aom Oki_Nor Osa_Bar Osa_Oki Osa_Kum Osa_Aom Osa_Nor </span>
<span class="co">##    6794    6697    8315    6902    6118    6707    6794    6697    7644    5981 </span>
<span class="co">## Bar_Osa Bar_Oki Bar_Kum Bar_Aom Bar_Nor Ply_Ros </span>
<span class="co">##    6707    6697    6643    6804    6151    2582</span></code></pre>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">orthoPairs_one2oneInOiks</span>, <span class="va">length</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Oki_Osa Oki_Bar Oki_Kum Oki_Aom Oki_Nor Osa_Bar Osa_Oki Osa_Kum Osa_Aom Osa_Nor </span>
<span class="co">##    5162    5162    5162    5162    5162    5162    5162    5162    5162    5162 </span>
<span class="co">## Bar_Osa Bar_Oki Bar_Kum Bar_Aom Bar_Nor </span>
<span class="co">##    5162    5162    5162    5162    5162</span></code></pre>
</div>
</div>
<div class="section level3">
<h3 id="remove-overlapping-annotations">Remove overlapping annotations<a class="anchor" aria-label="anchor" href="#remove-overlapping-annotations"></a>
</h3>
<p>Some transcript annotations overlap and this breaks the coalescing algorithm.</p>
<p>Here we produce non-overlapping versions of the objects.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">transcripts</span><span class="op">$</span><span class="va">Bar2.p4</span> |&gt; <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## [1] 15741</span></code></pre>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">transcripts</span><span class="op">$</span><span class="va">Bar2.p4</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/inter-range-methods.html" class="external-link">reduce</a></span><span class="op">(</span>min <span class="op">=</span> <span class="fl">0</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## [1] 14262</span></code></pre>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">flagOverlaps</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">gr</span><span class="op">)</span> <span class="op">{</span>
  <span class="co"># Find overlaps</span>
  <span class="va">ov</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/findOverlaps-methods.html" class="external-link">findOverlaps</a></span><span class="op">(</span><span class="va">gr</span><span class="op">)</span>
  <span class="co"># Index of self hits in the gr object</span>
  <span class="va">idx</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Hits-class.html" class="external-link">queryHits</a></span><span class="op">(</span><span class="va">ov</span><span class="op">[</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Hits-class.html" class="external-link">isSelfHit</a></span><span class="op">(</span><span class="va">ov</span><span class="op">)</span><span class="op">]</span><span class="op">)</span>
  <span class="co"># Convert to Boolean indexing the gr object</span>
  <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq_along</a></span><span class="op">(</span><span class="va">gr</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html" class="external-link">%in%</a></span> <span class="va">idx</span>
<span class="op">}</span>

<span class="va">transcripts</span><span class="op">$</span><span class="va">Bar2.p4</span><span class="op">[</span><span class="fu">flagOverlaps</span><span class="op">(</span><span class="va">transcripts</span><span class="op">$</span><span class="va">Bar2.p4</span><span class="op">)</span><span class="op">]</span></code></pre></div>
<pre><code><span class="co">## GRanges object with 2668 ranges and 5 metadata columns:</span>
<span class="co">##             seqnames          ranges strand |     tx_id     tx_name</span>
<span class="co">##                &lt;Rle&gt;       &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt; &lt;character&gt;</span>
<span class="co">##     g107.t2     Chr1   495446-496292      + |        58     g107.t2</span>
<span class="co">##     g107.t1     Chr1   495446-496688      + |        59     g107.t1</span>
<span class="co">##     g138.t1     Chr1   687958-689814      + |        76     g138.t1</span>
<span class="co">##     g138.t2     Chr1   687958-689814      + |        77     g138.t2</span>
<span class="co">##     g147.t2     Chr1   748328-751932      + |        82     g147.t2</span>
<span class="co">##         ...      ...             ...    ... .       ...         ...</span>
<span class="co">##   g13975.t2      YSR 4098404-4105602      - |     15657   g13975.t2</span>
<span class="co">##   g14058.t1      YSR 4778101-4778538      - |     15701   g14058.t1</span>
<span class="co">##   g14059.t1      YSR 4778480-4782151      - |     15702   g14059.t1</span>
<span class="co">##   g14064.t2      YSR 4817799-4819944      - |     15705   g14064.t2</span>
<span class="co">##   g14064.t1      YSR 4817799-4820671      - |     15706   g14064.t1</span>
<span class="co">##             dNdS_GUIDANCE2 dNdS_HmmCleaner dNdS_PRANK</span>
<span class="co">##                  &lt;numeric&gt;       &lt;numeric&gt;  &lt;numeric&gt;</span>
<span class="co">##     g107.t2             NA              NA         NA</span>
<span class="co">##     g107.t1             NA              NA         NA</span>
<span class="co">##     g138.t1             NA              NA         NA</span>
<span class="co">##     g138.t2        0.05054         0.05386    0.05386</span>
<span class="co">##     g147.t2             NA              NA         NA</span>
<span class="co">##         ...            ...             ...        ...</span>
<span class="co">##   g13975.t2             NA              NA         NA</span>
<span class="co">##   g14058.t1             NA              NA         NA</span>
<span class="co">##   g14059.t1             NA              NA         NA</span>
<span class="co">##   g14064.t2             NA              NA         NA</span>
<span class="co">##   g14064.t1             NA              NA         NA</span>
<span class="co">##   -------</span>
<span class="co">##   seqinfo: 68 sequences from Bar2.p4 genome</span></code></pre>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Show examples</span>
<span class="va">orthoPairs</span><span class="op">$</span><span class="va">Oki_Osa</span><span class="op">[</span><span class="fu">flagOverlaps</span><span class="op">(</span><span class="va">orthoPairs</span><span class="op">$</span><span class="va">Oki_Osa</span><span class="op">)</span><span class="op">]</span></code></pre></div>
<pre><code><span class="co">## GBreaks object with 93 ranges and 9 metadata columns:</span>
<span class="co">##             seqnames            ranges strand |     tx_id     tx_name</span>
<span class="co">##                &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt; &lt;character&gt;</span>
<span class="co">##     g499.t1     chr1   2110061-2117154      * |       272     g499.t1</span>
<span class="co">##     g500.t1     chr1   2115711-2116788      * |      2381     g500.t1</span>
<span class="co">##     g544.t1     chr1   2238352-2256657      * |       297     g544.t1</span>
<span class="co">##     g545.t1     chr1   2251048-2252855      * |      2402     g545.t1</span>
<span class="co">##     g985.t1     chr1   3781719-3784744      * |       518     g985.t1</span>
<span class="co">##         ...      ...               ...    ... .       ...         ...</span>
<span class="co">##   g15884.t1      XSR   8785428-8786009      * |     15900   g15884.t1</span>
<span class="co">##   g16057.t1      XSR   9342752-9345613      * |     18123   g16057.t1</span>
<span class="co">##   g16058.t1      XSR   9344384-9344779      * |     15985   g16058.t1</span>
<span class="co">##   g16592.t1      XSR 10958372-10959713      * |     18454   g16592.t1</span>
<span class="co">##   g16593.t1      XSR 10958914-10959511      * |     16264   g16593.t1</span>
<span class="co">##             dNdS_GUIDANCE2 dNdS_HmmCleaner dNdS_PRANK                 query</span>
<span class="co">##                  &lt;numeric&gt;       &lt;numeric&gt;  &lt;numeric&gt;             &lt;GRanges&gt;</span>
<span class="co">##     g499.t1         0.0299         0.03004     0.0329    Chr1:717839-724867</span>
<span class="co">##     g500.t1             NA              NA         NA    Chr1:718410-719881</span>
<span class="co">##     g544.t1             NA              NA         NA  Chr1:1777743-1781983</span>
<span class="co">##     g545.t1             NA              NA         NA  Chr1:1783570-1784116</span>
<span class="co">##     g985.t1             NA              NA         NA  Chr1:2597661-2601166</span>
<span class="co">##         ...            ...             ...        ...                   ...</span>
<span class="co">##   g15884.t1        0.06523         0.09376    0.08999   XSR:1917337-1918103</span>
<span class="co">##   g16057.t1             NA              NA         NA   XSR:5950541-5952986</span>
<span class="co">##   g16058.t1        0.00761         0.00524    0.00524   XSR:5950957-5951353</span>
<span class="co">##   g16592.t1             NA              NA         NA XSR:10797233-10798365</span>
<span class="co">##   g16593.t1             NA              NA         NA XSR:10797613-10797896</span>
<span class="co">##                        HOG          OG      Arm</span>
<span class="co">##                &lt;character&gt; &lt;character&gt; &lt;factor&gt;</span>
<span class="co">##     g499.t1 N19.HOG0002425   OG0000263    short</span>
<span class="co">##     g500.t1 N19.HOG0007945   OG0003451    short</span>
<span class="co">##     g544.t1 N19.HOG0003237   OG0000486    short</span>
<span class="co">##     g545.t1 N19.HOG0015828   OG0019308    short</span>
<span class="co">##     g985.t1 N19.HOG0012972   OG0011801    short</span>
<span class="co">##         ...            ...         ...      ...</span>
<span class="co">##   g15884.t1 N19.HOG0013867   OG0014177      XSR</span>
<span class="co">##   g16057.t1 N19.HOG0007856   OG0003342      XSR</span>
<span class="co">##   g16058.t1 N19.HOG0013727   OG0013149      XSR</span>
<span class="co">##   g16592.t1 N19.HOG0016102   OG0020123      XSR</span>
<span class="co">##   g16593.t1 N19.HOG0016845   OG0024775      XSR</span>
<span class="co">##   -------</span>
<span class="co">##   seqinfo: 19 sequences from OKI2018.I69 genome</span></code></pre>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">orthoPairs</span><span class="op">$</span><span class="va">Oki_Osa</span><span class="op">$</span><span class="va">query</span><span class="op">[</span><span class="fu">flagOverlaps</span><span class="op">(</span><span class="va">orthoPairs</span><span class="op">$</span><span class="va">Oki_Osa</span><span class="op">$</span><span class="va">query</span><span class="op">)</span><span class="op">]</span></code></pre></div>
<pre><code><span class="co">## GRanges object with 80 ranges and 5 metadata columns:</span>
<span class="co">##             seqnames            ranges strand |     tx_id     tx_name</span>
<span class="co">##                &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt; &lt;character&gt;</span>
<span class="co">##     g165.t1     Chr1     717839-724867      * |      1760     g165.t1</span>
<span class="co">##     g166.t1     Chr1     718410-719881      * |        84     g166.t1</span>
<span class="co">##     g343.t1     Chr1   1555362-1560721      * |       185     g343.t1</span>
<span class="co">##     g344.t1     Chr1   1555709-1558669      * |      1852     g344.t1</span>
<span class="co">##    g1318.t1     Chr1   5426128-5431903      * |       698    g1318.t1</span>
<span class="co">##         ...      ...               ...    ... .       ...         ...</span>
<span class="co">##   g14741.t1      XSR 10797613-10797896      * |     16354   g14741.t1</span>
<span class="co">##   g13032.t1      XSR   5203787-5206862      * |     13405   g13032.t1</span>
<span class="co">##   g13031.t2      XSR   5199352-5210190      * |     15385   g13031.t2</span>
<span class="co">##   g13679.t1      XSR   7256274-7260299      * |     13797   g13679.t1</span>
<span class="co">##   g13680.t1      XSR   7258070-7259639      * |     15749   g13680.t1</span>
<span class="co">##             dNdS_GUIDANCE2 dNdS_HmmCleaner dNdS_PRANK</span>
<span class="co">##                  &lt;numeric&gt;       &lt;numeric&gt;  &lt;numeric&gt;</span>
<span class="co">##     g165.t1        0.02990         0.03004    0.03290</span>
<span class="co">##     g166.t1             NA              NA         NA</span>
<span class="co">##     g343.t1             NA              NA         NA</span>
<span class="co">##     g344.t1        0.04569         0.05541    0.05541</span>
<span class="co">##    g1318.t1             NA              NA         NA</span>
<span class="co">##         ...            ...             ...        ...</span>
<span class="co">##   g14741.t1             NA              NA         NA</span>
<span class="co">##   g13032.t1             NA              NA         NA</span>
<span class="co">##   g13031.t2        0.00897         0.00897    0.00897</span>
<span class="co">##   g13679.t1             NA              NA         NA</span>
<span class="co">##   g13680.t1             NA              NA         NA</span>
<span class="co">##   -------</span>
<span class="co">##   seqinfo: 483 sequences from OSKA2016v1.9 genome</span></code></pre>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Remove and show how much was removed</span>

<span class="va">orthoPairsFiltered</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">orthoPairs</span>, <span class="kw">function</span><span class="op">(</span><span class="va">gb</span><span class="op">)</span> <span class="op">{</span>
  <span class="va">gb</span> <span class="op">&lt;-</span> <span class="va">gb</span><span class="op">[</span><span class="op">!</span><span class="fu">flagOverlaps</span><span class="op">(</span><span class="va">gb</span><span class="op">)</span><span class="op">]</span>
  <span class="va">gb</span> <span class="op">&lt;-</span> <span class="va">gb</span><span class="op">[</span><span class="op">!</span><span class="fu">flagOverlaps</span><span class="op">(</span><span class="va">gb</span><span class="op">$</span><span class="va">query</span><span class="op">)</span><span class="op">]</span>
  <span class="va">gb</span>
<span class="op">}</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/SimpleList-class.html" class="external-link">SimpleList</a></span><span class="op">(</span><span class="op">)</span>

<span class="va">orthoPairsFiltered_core</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">orthoPairs_core</span>, <span class="kw">function</span><span class="op">(</span><span class="va">gb</span><span class="op">)</span> <span class="op">{</span>
  <span class="va">gb</span> <span class="op">&lt;-</span> <span class="va">gb</span><span class="op">[</span><span class="op">!</span><span class="fu">flagOverlaps</span><span class="op">(</span><span class="va">gb</span><span class="op">)</span><span class="op">]</span>
  <span class="va">gb</span> <span class="op">&lt;-</span> <span class="va">gb</span><span class="op">[</span><span class="op">!</span><span class="fu">flagOverlaps</span><span class="op">(</span><span class="va">gb</span><span class="op">$</span><span class="va">query</span><span class="op">)</span><span class="op">]</span>
  <span class="va">gb</span>
<span class="op">}</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/SimpleList-class.html" class="external-link">SimpleList</a></span><span class="op">(</span><span class="op">)</span>

<span class="va">orthoPairs_one2oneInOiks_Filtered</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">orthoPairs_one2oneInOiks</span>, <span class="kw">function</span><span class="op">(</span><span class="va">gb</span><span class="op">)</span> <span class="op">{</span>
  <span class="va">gb</span> <span class="op">&lt;-</span> <span class="va">gb</span><span class="op">[</span><span class="op">!</span><span class="fu">flagOverlaps</span><span class="op">(</span><span class="va">gb</span><span class="op">)</span><span class="op">]</span>
  <span class="va">gb</span> <span class="op">&lt;-</span> <span class="va">gb</span><span class="op">[</span><span class="op">!</span><span class="fu">flagOverlaps</span><span class="op">(</span><span class="va">gb</span><span class="op">$</span><span class="va">query</span><span class="op">)</span><span class="op">]</span>
  <span class="va">gb</span>
<span class="op">}</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/SimpleList-class.html" class="external-link">SimpleList</a></span><span class="op">(</span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">orthoPairs</span>, <span class="va">length</span><span class="op">)</span>               <span class="op">-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">orthoPairsFiltered</span>, <span class="va">length</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Oki_Osa Oki_Bar Oki_Kum Oki_Aom Oki_Nor Osa_Bar Osa_Oki Osa_Kum Osa_Aom Osa_Nor </span>
<span class="co">##     132     131     202     166     119      98     132     114     121     103 </span>
<span class="co">## Bar_Osa Bar_Oki Bar_Kum Bar_Aom Bar_Nor Ply_Ros </span>
<span class="co">##      98     132     119     108     125      26</span></code></pre>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">orthoPairs_core</span>, <span class="va">length</span><span class="op">)</span>          <span class="op">-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">orthoPairsFiltered_core</span>, <span class="va">length</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Oki_Osa Oki_Bar Oki_Kum Oki_Aom Oki_Nor Osa_Bar Osa_Oki Osa_Kum Osa_Aom Osa_Nor </span>
<span class="co">##      74      85     125      93      58      72      74      67      75      48 </span>
<span class="co">## Bar_Osa Bar_Oki Bar_Kum Bar_Aom Bar_Nor Ply_Ros </span>
<span class="co">##      72      85      85      72      64      10</span></code></pre>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">orthoPairs_one2oneInOiks</span>, <span class="va">length</span><span class="op">)</span> <span class="op">-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">orthoPairs_one2oneInOiks_Filtered</span>, <span class="va">length</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Oki_Osa Oki_Bar Oki_Kum Oki_Aom Oki_Nor Osa_Bar Osa_Oki Osa_Kum Osa_Aom Osa_Nor </span>
<span class="co">##      18      26      14      18      38      26      18      20      16      44 </span>
<span class="co">## Bar_Osa Bar_Oki Bar_Kum Bar_Aom Bar_Nor </span>
<span class="co">##      26      26      26      26      52</span></code></pre>
</div>
<div class="section level3">
<h3 id="coalesce-syntenic-blocks-and-annotate-them">Coalesce syntenic blocks and annotate them<a class="anchor" aria-label="anchor" href="#coalesce-syntenic-blocks-and-annotate-them"></a>
</h3>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">coalOrtho</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">gb</span><span class="op">)</span> <span class="op">{</span>
  <span class="co"># Coalesce</span>
  <span class="va">coal</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/coalesce_contigs.html" class="external-link">coalesce_contigs</a></span><span class="op">(</span><span class="va">gb</span><span class="op">)</span>
  
  <span class="co"># Map annotations to the coalesced blocks</span>
  <span class="va">ov</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/findOverlaps-methods.html" class="external-link">findOverlaps</a></span><span class="op">(</span><span class="va">coal</span>, <span class="va">gb</span><span class="op">)</span>
  
  <span class="co"># Extract gene names mapped to each blocks</span>
  <span class="va">coal</span><span class="op">$</span><span class="va">geneNames</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/AtomicList-class.html" class="external-link">CharacterList</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/splitAsList.html" class="external-link">split</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">gb</span><span class="op">)</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Hits-class.html" class="external-link">subjectHits</a></span><span class="op">(</span><span class="va">ov</span><span class="op">)</span><span class="op">]</span>, <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Hits-class.html" class="external-link">queryHits</a></span><span class="op">(</span><span class="va">ov</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/r/base/unname.html" class="external-link">unname</a></span><span class="op">(</span><span class="op">)</span>
  <span class="co"># We can use the `ov` object for query ranges as well because of the way `coal` was constructed.</span>
  <span class="va">coal</span><span class="op">$</span><span class="va">query</span><span class="op">$</span><span class="va">geneNames</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/AtomicList-class.html" class="external-link">CharacterList</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/splitAsList.html" class="external-link">split</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">gb</span><span class="op">$</span><span class="va">query</span><span class="op">)</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Hits-class.html" class="external-link">subjectHits</a></span><span class="op">(</span><span class="va">ov</span><span class="op">)</span><span class="op">]</span>, <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Hits-class.html" class="external-link">queryHits</a></span><span class="op">(</span><span class="va">ov</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/r/base/unname.html" class="external-link">unname</a></span><span class="op">(</span><span class="op">)</span>
  
  <span class="co"># Extract dNdS values</span>
  <span class="va">coal</span><span class="op">$</span><span class="va">dNdS_GUIDANCE2</span>  <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/AtomicList-class.html" class="external-link">NumericList</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/splitAsList.html" class="external-link">split</a></span><span class="op">(</span><span class="va">gb</span><span class="op">$</span><span class="va">dNdS_GUIDANCE2</span> <span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Hits-class.html" class="external-link">subjectHits</a></span><span class="op">(</span><span class="va">ov</span><span class="op">)</span><span class="op">]</span>, <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Hits-class.html" class="external-link">queryHits</a></span><span class="op">(</span><span class="va">ov</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
  <span class="va">coal</span><span class="op">$</span><span class="va">dNdS_HmmCleaner</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/AtomicList-class.html" class="external-link">NumericList</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/splitAsList.html" class="external-link">split</a></span><span class="op">(</span><span class="va">gb</span><span class="op">$</span><span class="va">dNdS_HmmCleaner</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Hits-class.html" class="external-link">subjectHits</a></span><span class="op">(</span><span class="va">ov</span><span class="op">)</span><span class="op">]</span>, <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Hits-class.html" class="external-link">queryHits</a></span><span class="op">(</span><span class="va">ov</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
  <span class="va">coal</span><span class="op">$</span><span class="va">dNdS_PRANK</span>      <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/AtomicList-class.html" class="external-link">NumericList</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/splitAsList.html" class="external-link">split</a></span><span class="op">(</span><span class="va">gb</span><span class="op">$</span><span class="va">dNdS_PRANK</span>     <span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Hits-class.html" class="external-link">subjectHits</a></span><span class="op">(</span><span class="va">ov</span><span class="op">)</span><span class="op">]</span>, <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Hits-class.html" class="external-link">queryHits</a></span><span class="op">(</span><span class="va">ov</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
  <span class="co"># We just copy dNdS values to query ranges because they are the same.</span>
  <span class="va">coal</span><span class="op">$</span><span class="va">query</span><span class="op">$</span><span class="va">dNdS_GUIDANCE2</span>  <span class="op">&lt;-</span> <span class="va">coal</span><span class="op">$</span><span class="va">dNdS_GUIDANCE2</span>
  <span class="va">coal</span><span class="op">$</span><span class="va">query</span><span class="op">$</span><span class="va">dNdS_HmmCleaner</span> <span class="op">&lt;-</span> <span class="va">coal</span><span class="op">$</span><span class="va">dNdS_HmmCleaner</span>
  <span class="va">coal</span><span class="op">$</span><span class="va">query</span><span class="op">$</span><span class="va">dNdS_PRANK</span>      <span class="op">&lt;-</span> <span class="va">coal</span><span class="op">$</span><span class="va">dNdS_PRANK</span>

  <span class="co"># Count genes per syntenic block</span>
  <span class="va">coal</span><span class="op">$</span><span class="va">geneNumber</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">coal</span><span class="op">$</span><span class="va">geneNames</span>, <span class="va">length</span><span class="op">)</span>
  
  <span class="va">coal</span>
<span class="op">}</span>

<span class="va">orthoBlocks</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">orthoPairsFiltered</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">15</span><span class="op">]</span>, <span class="va">coalOrtho</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/SimpleList-class.html" class="external-link">SimpleList</a></span><span class="op">(</span><span class="op">)</span>
<span class="va">orthoBlocks</span><span class="op">$</span><span class="va">Ply_Ros</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/coalesce_contigs.html" class="external-link">coalesce_contigs</a></span><span class="op">(</span><span class="va">orthoPairsFiltered</span><span class="op">$</span><span class="va">Ply_Ros</span><span class="op">)</span>
<span class="va">orthoBlocks</span><span class="op">[</span> <span class="fl">1</span><span class="op">:</span><span class="fl">15</span> <span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">orthoBlocks</span><span class="op">[</span> <span class="fl">1</span><span class="op">:</span><span class="fl">15</span> <span class="op">]</span>,  <span class="va">flagLongShort_</span>, <span class="va">longShort</span><span class="op">)</span>

<span class="va">orthoBlocks_core</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">orthoPairsFiltered_core</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">15</span><span class="op">]</span>, <span class="va">coalOrtho</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/SimpleList-class.html" class="external-link">SimpleList</a></span><span class="op">(</span><span class="op">)</span>
<span class="va">orthoBlocks_core</span><span class="op">$</span><span class="va">Ply_Ros</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/coalesce_contigs.html" class="external-link">coalesce_contigs</a></span><span class="op">(</span><span class="va">orthoPairsFiltered_core</span><span class="op">$</span><span class="va">Ply_Ros</span><span class="op">)</span>
<span class="va">orthoBlocks_core</span><span class="op">[</span> <span class="fl">1</span><span class="op">:</span><span class="fl">15</span> <span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">orthoBlocks_core</span><span class="op">[</span> <span class="fl">1</span><span class="op">:</span><span class="fl">15</span> <span class="op">]</span>,  <span class="va">flagLongShort_</span>, <span class="va">longShort</span><span class="op">)</span>

<span class="va">orthoBlocks_one2oneInOiks</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">orthoPairs_one2oneInOiks_Filtered</span>, <span class="va">coalOrtho</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/SimpleList-class.html" class="external-link">SimpleList</a></span><span class="op">(</span><span class="op">)</span>
<span class="va">orthoBlocks_one2oneInOiks</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">orthoBlocks_one2oneInOiks</span>,  <span class="va">flagLongShort_</span>, <span class="va">longShort</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/SimpleList-class.html" class="external-link">SimpleList</a></span><span class="op">(</span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">orthoPairsFiltered</span>, <span class="va">length</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Oki_Osa Oki_Bar Oki_Kum Oki_Aom Oki_Nor Osa_Bar Osa_Oki Osa_Kum Osa_Aom Osa_Nor </span>
<span class="co">##    9406    9142   11323    9438    8652    9074    9406    9351   10258    8312 </span>
<span class="co">## Bar_Osa Bar_Oki Bar_Kum Bar_Aom Bar_Nor Ply_Ros </span>
<span class="co">##    9074    9141    9104    9142    8383    5412</span></code></pre>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">orthoBlocks</span>, <span class="va">length</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Oki_Osa Oki_Bar Oki_Kum Oki_Aom Oki_Nor Osa_Bar Osa_Oki Osa_Kum Osa_Aom Osa_Nor </span>
<span class="co">##    3581    3375     316    3533    3514    1443    3581    3533     455    1806 </span>
<span class="co">## Bar_Osa Bar_Oki Bar_Kum Bar_Aom Bar_Nor Ply_Ros </span>
<span class="co">##    1443    3373    3333    1418     907    2185</span></code></pre>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">orthoPairsFiltered_core</span>, <span class="va">length</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Oki_Osa Oki_Bar Oki_Kum Oki_Aom Oki_Nor Osa_Bar Osa_Oki Osa_Kum Osa_Aom Osa_Nor </span>
<span class="co">##    6720    6612    8190    6809    6060    6635    6720    6630    7569    5933 </span>
<span class="co">## Bar_Osa Bar_Oki Bar_Kum Bar_Aom Bar_Nor Ply_Ros </span>
<span class="co">##    6635    6612    6558    6732    6087    2572</span></code></pre>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">orthoBlocks_core</span>, <span class="va">length</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Oki_Osa Oki_Bar Oki_Kum Oki_Aom Oki_Nor Osa_Bar Osa_Oki Osa_Kum Osa_Aom Osa_Nor </span>
<span class="co">##    2618    2484     226    2598    2496    1119    2618    2548     361    1403 </span>
<span class="co">## Bar_Osa Bar_Oki Bar_Kum Bar_Aom Bar_Nor Ply_Ros </span>
<span class="co">##    1119    2482    2449    1078     773     664</span></code></pre>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">orthoPairs_one2oneInOiks_Filtered</span>, <span class="va">length</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Oki_Osa Oki_Bar Oki_Kum Oki_Aom Oki_Nor Osa_Bar Osa_Oki Osa_Kum Osa_Aom Osa_Nor </span>
<span class="co">##    5144    5136    5148    5144    5124    5136    5144    5142    5146    5118 </span>
<span class="co">## Bar_Osa Bar_Oki Bar_Kum Bar_Aom Bar_Nor </span>
<span class="co">##    5136    5136    5136    5136    5110</span></code></pre>
<div class="sourceCode" id="cb38"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">orthoBlocks_one2oneInOiks</span>, <span class="va">length</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Oki_Osa Oki_Bar Oki_Kum Oki_Aom Oki_Nor Osa_Bar Osa_Oki Osa_Kum Osa_Aom Osa_Nor </span>
<span class="co">##    2074    2013      93    2058    2227     848    2074    2076     188    1204 </span>
<span class="co">## Bar_Osa Bar_Oki Bar_Kum Bar_Aom Bar_Nor </span>
<span class="co">##     848    2013    2013     826     601</span></code></pre>
</div>
<div class="section level3">
<h3 id="compute-bridge-regions">Compute bridge regions<a class="anchor" aria-label="anchor" href="#compute-bridge-regions"></a>
</h3>
<p>Let’s pre-compute bridge regions with the <code><a href="https://oist.github.io/GenomicBreaks/reference/bridgeRegions.html" class="external-link">GenomicBreaks::bridgeRegions()</a></code> function.</p>
<div class="sourceCode" id="cb40"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># oP_bridges               &lt;- sapply(orthoPairsFiltered,                bridgeRegions) |&gt; SimpleList()</span>
<span class="co"># oP_core_bridges          &lt;- sapply(orthoPairsFiltered_core,           bridgeRegions) |&gt; SimpleList()</span>
<span class="va">oP_one2oneInOiks_bridges</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">orthoPairs_one2oneInOiks_Filtered</span>, <span class="va">bridgeRegions</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/SimpleList-class.html" class="external-link">SimpleList</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="size-distribution-of-the-blocks">Size distribution of the blocks<a class="anchor" aria-label="anchor" href="#size-distribution-of-the-blocks"></a>
</h2>
<div class="sourceCode" id="cb41"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/graphics/hist.html" class="external-link">hist</a></span><span class="op">(</span><span class="va">orthoBlocks</span><span class="op">$</span><span class="va">Oki_Osa</span><span class="op">$</span><span class="va">geneNumber</span><span class="op">)</span></code></pre></div>
<p><img src="OrthogroupSyntenies_files/figure-html/quick_overview-1.png" width="700"></p>
<div class="sourceCode" id="cb42"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">orthoBlocks</span><span class="op">$</span><span class="va">Oki_Osa</span><span class="op">$</span><span class="va">geneNumber</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## </span>
<span class="co">##    1    2    3    4    5    6    7    8    9   10   11   12   13   14   15   16 </span>
<span class="co">## 2096  561  244  190   94   84   52   47   37   30   24   16   21   16    9    7 </span>
<span class="co">##   17   18   19   20   21   22   23   24   25   26   27   28   29   31   38   40 </span>
<span class="co">##   11   10    6    4    3    2    2    5    1    1    2    1    1    1    1    1 </span>
<span class="co">##   44 </span>
<span class="co">##    1</span></code></pre>
<div class="sourceCode" id="cb44"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">orthoBlocks</span><span class="op">$</span><span class="va">Oki_Osa</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/order.html" class="external-link">order</a></span><span class="op">(</span><span class="va">orthoBlocks</span><span class="op">$</span><span class="va">Oki_Osa</span><span class="op">$</span><span class="va">geneNumber</span><span class="op">)</span><span class="op">]</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">tail</a></span><span class="op">(</span><span class="fl">11</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## GBreaks object with 11 ranges and 8 metadata columns:</span>
<span class="co">##        seqnames            ranges strand |                 query     score</span>
<span class="co">##           &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; |             &lt;GRanges&gt; &lt;integer&gt;</span>
<span class="co">##    [1]      XSR   7131851-7238581      * |   XSR:3999405-4095337    106731</span>
<span class="co">##    [2]      PAR 15956464-16063993      * | PAR:10902456-10995025    107530</span>
<span class="co">##    [3]      XSR   3436117-3526092      * |   XSR:2172085-2264761     89976</span>
<span class="co">##    [4]      PAR 10906073-11000190      * |   PAR:9275473-9359780     94118</span>
<span class="co">##    [5]      XSR   7246798-7345874      * |   XSR:4104726-4194323     99077</span>
<span class="co">##    [6]      XSR   7562507-7713904      * |   XSR:3625761-3772339    151398</span>
<span class="co">##    [7]      XSR   6441868-6594575      * | XSR:10009542-10156377    152708</span>
<span class="co">##    [8]     chr2 14727816-14844427      * |  Chr2:8441071-8537126    116612</span>
<span class="co">##    [9]     chr2 10668710-10828262      * |  Chr2:5679751-5827304    159553</span>
<span class="co">##   [10]      PAR 15193339-15358002      * | PAR:11170991-11285909    164664</span>
<span class="co">##   [11]      XSR   6597685-6797170      * |  XSR:9796684-10005669    199486</span>
<span class="co">##                                geneNames              dNdS_GUIDANCE2</span>
<span class="co">##                          &lt;CharacterList&gt;               &lt;NumericList&gt;</span>
<span class="co">##    [1] g15372.t1,g15373.t1,g15375.t1,...      NA,0.01805,0.03746,...</span>
<span class="co">##    [2] g12997.t1,g12998.t1,g12999.t1,... 0.05858,0.00080,     NA,...</span>
<span class="co">##    [3] g14231.t1,g14232.t1,g14233.t1,...    0.0168,    NA,    NA,...</span>
<span class="co">##    [4] g11569.t1,g11570.t1,g11571.t1,... 0.02328,     NA,     NA,...</span>
<span class="co">##    [5] g15407.t1,g15408.t1,g15410.t1,... 0.00543,0.05808,     NA,...</span>
<span class="co">##    [6] g15509.t1,g15511.t1,g15512.t4,... 0.03440,0.09182,0.03833,...</span>
<span class="co">##    [7] g15154.t1,g15156.t1,g15157.t1,... 0.06059,     NA,0.02761,...</span>
<span class="co">##    [8]    g8160.t1,g8161.t1,g8163.t1,... 0.01678,0.03361,0.01543,...</span>
<span class="co">##    [9]    g6993.t1,g6996.t1,g6997.t2,...      NA,     NA,0.01518,...</span>
<span class="co">##   [10] g12780.t1,g12781.t1,g12782.t1,...                NA,NA,NA,...</span>
<span class="co">##   [11] g15204.t1,g15205.t1,g15206.t1,... 0.02776,0.02098,0.05115,...</span>
<span class="co">##                    dNdS_HmmCleaner                  dNdS_PRANK geneNumber</span>
<span class="co">##                      &lt;NumericList&gt;               &lt;NumericList&gt;  &lt;integer&gt;</span>
<span class="co">##    [1]      NA,0.01966,0.04001,...      NA,0.01939,0.04001,...         24</span>
<span class="co">##    [2] 0.05657,0.00080,     NA,... 0.05657,0.00080,     NA,...         25</span>
<span class="co">##    [3]    0.0168,    NA,    NA,... 0.01823,     NA,     NA,...         26</span>
<span class="co">##    [4] 0.02575,     NA,     NA,... 0.02575,     NA,     NA,...         27</span>
<span class="co">##    [5] 0.00712,0.05952,     NA,... 0.00893,0.05955,     NA,...         27</span>
<span class="co">##    [6] 0.03475,0.09710,0.04044,... 0.03525,0.09034,0.03789,...         28</span>
<span class="co">##    [7] 0.05943,     NA,0.03195,... 0.05943,     NA,0.03195,...         29</span>
<span class="co">##    [8] 0.01670,0.03300,0.01543,... 0.01938,0.03378,0.01543,...         31</span>
<span class="co">##    [9]      NA,     NA,0.01606,...      NA,     NA,0.01658,...         38</span>
<span class="co">##   [10]                NA,NA,NA,...                NA,NA,NA,...         40</span>
<span class="co">##   [11] 0.02776,0.02367,0.05515,... 0.02776,0.02364,0.05424,...         44</span>
<span class="co">##             Arm</span>
<span class="co">##        &lt;factor&gt;</span>
<span class="co">##    [1]     XSR </span>
<span class="co">##    [2]     long</span>
<span class="co">##    [3]     XSR </span>
<span class="co">##    [4]     long</span>
<span class="co">##    [5]     XSR </span>
<span class="co">##    [6]     XSR </span>
<span class="co">##    [7]     XSR </span>
<span class="co">##    [8]     long</span>
<span class="co">##    [9]     long</span>
<span class="co">##   [10]     long</span>
<span class="co">##   [11]     XSR </span>
<span class="co">##   -------</span>
<span class="co">##   seqinfo: 19 sequences from OKI2018.I69 genome</span></code></pre>
<div class="sourceCode" id="cb46"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/graphics/hist.html" class="external-link">hist</a></span><span class="op">(</span><span class="va">orthoBlocks</span><span class="op">$</span><span class="va">Oki_Kum</span><span class="op">$</span><span class="va">geneNumber</span><span class="op">)</span></code></pre></div>
<p><img src="OrthogroupSyntenies_files/figure-html/quick_overview-2.png" width="700"></p>
<div class="sourceCode" id="cb47"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">orthoBlocks</span><span class="op">$</span><span class="va">Oki_Kum</span><span class="op">$</span><span class="va">geneNumber</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## </span>
<span class="co">##   1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17  18  19  20 </span>
<span class="co">##  76  24  20  15   8   4   8   3   5   4   7   4   3   4   4   2   3   4   4   3 </span>
<span class="co">##  22  23  24  25  26  27  28  29  30  31  32  33  34  36  37  39  40  42  44  47 </span>
<span class="co">##   2   2   2   3   3   2   2   1   2   3   2   1   1   3   1   2   1   3   2   1 </span>
<span class="co">##  48  49  50  51  54  55  56  57  58  61  63  64  65  68  69  70  71  75  76  78 </span>
<span class="co">##   2   1   3   1   3   1   2   2   1   1   1   1   4   2   2   1   1   1   2   1 </span>
<span class="co">##  80  81  83  87  90  92  94  96  97 102 103 104 106 107 114 117 118 141 142 143 </span>
<span class="co">##   1   1   1   1   1   1   2   1   1   2   1   1   1   1   1   1   2   1   1   1 </span>
<span class="co">## 149 161 170 180 197 199 200 264 269 308 356 367 423 438 459 475 </span>
<span class="co">##   1   1   1   1   1   1   1   1   1   1   1   1   1   1   1   1</span></code></pre>
<div class="sourceCode" id="cb49"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">orthoBlocks</span><span class="op">$</span><span class="va">Oki_Kum</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/order.html" class="external-link">order</a></span><span class="op">(</span><span class="va">orthoBlocks</span><span class="op">$</span><span class="va">Oki_Kum</span><span class="op">$</span><span class="va">geneNumber</span><span class="op">)</span><span class="op">]</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">tail</a></span><span class="op">(</span><span class="fl">11</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## GBreaks object with 11 ranges and 8 metadata columns:</span>
<span class="co">##        seqnames            ranges strand |                        query</span>
<span class="co">##           &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; |                    &lt;GRanges&gt;</span>
<span class="co">##    [1]      PAR  9991049-10823433      * |       contig_28_1:906-916604</span>
<span class="co">##    [2]      XSR   1398297-2355032      * |  contig_42_1:1522014-2519191</span>
<span class="co">##    [3]     chr1   6729778-8085886      * |      contig_90_1:571-1399045</span>
<span class="co">##    [4]     chr1  8676861-10061915      * |    contig_3_1:618849-1981910</span>
<span class="co">##    [5]      PAR   7792256-9333620      * |   contig_82_1:846722-2300436</span>
<span class="co">##    [6]      PAR 11974020-13925498      * |  contig_28_1:2026940-3782149</span>
<span class="co">##    [7]      XSR   7904454-9304595      * |  contig_42_1:7880616-9254847</span>
<span class="co">##    [8]     chr2   5748389-7696653      * |    contig_24_1:12160-1868534</span>
<span class="co">##    [9]      XSR   4988262-6642828      * |  contig_42_1:4994508-6631430</span>
<span class="co">##   [10]      XSR  9308304-11123841      * | contig_42_1:9258557-11063801</span>
<span class="co">##   [11]      XSR   2649320-4576212      * |  contig_42_1:2787081-4661476</span>
<span class="co">##            score                         geneNames              dNdS_GUIDANCE2</span>
<span class="co">##        &lt;integer&gt;                   &lt;CharacterList&gt;               &lt;NumericList&gt;</span>
<span class="co">##    [1]    832385 g11259.t1,g11261.t1,g11262.t1,...      NA,     NA,0.01453,...</span>
<span class="co">##    [2]    956736 g13662.t1,g13663.t1,g13665.t1,...      NA,0.03728,0.01538,...</span>
<span class="co">##    [3]   1356109    g1763.t1,g1772.t1,g1773.t1,...                NA,NA,NA,...</span>
<span class="co">##    [4]   1385055    g2316.t1,g2317.t1,g2319.t1,...      NA,0.01342,0.01277,...</span>
<span class="co">##    [5]   1541365 g10595.t1,g10596.t1,g10597.t1,...      NA,0.01305,0.02206,...</span>
<span class="co">##    [6]   1951479 g11881.t1,g11883.t1,g11885.t1,...                NA,NA,NA,...</span>
<span class="co">##    [7]   1400142 g15616.t1,g15617.t1,g15619.t1,... 0.01284,0.03441,     NA,...</span>
<span class="co">##    [8]   1948265    g5514.t1,g5516.t1,g5518.t1,...                NA,NA,NA,...</span>
<span class="co">##    [9]   1654567 g14704.t1,g14705.t1,g14706.t1,... 0.00618,0.02254,     NA,...</span>
<span class="co">##   [10]   1815538 g16044.t1,g16046.t1,g16049.t1,...      NA,0.01639,0.02433,...</span>
<span class="co">##   [11]   1926893 g13995.t1,g13996.t1,g13998.t1,... 0.03604,0.10111,     NA,...</span>
<span class="co">##                    dNdS_HmmCleaner                  dNdS_PRANK geneNumber</span>
<span class="co">##                      &lt;NumericList&gt;               &lt;NumericList&gt;  &lt;integer&gt;</span>
<span class="co">##    [1]      NA,     NA,0.01482,...      NA,     NA,0.01568,...        199</span>
<span class="co">##    [2]      NA,0.03195,0.01978,...      NA,0.03195,0.01978,...        200</span>
<span class="co">##    [3]                NA,NA,NA,...                NA,NA,NA,...        264</span>
<span class="co">##    [4]      NA,0.01342,0.01277,...      NA,0.01530,0.01277,...        269</span>
<span class="co">##    [5]      NA,0.01227,0.02460,...      NA,0.01227,0.02460,...        308</span>
<span class="co">##    [6]                NA,NA,NA,...                NA,NA,NA,...        356</span>
<span class="co">##    [7] 0.01400,0.00844,     NA,... 0.01579,0.00844,     NA,...        367</span>
<span class="co">##    [8]                NA,NA,NA,...                NA,NA,NA,...        423</span>
<span class="co">##    [9] 0.00656,0.01781,     NA,... 0.00656,0.01781,     NA,...        438</span>
<span class="co">##   [10]      NA,0.02021,0.02433,...      NA,0.02021,0.02433,...        459</span>
<span class="co">##   [11] 0.03765,0.10611,     NA,... 0.03765,0.10611,     NA,...        475</span>
<span class="co">##             Arm</span>
<span class="co">##        &lt;factor&gt;</span>
<span class="co">##    [1]     long</span>
<span class="co">##    [2]     XSR </span>
<span class="co">##    [3]     long</span>
<span class="co">##    [4]     long</span>
<span class="co">##    [5]     long</span>
<span class="co">##    [6]     long</span>
<span class="co">##    [7]     XSR </span>
<span class="co">##    [8]     long</span>
<span class="co">##    [9]     XSR </span>
<span class="co">##   [10]     XSR </span>
<span class="co">##   [11]     XSR </span>
<span class="co">##   -------</span>
<span class="co">##   seqinfo: 19 sequences from OKI2018.I69 genome</span></code></pre>
</div>
<div class="section level2">
<h2 id="gene-order-conservation-is-smaller-in-short-arms">Gene Order Conservation is smaller in short arms<a class="anchor" aria-label="anchor" href="#gene-order-conservation-is-smaller-in-short-arms"></a>
</h2>
<div class="sourceCode" id="cb51"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">orthoBlocks_LSnoNA</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">orthoBlocks</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">15</span><span class="op">]</span>, <span class="kw">function</span><span class="op">(</span><span class="va">gb</span><span class="op">)</span> <span class="va">gb</span><span class="op">[</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">gb</span><span class="op">$</span><span class="va">Arm</span><span class="op">)</span><span class="op">]</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/SimpleList-class.html" class="external-link">SimpleList</a></span><span class="op">(</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">orthoBlocks_LSnoNA</span>, <span class="va">length</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Oki_Osa Oki_Bar Oki_Kum Oki_Aom Oki_Nor Osa_Bar Osa_Oki Osa_Kum Osa_Aom Osa_Nor </span>
<span class="co">##    3579    3373     313    3531    3512    1402    3524    3480     410    1763 </span>
<span class="co">## Bar_Osa Bar_Oki Bar_Kum Bar_Aom Bar_Nor </span>
<span class="co">##    1423    3350    3308    1398     892</span></code></pre>
<div class="sourceCode" id="cb53"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">df</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">orthoBlocks_LSnoNA</span>,
             <span class="kw">function</span><span class="op">(</span><span class="va">gb</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/tapply.html" class="external-link">tapply</a></span><span class="op">(</span><span class="va">gb</span>, <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqnames</a></span><span class="op">(</span><span class="va">gb</span><span class="op">)</span>, <span class="va">gb</span><span class="op">$</span><span class="va">Arm</span><span class="op">)</span>, <span class="va">GOC</span><span class="op">)</span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/r/base/apply.html" class="external-link">apply</a></span><span class="op">(</span><span class="va">df</span>, <span class="fl">2</span>, \<span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="op">{</span>
  <span class="va">tst</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/t.test.html" class="external-link">t.test</a></span><span class="op">(</span><span class="va">x</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">3</span>,<span class="fl">5</span><span class="op">)</span><span class="op">]</span>, <span class="va">x</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">2</span>,<span class="fl">4</span>,<span class="fl">6</span><span class="op">)</span><span class="op">]</span><span class="op">)</span>
  <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span>pval<span class="op">=</span><span class="va">tst</span><span class="op">$</span><span class="va">p.value</span>, mx<span class="op">=</span><span class="va">tst</span><span class="op">$</span><span class="va">estimate</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>, my<span class="op">=</span><span class="va">tst</span><span class="op">$</span><span class="va">estimate</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span><span class="op">)</span>
<span class="op">}</span><span class="op">)</span> </code></pre></div>
<pre><code><span class="co">##                   Oki_Osa     Oki_Bar   Oki_Kum    Oki_Aom     Oki_Nor</span>
<span class="co">## pval         0.0003445748 0.001195248 0.4563762 0.01734870 0.008449444</span>
<span class="co">## mx.mean of x 0.1853163526 0.181080056 0.5647566 0.18213329 0.174234572</span>
<span class="co">## my.mean of y 0.0629330841 0.061442980 0.6188030 0.06829154 0.062364224</span>
<span class="co">##                 Osa_Bar     Osa_Oki    Osa_Kum   Osa_Aom    Osa_Nor   Bar_Osa</span>
<span class="co">## pval         0.05013474 0.002520974 0.00755681 0.0337040 0.01972651 0.1684680</span>
<span class="co">## mx.mean of x 0.42449520 0.196644110 0.18306629 0.7109280 0.32803812 0.3897294</span>
<span class="co">## my.mean of y 0.28083551 0.073813557 0.07284716 0.5246518 0.20384140 0.2739641</span>
<span class="co">##                   Bar_Oki     Bar_Kum    Bar_Aom   Bar_Nor</span>
<span class="co">## pval         0.0007675276 0.008239524 0.09425889 0.7257221</span>
<span class="co">## mx.mean of x 0.1855144293 0.183651905 0.41363040 0.3391719</span>
<span class="co">## my.mean of y 0.0709205522 0.070638469 0.27714160 0.3885798</span></code></pre>
<details><p><summary>Click to expand results on a microbenchmark comparing <code>sapply(split)</code> and <code>tapply</code></summary></p>
<pre><code>&gt; microbenchmark::microbenchmark(
  times = 10,
  sapply = sapply(split(x, paste(seqnames(x), x$Arm)), GOC),
  tapply = tapply(x, paste(seqnames(x), x$Arm), GOC)
)
Unit: seconds
   expr      min       lq     mean   median       uq      max neval
 sapply 6.853530 6.921293 7.050267 6.949315 7.270793 7.477390    10
 tapply 6.921118 6.923827 7.028373 6.939487 6.968288 7.431115    10</code></pre>
</details>
</div>
<div class="section level2">
<h2 id="fraction-of-the-genome-covered-or-syntenic">Fraction of the genome covered or syntenic<a class="anchor" aria-label="anchor" href="#fraction-of-the-genome-covered-or-syntenic"></a>
</h2>
<div class="sourceCode" id="cb56"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">uncov</span>  <span class="op">&lt;-</span> <span class="fu">BiocParallel</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocParallel/man/bplapply.html" class="external-link">bplapply</a></span><span class="op">(</span><span class="va">orthoPairsFiltered</span>, <span class="va">cleanGaps</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/SimpleList-class.html" class="external-link">SimpleList</a></span><span class="op">(</span><span class="op">)</span>
<span class="va">unsyn</span>  <span class="op">&lt;-</span> <span class="fu">BiocParallel</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocParallel/man/bplapply.html" class="external-link">bplapply</a></span><span class="op">(</span><span class="va">orthoBlocks</span>,        <span class="va">cleanGaps</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/SimpleList-class.html" class="external-link">SimpleList</a></span><span class="op">(</span><span class="op">)</span>

  <span class="va">covered_tot</span>    <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">orthoPairsFiltered</span>,   \<span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/start.html" class="external-link">width</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
  <span class="va">syntenic_tot</span>   <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">orthoBlocks</span>,          \<span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/start.html" class="external-link">width</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
<span class="va">uncovered_tot</span>    <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">uncov</span>,                \<span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/start.html" class="external-link">width</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
<span class="va">unsyntenic_tot</span>   <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">unsyn</span>,                \<span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/start.html" class="external-link">width</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>

<span class="op">(</span><span class="va">syn_summary</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/cbind.html" class="external-link">cbind</a></span><span class="op">(</span>
  covered_frac   <span class="op">=</span> <span class="va">covered_tot</span>   <span class="op">/</span> <span class="op">(</span> <span class="va">covered_tot</span>   <span class="op">+</span> <span class="va">uncovered_tot</span> <span class="op">)</span> <span class="op">*</span> <span class="fl">100</span>,
  orthogenes_n   <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">orthoPairsFiltered</span>, <span class="va">length</span><span class="op">)</span>,
  syntenic_frac  <span class="op">=</span> <span class="va">syntenic_tot</span>  <span class="op">/</span> <span class="op">(</span> <span class="va">syntenic_tot</span>  <span class="op">+</span> <span class="va">unsyntenic_tot</span> <span class="op">)</span> <span class="op">*</span> <span class="fl">100</span>,
  syntenic_reg   <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">orthoBlocks</span>, <span class="va">length</span><span class="op">)</span>
 <span class="op">)</span>|&gt; <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">##         covered_frac orthogenes_n syntenic_frac syntenic_reg</span>
<span class="co">## Oki_Osa     38.33865         9406      61.29621         3581</span>
<span class="co">## Oki_Bar     38.21979         9142      60.42189         3375</span>
<span class="co">## Oki_Kum     43.70838        11323      92.95563          316</span>
<span class="co">## Oki_Aom     38.60839         9438      61.35116         3533</span>
<span class="co">## Oki_Nor     34.90562         8652      54.26659         3514</span>
<span class="co">## Osa_Bar     42.10034         9074      76.09101         1443</span>
<span class="co">## Osa_Oki     43.87692         9406      65.46897         3581</span>
<span class="co">## Osa_Kum     42.96814         9351      63.74943         3533</span>
<span class="co">## Osa_Aom     45.38916        10258      88.99277          455</span>
<span class="co">## Osa_Nor     38.20910         8312      69.36702         1806</span>
<span class="co">## Bar_Osa     40.94778         9074      73.69212         1443</span>
<span class="co">## Bar_Oki     41.86938         9141      61.92448         3373</span>
<span class="co">## Bar_Kum     42.24314         9104      62.47976         3333</span>
<span class="co">## Bar_Aom     41.88404         9142      74.65392         1418</span>
<span class="co">## Bar_Nor     38.13315         8383      75.49597          907</span>
<span class="co">## Ply_Ros     25.53249         5412      56.86698         2185</span></code></pre>
<div class="sourceCode" id="cb58"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">syn_summary</span>, \<span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/tapply.html" class="external-link">tapply</a></span><span class="op">(</span><span class="va">x</span>, <span class="fu"><a href="https://rdrr.io/r/base/row.names.html" class="external-link">row.names</a></span><span class="op">(</span><span class="va">syn_summary</span><span class="op">)</span> |&gt; <span class="fu">OikScrambling</span><span class="fu">:::</span><span class="fu"><a href="../reference/compDistance.html">compDistance</a></span><span class="op">(</span><span class="op">)</span>, <span class="va">mean</span><span class="op">)</span><span class="op">)</span>   |&gt; <span class="fu"><a href="https://rdrr.io/r/base/Round.html" class="external-link">round</a></span><span class="op">(</span><span class="fl">1</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">##               covered_frac orthogenes_n syntenic_frac syntenic_reg</span>
<span class="co">## In same pop           42.4       9988.0          85.8        559.3</span>
<span class="co">## Int – Int             25.5       5412.0          56.9       2185.0</span>
<span class="co">## North – North         40.8       8900.5          73.5       1527.5</span>
<span class="co">## Oki – North           40.1       9205.0          61.4       3477.9</span></code></pre>
<div class="sourceCode" id="cb60"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">syn_summary</span>, \<span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/tapply.html" class="external-link">tapply</a></span><span class="op">(</span><span class="va">x</span>, <span class="fu"><a href="https://rdrr.io/r/base/row.names.html" class="external-link">row.names</a></span><span class="op">(</span><span class="va">syn_summary</span><span class="op">)</span> |&gt; <span class="fu">OikScrambling</span><span class="fu">:::</span><span class="fu"><a href="../reference/compDistance.html">compDistance</a></span><span class="op">(</span><span class="op">)</span>, <span class="va">median</span><span class="op">)</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/r/base/Round.html" class="external-link">round</a></span><span class="op">(</span><span class="fl">1</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">##               covered_frac orthogenes_n syntenic_frac syntenic_reg</span>
<span class="co">## In same pop           43.7      10258.0          89.0        455.0</span>
<span class="co">## Int – Int             25.5       5412.0          56.9       2185.0</span>
<span class="co">## North – North         41.4       9074.0          74.2       1443.0</span>
<span class="co">## Oki – North           40.2       9246.5          61.6       3523.5</span></code></pre>
<div class="sourceCode" id="cb62"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">syn_summary</span>, \<span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/tapply.html" class="external-link">tapply</a></span><span class="op">(</span><span class="va">x</span>, <span class="fu"><a href="https://rdrr.io/r/base/row.names.html" class="external-link">row.names</a></span><span class="op">(</span><span class="va">syn_summary</span><span class="op">)</span> |&gt; <span class="fu">OikScrambling</span><span class="fu">:::</span><span class="fu"><a href="../reference/compDistance.html">compDistance</a></span><span class="op">(</span><span class="op">)</span>, <span class="va">sd</span><span class="op">)</span><span class="op">)</span>     |&gt; <span class="fu"><a href="https://rdrr.io/r/base/Round.html" class="external-link">round</a></span><span class="op">(</span><span class="fl">1</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">##               covered_frac orthogenes_n syntenic_frac syntenic_reg</span>
<span class="co">## In same pop            3.8       1488.5           9.2        309.0</span>
<span class="co">## Int – Int               NA           NA            NA           NA</span>
<span class="co">## North – North          1.8        393.6           2.9        186.0</span>
<span class="co">## Oki – North            3.1        262.0           3.3        100.9</span></code></pre>
<p>Unfortunately, there are not enough transcripts from Ciona that are part of the pan-tunicate orthogroup set, which makes the comparison difficult between Oikopleura and Ciona…</p>
<div class="sourceCode" id="cb64"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">uncov</span>  <span class="op">&lt;-</span> <span class="fu">BiocParallel</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocParallel/man/bplapply.html" class="external-link">bplapply</a></span><span class="op">(</span><span class="va">orthoPairsFiltered_core</span>, <span class="va">cleanGaps</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/SimpleList-class.html" class="external-link">SimpleList</a></span><span class="op">(</span><span class="op">)</span>
<span class="va">unsyn</span>  <span class="op">&lt;-</span> <span class="fu">BiocParallel</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocParallel/man/bplapply.html" class="external-link">bplapply</a></span><span class="op">(</span><span class="va">orthoBlocks_core</span>,        <span class="va">cleanGaps</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/SimpleList-class.html" class="external-link">SimpleList</a></span><span class="op">(</span><span class="op">)</span>

  <span class="va">covered_tot</span>    <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">orthoPairsFiltered_core</span>,   \<span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/start.html" class="external-link">width</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
  <span class="va">syntenic_tot</span>   <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">orthoBlocks_core</span>,          \<span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/start.html" class="external-link">width</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
<span class="va">uncovered_tot</span>    <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">uncov</span>,                     \<span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/start.html" class="external-link">width</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
<span class="va">unsyntenic_tot</span>   <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">unsyn</span>,                     \<span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/start.html" class="external-link">width</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>

<span class="op">(</span><span class="va">syn_summary_core</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/cbind.html" class="external-link">cbind</a></span><span class="op">(</span>
  covered_frac   <span class="op">=</span> <span class="va">covered_tot</span>   <span class="op">/</span> <span class="op">(</span> <span class="va">covered_tot</span>   <span class="op">+</span> <span class="va">uncovered_tot</span> <span class="op">)</span> <span class="op">*</span> <span class="fl">100</span>,
  orthogenes_n   <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">orthoPairsFiltered_core</span>, <span class="va">length</span><span class="op">)</span>,
  syntenic_frac  <span class="op">=</span> <span class="va">syntenic_tot</span>  <span class="op">/</span> <span class="op">(</span> <span class="va">syntenic_tot</span>  <span class="op">+</span> <span class="va">unsyntenic_tot</span> <span class="op">)</span> <span class="op">*</span> <span class="fl">100</span>,
  syntenic_reg   <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">orthoBlocks_core</span>, <span class="va">length</span><span class="op">)</span>
 <span class="op">)</span>|&gt; <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">##         covered_frac orthogenes_n syntenic_frac syntenic_reg</span>
<span class="co">## Oki_Osa     27.48424         6720      51.25475         2618</span>
<span class="co">## Oki_Bar     28.30197         6612      53.38051         2484</span>
<span class="co">## Oki_Kum     31.22509         8190      91.90604          226</span>
<span class="co">## Oki_Aom     28.75649         6809      53.16578         2598</span>
<span class="co">## Oki_Nor     24.52914         6060      43.63803         2496</span>
<span class="co">## Osa_Bar     31.13847         6635      70.58493         1119</span>
<span class="co">## Osa_Oki     30.76206         6720      53.18648         2618</span>
<span class="co">## Osa_Kum     30.69252         6630      53.01613         2548</span>
<span class="co">## Osa_Aom     33.29049         7569      86.52898          361</span>
<span class="co">## Osa_Nor     27.50044         5933      62.12065         1403</span>
<span class="co">## Bar_Osa     30.15263         6635      67.64302         1119</span>
<span class="co">## Bar_Oki     30.84109         6612      52.63710         2482</span>
<span class="co">## Bar_Kum     31.17019         6558      52.57260         2449</span>
<span class="co">## Bar_Aom     31.42929         6732      69.47170         1078</span>
<span class="co">## Bar_Nor     27.93441         6087      70.48772          773</span>
<span class="co">## Ply_Ros     11.39509         2572      50.30680          664</span></code></pre>
<div class="sourceCode" id="cb66"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">syn_summary_core</span>, \<span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/tapply.html" class="external-link">tapply</a></span><span class="op">(</span><span class="va">x</span>, <span class="fu"><a href="https://rdrr.io/r/base/row.names.html" class="external-link">row.names</a></span><span class="op">(</span><span class="va">syn_summary_core</span><span class="op">)</span> |&gt; <span class="fu">OikScrambling</span><span class="fu">:::</span><span class="fu"><a href="../reference/compDistance.html">compDistance</a></span><span class="op">(</span><span class="op">)</span>, <span class="va">mean</span><span class="op">)</span><span class="op">)</span>   |&gt; <span class="fu"><a href="https://rdrr.io/r/base/Round.html" class="external-link">round</a></span><span class="op">(</span><span class="fl">1</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">##               covered_frac orthogenes_n syntenic_frac syntenic_reg</span>
<span class="co">## In same pop           30.8       7282.0          83.0        453.3</span>
<span class="co">## Int – Int             11.4       2572.0          50.3        664.0</span>
<span class="co">## North – North         30.1       6483.8          67.5       1179.8</span>
<span class="co">## Oki – North           29.1       6590.1          51.6       2536.6</span></code></pre>
<div class="sourceCode" id="cb68"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">syn_summary_core</span>, \<span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/tapply.html" class="external-link">tapply</a></span><span class="op">(</span><span class="va">x</span>, <span class="fu"><a href="https://rdrr.io/r/base/row.names.html" class="external-link">row.names</a></span><span class="op">(</span><span class="va">syn_summary_core</span><span class="op">)</span> |&gt; <span class="fu">OikScrambling</span><span class="fu">:::</span><span class="fu"><a href="../reference/compDistance.html">compDistance</a></span><span class="op">(</span><span class="op">)</span>, <span class="va">median</span><span class="op">)</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/r/base/Round.html" class="external-link">round</a></span><span class="op">(</span><span class="fl">1</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">##               covered_frac orthogenes_n syntenic_frac syntenic_reg</span>
<span class="co">## In same pop           31.2         7569          86.5          361</span>
<span class="co">## Int – Int             11.4         2572          50.3          664</span>
<span class="co">## North – North         30.6         6635          68.6         1119</span>
<span class="co">## Oki – North           29.7         6621          52.8         2522</span></code></pre>
<div class="sourceCode" id="cb70"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">syn_summary_core</span>, \<span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/tapply.html" class="external-link">tapply</a></span><span class="op">(</span><span class="va">x</span>, <span class="fu"><a href="https://rdrr.io/r/base/row.names.html" class="external-link">row.names</a></span><span class="op">(</span><span class="va">syn_summary_core</span><span class="op">)</span> |&gt; <span class="fu">OikScrambling</span><span class="fu">:::</span><span class="fu"><a href="../reference/compDistance.html">compDistance</a></span><span class="op">(</span><span class="op">)</span>, <span class="va">sd</span><span class="op">)</span><span class="op">)</span>     |&gt; <span class="fu"><a href="https://rdrr.io/r/base/Round.html" class="external-link">round</a></span><span class="op">(</span><span class="fl">1</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">##               covered_frac orthogenes_n syntenic_frac syntenic_reg</span>
<span class="co">## In same pop            2.7       1080.5          11.1        284.9</span>
<span class="co">## Int – Int               NA           NA            NA           NA</span>
<span class="co">## North – North          1.8        370.0           3.8        150.1</span>
<span class="co">## Oki – North            2.3        228.7           3.3         67.8</span></code></pre>
</div>
<div class="section level2">
<h2 id="agreement-with-nucleotide-alignments">Agreement with nucleotide alignments<a class="anchor" aria-label="anchor" href="#agreement-with-nucleotide-alignments"></a>
</h2>
<div class="sourceCode" id="cb72"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="op">(</span><span class="va">gb_P</span> <span class="op">&lt;-</span> <span class="va">orthoBlocks</span><span class="op">$</span><span class="va">Bar_Osa</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## GBreaks object with 1443 ranges and 8 metadata columns:</span>
<span class="co">##          seqnames          ranges strand |                query     score</span>
<span class="co">##             &lt;Rle&gt;       &lt;IRanges&gt;  &lt;Rle&gt; |            &lt;GRanges&gt; &lt;integer&gt;</span>
<span class="co">##      [1]     Chr1     17796-21383      * | Chr1:3581251-3591380      3588</span>
<span class="co">##      [2]     Chr1     27418-60748      * | Chr1:3596010-3622624     33331</span>
<span class="co">##      [3]     Chr1     69763-74781      * | Chr1:3644537-3650503      5019</span>
<span class="co">##      [4]     Chr1     87558-91675      * | Chr1:3818944-3824036      4118</span>
<span class="co">##      [5]     Chr1    92754-103548      * | Chr1:3833970-3843493     10795</span>
<span class="co">##      ...      ...             ...    ... .                  ...       ...</span>
<span class="co">##   [1439]      YSR 4033286-4033798      * |  YSR:1457970-1458977       513</span>
<span class="co">##   [1440]      YSR 4086487-4086957      * | Chr1:8499823-8501898       471</span>
<span class="co">##   [1441]      YSR 4321609-4322813      * |  YSR:1599784-1602684      1205</span>
<span class="co">##   [1442]      YSR 4865878-4866414      * |    YSR:434383-434709       537</span>
<span class="co">##   [1443]      YSR 5097970-5098632      * |  YSR:1729503-1738428       663</span>
<span class="co">##                         geneNames              dNdS_GUIDANCE2</span>
<span class="co">##                   &lt;CharacterList&gt;               &lt;NumericList&gt;</span>
<span class="co">##      [1]              g1.t1,g2.t1             0.04071,0.04756</span>
<span class="co">##      [2]    g4.t1,g5.t1,g6.t1,...                NA,NA,NA,...</span>
<span class="co">##      [3] g16.t1,g17.t1,g18.t1,... 0.00746,0.03727,0.06337,...</span>
<span class="co">##      [4]            g22.t1,g23.t1                       NA,NA</span>
<span class="co">##      [5]            g24.t1,g25.t1                  NA,0.04723</span>
<span class="co">##      ...                      ...                         ...</span>
<span class="co">##   [1439]                g13965.t1                          NA</span>
<span class="co">##   [1440]                g13972.t1                          NA</span>
<span class="co">##   [1441]                g13991.t1                          NA</span>
<span class="co">##   [1442]                g14068.t1                          NA</span>
<span class="co">##   [1443]                g14100.t1                          NA</span>
<span class="co">##                      dNdS_HmmCleaner                  dNdS_PRANK geneNumber</span>
<span class="co">##                        &lt;NumericList&gt;               &lt;NumericList&gt;  &lt;integer&gt;</span>
<span class="co">##      [1]             0.04071,0.05590             0.04071,0.05407          2</span>
<span class="co">##      [2]                NA,NA,NA,...                NA,NA,NA,...          8</span>
<span class="co">##      [3] 0.00746,0.04309,0.06258,... 0.00746,0.04309,0.06258,...          4</span>
<span class="co">##      [4]                       NA,NA                       NA,NA          2</span>
<span class="co">##      [5]                  NA,0.04214                  NA,0.04431          2</span>
<span class="co">##      ...                         ...                         ...        ...</span>
<span class="co">##   [1439]                          NA                          NA          1</span>
<span class="co">##   [1440]                          NA                          NA          1</span>
<span class="co">##   [1441]                          NA                          NA          1</span>
<span class="co">##   [1442]                          NA                          NA          1</span>
<span class="co">##   [1443]                          NA                          NA          1</span>
<span class="co">##               Arm</span>
<span class="co">##          &lt;factor&gt;</span>
<span class="co">##      [1]    short</span>
<span class="co">##      [2]    short</span>
<span class="co">##      [3]    short</span>
<span class="co">##      [4]    short</span>
<span class="co">##      [5]    short</span>
<span class="co">##      ...      ...</span>
<span class="co">##   [1439]      YSR</span>
<span class="co">##   [1440]      YSR</span>
<span class="co">##   [1441]      YSR</span>
<span class="co">##   [1442]      YSR</span>
<span class="co">##   [1443]      YSR</span>
<span class="co">##   -------</span>
<span class="co">##   seqinfo: 68 sequences from Bar2.p4 genome</span></code></pre>
<div class="sourceCode" id="cb74"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/start.html" class="external-link">width</a></span><span class="op">(</span><span class="va">gb_P</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## [1] 39929544</span></code></pre>
<div class="sourceCode" id="cb76"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="op">(</span><span class="va">gb_N</span> <span class="op">&lt;-</span> <span class="va">coa</span><span class="op">$</span><span class="va">Bar_Osa</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## GBreaks object with 4073 ranges and 8 metadata columns:</span>
<span class="co">##          seqnames          ranges strand |                query     score</span>
<span class="co">##             &lt;Rle&gt;       &lt;IRanges&gt;  &lt;Rle&gt; |            &lt;GRanges&gt; &lt;integer&gt;</span>
<span class="co">##      [1]     Chr1     15573-21624      - | Chr1:3585703-3594761      6052</span>
<span class="co">##      [2]     Chr1     22042-22295      + | Chr1:3585140-3585392       254</span>
<span class="co">##      [3]     Chr1     22421-24048      - | Chr1:3583028-3584936      1628</span>
<span class="co">##      [4]     Chr1     27262-34408      + | Chr1:3595848-3601817      7147</span>
<span class="co">##      [5]     Chr1     34435-34998      - | Chr1:3602092-3602509       564</span>
<span class="co">##      ...      ...             ...    ... .                  ...       ...</span>
<span class="co">##   [4069]      YSR 5273802-5274176      - |  YSR:1632176-1632556       375</span>
<span class="co">##   [4070]      YSR 5288325-5289549      - |  YSR:1086332-1087481      1225</span>
<span class="co">##   [4071]      YSR 5307497-5310556      - | Chr1:9561929-9564982      3060</span>
<span class="co">##   [4072]      YSR 5322809-5325938      - |  PAR:5433966-5437098      3130</span>
<span class="co">##   [4073]      YSR 5335965-5336405      - |   Chr2:928224-928659       441</span>
<span class="co">##               Arm                rep   repOvlp transcripts        flag</span>
<span class="co">##          &lt;factor&gt;    &lt;CharacterList&gt; &lt;integer&gt;       &lt;Rle&gt; &lt;character&gt;</span>
<span class="co">##      [1]    short tandem,unknown,rnd       210       g2.t1         Inv</span>
<span class="co">##      [2]    short               &lt;NA&gt;         0        &lt;NA&gt;        &lt;NA&gt;</span>
<span class="co">##      [3]    short               &lt;NA&gt;         0        &lt;NA&gt;        &lt;NA&gt;</span>
<span class="co">##      [4]    short                rnd       390        &lt;NA&gt;         Inv</span>
<span class="co">##      [5]    short                rnd         0       g5.t1        &lt;NA&gt;</span>
<span class="co">##      ...      ...                ...       ...         ...         ...</span>
<span class="co">##   [4069]      YSR               &lt;NA&gt;         0        &lt;NA&gt;        &lt;NA&gt;</span>
<span class="co">##   [4070]      YSR               &lt;NA&gt;         0   g14124.t1        &lt;NA&gt;</span>
<span class="co">##   [4071]      YSR                rnd         0   g14124.t1        &lt;NA&gt;</span>
<span class="co">##   [4072]      YSR                rnd         0        &lt;NA&gt;        &lt;NA&gt;</span>
<span class="co">##   [4073]      YSR                rnd       304        &lt;NA&gt;        &lt;NA&gt;</span>
<span class="co">##             nonCoa</span>
<span class="co">##          &lt;logical&gt;</span>
<span class="co">##      [1]     FALSE</span>
<span class="co">##      [2]      TRUE</span>
<span class="co">##      [3]     FALSE</span>
<span class="co">##      [4]     FALSE</span>
<span class="co">##      [5]      TRUE</span>
<span class="co">##      ...       ...</span>
<span class="co">##   [4069]      TRUE</span>
<span class="co">##   [4070]      TRUE</span>
<span class="co">##   [4071]      TRUE</span>
<span class="co">##   [4072]      TRUE</span>
<span class="co">##   [4073]      TRUE</span>
<span class="co">##   -------</span>
<span class="co">##   seqinfo: 68 sequences from Bar2.p4 genome</span></code></pre>
<div class="sourceCode" id="cb78"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/start.html" class="external-link">width</a></span><span class="op">(</span><span class="va">gb_N</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## [1] 45289525</span></code></pre>
<div class="sourceCode" id="cb80"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="op">(</span><span class="va">ov</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/findOverlaps-methods.html" class="external-link">subsetByOverlaps</a></span><span class="op">(</span><span class="va">gb_P</span>, <span class="va">gb_N</span>, type <span class="op">=</span> <span class="st">"within"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## GBreaks object with 918 ranges and 8 metadata columns:</span>
<span class="co">##         seqnames          ranges strand |                query     score</span>
<span class="co">##            &lt;Rle&gt;       &lt;IRanges&gt;  &lt;Rle&gt; |            &lt;GRanges&gt; &lt;integer&gt;</span>
<span class="co">##     [1]     Chr1     17796-21383      * | Chr1:3581251-3591380      3588</span>
<span class="co">##     [2]     Chr1     69763-74781      * | Chr1:3644537-3650503      5019</span>
<span class="co">##     [3]     Chr1     87558-91675      * | Chr1:3818944-3824036      4118</span>
<span class="co">##     [4]     Chr1    92754-103548      * | Chr1:3833970-3843493     10795</span>
<span class="co">##     [5]     Chr1   103823-135737      * | Chr1:2195467-2232252     31915</span>
<span class="co">##     ...      ...             ...    ... .                  ...       ...</span>
<span class="co">##   [914]      YSR 3597539-3598992      * |   Chr1:269069-271045      1454</span>
<span class="co">##   [915]      YSR 3599919-3601165      * | Chr1:4173793-4175304      1247</span>
<span class="co">##   [916]      YSR 3631705-3633609      * | Chr1:4315390-4317451      1905</span>
<span class="co">##   [917]      YSR 4086487-4086957      * | Chr1:8499823-8501898       471</span>
<span class="co">##   [918]      YSR 4865878-4866414      * |    YSR:434383-434709       537</span>
<span class="co">##                        geneNames              dNdS_GUIDANCE2</span>
<span class="co">##                  &lt;CharacterList&gt;               &lt;NumericList&gt;</span>
<span class="co">##     [1]              g1.t1,g2.t1             0.04071,0.04756</span>
<span class="co">##     [2] g16.t1,g17.t1,g18.t1,... 0.00746,0.03727,0.06337,...</span>
<span class="co">##     [3]            g22.t1,g23.t1                       NA,NA</span>
<span class="co">##     [4]            g24.t1,g25.t1                  NA,0.04723</span>
<span class="co">##     [5] g26.t1,g27.t1,g28.t1,... 0.02336,     NA,0.03624,...</span>
<span class="co">##     ...                      ...                         ...</span>
<span class="co">##   [914]                g13905.t1                     0.03202</span>
<span class="co">##   [915]                g13906.t1                          NA</span>
<span class="co">##   [916]                g13910.t1                     0.02976</span>
<span class="co">##   [917]                g13972.t1                          NA</span>
<span class="co">##   [918]                g14068.t1                          NA</span>
<span class="co">##                     dNdS_HmmCleaner                  dNdS_PRANK geneNumber</span>
<span class="co">##                       &lt;NumericList&gt;               &lt;NumericList&gt;  &lt;integer&gt;</span>
<span class="co">##     [1]             0.04071,0.05590             0.04071,0.05407          2</span>
<span class="co">##     [2] 0.00746,0.04309,0.06258,... 0.00746,0.04309,0.06258,...          4</span>
<span class="co">##     [3]                       NA,NA                       NA,NA          2</span>
<span class="co">##     [4]                  NA,0.04214                  NA,0.04431          2</span>
<span class="co">##     [5] 0.02316,     NA,0.03757,... 0.02511,     NA,0.03837,...          5</span>
<span class="co">##     ...                         ...                         ...        ...</span>
<span class="co">##   [914]                     0.03535                     0.03535          1</span>
<span class="co">##   [915]                          NA                          NA          1</span>
<span class="co">##   [916]                     0.02967                     0.03291          1</span>
<span class="co">##   [917]                          NA                          NA          1</span>
<span class="co">##   [918]                          NA                          NA          1</span>
<span class="co">##              Arm</span>
<span class="co">##         &lt;factor&gt;</span>
<span class="co">##     [1]    short</span>
<span class="co">##     [2]    short</span>
<span class="co">##     [3]    short</span>
<span class="co">##     [4]    short</span>
<span class="co">##     [5]    short</span>
<span class="co">##     ...      ...</span>
<span class="co">##   [914]      YSR</span>
<span class="co">##   [915]      YSR</span>
<span class="co">##   [916]      YSR</span>
<span class="co">##   [917]      YSR</span>
<span class="co">##   [918]      YSR</span>
<span class="co">##   -------</span>
<span class="co">##   seqinfo: 68 sequences from Bar2.p4 genome</span></code></pre>
<div class="sourceCode" id="cb82"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="op">(</span><span class="va">jaccard</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/start.html" class="external-link">width</a></span><span class="op">(</span><span class="va">ov</span><span class="op">)</span><span class="op">)</span> <span class="op">/</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/start.html" class="external-link">width</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/inter-range-methods.html" class="external-link">reduce</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="va">gb_P</span>, <span class="va">gb_N</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## [1] 0.1434809</span></code></pre>
<div class="sourceCode" id="cb84"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">gb_P</span> <span class="op">&lt;-</span> <span class="va">gb_P</span><span class="op">[</span><span class="va">gb_P</span><span class="op">$</span><span class="va">geneNumber</span> <span class="op">&gt;</span> <span class="fl">1</span><span class="op">]</span>
<span class="va">gb_P</span></code></pre></div>
<pre><code><span class="co">## GBreaks object with 867 ranges and 8 metadata columns:</span>
<span class="co">##         seqnames            ranges strand |                  query     score</span>
<span class="co">##            &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; |              &lt;GRanges&gt; &lt;integer&gt;</span>
<span class="co">##     [1]     Chr1       17796-21383      * |   Chr1:3581251-3591380      3588</span>
<span class="co">##     [2]     Chr1       27418-60748      * |   Chr1:3596010-3622624     33331</span>
<span class="co">##     [3]     Chr1       69763-74781      * |   Chr1:3644537-3650503      5019</span>
<span class="co">##     [4]     Chr1       87558-91675      * |   Chr1:3818944-3824036      4118</span>
<span class="co">##     [5]     Chr1      92754-103548      * |   Chr1:3833970-3843493     10795</span>
<span class="co">##     ...      ...               ...    ... .                    ...       ...</span>
<span class="co">##   [863]      XSR 12286992-12366325      * |    XSR:9201073-9267597     79334</span>
<span class="co">##   [864]      YSR     268825-302025      * |   Chr1:1911761-1929212     33201</span>
<span class="co">##   [865]      YSR     518298-532357      * | Chr2:12195648-12210212     14060</span>
<span class="co">##   [866]      YSR   1387847-1390918      * |   Chr1:1811359-1814424      3072</span>
<span class="co">##   [867]      YSR   2712461-2719315      * |    YSR:1128232-1134212      6855</span>
<span class="co">##                                 geneNames              dNdS_GUIDANCE2</span>
<span class="co">##                           &lt;CharacterList&gt;               &lt;NumericList&gt;</span>
<span class="co">##     [1]                       g1.t1,g2.t1             0.04071,0.04756</span>
<span class="co">##     [2]             g4.t1,g5.t1,g6.t1,...                NA,NA,NA,...</span>
<span class="co">##     [3]          g16.t1,g17.t1,g18.t1,... 0.00746,0.03727,0.06337,...</span>
<span class="co">##     [4]                     g22.t1,g23.t1                       NA,NA</span>
<span class="co">##     [5]                     g24.t1,g25.t1                  NA,0.04723</span>
<span class="co">##     ...                               ...                         ...</span>
<span class="co">##   [863] g13291.t1,g13292.t2,g13293.t2,...                NA,NA,NA,...</span>
<span class="co">##   [864] g13345.t1,g13347.t1,g13348.t1,...                NA,NA,NA,...</span>
<span class="co">##   [865]               g13390.t1,g13392.t1                       NA,NA</span>
<span class="co">##   [866]               g13542.t1,g13543.t1                       NA,NA</span>
<span class="co">##   [867]               g13745.t1,g13747.t1                       NA,NA</span>
<span class="co">##                     dNdS_HmmCleaner                  dNdS_PRANK geneNumber</span>
<span class="co">##                       &lt;NumericList&gt;               &lt;NumericList&gt;  &lt;integer&gt;</span>
<span class="co">##     [1]             0.04071,0.05590             0.04071,0.05407          2</span>
<span class="co">##     [2]                NA,NA,NA,...                NA,NA,NA,...          8</span>
<span class="co">##     [3] 0.00746,0.04309,0.06258,... 0.00746,0.04309,0.06258,...          4</span>
<span class="co">##     [4]                       NA,NA                       NA,NA          2</span>
<span class="co">##     [5]                  NA,0.04214                  NA,0.04431          2</span>
<span class="co">##     ...                         ...                         ...        ...</span>
<span class="co">##   [863]                NA,NA,NA,...                NA,NA,NA,...         20</span>
<span class="co">##   [864]                NA,NA,NA,...                NA,NA,NA,...          4</span>
<span class="co">##   [865]                       NA,NA                       NA,NA          2</span>
<span class="co">##   [866]                       NA,NA                       NA,NA          2</span>
<span class="co">##   [867]                       NA,NA                       NA,NA          2</span>
<span class="co">##              Arm</span>
<span class="co">##         &lt;factor&gt;</span>
<span class="co">##     [1]    short</span>
<span class="co">##     [2]    short</span>
<span class="co">##     [3]    short</span>
<span class="co">##     [4]    short</span>
<span class="co">##     [5]    short</span>
<span class="co">##     ...      ...</span>
<span class="co">##   [863]      XSR</span>
<span class="co">##   [864]      YSR</span>
<span class="co">##   [865]      YSR</span>
<span class="co">##   [866]      YSR</span>
<span class="co">##   [867]      YSR</span>
<span class="co">##   -------</span>
<span class="co">##   seqinfo: 68 sequences from Bar2.p4 genome</span></code></pre>
<div class="sourceCode" id="cb86"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="va">gb_P</span><span class="op">$</span><span class="va">geneNumber</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## [1] 8498</span></code></pre>
<div class="sourceCode" id="cb88"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="op">(</span><span class="va">valid</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/findOverlaps-methods.html" class="external-link">subsetByOverlaps</a></span><span class="op">(</span><span class="va">gb_P</span>, <span class="va">gb_N</span>, type <span class="op">=</span> <span class="st">"within"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## GBreaks object with 462 ranges and 8 metadata columns:</span>
<span class="co">##         seqnames            ranges strand |                query     score</span>
<span class="co">##            &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; |            &lt;GRanges&gt; &lt;integer&gt;</span>
<span class="co">##     [1]     Chr1       17796-21383      * | Chr1:3581251-3591380      3588</span>
<span class="co">##     [2]     Chr1       69763-74781      * | Chr1:3644537-3650503      5019</span>
<span class="co">##     [3]     Chr1       87558-91675      * | Chr1:3818944-3824036      4118</span>
<span class="co">##     [4]     Chr1      92754-103548      * | Chr1:3833970-3843493     10795</span>
<span class="co">##     [5]     Chr1     103823-135737      * | Chr1:2195467-2232252     31915</span>
<span class="co">##     ...      ...               ...    ... .                  ...       ...</span>
<span class="co">##   [458]      XSR 12170086-12180178      * |  XSR:9269610-9279934     10093</span>
<span class="co">##   [459]      XSR 12180804-12192236      * |    XSR:811181-825617     11433</span>
<span class="co">##   [460]      XSR 12279536-12286707      * |  XSR:1674570-1684997      7172</span>
<span class="co">##   [461]      XSR 12286992-12366325      * |  XSR:9201073-9267597     79334</span>
<span class="co">##   [462]      YSR   1387847-1390918      * | Chr1:1811359-1814424      3072</span>
<span class="co">##                                 geneNames              dNdS_GUIDANCE2</span>
<span class="co">##                           &lt;CharacterList&gt;               &lt;NumericList&gt;</span>
<span class="co">##     [1]                       g1.t1,g2.t1             0.04071,0.04756</span>
<span class="co">##     [2]          g16.t1,g17.t1,g18.t1,... 0.00746,0.03727,0.06337,...</span>
<span class="co">##     [3]                     g22.t1,g23.t1                       NA,NA</span>
<span class="co">##     [4]                     g24.t1,g25.t1                  NA,0.04723</span>
<span class="co">##     [5]          g26.t1,g27.t1,g28.t1,... 0.02336,     NA,0.03624,...</span>
<span class="co">##     ...                               ...                         ...</span>
<span class="co">##   [458] g13261.t1,g13262.t1,g13263.t1,... 0.01210,0.04173,     NA,...</span>
<span class="co">##   [459]               g13265.t1,g13267.t1             0.00881,0.05570</span>
<span class="co">##   [460]               g13289.t1,g13290.t1                  NA,0.07028</span>
<span class="co">##   [461] g13291.t1,g13292.t2,g13293.t2,...                NA,NA,NA,...</span>
<span class="co">##   [462]               g13542.t1,g13543.t1                       NA,NA</span>
<span class="co">##                     dNdS_HmmCleaner                  dNdS_PRANK geneNumber</span>
<span class="co">##                       &lt;NumericList&gt;               &lt;NumericList&gt;  &lt;integer&gt;</span>
<span class="co">##     [1]             0.04071,0.05590             0.04071,0.05407          2</span>
<span class="co">##     [2] 0.00746,0.04309,0.06258,... 0.00746,0.04309,0.06258,...          4</span>
<span class="co">##     [3]                       NA,NA                       NA,NA          2</span>
<span class="co">##     [4]                  NA,0.04214                  NA,0.04431          2</span>
<span class="co">##     [5] 0.02316,     NA,0.03757,... 0.02511,     NA,0.03837,...          5</span>
<span class="co">##     ...                         ...                         ...        ...</span>
<span class="co">##   [458] 0.01319,0.04173,     NA,... 0.01319,0.04173,     NA,...          4</span>
<span class="co">##   [459]             0.00885,0.05431             0.00878,0.05206          2</span>
<span class="co">##   [460]                  NA,0.06831                  NA,0.06831          2</span>
<span class="co">##   [461]                NA,NA,NA,...                NA,NA,NA,...         20</span>
<span class="co">##   [462]                       NA,NA                       NA,NA          2</span>
<span class="co">##              Arm</span>
<span class="co">##         &lt;factor&gt;</span>
<span class="co">##     [1]    short</span>
<span class="co">##     [2]    short</span>
<span class="co">##     [3]    short</span>
<span class="co">##     [4]    short</span>
<span class="co">##     [5]    short</span>
<span class="co">##     ...      ...</span>
<span class="co">##   [458]      XSR</span>
<span class="co">##   [459]      XSR</span>
<span class="co">##   [460]      XSR</span>
<span class="co">##   [461]      XSR</span>
<span class="co">##   [462]      YSR</span>
<span class="co">##   -------</span>
<span class="co">##   seqinfo: 68 sequences from Bar2.p4 genome</span></code></pre>
<div class="sourceCode" id="cb90"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="va">valid</span><span class="op">$</span><span class="va">geneNumber</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## [1] 2825</span></code></pre>
<div class="sourceCode" id="cb92"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/start.html" class="external-link">width</a></span><span class="op">(</span><span class="va">valid</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## [1] 11168468</span></code></pre>
<p>Question: what does interrupt the colinearity of nucleotide regions in gene-centric colinear regions ?</p>
</div>
<div class="section level2">
<h2 id="xsr-plot">XSR plot<a class="anchor" aria-label="anchor" href="#xsr-plot"></a>
</h2>
<p>Ugly…</p>
<div class="sourceCode" id="cb94"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">XSR_HOGs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">orthoPairs_one2oneInOiks</span>, \<span class="op">(</span><span class="va">gb</span><span class="op">)</span> <span class="op">{</span>
  <span class="va">gb</span> <span class="op">&lt;-</span> <span class="va">gb</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqnames</a></span><span class="op">(</span><span class="va">gb</span><span class="op">)</span> <span class="op">==</span> <span class="st">"XSR"</span><span class="op">]</span>
  <span class="co"># Keep biggest match only</span>
  <span class="va">chrQ</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/tapply.html" class="external-link">tapply</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/start.html" class="external-link">width</a></span><span class="op">(</span><span class="va">gb</span><span class="op">$</span><span class="va">query</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqnames</a></span><span class="op">(</span><span class="va">gb</span><span class="op">$</span><span class="va">query</span><span class="op">)</span>, <span class="va">sum</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/sort.html" class="external-link">sort</a></span><span class="op">(</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">tail</a></span><span class="op">(</span><span class="fl">1</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="op">)</span>
  <span class="va">gb</span> <span class="op">&lt;-</span> <span class="va">gb</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqnames</a></span><span class="op">(</span><span class="va">gb</span><span class="op">$</span><span class="va">query</span><span class="op">)</span> <span class="op">==</span> <span class="va">chrQ</span><span class="op">]</span>
  <span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqlevels</a></span><span class="op">(</span><span class="va">gb</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqlevelsInUse</a></span><span class="op">(</span><span class="va">gb</span><span class="op">)</span>
  <span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqlevels</a></span><span class="op">(</span><span class="va">gb</span><span class="op">$</span><span class="va">query</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqlevelsInUse</a></span><span class="op">(</span><span class="va">gb</span><span class="op">$</span><span class="va">query</span><span class="op">)</span>
  <span class="va">gb</span>
<span class="op">}</span><span class="op">)</span>

<span class="va">removeSeqInfoGR</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">gr</span><span class="op">)</span> 
  <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a></span><span class="op">(</span>seqnames <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqnames</a></span><span class="op">(</span><span class="va">gr</span><span class="op">)</span>, ranges <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/range-squeezers.html" class="external-link">ranges</a></span><span class="op">(</span><span class="va">gr</span><span class="op">)</span>, strand <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/strand.html" class="external-link">strand</a></span><span class="op">(</span><span class="va">gr</span><span class="op">)</span><span class="op">)</span>

<span class="va">removeSeqInfoGB</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">gb</span><span class="op">)</span>
  <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/GBreaks-class.html" class="external-link">GBreaks</a></span><span class="op">(</span>target <span class="op">=</span> <span class="fu">removeSeqInfoGR</span><span class="op">(</span><span class="va">gb</span><span class="op">)</span>, query <span class="op">=</span> <span class="fu">removeSeqInfoGR</span><span class="op">(</span><span class="va">gb</span><span class="op">$</span><span class="va">query</span><span class="op">)</span><span class="op">)</span>

<span class="va">XSR_toPlot</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">XSR_HOGs</span>, <span class="va">removeSeqInfoGB</span><span class="op">)</span>
<span class="va">XSR_toPlot</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/as.html" class="external-link">as</a></span><span class="op">(</span><span class="va">XSR_toPlot</span>, <span class="st">"GRangesList"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqlevels</a></span><span class="op">(</span><span class="va">XSR_toPlot</span><span class="op">$</span><span class="va">Oki_Osa</span><span class="op">)</span>       <span class="op">&lt;-</span> <span class="st">"XSR_01_Oki"</span>
<span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqlevels</a></span><span class="op">(</span><span class="va">XSR_toPlot</span><span class="op">$</span><span class="va">Oki_Osa</span><span class="op">$</span><span class="va">query</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="st">"XSR_03_Osa"</span>
<span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqlevels</a></span><span class="op">(</span><span class="va">XSR_toPlot</span><span class="op">$</span><span class="va">Oki_Bar</span><span class="op">)</span>       <span class="op">&lt;-</span> <span class="st">"XSR_01_Oki"</span>
<span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqlevels</a></span><span class="op">(</span><span class="va">XSR_toPlot</span><span class="op">$</span><span class="va">Oki_Bar</span><span class="op">$</span><span class="va">query</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="st">"XSR_06_Bar"</span>
<span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqlevels</a></span><span class="op">(</span><span class="va">XSR_toPlot</span><span class="op">$</span><span class="va">Oki_Kum</span><span class="op">)</span>       <span class="op">&lt;-</span> <span class="st">"XSR_01_Oki"</span>
<span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqlevels</a></span><span class="op">(</span><span class="va">XSR_toPlot</span><span class="op">$</span><span class="va">Oki_Kum</span><span class="op">$</span><span class="va">query</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="st">"XSR_02_Kum"</span>
<span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqlevels</a></span><span class="op">(</span><span class="va">XSR_toPlot</span><span class="op">$</span><span class="va">Oki_Aom</span><span class="op">)</span>       <span class="op">&lt;-</span> <span class="st">"XSR_01_Oki"</span>
<span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqlevels</a></span><span class="op">(</span><span class="va">XSR_toPlot</span><span class="op">$</span><span class="va">Oki_Aom</span><span class="op">$</span><span class="va">query</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="st">"XSR_04_Aom"</span>
<span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqlevels</a></span><span class="op">(</span><span class="va">XSR_toPlot</span><span class="op">$</span><span class="va">Oki_Nor</span><span class="op">)</span>       <span class="op">&lt;-</span> <span class="st">"XSR_01_Oki"</span>
<span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqlevels</a></span><span class="op">(</span><span class="va">XSR_toPlot</span><span class="op">$</span><span class="va">Oki_Nor</span><span class="op">$</span><span class="va">query</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="st">"XSR_06_Nor"</span>

<span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqlevels</a></span><span class="op">(</span><span class="va">XSR_toPlot</span><span class="op">$</span><span class="va">Osa_Oki</span><span class="op">)</span>       <span class="op">&lt;-</span> <span class="st">"XSR_03_Osa"</span>
<span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqlevels</a></span><span class="op">(</span><span class="va">XSR_toPlot</span><span class="op">$</span><span class="va">Osa_Oki</span><span class="op">$</span><span class="va">query</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="st">"XSR_01_Oki"</span>
<span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqlevels</a></span><span class="op">(</span><span class="va">XSR_toPlot</span><span class="op">$</span><span class="va">Osa_Bar</span><span class="op">)</span>       <span class="op">&lt;-</span> <span class="st">"XSR_03_Osa"</span>
<span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqlevels</a></span><span class="op">(</span><span class="va">XSR_toPlot</span><span class="op">$</span><span class="va">Osa_Bar</span><span class="op">$</span><span class="va">query</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="st">"XSR_06_Bar"</span>
<span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqlevels</a></span><span class="op">(</span><span class="va">XSR_toPlot</span><span class="op">$</span><span class="va">Osa_Kum</span><span class="op">)</span>       <span class="op">&lt;-</span> <span class="st">"XSR_03_Osa"</span>
<span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqlevels</a></span><span class="op">(</span><span class="va">XSR_toPlot</span><span class="op">$</span><span class="va">Osa_Kum</span><span class="op">$</span><span class="va">query</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="st">"XSR_02_Kum"</span>
<span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqlevels</a></span><span class="op">(</span><span class="va">XSR_toPlot</span><span class="op">$</span><span class="va">Osa_Aom</span><span class="op">)</span>       <span class="op">&lt;-</span> <span class="st">"XSR_03_Osa"</span>
<span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqlevels</a></span><span class="op">(</span><span class="va">XSR_toPlot</span><span class="op">$</span><span class="va">Osa_Aom</span><span class="op">$</span><span class="va">query</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="st">"XSR_04_Aom"</span>
<span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqlevels</a></span><span class="op">(</span><span class="va">XSR_toPlot</span><span class="op">$</span><span class="va">Osa_Nor</span><span class="op">)</span>       <span class="op">&lt;-</span> <span class="st">"XSR_03_Osa"</span>
<span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqlevels</a></span><span class="op">(</span><span class="va">XSR_toPlot</span><span class="op">$</span><span class="va">Osa_Nor</span><span class="op">$</span><span class="va">query</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="st">"XSR_06_Nor"</span>

<span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqlevels</a></span><span class="op">(</span><span class="va">XSR_toPlot</span><span class="op">$</span><span class="va">Bar_Osa</span><span class="op">)</span>       <span class="op">&lt;-</span> <span class="st">"XSR_06_Bar"</span>
<span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqlevels</a></span><span class="op">(</span><span class="va">XSR_toPlot</span><span class="op">$</span><span class="va">Bar_Osa</span><span class="op">$</span><span class="va">query</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="st">"XSR_03_Osa"</span>
<span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqlevels</a></span><span class="op">(</span><span class="va">XSR_toPlot</span><span class="op">$</span><span class="va">Bar_Oki</span><span class="op">)</span>       <span class="op">&lt;-</span> <span class="st">"XSR_06_Bar"</span>
<span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqlevels</a></span><span class="op">(</span><span class="va">XSR_toPlot</span><span class="op">$</span><span class="va">Bar_Oki</span><span class="op">$</span><span class="va">query</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="st">"XSR_01_Oki"</span>
<span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqlevels</a></span><span class="op">(</span><span class="va">XSR_toPlot</span><span class="op">$</span><span class="va">Bar_Kum</span><span class="op">)</span>       <span class="op">&lt;-</span> <span class="st">"XSR_06_Bar"</span>
<span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqlevels</a></span><span class="op">(</span><span class="va">XSR_toPlot</span><span class="op">$</span><span class="va">Bar_Kum</span><span class="op">$</span><span class="va">query</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="st">"XSR_02_Kum"</span>
<span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqlevels</a></span><span class="op">(</span><span class="va">XSR_toPlot</span><span class="op">$</span><span class="va">Bar_Aom</span><span class="op">)</span>       <span class="op">&lt;-</span> <span class="st">"XSR_06_Bar"</span>
<span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqlevels</a></span><span class="op">(</span><span class="va">XSR_toPlot</span><span class="op">$</span><span class="va">Bar_Aom</span><span class="op">$</span><span class="va">query</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="st">"XSR_04_Aom"</span>
<span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqlevels</a></span><span class="op">(</span><span class="va">XSR_toPlot</span><span class="op">$</span><span class="va">Bar_Nor</span><span class="op">)</span>       <span class="op">&lt;-</span> <span class="st">"XSR_06_Bar"</span>
<span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqlevels</a></span><span class="op">(</span><span class="va">XSR_toPlot</span><span class="op">$</span><span class="va">Bar_Nor</span><span class="op">$</span><span class="va">query</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="st">"XSR_06_Nor"</span>

<span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/makeOxfordPlots.html" class="external-link">makeOxfordPlots</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/unlist.html" class="external-link">unlist</a></span><span class="op">(</span><span class="va">XSR_toPlot</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Warning in .Seqinfo.mergexy(x, y): The 2 combined objects have no sequence levels in common. (Use</span>
<span class="co">##   suppressWarnings() to suppress this warning.)</span>

<span class="co">## Warning in .Seqinfo.mergexy(x, y): The 2 combined objects have no sequence levels in common. (Use</span>
<span class="co">##   suppressWarnings() to suppress this warning.)</span>

<span class="co">## Warning in .Seqinfo.mergexy(x, y): The 2 combined objects have no sequence levels in common. (Use</span>
<span class="co">##   suppressWarnings() to suppress this warning.)</span>

<span class="co">## Warning in .Seqinfo.mergexy(x, y): The 2 combined objects have no sequence levels in common. (Use</span>
<span class="co">##   suppressWarnings() to suppress this warning.)</span>

<span class="co">## Warning in .Seqinfo.mergexy(x, y): The 2 combined objects have no sequence levels in common. (Use</span>
<span class="co">##   suppressWarnings() to suppress this warning.)</span>

<span class="co">## Warning in .Seqinfo.mergexy(x, y): The 2 combined objects have no sequence levels in common. (Use</span>
<span class="co">##   suppressWarnings() to suppress this warning.)</span>

<span class="co">## Warning in .Seqinfo.mergexy(x, y): The 2 combined objects have no sequence levels in common. (Use</span>
<span class="co">##   suppressWarnings() to suppress this warning.)</span></code></pre>
<p><img src="OrthogroupSyntenies_files/figure-html/XSR_plot-1.png" width="700"></p>
</div>
<div class="section level2">
<h2 id="hox-genes">HOX genes<a class="anchor" aria-label="anchor" href="#hox-genes"></a>
</h2>
<div class="section level3">
<h3 id="hox-4">Hox 4<a class="anchor" aria-label="anchor" href="#hox-4"></a>
</h3>
<p>Appears to be duplicated in Bar so missing from HOG <code>N19.HOG0000192</code> in that genome.</p>
<div class="sourceCode" id="cb96"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">orthoPairs</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">15</span><span class="op">]</span>, \<span class="op">(</span><span class="va">gb</span><span class="op">)</span> <span class="op">{</span>
  <span class="va">x</span> <span class="op">&lt;-</span> <span class="va">gb</span> |&gt; <span class="fu">plyranges</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/plyranges/man/tidyverse-reexports.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">HOG</span> <span class="op">==</span> <span class="st">"N19.HOG0000192"</span><span class="op">)</span>
  <span class="kw">if</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="op">==</span> <span class="fl">0</span><span class="op">)</span> <span class="kw"><a href="https://rdrr.io/r/base/function.html" class="external-link">return</a></span><span class="op">(</span><span class="cn">NULL</span><span class="op">)</span>
  <span class="va">x</span><span class="op">}</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## $Oki_Osa</span>
<span class="co">## GBreaks object with 1 range and 9 metadata columns:</span>
<span class="co">##            seqnames          ranges strand |     tx_id     tx_name</span>
<span class="co">##               &lt;Rle&gt;       &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt; &lt;character&gt;</span>
<span class="co">##   g5193.t1     chr2 4686018-4687168      * |      7545    g5193.t1</span>
<span class="co">##            dNdS_GUIDANCE2 dNdS_HmmCleaner dNdS_PRANK                query</span>
<span class="co">##                 &lt;numeric&gt;       &lt;numeric&gt;  &lt;numeric&gt;            &lt;GRanges&gt;</span>
<span class="co">##   g5193.t1             NA              NA         NA Chr2:1156287-1157412</span>
<span class="co">##                       HOG          OG      Arm</span>
<span class="co">##               &lt;character&gt; &lt;character&gt; &lt;factor&gt;</span>
<span class="co">##   g5193.t1 N19.HOG0000192   OG0000004    short</span>
<span class="co">##   -------</span>
<span class="co">##   seqinfo: 19 sequences from OKI2018.I69 genome</span>
<span class="co">## </span>
<span class="co">## $Oki_Bar</span>
<span class="co">## NULL</span>
<span class="co">## </span>
<span class="co">## $Oki_Kum</span>
<span class="co">## GBreaks object with 1 range and 9 metadata columns:</span>
<span class="co">##            seqnames          ranges strand |     tx_id     tx_name</span>
<span class="co">##               &lt;Rle&gt;       &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt; &lt;character&gt;</span>
<span class="co">##   g5193.t1     chr2 4686018-4687168      * |      7545    g5193.t1</span>
<span class="co">##            dNdS_GUIDANCE2 dNdS_HmmCleaner dNdS_PRANK</span>
<span class="co">##                 &lt;numeric&gt;       &lt;numeric&gt;  &lt;numeric&gt;</span>
<span class="co">##   g5193.t1             NA              NA         NA</span>
<span class="co">##                                  query            HOG          OG      Arm</span>
<span class="co">##                              &lt;GRanges&gt;    &lt;character&gt; &lt;character&gt; &lt;factor&gt;</span>
<span class="co">##   g5193.t1 contig_16_1:4184659-4186299 N19.HOG0000192   OG0000004    short</span>
<span class="co">##   -------</span>
<span class="co">##   seqinfo: 19 sequences from OKI2018.I69 genome</span>
<span class="co">## </span>
<span class="co">## $Oki_Aom</span>
<span class="co">## NULL</span>
<span class="co">## </span>
<span class="co">## $Oki_Nor</span>
<span class="co">## GBreaks object with 1 range and 9 metadata columns:</span>
<span class="co">##            seqnames          ranges strand |     tx_id     tx_name</span>
<span class="co">##               &lt;Rle&gt;       &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt; &lt;character&gt;</span>
<span class="co">##   g5193.t1     chr2 4686018-4687168      * |      7545    g5193.t1</span>
<span class="co">##            dNdS_GUIDANCE2 dNdS_HmmCleaner dNdS_PRANK                     query</span>
<span class="co">##                 &lt;numeric&gt;       &lt;numeric&gt;  &lt;numeric&gt;                 &lt;GRanges&gt;</span>
<span class="co">##   g5193.t1             NA              NA         NA scaffold_47:176335-180537</span>
<span class="co">##                       HOG          OG      Arm</span>
<span class="co">##               &lt;character&gt; &lt;character&gt; &lt;factor&gt;</span>
<span class="co">##   g5193.t1 N19.HOG0000192   OG0000004    short</span>
<span class="co">##   -------</span>
<span class="co">##   seqinfo: 19 sequences from OKI2018.I69 genome</span>
<span class="co">## </span>
<span class="co">## $Osa_Bar</span>
<span class="co">## NULL</span>
<span class="co">## </span>
<span class="co">## $Osa_Oki</span>
<span class="co">## GBreaks object with 1 range and 9 metadata columns:</span>
<span class="co">##            seqnames          ranges strand |     tx_id     tx_name</span>
<span class="co">##               &lt;Rle&gt;       &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt; &lt;character&gt;</span>
<span class="co">##   g3346.t1     Chr2 1156287-1157412      * |      5799    g3346.t1</span>
<span class="co">##            dNdS_GUIDANCE2 dNdS_HmmCleaner dNdS_PRANK                query</span>
<span class="co">##                 &lt;numeric&gt;       &lt;numeric&gt;  &lt;numeric&gt;            &lt;GRanges&gt;</span>
<span class="co">##   g3346.t1             NA              NA         NA chr2:4686018-4687168</span>
<span class="co">##                       HOG          OG      Arm</span>
<span class="co">##               &lt;character&gt; &lt;character&gt; &lt;factor&gt;</span>
<span class="co">##   g3346.t1 N19.HOG0000192   OG0000004    short</span>
<span class="co">##   -------</span>
<span class="co">##   seqinfo: 483 sequences from OSKA2016v1.9 genome</span>
<span class="co">## </span>
<span class="co">## $Osa_Kum</span>
<span class="co">## GBreaks object with 1 range and 9 metadata columns:</span>
<span class="co">##            seqnames          ranges strand |     tx_id     tx_name</span>
<span class="co">##               &lt;Rle&gt;       &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt; &lt;character&gt;</span>
<span class="co">##   g3346.t1     Chr2 1156287-1157412      * |      5799    g3346.t1</span>
<span class="co">##            dNdS_GUIDANCE2 dNdS_HmmCleaner dNdS_PRANK</span>
<span class="co">##                 &lt;numeric&gt;       &lt;numeric&gt;  &lt;numeric&gt;</span>
<span class="co">##   g3346.t1             NA              NA         NA</span>
<span class="co">##                                  query            HOG          OG      Arm</span>
<span class="co">##                              &lt;GRanges&gt;    &lt;character&gt; &lt;character&gt; &lt;factor&gt;</span>
<span class="co">##   g3346.t1 contig_16_1:4184659-4186299 N19.HOG0000192   OG0000004    short</span>
<span class="co">##   -------</span>
<span class="co">##   seqinfo: 483 sequences from OSKA2016v1.9 genome</span>
<span class="co">## </span>
<span class="co">## $Osa_Aom</span>
<span class="co">## NULL</span>
<span class="co">## </span>
<span class="co">## $Osa_Nor</span>
<span class="co">## GBreaks object with 1 range and 9 metadata columns:</span>
<span class="co">##            seqnames          ranges strand |     tx_id     tx_name</span>
<span class="co">##               &lt;Rle&gt;       &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt; &lt;character&gt;</span>
<span class="co">##   g3346.t1     Chr2 1156287-1157412      * |      5799    g3346.t1</span>
<span class="co">##            dNdS_GUIDANCE2 dNdS_HmmCleaner dNdS_PRANK                     query</span>
<span class="co">##                 &lt;numeric&gt;       &lt;numeric&gt;  &lt;numeric&gt;                 &lt;GRanges&gt;</span>
<span class="co">##   g3346.t1             NA              NA         NA scaffold_47:176335-180537</span>
<span class="co">##                       HOG          OG      Arm</span>
<span class="co">##               &lt;character&gt; &lt;character&gt; &lt;factor&gt;</span>
<span class="co">##   g3346.t1 N19.HOG0000192   OG0000004    short</span>
<span class="co">##   -------</span>
<span class="co">##   seqinfo: 483 sequences from OSKA2016v1.9 genome</span>
<span class="co">## </span>
<span class="co">## $Bar_Osa</span>
<span class="co">## NULL</span>
<span class="co">## </span>
<span class="co">## $Bar_Oki</span>
<span class="co">## NULL</span>
<span class="co">## </span>
<span class="co">## $Bar_Kum</span>
<span class="co">## NULL</span>
<span class="co">## </span>
<span class="co">## $Bar_Aom</span>
<span class="co">## NULL</span>
<span class="co">## </span>
<span class="co">## $Bar_Nor</span>
<span class="co">## NULL</span></code></pre>
<div class="sourceCode" id="cb98"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">Hox4</span> <span class="op">&lt;-</span> <span class="va">orthoPairs</span><span class="op">$</span><span class="va">Oki_Osa</span> |&gt; <span class="fu">plyranges</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/plyranges/man/tidyverse-reexports.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">HOG</span> <span class="op">==</span> <span class="st">"N19.HOG0000192"</span><span class="op">)</span>
<span class="va">Hox4_1e4</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/GBreaks-class.html" class="external-link">GBreaks</a></span><span class="op">(</span>target <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/genomic-range-squeezers.html" class="external-link">granges</a></span><span class="op">(</span><span class="va">Hox4</span><span class="op">)</span> <span class="op">+</span> <span class="fl">1e4</span>, query <span class="op">=</span> <span class="va">Hox4</span><span class="op">$</span><span class="va">query</span> <span class="op">+</span> <span class="fl">1e4</span><span class="op">)</span>
<span class="va">Hox4_4e4</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/GBreaks-class.html" class="external-link">GBreaks</a></span><span class="op">(</span>target <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/genomic-range-squeezers.html" class="external-link">granges</a></span><span class="op">(</span><span class="va">Hox4</span><span class="op">)</span> <span class="op">+</span> <span class="fl">4e4</span>, query <span class="op">=</span> <span class="va">Hox4</span><span class="op">$</span><span class="va">query</span> <span class="op">+</span> <span class="fl">4e4</span><span class="op">)</span>
<span class="va">Hox4_1e5</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/GBreaks-class.html" class="external-link">GBreaks</a></span><span class="op">(</span>target <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/genomic-range-squeezers.html" class="external-link">granges</a></span><span class="op">(</span><span class="va">Hox4</span><span class="op">)</span> <span class="op">+</span> <span class="fl">1e5</span>, query <span class="op">=</span> <span class="va">Hox4</span><span class="op">$</span><span class="va">query</span> <span class="op">+</span> <span class="fl">1e5</span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/findOverlaps-methods.html" class="external-link">subsetByOverlaps</a></span><span class="op">(</span><span class="va">coa</span><span class="op">$</span><span class="va">Oki_Osa</span>, <span class="va">Hox4_1e4</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/plotApairOfChrs.html" class="external-link">plotApairOfChrs</a></span><span class="op">(</span><span class="st">"chr2"</span>, xlim <span class="op">=</span> <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/gb2xlim.html" class="external-link">gb2xlim</a></span><span class="op">(</span><span class="va">Hox4_1e4</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="OrthogroupSyntenies_files/figure-html/Hox4-1.svg" width="700"></p>
<div class="sourceCode" id="cb99"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/findOverlaps-methods.html" class="external-link">subsetByOverlaps</a></span><span class="op">(</span><span class="va">coa</span><span class="op">$</span><span class="va">Oki_Osa</span>, <span class="va">Hox4_4e4</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/plotApairOfChrs.html" class="external-link">plotApairOfChrs</a></span><span class="op">(</span><span class="st">"chr2"</span>, xlim <span class="op">=</span> <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/gb2xlim.html" class="external-link">gb2xlim</a></span><span class="op">(</span><span class="va">Hox4_4e4</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="OrthogroupSyntenies_files/figure-html/Hox4-2.svg" width="700"></p>
<div class="sourceCode" id="cb100"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/findOverlaps-methods.html" class="external-link">subsetByOverlaps</a></span><span class="op">(</span><span class="va">coa</span><span class="op">$</span><span class="va">Oki_Osa</span>, <span class="va">Hox4_1e5</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/plotApairOfChrs.html" class="external-link">plotApairOfChrs</a></span><span class="op">(</span><span class="st">"chr2"</span>, xlim <span class="op">=</span> <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/gb2xlim.html" class="external-link">gb2xlim</a></span><span class="op">(</span><span class="va">Hox4_1e5</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="OrthogroupSyntenies_files/figure-html/Hox4-3.svg" width="700"></p>
<div class="sourceCode" id="cb101"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## Genes</span>
<span class="va">Hox4_4e4.genes</span> <span class="op">&lt;-</span> <span class="va">transcripts</span><span class="op">$</span><span class="va">OKI2018.I69</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/findOverlaps-methods.html" class="external-link">subsetByOverlaps</a></span><span class="op">(</span><span class="va">Hox4_4e4</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/inter-range-methods.html" class="external-link">range</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span>
<span class="va">mids</span> <span class="op">&lt;-</span> <span class="va">Hox4_4e4.genes</span> |&gt; <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/gr2dna_seg.html" class="external-link">gr2dna_seg</a></span><span class="op">(</span><span class="op">)</span> |&gt; <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>mid <span class="op">=</span> <span class="op">(</span><span class="va">start</span> <span class="op">+</span> <span class="va">end</span><span class="op">)</span> <span class="op">/</span> <span class="fl">2</span><span class="op">)</span> |&gt; <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/pull.html" class="external-link">pull</a></span><span class="op">(</span><span class="va">mid</span><span class="op">)</span>
<span class="va">text</span> <span class="op">&lt;-</span> <span class="va">Hox4_4e4.genes</span><span class="op">$</span><span class="va">tx_name</span> |&gt; <span class="fu"><a href="https://rdrr.io/r/base/character.html" class="external-link">as.character</a></span><span class="op">(</span><span class="op">)</span>
<span class="va">annot</span> <span class="op">&lt;-</span> <span class="fu">genoPlotR</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/genoPlotR/man/annotation.html" class="external-link">annotation</a></span><span class="op">(</span>x1<span class="op">=</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/start.html" class="external-link">start</a></span><span class="op">(</span><span class="va">Hox4_4e4.genes</span><span class="op">)</span>, x2<span class="op">=</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/start.html" class="external-link">end</a></span><span class="op">(</span><span class="va">Hox4_4e4.genes</span><span class="op">)</span>, text<span class="op">=</span><span class="va">text</span>, rot<span class="op">=</span><span class="fl">30</span><span class="op">)</span>
<span class="va">gbs</span><span class="op">$</span><span class="va">Oki_Osa</span> |&gt; <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/plotApairOfChrs.html" class="external-link">plotApairOfChrs</a></span><span class="op">(</span><span class="st">"chr2"</span>, xlim <span class="op">=</span> <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/gb2xlim.html" class="external-link">gb2xlim</a></span><span class="op">(</span><span class="va">Hox4_4e4</span><span class="op">)</span>, annotations <span class="op">=</span> <span class="va">annot</span><span class="op">)</span></code></pre></div>
<p><img src="OrthogroupSyntenies_files/figure-html/Hox4-4.svg" width="700"></p>
</div>
<div class="section level3">
<h3 id="hox-1">Hox 1<a class="anchor" aria-label="anchor" href="#hox-1"></a>
</h3>
<p>HOG <code>N19.HOG0000193</code>.</p>
<div class="sourceCode" id="cb102"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">orthoPairs</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">15</span><span class="op">]</span>, \<span class="op">(</span><span class="va">gb</span><span class="op">)</span> <span class="op">{</span>
  <span class="va">x</span> <span class="op">&lt;-</span> <span class="va">gb</span> |&gt; <span class="fu">plyranges</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/plyranges/man/tidyverse-reexports.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">HOG</span> <span class="op">==</span> <span class="st">"N19.HOG0000193"</span><span class="op">)</span>
  <span class="kw">if</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="op">==</span> <span class="fl">0</span><span class="op">)</span> <span class="kw"><a href="https://rdrr.io/r/base/function.html" class="external-link">return</a></span><span class="op">(</span><span class="cn">NULL</span><span class="op">)</span>
  <span class="va">x</span><span class="op">}</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## $Oki_Osa</span>
<span class="co">## GBreaks object with 1 range and 9 metadata columns:</span>
<span class="co">##            seqnames            ranges strand |     tx_id     tx_name</span>
<span class="co">##               &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt; &lt;character&gt;</span>
<span class="co">##   g8043.t1     chr2 14331701-14339427      * |      9021    g8043.t1</span>
<span class="co">##            dNdS_GUIDANCE2 dNdS_HmmCleaner dNdS_PRANK                  query</span>
<span class="co">##                 &lt;numeric&gt;       &lt;numeric&gt;  &lt;numeric&gt;              &lt;GRanges&gt;</span>
<span class="co">##   g8043.t1         0.0172         0.01745    0.01745 Chr2:11876406-11881984</span>
<span class="co">##                       HOG          OG      Arm</span>
<span class="co">##               &lt;character&gt; &lt;character&gt; &lt;factor&gt;</span>
<span class="co">##   g8043.t1 N19.HOG0000193   OG0000004     long</span>
<span class="co">##   -------</span>
<span class="co">##   seqinfo: 19 sequences from OKI2018.I69 genome</span>
<span class="co">## </span>
<span class="co">## $Oki_Bar</span>
<span class="co">## GBreaks object with 1 range and 9 metadata columns:</span>
<span class="co">##            seqnames            ranges strand |     tx_id     tx_name</span>
<span class="co">##               &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt; &lt;character&gt;</span>
<span class="co">##   g8043.t1     chr2 14331701-14339427      * |      9021    g8043.t1</span>
<span class="co">##            dNdS_GUIDANCE2 dNdS_HmmCleaner dNdS_PRANK                query</span>
<span class="co">##                 &lt;numeric&gt;       &lt;numeric&gt;  &lt;numeric&gt;            &lt;GRanges&gt;</span>
<span class="co">##   g8043.t1         0.0172         0.01745    0.01745 Chr2:6198535-6201003</span>
<span class="co">##                       HOG          OG      Arm</span>
<span class="co">##               &lt;character&gt; &lt;character&gt; &lt;factor&gt;</span>
<span class="co">##   g8043.t1 N19.HOG0000193   OG0000004     long</span>
<span class="co">##   -------</span>
<span class="co">##   seqinfo: 19 sequences from OKI2018.I69 genome</span>
<span class="co">## </span>
<span class="co">## $Oki_Kum</span>
<span class="co">## GBreaks object with 1 range and 9 metadata columns:</span>
<span class="co">##            seqnames            ranges strand |     tx_id     tx_name</span>
<span class="co">##               &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt; &lt;character&gt;</span>
<span class="co">##   g8043.t1     chr2 14331701-14339427      * |      9021    g8043.t1</span>
<span class="co">##            dNdS_GUIDANCE2 dNdS_HmmCleaner dNdS_PRANK</span>
<span class="co">##                 &lt;numeric&gt;       &lt;numeric&gt;  &lt;numeric&gt;</span>
<span class="co">##   g8043.t1         0.0172         0.01745    0.01745</span>
<span class="co">##                                  query            HOG          OG      Arm</span>
<span class="co">##                              &lt;GRanges&gt;    &lt;character&gt; &lt;character&gt; &lt;factor&gt;</span>
<span class="co">##   g8043.t1 contig_81_1:1615872-1623410 N19.HOG0000193   OG0000004     long</span>
<span class="co">##   -------</span>
<span class="co">##   seqinfo: 19 sequences from OKI2018.I69 genome</span>
<span class="co">## </span>
<span class="co">## $Oki_Aom</span>
<span class="co">## GBreaks object with 1 range and 9 metadata columns:</span>
<span class="co">##            seqnames            ranges strand |     tx_id     tx_name</span>
<span class="co">##               &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt; &lt;character&gt;</span>
<span class="co">##   g8043.t1     chr2 14331701-14339427      * |      9021    g8043.t1</span>
<span class="co">##            dNdS_GUIDANCE2 dNdS_HmmCleaner dNdS_PRANK                     query</span>
<span class="co">##                 &lt;numeric&gt;       &lt;numeric&gt;  &lt;numeric&gt;                 &lt;GRanges&gt;</span>
<span class="co">##   g8043.t1         0.0172         0.01745    0.01745 contig_22_1:289322-292719</span>
<span class="co">##                       HOG          OG      Arm</span>
<span class="co">##               &lt;character&gt; &lt;character&gt; &lt;factor&gt;</span>
<span class="co">##   g8043.t1 N19.HOG0000193   OG0000004     long</span>
<span class="co">##   -------</span>
<span class="co">##   seqinfo: 19 sequences from OKI2018.I69 genome</span>
<span class="co">## </span>
<span class="co">## $Oki_Nor</span>
<span class="co">## GBreaks object with 1 range and 9 metadata columns:</span>
<span class="co">##            seqnames            ranges strand |     tx_id     tx_name</span>
<span class="co">##               &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt; &lt;character&gt;</span>
<span class="co">##   g8043.t1     chr2 14331701-14339427      * |      9021    g8043.t1</span>
<span class="co">##            dNdS_GUIDANCE2 dNdS_HmmCleaner dNdS_PRANK                    query</span>
<span class="co">##                 &lt;numeric&gt;       &lt;numeric&gt;  &lt;numeric&gt;                &lt;GRanges&gt;</span>
<span class="co">##   g8043.t1         0.0172         0.01745    0.01745 scaffold_9:221719-225168</span>
<span class="co">##                       HOG          OG      Arm</span>
<span class="co">##               &lt;character&gt; &lt;character&gt; &lt;factor&gt;</span>
<span class="co">##   g8043.t1 N19.HOG0000193   OG0000004     long</span>
<span class="co">##   -------</span>
<span class="co">##   seqinfo: 19 sequences from OKI2018.I69 genome</span>
<span class="co">## </span>
<span class="co">## $Osa_Bar</span>
<span class="co">## GBreaks object with 1 range and 9 metadata columns:</span>
<span class="co">##            seqnames            ranges strand |     tx_id     tx_name</span>
<span class="co">##               &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt; &lt;character&gt;</span>
<span class="co">##   g6520.t1     Chr2 11876406-11881984      * |      7447    g6520.t1</span>
<span class="co">##            dNdS_GUIDANCE2 dNdS_HmmCleaner dNdS_PRANK                query</span>
<span class="co">##                 &lt;numeric&gt;       &lt;numeric&gt;  &lt;numeric&gt;            &lt;GRanges&gt;</span>
<span class="co">##   g6520.t1         0.0172         0.01745    0.01745 Chr2:6198535-6201003</span>
<span class="co">##                       HOG          OG      Arm</span>
<span class="co">##               &lt;character&gt; &lt;character&gt; &lt;factor&gt;</span>
<span class="co">##   g6520.t1 N19.HOG0000193   OG0000004     long</span>
<span class="co">##   -------</span>
<span class="co">##   seqinfo: 483 sequences from OSKA2016v1.9 genome</span>
<span class="co">## </span>
<span class="co">## $Osa_Oki</span>
<span class="co">## GBreaks object with 1 range and 9 metadata columns:</span>
<span class="co">##            seqnames            ranges strand |     tx_id     tx_name</span>
<span class="co">##               &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt; &lt;character&gt;</span>
<span class="co">##   g6520.t1     Chr2 11876406-11881984      * |      7447    g6520.t1</span>
<span class="co">##            dNdS_GUIDANCE2 dNdS_HmmCleaner dNdS_PRANK                  query</span>
<span class="co">##                 &lt;numeric&gt;       &lt;numeric&gt;  &lt;numeric&gt;              &lt;GRanges&gt;</span>
<span class="co">##   g6520.t1         0.0172         0.01745    0.01745 chr2:14331701-14339427</span>
<span class="co">##                       HOG          OG      Arm</span>
<span class="co">##               &lt;character&gt; &lt;character&gt; &lt;factor&gt;</span>
<span class="co">##   g6520.t1 N19.HOG0000193   OG0000004     long</span>
<span class="co">##   -------</span>
<span class="co">##   seqinfo: 483 sequences from OSKA2016v1.9 genome</span>
<span class="co">## </span>
<span class="co">## $Osa_Kum</span>
<span class="co">## GBreaks object with 1 range and 9 metadata columns:</span>
<span class="co">##            seqnames            ranges strand |     tx_id     tx_name</span>
<span class="co">##               &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt; &lt;character&gt;</span>
<span class="co">##   g6520.t1     Chr2 11876406-11881984      * |      7447    g6520.t1</span>
<span class="co">##            dNdS_GUIDANCE2 dNdS_HmmCleaner dNdS_PRANK</span>
<span class="co">##                 &lt;numeric&gt;       &lt;numeric&gt;  &lt;numeric&gt;</span>
<span class="co">##   g6520.t1         0.0172         0.01745    0.01745</span>
<span class="co">##                                  query            HOG          OG      Arm</span>
<span class="co">##                              &lt;GRanges&gt;    &lt;character&gt; &lt;character&gt; &lt;factor&gt;</span>
<span class="co">##   g6520.t1 contig_81_1:1615872-1623410 N19.HOG0000193   OG0000004     long</span>
<span class="co">##   -------</span>
<span class="co">##   seqinfo: 483 sequences from OSKA2016v1.9 genome</span>
<span class="co">## </span>
<span class="co">## $Osa_Aom</span>
<span class="co">## GBreaks object with 1 range and 9 metadata columns:</span>
<span class="co">##            seqnames            ranges strand |     tx_id     tx_name</span>
<span class="co">##               &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt; &lt;character&gt;</span>
<span class="co">##   g6520.t1     Chr2 11876406-11881984      * |      7447    g6520.t1</span>
<span class="co">##            dNdS_GUIDANCE2 dNdS_HmmCleaner dNdS_PRANK                     query</span>
<span class="co">##                 &lt;numeric&gt;       &lt;numeric&gt;  &lt;numeric&gt;                 &lt;GRanges&gt;</span>
<span class="co">##   g6520.t1         0.0172         0.01745    0.01745 contig_22_1:289322-292719</span>
<span class="co">##                       HOG          OG      Arm</span>
<span class="co">##               &lt;character&gt; &lt;character&gt; &lt;factor&gt;</span>
<span class="co">##   g6520.t1 N19.HOG0000193   OG0000004     long</span>
<span class="co">##   -------</span>
<span class="co">##   seqinfo: 483 sequences from OSKA2016v1.9 genome</span>
<span class="co">## </span>
<span class="co">## $Osa_Nor</span>
<span class="co">## GBreaks object with 1 range and 9 metadata columns:</span>
<span class="co">##            seqnames            ranges strand |     tx_id     tx_name</span>
<span class="co">##               &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt; &lt;character&gt;</span>
<span class="co">##   g6520.t1     Chr2 11876406-11881984      * |      7447    g6520.t1</span>
<span class="co">##            dNdS_GUIDANCE2 dNdS_HmmCleaner dNdS_PRANK                    query</span>
<span class="co">##                 &lt;numeric&gt;       &lt;numeric&gt;  &lt;numeric&gt;                &lt;GRanges&gt;</span>
<span class="co">##   g6520.t1         0.0172         0.01745    0.01745 scaffold_9:221719-225168</span>
<span class="co">##                       HOG          OG      Arm</span>
<span class="co">##               &lt;character&gt; &lt;character&gt; &lt;factor&gt;</span>
<span class="co">##   g6520.t1 N19.HOG0000193   OG0000004     long</span>
<span class="co">##   -------</span>
<span class="co">##   seqinfo: 483 sequences from OSKA2016v1.9 genome</span>
<span class="co">## </span>
<span class="co">## $Bar_Osa</span>
<span class="co">## GBreaks object with 1 range and 9 metadata columns:</span>
<span class="co">##            seqnames          ranges strand |     tx_id     tx_name</span>
<span class="co">##               &lt;Rle&gt;       &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt; &lt;character&gt;</span>
<span class="co">##   g4409.t1     Chr2 6198535-6201003      * |      3884    g4409.t1</span>
<span class="co">##            dNdS_GUIDANCE2 dNdS_HmmCleaner dNdS_PRANK                  query</span>
<span class="co">##                 &lt;numeric&gt;       &lt;numeric&gt;  &lt;numeric&gt;              &lt;GRanges&gt;</span>
<span class="co">##   g4409.t1         0.0172         0.01745    0.01745 Chr2:11876406-11881984</span>
<span class="co">##                       HOG          OG      Arm</span>
<span class="co">##               &lt;character&gt; &lt;character&gt; &lt;factor&gt;</span>
<span class="co">##   g4409.t1 N19.HOG0000193   OG0000004     long</span>
<span class="co">##   -------</span>
<span class="co">##   seqinfo: 68 sequences from Bar2.p4 genome</span>
<span class="co">## </span>
<span class="co">## $Bar_Oki</span>
<span class="co">## GBreaks object with 1 range and 9 metadata columns:</span>
<span class="co">##            seqnames          ranges strand |     tx_id     tx_name</span>
<span class="co">##               &lt;Rle&gt;       &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt; &lt;character&gt;</span>
<span class="co">##   g4409.t1     Chr2 6198535-6201003      * |      3884    g4409.t1</span>
<span class="co">##            dNdS_GUIDANCE2 dNdS_HmmCleaner dNdS_PRANK                  query</span>
<span class="co">##                 &lt;numeric&gt;       &lt;numeric&gt;  &lt;numeric&gt;              &lt;GRanges&gt;</span>
<span class="co">##   g4409.t1         0.0172         0.01745    0.01745 chr2:14331701-14339427</span>
<span class="co">##                       HOG          OG      Arm</span>
<span class="co">##               &lt;character&gt; &lt;character&gt; &lt;factor&gt;</span>
<span class="co">##   g4409.t1 N19.HOG0000193   OG0000004     long</span>
<span class="co">##   -------</span>
<span class="co">##   seqinfo: 68 sequences from Bar2.p4 genome</span>
<span class="co">## </span>
<span class="co">## $Bar_Kum</span>
<span class="co">## GBreaks object with 1 range and 9 metadata columns:</span>
<span class="co">##            seqnames          ranges strand |     tx_id     tx_name</span>
<span class="co">##               &lt;Rle&gt;       &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt; &lt;character&gt;</span>
<span class="co">##   g4409.t1     Chr2 6198535-6201003      * |      3884    g4409.t1</span>
<span class="co">##            dNdS_GUIDANCE2 dNdS_HmmCleaner dNdS_PRANK</span>
<span class="co">##                 &lt;numeric&gt;       &lt;numeric&gt;  &lt;numeric&gt;</span>
<span class="co">##   g4409.t1         0.0172         0.01745    0.01745</span>
<span class="co">##                                  query            HOG          OG      Arm</span>
<span class="co">##                              &lt;GRanges&gt;    &lt;character&gt; &lt;character&gt; &lt;factor&gt;</span>
<span class="co">##   g4409.t1 contig_81_1:1615872-1623410 N19.HOG0000193   OG0000004     long</span>
<span class="co">##   -------</span>
<span class="co">##   seqinfo: 68 sequences from Bar2.p4 genome</span>
<span class="co">## </span>
<span class="co">## $Bar_Aom</span>
<span class="co">## GBreaks object with 1 range and 9 metadata columns:</span>
<span class="co">##            seqnames          ranges strand |     tx_id     tx_name</span>
<span class="co">##               &lt;Rle&gt;       &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt; &lt;character&gt;</span>
<span class="co">##   g4409.t1     Chr2 6198535-6201003      * |      3884    g4409.t1</span>
<span class="co">##            dNdS_GUIDANCE2 dNdS_HmmCleaner dNdS_PRANK                     query</span>
<span class="co">##                 &lt;numeric&gt;       &lt;numeric&gt;  &lt;numeric&gt;                 &lt;GRanges&gt;</span>
<span class="co">##   g4409.t1         0.0172         0.01745    0.01745 contig_22_1:289322-292719</span>
<span class="co">##                       HOG          OG      Arm</span>
<span class="co">##               &lt;character&gt; &lt;character&gt; &lt;factor&gt;</span>
<span class="co">##   g4409.t1 N19.HOG0000193   OG0000004     long</span>
<span class="co">##   -------</span>
<span class="co">##   seqinfo: 68 sequences from Bar2.p4 genome</span>
<span class="co">## </span>
<span class="co">## $Bar_Nor</span>
<span class="co">## GBreaks object with 1 range and 9 metadata columns:</span>
<span class="co">##            seqnames          ranges strand |     tx_id     tx_name</span>
<span class="co">##               &lt;Rle&gt;       &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt; &lt;character&gt;</span>
<span class="co">##   g4409.t1     Chr2 6198535-6201003      * |      3884    g4409.t1</span>
<span class="co">##            dNdS_GUIDANCE2 dNdS_HmmCleaner dNdS_PRANK                    query</span>
<span class="co">##                 &lt;numeric&gt;       &lt;numeric&gt;  &lt;numeric&gt;                &lt;GRanges&gt;</span>
<span class="co">##   g4409.t1         0.0172         0.01745    0.01745 scaffold_9:221719-225168</span>
<span class="co">##                       HOG          OG      Arm</span>
<span class="co">##               &lt;character&gt; &lt;character&gt; &lt;factor&gt;</span>
<span class="co">##   g4409.t1 N19.HOG0000193   OG0000004     long</span>
<span class="co">##   -------</span>
<span class="co">##   seqinfo: 68 sequences from Bar2.p4 genome</span></code></pre>
<div class="sourceCode" id="cb104"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">Hox1</span> <span class="op">&lt;-</span> <span class="va">orthoPairs</span><span class="op">$</span><span class="va">Oki_Osa</span> |&gt; <span class="fu">plyranges</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/plyranges/man/tidyverse-reexports.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">HOG</span> <span class="op">==</span> <span class="st">"N19.HOG0000193"</span><span class="op">)</span>
<span class="va">Hox1_1e4</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/GBreaks-class.html" class="external-link">GBreaks</a></span><span class="op">(</span>target <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/genomic-range-squeezers.html" class="external-link">granges</a></span><span class="op">(</span><span class="va">Hox1</span><span class="op">)</span> <span class="op">+</span> <span class="fl">1e4</span>, query <span class="op">=</span> <span class="va">Hox1</span><span class="op">$</span><span class="va">query</span> <span class="op">+</span> <span class="fl">1e4</span><span class="op">)</span>
<span class="va">Hox1_4e4</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/GBreaks-class.html" class="external-link">GBreaks</a></span><span class="op">(</span>target <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/genomic-range-squeezers.html" class="external-link">granges</a></span><span class="op">(</span><span class="va">Hox1</span><span class="op">)</span> <span class="op">+</span> <span class="fl">4e4</span>, query <span class="op">=</span> <span class="va">Hox1</span><span class="op">$</span><span class="va">query</span> <span class="op">+</span> <span class="fl">4e4</span><span class="op">)</span>
<span class="va">Hox1_1e5</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/GBreaks-class.html" class="external-link">GBreaks</a></span><span class="op">(</span>target <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/genomic-range-squeezers.html" class="external-link">granges</a></span><span class="op">(</span><span class="va">Hox1</span><span class="op">)</span> <span class="op">+</span> <span class="fl">1e5</span>, query <span class="op">=</span> <span class="va">Hox1</span><span class="op">$</span><span class="va">query</span> <span class="op">+</span> <span class="fl">1e5</span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/findOverlaps-methods.html" class="external-link">subsetByOverlaps</a></span><span class="op">(</span><span class="va">coa</span><span class="op">$</span><span class="va">Oki_Osa</span>, <span class="va">Hox1_1e4</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/plotApairOfChrs.html" class="external-link">plotApairOfChrs</a></span><span class="op">(</span><span class="st">"chr2"</span>, xlim <span class="op">=</span> <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/gb2xlim.html" class="external-link">gb2xlim</a></span><span class="op">(</span><span class="va">Hox1_1e4</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="OrthogroupSyntenies_files/figure-html/Hox1-1.svg" width="700"></p>
<div class="sourceCode" id="cb105"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/findOverlaps-methods.html" class="external-link">subsetByOverlaps</a></span><span class="op">(</span><span class="va">coa</span><span class="op">$</span><span class="va">Oki_Osa</span>, <span class="va">Hox1_4e4</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/plotApairOfChrs.html" class="external-link">plotApairOfChrs</a></span><span class="op">(</span><span class="st">"chr2"</span>, xlim <span class="op">=</span> <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/gb2xlim.html" class="external-link">gb2xlim</a></span><span class="op">(</span><span class="va">Hox1_4e4</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="OrthogroupSyntenies_files/figure-html/Hox1-2.svg" width="700"></p>
<div class="sourceCode" id="cb106"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/findOverlaps-methods.html" class="external-link">subsetByOverlaps</a></span><span class="op">(</span><span class="va">coa</span><span class="op">$</span><span class="va">Oki_Osa</span>, <span class="va">Hox1_1e5</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/plotApairOfChrs.html" class="external-link">plotApairOfChrs</a></span><span class="op">(</span><span class="st">"chr2"</span>, xlim <span class="op">=</span> <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/gb2xlim.html" class="external-link">gb2xlim</a></span><span class="op">(</span><span class="va">Hox1_1e5</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="OrthogroupSyntenies_files/figure-html/Hox1-3.svg" width="700"></p>
<div class="sourceCode" id="cb107"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## Genes</span>
<span class="va">Hox1_4e4.genes</span> <span class="op">&lt;-</span> <span class="va">transcripts</span><span class="op">$</span><span class="va">OKI2018.I69</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/findOverlaps-methods.html" class="external-link">subsetByOverlaps</a></span><span class="op">(</span><span class="va">Hox1_4e4</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/inter-range-methods.html" class="external-link">range</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span>
<span class="va">mids</span> <span class="op">&lt;-</span> <span class="va">Hox1_4e4.genes</span> |&gt; <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/gr2dna_seg.html" class="external-link">gr2dna_seg</a></span><span class="op">(</span><span class="op">)</span> |&gt; <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>mid <span class="op">=</span> <span class="op">(</span><span class="va">start</span> <span class="op">+</span> <span class="va">end</span><span class="op">)</span> <span class="op">/</span> <span class="fl">2</span><span class="op">)</span> |&gt; <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/pull.html" class="external-link">pull</a></span><span class="op">(</span><span class="va">mid</span><span class="op">)</span>
<span class="va">text</span> <span class="op">&lt;-</span> <span class="va">Hox1_4e4.genes</span><span class="op">$</span><span class="va">tx_name</span> |&gt; <span class="fu"><a href="https://rdrr.io/r/base/character.html" class="external-link">as.character</a></span><span class="op">(</span><span class="op">)</span>
<span class="va">annot</span> <span class="op">&lt;-</span> <span class="fu">genoPlotR</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/genoPlotR/man/annotation.html" class="external-link">annotation</a></span><span class="op">(</span>x1<span class="op">=</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/start.html" class="external-link">start</a></span><span class="op">(</span><span class="va">Hox1_4e4.genes</span><span class="op">)</span>, x2<span class="op">=</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/start.html" class="external-link">end</a></span><span class="op">(</span><span class="va">Hox1_4e4.genes</span><span class="op">)</span>, text<span class="op">=</span><span class="va">text</span>, rot<span class="op">=</span><span class="fl">30</span><span class="op">)</span>
<span class="va">gbs</span><span class="op">$</span><span class="va">Oki_Osa</span> |&gt; <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/plotApairOfChrs.html" class="external-link">plotApairOfChrs</a></span><span class="op">(</span><span class="st">"chr2"</span>, xlim <span class="op">=</span> <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/gb2xlim.html" class="external-link">gb2xlim</a></span><span class="op">(</span><span class="va">Hox1_4e4</span><span class="op">)</span>, annotations <span class="op">=</span> <span class="va">annot</span><span class="op">)</span></code></pre></div>
<p><img src="OrthogroupSyntenies_files/figure-html/Hox1-4.svg" width="700"></p>
</div>
</div>
<div class="section level2">
<h2 id="ancestral-states">Ancestral states<a class="anchor" aria-label="anchor" href="#ancestral-states"></a>
</h2>
<p>Prototype 1</p>
<p>In the concept case below, B has moved in the North Atlantic clade.</p>
<ul>
<li>Oki: A-B-C</li>
<li>Osa: A-B-C</li>
<li>Bar: A-C, and B can be found somewhere else</li>
</ul>
<p>As we work on alignment pairs, this translates to:</p>
<ul>
<li>Oki_Osa: A-B-C</li>
<li>Oki_Bar: A, B, C</li>
</ul>
<p>To find such cases I search for Oki-Bar orthoblock breakpoints that fall within an Oki-Osa orthoblock.</p>
<div class="sourceCode" id="cb108"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">reIndex</span> <span class="op">&lt;-</span> \<span class="op">(</span><span class="va">gb</span><span class="op">)</span> <span class="op">{</span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">gb</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq_along</a></span><span class="op">(</span><span class="va">gb</span><span class="op">)</span> ; <span class="va">gb</span><span class="op">}</span>
<span class="va">barBreaks</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">orthoBlocks_one2oneInOiks</span>, <span class="va">get_bps</span>, direction <span class="op">=</span> <span class="st">"mid"</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/SimpleList-class.html" class="external-link">SimpleList</a></span><span class="op">(</span><span class="op">)</span>


<span class="va">ROIs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/SimpleList-class.html" class="external-link">SimpleList</a></span><span class="op">(</span><span class="op">)</span>
<span class="va">findROIs</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">g1</span>, <span class="va">g2</span>, <span class="va">g3</span><span class="op">)</span> <span class="op">{</span>
  <span class="va">pair1_2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="va">g1</span>, sep <span class="op">=</span> <span class="st">'_'</span>, <span class="va">g2</span><span class="op">)</span>
  <span class="va">pair1_3</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="va">g1</span>, sep <span class="op">=</span> <span class="st">'_'</span>, <span class="va">g3</span><span class="op">)</span>
  <span class="va">gb</span> <span class="op">&lt;-</span> <span class="va">orthoBlocks_one2oneInOiks</span><span class="op">[[</span><span class="va">pair1_2</span><span class="op">]</span><span class="op">]</span> |&gt;
    <span class="fu">reIndex</span><span class="op">(</span><span class="op">)</span> |&gt;
    <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/findOverlaps-methods.html" class="external-link">subsetByOverlaps</a></span><span class="op">(</span><span class="va">barBreaks</span><span class="op">[[</span><span class="va">pair1_3</span><span class="op">]</span><span class="op">]</span><span class="op">)</span> |&gt;
    <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/subset.html" class="external-link">subset</a></span><span class="op">(</span><span class="va">geneNumber</span> <span class="op">&gt;</span> <span class="fl">2</span><span class="op">)</span>
  <span class="co"># Drop columns to ease browsing on single lines.</span>
  <span class="cn">NULL</span> <span class="op">-&gt;</span> <span class="va">gb</span><span class="op">$</span><span class="va">dNdS_GUIDANCE2</span> <span class="op">-&gt;</span> <span class="va">gb</span><span class="op">$</span><span class="va">dNdS_HmmCleaner</span> <span class="op">-&gt;</span> <span class="va">gb</span><span class="op">$</span><span class="va">dNdS_PRANK</span>
  <span class="va">gb</span>
<span class="op">}</span>
<span class="va">ROIs</span><span class="op">$</span><span class="va">Oki_Osa_Bar</span> <span class="op">&lt;-</span> <span class="fu">findROIs</span><span class="op">(</span><span class="st">"Oki"</span>, <span class="st">"Osa"</span>, <span class="st">"Bar"</span><span class="op">)</span>
<span class="va">ROIs</span><span class="op">$</span><span class="va">Oki_Bar_Osa</span> <span class="op">&lt;-</span> <span class="fu">findROIs</span><span class="op">(</span><span class="st">"Oki"</span>, <span class="st">"Bar"</span>, <span class="st">"Osa"</span><span class="op">)</span>
<span class="va">ROIs</span><span class="op">$</span><span class="va">Oki_Aom_Nor</span> <span class="op">&lt;-</span> <span class="fu">findROIs</span><span class="op">(</span><span class="st">"Oki"</span>, <span class="st">"Aom"</span>, <span class="st">"Nor"</span><span class="op">)</span>

<span class="va">findGene</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">gb</span>, <span class="va">gene</span><span class="op">)</span> <span class="op">{</span>
  <span class="kw">if</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/NULL.html" class="external-link">is.null</a></span><span class="op">(</span><span class="va">gb</span><span class="op">$</span><span class="va">geneNames</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span>
    <span class="va">ids</span> <span class="op">&lt;-</span> <span class="va">gb</span><span class="op">$</span><span class="va">tx_name</span>
    <span class="va">index</span> <span class="op">&lt;-</span> <span class="va">gb</span><span class="op">$</span><span class="va">tx_name</span> <span class="op"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html" class="external-link">%in%</a></span> <span class="va">gene</span>
  <span class="op">}</span> <span class="kw">else</span> <span class="op">{</span>
    <span class="va">ids</span> <span class="op">&lt;-</span> <span class="va">gb</span><span class="op">$</span><span class="va">geneNames</span>
    <span class="va">index</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/any.html" class="external-link">any</a></span><span class="op">(</span><span class="va">ids</span> <span class="op"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html" class="external-link">%in%</a></span> <span class="va">gene</span><span class="op">)</span>
  <span class="op">}</span>
  <span class="va">gb</span><span class="op">[</span><span class="va">index</span><span class="op">]</span>
<span class="op">}</span>

<span class="co"># Genes of Interest</span>

<span class="op">(</span><span class="va">GoI1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">head</a></span><span class="op">(</span><span class="va">ROIs</span><span class="op">$</span><span class="va">Oki_Osa_Bar</span><span class="op">$</span><span class="va">geneNames</span>, <span class="fl">1</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/unlist.html" class="external-link">unlist</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## [1] "g1206.t1" "g1207.t1" "g1211.t1"</span></code></pre>
<div class="sourceCode" id="cb110"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">GoI2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"g1205.t1"</span>, <span class="st">"g1208.t1"</span>, <span class="st">"g1209.t1"</span>, <span class="st">"g1210.t1"</span><span class="op">)</span> <span class="co"># To provide context</span>

<span class="co"># Exploring the region</span>

<span class="va">orthoBlocks</span><span class="op">$</span><span class="va">Oki_Osa</span> |&gt; <span class="fu">findGene</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="va">GoI1</span>, <span class="va">GoI2</span><span class="op">)</span><span class="op">)</span> <span class="co"># g1206.t1 g1207.t1 g1208.t1 g1211.t1 are in chr1:4573346-4588029 | Chr1:402525-416983  </span></code></pre></div>
<pre><code><span class="co">## GBreaks object with 1 range and 8 metadata columns:</span>
<span class="co">##       seqnames          ranges strand |              query     score</span>
<span class="co">##          &lt;Rle&gt;       &lt;IRanges&gt;  &lt;Rle&gt; |          &lt;GRanges&gt; &lt;integer&gt;</span>
<span class="co">##   [1]     chr1 4573346-4588029      * | Chr1:402525-416983     14684</span>
<span class="co">##                            geneNames              dNdS_GUIDANCE2</span>
<span class="co">##                      &lt;CharacterList&gt;               &lt;NumericList&gt;</span>
<span class="co">##   [1] g1206.t1,g1207.t1,g1208.t1,... 0.04258,0.01913,     NA,...</span>
<span class="co">##                   dNdS_HmmCleaner                  dNdS_PRANK geneNumber</span>
<span class="co">##                     &lt;NumericList&gt;               &lt;NumericList&gt;  &lt;integer&gt;</span>
<span class="co">##   [1] 0.04725,0.01899,     NA,... 0.04580,0.02468,     NA,...          4</span>
<span class="co">##            Arm</span>
<span class="co">##       &lt;factor&gt;</span>
<span class="co">##   [1]    short</span>
<span class="co">##   -------</span>
<span class="co">##   seqinfo: 19 sequences from OKI2018.I69 genome</span></code></pre>
<div class="sourceCode" id="cb112"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">orthoPairs</span> <span class="op">$</span><span class="va">Oki_Osa</span> |&gt; <span class="fu">findGene</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="va">GoI1</span>, <span class="va">GoI2</span><span class="op">)</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="op">)</span> <span class="co"># Individual coordinates of the genes</span></code></pre></div>
<pre><code><span class="co">##          seqnames   start     end width strand tx_id  tx_name dNdS_GUIDANCE2</span>
<span class="co">## g1206.t1     chr1 4573346 4575265  1920      *   631 g1206.t1        0.04258</span>
<span class="co">## g1207.t1     chr1 4575321 4577337  2017      *  2777 g1207.t1        0.01913</span>
<span class="co">## g1208.t1     chr1 4577995 4578612   618      *   632 g1208.t1             NA</span>
<span class="co">## g1211.t1     chr1 4581484 4588029  6546      *  2778 g1211.t1        0.02438</span>
<span class="co">##          dNdS_HmmCleaner dNdS_PRANK query.seqnames query.start query.end</span>
<span class="co">## g1206.t1         0.04725    0.04580           Chr1      414718    416983</span>
<span class="co">## g1207.t1         0.01899    0.02468           Chr1      410741    413759</span>
<span class="co">## g1208.t1              NA         NA           Chr1      409351    409953</span>
<span class="co">## g1211.t1         0.02420    0.02526           Chr1      402525    406012</span>
<span class="co">##          query.width query.strand query.tx_id query.tx_name</span>
<span class="co">## g1206.t1        2266            *        1722        g96.t1</span>
<span class="co">## g1207.t1        3019            *          52        g95.t1</span>
<span class="co">## g1208.t1         603            *        1721        g94.t1</span>
<span class="co">## g1211.t1        3488            *          50        g92.t1</span>
<span class="co">##          query.dNdS_GUIDANCE2 query.dNdS_HmmCleaner query.dNdS_PRANK</span>
<span class="co">## g1206.t1              0.04258               0.04725          0.04580</span>
<span class="co">## g1207.t1              0.01913               0.01899          0.02468</span>
<span class="co">## g1208.t1                   NA                    NA               NA</span>
<span class="co">## g1211.t1              0.02438               0.02420          0.02526</span>
<span class="co">##                     HOG        OG   Arm</span>
<span class="co">## g1206.t1 N19.HOG0003732 OG0000664 short</span>
<span class="co">## g1207.t1 N19.HOG0010718 OG0006871 short</span>
<span class="co">## g1208.t1 N19.HOG0015251 OG0016820 short</span>
<span class="co">## g1211.t1 N19.HOG0008357 OG0003884 short</span></code></pre>
<div class="sourceCode" id="cb114"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">orthoBlocks</span><span class="op">$</span><span class="va">Oki_Bar</span> |&gt; <span class="fu">findGene</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="va">GoI1</span>, <span class="va">GoI2</span><span class="op">)</span><span class="op">)</span> <span class="co"># g1205.t1 g1206.t1 g1207.t1 g1208.t1 g1211.t1 are in chr1 4571907-4588029 | Chr1:582810-2680665</span></code></pre></div>
<pre><code><span class="co">## GBreaks object with 3 ranges and 8 metadata columns:</span>
<span class="co">##       seqnames          ranges strand |                query     score</span>
<span class="co">##          &lt;Rle&gt;       &lt;IRanges&gt;  &lt;Rle&gt; |            &lt;GRanges&gt; &lt;integer&gt;</span>
<span class="co">##   [1]     chr1 4571907-4573262      * |   Chr1:582810-583163      1356</span>
<span class="co">##   [2]     chr1 4573346-4577337      * | Chr1:2669327-2674502      3992</span>
<span class="co">##   [3]     chr1 4577995-4588029      * | Chr1:2675030-2680665     10035</span>
<span class="co">##               geneNames  dNdS_GUIDANCE2 dNdS_HmmCleaner      dNdS_PRANK</span>
<span class="co">##         &lt;CharacterList&gt;   &lt;NumericList&gt;   &lt;NumericList&gt;   &lt;NumericList&gt;</span>
<span class="co">##   [1]          g1205.t1              NA              NA              NA</span>
<span class="co">##   [2] g1206.t1,g1207.t1 0.04258,0.01913 0.04725,0.01899 0.04580,0.02468</span>
<span class="co">##   [3] g1208.t1,g1211.t1      NA,0.02438       NA,0.0242      NA,0.02526</span>
<span class="co">##       geneNumber      Arm</span>
<span class="co">##        &lt;integer&gt; &lt;factor&gt;</span>
<span class="co">##   [1]          1    short</span>
<span class="co">##   [2]          2    short</span>
<span class="co">##   [3]          2    short</span>
<span class="co">##   -------</span>
<span class="co">##   seqinfo: 19 sequences from OKI2018.I69 genome</span></code></pre>
<div class="sourceCode" id="cb116"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">orthoPairs</span> <span class="op">$</span><span class="va">Oki_Bar</span> |&gt; <span class="fu">findGene</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="va">GoI1</span>, <span class="va">GoI2</span><span class="op">)</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="op">)</span> <span class="co"># Individual coordinates of the genes</span></code></pre></div>
<pre><code><span class="co">##          seqnames   start     end width strand tx_id  tx_name dNdS_GUIDANCE2</span>
<span class="co">## g1205.t1     chr1 4571907 4573262  1356      *   630 g1205.t1             NA</span>
<span class="co">## g1206.t1     chr1 4573346 4575265  1920      *   631 g1206.t1        0.04258</span>
<span class="co">## g1207.t1     chr1 4575321 4577337  2017      *  2777 g1207.t1        0.01913</span>
<span class="co">## g1208.t1     chr1 4577995 4578612   618      *   632 g1208.t1             NA</span>
<span class="co">## g1211.t1     chr1 4581484 4588029  6546      *  2778 g1211.t1        0.02438</span>
<span class="co">##          dNdS_HmmCleaner dNdS_PRANK query.seqnames query.start query.end</span>
<span class="co">## g1205.t1              NA         NA           Chr1      582810    583163</span>
<span class="co">## g1206.t1         0.04725    0.04580           Chr1     2672887   2674502</span>
<span class="co">## g1207.t1         0.01899    0.02468           Chr1     2669327   2672825</span>
<span class="co">## g1208.t1              NA         NA           Chr1     2675030   2675328</span>
<span class="co">## g1211.t1         0.02420    0.02526           Chr1     2678743   2680665</span>
<span class="co">##          query.width query.strand query.tx_id query.tx_name</span>
<span class="co">## g1205.t1         354            *        1566       g122.t1</span>
<span class="co">## g1206.t1        1616            *        1842       g651.t1</span>
<span class="co">## g1207.t1        3499            *         356       g650.t1</span>
<span class="co">## g1208.t1         299            *         357       g652.t1</span>
<span class="co">## g1211.t1        1923            *        1844       g654.t1</span>
<span class="co">##          query.dNdS_GUIDANCE2 query.dNdS_HmmCleaner query.dNdS_PRANK</span>
<span class="co">## g1205.t1                   NA                    NA               NA</span>
<span class="co">## g1206.t1              0.04258               0.04725          0.04580</span>
<span class="co">## g1207.t1              0.01913               0.01899          0.02468</span>
<span class="co">## g1208.t1                   NA                    NA               NA</span>
<span class="co">## g1211.t1              0.02438               0.02420          0.02526</span>
<span class="co">##                     HOG        OG   Arm</span>
<span class="co">## g1205.t1 N19.HOG0002360 OG0000249 short</span>
<span class="co">## g1206.t1 N19.HOG0003732 OG0000664 short</span>
<span class="co">## g1207.t1 N19.HOG0010718 OG0006871 short</span>
<span class="co">## g1208.t1 N19.HOG0015251 OG0016820 short</span>
<span class="co">## g1211.t1 N19.HOG0008357 OG0003884 short</span></code></pre>
<div class="sourceCode" id="cb118"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">orthoBlocks</span><span class="op">$</span><span class="va">Bar_Oki</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/findOverlaps-methods.html" class="external-link">subsetByOverlaps</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a></span><span class="op">(</span><span class="st">"Chr1:2669327-2680665"</span><span class="op">)</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/swap.html" class="external-link">swap</a></span><span class="op">(</span><span class="op">)</span>  <span class="co"># Let's forget about g1205.t1</span></code></pre></div>
<pre><code><span class="co">## GBreaks object with 2 ranges and 5 metadata columns:</span>
<span class="co">##       seqnames          ranges strand |         geneNames  dNdS_GUIDANCE2</span>
<span class="co">##          &lt;Rle&gt;       &lt;IRanges&gt;  &lt;Rle&gt; |   &lt;CharacterList&gt;   &lt;NumericList&gt;</span>
<span class="co">##   [1]     chr1 4573346-4577337      * | g1207.t1,g1206.t1 0.01913,0.04258</span>
<span class="co">##   [2]     chr1 4577995-4588029      * | g1208.t1,g1211.t1      NA,0.02438</span>
<span class="co">##       dNdS_HmmCleaner      dNdS_PRANK                query</span>
<span class="co">##         &lt;NumericList&gt;   &lt;NumericList&gt;            &lt;GRanges&gt;</span>
<span class="co">##   [1] 0.01899,0.04725 0.02468,0.04580 Chr1:2669327-2674502</span>
<span class="co">##   [2]       NA,0.0242      NA,0.02526 Chr1:2675030-2680665</span>
<span class="co">##   -------</span>
<span class="co">##   seqinfo: 19 sequences from OKI2018.I69 genome</span></code></pre>
<div class="sourceCode" id="cb120"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">orthoBlocks</span><span class="op">$</span><span class="va">Bar_Oki</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/findOverlaps-methods.html" class="external-link">subsetByOverlaps</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a></span><span class="op">(</span><span class="st">"Chr1:2669327-2680665"</span><span class="op">)</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/swap.html" class="external-link">swap</a></span><span class="op">(</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/cleanGaps.html" class="external-link">cleanGaps</a></span><span class="op">(</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/start.html" class="external-link">width</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## [1] 657</span></code></pre>
<pre><code>Oki         Osa       Bar  
g1206.t1    g96.t1    g651.t1
g1207.t1    g95.t1    g650.t1
g1208.t1    g94.t1    g652.t1
g1211.t1    g92.t1    g654.t1 </code></pre>
<div class="sourceCode" id="cb123"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">### Between "g1207.t1" and "g1208.t1", there is a repeat region, where the Bar genome broke, but that is still in the Osa genome (bri)</span>

<span class="va">unal</span><span class="op">$</span><span class="va">Oki_Osa</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/findOverlaps-methods.html" class="external-link">subsetByOverlaps</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a></span><span class="op">(</span><span class="st">"chr1"</span>, <span class="op">(</span><span class="fl">4577337</span> <span class="op">+</span> <span class="fl">4577995</span><span class="op">)</span><span class="op">/</span><span class="fl">2</span><span class="op">)</span><span class="op">)</span> <span class="co"># between g1207.t1 and g1208.t1 on Oki</span></code></pre></div>
<pre><code><span class="co">## GRanges object with 1 range and 4 metadata columns:</span>
<span class="co">##       seqnames          ranges strand |      Arm             rep transcripts</span>
<span class="co">##          &lt;Rle&gt;       &lt;IRanges&gt;  &lt;Rle&gt; | &lt;factor&gt; &lt;CharacterList&gt;       &lt;Rle&gt;</span>
<span class="co">##   [1]     chr1 4577462-4577685      * |    short         unknown        &lt;NA&gt;</span>
<span class="co">##          nonCoa</span>
<span class="co">##       &lt;logical&gt;</span>
<span class="co">##   [1]     FALSE</span>
<span class="co">##   -------</span>
<span class="co">##   seqinfo: 19 sequences from OKI2018.I69 genome</span></code></pre>
<div class="sourceCode" id="cb125"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">unal</span><span class="op">$</span><span class="va">Osa_Oki</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/findOverlaps-methods.html" class="external-link">subsetByOverlaps</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a></span><span class="op">(</span><span class="st">"Chr1"</span>, <span class="op">(</span> <span class="fl">409953</span> <span class="op">+</span> <span class="fl">410741</span> <span class="op">)</span><span class="op">/</span><span class="fl">2</span><span class="op">)</span><span class="op">)</span> <span class="co"># between g95.t1 and g94.t1 on Osa</span></code></pre></div>
<pre><code><span class="co">## GRanges object with 1 range and 4 metadata columns:</span>
<span class="co">##       seqnames        ranges strand |      Arm             rep transcripts</span>
<span class="co">##          &lt;Rle&gt;     &lt;IRanges&gt;  &lt;Rle&gt; | &lt;factor&gt; &lt;CharacterList&gt;       &lt;Rle&gt;</span>
<span class="co">##   [1]     Chr1 410306-410620      * |    short     unknown,rnd        &lt;NA&gt;</span>
<span class="co">##          nonCoa</span>
<span class="co">##       &lt;logical&gt;</span>
<span class="co">##   [1]     FALSE</span>
<span class="co">##   -------</span>
<span class="co">##   seqinfo: 483 sequences from OSKA2016v1.9 genome</span></code></pre>
<div class="sourceCode" id="cb127"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Or just:</span>

<span class="op">(</span><span class="va">BP1_Oki_Osa</span> <span class="op">&lt;-</span> <span class="va">bri</span><span class="op">$</span><span class="va">Oki_Osa</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/findOverlaps-methods.html" class="external-link">subsetByOverlaps</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a></span><span class="op">(</span><span class="st">"chr1"</span>, <span class="op">(</span><span class="fl">4577337</span> <span class="op">+</span> <span class="fl">4577995</span><span class="op">)</span><span class="op">/</span><span class="fl">2</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="co"># between g1207.t1 and g1208.t1 on Oki</span></code></pre></div>
<pre><code><span class="co">## GBreaks object with 1 range and 5 metadata columns:</span>
<span class="co">##       seqnames          ranges strand |              query      Arm</span>
<span class="co">##          &lt;Rle&gt;       &lt;IRanges&gt;  &lt;Rle&gt; |          &lt;GRanges&gt; &lt;factor&gt;</span>
<span class="co">##   [1]     chr1 4577462-4577685      * | Chr1:410306-410620    short</span>
<span class="co">##                   rep   repOvlp transcripts</span>
<span class="co">##       &lt;CharacterList&gt; &lt;integer&gt;       &lt;Rle&gt;</span>
<span class="co">##   [1]         unknown       110        &lt;NA&gt;</span>
<span class="co">##   -------</span>
<span class="co">##   seqinfo: 19 sequences from OKI2018.I69 genome</span></code></pre>
<div class="sourceCode" id="cb129"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Now on Bar:</span>
<span class="va">gbs</span><span class="op">$</span><span class="va">Bar_Oki</span> |&gt; <span class="fu">reIndex</span><span class="op">(</span><span class="op">)</span>|&gt; <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/findOverlaps-methods.html" class="external-link">subsetByOverlaps</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a></span><span class="op">(</span><span class="st">"Chr1"</span>, <span class="op">(</span><span class="fl">2674502</span> <span class="op">+</span> <span class="fl">2675030</span><span class="op">)</span><span class="op">/</span><span class="fl">2</span><span class="op">)</span><span class="op">)</span> <span class="co"># between g651.t1 and  g652.t1 on Bar</span></code></pre></div>
<pre><code><span class="co">## GBreaks object with 1 range and 8 metadata columns:</span>
<span class="co">##        seqnames          ranges strand |     score                query</span>
<span class="co">##           &lt;Rle&gt;       &lt;IRanges&gt;  &lt;Rle&gt; | &lt;numeric&gt;            &lt;GRanges&gt;</span>
<span class="co">##   2026     Chr1 2674688-2675041      + |       515 chr1:4577686-4578054</span>
<span class="co">##             Arm             rep   repOvlp transcripts        flag    nonCoa</span>
<span class="co">##        &lt;factor&gt; &lt;CharacterList&gt; &lt;integer&gt;       &lt;Rle&gt; &lt;character&gt; &lt;logical&gt;</span>
<span class="co">##   2026    short            &lt;NA&gt;         0     g652.t1         Col     FALSE</span>
<span class="co">##   -------</span>
<span class="co">##   seqinfo: 68 sequences from Bar2.p4 genome</span></code></pre>
<div class="sourceCode" id="cb131"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">coa</span><span class="op">$</span><span class="va">Bar_Oki</span> |&gt; <span class="fu">reIndex</span><span class="op">(</span><span class="op">)</span>|&gt; <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/findOverlaps-methods.html" class="external-link">subsetByOverlaps</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a></span><span class="op">(</span><span class="st">"Chr1"</span>, <span class="op">(</span><span class="fl">2674502</span> <span class="op">+</span> <span class="fl">2675030</span><span class="op">)</span><span class="op">/</span><span class="fl">2</span><span class="op">)</span><span class="op">)</span> <span class="co"># between g651.t1 and  g652.t1 on Bar</span></code></pre></div>
<pre><code><span class="co">## GBreaks object with 1 range and 8 metadata columns:</span>
<span class="co">##       seqnames          ranges strand |                query     score      Arm</span>
<span class="co">##          &lt;Rle&gt;       &lt;IRanges&gt;  &lt;Rle&gt; |            &lt;GRanges&gt; &lt;integer&gt; &lt;factor&gt;</span>
<span class="co">##   893     Chr1 2674688-2680680      + | chr1:4577686-4588063      5993    short</span>
<span class="co">##                   rep   repOvlp transcripts        flag    nonCoa</span>
<span class="co">##       &lt;CharacterList&gt; &lt;integer&gt;       &lt;Rle&gt; &lt;character&gt; &lt;logical&gt;</span>
<span class="co">##   893     unknown,rnd      1060     g652.t1        &lt;NA&gt;     FALSE</span>
<span class="co">##   -------</span>
<span class="co">##   seqinfo: 68 sequences from Bar2.p4 genome</span></code></pre>
<div class="sourceCode" id="cb133"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="op">(</span><span class="va">BP1_Bar</span> <span class="op">&lt;-</span> <span class="va">unal</span><span class="op">$</span><span class="va">Bar_Oki</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/findOverlaps-methods.html" class="external-link">subsetByOverlaps</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a></span><span class="op">(</span><span class="st">"Chr1"</span>, <span class="op">(</span><span class="fl">2674538</span> <span class="op">+</span> <span class="fl">2674688</span><span class="op">)</span><span class="op">/</span><span class="fl">2</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="co"># between g651.t1 and  g652.t1 on Bar according to gbs or coa</span></code></pre></div>
<pre><code><span class="co">## GRanges object with 1 range and 4 metadata columns:</span>
<span class="co">##       seqnames          ranges strand |      Arm             rep transcripts</span>
<span class="co">##          &lt;Rle&gt;       &lt;IRanges&gt;  &lt;Rle&gt; | &lt;factor&gt; &lt;CharacterList&gt;       &lt;Rle&gt;</span>
<span class="co">##   [1]     Chr1 2674539-2674687      * |    short            &lt;NA&gt;        &lt;NA&gt;</span>
<span class="co">##          nonCoa</span>
<span class="co">##       &lt;logical&gt;</span>
<span class="co">##   [1]      TRUE</span>
<span class="co">##   -------</span>
<span class="co">##   seqinfo: 68 sequences from Bar2.p4 genome</span></code></pre>
<div class="sourceCode" id="cb135"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Nothing noticeable on either strand</span>
<span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/pairwiseAlignment.html" class="external-link">pairwiseAlignment</a></span><span class="op">(</span>type <span class="op">=</span><span class="st">"local"</span>, <span class="va">BP1_Oki_Osa</span> |&gt; <span class="fu">plyranges</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/plyranges/man/tidyverse-reexports.html" class="external-link">mutate</a></span><span class="op">(</span>strand <span class="op">=</span> <span class="st">'-'</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Local PairwiseAlignmentsSingleSubject (1 of 1)</span>
<span class="co">## pattern: [164] <span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">TTTT</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">TT</span></span>
<span class="co">## subject: [182] <span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">TTTT</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">TT</span></span>
<span class="co">## score: 17.8358</span></code></pre>
<div class="sourceCode" id="cb137"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/pairwiseAlignment.html" class="external-link">pairwiseAlignment</a></span><span class="op">(</span>type <span class="op">=</span><span class="st">"local"</span>, <span class="va">BP1_Oki_Osa</span> |&gt; <span class="fu">plyranges</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/plyranges/man/tidyverse-reexports.html" class="external-link">mutate</a></span><span class="op">(</span>strand <span class="op">=</span> <span class="st">'+'</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Local PairwiseAlignmentsSingleSubject (1 of 1)</span>
<span class="co">## pattern: [37] <span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">GG</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">CC</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span></span>
<span class="co">## subject: [23] <span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">CC</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span></span>
<span class="co">## score: 15.90003</span></code></pre>
<div class="sourceCode" id="cb139"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/pairwiseAlignment.html" class="external-link">pairwiseAlignment</a></span><span class="op">(</span>type <span class="op">=</span><span class="st">"local"</span>, <span class="va">BP1_Oki_Osa</span> <span class="op">$</span><span class="va">query</span>, <span class="va">BP1_Bar</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Local PairwiseAlignmentsSingleSubject (1 of 1)</span>
<span class="co">## pattern: [188] <span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">TTTTTT</span><span style="color: #000000; background-color: #AFFFAF;">C</span></span>
<span class="co">## subject:  [31] <span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">TTTTTT</span><span style="color: #000000; background-color: #AFFFAF;">C</span></span>
<span class="co">## score: 15.85405</span></code></pre>
<div class="sourceCode" id="cb141"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/pairwiseAlignment.html" class="external-link">pairwiseAlignment</a></span><span class="op">(</span>type <span class="op">=</span><span class="st">"global"</span>, <span class="va">BP1_Oki_Osa</span> <span class="op">$</span><span class="va">query</span>, <span class="va">BP1_Bar</span>  |&gt; <span class="fu">plyranges</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/plyranges/man/tidyverse-reexports.html" class="external-link">mutate</a></span><span class="op">(</span>strand <span class="op">=</span> <span class="st">'-'</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Global PairwiseAlignmentsSingleSubject (1 of 1)</span>
<span class="co">## pattern: <span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">TTT</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #AFFFFF;">GGG</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">AAA</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #AFFFAF;">CC</span><span style="color: #000000; background-color: #FFAFAF;">AAAA</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">CC</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span>...<span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">CC</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">AAA</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">CC</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">TTTT</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">C</span></span>
<span class="co">## subject: <span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">G</span>----<span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">AAA</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">AA</span>-<span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #FFD7AF;">T</span>...<span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">CC</span><span style="color: #000000; background-color: #AFFFFF;">G</span>-----------------<span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">TTTT</span><span style="color: #000000; background-color: #AFFFFF;">G</span>------</span>
<span class="co">## score: -867.5856</span></code></pre>
<div class="sourceCode" id="cb143"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">oP_one2oneInOiks_bridges</span><span class="op">$</span><span class="va">Oki_Osa</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/findOverlaps-methods.html" class="external-link">subsetByOverlaps</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a></span><span class="op">(</span><span class="st">"chr1"</span>, <span class="op">(</span><span class="fl">4577337</span> <span class="op">+</span> <span class="fl">4577995</span><span class="op">)</span><span class="op">/</span><span class="fl">2</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## GBreaks object with 1 range and 1 metadata column:</span>
<span class="co">##       seqnames          ranges strand |              query</span>
<span class="co">##          &lt;Rle&gt;       &lt;IRanges&gt;  &lt;Rle&gt; |          &lt;GRanges&gt;</span>
<span class="co">##   [1]     chr1 4577338-4581483      * | Chr1:413760-414717</span>
<span class="co">##   -------</span>
<span class="co">##   seqinfo: 19 sequences from OKI2018.I69 genome</span></code></pre>
<p>I can also search for orthologs that are marked “Col” in Oki_Osa and “Tra” in Osa_Bar.</p>
</div>
<div class="section level2">
<h2 id="session-information">Session information<a class="anchor" aria-label="anchor" href="#session-information"></a>
</h2>
<div class="sourceCode" id="cb145"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## R version 4.1.2 (2021-11-01)</span>
<span class="co">## Platform: x86_64-pc-linux-gnu (64-bit)</span>
<span class="co">## Running under: Debian GNU/Linux bookworm/sid</span>
<span class="co">## </span>
<span class="co">## Matrix products: default</span>
<span class="co">## BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.10.0</span>
<span class="co">## LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.10.0</span>
<span class="co">## </span>
<span class="co">## locale:</span>
<span class="co">##  [1] LC_CTYPE=en_GB.UTF-8       LC_NUMERIC=C              </span>
<span class="co">##  [3] LC_TIME=en_GB.UTF-8        LC_COLLATE=en_GB.UTF-8    </span>
<span class="co">##  [5] LC_MONETARY=en_GB.UTF-8    LC_MESSAGES=en_GB.UTF-8   </span>
<span class="co">##  [7] LC_PAPER=en_GB.UTF-8       LC_NAME=C                 </span>
<span class="co">##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span>
<span class="co">## [11] LC_MEASUREMENT=en_GB.UTF-8 LC_IDENTIFICATION=C       </span>
<span class="co">## </span>
<span class="co">## attached base packages:</span>
<span class="co">## [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span>
<span class="co">## [8] base     </span>
<span class="co">## </span>
<span class="co">## other attached packages:</span>
<span class="co">##  [1] BSgenome.Oidioi.genoscope.OdB3_1.0.0   </span>
<span class="co">##  [2] BSgenome.Oidioi.OIST.AOM.5.5f_1.0.1    </span>
<span class="co">##  [3] BSgenome.Oidioi.OIST.KUM.M3.7f_1.0.1   </span>
<span class="co">##  [4] BSgenome.Oidioi.OIST.Bar2.p4_1.0.1     </span>
<span class="co">##  [5] BSgenome.Oidioi.OIST.OSKA2016v1.9_1.0.0</span>
<span class="co">##  [6] BSgenome.Oidioi.OIST.OKI2018.I69_1.0.1 </span>
<span class="co">##  [7] OikScrambling_3.0.0                    </span>
<span class="co">##  [8] ggplot2_3.3.5                          </span>
<span class="co">##  [9] GenomicBreaks_0.13.1                   </span>
<span class="co">## [10] BSgenome_1.62.0                        </span>
<span class="co">## [11] rtracklayer_1.54.0                     </span>
<span class="co">## [12] Biostrings_2.62.0                      </span>
<span class="co">## [13] XVector_0.34.0                         </span>
<span class="co">## [14] GenomicRanges_1.46.1                   </span>
<span class="co">## [15] GenomeInfoDb_1.30.1                    </span>
<span class="co">## [16] IRanges_2.28.0                         </span>
<span class="co">## [17] S4Vectors_0.32.3                       </span>
<span class="co">## [18] BiocGenerics_0.40.0                    </span>
<span class="co">## </span>
<span class="co">## loaded via a namespace (and not attached):</span>
<span class="co">##   [1] backports_1.4.1             Hmisc_4.6-0                </span>
<span class="co">##   [3] BiocFileCache_2.2.1         systemfonts_1.0.3          </span>
<span class="co">##   [5] plyr_1.8.6                  splines_4.1.2              </span>
<span class="co">##   [7] BiocParallel_1.28.3         digest_0.6.29              </span>
<span class="co">##   [9] htmltools_0.5.2             tiff_0.1-11                </span>
<span class="co">##  [11] GO.db_3.14.0                gdata_2.18.0               </span>
<span class="co">##  [13] fansi_1.0.2                 magrittr_2.0.2             </span>
<span class="co">##  [15] checkmate_2.0.0             memoise_2.0.1              </span>
<span class="co">##  [17] cluster_2.1.2               tzdb_0.2.0                 </span>
<span class="co">##  [19] readr_2.1.2                 annotate_1.72.0            </span>
<span class="co">##  [21] matrixStats_0.61.0          R.utils_2.11.0             </span>
<span class="co">##  [23] pkgdown_2.0.2               prettyunits_1.1.1          </span>
<span class="co">##  [25] jpeg_0.1-9                  colorspace_2.0-3           </span>
<span class="co">##  [27] rappdirs_0.3.3              blob_1.2.2                 </span>
<span class="co">##  [29] textshaping_0.3.6           xfun_0.30                  </span>
<span class="co">##  [31] dplyr_1.0.8                 crayon_1.5.0               </span>
<span class="co">##  [33] RCurl_1.98-1.6              jsonlite_1.8.0             </span>
<span class="co">##  [35] lme4_1.1-28                 survival_3.2-13            </span>
<span class="co">##  [37] glue_1.6.2                  gtable_0.3.0               </span>
<span class="co">##  [39] zlibbioc_1.40.0             DelayedArray_0.20.0        </span>
<span class="co">##  [41] plyranges_1.14.0            weights_1.0.4              </span>
<span class="co">##  [43] abind_1.4-5                 scales_1.1.1               </span>
<span class="co">##  [45] DBI_1.1.2                   Rcpp_1.0.8.3               </span>
<span class="co">##  [47] plotrix_3.8-2               progress_1.2.2             </span>
<span class="co">##  [49] genoPlotR_0.8.11            xtable_1.8-4               </span>
<span class="co">##  [51] htmlTable_2.4.0             foreign_0.8-81             </span>
<span class="co">##  [53] bit_4.0.4                   Formula_1.2-4              </span>
<span class="co">##  [55] htmlwidgets_1.5.4           httr_1.4.2                 </span>
<span class="co">##  [57] RColorBrewer_1.1-2          ellipsis_0.3.2             </span>
<span class="co">##  [59] mice_3.14.0                 farver_2.1.0               </span>
<span class="co">##  [61] pkgconfig_2.0.3             XML_3.99-0.9               </span>
<span class="co">##  [63] R.methodsS3_1.8.1           dbplyr_2.1.1               </span>
<span class="co">##  [65] nnet_7.3-16                 sass_0.4.0                 </span>
<span class="co">##  [67] locfit_1.5-9.5              utf8_1.2.2                 </span>
<span class="co">##  [69] tidyselect_1.1.2            rlang_1.0.2                </span>
<span class="co">##  [71] reshape2_1.4.4              AnnotationDbi_1.56.2       </span>
<span class="co">##  [73] munsell_0.5.0               tools_4.1.2                </span>
<span class="co">##  [75] cachem_1.0.6                cli_3.2.0                  </span>
<span class="co">##  [77] generics_0.1.2              RSQLite_2.2.10             </span>
<span class="co">##  [79] ade4_1.7-18                 broom_0.7.12               </span>
<span class="co">##  [81] evaluate_0.15               stringr_1.4.0              </span>
<span class="co">##  [83] fastmap_1.1.0               fftwtools_0.9-11           </span>
<span class="co">##  [85] yaml_2.3.5                  ragg_1.2.1                 </span>
<span class="co">##  [87] knitr_1.37                  bit64_4.0.5                </span>
<span class="co">##  [89] fs_1.5.2                    purrr_0.3.4                </span>
<span class="co">##  [91] heatmaps_1.18.0             KEGGREST_1.34.0            </span>
<span class="co">##  [93] nlme_3.1-153                R.oo_1.24.0                </span>
<span class="co">##  [95] poweRlaw_0.70.6             xml2_1.3.3                 </span>
<span class="co">##  [97] pracma_2.3.8                biomaRt_2.50.3             </span>
<span class="co">##  [99] compiler_4.1.2              rstudioapi_0.13            </span>
<span class="co">## [101] filelock_1.0.2              curl_4.3.2                 </span>
<span class="co">## [103] png_0.1-7                   tibble_3.1.6               </span>
<span class="co">## [105] bslib_0.3.1                 stringi_1.7.6              </span>
<span class="co">## [107] highr_0.9                   GenomicFeatures_1.46.5     </span>
<span class="co">## [109] desc_1.4.1                  lattice_0.20-45            </span>
<span class="co">## [111] CNEr_1.30.0                 Matrix_1.4-0               </span>
<span class="co">## [113] nloptr_1.2.2.3              vctrs_0.3.8                </span>
<span class="co">## [115] pillar_1.7.0                lifecycle_1.0.1            </span>
<span class="co">## [117] jquerylib_0.1.4             data.table_1.14.2          </span>
<span class="co">## [119] bitops_1.0-7                R6_2.5.1                   </span>
<span class="co">## [121] BiocIO_1.4.0                latticeExtra_0.6-29        </span>
<span class="co">## [123] KernSmooth_2.23-20          gridExtra_2.3              </span>
<span class="co">## [125] boot_1.3-28                 gtools_3.9.2               </span>
<span class="co">## [127] MASS_7.3-54                 assertthat_0.2.1           </span>
<span class="co">## [129] SummarizedExperiment_1.24.0 rprojroot_2.0.2            </span>
<span class="co">## [131] rjson_0.2.21                withr_2.5.0                </span>
<span class="co">## [133] GenomicAlignments_1.30.0    Rsamtools_2.10.0           </span>
<span class="co">## [135] GenomeInfoDbData_1.2.7      parallel_4.1.2             </span>
<span class="co">## [137] hms_1.1.1                   EBImage_4.36.0             </span>
<span class="co">## [139] grid_4.1.2                  rpart_4.1-15               </span>
<span class="co">## [141] minqa_1.2.4                 tidyr_1.2.0                </span>
<span class="co">## [143] rmarkdown_2.11              MatrixGenerics_1.6.0       </span>
<span class="co">## [145] Biobase_2.54.0              base64enc_0.1-3            </span>
<span class="co">## [147] restfulr_0.0.13</span></code></pre>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Aleksandra Bliznina, Michael Mansfield, Charles Plessy.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.2.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
