<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Orthogroup syntenies â€¢ OikScrambling</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Orthogroup syntenies">
<meta property="og:description" content="OikScrambling">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">OikScrambling</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.3.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/OikScrambling.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/ColinearityInterruptors.html">Regions interrupting colinearity</a>
    </li>
    <li>
      <a href="../articles/DistanceMeasures.html">Distance Measures</a>
    </li>
    <li>
      <a href="../articles/Inversions.html">Inversions</a>
    </li>
    <li>
      <a href="../articles/LoadGenomicBreaks.html">Load genomic break data</a>
    </li>
    <li>
      <a href="../articles/OrthogroupSyntenies.html">Orthogroup syntenies</a>
    </li>
    <li>
      <a href="../articles/PWM.html">Motif finding and analysis</a>
    </li>
    <li>
      <a href="../articles/ParallelPlots.html">Whole-chromosome parallel plots</a>
    </li>
    <li>
      <a href="../articles/RegionWidths.html">Width of the aligned and mapped regions.</a>
    </li>
    <li>
      <a href="../articles/RepeatRegions.html">Repeat regions</a>
    </li>
    <li>
      <a href="../articles/Translocations.html">Translocations</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/oist/LuscombeU_OikScrambling/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="OrthogroupSyntenies_files/header-attrs-2.11/header-attrs.js"></script><script src="OrthogroupSyntenies_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Orthogroup syntenies</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/oist/LuscombeU_OikScrambling/blob/master/vignettes/OrthogroupSyntenies.Rmd"><code>vignettes/OrthogroupSyntenies.Rmd</code></a></small>
      <div class="hidden name"><code>OrthogroupSyntenies.Rmd</code></div>

    </div>

    
    
<div id="load-packages-and-data" class="section level1">
<h1 class="hasAnchor">
<a href="#load-packages-and-data" class="anchor"></a>Load packages and data</h1>
<div id="r-packages" class="section level2">
<h2 class="hasAnchor">
<a href="#r-packages" class="anchor"></a>R packages</h2>
<p>Install the <em>GenomicBreaks</em> package with:</p>
<pre><code><a href="https://devtools.r-lib.org//reference/remote-reexports.html">devtools::install_github("oist/GenomicBreaks")</a></code></pre>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/message.html">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="op">{</span>
  <span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="st">'BreakpointsData'</span><span class="op">)</span>
  <span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="st"><a href="https://oist.github.io/GenomicBreaks">'GenomicBreaks'</a></span><span class="op">)</span>
  <span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="st"><a href="https://bioconductor.org/packages/BSgenome">'BSgenome'</a></span><span class="op">)</span>
  <span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="st"><a href="https://bioconductor.org/packages/GenomicFeatures">'GenomicFeatures'</a></span><span class="op">)</span>
<span class="op">}</span><span class="op">)</span></code></pre></div>
</div>
<div id="bsgenome-packages" class="section level2">
<h2 class="hasAnchor">
<a href="#bsgenome-packages" class="anchor"></a>BSgenome packages</h2>
<p>Example command to install one <em>Oikopleura</em> BSgenome package if you do not yet have it:</p>
<pre><code><a href="https://rdrr.io/r/utils/install.packages.html">install.packages("BSgenome.Odioica.local.OKI2018.I69", repos="https://oist.github.io/plessy_oikgenomes_drat/")</a></code></pre>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="st">"BSgenome.Odioica.local.OKI2018.I69"</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="st">"BSgenome.Odioica.local.OSKA2016v1.9"</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="st">"BSgenome.Odioica.local.Bar2.p4"</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="st">"BSgenome.Odioica.local.KUM.M3"</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="st">"BSgenome.Odioica.local.AOM.5"</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="st">"BSgenome.Odioica.local.Odioica.reference.v3.0"</span><span class="op">)</span>

<span class="va">genomes</span> <span class="op">&lt;-</span> <span class="fu">SimpleList</span><span class="op">(</span>
  <span class="co"># Chromosome assemblies</span>
  Oki <span class="op">=</span> <span class="va">OKI2018_I69</span>, Osa <span class="op">=</span> <span class="va">OSKA2016v1.9</span>, Bar <span class="op">=</span> <span class="va">Bar2_p4</span>,
  <span class="co"># Less contiguous assemblies</span>
  Kum <span class="op">=</span> <span class="va">KUM_M3</span>, Aom <span class="op">=</span> <span class="va">AOM_5</span>, Nor <span class="op">=</span> <span class="va">OdB3</span><span class="op">)</span>

<span class="co">## Genome lengths</span>
<span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="va">genomes</span>, \<span class="op">(</span><span class="va">g</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="fu">seqlengths</span><span class="op">(</span><span class="fu">seqinfo</span><span class="op">(</span><span class="va">g</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code>##      Oki      Osa      Bar      Kum      Aom      Nor 
## 64281565 56625162 55793437 64653574 56753784 70471451</code></pre>
</div>
<div id="annotations" class="section level2">
<h2 class="hasAnchor">
<a href="#annotations" class="anchor"></a>Annotations</h2>
<p>The annotations from <code>/bucket/LuscombeU/common/Breakpoints/Annotations</code> are distributed in the same drat repository as the alignment data.</p>
<pre><code><a href="https://rdrr.io/r/utils/install.packages.html">install.packages("BreakpointsData", repos="https://oist.github.io/plessy_oikgenomes_drat/")</a></code></pre>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">annots</span> <span class="op">&lt;-</span> <span class="fu">SimpleList</span><span class="op">(</span><span class="op">)</span>

<span class="va">gff2txdb</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">file</span>, <span class="va">genome</span><span class="op">)</span> <span class="op">{</span>
  <span class="va">file</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="st">"extdata/Annotations/"</span>, <span class="va">file</span><span class="op">)</span>, package <span class="op">=</span> <span class="st">"BreakpointsData"</span><span class="op">)</span>
  <span class="va">tx</span> <span class="op">&lt;-</span> <span class="fu">rtracklayer</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/rtracklayer/man/GFFFile-class.html">import.gff</a></span><span class="op">(</span><span class="va">file</span><span class="op">)</span>
  <span class="va">tx</span> <span class="op">&lt;-</span> <span class="va">tx</span><span class="op">[</span><span class="fu">seqnames</span><span class="op">(</span><span class="va">tx</span><span class="op">)</span> <span class="op">%in%</span> <span class="fu">seqnames</span><span class="op">(</span><span class="va">genome</span><span class="op">)</span><span class="op">]</span>
  <span class="va">tx</span> <span class="op">&lt;-</span> <span class="fu">GRanges</span><span class="op">(</span><span class="va">tx</span>, seqinfo <span class="op">=</span> <span class="fu">seqinfo</span><span class="op">(</span><span class="va">genome</span><span class="op">)</span><span class="op">)</span>
  <span class="va">tx</span> <span class="op">&lt;-</span> <span class="fu">GenomicFeatures</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/GenomicFeatures/man/makeTxDbFromGRanges.html">makeTxDbFromGRanges</a></span><span class="op">(</span><span class="va">tx</span><span class="op">)</span>
<span class="op">}</span>

<span class="va">gff2txdb_Norway</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">file</span>, <span class="va">genome</span><span class="op">)</span> <span class="op">{</span>
  <span class="va">file</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="st">"extdata/Annotations/"</span>, <span class="va">file</span><span class="op">)</span>, package <span class="op">=</span> <span class="st">"BreakpointsData"</span><span class="op">)</span>
  <span class="va">tx</span> <span class="op">&lt;-</span> <span class="fu">rtracklayer</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/rtracklayer/man/GFFFile-class.html">import.gff</a></span><span class="op">(</span><span class="va">file</span><span class="op">)</span>
  <span class="va">tx</span> <span class="op">&lt;-</span> <span class="va">tx</span><span class="op">[</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span><span class="op">(</span><span class="va">tx</span><span class="op">$</span><span class="va">mRNA</span><span class="op">)</span><span class="op">]</span>
  <span class="co"># Remove "scaffoldA" objects in the OdB3 annotation</span>
  <span class="va">tx</span> <span class="op">&lt;-</span> <span class="va">tx</span><span class="op">[</span><span class="fu">seqnames</span><span class="op">(</span><span class="va">tx</span><span class="op">)</span> <span class="op">%in%</span> <span class="fu">seqnames</span><span class="op">(</span><span class="va">genome</span><span class="op">)</span><span class="op">]</span>
  <span class="va">tx</span><span class="op">$</span><span class="va">ID</span> <span class="op">&lt;-</span> <span class="va">tx</span><span class="op">$</span><span class="va">mRNA</span>
  <span class="va">tx</span> <span class="op">&lt;-</span> <span class="fu">GRanges</span><span class="op">(</span><span class="va">tx</span>, seqinfo <span class="op">=</span> <span class="fu">seqinfo</span><span class="op">(</span><span class="va">genome</span><span class="op">)</span><span class="op">)</span>
  <span class="va">tx</span> <span class="op">&lt;-</span> <span class="fu">GenomicFeatures</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/GenomicFeatures/man/makeTxDbFromGRanges.html">makeTxDbFromGRanges</a></span><span class="op">(</span><span class="va">tx</span><span class="op">)</span>
<span class="op">}</span>

<span class="va">annots</span><span class="op">$</span><span class="va">Oki</span> <span class="op">&lt;-</span> <span class="fu">gff2txdb</span><span class="op">(</span><span class="st">"OKI2018_I69.v2/OKI2018_I69.v2.gm.gff.gz"</span>,  <span class="va">OKI2018_I69</span><span class="op">)</span>
<span class="va">annots</span><span class="op">$</span><span class="va">Osa</span> <span class="op">&lt;-</span> <span class="fu">gff2txdb</span><span class="op">(</span><span class="st">"OSKA2016v1.9/OSKA2016v1.9.gm.gff.gz"</span>,      <span class="va">OSKA2016v1.9</span><span class="op">)</span>
<span class="va">annots</span><span class="op">$</span><span class="va">Bar</span> <span class="op">&lt;-</span> <span class="fu">gff2txdb</span><span class="op">(</span><span class="st">"Bar2_p4.Flye/Bar2_p4.Flye.gm.gff.gz"</span>,      <span class="va">Bar2_p4</span><span class="op">)</span>
<span class="va">annots</span><span class="op">$</span><span class="va">Kum</span> <span class="op">&lt;-</span> <span class="fu">gff2txdb</span><span class="op">(</span><span class="st">"KUM-M3-7f/KUM-M3-7f.gm.gff.gz"</span>,            <span class="va">KUM_M3</span><span class="op">)</span>
<span class="va">annots</span><span class="op">$</span><span class="va">Aom</span> <span class="op">&lt;-</span> <span class="fu">gff2txdb</span><span class="op">(</span><span class="st">"AOM-5-5f/AOM-5-5f.gm.gff.gz"</span>,              <span class="va">AOM_5</span><span class="op">)</span>
<span class="va">annots</span><span class="op">$</span><span class="va">Nor</span> <span class="op">&lt;-</span> <span class="fu">gff2txdb_Norway</span><span class="op">(</span><span class="st">"OdB3/Oikopleura_annot_v1.0.gff.gz"</span>, <span class="va">OdB3</span><span class="op">)</span></code></pre></div>
</div>
<div id="orthogroups" class="section level2">
<h2 class="hasAnchor">
<a href="#orthogroups" class="anchor"></a>Orthogroups</h2>
<p>Load Orthogroups for <em>Oikopleura</em> from the <code>BreakpointsData</code> package with the <code><a href="https://oist.github.io/GenomicBreaks/reference/load_one_to_ones.html">GenomicBreaks::load_one_to_ones()</a></code> function. Original file is in <code>/bucket/LuscombeU/common/Breakpoints/Orthologues/selectedTunicates+Amph+vertebrates_lp_100clstr_blast/OrthoFinder/Results_Sep07/Phylogenetic_Hierarchical_Orthogroups/N6.tsv</code>.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Example for loading all orthogroups:</span>
<span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/load_one_to_ones.html">load_one_to_ones</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata/OrthoFinder/N6.tsv.gz"</span>, package <span class="op">=</span> <span class="st">"BreakpointsData"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code>## DataFrame with 5359 rows and 9 columns
##                HOG          OG Gene Tree Parent Clade
##        &lt;character&gt; &lt;character&gt;            &lt;character&gt;
## 1    N6.HOG0000000   OG0000000                     n6
## 2    N6.HOG0000001   OG0000000                    n11
## 3    N6.HOG0000002   OG0000000                    n89
## 4    N6.HOG0000003   OG0000000                   n136
## 5    N6.HOG0000007   OG0000000                   n245
## ...            ...         ...                    ...
## 5355 N6.HOG0012841   OG0013622                     n0
## 5356 N6.HOG0012842   OG0013623                     n0
## 5357 N6.HOG0012843   OG0013624                     n0
## 5358 N6.HOG0012844   OG0013625                     n0
## 5359 N6.HOG0012845   OG0013626                     n0
##      AOM-5-5f.prot.longest.fa_1 Bar2_p4.Flye.prot.longest.fa_1
##                     &lt;character&gt;                    &lt;character&gt;
## 1             AOM-5-5f.g9988.t1         Bar2_p4_Flye.g12314.t1
## 2             AOM-5-5f.g1151.t1          Bar2_p4_Flye.g6775.t1
## 3             AOM-5-5f.g4268.t2           Bar2_p4_Flye.g457.t1
## 4             AOM-5-5f.g4051.t1           Bar2_p4_Flye.g567.t1
## 5             AOM-5-5f.g2181.t1          Bar2_p4_Flye.g8280.t1
## ...                         ...                            ...
## 5355          AOM-5-5f.g9983.t1         Bar2_p4_Flye.g12308.t1
## 5356          AOM-5-5f.g9984.t1         Bar2_p4_Flye.g12310.t2
## 5357          AOM-5-5f.g9985.t1         Bar2_p4_Flye.g12311.t1
## 5358          AOM-5-5f.g9987.t1         Bar2_p4_Flye.g12313.t1
## 5359          AOM-5-5f.g9990.t1         Bar2_p4_Flye.g12316.t1
##      KUM-M3-7f.prot.longest.fa_1 OKI2018_I69.v2.prot.longest.fa_1
##                      &lt;character&gt;                      &lt;character&gt;
## 1            KUM-M3-7f.g12059.t2           OKI2018_I69.v2.g1612..
## 2             KUM-M3-7f.g8561.t1           OKI2018_I69.v2.g9389..
## 3             KUM-M3-7f.g3690.t1           OKI2018_I69.v2.g107.t1
## 4             KUM-M3-7f.g4428.t1           OKI2018_I69.v2.g924.t1
## 5             KUM-M3-7f.g5128.t1           OKI2018_I69.v2.g1156..
## ...                          ...                              ...
## 5355         KUM-M3-7f.g12065.t1           OKI2018_I69.v2.g1612..
## 5356         KUM-M3-7f.g12063.t1           OKI2018_I69.v2.g1612..
## 5357         KUM-M3-7f.g12062.t1           OKI2018_I69.v2.g1612..
## 5358         KUM-M3-7f.g12060.t1           OKI2018_I69.v2.g1612..
## 5359         KUM-M3-7f.g12057.t1           OKI2018_I69.v2.g1612..
##      OSKA2016v1.9.prot.longest.fa_1    OdB3.v1.0.prot.fa_1
##                         &lt;character&gt;            &lt;character&gt;
## 1            OSKA2016v1.9.g12765.t1 OdB3.GSOIDT00006516001
## 2             OSKA2016v1.9.g8621.t1 OdB3.GSOIDT00014232001
## 3              OSKA2016v1.9.g457.t1 OdB3.GSOIDT00006315001
## 4             OSKA2016v1.9.g1017.t1 OdB3.GSOIDT00002732001
## 5             OSKA2016v1.9.g9587.t1 OdB3.GSOIDT00003802001
## ...                             ...                    ...
## 5355         OSKA2016v1.9.g12771.t1 OdB3.GSOIDT00006510001
## 5356         OSKA2016v1.9.g12769.t1 OdB3.GSOIDT00006512001
## 5357         OSKA2016v1.9.g12768.t1 OdB3.GSOIDT00006513001
## 5358         OSKA2016v1.9.g12766.t1 OdB3.GSOIDT00006515001
## 5359         OSKA2016v1.9.g12763.t1 OdB3.GSOIDT00006520001</code></pre>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Example for loading a pair</span>
<span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/load_one_to_ones.html">load_one_to_ones</a></span><span class="op">(</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata/OrthoFinder/N6.tsv.gz"</span>, package <span class="op">=</span> <span class="st">"BreakpointsData"</span><span class="op">)</span>
                , <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Bar2_p4.Flye.prot.longest.fa_1"</span>, <span class="st">"OSKA2016v1.9.prot.longest.fa_1"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code>## DataFrame with 9217 rows and 5 columns
##                HOG          OG Gene Tree Parent Clade
##        &lt;character&gt; &lt;character&gt;            &lt;character&gt;
## 1    N6.HOG0000000   OG0000000                     n6
## 2    N6.HOG0000001   OG0000000                    n11
## 3    N6.HOG0000002   OG0000000                    n89
## 4    N6.HOG0000003   OG0000000                   n136
## 5    N6.HOG0000004   OG0000000                   n159
## ...            ...         ...                    ...
## 9213 N6.HOG0016671   OG0022509                      -
## 9214 N6.HOG0016676   OG0022514                      -
## 9215 N6.HOG0016685   OG0022523                      -
## 9216 N6.HOG0016687   OG0022525                      -
## 9217 N6.HOG0016688   OG0022526                      -
##      Bar2_p4.Flye.prot.longest.fa_1 OSKA2016v1.9.prot.longest.fa_1
##                         &lt;character&gt;                    &lt;character&gt;
## 1            Bar2_p4_Flye.g12314.t1         OSKA2016v1.9.g12765.t1
## 2             Bar2_p4_Flye.g6775.t1          OSKA2016v1.9.g8621.t1
## 3              Bar2_p4_Flye.g457.t1           OSKA2016v1.9.g457.t1
## 4              Bar2_p4_Flye.g567.t1          OSKA2016v1.9.g1017.t1
## 5             Bar2_p4_Flye.g1107.t1          OSKA2016v1.9.g2469.t1
## ...                             ...                            ...
## 9213          Bar2_p4_Flye.g8448.t1          OSKA2016v1.9.g9813.t1
## 9214          Bar2_p4_Flye.g8611.t1          OSKA2016v1.9.g9658.t1
## 9215          Bar2_p4_Flye.g9382.t1         OSKA2016v1.9.g10500.t1
## 9216          Bar2_p4_Flye.g9616.t1          OSKA2016v1.9.g9098.t1
## 9217          Bar2_p4_Flye.g9635.t1          OSKA2016v1.9.g9126.t1</code></pre>
</div>
</div>
<div id="prepare-gbreak-objects" class="section level1">
<h1 class="hasAnchor">
<a href="#prepare-gbreak-objects" class="anchor"></a>Prepare GBreak objects</h1>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">orthoPairToGBreaks</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">name1</span>, <span class="va">name2</span>, <span class="va">annot1</span>, <span class="va">annot2</span><span class="op">)</span> <span class="op">{</span>
  <span class="va">orthoPairs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/load_one_to_ones.html">load_one_to_ones</a></span><span class="op">(</span>
    <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata/OrthoFinder/N6.tsv.gz"</span>, package <span class="op">=</span> <span class="st">"BreakpointsData"</span><span class="op">)</span>,
    <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">name1</span>, <span class="va">name2</span><span class="op">)</span><span class="op">)</span>

  <span class="va">IDs2GRanges</span> <span class="op">&lt;-</span> <span class="kw">function</span> <span class="op">(</span><span class="va">IDs</span>, <span class="va">annot</span><span class="op">)</span> <span class="op">{</span>
    <span class="va">prefix</span> <span class="op">&lt;-</span> <span class="fu">Biobase</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/lcSuffix.html">lcPrefix</a></span><span class="op">(</span><span class="va">IDs</span><span class="op">)</span>                  <span class="co"># Guess prefix in transcript IDs from HOG files</span>
    <span class="va">not_prefix</span> <span class="op">&lt;-</span> <span class="fu">Biobase</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/lcSuffix.html">lcPrefix</a></span><span class="op">(</span><span class="va">annot</span><span class="op">$</span><span class="va">tx_name</span><span class="op">)</span>    <span class="co"># Remove trailing characters that are part of the name</span>
    <span class="va">not_prefix</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">not_prefix</span>,<span class="st">"$"</span><span class="op">)</span>              <span class="co"># Anchor to end of the string</span>
    <span class="va">prefix</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html">sub</a></span><span class="op">(</span><span class="va">not_prefix</span>, <span class="st">""</span>, <span class="va">prefix</span><span class="op">)</span>             <span class="co"># Finalise prefix</span>
    <span class="va">IDs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html">sub</a></span><span class="op">(</span><span class="va">prefix</span>, <span class="st">""</span>, <span class="va">IDs</span><span class="op">)</span>                       <span class="co"># Remove prefix from IDS</span>
    <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">annot</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">annot</span><span class="op">$</span><span class="va">tx_name</span>                     <span class="co"># Store transcript name in names slot</span>
    <span class="va">gr</span> <span class="op">&lt;-</span> <span class="fu">granges</span><span class="op">(</span><span class="va">annot</span><span class="op">[</span><span class="va">IDs</span><span class="op">]</span><span class="op">)</span>                         <span class="co"># Discard extra metadata and sort by ID</span>
    <span class="fu">strand</span><span class="op">(</span><span class="va">gr</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="st">"*"</span>                                 <span class="co"># Make strandless</span>
    <span class="va">gr</span>                                                <span class="co"># Return the object</span>
  <span class="op">}</span>
  
  <span class="va">gb</span>       <span class="op">&lt;-</span> <span class="fu">IDs2GRanges</span><span class="op">(</span><span class="va">orthoPairs</span><span class="op">[</span>,<span class="va">name1</span><span class="op">]</span>, <span class="fu"><a href="https://rdrr.io/pkg/GenomicFeatures/man/transcripts.html">transcripts</a></span><span class="op">(</span><span class="va">annot1</span><span class="op">)</span><span class="op">)</span>
  <span class="va">gb</span><span class="op">$</span><span class="va">query</span> <span class="op">&lt;-</span> <span class="fu">IDs2GRanges</span><span class="op">(</span><span class="va">orthoPairs</span><span class="op">[</span>,<span class="va">name2</span><span class="op">]</span>, <span class="fu"><a href="https://rdrr.io/pkg/GenomicFeatures/man/transcripts.html">transcripts</a></span><span class="op">(</span><span class="va">annot2</span><span class="op">)</span><span class="op">)</span>
  <span class="va">gb</span> <span class="op">&lt;-</span> <span class="fu">GenomicBreaks</span><span class="fu">:::</span><span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/GBreaks-class.html">GBreaks</a></span><span class="op">(</span><span class="va">gb</span><span class="op">)</span>
  <span class="fu"><a href="https://rdrr.io/r/base/sort.html">sort</a></span><span class="op">(</span><span class="va">gb</span><span class="op">)</span>
<span class="op">}</span>

<span class="va">orthoPairs</span> <span class="op">&lt;-</span> <span class="fu">SimpleList</span><span class="op">(</span><span class="op">)</span>
<span class="va">orthoPairs</span><span class="op">$</span><span class="va">Oki_Osa</span> <span class="op">&lt;-</span> <span class="fu">orthoPairToGBreaks</span><span class="op">(</span><span class="st">"OKI2018_I69.v2.prot.longest.fa_1"</span>, <span class="st">"OSKA2016v1.9.prot.longest.fa_1"</span>
                                        , <span class="va">annots</span><span class="op">$</span><span class="va">Oki</span>, <span class="va">annots</span><span class="op">$</span><span class="va">Osa</span><span class="op">)</span>
<span class="va">orthoPairs</span><span class="op">$</span><span class="va">Oki_Bar</span> <span class="op">&lt;-</span> <span class="fu">orthoPairToGBreaks</span><span class="op">(</span><span class="st">"OKI2018_I69.v2.prot.longest.fa_1"</span>, <span class="st">"Bar2_p4.Flye.prot.longest.fa_1"</span>
                                        , <span class="va">annots</span><span class="op">$</span><span class="va">Oki</span>, <span class="va">annots</span><span class="op">$</span><span class="va">Bar</span><span class="op">)</span>
<span class="va">orthoPairs</span><span class="op">$</span><span class="va">Oki_Kum</span> <span class="op">&lt;-</span> <span class="fu">orthoPairToGBreaks</span><span class="op">(</span><span class="st">"OKI2018_I69.v2.prot.longest.fa_1"</span>, <span class="st">"KUM-M3-7f.prot.longest.fa_1"</span>
                                        , <span class="va">annots</span><span class="op">$</span><span class="va">Oki</span>, <span class="va">annots</span><span class="op">$</span><span class="va">Kum</span><span class="op">)</span>
<span class="va">orthoPairs</span><span class="op">$</span><span class="va">Oki_Aom</span> <span class="op">&lt;-</span> <span class="fu">orthoPairToGBreaks</span><span class="op">(</span><span class="st">"OKI2018_I69.v2.prot.longest.fa_1"</span>, <span class="st">"AOM-5-5f.prot.longest.fa_1"</span>
                                        , <span class="va">annots</span><span class="op">$</span><span class="va">Oki</span>, <span class="va">annots</span><span class="op">$</span><span class="va">Aom</span><span class="op">)</span>
<span class="co">## Problem: some Norway transcripts come from alternative "scaffoldA" sequences that are not in the genome file.</span>
<span class="co"># orthoPairs$Bar_Nor &lt;- orthoPairToGBreaks("Bar2_p4.Flye.prot.longest.fa_1", "OdB3.v1.0.prot.fa_1"</span>
<span class="co">#                                         , annots$Bar, annots$Nor)</span>

<span class="va">orthoPairs</span><span class="op">$</span><span class="va">Osa_Bar</span> <span class="op">&lt;-</span> <span class="fu">orthoPairToGBreaks</span><span class="op">(</span><span class="st">"OSKA2016v1.9.prot.longest.fa_1"</span>, <span class="st">"Bar2_p4.Flye.prot.longest.fa_1"</span>
                                        , <span class="va">annots</span><span class="op">$</span><span class="va">Osa</span>, <span class="va">annots</span><span class="op">$</span><span class="va">Bar</span><span class="op">)</span>
<span class="va">orthoPairs</span><span class="op">$</span><span class="va">Osa_Oki</span> <span class="op">&lt;-</span> <span class="fu">orthoPairToGBreaks</span><span class="op">(</span><span class="st">"OSKA2016v1.9.prot.longest.fa_1"</span>, <span class="st">"OKI2018_I69.v2.prot.longest.fa_1"</span>
                                        , <span class="va">annots</span><span class="op">$</span><span class="va">Osa</span>, <span class="va">annots</span><span class="op">$</span><span class="va">Oki</span><span class="op">)</span>
<span class="va">orthoPairs</span><span class="op">$</span><span class="va">Osa_Kum</span> <span class="op">&lt;-</span> <span class="fu">orthoPairToGBreaks</span><span class="op">(</span><span class="st">"OSKA2016v1.9.prot.longest.fa_1"</span>, <span class="st">"KUM-M3-7f.prot.longest.fa_1"</span>
                                        , <span class="va">annots</span><span class="op">$</span><span class="va">Osa</span>, <span class="va">annots</span><span class="op">$</span><span class="va">Kum</span><span class="op">)</span>
<span class="va">orthoPairs</span><span class="op">$</span><span class="va">Osa_Aom</span> <span class="op">&lt;-</span> <span class="fu">orthoPairToGBreaks</span><span class="op">(</span><span class="st">"OSKA2016v1.9.prot.longest.fa_1"</span>, <span class="st">"AOM-5-5f.prot.longest.fa_1"</span>
                                        , <span class="va">annots</span><span class="op">$</span><span class="va">Osa</span>, <span class="va">annots</span><span class="op">$</span><span class="va">Aom</span><span class="op">)</span>
<span class="co">## Problem: some Norway transcripts come from alternative "scaffoldA" sequences that are not in the genome file.</span>
<span class="co"># orthoPairs$Bar_Nor &lt;- orthoPairToGBreaks("Bar2_p4.Flye.prot.longest.fa_1", "OdB3.v1.0.prot.fa_1"</span>
<span class="co">#                                         , annots$Bar, annots$Nor)</span>


<span class="va">orthoPairs</span><span class="op">$</span><span class="va">Bar_Osa</span> <span class="op">&lt;-</span> <span class="fu">orthoPairToGBreaks</span><span class="op">(</span><span class="st">"Bar2_p4.Flye.prot.longest.fa_1"</span>, <span class="st">"OSKA2016v1.9.prot.longest.fa_1"</span>
                                        , <span class="va">annots</span><span class="op">$</span><span class="va">Bar</span>, <span class="va">annots</span><span class="op">$</span><span class="va">Osa</span><span class="op">)</span>
<span class="va">orthoPairs</span><span class="op">$</span><span class="va">Bar_Oki</span> <span class="op">&lt;-</span> <span class="fu">orthoPairToGBreaks</span><span class="op">(</span><span class="st">"Bar2_p4.Flye.prot.longest.fa_1"</span>, <span class="st">"OKI2018_I69.v2.prot.longest.fa_1"</span>
                                        , <span class="va">annots</span><span class="op">$</span><span class="va">Bar</span>, <span class="va">annots</span><span class="op">$</span><span class="va">Oki</span><span class="op">)</span>
<span class="va">orthoPairs</span><span class="op">$</span><span class="va">Bar_Kum</span> <span class="op">&lt;-</span> <span class="fu">orthoPairToGBreaks</span><span class="op">(</span><span class="st">"Bar2_p4.Flye.prot.longest.fa_1"</span>, <span class="st">"KUM-M3-7f.prot.longest.fa_1"</span>
                                        , <span class="va">annots</span><span class="op">$</span><span class="va">Bar</span>, <span class="va">annots</span><span class="op">$</span><span class="va">Kum</span><span class="op">)</span>
<span class="va">orthoPairs</span><span class="op">$</span><span class="va">Bar_Aom</span> <span class="op">&lt;-</span> <span class="fu">orthoPairToGBreaks</span><span class="op">(</span><span class="st">"Bar2_p4.Flye.prot.longest.fa_1"</span>, <span class="st">"AOM-5-5f.prot.longest.fa_1"</span>
                                        , <span class="va">annots</span><span class="op">$</span><span class="va">Bar</span>, <span class="va">annots</span><span class="op">$</span><span class="va">Aom</span><span class="op">)</span>
<span class="co">## Problem: some Norway transcripts come from alternative "scaffoldA" sequences that are not in the genome file.</span>
<span class="co"># orthoPairs$Bar_Nor &lt;- orthoPairToGBreaks("Bar2_p4.Flye.prot.longest.fa_1", "OdB3.v1.0.prot.fa_1"</span>
<span class="co">#                                         , annots$Bar, annots$Nor)</span>

<span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="va">orthoPairs</span>, <span class="va">length</span><span class="op">)</span></code></pre></div>
<pre><code>## Oki_Osa Oki_Bar Oki_Kum Oki_Aom Osa_Bar Osa_Oki Osa_Kum Osa_Aom Bar_Osa Bar_Oki 
##    9637    9390   11624    9683    9217    9637    9498   10383    9217    9390 
## Bar_Kum Bar_Aom 
##    9297    9275</code></pre>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/load.html">load</a></span><span class="op">(</span><span class="st">"BreakPoints.Rdata"</span><span class="op">)</span>
<span class="va">orthoPairs</span><span class="op">[</span> <span class="fl">1</span><span class="op">:</span><span class="fl">4</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="va">orthoPairs</span><span class="op">[</span> <span class="fl">1</span><span class="op">:</span><span class="fl">4</span><span class="op">]</span>,  <span class="va">flagLongShort</span>, <span class="va">longShort</span><span class="op">$</span><span class="va">Oki</span><span class="op">)</span>
<span class="va">orthoPairs</span><span class="op">[</span> <span class="fl">5</span><span class="op">:</span><span class="fl">8</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="va">orthoPairs</span><span class="op">[</span> <span class="fl">5</span><span class="op">:</span><span class="fl">8</span><span class="op">]</span>,  <span class="va">flagLongShort</span>, <span class="va">longShort</span><span class="op">$</span><span class="va">Osa</span><span class="op">)</span>
<span class="va">orthoPairs</span><span class="op">[</span><span class="fl">9</span><span class="op">:</span><span class="fl">11</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="va">orthoPairs</span><span class="op">[</span><span class="fl">9</span><span class="op">:</span><span class="fl">11</span><span class="op">]</span>,  <span class="va">flagLongShort</span>, <span class="va">longShort</span><span class="op">$</span><span class="va">Bar</span><span class="op">)</span></code></pre></div>
<div id="problem-some-annotations-overlap" class="section level2">
<h2 class="hasAnchor">
<a href="#problem-some-annotations-overlap" class="anchor"></a>Problem: some annotations overlap</h2>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/GenomicFeatures/man/transcripts.html">transcripts</a></span><span class="op">(</span><span class="va">annots</span><span class="op">$</span><span class="va">Bar</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] 15741</code></pre>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/GenomicFeatures/man/transcripts.html">transcripts</a></span><span class="op">(</span><span class="va">annots</span><span class="op">$</span><span class="va">Bar</span><span class="op">)</span> |&gt; <span class="fu">reduce</span><span class="op">(</span>min <span class="op">=</span> <span class="fl">0</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] 14262</code></pre>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">flagOverlaps</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">gr</span><span class="op">)</span> <span class="op">{</span>
  <span class="co"># Find overlaps</span>
  <span class="va">ov</span> <span class="op">&lt;-</span> <span class="fu">findOverlaps</span><span class="op">(</span><span class="va">gr</span><span class="op">)</span>
  <span class="co"># Index of self hits in the gr object</span>
  <span class="va">idx</span> <span class="op">&lt;-</span> <span class="fu">queryHits</span><span class="op">(</span><span class="va">ov</span><span class="op">[</span><span class="op">!</span><span class="fu">isSelfHit</span><span class="op">(</span><span class="va">ov</span><span class="op">)</span><span class="op">]</span><span class="op">)</span>
  <span class="co"># Convert to Boolean indexing the gr object</span>
  <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq_along</a></span><span class="op">(</span><span class="va">gr</span><span class="op">)</span> <span class="op">%in%</span> <span class="va">idx</span>
<span class="op">}</span>

<span class="fu"><a href="https://rdrr.io/pkg/GenomicFeatures/man/transcripts.html">transcripts</a></span><span class="op">(</span><span class="va">annots</span><span class="op">$</span><span class="va">Bar</span><span class="op">)</span><span class="op">[</span><span class="fu">flagOverlaps</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/GenomicFeatures/man/transcripts.html">transcripts</a></span><span class="op">(</span><span class="va">annots</span><span class="op">$</span><span class="va">Bar</span><span class="op">)</span><span class="op">)</span><span class="op">]</span></code></pre></div>
<pre><code>## GRanges object with 2668 ranges and 2 metadata columns:
##          seqnames          ranges strand |     tx_id     tx_name
##             &lt;Rle&gt;       &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt; &lt;character&gt;
##      [1]     Chr1   495446-496292      + |        58     g107.t2
##      [2]     Chr1   495446-496688      + |        59     g107.t1
##      [3]     Chr1   687958-689814      + |        76     g138.t1
##      [4]     Chr1   687958-689814      + |        77     g138.t2
##      [5]     Chr1   748328-751932      + |        82     g147.t2
##      ...      ...             ...    ... .       ...         ...
##   [2664]      YSR 4098404-4105602      - |     15657   g13975.t2
##   [2665]      YSR 4778101-4778538      - |     15701   g14058.t1
##   [2666]      YSR 4778480-4782151      - |     15702   g14059.t1
##   [2667]      YSR 4817799-4819944      - |     15705   g14064.t2
##   [2668]      YSR 4817799-4820671      - |     15706   g14064.t1
##   -------
##   seqinfo: 68 sequences from Bar2_p4 genome</code></pre>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Show examples</span>
<span class="va">orthoPairs</span><span class="op">$</span><span class="va">Oki_Osa</span><span class="op">[</span><span class="fu">flagOverlaps</span><span class="op">(</span><span class="va">orthoPairs</span><span class="op">$</span><span class="va">Oki_Osa</span><span class="op">)</span><span class="op">]</span></code></pre></div>
<pre><code>## GBreaks object with 96 ranges and 2 metadata columns:
##             seqnames            ranges strand |                 query      Arm
##                &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; |             &lt;GRanges&gt; &lt;factor&gt;
##     g499.t1     chr1   2110061-2117154      * |    Chr1:717839-724867    short
##     g500.t1     chr1   2115711-2116788      * |    Chr1:718410-719881    short
##     g544.t1     chr1   2238352-2256657      * |  Chr1:1777743-1781983    short
##     g545.t1     chr1   2251048-2252855      * |  Chr1:1783570-1784116    short
##     g985.t1     chr1   3781719-3784744      * |  Chr1:2597661-2601166    short
##         ...      ...               ...    ... .                   ...      ...
##   g15892.t1      XSR   8806041-8807773      * |   XSR:7397688-7398997      XSR
##   g16057.t1      XSR   9342752-9345613      * |   XSR:5950541-5952986      XSR
##   g16058.t1      XSR   9344384-9344779      * |   XSR:5950957-5951353      XSR
##   g16592.t1      XSR 10958372-10959713      * | XSR:10797233-10798365      XSR
##   g16593.t1      XSR 10958914-10959511      * | XSR:10797613-10797896      XSR
##   -------
##   seqinfo: 19 sequences from OKI2018_I69 genome</code></pre>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">orthoPairs</span><span class="op">$</span><span class="va">Oki_Osa</span><span class="op">$</span><span class="va">query</span><span class="op">[</span><span class="fu">flagOverlaps</span><span class="op">(</span><span class="va">orthoPairs</span><span class="op">$</span><span class="va">Oki_Osa</span><span class="op">$</span><span class="va">query</span><span class="op">)</span><span class="op">]</span></code></pre></div>
<pre><code>## GRanges object with 78 ranges and 0 metadata columns:
##             seqnames            ranges strand
##                &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt;
##     g165.t1     Chr1     717839-724867      *
##     g166.t1     Chr1     718410-719881      *
##     g993.t1     Chr1   4257010-4258454      *
##     g992.t1     Chr1   4256733-4260671      *
##     g343.t1     Chr1   1555362-1560721      *
##         ...      ...               ...    ...
##   g13270.t1      XSR   5950957-5951353      *
##   g14740.t1      XSR 10797233-10798365      *
##   g14741.t1      XSR 10797613-10797896      *
##   g13679.t1      XSR   7256274-7260299      *
##   g13680.t1      XSR   7258070-7259639      *
##   -------
##   seqinfo: 483 sequences from OSKA2016v1.9 genome</code></pre>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Remove and show how much</span>

<span class="va">orthoPairsFiltered</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="va">orthoPairs</span>, <span class="kw">function</span><span class="op">(</span><span class="va">gb</span><span class="op">)</span> <span class="op">{</span>
  <span class="va">gb</span> <span class="op">&lt;-</span> <span class="va">gb</span><span class="op">[</span><span class="op">!</span><span class="fu">flagOverlaps</span><span class="op">(</span><span class="va">gb</span><span class="op">)</span><span class="op">]</span>
  <span class="va">gb</span> <span class="op">&lt;-</span> <span class="va">gb</span><span class="op">[</span><span class="op">!</span><span class="fu">flagOverlaps</span><span class="op">(</span><span class="va">gb</span><span class="op">$</span><span class="va">query</span><span class="op">)</span><span class="op">]</span>
  <span class="va">gb</span>
<span class="op">}</span><span class="op">)</span> |&gt; <span class="fu">SimpleList</span><span class="op">(</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="va">orthoPairsFiltered</span>, <span class="va">length</span><span class="op">)</span></code></pre></div>
<pre><code>## Oki_Osa Oki_Bar Oki_Kum Oki_Aom Osa_Bar Osa_Oki Osa_Kum Osa_Aom Bar_Osa Bar_Oki 
##    9500    9269   11421    9515    9124    9500    9384   10252    9124    9268 
## Bar_Kum Bar_Aom 
##    9184    9172</code></pre>
</div>
<div id="coalesce-syntenic-blocks-and-annotate-them" class="section level2">
<h2 class="hasAnchor">
<a href="#coalesce-syntenic-blocks-and-annotate-them" class="anchor"></a>Coalesce syntenic blocks and annotate them</h2>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">coalOrtho</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">gb</span><span class="op">)</span> <span class="op">{</span>
  <span class="co"># Coalesce</span>
  <span class="va">coal</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/coalesce_contigs.html">coalesce_contigs</a></span><span class="op">(</span><span class="va">gb</span><span class="op">)</span>
  
  <span class="co"># Map annotations to the coalesced blocks</span>
  <span class="va">ov</span> <span class="op">&lt;-</span> <span class="fu">findOverlaps</span><span class="op">(</span><span class="va">coal</span>, <span class="va">gb</span><span class="op">)</span>
  
  <span class="co"># Extract gene names mapped to each blocks</span>
  <span class="va">geneNames</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/split.html">split</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">gb</span><span class="op">)</span><span class="op">[</span><span class="fu">subjectHits</span><span class="op">(</span><span class="va">ov</span><span class="op">)</span><span class="op">]</span>, <span class="fu">queryHits</span><span class="op">(</span><span class="va">ov</span><span class="op">)</span><span class="op">)</span>
  
  <span class="co"># Concatenate them as single strings and graft them to the coal object</span>
  <span class="va">coal</span><span class="op">$</span><span class="va">geneNames</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="va">geneNames</span>, <span class="va">paste</span>, collapse <span class="op">=</span> <span class="st">", "</span><span class="op">)</span>
  
  <span class="co"># Count genes per syntenic block</span>
  <span class="va">coal</span><span class="op">$</span><span class="va">geneNumber</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="va">geneNames</span>, <span class="va">length</span><span class="op">)</span>
  
  <span class="va">coal</span>
<span class="op">}</span>
<span class="va">orthoBlocks</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="va">orthoPairsFiltered</span>, <span class="va">coalOrtho</span><span class="op">)</span> |&gt; <span class="fu">SimpleList</span><span class="op">(</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="va">orthoBlocks</span>, <span class="va">length</span><span class="op">)</span></code></pre></div>
<pre><code>## Oki_Osa Oki_Bar Oki_Kum Oki_Aom Osa_Bar Osa_Oki Osa_Kum Osa_Aom Bar_Osa Bar_Oki 
##    6632    6552    2731    6756    6364    6632    6522    6241    6364    6550 
## Bar_Kum Bar_Aom 
##    6604    4847</code></pre>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">orthoBlocks</span><span class="op">[</span> <span class="fl">1</span><span class="op">:</span><span class="fl">4</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="va">orthoBlocks</span><span class="op">[</span> <span class="fl">1</span><span class="op">:</span><span class="fl">4</span><span class="op">]</span>,  <span class="va">flagLongShort</span>, <span class="va">longShort</span><span class="op">$</span><span class="va">Oki</span><span class="op">)</span>
<span class="va">orthoBlocks</span><span class="op">[</span> <span class="fl">5</span><span class="op">:</span><span class="fl">8</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="va">orthoBlocks</span><span class="op">[</span> <span class="fl">5</span><span class="op">:</span><span class="fl">8</span><span class="op">]</span>,  <span class="va">flagLongShort</span>, <span class="va">longShort</span><span class="op">$</span><span class="va">Osa</span><span class="op">)</span>
<span class="va">orthoBlocks</span><span class="op">[</span><span class="fl">9</span><span class="op">:</span><span class="fl">12</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="va">orthoBlocks</span><span class="op">[</span><span class="fl">9</span><span class="op">:</span><span class="fl">12</span><span class="op">]</span>,  <span class="va">flagLongShort</span>, <span class="va">longShort</span><span class="op">$</span><span class="va">Bar</span><span class="op">)</span></code></pre></div>
</div>
</div>
<div id="quick-overview-of-the-results" class="section level1">
<h1 class="hasAnchor">
<a href="#quick-overview-of-the-results" class="anchor"></a>Quick overview of the results</h1>
<div id="size-distribution-of-the-blocks" class="section level2">
<h2 class="hasAnchor">
<a href="#size-distribution-of-the-blocks" class="anchor"></a>Size distribution of the blocks</h2>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/graphics/stem.html">stem</a></span><span class="op">(</span><span class="va">orthoBlocks</span><span class="op">$</span><span class="va">Oki_Osa</span><span class="op">$</span><span class="va">geneNumber</span><span class="op">)</span></code></pre></div>
<pre><code>## 
##   The decimal point is at the |
## 
##    0 | 00000000000000000000000000000000000000000000000000000000000000000000+5828
##    2 | 00000000000000000000000000000000000000000000000000000000000000000000+307
##    4 | 00000000000000000000000000000000000000000000000000000000000000000000+71
##    6 | 0000000000000000000000000000000000000000000000000000000000000
##    8 | 0000000000000000000000000000000000000
##   10 | 0000000000000000000000000
##   12 | 000000000000000000
##   14 | 0000000000000
##   16 | 000000000
##   18 | 00000000
##   20 | 0000
##   22 | 00
##   24 | 000
##   26 | 000
##   28 | 
##   30 | 0
##   32 | 0
##   34 | 
##   36 | 0</code></pre>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span><span class="op">(</span><span class="va">orthoBlocks</span><span class="op">$</span><span class="va">Oki_Osa</span><span class="op">$</span><span class="va">geneNumber</span><span class="op">)</span></code></pre></div>
<pre><code>## 
##    1    2    3    4    5    6    7    8    9   10   11   12   13   14   15   16 
## 5908  280  107   88   63   42   19   22   15   14   11    8   10   11    2    6 
##   17   18   19   20   21   22   23   24   25   26   27   31   33   36 
##    3    4    4    1    3    1    1    1    2    1    2    1    1    1</code></pre>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">orthoBlocks</span><span class="op">$</span><span class="va">Oki_Osa</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/order.html">order</a></span><span class="op">(</span><span class="va">orthoBlocks</span><span class="op">$</span><span class="va">Oki_Osa</span><span class="op">$</span><span class="va">geneNumber</span><span class="op">)</span><span class="op">]</span> |&gt; <span class="fu"><a href="https://rdrr.io/r/utils/head.html">tail</a></span><span class="op">(</span><span class="fl">11</span><span class="op">)</span></code></pre></div>
<pre><code>## GBreaks object with 11 ranges and 5 metadata columns:
##        seqnames            ranges strand |                query     score
##           &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; |            &lt;GRanges&gt; &lt;integer&gt;
##    [1]      XSR   4528928-4645908      * |  XSR:2954670-3018425    116981
##    [2]     chr2   6118697-6208680      * | Chr2:8974962-9049921     89984
##    [3]      PAR 10906073-10997442      * |  PAR:9275473-9353548     91370
##    [4]     chr2   5838991-5959986      * | Chr2:5470156-5575948    120996
##    [5]      PAR 16599874-16699948      * |  PAR:7442672-7520546    100075
##    [6]      XSR   5784689-5900834      * |  XSR:2547537-2661225    116146
##    [7]      XSR   3436117-3526092      * |  XSR:2172085-2264761     89976
##    [8]      XSR   7246798-7345874      * |  XSR:4104726-4194323     99077
##    [9]      XSR   7562507-7713904      * |  XSR:3625761-3772339    151398
##   [10]     chr2 14727816-14844427      * | Chr2:8441071-8537126    116612
##   [11]     chr2 10668710-10828262      * | Chr2:5679751-5827304    159553
##                     geneNames geneNumber      Arm
##                   &lt;character&gt;  &lt;integer&gt; &lt;factor&gt;
##    [1] g14568.t1, g14569.t1..         22     XSR 
##    [2] g5638.t1, g5640.t1, ..         23     long
##    [3] g11569.t1, g11570.t1..         24     long
##    [4] g5558.t1, g5559.t1, ..         25     long
##    [5] g13147.t1, g13152.t1..         25     long
##    [6] g14935.t1, g14939.t1..         26     XSR 
##    [7] g14231.t1, g14232.t1..         27     XSR 
##    [8] g15407.t1, g15408.t1..         27     XSR 
##    [9] g15509.t1, g15510.t1..         31     XSR 
##   [10] g8160.t1, g8161.t1, ..         33     long
##   [11] g6993.t1, g6996.t1, ..         36     long
##   -------
##   seqinfo: 19 sequences from OKI2018_I69 genome</code></pre>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/graphics/stem.html">stem</a></span><span class="op">(</span><span class="va">orthoBlocks</span><span class="op">$</span><span class="va">Oki_Kum</span><span class="op">$</span><span class="va">geneNumber</span><span class="op">)</span></code></pre></div>
<pre><code>## 
##   The decimal point is 2 digit(s) to the right of the |
## 
##   0 | 00000000000000000000000000000000000000000000000000000000000000000000+2598
##   0 | 55555555566666677778889999999
##   1 | 01111244
##   1 | 556689
##   2 | 14
##   2 | 577
##   3 | 22
##   3 | 7
##   4 | 
##   4 | 
##   5 | 
##   5 | 
##   6 | 
##   6 | 
##   7 | 2
##   7 | 
##   8 | 3</code></pre>
<div class="sourceCode" id="cb38"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span><span class="op">(</span><span class="va">orthoBlocks</span><span class="op">$</span><span class="va">Oki_Kum</span><span class="op">$</span><span class="va">geneNumber</span><span class="op">)</span></code></pre></div>
<pre><code>## 
##    1    2    3    4    5    6    7    8   10   11   12   13   14   15   17   18 
## 2598    6    2   10    3    2    5    7    2    3    1    2    3    3    4    1 
##   19   20   22   24   26   27   28   29   30   31   34   36   37   41   44   47 
##    2    4    3    2    3    1    1    1    1    1    1    1    1    2    2    2 
##   48   49   50   51   53   54   56   58   59   61   64   66   67   71   74   75 
##    1    1    1    1    2    1    1    2    1    1    1    1    1    1    1    1 
##   77   78   85   89   92   93   94  102  107  108  110  112  115  135  138  145 
##    1    1    2    1    1    1    2    1    1    1    1    1    1    1    1    1 
##  154  155  164  179  190  211  244  246  266  268  318  323  365  717  831 
##    1    1    1    1    1    1    1    1    1    1    1    1    1    1    1</code></pre>
<div class="sourceCode" id="cb40"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">orthoBlocks</span><span class="op">$</span><span class="va">Oki_Kum</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/order.html">order</a></span><span class="op">(</span><span class="va">orthoBlocks</span><span class="op">$</span><span class="va">Oki_Kum</span><span class="op">$</span><span class="va">geneNumber</span><span class="op">)</span><span class="op">]</span> |&gt; <span class="fu"><a href="https://rdrr.io/r/utils/head.html">tail</a></span><span class="op">(</span><span class="fl">11</span><span class="op">)</span></code></pre></div>
<pre><code>## GBreaks object with 11 ranges and 5 metadata columns:
##        seqnames            ranges strand |                        query
##           &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; |                    &lt;GRanges&gt;
##    [1]     chr2   8889881-9716002      * |  contig_24_1:3004262-3811617
##    [2]     chr1 11819952-12852434      * |   contig_3_1:3591774-4660565
##    [3]     chr1 10216425-11360460      * |   contig_3_1:2089417-3101980
##    [4]     chr2   2258733-3555695      * |  contig_16_1:1842424-3134414
##    [5]     chr1   8457311-9749351      * |    contig_3_1:395673-1641942
##    [6]     chr2   5748389-6937702      * |    contig_24_1:12160-1155646
##    [7]      PAR   7792256-9333620      * |   contig_82_1:846722-2300436
##    [8]      XSR   5481685-6642828      * |  contig_42_1:5471798-6631430
##    [9]      PAR 11974020-13925498      * |  contig_28_1:2026940-3782149
##   [10]      XSR   1398297-4576212      * |  contig_42_1:1522014-4661476
##   [11]      XSR  7904454-11123841      * | contig_42_1:7880616-11063801
##            score              geneNames geneNumber      Arm
##        &lt;integer&gt;            &lt;character&gt;  &lt;integer&gt; &lt;factor&gt;
##    [1]    826122 g6461.t1, g6462.t1, ..        190    long 
##    [2]   1032483 g3221.t1, g3222.t1, ..        211    long 
##    [3]   1144036 g2725.t1, g2726.t1, ..        244    long 
##    [4]   1296963 g4487.t1, g4488.t1, ..        246    short
##    [5]   1292041 g2255.t1, g2256.t1, ..        266    long 
##    [6]   1189314 g5514.t1, g5515.t1, ..        268    long 
##    [7]   1541365 g10595.t1, g10596.t1..        318    long 
##    [8]   1161144 g14846.t1, g14847.t1..        323    XSR  
##    [9]   1951479 g11881.t1, g11883.t1..        365    long 
##   [10]   3177916 g13662.t1, g13663.t1..        717    XSR  
##   [11]   3219388 g15616.t1, g15617.t1..        831    XSR  
##   -------
##   seqinfo: 19 sequences from OKI2018_I69 genome</code></pre>
</div>
<div id="gene-order-conservation-is-smaller-in-short-arms" class="section level2">
<h2 class="hasAnchor">
<a href="#gene-order-conservation-is-smaller-in-short-arms" class="anchor"></a>Gene Order Conservation is smaller in short arms</h2>
<div class="sourceCode" id="cb42"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">orthoBlocks_LSnoNA</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="va">orthoBlocks</span>, <span class="kw">function</span><span class="op">(</span><span class="va">gb</span><span class="op">)</span> <span class="va">gb</span><span class="op">[</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span><span class="op">(</span><span class="va">gb</span><span class="op">$</span><span class="va">Arm</span><span class="op">)</span><span class="op">]</span><span class="op">)</span> |&gt; <span class="fu">SimpleList</span><span class="op">(</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="va">orthoBlocks_LSnoNA</span>, <span class="va">length</span><span class="op">)</span></code></pre></div>
<pre><code>## Oki_Osa Oki_Bar Oki_Kum Oki_Aom Osa_Bar Osa_Oki Osa_Kum Osa_Aom Bar_Osa Bar_Oki 
##    6626    6548    2728    6754    6327    6575    6467    6191    6332    6513 
## Bar_Kum Bar_Aom 
##    6563    4817</code></pre>
<div class="sourceCode" id="cb44"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">df</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="va">orthoBlocks_LSnoNA</span>,
             <span class="kw">function</span><span class="op">(</span><span class="va">gb</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/base/tapply.html">tapply</a></span><span class="op">(</span><span class="va">gb</span>, <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span><span class="op">(</span><span class="fu">seqnames</span><span class="op">(</span><span class="va">gb</span><span class="op">)</span>, <span class="va">gb</span><span class="op">$</span><span class="va">Arm</span><span class="op">)</span>, <span class="va">GOC</span><span class="op">)</span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/r/base/apply.html">apply</a></span><span class="op">(</span><span class="va">df</span>, <span class="fl">2</span>, \<span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="op">{</span>
  <span class="va">tst</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/t.test.html">t.test</a></span><span class="op">(</span><span class="va">x</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">3</span>,<span class="fl">5</span><span class="op">)</span><span class="op">]</span>, <span class="va">x</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">2</span>,<span class="fl">4</span>,<span class="fl">6</span><span class="op">)</span><span class="op">]</span><span class="op">)</span>
  <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span>pval<span class="op">=</span><span class="va">tst</span><span class="op">$</span><span class="va">p.value</span>, mx<span class="op">=</span><span class="va">tst</span><span class="op">$</span><span class="va">estimate</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>, my<span class="op">=</span><span class="va">tst</span><span class="op">$</span><span class="va">estimate</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span><span class="op">)</span>
<span class="op">}</span><span class="op">)</span> </code></pre></div>
<pre><code>##                   Oki_Osa     Oki_Bar   Oki_Kum     Oki_Aom    Osa_Bar
## pval         0.0008685401 0.005961179 0.3538487 0.001833285 0.05533798
## mx.mean of x 0.6287411452 0.632541475 0.9470086 0.646646734 0.90172809
## my.mean of y 0.2264427685 0.220715694 0.8521385 0.230918802 0.58727154
##                  Osa_Oki    Osa_Kum    Osa_Aom    Bar_Osa     Bar_Oki
## pval         0.001451611 0.00103602 0.07480647 0.04515765 0.004094705
## mx.mean of x 0.632640617 0.62661540 0.96423909 0.89762371 0.633444544
## my.mean of y 0.236163526 0.22695506 0.91595262 0.58585543 0.231771014
##                 Bar_Kum   Bar_Aom
## pval         0.00958392 0.0289898
## mx.mean of x 0.64301879 0.8854833
## my.mean of y 0.23854103 0.5791719</code></pre>
</div>
<div id="agreement-with-nucleotide-alignments" class="section level2">
<h2 class="hasAnchor">
<a href="#agreement-with-nucleotide-alignments" class="anchor"></a>Agreement with nucleotide alignments</h2>
<div class="sourceCode" id="cb46"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="op">(</span><span class="va">gb_P</span> <span class="op">&lt;-</span> <span class="va">orthoBlocks</span><span class="op">$</span><span class="va">Bar_Osa</span><span class="op">)</span></code></pre></div>
<pre><code>## GBreaks object with 6364 ranges and 5 metadata columns:
##          seqnames          ranges strand |                query     score
##             &lt;Rle&gt;       &lt;IRanges&gt;  &lt;Rle&gt; |            &lt;GRanges&gt; &lt;integer&gt;
##      [1]     Chr1     17796-19101      * | Chr1:3589156-3591380      1306
##      [2]     Chr1     19387-21383      * | Chr1:3581251-3588847      1997
##      [3]     Chr1     27418-60748      * | Chr1:3596010-3622624     33331
##      [4]     Chr1     69763-70357      * | Chr1:3649897-3650503       595
##      [5]     Chr1     71501-72381      * | Chr1:3648112-3648637       881
##      ...      ...             ...    ... .                  ...       ...
##   [6360]      YSR 3599919-3601165      * | Chr1:4173793-4175304      1247
##   [6361]      YSR 3631705-3633609      * | Chr1:4315390-4317451      1905
##   [6362]      YSR 4033286-4033798      * |  YSR:1457970-1458977       513
##   [6363]      YSR 4086487-4086957      * | Chr1:8499823-8501898       471
##   [6364]      YSR 4865878-4866414      * |    YSR:434383-434709       537
##                       geneNames geneNumber      Arm
##                     &lt;character&gt;  &lt;integer&gt; &lt;factor&gt;
##      [1]                  g1.t1          1    short
##      [2]                  g2.t1          1    short
##      [3] g4.t1, g5.t1, g6.t1,..          8    short
##      [4]                 g16.t1          1    short
##      [5]                 g17.t1          1    short
##      ...                    ...        ...      ...
##   [6360]              g13906.t1          1      YSR
##   [6361]              g13910.t1          1      YSR
##   [6362]              g13965.t1          1      YSR
##   [6363]              g13972.t1          1      YSR
##   [6364]              g14068.t1          1      YSR
##   -------
##   seqinfo: 68 sequences from Bar2_p4 genome</code></pre>
<div class="sourceCode" id="cb48"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="fu">width</span><span class="op">(</span><span class="va">gb_P</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] 28694224</code></pre>
<div class="sourceCode" id="cb50"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="op">(</span><span class="va">gb_N</span> <span class="op">&lt;-</span> <span class="va">coa</span><span class="op">$</span><span class="va">Bar_Osa</span><span class="op">)</span></code></pre></div>
<pre><code>## GBreaks object with 4073 ranges and 6 metadata columns:
##          seqnames          ranges strand |                query     score
##             &lt;Rle&gt;       &lt;IRanges&gt;  &lt;Rle&gt; |            &lt;GRanges&gt; &lt;integer&gt;
##      [1]     Chr1     15573-21624      - | Chr1:3585703-3594761      6052
##      [2]     Chr1     22042-22295      + | Chr1:3585140-3585392       254
##      [3]     Chr1     22421-24048      - | Chr1:3583028-3584936      1628
##      [4]     Chr1     27262-34408      + | Chr1:3595848-3601817      7147
##      [5]     Chr1     34435-34998      - | Chr1:3602092-3602509       564
##      ...      ...             ...    ... .                  ...       ...
##   [4069]      YSR 5273802-5274176      - |  YSR:1632176-1632556       375
##   [4070]      YSR 5288325-5289549      - |  YSR:1086332-1087481      1225
##   [4071]      YSR 5307497-5310556      - | Chr1:9561929-9564982      3060
##   [4072]      YSR 5322809-5325938      - |  PAR:5433966-5437098      3130
##   [4073]      YSR 5335965-5336405      - |   Chr2:928224-928659       441
##               Arm             rep  genes        flag
##          &lt;factor&gt; &lt;CharacterList&gt;  &lt;Rle&gt; &lt;character&gt;
##      [1]    short             rnd     g2         Inv
##      [2]    short            &lt;NA&gt;   &lt;NA&gt;        &lt;NA&gt;
##      [3]    short            &lt;NA&gt;   &lt;NA&gt;        &lt;NA&gt;
##      [4]    short             rnd   &lt;NA&gt;         Inv
##      [5]    short            &lt;NA&gt;     g5        &lt;NA&gt;
##      ...      ...             ...    ...         ...
##   [4069]      YSR            &lt;NA&gt;   &lt;NA&gt;        &lt;NA&gt;
##   [4070]      YSR            &lt;NA&gt; g14124        &lt;NA&gt;
##   [4071]      YSR            &lt;NA&gt; g14124        &lt;NA&gt;
##   [4072]      YSR            &lt;NA&gt;   &lt;NA&gt;        &lt;NA&gt;
##   [4073]      YSR             rnd   &lt;NA&gt;        &lt;NA&gt;
##   -------
##   seqinfo: 68 sequences from Bar2_p4 genome</code></pre>
<div class="sourceCode" id="cb52"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="fu">width</span><span class="op">(</span><span class="va">gb_N</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] 45289525</code></pre>
<div class="sourceCode" id="cb54"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="op">(</span><span class="va">ov</span> <span class="op">&lt;-</span> <span class="fu">subsetByOverlaps</span><span class="op">(</span><span class="va">gb_P</span>, <span class="va">gb_N</span>, type <span class="op">=</span> <span class="st">"within"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code>## GBreaks object with 5944 ranges and 5 metadata columns:
##          seqnames          ranges strand |                query     score
##             &lt;Rle&gt;       &lt;IRanges&gt;  &lt;Rle&gt; |            &lt;GRanges&gt; &lt;integer&gt;
##      [1]     Chr1     17796-19101      * | Chr1:3589156-3591380      1306
##      [2]     Chr1     19387-21383      * | Chr1:3581251-3588847      1997
##      [3]     Chr1     69763-70357      * | Chr1:3649897-3650503       595
##      [4]     Chr1     71501-72381      * | Chr1:3648112-3648637       881
##      [5]     Chr1     72537-73649      * | Chr1:3645566-3647968      1113
##      ...      ...             ...    ... .                  ...       ...
##   [5940]      YSR 3597539-3598992      * |   Chr1:269069-271045      1454
##   [5941]      YSR 3599919-3601165      * | Chr1:4173793-4175304      1247
##   [5942]      YSR 3631705-3633609      * | Chr1:4315390-4317451      1905
##   [5943]      YSR 4086487-4086957      * | Chr1:8499823-8501898       471
##   [5944]      YSR 4865878-4866414      * |    YSR:434383-434709       537
##            geneNames geneNumber      Arm
##          &lt;character&gt;  &lt;integer&gt; &lt;factor&gt;
##      [1]       g1.t1          1    short
##      [2]       g2.t1          1    short
##      [3]      g16.t1          1    short
##      [4]      g17.t1          1    short
##      [5]      g18.t1          1    short
##      ...         ...        ...      ...
##   [5940]   g13905.t1          1      YSR
##   [5941]   g13906.t1          1      YSR
##   [5942]   g13910.t1          1      YSR
##   [5943]   g13972.t1          1      YSR
##   [5944]   g14068.t1          1      YSR
##   -------
##   seqinfo: 68 sequences from Bar2_p4 genome</code></pre>
<div class="sourceCode" id="cb56"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="op">(</span><span class="va">jaccard</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="fu">width</span><span class="op">(</span><span class="va">ov</span><span class="op">)</span><span class="op">)</span> <span class="op">/</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="fu">width</span><span class="op">(</span><span class="fu">reduce</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">gb_P</span>, <span class="va">gb_N</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] 0.2490777</code></pre>
<div class="sourceCode" id="cb58"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">gb_P</span> <span class="op">&lt;-</span> <span class="va">gb_P</span><span class="op">[</span><span class="va">gb_P</span><span class="op">$</span><span class="va">geneNumber</span> <span class="op">&gt;</span> <span class="fl">1</span><span class="op">]</span>
<span class="va">gb_P</span></code></pre></div>
<pre><code>## GBreaks object with 405 ranges and 5 metadata columns:
##         seqnames            ranges strand |                  query     score
##            &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; |              &lt;GRanges&gt; &lt;integer&gt;
##     [1]     Chr1       27418-60748      * |   Chr1:3596010-3622624     33331
##     [2]     Chr1       87558-91675      * |   Chr1:3818944-3824036      4118
##     [3]     Chr1     232572-233634      * |   Chr1:1279025-1280616      1063
##     [4]     Chr1     291713-332464      * |   Chr1:2941852-2986198     40752
##     [5]     Chr1     442856-457337      * |   Chr1:3099876-3116080     14482
##     ...      ...               ...    ... .                    ...       ...
##   [401]      XSR 12002949-12013527      * |  XSR:10324639-10335467     10579
##   [402]      XSR 12026997-12044817      * |    XSR:8178353-8191672     17821
##   [403]      XSR 12204593-12269268      * |    XSR:1089754-1157686     64676
##   [404]      YSR     518298-532357      * | Chr2:12195648-12210212     14060
##   [405]      YSR   2712461-2719315      * |    YSR:1128232-1134212      6855
##                      geneNames geneNumber      Arm
##                    &lt;character&gt;  &lt;integer&gt; &lt;factor&gt;
##     [1] g4.t1, g5.t1, g6.t1,..          8    short
##     [2]         g22.t1, g23.t1          2    short
##     [3]         g61.t1, g62.t1          2    short
##     [4] g72.t1, g73.t1, g74...          5    short
##     [5]         g97.t1, g99.t1          2    short
##     ...                    ...        ...      ...
##   [401] g13227.t1, g13228.t1..          5      XSR
##   [402] g13237.t1, g13238.t1..          4      XSR
##   [403] g13272.t1, g13273.t2..         14      XSR
##   [404]   g13390.t1, g13392.t1          2      YSR
##   [405]   g13745.t1, g13747.t1          2      YSR
##   -------
##   seqinfo: 68 sequences from Bar2_p4 genome</code></pre>
<div class="sourceCode" id="cb60"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">gb_P</span><span class="op">$</span><span class="va">geneNumber</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] 3165</code></pre>
<div class="sourceCode" id="cb62"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="op">(</span><span class="va">valid</span> <span class="op">&lt;-</span> <span class="fu">subsetByOverlaps</span><span class="op">(</span><span class="va">gb_P</span>, <span class="va">gb_N</span>, type <span class="op">=</span> <span class="st">"within"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code>## GBreaks object with 239 ranges and 5 metadata columns:
##         seqnames            ranges strand |                 query     score
##            &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; |             &lt;GRanges&gt; &lt;integer&gt;
##     [1]     Chr1       87558-91675      * |  Chr1:3818944-3824036      4118
##     [2]     Chr1     442856-457337      * |  Chr1:3099876-3116080     14482
##     [3]     Chr1     881296-888558      * |  Chr1:2110277-2120600      7263
##     [4]     Chr1     905933-925391      * |  Chr1:4012642-4036759     19459
##     [5]     Chr1   1032978-1044489      * |  Chr1:3879847-3895767     11512
##     ...      ...               ...    ... .                   ...       ...
##   [235]      XSR 11491782-11499863      * |     XSR:761522-772346      8082
##   [236]      XSR 11711777-11734215      * | XSR:11045356-11071532     22439
##   [237]      XSR 11811894-11864706      * | XSR:11109942-11160056     52813
##   [238]      XSR 12002949-12013527      * | XSR:10324639-10335467     10579
##   [239]      XSR 12026997-12044817      * |   XSR:8178353-8191672     17821
##                      geneNames geneNumber      Arm
##                    &lt;character&gt;  &lt;integer&gt; &lt;factor&gt;
##     [1]         g22.t1, g23.t1          2    short
##     [2]         g97.t1, g99.t1          2    short
##     [3]       g178.t1, g179.t1          2    short
##     [4] g183.t1, g185.t1, g1..          8    short
##     [5] g214.t1, g215.t2, g2..          3    short
##     ...                    ...        ...      ...
##   [235]   g13111.t1, g13112.t1          2      XSR
##   [236] g13159.t1, g13162.t1..          5      XSR
##   [237] g13189.t1, g13190.t1..          8      XSR
##   [238] g13227.t1, g13228.t1..          5      XSR
##   [239] g13237.t1, g13238.t1..          4      XSR
##   -------
##   seqinfo: 68 sequences from Bar2_p4 genome</code></pre>
<div class="sourceCode" id="cb64"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">valid</span><span class="op">$</span><span class="va">geneNumber</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] 1229</code></pre>
<div class="sourceCode" id="cb66"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="fu">width</span><span class="op">(</span><span class="va">valid</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] 4777278</code></pre>
<p>Question: what does interrupt the colinearity of nucleotide regions in gene-centric colinear regions ?</p>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Charles Plessy.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
