<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Whole-chromosome parallel plots • OikScrambling</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Whole-chromosome parallel plots">
<meta property="og:description" content="OikScrambling">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">OikScrambling</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.6.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/OikScrambling.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/CoalescingCutoffs.html">Modifying coalescent cutoffs</a>
    </li>
    <li>
      <a href="../articles/ColinearityInterruptors.html">Regions interrupting colinearity</a>
    </li>
    <li>
      <a href="../articles/DistanceMeasures.html">Distance Measures</a>
    </li>
    <li>
      <a href="../articles/GenomicFeatures.html">Introduction to the OikScrambling package</a>
    </li>
    <li>
      <a href="../articles/Inversions.html">Inversions</a>
    </li>
    <li>
      <a href="../articles/LoadGenomicBreaks.html">Load genomic break data</a>
    </li>
    <li>
      <a href="../articles/OrthogroupSyntenies.html">Orthogroup syntenies</a>
    </li>
    <li>
      <a href="../articles/OxfordPlots.html">Whole-genome dot-plots</a>
    </li>
    <li>
      <a href="../articles/PWM.html">Motif finding and analysis</a>
    </li>
    <li>
      <a href="../articles/ParallelPlots.html">Whole-chromosome parallel plots</a>
    </li>
    <li>
      <a href="../articles/RegionWidths.html">Width of the aligned and mapped regions.</a>
    </li>
    <li>
      <a href="../articles/RepeatRegions.html">Repeat regions</a>
    </li>
    <li>
      <a href="../articles/Translocations.html">Translocations</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/oist/LuscombeU_OikScrambling/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="ParallelPlots_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Whole-chromosome parallel plots</h1>
                        <h4 data-toc-skip class="author">Charles Plessy</h4>
            
            <h4 data-toc-skip class="date">13 December, 2021</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/oist/LuscombeU_OikScrambling/blob/HEAD/vignettes/ParallelPlots.Rmd" class="external-link"><code>vignettes/ParallelPlots.Rmd</code></a></small>
      <div class="hidden name"><code>ParallelPlots.Rmd</code></div>

    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">knitr</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/knitr/man/opts_chunk.html" class="external-link">opts_chunk</a></span><span class="op">$</span><span class="fu">set</span><span class="op">(</span>cache <span class="op">=</span> <span class="cn">TRUE</span>, cache.lazy <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
<span class="fu">knitr</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/knitr/man/opts_knit.html" class="external-link">opts_knit</a></span><span class="op">$</span><span class="fu">set</span><span class="op">(</span>verbose <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<div class="section level1">
<h1 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h1>
<p>In this vignette I plot whole chromosomes in parallel to reflect synteny and the difference of scrambling between short and long arms.</p>
<p>The core functions used here are maintained in our <em>GenomicBreaks</em> R package, which is fully documented at: <a href="https://oist.github.io/GenomicBreaks" class="external-link uri">https://oist.github.io/GenomicBreaks</a>.</p>
<div class="section level2">
<h2 id="load-pacakges">Load pacakges<a class="anchor" aria-label="anchor" href="#load-pacakges"></a>
</h2>
</div>
</div>
<div class="section level1">
<h1 id="load-r-pacakges-and-data">Load R pacakges and data<a class="anchor" aria-label="anchor" href="#load-r-pacakges-and-data"></a>
</h1>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/oist/LuscombeU_OikScrambling" class="external-link">'OikScrambling'</a></span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="op">)</span>
<span class="op">(</span><span class="va">genomes</span> <span class="op">&lt;-</span> <span class="fu">OikScrambling</span><span class="fu">:::</span><span class="fu"><a href="../reference/loadAllGenomes.html">loadAllGenomes</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## List of length 6</span>
<span class="co">## names(6): Oki Osa Bar Kum Aom Nor</span></code></pre>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/load.html" class="external-link">load</a></span><span class="op">(</span><span class="st">"BreakPoints.Rdata"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">##  [1] "gbs"       "unal"      "coa"       "unmap"     "unalMap"   "tra"      </span>
<span class="co">##  [7] "tra2"      "coa2"      "unmap2"    "longShort"</span></code></pre>
<div class="section level3">
<h3 id="phylogenetic-cladogram">Phylogenetic cladogram<a class="anchor" aria-label="anchor" href="#phylogenetic-cladogram"></a>
</h3>
<p>We assume that the <em>North Pacific</em> and the <em>Atlantic</em> species are more related to each other than to the <em>Okinawan</em> species.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"ade4"</span><span class="op">)</span>
<span class="va">treeLeaf</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">name</span>, <span class="va">length</span><span class="op">=</span><span class="cn">NULL</span><span class="op">)</span> <span class="op">{</span>
  <span class="kw">if</span><span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NULL.html" class="external-link">is.null</a></span><span class="op">(</span><span class="va">length</span><span class="op">)</span><span class="op">)</span> <span class="va">length</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">':'</span>, <span class="va">length</span><span class="op">)</span>
  <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">name</span>, <span class="va">length</span><span class="op">)</span>
<span class="op">}</span>

<span class="va">treeNode</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">branch1</span>, <span class="va">branch2</span>, <span class="va">length</span> <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span> <span class="op">{</span>
  <span class="kw">if</span><span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NULL.html" class="external-link">is.null</a></span><span class="op">(</span><span class="va">length</span><span class="op">)</span><span class="op">)</span> <span class="va">length</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">':'</span>, <span class="va">length</span><span class="op">)</span>
  <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">'('</span>, <span class="va">branch1</span>, <span class="st">','</span>, <span class="va">branch2</span>, <span class="st">')'</span>, <span class="va">length</span><span class="op">)</span>
<span class="op">}</span>

<span class="va">addRoot</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">branch</span><span class="op">)</span>  <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">branch</span>, <span class="st">";"</span><span class="op">)</span>

<span class="va">tree</span> <span class="op">&lt;-</span>
  <span class="fu">addRoot</span><span class="op">(</span>
    <span class="fu">treeNode</span><span class="op">(</span>
      <span class="fu">treeNode</span><span class="op">(</span> length <span class="op">=</span> <span class="fl">2</span>,
        <span class="fu">treeLeaf</span><span class="op">(</span><span class="st">"Okinawa"</span>, <span class="fl">1</span><span class="op">)</span>,
        <span class="fu">treeLeaf</span><span class="op">(</span><span class="st">"Kume"</span>, <span class="fl">1</span><span class="op">)</span>
      <span class="op">)</span>,
      <span class="fu">treeNode</span><span class="op">(</span> length <span class="op">=</span> <span class="fl">1</span>,
        <span class="fu">treeNode</span><span class="op">(</span> length <span class="op">=</span> <span class="fl">1</span>,
          <span class="fu">treeLeaf</span><span class="op">(</span><span class="st">"Osaka"</span>, <span class="fl">1</span><span class="op">)</span>,
          <span class="fu">treeLeaf</span><span class="op">(</span><span class="st">"Aomori"</span>, <span class="fl">1</span><span class="op">)</span>
        <span class="op">)</span>,
        <span class="fu">treeNode</span><span class="op">(</span> length <span class="op">=</span><span class="fl">1</span>,
          <span class="fu">treeLeaf</span><span class="op">(</span><span class="st">"Norway"</span>, <span class="fl">1</span><span class="op">)</span>,
          <span class="fu">treeLeaf</span><span class="op">(</span><span class="st">"Barcelona"</span>, <span class="fl">1</span><span class="op">)</span>
        <span class="op">)</span>
      <span class="op">)</span>
    <span class="op">)</span>
  <span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="fu">ade4</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ade4/man/newick2phylog.html" class="external-link">newick2phylog</a></span><span class="op">(</span><span class="va">tree</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="ParallelPlots_files/figure-html/phylogenetic_tree-1.svg" width="576"></p>
</div>
<div class="section level2">
<h2 id="genome-plots-with-genoplotr">Genome plots with genoPlotR<a class="anchor" aria-label="anchor" href="#genome-plots-with-genoplotr"></a>
</h2>
<div class="section level3">
<h3 id="pairs-of-chromosome-1s-">Pairs of chromosome 1s.<a class="anchor" aria-label="anchor" href="#pairs-of-chromosome-1s-"></a>
</h3>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/plotApairOfChrs.html" class="external-link">plotApairOfChrs</a></span><span class="op">(</span><span class="va">coa</span><span class="op">$</span><span class="va">Oki_Bar</span>, <span class="st">"chr1"</span>, main <span class="op">=</span> <span class="st">"Oki vs Bar"</span><span class="op">)</span></code></pre></div>
<p><img src="ParallelPlots_files/figure-html/Oki_Bar_chr1-1.png" width="1440"></p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/plotApairOfChrs.html" class="external-link">plotApairOfChrs</a></span><span class="op">(</span><span class="va">coa</span><span class="op">$</span><span class="va">Oki_Osa</span>, <span class="st">"chr1"</span>, main <span class="op">=</span> <span class="st">"Oki vs Osa"</span><span class="op">)</span></code></pre></div>
<p><img src="ParallelPlots_files/figure-html/Oki_Osa_chr1-1.png" width="1440"></p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/plotApairOfChrs.html" class="external-link">plotApairOfChrs</a></span><span class="op">(</span><span class="va">coa</span><span class="op">$</span><span class="va">Osa_Oki</span>, <span class="st">"Chr1"</span>, main <span class="op">=</span> <span class="st">"Osa vs Oki"</span><span class="op">)</span></code></pre></div>
<p><img src="ParallelPlots_files/figure-html/Osa_Oki_chr1-1.png" width="1440"></p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/plotApairOfChrs.html" class="external-link">plotApairOfChrs</a></span><span class="op">(</span><span class="va">coa</span><span class="op">$</span><span class="va">Osa_Bar</span>, <span class="st">"Chr1"</span>, main <span class="op">=</span> <span class="st">"Osa vs Bar"</span><span class="op">)</span></code></pre></div>
<p><img src="ParallelPlots_files/figure-html/Osa_Bar_chr1-1.png" width="1440"></p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/plotApairOfChrs.html" class="external-link">plotApairOfChrs</a></span><span class="op">(</span><span class="va">coa2</span><span class="op">$</span><span class="va">Osa_Bar</span>, <span class="st">"Chr1"</span>, main <span class="op">=</span> <span class="st">"Osa vs Bar (double-collapsed"</span><span class="op">)</span></code></pre></div>
<p><img src="ParallelPlots_files/figure-html/Oki_Bar_chr1_coa2-1.png" width="1440"></p>
<p>Kume’s <code>chr1</code> is mostly comprised of 4 large contigs.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">x</span> <span class="op">&lt;-</span> <span class="va">coa</span><span class="op">$</span><span class="va">Oki_Kum</span>
<span class="va">x</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqnames</a></span><span class="op">(</span><span class="va">x</span><span class="op">$</span><span class="va">query</span><span class="op">)</span> <span class="op">==</span> <span class="st">"contig_90_1"</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/reverse.html" class="external-link">reverse</a></span><span class="op">(</span>query <span class="op">=</span> <span class="cn">TRUE</span>, <span class="va">x</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqnames</a></span><span class="op">(</span><span class="va">x</span><span class="op">$</span><span class="va">query</span><span class="op">)</span> <span class="op">==</span> <span class="st">"contig_90_1"</span><span class="op">]</span><span class="op">)</span>
<span class="va">x</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqnames</a></span><span class="op">(</span><span class="va">x</span><span class="op">$</span><span class="va">query</span><span class="op">)</span> <span class="op">==</span> <span class="st">"contig_88_1"</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/reverse.html" class="external-link">reverse</a></span><span class="op">(</span>query <span class="op">=</span> <span class="cn">TRUE</span>, <span class="va">x</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqnames</a></span><span class="op">(</span><span class="va">x</span><span class="op">$</span><span class="va">query</span><span class="op">)</span> <span class="op">==</span> <span class="st">"contig_88_1"</span><span class="op">]</span><span class="op">)</span>

<span class="va">x</span><span class="op">$</span><span class="va">query</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/mergeSeqLevels.html" class="external-link">mergeSeqLevels</a></span><span class="op">(</span><span class="va">x</span><span class="op">$</span><span class="va">query</span>, <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"contig_27_1"</span>, <span class="st">"contig_90_1"</span>, <span class="st">"contig_3_1"</span>, <span class="st">"contig_88_1"</span><span class="op">)</span>, <span class="st">"new_scaffold"</span><span class="op">)</span>
<span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/plotApairOfChrs.html" class="external-link">plotApairOfChrs</a></span><span class="op">(</span><span class="va">x</span>, <span class="st">"chr1"</span>, dna_seg_scale<span class="op">=</span><span class="cn">TRUE</span>, main <span class="op">=</span> <span class="st">"Oki vs Kum"</span><span class="op">)</span></code></pre></div>
<p><img src="ParallelPlots_files/figure-html/coa_based-1.png" width="1440"></p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">x</span> <span class="op">&lt;-</span> <span class="va">coa2</span><span class="op">$</span><span class="va">Oki_Kum</span>
<span class="va">x</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqnames</a></span><span class="op">(</span><span class="va">x</span><span class="op">$</span><span class="va">query</span><span class="op">)</span> <span class="op">==</span> <span class="st">"contig_90_1"</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/reverse.html" class="external-link">reverse</a></span><span class="op">(</span>query <span class="op">=</span> <span class="cn">TRUE</span>, <span class="va">x</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqnames</a></span><span class="op">(</span><span class="va">x</span><span class="op">$</span><span class="va">query</span><span class="op">)</span> <span class="op">==</span> <span class="st">"contig_90_1"</span><span class="op">]</span><span class="op">)</span>
<span class="va">x</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqnames</a></span><span class="op">(</span><span class="va">x</span><span class="op">$</span><span class="va">query</span><span class="op">)</span> <span class="op">==</span> <span class="st">"contig_88_1"</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/reverse.html" class="external-link">reverse</a></span><span class="op">(</span>query <span class="op">=</span> <span class="cn">TRUE</span>, <span class="va">x</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqnames</a></span><span class="op">(</span><span class="va">x</span><span class="op">$</span><span class="va">query</span><span class="op">)</span> <span class="op">==</span> <span class="st">"contig_88_1"</span><span class="op">]</span><span class="op">)</span>

<span class="va">x</span><span class="op">$</span><span class="va">query</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/mergeSeqLevels.html" class="external-link">mergeSeqLevels</a></span><span class="op">(</span><span class="va">x</span><span class="op">$</span><span class="va">query</span>, <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"contig_27_1"</span>, <span class="st">"contig_90_1"</span>, <span class="st">"contig_3_1"</span>, <span class="st">"contig_88_1"</span><span class="op">)</span>, <span class="st">"new_scaffold"</span><span class="op">)</span>
<span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/plotApairOfChrs.html" class="external-link">plotApairOfChrs</a></span><span class="op">(</span><span class="va">x</span>, <span class="st">"chr1"</span>, dna_seg_scale<span class="op">=</span><span class="cn">TRUE</span>, main <span class="op">=</span> <span class="st">"Oki vs Kum (double-collapsed"</span><span class="op">)</span></code></pre></div>
<p><img src="ParallelPlots_files/figure-html/coa2_based2-1.png" width="1440"></p>
</div>
</div>
<div class="section level2">
<h2 id="plot-chr1-in-three-species">Plot chr1 in three species<a class="anchor" aria-label="anchor" href="#plot-chr1-in-three-species"></a>
</h2>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">doubleSubset</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">gb</span>, <span class="va">a</span>, <span class="va">b</span><span class="op">)</span> <span class="op">{</span>
  <span class="va">gb</span> <span class="op">&lt;-</span> <span class="va">gb</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqnames</a></span><span class="op">(</span><span class="va">gb</span><span class="op">)</span> <span class="op">==</span> <span class="va">a</span><span class="op">]</span>
  <span class="va">gb</span> <span class="op">&lt;-</span> <span class="va">gb</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqnames</a></span><span class="op">(</span><span class="va">gb</span><span class="op">$</span><span class="va">query</span><span class="op">)</span> <span class="op">==</span> <span class="va">b</span><span class="op">]</span>
  <span class="va">gb</span>
<span class="op">}</span>

<span class="va">dsList</span> <span class="op">&lt;-</span>   <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>
    OKI <span class="op">=</span> <span class="fu">doubleSubset</span><span class="op">(</span><span class="va">coa</span><span class="op">$</span><span class="va">Oki_Osa</span>, <span class="st">"chr1"</span>, <span class="st">"Chr1"</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/gr2dna_seg.html" class="external-link">gr2dna_seg</a></span><span class="op">(</span><span class="op">)</span>,
    OSA <span class="op">=</span> <span class="fu">doubleSubset</span><span class="op">(</span><span class="va">coa</span><span class="op">$</span><span class="va">Osa_Bar</span>, <span class="st">"Chr1"</span>, <span class="st">"Chr1"</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/gr2dna_seg.html" class="external-link">gr2dna_seg</a></span><span class="op">(</span><span class="op">)</span>,
    BAR <span class="op">=</span> <span class="fu">doubleSubset</span><span class="op">(</span><span class="va">coa</span><span class="op">$</span><span class="va">Bar_Osa</span>, <span class="st">"Chr1"</span>, <span class="st">"Chr1"</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/gr2dna_seg.html" class="external-link">gr2dna_seg</a></span><span class="op">(</span><span class="op">)</span>
<span class="op">)</span>

<span class="va">compList</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>
    <span class="fu">doubleSubset</span><span class="op">(</span><span class="va">coa</span><span class="op">$</span><span class="va">Oki_Osa</span>, <span class="st">"chr1"</span>, <span class="st">"Chr1"</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/gb2comp.html" class="external-link">gb2comp</a></span><span class="op">(</span><span class="op">)</span>,
    Osa_Bar <span class="op">=</span> <span class="fu">doubleSubset</span><span class="op">(</span><span class="va">coa</span><span class="op">$</span><span class="va">Osa_Bar</span>, <span class="st">"Chr1"</span>, <span class="st">"Chr1"</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/gb2comp.html" class="external-link">gb2comp</a></span><span class="op">(</span><span class="op">)</span>
  <span class="op">)</span>

<span class="fu">genoPlotR</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/genoPlotR/man/plot_gene_map.html" class="external-link">plot_gene_map</a></span><span class="op">(</span><span class="va">dsList</span>, <span class="va">compList</span>, dna_seg_scale<span class="op">=</span><span class="cn">TRUE</span>, tree<span class="op">=</span><span class="fu">ade4</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ade4/man/newick2phylog.html" class="external-link">newick2phylog</a></span><span class="op">(</span><span class="fu">addRoot</span><span class="op">(</span><span class="fu">treeNode</span><span class="op">(</span><span class="st">"OKI"</span>, <span class="fu">treeNode</span><span class="op">(</span><span class="st">"OSA"</span>, <span class="st">"BAR"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>,
                         main <span class="op">=</span> <span class="st">"Chromosome 1 in three Oikopleura genome assemblies"</span><span class="op">)</span></code></pre></div>
<p><img src="ParallelPlots_files/figure-html/plot_intersect_GBs_with_tree-1.png" width="1440"></p>
</div>
<div class="section level2">
<h2 id="plot-chr1-in-four-species-double-collapsed">Plot chr1 in four species (double-collapsed)<a class="anchor" aria-label="anchor" href="#plot-chr1-in-four-species-double-collapsed"></a>
</h2>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">KumChr1</span> <span class="op">&lt;-</span><span class="va">coa2</span><span class="op">$</span><span class="va">Oki_Kum</span>
<span class="va">KumChr1</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqnames</a></span><span class="op">(</span><span class="va">KumChr1</span><span class="op">$</span><span class="va">query</span><span class="op">)</span> <span class="op">==</span> <span class="st">"contig_90_1"</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/reverse.html" class="external-link">reverse</a></span><span class="op">(</span>query <span class="op">=</span> <span class="cn">TRUE</span>, <span class="va">KumChr1</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqnames</a></span><span class="op">(</span><span class="va">KumChr1</span><span class="op">$</span><span class="va">query</span><span class="op">)</span> <span class="op">==</span> <span class="st">"contig_90_1"</span><span class="op">]</span><span class="op">)</span>
<span class="va">KumChr1</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqnames</a></span><span class="op">(</span><span class="va">KumChr1</span><span class="op">$</span><span class="va">query</span><span class="op">)</span> <span class="op">==</span> <span class="st">"contig_88_1"</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/reverse.html" class="external-link">reverse</a></span><span class="op">(</span>query <span class="op">=</span> <span class="cn">TRUE</span>, <span class="va">KumChr1</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqnames</a></span><span class="op">(</span><span class="va">KumChr1</span><span class="op">$</span><span class="va">query</span><span class="op">)</span> <span class="op">==</span> <span class="st">"contig_88_1"</span><span class="op">]</span><span class="op">)</span>
<span class="va">KumChr1</span><span class="op">$</span><span class="va">query</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/mergeSeqLevels.html" class="external-link">mergeSeqLevels</a></span><span class="op">(</span><span class="va">KumChr1</span><span class="op">$</span><span class="va">query</span>, <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"contig_27_1"</span>, <span class="st">"contig_90_1"</span>, <span class="st">"contig_3_1"</span>, <span class="st">"contig_88_1"</span><span class="op">)</span>, <span class="st">"Chr1"</span><span class="op">)</span>

<span class="va">dsList</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>
    KUM <span class="op">=</span> <span class="fu">doubleSubset</span><span class="op">(</span><span class="va">KumChr1</span>,     <span class="st">"chr1"</span>, <span class="st">"Chr1"</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/gr2dna_seg.html" class="external-link">gr2dna_seg</a></span><span class="op">(</span><span class="op">)</span>,
    OKI <span class="op">=</span> <span class="fu">doubleSubset</span><span class="op">(</span><span class="va">coa2</span><span class="op">$</span><span class="va">Oki_Osa</span>, <span class="st">"chr1"</span>, <span class="st">"Chr1"</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/gr2dna_seg.html" class="external-link">gr2dna_seg</a></span><span class="op">(</span><span class="op">)</span>,
    OSA <span class="op">=</span> <span class="fu">doubleSubset</span><span class="op">(</span><span class="va">coa2</span><span class="op">$</span><span class="va">Osa_Bar</span>, <span class="st">"Chr1"</span>, <span class="st">"Chr1"</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/gr2dna_seg.html" class="external-link">gr2dna_seg</a></span><span class="op">(</span><span class="op">)</span>,
    BAR <span class="op">=</span> <span class="fu">doubleSubset</span><span class="op">(</span><span class="va">coa2</span><span class="op">$</span><span class="va">Bar_Osa</span>, <span class="st">"Chr1"</span>, <span class="st">"Chr1"</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/gr2dna_seg.html" class="external-link">gr2dna_seg</a></span><span class="op">(</span><span class="op">)</span>
<span class="op">)</span>

<span class="va">compList</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>
    Oki_Kum <span class="op">=</span> <span class="fu">doubleSubset</span><span class="op">(</span><span class="va">KumChr1</span>,     <span class="st">"chr1"</span>, <span class="st">"Chr1"</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/gb2comp.html" class="external-link">gb2comp</a></span><span class="op">(</span><span class="op">)</span>,
    Oki_Osa <span class="op">=</span> <span class="fu">doubleSubset</span><span class="op">(</span><span class="va">coa2</span><span class="op">$</span><span class="va">Oki_Osa</span>, <span class="st">"chr1"</span>, <span class="st">"Chr1"</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/gb2comp.html" class="external-link">gb2comp</a></span><span class="op">(</span><span class="op">)</span>,
    Osa_Bar <span class="op">=</span> <span class="fu">doubleSubset</span><span class="op">(</span><span class="va">coa2</span><span class="op">$</span><span class="va">Osa_Bar</span>, <span class="st">"Chr1"</span>, <span class="st">"Chr1"</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/gb2comp.html" class="external-link">gb2comp</a></span><span class="op">(</span><span class="op">)</span>
  <span class="op">)</span>

<span class="fu">genoPlotR</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/genoPlotR/man/plot_gene_map.html" class="external-link">plot_gene_map</a></span><span class="op">(</span><span class="va">dsList</span>, <span class="va">compList</span>, dna_seg_scale <span class="op">=</span> <span class="cn">TRUE</span>,
                         tree <span class="op">=</span> <span class="fu">ade4</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ade4/man/newick2phylog.html" class="external-link">newick2phylog</a></span><span class="op">(</span><span class="fu">addRoot</span><span class="op">(</span><span class="fu">treeNode</span><span class="op">(</span><span class="fu">treeNode</span><span class="op">(</span><span class="st">"OKI"</span>, <span class="st">"KUM"</span><span class="op">)</span>, <span class="fu">treeNode</span><span class="op">(</span><span class="st">"OSA"</span>, <span class="st">"BAR"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>,
                         main <span class="op">=</span> <span class="st">"Chromosome 1 in four Oikopleura genome assemblies"</span><span class="op">)</span></code></pre></div>
<p><img src="ParallelPlots_files/figure-html/plot_intersect_GBs_with_tree_with_Kume-1.svg" width="1440"></p>
</div>
<div class="section level2">
<h2 id="region-that-is-conserved-between-osaka-and-barcelona">Region that is conserved between Osaka and Barcelona<a class="anchor" aria-label="anchor" href="#region-that-is-conserved-between-osaka-and-barcelona"></a>
</h2>
<p>Since Okinawa is the outgroup, let’s look for regions of it that are aligned on both the Osaka and the Barcelona genomes.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">x</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/findOverlaps-methods.html" class="external-link">subsetByOverlaps</a></span><span class="op">(</span><span class="va">coa2</span><span class="op">$</span><span class="va">Oki_Osa</span>, <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/genomic-range-squeezers.html" class="external-link">granges</a></span><span class="op">(</span><span class="va">coa2</span><span class="op">$</span><span class="va">Oki_Bar</span><span class="op">)</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">subset</a></span><span class="op">(</span><span class="va">width</span> <span class="op">&gt;</span> <span class="fl">1e5</span><span class="op">)</span>
<span class="va">y</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/findOverlaps-methods.html" class="external-link">subsetByOverlaps</a></span><span class="op">(</span><span class="va">coa2</span><span class="op">$</span><span class="va">Oki_Bar</span>, <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/genomic-range-squeezers.html" class="external-link">granges</a></span><span class="op">(</span><span class="va">coa2</span><span class="op">$</span><span class="va">Oki_Osa</span><span class="op">)</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">subset</a></span><span class="op">(</span><span class="va">width</span> <span class="op">&gt;</span> <span class="fl">1e5</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/findOverlaps-methods.html" class="external-link">findOverlapPairs</a></span><span class="op">(</span><span class="va">x</span>,<span class="va">y</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Pairs object with 4 pairs and 0 metadata columns:</span>
<span class="co">##                          first                   second</span>
<span class="co">##                      &lt;GBreaks&gt;                &lt;GBreaks&gt;</span>
<span class="co">##   [1] chr2:10664980-10829335:+ chr2:10664980-10828981:+</span>
<span class="co">##   [2] chr2:13895789-14003399:- chr2:13895789-14003408:-</span>
<span class="co">##   [3]  PAR:15193293-15358116:-  PAR:15193294-15358116:-</span>
<span class="co">##   [4]  XSR:10090052-10210973:-  XSR:10090052-10210972:-</span></code></pre>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/findOverlaps-methods.html" class="external-link">subsetByOverlaps</a></span><span class="op">(</span><span class="va">coa2</span><span class="op">$</span><span class="va">Oki_Osa</span>, <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/inter-range-methods.html" class="external-link">gaps</a></span><span class="op">(</span><span class="va">coa2</span><span class="op">$</span><span class="va">Oki_Bar</span><span class="op">)</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">subset</a></span><span class="op">(</span><span class="va">width</span> <span class="op">&gt;</span> <span class="fl">1e5</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## GBreaks object with 16 ranges and 5 metadata columns:</span>
<span class="co">##        seqnames            ranges strand |                  query      Arm</span>
<span class="co">##           &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; |              &lt;GRanges&gt; &lt;factor&gt;</span>
<span class="co">##    [1]     chr2   5825747-5966560      + |   Chr2:5465216-5578318     long</span>
<span class="co">##    [2]     chr2 10361302-10481638      + | Chr2:11648925-11718071     long</span>
<span class="co">##    [3]     chr2 10664980-10829335      + |   Chr2:5679693-5828385     long</span>
<span class="co">##    [4]     chr2 13895789-14003399      - | Chr2:11294781-11373503     long</span>
<span class="co">##    [5]     chr2 14726918-14844516      + |   Chr2:8440297-8537610     long</span>
<span class="co">##    ...      ...               ...    ... .                    ...      ...</span>
<span class="co">##   [12]      XSR   7236669-7346658      + |    XSR:4093395-4195157      XSR</span>
<span class="co">##   [13]      XSR   7543498-7653200      + |    XSR:3606737-3707501      XSR</span>
<span class="co">##   [14]      XSR   8709632-8820503      + |    XSR:7297396-7413239      XSR</span>
<span class="co">##   [15]      XSR   8929901-9075791      - |    XSR:7506741-7641724      XSR</span>
<span class="co">##   [16]      XSR 10090052-10210973      - |  XSR:10484001-10612402      XSR</span>
<span class="co">##                                  rep            transcripts        flag</span>
<span class="co">##                      &lt;CharacterList&gt;                  &lt;Rle&gt; &lt;character&gt;</span>
<span class="co">##    [1]        tandem,unknown,rnd,... g5554.t1;g5556.t1;g5..        &lt;NA&gt;</span>
<span class="co">##    [2]  tandem,rnd,LowComplexity,... g6889.t1;g6890.t1;g6..        &lt;NA&gt;</span>
<span class="co">##    [3] unknown,rnd,LowComplexity,... g7009.t1;g7010.t1;g7..        &lt;NA&gt;</span>
<span class="co">##    [4]             unknown,rnd,ltr-1 g7918.t1;g7922.t1;g7..        &lt;NA&gt;</span>
<span class="co">##    [5]        tandem,rnd,unknown,... g8164.t1;g8165.t1;g8..        &lt;NA&gt;</span>
<span class="co">##    ...                           ...                    ...         ...</span>
<span class="co">##   [12]      tandem,rnd,LowComplexity g15405.t1;g15406.t1;..         Tra</span>
<span class="co">##   [13]        tandem,rnd,unknown,... g15508.t1;g15509.t1;..        &lt;NA&gt;</span>
<span class="co">##   [14]      rnd,tandem,LowComplexity g15864.t1;g15865.t1;..        &lt;NA&gt;</span>
<span class="co">##   [15]             unknown,rnd,ltr-1 g15934.t1;g15935.t1;..        &lt;NA&gt;</span>
<span class="co">##   [16]                   rnd,unknown g16292.t1;g16293.t1;..        &lt;NA&gt;</span>
<span class="co">##   -------</span>
<span class="co">##   seqinfo: 19 sequences from OKI2018_I69 genome</span></code></pre>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">a</span> <span class="op">&lt;-</span> <span class="va">coa</span><span class="op">$</span><span class="va">Oki_Osa</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/findOverlaps-methods.html" class="external-link">subsetByOverlaps</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a></span><span class="op">(</span><span class="st">"chr2:10000000-11000000"</span><span class="op">)</span><span class="op">)</span> |&gt; <span class="fu">doubleSubset</span><span class="op">(</span><span class="st">"chr2"</span>, <span class="st">"Chr2"</span><span class="op">)</span>
<span class="va">b</span> <span class="op">&lt;-</span> <span class="va">coa</span><span class="op">$</span><span class="va">Oki_Bar</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/findOverlaps-methods.html" class="external-link">subsetByOverlaps</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a></span><span class="op">(</span><span class="st">"chr2:10000000-11000000"</span><span class="op">)</span><span class="op">)</span> |&gt; <span class="fu">doubleSubset</span><span class="op">(</span><span class="st">"chr2"</span>, <span class="st">"Chr2"</span><span class="op">)</span>
<span class="va">c</span> <span class="op">&lt;-</span> <span class="va">coa</span><span class="op">$</span><span class="va">Bar_Osa</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/findOverlaps-methods.html" class="external-link">subsetByOverlaps</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/inter-range-methods.html" class="external-link">range</a></span><span class="op">(</span><span class="va">b</span><span class="op">$</span><span class="va">query</span><span class="op">)</span><span class="op">)</span>                    |&gt; <span class="fu">doubleSubset</span><span class="op">(</span><span class="st">"Chr2"</span>, <span class="st">"Chr2"</span><span class="op">)</span>

<span class="va">dsList</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>
  OSA <span class="op">=</span> <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/swap.html" class="external-link">swap</a></span><span class="op">(</span><span class="va">a</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/gr2dna_seg.html" class="external-link">gr2dna_seg</a></span><span class="op">(</span><span class="op">)</span>,
  OKI <span class="op">=</span> <span class="va">a</span> |&gt; <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/gr2dna_seg.html" class="external-link">gr2dna_seg</a></span><span class="op">(</span><span class="op">)</span>,
  BAR <span class="op">=</span> <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/swap.html" class="external-link">swap</a></span><span class="op">(</span><span class="va">b</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/gr2dna_seg.html" class="external-link">gr2dna_seg</a></span><span class="op">(</span><span class="op">)</span>,
  Osa <span class="op">=</span> <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/swap.html" class="external-link">swap</a></span><span class="op">(</span><span class="va">c</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/gr2dna_seg.html" class="external-link">gr2dna_seg</a></span><span class="op">(</span><span class="op">)</span>
<span class="op">)</span>

<span class="va">compList</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>
  <span class="va">a</span> |&gt; <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/swap.html" class="external-link">swap</a></span><span class="op">(</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/gb2comp.html" class="external-link">gb2comp</a></span><span class="op">(</span><span class="op">)</span>,
  <span class="va">b</span> |&gt; <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/gb2comp.html" class="external-link">gb2comp</a></span><span class="op">(</span><span class="op">)</span>,
  <span class="va">c</span> |&gt; <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/gb2comp.html" class="external-link">gb2comp</a></span><span class="op">(</span><span class="op">)</span>
<span class="op">)</span>

<span class="fu">genoPlotR</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/genoPlotR/man/plot_gene_map.html" class="external-link">plot_gene_map</a></span><span class="op">(</span><span class="va">dsList</span>, <span class="va">compList</span>, dna_seg_scale<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<p><img src="ParallelPlots_files/figure-html/combine_GBs-1.png" width="700"></p>
</div>
</div>
<div class="section level1">
<h1 id="other-species">Other species<a class="anchor" aria-label="anchor" href="#other-species"></a>
</h1>
<div class="section level2">
<h2 id="ciona">Ciona<a class="anchor" aria-label="anchor" href="#ciona"></a>
</h2>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/plotApairOfChrs.html" class="external-link">plotApairOfChrs</a></span><span class="op">(</span><span class="va">coa2</span><span class="op">$</span><span class="va">Ply_Ros</span>, <span class="st">"BNJZ01000001.1"</span>, main <span class="op">=</span> <span class="st">"Ciona intestinalis Plymouth vs Roscoff"</span><span class="op">)</span></code></pre></div>
<p><img src="ParallelPlots_files/figure-html/Ciona_Ply_Ros_chr1-1.png" width="700"></p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/plotApairOfChrs.html" class="external-link">plotApairOfChrs</a></span><span class="op">(</span><span class="va">coa2</span><span class="op">$</span><span class="va">Ply_Rob</span>, <span class="st">"BNJZ01000001.1"</span>, main <span class="op">=</span> <span class="st">"Ciona intestinalis Plymouth vs Ciona robusta"</span><span class="op">)</span></code></pre></div>
<p><img src="ParallelPlots_files/figure-html/Ciona_Ply_Rob_chr1-1.png" width="700"></p>
<p>Unfortunately, <em>C. savignyi</em>’s genome assembly is too fragmented. Here is a plot with the largest fragment.</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/plotApairOfChrs.html" class="external-link">plotApairOfChrs</a></span><span class="op">(</span><span class="va">coa2</span><span class="op">$</span><span class="va">Ply_Sav</span>, <span class="st">"BNJZ01000001.1"</span>, main <span class="op">=</span> <span class="st">"Ciona intestinalis Plymouth vs Ciona savigny"</span><span class="op">)</span></code></pre></div>
<p><img src="ParallelPlots_files/figure-html/Ciona_Ply_Sav_chr1-1.png" width="700"></p>
</div>
</div>
<div class="section level1">
<h1 id="session-information">Session information<a class="anchor" aria-label="anchor" href="#session-information"></a>
</h1>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## R version 4.1.2 (2021-11-01)</span>
<span class="co">## Platform: x86_64-pc-linux-gnu (64-bit)</span>
<span class="co">## Running under: Debian GNU/Linux bookworm/sid</span>
<span class="co">## </span>
<span class="co">## Matrix products: default</span>
<span class="co">## BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.10.0</span>
<span class="co">## LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.10.0</span>
<span class="co">## </span>
<span class="co">## locale:</span>
<span class="co">##  [1] LC_CTYPE=en_GB.UTF-8       LC_NUMERIC=C              </span>
<span class="co">##  [3] LC_TIME=en_GB.UTF-8        LC_COLLATE=en_GB.UTF-8    </span>
<span class="co">##  [5] LC_MONETARY=en_GB.UTF-8    LC_MESSAGES=en_GB.UTF-8   </span>
<span class="co">##  [7] LC_PAPER=en_GB.UTF-8       LC_NAME=C                 </span>
<span class="co">##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span>
<span class="co">## [11] LC_MEASUREMENT=en_GB.UTF-8 LC_IDENTIFICATION=C       </span>
<span class="co">## </span>
<span class="co">## attached base packages:</span>
<span class="co">## [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span>
<span class="co">## [8] base     </span>
<span class="co">## </span>
<span class="co">## other attached packages:</span>
<span class="co">##  [1] BSgenome.Odioica.local.Odioica.reference.v3.0_1.0.1</span>
<span class="co">##  [2] BSgenome.Odioica.local.AOM.5_1.0.0                 </span>
<span class="co">##  [3] BSgenome.Odioica.local.KUM.M3_1.0.0                </span>
<span class="co">##  [4] BSgenome.Odioica.local.Bar2.p4_1.0.0               </span>
<span class="co">##  [5] BSgenome.Odioica.local.OSKA2016v1.9_1.0.0          </span>
<span class="co">##  [6] BSgenome.Odioica.local.OKI2018.I69_1.0.0           </span>
<span class="co">##  [7] OikScrambling_0.6.2                                </span>
<span class="co">##  [8] ggplot2_3.3.5                                      </span>
<span class="co">##  [9] GenomicBreaks_0.12.0                               </span>
<span class="co">## [10] BSgenome_1.62.0                                    </span>
<span class="co">## [11] rtracklayer_1.54.0                                 </span>
<span class="co">## [12] Biostrings_2.62.0                                  </span>
<span class="co">## [13] XVector_0.34.0                                     </span>
<span class="co">## [14] GenomicRanges_1.46.1                               </span>
<span class="co">## [15] GenomeInfoDb_1.30.0                                </span>
<span class="co">## [16] IRanges_2.28.0                                     </span>
<span class="co">## [17] S4Vectors_0.32.3                                   </span>
<span class="co">## [18] BiocGenerics_0.40.0                                </span>
<span class="co">## </span>
<span class="co">## loaded via a namespace (and not attached):</span>
<span class="co">##   [1] backports_1.4.0             Hmisc_4.6-0                </span>
<span class="co">##   [3] systemfonts_1.0.3           plyr_1.8.6                 </span>
<span class="co">##   [5] splines_4.1.2               BiocParallel_1.28.3        </span>
<span class="co">##   [7] digest_0.6.29               htmltools_0.5.2            </span>
<span class="co">##   [9] tiff_0.1-10                 GO.db_3.14.0               </span>
<span class="co">##  [11] gdata_2.18.0                fansi_0.5.0                </span>
<span class="co">##  [13] magrittr_2.0.1              checkmate_2.0.0            </span>
<span class="co">##  [15] memoise_2.0.1               cluster_2.1.2              </span>
<span class="co">##  [17] tzdb_0.2.0                  readr_2.1.1                </span>
<span class="co">##  [19] annotate_1.72.0             matrixStats_0.61.0         </span>
<span class="co">##  [21] R.utils_2.11.0              pkgdown_2.0.1              </span>
<span class="co">##  [23] jpeg_0.1-9                  colorspace_2.0-2           </span>
<span class="co">##  [25] blob_1.2.2                  textshaping_0.3.6          </span>
<span class="co">##  [27] xfun_0.28                   dplyr_1.0.7                </span>
<span class="co">##  [29] crayon_1.4.2                RCurl_1.98-1.5             </span>
<span class="co">##  [31] jsonlite_1.7.2              lme4_1.1-27.1              </span>
<span class="co">##  [33] survival_3.2-13             glue_1.5.1                 </span>
<span class="co">##  [35] gtable_0.3.0                zlibbioc_1.40.0            </span>
<span class="co">##  [37] DelayedArray_0.20.0         weights_1.0.4              </span>
<span class="co">##  [39] abind_1.4-5                 scales_1.1.1               </span>
<span class="co">##  [41] DBI_1.1.1                   Rcpp_1.0.7                 </span>
<span class="co">##  [43] plotrix_3.8-2               genoPlotR_0.8.11           </span>
<span class="co">##  [45] xtable_1.8-4                htmlTable_2.3.0            </span>
<span class="co">##  [47] foreign_0.8-81              bit_4.0.4                  </span>
<span class="co">##  [49] Formula_1.2-4               htmlwidgets_1.5.4          </span>
<span class="co">##  [51] httr_1.4.2                  RColorBrewer_1.1-2         </span>
<span class="co">##  [53] ellipsis_0.3.2              mice_3.14.0                </span>
<span class="co">##  [55] pkgconfig_2.0.3             XML_3.99-0.8               </span>
<span class="co">##  [57] R.methodsS3_1.8.1           nnet_7.3-16                </span>
<span class="co">##  [59] sass_0.4.0                  locfit_1.5-9.4             </span>
<span class="co">##  [61] utf8_1.2.2                  tidyselect_1.1.1           </span>
<span class="co">##  [63] rlang_0.4.12                reshape2_1.4.4             </span>
<span class="co">##  [65] AnnotationDbi_1.56.2        munsell_0.5.0              </span>
<span class="co">##  [67] tools_4.1.2                 cachem_1.0.6               </span>
<span class="co">##  [69] generics_0.1.1              RSQLite_2.2.9              </span>
<span class="co">##  [71] ade4_1.7-18                 broom_0.7.10               </span>
<span class="co">##  [73] evaluate_0.14               stringr_1.4.0              </span>
<span class="co">##  [75] fastmap_1.1.0               fftwtools_0.9-11           </span>
<span class="co">##  [77] yaml_2.2.1                  ragg_1.2.1                 </span>
<span class="co">##  [79] knitr_1.36                  bit64_4.0.5                </span>
<span class="co">##  [81] fs_1.5.0                    purrr_0.3.4                </span>
<span class="co">##  [83] heatmaps_1.18.0             KEGGREST_1.34.0            </span>
<span class="co">##  [85] nlme_3.1-153                R.oo_1.24.0                </span>
<span class="co">##  [87] poweRlaw_0.70.6             pracma_2.3.6               </span>
<span class="co">##  [89] compiler_4.1.2              rstudioapi_0.13            </span>
<span class="co">##  [91] png_0.1-7                   tibble_3.1.6               </span>
<span class="co">##  [93] bslib_0.3.1                 stringi_1.7.6              </span>
<span class="co">##  [95] highr_0.9                   desc_1.4.0                 </span>
<span class="co">##  [97] lattice_0.20-45             CNEr_1.30.0                </span>
<span class="co">##  [99] Matrix_1.3-4                nloptr_1.2.2.3             </span>
<span class="co">## [101] vctrs_0.3.8                 pillar_1.6.4               </span>
<span class="co">## [103] lifecycle_1.0.1             jquerylib_0.1.4            </span>
<span class="co">## [105] data.table_1.14.2           bitops_1.0-7               </span>
<span class="co">## [107] R6_2.5.1                    BiocIO_1.4.0               </span>
<span class="co">## [109] latticeExtra_0.6-29         KernSmooth_2.23-20         </span>
<span class="co">## [111] gridExtra_2.3               codetools_0.2-18           </span>
<span class="co">## [113] boot_1.3-28                 MASS_7.3-54                </span>
<span class="co">## [115] gtools_3.9.2                assertthat_0.2.1           </span>
<span class="co">## [117] SummarizedExperiment_1.24.0 rprojroot_2.0.2            </span>
<span class="co">## [119] rjson_0.2.20                withr_2.4.3                </span>
<span class="co">## [121] GenomicAlignments_1.30.0    Rsamtools_2.10.0           </span>
<span class="co">## [123] GenomeInfoDbData_1.2.7      parallel_4.1.2             </span>
<span class="co">## [125] hms_1.1.1                   EBImage_4.36.0             </span>
<span class="co">## [127] grid_4.1.2                  rpart_4.1-15               </span>
<span class="co">## [129] minqa_1.2.4                 tidyr_1.1.4                </span>
<span class="co">## [131] rmarkdown_2.11              MatrixGenerics_1.6.0       </span>
<span class="co">## [133] Biobase_2.54.0              base64enc_0.1-3            </span>
<span class="co">## [135] restfulr_0.0.13</span></code></pre>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Charles Plessy.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.1.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
