<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Motif search and analysis • OikScrambling</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Motif search and analysis">
<meta property="og:description" content="OikScrambling">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">OikScrambling</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.0.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/OikScrambling.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/CNEs.html">Conserved non-coding elements</a>
    </li>
    <li>
      <a href="../articles/CoalescingCutoffs.html">Modifying coalescent cutoffs</a>
    </li>
    <li>
      <a href="../articles/ColinearityInterruptors.html">Regions interrupting colinearity</a>
    </li>
    <li>
      <a href="../articles/DistanceMeasures.html">Distance Measures</a>
    </li>
    <li>
      <a href="../articles/GenomicFeatures.html">Genomic features of the breakpoints</a>
    </li>
    <li>
      <a href="../articles/Inversions.html">Inversions</a>
    </li>
    <li>
      <a href="../articles/LoadGenomicBreaks.html">Load genomic break data</a>
    </li>
    <li>
      <a href="../articles/OrthogroupSyntenies.html">Orthogroup syntenies</a>
    </li>
    <li>
      <a href="../articles/OxfordPlots.html">Whole-genome dot-plots</a>
    </li>
    <li>
      <a href="../articles/PWM.html">Motif search and analysis</a>
    </li>
    <li>
      <a href="../articles/ParallelPlots.html">Whole-chromosome parallel plots</a>
    </li>
    <li>
      <a href="../articles/RegionWidths.html">Width of the aligned and mapped regions.</a>
    </li>
    <li>
      <a href="../articles/RegionsOfInterest.html">Regions of interest</a>
    </li>
    <li>
      <a href="../articles/RepeatRegions.html">Repeat regions</a>
    </li>
    <li>
      <a href="../articles/SegmentMovements.html">Segment movement distances</a>
    </li>
    <li>
      <a href="../articles/TFBS.html">Transcription factor binding sites</a>
    </li>
    <li>
      <a href="../articles/Translocations.html">Translocations</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/oist/LuscombeU_OikScrambling/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="PWM_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Motif search and analysis</h1>
                        <h4 data-toc-skip class="author">Charles Plessy</h4>
            
            <h4 data-toc-skip class="date">16 February, 2022</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/oist/LuscombeU_OikScrambling/blob/HEAD/vignettes/PWM.Rmd" class="external-link"><code>vignettes/PWM.Rmd</code></a></small>
      <div class="hidden name"><code>PWM.Rmd</code></div>

    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">knitr</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/knitr/man/opts_chunk.html" class="external-link">opts_chunk</a></span><span class="op">$</span><span class="fu">set</span><span class="op">(</span>cache <span class="op">=</span> <span class="cn">TRUE</span>, include <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
<span class="fu">knitr</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/knitr/man/opts_knit.html" class="external-link">opts_knit</a></span><span class="op">$</span><span class="fu">set</span><span class="op">(</span>verbose <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<div class="section level1">
<h1 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h1>
<p>Search for motifs in regions near genomic breaks. By default the computations that take time are loaded from R objects. The source code to generate them is only executed when the object files can not be found.</p>
</div>
<div class="section level1">
<h1 id="load-pacakges">Load pacakges<a class="anchor" aria-label="anchor" href="#load-pacakges"></a>
</h1>
<p>Core packages that provide functions we use a lot.</p>
</div>
<div class="section level1">
<h1 id="motif-searches">Motif searches<a class="anchor" aria-label="anchor" href="#motif-searches"></a>
</h1>
<p><a href="https://www.niehs.nih.gov/research/resources/software/biostatistics/gadem/index.cfm" class="external-link">GADEM</a>’s documentation states:</p>
<blockquote>
<p><em>The background sequences are simulated using the [a,c,g,t] frequencies in the input sequences, with length matched between the two sets. The background sequences are used as the random sequences for assessing motif enrichment in the input data.</em></p>
</blockquote>
<div class="section level2">
<h2 id="aligned-non-coalesced-regions">Aligned non-coalesced regions<a class="anchor" aria-label="anchor" href="#aligned-non-coalesced-regions"></a>
</h2>
<p>Aligned regions that do not coalesce into syntenic blocks are relatively depleted in genes, and show no enrichment of exons versus introns. They are also depleted in repeats, and are mildly enriched in CNEs. Altogether, they can not be summarised in one feature. So let’s see if there are motifs enriched there.</p>
<p>See <code>vignette("GenomicFeatuers", package = "OikScrambling")</code> for details on these regions.</p>
</div>
<div class="section level2">
<h2 id="unmapped">Unmapped<a class="anchor" aria-label="anchor" href="#unmapped"></a>
</h2>
<p>There regions are not aligned and not within a synteny block.</p>
</div>
<div class="section level2">
<h2 id="unaligned-regions-near-inversions">Unaligned regions near inversions<a class="anchor" aria-label="anchor" href="#unaligned-regions-near-inversions"></a>
</h2>
<p>I wanted to search search for motifs near inversions to see if they can lead to an explanation.</p>
<p>See <code><a href="../articles/Inversions.html">vignette("Inversions", package = "OikScrambling")</a></code> for details on inversion.</p>
<div class="section level3">
<h3 id="isolate-the-left-side-gaps-in-inversions">Isolate the left-side gaps in inversions<a class="anchor" aria-label="anchor" href="#isolate-the-left-side-gaps-in-inversions"></a>
</h3>
</div>
</div>
<div class="section level2">
<h2 id="inversions">Inversions<a class="anchor" aria-label="anchor" href="#inversions"></a>
</h2>
</div>
<div class="section level2">
<h2 id="promoters">Promoters<a class="anchor" aria-label="anchor" href="#promoters"></a>
</h2>
</div>
<div class="section level2">
<h2 id="introns">Introns<a class="anchor" aria-label="anchor" href="#introns"></a>
</h2>
</div>
<div class="section level2">
<h2 id="summary">Summary<a class="anchor" aria-label="anchor" href="#summary"></a>
</h2>
<p>Number of motifs found</p>
</div>
</div>
<div class="section level1">
<h1 id="display">Display<a class="anchor" aria-label="anchor" href="#display"></a>
</h1>
<p>Service functions.</p>
<div class="section level2">
<h2 id="aligned-non-coalesced-regions-1">Aligned non-coalesced regions<a class="anchor" aria-label="anchor" href="#aligned-non-coalesced-regions-1"></a>
</h2>
<p>Many comparison return a motif that looks like <code>AAAA.....AAAA</code>, and some also return <code>AAGAA</code> (or their reverse-complement).</p>
<p>The Oki - Kume comparison shows T-rich motifs (note that they could have been A-rich reverse-complement).</p>
<p>Sometimes a weaker A-rich / T-rich motif is also found.</p>
<p>Here are a few graphical examples.</p>
</div>
<div class="section level2">
<h2 id="unaligned-non-mapped-regions">Unaligned non-mapped regions<a class="anchor" aria-label="anchor" href="#unaligned-non-mapped-regions"></a>
</h2>
<p>We still find <code>AAAA.....AAAA</code>,<code>AAGAA</code>, <code>A</code>-rich (or their reverse-complement), but not as systematicaly.</p>
<p>We start to see species-specific motifs:</p>
<ul>
<li>Variations over <code>GsCsCkwmGsGsC</code> in some Oki -&gt; other comparisons.</li>
<li>Variations over <code>GCT..GCT</code> in some Nor -&gt; other comparisons (pay attention the reverse complements).</li>
</ul>
<p>Here are a few graphical examples.</p>
</div>
<div class="section level2">
<h2 id="unaligned-regions-left-to-inversion">Unaligned regions left to inversion<a class="anchor" aria-label="anchor" href="#unaligned-regions-left-to-inversion"></a>
</h2>
<ul>
<li>We still find <code>AAAA.....AAAA</code>,<code>AAGAA</code>, <code>A</code>-rich (or their reverse-complement).</li>
<li>We see palindromic motifs (<code>CCTGA.TCAGG</code>, <code>aaGCCGc...gCGGCtt</code>) in the Okinawan genome.</li>
<li>Two other motifs are found when comparing Oki to Aom (<code>nGAGGCmACGmsCGTTCn</code>) and to Nor <code>nACGCyCryTCATTCTwTy</code>).</li>
<li>In the Osaka genome, we see a the <code>GCT..GCT</code> motif again.</li>
</ul>
<p>Here are a few graphical examples.</p>
<div class="section level3">
<h3 id="cctga-tcagg">
<code>CCTGA.TCAGG</code><a class="anchor" aria-label="anchor" href="#cctga-tcagg"></a>
</h3>
</div>
<div class="section level3">
<h3 id="aagccgc---gcggctt">
<code>aaGCCGc...gCGGCtt</code><a class="anchor" aria-label="anchor" href="#aagccgc---gcggctt"></a>
</h3>
</div>
<div class="section level3">
<h3 id="other-from-okinawan-genome">Other from Okinawan genome<a class="anchor" aria-label="anchor" href="#other-from-okinawan-genome"></a>
</h3>
</div>
<div class="section level3">
<h3 id="gct--gct">
<code>GCT..GCT</code><a class="anchor" aria-label="anchor" href="#gct--gct"></a>
</h3>
</div>
<div class="section level3">
<h3 id="gcttcgcttc">
<code>GCTTCGCTTC</code><a class="anchor" aria-label="anchor" href="#gcttcgcttc"></a>
</h3>
</div>
<div class="section level3">
<h3 id="caa----ttggtca">
<code>CAA....TTGGTCA</code><a class="anchor" aria-label="anchor" href="#caa----ttggtca"></a>
</h3>
</div>
</div>
</div>
<div class="section level1">
<h1 id="at-richness">AT richness<a class="anchor" aria-label="anchor" href="#at-richness"></a>
</h1>
<p>Unmapped regions have a much higher AT-richness than aligned ones.</p>
<p>A palindromic (rAAGCsGCwwmkCGrCTTyn) motif is found in the left gaps.</p>
<p>Or could it be <code>GCCGCnnnGCGGC</code> ? Its frequency in genomes is something like:</p>
<pre><code>  Oki   Osa   Bar   Kum   Aom   Nor 
10827  2273  1967 10880  2210  2247 </code></pre>
<p>On the other hand, a motif containing GCGAAGCGAA identified in Osa_Oki shows:</p>
<pre><code>  Oki   Osa   Bar   Kum   Aom   Nor 
  483 21360 23340   501 21594 16413 </code></pre>
</div>
<div class="section level1">
<h1 id="search-for-the-motifs-in-jaspar">Search for the motifs in JASPAR<a class="anchor" aria-label="anchor" href="#search-for-the-motifs-in-jaspar"></a>
</h1>
<p>But a search in Jaspar with either the full motif or its halof did not reveal strong hits. Visual inspection of PRDM9 sequence logos via Google did not show similarity either.</p>
<pre><code><span class="co"># writeXStringSet(lgapsSeq_Oki_O, file="firstGaps.fasta")</span>
<span class="co"># I used this FASTA export to search for motifs with GLAM2, and it also</span>
<span class="co"># returned A/T polymers.</span></code></pre>
<p>The GLAM2 motif finder was run on this file, and then to screen the genome. Hits coordinates were encoded in a BED file uploaded to the ZENBU genome browser. Visual inspection gave me the impression that the motif might be related to repeats.</p>
<p>GLAM2 search of the gap sequences did not return a motif similar to “AAGCsGCwwmkCGrCTTyn”.</p>
<div class="section level3">
<h3 id="whole-genome-search-of-the-motif">Whole-genome search of the motif<a class="anchor" aria-label="anchor" href="#whole-genome-search-of-the-motif"></a>
</h3>
<p>Disappointingly, the motif is found equally in gaps and inversions when searched with its PWM.</p>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Aleksandra Bliznina, Michael Mansfield, Charles Plessy.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.1.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
