<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Width of the aligned and mapped regions. • OikScrambling</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Width of the aligned and mapped regions.">
<meta property="og:description" content="OikScrambling">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">OikScrambling</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.8.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/OikScrambling.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/CoalescingCutoffs.html">Modifying coalescent cutoffs</a>
    </li>
    <li>
      <a href="../articles/ColinearityInterruptors.html">Regions interrupting colinearity</a>
    </li>
    <li>
      <a href="../articles/DistanceMeasures.html">Distance Measures</a>
    </li>
    <li>
      <a href="../articles/GenomicFeatures.html">Genomic features of the breakpoints</a>
    </li>
    <li>
      <a href="../articles/Inversions.html">Inversions</a>
    </li>
    <li>
      <a href="../articles/LoadGenomicBreaks.html">Load genomic break data</a>
    </li>
    <li>
      <a href="../articles/OrthogroupSyntenies.html">Orthogroup syntenies</a>
    </li>
    <li>
      <a href="../articles/OxfordPlots.html">Whole-genome dot-plots</a>
    </li>
    <li>
      <a href="../articles/PWM.html">Motif finding and analysis</a>
    </li>
    <li>
      <a href="../articles/ParallelPlots.html">Whole-chromosome parallel plots</a>
    </li>
    <li>
      <a href="../articles/RegionWidths.html">Width of the aligned and mapped regions.</a>
    </li>
    <li>
      <a href="../articles/RepeatRegions.html">Repeat regions</a>
    </li>
    <li>
      <a href="../articles/SegmentMovements.html">Segment movement distances</a>
    </li>
    <li>
      <a href="../articles/Translocations.html">Translocations</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/oist/LuscombeU_OikScrambling/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="RegionWidths_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Width of the aligned and mapped regions.</h1>
                        <h4 data-toc-skip class="author">Charles Plessy</h4>
            
            <h4 data-toc-skip class="date">25 January, 2022</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/oist/LuscombeU_OikScrambling/blob/HEAD/vignettes/RegionWidths.Rmd" class="external-link"><code>vignettes/RegionWidths.Rmd</code></a></small>
      <div class="hidden name"><code>RegionWidths.Rmd</code></div>

    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">knitr</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/knitr/man/opts_chunk.html" class="external-link">opts_chunk</a></span><span class="op">$</span><span class="fu">set</span><span class="op">(</span>cache <span class="op">=</span> <span class="cn">TRUE</span>, cache.lazy <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
<span class="fu">knitr</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/knitr/man/opts_knit.html" class="external-link">opts_knit</a></span><span class="op">$</span><span class="fu">set</span><span class="op">(</span>verbose <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<div class="section level1">
<h1 id="load-r-pacakges-and-data">Load R pacakges and data<a class="anchor" aria-label="anchor" href="#load-r-pacakges-and-data"></a>
</h1>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/oist/LuscombeU_OikScrambling" class="external-link">'OikScrambling'</a></span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://ggplot2.tidyverse.org" class="external-link">'ggplot2'</a></span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">'patchwork'</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="op">)</span>
<span class="va">genomes</span> <span class="op">&lt;-</span> <span class="fu">OikScrambling</span><span class="fu">:::</span><span class="fu"><a href="../reference/loadAllGenomes.html">loadAllGenomes</a></span><span class="op">(</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/load.html" class="external-link">load</a></span><span class="op">(</span><span class="st">"BreakPoints.Rdata"</span><span class="op">)</span></code></pre></div>
<p>See <code><a href="../articles/LoadGenomicBreaks.html">vignette("LoadGenomicBreaks", package = "OikScrambling")</a></code> for how the different GBreaks objects are prepared.</p>
</div>
<div class="section level1">
<h1 id="precompute-widths">Precompute widths<a class="anchor" aria-label="anchor" href="#precompute-widths"></a>
</h1>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">widths</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/SimpleList-class.html" class="external-link">SimpleList</a></span><span class="op">(</span><span class="op">)</span>
<span class="va">makeWidthsDF</span> <span class="op">&lt;-</span> <span class="kw">function</span> <span class="op">(</span><span class="va">objList</span><span class="op">)</span> <span class="op">{</span>
  <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">objList</span><span class="op">)</span>, \<span class="op">(</span><span class="va">name</span><span class="op">)</span> <span class="op">{</span>
    <span class="va">DF</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">mcols</a></span><span class="op">(</span><span class="va">objList</span><span class="op">[[</span><span class="va">name</span><span class="op">]</span><span class="op">]</span><span class="op">)</span>
    <span class="kw">if</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">DF</span><span class="op">)</span> <span class="op">==</span> <span class="fl">0</span><span class="op">)</span> <span class="kw"><a href="https://rdrr.io/r/base/function.html" class="external-link">return</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/DataFrame-class.html" class="external-link">DataFrame</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span>
    <span class="kw">if</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/NULL.html" class="external-link">is.null</a></span><span class="op">(</span><span class="va">DF</span><span class="op">$</span><span class="va">Arm</span><span class="op">)</span>   <span class="op">)</span> <span class="va">DF</span><span class="op">$</span><span class="va">Arm</span>     <span class="op">&lt;-</span> <span class="cn">NA</span>
    <span class="kw">if</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/NULL.html" class="external-link">is.null</a></span><span class="op">(</span><span class="va">DF</span><span class="op">$</span><span class="va">flag</span><span class="op">)</span><span class="op">)</span>   <span class="va">DF</span><span class="op">$</span><span class="va">flag</span>    <span class="op">&lt;-</span> <span class="cn">NA</span>
    <span class="kw">if</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/NULL.html" class="external-link">is.null</a></span><span class="op">(</span><span class="va">DF</span><span class="op">$</span><span class="va">nonCoa</span><span class="op">)</span><span class="op">)</span> <span class="va">DF</span><span class="op">$</span><span class="va">nonCoa</span>  <span class="op">&lt;-</span> <span class="cn">NA</span>
    <span class="va">DF</span><span class="op">$</span><span class="va">name</span>  <span class="op">&lt;-</span> <span class="va">name</span>
    <span class="va">DF</span><span class="op">$</span><span class="va">width</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/IPosRanges-class.html" class="external-link">width</a></span><span class="op">(</span><span class="va">objList</span><span class="op">[[</span><span class="va">name</span><span class="op">]</span><span class="op">]</span><span class="op">)</span>
    <span class="va">DF</span><span class="op">$</span><span class="va">chr</span>   <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqnames</a></span><span class="op">(</span><span class="va">objList</span><span class="op">[[</span><span class="va">name</span><span class="op">]</span><span class="op">]</span><span class="op">)</span> |&gt;
      <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">sub</a></span><span class="op">(</span>pat <span class="op">=</span> <span class="st">"Chr"</span>,    rep <span class="op">=</span> <span class="st">"chr"</span><span class="op">)</span>    |&gt;
      <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">sub</a></span><span class="op">(</span>pat <span class="op">=</span> <span class="st">"BNJZ.*"</span>, rep <span class="op">=</span> <span class="st">"Ply"</span><span class="op">)</span>    |&gt;
      <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">sub</a></span><span class="op">(</span>pat <span class="op">=</span> <span class="st">"BNKA.*"</span>, rep <span class="op">=</span> <span class="st">"Ros"</span><span class="op">)</span>    |&gt;
      <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">sub</a></span><span class="op">(</span>pat <span class="op">=</span> <span class="st">"BJTB.*"</span>, rep <span class="op">=</span> <span class="st">"Rob"</span><span class="op">)</span>    |&gt;
      <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">sub</a></span><span class="op">(</span>pat <span class="op">=</span> <span class="st">"^R.*"</span>,   rep <span class="op">=</span> <span class="st">"Sav"</span><span class="op">)</span>    |&gt;
      <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">sub</a></span><span class="op">(</span>pat <span class="op">=</span> <span class="st">"^2L$"</span>,   rep <span class="op">=</span> <span class="st">"Dro"</span><span class="op">)</span>    |&gt;
      <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">sub</a></span><span class="op">(</span>pat <span class="op">=</span> <span class="st">"^2R$"</span>,   rep <span class="op">=</span> <span class="st">"Dro"</span><span class="op">)</span>    |&gt;
      <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">sub</a></span><span class="op">(</span>pat <span class="op">=</span> <span class="st">"^3L$"</span>,   rep <span class="op">=</span> <span class="st">"Dro"</span><span class="op">)</span>    |&gt;
      <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">sub</a></span><span class="op">(</span>pat <span class="op">=</span> <span class="st">"^3R$"</span>,   rep <span class="op">=</span> <span class="st">"Dro"</span><span class="op">)</span>    |&gt;
      <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">sub</a></span><span class="op">(</span>pat <span class="op">=</span> <span class="st">"^4$"</span>,    rep <span class="op">=</span> <span class="st">"Dro"</span><span class="op">)</span>    |&gt;
      <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">sub</a></span><span class="op">(</span>pat <span class="op">=</span> <span class="st">"^X$"</span>,    rep <span class="op">=</span> <span class="st">"Dro"</span><span class="op">)</span>    |&gt;
      <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">sub</a></span><span class="op">(</span>pat <span class="op">=</span> <span class="st">"^Y$"</span>,    rep <span class="op">=</span> <span class="st">"Dro"</span><span class="op">)</span>   

    <span class="va">DF</span><span class="op">$</span><span class="va">query</span> <span class="op">&lt;-</span> <span class="va">DF</span><span class="op">$</span><span class="va">score</span> <span class="op">&lt;-</span> <span class="va">DF</span><span class="op">$</span><span class="va">rep</span> <span class="op">&lt;-</span> <span class="va">DF</span><span class="op">$</span><span class="va">repOvlp</span> <span class="op">&lt;-</span> <span class="va">DF</span><span class="op">$</span><span class="va">transcripts</span> <span class="op">&lt;-</span> <span class="cn">NULL</span>
    <span class="va">DF</span><span class="op">$</span><span class="va">dist</span>  <span class="op">&lt;-</span> <span class="fu">OikScrambling</span><span class="fu">:::</span><span class="fu"><a href="../reference/compDistance.html">compDistance</a></span><span class="op">(</span><span class="va">DF</span><span class="op">$</span><span class="va">name</span><span class="op">)</span>
    <span class="va">DF</span><span class="op">$</span><span class="va">distS</span> <span class="op">&lt;-</span> <span class="fu">OikScrambling</span><span class="fu">:::</span><span class="fu"><a href="../reference/compDistance.html">compDistance</a></span><span class="op">(</span><span class="va">DF</span><span class="op">$</span><span class="va">name</span>, short <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
    <span class="va">DF</span><span class="op">$</span><span class="va">genus</span> <span class="op">&lt;-</span> <span class="fu">OikScrambling</span><span class="fu">:::</span><span class="fu"><a href="../reference/compGenus.html">compGenus</a></span><span class="op">(</span><span class="va">DF</span><span class="op">$</span><span class="va">name</span><span class="op">)</span>
    <span class="va">DF</span><span class="op">$</span><span class="va">class</span> <span class="op">&lt;-</span> <span class="fu">OikScrambling</span><span class="fu">:::</span><span class="fu"><a href="../reference/compDistClass.html">compDistClass</a></span><span class="op">(</span><span class="va">DF</span><span class="op">$</span><span class="va">name</span><span class="op">)</span>
    <span class="va">DF</span>
  <span class="op">}</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/do.call.html" class="external-link">do.call</a></span><span class="op">(</span>what<span class="op">=</span><span class="va">rbind</span><span class="op">)</span>
<span class="op">}</span>

<span class="va">widths</span><span class="op">$</span><span class="va">gbs</span>     <span class="op">&lt;-</span> <span class="fu">makeWidthsDF</span><span class="op">(</span><span class="va">gbs</span><span class="op">)</span>     |&gt; <span class="fu">tibble</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/as_tibble.html" class="external-link">as_tibble</a></span><span class="op">(</span><span class="op">)</span>
<span class="va">widths</span><span class="op">$</span><span class="va">unal</span>    <span class="op">&lt;-</span> <span class="fu">makeWidthsDF</span><span class="op">(</span><span class="va">unal</span><span class="op">)</span>    |&gt; <span class="fu">tibble</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/as_tibble.html" class="external-link">as_tibble</a></span><span class="op">(</span><span class="op">)</span>
<span class="va">widths</span><span class="op">$</span><span class="va">coa</span>     <span class="op">&lt;-</span> <span class="fu">makeWidthsDF</span><span class="op">(</span><span class="va">coa</span><span class="op">)</span>     |&gt; <span class="fu">tibble</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/as_tibble.html" class="external-link">as_tibble</a></span><span class="op">(</span><span class="op">)</span>
<span class="va">widths</span><span class="op">$</span><span class="va">unmap</span>   <span class="op">&lt;-</span> <span class="fu">makeWidthsDF</span><span class="op">(</span><span class="va">unmap</span><span class="op">)</span>   |&gt; <span class="fu">tibble</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/as_tibble.html" class="external-link">as_tibble</a></span><span class="op">(</span><span class="op">)</span>
<span class="va">widths</span><span class="op">$</span><span class="va">unalMap</span> <span class="op">&lt;-</span> <span class="fu">makeWidthsDF</span><span class="op">(</span><span class="va">unalMap</span><span class="op">)</span> |&gt; <span class="fu">tibble</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/as_tibble.html" class="external-link">as_tibble</a></span><span class="op">(</span><span class="op">)</span>
<span class="va">widths</span><span class="op">$</span><span class="va">tra</span>     <span class="op">&lt;-</span> <span class="fu">makeWidthsDF</span><span class="op">(</span><span class="va">tra</span><span class="op">)</span>     |&gt; <span class="fu">tibble</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/as_tibble.html" class="external-link">as_tibble</a></span><span class="op">(</span><span class="op">)</span>
<span class="va">widths</span><span class="op">$</span><span class="va">tra2</span>    <span class="op">&lt;-</span> <span class="fu">makeWidthsDF</span><span class="op">(</span><span class="va">tra2</span><span class="op">)</span>    |&gt; <span class="fu">tibble</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/as_tibble.html" class="external-link">as_tibble</a></span><span class="op">(</span><span class="op">)</span>
<span class="va">widths</span><span class="op">$</span><span class="va">coa2</span>    <span class="op">&lt;-</span> <span class="fu">makeWidthsDF</span><span class="op">(</span><span class="va">coa2</span><span class="op">)</span>    |&gt; <span class="fu">tibble</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/as_tibble.html" class="external-link">as_tibble</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">names_table</span> <span class="op">&lt;-</span> <span class="fu">tibble</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/tibble.html" class="external-link">tibble</a></span><span class="op">(</span>
  pairnames <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Factor-class.html" class="external-link">levels</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="op">(</span><span class="va">widths</span><span class="op">$</span><span class="va">gbs</span><span class="op">$</span><span class="va">name</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>,
  class     <span class="op">=</span> <span class="fu">OikScrambling</span><span class="fu">:::</span><span class="fu"><a href="../reference/compDistClass.html">compDistClass</a></span><span class="op">(</span><span class="va">pairnames</span><span class="op">)</span>,
  distS     <span class="op">=</span> <span class="fu">OikScrambling</span><span class="fu">:::</span><span class="fu"><a href="../reference/compDistance.html">compDistance</a></span><span class="op">(</span><span class="va">pairnames</span>, short <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>,
  target    <span class="op">=</span> <span class="va">pairnames</span> |&gt; <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">sub</a></span><span class="op">(</span>pat<span class="op">=</span><span class="st">"_.*"</span>, rep<span class="op">=</span><span class="st">""</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="op">)</span>,
  query     <span class="op">=</span> <span class="va">pairnames</span> |&gt; <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">sub</a></span><span class="op">(</span>pat<span class="op">=</span><span class="st">".*_"</span>, rep<span class="op">=</span><span class="st">""</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="op">)</span>
<span class="op">)</span></code></pre></div>
</div>
<div class="section level1">
<h1 id="plot-all-width-distributions">Plot all width distributions<a class="anchor" aria-label="anchor" href="#plot-all-width-distributions"></a>
</h1>
<p>This is useful to verify by eye that collections of distributions in the same distance group resemble each other.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">gg_freq_poly</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">tibble</span><span class="op">)</span> <span class="op">{</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">tibble</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span><span class="va">width</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_histogram.html" class="external-link">geom_freqpoly</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_continuous.html" class="external-link">scale_x_log10</a></span><span class="op">(</span><span class="op">)</span>
<span class="op">}</span>

<span class="va">w_df</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/cbind.html" class="external-link">rbind</a></span><span class="op">(</span><span class="va">widths</span><span class="op">$</span><span class="va">gbs</span>     |&gt; <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="op">)</span>|&gt; <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>what <span class="op">=</span> <span class="st">"aligned"</span><span class="op">)</span>,
                   <span class="va">widths</span><span class="op">$</span><span class="va">unal</span>    |&gt; <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="op">)</span>|&gt; <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>what <span class="op">=</span> <span class="st">"unaligned"</span><span class="op">)</span>,
                   <span class="va">widths</span><span class="op">$</span><span class="va">coa</span>     |&gt; <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="op">)</span>|&gt; <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>what <span class="op">=</span> <span class="st">"mapped"</span><span class="op">)</span>,
                   <span class="va">widths</span><span class="op">$</span><span class="va">unalMap</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="op">)</span>|&gt; <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>what <span class="op">=</span> <span class="st">"mapped unaligned"</span><span class="op">)</span>,
                   <span class="va">widths</span><span class="op">$</span><span class="va">unmap</span>   |&gt; <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="op">)</span>|&gt; <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>what <span class="op">=</span> <span class="st">"unmaped"</span><span class="op">)</span>,
                   <span class="va">widths</span><span class="op">$</span><span class="va">coa2</span>    |&gt; <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="op">)</span>|&gt; <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>what <span class="op">=</span> <span class="st">"doublecoalesced"</span><span class="op">)</span><span class="op">)</span>


<span class="va">widths_all_tibble</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/cbind.html" class="external-link">rbind</a></span><span class="op">(</span><span class="va">widths</span><span class="op">$</span><span class="va">gbs</span> |&gt; <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>what <span class="op">=</span> <span class="st">"aligned"</span><span class="op">)</span>,
                       <span class="va">widths</span><span class="op">$</span><span class="va">unal</span>    |&gt; <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>what <span class="op">=</span> <span class="st">"unaligned"</span><span class="op">)</span>,
                       <span class="va">widths</span><span class="op">$</span><span class="va">coa</span>     |&gt; <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>what <span class="op">=</span> <span class="st">"mapped"</span><span class="op">)</span>,
                       <span class="va">widths</span><span class="op">$</span><span class="va">unalMap</span> |&gt; <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>what <span class="op">=</span> <span class="st">"mapped unaligned"</span><span class="op">)</span>,
                       <span class="va">widths</span><span class="op">$</span><span class="va">unmap</span>   |&gt; <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>what <span class="op">=</span> <span class="st">"unmaped"</span><span class="op">)</span>,
                       <span class="va">widths</span><span class="op">$</span><span class="va">coa2</span>    |&gt; <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>what <span class="op">=</span> <span class="st">"doublecoalesced"</span><span class="op">)</span><span class="op">)</span>

<span class="va">p</span> <span class="op">&lt;-</span><span class="fu">gg_freq_poly</span><span class="op">(</span><span class="va">widths_all_tibble</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>col <span class="op">=</span> <span class="va">what</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Distribution of alignment width before and after collapsing"</span><span class="op">)</span>

<span class="co"># Showing all individual distributions, to demonstrate we can pool some.</span>
<span class="va">p</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html" class="external-link">facet_wrap</a></span><span class="op">(</span><span class="op">~</span><span class="va">class</span> <span class="op">+</span> <span class="va">genus</span> <span class="op">+</span> <span class="va">distS</span> <span class="op">+</span> <span class="va">name</span>, scales <span class="op">=</span> <span class="st">"free_y"</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.</span></code></pre>
<p><img src="RegionWidths_files/figure-html/width_after_processing_steps-1.png" width="1152"></p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Aggregating by distance class</span>
<span class="va">p_droso</span> <span class="op">&lt;-</span> <span class="fu">gg_freq_poly</span><span class="op">(</span><span class="va">widths_all_tibble</span> |&gt; <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">genus</span> <span class="op">==</span> <span class="st">"Drosophila"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>col <span class="op">=</span> <span class="va">what</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html" class="external-link">facet_wrap</a></span><span class="op">(</span><span class="op">~</span><span class="va">genus</span> <span class="op">+</span> <span class="va">class</span>, nrow <span class="op">=</span> <span class="fl">1</span>, scales <span class="op">=</span> <span class="st">"free_y"</span><span class="op">)</span>
<span class="va">p_ciona</span> <span class="op">&lt;-</span> <span class="fu">gg_freq_poly</span><span class="op">(</span><span class="va">widths_all_tibble</span> |&gt; <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">genus</span> <span class="op">==</span> <span class="st">"Ciona"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>col <span class="op">=</span> <span class="va">what</span><span class="op">)</span>      <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html" class="external-link">facet_wrap</a></span><span class="op">(</span><span class="op">~</span><span class="va">genus</span> <span class="op">+</span> <span class="va">class</span>, nrow <span class="op">=</span> <span class="fl">1</span>, scales <span class="op">=</span> <span class="st">"free_y"</span><span class="op">)</span>
<span class="va">p_oikop</span> <span class="op">&lt;-</span> <span class="fu">gg_freq_poly</span><span class="op">(</span><span class="va">widths_all_tibble</span> |&gt; <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">genus</span> <span class="op">==</span> <span class="st">"Oikopleura"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>col <span class="op">=</span> <span class="va">what</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html" class="external-link">facet_wrap</a></span><span class="op">(</span><span class="op">~</span><span class="va">genus</span> <span class="op">+</span> <span class="va">class</span>, nrow <span class="op">=</span> <span class="fl">1</span>, scales <span class="op">=</span> <span class="st">"free_y"</span><span class="op">)</span>

<span class="va">layout</span> <span class="op">&lt;-</span> <span class="st">"
AAAA
BBB#
CCC#
"</span>

<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://patchwork.data-imaginist.com" class="external-link">"patchwork"</a></span><span class="op">)</span>
<span class="op">(</span><span class="va">p_droso</span> <span class="op">/</span> <span class="va">p_ciona</span> <span class="op">/</span> <span class="va">p_oikop</span><span class="op">)</span> <span class="op">+</span>   <span class="fu"><a href="https://patchwork.data-imaginist.com/reference/plot_layout.html" class="external-link">plot_layout</a></span><span class="op">(</span>design <span class="op">=</span> <span class="va">layout</span>, guides <span class="op">=</span> <span class="st">'collect'</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.</span>
<span class="co">## `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.</span>
<span class="co">## `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.</span></code></pre>
<p><img src="RegionWidths_files/figure-html/width_after_processing_steps_by_class_and_genus-1.png" width="1152"></p>
</div>
<div class="section level1">
<h1 id="width-distribution-of-aligned-and-mapped-regions">Width distribution of aligned and mapped regions<a class="anchor" aria-label="anchor" href="#width-distribution-of-aligned-and-mapped-regions"></a>
</h1>
<div class="section level2">
<h2 id="aligned-regions">Aligned regions<a class="anchor" aria-label="anchor" href="#aligned-regions"></a>
</h2>
<div class="section level3">
<h3 id="whole-genomes">Whole genomes<a class="anchor" aria-label="anchor" href="#whole-genomes"></a>
</h3>
<p><em>Aligned</em> means “before coalescing” and <em>mapped</em> means “after coalescing”.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">gg_freq_poly</span><span class="op">(</span><span class="va">widths</span><span class="op">$</span><span class="va">gbs</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span><span class="va">width</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes_eval.html" class="external-link">after_stat</a></span><span class="op">(</span><span class="va">density</span><span class="op">)</span>, col<span class="op">=</span><span class="va">name</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html" class="external-link">facet_wrap</a></span><span class="op">(</span><span class="op">~</span><span class="va">genus</span> <span class="op">+</span> <span class="va">class</span>, scales <span class="op">=</span> <span class="st">"free_y"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Distribution of aligned region widths"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html" class="external-link">scale_color_manual</a></span><span class="op">(</span>values <span class="op">=</span> <span class="va">names_table</span><span class="op">$</span><span class="va">class</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.</span></code></pre>
<p><img src="RegionWidths_files/figure-html/width_before_coalescing-1.png" width="960"></p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">gg_freq_poly</span><span class="op">(</span><span class="va">widths</span><span class="op">$</span><span class="va">gbs</span> |&gt; <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">genus</span> <span class="op">==</span> <span class="st">"Oikopleura"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>col<span class="op">=</span><span class="va">name</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html" class="external-link">facet_wrap</a></span><span class="op">(</span><span class="op">~</span><span class="va">dist</span>, scales <span class="op">=</span> <span class="st">"free_y"</span>, nrow <span class="op">=</span> <span class="fl">1</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Distribution of aligned region widths"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html" class="external-link">scale_color_manual</a></span><span class="op">(</span>values <span class="op">=</span> <span class="va">names_table</span><span class="op">$</span><span class="va">query</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.</span></code></pre>
<p><img src="RegionWidths_files/figure-html/width_before_coalescing-2.png" width="960"></p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">gg_freq_poly</span><span class="op">(</span><span class="va">widths</span><span class="op">$</span><span class="va">gbs</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span><span class="va">width</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes_eval.html" class="external-link">after_stat</a></span><span class="op">(</span><span class="va">density</span><span class="op">)</span>, col<span class="op">=</span><span class="va">class</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html" class="external-link">facet_wrap</a></span><span class="op">(</span><span class="op">~</span><span class="va">genus</span>, scales <span class="op">=</span> <span class="st">"free_y"</span>, ncol <span class="op">=</span> <span class="fl">1</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Distribution of aligned region widths"</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.</span></code></pre>
<p><img src="RegionWidths_files/figure-html/width_before_coalescing2-1.png" width="576"></p>
</div>
<div class="section level3">
<h3 id="faceted-by-arms">Faceted by arms<a class="anchor" aria-label="anchor" href="#faceted-by-arms"></a>
</h3>
<p>The distribution of widths is not the same on long and short arms.</p>
<p>Note that for this reason, it would be unwise to filter by width on some operations such as coalescing.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">gg_freq_poly</span><span class="op">(</span><span class="va">widths</span><span class="op">$</span><span class="va">gbs</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>col<span class="op">=</span><span class="va">Arm</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html" class="external-link">facet_wrap</a></span><span class="op">(</span><span class="op">~</span><span class="va">chr</span>, scales <span class="op">=</span> <span class="st">"free_y"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Distribution of aligned region widths"</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.</span></code></pre>
<p><img src="RegionWidths_files/figure-html/widths_gbs_by_arm-1.png" width="700"></p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">gg_freq_poly</span><span class="op">(</span><span class="va">widths</span><span class="op">$</span><span class="va">gbs</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">subset</a></span><span class="op">(</span><span class="va">chr</span> <span class="op">==</span> <span class="st">"chr1"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>col<span class="op">=</span><span class="va">Arm</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html" class="external-link">facet_wrap</a></span><span class="op">(</span><span class="op">~</span><span class="va">name</span>, scales <span class="op">=</span> <span class="st">"free_y"</span>, ncol<span class="op">=</span><span class="fl">5</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Distribution of aligned region widths on chr1"</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.</span></code></pre>
<p><img src="RegionWidths_files/figure-html/widths_gbs_by_arm-2.png" width="700"></p>
<p>Width distribution per arm follows similar pattern on chr2 (first graph). Pairs at a similar evolutionary distance show similar profile, with the exception of Osa_Aom, where the number of alignments on short arms is higher. Nevertheless, the trend of widths being longer on long arms stay true, so we can average them in the plot below.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">gg_freq_poly</span><span class="op">(</span><span class="va">widths</span><span class="op">$</span><span class="va">gbs</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">subset</a></span><span class="op">(</span><span class="va">chr</span> <span class="op"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"chr1"</span>, <span class="st">"chr2"</span>, <span class="st">"PAR"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>col<span class="op">=</span><span class="va">Arm</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html" class="external-link">facet_wrap</a></span><span class="op">(</span><span class="op">~</span><span class="va">dist</span>, scales <span class="op">=</span> <span class="st">"free_y"</span>, nrow<span class="op">=</span><span class="fl">1</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Distribution of alignment widths before collapsing, split by distance group"</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.</span></code></pre>
<p><img src="RegionWidths_files/figure-html/widths_gbs2_by_arm-1.png" width="700"></p>
</div>
</div>
<div class="section level2">
<h2 id="unaligned-regions">Unaligned regions<a class="anchor" aria-label="anchor" href="#unaligned-regions"></a>
</h2>
<p>Unaligned regions show no difference of width between arms, except perhaps at the “same population” level.</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">gg_freq_poly</span><span class="op">(</span><span class="va">widths</span><span class="op">$</span><span class="va">unal</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>col<span class="op">=</span><span class="va">Arm</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html" class="external-link">facet_wrap</a></span><span class="op">(</span><span class="op">~</span><span class="va">chr</span>, scales <span class="op">=</span> <span class="st">"free_y"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Distribution of unaligned region widths"</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.</span></code></pre>
<p><img src="RegionWidths_files/figure-html/widths_unaligned-1.png" width="700"></p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">gg_freq_poly</span><span class="op">(</span><span class="va">widths</span><span class="op">$</span><span class="va">unal</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">subset</a></span><span class="op">(</span><span class="va">chr</span> <span class="op">==</span> <span class="st">"chr1"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>col<span class="op">=</span><span class="va">Arm</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html" class="external-link">facet_wrap</a></span><span class="op">(</span><span class="op">~</span><span class="va">name</span>, scales <span class="op">=</span> <span class="st">"free_y"</span>, ncol<span class="op">=</span><span class="fl">5</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Distribution of unaligned region widths"</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.</span></code></pre>
<p><img src="RegionWidths_files/figure-html/widths_unaligned-2.png" width="700"></p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">gg_freq_poly</span><span class="op">(</span><span class="va">widths</span><span class="op">$</span><span class="va">unal</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">subset</a></span><span class="op">(</span><span class="va">chr</span> <span class="op"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"chr1"</span>, <span class="st">"chr2"</span>, <span class="st">"PAR"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>col<span class="op">=</span><span class="va">Arm</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html" class="external-link">facet_wrap</a></span><span class="op">(</span><span class="op">~</span><span class="va">dist</span>, scales <span class="op">=</span> <span class="st">"free_y"</span>, ncol<span class="op">=</span><span class="fl">1</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Distribution of unaligned region widths"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_abline.html" class="external-link">geom_vline</a></span><span class="op">(</span>xintercept <span class="op">=</span> <span class="fl">300</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_continuous.html" class="external-link">scale_x_log10</a></span><span class="op">(</span><span class="st">"Width (vertical bar at 300 bp)"</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Scale for 'x' is already present. Adding another scale for 'x', which will</span>
<span class="co">## replace the existing scale.</span>
<span class="co">## `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.</span></code></pre>
<p><img src="RegionWidths_files/figure-html/widths_unaligned-3.png" width="700"></p>
<p>Interestingly, only in Oikopleura do the width distributions from pairs of different distance peak at the same size. Could that be because the genome is very compact?</p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">gg_freq_poly</span><span class="op">(</span><span class="va">widths</span><span class="op">$</span><span class="va">unal</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">subset</a></span><span class="op">(</span><span class="va">chr</span> <span class="op"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"chr1"</span>, <span class="st">"chr2"</span>, <span class="st">"PAR"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>col<span class="op">=</span><span class="va">name</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html" class="external-link">facet_wrap</a></span><span class="op">(</span><span class="op">~</span><span class="va">dist</span>, scales <span class="op">=</span> <span class="st">"free_y"</span>, ncol<span class="op">=</span><span class="fl">1</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Distribution of unaligned region widths"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_abline.html" class="external-link">geom_vline</a></span><span class="op">(</span>xintercept <span class="op">=</span> <span class="fl">300</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_continuous.html" class="external-link">scale_x_log10</a></span><span class="op">(</span><span class="st">"Width (vertical bar at 300 bp)"</span><span class="op">)</span>  <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html" class="external-link">scale_color_manual</a></span><span class="op">(</span>values <span class="op">=</span> <span class="va">names_table</span><span class="op">$</span><span class="va">query</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Scale for 'x' is already present. Adding another scale for 'x', which will</span>
<span class="co">## replace the existing scale.</span></code></pre>
<pre><code><span class="co">## `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.</span></code></pre>
<p><img src="RegionWidths_files/figure-html/widths_unaligned2-1.png" width="700"></p>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">gg_freq_poly</span><span class="op">(</span><span class="va">widths</span><span class="op">$</span><span class="va">unal</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>col<span class="op">=</span><span class="va">class</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html" class="external-link">facet_wrap</a></span><span class="op">(</span><span class="op">~</span><span class="va">genus</span>, scales <span class="op">=</span> <span class="st">"free_y"</span>, ncol<span class="op">=</span><span class="fl">1</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Distribution of unaligned region widths"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_abline.html" class="external-link">geom_vline</a></span><span class="op">(</span>xintercept <span class="op">=</span> <span class="fl">300</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_continuous.html" class="external-link">scale_x_log10</a></span><span class="op">(</span><span class="st">"Width (vertical bar at 300 bp)"</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Scale for 'x' is already present. Adding another scale for 'x', which will</span>
<span class="co">## replace the existing scale.</span>
<span class="co">## `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.</span></code></pre>
<p><img src="RegionWidths_files/figure-html/widths_unaligned2-2.png" width="700"></p>
</div>
<div class="section level2">
<h2 id="mapped-coalesced-regions">Mapped (coalesced) regions<a class="anchor" aria-label="anchor" href="#mapped-coalesced-regions"></a>
</h2>
<p>After coalescing, some width distribution become clearly bimodal. The distributions are hard to compare between genera and between evolutionary distance groups, and even within group for the “same population” <em>Oikopleura</em>.</p>
<p>The plots below display the data in details for the reader to make their opinion about the above statement.</p>
<div class="section level3">
<h3 id="whole-data">Whole data<a class="anchor" aria-label="anchor" href="#whole-data"></a>
</h3>
<div class="section level4">
<h4 id="width-distributions-">Width distributions.<a class="anchor" aria-label="anchor" href="#width-distributions-"></a>
</h4>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Using after_stat(density) to make more meaningful comparisons between genus</span>
<span class="fu">gg_freq_poly</span><span class="op">(</span><span class="va">widths</span><span class="op">$</span><span class="va">coa</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span><span class="va">width</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes_eval.html" class="external-link">after_stat</a></span><span class="op">(</span><span class="va">density</span><span class="op">)</span>, col<span class="op">=</span><span class="va">name</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html" class="external-link">facet_wrap</a></span><span class="op">(</span><span class="op">~</span><span class="va">genus</span> <span class="op">+</span> <span class="va">class</span>, scales <span class="op">=</span> <span class="st">"free_y"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Distribution of mapped region widths"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html" class="external-link">scale_color_manual</a></span><span class="op">(</span>values <span class="op">=</span> <span class="va">names_table</span><span class="op">$</span><span class="va">query</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.</span></code></pre>
<p><img src="RegionWidths_files/figure-html/width_after_coalescing-1.png" width="1152"></p>
<p>Faceting by genus only. Why is North / North (“intermediate”) not bimodal?</p>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">gg_freq_poly</span><span class="op">(</span><span class="va">widths</span><span class="op">$</span><span class="va">coa</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span><span class="va">width</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes_eval.html" class="external-link">after_stat</a></span><span class="op">(</span><span class="va">density</span><span class="op">)</span>, col<span class="op">=</span><span class="va">class</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html" class="external-link">facet_wrap</a></span><span class="op">(</span><span class="op">~</span><span class="va">genus</span>, scales <span class="op">=</span> <span class="st">"free_y"</span>, ncol <span class="op">=</span> <span class="fl">1</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Distribution of mapped region widths"</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.</span></code></pre>
<p><img src="RegionWidths_files/figure-html/width_after_coalescing2-1.png" width="768"></p>
<p>What is in the first peak of the bimodal distributions ?</p>
</div>
<div class="section level4">
<h4 id="empirical-cumulative-distributions">Empirical cumulative distributions<a class="anchor" aria-label="anchor" href="#empirical-cumulative-distributions"></a>
</h4>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">widths</span><span class="op">$</span><span class="va">coa</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span><span class="va">width</span>, col <span class="op">=</span> <span class="va">genus</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/stat_ecdf.html" class="external-link">stat_ecdf</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_continuous.html" class="external-link">scale_x_log10</a></span><span class="op">(</span>limits <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">50</span>,<span class="fl">5e6</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html" class="external-link">facet_wrap</a></span><span class="op">(</span><span class="op">~</span><span class="va">class</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Warning: Removed 13 rows containing non-finite values (stat_ecdf).</span></code></pre>
<p><img src="RegionWidths_files/figure-html/width_after_coalescing_ecdf-1.png" width="1152"></p>
<div class="sourceCode" id="cb38"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">widths</span><span class="op">$</span><span class="va">coa</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span><span class="va">width</span>, col <span class="op">=</span> <span class="va">class</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/stat_ecdf.html" class="external-link">stat_ecdf</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_continuous.html" class="external-link">scale_x_log10</a></span><span class="op">(</span>limits <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">50</span>,<span class="fl">5e6</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html" class="external-link">facet_wrap</a></span><span class="op">(</span><span class="op">~</span><span class="va">genus</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Warning: Removed 13 rows containing non-finite values (stat_ecdf).</span></code></pre>
<p><img src="RegionWidths_files/figure-html/width_after_coalescing_ecdf-2.png" width="1152"></p>
</div>
<div class="section level4">
<h4 id="cumulative-sums-">Cumulative sums.<a class="anchor" aria-label="anchor" href="#cumulative-sums-"></a>
</h4>
<p>Note: there is a bug: the lines jump between replicates instead of plotting them separately</p>
<div class="sourceCode" id="cb40"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">widths</span><span class="op">$</span><span class="va">coa</span> |&gt;
  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/arrange.html" class="external-link">arrange</a></span><span class="op">(</span><span class="va">name</span>, <span class="va">width</span><span class="op">)</span> |&gt;
  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html" class="external-link">group_by</a></span><span class="op">(</span><span class="va">name</span><span class="op">)</span> |&gt;
  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>cumsum <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/cumsum.html" class="external-link">cumsum</a></span><span class="op">(</span><span class="va">width</span><span class="op">)</span><span class="op">)</span> |&gt;
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html" class="external-link">facet_wrap</a></span><span class="op">(</span><span class="op">~</span><span class="va">class</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">width</span>, y <span class="op">=</span> <span class="va">cumsum</span>, col <span class="op">=</span> <span class="va">genus</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_path.html" class="external-link">geom_line</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_continuous.html" class="external-link">scale_x_log10</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_continuous.html" class="external-link">scale_y_log10</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<p><img src="RegionWidths_files/figure-html/width_after_coalescing_cumsum-1.png" width="1152"></p>
<div class="sourceCode" id="cb41"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">widths</span><span class="op">$</span><span class="va">coa</span> |&gt;
  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/arrange.html" class="external-link">arrange</a></span><span class="op">(</span><span class="va">name</span>, <span class="va">width</span><span class="op">)</span> |&gt;
  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html" class="external-link">group_by</a></span><span class="op">(</span><span class="va">name</span><span class="op">)</span> |&gt;
  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>cumsum <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/cumsum.html" class="external-link">cumsum</a></span><span class="op">(</span><span class="va">width</span><span class="op">)</span><span class="op">)</span> |&gt;
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html" class="external-link">facet_wrap</a></span><span class="op">(</span><span class="op">~</span><span class="va">genus</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">width</span>, y <span class="op">=</span> <span class="va">cumsum</span>, col <span class="op">=</span> <span class="va">class</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_path.html" class="external-link">geom_line</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_continuous.html" class="external-link">scale_x_log10</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_continuous.html" class="external-link">scale_y_log10</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<p><img src="RegionWidths_files/figure-html/width_after_coalescing_cumsum-2.png" width="1152"></p>
</div>
</div>
<div class="section level3">
<h3 id="by-arm">By arm<a class="anchor" aria-label="anchor" href="#by-arm"></a>
</h3>
<p>When faceted by chromosome and arm type, bimodality is also observed. There is a small peak under 500 bp, found on all arms except YSR. The second peak shows two different distributions for long and short arms.</p>
<div class="sourceCode" id="cb42"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">gg_freq_poly</span><span class="op">(</span><span class="va">widths</span><span class="op">$</span><span class="va">coa</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>col<span class="op">=</span><span class="va">Arm</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html" class="external-link">facet_wrap</a></span><span class="op">(</span><span class="op">~</span><span class="va">chr</span>, scales <span class="op">=</span> <span class="st">"free_y"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Distribution of alignment widths after collapsing, split by arm."</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.</span></code></pre>
<p><img src="RegionWidths_files/figure-html/width_after_coalescing3-1.png" width="700"></p>
<div class="sourceCode" id="cb44"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">gg_freq_poly</span><span class="op">(</span><span class="va">widths</span><span class="op">$</span><span class="va">coa</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>col<span class="op">=</span><span class="va">flag</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html" class="external-link">facet_wrap</a></span><span class="op">(</span><span class="op">~</span><span class="va">chr</span>, scales <span class="op">=</span> <span class="st">"free_y"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Distribution of alignment widths after collapsing, split by flag."</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.</span></code></pre>
<p><img src="RegionWidths_files/figure-html/width_after_coalescing3-2.png" width="700"></p>
<div class="sourceCode" id="cb46"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">gg_freq_poly</span><span class="op">(</span><span class="va">widths</span><span class="op">$</span><span class="va">coa</span> <span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>col<span class="op">=</span><span class="va">Arm</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html" class="external-link">facet_wrap</a></span><span class="op">(</span><span class="op">~</span><span class="va">name</span>, scales <span class="op">=</span> <span class="st">"free_y"</span>, ncol<span class="op">=</span><span class="fl">5</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Distribution of alignment widths on chr1 after collapsing, split by genome pair"</span><span class="op">)</span><span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_abline.html" class="external-link">geom_vline</a></span><span class="op">(</span>xintercept <span class="op">=</span> <span class="fl">500</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.</span></code></pre>
<p><img src="RegionWidths_files/figure-html/width_after_coalescing3-3.png" width="700"></p>
<div class="sourceCode" id="cb48"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">gg_freq_poly</span><span class="op">(</span><span class="va">widths</span><span class="op">$</span><span class="va">coa</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">subset</a></span><span class="op">(</span><span class="va">chr</span> <span class="op"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"chr1"</span>, <span class="st">"chr2"</span>, <span class="st">"PAR"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>col<span class="op">=</span><span class="va">Arm</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html" class="external-link">facet_wrap</a></span><span class="op">(</span><span class="op">~</span><span class="va">dist</span>, scales <span class="op">=</span> <span class="st">"free_y"</span>, ncol<span class="op">=</span><span class="fl">1</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Distribution of alignment widths after collapsing, split by distance group"</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.</span></code></pre>
<p><img src="RegionWidths_files/figure-html/width_after_coalescing3-4.png" width="700"></p>
<div class="sourceCode" id="cb50"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">gg_freq_poly</span><span class="op">(</span><span class="va">widths</span><span class="op">$</span><span class="va">coa</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">subset</a></span><span class="op">(</span><span class="va">chr</span> <span class="op"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"chr1"</span>, <span class="st">"chr2"</span>, <span class="st">"PAR"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>col<span class="op">=</span><span class="va">flag</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html" class="external-link">facet_wrap</a></span><span class="op">(</span><span class="op">~</span><span class="va">class</span> <span class="op">+</span> <span class="va">flag</span>, scales <span class="op">=</span> <span class="st">"free_y"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Distribution of alignment widths after collapsing, split by distance group"</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.</span></code></pre>
<p><img src="RegionWidths_files/figure-html/width_after_coalescing3-5.png" width="700"></p>
<p>Therefore, if the first peak represented events that we do not want to consider breaking synteny (either alignment artefacts or insertion/transpositions of short mobile elements), we could re-coalesce once after removing the shortest intervals. We tried some width cutoffs on the coalescing step, but at 500 bp, the coverage of the genome by the resulting mapped regions is already decreasing (<code><a href="../articles/CoalescingCutoffs.html">vignette("CoalescingCutoffs", package = "OikScrambling")</a></code>).</p>
</div>
<div class="section level3">
<h3 id="non-coalesced-mapped-regions">Non-coalesced mapped regions<a class="anchor" aria-label="anchor" href="#non-coalesced-mapped-regions"></a>
</h3>
<p>The mapped regions that are found identical in the aligned set are those that were not coalesced… Are they the small peak ?</p>
<div class="sourceCode" id="cb52"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">widths</span><span class="op">$</span><span class="va">coa</span><span class="op">$</span><span class="va">nonCoa</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">coa</span>, \<span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="va">x</span><span class="op">$</span><span class="va">nonCoa</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/unlist.html" class="external-link">unlist</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level3">
<h3 id="focus-on-oki-north-comparisons">Focus on Oki – North comparisons<a class="anchor" aria-label="anchor" href="#focus-on-oki-north-comparisons"></a>
</h3>
<div class="sourceCode" id="cb53"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">p_all</span> <span class="op">&lt;-</span> <span class="fu">gg_freq_poly</span><span class="op">(</span><span class="va">widths</span><span class="op">$</span><span class="va">coa</span> |&gt; <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">dist</span> <span class="op">==</span> <span class="st">"Oki – North"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>col<span class="op">=</span><span class="va">name</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html" class="external-link">facet_wrap</a></span><span class="op">(</span><span class="op">~</span><span class="va">dist</span>, scales <span class="op">=</span> <span class="st">"free_y"</span>, nrow <span class="op">=</span> <span class="fl">1</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Distribution of mapped region widths (vline at 600)"</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_bw</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_abline.html" class="external-link">geom_vline</a></span><span class="op">(</span>xintercept <span class="op">=</span> <span class="fl">600</span><span class="op">)</span>

<span class="va">p_coa_map</span> <span class="op">&lt;-</span> <span class="fu">gg_freq_poly</span><span class="op">(</span><span class="va">widths</span><span class="op">$</span><span class="va">coa</span> |&gt; <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">dist</span> <span class="op">==</span> <span class="st">"Oki – North"</span>, <span class="va">Arm</span> <span class="op"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"short"</span>, <span class="st">"long"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>col<span class="op">=</span><span class="va">name</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"FALSE = collapsed, TRUE = non-collapsed"</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_bw</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_abline.html" class="external-link">geom_vline</a></span><span class="op">(</span>xintercept <span class="op">=</span> <span class="fl">600</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html" class="external-link">facet_wrap</a></span><span class="op">(</span><span class="op">~</span><span class="va">nonCoa</span> <span class="op">+</span> <span class="va">Arm</span><span class="op">)</span>

<span class="va">p_all</span> <span class="op">+</span> <span class="va">p_coa_map</span> <span class="op">+</span> <span class="fu"><a href="https://patchwork.data-imaginist.com/reference/plot_layout.html" class="external-link">plot_layout</a></span><span class="op">(</span>guides <span class="op">=</span> <span class="st">'collect'</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_bw</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.</span>
<span class="co">## `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.</span></code></pre>
<p><img src="RegionWidths_files/figure-html/mapped_region_widths_Oki_vs_North-1.png" width="700"></p>
<p>The short-width mapped peak is made of aligned regions that can not be coalesced.</p>
<p>We know from <code><a href="../articles/RepeatRegions.html">vignette("RepeatRegions", package = "OikScrambling")</a></code> that these short mapped regions are not repeats.</p>
</div>
<div class="section level3">
<h3 id="compound-pictures">Compound pictures<a class="anchor" aria-label="anchor" href="#compound-pictures"></a>
</h3>
<p>Let’s put into parallel “aligned”, “unaligned” and “mapped”</p>
<div class="sourceCode" id="cb55"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">p_ali</span> <span class="op">&lt;-</span> <span class="fu">gg_freq_poly</span><span class="op">(</span><span class="va">widths</span><span class="op">$</span><span class="va">gbs</span> |&gt; <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="op">!</span> <span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">genus</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span><span class="va">width</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes_eval.html" class="external-link">after_stat</a></span><span class="op">(</span><span class="va">density</span><span class="op">)</span>, col<span class="op">=</span><span class="va">class</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_bw</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html" class="external-link">facet_wrap</a></span><span class="op">(</span><span class="op">~</span><span class="va">genus</span>, scales <span class="op">=</span> <span class="st">"free_y"</span>, ncol <span class="op">=</span> <span class="fl">1</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Aligned"</span><span class="op">)</span>

<span class="va">p_una</span> <span class="op">&lt;-</span> <span class="fu">gg_freq_poly</span><span class="op">(</span><span class="va">widths</span><span class="op">$</span><span class="va">unal</span> |&gt; <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="op">!</span> <span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">genus</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span><span class="va">width</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes_eval.html" class="external-link">after_stat</a></span><span class="op">(</span><span class="va">density</span><span class="op">)</span>, col<span class="op">=</span><span class="va">class</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_bw</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html" class="external-link">facet_wrap</a></span><span class="op">(</span><span class="op">~</span><span class="va">genus</span>, scales <span class="op">=</span> <span class="st">"free_y"</span>, ncol <span class="op">=</span> <span class="fl">1</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Not aligned"</span><span class="op">)</span>

<span class="va">p_map</span> <span class="op">&lt;-</span> <span class="fu">gg_freq_poly</span><span class="op">(</span><span class="va">widths</span><span class="op">$</span><span class="va">coa</span> |&gt; <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="op">!</span> <span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">genus</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span><span class="va">width</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes_eval.html" class="external-link">after_stat</a></span><span class="op">(</span><span class="va">density</span><span class="op">)</span>, col<span class="op">=</span><span class="va">class</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_bw</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html" class="external-link">facet_wrap</a></span><span class="op">(</span><span class="op">~</span><span class="va">genus</span>, scales <span class="op">=</span> <span class="st">"free_y"</span>, ncol <span class="op">=</span> <span class="fl">1</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Mapped"</span><span class="op">)</span>

<span class="co"># Patchwork</span>
<span class="op">(</span><span class="va">p_ali</span> <span class="op">|</span> <span class="va">p_una</span> <span class="op">|</span> <span class="va">p_map</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://patchwork.data-imaginist.com/reference/plot_layout.html" class="external-link">plot_layout</a></span><span class="op">(</span>ncol <span class="op">=</span> <span class="fl">3</span>, guides <span class="op">=</span> <span class="st">'collect'</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_bw</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.</span>
<span class="co">## `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.</span>
<span class="co">## `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.</span></code></pre>
<p><img src="RegionWidths_files/figure-html/compound1-1.png" width="700"></p>
<div class="sourceCode" id="cb57"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">p_ali2</span> <span class="op">&lt;-</span> <span class="fu">gg_freq_poly</span><span class="op">(</span><span class="va">widths</span><span class="op">$</span><span class="va">coa</span> |&gt; <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">dist</span> <span class="op">==</span> <span class="st">"Oki – North"</span>, <span class="va">nonCoa</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span><span class="va">width</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes_eval.html" class="external-link">after_stat</a></span><span class="op">(</span><span class="va">density</span><span class="op">)</span>, col<span class="op">=</span><span class="va">name</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_bw</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_continuous.html" class="external-link">scale_x_log10</a></span><span class="op">(</span>limits <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">5e5</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Aligned (not collapsed)"</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Scale for 'x' is already present. Adding another scale for 'x', which will</span>
<span class="co">## replace the existing scale.</span></code></pre>
<div class="sourceCode" id="cb59"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">p_una2</span> <span class="op">&lt;-</span> <span class="fu">gg_freq_poly</span><span class="op">(</span><span class="va">widths</span><span class="op">$</span><span class="va">unmap</span> |&gt; <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">dist</span> <span class="op">==</span> <span class="st">"Oki – North"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span><span class="va">width</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes_eval.html" class="external-link">after_stat</a></span><span class="op">(</span><span class="va">density</span><span class="op">)</span>, col<span class="op">=</span><span class="va">name</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_bw</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_continuous.html" class="external-link">scale_x_log10</a></span><span class="op">(</span>limits <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">5e5</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Not mapped"</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Scale for 'x' is already present. Adding another scale for 'x', which will</span>
<span class="co">## replace the existing scale.</span></code></pre>
<div class="sourceCode" id="cb61"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">p_map2</span> <span class="op">&lt;-</span> <span class="fu">gg_freq_poly</span><span class="op">(</span><span class="va">widths</span><span class="op">$</span><span class="va">coa</span> |&gt; <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">dist</span> <span class="op">==</span> <span class="st">"Oki – North"</span>, <span class="op">!</span><span class="va">nonCoa</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span><span class="va">width</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes_eval.html" class="external-link">after_stat</a></span><span class="op">(</span><span class="va">density</span><span class="op">)</span>, col<span class="op">=</span><span class="va">name</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_bw</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_continuous.html" class="external-link">scale_x_log10</a></span><span class="op">(</span>limits <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">5e5</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Mapped"</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Scale for 'x' is already present. Adding another scale for 'x', which will</span>
<span class="co">## replace the existing scale.</span></code></pre>
<div class="sourceCode" id="cb63"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">p_unalmap2</span> <span class="op">&lt;-</span> <span class="fu">gg_freq_poly</span><span class="op">(</span><span class="va">widths</span><span class="op">$</span><span class="va">unalMap</span> |&gt; <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">dist</span> <span class="op">==</span> <span class="st">"Oki – North"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span><span class="va">width</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes_eval.html" class="external-link">after_stat</a></span><span class="op">(</span><span class="va">density</span><span class="op">)</span>, col<span class="op">=</span><span class="va">name</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_bw</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_continuous.html" class="external-link">scale_x_log10</a></span><span class="op">(</span>limits <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">5e5</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Double gaps"</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Scale for 'x' is already present. Adding another scale for 'x', which will</span>
<span class="co">## replace the existing scale.</span></code></pre>
<div class="sourceCode" id="cb65"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Patchwork</span>
<span class="op">(</span><span class="va">p_ali2</span> <span class="op">|</span> <span class="va">p_una2</span> <span class="op">|</span> <span class="va">p_map2</span> <span class="op">|</span> <span class="va">p_unalmap2</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://patchwork.data-imaginist.com/reference/plot_layout.html" class="external-link">plot_layout</a></span><span class="op">(</span>ncol <span class="op">=</span> <span class="fl">4</span>, guides <span class="op">=</span> <span class="st">'collect'</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_bw</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.</span></code></pre>
<pre><code><span class="co">## Warning: Removed 16 row(s) containing missing values (geom_path).</span></code></pre>
<pre><code><span class="co">## `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.</span></code></pre>
<pre><code><span class="co">## Warning: Removed 16 row(s) containing missing values (geom_path).</span></code></pre>
<pre><code><span class="co">## `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.</span></code></pre>
<pre><code><span class="co">## Warning: Removed 16 row(s) containing missing values (geom_path).</span></code></pre>
<pre><code><span class="co">## `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.</span></code></pre>
<pre><code><span class="co">## Warning: Removed 16 row(s) containing missing values (geom_path).</span></code></pre>
<p><img src="RegionWidths_files/figure-html/compound2-1.png" width="700"></p>
</div>
</div>
<div class="section level2">
<h2 id="unmapped-regions">Unmapped regions<a class="anchor" aria-label="anchor" href="#unmapped-regions"></a>
</h2>
<p>Unmapped regions are a bit longer on the short arm in Oki – North comparisons…</p>
<div class="sourceCode" id="cb74"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">gg_freq_poly</span><span class="op">(</span><span class="va">widths</span><span class="op">$</span><span class="va">unmap</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>col<span class="op">=</span><span class="va">Arm</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html" class="external-link">facet_wrap</a></span><span class="op">(</span><span class="op">~</span><span class="va">chr</span>, scales <span class="op">=</span> <span class="st">"free_y"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Distribution of unmapped regions width after collapsing, split by arm."</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.</span></code></pre>
<p><img src="RegionWidths_files/figure-html/width_of_unmapped_regions-1.png" width="700"></p>
<div class="sourceCode" id="cb76"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">gg_freq_poly</span><span class="op">(</span><span class="va">widths</span><span class="op">$</span><span class="va">unmap</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>col<span class="op">=</span><span class="va">Arm</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html" class="external-link">facet_wrap</a></span><span class="op">(</span><span class="op">~</span><span class="va">name</span>, scales <span class="op">=</span> <span class="st">"free_y"</span>, ncol<span class="op">=</span><span class="fl">5</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Distribution of unmapped regions width after collapsing, split by genome pair"</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.</span></code></pre>
<p><img src="RegionWidths_files/figure-html/width_of_unmapped_regions-2.png" width="700"></p>
<div class="sourceCode" id="cb78"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">gg_freq_poly</span><span class="op">(</span><span class="va">widths</span><span class="op">$</span><span class="va">unmap</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">subset</a></span><span class="op">(</span><span class="va">chr</span> <span class="op"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"chr1"</span>, <span class="st">"chr2"</span>, <span class="st">"PAR"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>col<span class="op">=</span><span class="va">Arm</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html" class="external-link">facet_wrap</a></span><span class="op">(</span><span class="op">~</span><span class="va">dist</span>, scales <span class="op">=</span> <span class="st">"free_y"</span>, nrow<span class="op">=</span><span class="fl">1</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Distribution of unmapped region widths after collapsing, split by distance group"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_abline.html" class="external-link">geom_vline</a></span><span class="op">(</span>xintercept <span class="op">=</span> <span class="fl">300</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_continuous.html" class="external-link">scale_x_log10</a></span><span class="op">(</span><span class="st">"Width (vertical bar at 300 bp)"</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Scale for 'x' is already present. Adding another scale for 'x', which will</span>
<span class="co">## replace the existing scale.</span>
<span class="co">## `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.</span></code></pre>
<p><img src="RegionWidths_files/figure-html/width_of_unmapped_regions-3.png" width="700"></p>
</div>
<div class="section level2">
<h2 id="mapped-unaligned-regions">Mapped unaligned regions<a class="anchor" aria-label="anchor" href="#mapped-unaligned-regions"></a>
</h2>
<p>Note the short peak when comparing pairs within the same population.</p>
<div class="sourceCode" id="cb80"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">gg_freq_poly</span><span class="op">(</span><span class="va">widths</span><span class="op">$</span><span class="va">unalMap</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>col<span class="op">=</span><span class="va">Arm</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html" class="external-link">facet_wrap</a></span><span class="op">(</span><span class="op">~</span><span class="va">chr</span>, scales <span class="op">=</span> <span class="st">"free_y"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Distribution of widths of mapped unaligned regions, split by arm."</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.</span></code></pre>
<p><img src="RegionWidths_files/figure-html/width_of_unAlmapped_regions-1.png" width="700"></p>
<div class="sourceCode" id="cb82"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">gg_freq_poly</span><span class="op">(</span><span class="va">widths</span><span class="op">$</span><span class="va">unalMap</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>col<span class="op">=</span><span class="va">Arm</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html" class="external-link">facet_wrap</a></span><span class="op">(</span><span class="op">~</span><span class="va">name</span>, scales <span class="op">=</span> <span class="st">"free_y"</span>, ncol<span class="op">=</span><span class="fl">5</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Distribution of widths of mapped unaligned regions, split by genome pair"</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.</span></code></pre>
<p><img src="RegionWidths_files/figure-html/width_of_unAlmapped_regions-2.png" width="700"></p>
<div class="sourceCode" id="cb84"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">gg_freq_poly</span><span class="op">(</span><span class="va">widths</span><span class="op">$</span><span class="va">unalMap</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">subset</a></span><span class="op">(</span><span class="va">chr</span> <span class="op"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"chr1"</span>, <span class="st">"chr2"</span>, <span class="st">"PAR"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>col<span class="op">=</span><span class="va">Arm</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html" class="external-link">facet_wrap</a></span><span class="op">(</span><span class="op">~</span><span class="va">dist</span>, scales <span class="op">=</span> <span class="st">"free_y"</span>, nrow<span class="op">=</span><span class="fl">1</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Distribution of widths of mapped unaligned regions, split by distance group"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_abline.html" class="external-link">geom_vline</a></span><span class="op">(</span>xintercept <span class="op">=</span> <span class="fl">300</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_continuous.html" class="external-link">scale_x_log10</a></span><span class="op">(</span><span class="st">"Width (vertical bar at 300 bp)"</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Scale for 'x' is already present. Adding another scale for 'x', which will</span>
<span class="co">## replace the existing scale.</span>
<span class="co">## `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.</span></code></pre>
<p><img src="RegionWidths_files/figure-html/width_of_unAlmapped_regions-3.png" width="700"></p>
</div>
<div class="section level2">
<h2 id="translocated-regions">Translocated regions<a class="anchor" aria-label="anchor" href="#translocated-regions"></a>
</h2>
<div class="section level3">
<h3 id="whole-region">Whole region<a class="anchor" aria-label="anchor" href="#whole-region"></a>
</h3>
<p>Reminder: here we measure the distance between the two colinear pairs that are separated by the translocated pair.</p>
<div class="sourceCode" id="cb86"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">gg_freq_poly</span><span class="op">(</span><span class="va">widths</span><span class="op">$</span><span class="va">tra</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>col<span class="op">=</span><span class="va">Arm</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html" class="external-link">facet_wrap</a></span><span class="op">(</span><span class="op">~</span><span class="va">chr</span>, scales <span class="op">=</span> <span class="st">"free_y"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Distribution of widths of translocated regions, split by arm."</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.</span></code></pre>
<p><img src="RegionWidths_files/figure-html/width_of_translocated_regions-1.png" width="700"></p>
<div class="sourceCode" id="cb88"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">gg_freq_poly</span><span class="op">(</span><span class="va">widths</span><span class="op">$</span><span class="va">tra</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>col<span class="op">=</span><span class="va">Arm</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html" class="external-link">facet_wrap</a></span><span class="op">(</span><span class="op">~</span><span class="va">name</span>, scales <span class="op">=</span> <span class="st">"free_y"</span>, ncol<span class="op">=</span><span class="fl">5</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Distribution of widths of translocated regions, split by genome pair"</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.</span></code></pre>
<p><img src="RegionWidths_files/figure-html/width_of_translocated_regions-2.png" width="700"></p>
<div class="sourceCode" id="cb90"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">gg_freq_poly</span><span class="op">(</span><span class="va">widths</span><span class="op">$</span><span class="va">tra</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">subset</a></span><span class="op">(</span><span class="va">chr</span> <span class="op"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"chr1"</span>, <span class="st">"chr2"</span>, <span class="st">"PAR"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>col<span class="op">=</span><span class="va">Arm</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html" class="external-link">facet_wrap</a></span><span class="op">(</span><span class="op">~</span><span class="va">dist</span>, scales <span class="op">=</span> <span class="st">"free_y"</span>, nrow<span class="op">=</span><span class="fl">1</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Distribution of widths of translocated regions, split by distance group"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_abline.html" class="external-link">geom_vline</a></span><span class="op">(</span>xintercept <span class="op">=</span> <span class="fl">300</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_continuous.html" class="external-link">scale_x_log10</a></span><span class="op">(</span><span class="st">"Width (vertical bar at 300 bp)"</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Scale for 'x' is already present. Adding another scale for 'x', which will</span>
<span class="co">## replace the existing scale.</span>
<span class="co">## `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.</span></code></pre>
<p><img src="RegionWidths_files/figure-html/width_of_translocated_regions-3.png" width="700"></p>
</div>
<div class="section level3">
<h3 id="whole-region-after-removal">Whole region, after removal<a class="anchor" aria-label="anchor" href="#whole-region-after-removal"></a>
</h3>
<p>Small peak in <em>“same pop”</em> disapears by design. Total number in the main peak also decreases, because they may pass the size filter on the <em>query</em> genome.</p>
<div class="sourceCode" id="cb92"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">gg_freq_poly</span><span class="op">(</span><span class="va">widths</span><span class="op">$</span><span class="va">tra2</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>col<span class="op">=</span><span class="va">Arm</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html" class="external-link">facet_wrap</a></span><span class="op">(</span><span class="op">~</span><span class="va">chr</span>, scales <span class="op">=</span> <span class="st">"free_y"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Distribution of widths of translocated regions (after removal), split by arm."</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.</span></code></pre>
<p><img src="RegionWidths_files/figure-html/width_of_translocated_regions_rem-1.png" width="700"></p>
<div class="sourceCode" id="cb94"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">gg_freq_poly</span><span class="op">(</span><span class="va">widths</span><span class="op">$</span><span class="va">tra2</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>col<span class="op">=</span><span class="va">Arm</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html" class="external-link">facet_wrap</a></span><span class="op">(</span><span class="op">~</span><span class="va">name</span>, scales <span class="op">=</span> <span class="st">"free_y"</span>, ncol<span class="op">=</span><span class="fl">5</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Distribution of widths of translocated regions (after removal), split by genome pair"</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.</span></code></pre>
<p><img src="RegionWidths_files/figure-html/width_of_translocated_regions_rem-2.png" width="700"></p>
<div class="sourceCode" id="cb96"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">gg_freq_poly</span><span class="op">(</span><span class="va">widths</span><span class="op">$</span><span class="va">tra2</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">subset</a></span><span class="op">(</span><span class="va">chr</span> <span class="op"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"chr1"</span>, <span class="st">"chr2"</span>, <span class="st">"PAR"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>col<span class="op">=</span><span class="va">Arm</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html" class="external-link">facet_wrap</a></span><span class="op">(</span><span class="op">~</span><span class="va">dist</span>, scales <span class="op">=</span> <span class="st">"free_y"</span>, nrow<span class="op">=</span><span class="fl">1</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Distribution of widths of translocated regions (after removal), split by distance group"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_abline.html" class="external-link">geom_vline</a></span><span class="op">(</span>xintercept <span class="op">=</span> <span class="fl">300</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_continuous.html" class="external-link">scale_x_log10</a></span><span class="op">(</span><span class="st">"Width (vertical bar at 300 bp)"</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Scale for 'x' is already present. Adding another scale for 'x', which will</span>
<span class="co">## replace the existing scale.</span>
<span class="co">## `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.</span></code></pre>
<p><img src="RegionWidths_files/figure-html/width_of_translocated_regions_rem-3.png" width="700"></p>
</div>
<div class="section level3">
<h3 id="aligned-before-removal">Aligned, before removal<a class="anchor" aria-label="anchor" href="#aligned-before-removal"></a>
</h3>
<div class="sourceCode" id="cb98"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">widths</span><span class="op">$</span><span class="va">tra_coa_aln</span>   <span class="op">&lt;-</span> <span class="fu">makeWidthsDF</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">coa</span>, \<span class="op">(</span><span class="va">gb</span><span class="op">)</span> <span class="va">gb</span> |&gt; <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/flagTranslocations.html" class="external-link">flagTranslocations</a></span><span class="op">(</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/filterTranslocations.html" class="external-link">filterTranslocations</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> |&gt; <span class="fu">tibble</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/as_tibble.html" class="external-link">as_tibble</a></span><span class="op">(</span><span class="op">)</span>

<span class="fu">gg_freq_poly</span><span class="op">(</span><span class="va">widths</span><span class="op">$</span><span class="va">tra_coa_aln</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>col<span class="op">=</span><span class="va">Arm</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html" class="external-link">facet_wrap</a></span><span class="op">(</span><span class="op">~</span><span class="va">chr</span>, scales <span class="op">=</span> <span class="st">"free_y"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Distribution of widths of translocated regions (aligned), split by arm."</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.</span></code></pre>
<p><img src="RegionWidths_files/figure-html/width_of_translocated_regions_coa_aligned-1.png" width="700"></p>
<div class="sourceCode" id="cb100"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">gg_freq_poly</span><span class="op">(</span><span class="va">widths</span><span class="op">$</span><span class="va">tra_coa_aln</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>col<span class="op">=</span><span class="va">Arm</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html" class="external-link">facet_wrap</a></span><span class="op">(</span><span class="op">~</span><span class="va">name</span>, scales <span class="op">=</span> <span class="st">"free_y"</span>, ncol<span class="op">=</span><span class="fl">5</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Distribution of widths of translocated regions (aligned), split by genome pair"</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.</span></code></pre>
<p><img src="RegionWidths_files/figure-html/width_of_translocated_regions_coa_aligned-2.png" width="700"></p>
<div class="sourceCode" id="cb102"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">gg_freq_poly</span><span class="op">(</span><span class="va">widths</span><span class="op">$</span><span class="va">tra_coa_aln</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">subset</a></span><span class="op">(</span><span class="va">chr</span> <span class="op"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"chr1"</span>, <span class="st">"chr2"</span>, <span class="st">"PAR"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>col<span class="op">=</span><span class="va">Arm</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html" class="external-link">facet_wrap</a></span><span class="op">(</span><span class="op">~</span><span class="va">dist</span>, scales <span class="op">=</span> <span class="st">"free_y"</span>, nrow<span class="op">=</span><span class="fl">1</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Distribution of widths of translocated regions (aligned), split by distance group"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_abline.html" class="external-link">geom_vline</a></span><span class="op">(</span>xintercept <span class="op">=</span> <span class="fl">300</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_continuous.html" class="external-link">scale_x_log10</a></span><span class="op">(</span><span class="st">"Width (vertical bar at 300 bp)"</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Scale for 'x' is already present. Adding another scale for 'x', which will</span>
<span class="co">## replace the existing scale.</span>
<span class="co">## `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.</span></code></pre>
<p><img src="RegionWidths_files/figure-html/width_of_translocated_regions_coa_aligned-3.png" width="700"></p>
</div>
<div class="section level3">
<h3 id="aligned-after-removal">Aligned, after removal<a class="anchor" aria-label="anchor" href="#aligned-after-removal"></a>
</h3>
<div class="sourceCode" id="cb104"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">widths</span><span class="op">$</span><span class="va">tra_coa2_aln</span>   <span class="op">&lt;-</span> <span class="fu">makeWidthsDF</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">coa2</span>, \<span class="op">(</span><span class="va">gb</span><span class="op">)</span> <span class="va">gb</span> |&gt; <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/flagTranslocations.html" class="external-link">flagTranslocations</a></span><span class="op">(</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/filterTranslocations.html" class="external-link">filterTranslocations</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> |&gt; <span class="fu">tibble</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/as_tibble.html" class="external-link">as_tibble</a></span><span class="op">(</span><span class="op">)</span>

<span class="fu">gg_freq_poly</span><span class="op">(</span><span class="va">widths</span><span class="op">$</span><span class="va">tra_coa2_aln</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>col<span class="op">=</span><span class="va">Arm</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html" class="external-link">facet_wrap</a></span><span class="op">(</span><span class="op">~</span><span class="va">chr</span>, scales <span class="op">=</span> <span class="st">"free_y"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Distribution of widths of translocated regions (aligned, after removal), split by arm."</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.</span></code></pre>
<p><img src="RegionWidths_files/figure-html/width_of_translocated_regions_coa2_aligned-1.png" width="700"></p>
<div class="sourceCode" id="cb106"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">gg_freq_poly</span><span class="op">(</span><span class="va">widths</span><span class="op">$</span><span class="va">tra_coa2_aln</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>col<span class="op">=</span><span class="va">Arm</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html" class="external-link">facet_wrap</a></span><span class="op">(</span><span class="op">~</span><span class="va">name</span>, scales <span class="op">=</span> <span class="st">"free_y"</span>, ncol<span class="op">=</span><span class="fl">5</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Distribution of widths of translocated regions (aligned, after removal), split by genome pair"</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.</span></code></pre>
<p><img src="RegionWidths_files/figure-html/width_of_translocated_regions_coa2_aligned-2.png" width="700"></p>
<div class="sourceCode" id="cb108"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">gg_freq_poly</span><span class="op">(</span><span class="va">widths</span><span class="op">$</span><span class="va">tra_coa2_aln</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">subset</a></span><span class="op">(</span><span class="va">chr</span> <span class="op"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"chr1"</span>, <span class="st">"chr2"</span>, <span class="st">"PAR"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>col<span class="op">=</span><span class="va">Arm</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html" class="external-link">facet_wrap</a></span><span class="op">(</span><span class="op">~</span><span class="va">dist</span>, scales <span class="op">=</span> <span class="st">"free_y"</span>, nrow<span class="op">=</span><span class="fl">1</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Distribution of widths of translocated regions (aligned, after removal), split by distance group"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_abline.html" class="external-link">geom_vline</a></span><span class="op">(</span>xintercept <span class="op">=</span> <span class="fl">300</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_continuous.html" class="external-link">scale_x_log10</a></span><span class="op">(</span><span class="st">"Width (vertical bar at 300 bp)"</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Scale for 'x' is already present. Adding another scale for 'x', which will</span>
<span class="co">## replace the existing scale.</span>
<span class="co">## `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.</span></code></pre>
<p><img src="RegionWidths_files/figure-html/width_of_translocated_regions_coa2_aligned-3.png" width="700"></p>
</div>
</div>
<div class="section level2">
<h2 id="double-coalesced-regions">Double-coalesced regions<a class="anchor" aria-label="anchor" href="#double-coalesced-regions"></a>
</h2>
<div class="sourceCode" id="cb110"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">gg_freq_poly</span><span class="op">(</span><span class="va">widths</span><span class="op">$</span><span class="va">coa2</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>col<span class="op">=</span><span class="va">Arm</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html" class="external-link">facet_wrap</a></span><span class="op">(</span><span class="op">~</span><span class="va">chr</span>, scales <span class="op">=</span> <span class="st">"free_y"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Distribution of alignment widths after double-collapsing, split by arm."</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.</span></code></pre>
<p><img src="RegionWidths_files/figure-html/re_coalesce-1.png" width="700"></p>
<div class="sourceCode" id="cb112"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">gg_freq_poly</span><span class="op">(</span><span class="va">widths</span><span class="op">$</span><span class="va">coa2</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>col<span class="op">=</span><span class="va">Arm</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html" class="external-link">facet_wrap</a></span><span class="op">(</span><span class="op">~</span><span class="va">name</span>, scales <span class="op">=</span> <span class="st">"free_y"</span>, ncol<span class="op">=</span><span class="fl">5</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Distribution of widthsof alignment widths after double-collapsing, split by genome pair"</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.</span></code></pre>
<p><img src="RegionWidths_files/figure-html/re_coalesce-2.png" width="700"></p>
<div class="sourceCode" id="cb114"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">gg_freq_poly</span><span class="op">(</span><span class="va">widths</span><span class="op">$</span><span class="va">coa2</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">subset</a></span><span class="op">(</span><span class="va">chr</span> <span class="op"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"chr1"</span>, <span class="st">"chr2"</span>, <span class="st">"PAR"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>col<span class="op">=</span><span class="va">Arm</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html" class="external-link">facet_wrap</a></span><span class="op">(</span><span class="op">~</span><span class="va">dist</span>, scales <span class="op">=</span> <span class="st">"free_y"</span>, nrow<span class="op">=</span><span class="fl">1</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Distribution of alignment widths after double-collapsing, split by distance group"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_abline.html" class="external-link">geom_vline</a></span><span class="op">(</span>xintercept <span class="op">=</span> <span class="fl">500</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_continuous.html" class="external-link">scale_x_log10</a></span><span class="op">(</span><span class="st">"Width (vertical bar at 500 bp)"</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Scale for 'x' is already present. Adding another scale for 'x', which will</span>
<span class="co">## replace the existing scale.</span>
<span class="co">## `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.</span></code></pre>
<p><img src="RegionWidths_files/figure-html/re_coalesce-3.png" width="700"></p>
</div>
</div>
<div class="section level1">
<h1 id="paired-comparison-of-widths-in-target-and-query-genomes-">Paired comparison of widths in target and query genomes.<a class="anchor" aria-label="anchor" href="#paired-comparison-of-widths-in-target-and-query-genomes-"></a>
</h1>
<div class="sourceCode" id="cb116"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">GBreaks2widthTibble</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">gb</span>, <span class="va">pair</span> <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span> <span class="op">{</span>
  <span class="va">df</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="va">gb</span><span class="op">)</span>
  <span class="kw">if</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/NULL.html" class="external-link">is.null</a></span><span class="op">(</span><span class="va">df</span><span class="op">$</span><span class="va">Arm</span><span class="op">)</span><span class="op">)</span>       <span class="va">df</span><span class="op">$</span><span class="va">Arm</span>        <span class="op">&lt;-</span> <span class="cn">NA</span>
  <span class="kw">if</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/NULL.html" class="external-link">is.null</a></span><span class="op">(</span><span class="va">df</span><span class="op">$</span><span class="va">rep</span><span class="op">)</span><span class="op">)</span>       <span class="va">df</span><span class="op">$</span><span class="va">rep</span>        <span class="op">&lt;-</span> <span class="cn">NA</span>
  <span class="kw">if</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/NULL.html" class="external-link">is.null</a></span><span class="op">(</span><span class="va">df</span><span class="op">$</span><span class="va">query.rep</span><span class="op">)</span><span class="op">)</span> <span class="va">df</span><span class="op">$</span><span class="va">query.rep</span>  <span class="op">&lt;-</span> <span class="cn">NA</span>
  <span class="fu">tibble</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/tibble.html" class="external-link">tibble</a></span><span class="op">(</span>target     <span class="op">=</span> <span class="va">df</span><span class="op">$</span><span class="va">width</span>,        query     <span class="op">=</span> <span class="va">df</span><span class="op">$</span><span class="va">query.width</span>,
                 target.rep <span class="op">=</span> <span class="op">!</span> <span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">df</span><span class="op">$</span><span class="va">rep</span><span class="op">)</span>, query.rep <span class="op">=</span> <span class="op">!</span> <span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">df</span><span class="op">$</span><span class="va">query.rep</span><span class="op">)</span>,  <span class="co"># TRUE if overlaps a repeat</span>
                 pair <span class="op">=</span> <span class="va">pair</span>, Arm <span class="op">=</span> <span class="va">df</span><span class="op">$</span><span class="va">Arm</span>,
                 dist <span class="op">=</span> <span class="fu">OikScrambling</span><span class="fu">:::</span><span class="fu"><a href="../reference/compDistance.html">compDistance</a></span><span class="op">(</span><span class="va">pair</span><span class="op">)</span><span class="op">)</span>
<span class="op">}</span>

<span class="va">gbs.w2w</span>     <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/do.call.html" class="external-link">do.call</a></span><span class="op">(</span><span class="va">rbind</span>, <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">gbs</span><span class="op">)</span>,      \<span class="op">(</span><span class="va">name</span><span class="op">)</span> <span class="fu">GBreaks2widthTibble</span><span class="op">(</span><span class="va">gbs</span><span class="op">[[</span><span class="va">name</span><span class="op">]</span><span class="op">]</span>,      pair <span class="op">=</span> <span class="va">name</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
<span class="va">unalMap.w2s</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/do.call.html" class="external-link">do.call</a></span><span class="op">(</span><span class="va">rbind</span>, <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">unalMap</span><span class="op">)</span>,  \<span class="op">(</span><span class="va">name</span><span class="op">)</span> <span class="fu">GBreaks2widthTibble</span><span class="op">(</span><span class="va">unalMap</span><span class="op">[[</span><span class="va">name</span><span class="op">]</span><span class="op">]</span>,  pair <span class="op">=</span> <span class="va">name</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
<span class="va">tra.w2s</span>     <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/do.call.html" class="external-link">do.call</a></span><span class="op">(</span><span class="va">rbind</span>, <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">tra</span><span class="op">)</span>,      \<span class="op">(</span><span class="va">name</span><span class="op">)</span> <span class="fu">GBreaks2widthTibble</span><span class="op">(</span><span class="va">tra</span><span class="op">[[</span><span class="va">name</span><span class="op">]</span><span class="op">]</span>,      pair <span class="op">=</span> <span class="va">name</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
<span class="va">tra2.w2s</span>    <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/do.call.html" class="external-link">do.call</a></span><span class="op">(</span><span class="va">rbind</span>, <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">tra2</span><span class="op">)</span>,     \<span class="op">(</span><span class="va">name</span><span class="op">)</span> <span class="fu">GBreaks2widthTibble</span><span class="op">(</span><span class="va">tra2</span><span class="op">[[</span><span class="va">name</span><span class="op">]</span><span class="op">]</span>,     pair <span class="op">=</span> <span class="va">name</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/cbind.html" class="external-link">rbind</a></span> <span class="op">(</span>
  aligned <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/table.html" class="external-link">table</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/paste.html" class="external-link">paste</a></span><span class="op">(</span>    <span class="va">gbs.w2w</span><span class="op">$</span><span class="va">target.rep</span>,     <span class="va">gbs.w2w</span><span class="op">$</span><span class="va">query.rep</span><span class="op">)</span><span class="op">)</span>,
  unalMap <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/table.html" class="external-link">table</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="va">unalMap.w2s</span><span class="op">$</span><span class="va">target.rep</span>, <span class="va">unalMap.w2s</span><span class="op">$</span><span class="va">query.rep</span><span class="op">)</span><span class="op">)</span>,
  transpo <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/table.html" class="external-link">table</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/paste.html" class="external-link">paste</a></span><span class="op">(</span>    <span class="va">tra.w2s</span><span class="op">$</span><span class="va">target.rep</span>,     <span class="va">tra.w2s</span><span class="op">$</span><span class="va">query.rep</span><span class="op">)</span><span class="op">)</span>
<span class="op">)</span></code></pre></div>
<pre><code><span class="co">##         FALSE FALSE FALSE TRUE TRUE FALSE TRUE TRUE</span>
<span class="co">## aligned      768648      41624      41943     52103</span>
<span class="co">## unalMap      553642      84463      82180     35726</span>
<span class="co">## transpo        7726       1973       2926      3016</span></code></pre>
<div class="sourceCode" id="cb118"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ggPlotW2W</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">tibble</span><span class="op">)</span> <span class="op">{</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">tibble</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span><span class="va">target</span>, <span class="va">query</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span>alpha <span class="op">=</span> <span class="fl">0.05</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_density_2d.html" class="external-link">geom_density_2d</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_continuous.html" class="external-link">scale_x_log10</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_continuous.html" class="external-link">scale_y_log10</a></span><span class="op">(</span><span class="op">)</span>
<span class="op">}</span>

<span class="fu">ggPlotW2W</span><span class="op">(</span><span class="va">gbs.w2w</span> |&gt; <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">Arm</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html" class="external-link">facet_wrap</a></span><span class="op">(</span><span class="op">~</span><span class="va">Arm</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Width of the target and query aligned regions"</span><span class="op">)</span></code></pre></div>
<p><img src="RegionWidths_files/figure-html/paired_width_analysis-1.png" width="700"></p>
<div class="sourceCode" id="cb119"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">ggPlotW2W</span><span class="op">(</span><span class="va">unalMap.w2s</span> |&gt; <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">Arm</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html" class="external-link">facet_wrap</a></span><span class="op">(</span><span class="op">~</span><span class="va">Arm</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Width of the target and query mapped unaligned regions"</span><span class="op">)</span></code></pre></div>
<p><img src="RegionWidths_files/figure-html/paired_width_analysis-2.png" width="700"></p>
<div class="sourceCode" id="cb120"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">ggPlotW2W</span><span class="op">(</span><span class="va">gbs.w2w</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html" class="external-link">facet_wrap</a></span><span class="op">(</span><span class="op">~</span><span class="va">pair</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Width of the target and query aligned regions"</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Warning: stat_contour(): Zero contours were generated</span></code></pre>
<pre><code><span class="co">## Warning in min(x): no non-missing arguments to min; returning Inf</span></code></pre>
<pre><code><span class="co">## Warning in max(x): no non-missing arguments to max; returning -Inf</span></code></pre>
<pre><code><span class="co">## Warning: stat_contour(): Zero contours were generated</span></code></pre>
<pre><code><span class="co">## Warning in min(x): no non-missing arguments to min; returning Inf</span></code></pre>
<pre><code><span class="co">## Warning in max(x): no non-missing arguments to max; returning -Inf</span></code></pre>
<pre><code><span class="co">## Warning: stat_contour(): Zero contours were generated</span></code></pre>
<pre><code><span class="co">## Warning in min(x): no non-missing arguments to min; returning Inf</span></code></pre>
<pre><code><span class="co">## Warning in max(x): no non-missing arguments to max; returning -Inf</span></code></pre>
<pre><code><span class="co">## Warning: stat_contour(): Zero contours were generated</span></code></pre>
<pre><code><span class="co">## Warning in min(x): no non-missing arguments to min; returning Inf</span></code></pre>
<pre><code><span class="co">## Warning in max(x): no non-missing arguments to max; returning -Inf</span></code></pre>
<pre><code><span class="co">## Warning: stat_contour(): Zero contours were generated</span></code></pre>
<pre><code><span class="co">## Warning in min(x): no non-missing arguments to min; returning Inf</span></code></pre>
<pre><code><span class="co">## Warning in max(x): no non-missing arguments to max; returning -Inf</span></code></pre>
<pre><code><span class="co">## Warning: stat_contour(): Zero contours were generated</span></code></pre>
<pre><code><span class="co">## Warning in min(x): no non-missing arguments to min; returning Inf</span></code></pre>
<pre><code><span class="co">## Warning in max(x): no non-missing arguments to max; returning -Inf</span></code></pre>
<pre><code><span class="co">## Warning: stat_contour(): Zero contours were generated</span></code></pre>
<pre><code><span class="co">## Warning in min(x): no non-missing arguments to min; returning Inf</span></code></pre>
<pre><code><span class="co">## Warning in max(x): no non-missing arguments to max; returning -Inf</span></code></pre>
<pre><code><span class="co">## Warning: stat_contour(): Zero contours were generated</span></code></pre>
<pre><code><span class="co">## Warning in min(x): no non-missing arguments to min; returning Inf</span></code></pre>
<pre><code><span class="co">## Warning in max(x): no non-missing arguments to max; returning -Inf</span></code></pre>
<pre><code><span class="co">## Warning: stat_contour(): Zero contours were generated</span></code></pre>
<pre><code><span class="co">## Warning in min(x): no non-missing arguments to min; returning Inf</span></code></pre>
<pre><code><span class="co">## Warning in max(x): no non-missing arguments to max; returning -Inf</span></code></pre>
<pre><code><span class="co">## Warning: stat_contour(): Zero contours were generated</span></code></pre>
<pre><code><span class="co">## Warning in min(x): no non-missing arguments to min; returning Inf</span></code></pre>
<pre><code><span class="co">## Warning in max(x): no non-missing arguments to max; returning -Inf</span></code></pre>
<pre><code><span class="co">## Warning: stat_contour(): Zero contours were generated</span></code></pre>
<pre><code><span class="co">## Warning in min(x): no non-missing arguments to min; returning Inf</span></code></pre>
<pre><code><span class="co">## Warning in max(x): no non-missing arguments to max; returning -Inf</span></code></pre>
<pre><code><span class="co">## Warning: stat_contour(): Zero contours were generated</span></code></pre>
<pre><code><span class="co">## Warning in min(x): no non-missing arguments to min; returning Inf</span></code></pre>
<pre><code><span class="co">## Warning in max(x): no non-missing arguments to max; returning -Inf</span></code></pre>
<pre><code><span class="co">## Warning: stat_contour(): Zero contours were generated</span></code></pre>
<pre><code><span class="co">## Warning in min(x): no non-missing arguments to min; returning Inf</span></code></pre>
<pre><code><span class="co">## Warning in max(x): no non-missing arguments to max; returning -Inf</span></code></pre>
<pre><code><span class="co">## Warning: stat_contour(): Zero contours were generated</span></code></pre>
<pre><code><span class="co">## Warning in min(x): no non-missing arguments to min; returning Inf</span></code></pre>
<pre><code><span class="co">## Warning in max(x): no non-missing arguments to max; returning -Inf</span></code></pre>
<pre><code><span class="co">## Warning: stat_contour(): Zero contours were generated</span></code></pre>
<pre><code><span class="co">## Warning in min(x): no non-missing arguments to min; returning Inf</span></code></pre>
<pre><code><span class="co">## Warning in max(x): no non-missing arguments to max; returning -Inf</span></code></pre>
<pre><code><span class="co">## Warning: stat_contour(): Zero contours were generated</span></code></pre>
<pre><code><span class="co">## Warning in min(x): no non-missing arguments to min; returning Inf</span></code></pre>
<pre><code><span class="co">## Warning in max(x): no non-missing arguments to max; returning -Inf</span></code></pre>
<pre><code><span class="co">## Warning: stat_contour(): Zero contours were generated</span></code></pre>
<pre><code><span class="co">## Warning in min(x): no non-missing arguments to min; returning Inf</span></code></pre>
<pre><code><span class="co">## Warning in max(x): no non-missing arguments to max; returning -Inf</span></code></pre>
<pre><code><span class="co">## Warning: stat_contour(): Zero contours were generated</span></code></pre>
<pre><code><span class="co">## Warning in min(x): no non-missing arguments to min; returning Inf</span></code></pre>
<pre><code><span class="co">## Warning in max(x): no non-missing arguments to max; returning -Inf</span></code></pre>
<pre><code><span class="co">## Warning: stat_contour(): Zero contours were generated</span></code></pre>
<pre><code><span class="co">## Warning in min(x): no non-missing arguments to min; returning Inf</span></code></pre>
<pre><code><span class="co">## Warning in max(x): no non-missing arguments to max; returning -Inf</span></code></pre>
<pre><code><span class="co">## Warning: stat_contour(): Zero contours were generated</span></code></pre>
<pre><code><span class="co">## Warning in min(x): no non-missing arguments to min; returning Inf</span></code></pre>
<pre><code><span class="co">## Warning in max(x): no non-missing arguments to max; returning -Inf</span></code></pre>
<pre><code><span class="co">## Warning: stat_contour(): Zero contours were generated</span></code></pre>
<pre><code><span class="co">## Warning in min(x): no non-missing arguments to min; returning Inf</span></code></pre>
<pre><code><span class="co">## Warning in max(x): no non-missing arguments to max; returning -Inf</span></code></pre>
<pre><code><span class="co">## Warning: stat_contour(): Zero contours were generated</span></code></pre>
<pre><code><span class="co">## Warning in min(x): no non-missing arguments to min; returning Inf</span></code></pre>
<pre><code><span class="co">## Warning in max(x): no non-missing arguments to max; returning -Inf</span></code></pre>
<p><img src="RegionWidths_files/figure-html/paired_width_analysis-3.png" width="700"></p>
<div class="sourceCode" id="cb187"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">ggPlotW2W</span><span class="op">(</span><span class="va">gbs.w2w</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html" class="external-link">facet_wrap</a></span><span class="op">(</span><span class="op">~</span><span class="va">dist</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Width of the target and query aligned regions"</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Warning: stat_contour(): Zero contours were generated</span></code></pre>
<pre><code><span class="co">## Warning in min(x): no non-missing arguments to min; returning Inf</span></code></pre>
<pre><code><span class="co">## Warning in max(x): no non-missing arguments to max; returning -Inf</span></code></pre>
<pre><code><span class="co">## Warning: stat_contour(): Zero contours were generated</span></code></pre>
<pre><code><span class="co">## Warning in min(x): no non-missing arguments to min; returning Inf</span></code></pre>
<pre><code><span class="co">## Warning in max(x): no non-missing arguments to max; returning -Inf</span></code></pre>
<pre><code><span class="co">## Warning: stat_contour(): Zero contours were generated</span></code></pre>
<pre><code><span class="co">## Warning in min(x): no non-missing arguments to min; returning Inf</span></code></pre>
<pre><code><span class="co">## Warning in max(x): no non-missing arguments to max; returning -Inf</span></code></pre>
<pre><code><span class="co">## Warning: stat_contour(): Zero contours were generated</span></code></pre>
<pre><code><span class="co">## Warning in min(x): no non-missing arguments to min; returning Inf</span></code></pre>
<pre><code><span class="co">## Warning in max(x): no non-missing arguments to max; returning -Inf</span></code></pre>
<pre><code><span class="co">## Warning: stat_contour(): Zero contours were generated</span></code></pre>
<pre><code><span class="co">## Warning in min(x): no non-missing arguments to min; returning Inf</span></code></pre>
<pre><code><span class="co">## Warning in max(x): no non-missing arguments to max; returning -Inf</span></code></pre>
<pre><code><span class="co">## Warning: stat_contour(): Zero contours were generated</span></code></pre>
<pre><code><span class="co">## Warning in min(x): no non-missing arguments to min; returning Inf</span></code></pre>
<pre><code><span class="co">## Warning in max(x): no non-missing arguments to max; returning -Inf</span></code></pre>
<p><img src="RegionWidths_files/figure-html/paired_width_analysis-4.png" width="700"></p>
<div class="sourceCode" id="cb206"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">ggPlotW2W</span><span class="op">(</span><span class="va">unalMap.w2s</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html" class="external-link">facet_wrap</a></span><span class="op">(</span><span class="op">~</span><span class="va">pair</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Width of the target and query mapped unaligned regions"</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Warning: stat_contour(): Zero contours were generated</span></code></pre>
<pre><code><span class="co">## Warning in min(x): no non-missing arguments to min; returning Inf</span></code></pre>
<pre><code><span class="co">## Warning in max(x): no non-missing arguments to max; returning -Inf</span></code></pre>
<pre><code><span class="co">## Warning: stat_contour(): Zero contours were generated</span></code></pre>
<pre><code><span class="co">## Warning in min(x): no non-missing arguments to min; returning Inf</span></code></pre>
<pre><code><span class="co">## Warning in max(x): no non-missing arguments to max; returning -Inf</span></code></pre>
<p><img src="RegionWidths_files/figure-html/paired_width_analysis-5.png" width="700"></p>
<div class="sourceCode" id="cb213"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">ggPlotW2W</span><span class="op">(</span><span class="va">tra.w2s</span> |&gt; <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">Arm</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html" class="external-link">facet_wrap</a></span><span class="op">(</span><span class="op">~</span><span class="va">pair</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Width of the target and query translocated regions"</span><span class="op">)</span></code></pre></div>
<p><img src="RegionWidths_files/figure-html/paired_width_analysis-6.png" width="700"></p>
<div class="sourceCode" id="cb214"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">ggPlotW2W</span><span class="op">(</span><span class="va">tra.w2s</span> |&gt; <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">Arm</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html" class="external-link">facet_wrap</a></span><span class="op">(</span><span class="op">~</span><span class="va">dist</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Width of the target and query translocated regions"</span><span class="op">)</span></code></pre></div>
<p><img src="RegionWidths_files/figure-html/paired_width_analysis-7.png" width="700"></p>
<div class="sourceCode" id="cb215"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">ggPlotW2W</span><span class="op">(</span><span class="va">tra2.w2s</span> |&gt; <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">Arm</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html" class="external-link">facet_wrap</a></span><span class="op">(</span><span class="op">~</span><span class="va">pair</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Width of the target and query translocated regions (after removal)"</span><span class="op">)</span></code></pre></div>
<p><img src="RegionWidths_files/figure-html/paired_width_analysis-8.png" width="700"></p>
<div class="sourceCode" id="cb216"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">ggPlotW2W</span><span class="op">(</span><span class="va">tra2.w2s</span> |&gt; <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">Arm</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html" class="external-link">facet_wrap</a></span><span class="op">(</span><span class="op">~</span><span class="va">dist</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Width of the target and query translocated regions (after removal)"</span><span class="op">)</span></code></pre></div>
<p><img src="RegionWidths_files/figure-html/paired_width_analysis-9.png" width="700"></p>
<div class="sourceCode" id="cb217"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">ggPlotW2W</span><span class="op">(</span><span class="va">gbs.w2w</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html" class="external-link">facet_wrap</a></span><span class="op">(</span><span class="op">~</span><span class="va">query.rep</span> <span class="op">+</span> <span class="va">target.rep</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Width of the target and query aligned regions, matching or not matching repeats"</span><span class="op">)</span></code></pre></div>
<p><img src="RegionWidths_files/figure-html/paired_width_analysis-10.png" width="700"></p>
<div class="sourceCode" id="cb218"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">ggPlotW2W</span><span class="op">(</span><span class="va">unalMap.w2s</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html" class="external-link">facet_wrap</a></span><span class="op">(</span><span class="op">~</span><span class="va">query.rep</span> <span class="op">+</span> <span class="va">target.rep</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Width of the target and query mapped unaligned regions, matching or not matching repeats"</span><span class="op">)</span></code></pre></div>
<p><img src="RegionWidths_files/figure-html/paired_width_analysis-11.png" width="700"></p>
<div class="sourceCode" id="cb219"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">ggPlotW2W</span><span class="op">(</span><span class="va">tra.w2s</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html" class="external-link">facet_wrap</a></span><span class="op">(</span><span class="op">~</span><span class="va">dist</span> <span class="op">+</span> <span class="va">query.rep</span> <span class="op">+</span> <span class="va">target.rep</span>, nrow <span class="op">=</span> <span class="fl">3</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Width of the target and query translocated regions, matching or not matching repeats"</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Warning: stat_contour(): Zero contours were generated</span></code></pre>
<pre><code><span class="co">## Warning in min(x): no non-missing arguments to min; returning Inf</span></code></pre>
<pre><code><span class="co">## Warning in max(x): no non-missing arguments to max; returning -Inf</span></code></pre>
<p><img src="RegionWidths_files/figure-html/paired_width_analysis-12.png" width="700"></p>
<div class="sourceCode" id="cb223"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">unalMap.w2s</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span><span class="va">target</span> <span class="op">/</span> <span class="va">query</span>, col <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="va">target.rep</span>, <span class="va">query.rep</span><span class="op">)</span> <span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_continuous.html" class="external-link">scale_x_log10</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_histogram.html" class="external-link">geom_freqpoly</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"In presence of a repeat, target width tends to be longer than query"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_abline.html" class="external-link">geom_vline</a></span><span class="op">(</span>xintercept <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.</span></code></pre>
<p><img src="RegionWidths_files/figure-html/paired_width_analysis-13.png" width="700"></p>
</div>
<div class="section level1">
<h1 id="structural-variants">Structural variants<a class="anchor" aria-label="anchor" href="#structural-variants"></a>
</h1>
<div class="section level2">
<h2 id="width-of-inverted-regions">Width of inverted regions<a class="anchor" aria-label="anchor" href="#width-of-inverted-regions"></a>
</h2>
<p>We know that we can detect more inversions after coalescing and removing translocations (<code><a href="../articles/Inversions.html">vignette("Inversions", package = "OikScrambling")</a></code>).</p>
<p>Many inversions are shorter than 1 kbp. Size may differ slightly between long and short arms.</p>
<div class="sourceCode" id="cb225"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">invs.gbs</span>  <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">gbs</span>,  <span class="kw">function</span><span class="op">(</span><span class="va">gb</span><span class="op">)</span> <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/filterInversions.html" class="external-link">filterInversions</a></span><span class="op">(</span><span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/flagInversions.html" class="external-link">flagInversions</a></span><span class="op">(</span><span class="va">gb</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/SimpleList-class.html" class="external-link">SimpleList</a></span><span class="op">(</span><span class="op">)</span>
<span class="va">invs.coa</span>  <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">coa</span>,  <span class="kw">function</span><span class="op">(</span><span class="va">gb</span><span class="op">)</span> <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/filterInversions.html" class="external-link">filterInversions</a></span><span class="op">(</span><span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/flagInversions.html" class="external-link">flagInversions</a></span><span class="op">(</span><span class="va">gb</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/SimpleList-class.html" class="external-link">SimpleList</a></span><span class="op">(</span><span class="op">)</span>
<span class="va">invs.coa2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">coa2</span>, <span class="kw">function</span><span class="op">(</span><span class="va">gb</span><span class="op">)</span> <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/filterInversions.html" class="external-link">filterInversions</a></span><span class="op">(</span><span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/flagInversions.html" class="external-link">flagInversions</a></span><span class="op">(</span><span class="va">gb</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/SimpleList-class.html" class="external-link">SimpleList</a></span><span class="op">(</span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">invs.gbs</span>,  <span class="va">length</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Oki_Osa Oki_Bar Oki_Kum Oki_Aom Oki_Nor Osa_Oki Osa_Bar Osa_Kum Osa_Aom Osa_Nor </span>
<span class="co">##     614     604      89     608     541     613     241     620      56     212 </span>
<span class="co">## Bar_Oki Bar_Osa Bar_Kum Bar_Aom Bar_Nor Ply_Ros Ply_Rob Ply_Sav Ply_Oki Rob_Ros </span>
<span class="co">##     597     237     608     237      19      42      58      16       0      50 </span>
<span class="co">## Rob_Ply Rob_Sav Rob_Oki Dme_Dbu Dme_Dsu Dme_Dya Dme_Dma </span>
<span class="co">##      58      14       0     697      20      13      12</span></code></pre>
<div class="sourceCode" id="cb227"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">invs.coa</span>,  <span class="va">length</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Oki_Osa Oki_Bar Oki_Kum Oki_Aom Oki_Nor Osa_Oki Osa_Bar Osa_Kum Osa_Aom Osa_Nor </span>
<span class="co">##     670     675     105     665     591     671     321     678      73     275 </span>
<span class="co">## Bar_Oki Bar_Osa Bar_Kum Bar_Aom Bar_Nor Ply_Ros Ply_Rob Ply_Sav Ply_Oki Rob_Ros </span>
<span class="co">##     667     319     686     319      24      51      71      92       0      65 </span>
<span class="co">## Rob_Ply Rob_Sav Rob_Oki Dme_Dbu Dme_Dsu Dme_Dya Dme_Dma </span>
<span class="co">##      70      92       0    1003      44      21      16</span></code></pre>
<div class="sourceCode" id="cb229"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">invs.coa2</span>, <span class="va">length</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Oki_Osa Oki_Bar Oki_Kum Oki_Aom Oki_Nor Osa_Oki Osa_Bar Osa_Kum Osa_Aom Osa_Nor </span>
<span class="co">##     704     705     122     695     611     706     330     708      79     279 </span>
<span class="co">## Bar_Oki Bar_Osa Bar_Kum Bar_Aom Bar_Nor Ply_Ros Ply_Rob Ply_Sav Ply_Oki Rob_Ros </span>
<span class="co">##     695     327     711     335      27      55      76      92       0      71 </span>
<span class="co">## Rob_Ply Rob_Sav Rob_Oki Dme_Dbu Dme_Dsu Dme_Dya Dme_Dma </span>
<span class="co">##      77      92       0    1003      45      23      16</span></code></pre>
<div class="sourceCode" id="cb231"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Sanity check that we detect inversions on both strands</span>
<span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">invs.coa2</span>, \<span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="op">{</span><span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Rle-class.html" class="external-link">decode</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/strand.html" class="external-link">strand</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">}</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">##   Oki_Osa Oki_Bar Oki_Kum Oki_Aom Oki_Nor Osa_Oki Osa_Bar Osa_Kum Osa_Aom</span>
<span class="co">## +     381     365      33     369     317     382     186     367      38</span>
<span class="co">## -     323     340      89     326     294     324     144     341      41</span>
<span class="co">## *       0       0       0       0       0       0       0       0       0</span>
<span class="co">##   Osa_Nor Bar_Oki Bar_Osa Bar_Kum Bar_Aom Bar_Nor Ply_Ros Ply_Rob Ply_Sav</span>
<span class="co">## +     147     354     182     365     164       7       2       6      51</span>
<span class="co">## -     132     341     145     346     171      20      53      70      41</span>
<span class="co">## *       0       0       0       0       0       0       0       0       0</span>
<span class="co">##   Ply_Oki Rob_Ros Rob_Ply Rob_Sav Rob_Oki Dme_Dbu Dme_Dsu Dme_Dya Dme_Dma</span>
<span class="co">## +       0       5       5      52       0     470      29      11       3</span>
<span class="co">## -       0      66      72      40       0     533      16      12      13</span>
<span class="co">## *       0       0       0       0       0       0       0       0       0</span></code></pre>
<div class="sourceCode" id="cb233"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">widths</span><span class="op">$</span><span class="va">invs.gbs</span>  <span class="op">&lt;-</span> <span class="fu">makeWidthsDF</span><span class="op">(</span><span class="va">invs.gbs</span><span class="op">)</span>  |&gt; <span class="fu">tibble</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/as_tibble.html" class="external-link">as_tibble</a></span><span class="op">(</span><span class="op">)</span>
<span class="va">widths</span><span class="op">$</span><span class="va">invs.coa</span>  <span class="op">&lt;-</span> <span class="fu">makeWidthsDF</span><span class="op">(</span><span class="va">invs.coa</span><span class="op">)</span>  |&gt; <span class="fu">tibble</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/as_tibble.html" class="external-link">as_tibble</a></span><span class="op">(</span><span class="op">)</span>
<span class="va">widths</span><span class="op">$</span><span class="va">invs.coa2</span> <span class="op">&lt;-</span> <span class="fu">makeWidthsDF</span><span class="op">(</span><span class="va">invs.coa2</span><span class="op">)</span> |&gt; <span class="fu">tibble</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/as_tibble.html" class="external-link">as_tibble</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb234"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">gg_freq_poly</span><span class="op">(</span><span class="va">widths</span><span class="op">$</span><span class="va">invs.gbs</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>col<span class="op">=</span><span class="va">Arm</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html" class="external-link">facet_wrap</a></span><span class="op">(</span><span class="op">~</span><span class="va">chr</span>, scales <span class="op">=</span> <span class="st">"free_y"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Distribution of alignment widths for inverted regions, using aligned regions, split by arm."</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.</span></code></pre>
<p><img src="RegionWidths_files/figure-html/inversion_width_aligned-1.png" width="700"></p>
<div class="sourceCode" id="cb236"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">gg_freq_poly</span><span class="op">(</span><span class="va">widths</span><span class="op">$</span><span class="va">invs.gbs</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>col<span class="op">=</span><span class="va">Arm</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html" class="external-link">facet_wrap</a></span><span class="op">(</span><span class="op">~</span><span class="va">name</span>, scales <span class="op">=</span> <span class="st">"free_y"</span>, ncol<span class="op">=</span><span class="fl">5</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Distribution of alignment widths for inverted regions, using aligned regions, split by genome pair"</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.</span></code></pre>
<p><img src="RegionWidths_files/figure-html/inversion_width_aligned-2.png" width="700"></p>
<div class="sourceCode" id="cb238"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">gg_freq_poly</span><span class="op">(</span><span class="va">widths</span><span class="op">$</span><span class="va">invs.gbs</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">subset</a></span><span class="op">(</span><span class="va">chr</span> <span class="op"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"chr1"</span>, <span class="st">"chr2"</span>, <span class="st">"PAR"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>col<span class="op">=</span><span class="va">Arm</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html" class="external-link">facet_wrap</a></span><span class="op">(</span><span class="op">~</span><span class="va">dist</span>, scales <span class="op">=</span> <span class="st">"free_y"</span>, nrow<span class="op">=</span><span class="fl">1</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Distribution of alignment widths for inverted regions, using aligned regions, split by distance group"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_abline.html" class="external-link">geom_vline</a></span><span class="op">(</span>xintercept <span class="op">=</span> <span class="fl">500</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_continuous.html" class="external-link">scale_x_log10</a></span><span class="op">(</span><span class="st">"Width (vertical bar at 500 bp)"</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Scale for 'x' is already present. Adding another scale for 'x', which will</span>
<span class="co">## replace the existing scale.</span>
<span class="co">## `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.</span></code></pre>
<p><img src="RegionWidths_files/figure-html/inversion_width_aligned-3.png" width="700"></p>
<div class="sourceCode" id="cb240"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">gg_freq_poly</span><span class="op">(</span><span class="va">widths</span><span class="op">$</span><span class="va">invs.coa</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>col<span class="op">=</span><span class="va">Arm</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html" class="external-link">facet_wrap</a></span><span class="op">(</span><span class="op">~</span><span class="va">chr</span>, scales <span class="op">=</span> <span class="st">"free_y"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Distribution of alignment widths for inverted regions, using mapped regions, split by arm."</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.</span></code></pre>
<p><img src="RegionWidths_files/figure-html/inversion_width_mapped-1.png" width="700"></p>
<div class="sourceCode" id="cb242"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">gg_freq_poly</span><span class="op">(</span><span class="va">widths</span><span class="op">$</span><span class="va">invs.coa</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>col<span class="op">=</span><span class="va">Arm</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html" class="external-link">facet_wrap</a></span><span class="op">(</span><span class="op">~</span><span class="va">name</span>, scales <span class="op">=</span> <span class="st">"free_y"</span>, ncol<span class="op">=</span><span class="fl">5</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Distribution of alignment widths for inverted regions, using mapped regions, split by genome pair"</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.</span></code></pre>
<p><img src="RegionWidths_files/figure-html/inversion_width_mapped-2.png" width="700"></p>
<div class="sourceCode" id="cb244"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">gg_freq_poly</span><span class="op">(</span><span class="va">widths</span><span class="op">$</span><span class="va">invs.coa</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">subset</a></span><span class="op">(</span><span class="va">chr</span> <span class="op"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"chr1"</span>, <span class="st">"chr2"</span>, <span class="st">"PAR"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>col<span class="op">=</span><span class="va">Arm</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html" class="external-link">facet_wrap</a></span><span class="op">(</span><span class="op">~</span><span class="va">dist</span>, scales <span class="op">=</span> <span class="st">"free_y"</span>, nrow<span class="op">=</span><span class="fl">1</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Distribution of alignment widths for inverted regions, using mapped regions, split by distance group"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_abline.html" class="external-link">geom_vline</a></span><span class="op">(</span>xintercept <span class="op">=</span> <span class="fl">500</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_continuous.html" class="external-link">scale_x_log10</a></span><span class="op">(</span><span class="st">"Width (vertical bar at 500 bp)"</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Scale for 'x' is already present. Adding another scale for 'x', which will</span>
<span class="co">## replace the existing scale.</span>
<span class="co">## `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.</span></code></pre>
<p><img src="RegionWidths_files/figure-html/inversion_width_mapped-3.png" width="700"></p>
<div class="sourceCode" id="cb246"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">gg_freq_poly</span><span class="op">(</span><span class="va">widths</span><span class="op">$</span><span class="va">invs.coa2</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>col<span class="op">=</span><span class="va">Arm</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html" class="external-link">facet_wrap</a></span><span class="op">(</span><span class="op">~</span><span class="va">chr</span>, scales <span class="op">=</span> <span class="st">"free_y"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Distribution of alignment widths for inverted regions, using doublemapped regions, split by arm."</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.</span></code></pre>
<p><img src="RegionWidths_files/figure-html/inversion_width_doublemapped-1.png" width="700"></p>
<div class="sourceCode" id="cb248"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">gg_freq_poly</span><span class="op">(</span><span class="va">widths</span><span class="op">$</span><span class="va">invs.coa2</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>col<span class="op">=</span><span class="va">Arm</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html" class="external-link">facet_wrap</a></span><span class="op">(</span><span class="op">~</span><span class="va">name</span>, scales <span class="op">=</span> <span class="st">"free_y"</span>, ncol<span class="op">=</span><span class="fl">5</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Distribution of alignment widths for inverted regions, using doublemapped regions, split by genome pair"</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.</span></code></pre>
<p><img src="RegionWidths_files/figure-html/inversion_width_doublemapped-2.png" width="700"></p>
<div class="sourceCode" id="cb250"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">gg_freq_poly</span><span class="op">(</span><span class="va">widths</span><span class="op">$</span><span class="va">invs.coa2</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">subset</a></span><span class="op">(</span><span class="va">chr</span> <span class="op"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"chr1"</span>, <span class="st">"chr2"</span>, <span class="st">"PAR"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>col<span class="op">=</span><span class="va">Arm</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html" class="external-link">facet_wrap</a></span><span class="op">(</span><span class="op">~</span><span class="va">dist</span>, scales <span class="op">=</span> <span class="st">"free_y"</span>, nrow<span class="op">=</span><span class="fl">1</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Distribution of alignment widths for inverted regions, using doublemapped regions, split by distance group"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_abline.html" class="external-link">geom_vline</a></span><span class="op">(</span>xintercept <span class="op">=</span> <span class="fl">500</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_continuous.html" class="external-link">scale_x_log10</a></span><span class="op">(</span><span class="st">"Width (vertical bar at 500 bp)"</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Scale for 'x' is already present. Adding another scale for 'x', which will</span>
<span class="co">## replace the existing scale.</span>
<span class="co">## `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.</span></code></pre>
<p><img src="RegionWidths_files/figure-html/inversion_width_doublemapped-3.png" width="700"></p>
</div>
<div class="section level2">
<h2 id="width-of-gaps-next-to-the-inverted-regions">Width of gaps next to the inverted regions<a class="anchor" aria-label="anchor" href="#width-of-gaps-next-to-the-inverted-regions"></a>
</h2>
<div class="sourceCode" id="cb252"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">inv.lgaps</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">coa</span>, <span class="va">leftInversionGaps</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/SimpleList-class.html" class="external-link">SimpleList</a></span><span class="op">(</span><span class="op">)</span>
<span class="va">inv.lgaps</span><span class="op">[</span> <span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span>  <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">inv.lgaps</span><span class="op">[</span> <span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span>,  <span class="va">flagLongShort</span>, <span class="va">longShort</span><span class="op">$</span><span class="va">Oki</span><span class="op">)</span>
<span class="va">inv.lgaps</span><span class="op">[</span> <span class="fl">6</span><span class="op">:</span><span class="fl">10</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">inv.lgaps</span><span class="op">[</span> <span class="fl">6</span><span class="op">:</span><span class="fl">10</span><span class="op">]</span>, <span class="va">flagLongShort</span>, <span class="va">longShort</span><span class="op">$</span><span class="va">Osa</span><span class="op">)</span>
<span class="va">inv.lgaps</span><span class="op">[</span><span class="fl">11</span><span class="op">:</span><span class="fl">15</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">inv.lgaps</span><span class="op">[</span><span class="fl">11</span><span class="op">:</span><span class="fl">15</span><span class="op">]</span>, <span class="va">flagLongShort</span>, <span class="va">longShort</span><span class="op">$</span><span class="va">Bar</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">inv.lgaps</span>, <span class="va">length</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Oki_Osa Oki_Bar Oki_Kum Oki_Aom Oki_Nor Osa_Oki Osa_Bar Osa_Kum Osa_Aom Osa_Nor </span>
<span class="co">##     669     675      96     665     591     669     319     678      69     274 </span>
<span class="co">## Bar_Oki Bar_Osa Bar_Kum Bar_Aom Bar_Nor Ply_Ros Ply_Rob Ply_Sav Ply_Oki Rob_Ros </span>
<span class="co">##     666     315     685     315      22      46      65      92       0      61 </span>
<span class="co">## Rob_Ply Rob_Sav Rob_Oki Dme_Dbu Dme_Dsu Dme_Dya Dme_Dma </span>
<span class="co">##      67      92       0    1003      44      21      14</span></code></pre>
<div class="sourceCode" id="cb254"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">widths</span><span class="op">$</span><span class="va">invs.lgaps</span> <span class="op">&lt;-</span> <span class="fu">makeWidthsDF</span><span class="op">(</span><span class="va">inv.lgaps</span><span class="op">)</span> |&gt; <span class="fu">tibble</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/as_tibble.html" class="external-link">as_tibble</a></span><span class="op">(</span><span class="op">)</span>

<span class="fu">gg_freq_poly</span><span class="op">(</span><span class="va">widths</span><span class="op">$</span><span class="va">invs.lgaps</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>col<span class="op">=</span><span class="va">Arm</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html" class="external-link">facet_wrap</a></span><span class="op">(</span><span class="op">~</span><span class="va">chr</span>, scales <span class="op">=</span> <span class="st">"free_y"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Distribution widths of left gaps next to inverted regions, split by arm."</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_abline.html" class="external-link">geom_vline</a></span><span class="op">(</span>xintercept <span class="op">=</span> <span class="fl">1000</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.</span></code></pre>
<p><img src="RegionWidths_files/figure-html/width_of_inversions_gaps-1.png" width="700"></p>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Aleksandra Bliznina, Michael Mansfield, Charles Plessy.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.1.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
