<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Regions of interest â€¢ OikScrambling</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Regions of interest">
<meta property="og:description" content="OikScrambling">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">OikScrambling</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">3.0.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/OikScrambling.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/CNEs.html">Conserved non-coding elements</a>
    </li>
    <li>
      <a href="../articles/CoalescingCutoffs.html">Modifying coalescent cutoffs</a>
    </li>
    <li>
      <a href="../articles/ColinearityInterruptors.html">Regions interrupting colinearity</a>
    </li>
    <li>
      <a href="../articles/DistanceMeasures.html">Distance Measures</a>
    </li>
    <li>
      <a href="../articles/GenomicFeatures.html">Genomic features of the breakpoints</a>
    </li>
    <li>
      <a href="../articles/Inversions.html">Inversions</a>
    </li>
    <li>
      <a href="../articles/LoadGenomicBreaks.html">Load genomic break data</a>
    </li>
    <li>
      <a href="../articles/OrthogroupSyntenies.html">Orthogroup syntenies</a>
    </li>
    <li>
      <a href="../articles/OxfordPlots.html">Whole-genome dot-plots</a>
    </li>
    <li>
      <a href="../articles/PAC3.html">Region of interest: PAC3</a>
    </li>
    <li>
      <a href="../articles/PWM.html">Motif search and analysis</a>
    </li>
    <li>
      <a href="../articles/ParallelPlots.html">Whole-chromosome parallel plots</a>
    </li>
    <li>
      <a href="../articles/RegionWidths.html">Width of the aligned and mapped regions.</a>
    </li>
    <li>
      <a href="../articles/RegionsOfInterest.html">Regions of interest</a>
    </li>
    <li>
      <a href="../articles/RepeatRegions.html">Repeat regions</a>
    </li>
    <li>
      <a href="../articles/SegmentMovements.html">Segment movement distances</a>
    </li>
    <li>
      <a href="../articles/TFBS.html">Transcription factor binding sites</a>
    </li>
    <li>
      <a href="../articles/TandemSiteDuplications.html">Tandem site duplications</a>
    </li>
    <li>
      <a href="../articles/Translocations.html">Translocations</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/oist/LuscombeU_OikScrambling/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="RegionsOfInterest_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Regions of interest</h1>
                        <h4 data-toc-skip class="author">Charles Plessy</h4>
            
            <h4 data-toc-skip class="date">30 March, 2022</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/oist/LuscombeU_OikScrambling/blob/HEAD/vignettes/RegionsOfInterest.Rmd" class="external-link"><code>vignettes/RegionsOfInterest.Rmd</code></a></small>
      <div class="hidden name"><code>RegionsOfInterest.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="load-pacakges-and-data">Load pacakges and data<a class="anchor" aria-label="anchor" href="#load-pacakges-and-data"></a>
</h2>
<p>See the <code><a href="../articles/OikScrambling.html">vignette("OikScrambling", package = "OikScrambling")</a></code> for general details on package and data load.</p>
<p>See <code><a href="../articles/LoadGenomicBreaks.html">vignette("LoadGenomicBreaks", package = "OikScrambling")</a></code> for how the different <code>GBreaks</code> objects are prepared.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="op">{</span>
  <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://oist.github.io/GenomicBreaks" class="external-link">'GenomicBreaks'</a></span><span class="op">)</span>
  <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://ggplot2.tidyverse.org" class="external-link">'ggplot2'</a></span><span class="op">)</span>
  <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st">"BreakpointsData"</span><span class="op">)</span>
  <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://bioconductor.org/packages/Biostrings" class="external-link">"Biostrings"</a></span><span class="op">)</span>
<span class="op">}</span><span class="op">)</span>
<span class="va">genomes</span> <span class="op">&lt;-</span> <span class="fu">OikScrambling</span><span class="fu">:::</span><span class="fu"><a href="../reference/loadAllGenomes.html">loadAllGenomes</a></span><span class="op">(</span>compat <span class="op">=</span> <span class="cn">F</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/load.html" class="external-link">load</a></span><span class="op">(</span><span class="st">"BreakPoints.Rdata"</span><span class="op">)</span>
<span class="va">reps</span> <span class="op">&lt;-</span> <span class="fu">OikScrambling</span><span class="fu">:::</span><span class="fu"><a href="../reference/loadAllRepeats.html">loadAllRepeats</a></span><span class="op">(</span>compat <span class="op">=</span> <span class="cn">F</span><span class="op">)</span>
<span class="va">transcripts</span> <span class="op">&lt;-</span> <span class="fu">OikScrambling</span><span class="fu">:::</span><span class="fu"><a href="../reference/loadAllTranscriptsGR.html">loadAllTranscriptsGR</a></span><span class="op">(</span>compat <span class="op">=</span> <span class="cn">F</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="the-problem">The problem:<a class="anchor" aria-label="anchor" href="#the-problem"></a>
</h2>
<p>The plot below shows high colinearity between Oki and Kum, but the <code>GenomicBreaks</code> objects are still scattered in small pieces. What is interrupting colinearity?</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/plotApairOfChrs.html" class="external-link">plotApairOfChrs</a></span><span class="op">(</span><span class="va">coa</span><span class="op">$</span><span class="va">Oki_Kum</span>, <span class="st">"chr1"</span>, main <span class="op">=</span> <span class="st">"Oki â€“ Kum"</span><span class="op">)</span></code></pre></div>
<p><img src="RegionsOfInterest_files/figure-html/parallel_chr_plots_Oki_Kum-1.png" width="700"></p>
<p>In this vignette, we look at a couple of example regions in details. These regions helped me to decide how to process the data in <em>Load Genomic Breaks</em> (<code><a href="../articles/LoadGenomicBreaks.html">vignette("LoadGenomicBreaks", package = "OikScrambling")</a></code> and to decide which analysis to prioritise elsewhere.</p>
</div>
<div class="section level2">
<h2 id="alignments-are-real-region-of-interest-1">Alignments are real (Region of Interest 1)<a class="anchor" aria-label="anchor" href="#alignments-are-real-region-of-interest-1"></a>
</h2>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># The best Kume contig matching ot chr1 is contig_3_1</span>
<span class="va">bestMatch</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">gb</span><span class="op">)</span>
  <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/tapply.html" class="external-link">tapply</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/start.html" class="external-link">width</a></span><span class="op">(</span><span class="va">gb</span><span class="op">$</span><span class="va">query</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqnames</a></span><span class="op">(</span><span class="va">gb</span><span class="op">$</span><span class="va">query</span><span class="op">)</span>, <span class="va">sum</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/sort.html" class="external-link">sort</a></span><span class="op">(</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">tail</a></span><span class="op">(</span><span class="fl">1</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="op">)</span>

<span class="va">coa</span><span class="op">$</span><span class="va">Oki_Kum</span> |&gt; <span class="fu">plyranges</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/plyranges/man/tidyverse-reexports.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">seqnames</span> <span class="op">==</span> <span class="st">"chr1"</span><span class="op">)</span> |&gt; <span class="fu">bestMatch</span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## [1] "contig_3_1"</span></code></pre>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># contig_3_1 appears to map roughly in the 8,000,000 â€“ 13,000,000 region</span>
<span class="va">coa</span><span class="op">$</span><span class="va">Oki_Kum</span> |&gt; <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/flagAll.html" class="external-link">flagAll</a></span><span class="op">(</span><span class="op">)</span> |&gt; <span class="fu">plyranges</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/plyranges/man/tidyverse-reexports.html" class="external-link">filter</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqnames</a></span><span class="op">(</span><span class="va">query</span><span class="op">)</span> <span class="op">==</span> <span class="st">"contig_3_1"</span><span class="op">)</span> |&gt;
  <span class="fu">plyranges</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/plyranges/man/tidyverse-reexports.html" class="external-link">arrange</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/start.html" class="external-link">start</a></span><span class="op">(</span><span class="va">query</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## GBreaks object with 303 ranges and 8 metadata columns:</span>
<span class="co">##         seqnames            ranges strand |                      query</span>
<span class="co">##            &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; |                  &lt;GRanges&gt;</span>
<span class="co">##     [1]     chr1   8086429-8100669      + |       contig_3_1:714-16270</span>
<span class="co">##     [2]     chr1   3870099-3870324      - |     contig_3_1:16561-16812</span>
<span class="co">##     [3]     chr1   8100696-8104950      + |     contig_3_1:16822-21079</span>
<span class="co">##     [4]     chr1   8106235-8126929      + |     contig_3_1:21198-40210</span>
<span class="co">##     [5]      PAR   9620326-9621010      - |     contig_3_1:40220-40912</span>
<span class="co">##     ...      ...               ...    ... .                        ...</span>
<span class="co">##   [299]     chr1 12792338-12796149      + | contig_3_1:4603103-4606912</span>
<span class="co">##   [300]     chr1 12799027-12831934      + | contig_3_1:4606913-4641495</span>
<span class="co">##   [301]     chr1 12833633-12852708      + | contig_3_1:4641498-4657954</span>
<span class="co">##   [302]     chr1 12866392-12879379      + | contig_3_1:4657955-4669933</span>
<span class="co">##   [303]     chr1 12881855-12897271      + | contig_3_1:4669935-4684841</span>
<span class="co">##             score      Arm                           rep   repOvlp</span>
<span class="co">##         &lt;integer&gt; &lt;factor&gt;               &lt;CharacterList&gt; &lt;integer&gt;</span>
<span class="co">##     [1]     14241    long                         tandem       121</span>
<span class="co">##     [2]       226    short                           rnd        85</span>
<span class="co">##     [3]      4255    long                     rnd,tandem       114</span>
<span class="co">##     [4]     20695    long                    rnd,unknown       711</span>
<span class="co">##     [5]       685    long                        unknown       685</span>
<span class="co">##     ...       ...      ...                           ...       ...</span>
<span class="co">##   [299]      3812     long                          &lt;NA&gt;         0</span>
<span class="co">##   [300]     32908     long LowComplexity,rnd,unknown,...      2219</span>
<span class="co">##   [301]     19076     long            tandem,rnd,unknown       933</span>
<span class="co">##   [302]     12988     long            unknown,tandem,rnd      1423</span>
<span class="co">##   [303]     15417     long  tandem,LowComplexity,rnd,...       688</span>
<span class="co">##                    transcripts        flag    nonCoa</span>
<span class="co">##                          &lt;Rle&gt; &lt;character&gt; &lt;logical&gt;</span>
<span class="co">##     [1]               g2146.t1        &lt;NA&gt;     FALSE</span>
<span class="co">##     [2]                   &lt;NA&gt;        &lt;NA&gt;      TRUE</span>
<span class="co">##     [3] g2151.t1;g2152.t1;g2..         Tra      TRUE</span>
<span class="co">##     [4] g2153.t1;g2157.t1;g2..        &lt;NA&gt;     FALSE</span>
<span class="co">##     [5]                   &lt;NA&gt;        &lt;NA&gt;      TRUE</span>
<span class="co">##     ...                    ...         ...       ...</span>
<span class="co">##   [299]                   &lt;NA&gt;        &lt;NA&gt;      TRUE</span>
<span class="co">##   [300] g3507.t1;g3511.t2;g3..         Tra     FALSE</span>
<span class="co">##   [301]      g3513.t1;g3515.t1         Tra     FALSE</span>
<span class="co">##   [302]                   &lt;NA&gt;         Tra     FALSE</span>
<span class="co">##   [303] g3531.t1;g3532.t1;g3..        &lt;NA&gt;     FALSE</span>
<span class="co">##   -------</span>
<span class="co">##   seqinfo: 19 sequences from OKI2018.I69 genome</span></code></pre>
<p>How about the second alignment pair, mapping 226 bases of <code>contig_3_1</code> to the short arm?</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ROI1</span> <span class="op">&lt;-</span> <span class="va">coa</span><span class="op">$</span><span class="va">Oki_Kum</span> |&gt;
  <span class="fu">plyranges</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/plyranges/man/tidyverse-reexports.html" class="external-link">filter</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqnames</a></span><span class="op">(</span><span class="va">query</span><span class="op">)</span> <span class="op">==</span> <span class="st">"contig_3_1"</span>, <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/start.html" class="external-link">end</a></span><span class="op">(</span><span class="va">query</span><span class="op">)</span> <span class="op">&lt;=</span> <span class="fl">21079</span><span class="op">)</span> |&gt;
  <span class="fu">plyranges</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/plyranges/man/tidyverse-reexports.html" class="external-link">arrange</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/start.html" class="external-link">start</a></span><span class="op">(</span><span class="va">query</span><span class="op">)</span><span class="op">)</span>
<span class="va">ROI1</span></code></pre></div>
<pre><code><span class="co">## GBreaks object with 3 ranges and 8 metadata columns:</span>
<span class="co">##       seqnames          ranges strand |                  query     score</span>
<span class="co">##          &lt;Rle&gt;       &lt;IRanges&gt;  &lt;Rle&gt; |              &lt;GRanges&gt; &lt;integer&gt;</span>
<span class="co">##   [1]     chr1 8086429-8100669      + |   contig_3_1:714-16270     14241</span>
<span class="co">##   [2]     chr1 3870099-3870324      - | contig_3_1:16561-16812       226</span>
<span class="co">##   [3]     chr1 8100696-8104950      + | contig_3_1:16822-21079      4255</span>
<span class="co">##            Arm             rep   repOvlp            transcripts        flag</span>
<span class="co">##       &lt;factor&gt; &lt;CharacterList&gt; &lt;integer&gt;                  &lt;Rle&gt; &lt;character&gt;</span>
<span class="co">##   [1]    long           tandem       121               g2146.t1        &lt;NA&gt;</span>
<span class="co">##   [2]    short             rnd        85                   &lt;NA&gt;        &lt;NA&gt;</span>
<span class="co">##   [3]    long       rnd,tandem       114 g2151.t1;g2152.t1;g2..         Tra</span>
<span class="co">##          nonCoa</span>
<span class="co">##       &lt;logical&gt;</span>
<span class="co">##   [1]     FALSE</span>
<span class="co">##   [2]      TRUE</span>
<span class="co">##   [3]      TRUE</span>
<span class="co">##   -------</span>
<span class="co">##   seqinfo: 19 sequences from OKI2018.I69 genome</span></code></pre>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">coa</span><span class="op">$</span><span class="va">Oki_Kum</span> |&gt; <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/plotApairOfChrs.html" class="external-link">plotApairOfChrs</a></span><span class="op">(</span><span class="st">"chr1"</span>, <span class="st">"contig_3_1"</span>, xlim <span class="op">=</span> <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/gb2xlim.html" class="external-link">gb2xlim</a></span><span class="op">(</span><span class="va">ROI1</span><span class="op">[</span><span class="op">-</span><span class="fl">2</span><span class="op">]</span><span class="op">)</span>, main <span class="op">=</span> <span class="st">"chr1 vs contig_3_1 at ROI 1"</span><span class="op">)</span></code></pre></div>
<p><img src="RegionsOfInterest_files/figure-html/re_align_a_clolinearity_breaker-1.png" width="700"></p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">coa</span><span class="op">$</span><span class="va">Oki_Kum</span> |&gt; <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/plotApairOfChrs.html" class="external-link">plotApairOfChrs</a></span><span class="op">(</span><span class="st">"chr1"</span>, <span class="st">"contig_3_1"</span>, xlim <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">3838658</span>, <span class="fl">3881203</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">21079</span><span class="op">)</span><span class="op">)</span>, main <span class="op">=</span> <span class="st">"chr1 matching contig_3_1 in the short arm"</span><span class="op">)</span></code></pre></div>
<p><img src="RegionsOfInterest_files/figure-html/re_align_a_clolinearity_breaker-2.png" width="700"></p>
<p>Is the alignment convincing ?</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Good match where reported</span>
<span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/pairwiseAlignment.html" class="external-link">pairwiseAlignment</a></span><span class="op">(</span><span class="va">ROI1</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Global PairwiseAlignmentsSingleSubject (1 of 1)</span>
<span class="co">## pattern: <span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">AAA</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span>-------------...<span style="color: #000000; background-color: #AFFFAF;">CC</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">GG</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">AAA</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">GG</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">CCCCC</span></span>
<span class="co">## subject: <span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #FFD7AF;">TTT</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">AAA</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #AFFFAF;">CC</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #AFFFAF;">C</span>...<span style="color: #000000; background-color: #AFFFAF;">CC</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">GG</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">AAA</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">GG</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">CCCCC</span></span>
<span class="co">## score: 262.3701</span></code></pre>
<details><p><summary>See the full alignment with 50 bp of flanking regions added to contrast</summary></p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/pairwiseAlignment.html" class="external-link">pairwiseAlignment</a></span><span class="op">(</span><span class="va">ROI1</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span> <span class="op">+</span> <span class="fl">50</span>, <span class="va">ROI1</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span><span class="op">$</span><span class="va">query</span> <span class="op">+</span> <span class="fl">50</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/PairwiseAlignments-io.html" class="external-link">writePairwiseAlignments</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## ########################################</span>
<span class="co">## # Program: Biostrings (version 2.62.0), a Bioconductor package</span>
<span class="co">## # Rundate: Wed Mar 30 19:15:28 2022</span>
<span class="co">## ########################################</span>
<span class="co">## #=======================================</span>
<span class="co">## #</span>
<span class="co">## # Aligned_sequences: 2</span>
<span class="co">## # 1: P1</span>
<span class="co">## # 2: S1</span>
<span class="co">## # Matrix: NA</span>
<span class="co">## # Gap_penalty: 14.0</span>
<span class="co">## # Extend_penalty: 4.0</span>
<span class="co">## #</span>
<span class="co">## # Length: 358</span>
<span class="co">## # Identity:     279/358 (77.9%)</span>
<span class="co">## # Similarity:    NA/358 (NA%)</span>
<span class="co">## # Gaps:          38/358 (10.6%)</span>
<span class="co">## # Score: 9.038877</span>
<span class="co">## #</span>
<span class="co">## #</span>
<span class="co">## #=======================================</span>
<span class="co">## </span>
<span class="co">## P1                 1 GGGATTG-GACGA---AAGTA-----ACGGAAA-----TAAAAATATCTG     36</span>
<span class="co">##                       |  ||| | |||    |||      | ||| |     |  | | ||  |</span>
<span class="co">## S1                 1 AGCGTTGCGTCGAGTCGAGTCGGGCCAGGGAGAAGCCCTCTATA-ATAAG     49</span>
<span class="co">## </span>
<span class="co">## P1                37 GTTTCGG-TGCTTTATCACTGATAAATTATC-----------AGTGACCT     74</span>
<span class="co">##                      ||  | | |  ||||||| |||||||||| |           | ||||||</span>
<span class="co">## S1                50 GTCACTGATAATTTATCAGTGATAAATTAACCTTCTCTCATAAATGACCT     99</span>
<span class="co">## </span>
<span class="co">## P1                75 TCTCCCTTCAGGATATATAATAACCCCTTTCTTTCAATAGTTATACTTAT    124</span>
<span class="co">##                      ||||||||||||||||||||||||||||||||||||||||||||||||||</span>
<span class="co">## S1               100 TCTCCCTTCAGGATATATAATAACCCCTTTCTTTCAATAGTTATACTTAT    149</span>
<span class="co">## </span>
<span class="co">## P1               125 TTTTT-AGTATTATTCTGAATTTTTT-TAAAATTATCATTTGTCAATTTT    172</span>
<span class="co">##                      ||||| ||||| |||||||||||||| | |||||||||||||||||||||</span>
<span class="co">## S1               150 TTTTTTAGTATAATTCTGAATTTTTTCTGAAATTATCATTTGTCAATTTT    199</span>
<span class="co">## </span>
<span class="co">## P1               173 ATCGGCAAGATGCACCTTGGGAGCCCATTGCGTATTTATCGGATCTCAAA    222</span>
<span class="co">##                      ||||| |||||||||| |||||||||||||||||||||||||||||||||</span>
<span class="co">## S1               200 ATCGG-AAGATGCACCGTGGGAGCCCATTGCGTATTTATCGGATCTCAAA    248</span>
<span class="co">## </span>
<span class="co">## P1               223 GGATTAGCGCTTAGCATTGCCCATATTACACACTGGTCAAAGTGGTCAGC    272</span>
<span class="co">##                      ||||||||||||||||||||||||||||||||||||||||||||||||||</span>
<span class="co">## S1               249 GGATTAGCGCTTAGCATTGCCCATATTACACACTGGTCAAAGTGGTCAGC    298</span>
<span class="co">## </span>
<span class="co">## P1               273 CCCCTTCGAACCAA--CTCA-AGTGATG-GTCTCTTACTTCTGCGGAACG    318</span>
<span class="co">##                      ||||    ||  ||  |||| | ||||| || |   | || | || ||| </span>
<span class="co">## S1               299 CCCCAAA-AATGAAGGCTCATAATGATGAGTGTGCAAATTTTTCGTAAC-    346</span>
<span class="co">## </span>
<span class="co">## P1               319 AACGGGGC    326</span>
<span class="co">##                        |   ||</span>
<span class="co">## S1               347 --CTTTGC    352</span>
<span class="co">## </span>
<span class="co">## </span>
<span class="co">## #---------------------------------------</span>
<span class="co">## #---------------------------------------</span></code></pre>
</details><p>No good match between the flanking pairs.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/pairwiseAlignment.html" class="external-link">pairwiseAlignment</a></span><span class="op">(</span><span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/getSeq.html" class="external-link">getSeq</a></span><span class="op">(</span><span class="va">ROI1</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span><span class="op">$</span><span class="va">query</span><span class="op">)</span>, <span class="va">genomes</span><span class="op">$</span><span class="va">OKI2018.I69</span><span class="op">$</span><span class="va">chr1</span><span class="op">[</span><span class="fl">8086429</span><span class="op">:</span><span class="fl">8104950</span><span class="op">]</span>, type<span class="op">=</span><span class="st">"local"</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Local PairwiseAlignmentsSingleSubject (1 of 1)</span>
<span class="co">## pattern:   [113] <span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #FFD7AF;">TTTTTT</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">G</span></span>
<span class="co">## subject: [11595] <span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">GG</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">TTTTTT</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">G</span></span>
<span class="co">## score: 23.82705</span></code></pre>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/pairwiseAlignment.html" class="external-link">pairwiseAlignment</a></span><span class="op">(</span><span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/getSeq.html" class="external-link">getSeq</a></span><span class="op">(</span><span class="va">genomes</span><span class="op">$</span><span class="va">KUM.M3.7f</span>, <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a></span><span class="op">(</span><span class="st">"contig_3_1:16561-16812:-"</span><span class="op">)</span><span class="op">)</span>, <span class="va">genomes</span><span class="op">$</span><span class="va">OKI2018.I69</span><span class="op">$</span><span class="va">chr1</span><span class="op">[</span><span class="fl">8086429</span><span class="op">:</span><span class="fl">8104950</span><span class="op">]</span>, type<span class="op">=</span><span class="st">"local"</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Local PairwiseAlignmentsSingleSubject (1 of 1)</span>
<span class="co">## pattern:  [221] <span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">GG</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #FFD7AF;">TTT</span><span style="color: #000000; background-color: #FFAFAF;">A</span></span>
<span class="co">## subject: [1978] <span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">GG</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #FFD7AF;">TTT</span><span style="color: #000000; background-color: #FFAFAF;">A</span></span>
<span class="co">## score: 21.79932</span></code></pre>
<p>Yes, the alignment looks real!</p>
</div>
<div class="section level2">
<h2 id="colinearity-interrupted-by-inversions-and-translocations-region-of-interest-2">Colinearity interrupted by inversions and translocations (Region of Interest 2)<a class="anchor" aria-label="anchor" href="#colinearity-interrupted-by-inversions-and-translocations-region-of-interest-2"></a>
</h2>
<p>Letâ€™s look at the area <code>chr1:8080000-12900000</code> covered by <code>contig_3_1</code>. Can we coalesce it once we only keep the paired matches? The answer is noâ€¦</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">coa</span><span class="op">$</span><span class="va">Oki_Kum</span> |&gt; <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/flagAll.html" class="external-link">flagAll</a></span><span class="op">(</span><span class="op">)</span> |&gt; <span class="fu">plyranges</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/plyranges/man/tidyverse-reexports.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">seqnames</span> <span class="op">==</span> <span class="st">"chr1"</span>, <span class="va">start</span> <span class="op">&gt;</span> <span class="fl">8080000</span>, <span class="va">end</span> <span class="op">&lt;</span> <span class="fl">12900000</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## GBreaks object with 303 ranges and 8 metadata columns:</span>
<span class="co">##         seqnames            ranges strand |                       query</span>
<span class="co">##            &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; |                   &lt;GRanges&gt;</span>
<span class="co">##     [1]     chr1   8086429-8100669      + |        contig_3_1:714-16270</span>
<span class="co">##     [2]     chr1   8100696-8104950      + |      contig_3_1:16822-21079</span>
<span class="co">##     [3]     chr1   8105493-8106187      - | contig_29_1:1884094-1884784</span>
<span class="co">##     [4]     chr1   8106235-8126929      + |      contig_3_1:21198-40210</span>
<span class="co">##     [5]     chr1   8126930-8192136      + |     contig_3_1:40913-120953</span>
<span class="co">##     ...      ...               ...    ... .                         ...</span>
<span class="co">##   [299]     chr1 12852734-12866391      + | contig_82_1:2309881-2322607</span>
<span class="co">##   [300]     chr1 12866392-12879379      + |  contig_3_1:4657955-4669933</span>
<span class="co">##   [301]     chr1 12880951-12881290      + | contig_27_1:4164345-4164661</span>
<span class="co">##   [302]     chr1 12881855-12897271      + |  contig_3_1:4669935-4684841</span>
<span class="co">##   [303]     chr1 12897273-12899763      - | contig_88_1:1125832-1128287</span>
<span class="co">##             score      Arm                          rep   repOvlp</span>
<span class="co">##         &lt;integer&gt; &lt;factor&gt;              &lt;CharacterList&gt; &lt;integer&gt;</span>
<span class="co">##     [1]     14241     long                       tandem       121</span>
<span class="co">##     [2]      4255     long                   rnd,tandem       114</span>
<span class="co">##     [3]       695     long                      unknown         0</span>
<span class="co">##     [4]     20695     long                  rnd,unknown       711</span>
<span class="co">##     [5]     65207     long         LowComplexity,tandem       620</span>
<span class="co">##     ...       ...      ...                          ...       ...</span>
<span class="co">##   [299]     13658     long           unknown,rnd,tandem       131</span>
<span class="co">##   [300]     12988     long           unknown,tandem,rnd      1423</span>
<span class="co">##   [301]       340     long                          rnd         0</span>
<span class="co">##   [302]     15417     long tandem,LowComplexity,rnd,...       688</span>
<span class="co">##   [303]      2491     long                          rnd         0</span>
<span class="co">##                    transcripts        flag    nonCoa</span>
<span class="co">##                          &lt;Rle&gt; &lt;character&gt; &lt;logical&gt;</span>
<span class="co">##     [1]               g2146.t1        &lt;NA&gt;     FALSE</span>
<span class="co">##     [2] g2151.t1;g2152.t1;g2..         Tra      TRUE</span>
<span class="co">##     [3]                   &lt;NA&gt;        &lt;NA&gt;      TRUE</span>
<span class="co">##     [4] g2153.t1;g2157.t1;g2..        &lt;NA&gt;     FALSE</span>
<span class="co">##     [5] g2160.t1;g2161.t1;g2..        &lt;NA&gt;     FALSE</span>
<span class="co">##     ...                    ...         ...       ...</span>
<span class="co">##   [299] g3518.t1;g3519.t1;g3..        &lt;NA&gt;     FALSE</span>
<span class="co">##   [300]                   &lt;NA&gt;         Tra     FALSE</span>
<span class="co">##   [301]                   &lt;NA&gt;        &lt;NA&gt;      TRUE</span>
<span class="co">##   [302] g3531.t1;g3532.t1;g3..        &lt;NA&gt;     FALSE</span>
<span class="co">##   [303]                   &lt;NA&gt;         Tra      TRUE</span>
<span class="co">##   -------</span>
<span class="co">##   seqinfo: 19 sequences from OKI2018.I69 genome</span></code></pre>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Even removing all but the best match on the query is not enough to collapse all</span>
<span class="va">coa</span><span class="op">$</span><span class="va">Oki_Kum</span> |&gt;
  <span class="fu">plyranges</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/plyranges/man/tidyverse-reexports.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">seqnames</span> <span class="op">==</span> <span class="st">"chr1"</span>, <span class="va">start</span> <span class="op">&gt;</span> <span class="fl">8080000</span>, <span class="va">end</span> <span class="op">&lt;</span> <span class="fl">12900000</span><span class="op">)</span> |&gt;
  <span class="fu">plyranges</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/plyranges/man/tidyverse-reexports.html" class="external-link">filter</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqnames</a></span><span class="op">(</span><span class="va">query</span><span class="op">)</span> <span class="op">==</span> <span class="st">"contig_3_1"</span><span class="op">)</span> |&gt;
  <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/coalesce_contigs.html" class="external-link">coalesce_contigs</a></span><span class="op">(</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/flagAll.html" class="external-link">flagAll</a></span><span class="op">(</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">head</a></span><span class="op">(</span><span class="fl">10</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## GBreaks object with 10 ranges and 3 metadata columns:</span>
<span class="co">##        seqnames          ranges strand |                      query     score</span>
<span class="co">##           &lt;Rle&gt;       &lt;IRanges&gt;  &lt;Rle&gt; |                  &lt;GRanges&gt; &lt;integer&gt;</span>
<span class="co">##    [1]     chr1 8086429-8483438      + |      contig_3_1:714-425640    397010</span>
<span class="co">##    [2]     chr1 8485418-8487388      - |   contig_3_1:428514-429368      1971</span>
<span class="co">##    [3]     chr1 8489327-8883052      + |   contig_3_1:432370-835845    393726</span>
<span class="co">##    [4]     chr1 8883053-8945740      + |   contig_3_1:836919-893789     62688</span>
<span class="co">##    [5]     chr1 8953814-8955498      + |   contig_3_1:905043-906751      1685</span>
<span class="co">##    [6]     chr1 8957904-8966137      + |   contig_3_1:896965-905042      8234</span>
<span class="co">##    [7]     chr1 8967756-9470136      + |  contig_3_1:906754-1383986    502381</span>
<span class="co">##    [8]     chr1 9470145-9470890      - | contig_3_1:3260388-3261159       746</span>
<span class="co">##    [9]     chr1 9470891-9891134      + | contig_3_1:1383987-1808624    420244</span>
<span class="co">##   [10]     chr1 9891445-9891819      - | contig_3_1:1808635-1809010       375</span>
<span class="co">##               flag</span>
<span class="co">##        &lt;character&gt;</span>
<span class="co">##    [1]         Inv</span>
<span class="co">##    [2]        &lt;NA&gt;</span>
<span class="co">##    [3]        &lt;NA&gt;</span>
<span class="co">##    [4]         Tra</span>
<span class="co">##    [5]         Tra</span>
<span class="co">##    [6]        &lt;NA&gt;</span>
<span class="co">##    [7]         Tra</span>
<span class="co">##    [8]        &lt;NA&gt;</span>
<span class="co">##    [9]         Inv</span>
<span class="co">##   [10]        &lt;NA&gt;</span>
<span class="co">##   -------</span>
<span class="co">##   seqinfo: 19 sequences from OKI2018.I69 genome</span></code></pre>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">coa</span><span class="op">$</span><span class="va">Oki_Kum</span> |&gt;
  <span class="fu">plyranges</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/plyranges/man/tidyverse-reexports.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">seqnames</span> <span class="op">==</span> <span class="st">"chr1"</span>, <span class="va">start</span> <span class="op">&gt;</span> <span class="fl">8080000</span>, <span class="va">end</span> <span class="op">&lt;</span> <span class="fl">12900000</span><span class="op">)</span> |&gt;
  <span class="fu">plyranges</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/plyranges/man/tidyverse-reexports.html" class="external-link">filter</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqnames</a></span><span class="op">(</span><span class="va">query</span><span class="op">)</span> <span class="op">==</span> <span class="st">"contig_3_1"</span><span class="op">)</span>|&gt;
  <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/coalesce_contigs.html" class="external-link">coalesce_contigs</a></span><span class="op">(</span><span class="op">)</span> |&gt;  <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/plotApairOfChrs.html" class="external-link">plotApairOfChrs</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<p><img src="RegionsOfInterest_files/figure-html/ROI2-1.png" width="700"></p>
<p>Colinearity appears to be interrupted by many inversions and translocations.</p>
</div>
<div class="section level2">
<h2 id="well-aligned-repeats-are-kept-in-the-one-to-one-graph-region-of-interest-3">Well-aligned repeats are kept in the one-to-one graph (Region of interest 3)<a class="anchor" aria-label="anchor" href="#well-aligned-repeats-are-kept-in-the-one-to-one-graph-region-of-interest-3"></a>
</h2>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">gbs</span><span class="op">$</span><span class="va">Oki_Kum</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">##   seqnames start   end width strand score query.seqnames query.start query.end</span>
<span class="co">## 1     chr1  7785 15657  7873      + 43650    contig_10_1       52244     60120</span>
<span class="co">##   query.width query.strand query.rep query.repOvlp query.transcripts   Arm rep</span>
<span class="co">## 1        7877            *       rnd          7738              &lt;NA&gt; short rnd</span>
<span class="co">##   repOvlp transcripts flag nonCoa</span>
<span class="co">## 1    6740        &lt;NA&gt; &lt;NA&gt;   TRUE</span></code></pre>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/pairwiseAlignment.html" class="external-link">pairwiseAlignment</a></span><span class="op">(</span><span class="va">gbs</span><span class="op">$</span><span class="va">Oki_Kum</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Global PairwiseAlignmentsSingleSubject (1 of 1)</span>
<span class="co">## pattern: <span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">AAAAAA</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">AA</span>...<span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">CC</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #AFFFFF;">GG</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">A</span></span>
<span class="co">## subject: <span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">AAAAAA</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">AA</span>...<span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">CC</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #AFFFFF;">GG</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">A</span></span>
<span class="co">## score: 14961.67</span></code></pre>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/findOverlaps-methods.html" class="external-link">subsetByOverlaps</a></span><span class="op">(</span><span class="va">reps</span><span class="op">$</span><span class="va">OKI2018.I69</span>, <span class="va">gbs</span><span class="op">$</span><span class="va">Oki_Kum</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## GRanges object with 5 ranges and 6 metadata columns:</span>
<span class="co">##       seqnames      ranges strand |       source       type     score     phase</span>
<span class="co">##          &lt;Rle&gt;   &lt;IRanges&gt;  &lt;Rle&gt; |     &lt;factor&gt;   &lt;factor&gt; &lt;numeric&gt; &lt;integer&gt;</span>
<span class="co">##   [1]     chr1   7783-7893      + | RepeatMasker similarity      16.8      &lt;NA&gt;</span>
<span class="co">##   [2]     chr1  8181-13174      + | RepeatMasker similarity      25.6      &lt;NA&gt;</span>
<span class="co">##   [3]     chr1 13538-13669      + | RepeatMasker similarity      22.0      &lt;NA&gt;</span>
<span class="co">##   [4]     chr1 13971-14841      + | RepeatMasker similarity      22.2      &lt;NA&gt;</span>
<span class="co">##   [5]     chr1 15005-15638      + | RepeatMasker similarity      18.2      &lt;NA&gt;</span>
<span class="co">##                       Target    Class</span>
<span class="co">##                  &lt;character&gt; &lt;factor&gt;</span>
<span class="co">##   [1] Motif:rnd-1_family-1..      rnd</span>
<span class="co">##   [2] Motif:rnd-1_family-1..      rnd</span>
<span class="co">##   [3] Motif:rnd-1_family-3..      rnd</span>
<span class="co">##   [4] Motif:rnd-1_family-1..      rnd</span>
<span class="co">##   [5] Motif:rnd-1_family-2..      rnd</span>
<span class="co">##   -------</span>
<span class="co">##   seqinfo: 19 sequences from OKI2018.I69 genome</span></code></pre>
</div>
<div class="section level2">
<h2 id="unaligned-mapped-region-roi4">Unaligned mapped region: ROI4<a class="anchor" aria-label="anchor" href="#unaligned-mapped-region-roi4"></a>
</h2>
<p>Letâ€™s check examples of unaligned mapped regions in function of presence of repeats.</p>
<div class="section level3">
<h3 id="falsetrue">FALSE/TRUE<a class="anchor" aria-label="anchor" href="#falsetrue"></a>
</h3>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">bri</span><span class="op">$</span><span class="va">Oki_Osa</span> <span class="op">-&gt;</span> <span class="va">x</span>
<span class="va">x</span><span class="op">$</span><span class="va">Class</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">x</span><span class="op">$</span><span class="va">rep</span>, <span class="va">anyNA</span><span class="op">)</span>, <span class="op">!</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">x</span><span class="op">$</span><span class="va">query</span><span class="op">$</span><span class="va">rep</span>, <span class="va">anyNA</span><span class="op">)</span><span class="op">)</span>
<span class="va">x</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">##   seqnames  start    end width strand query.seqnames query.start query.end</span>
<span class="co">## 1     chr1 552336 552382    47      *           Chr1     1730796   1731144</span>
<span class="co">## 2     chr1 552648 552906   259      *           Chr1     1730116   1730509</span>
<span class="co">## 3     chr1 553835 553872    38      *           Chr1     1728810   1729168</span>
<span class="co">## 4     chr1 554135 554167    33      *           Chr1     1728322   1728529</span>
<span class="co">## 5     chr1 554366 554545   180      *           Chr1     1727928   1728108</span>
<span class="co">## 6     chr1 554693 555190   498      *           Chr1     1727332   1727780</span>
<span class="co">##   query.width query.strand query.Arm query.rep query.repOvlp query.transcripts</span>
<span class="co">## 1         349            *     short       rnd           388              &lt;NA&gt;</span>
<span class="co">## 2         394            *     short    tandem           237              &lt;NA&gt;</span>
<span class="co">## 3         359            *     short        NA             0           g385.t1</span>
<span class="co">## 4         208            *     short        NA             0           g385.t1</span>
<span class="co">## 5         181            *     short        NA             0           g384.t1</span>
<span class="co">## 6         449            *     short        NA             0           g384.t1</span>
<span class="co">##     Arm rep repOvlp transcripts       Class</span>
<span class="co">## 1 short  NA       0      g56.t1  FALSE TRUE</span>
<span class="co">## 2 short  NA       0        &lt;NA&gt;  FALSE TRUE</span>
<span class="co">## 3 short  NA       0        &lt;NA&gt; FALSE FALSE</span>
<span class="co">## 4 short  NA       0        &lt;NA&gt; FALSE FALSE</span>
<span class="co">## 5 short  NA       0        &lt;NA&gt; FALSE FALSE</span>
<span class="co">## 6 short rnd     162        &lt;NA&gt;  TRUE FALSE</span></code></pre>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ROI4_1</span> <span class="op">&lt;-</span> <span class="va">x</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>

<span class="co"># Let's recover the flanking regions</span>
<span class="co"># We expand the bri regions of 1 base on each side to promote overlaps with gbs regions)</span>
<span class="va">gbs</span><span class="op">$</span><span class="va">Oki_Osa</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/findOverlaps-methods.html" class="external-link">subsetByOverlaps</a></span><span class="op">(</span><span class="va">ROI4_1</span> <span class="op">+</span> <span class="fl">1</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">##   seqnames  start    end width strand score query.seqnames query.start</span>
<span class="co">## 1     chr1 551988 552335   348      -   680           Chr1     1731145</span>
<span class="co">## 2     chr1 552383 552647   265      -   416           Chr1     1730510</span>
<span class="co">##   query.end query.width query.strand query.Arm query.rep query.repOvlp</span>
<span class="co">## 1   1731548         404            *     short        NA             0</span>
<span class="co">## 2   1730795         286            *     short       rnd             4</span>
<span class="co">##   query.transcripts   Arm rep repOvlp transcripts flag nonCoa</span>
<span class="co">## 1              &lt;NA&gt; short  NA       0      g56.t1  Col  FALSE</span>
<span class="co">## 2              &lt;NA&gt; short  NA       0      g56.t1  Col  FALSE</span></code></pre>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">gbs</span><span class="op">$</span><span class="va">Oki_Osa</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/findOverlaps-methods.html" class="external-link">subsetByOverlaps</a></span><span class="op">(</span><span class="va">ROI4_1</span> <span class="op">+</span> <span class="fl">1</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/inter-range-methods.html" class="external-link">range</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## GBreaks object with 1 range and 1 metadata column:</span>
<span class="co">##       seqnames        ranges strand |                query</span>
<span class="co">##          &lt;Rle&gt;     &lt;IRanges&gt;  &lt;Rle&gt; |            &lt;GRanges&gt;</span>
<span class="co">##   [1]     chr1 551988-552647      * | Chr1:1730510-1731548</span>
<span class="co">##   -------</span>
<span class="co">##   seqinfo: 19 sequences from OKI2018.I69 genome</span></code></pre>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># We confirm that unalmap regions can not be mapped</span>
<span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/pairwiseAlignment.html" class="external-link">pairwiseAlignment</a></span><span class="op">(</span><span class="va">ROI4_1</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Global PairwiseAlignmentsSingleSubject (1 of 1)</span>
<span class="co">## pattern: <span style="color: #000000; background-color: #FFD7AF;">T</span>------------<span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">CC</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">CC</span>---------------...---------------<span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span>-<span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">TT</span>-------<span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">GG</span><span style="color: #000000; background-color: #FFD7AF;">T</span></span>
<span class="co">## subject: <span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #AFFFFF;">GGGG</span><span style="color: #000000; background-color: #FFAFAF;">AAA</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">CC</span><span style="color: #000000; background-color: #FFAFAF;">AA</span>...<span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">TTTT</span><span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span></span>
<span class="co">## score: -1254.263</span></code></pre>
<div class="sourceCode" id="cb38"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/pairwiseAlignment.html" class="external-link">pairwiseAlignment</a></span><span class="op">(</span><span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/getSeq.html" class="external-link">getSeq</a></span><span class="op">(</span><span class="va">ROI4_1</span><span class="op">)</span>, <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/getSeq.html" class="external-link">getSeq</a></span><span class="op">(</span><span class="va">ROI4_1</span><span class="op">$</span><span class="va">query</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/reverseComplement.html" class="external-link">reverseComplement</a></span><span class="op">(</span><span class="op">)</span> <span class="op">)</span> </code></pre></div>
<pre><code><span class="co">## Global PairwiseAlignmentsSingleSubject (1 of 1)</span>
<span class="co">## pattern: ---------<span style="color: #000000; background-color: #FFD7AF;">TT</span>-----------------------...<span style="color: #000000; background-color: #FFD7AF;">TTT</span><span style="color: #000000; background-color: #AFFFFF;">GG</span><span style="color: #000000; background-color: #FFD7AF;">T</span>----------------------------</span>
<span class="co">## subject: <span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #FFAFAF;">AAAA</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">AA</span>...<span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #AFFFFF;">GG</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">TTT</span><span style="color: #000000; background-color: #AFFFAF;">CCCC</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">A</span></span>
<span class="co">## score: -1260.025</span></code></pre>
<div class="sourceCode" id="cb40"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># We confirm that flanking regions map well</span>
<span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/pairwiseAlignment.html" class="external-link">pairwiseAlignment</a></span><span class="op">(</span><span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/getSeq.html" class="external-link">getSeq</a></span><span class="op">(</span><span class="va">ROI4_1</span> <span class="op">+</span> <span class="fl">100</span><span class="op">)</span>, <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/getSeq.html" class="external-link">getSeq</a></span><span class="op">(</span><span class="va">ROI4_1</span><span class="op">$</span><span class="va">query</span> <span class="op">+</span> <span class="fl">100</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/reverseComplement.html" class="external-link">reverseComplement</a></span><span class="op">(</span><span class="op">)</span>, type<span class="op">=</span><span class="st">"global"</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Global PairwiseAlignmentsSingleSubject (1 of 1)</span>
<span class="co">## pattern: <span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">CCC</span><span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">CC</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">CC</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">GGGG</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #AFFFFF;">GGG</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">A</span>...<span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">AAAA</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">CC</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">G</span>-</span>
<span class="co">## subject: <span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">CCC</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">CC</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">CC</span><span style="color: #000000; background-color: #AFFFFF;">GG</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #AFFFFF;">GG</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">A</span>...<span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">TTT</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">C</span>--<span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">AA</span>---<span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span></span>
<span class="co">## score: -1312.463</span></code></pre>
<div class="sourceCode" id="cb42"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Only one hit</span>
<span class="co">#vmatchPattern(genomes$Osa$Chr1[1730795:1731145], genomes$Osa, with.indels = TRUE, max.mismatch = 10)</span></code></pre></div>
</div>
<div class="section level3">
<h3 id="falsefalse">FALSE/FALSE<a class="anchor" aria-label="anchor" href="#falsefalse"></a>
</h3>
<div class="sourceCode" id="cb43"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ROI4_2</span> <span class="op">&lt;-</span> <span class="va">x</span><span class="op">[</span><span class="fl">5</span><span class="op">]</span>
<span class="co"># As character for convenience of BLAST searching.</span>
<span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/getSeq.html" class="external-link">getSeq</a></span><span class="op">(</span><span class="va">ROI4_2</span><span class="op">)</span>      |&gt; <span class="fu"><a href="https://rdrr.io/r/base/character.html" class="external-link">as.character</a></span><span class="op">(</span><span class="op">)</span>  <span class="co"># Only one perfect BLAST hit at the NCBI, to the Oki genome.</span></code></pre></div>
<pre><code><span class="co">## [1] "AAAGGCTGCTCTAGAATTTTTGATCAAGCGGCAATTATCATTTGCTTGAATTTTATGCCAAAACTTCCTTCTTTGCATTTCGCCGAAAACACTGATAAGAAACGTTTTCGGCAAAAGCGGAAAGTGTCAACTTCCGCGAAAGGCGCCGTTCAATTTTCTAGTGCCTTTTCGAGCCCATTA"</span></code></pre>
<div class="sourceCode" id="cb45"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/getSeq.html" class="external-link">getSeq</a></span><span class="op">(</span><span class="va">ROI4_2</span><span class="op">$</span><span class="va">query</span><span class="op">)</span>|&gt; <span class="fu"><a href="https://rdrr.io/r/base/character.html" class="external-link">as.character</a></span><span class="op">(</span><span class="op">)</span>  <span class="co"># No relevant BLAST hit at the NCBI (evalues &gt;= 1e-4)</span></code></pre></div>
<pre><code><span class="co">## [1] "CTCAGGATCGACATGCTTCTTAGAAACCACTGATTTTTCGAAGAAAGGTGCGAAATTTTAAGATCAAGAAAAATCTTAAATTCAACAATTTTTTTTGAAAATGGAAAAAAAAATTTTTGTCACAGCGCGATTGAACGCTAACCCTTCCCATTTCTTCAATCTTTTTCCGATACGAAGCATG"</span></code></pre>
<div class="sourceCode" id="cb47"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/matchPattern.html" class="external-link">vmatchPattern</a></span><span class="op">(</span><span class="va">genomes</span><span class="op">$</span><span class="va">OKI2018.I69</span><span class="op">$</span><span class="va">chr1</span><span class="op">[</span><span class="fl">554365</span><span class="op">:</span><span class="fl">554546</span><span class="op">]</span>, <span class="va">genomes</span><span class="op">$</span><span class="va">OKI2018.I69</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## GRanges object with 1 range and 0 metadata columns:</span>
<span class="co">##       seqnames        ranges strand</span>
<span class="co">##          &lt;Rle&gt;     &lt;IRanges&gt;  &lt;Rle&gt;</span>
<span class="co">##   [1]     chr1 554365-554546      +</span>
<span class="co">##   -------</span>
<span class="co">##   seqinfo: 19 sequences from OKI2018.I69 genome</span></code></pre>
<div class="sourceCode" id="cb49"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># unaligned mapped region (+ / -)</span>
<span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/pairwiseAlignment.html" class="external-link">pairwiseAlignment</a></span><span class="op">(</span><span class="va">ROI4_2</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Global PairwiseAlignmentsSingleSubject (1 of 1)</span>
<span class="co">## pattern: <span style="color: #000000; background-color: #FFAFAF;">AA</span>-<span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">GG</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #FFD7AF;">TTTTT</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #AFFFFF;">GG</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">A</span>...<span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #FFD7AF;">TTTT</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span>--<span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">CC</span><span style="color: #000000; background-color: #FFD7AF;">TTTT</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">CCC</span><span style="color: #000000; background-color: #FFAFAF;">A</span>---<span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #FFAFAF;">A</span></span>
<span class="co">## subject: <span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">GG</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">AAA</span><span style="color: #000000; background-color: #AFFFAF;">CC</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">G</span>-<span style="color: #000000; background-color: #FFAFAF;">A</span>...<span style="color: #000000; background-color: #AFFFAF;">CC</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">TTT</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">TTTTT</span><span style="color: #000000; background-color: #AFFFAF;">CC</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">G</span></span>
<span class="co">## score: -477.1049</span></code></pre>
<div class="sourceCode" id="cb51"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/pairwiseAlignment.html" class="external-link">pairwiseAlignment</a></span><span class="op">(</span><span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/getSeq.html" class="external-link">getSeq</a></span><span class="op">(</span><span class="va">ROI4_2</span><span class="op">)</span>, <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/getSeq.html" class="external-link">getSeq</a></span><span class="op">(</span><span class="va">ROI4_2</span><span class="op">$</span><span class="va">query</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/reverseComplement.html" class="external-link">reverseComplement</a></span><span class="op">(</span><span class="op">)</span> <span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Global PairwiseAlignmentsSingleSubject (1 of 1)</span>
<span class="co">## pattern: <span style="color: #000000; background-color: #FFAFAF;">AAA</span><span style="color: #000000; background-color: #AFFFFF;">GG</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span>---<span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #FFD7AF;">TTTTT</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #AFFFFF;">GG</span>...<span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #FFD7AF;">T</span>--<span style="color: #000000; background-color: #FFD7AF;">TTT</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">A</span>-<span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">CC</span><span style="color: #000000; background-color: #FFD7AF;">TTTT</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">CCC</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #FFAFAF;">A</span>-</span>
<span class="co">## subject: <span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #AFFFFF;">GG</span><span style="color: #000000; background-color: #FFAFAF;">AAAAA</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">AAA</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">GG</span>...<span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">GG</span><span style="color: #000000; background-color: #FFD7AF;">TTT</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">CC</span>--<span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span></span>
<span class="co">## score: -429.3151</span></code></pre>
<div class="sourceCode" id="cb53"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Identity of ~50% but no striking difference between strands</span>

<span class="co"># unaligned mapped region with flanking colinear alignments.</span>
<span class="va">gbs</span><span class="op">$</span><span class="va">Oki_Osa</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/findOverlaps-methods.html" class="external-link">subsetByOverlaps</a></span><span class="op">(</span><span class="va">ROI4_2</span> <span class="op">+</span> <span class="fl">1</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## GBreaks object with 2 ranges and 8 metadata columns:</span>
<span class="co">##       seqnames        ranges strand |     score                query      Arm</span>
<span class="co">##          &lt;Rle&gt;     &lt;IRanges&gt;  &lt;Rle&gt; | &lt;numeric&gt;            &lt;GRanges&gt; &lt;factor&gt;</span>
<span class="co">##   [1]     chr1 554168-554365      - |       450 Chr1:1728109-1728321    short</span>
<span class="co">##   [2]     chr1 554546-554692      - |       564 Chr1:1727781-1727927    short</span>
<span class="co">##                   rep   repOvlp transcripts        flag    nonCoa</span>
<span class="co">##       &lt;CharacterList&gt; &lt;integer&gt;       &lt;Rle&gt; &lt;character&gt; &lt;logical&gt;</span>
<span class="co">##   [1]            &lt;NA&gt;         0        &lt;NA&gt;         Col     FALSE</span>
<span class="co">##   [2]            &lt;NA&gt;         0        &lt;NA&gt;         Col     FALSE</span>
<span class="co">##   -------</span>
<span class="co">##   seqinfo: 19 sequences from OKI2018.I69 genome</span></code></pre>
<div class="sourceCode" id="cb55"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/pairwiseAlignment.html" class="external-link">pairwiseAlignment</a></span><span class="op">(</span><span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/getSeq.html" class="external-link">getSeq</a></span><span class="op">(</span><span class="va">ROI4_2</span> <span class="op">+</span> <span class="fl">150</span><span class="op">)</span>, <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/getSeq.html" class="external-link">getSeq</a></span><span class="op">(</span><span class="va">ROI4_2</span><span class="op">$</span><span class="va">query</span> <span class="op">+</span> <span class="fl">150</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/reverseComplement.html" class="external-link">reverseComplement</a></span><span class="op">(</span><span class="op">)</span>, type<span class="op">=</span><span class="st">"global"</span><span class="op">)</span>|&gt; <span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/PairwiseAlignments-io.html" class="external-link">writePairwiseAlignments</a></span><span class="op">(</span>b<span class="op">=</span><span class="fl">60</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## ########################################</span>
<span class="co">## # Program: Biostrings (version 2.62.0), a Bioconductor package</span>
<span class="co">## # Rundate: Wed Mar 30 19:15:51 2022</span>
<span class="co">## ########################################</span>
<span class="co">## #=======================================</span>
<span class="co">## #</span>
<span class="co">## # Aligned_sequences: 2</span>
<span class="co">## # 1: P1</span>
<span class="co">## # 2: S1</span>
<span class="co">## # Matrix: NA</span>
<span class="co">## # Gap_penalty: 14.0</span>
<span class="co">## # Extend_penalty: 4.0</span>
<span class="co">## #</span>
<span class="co">## # Length: 513</span>
<span class="co">## # Identity:     344/513 (67.1%)</span>
<span class="co">## # Similarity:    NA/513 (NA%)</span>
<span class="co">## # Gaps:          65/513 (12.7%)</span>
<span class="co">## # Score: -421.8027</span>
<span class="co">## #</span>
<span class="co">## #</span>
<span class="co">## #=======================================</span>
<span class="co">## </span>
<span class="co">## P1                 1 GAAAAGAACCCCTCCGATCTCGAAATCGATCATAATTGGAACTTTGTTTTTTATTTTACG     60</span>
<span class="co">##                       |               | || ||||||| ||||||||||||| |||||  ||||||  |</span>
<span class="co">## S1                 1 AA---------------TTTCAAAATCGACCATAATTGGAACTCTGTTTACTATTTTTTG     45</span>
<span class="co">## </span>
<span class="co">## P1                61 AATGCTCATCTTTT-------TATTTTTTCGAAAATGGAATTTTTTTTTG-CAG------    106</span>
<span class="co">##                       |||||||| ||||       ||||||||| |   ||||||||||      |||      </span>
<span class="co">## S1                46 GATGCTCATTTTTTATAAGAATATTTTTTC-AGCGTGGAATTTTTGCAGATCAGGAAAAC    104</span>
<span class="co">## </span>
<span class="co">## P1               107 AAGGAAGGTT-TGAATTTTTCATTTTTGGCGGA-CAATGGGCTCAAAAAGGCTGCTCTAG    164</span>
<span class="co">##                      ||||||| |  ||||||||| |||  ||||||| |||  ||||| | | |  |  | | |</span>
<span class="co">## S1               105 AAGGAAGCTCGTGAATTTTTTATTCGTGGCGGAACAAAAGGCTCGACATGCTTCGTATCG    164</span>
<span class="co">## </span>
<span class="co">## P1               165 ---AATTTTTGATCAAGCGGCAA---TTATCATTTGCTTGAATTTTATGCCAAAACTTCC    218</span>
<span class="co">##                         ||   ||||  ||  || ||   ||| | ||   | |     | || ||||| ||  </span>
<span class="co">## S1               165 GAAAAAGATTGAAGAAATGGGAAGGGTTAGCGTTCAATCGCGC--TGTGACAAAAATTTT    222</span>
<span class="co">## </span>
<span class="co">## P1               219 TTCTTTGCATTTCGCCGAAAACACTGAT-AAGAAACGTTTTCGGCAAAAGCGGAAAGTGT    277</span>
<span class="co">##                      || ||| |||||  |  |||| | || | ||   | | |||   |   | |  ||| | |</span>
<span class="co">## S1               223 TT-TTTCCATTTT-CAAAAAAAATTGTTGAATTTAAGATTTTT-CTTGATCTTAAAATTT    279</span>
<span class="co">## </span>
<span class="co">## P1               278 C--AACTTCC-GCGAAAGGCGCCGTTCAAT--TTTCTA-GTGCCTTTTCGAGCCCATTA-    330</span>
<span class="co">##                      |  | ||| |  |||||        ||| |  |||||| |   | | |||| ||  | | </span>
<span class="co">## S1               280 CGCACCTTTCTTCGAAAAA------TCAGTGGTTTCTAAGAAGCATGTCGATCC--TGAG    331</span>
<span class="co">## </span>
<span class="co">## P1               331 AGCTGCTGCAGCAGATGCATGGAACTACGTGTTTGACGACTTTGAAAAAGGGATTTCGGC    390</span>
<span class="co">##                      ||||  ||||||| || |||||||||||||  ||||||||||||||||||||||||||||</span>
<span class="co">## S1               332 AGCTTGTGCAGCAAATACATGGAACTACGTAATTGACGACTTTGAAAAAGGGATTTCGGC    391</span>
<span class="co">## </span>
<span class="co">## P1               391 CGCTGCTTTGAGTCGC---GCTGGACCGCGGGGCGTCGCCTGATCCGCTTTCGTTGATCT    447</span>
<span class="co">##                      || |   ||| ||||    |||||||||||||||||||||||||||||||||||||||||</span>
<span class="co">## S1               392 CGGT---TTGCGTCGATGAGCTGGACCGCGGGGCGTCGCCTGATCCGCTTTCGTTGATCT    448</span>
<span class="co">## </span>
<span class="co">## P1               448 TCGCATTGCTCGACAGTTTGCGCTCCATCTTGT    480</span>
<span class="co">##                      ||||||||||||||||||||||||||||||   </span>
<span class="co">## S1               449 TCGCATTGCTCGACAGTTTGCGCTCCATCTGTA    481</span>
<span class="co">## </span>
<span class="co">## </span>
<span class="co">## #---------------------------------------</span>
<span class="co">## #---------------------------------------</span></code></pre>
</div>
<div class="section level3">
<h3 id="truetrue">TRUE/TRUE<a class="anchor" aria-label="anchor" href="#truetrue"></a>
</h3>
<div class="sourceCode" id="cb57"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ROI4_3</span> <span class="op">&lt;-</span> <span class="va">x</span><span class="op">[</span><span class="fl">12</span><span class="op">]</span>
<span class="va">ROI4_3</span> |&gt; <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">##   seqnames  start    end width strand query.seqnames query.start query.end</span>
<span class="co">## 1     chr1 561647 562123   477      *           Chr1     1722713   1723219</span>
<span class="co">##   query.width query.strand query.Arm query.rep query.repOvlp</span>
<span class="co">## 1         507            *     short       rnd           195</span>
<span class="co">##         query.transcripts   Arm    rep repOvlp          transcripts     Class</span>
<span class="co">## 1 g383.t2;g383.t1;g383.t3 short tandem     158 g57.t3;g57.t1;g57.t2 TRUE TRUE</span></code></pre>
<div class="sourceCode" id="cb59"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># As character for convenience of BLAST searching.</span>
<span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/getSeq.html" class="external-link">getSeq</a></span><span class="op">(</span><span class="va">ROI4_3</span><span class="op">)</span>      |&gt; <span class="fu"><a href="https://rdrr.io/r/base/character.html" class="external-link">as.character</a></span><span class="op">(</span><span class="op">)</span>  <span class="co"># Only one perfect BLAST hit at the NCBI, to the Oki genome.</span></code></pre></div>
<pre><code><span class="co">## [1] "ATCATAGCAATCGGTATTTTGGTAGAGATTAACGAGCGAAAAGGAACATCCAGTAGCTCTTCAGTCCCTTGAATCCTTCGTGAAATGTGGTTGTTGTTGTTCCTGACGAATTTGAAGTCGTTGTCTTAGTCGTTGTTGTTGTTGAACGTGTAGTGGTTTTTTTAGTGGTTGTTGTCGATGTCAATGATAATGTGGATGTTGTTGTTTTTGTTGTTTGCGTAGACATTGTTGTTGATGATTTTGAGGGTGCTGGCGTATTGGAAATAGAACTTGTCATCTGATAAGTCGACAGATTTGTCATGTCAGTCGTCAGGGATGATTCTGCCGTTGTTAGATTGTCAACACTCGGATCGAGATCTTCATTTTTTCCAAGGCAAAGAGGAATCGTGATTCCAAGAATTGTTACTATCACTGCAGAAACTTCATTTCACACCGAAATTGTTATTCAAATATAATTCTTAAATACCTAGCGCGAAT"</span></code></pre>
<div class="sourceCode" id="cb61"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/getSeq.html" class="external-link">getSeq</a></span><span class="op">(</span><span class="va">ROI4_3</span><span class="op">$</span><span class="va">query</span><span class="op">)</span>|&gt; <span class="fu"><a href="https://rdrr.io/r/base/character.html" class="external-link">as.character</a></span><span class="op">(</span><span class="op">)</span>  <span class="co"># No relevant BLAST hit at the NCBI (evalues &gt;= 1e-4)</span></code></pre></div>
<pre><code><span class="co">## [1] "CTTGTTTATGAGTTAAAAAAAACTTTTTAATATATCTTTGCTTCAAAAGCTTGAATTAAGTTACGACGATTATTATTGTTATCCTCGGATGTATCCCGGTTTTTATCGAAAAAAACAATGAAATAACCACAAACTCGACAACAATGCCTGACTCAACAACAATTATCATCACTACAACGACTGTTCCTGACTTAATACTATTGAACGAAACTCTGACAACAACAACACTGACCTCGAAATCGAGCGCCACTTTGTCAACCACTACAAAAACTATTACAACCAACGCAACAACGACTTCCGCGACTACAACGACAACAACGAGTTCAACAACAACTACAACAACATCGACAACACCAACAACAACCGCGACCGGTTCTTCTACGACTACAACTCCAACCACAACATCAACAACATTGAACACAACGTCAACGACATCAACAACAACATCAAAATGTGTTCCATTTGACGAGTACGACGATATTTGTCCGTAATTTCTGGCAATTTGTC"</span></code></pre>
<div class="sourceCode" id="cb63"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># unaligned mapped region (+ / -)</span>
<span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/pairwiseAlignment.html" class="external-link">pairwiseAlignment</a></span><span class="op">(</span><span class="va">ROI4_3</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Global PairwiseAlignmentsSingleSubject (1 of 1)</span>
<span class="co">## pattern: <span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #AFFFFF;">GG</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">TTTT</span><span style="color: #000000; background-color: #AFFFFF;">GG</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #AFFFFF;">G</span>...-<span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #FFAFAF;">AAA</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">CC</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #FFD7AF;">T</span>-----</span>
<span class="co">## subject: <span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">TT</span>----------<span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">TTT</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #FFAFAF;">AAAAAAAA</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span>...<span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">TTT</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">CC</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #FFD7AF;">TTT</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">GG</span><span style="color: #000000; background-color: #AFFFAF;">C</span>---<span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #FFD7AF;">TTT</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span></span>
<span class="co">## score: -1463.191</span></code></pre>
<div class="sourceCode" id="cb65"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/pairwiseAlignment.html" class="external-link">pairwiseAlignment</a></span><span class="op">(</span><span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/getSeq.html" class="external-link">getSeq</a></span><span class="op">(</span><span class="va">ROI4_3</span><span class="op">)</span>, <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/getSeq.html" class="external-link">getSeq</a></span><span class="op">(</span><span class="va">ROI4_3</span><span class="op">$</span><span class="va">query</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/reverseComplement.html" class="external-link">reverseComplement</a></span><span class="op">(</span><span class="op">)</span> <span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Global PairwiseAlignmentsSingleSubject (1 of 1)</span>
<span class="co">## pattern: <span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #AFFFFF;">GG</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">TTTT</span><span style="color: #000000; background-color: #AFFFFF;">GG</span>-----<span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span>...<span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">AAA</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #FFAFAF;">AAA</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">CC</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #FFD7AF;">T</span></span>
<span class="co">## subject: <span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">AAA</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">CC</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">AAA</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #AFFFFF;">GG</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">AAA</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span>...<span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #FFAFAF;">AAAAA</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">TTTTTTTT</span><span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">AAA</span><span style="color: #000000; background-color: #AFFFAF;">C</span>---<span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #AFFFFF;">G</span></span>
<span class="co">## score: -1006.199</span></code></pre>
<div class="sourceCode" id="cb67"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Identity of ~50% but no striking difference between strands</span>

<span class="co"># unaligned mapped region with flanking colinear alignments.</span>
<span class="va">gbs</span><span class="op">$</span><span class="va">Oki_Osa</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/findOverlaps-methods.html" class="external-link">subsetByOverlaps</a></span><span class="op">(</span><span class="va">ROI4_3</span> <span class="op">+</span> <span class="fl">1</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## GBreaks object with 2 ranges and 8 metadata columns:</span>
<span class="co">##       seqnames        ranges strand |     score                query      Arm</span>
<span class="co">##          &lt;Rle&gt;     &lt;IRanges&gt;  &lt;Rle&gt; | &lt;numeric&gt;            &lt;GRanges&gt; &lt;factor&gt;</span>
<span class="co">##   [1]     chr1 561142-561646      - |      1387 Chr1:1723220-1723727    short</span>
<span class="co">##   [2]     chr1 562124-563743      - |      2869 Chr1:1720807-1722712    short</span>
<span class="co">##                   rep   repOvlp transcripts        flag    nonCoa</span>
<span class="co">##       &lt;CharacterList&gt; &lt;integer&gt;       &lt;Rle&gt; &lt;character&gt; &lt;logical&gt;</span>
<span class="co">##   [1]            &lt;NA&gt;         0        &lt;NA&gt;         Col     FALSE</span>
<span class="co">##   [2]   LowComplexity         0        &lt;NA&gt;         Col     FALSE</span>
<span class="co">##   -------</span>
<span class="co">##   seqinfo: 19 sequences from OKI2018.I69 genome</span></code></pre>
<div class="sourceCode" id="cb69"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/pairwiseAlignment.html" class="external-link">pairwiseAlignment</a></span><span class="op">(</span><span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/getSeq.html" class="external-link">getSeq</a></span><span class="op">(</span><span class="va">ROI4_3</span> <span class="op">+</span> <span class="fl">150</span><span class="op">)</span>, <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/getSeq.html" class="external-link">getSeq</a></span><span class="op">(</span><span class="va">ROI4_3</span><span class="op">$</span><span class="va">query</span> <span class="op">+</span> <span class="fl">150</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/reverseComplement.html" class="external-link">reverseComplement</a></span><span class="op">(</span><span class="op">)</span>, type<span class="op">=</span><span class="st">"global"</span><span class="op">)</span>|&gt; <span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/PairwiseAlignments-io.html" class="external-link">writePairwiseAlignments</a></span><span class="op">(</span>b<span class="op">=</span><span class="fl">60</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## ########################################</span>
<span class="co">## # Program: Biostrings (version 2.62.0), a Bioconductor package</span>
<span class="co">## # Rundate: Wed Mar 30 19:15:53 2022</span>
<span class="co">## ########################################</span>
<span class="co">## #=======================================</span>
<span class="co">## #</span>
<span class="co">## # Aligned_sequences: 2</span>
<span class="co">## # 1: P1</span>
<span class="co">## # 2: S1</span>
<span class="co">## # Matrix: NA</span>
<span class="co">## # Gap_penalty: 14.0</span>
<span class="co">## # Extend_penalty: 4.0</span>
<span class="co">## #</span>
<span class="co">## # Length: 839</span>
<span class="co">## # Identity:     499/839 (59.5%)</span>
<span class="co">## # Similarity:    NA/839 (NA%)</span>
<span class="co">## # Gaps:          94/839 (11.2%)</span>
<span class="co">## # Score: -1188.32</span>
<span class="co">## #</span>
<span class="co">## #</span>
<span class="co">## #=======================================</span>
<span class="co">## </span>
<span class="co">## P1                 1 ATTATCATTCCGTATAAAAATAGAGAGGAGCATCTGCGCGCCCTGCTGAATACATTGCAT     60</span>
<span class="co">##                      ||||| |||||||| || ||| ||||||| || ||||||||  |||| ||||||||||||</span>
<span class="co">## S1                 1 ATTATTATTCCGTACAAGAATCGAGAGGAACACCTGCGCGCGTTGCTTAATACATTGCAT     60</span>
<span class="co">## </span>
<span class="co">## P1                61 CCAATTCTTCAAAGACAAAACACCGCATACTGCATTTATGTGGGCGAGCAGCATGACGAT    120</span>
<span class="co">##                      || | ||| ||  | || |||||||| |||||||||||||||| |||||| |||||||||</span>
<span class="co">## S1                61 CCGACTCTGCAGCGGCAGAACACCGCGTACTGCATTTATGTGGCCGAGCAACATGACGAT    120</span>
<span class="co">## </span>
<span class="co">## P1               121 GGCAGATTTAATAAAGGTTTGTTTATTTTTATCATAGCAATCGGTATTTTGG-----TAG    175</span>
<span class="co">##                      ||||||||||| ||||||  ||||  ||||  || |      |  |||  ||     || </span>
<span class="co">## S1               121 GGCAGATTTAACAAAGGTCAGTTTTATTTTGACAAATTGCCAGAAATTACGGACAAATAT    180</span>
<span class="co">## </span>
<span class="co">## P1               176 AGATTAACGAGCGAAAAGGAACA-----TCCAGTAGCTCTTCAGTCCCTTGAATCCTTC-    229</span>
<span class="co">##                       |    ||  |  ||| ||||||     |   || | | || |   | ||||  | ||  </span>
<span class="co">## S1               181 CGTCGTACTCGTCAAATGGAACACATTTTGATGTTGTTGTTGATGTCGTTGA--CGTTGT    238</span>
<span class="co">## </span>
<span class="co">## P1               230 GTGAAATGTGGTTGTTGTTGTTCCTGACGAATTTGAAGTCGTTGTCTTA---GTCGTTGT    286</span>
<span class="co">##                      ||  ||||| |||| ||||||   ||  |   ||| |||||| |    |   ||||  ||</span>
<span class="co">## S1               239 GTTCAATGTTGTTGATGTTGTGGTTGGAG---TTGTAGTCGTAGAAGAACCGGTCGCGGT    295</span>
<span class="co">## </span>
<span class="co">## P1               287 TGTTGTTGAACGTGTAGTGGTTTTTTTAGTGGTTGTTG------TCGATGTCAATGATAA    340</span>
<span class="co">##                      ||||||||    ||| |  ||| || |||| |||||||      | | ||||  || || </span>
<span class="co">## S1               296 TGTTGTTGGTGTTGTCGATGTTGTTGTAGTTGTTGTTGAACTCGTTGTTGTCGTTG-TAG    354</span>
<span class="co">## </span>
<span class="co">## P1               341 T-GTGGATGTTGTTGTTTTTGTTGTTTGCG-TAGACATTGTTGTTGATGATTTTGAGGGT    398</span>
<span class="co">##                      | | ||| || ||||||   |||| |||   |||   |||| || | |||    | ||  </span>
<span class="co">## S1               355 TCGCGGAAGTCGTTGTTGC-GTTGGTTGTAATAGTTTTTGTAGTGGTTGACAAAGTGGC-    412</span>
<span class="co">## </span>
<span class="co">## P1               399 GCTGGCGTATTGGAAATA---GAACTTGTCATCTGA---TAAGTCGACAGATTT--GTCA    450</span>
<span class="co">##                      |||  ||  || ||  |    |   ||||  || ||   |   || | ||  ||  ||||</span>
<span class="co">## S1               413 GCT--CGATTTCGAGGTCAGTGTTGTTGTTGTCAGAGTTTCGTTCAATAGTATTAAGTCA    470</span>
<span class="co">## </span>
<span class="co">## P1               451 TGT-CAGTCGTC--AGGGATGATTCTGCCGTTGTTAGATTGTCAA-CACTCGGATCGAGA    506</span>
<span class="co">##                       |  |||||||   || ||||||  |   |||||| ||  ||||  || |    ||||| </span>
<span class="co">## S1               471 GGAACAGTCGTTGTAGTGATGATAATT--GTTGTT-GA--GTCAGGCATTGTTGTCGAGT    525</span>
<span class="co">## </span>
<span class="co">## P1               507 TC--------TTCATT-TTTTCCAAGGCAAAGAGGAATCGTGAT---TCCAAGAATTGTT    554</span>
<span class="co">##                      |         |||||| ||||    |  |||    || || |||   ||| || ||    </span>
<span class="co">## S1               526 TTGTGGTTATTTCATTGTTTTTTTCGATAAA----AACCGGGATACATCCGAGGATAACA    581</span>
<span class="co">## </span>
<span class="co">## P1               555 ACTATCACTGCAGAAACTTCATTTCAC------ACCGAAATTGTTATTCAAATATAATTC    608</span>
<span class="co">##                      |  || |  |  | ||||| |||  |       |   |||    |||| |||  |  || </span>
<span class="co">## S1               582 ATAATAATCGTCGTAACTTAATTCAAGCTTTTGAAGCAAAGATATATTAAAAAGTTTTTT    641</span>
<span class="co">## </span>
<span class="co">## P1               609 TTAAATACCTAGCGCGAATCCAACAATAATCCTTTTTCGTTTTCTCCAACCTCTTTTGTA    668</span>
<span class="co">##                      |||| |    | |   || || |  | || || |||||| |||||||| ||||||| | |</span>
<span class="co">## S1               642 TTAACTCATAAAC---AAGCCGATGAGAACCCGTTTTCGCTTTCTCCATCCTCTTT-GAA    697</span>
<span class="co">## </span>
<span class="co">## P1               669 AATGTTTGCAATATTATATTCTTCGAGAAAGAATATTTATAAGTTTAATT----AATGCT    724</span>
<span class="co">##                      ||||||  || ||  |       | ||   ||| ||||| | |  |  ||    ||   |</span>
<span class="co">## S1               698 AATGTT--CAGTAAGA-------CTAGTTTGAAAATTTACACGCGTCTTTGCAGAACATT    748</span>
<span class="co">## </span>
<span class="co">## P1               725 GAGT-GAATATTCGTCTATAAACATTTTCTTGACA-CACTCACAAAAAATCCGTT----    777</span>
<span class="co">##                       | | |||  ||   | | |||   ||||||  |  |  |    | |    | ||    </span>
<span class="co">## S1               749 AACTTGAAACTTTTACAAAAAATCCTTTCTTCTCGTCTGTGGTCATATTGACTTTTTCA    807</span>
<span class="co">## </span>
<span class="co">## </span>
<span class="co">## #---------------------------------------</span>
<span class="co">## #---------------------------------------</span></code></pre>
</div>
</div>
<div class="section level2">
<h2 id="what-still-interrupts-colinearity-once-repeats-are-ruled-out-roi5">What still interrupts colinearity once repeats are ruled out ? (ROI5)<a class="anchor" aria-label="anchor" href="#what-still-interrupts-colinearity-once-repeats-are-ruled-out-roi5"></a>
</h2>
<p>We start with a random region</p>
<div class="sourceCode" id="cb71"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">coa</span><span class="op">$</span><span class="va">Oki_Kum</span> |&gt; <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/flagAll.html" class="external-link">flagAll</a></span><span class="op">(</span><span class="op">)</span> |&gt; <span class="fu">plyranges</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/plyranges/man/tidyverse-reexports.html" class="external-link">slice</a></span><span class="op">(</span><span class="fl">200</span><span class="op">:</span><span class="fl">210</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">##    seqnames   start     end width strand query.seqnames query.start query.end</span>
<span class="co">## 1      chr1 1436172 1436279   108      +     contig_3_1     4439096   4439203</span>
<span class="co">## 2      chr1 1436544 1466354 29811      +    contig_27_1     1409471   1440844</span>
<span class="co">## 3      chr1 1466359 1470295  3937      +    contig_27_1     1441222   1445179</span>
<span class="co">## 4      chr1 1470311 1471259   949      +    contig_27_1     1445585   1446558</span>
<span class="co">## 5      chr1 1471276 1478839  7564      +    contig_27_1     1447542   1456167</span>
<span class="co">## 6      chr1 1478840 1479523   684      -    contig_11_1      467930    468616</span>
<span class="co">## 7      chr1 1479524 1481440  1917      +    contig_27_1     1456168   1458038</span>
<span class="co">## 8      chr1 1483131 1483268   138      +    contig_43_1       62949     63085</span>
<span class="co">## 9      chr1 1484885 1485855   971      -    contig_16_1     4442527   4444265</span>
<span class="co">## 10     chr1 1485856 1486085   230      -    contig_16_1     4441532   4441769</span>
<span class="co">## 11     chr1 1486156 1486497   342      +    contig_16_1     4446412   4446746</span>
<span class="co">##    query.width query.strand    query.rep query.repOvlp</span>
<span class="co">## 1          108            *          rnd           108</span>
<span class="co">## 2        31374            * tandem, ....          3619</span>
<span class="co">## 3         3958            * rnd, unknown           676</span>
<span class="co">## 4          974            *           NA             0</span>
<span class="co">## 5         8626            * rnd, unk....          2355</span>
<span class="co">## 6          687            *      unknown           684</span>
<span class="co">## 7         1871            * rnd, unknown          1698</span>
<span class="co">## 8          137            *           NA             0</span>
<span class="co">## 9         1739            * unknown, rnd           353</span>
<span class="co">## 10         238            *           NA             0</span>
<span class="co">## 11         335            *           NA             0</span>
<span class="co">##                                                                                              query.transcripts</span>
<span class="co">## 1                                                                                                     g8176.t1</span>
<span class="co">## 2  g3891.t1;g3892.t1;g3893.t1;g3894.t1;g3895.t1;g3898.t1;g3899.t1;g3896.t1;g3897.t1;g3900.t2;g3900.t1;g3901.t1</span>
<span class="co">## 3                                                                                   g3902.t1;g3902.t2;g3901.t1</span>
<span class="co">## 4                                                                                            g3902.t1;g3902.t2</span>
<span class="co">## 5                                                                                            g3904.t1;g3903.t1</span>
<span class="co">## 6                                                                                                         &lt;NA&gt;</span>
<span class="co">## 7                                                                                                         &lt;NA&gt;</span>
<span class="co">## 8                                                                                                    g12993.t1</span>
<span class="co">## 9                                                                                                         &lt;NA&gt;</span>
<span class="co">## 10                                                                                                        &lt;NA&gt;</span>
<span class="co">## 11                                                                                                        &lt;NA&gt;</span>
<span class="co">##    score   Arm          rep repOvlp</span>
<span class="co">## 1    108 short          rnd     108</span>
<span class="co">## 2  29811 short tandem, ....    1178</span>
<span class="co">## 3   3937 short rnd, unknown     697</span>
<span class="co">## 4    949 short           NA       0</span>
<span class="co">## 5   7564 short rnd, unk....     562</span>
<span class="co">## 6    684 short          rnd     684</span>
<span class="co">## 7   1917 short rnd, unk....    1592</span>
<span class="co">## 8    138 short           NA       0</span>
<span class="co">## 9    971 short       tandem       0</span>
<span class="co">## 10   230 short           NA       0</span>
<span class="co">## 11   342 short           NA       0</span>
<span class="co">##                                                        transcripts flag nonCoa</span>
<span class="co">## 1                                                          g302.t1 &lt;NA&gt;   TRUE</span>
<span class="co">## 2  g302.t1;g303.t1;g304.t1;g306.t1;g307.t1;g308.t1;g313.t1;g314.t1 &lt;NA&gt;  FALSE</span>
<span class="co">## 3                                                  g319.t1;g319.t2 &lt;NA&gt;  FALSE</span>
<span class="co">## 4                                                  g319.t1;g319.t2 &lt;NA&gt;   TRUE</span>
<span class="co">## 5                                                          g321.t1  Tra  FALSE</span>
<span class="co">## 6                                                             &lt;NA&gt; &lt;NA&gt;   TRUE</span>
<span class="co">## 7                                                             &lt;NA&gt; &lt;NA&gt;   TRUE</span>
<span class="co">## 8                                                             &lt;NA&gt; &lt;NA&gt;   TRUE</span>
<span class="co">## 9                                                             &lt;NA&gt; &lt;NA&gt;  FALSE</span>
<span class="co">## 10                                                            &lt;NA&gt; &lt;NA&gt;   TRUE</span>
<span class="co">## 11                                                            &lt;NA&gt; &lt;NA&gt;   TRUE</span></code></pre>
<p>Letâ€™s focus on alignments between chr1 and contig_27_1:</p>
<div class="sourceCode" id="cb73"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ROI5.range</span> <span class="op">&lt;-</span> <span class="va">coa</span><span class="op">$</span><span class="va">Oki_Kum</span><span class="op">[</span><span class="fl">200</span><span class="op">:</span><span class="fl">210</span><span class="op">]</span> |&gt; <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">seqnames</span> <span class="op">==</span> <span class="st">"chr1"</span>, <span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqnames</a></span><span class="op">(</span><span class="va">query</span><span class="op">)</span> <span class="op">==</span> <span class="st">"contig_27_1"</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/inter-range-methods.html" class="external-link">range</a></span><span class="op">(</span><span class="op">)</span>
<span class="va">ROI5.range</span></code></pre></div>
<pre><code><span class="co">## GBreaks object with 1 range and 1 metadata column:</span>
<span class="co">##       seqnames          ranges strand |                       query</span>
<span class="co">##          &lt;Rle&gt;       &lt;IRanges&gt;  &lt;Rle&gt; |                   &lt;GRanges&gt;</span>
<span class="co">##   [1]     chr1 1436544-1481440      * | contig_27_1:1409471-1458038</span>
<span class="co">##   -------</span>
<span class="co">##   seqinfo: 19 sequences from OKI2018.I69 genome</span></code></pre>
<div class="sourceCode" id="cb75"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ROI5</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/findOverlaps-methods.html" class="external-link">subsetByOverlaps</a></span><span class="op">(</span><span class="va">coa</span><span class="op">$</span><span class="va">Oki_Kum</span>, <span class="va">ROI5.range</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/sort.html" class="external-link">sort</a></span><span class="op">(</span>ignore.strand<span class="op">=</span><span class="cn">T</span><span class="op">)</span>

<span class="co"># Let's capture all the pairs that overlap that range:</span>
<span class="va">ROI5</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/sort.html" class="external-link">sort</a></span><span class="op">(</span>ignore.strand<span class="op">=</span><span class="cn">T</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">##   seqnames   start     end width strand query.seqnames query.start query.end</span>
<span class="co">## 1     chr1 1436544 1466354 29811      +    contig_27_1     1409471   1440844</span>
<span class="co">## 2     chr1 1466359 1470295  3937      +    contig_27_1     1441222   1445179</span>
<span class="co">## 3     chr1 1470311 1471259   949      +    contig_27_1     1445585   1446558</span>
<span class="co">## 4     chr1 1471276 1478839  7564      +    contig_27_1     1447542   1456167</span>
<span class="co">## 5     chr1 1478840 1479523   684      -    contig_11_1      467930    468616</span>
<span class="co">## 6     chr1 1479524 1481440  1917      +    contig_27_1     1456168   1458038</span>
<span class="co">## 7     chr1 2405232 2405393   162      -    contig_27_1     1445390   1445562</span>
<span class="co">## 8     chr1 9897740 9898012   273      +    contig_27_1     1447046   1447318</span>
<span class="co">## 9     chr2 3010219 3010391   173      -    contig_27_1     1441042   1441221</span>
<span class="co">##   query.width query.strand    query.rep query.repOvlp</span>
<span class="co">## 1       31374            * tandem, ....          3619</span>
<span class="co">## 2        3958            * rnd, unknown           676</span>
<span class="co">## 3         974            *           NA             0</span>
<span class="co">## 4        8626            * rnd, unk....          2355</span>
<span class="co">## 5         687            *      unknown           684</span>
<span class="co">## 6        1871            * rnd, unknown          1698</span>
<span class="co">## 7         173            *  tandem, rnd            68</span>
<span class="co">## 8         273            * rnd, unknown             2</span>
<span class="co">## 9         180            *          rnd           180</span>
<span class="co">##                                                                                             query.transcripts</span>
<span class="co">## 1 g3891.t1;g3892.t1;g3893.t1;g3894.t1;g3895.t1;g3898.t1;g3899.t1;g3896.t1;g3897.t1;g3900.t2;g3900.t1;g3901.t1</span>
<span class="co">## 2                                                                                  g3902.t1;g3902.t2;g3901.t1</span>
<span class="co">## 3                                                                                           g3902.t1;g3902.t2</span>
<span class="co">## 4                                                                                           g3904.t1;g3903.t1</span>
<span class="co">## 5                                                                                                        &lt;NA&gt;</span>
<span class="co">## 6                                                                                                        &lt;NA&gt;</span>
<span class="co">## 7                                                                                           g3902.t1;g3902.t2</span>
<span class="co">## 8                                                                                                        &lt;NA&gt;</span>
<span class="co">## 9                                                                                                    g3901.t1</span>
<span class="co">##   score   Arm          rep repOvlp</span>
<span class="co">## 1 29811 short tandem, ....    1178</span>
<span class="co">## 2  3937 short rnd, unknown     697</span>
<span class="co">## 3   949 short           NA       0</span>
<span class="co">## 4  7564 short rnd, unk....     562</span>
<span class="co">## 5   684 short          rnd     684</span>
<span class="co">## 6  1917 short rnd, unk....    1592</span>
<span class="co">## 7   162 short           NA       0</span>
<span class="co">## 8   273  long       tandem      24</span>
<span class="co">## 9   173 short           NA       0</span>
<span class="co">##                                                       transcripts flag nonCoa</span>
<span class="co">## 1 g302.t1;g303.t1;g304.t1;g306.t1;g307.t1;g308.t1;g313.t1;g314.t1 &lt;NA&gt;  FALSE</span>
<span class="co">## 2                                                 g319.t1;g319.t2 &lt;NA&gt;  FALSE</span>
<span class="co">## 3                                                 g319.t1;g319.t2 &lt;NA&gt;   TRUE</span>
<span class="co">## 4                                                         g321.t1  Tra  FALSE</span>
<span class="co">## 5                                                            &lt;NA&gt; &lt;NA&gt;   TRUE</span>
<span class="co">## 6                                                            &lt;NA&gt; &lt;NA&gt;   TRUE</span>
<span class="co">## 7                                                            &lt;NA&gt; &lt;NA&gt;   TRUE</span>
<span class="co">## 8                                                        g2641.t1 &lt;NA&gt;   TRUE</span>
<span class="co">## 9                                                        g4701.t1 &lt;NA&gt;   TRUE</span></code></pre>
<div class="sourceCode" id="cb77"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ROI5</span> |&gt; <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/arrange.html" class="external-link">arrange</a></span><span class="op">(</span><span class="va">query</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">##   seqnames   start     end width strand query.seqnames query.start query.end</span>
<span class="co">## 1     chr1 1478840 1479523   684      -    contig_11_1      467930    468616</span>
<span class="co">## 2     chr1 1436544 1466354 29811      +    contig_27_1     1409471   1440844</span>
<span class="co">## 3     chr2 3010219 3010391   173      -    contig_27_1     1441042   1441221</span>
<span class="co">## 4     chr1 1466359 1470295  3937      +    contig_27_1     1441222   1445179</span>
<span class="co">## 5     chr1 2405232 2405393   162      -    contig_27_1     1445390   1445562</span>
<span class="co">## 6     chr1 1470311 1471259   949      +    contig_27_1     1445585   1446558</span>
<span class="co">## 7     chr1 9897740 9898012   273      +    contig_27_1     1447046   1447318</span>
<span class="co">## 8     chr1 1471276 1478839  7564      +    contig_27_1     1447542   1456167</span>
<span class="co">## 9     chr1 1479524 1481440  1917      +    contig_27_1     1456168   1458038</span>
<span class="co">##   query.width query.strand    query.rep query.repOvlp</span>
<span class="co">## 1         687            *      unknown           684</span>
<span class="co">## 2       31374            * tandem, ....          3619</span>
<span class="co">## 3         180            *          rnd           180</span>
<span class="co">## 4        3958            * rnd, unknown           676</span>
<span class="co">## 5         173            *  tandem, rnd            68</span>
<span class="co">## 6         974            *           NA             0</span>
<span class="co">## 7         273            * rnd, unknown             2</span>
<span class="co">## 8        8626            * rnd, unk....          2355</span>
<span class="co">## 9        1871            * rnd, unknown          1698</span>
<span class="co">##                                                                                             query.transcripts</span>
<span class="co">## 1                                                                                                        &lt;NA&gt;</span>
<span class="co">## 2 g3891.t1;g3892.t1;g3893.t1;g3894.t1;g3895.t1;g3898.t1;g3899.t1;g3896.t1;g3897.t1;g3900.t2;g3900.t1;g3901.t1</span>
<span class="co">## 3                                                                                                    g3901.t1</span>
<span class="co">## 4                                                                                  g3902.t1;g3902.t2;g3901.t1</span>
<span class="co">## 5                                                                                           g3902.t1;g3902.t2</span>
<span class="co">## 6                                                                                           g3902.t1;g3902.t2</span>
<span class="co">## 7                                                                                                        &lt;NA&gt;</span>
<span class="co">## 8                                                                                           g3904.t1;g3903.t1</span>
<span class="co">## 9                                                                                                        &lt;NA&gt;</span>
<span class="co">##   score   Arm          rep repOvlp</span>
<span class="co">## 1   684 short          rnd     684</span>
<span class="co">## 2 29811 short tandem, ....    1178</span>
<span class="co">## 3   173 short           NA       0</span>
<span class="co">## 4  3937 short rnd, unknown     697</span>
<span class="co">## 5   162 short           NA       0</span>
<span class="co">## 6   949 short           NA       0</span>
<span class="co">## 7   273  long       tandem      24</span>
<span class="co">## 8  7564 short rnd, unk....     562</span>
<span class="co">## 9  1917 short rnd, unk....    1592</span>
<span class="co">##                                                       transcripts flag nonCoa</span>
<span class="co">## 1                                                            &lt;NA&gt; &lt;NA&gt;   TRUE</span>
<span class="co">## 2 g302.t1;g303.t1;g304.t1;g306.t1;g307.t1;g308.t1;g313.t1;g314.t1 &lt;NA&gt;  FALSE</span>
<span class="co">## 3                                                        g4701.t1 &lt;NA&gt;   TRUE</span>
<span class="co">## 4                                                 g319.t1;g319.t2 &lt;NA&gt;  FALSE</span>
<span class="co">## 5                                                            &lt;NA&gt; &lt;NA&gt;   TRUE</span>
<span class="co">## 6                                                 g319.t1;g319.t2 &lt;NA&gt;   TRUE</span>
<span class="co">## 7                                                        g2641.t1 &lt;NA&gt;   TRUE</span>
<span class="co">## 8                                                         g321.t1  Tra  FALSE</span>
<span class="co">## 9                                                            &lt;NA&gt; &lt;NA&gt;   TRUE</span></code></pre>
<div class="sourceCode" id="cb79"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ROI5</span> |&gt; <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/plotApairOfChrs.html" class="external-link">plotApairOfChrs</a></span><span class="op">(</span>xlim <span class="op">=</span> <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/gb2xlim.html" class="external-link">gb2xlim</a></span><span class="op">(</span><span class="va">ROI5.range</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="RegionsOfInterest_files/figure-html/ROI5-1.png" width="700"></p>
<p>Very short gaps between successive ranges in ROI5</p>
<div class="sourceCode" id="cb80"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/cleanGaps.html" class="external-link">cleanGaps</a></span><span class="op">(</span><span class="va">ROI5</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">##   seqnames   start     end   width strand</span>
<span class="co">## 1     chr1 1466355 1466358       4      *</span>
<span class="co">## 2     chr1 1470296 1470310      15      *</span>
<span class="co">## 3     chr1 1471260 1471275      16      *</span>
<span class="co">## 4     chr1 1481441 2405231  923791      *</span>
<span class="co">## 5     chr1 2405394 9897739 7492346      *</span></code></pre>
<div class="sourceCode" id="cb82"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ROI5</span> |&gt; <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/flagAll.html" class="external-link">flagAll</a></span><span class="op">(</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/dist2next.html" class="external-link">dist2next</a></span><span class="op">(</span> ignore.strand <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">##   seqnames   start     end width strand query.seqnames query.start query.end</span>
<span class="co">## 1     chr1 1436544 1466354 29811      +    contig_27_1     1409471   1440844</span>
<span class="co">## 2     chr1 1466359 1470295  3937      +    contig_27_1     1441222   1445179</span>
<span class="co">## 3     chr1 1470311 1471259   949      +    contig_27_1     1445585   1446558</span>
<span class="co">## 4     chr1 1471276 1478839  7564      +    contig_27_1     1447542   1456167</span>
<span class="co">## 5     chr1 1478840 1479523   684      -    contig_11_1      467930    468616</span>
<span class="co">## 6     chr1 1479524 1481440  1917      +    contig_27_1     1456168   1458038</span>
<span class="co">## 7     chr1 2405232 2405393   162      -    contig_27_1     1445390   1445562</span>
<span class="co">## 8     chr1 9897740 9898012   273      +    contig_27_1     1447046   1447318</span>
<span class="co">## 9     chr2 3010219 3010391   173      -    contig_27_1     1441042   1441221</span>
<span class="co">##   query.width query.strand    query.rep query.repOvlp</span>
<span class="co">## 1       31374            * tandem, ....          3619</span>
<span class="co">## 2        3958            * rnd, unknown           676</span>
<span class="co">## 3         974            *           NA             0</span>
<span class="co">## 4        8626            * rnd, unk....          2355</span>
<span class="co">## 5         687            *      unknown           684</span>
<span class="co">## 6        1871            * rnd, unknown          1698</span>
<span class="co">## 7         173            *  tandem, rnd            68</span>
<span class="co">## 8         273            * rnd, unknown             2</span>
<span class="co">## 9         180            *          rnd           180</span>
<span class="co">##                                                                                             query.transcripts</span>
<span class="co">## 1 g3891.t1;g3892.t1;g3893.t1;g3894.t1;g3895.t1;g3898.t1;g3899.t1;g3896.t1;g3897.t1;g3900.t2;g3900.t1;g3901.t1</span>
<span class="co">## 2                                                                                  g3902.t1;g3902.t2;g3901.t1</span>
<span class="co">## 3                                                                                           g3902.t1;g3902.t2</span>
<span class="co">## 4                                                                                           g3904.t1;g3903.t1</span>
<span class="co">## 5                                                                                                        &lt;NA&gt;</span>
<span class="co">## 6                                                                                                        &lt;NA&gt;</span>
<span class="co">## 7                                                                                           g3902.t1;g3902.t2</span>
<span class="co">## 8                                                                                                        &lt;NA&gt;</span>
<span class="co">## 9                                                                                                    g3901.t1</span>
<span class="co">##   score   Arm          rep repOvlp</span>
<span class="co">## 1 29811 short tandem, ....    1178</span>
<span class="co">## 2  3937 short rnd, unknown     697</span>
<span class="co">## 3   949 short           NA       0</span>
<span class="co">## 4  7564 short rnd, unk....     562</span>
<span class="co">## 5   684 short          rnd     684</span>
<span class="co">## 6  1917 short rnd, unk....    1592</span>
<span class="co">## 7   162 short           NA       0</span>
<span class="co">## 8   273  long       tandem      24</span>
<span class="co">## 9   173 short           NA       0</span>
<span class="co">##                                                       transcripts flag nonCoa</span>
<span class="co">## 1 g302.t1;g303.t1;g304.t1;g306.t1;g307.t1;g308.t1;g313.t1;g314.t1 &lt;NA&gt;  FALSE</span>
<span class="co">## 2                                                 g319.t1;g319.t2 &lt;NA&gt;  FALSE</span>
<span class="co">## 3                                                 g319.t1;g319.t2 &lt;NA&gt;   TRUE</span>
<span class="co">## 4                                                         g321.t1  Tra  FALSE</span>
<span class="co">## 5                                                            &lt;NA&gt; &lt;NA&gt;   TRUE</span>
<span class="co">## 6                                                            &lt;NA&gt; &lt;NA&gt;   TRUE</span>
<span class="co">## 7                                                            &lt;NA&gt; &lt;NA&gt;   TRUE</span>
<span class="co">## 8                                                        g2641.t1 &lt;NA&gt;   TRUE</span>
<span class="co">## 9                                                        g4701.t1 &lt;NA&gt;   TRUE</span>
<span class="co">##     tdist qdist</span>
<span class="co">## 1       5   378</span>
<span class="co">## 2      16   406</span>
<span class="co">## 3      17   984</span>
<span class="co">## 4       1    NA</span>
<span class="co">## 5       1    NA</span>
<span class="co">## 6  923792 10606</span>
<span class="co">## 7 7492347  1484</span>
<span class="co">## 8      NA  5825</span>
<span class="co">## 9     Inf   Inf</span></code></pre>
<div class="sourceCode" id="cb84"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ROI5</span> |&gt; <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/swap.html" class="external-link">swap</a></span><span class="op">(</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/sort.html" class="external-link">sort</a></span><span class="op">(</span>ignore.strand <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/flagAll.html" class="external-link">flagAll</a></span><span class="op">(</span><span class="op">)</span>|&gt; <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/dist2next.html" class="external-link">dist2next</a></span><span class="op">(</span> ignore.strand <span class="op">=</span> <span class="cn">TRUE</span>, step <span class="op">=</span><span class="fl">2</span><span class="op">)</span>  |&gt; <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">##      seqnames   start     end width strand          rep repOvlp</span>
<span class="co">## 1 contig_11_1  467930  468616   687      -      unknown     684</span>
<span class="co">## 2 contig_27_1 1409471 1440844 31374      + tandem, ....    3619</span>
<span class="co">## 3 contig_27_1 1441042 1441221   180      -          rnd     180</span>
<span class="co">## 4 contig_27_1 1441222 1445179  3958      + rnd, unknown     676</span>
<span class="co">## 5 contig_27_1 1445390 1445562   173      -  tandem, rnd      68</span>
<span class="co">## 6 contig_27_1 1445585 1446558   974      +           NA       0</span>
<span class="co">## 7 contig_27_1 1447046 1447318   273      + rnd, unknown       2</span>
<span class="co">## 8 contig_27_1 1447542 1456167  8626      + rnd, unk....    2355</span>
<span class="co">## 9 contig_27_1 1456168 1458038  1871      + rnd, unknown    1698</span>
<span class="co">##                                                                                                   transcripts</span>
<span class="co">## 1                                                                                                        &lt;NA&gt;</span>
<span class="co">## 2 g3891.t1;g3892.t1;g3893.t1;g3894.t1;g3895.t1;g3898.t1;g3899.t1;g3896.t1;g3897.t1;g3900.t2;g3900.t1;g3901.t1</span>
<span class="co">## 3                                                                                                    g3901.t1</span>
<span class="co">## 4                                                                                  g3902.t1;g3902.t2;g3901.t1</span>
<span class="co">## 5                                                                                           g3902.t1;g3902.t2</span>
<span class="co">## 6                                                                                           g3902.t1;g3902.t2</span>
<span class="co">## 7                                                                                                        &lt;NA&gt;</span>
<span class="co">## 8                                                                                           g3904.t1;g3903.t1</span>
<span class="co">## 9                                                                                                        &lt;NA&gt;</span>
<span class="co">##   query.seqnames query.start query.end query.width query.strand flag tdist</span>
<span class="co">## 1           chr1     1478840   1479523         684            * &lt;NA&gt;    NA</span>
<span class="co">## 2           chr1     1436544   1466354       29811            *  Tra   378</span>
<span class="co">## 3           chr2     3010219   3010391         173            * &lt;NA&gt;  4169</span>
<span class="co">## 4           chr1     1466359   1470295        3937            *  Tra   406</span>
<span class="co">## 5           chr1     2405232   2405393         162            * &lt;NA&gt;  1484</span>
<span class="co">## 6           chr1     1470311   1471259         949            *  Tra   984</span>
<span class="co">## 7           chr1     9897740   9898012         273            * &lt;NA&gt;  8850</span>
<span class="co">## 8           chr1     1471276   1478839        7564            * &lt;NA&gt;   Inf</span>
<span class="co">## 9           chr1     1479524   1481440        1917            * &lt;NA&gt;   Inf</span>
<span class="co">##     qdist</span>
<span class="co">## 1      NA</span>
<span class="co">## 2       5</span>
<span class="co">## 3      NA</span>
<span class="co">## 4      16</span>
<span class="co">## 5 7492347</span>
<span class="co">## 6      17</span>
<span class="co">## 7 8416300</span>
<span class="co">## 8     Inf</span>
<span class="co">## 9     Inf</span></code></pre>
<p>Can we coalesce better by removing the translocations ?</p>
<div class="sourceCode" id="cb86"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">removeTranslocations</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">gb</span><span class="op">)</span> <span class="op">{</span>
  <span class="va">gb</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/flagTranslocations.html" class="external-link">flagTranslocations</a></span><span class="op">(</span><span class="va">gb</span><span class="op">)</span>
  <span class="kw">if</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/any.html" class="external-link">any</a></span><span class="op">(</span><span class="va">gb</span><span class="op">$</span><span class="va">tra</span><span class="op">)</span><span class="op">)</span>
    <span class="va">gb</span> <span class="op">&lt;-</span> <span class="va">gb</span><span class="op">[</span><span class="op">-</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/which.html" class="external-link">which</a></span><span class="op">(</span><span class="va">gb</span><span class="op">$</span><span class="va">tra</span><span class="op">)</span> <span class="op">+</span> <span class="fl">1</span><span class="op">)</span><span class="op">]</span>
  <span class="va">gb</span>
<span class="op">}</span>

<span class="va">ROI5</span> |&gt; <span class="fu">removeTranslocations</span><span class="op">(</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/coalesce_contigs.html" class="external-link">coalesce_contigs</a></span><span class="op">(</span><span class="op">)</span> |&gt;
  <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/swap.html" class="external-link">swap</a></span><span class="op">(</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/sort.html" class="external-link">sort</a></span><span class="op">(</span>i<span class="op">=</span><span class="cn">T</span><span class="op">)</span> |&gt; <span class="fu">removeTranslocations</span><span class="op">(</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/coalesce_contigs.html" class="external-link">coalesce_contigs</a></span><span class="op">(</span><span class="op">)</span> |&gt;
  <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/swap.html" class="external-link">swap</a></span><span class="op">(</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/sort.html" class="external-link">sort</a></span><span class="op">(</span>i<span class="op">=</span><span class="cn">T</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/plotApairOfChrs.html" class="external-link">plotApairOfChrs</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<p><img src="RegionsOfInterest_files/figure-html/ROI5_removeTranslocations-1.png" width="700"></p>
<p>This is now implemented in the <code>coa2</code> objects, with a size treshold of 200.</p>
<div class="sourceCode" id="cb87"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/findOverlaps-methods.html" class="external-link">subsetByOverlaps</a></span><span class="op">(</span><span class="va">coa2</span><span class="op">$</span><span class="va">Oki_Kum</span>, <span class="va">ROI5.range</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/plotApairOfChrs.html" class="external-link">plotApairOfChrs</a></span><span class="op">(</span>xlim <span class="op">=</span> <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/gb2xlim.html" class="external-link">gb2xlim</a></span><span class="op">(</span><span class="va">ROI5.range</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="RegionsOfInterest_files/figure-html/ROI5_coa2-1.png" width="700"></p>
</div>
<div class="section level2">
<h2 id="what-still-interrupts-colinearity-once-translocations-are-removed-roi6">What still interrupts colinearity once translocations are removed ? (ROI6)<a class="anchor" aria-label="anchor" href="#what-still-interrupts-colinearity-once-translocations-are-removed-roi6"></a>
</h2>
<p>Inversions â€¦</p>
<div class="sourceCode" id="cb88"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># We start with a broader random region</span>
<span class="op">(</span><span class="va">ROI6</span> <span class="op">&lt;-</span> <span class="va">coa2</span><span class="op">$</span><span class="va">Oki_Kum</span> |&gt; <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/flagAll.html" class="external-link">flagAll</a></span><span class="op">(</span><span class="op">)</span> |&gt; <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/slice.html" class="external-link">slice</a></span><span class="op">(</span><span class="fl">160</span><span class="op">:</span><span class="fl">200</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## GBreaks object with 41 ranges and 6 metadata columns:</span>
<span class="co">##        seqnames          ranges strand |                       query      Arm</span>
<span class="co">##           &lt;Rle&gt;       &lt;IRanges&gt;  &lt;Rle&gt; |                   &lt;GRanges&gt; &lt;factor&gt;</span>
<span class="co">##    [1]     chr1 3881205-3906741      + | contig_27_1:3631331-3656195    short</span>
<span class="co">##    [2]     chr1 3906770-3929141      - | contig_27_1:3656207-3666222    short</span>
<span class="co">##    [3]     chr1 3929230-3969484      + | contig_27_1:3666267-3704635    short</span>
<span class="co">##    [4]     chr1 3969656-3971676      - | contig_27_1:3704755-3707751    short</span>
<span class="co">##    [5]     chr1 3971851-3972180      + | contig_27_1:3707792-3708116    short</span>
<span class="co">##    ...      ...             ...    ... .                         ...      ...</span>
<span class="co">##   [37]     chr1 4675996-4676385      + |   contig_10_1:126416-126806    short</span>
<span class="co">##   [38]     chr1 4680756-4680883      + |   contig_13_1:288535-288663    short</span>
<span class="co">##   [39]     chr1 4682663-4683297      + | contig_27_1:4367224-4367927    short</span>
<span class="co">##   [40]     chr1 4685570-4769819      + | contig_27_1:4368942-4459904    short</span>
<span class="co">##   [41]     chr1 4770023-4770969      + | contig_27_1:4463373-4464345    short</span>
<span class="co">##                           rep   repOvlp            transcripts        flag</span>
<span class="co">##               &lt;CharacterList&gt; &lt;integer&gt;                  &lt;Rle&gt; &lt;character&gt;</span>
<span class="co">##    [1]     rnd,unknown,tandem      1421 g1018.t1;g1024.t1;g1..         Inv</span>
<span class="co">##    [2]   tandem,ltr-1,rnd,...      2876      g1030.t1;g1032.t1        &lt;NA&gt;</span>
<span class="co">##    [3]   rnd,ltr-1,tandem,...      1477 g1036.t1;g1037.t2;g1..         Inv</span>
<span class="co">##    [4]                   &lt;NA&gt;         0                   &lt;NA&gt;        &lt;NA&gt;</span>
<span class="co">##    [5]                   &lt;NA&gt;         0                   &lt;NA&gt;         Inv</span>
<span class="co">##    ...                    ...       ...                    ...         ...</span>
<span class="co">##   [37]                    rnd       390                   &lt;NA&gt;        &lt;NA&gt;</span>
<span class="co">##   [38]                   &lt;NA&gt;         0                   &lt;NA&gt;        &lt;NA&gt;</span>
<span class="co">##   [39]                   &lt;NA&gt;         0                   &lt;NA&gt;        &lt;NA&gt;</span>
<span class="co">##   [40] rnd,tandem,unknown,...      9359 g1242.t1;g1243.t1;g1..        &lt;NA&gt;</span>
<span class="co">##   [41]                unknown        94                   &lt;NA&gt;         Tra</span>
<span class="co">##   -------</span>
<span class="co">##   seqinfo: 19 sequences from OKI2018.I69 genome</span></code></pre>
<div class="sourceCode" id="cb90"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ROI6</span> |&gt; <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/plotApairOfChrs.html" class="external-link">plotApairOfChrs</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<p><img src="RegionsOfInterest_files/figure-html/ROI6-1.png" width="700"></p>
<div class="sourceCode" id="cb91"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">coa2</span><span class="op">$</span><span class="va">Oki_Kum</span> |&gt; <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/plotApairOfChrs.html" class="external-link">plotApairOfChrs</a></span><span class="op">(</span>chrT <span class="op">=</span> <span class="st">"chr1"</span>, chrQ <span class="op">=</span> <span class="st">"contig_27_1"</span>, xlim <span class="op">=</span> <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/gb2xlim.html" class="external-link">gb2xlim</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/inter-range-methods.html" class="external-link">range</a></span><span class="op">(</span><span class="va">ROI6</span><span class="op">)</span><span class="op">[</span><span class="fl">7</span><span class="op">]</span><span class="op">)</span>, main <span class="op">=</span> <span class="st">"Double-collapsed"</span><span class="op">)</span></code></pre></div>
<p><img src="RegionsOfInterest_files/figure-html/ROI6-2.png" width="700"></p>
<div class="sourceCode" id="cb92"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">coa</span> <span class="op">$</span><span class="va">Oki_Kum</span> |&gt; <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/plotApairOfChrs.html" class="external-link">plotApairOfChrs</a></span><span class="op">(</span>chrT <span class="op">=</span> <span class="st">"chr1"</span>, chrQ <span class="op">=</span> <span class="st">"contig_27_1"</span>, xlim <span class="op">=</span> <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/gb2xlim.html" class="external-link">gb2xlim</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/inter-range-methods.html" class="external-link">range</a></span><span class="op">(</span><span class="va">ROI6</span><span class="op">)</span><span class="op">[</span><span class="fl">7</span><span class="op">]</span><span class="op">)</span>, main <span class="op">=</span> <span class="st">"Collapsed"</span><span class="op">)</span></code></pre></div>
<p><img src="RegionsOfInterest_files/figure-html/ROI6-3.png" width="700"></p>
<div class="sourceCode" id="cb93"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">gbs</span> <span class="op">$</span><span class="va">Oki_Kum</span> |&gt; <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/plotApairOfChrs.html" class="external-link">plotApairOfChrs</a></span><span class="op">(</span>chrT <span class="op">=</span> <span class="st">"chr1"</span>, chrQ <span class="op">=</span> <span class="st">"contig_27_1"</span>, xlim <span class="op">=</span> <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/gb2xlim.html" class="external-link">gb2xlim</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/inter-range-methods.html" class="external-link">range</a></span><span class="op">(</span><span class="va">ROI6</span><span class="op">)</span><span class="op">[</span><span class="fl">7</span><span class="op">]</span><span class="op">)</span>, main <span class="op">=</span> <span class="st">"Mapped"</span><span class="op">)</span></code></pre></div>
<p><img src="RegionsOfInterest_files/figure-html/ROI6-4.png" width="700"></p>
<div class="sourceCode" id="cb94"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Sometimes, everything that remains is inversions</span>
<span class="va">ROI6</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">6</span><span class="op">]</span> |&gt; <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/plotApairOfChrs.html" class="external-link">plotApairOfChrs</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<p><img src="RegionsOfInterest_files/figure-html/ROI6-5.png" width="700"></p>
<div class="sourceCode" id="cb95"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ROI6</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">6</span><span class="op">]</span> |&gt; <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>strand <span class="op">=</span> <span class="st">'*'</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/coalesce_contigs.html" class="external-link">coalesce_contigs</a></span><span class="op">(</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/plotApairOfChrs.html" class="external-link">plotApairOfChrs</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<p><img src="RegionsOfInterest_files/figure-html/ROI6-6.png" width="700"></p>
</div>
<div class="section level2">
<h2 id="multiple-inversions-in-a-row-roi7">Multiple inversions in a row (ROI7)<a class="anchor" aria-label="anchor" href="#multiple-inversions-in-a-row-roi7"></a>
</h2>
<div class="sourceCode" id="cb96"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ROI7</span> <span class="op">&lt;-</span> <span class="va">coa</span><span class="op">$</span><span class="va">Oki_Osa</span> |&gt; <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/flagInversions.html" class="external-link">flagInversions</a></span><span class="op">(</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/dist2next.html" class="external-link">dist2next</a></span><span class="op">(</span>ignore.strand <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>

<span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/showInversions.html" class="external-link">showInversions</a></span><span class="op">(</span><span class="va">ROI7</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## GBreaks object with 1826 ranges and 11 metadata columns:</span>
<span class="co">##        seqnames            ranges strand |                query     score</span>
<span class="co">##           &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; |            &lt;GRanges&gt; &lt;integer&gt;</span>
<span class="co">##     16     chr1     592887-593096      - | Chr1:1309140-1309328       210</span>
<span class="co">##     17     chr1     593196-593292      + | Chr1:1308820-1308912        97</span>
<span class="co">##     18     chr1     594240-603127      - | Chr1:1296981-1308607      8888</span>
<span class="co">##     37     chr1     741910-744560      + | Chr1:2036766-2039472      2651</span>
<span class="co">##     38     chr1     744937-745142      - | Chr1:2039897-2040113       206</span>
<span class="co">##    ...      ...               ...    ... .                  ...       ...</span>
<span class="co">##   8642      XSR 11853592-11853924      + |  XSR:8431875-8432251       333</span>
<span class="co">##   8643      XSR 11854037-11855928      - |  XSR:8429198-8431487      1892</span>
<span class="co">##   8695      XSR 12324700-12329600      - |  XSR:8992784-8997031      4901</span>
<span class="co">##   8696      XSR 12331614-12331778      + |  XSR:8990804-8990961       165</span>
<span class="co">##   8697      XSR 12335389-12337572      - |  XSR:8987052-8988968      2184</span>
<span class="co">##             Arm                    rep   repOvlp transcripts        flag</span>
<span class="co">##        &lt;factor&gt;        &lt;CharacterList&gt; &lt;integer&gt;       &lt;Rle&gt; &lt;character&gt;</span>
<span class="co">##     16    short                   &lt;NA&gt;         0        &lt;NA&gt;         Inv</span>
<span class="co">##     17    short                   &lt;NA&gt;         0        &lt;NA&gt;        &lt;NA&gt;</span>
<span class="co">##     18    short tandem,unknown,rnd,...        31      g68.t1        &lt;NA&gt;</span>
<span class="co">##     37    short                 tandem        68        &lt;NA&gt;         Inv</span>
<span class="co">##     38    short                   &lt;NA&gt;         0        &lt;NA&gt;        &lt;NA&gt;</span>
<span class="co">##    ...      ...                    ...       ...         ...         ...</span>
<span class="co">##   8642      XSR                   &lt;NA&gt;         0   g16813.t1        &lt;NA&gt;</span>
<span class="co">##   8643      XSR                 tandem         0        &lt;NA&gt;        &lt;NA&gt;</span>
<span class="co">##   8695      XSR     tandem,unknown,rnd        26   g16922.t1         Inv</span>
<span class="co">##   8696      XSR                   &lt;NA&gt;         0        &lt;NA&gt;        &lt;NA&gt;</span>
<span class="co">##   8697      XSR                 tandem         0   g16922.t1        &lt;NA&gt;</span>
<span class="co">##           nonCoa   inv     tdist     qdist</span>
<span class="co">##        &lt;logical&gt; &lt;Rle&gt; &lt;numeric&gt; &lt;numeric&gt;</span>
<span class="co">##     16      TRUE  TRUE       100       228</span>
<span class="co">##     17      TRUE FALSE       948       213</span>
<span class="co">##     18     FALSE FALSE     29392        NA</span>
<span class="co">##     37      TRUE  TRUE       377       425</span>
<span class="co">##     38      TRUE FALSE       440       387</span>
<span class="co">##    ...       ...   ...       ...       ...</span>
<span class="co">##   8642      TRUE FALSE       113       388</span>
<span class="co">##   8643     FALSE FALSE        93   1180869</span>
<span class="co">##   8695     FALSE  TRUE      2014      1823</span>
<span class="co">##   8696      TRUE FALSE      3611      1836</span>
<span class="co">##   8697     FALSE FALSE      1165       356</span>
<span class="co">##   -------</span>
<span class="co">##   seqinfo: 19 sequences from OKI2018.I69 genome</span></code></pre>
<div class="sourceCode" id="cb98"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ROI7</span><span class="op">[</span><span class="fl">35</span><span class="op">:</span><span class="fl">45</span><span class="op">]</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">##    seqnames  start    end width strand query.seqnames query.start query.end</span>
<span class="co">## 1      chr1 736824 738412  1589      +           Chr1     2033491   2036651</span>
<span class="co">## 2      chr1 738882 741671  2790      +           Chr1     2155668   2159253</span>
<span class="co">## 3      chr1 741910 744560  2651      +           Chr1     2036766   2039472</span>
<span class="co">## 4      chr1 744937 745142   206      -           Chr1     2039897   2040113</span>
<span class="co">## 5      chr1 745582 745862   281      +           Chr1     2040500   2040775</span>
<span class="co">## 6      chr1 747713 747961   249      -           Chr1     2041371   2041628</span>
<span class="co">## 7      chr1 750585 751341   757      +           Chr1     2042328   2043943</span>
<span class="co">## 8      chr1 752126 752262   137      -           Chr1     2044495   2044628</span>
<span class="co">## 9      chr1 753366 768703 15338      +           Chr1     2045148   2050583</span>
<span class="co">## 10     chr1 768953 773691  4739      -           Chr1     4442048   4448098</span>
<span class="co">## 11     chr1 776248 779526  3279      -           Chr1     3333268   3337597</span>
<span class="co">##    query.width query.strand query.Arm    query.rep query.repOvlp</span>
<span class="co">## 1         3161            *     short rnd, tan....          1634</span>
<span class="co">## 2         3586            *     short rnd, unknown           557</span>
<span class="co">## 3         2707            *     short       tandem           160</span>
<span class="co">## 4          217            *     short           NA             0</span>
<span class="co">## 5          276            *     short           NA             0</span>
<span class="co">## 6          258            *     short           NA             0</span>
<span class="co">## 7         1616            *     short LowCompl....           152</span>
<span class="co">## 8          134            *     short           NA             0</span>
<span class="co">## 9         5436            *     short rnd, Low....           768</span>
<span class="co">## 10        6051            *     short          rnd          1052</span>
<span class="co">## 11        4330            *     short  rnd, tandem          1147</span>
<span class="co">##             query.transcripts score   Arm          rep repOvlp     transcripts</span>
<span class="co">## 1                     g456.t1  1589 short          rnd      98            &lt;NA&gt;</span>
<span class="co">## 2                     g492.t1  2790 short           NA       0         g106.t1</span>
<span class="co">## 3                     g457.t1  2651 short       tandem      68            &lt;NA&gt;</span>
<span class="co">## 4                        &lt;NA&gt;   206 short           NA       0            &lt;NA&gt;</span>
<span class="co">## 5                        &lt;NA&gt;   281 short           NA       0            &lt;NA&gt;</span>
<span class="co">## 6                        &lt;NA&gt;   249 short           NA       0            &lt;NA&gt;</span>
<span class="co">## 7                        &lt;NA&gt;   757 short           NA       0            &lt;NA&gt;</span>
<span class="co">## 8                        &lt;NA&gt;   137 short           NA       0            &lt;NA&gt;</span>
<span class="co">## 9             g458.t1;g459.t1 15338 short tandem, ....    3169            &lt;NA&gt;</span>
<span class="co">## 10 g1051.t1;g1052.t1;g1053.t1  4739 short      unknown       0 g111.t1;g112.t1</span>
<span class="co">## 11                    g750.t1  3279 short tandem, ....      77         g115.t1</span>
<span class="co">##    flag nonCoa   inv tdist   qdist</span>
<span class="co">## 1   Tra  FALSE FALSE   470  119017</span>
<span class="co">## 2  &lt;NA&gt;  FALSE FALSE   239  116196</span>
<span class="co">## 3   Inv   TRUE  TRUE   377     425</span>
<span class="co">## 4  &lt;NA&gt;   TRUE FALSE   440     387</span>
<span class="co">## 5   Inv   TRUE  TRUE  1851     596</span>
<span class="co">## 6  &lt;NA&gt;   TRUE FALSE  2624     700</span>
<span class="co">## 7   Inv  FALSE  TRUE   785     552</span>
<span class="co">## 8  &lt;NA&gt;   TRUE FALSE  1104     520</span>
<span class="co">## 9  &lt;NA&gt;  FALSE FALSE   250 2391465</span>
<span class="co">## 10 &lt;NA&gt;  FALSE FALSE  2557 1104451</span>
<span class="co">## 11 &lt;NA&gt;  FALSE FALSE   240 2506121</span></code></pre>
<div class="sourceCode" id="cb100"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ROI7</span><span class="op">[</span><span class="fl">37</span><span class="op">:</span><span class="fl">43</span><span class="op">]</span> |&gt; <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/plotApairOfChrs.html" class="external-link">plotApairOfChrs</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<p><img src="RegionsOfInterest_files/figure-html/unnamed-chunk-1-1.png" width="700"></p>
<div class="sourceCode" id="cb101"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## Annotations</span>
<span class="va">ROI7.reps</span> <span class="op">&lt;-</span> <span class="va">reps</span><span class="op">$</span><span class="va">OKI2018.I69</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/findOverlaps-methods.html" class="external-link">subsetByOverlaps</a></span><span class="op">(</span><span class="va">ROI7</span><span class="op">[</span><span class="fl">37</span><span class="op">:</span><span class="fl">43</span><span class="op">]</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/inter-range-methods.html" class="external-link">range</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span>
<span class="va">mids</span> <span class="op">&lt;-</span> <span class="va">ROI7.reps</span> |&gt; <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/gr2dna_seg.html" class="external-link">gr2dna_seg</a></span><span class="op">(</span><span class="op">)</span> |&gt; <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>mid <span class="op">=</span> <span class="op">(</span><span class="va">start</span> <span class="op">+</span> <span class="va">end</span><span class="op">)</span> <span class="op">/</span> <span class="fl">2</span><span class="op">)</span> |&gt; <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/pull.html" class="external-link">pull</a></span><span class="op">(</span><span class="va">mid</span><span class="op">)</span>
<span class="va">text</span> <span class="op">&lt;-</span> <span class="va">ROI7.reps</span><span class="op">$</span><span class="va">Class</span> |&gt; <span class="fu"><a href="https://rdrr.io/r/base/character.html" class="external-link">as.character</a></span><span class="op">(</span><span class="op">)</span>
<span class="va">annot</span> <span class="op">&lt;-</span> <span class="fu">genoPlotR</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/genoPlotR/man/annotation.html" class="external-link">annotation</a></span><span class="op">(</span>x1<span class="op">=</span><span class="va">mids</span>, text<span class="op">=</span><span class="va">text</span>, rot<span class="op">=</span><span class="fl">30</span><span class="op">)</span>
<span class="va">ROI7</span><span class="op">[</span><span class="fl">37</span><span class="op">:</span><span class="fl">43</span><span class="op">]</span> |&gt; <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/plotApairOfChrs.html" class="external-link">plotApairOfChrs</a></span><span class="op">(</span>annotations <span class="op">=</span> <span class="va">annot</span><span class="op">)</span></code></pre></div>
<p><img src="RegionsOfInterest_files/figure-html/unnamed-chunk-1-2.png" width="700"></p>
<div class="sourceCode" id="cb102"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## Genes</span>
<span class="va">ROI7.genes</span> <span class="op">&lt;-</span> <span class="va">transcripts</span><span class="op">$</span><span class="va">OKI2018.I69</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/findOverlaps-methods.html" class="external-link">subsetByOverlaps</a></span><span class="op">(</span><span class="va">ROI7</span><span class="op">[</span><span class="fl">37</span><span class="op">:</span><span class="fl">43</span><span class="op">]</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/inter-range-methods.html" class="external-link">range</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span>
<span class="va">mids</span> <span class="op">&lt;-</span> <span class="va">ROI7.genes</span> |&gt; <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/gr2dna_seg.html" class="external-link">gr2dna_seg</a></span><span class="op">(</span><span class="op">)</span> |&gt; <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>mid <span class="op">=</span> <span class="op">(</span><span class="va">start</span> <span class="op">+</span> <span class="va">end</span><span class="op">)</span> <span class="op">/</span> <span class="fl">2</span><span class="op">)</span> |&gt; <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/pull.html" class="external-link">pull</a></span><span class="op">(</span><span class="va">mid</span><span class="op">)</span>
<span class="va">text</span> <span class="op">&lt;-</span> <span class="va">ROI7.genes</span><span class="op">$</span><span class="va">tx_name</span> |&gt; <span class="fu"><a href="https://rdrr.io/r/base/character.html" class="external-link">as.character</a></span><span class="op">(</span><span class="op">)</span>
<span class="va">annot</span> <span class="op">&lt;-</span> <span class="fu">genoPlotR</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/genoPlotR/man/annotation.html" class="external-link">annotation</a></span><span class="op">(</span>x1<span class="op">=</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/start.html" class="external-link">start</a></span><span class="op">(</span><span class="va">ROI7.genes</span><span class="op">)</span>, x2<span class="op">=</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/start.html" class="external-link">end</a></span><span class="op">(</span><span class="va">ROI7.genes</span><span class="op">)</span>, text<span class="op">=</span><span class="va">text</span>, rot<span class="op">=</span><span class="fl">30</span><span class="op">)</span>
<span class="va">ROI7</span><span class="op">[</span><span class="fl">37</span><span class="op">:</span><span class="fl">43</span><span class="op">]</span> |&gt; <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/plotApairOfChrs.html" class="external-link">plotApairOfChrs</a></span><span class="op">(</span>annotations <span class="op">=</span> <span class="va">annot</span><span class="op">)</span></code></pre></div>
<p><img src="RegionsOfInterest_files/figure-html/unnamed-chunk-1-3.png" width="700"></p>
</div>
<div class="section level2">
<h2 id="double-inversions">Double inversions<a class="anchor" aria-label="anchor" href="#double-inversions"></a>
</h2>
<p>In these examples, the inversions happened on the <em>target</em> genome, therefore only in this genome the names of the ranges were changed.</p>
<div class="sourceCode" id="cb103"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">exampleDoubleInversion</span> |&gt; <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/plotApairOfChrs.html" class="external-link">plotApairOfChrs</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<p><img src="RegionsOfInterest_files/figure-html/ROI8.T-1.png" width="700"></p>
<div class="sourceCode" id="cb104"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">x0</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/GBreaks-class.html" class="external-link">GBreaks</a></span><span class="op">(</span>target <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"T:1:+"</span>, <span class="st">"T:2:+"</span>, <span class="st">"T:3:+"</span><span class="op">)</span><span class="op">)</span>,
              query  <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Q:1"</span>,   <span class="st">"Q:2"</span>,   <span class="st">"Q:3"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>

<span class="va">x1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/GBreaks-class.html" class="external-link">GBreaks</a></span><span class="op">(</span>target <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"T:1:-"</span>, <span class="st">"T:2:-"</span>, <span class="st">"T:3:+"</span><span class="op">)</span><span class="op">)</span>,
              query  <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Q:2"</span>,   <span class="st">"Q:1"</span>,   <span class="st">"Q:3"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>

<span class="va">x2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/GBreaks-class.html" class="external-link">GBreaks</a></span><span class="op">(</span>target <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"T:1:-"</span>, <span class="st">"T:2:+"</span>, <span class="st">"T:3:-"</span><span class="op">)</span><span class="op">)</span>,
              query  <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Q:2"</span>,   <span class="st">"Q:3"</span>,   <span class="st">"Q:1"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>

<span class="va">y0</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/GBreaks-class.html" class="external-link">GBreaks</a></span><span class="op">(</span>target <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span>A<span class="op">=</span><span class="st">"T:10-15:+"</span>, B<span class="op">=</span><span class="st">"T:20-25:+"</span>, C<span class="op">=</span><span class="st">"T:30-35:+"</span><span class="op">)</span><span class="op">)</span>,
              query  <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span>A<span class="op">=</span><span class="st">"Q:10-15"</span>,   B<span class="op">=</span><span class="st">"Q:20-25"</span>,   C<span class="op">=</span><span class="st">"Q:30-35"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>

<span class="va">y1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/GBreaks-class.html" class="external-link">GBreaks</a></span><span class="op">(</span>target <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span>A<span class="op">=</span><span class="st">"T:10-15:-"</span>, B<span class="op">=</span><span class="st">"T:20-25:-"</span>, C<span class="op">=</span><span class="st">"T:30-35:+"</span><span class="op">)</span><span class="op">)</span>,
              query  <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span>B<span class="op">=</span><span class="st">"Q:20-25"</span>,   A<span class="op">=</span><span class="st">"Q:10-15"</span>,   C<span class="op">=</span><span class="st">"Q:30-35"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>

<span class="va">y2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/GBreaks-class.html" class="external-link">GBreaks</a></span><span class="op">(</span>target <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span>A<span class="op">=</span><span class="st">"T:10-15:-"</span>, B<span class="op">=</span><span class="st">"T:20-25:+"</span>, C<span class="op">=</span><span class="st">"T:30-35:-"</span><span class="op">)</span><span class="op">)</span>,
              query  <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span>B<span class="op">=</span><span class="st">"Q:20-25"</span>,   C<span class="op">=</span><span class="st">"Q:30-35"</span>,   A<span class="op">=</span><span class="st">"Q:10-15"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>


<span class="va">a0</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/GBreaks-class.html" class="external-link">GBreaks</a></span><span class="op">(</span>target <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span>`ABC+`<span class="op">=</span><span class="st">"T:10-39:+"</span><span class="op">)</span><span class="op">)</span>,
              query  <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span>`ABC+`<span class="op">=</span><span class="st">"Q:10-39"</span>  <span class="op">)</span><span class="op">)</span><span class="op">)</span>

<span class="va">a1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/GBreaks-class.html" class="external-link">GBreaks</a></span><span class="op">(</span>target <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span>`AB+`<span class="op">=</span><span class="st">"T:10-29:-"</span>, `C+`<span class="op">=</span><span class="st">"T:30-39:+"</span><span class="op">)</span><span class="op">)</span>,
              query  <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span>`BA-`<span class="op">=</span><span class="st">"Q:10-29"</span>,   `C+`<span class="op">=</span><span class="st">"Q:30-39"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>

<span class="va">a2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/GBreaks-class.html" class="external-link">GBreaks</a></span><span class="op">(</span>target <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span>`A+`<span class="op">=</span><span class="st">"T:10-19:+"</span>, `B+`<span class="op">=</span><span class="st">"T:20-29:-"</span>, `C+`<span class="op">=</span><span class="st">"T:30-39:-"</span><span class="op">)</span><span class="op">)</span>,
              query  <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span>`A+`<span class="op">=</span><span class="st">"Q:30-39"</span>,   `B-`<span class="op">=</span><span class="st">"Q:10-19"</span>,   `C-`<span class="op">=</span><span class="st">"Q:20-29"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p>Letâ€™s start with a simpler example, where the inversions happen on the <em>query</em> genome.</p>
<div class="sourceCode" id="cb105"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">exampleDoubleInversion</span> |&gt; <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/plotApairOfChrs.html" class="external-link">plotApairOfChrs</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<p><img src="RegionsOfInterest_files/figure-html/ROI8.Q-1.png" width="700"></p>
<div class="sourceCode" id="cb106"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Start colinear</span>
<span class="va">z0</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/GBreaks-class.html" class="external-link">GBreaks</a></span><span class="op">(</span>target <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span>A<span class="op">=</span><span class="st">"T:10-15:+"</span>, B<span class="op">=</span><span class="st">"T:20-25:+"</span>, C<span class="op">=</span><span class="st">"T:30-35:+"</span><span class="op">)</span><span class="op">)</span>,
              query  <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span>A<span class="op">=</span><span class="st">"Q:10-15"</span>,   B<span class="op">=</span><span class="st">"Q:20-25"</span>,   C<span class="op">=</span><span class="st">"Q:30-35"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>

<span class="co"># Swap coordinates of A and B on the query and flip strands</span>
<span class="va">z1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/GBreaks-class.html" class="external-link">GBreaks</a></span><span class="op">(</span>target <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span>A<span class="op">=</span><span class="st">"T:10-15:-"</span>, B<span class="op">=</span><span class="st">"T:20-25:-"</span>, C<span class="op">=</span><span class="st">"T:30-35:+"</span><span class="op">)</span><span class="op">)</span>,
              query  <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span>A<span class="op">=</span><span class="st">"Q:20-25"</span>,   B<span class="op">=</span><span class="st">"Q:10-15"</span>,   C<span class="op">=</span><span class="st">"Q:30-35"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>

<span class="co"># Now query order is B - A - C.  Swap A and C and flip strands</span>
<span class="va">z2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/GBreaks-class.html" class="external-link">GBreaks</a></span><span class="op">(</span>target <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span>A<span class="op">=</span><span class="st">"T:10-15:+"</span>, B<span class="op">=</span><span class="st">"T:20-25:-"</span>, C<span class="op">=</span><span class="st">"T:30-35:-"</span><span class="op">)</span><span class="op">)</span>,
              query  <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span>A<span class="op">=</span><span class="st">"Q:30-35"</span>,   B<span class="op">=</span><span class="st">"Q:10-15"</span>,   C<span class="op">=</span><span class="st">"Q:20-25"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>

<span class="co"># We have the same configuration, plus flanking regions, in exampleDoubleInversion</span>
<span class="va">exampleDoubleInversion</span></code></pre></div>
<pre><code><span class="co">## GBreaks object with 5 ranges and 1 metadata column:</span>
<span class="co">##       seqnames    ranges strand |        query</span>
<span class="co">##          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; |    &lt;GRanges&gt;</span>
<span class="co">##   [1]     chrA   100-190      + | chrB:100-190</span>
<span class="co">##   [2]     chrA   200-290      + | chrB:400-490</span>
<span class="co">##   [3]     chrA   300-390      - | chrB:200-290</span>
<span class="co">##   [4]     chrA   400-490      - | chrB:300-390</span>
<span class="co">##   [5]     chrA   500-590      + | chrB:500-590</span>
<span class="co">##   -------</span>
<span class="co">##   seqinfo: 1 sequence from an unspecified genome</span></code></pre>
<div class="sourceCode" id="cb108"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ROI8</span> <span class="op">&lt;-</span> <span class="va">coa2</span><span class="op">$</span><span class="va">Oki_Osa</span><span class="op">[</span><span class="fl">1420</span><span class="op">:</span><span class="fl">1430</span><span class="op">]</span>
<span class="va">ROI8</span> |&gt; <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/flagAll.html" class="external-link">flagAll</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## GBreaks object with 11 ranges and 6 metadata columns:</span>
<span class="co">##        seqnames          ranges strand |                query      Arm</span>
<span class="co">##           &lt;Rle&gt;       &lt;IRanges&gt;  &lt;Rle&gt; |            &lt;GRanges&gt; &lt;factor&gt;</span>
<span class="co">##    [1]     chr1 7925162-7925707      - | Chr1:5457033-5457486     long</span>
<span class="co">##    [2]     chr1 7926105-7926315      + | Chr1:5456632-5456843     long</span>
<span class="co">##    [3]     chr1 7926445-7935682      - | Chr1:5450903-5456340     long</span>
<span class="co">##    [4]     chr1 7935883-7936233      + | Chr1:5450455-5450759     long</span>
<span class="co">##    [5]     chr1 7936510-7937713      - | Chr1:5449692-5450249     long</span>
<span class="co">##    [6]     chr1 7937859-7938193      + | Chr1:5448891-5449199     long</span>
<span class="co">##    [7]     chr1 7938294-7938871      + | Chr1:5446199-5446704     long</span>
<span class="co">##    [8]     chr1 7938916-7939612      - | Chr1:5446876-5447698     long</span>
<span class="co">##    [9]     chr1 7940004-7940220      + | Chr1:5445781-5445992     long</span>
<span class="co">##   [10]     chr1 7940630-7940807      - | Chr1:5447947-5448116     long</span>
<span class="co">##   [11]     chr1 7941410-7942546      + | Chr1:5444239-5445410     long</span>
<span class="co">##                       rep   repOvlp       transcripts        flag</span>
<span class="co">##           &lt;CharacterList&gt; &lt;integer&gt;             &lt;Rle&gt; &lt;character&gt;</span>
<span class="co">##    [1]               &lt;NA&gt;         0          g2110.t1         Inv</span>
<span class="co">##    [2]               &lt;NA&gt;         0              &lt;NA&gt;        &lt;NA&gt;</span>
<span class="co">##    [3] unknown,rnd,tandem      1107 g2111.t1;g2111.t2         Inv</span>
<span class="co">##    [4]               &lt;NA&gt;         0              &lt;NA&gt;        &lt;NA&gt;</span>
<span class="co">##    [5]               &lt;NA&gt;         0 g2111.t1;g2111.t2        &lt;NA&gt;</span>
<span class="co">##    [6]               &lt;NA&gt;         0              &lt;NA&gt;        &lt;NA&gt;</span>
<span class="co">##    [7]               &lt;NA&gt;         0              &lt;NA&gt;        &lt;NA&gt;</span>
<span class="co">##    [8]               &lt;NA&gt;         0 g2111.t1;g2111.t2        &lt;NA&gt;</span>
<span class="co">##    [9]               &lt;NA&gt;         0              &lt;NA&gt;        &lt;NA&gt;</span>
<span class="co">##   [10]               &lt;NA&gt;         0              &lt;NA&gt;        &lt;NA&gt;</span>
<span class="co">##   [11]               &lt;NA&gt;         0              &lt;NA&gt;        &lt;NA&gt;</span>
<span class="co">##   -------</span>
<span class="co">##   seqinfo: 19 sequences from OKI2018.I69 genome</span></code></pre>
<div class="sourceCode" id="cb110"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ROI8</span> |&gt; <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/plotApairOfChrs.html" class="external-link">plotApairOfChrs</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<p><img src="RegionsOfInterest_files/figure-html/ROI8.Q-2.png" width="700"></p>
<div class="sourceCode" id="cb111"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ROI8</span> |&gt; <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/reverse.html" class="external-link">reverse</a></span><span class="op">(</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/sort.html" class="external-link">sort</a></span><span class="op">(</span>i<span class="op">=</span><span class="cn">T</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/plotApairOfChrs.html" class="external-link">plotApairOfChrs</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<p><img src="RegionsOfInterest_files/figure-html/ROI8.Q-3.png" width="700"></p>
<div class="sourceCode" id="cb112"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ROI8</span> |&gt; <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/reverse.html" class="external-link">reverse</a></span><span class="op">(</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/sort.html" class="external-link">sort</a></span><span class="op">(</span>i<span class="op">=</span><span class="cn">T</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/flagAll.html" class="external-link">flagAll</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## GBreaks object with 11 ranges and 6 metadata columns:</span>
<span class="co">##        seqnames          ranges strand |                query      Arm</span>
<span class="co">##           &lt;Rle&gt;       &lt;IRanges&gt;  &lt;Rle&gt; |            &lt;GRanges&gt; &lt;factor&gt;</span>
<span class="co">##    [1]     chr1 6590477-6591613      - | Chr1:5444239-5445410     long</span>
<span class="co">##    [2]     chr1 6592216-6592393      + | Chr1:5447947-5448116     long</span>
<span class="co">##    [3]     chr1 6592803-6593019      - | Chr1:5445781-5445992     long</span>
<span class="co">##    [4]     chr1 6593411-6594107      + | Chr1:5446876-5447698     long</span>
<span class="co">##    [5]     chr1 6594152-6594729      - | Chr1:5446199-5446704     long</span>
<span class="co">##    [6]     chr1 6594830-6595164      - | Chr1:5448891-5449199     long</span>
<span class="co">##    [7]     chr1 6595310-6596513      + | Chr1:5449692-5450249     long</span>
<span class="co">##    [8]     chr1 6596790-6597140      - | Chr1:5450455-5450759     long</span>
<span class="co">##    [9]     chr1 6597341-6606578      + | Chr1:5450903-5456340     long</span>
<span class="co">##   [10]     chr1 6606708-6606918      - | Chr1:5456632-5456843     long</span>
<span class="co">##   [11]     chr1 6607316-6607861      + | Chr1:5457033-5457486     long</span>
<span class="co">##                       rep   repOvlp       transcripts        flag</span>
<span class="co">##           &lt;CharacterList&gt; &lt;integer&gt;             &lt;Rle&gt; &lt;character&gt;</span>
<span class="co">##    [1]               &lt;NA&gt;         0              &lt;NA&gt;        &lt;NA&gt;</span>
<span class="co">##    [2]               &lt;NA&gt;         0              &lt;NA&gt;        &lt;NA&gt;</span>
<span class="co">##    [3]               &lt;NA&gt;         0              &lt;NA&gt;        &lt;NA&gt;</span>
<span class="co">##    [4]               &lt;NA&gt;         0 g2111.t1;g2111.t2        &lt;NA&gt;</span>
<span class="co">##    [5]               &lt;NA&gt;         0              &lt;NA&gt;        &lt;NA&gt;</span>
<span class="co">##    [6]               &lt;NA&gt;         0              &lt;NA&gt;        &lt;NA&gt;</span>
<span class="co">##    [7]               &lt;NA&gt;         0 g2111.t1;g2111.t2         Inv</span>
<span class="co">##    [8]               &lt;NA&gt;         0              &lt;NA&gt;        &lt;NA&gt;</span>
<span class="co">##    [9] unknown,rnd,tandem      1107 g2111.t1;g2111.t2         Inv</span>
<span class="co">##   [10]               &lt;NA&gt;         0              &lt;NA&gt;        &lt;NA&gt;</span>
<span class="co">##   [11]               &lt;NA&gt;         0          g2110.t1        &lt;NA&gt;</span>
<span class="co">##   -------</span>
<span class="co">##   seqinfo: 19 sequences from OKI2018.I69 genome</span></code></pre>
</div>
<div class="section level2">
<h2 id="filling-inversions-to-better-see-the-rest-roi9">Filling inversions to better see the rest (ROI9)<a class="anchor" aria-label="anchor" href="#filling-inversions-to-better-see-the-rest-roi9"></a>
</h2>
<p>We will not fill inversions in objects we analyse, because we do not want to erase their breakpoints.</p>
<div class="sourceCode" id="cb114"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">fillInversions</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">gb</span><span class="op">)</span> <span class="op">{</span>
  <span class="va">Invs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/which.html" class="external-link">which</a></span><span class="op">(</span><span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/flagInversions.html" class="external-link">flagInversions</a></span><span class="op">(</span><span class="va">gb</span><span class="op">)</span><span class="op">$</span><span class="va">inv</span><span class="op">)</span> <span class="op">+</span> <span class="fl">1</span>
  <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/strand.html" class="external-link">strand</a></span><span class="op">(</span><span class="va">gb</span><span class="op">)</span><span class="op">[</span><span class="va">Invs</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html" class="external-link">ifelse</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/strand.html" class="external-link">strand</a></span><span class="op">(</span><span class="va">gb</span><span class="op">)</span><span class="op">[</span><span class="va">Invs</span><span class="op">]</span> <span class="op">==</span> <span class="st">"+"</span>, <span class="st">"-"</span>, <span class="st">"+"</span><span class="op">)</span>
  <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/coalesce_contigs.html" class="external-link">coalesce_contigs</a></span><span class="op">(</span><span class="va">gb</span><span class="op">)</span>
<span class="op">}</span>

<span class="va">ROI9.all</span> <span class="op">&lt;-</span> <span class="fu">fillInversions</span><span class="op">(</span><span class="va">coa2</span><span class="op">$</span><span class="va">Oki_Kum</span><span class="op">)</span>

<span class="va">ROI9</span> <span class="op">&lt;-</span> <span class="va">ROI9.all</span> |&gt;
  <span class="fu">plyranges</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/plyranges/man/tidyverse-reexports.html" class="external-link">filter</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqnames</a></span><span class="op">(</span><span class="va">query</span><span class="op">)</span> <span class="op">==</span> <span class="st">"contig_3_1"</span><span class="op">)</span> |&gt;
  <span class="fu">plyranges</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/plyranges/man/tidyverse-reexports.html" class="external-link">arrange</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/start.html" class="external-link">start</a></span><span class="op">(</span><span class="va">query</span><span class="op">)</span><span class="op">)</span> |&gt;
  <span class="fu">plyranges</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/plyranges/man/tidyverse-reexports.html" class="external-link">slice</a></span><span class="op">(</span><span class="fl">10</span><span class="op">:</span><span class="fl">20</span><span class="op">)</span>

<span class="va">ROI9.range</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/inter-range-methods.html" class="external-link">range</a></span><span class="op">(</span><span class="va">ROI9</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>

<span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/plotApairOfChrs.html" class="external-link">plotApairOfChrs</a></span><span class="op">(</span><span class="va">ROI9</span><span class="op">)</span></code></pre></div>
<p><img src="RegionsOfInterest_files/figure-html/unnamed-chunk-2-1.png" width="700"></p>
<div class="sourceCode" id="cb115"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/plotApairOfChrs.html" class="external-link">plotApairOfChrs</a></span><span class="op">(</span><span class="va">ROI9.all</span>, xlim <span class="op">=</span> <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/gb2xlim.html" class="external-link">gb2xlim</a></span><span class="op">(</span><span class="va">ROI9.range</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="RegionsOfInterest_files/figure-html/unnamed-chunk-2-2.png" width="700"></p>
<div class="sourceCode" id="cb116"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/findOverlaps-methods.html" class="external-link">subsetByOverlaps</a></span><span class="op">(</span><span class="va">ROI9.all</span>, <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/genomic-range-squeezers.html" class="external-link">granges</a></span><span class="op">(</span><span class="va">ROI9.range</span><span class="op">)</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/flagAll.html" class="external-link">flagAll</a></span><span class="op">(</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">##    seqnames    start      end  width strand query.seqnames query.start</span>
<span class="co">## 1      chr1  8831594  8904792  73199      +     contig_3_1      781316</span>
<span class="co">## 2      chr1  8904796  8905391    596      -    contig_87_1      129264</span>
<span class="co">## 3      chr1  8907966  9178393 270428      +     contig_3_1      857220</span>
<span class="co">## 4      chr1  9179294  9179883    590      -    contig_28_1     2686583</span>
<span class="co">## 5      chr1  9180258  9238680  58423      +     contig_3_1     1103683</span>
<span class="co">## 6      chr1  9238754  9238914    161      -    contig_11_1       64537</span>
<span class="co">## 7      chr1  9247203  9756276 509074      +     contig_3_1     1170666</span>
<span class="co">## 8      chr1  9756672 10487037 730366      +     contig_3_1     1665784</span>
<span class="co">## 9      chr1 10489186 10490144    959      -    contig_82_1      469347</span>
<span class="co">## 10     chr1 10490223 10579632  89410      +     contig_3_1     2335682</span>
<span class="co">## 11     chr1 10581313 10582002    690      +    contig_42_1    12447665</span>
<span class="co">## 12     chr1 10583588 10707107 123520      +     contig_3_1     2422786</span>
<span class="co">## 13     chr1 10707650 10708228    579      -    contig_43_1      542809</span>
<span class="co">## 14     chr1 10709490 10826652 117163      +     contig_3_1     2509221</span>
<span class="co">## 15     chr1 10829074 10829372    299      +    contig_28_1     4919962</span>
<span class="co">## 16     chr1 10830830 10831185    356      -    contig_28_1     4917700</span>
<span class="co">## 17     chr1 10831349 10832281    933      +    contig_28_1     4921734</span>
<span class="co">## 18     chr1 10832293 11220548 388256      +     contig_3_1     2614209</span>
<span class="co">## 19     chr1 11223351 11224198    848      -    contig_90_1      515726</span>
<span class="co">## 20     chr1 11224199 11224556    358      +    contig_29_1     2413024</span>
<span class="co">## 21     chr1 11225607 11565085 339479      +     contig_3_1     2973842</span>
<span class="co">##    query.end query.width query.strand  score flag</span>
<span class="co">## 1     857219       75904            *  73199  Tra</span>
<span class="co">## 2     129861         598            *    596 &lt;NA&gt;</span>
<span class="co">## 3    1103682      246463            * 270428  Tra</span>
<span class="co">## 4    2687173         591            *    590 &lt;NA&gt;</span>
<span class="co">## 5    1169654       65972            *  58423  Tra</span>
<span class="co">## 6      64695         159            *    161 &lt;NA&gt;</span>
<span class="co">## 7    1648012      477347            * 509074  Col</span>
<span class="co">## 8    2335681      669898            * 730366  Tra</span>
<span class="co">## 9     470292         946            *    959 &lt;NA&gt;</span>
<span class="co">## 10   2422784       87103            *  89410  Tra</span>
<span class="co">## 11  12448361         697            *    690 &lt;NA&gt;</span>
<span class="co">## 12   2509220       86435            * 123520  Tra</span>
<span class="co">## 13    543388         580            *    579 &lt;NA&gt;</span>
<span class="co">## 14   2614206      104986            * 117163 &lt;NA&gt;</span>
<span class="co">## 15   4920260         299            *    299  Tra</span>
<span class="co">## 16   4918055         356            *    356 &lt;NA&gt;</span>
<span class="co">## 17   4922665         932            *    933 &lt;NA&gt;</span>
<span class="co">## 18   2973841      359633            * 388256 &lt;NA&gt;</span>
<span class="co">## 19    516573         848            *    848 &lt;NA&gt;</span>
<span class="co">## 20   2413382         359            *    358 &lt;NA&gt;</span>
<span class="co">## 21   3316589      342748            * 339479 &lt;NA&gt;</span></code></pre>
<div class="sourceCode" id="cb118"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/findOverlaps-methods.html" class="external-link">subsetByOverlaps</a></span><span class="op">(</span><span class="va">ROI9.all</span> |&gt; <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/swap.html" class="external-link">swap</a></span><span class="op">(</span><span class="op">)</span>, <span class="va">ROI9.range</span><span class="op">$</span><span class="va">query</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## GBreaks object with 11 ranges and 1 metadata column:</span>
<span class="co">##          seqnames          ranges strand |                  query</span>
<span class="co">##             &lt;Rle&gt;       &lt;IRanges&gt;  &lt;Rle&gt; |              &lt;GRanges&gt;</span>
<span class="co">##    [1] contig_3_1   781316-857219      + |   chr1:8831594-8904792</span>
<span class="co">##    [2] contig_3_1  857220-1103682      + |   chr1:8907966-9178393</span>
<span class="co">##    [3] contig_3_1 1103683-1169654      + |   chr1:9180258-9238680</span>
<span class="co">##    [4] contig_3_1 1170666-1648012      + |   chr1:9247203-9756276</span>
<span class="co">##    [5] contig_3_1 1665784-2335681      + |  chr1:9756672-10487037</span>
<span class="co">##    [6] contig_3_1 2335682-2422784      + | chr1:10490223-10579632</span>
<span class="co">##    [7] contig_3_1 2422786-2509220      + | chr1:10583588-10707107</span>
<span class="co">##    [8] contig_3_1 2509221-2614206      + | chr1:10709490-10826652</span>
<span class="co">##    [9] contig_3_1 2614209-2973841      + | chr1:10832293-11220548</span>
<span class="co">##   [10] contig_3_1 2973842-3316589      + | chr1:11225607-11565085</span>
<span class="co">##   [11] contig_3_1 1652225-1652940      + |    YSR:2724777-2725492</span>
<span class="co">##   -------</span>
<span class="co">##   seqinfo: 55 sequences from KUM.M3.7f genome</span></code></pre>
<div class="sourceCode" id="cb120"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">x</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/cbind.html" class="external-link">rbind</a></span><span class="op">(</span>
  <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">gbs</span>, \<span class="op">(</span><span class="va">gb</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/table.html" class="external-link">table</a></span><span class="op">(</span><span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/flagPairs.html" class="external-link">flagPairs</a></span><span class="op">(</span><span class="va">gb</span><span class="op">)</span><span class="op">$</span><span class="va">pairs</span><span class="op">)</span><span class="op">)</span>
<span class="op">)</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="roi-10-small-unaligned-inversion-in-oki-kum-alignment">ROI 10: small unaligned inversion in Oki-Kum alignment<a class="anchor" aria-label="anchor" href="#roi-10-small-unaligned-inversion-in-oki-kum-alignment"></a>
</h2>
<div class="section level3">
<h3 id="the-small-unaligned-inversion">The small unaligned inversion<a class="anchor" aria-label="anchor" href="#the-small-unaligned-inversion"></a>
</h3>
<p>Martin reported a small unaligned inversion at PAR:80769-81032 in the Oki â€“ Kum alignment (version 3). Note that it will not be visible on the plot if this vignette is rebuilt with an alignment updated to include these inversions.</p>
<div class="sourceCode" id="cb121"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ROI10</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a></span><span class="op">(</span><span class="st">"PAR:80769-81032:-"</span><span class="op">)</span>
<span class="va">ROI10_inv</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/findOverlaps-methods.html" class="external-link">subsetByOverlaps</a></span><span class="op">(</span><span class="va">gbs</span><span class="op">$</span><span class="va">Oki_Kum</span>, <span class="va">ROI10</span> <span class="op">+</span> <span class="fl">1e2</span><span class="op">)</span>

<span class="va">ROI10_inv</span> |&gt; <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/plotApairOfChrs.html" class="external-link">plotApairOfChrs</a></span><span class="op">(</span>main <span class="op">=</span> <span class="st">"A short unaligned region that is actually an inversion within the inversion."</span><span class="op">)</span></code></pre></div>
<p><img src="RegionsOfInterest_files/figure-html/ROI10-1.png" width="700"></p>
<div class="sourceCode" id="cb122"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ROI10_inv</span>       |&gt; <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/cleanGaps.html" class="external-link">cleanGaps</a></span><span class="op">(</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/r/base/character.html" class="external-link">as.character</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## [1] "PAR:80780-81022"</span></code></pre>
<div class="sourceCode" id="cb124"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ROI10_inv</span><span class="op">$</span><span class="va">query</span> |&gt; <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/cleanGaps.html" class="external-link">cleanGaps</a></span><span class="op">(</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/r/base/character.html" class="external-link">as.character</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## [1] "contig_57_1:58108-58350"</span></code></pre>
<div class="sourceCode" id="cb126"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ROI10_unal</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/findOverlaps-methods.html" class="external-link">subsetByOverlaps</a></span><span class="op">(</span><span class="va">bri</span><span class="op">$</span><span class="va">Oki_Kum</span>, <span class="va">ROI10</span> <span class="op">+</span> <span class="fl">1e2</span><span class="op">)</span></code></pre></div>
<p>Interestingly, this small inversion is flanked by an inverted repeat.</p>
<div class="sourceCode" id="cb127"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/pairwiseAlignment.html" class="external-link">pairwiseAlignment</a></span><span class="op">(</span><span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/getSeq.html" class="external-link">getSeq</a></span><span class="op">(</span><span class="va">ROI10_unal</span>  <span class="op">)</span>,
                  <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/getSeq.html" class="external-link">getSeq</a></span><span class="op">(</span><span class="va">ROI10_unal</span><span class="op">$</span><span class="va">query</span> <span class="op">)</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/PairwiseAlignments-io.html" class="external-link">writePairwiseAlignments</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## ########################################</span>
<span class="co">## # Program: Biostrings (version 2.62.0), a Bioconductor package</span>
<span class="co">## # Rundate: Wed Mar 30 19:17:39 2022</span>
<span class="co">## ########################################</span>
<span class="co">## #=======================================</span>
<span class="co">## #</span>
<span class="co">## # Aligned_sequences: 2</span>
<span class="co">## # 1: P1</span>
<span class="co">## # 2: S1</span>
<span class="co">## # Matrix: NA</span>
<span class="co">## # Gap_penalty: 14.0</span>
<span class="co">## # Extend_penalty: 4.0</span>
<span class="co">## #</span>
<span class="co">## # Length: 243</span>
<span class="co">## # Identity:     243/243 (100.0%)</span>
<span class="co">## # Similarity:    NA/243 (NA%)</span>
<span class="co">## # Gaps:           0/243 (0.0%)</span>
<span class="co">## # Score: 481.5658</span>
<span class="co">## #</span>
<span class="co">## #</span>
<span class="co">## #=======================================</span>
<span class="co">## </span>
<span class="co">## P1                 1 ACTGTCTTGTAGAGAATCGTCAGTTCTACAAATCCTGTTTTTCAGATTTT     50</span>
<span class="co">##                      ||||||||||||||||||||||||||||||||||||||||||||||||||</span>
<span class="co">## S1                 1 ACTGTCTTGTAGAGAATCGTCAGTTCTACAAATCCTGTTTTTCAGATTTT     50</span>
<span class="co">## </span>
<span class="co">## P1                51 TGTTTGAGGCTTTCATTAATTTTACATCGCAATTTGTAAAGAGCGCTACT    100</span>
<span class="co">##                      ||||||||||||||||||||||||||||||||||||||||||||||||||</span>
<span class="co">## S1                51 TGTTTGAGGCTTTCATTAATTTTACATCGCAATTTGTAAAGAGCGCTACT    100</span>
<span class="co">## </span>
<span class="co">## P1               101 TTTTCTCGCGCCTCTCCGCTCTCTATTTCTCACGCCACGACCTCACATCG    150</span>
<span class="co">##                      ||||||||||||||||||||||||||||||||||||||||||||||||||</span>
<span class="co">## S1               101 TTTTCTCGCGCCTCTCCGCTCTCTATTTCTCACGCCACGACCTCACATCG    150</span>
<span class="co">## </span>
<span class="co">## P1               151 CTTACTTCAAGGTCGCATATTAAAATCCGAAGATCTTAAAACAAAAATCT    200</span>
<span class="co">##                      ||||||||||||||||||||||||||||||||||||||||||||||||||</span>
<span class="co">## S1               151 CTTACTTCAAGGTCGCATATTAAAATCCGAAGATCTTAAAACAAAAATCT    200</span>
<span class="co">## </span>
<span class="co">## P1               201 GAAAAACAGGATTTGTAGAACTGACGATTCTCTACAAGACAGT    243</span>
<span class="co">##                      |||||||||||||||||||||||||||||||||||||||||||</span>
<span class="co">## S1               201 GAAAAACAGGATTTGTAGAACTGACGATTCTCTACAAGACAGT    243</span>
<span class="co">## </span>
<span class="co">## </span>
<span class="co">## #---------------------------------------</span>
<span class="co">## #---------------------------------------</span></code></pre>
<div class="sourceCode" id="cb129"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/pairwiseAlignment.html" class="external-link">pairwiseAlignment</a></span><span class="op">(</span><span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/getSeq.html" class="external-link">getSeq</a></span><span class="op">(</span><span class="va">ROI10_unal</span>  <span class="op">)</span>,
                  <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/getSeq.html" class="external-link">getSeq</a></span><span class="op">(</span><span class="va">ROI10_unal</span><span class="op">$</span><span class="va">query</span> <span class="op">)</span>|&gt; <span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/reverseComplement.html" class="external-link">reverseComplement</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/PairwiseAlignments-io.html" class="external-link">writePairwiseAlignments</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## ########################################</span>
<span class="co">## # Program: Biostrings (version 2.62.0), a Bioconductor package</span>
<span class="co">## # Rundate: Wed Mar 30 19:17:39 2022</span>
<span class="co">## ########################################</span>
<span class="co">## #=======================================</span>
<span class="co">## #</span>
<span class="co">## # Aligned_sequences: 2</span>
<span class="co">## # 1: P1</span>
<span class="co">## # 2: S1</span>
<span class="co">## # Matrix: NA</span>
<span class="co">## # Gap_penalty: 14.0</span>
<span class="co">## # Extend_penalty: 4.0</span>
<span class="co">## #</span>
<span class="co">## # Length: 246</span>
<span class="co">## # Identity:     170/246 (69.1%)</span>
<span class="co">## # Similarity:    NA/246 (NA%)</span>
<span class="co">## # Gaps:           6/246 (2.4%)</span>
<span class="co">## # Score: -140.052</span>
<span class="co">## #</span>
<span class="co">## #</span>
<span class="co">## #=======================================</span>
<span class="co">## </span>
<span class="co">## P1                 1 ACTGTCTTGTAGAGAATCGTCAGTTCTACAAATCCTGTTTTTCAGATTTT     50</span>
<span class="co">##                      ||||||||||||||||||||||||||||||||||||||||||||||||||</span>
<span class="co">## S1                 1 ACTGTCTTGTAGAGAATCGTCAGTTCTACAAATCCTGTTTTTCAGATTTT     50</span>
<span class="co">## </span>
<span class="co">## P1                51 TGTTTGAGGCTTTCATTAATTTTACATCGCAATTTGTAAAGAGCGCTACT    100</span>
<span class="co">##                      ||||| | | | |    | ||| | ||   |  ||| |   |||| |   </span>
<span class="co">## S1                51 TGTTTTAAGATCTTCGGATTTTAATATGCGACCTTGAAGTAAGCGATG--     98</span>
<span class="co">## </span>
<span class="co">## P1               101 TTTTCTCGCGCCTCTCCGCTCTCTATTTCTCAC-GCCACGACCTCACAT-    148</span>
<span class="co">##                      |    ||| | |  |  |   |  |   |  |  | | |||    |  | </span>
<span class="co">## S1                99 TGAGGTCGTGGCG-TGAGAAATAGAGAGCGGAGAGGCGCGAGAAAAAGTA    147</span>
<span class="co">## </span>
<span class="co">## P1               149 -CGCTTACTTCAAGGTCGCATATTAAAATCCGAAGATCTTAAAACAAAAA    197</span>
<span class="co">##                       ||||   | |||  |   || | ||| |    | | | | |||||||||</span>
<span class="co">## S1               148 GCGCTCTTTACAAATTGCGATGTAAAATTAATGAAAGCCTCAAACAAAAA    197</span>
<span class="co">## </span>
<span class="co">## P1               198 TCTGAAAAACAGGATTTGTAGAACTGACGATTCTCTACAAGACAGT    243</span>
<span class="co">##                      ||||||||||||||||||||||||||||||||||||||||||||||</span>
<span class="co">## S1               198 TCTGAAAAACAGGATTTGTAGAACTGACGATTCTCTACAAGACAGT    243</span>
<span class="co">## </span>
<span class="co">## </span>
<span class="co">## #---------------------------------------</span>
<span class="co">## #---------------------------------------</span></code></pre>
<div class="sourceCode" id="cb131"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/pairwiseAlignment.html" class="external-link">pairwiseAlignment</a></span><span class="op">(</span><span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/getSeq.html" class="external-link">getSeq</a></span><span class="op">(</span><span class="va">ROI10_unal</span>  <span class="op">)</span>,
                  <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/getSeq.html" class="external-link">getSeq</a></span><span class="op">(</span><span class="va">ROI10_unal</span> <span class="op">)</span>|&gt; <span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/reverseComplement.html" class="external-link">reverseComplement</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/PairwiseAlignments-io.html" class="external-link">writePairwiseAlignments</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## ########################################</span>
<span class="co">## # Program: Biostrings (version 2.62.0), a Bioconductor package</span>
<span class="co">## # Rundate: Wed Mar 30 19:17:39 2022</span>
<span class="co">## ########################################</span>
<span class="co">## #=======================================</span>
<span class="co">## #</span>
<span class="co">## # Aligned_sequences: 2</span>
<span class="co">## # 1: P1</span>
<span class="co">## # 2: S1</span>
<span class="co">## # Matrix: NA</span>
<span class="co">## # Gap_penalty: 14.0</span>
<span class="co">## # Extend_penalty: 4.0</span>
<span class="co">## #</span>
<span class="co">## # Length: 246</span>
<span class="co">## # Identity:     170/246 (69.1%)</span>
<span class="co">## # Similarity:    NA/246 (NA%)</span>
<span class="co">## # Gaps:           6/246 (2.4%)</span>
<span class="co">## # Score: -140.052</span>
<span class="co">## #</span>
<span class="co">## #</span>
<span class="co">## #=======================================</span>
<span class="co">## </span>
<span class="co">## P1                 1 ACTGTCTTGTAGAGAATCGTCAGTTCTACAAATCCTGTTTTTCAGATTTT     50</span>
<span class="co">##                      ||||||||||||||||||||||||||||||||||||||||||||||||||</span>
<span class="co">## S1                 1 ACTGTCTTGTAGAGAATCGTCAGTTCTACAAATCCTGTTTTTCAGATTTT     50</span>
<span class="co">## </span>
<span class="co">## P1                51 TGTTTGAGGCTTTCATTAATTTTACATCGCAATTTGTAAAGAGCGCTACT    100</span>
<span class="co">##                      ||||| | | | |    | ||| | ||   |  ||| |   |||| |   </span>
<span class="co">## S1                51 TGTTTTAAGATCTTCGGATTTTAATATGCGACCTTGAAGTAAGCGATG--     98</span>
<span class="co">## </span>
<span class="co">## P1               101 TTTTCTCGCGCCTCTCCGCTCTCTATTTCTCAC-GCCACGACCTCACAT-    148</span>
<span class="co">##                      |    ||| | |  |  |   |  |   |  |  | | |||    |  | </span>
<span class="co">## S1                99 TGAGGTCGTGGCG-TGAGAAATAGAGAGCGGAGAGGCGCGAGAAAAAGTA    147</span>
<span class="co">## </span>
<span class="co">## P1               149 -CGCTTACTTCAAGGTCGCATATTAAAATCCGAAGATCTTAAAACAAAAA    197</span>
<span class="co">##                       ||||   | |||  |   || | ||| |    | | | | |||||||||</span>
<span class="co">## S1               148 GCGCTCTTTACAAATTGCGATGTAAAATTAATGAAAGCCTCAAACAAAAA    197</span>
<span class="co">## </span>
<span class="co">## P1               198 TCTGAAAAACAGGATTTGTAGAACTGACGATTCTCTACAAGACAGT    243</span>
<span class="co">##                      ||||||||||||||||||||||||||||||||||||||||||||||</span>
<span class="co">## S1               198 TCTGAAAAACAGGATTTGTAGAACTGACGATTCTCTACAAGACAGT    243</span>
<span class="co">## </span>
<span class="co">## </span>
<span class="co">## #---------------------------------------</span>
<span class="co">## #---------------------------------------</span></code></pre>
<div class="sourceCode" id="cb133"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/getSeq.html" class="external-link">getSeq</a></span><span class="op">(</span><span class="va">genomes</span><span class="op">$</span><span class="va">OKI2018.I69</span>, <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a></span><span class="op">(</span><span class="st">"PAR:80523-80834"</span><span class="op">)</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/r/base/character.html" class="external-link">as.character</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## [1] "TTAAGGGTAATGGTATACATCGGCGCGGCGCTGCTCTCTCGATCCGCGCTCCGCCAGAACCTATTTCTCACGAGGCCGTAAAATCTGAACGGATAAAGATTTTTTCAAAATTCAAAAAGATTCTGAATAATTAAAGTCTTTTCTACTAACACAGCAAAATTGGTAATGATTGTCAAAGAGGAACTATCTGAAAATTGCCAGTTTCCTTCAAATGACATTTTCTTCAACATTTCTCATGCAAAATTTTTGAAAATTACACTGTCTTGTAGAGAATCGTCAGTTCTACAAATCCTGTTTTTCAGATTTTTGTTT"</span></code></pre>
<p>This sequence matches <code>MITE_oki2018_i69_4_20065_S1_MITE#DNA/MITE</code> in Sashaâ€™s repeat annotation.</p>
<p>~20 matches on the Okinawa genome. Uncomment to check.</p>
<div class="sourceCode" id="cb135"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#vmatchPattern(genomes$OKI2018.I69$PAR[80523:80834], genomes$OKI2018.I69, with.indels = TRUE, max.mismatch = 20) |&gt; sort(ignore.strand = TRUE) |&gt; as.data.frame()</span></code></pre></div>
</div>
<div class="section level3">
<h3 id="other-interesting-things-in-the-area">Other interesting things in the area<a class="anchor" aria-label="anchor" href="#other-interesting-things-in-the-area"></a>
</h3>
<p>Expanding the area broader, there is this interesting pattern where there is only one gap on the <em>Oki</em> side and two gaps on the <em>Kum</em> side. This is because the alignments on the <em>Oki</em> side are directly contacting each other.</p>
<p>(Note: in the meantime this was explored further in <code><a href="../articles/TandemSiteDuplications.html">vignette("TandemSiteDuplications", package = "OikScrambling")</a></code> )</p>
<div class="sourceCode" id="cb136"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ROI10_ins</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/findOverlaps-methods.html" class="external-link">subsetByOverlaps</a></span><span class="op">(</span><span class="va">gbs</span><span class="op">$</span><span class="va">Oki_Kum</span>, <span class="va">ROI10</span> <span class="op">+</span> <span class="fl">3e3</span><span class="op">)</span>
<span class="va">ROI10_ins</span> |&gt; <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/plotApairOfChrs.html" class="external-link">plotApairOfChrs</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<p><img src="RegionsOfInterest_files/figure-html/ROI10_insertion-1.png" width="700"></p>
<div class="sourceCode" id="cb137"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ROI10_ins</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">##   seqnames start   end width strand score query.seqnames query.start query.end</span>
<span class="co">## 1      PAR 75427 80779  5353      - 30134    contig_57_1       58351     63716</span>
<span class="co">## 2      PAR 81023 83450  2428      - 13433    contig_57_1       55648     58107</span>
<span class="co">## 3      PAR 83451 85486  2036      - 11492    contig_57_1       53302     55336</span>
<span class="co">##   query.width query.strand    query.rep query.repOvlp query.transcripts   Arm</span>
<span class="co">## 1        5366            * unknown, rnd          2235              &lt;NA&gt; short</span>
<span class="co">## 2        2460            * rnd, unknown          2338              &lt;NA&gt; short</span>
<span class="co">## 3        2035            *          rnd           697              &lt;NA&gt; short</span>
<span class="co">##            rep repOvlp transcripts flag nonCoa</span>
<span class="co">## 1 rnd, unknown     915        &lt;NA&gt;  Col  FALSE</span>
<span class="co">## 2 unknown,....    1720        &lt;NA&gt;  Col  FALSE</span>
<span class="co">## 3   ltr-1, rnd      92        &lt;NA&gt;  Col  FALSE</span></code></pre>
<div class="sourceCode" id="cb139"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ROI10_ins</span> |&gt; <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/cleanGaps.html" class="external-link">cleanGaps</a></span><span class="op">(</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">##   seqnames start   end width strand</span>
<span class="co">## 1      PAR 80780 81022   243      *</span></code></pre>
<div class="sourceCode" id="cb141"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ROI10_ins</span><span class="op">$</span><span class="va">query</span> |&gt; <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/cleanGaps.html" class="external-link">cleanGaps</a></span><span class="op">(</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">##      seqnames start   end width strand</span>
<span class="co">## 1 contig_57_1 55337 55647   311      *</span>
<span class="co">## 2 contig_57_1 58108 58350   243      *</span></code></pre>
<div class="sourceCode" id="cb143"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">tail</a></span><span class="op">(</span><span class="va">ROI10_ins</span>, <span class="fl">2</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/plotApairOfChrs.html" class="external-link">plotApairOfChrs</a></span><span class="op">(</span>main <span class="op">=</span> <span class="st">"An insertion in the Kume genome"</span><span class="op">)</span></code></pre></div>
<p><img src="RegionsOfInterest_files/figure-html/ROI10_insertion-2.png" width="700"></p>
<div class="sourceCode" id="cb144"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">tail</a></span><span class="op">(</span><span class="va">ROI10_ins</span>, <span class="fl">2</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/swap.html" class="external-link">swap</a></span><span class="op">(</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/cleanGaps.html" class="external-link">cleanGaps</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## GRanges object with 1 range and 0 metadata columns:</span>
<span class="co">##          seqnames      ranges strand</span>
<span class="co">##             &lt;Rle&gt;   &lt;IRanges&gt;  &lt;Rle&gt;</span>
<span class="co">##   [1] contig_57_1 55337-55647      *</span>
<span class="co">##   -------</span>
<span class="co">##   seqinfo: 55 sequences from KUM.M3.7f genome</span></code></pre>
<div class="sourceCode" id="cb146"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="op">(</span><span class="va">ROI10_ins_bri</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/findOverlaps-methods.html" class="external-link">subsetByOverlaps</a></span><span class="op">(</span><span class="va">bri</span><span class="op">$</span><span class="va">Oki_Kum</span> <span class="op">+</span> <span class="fl">1</span>, <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">tail</a></span><span class="op">(</span><span class="va">ROI10_ins</span>, <span class="fl">2</span><span class="op">)</span> <span class="op">)</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## GBreaks object with 1 range and 5 metadata columns:</span>
<span class="co">##       seqnames      ranges strand |                   query      Arm</span>
<span class="co">##          &lt;Rle&gt;   &lt;IRanges&gt;  &lt;Rle&gt; |               &lt;GRanges&gt; &lt;factor&gt;</span>
<span class="co">##   [1]      PAR 83450-83451      * | contig_57_1:55337-55647    short</span>
<span class="co">##                   rep   repOvlp transcripts</span>
<span class="co">##       &lt;CharacterList&gt; &lt;integer&gt;       &lt;Rle&gt;</span>
<span class="co">##   [1]           ltr-1         0        &lt;NA&gt;</span>
<span class="co">##   -------</span>
<span class="co">##   seqinfo: 19 sequences from OKI2018.I69 genome</span></code></pre>
<div class="sourceCode" id="cb148"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Left side</span>
<span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/pairwiseAlignment.html" class="external-link">pairwiseAlignment</a></span><span class="op">(</span><span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/getSeq.html" class="external-link">getSeq</a></span><span class="op">(</span><span class="va">genomes</span><span class="op">$</span><span class="va">OKI2018.I69</span>, <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a></span><span class="op">(</span><span class="st">"PAR:83350-83550"</span><span class="op">)</span><span class="op">)</span>,
                  <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/getSeq.html" class="external-link">getSeq</a></span><span class="op">(</span><span class="va">genomes</span><span class="op">$</span><span class="va">KUM.M3.7f</span>, <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a></span><span class="op">(</span><span class="st">"contig_57_1:55548-55748"</span><span class="op">)</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/reverseComplement.html" class="external-link">reverseComplement</a></span><span class="op">(</span><span class="op">)</span>,
                  type <span class="op">=</span> <span class="st">"global"</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/PairwiseAlignments-io.html" class="external-link">writePairwiseAlignments</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## ########################################</span>
<span class="co">## # Program: Biostrings (version 2.62.0), a Bioconductor package</span>
<span class="co">## # Rundate: Wed Mar 30 19:17:41 2022</span>
<span class="co">## ########################################</span>
<span class="co">## #=======================================</span>
<span class="co">## #</span>
<span class="co">## # Aligned_sequences: 2</span>
<span class="co">## # 1: P1</span>
<span class="co">## # 2: S1</span>
<span class="co">## # Matrix: NA</span>
<span class="co">## # Gap_penalty: 14.0</span>
<span class="co">## # Extend_penalty: 4.0</span>
<span class="co">## #</span>
<span class="co">## # Length: 212</span>
<span class="co">## # Identity:     151/212 (71.2%)</span>
<span class="co">## # Similarity:    NA/212 (NA%)</span>
<span class="co">## # Gaps:          22/212 (10.4%)</span>
<span class="co">## # Score: -78.8276</span>
<span class="co">## #</span>
<span class="co">## #</span>
<span class="co">## #=======================================</span>
<span class="co">## </span>
<span class="co">## P1                 1 AGAAAAGACTTTTAAAATATGTATTCCTAAATTGACTCGAAATGTTCCAA     50</span>
<span class="co">##                      ||||||||||||||||||||||||||||||||||||||||||||||||||</span>
<span class="co">## S1                 1 AGAAAAGACTTTTAAAATATGTATTCCTAAATTGACTCGAAATGTTCCAA     50</span>
<span class="co">## </span>
<span class="co">## P1                51 CTTAAGAGGACTTTCTCTTAGCCCTTGTCATCTCAAAAGGCCTCTTAAGA    100</span>
<span class="co">##                      ||||||||||||||||||||||||||||||||||||||||||||||||||</span>
<span class="co">## S1                51 CTTAAGAGGACTTTCTCTTAGCCCTTGTCATCTCAAAAGGCCTCTTAAGA    100</span>
<span class="co">## </span>
<span class="co">## P1               101 TGGATACCAACTGTATTTAGATAGAGACATTATACCAATGATCATCTCGC    150</span>
<span class="co">##                      ||| |       |||    |   | |||  || |    || | |   || </span>
<span class="co">## S1               101 TGGGT-------GTAGGGCG---GCGACGGTACAAATTTG-TAACGGCGG    139</span>
<span class="co">## </span>
<span class="co">## P1               151 TTAAAATTTGT---TCGAGTCATCAAATTCTTGA------GTTATGGGGC    191</span>
<span class="co">##                      |  ||||||||    |  ||    ||||| |||       | ||   | |</span>
<span class="co">## S1               140 TACAAATTTGTAACGCCGGTACAAAAATTTTTGTACCGCCGGTACAAGAC    189</span>
<span class="co">## </span>
<span class="co">## P1               192 TCAGAAACAA--    201</span>
<span class="co">##                        | | | ||  </span>
<span class="co">## S1               190 AAAAATAGAATC    201</span>
<span class="co">## </span>
<span class="co">## </span>
<span class="co">## #---------------------------------------</span>
<span class="co">## #---------------------------------------</span></code></pre>
<div class="sourceCode" id="cb150"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Right side</span>
<span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/pairwiseAlignment.html" class="external-link">pairwiseAlignment</a></span><span class="op">(</span><span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/getSeq.html" class="external-link">getSeq</a></span><span class="op">(</span><span class="va">genomes</span><span class="op">$</span><span class="va">OKI2018.I69</span>, <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a></span><span class="op">(</span><span class="st">"PAR:83351-83551"</span><span class="op">)</span><span class="op">)</span>,
                  <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/getSeq.html" class="external-link">getSeq</a></span><span class="op">(</span><span class="va">genomes</span><span class="op">$</span><span class="va">KUM.M3.7f</span>, <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a></span><span class="op">(</span><span class="st">"contig_57_1:55236-55436"</span><span class="op">)</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/reverseComplement.html" class="external-link">reverseComplement</a></span><span class="op">(</span><span class="op">)</span>,
                  type <span class="op">=</span> <span class="st">"global"</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/PairwiseAlignments-io.html" class="external-link">writePairwiseAlignments</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## ########################################</span>
<span class="co">## # Program: Biostrings (version 2.62.0), a Bioconductor package</span>
<span class="co">## # Rundate: Wed Mar 30 19:17:42 2022</span>
<span class="co">## ########################################</span>
<span class="co">## #=======================================</span>
<span class="co">## #</span>
<span class="co">## # Aligned_sequences: 2</span>
<span class="co">## # 1: P1</span>
<span class="co">## # 2: S1</span>
<span class="co">## # Matrix: NA</span>
<span class="co">## # Gap_penalty: 14.0</span>
<span class="co">## # Extend_penalty: 4.0</span>
<span class="co">## #</span>
<span class="co">## # Length: 203</span>
<span class="co">## # Identity:     146/203 (71.9%)</span>
<span class="co">## # Similarity:    NA/203 (NA%)</span>
<span class="co">## # Gaps:           4/203 (2.0%)</span>
<span class="co">## # Score: -59.32605</span>
<span class="co">## #</span>
<span class="co">## #</span>
<span class="co">## #=======================================</span>
<span class="co">## </span>
<span class="co">## P1                 1 --GAAAAGACTTTTAAAATATGTATTCCTAAATTGACTCGAAATGTTCCA     48</span>
<span class="co">##                        |  || ||  ||   || | |  | |       ||   |  ||| || </span>
<span class="co">## S1                 1 TTGTCAAAACAATTGCGATTTTTTGTACCGGCGGTACAAAATTTGTACCC     50</span>
<span class="co">## </span>
<span class="co">## P1                49 ACTTAAGAGGACTTTCTCTTAGCCCTTGTCATCTCAAAAGGCCTCTTAAG     98</span>
<span class="co">##                       |   | |  | || | |||       ||   | |      || ||||||</span>
<span class="co">## S1                51 GCGGTACA--AATTCCACTTTTGTACCGTGGCCGCCCGCCCCCCCTTAAG     98</span>
<span class="co">## </span>
<span class="co">## P1                99 ATGGATACCAACTGTATTTAGATAGAGACATTATACCAATGATCATCTCG    148</span>
<span class="co">##                      ||||||||||||||||||||||||||||||||||||||||||||||||||</span>
<span class="co">## S1                99 ATGGATACCAACTGTATTTAGATAGAGACATTATACCAATGATCATCTCG    148</span>
<span class="co">## </span>
<span class="co">## P1               149 CTTAAAATTTGTTCGAGTCATCAAATTCTTGAGTTATGGGGCTCAGAAAC    198</span>
<span class="co">##                      ||||||||||||||||||||||||||||||||||||||||||||||||||</span>
<span class="co">## S1               149 CTTAAAATTTGTTCGAGTCATCAAATTCTTGAGTTATGGGGCTCAGAAAC    198</span>
<span class="co">## </span>
<span class="co">## P1               199 AAC    201</span>
<span class="co">##                      |||</span>
<span class="co">## S1               199 AAC    201</span>
<span class="co">## </span>
<span class="co">## </span>
<span class="co">## #---------------------------------------</span>
<span class="co">## #---------------------------------------</span></code></pre>
<div class="sourceCode" id="cb152"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Both sides</span>
<span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/pairwiseAlignment.html" class="external-link">pairwiseAlignment</a></span><span class="op">(</span><span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/getSeq.html" class="external-link">getSeq</a></span><span class="op">(</span><span class="va">ROI10_ins_bri</span> <span class="op">+</span> <span class="fl">100</span><span class="op">)</span>,
                  <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/getSeq.html" class="external-link">getSeq</a></span><span class="op">(</span><span class="va">ROI10_ins_bri</span><span class="op">$</span><span class="va">query</span> <span class="op">+</span> <span class="fl">100</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/reverseComplement.html" class="external-link">reverseComplement</a></span><span class="op">(</span><span class="op">)</span>,
                  type <span class="op">=</span> <span class="st">"global"</span><span class="op">)</span> </code></pre></div>
<pre><code><span class="co">## Global PairwiseAlignmentsSingleSubject (1 of 1)</span>
<span class="co">## pattern: <span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">AAAA</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">TTTT</span><span style="color: #000000; background-color: #FFAFAF;">AAAA</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #AFFFAF;">CC</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">AAA</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #AFFFFF;">G</span>...<span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">AAA</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">GGGG</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">AAA</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #AFFFAF;">C</span></span>
<span class="co">## subject: -<span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">AAAA</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">TTTT</span><span style="color: #000000; background-color: #FFAFAF;">AAAA</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #AFFFAF;">CC</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">AAA</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #AFFFFF;">G</span>...<span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">AAA</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">GGGG</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">AAA</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">AA</span>-</span>
<span class="co">## score: -885.648</span></code></pre>
<div class="sourceCode" id="cb154"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/getSeq.html" class="external-link">getSeq</a></span><span class="op">(</span><span class="va">ROI10_ins_bri</span><span class="op">$</span><span class="va">query</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/r/base/character.html" class="external-link">as.character</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## [1] "ATCTTAAGGGGGGGCGGGCGGCCACGGTACAAAAGTGGAATTTGTACCGCGGGTACAAATTTTGTACCGCCGGTACAAAAAATCGCAATTGTTTTGACAATAAAAACCGATTTCTACCCATTATAATGCTCACTTTTGAACAGAACTCCACCTCAGCTATACCAGTAAGGAGTTTCCATCAATTCTAGCGCGAAGCGCACACTTTTTCCTTGATTCTATTTTTGTCTTGTACCGGCGGTACAAAAATTTTTGTACCGGCGTTACAAATTTGTACCGCCGTTACAAATTTGTACCGTCGCCGCCCTACACCC"</span></code></pre>
<p>Exemplar from the Okinawa genome (found by BLASTing the sequence output by <code>getSeq</code> above).</p>
<pre><code>&gt;RepeatInsertion
AAgggggggCGGGCGGCCACGGTACAAAAGTGGAATTTGTACCGCGGGTACAAATTTTGT
ACCGCCGGTACAAAAAATCGCAATTGTTTTGACAATAAAAACCGATTTCTACCCATTATA
ATGCTCACTTTTGAACAGAACTCCACCTCAGCTATACCAGTAAGGAGTTTCCATCAATTC
TAGCGCGAAGCGCACACTTTTTCCTTGATTCTATTTTTGTCTTGTACCGGCGGTACAAAA
ATTTTTGTACCGGCGTTACAAATTTGTACCGCCGTTACAAATTTGTACCGTCGCCGCCCT
ACACCC</code></pre>
</div>
</div>
<div class="section level2">
<h2 id="roi-11-highly-conserved-regions">ROI 11: highly conserved regions<a class="anchor" aria-label="anchor" href="#roi-11-highly-conserved-regions"></a>
</h2>
<p>Many mapped regions have a width shorter than 600 and form a distinct peak in width distribution plot (see <code><a href="../articles/RegionWidths.html">vignette("RegionWidths", package = "OikScrambling")</a></code> for details).</p>
<p>Surprisingly they appear to be very conserved.</p>
<div class="sourceCode" id="cb157"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="http://www.bioinf.jku.at/software/msa/" class="external-link">msa</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="http://yulab-smu.top/ggmsa/" class="external-link">ggmsa</a></span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Registered S3 methods overwritten by 'ggalt':</span>
<span class="co">##   method                  from   </span>
<span class="co">##   grid.draw.absoluteGrob  ggplot2</span>
<span class="co">##   grobHeight.absoluteGrob ggplot2</span>
<span class="co">##   grobWidth.absoluteGrob  ggplot2</span>
<span class="co">##   grobX.absoluteGrob      ggplot2</span>
<span class="co">##   grobY.absoluteGrob      ggplot2</span></code></pre>
<pre><code><span class="co">## ggmsa v1.1.4  Document: http://yulab-smu.top/ggmsa/</span>
<span class="co">## </span>
<span class="co">## If you use ggmsa in published research, please cite: DOI: 10.18129/B9.bioc.ggmsa</span></code></pre>
<div class="sourceCode" id="cb160"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">x</span> <span class="op">&lt;-</span> <span class="va">coa</span><span class="op">$</span><span class="va">Oki_Osa</span> |&gt; <span class="fu">plyranges</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/plyranges/man/tidyverse-reexports.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">width</span> <span class="op">&lt;</span> <span class="fl">600</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/sort.html" class="external-link">sort</a></span><span class="op">(</span>i<span class="op">=</span><span class="cn">T</span><span class="op">)</span>

<span class="va">roi</span> <span class="op">&lt;-</span> <span class="va">x</span><span class="op">[</span><span class="fl">4</span><span class="op">]</span>
<span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/getSeq.html" class="external-link">getSeq</a></span><span class="op">(</span><span class="va">roi</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## DNAStringSet object of length 1:</span>
<span class="co">##     width seq</span>
<span class="co">## [1]   222 <span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #AFFFFF;">GG</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #FFAFAF;">AAAA</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">TTT</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">GG</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #FFAFAF;">AAA</span><span style="color: #000000; background-color: #AFFFAF;">CC</span>...<span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">TTTTT</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">T</span></span></code></pre>
<div class="sourceCode" id="cb162"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/pairwiseAlignment.html" class="external-link">pairwiseAlignment</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/getSeq.html" class="external-link">getSeq</a></span><span class="op">(</span><span class="va">roi</span> <span class="op">+</span> <span class="fl">100</span><span class="op">)</span>,
                  <span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/getSeq.html" class="external-link">getSeq</a></span><span class="op">(</span><span class="va">roi</span><span class="op">$</span><span class="va">query</span> <span class="op">+</span> <span class="fl">100</span><span class="op">)</span>,
                  type <span class="op">=</span> <span class="st">"global"</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/PairwiseAlignments-io.html" class="external-link">writePairwiseAlignments</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## ########################################</span>
<span class="co">## # Program: Biostrings (version 2.62.0), a Bioconductor package</span>
<span class="co">## # Rundate: Wed Mar 30 19:17:47 2022</span>
<span class="co">## ########################################</span>
<span class="co">## #=======================================</span>
<span class="co">## #</span>
<span class="co">## # Aligned_sequences: 2</span>
<span class="co">## # 1: P1</span>
<span class="co">## # 2: S1</span>
<span class="co">## # Matrix: NA</span>
<span class="co">## # Gap_penalty: 14.0</span>
<span class="co">## # Extend_penalty: 4.0</span>
<span class="co">## #</span>
<span class="co">## # Length: 441</span>
<span class="co">## # Identity:     261/441 (59.2%)</span>
<span class="co">## # Similarity:    NA/441 (NA%)</span>
<span class="co">## # Gaps:          47/441 (10.7%)</span>
<span class="co">## # Score: -695.3687</span>
<span class="co">## #</span>
<span class="co">## #</span>
<span class="co">## #=======================================</span>
<span class="co">## </span>
<span class="co">## P1                 1 AGATTTCAAGACGCCTGCCCTTTTTCTTTGCCTAAAATGACCTCGAATTT     50</span>
<span class="co">##                       |  |  | || |  |     || || | |   | |||  | | |||| |</span>
<span class="co">## S1                 1 -GCGTAAATGAAGA-TCAAAGTTGTCATAGG--ACAATT-CTTAGAATAT     45</span>
<span class="co">## </span>
<span class="co">## P1                51 --AGTCTTTGAA---CCAATGTTATAAAAAAGCTAGA-AAGACAGAGAAA     94</span>
<span class="co">##                        | | ||||||   |   |||| | | ||| ||| | || | |  | | </span>
<span class="co">## S1                46 CCAATTTTTGAATAGCTCCTGTTTTTATAAATCTATAGAAAAAATGGTAC     95</span>
<span class="co">## </span>
<span class="co">## P1                95 GAGAGCCGGTCTTAAAATGTTTAGAGAAGTGGCTTAAACCCGGAGTATCA    144</span>
<span class="co">##                      ||  || ||||||||| | ||||||  | | |  ||| || |||||||||</span>
<span class="co">## S1                96 GATCGC-GGTCTTAAATTTTTTAGA--ACTCGACTAATCCTGGAGTATCA    142</span>
<span class="co">## </span>
<span class="co">## P1               145 CGTTGAACATTCTACACCGGTAGAGATAACTTTGAGAAATGATTTGAACA    194</span>
<span class="co">##                      |||| || || || |    || ||||||| ||||  |||||||| |||| </span>
<span class="co">## S1               143 CGTTAAAAATCCTGC----GT-GAGATAAGTTTGCAAAATGATTGGAACC    187</span>
<span class="co">## </span>
<span class="co">## P1               195 CACGAACACCTTGAAACTACTCCGGTTTCAGCTGTGAGCTAATTTCCGTT    244</span>
<span class="co">##                       |    |||||||   |||||  ||||||||||| ||||| |||| ||||</span>
<span class="co">## S1               188 TAGTTGCACCTTGTTGCTACTTTGGTTTCAGCTG-GAGCTTATTTGCGTT    236</span>
<span class="co">## </span>
<span class="co">## P1               245 TTTTCTCACGCCACTTTTTT---CAAGATCTAAAAAACAAAAATTAAAAA    291</span>
<span class="co">##                      ||     |    | |||      ||||||  | || |  ||  || |  |</span>
<span class="co">## S1               237 TTA----AAATTAATTTCCCAGACAAGATGCAGAACAGCAATTTTCAGTA    282</span>
<span class="co">## </span>
<span class="co">## P1               292 GCTGTTTTTCGAGCGCTTGAGATAATCAAGTCCCTGTTCCAGCGACTTTC    341</span>
<span class="co">##                       ||| ||||   || ||| ||||||||||||   | ||| ||  |||  |</span>
<span class="co">## S1               283 ACTGCTTTTTCTGCACTTCAGATAATCAAGTA--TTTTCTAGT-ACT--C    327</span>
<span class="co">## </span>
<span class="co">## P1               342 AAATC-AAG-CTGAAAT--TGAACGAAAAAAAACGAAAAAAGAGCCTTTG    387</span>
<span class="co">##                      |  || ||| || | ||  |||  ||  || |  |||  |   || |  |</span>
<span class="co">## S1               328 AGTTCTAAGACTCACATCTTGATAGAGCAACACTGAAGCA---GCATCAG    374</span>
<span class="co">## </span>
<span class="co">## P1               388 TCCGAAAAATAAAGA-TCAAAAAGAAAATCCTCGCT-----    422</span>
<span class="co">##                      |  || |   || || |  |      |||  |   |     </span>
<span class="co">## S1               375 T--GATATTGAACGACTTGAGTTTTCAATGTTTTGTGAAAA    413</span>
<span class="co">## </span>
<span class="co">## </span>
<span class="co">## #---------------------------------------</span>
<span class="co">## #---------------------------------------</span></code></pre>
<div class="sourceCode" id="cb164"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">roi</span> <span class="op">&lt;-</span> <span class="va">x</span><span class="op">[</span><span class="fl">10</span><span class="op">]</span>
<span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/getSeq.html" class="external-link">getSeq</a></span><span class="op">(</span><span class="va">roi</span><span class="op">)</span>       |&gt; <span class="fu"><a href="https://rdrr.io/r/base/character.html" class="external-link">as.character</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## [1] "ATTGTTTAGTCTGGGGTTTACAGAAGAGTTGTTTCTTCATGGAACCGGGCTGCTGTAAATCAGAGGACTCGGTCTCGGCCGAGCGCGCGCTGACCTCTAAAGCGGCGGCCCTAGCGCGCATCGTCAAGCGCAAATTTTTCAGAATTGAGCTTCCAAAATCAAACACCAGAAACGCCTTTCGCCGAAAAATCAAGGTTCCATTGAGT"</span></code></pre>
<div class="sourceCode" id="cb166"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/getSeq.html" class="external-link">getSeq</a></span><span class="op">(</span><span class="va">roi</span><span class="op">$</span><span class="va">query</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/r/base/character.html" class="external-link">as.character</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## [1] "ATTGTTTAGTCTGCTGTCACAGTAGAGTTGTTGCTTCATGGAACCAGCCCGCAGTAAATCAGAAGAGTCAGTCGCGGACGAGTGCGCGCTGACCTCTAAAGCGGCGGCCCCAACGCGCATCGTCAAGCGCAACATTTTCAGAATTGAGCTTCTGAAATCAAAAAGCAGAAACAATTTCGCCGAATAATCACTCAATTGAAAGACGGTTCCATTGAGT"</span></code></pre>
<div class="sourceCode" id="cb168"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">seqs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span>
  <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/findOverlaps-methods.html" class="external-link">subsetByOverlaps</a></span><span class="op">(</span><span class="va">coa</span><span class="op">$</span><span class="va">Oki_Osa</span>, <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/genomic-range-squeezers.html" class="external-link">granges</a></span><span class="op">(</span><span class="va">roi</span><span class="op">)</span><span class="op">)</span>                             |&gt; <span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/getSeq.html" class="external-link">getSeq</a></span><span class="op">(</span><span class="op">)</span>,
  <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/findOverlaps-methods.html" class="external-link">subsetByOverlaps</a></span><span class="op">(</span><span class="va">coa</span><span class="op">$</span><span class="va">Oki_Osa</span>, <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/genomic-range-squeezers.html" class="external-link">granges</a></span><span class="op">(</span><span class="va">roi</span><span class="op">)</span><span class="op">)</span><span class="op">$</span><span class="va">query</span>                       |&gt; <span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/getSeq.html" class="external-link">getSeq</a></span><span class="op">(</span><span class="op">)</span>,
  <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/findOverlaps-methods.html" class="external-link">subsetByOverlaps</a></span><span class="op">(</span><span class="va">coa</span><span class="op">$</span><span class="va">Oki_Aom</span>, <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/genomic-range-squeezers.html" class="external-link">granges</a></span><span class="op">(</span><span class="va">roi</span><span class="op">)</span>, ignore.strand <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">$</span><span class="va">query</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/getSeq.html" class="external-link">getSeq</a></span><span class="op">(</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/reverseComplement.html" class="external-link">reverseComplement</a></span><span class="op">(</span><span class="op">)</span>,
  <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/findOverlaps-methods.html" class="external-link">subsetByOverlaps</a></span><span class="op">(</span><span class="va">coa</span><span class="op">$</span><span class="va">Oki_Bar</span>, <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/genomic-range-squeezers.html" class="external-link">granges</a></span><span class="op">(</span><span class="va">roi</span><span class="op">)</span><span class="op">)</span><span class="op">$</span><span class="va">query</span>                       |&gt; <span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/getSeq.html" class="external-link">getSeq</a></span><span class="op">(</span><span class="op">)</span>,
  <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/findOverlaps-methods.html" class="external-link">subsetByOverlaps</a></span><span class="op">(</span><span class="va">coa</span><span class="op">$</span><span class="va">Oki_Nor</span>, <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/genomic-range-squeezers.html" class="external-link">granges</a></span><span class="op">(</span><span class="va">roi</span><span class="op">)</span><span class="op">)</span><span class="op">$</span><span class="va">query</span>                       |&gt; <span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/getSeq.html" class="external-link">getSeq</a></span><span class="op">(</span><span class="op">)</span>
<span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">seqs</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Oki"</span>, <span class="st">"Osa"</span>, <span class="st">"Aom"</span>, <span class="st">"Bar"</span>, <span class="st">"Nor"</span><span class="op">)</span>

<span class="op">(</span><span class="va">seqs_aln</span> <span class="op">&lt;-</span> <span class="fu">msa</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/msa/man/msaMuscle.html" class="external-link">msaMuscle</a></span><span class="op">(</span><span class="va">seqs</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/r/methods/as.html" class="external-link">as</a></span><span class="op">(</span><span class="st">"DNAMultipleAlignment"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## DNAMultipleAlignment with 5 rows and 230 columns</span>
<span class="co">##      aln                                                    names               </span>
<span class="co">## [1] -----<span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">TTT</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">GGGG</span><span style="color: #000000; background-color: #FFD7AF;">TTT</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">C</span>...------------<span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #AFFFAF;">CC</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">T</span>- Oki</span>
<span class="co">## [2] -----<span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">TTT</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">G</span>-<span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">C</span>...<span style="color: #000000; background-color: #FFAFAF;">AAA</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #AFFFFF;">G</span>-----<span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #AFFFAF;">CC</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">T</span>- Osa</span>
<span class="co">## [3] -----<span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">TTT</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">G</span>-<span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">C</span>...<span style="color: #000000; background-color: #FFAFAF;">AAA</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #AFFFAF;">CC</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">T</span>- Aom</span>
<span class="co">## [4] ---<span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">TTT</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span>-<span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">C</span>...<span style="color: #000000; background-color: #AFFFAF;">C</span>-----------<span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #AFFFAF;">CC</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">G</span> Bar</span>
<span class="co">## [5] <span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">TTT</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span>-<span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">C</span>...<span style="color: #000000; background-color: #AFFFFF;">G</span>-----------<span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #AFFFAF;">CC</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">G</span> Nor</span></code></pre>
<div class="sourceCode" id="cb170"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#from Biostrings:::.write.MultAlign</span>
<span class="va">showMultiAlign</span> <span class="op">&lt;-</span> <span class="kw">function</span> <span class="op">(</span><span class="va">x</span>, <span class="va">invertColMask</span> <span class="op">=</span> <span class="cn">F</span>, <span class="va">showRowNames</span> <span class="op">=</span> <span class="cn">T</span>, <span class="va">hideMaskedCols</span> <span class="op">=</span> <span class="cn">T</span><span class="op">)</span> 
<span class="op">{</span>
        <span class="va">msk</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/MultipleAlignment-class.html" class="external-link">colmask</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span>
        <span class="va">dims</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span>
        <span class="kw">if</span> <span class="op">(</span><span class="va">invertColMask</span> <span class="op">==</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">{</span>
            <span class="va">msk</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/inter-range-methods.html" class="external-link">gaps</a></span><span class="op">(</span><span class="va">msk</span>, start <span class="op">=</span> <span class="fl">1</span>, end <span class="op">=</span> <span class="va">dims</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span><span class="op">)</span>
        <span class="op">}</span>
        <span class="kw">if</span> <span class="op">(</span><span class="va">hideMaskedCols</span><span class="op">)</span> <span class="op">{</span>
            <span class="va">hasMask</span> <span class="op">&lt;-</span> <span class="cn">FALSE</span>
        <span class="op">}</span>
        <span class="kw">else</span> <span class="op">{</span>
            <span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/MultipleAlignment-class.html" class="external-link">colmask</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="cn">NULL</span>
            <span class="kw">if</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">msk</span><span class="op">)</span> <span class="op">&gt;</span> <span class="fl">0</span><span class="op">)</span> <span class="op">{</span>
                <span class="va">hasMask</span> <span class="op">&lt;-</span> <span class="cn">TRUE</span>
            <span class="op">}</span>
            <span class="kw">else</span> <span class="op">{</span>
                <span class="va">hasMask</span> <span class="op">&lt;-</span> <span class="cn">FALSE</span>
            <span class="op">}</span>
        <span class="op">}</span>
        <span class="kw">if</span> <span class="op">(</span><span class="va">hasMask</span><span class="op">)</span> <span class="op">{</span>
            <span class="va">dims</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="va">dims</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span> <span class="op">+</span> <span class="fl">1</span>
        <span class="op">}</span>
        <span class="va">ch</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/character.html" class="external-link">as.character</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span>
        <span class="va">ch</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/unlist.html" class="external-link">unlist</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">ch</span>, <span class="fu">Biostrings</span><span class="fu">:::</span><span class="va">.insertSpaces</span><span class="op">)</span><span class="op">)</span>
        <span class="kw">if</span> <span class="op">(</span><span class="va">hasMask</span><span class="op">)</span> <span class="op">{</span>
            <span class="va">mskInd</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/integer.html" class="external-link">as.integer</a></span><span class="op">(</span><span class="va">msk</span><span class="op">)</span>
            <span class="va">mskCh</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/character.html" class="external-link">as.character</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/replace.html" class="external-link">replace</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">rep</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span><span class="op">)</span>, 
                <span class="va">mskInd</span>, <span class="fl">0</span><span class="op">)</span><span class="op">)</span>, collapse <span class="op">=</span> <span class="st">""</span><span class="op">)</span>
            <span class="va">mskCh</span> <span class="op">&lt;-</span> <span class="fu">Biostrings</span><span class="fu">:::</span><span class="fu">.insertSpaces</span><span class="op">(</span><span class="va">mskCh</span><span class="op">)</span>
        <span class="op">}</span>
        <span class="va">names</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">ch</span><span class="op">)</span>
        <span class="va">ch</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">ch</span>, <span class="fu">Biostrings</span><span class="fu">:::</span><span class="va">.strChop</span>, chopsize <span class="op">=</span> <span class="fl">55</span>, simplify <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
        <span class="kw">if</span> <span class="op">(</span><span class="va">hasMask</span><span class="op">)</span> <span class="op">{</span>
            <span class="va">mskCh</span> <span class="op">&lt;-</span> <span class="fu">.strChop</span><span class="op">(</span><span class="va">mskCh</span>, chopsize <span class="op">=</span> <span class="fl">55</span><span class="op">)</span>
            <span class="va">ch</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>Mask <span class="op">=</span> <span class="va">mskCh</span><span class="op">)</span>, <span class="va">ch</span><span class="op">)</span>
        <span class="op">}</span>
        <span class="va">maxLen</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html" class="external-link">max</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/nchar.html" class="external-link">nchar</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">ch</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
        <span class="va">stockSpc</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">" "</span>, <span class="va">maxLen</span><span class="op">)</span>, collapse <span class="op">=</span> <span class="st">""</span><span class="op">)</span>
        <span class="va">bufferSpacing</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">name</span><span class="op">)</span> <span class="op">{</span>
            <span class="va">spc</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">" "</span>, <span class="va">maxLen</span> <span class="op">-</span> <span class="fu"><a href="https://rdrr.io/r/base/nchar.html" class="external-link">nchar</a></span><span class="op">(</span><span class="va">name</span><span class="op">)</span><span class="op">)</span>, collapse <span class="op">=</span> <span class="st">""</span><span class="op">)</span>
            <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="va">name</span>, <span class="va">spc</span>, sep <span class="op">=</span> <span class="st">""</span><span class="op">)</span>
        <span class="op">}</span>
        <span class="va">ch</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="va">ch</span>, <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">""</span>, <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">ch</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
        <span class="va">output</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/character.html" class="external-link">character</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">ch</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">)</span> <span class="op">*</span> <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">ch</span><span class="op">)</span><span class="op">)</span>
        <span class="kw">for</span> <span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq_len</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">ch</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span>
            <span class="kw">for</span> <span class="op">(</span><span class="va">j</span> <span class="kw">in</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq_len</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">ch</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span>
                <span class="kw">if</span> <span class="op">(</span><span class="va">i</span> <span class="op">==</span> <span class="fl">1</span><span class="op">)</span> <span class="op">{</span>
                  <span class="va">output</span><span class="op">[</span><span class="va">j</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/unlist.html" class="external-link">unlist</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">ch</span><span class="op">[</span><span class="va">j</span><span class="op">]</span><span class="op">)</span>, 
                    <span class="va">bufferSpacing</span><span class="op">)</span><span class="op">)</span>, <span class="st">"   "</span>, <span class="va">ch</span><span class="op">[[</span><span class="va">j</span><span class="op">]</span><span class="op">]</span><span class="op">[</span><span class="va">i</span><span class="op">]</span>, sep <span class="op">=</span> <span class="st">""</span><span class="op">)</span>
                <span class="op">}</span>
                <span class="kw">else</span> <span class="op">{</span>
                  <span class="kw">if</span> <span class="op">(</span><span class="va">showRowNames</span><span class="op">)</span> <span class="op">{</span>
                    <span class="va">output</span><span class="op">[</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">ch</span><span class="op">)</span> <span class="op">*</span> <span class="op">(</span><span class="va">i</span> <span class="op">-</span> <span class="fl">1</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="va">j</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/unlist.html" class="external-link">unlist</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">ch</span><span class="op">[</span><span class="va">j</span><span class="op">]</span><span class="op">)</span>, 
                      <span class="va">bufferSpacing</span><span class="op">)</span><span class="op">)</span>, <span class="st">"   "</span>, <span class="va">ch</span><span class="op">[[</span><span class="va">j</span><span class="op">]</span><span class="op">]</span><span class="op">[</span><span class="va">i</span><span class="op">]</span>, sep <span class="op">=</span> <span class="st">""</span><span class="op">)</span>
                  <span class="op">}</span>
                  <span class="kw">else</span> <span class="op">{</span>
                    <span class="va">output</span><span class="op">[</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">ch</span><span class="op">)</span> <span class="op">*</span> <span class="op">(</span><span class="va">i</span> <span class="op">-</span> <span class="fl">1</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="va">j</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="va">stockSpc</span>, 
                      <span class="st">"   "</span>, <span class="va">ch</span><span class="op">[[</span><span class="va">j</span><span class="op">]</span><span class="op">]</span><span class="op">[</span><span class="va">i</span><span class="op">]</span>, sep <span class="op">=</span> <span class="st">""</span><span class="op">)</span>
                  <span class="op">}</span>
                <span class="op">}</span>
            <span class="op">}</span>
        <span class="op">}</span>
        <span class="va">output</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">gsub</a></span><span class="op">(</span><span class="st">"\\s+$"</span>, <span class="st">""</span>, <span class="va">output</span><span class="op">)</span>
        <span class="va">output</span> <span class="op">&lt;-</span> <span class="va">output</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">output</span><span class="op">)</span> <span class="op">-</span> <span class="fl">1</span><span class="op">]</span>
        <span class="kw">if</span> <span class="op">(</span><span class="va">hasMask</span><span class="op">)</span> <span class="op">{</span>
            <span class="va">output</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="st">""</span>, <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="va">dims</span>, <span class="st">""</span><span class="op">)</span>, collapse <span class="op">=</span> <span class="st">" "</span><span class="op">)</span>, 
                collapse <span class="op">=</span> <span class="st">" "</span><span class="op">)</span>, <span class="va">output</span><span class="op">)</span>
        <span class="op">}</span>
        <span class="kw">else</span> <span class="op">{</span>
            <span class="va">output</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="st">""</span>, <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="va">dims</span>, collapse <span class="op">=</span> <span class="st">" "</span><span class="op">)</span>, 
                collapse <span class="op">=</span> <span class="st">" "</span><span class="op">)</span>, <span class="va">output</span><span class="op">)</span>
        <span class="op">}</span>
        <span class="va">output</span>
<span class="op">}</span>
<span class="va">seqs_aln</span> |&gt; <span class="fu">showMultiAlign</span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">##  [1] " 5 230"                                                      </span>
<span class="co">##  [2] "Oki   -----ATTGT TTAGTCTGGG GTTTACAGAA GAGTTGTTTC TTCATGGAAC"</span>
<span class="co">##  [3] "Osa   -----ATTGT TTAGTCTGCT G-TCACAGTA GAGTTGTTGC TTCATGGAAC"</span>
<span class="co">##  [4] "Aom   -----ATTGT TTAGTCTGCT G-TCACAGTA GAGTTGTTGC TTCATGGAAC"</span>
<span class="co">##  [5] "Bar   ---TTATTGT TTAGTCTGCA G-TCACAGTA GAGTTGTTGC TTCATGGAAC"</span>
<span class="co">##  [6] "Nor   AGATTATTGT TTAGTCTGCA G-TCACAGTA GAGTTGTTGC TTCATGGAAC"</span>
<span class="co">##  [7] ""                                                            </span>
<span class="co">##  [8] "Oki   CGGGCTGCTG TAAATCAGAG GACTCGGTCT CGGCCGAGCG CGCGCTGACC"</span>
<span class="co">##  [9] "Osa   CAGCCCGCAG TAAATCAGAA GAGTCAGTCG CGGACGAGTG CGCGCTGACC"</span>
<span class="co">## [10] "Aom   CAGCCCGCAG TAAATCAGAA GAGTCAGTCG CGGACGAGTG CGCGCTGACC"</span>
<span class="co">## [11] "Bar   CAGCCCGCAG TAAATCAGAA GACTAGGTCG CGGGCGACTG CGTGCTGACC"</span>
<span class="co">## [12] "Nor   CAGCCCGCAG TAAATCAGAA GACTCGGTCG CGGGCGACTG CGTGCTGACC"</span>
<span class="co">## [13] ""                                                            </span>
<span class="co">## [14] "Oki   TCTAAAGCGG CGGCCCTAGC GCGCATCGTC AAGCGCAAAT TTTTCAGAAT"</span>
<span class="co">## [15] "Osa   TCTAAAGCGG CGGCCCCAAC GCGCATCGTC AAGCGCAACA TTTTCAGAAT"</span>
<span class="co">## [16] "Aom   TCTAAAGCGG CGGCCCCAAC GCGCATCGTC AAGCGCAACA TTTTCAGAAT"</span>
<span class="co">## [17] "Bar   TCTAAAGCGG CGGCCTCAGC GCGCACCGTC GAGCGCAACA TTTTCAGAAT"</span>
<span class="co">## [18] "Nor   TCTAAAGCGG CGGCCTCAGC GCGCACCGTC AAGCGCAACA TTTTCAGAAT"</span>
<span class="co">## [19] ""                                                            </span>
<span class="co">## [20] "Oki   TGAGCTTCCA AAATCAAACA CCAGAAACGC CTTTCGCCGA AAAATCAAG-"</span>
<span class="co">## [21] "Osa   TGAGCTTCTG AAATCAAAAA GCAGAAACA- ATTTCGCCGA ATAATCACTC"</span>
<span class="co">## [22] "Aom   TGAGCTTCTG AAATCAAAAA GCAGAAACA- AATTCGCCGA ATAATCACTC"</span>
<span class="co">## [23] "Bar   TGAGCTTCTG AAATCAAAAA GCAGAAACA- TTTTCGCCGA ATAATCACTG"</span>
<span class="co">## [24] "Nor   TGAGCTTCTG AAATCAAAAA GCAGAAACA- TTTTCGCCGA ATAATCACTG"</span>
<span class="co">## [25] ""                                                            </span>
<span class="co">## [26] "Oki   ---------- -------GTT CCATTGAGT-"                      </span>
<span class="co">## [27] "Osa   AATTGAAAGA CG-----GTT CCATTGAGT-"                      </span>
<span class="co">## [28] "Aom   AATTGAAAGA CGCGATTGTT CCATTGAGT-"                      </span>
<span class="co">## [29] "Bar   AATCGC---- -------GTT CCATCGAATG"                      </span>
<span class="co">## [30] "Nor   AATCGG---- -------GTT CCATCGAATG"</span></code></pre>
<div class="sourceCode" id="cb172"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">roi</span> <span class="op">&lt;-</span> <span class="va">x</span><span class="op">[</span><span class="fl">11</span><span class="op">]</span>
<span class="va">seqs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span>
  <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/findOverlaps-methods.html" class="external-link">subsetByOverlaps</a></span><span class="op">(</span><span class="va">coa</span><span class="op">$</span><span class="va">Oki_Osa</span>, <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/genomic-range-squeezers.html" class="external-link">granges</a></span><span class="op">(</span><span class="va">roi</span><span class="op">)</span><span class="op">)</span>                             |&gt; <span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/getSeq.html" class="external-link">getSeq</a></span><span class="op">(</span><span class="op">)</span>,
  <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/findOverlaps-methods.html" class="external-link">subsetByOverlaps</a></span><span class="op">(</span><span class="va">coa</span><span class="op">$</span><span class="va">Oki_Osa</span>, <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/genomic-range-squeezers.html" class="external-link">granges</a></span><span class="op">(</span><span class="va">roi</span><span class="op">)</span><span class="op">)</span><span class="op">$</span><span class="va">query</span>                       |&gt; <span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/getSeq.html" class="external-link">getSeq</a></span><span class="op">(</span><span class="op">)</span>,
  <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/findOverlaps-methods.html" class="external-link">subsetByOverlaps</a></span><span class="op">(</span><span class="va">coa</span><span class="op">$</span><span class="va">Oki_Aom</span>, <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/genomic-range-squeezers.html" class="external-link">granges</a></span><span class="op">(</span><span class="va">roi</span><span class="op">)</span>, ignore.strand <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">$</span><span class="va">query</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/getSeq.html" class="external-link">getSeq</a></span><span class="op">(</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/reverseComplement.html" class="external-link">reverseComplement</a></span><span class="op">(</span><span class="op">)</span>,
  <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/findOverlaps-methods.html" class="external-link">subsetByOverlaps</a></span><span class="op">(</span><span class="va">coa</span><span class="op">$</span><span class="va">Oki_Bar</span>, <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/genomic-range-squeezers.html" class="external-link">granges</a></span><span class="op">(</span><span class="va">roi</span><span class="op">)</span><span class="op">)</span><span class="op">$</span><span class="va">query</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/intra-range-methods.html" class="external-link">resize</a></span><span class="op">(</span><span class="fl">300</span>, fix<span class="op">=</span><span class="st">"start"</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/getSeq.html" class="external-link">getSeq</a></span><span class="op">(</span><span class="op">)</span>,
  <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/findOverlaps-methods.html" class="external-link">subsetByOverlaps</a></span><span class="op">(</span><span class="va">coa</span><span class="op">$</span><span class="va">Oki_Nor</span>, <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/genomic-range-squeezers.html" class="external-link">granges</a></span><span class="op">(</span><span class="va">roi</span><span class="op">)</span><span class="op">)</span><span class="op">$</span><span class="va">query</span>                       |&gt; <span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/getSeq.html" class="external-link">getSeq</a></span><span class="op">(</span><span class="op">)</span>
<span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">seqs</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Oki"</span>, <span class="st">"Osa"</span>, <span class="st">"Aom"</span>, <span class="st">"Bar"</span>, <span class="st">"Nor"</span><span class="op">)</span>

<span class="op">(</span><span class="va">seqs_aln</span> <span class="op">&lt;-</span> <span class="fu">msa</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/msa/man/msaMuscle.html" class="external-link">msaMuscle</a></span><span class="op">(</span><span class="va">seqs</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/r/methods/as.html" class="external-link">as</a></span><span class="op">(</span><span class="st">"DNAMultipleAlignment"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## DNAMultipleAlignment with 5 rows and 311 columns</span>
<span class="co">##      aln                                                    names               </span>
<span class="co">## [1] <span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #AFFFFF;">GGGG</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #AFFFFF;">GG</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">G</span>...------------------------- Oki</span>
<span class="co">## [2] --------<span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #AFFFFF;">GGGG</span>...<span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">TTTTT</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">TTTT</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">C</span> Bar</span>
<span class="co">## [3] --------<span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #AFFFFF;">GGGG</span>...------------------------- Nor</span>
<span class="co">## [4] <span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #AFFFFF;">GG</span>-<span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #AFFFFF;">GG</span>-<span style="color: #000000; background-color: #AFFFFF;">G</span>...------------------------- Osa</span>
<span class="co">## [5] <span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #AFFFFF;">GG</span>-<span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #AFFFFF;">GGGG</span>...<span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">TTTT</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">A</span>--------- Aom</span></code></pre>
<div class="sourceCode" id="cb174"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">seqs_aln</span> |&gt; <span class="fu">showMultiAlign</span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">##  [1] " 5 311"                                                      </span>
<span class="co">##  [2] "Oki   GATAAGGGGC GCTCATAATA TCGGTGTTCT ATTTGTCGAC TCTAAACGGA"</span>
<span class="co">##  [3] "Bar   --------GC GATCATAATA TCGGGGCGTT ATTTGGCGAT TCTAAACGGA"</span>
<span class="co">##  [4] "Nor   --------GC GATCATAATA TCGGGGCGTT ATTTGGCGAT TCTAAACGGA"</span>
<span class="co">##  [5] "Osa   GATAAGCGG- GCTCATAATA TCGG-GCGTT ATTTGGCGAT TCTAAACGGA"</span>
<span class="co">##  [6] "Aom   GATAAGCGG- GCTCATAATA TCGGGGCGTT ATTTGGCGAT TCTAAACGGA"</span>
<span class="co">##  [7] ""                                                            </span>
<span class="co">##  [8] "Oki   GCCTGTTTAG GCAACGCAAT GTTCCACGAA GTCGGCGCTG ACACGGTGAG"</span>
<span class="co">##  [9] "Bar   GCCTGTTTAG GCAACGCAAT GTTCCACGAA GTCGGCGCTG ACACGGGTGC"</span>
<span class="co">## [10] "Nor   GCCTGTTTAG GCAACGCAAT GTTCCACGAA GTCGGCGCTG ACACGGGTGC"</span>
<span class="co">## [11] "Osa   GCCTGTTTAG GCAACGCAAT GTTCCACGAA GTCGGCGCTG ACACGGGTGC"</span>
<span class="co">## [12] "Aom   GCCTGTTTAG GCAACGCAAT GTTCCACGAA GTCGGCGCTG ACACGGGTGC"</span>
<span class="co">## [13] ""                                                            </span>
<span class="co">## [14] "Oki   CGGCAACGTG AGGCCGCGCA ACTCGGCCCC GGTCTCGTCG GTCTGGTGCA"</span>
<span class="co">## [15] "Bar   TCGCAACGTG AGGCCGCGCA ACTCGGTCT- GGTGGTG-CA GTACGCGGCG"</span>
<span class="co">## [16] "Nor   TCGCAACGTG AGGCCGCGCA ACTCGGTCT- GGTGGTG-CA GTACGCGGCG"</span>
<span class="co">## [17] "Osa   TGGCAACGTG AGGCCGCGCA ACTCGGTCT- GGTGGTG-CA GTGCGCCCCG"</span>
<span class="co">## [18] "Aom   TGGCAACGTG AGGCCGCGCA ACTCGGTCT- GGTGGTG-CA GTGCGCCCCG"</span>
<span class="co">## [19] ""                                                            </span>
<span class="co">## [20] "Oki   GTGTGGTGCT CAACTTAGAC GCGCAAAACA CACGTTAATT ACCACACTGC"</span>
<span class="co">## [21] "Bar   GTGTGGTGCT CAACTTAGAC GCGCAAAACA CACGTTAATT ACCACACTGC"</span>
<span class="co">## [22] "Nor   GTGTGGTGCT CAACTTAGAC GCGCAAAACA CACGTTAATT ACCACACTGC"</span>
<span class="co">## [23] "Osa   GTCTGGTGCT CAACTTAGAC GCGCAAAACA CACGTTAATT ACCACACTGC"</span>
<span class="co">## [24] "Aom   GTCTGGTGCT CAACTTAGAC GCGCAAAACA CACGTTAATT ACCACACTGC"</span>
<span class="co">## [25] ""                                                            </span>
<span class="co">## [26] "Oki   ACCACCACCA TTGCCAGGAC ACTTCCTGGA GCTAAATCCT CGACACCAGA"</span>
<span class="co">## [27] "Bar   ACCACCACCA TTGCCAGGAC GCTTCCTGGA GCTAAATCCT CGACACGAGA"</span>
<span class="co">## [28] "Nor   ACCACCACCA TTGCCAGGAC GCTTCCTGGA GCTAAATCCT CGACACGAGA"</span>
<span class="co">## [29] "Osa   ACCACCACCA TTGCCAGGAC GCTTCCTGGA GCTAAATCCT CGACACGAGA"</span>
<span class="co">## [30] "Aom   ACCACCACCA TTGCCAGGAC GCTTCCTGGA GCTAAATCCT CGACACGAGA"</span>
<span class="co">## [31] ""                                                            </span>
<span class="co">## [32] "Oki   GCACCTTGCA ATAATTCGCG CAAAATGGAA C--------- ----------"</span>
<span class="co">## [33] "Bar   GCGCCT-GCA ATAATTGTTG CGGGATTGAA CACGGCCTAA GCGCATTTTT"</span>
<span class="co">## [34] "Nor   GCGCCT-GCA ATAATTGTTG CGGGATTGAA C--------- ----------"</span>
<span class="co">## [35] "Osa   GCGCCT-GCA ATAATTGTTG CGGGATGGAA C--------- ----------"</span>
<span class="co">## [36] "Aom   GCGCCT-GCA ATAATTGTTG CGGGATGGAA CACGCACTAA GCGCACTTTT"</span>
<span class="co">## [37] ""                                                            </span>
<span class="co">## [38] "Oki   ---------- -"                                          </span>
<span class="co">## [39] "Bar   GCACTTTTCA C"                                          </span>
<span class="co">## [40] "Nor   ---------- -"                                          </span>
<span class="co">## [41] "Osa   ---------- -"                                          </span>
<span class="co">## [42] "Aom   CA-------- -"</span></code></pre>
<div class="section level3">
<h3 id="autocorrelations">Autocorrelations<a class="anchor" aria-label="anchor" href="#autocorrelations"></a>
</h3>
<p>There appears to be a small tendency that short width regions are followed by short width regions, but I am not sure if I understand the statistics well.</p>
<div class="sourceCode" id="cb176"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="http://yulab-smu.top/ggmsa/reference/plot-methods.html" class="external-link">plot</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/start.html" class="external-link">width</a></span><span class="op">(</span><span class="va">coa</span><span class="op">$</span><span class="va">Oki_Osa</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">tail</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/start.html" class="external-link">width</a></span><span class="op">(</span><span class="va">coa</span><span class="op">$</span><span class="va">Oki_Osa</span><span class="op">)</span>, <span class="op">-</span><span class="fl">1</span><span class="op">)</span>, <span class="cn">NA</span><span class="op">)</span>, log <span class="op">=</span> <span class="st">'xy'</span>, pch<span class="op">=</span><span class="st">'.'</span><span class="op">)</span></code></pre></div>
<p><img src="RegionsOfInterest_files/figure-html/autocorrelations-1.png" width="700"></p>
<div class="sourceCode" id="cb177"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/stats/acf.html" class="external-link">acf</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/start.html" class="external-link">width</a></span><span class="op">(</span><span class="va">coa</span><span class="op">$</span><span class="va">Oki_Osa</span><span class="op">)</span>, lag.max <span class="op">=</span> <span class="fl">1000</span><span class="op">)</span></code></pre></div>
<p><img src="RegionsOfInterest_files/figure-html/autocorrelations-2.png" width="700"></p>
<div class="sourceCode" id="cb178"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/stats/acf.html" class="external-link">acf</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/AnnotationDbi/man/Bimap-envirAPI.html" class="external-link">sample</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/start.html" class="external-link">width</a></span><span class="op">(</span><span class="va">coa</span><span class="op">$</span><span class="va">Oki_Osa</span><span class="op">)</span><span class="op">)</span>, lag.max <span class="op">=</span> <span class="fl">1000</span><span class="op">)</span></code></pre></div>
<p><img src="RegionsOfInterest_files/figure-html/autocorrelations-3.png" width="700"></p>
<div class="sourceCode" id="cb179"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/stats/acf.html" class="external-link">pacf</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/start.html" class="external-link">width</a></span><span class="op">(</span><span class="va">coa</span><span class="op">$</span><span class="va">Oki_Osa</span><span class="op">)</span>, lag.max <span class="op">=</span> <span class="fl">100</span><span class="op">)</span></code></pre></div>
<p><img src="RegionsOfInterest_files/figure-html/autocorrelations-4.png" width="700"></p>
<div class="sourceCode" id="cb180"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/stats/acf.html" class="external-link">pacf</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/AnnotationDbi/man/Bimap-envirAPI.html" class="external-link">sample</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/start.html" class="external-link">width</a></span><span class="op">(</span><span class="va">coa</span><span class="op">$</span><span class="va">Oki_Osa</span><span class="op">)</span><span class="op">)</span>, lag.max <span class="op">=</span> <span class="fl">100</span><span class="op">)</span></code></pre></div>
<p><img src="RegionsOfInterest_files/figure-html/autocorrelations-5.png" width="700"></p>
</div>
</div>
<div class="section level2">
<h2 id="plots-of-unalmap-regions">Plots of unalmap regions<a class="anchor" aria-label="anchor" href="#plots-of-unalmap-regions"></a>
</h2>
<p>Section to be moved to a more proper place later.</p>
<p>Here we plot the regions that are <em>not</em> aligned, but that could be mapped because they are flanked by colinear regions.</p>
<p>We see indels, but we also see blocks of some length that do not align. Are they CNVs ?</p>
<p>Regions of interest were cherry picked for being syntenic, but scrambled regions also exist.</p>
<div class="sourceCode" id="cb181"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">bri</span><span class="op">$</span><span class="va">Oki_Kum</span><span class="op">[</span><span class="fl">20</span><span class="op">:</span><span class="fl">30</span><span class="op">]</span>   |&gt; <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/plotApairOfChrs.html" class="external-link">plotApairOfChrs</a></span><span class="op">(</span>main <span class="op">=</span> <span class="st">"Okiâ€“Kum"</span><span class="op">)</span></code></pre></div>
<p><img src="RegionsOfInterest_files/figure-html/unalmap_plots-1.png" width="700"></p>
<div class="sourceCode" id="cb182"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">bri</span><span class="op">$</span><span class="va">Oki_Osa</span><span class="op">[</span><span class="fl">405</span><span class="op">:</span><span class="fl">413</span><span class="op">]</span> |&gt; <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/plotApairOfChrs.html" class="external-link">plotApairOfChrs</a></span><span class="op">(</span>main <span class="op">=</span> <span class="st">"Okiâ€“Osa"</span><span class="op">)</span></code></pre></div>
<p><img src="RegionsOfInterest_files/figure-html/unalmap_plots-2.png" width="700"></p>
<div class="sourceCode" id="cb183"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">bri</span><span class="op">$</span><span class="va">Osa_Aom</span><span class="op">[</span><span class="fl">40</span><span class="op">:</span><span class="fl">50</span><span class="op">]</span>   |&gt; <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/plotApairOfChrs.html" class="external-link">plotApairOfChrs</a></span><span class="op">(</span>main <span class="op">=</span> <span class="st">"Osaâ€“Aom"</span><span class="op">)</span></code></pre></div>
<p><img src="RegionsOfInterest_files/figure-html/unalmap_plots-3.png" width="700"></p>
</div>
<div class="section level2">
<h2 id="kumes-contig_11_1">Kumeâ€™s contig_11_1<a class="anchor" aria-label="anchor" href="#kumes-contig_11_1"></a>
</h2>
<p>It does not seem to be syntenic with anything on Oki</p>
<div class="sourceCode" id="cb184"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">gbs</span><span class="op">$</span><span class="va">Oki_Kum</span> |&gt; <span class="fu">plyranges</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/plyranges/man/tidyverse-reexports.html" class="external-link">filter</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqnames</a></span><span class="op">(</span><span class="va">query</span><span class="op">)</span> <span class="op">==</span> <span class="st">"contig_11_1"</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/plotApairOfChrs.html" class="external-link">plotApairOfChrs</a></span><span class="op">(</span>chrT <span class="op">=</span> <span class="st">"chr1"</span><span class="op">)</span></code></pre></div>
<p><img src="RegionsOfInterest_files/figure-html/Kum_contig_11_1-1.png" width="700"></p>
<div class="sourceCode" id="cb185"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">gbs</span><span class="op">$</span><span class="va">Oki_Kum</span> |&gt; <span class="fu">plyranges</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/plyranges/man/tidyverse-reexports.html" class="external-link">filter</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqnames</a></span><span class="op">(</span><span class="va">query</span><span class="op">)</span> <span class="op">==</span> <span class="st">"contig_11_1"</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/plotApairOfChrs.html" class="external-link">plotApairOfChrs</a></span><span class="op">(</span>chrT <span class="op">=</span> <span class="st">"chr2"</span><span class="op">)</span></code></pre></div>
<p><img src="RegionsOfInterest_files/figure-html/Kum_contig_11_1-2.png" width="700"></p>
<div class="sourceCode" id="cb186"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">gbs</span><span class="op">$</span><span class="va">Oki_Kum</span> |&gt; <span class="fu">plyranges</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/plyranges/man/tidyverse-reexports.html" class="external-link">filter</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqnames</a></span><span class="op">(</span><span class="va">query</span><span class="op">)</span> <span class="op">==</span> <span class="st">"contig_11_1"</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/plotApairOfChrs.html" class="external-link">plotApairOfChrs</a></span><span class="op">(</span>chrT <span class="op">=</span> <span class="st">"PAR"</span><span class="op">)</span></code></pre></div>
<p><img src="RegionsOfInterest_files/figure-html/Kum_contig_11_1-3.png" width="700"></p>
<div class="sourceCode" id="cb187"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">gbs</span><span class="op">$</span><span class="va">Oki_Kum</span> |&gt; <span class="fu">plyranges</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/plyranges/man/tidyverse-reexports.html" class="external-link">filter</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqnames</a></span><span class="op">(</span><span class="va">query</span><span class="op">)</span> <span class="op">==</span> <span class="st">"contig_11_1"</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/plotApairOfChrs.html" class="external-link">plotApairOfChrs</a></span><span class="op">(</span>chrT <span class="op">=</span> <span class="st">"XSR"</span><span class="op">)</span></code></pre></div>
<p><img src="RegionsOfInterest_files/figure-html/Kum_contig_11_1-4.png" width="700"></p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Aleksandra Bliznina, Michael Mansfield, Charles Plessy.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.2.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
