<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Regions of interest • OikScrambling</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Regions of interest">
<meta property="og:description" content="OikScrambling">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">OikScrambling</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.8.3</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/OikScrambling.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/CNEs.html">Conserved non-coding elements</a>
    </li>
    <li>
      <a href="../articles/CoalescingCutoffs.html">Modifying coalescent cutoffs</a>
    </li>
    <li>
      <a href="../articles/ColinearityInterruptors.html">Regions interrupting colinearity</a>
    </li>
    <li>
      <a href="../articles/DistanceMeasures.html">Distance Measures</a>
    </li>
    <li>
      <a href="../articles/GenomicFeatures.html">Genomic features of the breakpoints</a>
    </li>
    <li>
      <a href="../articles/Inversions.html">Inversions</a>
    </li>
    <li>
      <a href="../articles/LoadGenomicBreaks.html">Load genomic break data</a>
    </li>
    <li>
      <a href="../articles/OrthogroupSyntenies.html">Orthogroup syntenies</a>
    </li>
    <li>
      <a href="../articles/OxfordPlots.html">Whole-genome dot-plots</a>
    </li>
    <li>
      <a href="../articles/PWM.html">Motif finding and analysis</a>
    </li>
    <li>
      <a href="../articles/ParallelPlots.html">Whole-chromosome parallel plots</a>
    </li>
    <li>
      <a href="../articles/RegionWidths.html">Width of the aligned and mapped regions.</a>
    </li>
    <li>
      <a href="../articles/RegionsOfInterest.html">Regions of interest</a>
    </li>
    <li>
      <a href="../articles/RepeatRegions.html">Repeat regions</a>
    </li>
    <li>
      <a href="../articles/SegmentMovements.html">Segment movement distances</a>
    </li>
    <li>
      <a href="../articles/TFBS.html">Transcription factor binding sites</a>
    </li>
    <li>
      <a href="../articles/Translocations.html">Translocations</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/oist/LuscombeU_OikScrambling/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="RegionsOfInterest_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Regions of interest</h1>
                        <h4 data-toc-skip class="author">Charles Plessy</h4>
            
            <h4 data-toc-skip class="date">09 February, 2022</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/oist/LuscombeU_OikScrambling/blob/HEAD/vignettes/RegionsOfInterest.Rmd" class="external-link"><code>vignettes/RegionsOfInterest.Rmd</code></a></small>
      <div class="hidden name"><code>RegionsOfInterest.Rmd</code></div>

    </div>

    
    
<div class="section level1">
<h1 id="load-pacakges-and-data">Load pacakges and data<a class="anchor" aria-label="anchor" href="#load-pacakges-and-data"></a>
</h1>
<p>See the <code><a href="../articles/OikScrambling.html">vignette("OikScrambling", package = "OikScrambling")</a></code> for general details on package and data load.</p>
<p>See <code><a href="../articles/LoadGenomicBreaks.html">vignette("LoadGenomicBreaks", package = "OikScrambling")</a></code> for how the different <code>GBreaks</code> objects are prepared.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="op">{</span>
  <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://oist.github.io/GenomicBreaks" class="external-link">'GenomicBreaks'</a></span><span class="op">)</span>
  <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://ggplot2.tidyverse.org" class="external-link">'ggplot2'</a></span><span class="op">)</span>
  <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st">"BreakpointsData"</span><span class="op">)</span>
<span class="op">}</span><span class="op">)</span>
<span class="va">genomes</span> <span class="op">&lt;-</span> <span class="fu">OikScrambling</span><span class="fu">:::</span><span class="fu"><a href="../reference/loadAllGenomes.html">loadAllGenomes</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Loading required package: BSgenome</span></code></pre>
<pre><code><span class="co">## Loading required package: Biostrings</span></code></pre>
<pre><code><span class="co">## Loading required package: XVector</span></code></pre>
<pre><code><span class="co">## </span>
<span class="co">## Attaching package: 'Biostrings'</span></code></pre>
<pre><code><span class="co">## The following object is masked from 'package:base':</span>
<span class="co">## </span>
<span class="co">##     strsplit</span></code></pre>
<pre><code><span class="co">## Loading required package: rtracklayer</span></code></pre>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/load.html" class="external-link">load</a></span><span class="op">(</span><span class="st">"BreakPoints.Rdata"</span><span class="op">)</span>
<span class="va">reps</span> <span class="op">&lt;-</span> <span class="fu">OikScrambling</span><span class="fu">:::</span><span class="fu"><a href="../reference/loadAllRepeats.html">loadAllRepeats</a></span><span class="op">(</span><span class="op">)</span>
<span class="va">transcripts</span> <span class="op">&lt;-</span> <span class="fu">OikScrambling</span><span class="fu">:::</span><span class="fu"><a href="../reference/loadAllTranscriptsGR.html">loadAllTranscriptsGR</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level1">
<h1 id="purpose">Purpose<a class="anchor" aria-label="anchor" href="#purpose"></a>
</h1>
<p>Here we explore regions of interest that caught our attention.</p>
<p>We have collected in a separate vignette (<code><a href="../articles/ColinearityInterruptors.html">vignette("ColinearityInterruptors", package = "OikScrambling")</a></code>) regions of interest that interrupt colinearity in same-population alignments such as Oki vs Kum.</p>
<p>The ROIs are not in the order I studied them, because I reorganised the code in the meantime.</p>
</div>
<div class="section level1">
<h1 id="roi1-composite-transposon-and-tandem-site-duplication-">ROI1: composite transposon and tandem site duplication.<a class="anchor" aria-label="anchor" href="#roi1-composite-transposon-and-tandem-site-duplication-"></a>
</h1>
<div class="section level2">
<h2 id="details-on-the-region">Details on the region<a class="anchor" aria-label="anchor" href="#details-on-the-region"></a>
</h2>
<p>This was identified and analysed by Martin Frith. I am just transposing the report in Bioconductor’s context.</p>
<p>In <code>contig_27_1</code> (which is most of chr1’s short arm), there is an indel region: the aligned regions in the Okinawa genome are directly in contact, but on the Kume genome the aligned regions are separated by a 5146-bp gap.</p>
<p>(The regions align on opposite strands, so the plot looks strange if you are not familiar with it.)</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="op">(</span><span class="va">ROI1</span> <span class="op">&lt;-</span> <span class="va">gbs</span><span class="op">$</span><span class="va">Oki_Kum</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/findOverlaps-methods.html" class="external-link">subsetByOverlaps</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a></span><span class="op">(</span><span class="st">"chr1:3720291-3734643"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## GBreaks object with 2 ranges and 8 metadata columns:</span>
<span class="co">##       seqnames          ranges strand |     score                       query</span>
<span class="co">##          &lt;Rle&gt;       &lt;IRanges&gt;  &lt;Rle&gt; | &lt;numeric&gt;                   &lt;GRanges&gt;</span>
<span class="co">##   [1]     chr1 3720291-3726692      - |     36253 contig_27_1:1233370-1239790</span>
<span class="co">##   [2]     chr1 3726693-3734643      - |     44991 contig_27_1:1220273-1228223</span>
<span class="co">##            Arm                   rep   repOvlp transcripts        flag</span>
<span class="co">##       &lt;factor&gt;       &lt;CharacterList&gt; &lt;integer&gt;       &lt;Rle&gt; &lt;character&gt;</span>
<span class="co">##   [1]    short           unknown,rnd       674        &lt;NA&gt;         Col</span>
<span class="co">##   [2]    short rnd,ltr-1,unknown,...       886        &lt;NA&gt;        &lt;NA&gt;</span>
<span class="co">##          nonCoa</span>
<span class="co">##       &lt;logical&gt;</span>
<span class="co">##   [1]     FALSE</span>
<span class="co">##   [2]     FALSE</span>
<span class="co">##   -------</span>
<span class="co">##   seqinfo: 19 sequences from OKI2018_I69 genome</span></code></pre>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/plotApairOfChrs.html" class="external-link">plotApairOfChrs</a></span><span class="op">(</span><span class="va">ROI1</span><span class="op">)</span></code></pre></div>
<p><img src="RegionsOfInterest_files/figure-html/ROI1-1.png" width="700"></p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/cleanGaps.html" class="external-link">cleanGaps</a></span><span class="op">(</span><span class="va">ROI1</span><span class="op">$</span><span class="va">query</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">##      seqnames   start     end width strand</span>
<span class="co">## 1 contig_27_1 1228224 1233369  5146      *</span></code></pre>
<p>Interestingly, the unaligned region in Kume contains two repeat elements in inverted tandem. You can see them in the alignment below, of the left-side boundary of the unaligned region to the reverse-complement of its right side.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">pairwiseAlignment</span><span class="op">(</span><span class="fu">getSeq</span><span class="op">(</span><span class="va">genomes</span><span class="op">$</span><span class="va">Kum</span>, <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/cleanGaps.html" class="external-link">cleanGaps</a></span><span class="op">(</span><span class="va">ROI1</span><span class="op">$</span><span class="va">query</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/intra-range-methods.html" class="external-link">flank</a></span><span class="op">(</span><span class="fl">20</span>, both <span class="op">=</span> <span class="cn">T</span>, start <span class="op">=</span> <span class="cn">T</span><span class="op">)</span><span class="op">)</span>,
                  <span class="fu">getSeq</span><span class="op">(</span><span class="va">genomes</span><span class="op">$</span><span class="va">Kum</span>, <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/cleanGaps.html" class="external-link">cleanGaps</a></span><span class="op">(</span><span class="va">ROI1</span><span class="op">$</span><span class="va">query</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/intra-range-methods.html" class="external-link">flank</a></span><span class="op">(</span><span class="fl">20</span>, both <span class="op">=</span> <span class="cn">T</span>, start <span class="op">=</span> <span class="cn">F</span><span class="op">)</span><span class="op">)</span> |&gt; <span class="fu">reverseComplement</span><span class="op">(</span><span class="op">)</span>,
                  type<span class="op">=</span><span class="st">"global"</span><span class="op">)</span> |&gt; <span class="fu">writePairwiseAlignments</span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## ########################################</span>
<span class="co">## # Program: Biostrings (version 2.62.0), a Bioconductor package</span>
<span class="co">## # Rundate: Wed Feb  9 10:23:27 2022</span>
<span class="co">## ########################################</span>
<span class="co">## #=======================================</span>
<span class="co">## #</span>
<span class="co">## # Aligned_sequences: 2</span>
<span class="co">## # 1: P1</span>
<span class="co">## # 2: S1</span>
<span class="co">## # Matrix: NA</span>
<span class="co">## # Gap_penalty: 14.0</span>
<span class="co">## # Extend_penalty: 4.0</span>
<span class="co">## #</span>
<span class="co">## # Length: 44</span>
<span class="co">## # Identity:      23/44 (52.3%)</span>
<span class="co">## # Similarity:    NA/44 (NA%)</span>
<span class="co">## # Gaps:           8/44 (18.2%)</span>
<span class="co">## # Score: -93.11032</span>
<span class="co">## #</span>
<span class="co">## #</span>
<span class="co">## #=======================================</span>
<span class="co">## </span>
<span class="co">## P1                 1 CAGAGCTTTTTCAGATCCAAGATACGTAGGCTTACCGGAC----     40</span>
<span class="co">##                      |  ||    | |   ||  |  |   ||||||||||||||    </span>
<span class="co">## S1                 1 CGAAGAGAATGCG--TCGTATCTT--TAGGCTTACCGGACTAAA     40</span>
<span class="co">## </span>
<span class="co">## </span>
<span class="co">## #---------------------------------------</span>
<span class="co">## #---------------------------------------</span></code></pre>
<details><p><summary>Click here to see the full pairwise alignment between the Kume insert and its own reverse-complement.</summary></p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">pairwiseAlignment</span><span class="op">(</span><span class="fu">getSeq</span><span class="op">(</span><span class="va">genomes</span><span class="op">$</span><span class="va">Kum</span>, <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/cleanGaps.html" class="external-link">cleanGaps</a></span><span class="op">(</span><span class="va">ROI1</span><span class="op">$</span><span class="va">query</span><span class="op">)</span><span class="op">)</span>,
                  <span class="fu">getSeq</span><span class="op">(</span><span class="va">genomes</span><span class="op">$</span><span class="va">Kum</span>, <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/cleanGaps.html" class="external-link">cleanGaps</a></span><span class="op">(</span><span class="va">ROI1</span><span class="op">$</span><span class="va">query</span><span class="op">)</span><span class="op">)</span> |&gt; <span class="fu">reverseComplement</span><span class="op">(</span><span class="op">)</span>,
                  type<span class="op">=</span><span class="st">"global"</span><span class="op">)</span> |&gt; <span class="fu">writePairwiseAlignments</span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## ########################################</span>
<span class="co">## # Program: Biostrings (version 2.62.0), a Bioconductor package</span>
<span class="co">## # Rundate: Wed Feb  9 10:23:28 2022</span>
<span class="co">## ########################################</span>
<span class="co">## #=======================================</span>
<span class="co">## #</span>
<span class="co">## # Aligned_sequences: 2</span>
<span class="co">## # 1: P1</span>
<span class="co">## # 2: S1</span>
<span class="co">## # Matrix: NA</span>
<span class="co">## # Gap_penalty: 14.0</span>
<span class="co">## # Extend_penalty: 4.0</span>
<span class="co">## #</span>
<span class="co">## # Length: 5518</span>
<span class="co">## # Identity:    3398/5518 (61.6%)</span>
<span class="co">## # Similarity:    NA/5518 (NA%)</span>
<span class="co">## # Gaps:         744/5518 (13.5%)</span>
<span class="co">## # Score: -6519.19</span>
<span class="co">## #</span>
<span class="co">## #</span>
<span class="co">## #=======================================</span>
<span class="co">## </span>
<span class="co">## P1                 1 GATACGTAGGCTTACCGGACTAAAATTTGGCCGATTTTTCTAGCTTAGTG     50</span>
<span class="co">##                        ||    ||||||||||||||||||||||||||||||||||||||||||</span>
<span class="co">## S1                 1 TTTA----GGCTTACCGGACTAAAATTTGGCCGATTTTTCTAGCTTAGTG     46</span>
<span class="co">## </span>
<span class="co">## P1                51 GGGGGGTCATATGATACATTTTTAGAAAGCTCAGAAGCTGGGCTTTCATT    100</span>
<span class="co">##                      ||||||||||||||||||||||||||||||||||||||||||||||||||</span>
<span class="co">## S1                47 GGGGGGTCATATGATACATTTTTAGAAAGCTCAGAAGCTGGGCTTTCATT     96</span>
<span class="co">## </span>
<span class="co">## P1               101 TGAGCCTAAAATCAGGGGGTCTGGGTCATGCCCGCAAGGCAAATTATGGG    150</span>
<span class="co">##                      ||||||||||||||||||||||||||||||||||||||||||||||||||</span>
<span class="co">## S1                97 TGAGCCTAAAATCAGGGGGTCTGGGTCATGCCCGCAAGGCAAATTATGGG    146</span>
<span class="co">## </span>
<span class="co">## P1               151 ACCTGAAGTCCCAAAATAGGCACTAGAACTCTTCAAACTATTTTTTTGGA    200</span>
<span class="co">##                      ||||||||||||||||||||||||||||||||||||||||||||||||||</span>
<span class="co">## S1               147 ACCTGAAGTCCCAAAATAGGCACTAGAACTCTTCAAACTATTTTTTTGGA    196</span>
<span class="co">## </span>
<span class="co">## P1               201 GGCGGGCCGGCCTCCAAAACCGTCGAGACTTATATATTTAGAAAGCTCAG    250</span>
<span class="co">##                      ||||||||||||||||||||||||||||||||||||||||||||||||||</span>
<span class="co">## S1               197 GGCGGGCCGGCCTCCAAAACCGTCGAGACTTATATATTTAGAAAGCTCAG    246</span>
<span class="co">## </span>
<span class="co">## P1               251 GCTCTGGGCTTTCAGATGAGCCCAAGAAATCTTAGTGCAAATGTCTCTGC    300</span>
<span class="co">##                      ||||||||||||||||||||||||||||||||||||||||||||||||||</span>
<span class="co">## S1               247 GCTCTGGGCTTTCAGATGAGCCCAAGAAATCTTAGTGCAAATGTCTCTGC    296</span>
<span class="co">## </span>
<span class="co">## P1               301 GGCCACCTGGGGGCGCTAGAAGTTGGGTGAAAAACCGGAAGTTCCGGAAA    350</span>
<span class="co">##                      ||||||||||||||||||||||||||||||||||||||||||||||||||</span>
<span class="co">## S1               297 GGCCACCTGGGGGCGCTAGAAGTTGGGTGAAAAACCGGAAGTTCCGGAAA    346</span>
<span class="co">## </span>
<span class="co">## P1               351 ACTCGGATTTAGAATTCTAAACTCTTATATTATGTTTGTAAAACATAAAC    400</span>
<span class="co">##                      ||||||||||||||||||||||||||||||||||||||||||||||||||</span>
<span class="co">## S1               347 ACTCGGATTTAGAATTCTAAACTCTTATATTATGTTTGTAAAACATAAAC    396</span>
<span class="co">## </span>
<span class="co">## P1               401 AAACAACAATTGACGCTCAAAAATGATGAACCGCGGTCGAACCGGGTGGT    450</span>
<span class="co">##                      ||||||||||||||||||||||||||||||||||||||||||||||||||</span>
<span class="co">## S1               397 AAACAACAATTGACGCTCAAAAATGATGAACCGCGGTCGAACCGGGTGGT    446</span>
<span class="co">## </span>
<span class="co">## P1               451 CAACCAGGTGGTTCATCGCGGTTCAACTCTAGAAACCTTCTAAAACATGG    500</span>
<span class="co">##                      ||||||||||||||||||||||||||||||||||||| ||||||||||||</span>
<span class="co">## S1               447 CAACCAGGTGGTTCATCGCGGTTCAACTCTAGAAACCGTCTAAAACATGG    496</span>
<span class="co">## </span>
<span class="co">## P1               501 GAATATGGGTATTTTTCGACGGGAAATTGAATGAAAATGAAAATAAAAGT    550</span>
<span class="co">##                      ||||||||||||||||||||||||||||||||||||||||||||||||||</span>
<span class="co">## S1               497 GAATATGGGTATTTTTCGACGGGAAATTGAATGAAAATGAAAATAAAAGT    546</span>
<span class="co">## </span>
<span class="co">## P1               551 TGAACCACGGTCGAACCACGTGGTCAACCACCCGGTTAACCGCGGTTCAA    600</span>
<span class="co">##                      ||||||||||||||||||||||||||||||   |||| | ||||||||||</span>
<span class="co">## S1               547 TGAACCACGGTCGAACCACGTGGTCAACCAGGTGGTTCATCGCGGTTCAA    596</span>
<span class="co">## </span>
<span class="co">## P1               601 CTCTAGAAACCTTCTAAAACATGGGAATATGGGTATTTTTCGACGGGAAA    650</span>
<span class="co">##                      ||||||||  |||||||||| | | |||||||||||||||||||||||||</span>
<span class="co">## S1               597 CTCTAGAATTCTTCTAAAACCTTGCAATATGGGTATTTTTCGACGGGAAA    646</span>
<span class="co">## </span>
<span class="co">## P1               651 TTGAATGAAAATGAAAATAAAAGTTGAACCGCGGTCAAACCGGGTGGTCA    700</span>
<span class="co">##                      |||||||||| |||||||||||||| |||||||||| ||| |||||||||</span>
<span class="co">## S1               647 TTGAATGAAATTGAAAATAAAAGTTAAACCGCGGTCGAACTGGGTGGTCA    696</span>
<span class="co">## </span>
<span class="co">## P1               701 ACCAGGTGGTTCATCGCGGTTCAACTCTAGAAACCTTCTTAAACATGGCA    750</span>
<span class="co">##                      |||||||||||| | |||||||||||||||||||||||| |||||||  |</span>
<span class="co">## S1               697 ACCAGGTGGTTCCTTGCGGTTCAACTCTAGAAACCTTCTAAAACATGATA    746</span>
<span class="co">## </span>
<span class="co">## P1               751 ATATGGGTATTTTTCGACGGGAAATTTGATTTTTATGTTTTTTTTATAAG    800</span>
<span class="co">##                      |||| ||||||||||||||| ||| |  ||  | |  |    ||| | | </span>
<span class="co">## S1               747 ATATTGGTATTTTTCGACGGCAAAGTCTATAATAAGCTAAAATTTCTTAA    796</span>
<span class="co">## </span>
<span class="co">## P1               801 ATCAAAACAACAAAACAAGATAATTTCCAAAAAACCATTTTATTCATTCT    850</span>
<span class="co">##                       |  ||  |         | | |||    |||    |||| ||| |||  </span>
<span class="co">## S1               797 TTTGAATGA---------GTTTATTG---AAATTTTATTTCATTAATTGA    834</span>
<span class="co">## </span>
<span class="co">## P1               851 GTGGTTCAGA--AGGTTCAGAAGAAATTCCTTCAAAATCCATCTCGTC-G    897</span>
<span class="co">##                        ||||  |   |   |||       ||| | |   || |||||  |  |</span>
<span class="co">## S1               835 AAGGTTTCGTCCAAAATCACTTATTTTTCTTCCTTGATTCATCTTCTGAG    884</span>
<span class="co">## </span>
<span class="co">## P1               898 ACATTCTGTTCTTCTTCCTCGTTCATTTCGTCCAATTCCTTTTCGACTAT    947</span>
<span class="co">##                       || ||| | |  ||| | |  || | ||   |    |   ||||||   </span>
<span class="co">## S1               885 TCAATCT-TGCCACTTGCGCAGTCTTATCAGTCGGCGCTGATTCGACGCG    933</span>
<span class="co">## </span>
<span class="co">## P1               948 TTCTTCCACTTCTTCAACAATCTCTTCCTCTCGTTCTTCCTCGTCATCTT    997</span>
<span class="co">##                        || || |||  |||   ||| |||  |||| | |||| || ||| | |</span>
<span class="co">## S1               934 CGCT-CCGCTT--TCACTTATC-CTTTTTCTCTTCCTTC-TC-TCACCGT    977</span>
<span class="co">## </span>
<span class="co">## P1               998 CGTTGATTTCGATTTCAGCCTCTCGCTCGTTCTTGTCTTTGTATTTCATT   1047</span>
<span class="co">##                      | || ||||   |||     |||  ||  |||| | |  || ||  || |</span>
<span class="co">## S1               978 CATTTATTTTTCTTTT----TCTTTCTTTTTCTCG-CAATGGATG-CAAT   1021</span>
<span class="co">## </span>
<span class="co">## P1              1048 TGGTCCTTCTTTTCGAGTTCCTTTGCTTTCTTCTCATTTTTCGCTCGTTC   1097</span>
<span class="co">##                       | |  | ||      || ||  |  |||   |||| ||| | | ||  |</span>
<span class="co">## S1              1022 CGATGATCCTC-----GTACCACT--TTTAAGCTCAATTT-CACCCGAGC   1063</span>
<span class="co">## </span>
<span class="co">## P1              1098 AACCATCGCTTTTGTAATCTCTACAAGACATTCATCATCCCGAAGGTCTT   1147</span>
<span class="co">##                      ||  | |  | ||| |   |||  |||| |||| ||              </span>
<span class="co">## S1              1064 AATGACCATTCTTG-AGCATCTTGAAGAGATTCCTCC-------------   1099</span>
<span class="co">## </span>
<span class="co">## P1              1148 CAAGCCAGAGACTTGTTTTATTGTACTGAATGAGACATTGTTAATATAAT   1197</span>
<span class="co">##                              |||||  |             |||||| ||  | || | |||</span>
<span class="co">## S1              1100 --------AGACTCAT-------------ATGAGAAATC-TCAACAAAAT   1127</span>
<span class="co">## </span>
<span class="co">## P1              1198 AA--AAATAGGCCAAAATTACCATGGAAACAGCCACGACAAATGTTCTTC   1245</span>
<span class="co">##                       |  ||  ||||     ||||    || | ||  |  |    | | || |</span>
<span class="co">## S1              1128 GACGAAGAAGGC-----TTACGCGAGAGAAAGAAAGTA----TCTGCTCC   1168</span>
<span class="co">## </span>
<span class="co">## P1              1246 GAAGTGAAAGCCTCG-ATTGTGTTCTTTCCAGCTCCTTTAATGTCCCAAA   1294</span>
<span class="co">##                       || |    | |  | |   | || ||   ||||| |   || ||  || </span>
<span class="co">## S1              1169 TAATTCTGCGGCAAGGAGACTTTTGTTAGAAGCTCTT---ATTTCTGAAG   1215</span>
<span class="co">## </span>
<span class="co">## P1              1295 CAGGGCTTCCACGCGT-CGAGAAGTCGCTGTCAAAAAGCTGTTTTCGTTC   1343</span>
<span class="co">##                          | ||| | |  | |     || | | |||||        ||  || </span>
<span class="co">## S1              1216 GTAAGATTCAAAGGTTTCTCTTCGTTGATTTCAAA--------TTAATT-   1256</span>
<span class="co">## </span>
<span class="co">## P1              1344 TGGACTCGAGTTGCGATTGCGTCCCATTTTTCTTTCTCCTTCGT-GAACT   1392</span>
<span class="co">##                      |  |   ||   ||||  || |    ||||||    |||  ||| ||| |</span>
<span class="co">## S1              1257 TAAA---GAAAAGCGACCGCAT----TTTTTCGAAATCC--CGTCGAATT   1297</span>
<span class="co">## </span>
<span class="co">## P1              1393 TCTCAAAGCTGCGACCGACAGACTGCCTAGTAACGCTGTCCAAGTGCTTC   1442</span>
<span class="co">##                      ||     | |  |  ||| || | |  | | || ||    |||||     </span>
<span class="co">## S1              1298 TCAAGTGGGTCGGGACGAGAGGCCGTGT-GAAAAGC----CAAGTCGACA   1342</span>
<span class="co">## </span>
<span class="co">## P1              1443 TTTTGTTCACCATCAAGAGCTCGGTACTGGATGAGTAGTCGCTGCTTGAC   1492</span>
<span class="co">##                         | ||| | |  |||   |||| | || |  |  |  |||   |   |</span>
<span class="co">## S1              1343 AGATCTTCTCGA--AAG---TCGGAATTGAAAAA--AACCGCCAGTATCC   1385</span>
<span class="co">## </span>
<span class="co">## P1              1493 TGTT------TTTGTCTCATCGTTGTCGTCGTTGGCCAGTTCTTGAAGC-   1535</span>
<span class="co">##                      | |       ||||| |||      ||   |   ||||   | | |||| </span>
<span class="co">## S1              1386 TTTCGGGAAGTTTGT-TCAAGAACTTCCAAGGCAGCCACCACGTCAAGCG   1434</span>
<span class="co">## </span>
<span class="co">## P1              1536 -ATTCG----CCAGTAAATATTCTTCTCAGATGCCATCATTCGCTTGTAG   1580</span>
<span class="co">##                       | |||    | || ||    ||||| || ||  |  | ||| |  ||  </span>
<span class="co">## S1              1435 AAATCGGAATCAAGGAAGAGATCTTCACAAATCTCC-CTTTCTCAGGT--   1481</span>
<span class="co">## </span>
<span class="co">## P1              1581 TAGTC-TTCCAAATTCGCGATAGCTTCTACAATCTTTTTCAAGGAGCTTG   1629</span>
<span class="co">##                         || || |   ||| |  |   |||   ||| ||||      |||| |</span>
<span class="co">## S1              1482 ---TCATTACCTTTTCCC--TCCTTTCGTTAATATTTTAACTCAAGCTCG   1526</span>
<span class="co">## </span>
<span class="co">## P1              1630 --ATGTTTCACTGACAGCTGC--ATGATATTTCTTCA-AGGAGCCATAGA   1674</span>
<span class="co">##                        ||||   | |||  | ||   ||||  | | |||  | || ||||  |</span>
<span class="co">## S1              1527 CCATGTCCGA-TGAT-GATGACGATGACGTCTTTTCTGAAGAACCATCAA   1574</span>
<span class="co">## </span>
<span class="co">## P1              1675 AAATGGCAAATGGGATGCAAGCAAGGAGGAGAACGAGTCTTTTGTGGGAT   1724</span>
<span class="co">##                            ||  |      || || | ||   || ||| |||     || | </span>
<span class="co">## S1              1575 ------CATCT------CACGCGATGATCCGA-CGA-TCTCAGCCGGAAG   1610</span>
<span class="co">## </span>
<span class="co">## P1              1725 CAACATTCGATCAACAACAACAACAACAACAACAACAACTACAACTA-CA   1773</span>
<span class="co">##                       || |   ||  || || || || || || |  |    | | |  |  ||</span>
<span class="co">## S1              1611 AAAAAGAAGAAGAAAAAGAAGAAGAAGAAGACGAGTCGCGAAAGTTGTCA   1660</span>
<span class="co">## </span>
<span class="co">## P1              1774 ACTAC----AACTACAACTACAACTA-CAACTAGTGATAGATGAACCTTT   1818</span>
<span class="co">##                      ||  |    | | | |  | |||  | || |   |||    |||  | ||</span>
<span class="co">## S1              1661 ACATCGGCGAGCGAGACATTCAAAAAACAGCGGCTGAGCCTTGACTCATT   1710</span>
<span class="co">## </span>
<span class="co">## P1              1819 -----GCCT------------ATTTTCCTTTG-CAGCTTTCTTCATTTCC   1850</span>
<span class="co">##                           || |            ||||||    | |  ||||| ||    ||</span>
<span class="co">## S1              1711 AGACAGCGTCGATGGCCGTGGATTTTCGAAAGTCTCCTTTCATCCGAACC   1760</span>
<span class="co">## </span>
<span class="co">## P1              1851 TCATCATTTTCTTCGTGAGGCCAGA--ATGACTCCAGAAAATCGCGAAGA   1898</span>
<span class="co">##                      | ||||   |  || ||  |||     |||     |||| || || ||| </span>
<span class="co">## S1              1761 TGATCAAGATTCTC-TGTTGCCTTCTCATGCGCGGAGAACATGGC-AAGT   1808</span>
<span class="co">## </span>
<span class="co">## P1              1899 AATTTCGCCGTGCTACTAGAATCCCCTGCGTAGCCCTGTAGAAGCGTTC-   1947</span>
<span class="co">##                        ||  | |   || || |||   | || | |      | || | |||| </span>
<span class="co">## S1              1809 CGTT--GTCAAACTTCTTGAACGTCTTGTGCAAGTGCATCGATGAGTTCA   1856</span>
<span class="co">## </span>
<span class="co">## P1              1948 -GGAT-AGTAG-------------GCATGATCAA----AACC---AGGCT   1975</span>
<span class="co">##                       |||| ||  |             ||| ||||||    || |   ||| |</span>
<span class="co">## S1              1857 AGGATCAGACGCCCCCTTCGATCAGCACGATCAATGAGAAACGGTAGGGT   1906</span>
<span class="co">## </span>
<span class="co">## P1              1976 TCTCATG-TTGCTTGGGAT-GATCATGAATCTTG-AGCCCATGTCTT---   2019</span>
<span class="co">##                      |||| |  ||| ||   || ||   | ||||||  |||   ||  |    </span>
<span class="co">## S1              1907 TCTCTTCCTTGATTTCAATTGAAGCT-AATCTTTTAGCTACTGCATCGAT   1955</span>
<span class="co">## </span>
<span class="co">## P1              2020 --TTTTAAGATTTTTAATATTG------AAATCAAGGAAGAGTTTGACAC   2061</span>
<span class="co">##                        ||| | |   |   || | |      ||||  ||| || |   |||  </span>
<span class="co">## S1              1956 ACTTTGACGGAATCCCATCTGGCAAAAGAAATTGAGGCAGCG---GACGA   2002</span>
<span class="co">## </span>
<span class="co">## P1              2062 ACGATGATGATTTCGATGACACAAACGCATTCAGCACCATCGAAAAC-CC   2110</span>
<span class="co">##                        |  | || || |  |   || | |||   || |  | || | | | ||</span>
<span class="co">## S1              2003 GGGTGGCTGTTTGCTTTTGGACGA-CGC---CACCCTCGTCCATAGCGCC   2048</span>
<span class="co">## </span>
<span class="co">## P1              2111 AAGTCTGGCACCCAAAATGAATCAAACAAG--GACTTTTCAAAAAAAAAG   2158</span>
<span class="co">##                      ||| |   ||||  || ||  |  |  | |  | ||             |</span>
<span class="co">## S1              2049 AAGGC---CACCGGAATTGTCTTTATTACGCCGGCT-------------G   2082</span>
<span class="co">## </span>
<span class="co">## P1              2159 GGTTGAAAATTTTTCTAAGTTTTTCAAAGAAATAGTGCCGCCTGATAAGG   2208</span>
<span class="co">##                      ||  |    |||   |  |  | |   |||  | | |||||   | || |</span>
<span class="co">## S1              2083 GGAAGTCCCTTTGCATCGGCCTGTATGAGACTTTG-GCCGC---AAAAAG   2128</span>
<span class="co">## </span>
<span class="co">## P1              2209 CTAAGCCAGTTATCGGCAAAAATTTCTAGCGTCCTAAACGGCTGGTACTA   2258</span>
<span class="co">##                        ||| ||    || ||| |   |||| |          |||   |||  </span>
<span class="co">## S1              2129 TGAAGACATGGCTCCGCAGA---TTCTTG----------GGCATCTAC--   2163</span>
<span class="co">## </span>
<span class="co">## P1              2259 GGGAGTTTTCCACCTTATTTTTC-GCGCTGAATTC--AAAT----TTTAT   2301</span>
<span class="co">##                          ||| ||  |||| ||  || | ||| || ||  ||||    ||| |</span>
<span class="co">## S1              2164 ----GTTCTCTTCCTTTTTGGTCTGTGCTCAAGTCGAAAATCGCCTTTCT   2209</span>
<span class="co">## </span>
<span class="co">## P1              2302 GATAAGGAATCGCTAAAGTCTCCGACTC---TGCTATTACTTTTCATACT   2348</span>
<span class="co">##                        |   | || |||        | | ||   ||| || |  |  |||   </span>
<span class="co">## S1              2210 TGTGTCGGATAGCTGTGCAGCTCAAATCAAGTGCAATGAGCT--CAT---   2254</span>
<span class="co">## </span>
<span class="co">## P1              2349 CCCTCTAGAATTTCTCACTAGAATATTCAATAATTTCTAACATTTGAAAA   2398</span>
<span class="co">##                       | || | | | ||  |  |||| ||  |||     | | |         </span>
<span class="co">## S1              2255 -CATCAAAAGTCTCCAAGAAGAAGATGGAATCGACGCCATC---------   2294</span>
<span class="co">## </span>
<span class="co">## P1              2399 AGTGCTGGTTTGTTCCATTTTTGATTCTAGCCTGAGGTTTTTGAAGATGC   2448</span>
<span class="co">##                       |   |  ||||| | | |    |    ||||||        || |  | </span>
<span class="co">## S1              2295 -GCCATCCTTTGT-CAAATGCACAGCACAGCCTG--------GATG--GG   2332</span>
<span class="co">## </span>
<span class="co">## P1              2449 TGAATGCGTTTTTTACATCAAAATCATCATTGGGTGCTAAACTCTTCCTT   2498</span>
<span class="co">##                      | ||| | ||||               ||||          |||||||||</span>
<span class="co">## S1              2333 TAAATTCATTTT---------------CATT----------CTCTTCCTT   2357</span>
<span class="co">## </span>
<span class="co">## P1              2499 GATTTCAATAAT------AAAAATCCGTTCCTTGGCCAAACATTCTGCTT   2542</span>
<span class="co">##                      ||||||||| |       |||||| |   |||    |||| | |  ||| </span>
<span class="co">## S1              2358 GATTTCAATTACCTTTAGAAAAATACA--CCTA---CAAAGACTGGGCT-   2401</span>
<span class="co">## </span>
<span class="co">## P1              2543 CTGCATCCGAACCAGTCCTCAAGATAAGCTTCGAGGATGTTTCGCAGGTT   2592</span>
<span class="co">##                      | || | | ||  || || ||       |||| ||     | |  | | |</span>
<span class="co">## S1              2402 CAGCCTGCAAAAAAGACC-CA-------CTTCCAGCCGAATCCAGACGAT   2443</span>
<span class="co">## </span>
<span class="co">## P1              2593 GTCTCGATTGTGGCCTGTCAAGTTCAGCGTTCGCGAGCCGAAGAGTTTCA   2642</span>
<span class="co">##                      |  || ||   || |   |  | ||  | |   | |    |||  || ||</span>
<span class="co">## S1              2444 GCTTCAAT---GGACCAGCCTGATCCTCTTCTTCAA----AAGGCTTCCA   2486</span>
<span class="co">## </span>
<span class="co">## P1              2643 TGGTAGCCATGTGGACCTTGGAAGCCTT----TTGAAGAAGAGGATCAGG   2688</span>
<span class="co">##                       |||   ||||   ||| || ||  |||    | |   | |  || |  |</span>
<span class="co">## S1              2487 AGGTCCACATGGCTACCATGAAACTCTTCGGCTCGCGAACGCTGAACTTG   2536</span>
<span class="co">## </span>
<span class="co">## P1              2689 CTGGTCCA---TTGAAGCATCGTCTGGATTCGGCTGGAAG-------TG-   2727</span>
<span class="co">##                         | |||   | ||  || | |  | |     || ||||       || </span>
<span class="co">## S1              2537 ACAGGCCACAATCGAGACAACCTGCGAAACATCCTCGAAGCTTATCTTGA   2586</span>
<span class="co">## </span>
<span class="co">## P1              2728 GGTCTTTTTTGCAGGCTGA-GCCCAGTCTTTGT---AGGT--GTATTTTT   2771</span>
<span class="co">##                      || ||  || | | || || ||  | | ||||    |||   | ||||||</span>
<span class="co">## S1              2587 GGACTGGTTCGGATGCAGAAGCAGAATGTTTGGCCAAGGAACGGATTTTT   2636</span>
<span class="co">## </span>
<span class="co">## P1              2772 CTAAAGGTAATTGAAATCAAGGAAGAG----------AATGA--------   2803</span>
<span class="co">##                       | |      |||||||||||||||||          |||||        </span>
<span class="co">## S1              2637 ATTA------TTGAAATCAAGGAAGAGTTTAGCACCCAATGATGATTTTG   2680</span>
<span class="co">## </span>
<span class="co">## P1              2804 -------AAATGAATTTACC--CATC--------CAGGCTGTGCTGTGCA   2836</span>
<span class="co">##                             ||| | ||| | |  | ||        ||||||    |    |</span>
<span class="co">## S1              2681 ATGTAAAAAACGCATTCAGCATCTTCAAAAACCTCAGGCTAGAATCAAAA   2730</span>
<span class="co">## </span>
<span class="co">## P1              2837 TTTGA-CAAAGGATGGC----------GATGGCGTCGATTCCATCTTCTT   2875</span>
<span class="co">##                       | || ||||  |   |          | | |     |||  || |||| </span>
<span class="co">## S1              2731 ATGGAACAAACCAGCACTTTTTCAAATGTTAGAAATTATTGAATATTCTA   2780</span>
<span class="co">## </span>
<span class="co">## P1              2876 CTTGGAGACTTTTGATG----ATGAGC--TCATTGCACTTGATTTGAGCT   2919</span>
<span class="co">##                       |  || | | | || |    ||||    | || |||   || | |    </span>
<span class="co">## S1              2781 GTGAGAAATTCTAGAGGGAGTATGAAAAGTAATAGCA---GAGTCGGAGA   2827</span>
<span class="co">## </span>
<span class="co">## P1              2920 GCACAGCTATCCGACACAAGAAAGGCGATTTTCGACTTGAGCACAGACCA   2969</span>
<span class="co">##                          ||| || |   |  | |||    ||||  || || ||| | ||  |</span>
<span class="co">## S1              2828 CTTTAGCGATTCCTTATCATAAA----ATTT--GAATTCAGCGC-GAAAA   2870</span>
<span class="co">## </span>
<span class="co">## P1              2970 AAAAGGAAGAGAAC------GTAGATGCC----------CAAGAA---TC   3000</span>
<span class="co">##                      | ||||  || |||      |||   |||          | ||||   | </span>
<span class="co">## S1              2871 ATAAGGTGGAAAACTCCCTAGTACCAGCCGTTTAGGACGCTAGAAATTTT   2920</span>
<span class="co">## </span>
<span class="co">## P1              3001 TGCGGAGCCATGTCTTCACTTTTT---GCGGC-CAAAGTCTCATACAGGC   3046</span>
<span class="co">##                      ||| ||    || |||  | || |   ||||| | |  |||   | |  |</span>
<span class="co">## S1              2921 TGCCGATAACTGGCTTAGCCTTATCAGGCGGCACTATTTCTTTGAAAAAC   2970</span>
<span class="co">## </span>
<span class="co">## P1              3047 CGATGCAAAGGGACTTCCC-------------AGCCGGCGTAATAAAGAC   3083</span>
<span class="co">##                        |   |||    |  |||             || |   ||  |  |  |</span>
<span class="co">## S1              2971 TTAGAAAAATTTTCAACCCTTTTTTTTTGAAAAGTCCTTGT--TTGATTC   3018</span>
<span class="co">## </span>
<span class="co">## P1              3084 AATTCCGGTGGC---CTTGGCGCTATGGACGAGGGTG---GCGTC-GTCC   3126</span>
<span class="co">##                      | ||  |||| |   ||||| | | | || |  | ||   ||||  ||  </span>
<span class="co">## S1              3019 ATTTTGGGTGCCAGACTTGG-GTTTTCGATGGTGCTGAATGCGTTTGTGT   3067</span>
<span class="co">## </span>
<span class="co">## P1              3127 AAAAGCAAACAGCCACCCTCGTC---CGCTGCCTCAATTTCTTTTGCCAG   3173</span>
<span class="co">##                       |  | || || |  |    |||   | || |||  |||||  |      </span>
<span class="co">## S1              3068 CATCGAAATCATCATCGTGTGTCAAACTCTTCCTTGATTTCAAT------   3111</span>
<span class="co">## </span>
<span class="co">## P1              3174 ATGGGATTCCGTCAAAGTATCGATGCAGTAGCTAAAAGATT-AGCTTCAA   3222</span>
<span class="co">##                      ||   |   | | |||      |  ||   ||| || |||| |   ||| </span>
<span class="co">## S1              3112 ATTAAAAATCTTAAAAA-----AGACATGGGCTCAA-GATTCATGATCA-   3154</span>
<span class="co">## </span>
<span class="co">## P1              3223 TTGAAATCAAGGAAGAGAACCCTACCGTTTCTCATTGATCGTGCTGATCG   3272</span>
<span class="co">##                      |   || |||  | ||||| |||   ||||     ||||| |||  |   </span>
<span class="co">## S1              3155 TCCCAAGCAAC-ATGAGAAGCCTG--GTTT-----TGATCATGCCTA---   3193</span>
<span class="co">## </span>
<span class="co">## P1              3273 AAGGGGGCGTCTGATCCTTGAACTCATCGATGCACTTGCACAAGACGTTC   3322</span>
<span class="co">##                                || ||||  |||| | || |      | | || |   |||</span>
<span class="co">## S1              3194 ----------CT-ATCC--GAACGCTTCTACAGGGCTACGCAGGGGATTC   3230</span>
<span class="co">## </span>
<span class="co">## P1              3323 AAGAAGTTTGAC--AACGACTT-GCCATGTTCTCCGCGCATGAGAAGGCA   3369</span>
<span class="co">##                       || ||   | |  ||   ||| || || ||||     |||     ||| </span>
<span class="co">## S1              3231 TAGTAGCACGGCGAAATTTCTTCGCGATTTTCTGGAGTCATTCT--GGCC   3278</span>
<span class="co">## </span>
<span class="co">## P1              3370 ACA-GAGAATCTTGATCAGGTTCGGATGAAAGGAGACTTTCGAAAATCCA   3418</span>
<span class="co">##                       || ||  |   |||| |||    || |||||  | |    ||||||   </span>
<span class="co">## S1              3279 TCACGAAGAAAATGATGAGGAAATGAAGAAAGCTG-CAAAGGAAAATA--   3325</span>
<span class="co">## </span>
<span class="co">## P1              3419 CGGCCATCGACGCTGTCTAATGAGTCAAGGCTCAGCCGCTGTTTTTTGAA   3468</span>
<span class="co">##                       ||| |  |    | | | |  |||    | |  |  | |||  ||  | </span>
<span class="co">## S1              3326 -GGCAAAGGTTCATCTATCACTAGTTGTAGTT--GTAGTTGTAGTTGTAG   3372</span>
<span class="co">## </span>
<span class="co">## P1              3469 TGTCTCGCTCGCCGATGTTGACAACTTTCGCGACTCGTCTTCTTCTTCTT   3518</span>
<span class="co">##                      | | | | | |  | |||||                 | || || || ||</span>
<span class="co">## S1              3373 T-TGTAGTT-GTAGTTGTTG----------------TTGTTGTTGTTGTT   3404</span>
<span class="co">## </span>
<span class="co">## P1              3519 CTTTTTCTTCTTCTTTTTCTTCCGGCTGAGA-TCGT-CGGATCATCGCGT   3566</span>
<span class="co">##                       || ||  ||   | ||  | ||     ||| |||| |   || | || |</span>
<span class="co">## S1              3405 GTTGTTGATCGAATGTTGATCCCACAAAAGACTCGTTCTCCTCCTTGCTT   3454</span>
<span class="co">## </span>
<span class="co">## P1              3567 GAG-----------ATGTTG-ATGGTTCTTCAGAAAAGACGTCATCGTCA   3604</span>
<span class="co">##                      |             || ||  |||| || |  ||| | |  ||||   ||</span>
<span class="co">## S1              3455 GCATCCCATTTGCCATTTTCTATGGCTCCTT-GAAGAAATATCATG--CA   3501</span>
<span class="co">## </span>
<span class="co">## P1              3605 TCA-TCATCGGA-CATGGCGAGCTTGAGTTAAAATATTAACGAAAGGA--   3650</span>
<span class="co">##                       |  |||  | | |||  | ||||      |||| |||   |||   |  </span>
<span class="co">## S1              3502 GCTGTCAGTGAAACAT--CAAGCTCCTTGAAAAAGATTGTAGAAGCTATC   3549</span>
<span class="co">## </span>
<span class="co">## P1              3651 GGGAAAAGGTAATGA-----ACCTGAGAAAGGGAG-ATTTGTGAAGATCT   3694</span>
<span class="co">##                      | |||   | || ||     ||  | ||| |   | || || |||||   </span>
<span class="co">## S1              3550 GCGAATTTGGAA-GACTACTACAAGCGAATGATGGCATCTGAGAAGAATA   3598</span>
<span class="co">## </span>
<span class="co">## P1              3695 CTTCCTTGATTCCGATTTCGCTTGACGTGGTGGCTGCCTTGGAAGTTCTT   3744</span>
<span class="co">##                       || || |    ||| |  |||| | |   ||||   |   ||      |</span>
<span class="co">## S1              3599 TTTACTGG----CGAAT--GCTTCAAGAACTGGCCAACGACGACAACGAT   3642</span>
<span class="co">## </span>
<span class="co">## P1              3745 GA-ACAAACTTCCCGAAAGGATACTGGCGGTTTTT--TTCAATTCCGA-C   3790</span>
<span class="co">##                      || |||||       | ||   |   |||  |  |  | || | |||| |</span>
<span class="co">## S1              3643 GAGACAAAA------ACAGTCAAGCAGCGACTACTCATCCAGTACCGAGC   3686</span>
<span class="co">## </span>
<span class="co">## P1              3791 TTTCGA----GAAGATCTTGTCGACTTGG----CTTTTC-ACACGGCCTC   3831</span>
<span class="co">##                      | | ||    ||| |        ||||||    | || | |  | | || </span>
<span class="co">## S1              3687 TCTTGATGGTGAACAAAAGAAGCACTTGGACAGCGTTACTAGGCAGTCTG   3736</span>
<span class="co">## </span>
<span class="co">## P1              3832 TCGTCCCGACCCACTTGAAATTCGACG--GGATTTCGAAAAA----ATGC   3875</span>
<span class="co">##                      |||  |  | |     ||| ||| |||  |||    ||||||    | ||</span>
<span class="co">## S1              3737 TCGGTCGCAGCTTTGAGAAGTTC-ACGAAGGAGAAAGAAAAATGGGACGC   3785</span>
<span class="co">## </span>
<span class="co">## P1              3876 GGTCGCTTTTC---TTTA-AATTAA--------TTTGAAATCAACGAAGA   3913</span>
<span class="co">##                        ||||   ||   |  | ||  ||        ||||| | | ||     </span>
<span class="co">## S1              3786 AATCGCAACTCGAGTCCAGAACGAAAACAGCTTTTTGACAGCGACTTCTC   3835</span>
<span class="co">## </span>
<span class="co">## P1              3914 GAAACCTTTGAATCTTACCTTCAGAAATAA---GAGCTTCTAACAAAAGT   3960</span>
<span class="co">##                      ||  | |  ||| |     ||  || || |   |||||   || || |  </span>
<span class="co">## S1              3836 GACGCGTG-GAAGCCCTGTTTGGGACATTAAAGGAGCTGGAAAGAACACA   3884</span>
<span class="co">## </span>
<span class="co">## P1              3961 CTCCTTGCCGCAGAATTAGGAGCAGATACTTTCTTTCTCTCGC----GTA   4006</span>
<span class="co">##                       ||   ||     | || | || | ||   |  |  || |  |    |||</span>
<span class="co">## S1              3885 ATCGAGGCTTTC-ACTTCGAAGAACATTTGTCGTGGCTGTTTCCATGGTA   3933</span>
<span class="co">## </span>
<span class="co">## P1              4007 A-----GCCTTCTTCGTCATTTTGTTGAGATT-TCTCAT-----------   4039</span>
<span class="co">##                      |     ||||  ||  | ||| | || | | | ||||||           </span>
<span class="co">## S1              3934 ATTTTGGCCTATTT--TTATTATATTAACAATGTCTCATTCAGTACAATA   3981</span>
<span class="co">## </span>
<span class="co">## P1              4040 --ATGAGTCTG---------------------GAGGAATCTCTTCAAGAT   4066</span>
<span class="co">##                        |  |||||                      || |||| ||||  ||| </span>
<span class="co">## S1              3982 AAACAAGTCTCTGGCTTGAAGACCTTCGGGATGATGAATGTCTTGTAGAG   4031</span>
<span class="co">## </span>
<span class="co">## P1              4067 GCT-CAAGAATGGTCATTGCTCGGGTGAAAT-TGAGCTTAAA--AGTGGT   4112</span>
<span class="co">##                        | ||| |  | |  |||  || | ||||  ||||   |||  |  || </span>
<span class="co">## S1              4032 ATTACAAAAGCGATGGTTGAACGAGCGAAAAATGAGAAGAAAGCAAAGGA   4081</span>
<span class="co">## </span>
<span class="co">## P1              4113 ACG-----AGGATCATCGATTGCA-TCCATTG-CGAGAAAAAGAAAGAAA   4155</span>
<span class="co">##                      ||      || |  | | | || | | ||  | | ||||  ||  |||  </span>
<span class="co">## S1              4082 ACTCGAAAAGAAGGACCAAATGAAATACAAAGACAAGAACGAGCGAGAGG   4131</span>
<span class="co">## </span>
<span class="co">## P1              4156 AAGAAA----AATAAATGACGGTGA-GAGAAGGAA-GAGAAAAAG-GATA   4198</span>
<span class="co">##                        ||||    ||| || || | ||| ||| | ||| ||||  ||| ||| </span>
<span class="co">## S1              4132 CTGAAATCGAAATCAACGAAGATGACGAGGAAGAACGAGAGGAAGAGATT   4181</span>
<span class="co">## </span>
<span class="co">## P1              4199 AGTGAA--AGCGGA-GCGCGCGTCGAATCAGCGCCGACTGATAAGACTGC   4245</span>
<span class="co">##                        ||||  || ||| |     ||||||   |    |   || | ||  | </span>
<span class="co">## S1              4182 GTTGAAGAAGTGGAAGAAATAGTCGAAAAGGAATTGGACGAAATGAACGA   4231</span>
<span class="co">## </span>
<span class="co">## P1              4246 GCAAGTGGCA-AGATTGACTCAGAAGATGAATCAAGGAAGAAAAATAAGT   4294</span>
<span class="co">##                      | |||  | | ||| || |   | ||||| ||   | | |||       |</span>
<span class="co">## S1              4232 GGAAGAAGAACAGAATGTCGACG-AGATGGATTTTGAAGGAATTTCTTCT   4280</span>
<span class="co">## </span>
<span class="co">## P1              4295 GATTTTGGACGAAACCTTTCAATTAATGAAATAAAATTTC---AATAAAC   4341</span>
<span class="co">##                      ||   |   |  ||||    ||| ||| ||||    |||    ||| | |</span>
<span class="co">## S1              4281 GAACCTT--CTGAACCACAGAATGAATAAAATGGTTTTTTGGAAATTATC   4328</span>
<span class="co">## </span>
<span class="co">## P1              4342 TCATTCAAAT---------TAAGAAATTTTAGCTTATTATAGACTTTGCC   4382</span>
<span class="co">##                      |  ||    |         | | |||    |  | |  ||  | ||| ||</span>
<span class="co">## S1              4329 TTGTTTTGTTGTTTTGATCTTATAAAAAAAACATAAAAATCAAATTTCCC   4378</span>
<span class="co">## </span>
<span class="co">## P1              4383 GTCGAAAAATACCAATATTATCATGTTTTAGAAGGTTTCTAGAGTTGAAC   4432</span>
<span class="co">##                      ||||||||||||| |||||  ||||||| |||||||||||||||||||||</span>
<span class="co">## S1              4379 GTCGAAAAATACCCATATTGCCATGTTTAAGAAGGTTTCTAGAGTTGAAC   4428</span>
<span class="co">## </span>
<span class="co">## P1              4433 CGCAAGGAACCACCTGGTTGACCACCCAGTTCGACCGCGGTTTAACTTTT   4482</span>
<span class="co">##                      ||| | ||||||||||||||||||||| ||| |||||||||| |||||||</span>
<span class="co">## S1              4429 CGCGATGAACCACCTGGTTGACCACCCGGTTTGACCGCGGTTCAACTTTT   4478</span>
<span class="co">## </span>
<span class="co">## P1              4483 ATTTTCAATTTCATTCAATTTCCCGTCGAAAAATACCCATATTGCAAGGT   4532</span>
<span class="co">##                      ||||||| ||||||||||||||||||||||||||||||||||| | | ||</span>
<span class="co">## S1              4479 ATTTTCATTTTCATTCAATTTCCCGTCGAAAAATACCCATATTCCCATGT   4528</span>
<span class="co">## </span>
<span class="co">## P1              4533 TTTAGAAGAATTCTAGAGTTGAACCGCGATGAACCACCTGGTTGACCACG   4582</span>
<span class="co">##                      ||||||||  |||||||||||||||||| | ||||   ||||||||||||</span>
<span class="co">## S1              4529 TTTAGAAGGTTTCTAGAGTTGAACCGCGGTTAACCGGGTGGTTGACCACG   4578</span>
<span class="co">## </span>
<span class="co">## P1              4583 TGGTTCGACCGTGGTTCAACTTTTATTTTCATTTTCATTCAATTTCCCGT   4632</span>
<span class="co">##                      ||||||||||||||||||||||||||||||||||||||||||||||||||</span>
<span class="co">## S1              4579 TGGTTCGACCGTGGTTCAACTTTTATTTTCATTTTCATTCAATTTCCCGT   4628</span>
<span class="co">## </span>
<span class="co">## P1              4633 CGAAAAATACCCATATTCCCATGTTTTAGACGGTTTCTAGAGTTGAACCG   4682</span>
<span class="co">##                      |||||||||||||||||||||||||||||| |||||||||||||||||||</span>
<span class="co">## S1              4629 CGAAAAATACCCATATTCCCATGTTTTAGAAGGTTTCTAGAGTTGAACCG   4678</span>
<span class="co">## </span>
<span class="co">## P1              4683 CGATGAACCACCTGGTTGACCACCCGGTTCGACCGCGGTTCATCATTTTT   4732</span>
<span class="co">##                      ||||||||||||||||||||||||||||||||||||||||||||||||||</span>
<span class="co">## S1              4679 CGATGAACCACCTGGTTGACCACCCGGTTCGACCGCGGTTCATCATTTTT   4728</span>
<span class="co">## </span>
<span class="co">## P1              4733 GAGCGTCAATTGTTGTTTGTTTATGTTTTACAAACATAATATAAGAGTTT   4782</span>
<span class="co">##                      ||||||||||||||||||||||||||||||||||||||||||||||||||</span>
<span class="co">## S1              4729 GAGCGTCAATTGTTGTTTGTTTATGTTTTACAAACATAATATAAGAGTTT   4778</span>
<span class="co">## </span>
<span class="co">## P1              4783 AGAATTCTAAATCCGAGTTTTCCGGAACTTCCGGTTTTTCACCCAACTTC   4832</span>
<span class="co">##                      ||||||||||||||||||||||||||||||||||||||||||||||||||</span>
<span class="co">## S1              4779 AGAATTCTAAATCCGAGTTTTCCGGAACTTCCGGTTTTTCACCCAACTTC   4828</span>
<span class="co">## </span>
<span class="co">## P1              4833 TAGCGCCCCCAGGTGGCCGCAGAGACATTTGCACTAAGATTTCTTGGGCT   4882</span>
<span class="co">##                      ||||||||||||||||||||||||||||||||||||||||||||||||||</span>
<span class="co">## S1              4829 TAGCGCCCCCAGGTGGCCGCAGAGACATTTGCACTAAGATTTCTTGGGCT   4878</span>
<span class="co">## </span>
<span class="co">## P1              4883 CATCTGAAAGCCCAGAGCCTGAGCTTTCTAAATATATAAGTCTCGACGGT   4932</span>
<span class="co">##                      ||||||||||||||||||||||||||||||||||||||||||||||||||</span>
<span class="co">## S1              4879 CATCTGAAAGCCCAGAGCCTGAGCTTTCTAAATATATAAGTCTCGACGGT   4928</span>
<span class="co">## </span>
<span class="co">## P1              4933 TTTGGAGGCCGGCCCGCCTCCAAAAAAATAGTTTGAAGAGTTCTAGTGCC   4982</span>
<span class="co">##                      ||||||||||||||||||||||||||||||||||||||||||||||||||</span>
<span class="co">## S1              4929 TTTGGAGGCCGGCCCGCCTCCAAAAAAATAGTTTGAAGAGTTCTAGTGCC   4978</span>
<span class="co">## </span>
<span class="co">## P1              4983 TATTTTGGGACTTCAGGTCCCATAATTTGCCTTGCGGGCATGACCCAGAC   5032</span>
<span class="co">##                      ||||||||||||||||||||||||||||||||||||||||||||||||||</span>
<span class="co">## S1              4979 TATTTTGGGACTTCAGGTCCCATAATTTGCCTTGCGGGCATGACCCAGAC   5028</span>
<span class="co">## </span>
<span class="co">## P1              5033 CCCCTGATTTTAGGCTCAAATGAAAGCCCAGCTTCTGAGCTTTCTAAAAA   5082</span>
<span class="co">##                      ||||||||||||||||||||||||||||||||||||||||||||||||||</span>
<span class="co">## S1              5029 CCCCTGATTTTAGGCTCAAATGAAAGCCCAGCTTCTGAGCTTTCTAAAAA   5078</span>
<span class="co">## </span>
<span class="co">## P1              5083 TGTATCATATGACCCCCCCACTAAGCTAGAAAAATCGGCCAAATTTTAGT   5132</span>
<span class="co">##                      ||||||||||||||||||||||||||||||||||||||||||||||||||</span>
<span class="co">## S1              5079 TGTATCATATGACCCCCCCACTAAGCTAGAAAAATCGGCCAAATTTTAGT   5128</span>
<span class="co">## </span>
<span class="co">## P1              5133 CCGGTAAGCCT----AAA   5146</span>
<span class="co">##                      |||||||||||    |  </span>
<span class="co">## S1              5129 CCGGTAAGCCTACGTATC   5146</span>
<span class="co">## </span>
<span class="co">## </span>
<span class="co">## #---------------------------------------</span>
<span class="co">## #---------------------------------------</span></code></pre>
</details><p>Immediately upstream these repeats, there a tandem octamer <code>AAGATACG</code>. In the Okinawan genome, it is present as a single copy. You can see it in the alignment below, where the right-side boundary was not reverse-complemented.</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">pairwiseAlignment</span><span class="op">(</span><span class="fu">getSeq</span><span class="op">(</span><span class="va">genomes</span><span class="op">$</span><span class="va">Kum</span>, <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/cleanGaps.html" class="external-link">cleanGaps</a></span><span class="op">(</span><span class="va">ROI1</span><span class="op">$</span><span class="va">query</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/intra-range-methods.html" class="external-link">flank</a></span><span class="op">(</span><span class="fl">20</span>, both <span class="op">=</span> <span class="cn">T</span>, start <span class="op">=</span> <span class="cn">T</span><span class="op">)</span><span class="op">)</span>,
                  <span class="fu">getSeq</span><span class="op">(</span><span class="va">genomes</span><span class="op">$</span><span class="va">Kum</span>, <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/cleanGaps.html" class="external-link">cleanGaps</a></span><span class="op">(</span><span class="va">ROI1</span><span class="op">$</span><span class="va">query</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/intra-range-methods.html" class="external-link">flank</a></span><span class="op">(</span><span class="fl">20</span>, both <span class="op">=</span> <span class="cn">T</span>, start <span class="op">=</span> <span class="cn">F</span><span class="op">)</span><span class="op">)</span>,
                  type<span class="op">=</span><span class="st">"global"</span><span class="op">)</span> |&gt; <span class="fu">writePairwiseAlignments</span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## ########################################</span>
<span class="co">## # Program: Biostrings (version 2.62.0), a Bioconductor package</span>
<span class="co">## # Rundate: Wed Feb  9 10:23:29 2022</span>
<span class="co">## ########################################</span>
<span class="co">## #=======================================</span>
<span class="co">## #</span>
<span class="co">## # Aligned_sequences: 2</span>
<span class="co">## # 1: P1</span>
<span class="co">## # 2: S1</span>
<span class="co">## # Matrix: NA</span>
<span class="co">## # Gap_penalty: 14.0</span>
<span class="co">## # Extend_penalty: 4.0</span>
<span class="co">## #</span>
<span class="co">## # Length: 40</span>
<span class="co">## # Identity:      17/40 (42.5%)</span>
<span class="co">## # Similarity:    NA/40 (NA%)</span>
<span class="co">## # Gaps:           0/40 (0.0%)</span>
<span class="co">## # Score: -101.9937</span>
<span class="co">## #</span>
<span class="co">## #</span>
<span class="co">## #=======================================</span>
<span class="co">## </span>
<span class="co">## P1                 1 CAGAGCTTTTTCAGATCCAAGATACGTAGGCTTACCGGAC     40</span>
<span class="co">##                         ||     | ||    ||||||||  |  ||  |    </span>
<span class="co">## S1                 1 TTTAGTCCGGTAAGCCTAAAGATACGACGCATTCTCTTCG     40</span>
<span class="co">## </span>
<span class="co">## </span>
<span class="co">## #---------------------------------------</span>
<span class="co">## #---------------------------------------</span></code></pre>
<p>The same can be seen in the multiple alignment below comparising the left and right arms of the Kume alignments and their match to the Okinawa genome.</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Note that genome alignment is +/- therefore the order below is counter-intuitive.</span>
<span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>
  kum_right <span class="op">=</span> <span class="fu">getSeq</span><span class="op">(</span><span class="va">genomes</span><span class="op">$</span><span class="va">Kum</span>, <span class="va">ROI1</span><span class="op">$</span><span class="va">query</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/intra-range-methods.html" class="external-link">flank</a></span><span class="op">(</span><span class="fl">20</span>, both <span class="op">=</span> <span class="cn">T</span>, start <span class="op">=</span> <span class="cn">T</span><span class="op">)</span><span class="op">)</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span>,
  kum_left  <span class="op">=</span> <span class="fu">getSeq</span><span class="op">(</span><span class="va">genomes</span><span class="op">$</span><span class="va">Kum</span>, <span class="va">ROI1</span><span class="op">$</span><span class="va">query</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/intra-range-methods.html" class="external-link">flank</a></span><span class="op">(</span><span class="fl">20</span>, both <span class="op">=</span> <span class="cn">T</span>, start <span class="op">=</span> <span class="cn">F</span><span class="op">)</span><span class="op">)</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span>,
  oki       <span class="op">=</span> <span class="fu">getSeq</span><span class="op">(</span><span class="va">genomes</span><span class="op">$</span><span class="va">Oki</span>, <span class="va">ROI1</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/intra-range-methods.html" class="external-link">flank</a></span><span class="op">(</span><span class="fl">20</span>, both <span class="op">=</span> <span class="cn">T</span>, start <span class="op">=</span> <span class="cn">F</span><span class="op">)</span><span class="op">)</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span>
<span class="op">)</span> |&gt; <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/as.html" class="external-link">as</a></span><span class="op">(</span><span class="st">"DNAStringSet"</span><span class="op">)</span> |&gt; <span class="fu">msa</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/msa/man/msaClustalW.html" class="external-link">msaClustalW</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## use default substitution matrix</span></code></pre>
<pre><code><span class="co">## CLUSTAL 2.1  </span>
<span class="co">## </span>
<span class="co">## Call:</span>
<span class="co">##    msa::msaClustalW(as(list(kum_right = getSeq(genomes$Kum, flank(ROI1$query[1],     20, both = T, start = T))[[1]], kum_left = getSeq(genomes$Kum,     flank(ROI1$query[2], 20, both = T, start = F))[[1]], oki = getSeq(genomes$Oki,     flank(ROI1[2], 20, both = T, start = F))[[1]]), "DNAStringSet"))</span>
<span class="co">## </span>
<span class="co">## MsaDNAMultipleAlignment with 3 rows and 41 columns</span>
<span class="co">##     aln                                       names</span>
<span class="co">## [1] CAGAGCTTTTTCAGATCCAAGATACGTAGGCTTACCGGAC- kum_left</span>
<span class="co">## [2] CAGAGCTTTTTCAGATCCAAGATACG-ACGCATTCTCTTCG oki</span>
<span class="co">## [3] TTTAGTCCGGTAAGCCTAAAGATACG-ACGCATTCTCTTCG kum_right</span>
<span class="co">## Con CAGAGCTTTTTCAGATCCAAGATACG-ACGCATTCTCTTCG Consensus</span></code></pre>
<p>The microhomology does not appear to be more frequent in the Okinawan genome compared to the others.</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">genomes</span>, <span class="va">vmatchPattern</span>, pat <span class="op">=</span> <span class="st">"AAGATACG"</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">length</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">##  Oki  Osa  Bar  Kum  Aom  Nor </span>
<span class="co">## 1478 1334 1485 1493 1324 1730</span></code></pre>
</div>
<div class="section level2">
<h2 id="how-often-do-we-see-tandem-site-duplication-in-similar-indel-configurations">How often do we see tandem site duplication in similar indel configurations ?<a class="anchor" aria-label="anchor" href="#how-often-do-we-see-tandem-site-duplication-in-similar-indel-configurations"></a>
</h2>
<p>The <code>unalMap</code> objects contain unaligned regions flanked by colinear regions. Let’s search for other situations where one of the regions has a width of zero.</p>
<p>Note: <em>in order to represent zero-width intervals in this object, which would be impossible with <code>GRanges</code> objects, the unaligned regions were arbitrarly added one base of each flanking aligned regions</em>.</p>
<p>These regions are rare in between-species alignments (at most a few percents), and common in within-population alignments (~10 to 20 percents).</p>
<p>Note2: <em>the numbers would double if I would also inspect the query genome widths</em>.</p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">unalMap</span> |&gt; <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">sapply</a></span><span class="op">(</span>\<span class="op">(</span><span class="va">gb</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">gb</span><span class="op">[</span><span class="fu">width</span><span class="op">(</span><span class="va">gb</span><span class="op">)</span> <span class="op">&lt;=</span> <span class="fl">2</span><span class="op">]</span><span class="op">)</span><span class="op">)</span>  <span class="co"># See note above</span></code></pre></div>
<pre><code><span class="co">## Oki_Osa Oki_Bar Oki_Kum Oki_Aom Oki_Nor Osa_Oki Osa_Bar Osa_Kum Osa_Aom Osa_Nor </span>
<span class="co">##     194     139    1395     207     137     198     355     181    1227     394 </span>
<span class="co">## Bar_Oki Bar_Osa Bar_Kum Bar_Aom Bar_Nor Ply_Ros Ply_Rob Ply_Sav Ply_Oki Rob_Ros </span>
<span class="co">##     180     520     170     479    1785    3129    2483      30       1    2322 </span>
<span class="co">## Rob_Ply Rob_Sav Rob_Oki Dme_Dbu Dme_Dsu Dme_Dya Dme_Dma </span>
<span class="co">##    2311      32       1      18     116     276     504</span></code></pre>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/Round.html" class="external-link">round</a></span><span class="op">(</span><span class="va">unalMap</span> |&gt; <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">sapply</a></span><span class="op">(</span>\<span class="op">(</span><span class="va">gb</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">gb</span><span class="op">[</span><span class="fu">width</span><span class="op">(</span><span class="va">gb</span><span class="op">)</span> <span class="op">&lt;=</span> <span class="fl">2</span><span class="op">]</span><span class="op">)</span><span class="op">)</span> <span class="op">/</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">unalMap</span>, <span class="va">length</span><span class="op">)</span> <span class="op">*</span> <span class="fl">100</span> <span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Oki_Osa Oki_Bar Oki_Kum Oki_Aom Oki_Nor Osa_Oki Osa_Bar Osa_Kum Osa_Aom Osa_Nor </span>
<span class="co">##       1       1      14       1       1       1       2       1      12       2 </span>
<span class="co">## Bar_Oki Bar_Osa Bar_Kum Bar_Aom Bar_Nor Ply_Ros Ply_Rob Ply_Sav Ply_Oki Rob_Ros </span>
<span class="co">##       1       3       1       2      20       9       5       0       0       4 </span>
<span class="co">## Rob_Ply Rob_Sav Rob_Oki Dme_Dbu Dme_Dsu Dme_Dya Dme_Dma </span>
<span class="co">##       4       0       0       0       0       1       7</span></code></pre>
<p>Let’s inspect within-population alignments first. Here is an example with a <code>CATCTTTG</code> TSD.</p>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">gb</span> <span class="op">&lt;-</span> <span class="va">unalMap</span><span class="op">$</span><span class="va">Oki_Kum</span> |&gt; <span class="fu">plyranges</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/plyranges/man/tidyverse-reexports.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">width</span> <span class="op">==</span> <span class="fl">2</span><span class="op">)</span>
<span class="va">gb</span><span class="op">$</span><span class="va">query</span> |&gt; <span class="fu">width</span><span class="op">(</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">##    Min. 1st Qu.  Median    Mean 3rd Qu.    Max. </span>
<span class="co">##    51.0   343.5   577.0  1196.3   868.5 67179.0</span></code></pre>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">alignFlank</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">gb</span><span class="op">)</span>
  <span class="fu">pairwiseAlignment</span><span class="op">(</span><span class="fu">getSeq</span><span class="op">(</span><span class="va">genomes</span><span class="op">$</span><span class="va">Kum</span>, <span class="va">gb</span><span class="op">$</span><span class="va">query</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/intra-range-methods.html" class="external-link">flank</a></span><span class="op">(</span><span class="fl">20</span>, both <span class="op">=</span> <span class="cn">T</span>, start <span class="op">=</span> <span class="cn">T</span><span class="op">)</span><span class="op">)</span>,
                    <span class="fu">getSeq</span><span class="op">(</span><span class="va">genomes</span><span class="op">$</span><span class="va">Kum</span>, <span class="va">gb</span><span class="op">$</span><span class="va">query</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/intra-range-methods.html" class="external-link">flank</a></span><span class="op">(</span><span class="fl">20</span>, both <span class="op">=</span> <span class="cn">T</span>, start <span class="op">=</span> <span class="cn">F</span><span class="op">)</span><span class="op">)</span>,
                    type<span class="op">=</span><span class="st">"global"</span><span class="op">)</span> |&gt; <span class="fu">writePairwiseAlignments</span><span class="op">(</span><span class="op">)</span>

<span class="fu">alignFlank</span><span class="op">(</span><span class="va">gb</span><span class="op">[</span><span class="fl">4</span><span class="op">]</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## ########################################</span>
<span class="co">## # Program: Biostrings (version 2.62.0), a Bioconductor package</span>
<span class="co">## # Rundate: Wed Feb  9 10:24:32 2022</span>
<span class="co">## ########################################</span>
<span class="co">## #=======================================</span>
<span class="co">## #</span>
<span class="co">## # Aligned_sequences: 2</span>
<span class="co">## # 1: P1</span>
<span class="co">## # 2: S1</span>
<span class="co">## # Matrix: NA</span>
<span class="co">## # Gap_penalty: 14.0</span>
<span class="co">## # Extend_penalty: 4.0</span>
<span class="co">## #</span>
<span class="co">## # Length: 42</span>
<span class="co">## # Identity:      20/42 (47.6%)</span>
<span class="co">## # Similarity:    NA/42 (NA%)</span>
<span class="co">## # Gaps:           4/42 (9.5%)</span>
<span class="co">## # Score: -102.552</span>
<span class="co">## #</span>
<span class="co">## #</span>
<span class="co">## #=======================================</span>
<span class="co">## </span>
<span class="co">## P1                 1 AAAGCCAATTCGCGCAACCTTCATCTTTGGGTTGTAGGC--G     40</span>
<span class="co">##                       | |  |   |   | |||  ||||||||   |  |  |  |</span>
<span class="co">## S1                 1 CACGTTACCGCATACCACC--CATCTTTGCCCTTGACCCTTG     40</span>
<span class="co">## </span>
<span class="co">## </span>
<span class="co">## #---------------------------------------</span>
<span class="co">## #---------------------------------------</span></code></pre>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">alignFlank2</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">gb</span>, <span class="va">qGenome</span><span class="op">=</span><span class="va">genomes</span><span class="op">$</span><span class="va">Kum</span>, <span class="va">n</span> <span class="op">=</span> <span class="fl">8</span><span class="op">)</span> <span class="op">{</span>
  <span class="co"># Remove 1-bp padding</span>
  <span class="va">gb</span><span class="op">$</span><span class="va">query</span> <span class="op">&lt;-</span> <span class="va">gb</span><span class="op">$</span><span class="va">query</span> <span class="op">-</span><span class="fl">1</span>
  <span class="fu">pairwiseAlignment</span><span class="op">(</span><span class="fu">getSeq</span><span class="op">(</span><span class="va">qGenome</span>, <span class="va">gb</span><span class="op">$</span><span class="va">query</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/intra-range-methods.html" class="external-link">resize</a></span><span class="op">(</span><span class="va">n</span>, fix <span class="op">=</span> <span class="st">"start"</span><span class="op">)</span><span class="op">)</span>,
                    <span class="fu">getSeq</span><span class="op">(</span><span class="va">qGenome</span>, <span class="va">gb</span><span class="op">$</span><span class="va">query</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/intra-range-methods.html" class="external-link">flank</a></span><span class="op">(</span><span class="va">n</span>, start <span class="op">=</span> <span class="cn">F</span><span class="op">)</span><span class="op">)</span>,
                    type<span class="op">=</span><span class="st">"global"</span><span class="op">)</span>
<span class="op">}</span>
<span class="fu">alignFlank2</span><span class="op">(</span><span class="va">gb</span><span class="op">[</span><span class="fl">4</span><span class="op">]</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Global PairwiseAlignmentsSingleSubject (1 of 1)</span>
<span class="co">## pattern: <span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">TTT</span><span style="color: #000000; background-color: #AFFFFF;">G</span></span>
<span class="co">## subject: <span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">TTT</span><span style="color: #000000; background-color: #AFFFFF;">G</span></span>
<span class="co">## score: 15.85405</span></code></pre>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">msAlignFlank</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">gb</span>, <span class="va">tGenome</span> <span class="op">=</span> <span class="va">genomes</span><span class="op">$</span><span class="va">Oki</span>, <span class="va">qGenome</span><span class="op">=</span><span class="va">genomes</span><span class="op">$</span><span class="va">Kum</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>
  kum_left  <span class="op">=</span> <span class="fu">getSeq</span><span class="op">(</span><span class="va">qGenome</span>, <span class="va">gb</span><span class="op">$</span><span class="va">query</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/intra-range-methods.html" class="external-link">flank</a></span><span class="op">(</span><span class="fl">20</span>, both <span class="op">=</span> <span class="cn">T</span>, start <span class="op">=</span> <span class="cn">T</span><span class="op">)</span><span class="op">)</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span>,
  oki_left  <span class="op">=</span> <span class="fu">getSeq</span><span class="op">(</span><span class="va">tGenome</span>, <span class="va">gb</span>       |&gt; <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/intra-range-methods.html" class="external-link">flank</a></span><span class="op">(</span><span class="fl">20</span>, both <span class="op">=</span> <span class="cn">T</span>, start <span class="op">=</span> <span class="cn">F</span><span class="op">)</span><span class="op">)</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span>,
  kum_right <span class="op">=</span> <span class="fu">getSeq</span><span class="op">(</span><span class="va">qGenome</span>, <span class="va">gb</span><span class="op">$</span><span class="va">query</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/intra-range-methods.html" class="external-link">flank</a></span><span class="op">(</span><span class="fl">20</span>, both <span class="op">=</span> <span class="cn">T</span>, start <span class="op">=</span> <span class="cn">F</span><span class="op">)</span><span class="op">)</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span>
<span class="op">)</span> |&gt; <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/as.html" class="external-link">as</a></span><span class="op">(</span><span class="st">"DNAStringSet"</span><span class="op">)</span> |&gt; <span class="fu">msa</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/msa/man/msaClustalW.html" class="external-link">msaClustalW</a></span><span class="op">(</span>order <span class="op">=</span> <span class="st">"input"</span><span class="op">)</span>

<span class="fu">msAlignFlank</span><span class="op">(</span><span class="va">gb</span><span class="op">[</span><span class="fl">4</span><span class="op">]</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## use default substitution matrix</span></code></pre>
<pre><code><span class="co">## CLUSTAL 2.1  </span>
<span class="co">## </span>
<span class="co">## Call:</span>
<span class="co">##    msa::msaClustalW(as(list(kum_left = getSeq(qGenome, flank(gb$query,     20, both = T, start = T))[[1]], oki_left = getSeq(tGenome,     flank(gb, 20, both = T, start = F))[[1]], kum_right = getSeq(qGenome,     flank(gb$query, 20, both = T, start = F))[[1]]), "DNAStringSet"),     order = "input")</span>
<span class="co">## </span>
<span class="co">## MsaDNAMultipleAlignment with 3 rows and 42 columns</span>
<span class="co">##     aln                                        names</span>
<span class="co">## [1] AAAGCCAATTCGCGCAACCTTCATCTTTGGGTTGTAGGCG-- kum_left</span>
<span class="co">## [2] --GCCTAATTCGCGCAACCTTCATCTTTGCCCTTGACCCTTG oki_left</span>
<span class="co">## [3] CACGTTA--CCGCATACCACCCATCTTTGCCCTTGACCCTTG kum_right</span>
<span class="co">## Con ?A?GCTAATTCGCGCAACCTTCATCTTTGCCCTTGACCCTTG Consensus</span></code></pre>
<p>Let’s count the number of perfect 8-nt matches. See the variety in their sequences: none of them is found more than twice!</p>
<div class="sourceCode" id="cb38"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">alignFlank2</span><span class="op">(</span><span class="va">gb</span><span class="op">)</span> |&gt; <span class="fu">nedit</span><span class="op">(</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## </span>
<span class="co">##   0   1   2   3   4   5   6   7   8 </span>
<span class="co">## 609  99 119 188 179  95  66  27  13</span></code></pre>
<div class="sourceCode" id="cb40"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">gb</span><span class="op">$</span><span class="va">nedit</span> <span class="op">&lt;-</span> <span class="fu">alignFlank2</span><span class="op">(</span><span class="va">gb</span><span class="op">)</span> |&gt; <span class="fu">nedit</span><span class="op">(</span><span class="op">)</span> 
<span class="va">gb</span><span class="op">$</span><span class="va">consensus</span> <span class="op">&lt;-</span> <span class="fu">alignFlank2</span><span class="op">(</span><span class="va">gb</span><span class="op">)</span> |&gt; <span class="fu">compareStrings</span><span class="op">(</span><span class="op">)</span>

<span class="va">gb</span><span class="op">[</span><span class="va">gb</span><span class="op">$</span><span class="va">nedit</span><span class="op">==</span><span class="fl">0</span><span class="op">]</span><span class="op">$</span><span class="va">consensus</span> |&gt; <span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/r/base/sort.html" class="external-link">sort</a></span><span class="op">(</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">tail</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## </span>
<span class="co">## AGCTATTA CTTTTGAC GACAGAAT GCAATAAG GCTGGAAG GCTGGAAT </span>
<span class="co">##        2        2        2        2        2        2</span></code></pre>
<div class="sourceCode" id="cb42"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">gb</span><span class="op">[</span><span class="va">gb</span><span class="op">$</span><span class="va">nedit</span><span class="op">==</span><span class="fl">0</span><span class="op">]</span><span class="op">$</span><span class="va">consensus</span> |&gt; <span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## [1] 603</span></code></pre>
</div>
<div class="section level2">
<h2 id="remaining-questions">Remaining questions<a class="anchor" aria-label="anchor" href="#remaining-questions"></a>
</h2>
<p>Questions:</p>
<ul>
<li>How frequent is the inverted pair of repeats that we see?</li>
<li>Is there any hint of functionality for the sequence in the gap?</li>
<li>are the indels fixed in their respective populations?</li>
<li>Am I missing half of the microhomologies because of alignments to negative strands ?</li>
</ul>
</div>
</div>
<div class="section level1">
<h1 id="unaligned-mapped-region-roi4">Unaligned mapped region: ROI4<a class="anchor" aria-label="anchor" href="#unaligned-mapped-region-roi4"></a>
</h1>
<p>Let’s check examples of unaligned mapped regions in function of presence of repeats.</p>
<div class="section level2">
<h2 id="falsetrue">FALSE/TRUE<a class="anchor" aria-label="anchor" href="#falsetrue"></a>
</h2>
<div class="sourceCode" id="cb44"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">unalMap</span><span class="op">$</span><span class="va">Oki_Osa</span> <span class="op">-&gt;</span> <span class="va">x</span>
<span class="va">x</span><span class="op">$</span><span class="va">Class</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">x</span><span class="op">$</span><span class="va">rep</span>, <span class="va">anyNA</span><span class="op">)</span>, <span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">x</span><span class="op">$</span><span class="va">query</span><span class="op">$</span><span class="va">rep</span>, <span class="va">anyNA</span><span class="op">)</span><span class="op">)</span>
<span class="va">x</span> |&gt; <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">##   seqnames  start    end width strand query.seqnames query.start query.end</span>
<span class="co">## 1     chr1 552335 552383    49      *           Chr1     1730795   1731145</span>
<span class="co">## 2     chr1 552647 552907   261      *           Chr1     1730115   1730510</span>
<span class="co">## 3     chr1 553834 553873    40      *           Chr1     1728809   1729169</span>
<span class="co">## 4     chr1 554134 554168    35      *           Chr1     1728321   1728530</span>
<span class="co">## 5     chr1 554365 554546   182      *           Chr1     1727927   1728109</span>
<span class="co">## 6     chr1 554692 555191   500      *           Chr1     1727331   1727781</span>
<span class="co">##   query.width query.strand query.rep query.repOvlp query.transcripts   Arm rep</span>
<span class="co">## 1         351            *       rnd           389              &lt;NA&gt; short  NA</span>
<span class="co">## 2         396            *    tandem           237              &lt;NA&gt; short  NA</span>
<span class="co">## 3         361            *        NA             0           g385.t1 short  NA</span>
<span class="co">## 4         210            *        NA             0           g385.t1 short  NA</span>
<span class="co">## 5         183            *        NA             0           g384.t1 short  NA</span>
<span class="co">## 6         451            *        NA             0           g384.t1 short rnd</span>
<span class="co">##   repOvlp transcripts flag       Class</span>
<span class="co">## 1       0      g56.t1 &lt;NA&gt;  FALSE TRUE</span>
<span class="co">## 2       0        &lt;NA&gt; &lt;NA&gt;  FALSE TRUE</span>
<span class="co">## 3       0        &lt;NA&gt; &lt;NA&gt; FALSE FALSE</span>
<span class="co">## 4       0        &lt;NA&gt; &lt;NA&gt; FALSE FALSE</span>
<span class="co">## 5       0        &lt;NA&gt; &lt;NA&gt; FALSE FALSE</span>
<span class="co">## 6     162        &lt;NA&gt; &lt;NA&gt;  TRUE FALSE</span></code></pre>
<div class="sourceCode" id="cb46"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ROI4_1</span> <span class="op">&lt;-</span> <span class="va">x</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>

<span class="co"># Let's recover the flanking regions</span>
<span class="co"># (it works because the unalMap regions are expanded of 1 base on each side to avoid zero-length ranges)</span>
<span class="va">gbs</span><span class="op">$</span><span class="va">Oki_Osa</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/findOverlaps-methods.html" class="external-link">subsetByOverlaps</a></span><span class="op">(</span><span class="va">ROI4_1</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">##   seqnames  start    end width strand score query.seqnames query.start</span>
<span class="co">## 1     chr1 551988 552335   348      -   680           Chr1     1731145</span>
<span class="co">## 2     chr1 552383 552647   265      -   416           Chr1     1730510</span>
<span class="co">##   query.end query.width query.strand query.rep query.repOvlp query.transcripts</span>
<span class="co">## 1   1731548         404            *        NA             0              &lt;NA&gt;</span>
<span class="co">## 2   1730795         286            *       rnd             4              &lt;NA&gt;</span>
<span class="co">##     Arm rep repOvlp transcripts flag nonCoa</span>
<span class="co">## 1 short  NA       0      g56.t1  Col  FALSE</span>
<span class="co">## 2 short  NA       0      g56.t1  Col  FALSE</span></code></pre>
<div class="sourceCode" id="cb48"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">gbs</span><span class="op">$</span><span class="va">Oki_Osa</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/findOverlaps-methods.html" class="external-link">subsetByOverlaps</a></span><span class="op">(</span><span class="va">ROI4_1</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/inter-range-methods.html" class="external-link">range</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## GBreaks object with 1 range and 1 metadata column:</span>
<span class="co">##       seqnames        ranges strand |                query</span>
<span class="co">##          &lt;Rle&gt;     &lt;IRanges&gt;  &lt;Rle&gt; |            &lt;GRanges&gt;</span>
<span class="co">##   [1]     chr1 551988-552647      * | Chr1:1730510-1731548</span>
<span class="co">##   -------</span>
<span class="co">##   seqinfo: 19 sequences from OKI2018_I69 genome</span></code></pre>
<div class="sourceCode" id="cb50"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># We confirm that unalmap regions can not be mapped</span>
<span class="fu">pairwiseAlignment</span><span class="op">(</span><span class="va">genomes</span><span class="op">$</span><span class="va">Oki</span><span class="op">$</span><span class="va">chr1</span><span class="op">[</span><span class="fl">552335</span><span class="op">:</span><span class="fl">552383</span><span class="op">]</span>, <span class="va">genomes</span><span class="op">$</span><span class="va">Osa</span><span class="op">$</span><span class="va">Chr1</span><span class="op">[</span><span class="fl">1730795</span><span class="op">:</span><span class="fl">1731145</span><span class="op">]</span>, type<span class="op">=</span><span class="st">"global"</span><span class="op">)</span> |&gt; <span class="fu">writePairwiseAlignments</span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## ########################################</span>
<span class="co">## # Program: Biostrings (version 2.62.0), a Bioconductor package</span>
<span class="co">## # Rundate: Wed Feb  9 10:24:38 2022</span>
<span class="co">## ########################################</span>
<span class="co">## #=======================================</span>
<span class="co">## #</span>
<span class="co">## # Aligned_sequences: 2</span>
<span class="co">## # 1: P1</span>
<span class="co">## # 2: S1</span>
<span class="co">## # Matrix: NA</span>
<span class="co">## # Gap_penalty: 14.0</span>
<span class="co">## # Extend_penalty: 4.0</span>
<span class="co">## #</span>
<span class="co">## # Length: 351</span>
<span class="co">## # Identity:      44/351 (12.5%)</span>
<span class="co">## # Similarity:    NA/351 (NA%)</span>
<span class="co">## # Gaps:         302/351 (86.0%)</span>
<span class="co">## # Score: -1250.299</span>
<span class="co">## #</span>
<span class="co">## #</span>
<span class="co">## #=======================================</span>
<span class="co">## </span>
<span class="co">## P1                 1 TT------------TCCTCC------------------------------      8</span>
<span class="co">##                      ||            || | |                              </span>
<span class="co">## S1                 1 TTAGTTCAGTTAACTCATGCGGGGAAAGTAGCCAATAAGTTGTAAATTTT     50</span>
<span class="co">## </span>
<span class="co">## P1                 9 --------------------------------------------------      8</span>
<span class="co">##                                                                        </span>
<span class="co">## S1                51 CATTTACCTTAAAATGGCTGGAACGTAGCGAAACTAAAAACGCGAAGCGA    100</span>
<span class="co">## </span>
<span class="co">## P1                 9 --------------------------------------------------      8</span>
<span class="co">##                                                                        </span>
<span class="co">## S1               101 AGCTTCGAGCCAAAAATAAAAATATGAGACATTTTGAGGCGAAGCTTCGC    150</span>
<span class="co">## </span>
<span class="co">## P1                 9 ---------TTGATT-----------------------------------     14</span>
<span class="co">##                               ||||||                                   </span>
<span class="co">## S1               151 TTCACGCTTTTGATTGGTTGAGTGTGGCGAAGCTAAAAACGCGAAGCAAA    200</span>
<span class="co">## </span>
<span class="co">## P1                15 -----------TCAATTTT-------------------CGA---------     25</span>
<span class="co">##                                 | ||||||                   |||         </span>
<span class="co">## S1               201 GCTAATTTTCATTAATTTTGACGGGAGGCTTCGTTTTGCGATTTTAGCTT    250</span>
<span class="co">## </span>
<span class="co">## P1                26 ---------CA----TTTTT-----------GAAGA--------------     37</span>
<span class="co">##                               ||    |||||           |||||              </span>
<span class="co">## S1               251 CGCTACGCTCAGATGTTTTTAGCGAATTTGAGAAGACAACATTTTAGTCA    300</span>
<span class="co">## </span>
<span class="co">## P1                38 -------------------------------ATC-TATT-------TGGT     48</span>
<span class="co">##                                                     ||| ||||       |  |</span>
<span class="co">## S1               301 CTTTACCCGCAGGGGTTTAGTTTTAATGATTATCATATTAACTGCATACT    350</span>
<span class="co">## </span>
<span class="co">## P1                49 A     49</span>
<span class="co">##                      |</span>
<span class="co">## S1               351 A    351</span>
<span class="co">## </span>
<span class="co">## </span>
<span class="co">## #---------------------------------------</span>
<span class="co">## #---------------------------------------</span></code></pre>
<div class="sourceCode" id="cb52"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">pairwiseAlignment</span><span class="op">(</span><span class="va">genomes</span><span class="op">$</span><span class="va">Oki</span><span class="op">$</span><span class="va">chr1</span><span class="op">[</span><span class="fl">552335</span><span class="op">:</span><span class="fl">552383</span><span class="op">]</span>, <span class="va">genomes</span><span class="op">$</span><span class="va">Osa</span><span class="op">$</span><span class="va">Chr1</span><span class="op">[</span><span class="fl">1730795</span><span class="op">:</span><span class="fl">1731145</span><span class="op">]</span> |&gt; <span class="fu">reverseComplement</span><span class="op">(</span><span class="op">)</span>, type<span class="op">=</span><span class="st">"global"</span><span class="op">)</span> |&gt; <span class="fu">writePairwiseAlignments</span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## ########################################</span>
<span class="co">## # Program: Biostrings (version 2.62.0), a Bioconductor package</span>
<span class="co">## # Rundate: Wed Feb  9 10:24:38 2022</span>
<span class="co">## ########################################</span>
<span class="co">## #=======================================</span>
<span class="co">## #</span>
<span class="co">## # Aligned_sequences: 2</span>
<span class="co">## # 1: P1</span>
<span class="co">## # 2: S1</span>
<span class="co">## # Matrix: NA</span>
<span class="co">## # Gap_penalty: 14.0</span>
<span class="co">## # Extend_penalty: 4.0</span>
<span class="co">## #</span>
<span class="co">## # Length: 351</span>
<span class="co">## # Identity:      42/351 (12.0%)</span>
<span class="co">## # Similarity:    NA/351 (NA%)</span>
<span class="co">## # Gaps:         302/351 (86.0%)</span>
<span class="co">## # Score: -1256.061</span>
<span class="co">## #</span>
<span class="co">## #</span>
<span class="co">## #=======================================</span>
<span class="co">## </span>
<span class="co">## P1                 1 T---------TT--------------------------------------      3</span>
<span class="co">##                      |         ||                                      </span>
<span class="co">## S1                 1 TAGTATGCAGTTAATATGATAATCATTAAAACTAAACCCCTGCGGGTAAA     50</span>
<span class="co">## </span>
<span class="co">## P1                 4 --------------------------------------------------      3</span>
<span class="co">##                                                                        </span>
<span class="co">## S1                51 GTGACTAAAATGTTGTCTTCTCAAATTCGCTAAAAACATCTGAGCGTAGC    100</span>
<span class="co">## </span>
<span class="co">## P1                 4 -----------------------CCTCC----------------------      8</span>
<span class="co">##                                             |||||                      </span>
<span class="co">## S1               101 GAAGCTAAAATCGCAAAACGAAGCCTCCCGTCAAAATTAATGAAAATTAG    150</span>
<span class="co">## </span>
<span class="co">## P1                 9 --TTG---------------------------------------------     11</span>
<span class="co">##                        |||                                             </span>
<span class="co">## S1               151 CTTTGCTTCGCGTTTTTAGCTTCGCCACACTCAACCAATCAAAAGCGTGA    200</span>
<span class="co">## </span>
<span class="co">## P1                12 ----------------------------ATTTCAATTTT-----CGA---     25</span>
<span class="co">##                                                  ||||  |||||     |||   </span>
<span class="co">## S1               201 AGCGAAGCTTCGCCTCAAAATGTCTCATATTTTTATTTTTGGCTCGAAGC    250</span>
<span class="co">## </span>
<span class="co">## P1                26 -----------------------------------CATTTT--------T     32</span>
<span class="co">##                                                         ||||||        |</span>
<span class="co">## S1               251 TTCGCTTCGCGTTTTTAGTTTCGCTACGTTCCAGCCATTTTAAGGTAAAT    300</span>
<span class="co">## </span>
<span class="co">## P1                33 GAAG------AATCTATTTGGTA---------------------------     49</span>
<span class="co">##                      |||       ||  |||| | ||                           </span>
<span class="co">## S1               301 GAAAATTTACAACTTATTGGCTACTTTCCCCGCATGAGTTAACTGAACTA    350</span>
<span class="co">## </span>
<span class="co">## P1                50 -     49</span>
<span class="co">##                       </span>
<span class="co">## S1               351 A    351</span>
<span class="co">## </span>
<span class="co">## </span>
<span class="co">## #---------------------------------------</span>
<span class="co">## #---------------------------------------</span></code></pre>
<div class="sourceCode" id="cb54"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># We confirm that flanking regions map well</span>
<span class="fu">pairwiseAlignment</span><span class="op">(</span><span class="va">genomes</span><span class="op">$</span><span class="va">Oki</span><span class="op">$</span><span class="va">chr1</span><span class="op">[</span><span class="fl">551988</span><span class="op">:</span><span class="fl">552647</span><span class="op">]</span>, <span class="va">genomes</span><span class="op">$</span><span class="va">Osa</span><span class="op">$</span><span class="va">Chr1</span><span class="op">[</span><span class="fl">1730510</span><span class="op">:</span><span class="fl">1731548</span><span class="op">]</span> |&gt; <span class="fu">reverseComplement</span><span class="op">(</span><span class="op">)</span>, type<span class="op">=</span><span class="st">"global"</span><span class="op">)</span> |&gt; <span class="fu">writePairwiseAlignments</span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## ########################################</span>
<span class="co">## # Program: Biostrings (version 2.62.0), a Bioconductor package</span>
<span class="co">## # Rundate: Wed Feb  9 10:24:39 2022</span>
<span class="co">## ########################################</span>
<span class="co">## #=======================================</span>
<span class="co">## #</span>
<span class="co">## # Aligned_sequences: 2</span>
<span class="co">## # 1: P1</span>
<span class="co">## # 2: S1</span>
<span class="co">## # Matrix: NA</span>
<span class="co">## # Gap_penalty: 14.0</span>
<span class="co">## # Extend_penalty: 4.0</span>
<span class="co">## #</span>
<span class="co">## # Length: 1040</span>
<span class="co">## # Identity:     510/1040 (49.0%)</span>
<span class="co">## # Similarity:    NA/1040 (NA%)</span>
<span class="co">## # Gaps:         381/1040 (36.6%)</span>
<span class="co">## # Score: -1752.284</span>
<span class="co">## #</span>
<span class="co">## #</span>
<span class="co">## #=======================================</span>
<span class="co">## </span>
<span class="co">## P1                 1 TAAGATTTTATAAGTCCGTTATTCA-------------ATAAAATTCATT     37</span>
<span class="co">##                      ||| |||||| || ||| |||||               |||||||| |  </span>
<span class="co">## S1                 1 TAAAATTTTAGAAATCCATTATTTTTTTTTGAAGTTTGATAAAATTAAAA     50</span>
<span class="co">## </span>
<span class="co">## P1                38 TATTTCCTACTGTACAGGCGTGGCAAAACGTCTTCTTAGCGCATTGTCCA     87</span>
<span class="co">##                            |||||| ||||| || ||||| |||||||| |||||||| ||| </span>
<span class="co">## S1                51 ATACAACTACTGAACAGGAGTAGCAAATCGTCTTCTGAGCGCATTATCCG    100</span>
<span class="co">## </span>
<span class="co">## P1                88 TCGTATTTTTCTTGTCGAT-CTTGAAGCTATCCGAGTTCTCCTGGAACTT    136</span>
<span class="co">##                      ||||  |||||  ||| || || | | || |||| ||| || || ||   </span>
<span class="co">## S1               101 TCGTGCTTTTCCCGTCAATTCTAGTA-CTCTCCGCGTTTTCTTGAAATCG    149</span>
<span class="co">## </span>
<span class="co">## P1               137 GTGAATGATCGCTGGGACAGGCAAAGCAAAGCATAACACGCCAGTTAGCG    186</span>
<span class="co">##                       ||||| ||||| || || ||||  |||||||  || |||||  | ||||</span>
<span class="co">## S1               150 ATGAATAATCGCCGGAACTGGCAGCGCAAAGCTCAAAACGCCGCTCAGCG    199</span>
<span class="co">## </span>
<span class="co">## P1               187 CACAGAAACAGCTAGAAATG---GTT------TTGCGAAATCAGCGGCCG    227</span>
<span class="co">##                      | ||| | |||||| ||||    |||      |||| ||||||| |||||</span>
<span class="co">## S1               200 CGCAGCAGCAGCTAAAAATCAAAGTTAGATTCTTGCAAAATCAGTGGCCG    249</span>
<span class="co">## </span>
<span class="co">## P1               228 CA------GCGACAA-ATGG-------------ATTAA------------    245</span>
<span class="co">##                      |       || | || |||              |||||            </span>
<span class="co">## S1               250 CGAAACATGCAAGAACATGTTTTGCTAAATCGAATTAAAACTAGAATTTC    299</span>
<span class="co">## </span>
<span class="co">## P1               246 --CTCACCCAATTACGCGACCGCCAGGGGTTGTCGGGTAAATGTCCCCGT    293</span>
<span class="co">##                        || |||| | ||  |||||||| |  |||||||| ||||  || ||||</span>
<span class="co">## S1               300 TGCTGACCCGACTATTCGACCGCCGGCAGTTGTCGGATAAAGATCGCCGT    349</span>
<span class="co">## </span>
<span class="co">## P1               294 AGCCGATTGTAGACATTGTTACGATCGCCCACCAAATTCCGTCT--AAAA    341</span>
<span class="co">##                      |||| |||||||||||||||| ||| |||||||||||||| |||  ||||</span>
<span class="co">## S1               350 AGCCAATTGTAGACATTGTTATGATTGCCCACCAAATTCCATCTTCAAAA    399</span>
<span class="co">## </span>
<span class="co">## P1               342 A---------GCAGTT-------------TT-----------CCTCC---    355</span>
<span class="co">##                      |         ||||||             ||           ||| |   </span>
<span class="co">## S1               400 AAGTTAGTATGCAGTTAATATGATAATCATTAAAACTAAACCCCTGCGGG    449</span>
<span class="co">## </span>
<span class="co">## P1               356 ----------------TTGATTTC---AATT-------------------    367</span>
<span class="co">##                                      |||  |||   ||||                   </span>
<span class="co">## S1               450 TAAAGTGACTAAAATGTTGTCTTCTCAAATTCGCTAAAAACATCTGAGCG    499</span>
<span class="co">## </span>
<span class="co">## P1               368 ---------------------------------TTCGACATTTTTGAAGA    384</span>
<span class="co">##                                                        || | |||  |||| |</span>
<span class="co">## S1               500 TAGCGAAGCTAAAATCGCAAAACGAAGCCTCCCGTCAAAATTAATGAAAA    549</span>
<span class="co">## </span>
<span class="co">## P1               385 ATCTATTTGGTACCTG----------------------------------    400</span>
<span class="co">##                       |   |||| | |  |                                  </span>
<span class="co">## S1               550 TTAGCTTTGCTTCGCGTTTTTAGCTTCGCCACACTCAACCAATCAAAAGC    599</span>
<span class="co">## </span>
<span class="co">## P1               401 --GAATTGAAG---------AAAATTTATCAT---------------TCG    424</span>
<span class="co">##                        |||  ||||         ||||| | ||||               |||</span>
<span class="co">## S1               600 GTGAAGCGAAGCTTCGCCTCAAAATGTCTCATATTTTTATTTTTGGCTCG    649</span>
<span class="co">## </span>
<span class="co">## P1               425 -----TCGCAT-----------TTTCG-------------------AGCT    439</span>
<span class="co">##                           |||| |           |||||                   || |</span>
<span class="co">## S1               650 AAGCTTCGCTTCGCGTTTTTAGTTTCGCTACGTTCCAGCCATTTTAAGGT    699</span>
<span class="co">## </span>
<span class="co">## P1               440 AAAAAACAA---ACA--TTTTT----------------------------    456</span>
<span class="co">##                      |||  | ||   |||  || ||                            </span>
<span class="co">## S1               700 AAATGAAAATTTACAACTTATTGGCTACTTTCCCCGCATGAGTTAACTGA    749</span>
<span class="co">## </span>
<span class="co">## P1               457 ---------GAAGAC------------CATTAGTCGAA----------GA    475</span>
<span class="co">##                               ||| ||            |||| || | |          ||</span>
<span class="co">## S1               750 ACTAACCCGGAATACAGCTAAATTTTTCATTCGTTGCATTTTCGAGCTGA    799</span>
<span class="co">## </span>
<span class="co">## P1               476 ATG----------CTC-----AAAAC-------GCATACCTGGTAAATTA    503</span>
<span class="co">##                      ||           |||     ||||        | ||||||| ||||| |</span>
<span class="co">## S1               800 ATCACTTTTGAACCTCTTTAGAAAATATTTCTAGAATACCTGATAAATAA    849</span>
<span class="co">## </span>
<span class="co">## P1               504 CGGAGCCGTAAAATATGCCGGAAATGGAGAGAAGAATGAGCAGAAATGAG    553</span>
<span class="co">##                      | ||||| ||||||||||| ||||||| ||||||||| |||| ||| |||</span>
<span class="co">## S1               850 CAGAGCCATAAAATATGCCAGAAATGGCGAGAAGAATCAGCAAAAACGAG    899</span>
<span class="co">## </span>
<span class="co">## P1               554 ATCT---AAAAGT--------CTTCAAAGAGCAC--TTT---ATAAG-GT    586</span>
<span class="co">##                      ||||   ||||||        |||  ||  ||     ||   ||||| | </span>
<span class="co">## S1               900 ATCTGAAAAAAGTAATTTGCGCTTTGAACCGCCGCGATTCCGATAAGCGG    949</span>
<span class="co">## </span>
<span class="co">## P1               587 CGA----------------TTTGAGCTTACTTCTCTGGTACTATCAGCCA    620</span>
<span class="co">##                      | |                || || |  || || |  || || ||  |||</span>
<span class="co">## S1               950 CCACAGCGGAACGGCAAAATTAGAACAAACCTCGCGAGTGCTTTCGCCCA    999</span>
<span class="co">## </span>
<span class="co">## P1               621 TCGTTTTTAGCATTTGGCAAAGAGACTCATTTTTCTTGCT    660</span>
<span class="co">##                      ||||||| | ||| || || |||||||| || || |||||</span>
<span class="co">## S1              1000 TCGTTTTGATCATGTGACACAGAGACTCGTTGTTTTTGCT   1039</span>
<span class="co">## </span>
<span class="co">## </span>
<span class="co">## #---------------------------------------</span>
<span class="co">## #---------------------------------------</span></code></pre>
<div class="sourceCode" id="cb56"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Only one hit</span>
<span class="co">#vmatchPattern(genomes$Osa$Chr1[1730795:1731145], genomes$Osa, with.indels = TRUE, max.mismatch = 10)</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="falsefalse">FALSE/FALSE<a class="anchor" aria-label="anchor" href="#falsefalse"></a>
</h2>
<div class="sourceCode" id="cb57"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ROI4_2</span> <span class="op">&lt;-</span> <span class="va">x</span><span class="op">[</span><span class="fl">5</span><span class="op">]</span>
<span class="co"># As character for convenience of BLAST searching.</span>
<span class="va">genomes</span><span class="op">$</span><span class="va">Oki</span><span class="op">$</span><span class="va">chr1</span><span class="op">[</span><span class="fl">554365</span><span class="op">:</span><span class="fl">554546</span><span class="op">]</span>   |&gt; <span class="fu"><a href="https://rdrr.io/r/base/character.html" class="external-link">as.character</a></span><span class="op">(</span><span class="op">)</span>  <span class="co"># Only one perfect BLAST hit at the NCBI, to the Oki genome.</span></code></pre></div>
<pre><code><span class="co">## [1] "AAAAGGCTGCTCTAGAATTTTTGATCAAGCGGCAATTATCATTTGCTTGAATTTTATGCCAAAACTTCCTTCTTTGCATTTCGCCGAAAACACTGATAAGAAACGTTTTCGGCAAAAGCGGAAAGTGTCAACTTCCGCGAAAGGCGCCGTTCAATTTTCTAGTGCCTTTTCGAGCCCATTAA"</span></code></pre>
<div class="sourceCode" id="cb59"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">genomes</span><span class="op">$</span><span class="va">Osa</span><span class="op">$</span><span class="va">Chr1</span><span class="op">[</span><span class="fl">1727927</span><span class="op">:</span><span class="fl">1728109</span><span class="op">]</span> |&gt; <span class="fu"><a href="https://rdrr.io/r/base/character.html" class="external-link">as.character</a></span><span class="op">(</span><span class="op">)</span>  <span class="co"># No relevant BLAST hit at the NCBI (evalues &gt;= 1e-4)</span></code></pre></div>
<pre><code><span class="co">## [1] "TCTCAGGATCGACATGCTTCTTAGAAACCACTGATTTTTCGAAGAAAGGTGCGAAATTTTAAGATCAAGAAAAATCTTAAATTCAACAATTTTTTTTGAAAATGGAAAAAAAAATTTTTGTCACAGCGCGATTGAACGCTAACCCTTCCCATTTCTTCAATCTTTTTCCGATACGAAGCATGT"</span></code></pre>
<div class="sourceCode" id="cb61"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">vmatchPattern</span><span class="op">(</span><span class="va">genomes</span><span class="op">$</span><span class="va">Oki</span><span class="op">$</span><span class="va">chr1</span><span class="op">[</span><span class="fl">554365</span><span class="op">:</span><span class="fl">554546</span><span class="op">]</span>, <span class="va">genomes</span><span class="op">$</span><span class="va">Oki</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## GRanges object with 1 range and 0 metadata columns:</span>
<span class="co">##       seqnames        ranges strand</span>
<span class="co">##          &lt;Rle&gt;     &lt;IRanges&gt;  &lt;Rle&gt;</span>
<span class="co">##   [1]     chr1 554365-554546      +</span>
<span class="co">##   -------</span>
<span class="co">##   seqinfo: 19 sequences from OKI2018_I69 genome</span></code></pre>
<div class="sourceCode" id="cb63"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># unaligned mapped region (+ / -)</span>
<span class="fu">pairwiseAlignment</span><span class="op">(</span><span class="va">genomes</span><span class="op">$</span><span class="va">Oki</span><span class="op">$</span><span class="va">chr1</span><span class="op">[</span><span class="fl">554365</span><span class="op">:</span><span class="fl">554546</span><span class="op">]</span>, <span class="va">genomes</span><span class="op">$</span><span class="va">Osa</span><span class="op">$</span><span class="va">Chr1</span><span class="op">[</span><span class="fl">1727927</span><span class="op">:</span><span class="fl">1728109</span><span class="op">]</span>                       , type<span class="op">=</span><span class="st">"global"</span><span class="op">)</span> |&gt; <span class="fu">writePairwiseAlignments</span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## ########################################</span>
<span class="co">## # Program: Biostrings (version 2.62.0), a Bioconductor package</span>
<span class="co">## # Rundate: Wed Feb  9 10:24:40 2022</span>
<span class="co">## ########################################</span>
<span class="co">## #=======================================</span>
<span class="co">## #</span>
<span class="co">## # Aligned_sequences: 2</span>
<span class="co">## # 1: P1</span>
<span class="co">## # 2: S1</span>
<span class="co">## # Matrix: NA</span>
<span class="co">## # Gap_penalty: 14.0</span>
<span class="co">## # Extend_penalty: 4.0</span>
<span class="co">## #</span>
<span class="co">## # Length: 201</span>
<span class="co">## # Identity:      96/201 (47.8%)</span>
<span class="co">## # Similarity:    NA/201 (NA%)</span>
<span class="co">## # Gaps:          37/201 (18.4%)</span>
<span class="co">## # Score: -488.9035</span>
<span class="co">## #</span>
<span class="co">## #</span>
<span class="co">## #=======================================</span>
<span class="co">## </span>
<span class="co">## P1                 1 AAA-AGGCTGCTCTAGAATTTTTGATCAAGCGGCAATTATC------ATT     43</span>
<span class="co">##                          ||| |   |  |  | || ||     | | | || ||      |  </span>
<span class="co">## S1                 1 TCTCAGGATCGACATGCTTCTTAGAAACCACTG-ATTTTTCGAAGAAAGG     49</span>
<span class="co">## </span>
<span class="co">## P1                44 TGCTTGAATTTTATG--CCAAAACTTCCTTCTTTGCA-----TTTCGCCG     86</span>
<span class="co">##                      |||   ||||||| |  | | ||    |||   | ||     |||    |</span>
<span class="co">## S1                50 TGCGA-AATTTTAAGATCAAGAAAAATCTTAAATTCAACAATTTTTTTTG     98</span>
<span class="co">## </span>
<span class="co">## P1                87 AAAACACTGATAAGAAACGTTTTCGGCAAAAGCGGAAAGTGTCAACTTCC    136</span>
<span class="co">##                      ||||   ||  || |||  |||| | || | |||  |    | |||    </span>
<span class="co">## S1                99 AAAA---TGGAAAAAAAAATTTTTGTCACA-GCGCGAT---TGAAC----    137</span>
<span class="co">## </span>
<span class="co">## P1               137 GCGAAAGGCGCCGTTCAATTTTCT--AGTGCCTTTTCGAGCCCAT---TA    181</span>
<span class="co">##                      || ||     || |||   |||||  | |   ||| |||  | |    | </span>
<span class="co">## S1               138 GCTAA-----CCCTTCCCATTTCTTCAATCTTTTTCCGATACGAAGCATG    182</span>
<span class="co">## </span>
<span class="co">## P1               182 A    182</span>
<span class="co">##                       </span>
<span class="co">## S1               183 T    183</span>
<span class="co">## </span>
<span class="co">## </span>
<span class="co">## #---------------------------------------</span>
<span class="co">## #---------------------------------------</span></code></pre>
<div class="sourceCode" id="cb65"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">pairwiseAlignment</span><span class="op">(</span><span class="va">genomes</span><span class="op">$</span><span class="va">Oki</span><span class="op">$</span><span class="va">chr1</span><span class="op">[</span><span class="fl">554365</span><span class="op">:</span><span class="fl">554546</span><span class="op">]</span>, <span class="va">genomes</span><span class="op">$</span><span class="va">Osa</span><span class="op">$</span><span class="va">Chr1</span><span class="op">[</span><span class="fl">1727927</span><span class="op">:</span><span class="fl">1728109</span><span class="op">]</span> |&gt; <span class="fu">reverseComplement</span><span class="op">(</span><span class="op">)</span>, type<span class="op">=</span><span class="st">"global"</span><span class="op">)</span> |&gt; <span class="fu">writePairwiseAlignments</span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## ########################################</span>
<span class="co">## # Program: Biostrings (version 2.62.0), a Bioconductor package</span>
<span class="co">## # Rundate: Wed Feb  9 10:24:40 2022</span>
<span class="co">## ########################################</span>
<span class="co">## #=======================================</span>
<span class="co">## #</span>
<span class="co">## # Aligned_sequences: 2</span>
<span class="co">## # 1: P1</span>
<span class="co">## # 2: S1</span>
<span class="co">## # Matrix: NA</span>
<span class="co">## # Gap_penalty: 14.0</span>
<span class="co">## # Extend_penalty: 4.0</span>
<span class="co">## #</span>
<span class="co">## # Length: 196</span>
<span class="co">## # Identity:     104/196 (53.1%)</span>
<span class="co">## # Similarity:    NA/196 (NA%)</span>
<span class="co">## # Gaps:          27/196 (13.8%)</span>
<span class="co">## # Score: -425.3516</span>
<span class="co">## #</span>
<span class="co">## #</span>
<span class="co">## #=======================================</span>
<span class="co">## </span>
<span class="co">## P1                 1 AAAAGGCTGCTCTAG---AATTTTTGATCAAGCGGCAA---TTATCATTT     44</span>
<span class="co">##                      | | |  |  | | |   ||   ||||  ||  || ||   ||| | || </span>
<span class="co">## S1                 1 ACATGCTTCGTATCGGAAAAAGATTGAAGAAATGGGAAGGGTTAGCGTTC     50</span>
<span class="co">## </span>
<span class="co">## P1                45 GCTTGAATTTTATGCCAAAACTTCCTTCTTTGCATTTCGCCGAAAACACT     94</span>
<span class="co">##                        | |   | |  | ||||| ||  || ||| |||||  |  |||| | |</span>
<span class="co">## S1                51 AATCGCGCTGT--GACAAAAATTTTTT-TTTCCATTTT-CAAAAAAAATT     96</span>
<span class="co">## </span>
<span class="co">## P1                95 GAT-AAGAAACGTTTTCGGCAAAAGCGGAAAGTGT--CAACTTCCG-CGA    140</span>
<span class="co">##                      | | ||   | | |||   |   | |  ||| | |  || ||| |  |||</span>
<span class="co">## S1                97 GTTGAATTTAAGATTTTT-CTTGATCTTAAAATTTCGCACCTTTCTTCGA    145</span>
<span class="co">## </span>
<span class="co">## P1               141 AAGGCGCCGTTCAAT--TTTCTA-GTGCCTTTTCGAGCCCATTA-A    182</span>
<span class="co">##                      ||        ||| |  |||||| |   | | |||| ||  | | |</span>
<span class="co">## S1               146 AAAA------TCAGTGGTTTCTAAGAAGCATGTCGATCC--TGAGA    183</span>
<span class="co">## </span>
<span class="co">## </span>
<span class="co">## #---------------------------------------</span>
<span class="co">## #---------------------------------------</span></code></pre>
<div class="sourceCode" id="cb67"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Identity of ~50% but no striking difference between strands</span>

<span class="co"># unaligned mapped region with flanking colinear alignments.</span>
<span class="va">gbs</span><span class="op">$</span><span class="va">Oki_Osa</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/findOverlaps-methods.html" class="external-link">subsetByOverlaps</a></span><span class="op">(</span><span class="va">ROI4_2</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## GBreaks object with 2 ranges and 8 metadata columns:</span>
<span class="co">##       seqnames        ranges strand |     score                query      Arm</span>
<span class="co">##          &lt;Rle&gt;     &lt;IRanges&gt;  &lt;Rle&gt; | &lt;numeric&gt;            &lt;GRanges&gt; &lt;factor&gt;</span>
<span class="co">##   [1]     chr1 554168-554365      - |       450 Chr1:1728109-1728321    short</span>
<span class="co">##   [2]     chr1 554546-554692      - |       564 Chr1:1727781-1727927    short</span>
<span class="co">##                   rep   repOvlp transcripts        flag    nonCoa</span>
<span class="co">##       &lt;CharacterList&gt; &lt;integer&gt;       &lt;Rle&gt; &lt;character&gt; &lt;logical&gt;</span>
<span class="co">##   [1]            &lt;NA&gt;         0        &lt;NA&gt;         Col     FALSE</span>
<span class="co">##   [2]            &lt;NA&gt;         0        &lt;NA&gt;         Col     FALSE</span>
<span class="co">##   -------</span>
<span class="co">##   seqinfo: 19 sequences from OKI2018_I69 genome</span></code></pre>
<div class="sourceCode" id="cb69"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">pairwiseAlignment</span><span class="op">(</span><span class="va">genomes</span><span class="op">$</span><span class="va">Oki</span><span class="op">$</span><span class="va">chr1</span><span class="op">[</span><span class="fl">554168</span><span class="op">:</span><span class="fl">554692</span><span class="op">]</span>, <span class="va">genomes</span><span class="op">$</span><span class="va">Osa</span><span class="op">$</span><span class="va">Chr1</span><span class="op">[</span><span class="fl">1727781</span><span class="op">:</span><span class="fl">1728321</span><span class="op">]</span> |&gt; <span class="fu">reverseComplement</span><span class="op">(</span><span class="op">)</span>, type<span class="op">=</span><span class="st">"global"</span><span class="op">)</span> |&gt; <span class="fu">writePairwiseAlignments</span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## ########################################</span>
<span class="co">## # Program: Biostrings (version 2.62.0), a Bioconductor package</span>
<span class="co">## # Rundate: Wed Feb  9 10:24:40 2022</span>
<span class="co">## ########################################</span>
<span class="co">## #=======================================</span>
<span class="co">## #</span>
<span class="co">## # Aligned_sequences: 2</span>
<span class="co">## # 1: P1</span>
<span class="co">## # 2: S1</span>
<span class="co">## # Matrix: NA</span>
<span class="co">## # Gap_penalty: 14.0</span>
<span class="co">## # Extend_penalty: 4.0</span>
<span class="co">## #</span>
<span class="co">## # Length: 558</span>
<span class="co">## # Identity:     395/558 (70.8%)</span>
<span class="co">## # Similarity:    NA/558 (NA%)</span>
<span class="co">## # Gaps:          50/558 (9.0%)</span>
<span class="co">## # Score: -303.8273</span>
<span class="co">## #</span>
<span class="co">## #</span>
<span class="co">## #=======================================</span>
<span class="co">## </span>
<span class="co">## P1                 1 TACCAATAAGCTTTCAGGCGCGCGTTGTAGATTTTTCTCTTGAGTTCTGA     50</span>
<span class="co">##                      ||||||||| ||||| |||||   ||| | |||||||||||| |||||||</span>
<span class="co">## S1                 1 TACCAATAAACTTTCTGGCGCTTTTTGCAAATTTTTCTCTTGTGTTCTGA     50</span>
<span class="co">## </span>
<span class="co">## P1                51 AAAGAACCCCTCCGATCTCGAAATCGATCATAATTGGAACTTTGTTTTTT    100</span>
<span class="co">##                      | | ||  ||||| || || ||||||| ||||||||||||| |||||  |</span>
<span class="co">## S1                51 ATAAAATGCCTCCAATTTCAAAATCGACCATAATTGGAACTCTGTTTACT    100</span>
<span class="co">## </span>
<span class="co">## P1               101 ATTTTACGAATGCTCATCTTTT-------TATTTTTTCGAAAATGGAATT    143</span>
<span class="co">##                      |||||  | |||||||| ||||       ||||||||| |   |||||||</span>
<span class="co">## S1               101 ATTTTTTGGATGCTCATTTTTTATAAGAATATTTTTTC-AGCGTGGAATT    149</span>
<span class="co">## </span>
<span class="co">## P1               144 TTTTTTTG-CAG------AAGGAAGGTT-TGAATTTTTCATTTTTGGCGG    185</span>
<span class="co">##                      |||      |||      ||||||| |  ||||||||| |||  ||||||</span>
<span class="co">## S1               150 TTTGCAGATCAGGAAAACAAGGAAGCTCGTGAATTTTTTATTCGTGGCGG    199</span>
<span class="co">## </span>
<span class="co">## P1               186 A-CAATGGGCTCAAAAAGGCTGCTCTAG---AATTTTTGATCAAGCGGCA    231</span>
<span class="co">##                      | |||  ||||| | | |  |  | | |   ||   ||||  ||  || |</span>
<span class="co">## S1               200 AACAAAAGGCTCGACATGCTTCGTATCGGAAAAAGATTGAAGAAATGGGA    249</span>
<span class="co">## </span>
<span class="co">## P1               232 A---TTATCATTTGCTTGAATTTTATGCCAAAACTTCCTTCTTTGCATTT    278</span>
<span class="co">##                      |   ||| | ||   | |     | || ||||| ||  || ||| |||||</span>
<span class="co">## S1               250 AGGGTTAGCGTTCAATCGC--GCTGTGACAAAAATTTTTT-TTTCCATTT    296</span>
<span class="co">## </span>
<span class="co">## P1               279 CGCCGAAAACACTGAT-AAGAAACGTTTTCGGCAAAAGCGGAAAGTGTC-    326</span>
<span class="co">##                        |  |||| | || | ||   | | |||   |   | |  ||| | || </span>
<span class="co">## S1               297 T-CAAAAAAAATTGTTGAATTTAAGATTTTT-CTTGATCTTAAAATTTCG    344</span>
<span class="co">## </span>
<span class="co">## P1               327 AAC--TTCCGCGAAAGGCGCCGTTCAAT--TTTCTA-GTGCCTTTTCGAG    371</span>
<span class="co">##                       ||  |||  |||||        ||| |  |||||| |   | | |||| </span>
<span class="co">## S1               345 CACCTTTCTTCGAAAAA------TCAGTGGTTTCTAAGAAGCATGTCGAT    388</span>
<span class="co">## </span>
<span class="co">## P1               372 CCCATTA-AGCTGCTGCAGCAGATGCATGGAACTACGTGTTTGACGACTT    420</span>
<span class="co">##                      ||  | | ||||  ||||||| || |||||||||||||  ||||||||||</span>
<span class="co">## S1               389 CC--TGAGAGCTTGTGCAGCAAATACATGGAACTACGTAATTGACGACTT    436</span>
<span class="co">## </span>
<span class="co">## P1               421 TGAAAAAGGGATTTCGGCCGCTGCTTTGAGTCGC---GCTGGACCGCGGG    467</span>
<span class="co">##                      |||||||||||||||||||| |   ||| ||||    |||||||||||||</span>
<span class="co">## S1               437 TGAAAAAGGGATTTCGGCCGGT---TTGCGTCGATGAGCTGGACCGCGGG    483</span>
<span class="co">## </span>
<span class="co">## P1               468 GCGTCGCCTGATCCGCTTTCGTTGATCTTCGCATTGCTCGACAGTTTGCG    517</span>
<span class="co">##                      ||||||||||||||||||||||||||||||||||||||||||||||||||</span>
<span class="co">## S1               484 GCGTCGCCTGATCCGCTTTCGTTGATCTTCGCATTGCTCGACAGTTTGCG    533</span>
<span class="co">## </span>
<span class="co">## P1               518 CTCCATCT    525</span>
<span class="co">##                      ||||||||</span>
<span class="co">## S1               534 CTCCATCT    541</span>
<span class="co">## </span>
<span class="co">## </span>
<span class="co">## #---------------------------------------</span>
<span class="co">## #---------------------------------------</span></code></pre>
</div>
<div class="section level2">
<h2 id="truetrue">TRUE/TRUE<a class="anchor" aria-label="anchor" href="#truetrue"></a>
</h2>
<div class="sourceCode" id="cb71"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ROI4_3</span> <span class="op">&lt;-</span> <span class="va">x</span><span class="op">[</span><span class="fl">12</span><span class="op">]</span>
<span class="va">ROI4_3</span> |&gt; <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">##   seqnames  start    end width strand query.seqnames query.start query.end</span>
<span class="co">## 1     chr1 561646 562124   479      *           Chr1     1722712   1723220</span>
<span class="co">##   query.width query.strand query.rep query.repOvlp       query.transcripts</span>
<span class="co">## 1         509            *       rnd           195 g383.t2;g383.t1;g383.t3</span>
<span class="co">##     Arm    rep repOvlp          transcripts flag     Class</span>
<span class="co">## 1 short tandem     158 g57.t3;g57.t1;g57.t2 &lt;NA&gt; TRUE TRUE</span></code></pre>
<div class="sourceCode" id="cb73"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># As character for convenience of BLAST searching.</span>
<span class="va">genomes</span><span class="op">$</span><span class="va">Oki</span><span class="op">$</span><span class="va">chr1</span><span class="op">[</span><span class="fl">561646</span><span class="op">:</span><span class="fl">562124</span><span class="op">]</span>   |&gt; <span class="fu"><a href="https://rdrr.io/r/base/character.html" class="external-link">as.character</a></span><span class="op">(</span><span class="op">)</span>  <span class="co"># Only one perfect BLAST hit at the NCBI, to the Oki genome.</span></code></pre></div>
<pre><code><span class="co">## [1] "TATCATAGCAATCGGTATTTTGGTAGAGATTAACGAGCGAAAAGGAACATCCAGTAGCTCTTCAGTCCCTTGAATCCTTCGTGAAATGTGGTTGTTGTTGTTCCTGACGAATTTGAAGTCGTTGTCTTAGTCGTTGTTGTTGTTGAACGTGTAGTGGTTTTTTTAGTGGTTGTTGTCGATGTCAATGATAATGTGGATGTTGTTGTTTTTGTTGTTTGCGTAGACATTGTTGTTGATGATTTTGAGGGTGCTGGCGTATTGGAAATAGAACTTGTCATCTGATAAGTCGACAGATTTGTCATGTCAGTCGTCAGGGATGATTCTGCCGTTGTTAGATTGTCAACACTCGGATCGAGATCTTCATTTTTTCCAAGGCAAAGAGGAATCGTGATTCCAAGAATTGTTACTATCACTGCAGAAACTTCATTTCACACCGAAATTGTTATTCAAATATAATTCTTAAATACCTAGCGCGAATC"</span></code></pre>
<div class="sourceCode" id="cb75"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">genomes</span><span class="op">$</span><span class="va">Osa</span><span class="op">$</span><span class="va">Chr1</span><span class="op">[</span><span class="fl">1722712</span><span class="op">:</span><span class="fl">1723220</span><span class="op">]</span> |&gt; <span class="fu"><a href="https://rdrr.io/r/base/character.html" class="external-link">as.character</a></span><span class="op">(</span><span class="op">)</span>  <span class="co"># No relevant BLAST hit at the NCBI (evalues &gt;= 1e-4)</span></code></pre></div>
<pre><code><span class="co">## [1] "GCTTGTTTATGAGTTAAAAAAAACTTTTTAATATATCTTTGCTTCAAAAGCTTGAATTAAGTTACGACGATTATTATTGTTATCCTCGGATGTATCCCGGTTTTTATCGAAAAAAACAATGAAATAACCACAAACTCGACAACAATGCCTGACTCAACAACAATTATCATCACTACAACGACTGTTCCTGACTTAATACTATTGAACGAAACTCTGACAACAACAACACTGACCTCGAAATCGAGCGCCACTTTGTCAACCACTACAAAAACTATTACAACCAACGCAACAACGACTTCCGCGACTACAACGACAACAACGAGTTCAACAACAACTACAACAACATCGACAACACCAACAACAACCGCGACCGGTTCTTCTACGACTACAACTCCAACCACAACATCAACAACATTGAACACAACGTCAACGACATCAACAACAACATCAAAATGTGTTCCATTTGACGAGTACGACGATATTTGTCCGTAATTTCTGGCAATTTGTCA"</span></code></pre>
<div class="sourceCode" id="cb77"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># unaligned mapped region (+ / -)</span>
<span class="fu">pairwiseAlignment</span><span class="op">(</span><span class="va">genomes</span><span class="op">$</span><span class="va">Oki</span><span class="op">$</span><span class="va">chr1</span><span class="op">[</span><span class="fl">561646</span><span class="op">:</span><span class="fl">562124</span><span class="op">]</span>, <span class="va">genomes</span><span class="op">$</span><span class="va">Osa</span><span class="op">$</span><span class="va">Chr1</span><span class="op">[</span><span class="fl">1722712</span><span class="op">:</span><span class="fl">1723220</span><span class="op">]</span>                       , type<span class="op">=</span><span class="st">"global"</span><span class="op">)</span> |&gt; <span class="fu">writePairwiseAlignments</span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## ########################################</span>
<span class="co">## # Program: Biostrings (version 2.62.0), a Bioconductor package</span>
<span class="co">## # Rundate: Wed Feb  9 10:24:41 2022</span>
<span class="co">## ########################################</span>
<span class="co">## #=======================================</span>
<span class="co">## #</span>
<span class="co">## # Aligned_sequences: 2</span>
<span class="co">## # 1: P1</span>
<span class="co">## # 2: S1</span>
<span class="co">## # Matrix: NA</span>
<span class="co">## # Gap_penalty: 14.0</span>
<span class="co">## # Extend_penalty: 4.0</span>
<span class="co">## #</span>
<span class="co">## # Length: 557</span>
<span class="co">## # Identity:     230/557 (41.3%)</span>
<span class="co">## # Similarity:    NA/557 (NA%)</span>
<span class="co">## # Gaps:         126/557 (22.6%)</span>
<span class="co">## # Score: -1473.951</span>
<span class="co">## #</span>
<span class="co">## #</span>
<span class="co">## #=======================================</span>
<span class="co">## </span>
<span class="co">## P1                 1 TATCATAGCAATCGGTATTTTGGTAGAGATTAACGAGCGAAAAGGAACAT     50</span>
<span class="co">##                             ||          |||       || | |||  ||||  ||| |</span>
<span class="co">## S1                 1 -------GC----------TTG-------TTTATGAGTTAAAAAAAACTT     26</span>
<span class="co">## </span>
<span class="co">## P1                51 CCAG---TAGCT---CTTCAGTCCCTTGAATCCTTCGTGAAATGTGGTTG     94</span>
<span class="co">##                             || ||   |||||    |||||||  |  || |     | || </span>
<span class="co">## S1                27 TTTAATATATCTTTGCTTCAAAAGCTTGAAT--TAAGTTACGAC-GATTA     73</span>
<span class="co">## </span>
<span class="co">## P1                95 TTGTTGTTCCTGACGAATTTGAAGTCGTTGTCTTAGTCGTTGTTGTTGTT    144</span>
<span class="co">##                      || |||||     || || |    ||   || ||  |||          |</span>
<span class="co">## S1                74 TTATTGTTATCCTCGGATGTA---TCCCGGTTTTTATCGAAAAAAACAAT    120</span>
<span class="co">## </span>
<span class="co">## P1               145 GAA-----CGTGTAGTGGTTTTTTTAGTGGTTGTTGTCGATGTCAATGAT    189</span>
<span class="co">##                      |||     |    | | |       | ||  ||   || |   |||| ||</span>
<span class="co">## S1               121 GAAATAACCACAAACTCGACAAC--AATGCCTGAC-TCAACAACAATTAT    167</span>
<span class="co">## </span>
<span class="co">## P1               190 AATGTGGA-------TGTTGTTGTTTTTGTTGTTTGCGTAGACATTGTTG    232</span>
<span class="co">##                       ||    |       ||||  ||  ||  |  | |     || | | |  </span>
<span class="co">## S1               168 CATCACTACAACGACTGTTCCTGACTTAATACTATTGAACGAAACTCT--    215</span>
<span class="co">## </span>
<span class="co">## P1               233 TTGATGATTTTGAGGGTGCTGGCGTATTGGAAATAGAACTTGTCATCTGA    282</span>
<span class="co">##                        ||  |     |   ||    || | | ||     |  |||||| |   </span>
<span class="co">## S1               216 --GACAACAACAACACTGACCTCGAAATCGAGCGCCACTTTGTCAACCAC    263</span>
<span class="co">## </span>
<span class="co">## P1               283 TAAGTCGACAGATTTGTCATGTCAGTCGTCAGGGATGATTCTGCCGTTGT    332</span>
<span class="co">##                      ||   | | |  | |  ||    |  |  ||  ||     || |||    </span>
<span class="co">## S1               264 TA---CAAAAACTATTACAACCAACGCAACAACGA-----CTTCCGCGAC    305</span>
<span class="co">## </span>
<span class="co">## P1               333 TAGATTGTCAACACTCGGATCGAGATCTTC------ATTTTTTCCAAGGC    376</span>
<span class="co">##                      || |  | |||||    | || | | |  |      |   |   |||  |</span>
<span class="co">## S1               306 TACAACGACAACAACGAGTTCAACAACAACTACAACAACATCGACAACAC    355</span>
<span class="co">## </span>
<span class="co">## P1               377 AAAGAGGAATCGTGA-------TTCCAAGAATTGT--TACTATCACTGCA    417</span>
<span class="co">##                       || |  || || ||       ||| | || |     | | | |||  ||</span>
<span class="co">## S1               356 CAACAACAACCGCGACCGGTTCTTCTACGACTACAACTCCAACCACAACA    405</span>
<span class="co">## </span>
<span class="co">## P1               418 GAAACTTCATTTCACAC--------CGAAAT-------------------    440</span>
<span class="co">##                        |||  ||||  ||||        ||| ||                   </span>
<span class="co">## S1               406 TCAACAACATTGAACACAACGTCAACGACATCAACAACAACATCAAAATG    455</span>
<span class="co">## </span>
<span class="co">## P1               441 TGTT--ATTCAA-----------ATATAATTCTTAAATACCTAGCGCGAA    477</span>
<span class="co">##                      ||||  |||  |           ||||   ||   |||  || ||   ||</span>
<span class="co">## S1               456 TGTTCCATTTGACGAGTACGACGATATTTGTCCGTAATTTCTGGC---AA    502</span>
<span class="co">## </span>
<span class="co">## P1               478 TC-----    479</span>
<span class="co">##                      |      </span>
<span class="co">## S1               503 TTTGTCA    509</span>
<span class="co">## </span>
<span class="co">## </span>
<span class="co">## #---------------------------------------</span>
<span class="co">## #---------------------------------------</span></code></pre>
<div class="sourceCode" id="cb79"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">pairwiseAlignment</span><span class="op">(</span><span class="va">genomes</span><span class="op">$</span><span class="va">Oki</span><span class="op">$</span><span class="va">chr1</span><span class="op">[</span><span class="fl">561646</span><span class="op">:</span><span class="fl">562124</span><span class="op">]</span>, <span class="va">genomes</span><span class="op">$</span><span class="va">Osa</span><span class="op">$</span><span class="va">Chr1</span><span class="op">[</span><span class="fl">1722712</span><span class="op">:</span><span class="fl">1723220</span><span class="op">]</span> |&gt; <span class="fu">reverseComplement</span><span class="op">(</span><span class="op">)</span>, type<span class="op">=</span><span class="st">"global"</span><span class="op">)</span> |&gt; <span class="fu">writePairwiseAlignments</span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## ########################################</span>
<span class="co">## # Program: Biostrings (version 2.62.0), a Bioconductor package</span>
<span class="co">## # Rundate: Wed Feb  9 10:24:41 2022</span>
<span class="co">## ########################################</span>
<span class="co">## #=======================================</span>
<span class="co">## #</span>
<span class="co">## # Aligned_sequences: 2</span>
<span class="co">## # 1: P1</span>
<span class="co">## # 2: S1</span>
<span class="co">## # Matrix: NA</span>
<span class="co">## # Gap_penalty: 14.0</span>
<span class="co">## # Extend_penalty: 4.0</span>
<span class="co">## #</span>
<span class="co">## # Length: 531</span>
<span class="co">## # Identity:     288/531 (54.2%)</span>
<span class="co">## # Similarity:    NA/531 (NA%)</span>
<span class="co">## # Gaps:          74/531 (13.9%)</span>
<span class="co">## # Score: -1002.236</span>
<span class="co">## #</span>
<span class="co">## #</span>
<span class="co">## #=======================================</span>
<span class="co">## </span>
<span class="co">## P1                 1 TATCATAGCAATCGGTATTTTGG-----TAGAGATTAACGAGCGAAAAGG     45</span>
<span class="co">##                      |  || |      |  |||  ||     ||  |    ||  |  ||| ||</span>
<span class="co">## S1                 1 TGACAAATTGCCAGAAATTACGGACAAATATCGTCGTACTCGTCAAATGG     50</span>
<span class="co">## </span>
<span class="co">## P1                46 AACA-----TCCAGTAGCTCTTCAGTCCCTTGAATCCTTC-GTGAAATGT     89</span>
<span class="co">##                      ||||     |   || | | || |   | ||||  | ||  ||  |||||</span>
<span class="co">## S1                51 AACACATTTTGATGTTGTTGTTGATGTCGTTGA--CGTTGTGTTCAATGT     98</span>
<span class="co">## </span>
<span class="co">## P1                90 GGTTGTTGTTGTTCCTGACGAATTTGAAGTCGTTGTCTTA---GTCGTTG    136</span>
<span class="co">##                       |||| ||||||   ||  |   ||| |||||| |    |   ||||  |</span>
<span class="co">## S1                99 TGTTGATGTTGTGGTTGGAG---TTGTAGTCGTAGAAGAACCGGTCGCGG    145</span>
<span class="co">## </span>
<span class="co">## P1               137 TTGTTGTTGAACGTGTAGTGGTTTTTTTAGTGGTTGTTG------TCGAT    180</span>
<span class="co">##                      |||||||||    ||| |  ||| || |||| |||||||      | | |</span>
<span class="co">## S1               146 TTGTTGTTGGTGTTGTCGATGTTGTTGTAGTTGTTGTTGAACTCGTTGTT    195</span>
<span class="co">## </span>
<span class="co">## P1               181 GTCAATGATAAT-GTGGATGTTGTTGTTTTTGTTGTTTGCG-TAGACATT    228</span>
<span class="co">##                      |||  || || | | ||| || ||||||   |||| |||   |||   ||</span>
<span class="co">## S1               196 GTCGTTG-TAGTCGCGGAAGTCGTTGTTGC-GTTGGTTGTAATAGTTTTT    243</span>
<span class="co">## </span>
<span class="co">## P1               229 GTTGTTGATGATTTTGAGGGTGCTGGCGTATTGGAAATA---GAACTTGT    275</span>
<span class="co">##                      || || | |||    | ||  |||  ||  || ||  |    |   ||||</span>
<span class="co">## S1               244 GTAGTGGTTGACAAAGTGGC-GCT--CGATTTCGAGGTCAGTGTTGTTGT    290</span>
<span class="co">## </span>
<span class="co">## P1               276 CATCTGA---TAAGTCGACAGATTT--GTCATGT-CAGTCGTC--AGGGA    317</span>
<span class="co">##                        || ||   |   || | ||  ||  |||| |  |||||||   || ||</span>
<span class="co">## S1               291 TGTCAGAGTTTCGTTCAATAGTATTAAGTCAGGAACAGTCGTTGTAGTGA    340</span>
<span class="co">## </span>
<span class="co">## P1               318 TGATTCTGCCGTTGTTAGATTGTCAA-CACTCGGATCGAGATC-------    359</span>
<span class="co">##                      ||||  |   |||||| ||  ||||  || |    ||||| |        </span>
<span class="co">## S1               341 TGATAATT--GTTGTT-GA--GTCAGGCATTGTTGTCGAGTTTGTGGTTA    385</span>
<span class="co">## </span>
<span class="co">## P1               360 -TTCATT-TTTTCCAAGGCAAAGAGGAATCGTGAT---TCCAAGAATTGT    404</span>
<span class="co">##                       |||||| ||||    |  |||    || || |||   ||| || ||   </span>
<span class="co">## S1               386 TTTCATTGTTTTTTTCGATAAA----AACCGGGATACATCCGAGGATAAC    431</span>
<span class="co">## </span>
<span class="co">## P1               405 TACTATCACTGCAGAAACTTCATT------TCACACCGAAATTGTTATTC    448</span>
<span class="co">##                       |  || |  |  | ||||| |||      |   |   |||    |||| </span>
<span class="co">## S1               432 AATAATAATCGTCGTAACTTAATTCAAGCTTTTGAAGCAAAGATATATTA    481</span>
<span class="co">## </span>
<span class="co">## P1               449 AAATATAATTCTTAAATACCTAGCGCGAATC    479</span>
<span class="co">##                      |||  |  || |||| |    | |   || |</span>
<span class="co">## S1               482 AAAAGTTTTTTTTAACTCATAAAC---AAGC    509</span>
<span class="co">## </span>
<span class="co">## </span>
<span class="co">## #---------------------------------------</span>
<span class="co">## #---------------------------------------</span></code></pre>
<div class="sourceCode" id="cb81"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># unaligned mapped region with flanking colinear alignments.</span>
<span class="va">gbs</span><span class="op">$</span><span class="va">Oki_Osa</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/findOverlaps-methods.html" class="external-link">subsetByOverlaps</a></span><span class="op">(</span><span class="va">ROI4_3</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## GBreaks object with 2 ranges and 8 metadata columns:</span>
<span class="co">##       seqnames        ranges strand |     score                query      Arm</span>
<span class="co">##          &lt;Rle&gt;     &lt;IRanges&gt;  &lt;Rle&gt; | &lt;numeric&gt;            &lt;GRanges&gt; &lt;factor&gt;</span>
<span class="co">##   [1]     chr1 561142-561646      - |      1387 Chr1:1723220-1723727    short</span>
<span class="co">##   [2]     chr1 562124-563743      - |      2869 Chr1:1720807-1722712    short</span>
<span class="co">##                   rep   repOvlp transcripts        flag    nonCoa</span>
<span class="co">##       &lt;CharacterList&gt; &lt;integer&gt;       &lt;Rle&gt; &lt;character&gt; &lt;logical&gt;</span>
<span class="co">##   [1]            &lt;NA&gt;         0        &lt;NA&gt;         Col     FALSE</span>
<span class="co">##   [2]   LowComplexity         0        &lt;NA&gt;         Col     FALSE</span>
<span class="co">##   -------</span>
<span class="co">##   seqinfo: 19 sequences from OKI2018_I69 genome</span></code></pre>
<div class="sourceCode" id="cb83"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">pairwiseAlignment</span><span class="op">(</span><span class="va">genomes</span><span class="op">$</span><span class="va">Oki</span><span class="op">$</span><span class="va">chr1</span><span class="op">[</span><span class="fl">561142</span><span class="op">:</span><span class="fl">563743</span><span class="op">]</span>, <span class="va">genomes</span><span class="op">$</span><span class="va">Osa</span><span class="op">$</span><span class="va">Chr1</span><span class="op">[</span><span class="fl">1720807</span><span class="op">:</span><span class="fl">1723727</span><span class="op">]</span> |&gt; <span class="fu">reverseComplement</span><span class="op">(</span><span class="op">)</span>, type<span class="op">=</span><span class="st">"global"</span><span class="op">)</span> |&gt; <span class="fu">writePairwiseAlignments</span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## ########################################</span>
<span class="co">## # Program: Biostrings (version 2.62.0), a Bioconductor package</span>
<span class="co">## # Rundate: Wed Feb  9 10:24:41 2022</span>
<span class="co">## ########################################</span>
<span class="co">## #=======================================</span>
<span class="co">## #</span>
<span class="co">## # Aligned_sequences: 2</span>
<span class="co">## # 1: P1</span>
<span class="co">## # 2: S1</span>
<span class="co">## # Matrix: NA</span>
<span class="co">## # Gap_penalty: 14.0</span>
<span class="co">## # Extend_penalty: 4.0</span>
<span class="co">## #</span>
<span class="co">## # Length: 3007</span>
<span class="co">## # Identity:    1893/3007 (63.0%)</span>
<span class="co">## # Similarity:    NA/3007 (NA%)</span>
<span class="co">## # Gaps:         491/3007 (16.3%)</span>
<span class="co">## # Score: -2607.775</span>
<span class="co">## #</span>
<span class="co">## #</span>
<span class="co">## #=======================================</span>
<span class="co">## </span>
<span class="co">## P1                 1 TTATTTTAGGTTACTAGCGCGGCGGCGAGTAATACTGATGCTGCTCTTCC     50</span>
<span class="co">##                      || ||||||||||||||||||||||||||| ||||||||||| |||||||</span>
<span class="co">## S1                 1 TTTTTTTAGGTTACTAGCGCGGCGGCGAGTGATACTGATGCTTCTCTTCC     50</span>
<span class="co">## </span>
<span class="co">## P1                51 TCGGTGCCATCGTCCTAACCTACACAGATTCCGTCCGCCTTCTGCGAGAA    100</span>
<span class="co">##                      |||| || ||| || | || |||||||||||  | || ||||||  ||||</span>
<span class="co">## S1                51 TCGGCGCGATCCTCTTCACTTACACAGATTCGCTGCGTCTTCTGAAAGAA    100</span>
<span class="co">## </span>
<span class="co">## P1               101 CAAGAAGAATCGAATCGCGCGAAAAGAGACCTAGAAAAGCACCCTTTATC    150</span>
<span class="co">##                      ||||||||| | ||  | || ||  | || || |||||||||||  | ||</span>
<span class="co">## S1               101 CAAGAAGAAACAAACAGAGCCAAGCGCGATCTTGAAAAGCACCCACTTTC    150</span>
<span class="co">## </span>
<span class="co">## P1               151 GAAAGCTGCC------AGTCCTCAGAGCAATGTCCTTTCGATGGATCACA    194</span>
<span class="co">##                      || | || |       |||||||| |   |||| |||   ||| ||||| </span>
<span class="co">## S1               151 GAGAACTTCTCCAAAAAGTCCTCAAA---ATGTGCTTCAAATGAATCACG    197</span>
<span class="co">## </span>
<span class="co">## P1               195 AAGTGCCTCTTACGGAATGCGCTTACAATTCGCCAAAACTCATTGGCCCG    244</span>
<span class="co">##                      ||  | | ||| |||||||||  |||||||| || |||||| | ||| ||</span>
<span class="co">## S1               198 AAACGGCGCTTCCGGAATGCGAGTACAATTCTCCGAAACTCGTCGGCGCG    247</span>
<span class="co">## </span>
<span class="co">## P1               245 CTGAAAACAGATCCGAAGTACATTCCGGAATGGGATGAAATTGAAGTTGG    294</span>
<span class="co">##                      ||  |||  ||  | || |  ||||| |||||||| ||||| |||  |||</span>
<span class="co">## S1               248 CTTCAAATCGACACAAAATTAATTCCAGAATGGGACGAAATCGAAACTGG    297</span>
<span class="co">## </span>
<span class="co">## P1               295 CGAGACAAAGAATGGAGGTTGTTGGCAACCGAGCGATTGCGCGCAGAGAC    344</span>
<span class="co">##                       || || |  || ||||| || |||||||||   || |||   ||||| |</span>
<span class="co">## S1               298 AGAAACGAGAAACGGAGGCTGCTGGCAACCGGCAGACTGCCACCAGAGGC    347</span>
<span class="co">## </span>
<span class="co">## P1               345 AGAATGTGGCGATTATCATTCCGTATAAAAATAGAGAGGAGCATCTGCGC    394</span>
<span class="co">##                      |||||||||||||||| |||||||| || ||| ||||||| || ||||||</span>
<span class="co">## S1               348 AGAATGTGGCGATTATTATTCCGTACAAGAATCGAGAGGAACACCTGCGC    397</span>
<span class="co">## </span>
<span class="co">## P1               395 GCCCTGCTGAATACATTGCATCCAATTCTTCAAAGACAAAACACCGCATA    444</span>
<span class="co">##                      ||  |||| |||||||||||||| | ||| ||  | || |||||||| ||</span>
<span class="co">## S1               398 GCGTTGCTTAATACATTGCATCCGACTCTGCAGCGGCAGAACACCGCGTA    447</span>
<span class="co">## </span>
<span class="co">## P1               445 CTGCATTTATGTGGGCGAGCAGCATGACGATGGCAGATTTAATAAAGGTT    494</span>
<span class="co">##                      |||||||||||||| |||||| |||||||||||||||||||| |||||| </span>
<span class="co">## S1               448 CTGCATTTATGTGGCCGAGCAACATGACGATGGCAGATTTAACAAAGGTC    497</span>
<span class="co">## </span>
<span class="co">## P1               495 TGTTTATTTTTATCATAGCAATCGGTATTTTGG-----TAGAGATTAACG    539</span>
<span class="co">##                       ||||  ||||  || |      |  |||  ||     ||  |    || </span>
<span class="co">## S1               498 AGTTTTATTTTGACAAATTGCCAGAAATTACGGACAAATATCGTCGTACT    547</span>
<span class="co">## </span>
<span class="co">## P1               540 AGCGAAAAGGAACA-----TCCAGTAGCTCTTCAGTCCCTTGAATCCTTC    584</span>
<span class="co">##                       |  ||| ||||||     |   || | | || |   | ||||  | || </span>
<span class="co">## S1               548 CGTCAAATGGAACACATTTTGATGTTGTTGTTGATGTCGTTGA--CGTTG    595</span>
<span class="co">## </span>
<span class="co">## P1               585 -GTGAAATGTGGTTGTTGTTGTTCCTGACGAATTTGAAGTCGTTGTCTTA    633</span>
<span class="co">##                       ||  ||||| |||| ||||||   ||  |   ||| |||||| |    |</span>
<span class="co">## S1               596 TGTTCAATGTTGTTGATGTTGTGGTTGGAG---TTGTAGTCGTAGAAGAA    642</span>
<span class="co">## </span>
<span class="co">## P1               634 ---GTCGTTGTTGTTGTTGAACGTGTAGTGGTTTTTTTAGTGGTTGTTG-    679</span>
<span class="co">##                         ||||  ||||||||||    ||| |  ||| || |||| ||||||| </span>
<span class="co">## S1               643 CCGGTCGCGGTTGTTGTTGGTGTTGTCGATGTTGTTGTAGTTGTTGTTGA    692</span>
<span class="co">## </span>
<span class="co">## P1               680 -----TCGATGTCAATGATAAT-GTGGATGTTGTTGTTTTTGTTGTTTGC    723</span>
<span class="co">##                           | | ||||  || || | | ||| || ||||||   |||| ||| </span>
<span class="co">## S1               693 ACTCGTTGTTGTCGTTG-TAGTCGCGGAAGTCGTTGTTGC-GTTGGTTGT    740</span>
<span class="co">## </span>
<span class="co">## P1               724 G-TAGACATTGTTGTTGATGATTTTGAGGGTGCTGGCGTATTGGAAATA-    771</span>
<span class="co">##                        |||   |||| || | |||    | ||  |||  ||  || ||  |  </span>
<span class="co">## S1               741 AATAGTTTTTGTAGTGGTTGACAAAGTGGC-GCT--CGATTTCGAGGTCA    787</span>
<span class="co">## </span>
<span class="co">## P1               772 --GAACTTGTCATCTGA---TAAGTCGACAGATTT--GTCATGT-CAGTC    813</span>
<span class="co">##                        |   ||||  || ||   |   || | ||  ||  |||| |  |||||</span>
<span class="co">## S1               788 GTGTTGTTGTTGTCAGAGTTTCGTTCAATAGTATTAAGTCAGGAACAGTC    837</span>
<span class="co">## </span>
<span class="co">## P1               814 GTC--AGGGATGATTCTGCCGTTGTTAGATTGTCAA-CACTCGGATCGAG    860</span>
<span class="co">##                      ||   || ||||||  |   |||||| ||  ||||  || |    |||||</span>
<span class="co">## S1               838 GTTGTAGTGATGATAATT--GTTGTT-GA--GTCAGGCATTGTTGTCGAG    882</span>
<span class="co">## </span>
<span class="co">## P1               861 ATC--------TTCATT-TTTTCCAAGGCAAAGAGGAATCGTGAT---TC    898</span>
<span class="co">##                       |         |||||| ||||    |  |||    || || |||   ||</span>
<span class="co">## S1               883 TTTGTGGTTATTTCATTGTTTTTTTCGATAAA----AACCGGGATACATC    928</span>
<span class="co">## </span>
<span class="co">## P1               899 CAAGAATTGTTACTATCACTGCAGAAACTTCATTTCAC------ACCGAA    942</span>
<span class="co">##                      | || ||    |  || |  |  | ||||| |||  |       |   ||</span>
<span class="co">## S1               929 CGAGGATAACAATAATAATCGTCGTAACTTAATTCAAGCTTTTGAAGCAA    978</span>
<span class="co">## </span>
<span class="co">## P1               943 ATTGTTATTCAAATATAATTCTTAAATACCTAGCGCGAATCCAACAATAA    992</span>
<span class="co">##                      |    |||| |||  |  || |||| |    | |   || || |  | ||</span>
<span class="co">## S1               979 AGATATATTAAAAAGTTTTTTTTAACTCATAAAC---AAGCCGATGAGAA   1025</span>
<span class="co">## </span>
<span class="co">## P1               993 TCCTTTTTCGTTTTCTCCAACCTCTTTTGTAAATGTTTGCAATATTATAT   1042</span>
<span class="co">##                       || |||||| |||||||| ||||||| | |||||||  || ||  |   </span>
<span class="co">## S1              1026 CCCGTTTTCGCTTTCTCCATCCTCTTT-GAAAATGTT--CAGTAAGA---   1069</span>
<span class="co">## </span>
<span class="co">## P1              1043 TCTTCGAGAAAGAATATTTATAAGTTTAATTAATGCTGAGTGAATATTCG   1092</span>
<span class="co">##                          | ||   ||| ||||| | |  |  ||   || ||           </span>
<span class="co">## S1              1070 ----CTAGTTTGAAAATTTACACGCGTCTTT---GCAGA-----------   1101</span>
<span class="co">## </span>
<span class="co">## P1              1093 TCTATAAACATTTTCTTGACACACTCACAAAAAATCCGTTTTCCTCGTCT   1142</span>
<span class="co">##                             |||||  ||||| ||  | ||||||||||| || | |||||||</span>
<span class="co">## S1              1102 -------ACATTAACTTGAAACTTTTACAAAAAATCCTTTCTTCTCGTCT   1144</span>
<span class="co">## </span>
<span class="co">## P1              1143 TTTGTCATGGTATCGAAACTGATGCGAAAAATTTTTGTC-----TTCGAA   1187</span>
<span class="co">##                       | |||||   || ||  ||  | |    ||||| | ||     ||| ||</span>
<span class="co">## S1              1145 GTGGTCAT---ATTGA--CTTTTTC----AATTTCTTTCCAAAATTCAAA   1185</span>
<span class="co">## </span>
<span class="co">## P1              1188 T---TTTATAGTTATTTT-TAAGCCATCTGTCAAGATCTGCGTTATTTTT   1233</span>
<span class="co">##                          ||||| |  | ||  ||||||| ||||||  |  |   ||  ||||</span>
<span class="co">## S1              1186 CGCATTTATTGAAAATTGCTAAGCCACCTGTCATAAAGTT--TTTATTTT   1233</span>
<span class="co">## </span>
<span class="co">## P1              1234 CACACGAGCGTTG-AGATAAA-TTTATGATGACATTCGAATTGCACTCGA   1281</span>
<span class="co">##                      ||||||||| ||  || |||| ||||| ||||||   |  |  | |||  </span>
<span class="co">## S1              1234 CACACGAGCCTTCCAGCTAAAATTTATTATGACAA--GTTTGTCTCTC--   1279</span>
<span class="co">## </span>
<span class="co">## P1              1282 CACATTTTTAATCCC-ATCTTC-GGCCA----TAAAACTGCCAAAAAGCC   1325</span>
<span class="co">##                        | || | | |||  ||| |  ||||     ||||| || ||||  | |</span>
<span class="co">## S1              1280 --CCTTATGATTCCTGATCCTGAGGCCCCTCGTAAAAATGTCAAAGCGGC   1327</span>
<span class="co">## </span>
<span class="co">## P1              1326 GCAATTTTTCGCCTACCATGGATGAGTCT-----------AGGAATTTTT   1364</span>
<span class="co">##                      |  |  ||     |  | ||  | |||||           |  |||||||</span>
<span class="co">## S1              1328 GAGAAATTCAAATTTGC-TGATTTAGTCTGAAAATTTCAAAACAATTTTT   1376</span>
<span class="co">## </span>
<span class="co">## P1              1365 TC--AATTCTGATCGACATTTTTAGGAGCTGTGATGAACTCTGCCTTCAA   1412</span>
<span class="co">##                      ||  || |  |||  |  |||| ||| ||| | ||||| |||||||| ||</span>
<span class="co">## S1              1377 TCCAAAATTAGATTTATTTTTTCAGGCGCTATTATGAATTCTGCCTTTAA   1426</span>
<span class="co">## </span>
<span class="co">## P1              1413 AGAAGTTCTCAAAGAACATGCCTACGATTGCGTTATCTTCCACGACGTCG   1462</span>
<span class="co">##                      |||||||||||||||||| | |||||||||||| ||||||||||||||||</span>
<span class="co">## S1              1427 AGAAGTTCTCAAAGAACACGACTACGATTGCGTGATCTTCCACGACGTCG   1476</span>
<span class="co">## </span>
<span class="co">## P1              1463 ACATGCTCCCCGAAGACGACCGAAACATTTACCAGTGCGAATCGAATCCC   1512</span>
<span class="co">##                      ||||||||||||||||||| || ||||| |||||||| ||||||||||| </span>
<span class="co">## S1              1477 ACATGCTCCCCGAAGACGATCGCAACATCTACCAGTGTGAATCGAATCCG   1526</span>
<span class="co">## </span>
<span class="co">## P1              1513 GTCCATCTTTCTCCCCTCATCGACAAGTTCGATTACAAACCTTATGGAAC   1562</span>
<span class="co">##                      || ||||||||||| || |||||||| |||||||||||||||||||||||</span>
<span class="co">## S1              1527 GTTCATCTTTCTCCGCTGATCGACAAATTCGATTACAAACCTTATGGAAC   1576</span>
<span class="co">## </span>
<span class="co">## P1              1563 CGATTTCGGAGGAATCACCATGTTGCGGCCGGAGCATTTTATCGCGGCAA   1612</span>
<span class="co">##                       |||||||| |||||||| ||| | |||||||| ||||| ||||| ||||</span>
<span class="co">## S1              1577 TGATTTCGGCGGAATCACGATGCTCCGGCCGGAACATTTCATCGCCGCAA   1626</span>
<span class="co">## </span>
<span class="co">## P1              1613 ATGGAATGAGCAATCTTTTCTGGGGCTGGGGACGAGAAGACGATGACATG   1662</span>
<span class="co">##                      ||||||||||||||||||||||||| |||||||| |||||||||||||||</span>
<span class="co">## S1              1627 ATGGAATGAGCAATCTTTTCTGGGGATGGGGACGCGAAGACGATGACATG   1676</span>
<span class="co">## </span>
<span class="co">## P1              1663 CAATTTCGAGTTGGCAGATCTCCCTTCAATGGTAA---------------   1697</span>
<span class="co">##                      || ||| |||||||  |||| || || ||||||||               </span>
<span class="co">## S1              1677 CAGTTTAGAGTTGGACGATCGCCGTTTAATGGTAAGAATGTTGCAAATGT   1726</span>
<span class="co">## </span>
<span class="co">## P1              1698 --------------------------------------------------   1697</span>
<span class="co">##                                                                        </span>
<span class="co">## S1              1727 CGCCAAATGTCTCAAAATGTCTTAAAAATTATATAATTATCATTGATTAT   1776</span>
<span class="co">## </span>
<span class="co">## P1              1698 ATTTTTTCAATG------------------CCAATGT-------------   1716</span>
<span class="co">##                      || || ||| ||                  | |||||             </span>
<span class="co">## S1              1777 ATATTATCATTGGAAGGCTTAAAATGACCTCAAATGTCTCAGAATGTCAC   1826</span>
<span class="co">## </span>
<span class="co">## P1              1717 --------------------------------------------------   1716</span>
<span class="co">##                                                                        </span>
<span class="co">## S1              1827 AAAATGTCCGCGAAAATCACCAAATGTCTTCTGAAAATTTGAAGACTGGA   1876</span>
<span class="co">## </span>
<span class="co">## P1              1717 --TTTTCTCATT--ATTTTTCTAGTGACAAAACCGGTTAATTACGAGCAG   1762</span>
<span class="co">##                        |||||||| |  || ||||||||||| || || || || ||||| |||</span>
<span class="co">## S1              1877 CTTTTTCTCAATGCATATTTCTAGTGACGAAGCCAGTAAACTACGACCAG   1926</span>
<span class="co">## </span>
<span class="co">## P1              1763 GCCCGTTACAAAATGATCCCTCATCAGCACGCCTGGATCTTCAGAAATTT   1812</span>
<span class="co">##                      || || |||||||||||||||||||||||||||||||||||| | || ||</span>
<span class="co">## S1              1927 GCTCGATACAAAATGATCCCTCATCAGCACGCCTGGATCTTCCGCAACTT   1976</span>
<span class="co">## </span>
<span class="co">## P1              1813 CAAAATCCGCGACTCAACAACAGATGTCCGCTTTTTGCCTCCAGAGTACT   1862</span>
<span class="co">##                      ||||||| | || || || ||||| ||||| ||  | || ||||| || |</span>
<span class="co">## S1              1977 CAAAATCAGAGATTCGACCACAGACGTCCGATTCCTTCCCCCAGAATATT   2026</span>
<span class="co">## </span>
<span class="co">## P1              1863 TAGGACCGGATTGACGAGACTAA---TCAATACTTCTATCAACTTGTG--   1907</span>
<span class="co">##                      |||||   |||| |  |  | ||   |||| |   | || || ||  |  </span>
<span class="co">## S1              2027 TAGGAAAAGATTTAAAAAGCAAAAAATCAAAAA--CAATAAAGTTAAGAA   2074</span>
<span class="co">## </span>
<span class="co">## P1              1908 ---------GAATATTGAATTCTTGAATCAAAAAT-----AGAATAA---   1940</span>
<span class="co">##                               |||| |  |||| |   |||| | |      | |||||   </span>
<span class="co">## S1              2075 AATTATCTAGAATTTATAATTTTATTATCAGAGAAGTTTGATAATAAGTT   2124</span>
<span class="co">## </span>
<span class="co">## P1              1941 ---------------------------------AAAAATCAATCCTCAAA   1957</span>
<span class="co">##                                                       |||  | |||| |  | </span>
<span class="co">## S1              2125 TTAACATTTACTTTTGATAAGCATTTTAACAGCAAATTTTAATCTTATAC   2174</span>
<span class="co">## </span>
<span class="co">## P1              1958 AGTCAAATACAAAGAGCGCTCAACAGTAGAAGGCCTTACCAGTGTCAACT   2007</span>
<span class="co">##                      |||||||||||||||||| || || || ||||| || || || || || |</span>
<span class="co">## S1              2175 AGTCAAATACAAAGAGCGATCGACTGTCGAAGGTCTCACGAGCGTGAATT   2224</span>
<span class="co">## </span>
<span class="co">## P1              2008 ACAAAAACCTCGGAACCGACCGAAAATCGCTCTTTACTCATCTCGACCTT   2057</span>
<span class="co">##                      ||||||||||    || || || |    |||||| |||||||||||| ||</span>
<span class="co">## S1              2225 ACAAAAACCTGCAGACGGATCGCAGCCGGCTCTTCACTCATCTCGACATT   2274</span>
<span class="co">## </span>
<span class="co">## P1              2058 GAATTGCGAGAACTCGTTGTTGATTCCCTCAAAACCGAATTCATCAGCTC   2107</span>
<span class="co">##                      ||| | || || ||||| ||||| || ||  ||||||||||||  ||| |</span>
<span class="co">## S1              2275 GAACTCCGCGAGCTCGTCGTTGACTCTCTTCAAACCGAATTCACAAGCCC   2324</span>
<span class="co">## </span>
<span class="co">## P1              2108 TGGATCAAGCCAAACTCTGATCGACACGACGAAAAAGGAGTGCAGCTATA   2157</span>
<span class="co">##                       ||      | |||  ||||| || |||||||||||||||||||||||  </span>
<span class="co">## S1              2325 CGG------CAAAATGCTGATTGATACGACGAAAAAGGAGTGCAGCTACG   2368</span>
<span class="co">## </span>
<span class="co">## P1              2158 TAAAGATGGAAAACACTAGAGTCTGCGAGGAGTATGGACATACGATGGTG   2207</span>
<span class="co">##                      | ||  ||||||||||  || ||||||| || |||||||| |||||||| </span>
<span class="co">## S1              2369 TTAAACTGGAAAACACGCGAATCTGCGAAGAATATGGACACACGATGGTT   2418</span>
<span class="co">## </span>
<span class="co">## P1              2208 CTGAAGTACTTGAGAAAATTGCTGCTGAGTTATGATGACGCGGTGAAAAA   2257</span>
<span class="co">##                      ||||| ||  | ||||||  |||| |||| |  || || |||   || ||</span>
<span class="co">## S1              2419 CTGAAATATCTCAGAAAACAGCTGTTGAGCTTCGAAGATGCGAAAAAGAA   2468</span>
<span class="co">## </span>
<span class="co">## P1              2258 GTGTGATGAACTAGGGTA---GGAA------------------CTT----   2282</span>
<span class="co">##                       ||||| || || |||||   ||||                  |||    </span>
<span class="co">## S1              2469 ATGTGACGAGCTCGGGTAAGAGGAAACTAAGCGCGAAGCGAAGCTTCGCG   2518</span>
<span class="co">## </span>
<span class="co">## P1              2283 -------TAAC-------AATAACTTGA----AAG---------------   2299</span>
<span class="co">##                             |||        ||||  ||||    |||               </span>
<span class="co">## S1              2519 TCAAAAGTAAATATTCGAAATATTTTGAGGCGAAGCTTCGCTGCGCTCAG   2568</span>
<span class="co">## </span>
<span class="co">## P1              2300 -------------------------------ATTGATC------------   2306</span>
<span class="co">##                                                     |||| ||            </span>
<span class="co">## S1              2569 CAGTTTTCAGCGAAATTTAAGTTGACAACTAATTGGTCAGTTGATCCGCA   2618</span>
<span class="co">## </span>
<span class="co">## P1              2307 -----------TCTGTTTTAGCTACATGTGCACTGGATTCGCAGAAGACT   2345</span>
<span class="co">##                                 | | ||| || |||||||| ||||||||||||||||| |</span>
<span class="co">## S1              2619 GGGGTTAACCTTATTTTTAAGATACATGTGTACTGGATTCGCAGAAGATT   2668</span>
<span class="co">## </span>
<span class="co">## P1              2346 CTCTCGGAAAATACCGTCTTAGGGAGGTCACACAGCTTTTGTCAAGCGAT   2395</span>
<span class="co">##                      |||  |||||||| || ||  | || ||||| |||||| | || ||||| </span>
<span class="co">## S1              2669 CTCCGGGAAAATATCGCCTCCGAGAAGTCACGCAGCTTCTTTCTAGCGAC   2718</span>
<span class="co">## </span>
<span class="co">## P1              2396 AATACTCCTCAGGTAAGTGACTTTTCTAGCAACGAATAATTACTCGAAAC   2445</span>
<span class="co">##                      ||||| ||  | |||| | |   ||||  |  |    ||||| |  |   </span>
<span class="co">## S1              2719 AATACGCCAAACGTAA-TCAAAATTCTC-CTTCTCGAAATTATT--ATCT   2764</span>
<span class="co">## </span>
<span class="co">## P1              2446 TTCAGGATTGCAACACGAAGGGAAAGCACATCTTCCACAAATCCTGCCCA   2495</span>
<span class="co">##                      || |||||||||| ||||| ||||| ||||| | ||||||||| || || </span>
<span class="co">## S1              2765 TTTAGGATTGCAATACGAAAGGAAACCACATTTACCACAAATCATGTCCT   2814</span>
<span class="co">## </span>
<span class="co">## P1              2496 GGAGACCAATCATTTATCCAGGTTTCTCAAGATTTGACGATTCCAATTTC   2545</span>
<span class="co">##                      || ||||| ||||| || || |||||  | ||| | || ||||||||  |</span>
<span class="co">## S1              2815 GGCGACCAGTCATTCATTCAAGTTTCCGAGGATCTTACAATTCCAATCGC   2864</span>
<span class="co">## </span>
<span class="co">## P1              2546 TGATCAGCTGACACCGCCAAATTATGCTTTCACACATCGACTACAAGAAT   2595</span>
<span class="co">##                        |||||| ||||||||| ||||| ||||| |||||||| || ||||| |</span>
<span class="co">## S1              2865 AAATCAGCAGACACCGCCGAATTACGCTTTTACACATCGTCTTCAAGATT   2914</span>
<span class="co">## </span>
<span class="co">## P1              2596 GAGTTTC   2602</span>
<span class="co">##                      |||||||</span>
<span class="co">## S1              2915 GAGTTTC   2921</span>
<span class="co">## </span>
<span class="co">## </span>
<span class="co">## #---------------------------------------</span>
<span class="co">## #---------------------------------------</span></code></pre>
</div>
</div>
<div class="section level1">
<h1 id="multiple-inversions-in-a-row-roi7">Multiple inversions in a row (ROI7)<a class="anchor" aria-label="anchor" href="#multiple-inversions-in-a-row-roi7"></a>
</h1>
<div class="sourceCode" id="cb85"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ROI7</span> <span class="op">&lt;-</span> <span class="va">coa</span><span class="op">$</span><span class="va">Oki_Osa</span> |&gt; <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/flagInversions.html" class="external-link">flagInversions</a></span><span class="op">(</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/dist2next.html" class="external-link">dist2next</a></span><span class="op">(</span>ignore.strand <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>

<span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/showInversions.html" class="external-link">showInversions</a></span><span class="op">(</span><span class="va">ROI7</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## GBreaks object with 1826 ranges and 11 metadata columns:</span>
<span class="co">##        seqnames            ranges strand |                query     score</span>
<span class="co">##           &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; |            &lt;GRanges&gt; &lt;integer&gt;</span>
<span class="co">##     16     chr1     592887-593096      - | Chr1:1309140-1309328       210</span>
<span class="co">##     17     chr1     593196-593292      + | Chr1:1308820-1308912        97</span>
<span class="co">##     18     chr1     594240-603127      - | Chr1:1296981-1308607      8888</span>
<span class="co">##     37     chr1     741910-744560      + | Chr1:2036766-2039472      2651</span>
<span class="co">##     38     chr1     744937-745142      - | Chr1:2039897-2040113       206</span>
<span class="co">##    ...      ...               ...    ... .                  ...       ...</span>
<span class="co">##   8642      XSR 11853592-11853924      + |  XSR:8431875-8432251       333</span>
<span class="co">##   8643      XSR 11854037-11855928      - |  XSR:8429198-8431487      1892</span>
<span class="co">##   8695      XSR 12324700-12329600      - |  XSR:8992784-8997031      4901</span>
<span class="co">##   8696      XSR 12331614-12331778      + |  XSR:8990804-8990961       165</span>
<span class="co">##   8697      XSR 12335389-12337572      - |  XSR:8987052-8988968      2184</span>
<span class="co">##             Arm                    rep   repOvlp transcripts        flag</span>
<span class="co">##        &lt;factor&gt;        &lt;CharacterList&gt; &lt;integer&gt;       &lt;Rle&gt; &lt;character&gt;</span>
<span class="co">##     16    short                   &lt;NA&gt;         0        &lt;NA&gt;         Inv</span>
<span class="co">##     17    short                   &lt;NA&gt;         0        &lt;NA&gt;        &lt;NA&gt;</span>
<span class="co">##     18    short tandem,unknown,rnd,...        31      g68.t1        &lt;NA&gt;</span>
<span class="co">##     37    short                 tandem        68        &lt;NA&gt;         Inv</span>
<span class="co">##     38    short                   &lt;NA&gt;         0        &lt;NA&gt;        &lt;NA&gt;</span>
<span class="co">##    ...      ...                    ...       ...         ...         ...</span>
<span class="co">##   8642      XSR                   &lt;NA&gt;         0   g16813.t1        &lt;NA&gt;</span>
<span class="co">##   8643      XSR                 tandem         0        &lt;NA&gt;        &lt;NA&gt;</span>
<span class="co">##   8695      XSR     tandem,unknown,rnd        26   g16922.t1         Inv</span>
<span class="co">##   8696      XSR                   &lt;NA&gt;         0        &lt;NA&gt;        &lt;NA&gt;</span>
<span class="co">##   8697      XSR                 tandem         0   g16922.t1        &lt;NA&gt;</span>
<span class="co">##           nonCoa   inv     tdist     qdist</span>
<span class="co">##        &lt;logical&gt; &lt;Rle&gt; &lt;numeric&gt; &lt;numeric&gt;</span>
<span class="co">##     16      TRUE  TRUE       100       228</span>
<span class="co">##     17      TRUE FALSE       948       213</span>
<span class="co">##     18     FALSE FALSE     29392        NA</span>
<span class="co">##     37      TRUE  TRUE       377       425</span>
<span class="co">##     38      TRUE FALSE       440       387</span>
<span class="co">##    ...       ...   ...       ...       ...</span>
<span class="co">##   8642      TRUE FALSE       113       388</span>
<span class="co">##   8643     FALSE FALSE        93   1180869</span>
<span class="co">##   8695     FALSE  TRUE      2014      1823</span>
<span class="co">##   8696      TRUE FALSE      3611      1836</span>
<span class="co">##   8697     FALSE FALSE      1165       356</span>
<span class="co">##   -------</span>
<span class="co">##   seqinfo: 19 sequences from OKI2018_I69 genome</span></code></pre>
<div class="sourceCode" id="cb87"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ROI7</span><span class="op">[</span><span class="fl">35</span><span class="op">:</span><span class="fl">45</span><span class="op">]</span> |&gt; <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">##    seqnames  start    end width strand query.seqnames query.start query.end</span>
<span class="co">## 1      chr1 736824 738412  1589      +           Chr1     2033491   2036651</span>
<span class="co">## 2      chr1 738882 741671  2790      +           Chr1     2155668   2159253</span>
<span class="co">## 3      chr1 741910 744560  2651      +           Chr1     2036766   2039472</span>
<span class="co">## 4      chr1 744937 745142   206      -           Chr1     2039897   2040113</span>
<span class="co">## 5      chr1 745582 745862   281      +           Chr1     2040500   2040775</span>
<span class="co">## 6      chr1 747713 747961   249      -           Chr1     2041371   2041628</span>
<span class="co">## 7      chr1 750585 751341   757      +           Chr1     2042328   2043943</span>
<span class="co">## 8      chr1 752126 752262   137      -           Chr1     2044495   2044628</span>
<span class="co">## 9      chr1 753366 768703 15338      +           Chr1     2045148   2050583</span>
<span class="co">## 10     chr1 768953 773691  4739      -           Chr1     4442048   4448098</span>
<span class="co">## 11     chr1 776248 779526  3279      -           Chr1     3333268   3337597</span>
<span class="co">##    query.width query.strand    query.rep query.repOvlp</span>
<span class="co">## 1         3161            * rnd, tan....          1634</span>
<span class="co">## 2         3586            * rnd, unknown           557</span>
<span class="co">## 3         2707            *       tandem           160</span>
<span class="co">## 4          217            *           NA             0</span>
<span class="co">## 5          276            *           NA             0</span>
<span class="co">## 6          258            *           NA             0</span>
<span class="co">## 7         1616            * LowCompl....           152</span>
<span class="co">## 8          134            *           NA             0</span>
<span class="co">## 9         5436            * rnd, Low....           768</span>
<span class="co">## 10        6051            *          rnd          1052</span>
<span class="co">## 11        4330            *  rnd, tandem          1147</span>
<span class="co">##             query.transcripts score   Arm          rep repOvlp     transcripts</span>
<span class="co">## 1                     g456.t1  1589 short          rnd      98            &lt;NA&gt;</span>
<span class="co">## 2                     g492.t1  2790 short           NA       0         g106.t1</span>
<span class="co">## 3                     g457.t1  2651 short       tandem      68            &lt;NA&gt;</span>
<span class="co">## 4                        &lt;NA&gt;   206 short           NA       0            &lt;NA&gt;</span>
<span class="co">## 5                        &lt;NA&gt;   281 short           NA       0            &lt;NA&gt;</span>
<span class="co">## 6                        &lt;NA&gt;   249 short           NA       0            &lt;NA&gt;</span>
<span class="co">## 7                        &lt;NA&gt;   757 short           NA       0            &lt;NA&gt;</span>
<span class="co">## 8                        &lt;NA&gt;   137 short           NA       0            &lt;NA&gt;</span>
<span class="co">## 9             g458.t1;g459.t1 15338 short tandem, ....    3169            &lt;NA&gt;</span>
<span class="co">## 10 g1051.t1;g1052.t1;g1053.t1  4739 short      unknown       0 g111.t1;g112.t1</span>
<span class="co">## 11                    g750.t1  3279 short tandem, ....      77         g115.t1</span>
<span class="co">##    flag nonCoa   inv tdist   qdist</span>
<span class="co">## 1   Tra  FALSE FALSE   470  119017</span>
<span class="co">## 2  &lt;NA&gt;  FALSE FALSE   239  116196</span>
<span class="co">## 3   Inv   TRUE  TRUE   377     425</span>
<span class="co">## 4  &lt;NA&gt;   TRUE FALSE   440     387</span>
<span class="co">## 5   Inv   TRUE  TRUE  1851     596</span>
<span class="co">## 6  &lt;NA&gt;   TRUE FALSE  2624     700</span>
<span class="co">## 7   Inv  FALSE  TRUE   785     552</span>
<span class="co">## 8  &lt;NA&gt;   TRUE FALSE  1104     520</span>
<span class="co">## 9  &lt;NA&gt;  FALSE FALSE   250 2391465</span>
<span class="co">## 10 &lt;NA&gt;  FALSE FALSE  2557 1104451</span>
<span class="co">## 11 &lt;NA&gt;  FALSE FALSE   240 2506121</span></code></pre>
<div class="sourceCode" id="cb89"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ROI7</span><span class="op">[</span><span class="fl">37</span><span class="op">:</span><span class="fl">43</span><span class="op">]</span> |&gt; <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/plotApairOfChrs.html" class="external-link">plotApairOfChrs</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<p><img src="RegionsOfInterest_files/figure-html/unnamed-chunk-1-1.png" width="700"></p>
<div class="sourceCode" id="cb90"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## Annotations</span>
<span class="va">ROI7.reps</span> <span class="op">&lt;-</span> <span class="va">reps</span><span class="op">$</span><span class="va">Oki</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/findOverlaps-methods.html" class="external-link">subsetByOverlaps</a></span><span class="op">(</span><span class="va">ROI7</span><span class="op">[</span><span class="fl">37</span><span class="op">:</span><span class="fl">43</span><span class="op">]</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/inter-range-methods.html" class="external-link">range</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span>
<span class="va">mids</span> <span class="op">&lt;-</span> <span class="va">ROI7.reps</span> |&gt; <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/gr2dna_seg.html" class="external-link">gr2dna_seg</a></span><span class="op">(</span><span class="op">)</span> |&gt; <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>mid <span class="op">=</span> <span class="op">(</span><span class="va">start</span> <span class="op">+</span> <span class="va">end</span><span class="op">)</span> <span class="op">/</span> <span class="fl">2</span><span class="op">)</span> |&gt; <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/pull.html" class="external-link">pull</a></span><span class="op">(</span><span class="va">mid</span><span class="op">)</span>
<span class="va">text</span> <span class="op">&lt;-</span> <span class="va">ROI7.reps</span><span class="op">$</span><span class="va">Class</span> |&gt; <span class="fu"><a href="https://rdrr.io/r/base/character.html" class="external-link">as.character</a></span><span class="op">(</span><span class="op">)</span>
<span class="va">annot</span> <span class="op">&lt;-</span> <span class="fu">genoPlotR</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/genoPlotR/man/annotation.html" class="external-link">annotation</a></span><span class="op">(</span>x1<span class="op">=</span><span class="va">mids</span>, text<span class="op">=</span><span class="va">text</span>, rot<span class="op">=</span><span class="fl">30</span><span class="op">)</span>
<span class="va">ROI7</span><span class="op">[</span><span class="fl">37</span><span class="op">:</span><span class="fl">43</span><span class="op">]</span> |&gt; <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/plotApairOfChrs.html" class="external-link">plotApairOfChrs</a></span><span class="op">(</span>annotations <span class="op">=</span> <span class="va">annot</span><span class="op">)</span></code></pre></div>
<p><img src="RegionsOfInterest_files/figure-html/unnamed-chunk-1-2.png" width="700"></p>
<div class="sourceCode" id="cb91"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## Genes</span>
<span class="va">ROI7.genes</span> <span class="op">&lt;-</span> <span class="va">transcripts</span><span class="op">$</span><span class="va">Oki</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/findOverlaps-methods.html" class="external-link">subsetByOverlaps</a></span><span class="op">(</span><span class="va">ROI7</span><span class="op">[</span><span class="fl">37</span><span class="op">:</span><span class="fl">43</span><span class="op">]</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/inter-range-methods.html" class="external-link">range</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span>
<span class="va">mids</span> <span class="op">&lt;-</span> <span class="va">ROI7.genes</span> |&gt; <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/gr2dna_seg.html" class="external-link">gr2dna_seg</a></span><span class="op">(</span><span class="op">)</span> |&gt; <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>mid <span class="op">=</span> <span class="op">(</span><span class="va">start</span> <span class="op">+</span> <span class="va">end</span><span class="op">)</span> <span class="op">/</span> <span class="fl">2</span><span class="op">)</span> |&gt; <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/pull.html" class="external-link">pull</a></span><span class="op">(</span><span class="va">mid</span><span class="op">)</span>
<span class="va">text</span> <span class="op">&lt;-</span> <span class="va">ROI7.genes</span><span class="op">$</span><span class="va">tx_name</span> |&gt; <span class="fu"><a href="https://rdrr.io/r/base/character.html" class="external-link">as.character</a></span><span class="op">(</span><span class="op">)</span>
<span class="va">annot</span> <span class="op">&lt;-</span> <span class="fu">genoPlotR</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/genoPlotR/man/annotation.html" class="external-link">annotation</a></span><span class="op">(</span>x1<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/stats/start.html" class="external-link">start</a></span><span class="op">(</span><span class="va">ROI7.genes</span><span class="op">)</span>, x2<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/stats/start.html" class="external-link">end</a></span><span class="op">(</span><span class="va">ROI7.genes</span><span class="op">)</span>, text<span class="op">=</span><span class="va">text</span>, rot<span class="op">=</span><span class="fl">30</span><span class="op">)</span>
<span class="va">ROI7</span><span class="op">[</span><span class="fl">37</span><span class="op">:</span><span class="fl">43</span><span class="op">]</span> |&gt; <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/plotApairOfChrs.html" class="external-link">plotApairOfChrs</a></span><span class="op">(</span>annotations <span class="op">=</span> <span class="va">annot</span><span class="op">)</span></code></pre></div>
<p><img src="RegionsOfInterest_files/figure-html/unnamed-chunk-1-3.png" width="700"></p>
</div>
<div class="section level1">
<h1 id="double-inversions">Double inversions<a class="anchor" aria-label="anchor" href="#double-inversions"></a>
</h1>
<p>In these examples, the inversions happened on the <em>target</em> genome, therefore only in this genome the names of the ranges were changed.</p>
<div class="sourceCode" id="cb92"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">exampleDoubleInversion</span> |&gt; <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/plotApairOfChrs.html" class="external-link">plotApairOfChrs</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<p><img src="RegionsOfInterest_files/figure-html/ROI8.T-1.png" width="700"></p>
<div class="sourceCode" id="cb93"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">x0</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/GBreaks-class.html" class="external-link">GBreaks</a></span><span class="op">(</span>target <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"T:1:+"</span>, <span class="st">"T:2:+"</span>, <span class="st">"T:3:+"</span><span class="op">)</span><span class="op">)</span>,
              query  <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Q:1"</span>,   <span class="st">"Q:2"</span>,   <span class="st">"Q:3"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>

<span class="va">x1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/GBreaks-class.html" class="external-link">GBreaks</a></span><span class="op">(</span>target <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"T:1:-"</span>, <span class="st">"T:2:-"</span>, <span class="st">"T:3:+"</span><span class="op">)</span><span class="op">)</span>,
              query  <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Q:2"</span>,   <span class="st">"Q:1"</span>,   <span class="st">"Q:3"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>

<span class="va">x2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/GBreaks-class.html" class="external-link">GBreaks</a></span><span class="op">(</span>target <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"T:1:-"</span>, <span class="st">"T:2:+"</span>, <span class="st">"T:3:-"</span><span class="op">)</span><span class="op">)</span>,
              query  <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Q:2"</span>,   <span class="st">"Q:3"</span>,   <span class="st">"Q:1"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>

<span class="va">y0</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/GBreaks-class.html" class="external-link">GBreaks</a></span><span class="op">(</span>target <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>A<span class="op">=</span><span class="st">"T:10-15:+"</span>, B<span class="op">=</span><span class="st">"T:20-25:+"</span>, C<span class="op">=</span><span class="st">"T:30-35:+"</span><span class="op">)</span><span class="op">)</span>,
              query  <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>A<span class="op">=</span><span class="st">"Q:10-15"</span>,   B<span class="op">=</span><span class="st">"Q:20-25"</span>,   C<span class="op">=</span><span class="st">"Q:30-35"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>

<span class="va">y1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/GBreaks-class.html" class="external-link">GBreaks</a></span><span class="op">(</span>target <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>A<span class="op">=</span><span class="st">"T:10-15:-"</span>, B<span class="op">=</span><span class="st">"T:20-25:-"</span>, C<span class="op">=</span><span class="st">"T:30-35:+"</span><span class="op">)</span><span class="op">)</span>,
              query  <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>B<span class="op">=</span><span class="st">"Q:20-25"</span>,   A<span class="op">=</span><span class="st">"Q:10-15"</span>,   C<span class="op">=</span><span class="st">"Q:30-35"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>

<span class="va">y2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/GBreaks-class.html" class="external-link">GBreaks</a></span><span class="op">(</span>target <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>A<span class="op">=</span><span class="st">"T:10-15:-"</span>, B<span class="op">=</span><span class="st">"T:20-25:+"</span>, C<span class="op">=</span><span class="st">"T:30-35:-"</span><span class="op">)</span><span class="op">)</span>,
              query  <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>B<span class="op">=</span><span class="st">"Q:20-25"</span>,   C<span class="op">=</span><span class="st">"Q:30-35"</span>,   A<span class="op">=</span><span class="st">"Q:10-15"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>


<span class="va">a0</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/GBreaks-class.html" class="external-link">GBreaks</a></span><span class="op">(</span>target <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>`ABC+`<span class="op">=</span><span class="st">"T:10-39:+"</span><span class="op">)</span><span class="op">)</span>,
              query  <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>`ABC+`<span class="op">=</span><span class="st">"Q:10-39"</span>  <span class="op">)</span><span class="op">)</span><span class="op">)</span>

<span class="va">a1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/GBreaks-class.html" class="external-link">GBreaks</a></span><span class="op">(</span>target <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>`AB+`<span class="op">=</span><span class="st">"T:10-29:-"</span>, `C+`<span class="op">=</span><span class="st">"T:30-39:+"</span><span class="op">)</span><span class="op">)</span>,
              query  <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>`BA-`<span class="op">=</span><span class="st">"Q:10-29"</span>,   `C+`<span class="op">=</span><span class="st">"Q:30-39"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>

<span class="va">a2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/GBreaks-class.html" class="external-link">GBreaks</a></span><span class="op">(</span>target <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>`A+`<span class="op">=</span><span class="st">"T:10-19:+"</span>, `B+`<span class="op">=</span><span class="st">"T:20-29:-"</span>, `C+`<span class="op">=</span><span class="st">"T:30-39:-"</span><span class="op">)</span><span class="op">)</span>,
              query  <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>`A+`<span class="op">=</span><span class="st">"Q:30-39"</span>,   `B-`<span class="op">=</span><span class="st">"Q:10-19"</span>,   `C-`<span class="op">=</span><span class="st">"Q:20-29"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p>Let’s start with a simpler example, where the inversions happen on the <em>query</em> genome.</p>
<div class="sourceCode" id="cb94"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">exampleDoubleInversion</span> |&gt; <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/plotApairOfChrs.html" class="external-link">plotApairOfChrs</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<p><img src="RegionsOfInterest_files/figure-html/ROI8.Q-1.png" width="700"></p>
<div class="sourceCode" id="cb95"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Start colinear</span>
<span class="va">z0</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/GBreaks-class.html" class="external-link">GBreaks</a></span><span class="op">(</span>target <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>A<span class="op">=</span><span class="st">"T:10-15:+"</span>, B<span class="op">=</span><span class="st">"T:20-25:+"</span>, C<span class="op">=</span><span class="st">"T:30-35:+"</span><span class="op">)</span><span class="op">)</span>,
              query  <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>A<span class="op">=</span><span class="st">"Q:10-15"</span>,   B<span class="op">=</span><span class="st">"Q:20-25"</span>,   C<span class="op">=</span><span class="st">"Q:30-35"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>

<span class="co"># Swap coordinates of A and B on the query and flip strands</span>
<span class="va">z1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/GBreaks-class.html" class="external-link">GBreaks</a></span><span class="op">(</span>target <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>A<span class="op">=</span><span class="st">"T:10-15:-"</span>, B<span class="op">=</span><span class="st">"T:20-25:-"</span>, C<span class="op">=</span><span class="st">"T:30-35:+"</span><span class="op">)</span><span class="op">)</span>,
              query  <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>A<span class="op">=</span><span class="st">"Q:20-25"</span>,   B<span class="op">=</span><span class="st">"Q:10-15"</span>,   C<span class="op">=</span><span class="st">"Q:30-35"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>

<span class="co"># Now query order is B - A - C.  Swap A and C and flip strands</span>
<span class="va">z2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/GBreaks-class.html" class="external-link">GBreaks</a></span><span class="op">(</span>target <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>A<span class="op">=</span><span class="st">"T:10-15:+"</span>, B<span class="op">=</span><span class="st">"T:20-25:-"</span>, C<span class="op">=</span><span class="st">"T:30-35:-"</span><span class="op">)</span><span class="op">)</span>,
              query  <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>A<span class="op">=</span><span class="st">"Q:30-35"</span>,   B<span class="op">=</span><span class="st">"Q:10-15"</span>,   C<span class="op">=</span><span class="st">"Q:20-25"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>

<span class="co"># We have the same configuration, plus flanking regions, in exampleDoubleInversion</span>
<span class="va">exampleDoubleInversion</span></code></pre></div>
<pre><code><span class="co">## GBreaks object with 5 ranges and 1 metadata column:</span>
<span class="co">##       seqnames    ranges strand |        query</span>
<span class="co">##          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; |    &lt;GRanges&gt;</span>
<span class="co">##   [1]     chrA   100-190      + | chrB:100-190</span>
<span class="co">##   [2]     chrA   200-290      + | chrB:400-490</span>
<span class="co">##   [3]     chrA   300-390      - | chrB:200-290</span>
<span class="co">##   [4]     chrA   400-490      - | chrB:300-390</span>
<span class="co">##   [5]     chrA   500-590      + | chrB:500-590</span>
<span class="co">##   -------</span>
<span class="co">##   seqinfo: 1 sequence from an unspecified genome</span></code></pre>
<div class="sourceCode" id="cb97"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ROI8</span> <span class="op">&lt;-</span> <span class="va">coa2</span><span class="op">$</span><span class="va">Oki_Osa</span><span class="op">[</span><span class="fl">1420</span><span class="op">:</span><span class="fl">1430</span><span class="op">]</span>
<span class="va">ROI8</span> |&gt; <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/flagAll.html" class="external-link">flagAll</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## GBreaks object with 11 ranges and 6 metadata columns:</span>
<span class="co">##        seqnames          ranges strand |                query      Arm</span>
<span class="co">##           &lt;Rle&gt;       &lt;IRanges&gt;  &lt;Rle&gt; |            &lt;GRanges&gt; &lt;factor&gt;</span>
<span class="co">##    [1]     chr1 7925162-7925707      - | Chr1:5457033-5457486     long</span>
<span class="co">##    [2]     chr1 7926105-7926315      + | Chr1:5456632-5456843     long</span>
<span class="co">##    [3]     chr1 7926445-7935682      - | Chr1:5450903-5456340     long</span>
<span class="co">##    [4]     chr1 7935883-7936233      + | Chr1:5450455-5450759     long</span>
<span class="co">##    [5]     chr1 7936510-7937713      - | Chr1:5449692-5450249     long</span>
<span class="co">##    [6]     chr1 7937859-7938193      + | Chr1:5448891-5449199     long</span>
<span class="co">##    [7]     chr1 7938294-7938871      + | Chr1:5446199-5446704     long</span>
<span class="co">##    [8]     chr1 7938916-7939612      - | Chr1:5446876-5447698     long</span>
<span class="co">##    [9]     chr1 7940004-7940220      + | Chr1:5445781-5445992     long</span>
<span class="co">##   [10]     chr1 7940630-7940807      - | Chr1:5447947-5448116     long</span>
<span class="co">##   [11]     chr1 7941410-7942546      + | Chr1:5444239-5445410     long</span>
<span class="co">##                       rep   repOvlp       transcripts        flag</span>
<span class="co">##           &lt;CharacterList&gt; &lt;integer&gt;             &lt;Rle&gt; &lt;character&gt;</span>
<span class="co">##    [1]               &lt;NA&gt;         0          g2110.t1         Inv</span>
<span class="co">##    [2]               &lt;NA&gt;         0              &lt;NA&gt;        &lt;NA&gt;</span>
<span class="co">##    [3] unknown,rnd,tandem      1107 g2111.t1;g2111.t2         Inv</span>
<span class="co">##    [4]               &lt;NA&gt;         0              &lt;NA&gt;        &lt;NA&gt;</span>
<span class="co">##    [5]               &lt;NA&gt;         0 g2111.t1;g2111.t2        &lt;NA&gt;</span>
<span class="co">##    [6]               &lt;NA&gt;         0              &lt;NA&gt;        &lt;NA&gt;</span>
<span class="co">##    [7]               &lt;NA&gt;         0              &lt;NA&gt;        &lt;NA&gt;</span>
<span class="co">##    [8]               &lt;NA&gt;         0 g2111.t1;g2111.t2        &lt;NA&gt;</span>
<span class="co">##    [9]               &lt;NA&gt;         0              &lt;NA&gt;        &lt;NA&gt;</span>
<span class="co">##   [10]               &lt;NA&gt;         0              &lt;NA&gt;        &lt;NA&gt;</span>
<span class="co">##   [11]               &lt;NA&gt;         0              &lt;NA&gt;        &lt;NA&gt;</span>
<span class="co">##   -------</span>
<span class="co">##   seqinfo: 19 sequences from OKI2018_I69 genome</span></code></pre>
<div class="sourceCode" id="cb99"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ROI8</span> |&gt; <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/plotApairOfChrs.html" class="external-link">plotApairOfChrs</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<p><img src="RegionsOfInterest_files/figure-html/ROI8.Q-2.png" width="700"></p>
<div class="sourceCode" id="cb100"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ROI8</span> |&gt; <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/reverse.html" class="external-link">reverse</a></span><span class="op">(</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/r/base/sort.html" class="external-link">sort</a></span><span class="op">(</span>i<span class="op">=</span><span class="cn">T</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/plotApairOfChrs.html" class="external-link">plotApairOfChrs</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<p><img src="RegionsOfInterest_files/figure-html/ROI8.Q-3.png" width="700"></p>
<div class="sourceCode" id="cb101"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ROI8</span> |&gt; <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/reverse.html" class="external-link">reverse</a></span><span class="op">(</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/r/base/sort.html" class="external-link">sort</a></span><span class="op">(</span>i<span class="op">=</span><span class="cn">T</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/flagAll.html" class="external-link">flagAll</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## GBreaks object with 11 ranges and 6 metadata columns:</span>
<span class="co">##        seqnames          ranges strand |                query      Arm</span>
<span class="co">##           &lt;Rle&gt;       &lt;IRanges&gt;  &lt;Rle&gt; |            &lt;GRanges&gt; &lt;factor&gt;</span>
<span class="co">##    [1]     chr1 6590477-6591613      - | Chr1:5444239-5445410     long</span>
<span class="co">##    [2]     chr1 6592216-6592393      + | Chr1:5447947-5448116     long</span>
<span class="co">##    [3]     chr1 6592803-6593019      - | Chr1:5445781-5445992     long</span>
<span class="co">##    [4]     chr1 6593411-6594107      + | Chr1:5446876-5447698     long</span>
<span class="co">##    [5]     chr1 6594152-6594729      - | Chr1:5446199-5446704     long</span>
<span class="co">##    [6]     chr1 6594830-6595164      - | Chr1:5448891-5449199     long</span>
<span class="co">##    [7]     chr1 6595310-6596513      + | Chr1:5449692-5450249     long</span>
<span class="co">##    [8]     chr1 6596790-6597140      - | Chr1:5450455-5450759     long</span>
<span class="co">##    [9]     chr1 6597341-6606578      + | Chr1:5450903-5456340     long</span>
<span class="co">##   [10]     chr1 6606708-6606918      - | Chr1:5456632-5456843     long</span>
<span class="co">##   [11]     chr1 6607316-6607861      + | Chr1:5457033-5457486     long</span>
<span class="co">##                       rep   repOvlp       transcripts        flag</span>
<span class="co">##           &lt;CharacterList&gt; &lt;integer&gt;             &lt;Rle&gt; &lt;character&gt;</span>
<span class="co">##    [1]               &lt;NA&gt;         0              &lt;NA&gt;        &lt;NA&gt;</span>
<span class="co">##    [2]               &lt;NA&gt;         0              &lt;NA&gt;        &lt;NA&gt;</span>
<span class="co">##    [3]               &lt;NA&gt;         0              &lt;NA&gt;        &lt;NA&gt;</span>
<span class="co">##    [4]               &lt;NA&gt;         0 g2111.t1;g2111.t2        &lt;NA&gt;</span>
<span class="co">##    [5]               &lt;NA&gt;         0              &lt;NA&gt;        &lt;NA&gt;</span>
<span class="co">##    [6]               &lt;NA&gt;         0              &lt;NA&gt;        &lt;NA&gt;</span>
<span class="co">##    [7]               &lt;NA&gt;         0 g2111.t1;g2111.t2         Inv</span>
<span class="co">##    [8]               &lt;NA&gt;         0              &lt;NA&gt;        &lt;NA&gt;</span>
<span class="co">##    [9] unknown,rnd,tandem      1107 g2111.t1;g2111.t2         Inv</span>
<span class="co">##   [10]               &lt;NA&gt;         0              &lt;NA&gt;        &lt;NA&gt;</span>
<span class="co">##   [11]               &lt;NA&gt;         0          g2110.t1        &lt;NA&gt;</span>
<span class="co">##   -------</span>
<span class="co">##   seqinfo: 19 sequences from OKI2018_I69 genome</span></code></pre>
</div>
<div class="section level1">
<h1 id="filling-inversions-to-better-see-the-rest-roi9">Filling inversions to better see the rest (ROI9)<a class="anchor" aria-label="anchor" href="#filling-inversions-to-better-see-the-rest-roi9"></a>
</h1>
<p>We will not fill inversions in objects we analyse, because we do not want to erase their breakpoints.</p>
<div class="sourceCode" id="cb103"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">fillInversions</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">gb</span><span class="op">)</span> <span class="op">{</span>
  <span class="va">Invs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/which.html" class="external-link">which</a></span><span class="op">(</span><span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/flagInversions.html" class="external-link">flagInversions</a></span><span class="op">(</span><span class="va">gb</span><span class="op">)</span><span class="op">$</span><span class="va">inv</span><span class="op">)</span> <span class="op">+</span> <span class="fl">1</span>
  <span class="fu">strand</span><span class="op">(</span><span class="va">gb</span><span class="op">)</span><span class="op">[</span><span class="va">Invs</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html" class="external-link">ifelse</a></span><span class="op">(</span><span class="fu">strand</span><span class="op">(</span><span class="va">gb</span><span class="op">)</span><span class="op">[</span><span class="va">Invs</span><span class="op">]</span> <span class="op">==</span> <span class="st">"+"</span>, <span class="st">"-"</span>, <span class="st">"+"</span><span class="op">)</span>
  <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/coalesce_contigs.html" class="external-link">coalesce_contigs</a></span><span class="op">(</span><span class="va">gb</span><span class="op">)</span>
<span class="op">}</span>

<span class="va">ROI9.all</span> <span class="op">&lt;-</span> <span class="fu">fillInversions</span><span class="op">(</span><span class="va">coa2</span><span class="op">$</span><span class="va">Oki_Kum</span><span class="op">)</span>

<span class="va">ROI9</span> <span class="op">&lt;-</span> <span class="va">ROI9.all</span> |&gt;
  <span class="fu">plyranges</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/plyranges/man/tidyverse-reexports.html" class="external-link">filter</a></span><span class="op">(</span><span class="fu">seqnames</span><span class="op">(</span><span class="va">query</span><span class="op">)</span> <span class="op">==</span> <span class="st">"contig_3_1"</span><span class="op">)</span> |&gt;
  <span class="fu">plyranges</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/plyranges/man/tidyverse-reexports.html" class="external-link">arrange</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/start.html" class="external-link">start</a></span><span class="op">(</span><span class="va">query</span><span class="op">)</span><span class="op">)</span> |&gt;
  <span class="fu">plyranges</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/plyranges/man/tidyverse-reexports.html" class="external-link">slice</a></span><span class="op">(</span><span class="fl">10</span><span class="op">:</span><span class="fl">20</span><span class="op">)</span>

<span class="va">ROI9.range</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/inter-range-methods.html" class="external-link">range</a></span><span class="op">(</span><span class="va">ROI9</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>

<span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/plotApairOfChrs.html" class="external-link">plotApairOfChrs</a></span><span class="op">(</span><span class="va">ROI9</span><span class="op">)</span></code></pre></div>
<p><img src="RegionsOfInterest_files/figure-html/unnamed-chunk-2-1.png" width="700"></p>
<div class="sourceCode" id="cb104"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/plotApairOfChrs.html" class="external-link">plotApairOfChrs</a></span><span class="op">(</span><span class="va">ROI9.all</span>, xlim <span class="op">=</span> <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/gb2xlim.html" class="external-link">gb2xlim</a></span><span class="op">(</span><span class="va">ROI9.range</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="RegionsOfInterest_files/figure-html/unnamed-chunk-2-2.png" width="700"></p>
<div class="sourceCode" id="cb105"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/findOverlaps-methods.html" class="external-link">subsetByOverlaps</a></span><span class="op">(</span><span class="va">ROI9.all</span>, <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/genomic-range-squeezers.html" class="external-link">granges</a></span><span class="op">(</span><span class="va">ROI9.range</span><span class="op">)</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/flagAll.html" class="external-link">flagAll</a></span><span class="op">(</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">##    seqnames    start      end  width strand query.seqnames query.start</span>
<span class="co">## 1      chr1  8831594  8904792  73199      +     contig_3_1      781316</span>
<span class="co">## 2      chr1  8904796  8905391    596      -    contig_87_1      129264</span>
<span class="co">## 3      chr1  8907966  9178393 270428      +     contig_3_1      857220</span>
<span class="co">## 4      chr1  9179294  9179883    590      -    contig_28_1     2686583</span>
<span class="co">## 5      chr1  9180258  9238680  58423      +     contig_3_1     1103683</span>
<span class="co">## 6      chr1  9238754  9238914    161      -    contig_11_1       64537</span>
<span class="co">## 7      chr1  9247203  9756276 509074      +     contig_3_1     1170666</span>
<span class="co">## 8      chr1  9756672 10487037 730366      +     contig_3_1     1665784</span>
<span class="co">## 9      chr1 10489186 10490144    959      -    contig_82_1      469347</span>
<span class="co">## 10     chr1 10490223 10579632  89410      +     contig_3_1     2335682</span>
<span class="co">## 11     chr1 10581313 10582002    690      +    contig_42_1    12447665</span>
<span class="co">## 12     chr1 10583588 10707107 123520      +     contig_3_1     2422786</span>
<span class="co">## 13     chr1 10707650 10708228    579      -    contig_43_1      542809</span>
<span class="co">## 14     chr1 10709490 10826652 117163      +     contig_3_1     2509221</span>
<span class="co">## 15     chr1 10829074 10829372    299      +    contig_28_1     4919962</span>
<span class="co">## 16     chr1 10830830 10831185    356      -    contig_28_1     4917700</span>
<span class="co">## 17     chr1 10831349 10832281    933      +    contig_28_1     4921734</span>
<span class="co">## 18     chr1 10832293 11220548 388256      +     contig_3_1     2614209</span>
<span class="co">## 19     chr1 11223351 11224198    848      -    contig_90_1      515726</span>
<span class="co">## 20     chr1 11224199 11224556    358      +    contig_29_1     2413024</span>
<span class="co">## 21     chr1 11225607 11565085 339479      +     contig_3_1     2973842</span>
<span class="co">##    query.end query.width query.strand  score flag</span>
<span class="co">## 1     857219       75904            *  73199  Tra</span>
<span class="co">## 2     129861         598            *    596 &lt;NA&gt;</span>
<span class="co">## 3    1103682      246463            * 270428  Tra</span>
<span class="co">## 4    2687173         591            *    590 &lt;NA&gt;</span>
<span class="co">## 5    1169654       65972            *  58423  Tra</span>
<span class="co">## 6      64695         159            *    161 &lt;NA&gt;</span>
<span class="co">## 7    1648012      477347            * 509074  Col</span>
<span class="co">## 8    2335681      669898            * 730366  Tra</span>
<span class="co">## 9     470292         946            *    959 &lt;NA&gt;</span>
<span class="co">## 10   2422784       87103            *  89410  Tra</span>
<span class="co">## 11  12448361         697            *    690 &lt;NA&gt;</span>
<span class="co">## 12   2509220       86435            * 123520  Tra</span>
<span class="co">## 13    543388         580            *    579 &lt;NA&gt;</span>
<span class="co">## 14   2614206      104986            * 117163 &lt;NA&gt;</span>
<span class="co">## 15   4920260         299            *    299  Tra</span>
<span class="co">## 16   4918055         356            *    356 &lt;NA&gt;</span>
<span class="co">## 17   4922665         932            *    933 &lt;NA&gt;</span>
<span class="co">## 18   2973841      359633            * 388256 &lt;NA&gt;</span>
<span class="co">## 19    516573         848            *    848 &lt;NA&gt;</span>
<span class="co">## 20   2413382         359            *    358 &lt;NA&gt;</span>
<span class="co">## 21   3316589      342748            * 339479 &lt;NA&gt;</span></code></pre>
<div class="sourceCode" id="cb107"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/findOverlaps-methods.html" class="external-link">subsetByOverlaps</a></span><span class="op">(</span><span class="va">ROI9.all</span> |&gt; <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/swap.html" class="external-link">swap</a></span><span class="op">(</span><span class="op">)</span>, <span class="va">ROI9.range</span><span class="op">$</span><span class="va">query</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## GBreaks object with 11 ranges and 1 metadata column:</span>
<span class="co">##          seqnames          ranges strand |                  query</span>
<span class="co">##             &lt;Rle&gt;       &lt;IRanges&gt;  &lt;Rle&gt; |              &lt;GRanges&gt;</span>
<span class="co">##    [1] contig_3_1   781316-857219      + |   chr1:8831594-8904792</span>
<span class="co">##    [2] contig_3_1  857220-1103682      + |   chr1:8907966-9178393</span>
<span class="co">##    [3] contig_3_1 1103683-1169654      + |   chr1:9180258-9238680</span>
<span class="co">##    [4] contig_3_1 1170666-1648012      + |   chr1:9247203-9756276</span>
<span class="co">##    [5] contig_3_1 1665784-2335681      + |  chr1:9756672-10487037</span>
<span class="co">##    [6] contig_3_1 2335682-2422784      + | chr1:10490223-10579632</span>
<span class="co">##    [7] contig_3_1 2422786-2509220      + | chr1:10583588-10707107</span>
<span class="co">##    [8] contig_3_1 2509221-2614206      + | chr1:10709490-10826652</span>
<span class="co">##    [9] contig_3_1 2614209-2973841      + | chr1:10832293-11220548</span>
<span class="co">##   [10] contig_3_1 2973842-3316589      + | chr1:11225607-11565085</span>
<span class="co">##   [11] contig_3_1 1652225-1652940      + |    YSR:2724777-2725492</span>
<span class="co">##   -------</span>
<span class="co">##   seqinfo: 55 sequences from KUM-M3-7f genome</span></code></pre>
<div class="sourceCode" id="cb109"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">x</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">rbind</a></span><span class="op">(</span>
  <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">gbs</span>, \<span class="op">(</span><span class="va">gb</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/flagPairs.html" class="external-link">flagPairs</a></span><span class="op">(</span><span class="va">gb</span><span class="op">)</span><span class="op">$</span><span class="va">pairs</span><span class="op">)</span><span class="op">)</span>
<span class="op">)</span></code></pre></div>
</div>
<div class="section level1">
<h1 id="roi-11-highly-conserved-regions">ROI 11: highly conserved regions<a class="anchor" aria-label="anchor" href="#roi-11-highly-conserved-regions"></a>
</h1>
<p>Many mapped regions have a width shorter than 600 and form a distinct peak in width distribution plot (see <code><a href="../articles/RegionWidths.html">vignette("RegionWidths", package = "OikScrambling")</a></code> for details).</p>
<p>Surprisingly they appear to be very conserved.</p>
<div class="sourceCode" id="cb110"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="http://www.bioinf.jku.at/software/msa/" class="external-link">msa</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="http://yulab-smu.top/ggmsa/" class="external-link">ggmsa</a></span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Registered S3 methods overwritten by 'ggalt':</span>
<span class="co">##   method                  from   </span>
<span class="co">##   grid.draw.absoluteGrob  ggplot2</span>
<span class="co">##   grobHeight.absoluteGrob ggplot2</span>
<span class="co">##   grobWidth.absoluteGrob  ggplot2</span>
<span class="co">##   grobX.absoluteGrob      ggplot2</span>
<span class="co">##   grobY.absoluteGrob      ggplot2</span></code></pre>
<pre><code><span class="co">## ggmsa v1.1.4  Document: http://yulab-smu.top/ggmsa/</span>
<span class="co">## </span>
<span class="co">## If you use ggmsa in published research, please cite: DOI: 10.18129/B9.bioc.ggmsa</span></code></pre>
<div class="sourceCode" id="cb113"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">x</span> <span class="op">&lt;-</span> <span class="va">coa</span><span class="op">$</span><span class="va">Oki_Osa</span> |&gt; <span class="fu">plyranges</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/plyranges/man/tidyverse-reexports.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">width</span> <span class="op">&lt;</span> <span class="fl">600</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/r/base/sort.html" class="external-link">sort</a></span><span class="op">(</span>i<span class="op">=</span><span class="cn">T</span><span class="op">)</span>

<span class="va">roi</span> <span class="op">&lt;-</span> <span class="va">x</span><span class="op">[</span><span class="fl">4</span><span class="op">]</span>
<span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/getSeq.html" class="external-link">getSeq</a></span><span class="op">(</span><span class="va">genomes</span><span class="op">$</span><span class="va">Oki</span>, <span class="va">roi</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## DNAStringSet object of length 1:</span>
<span class="co">##     width seq</span>
<span class="co">## [1]   222 <span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #AFFFFF;">GG</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #FFAFAF;">AAAA</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">TTT</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">GG</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #FFAFAF;">AAA</span><span style="color: #000000; background-color: #AFFFAF;">CC</span>...<span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">TTTTT</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">T</span></span></code></pre>
<div class="sourceCode" id="cb115"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/pairwiseAlignment.html" class="external-link">pairwiseAlignment</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/getSeq.html" class="external-link">getSeq</a></span><span class="op">(</span><span class="va">genomes</span><span class="op">$</span><span class="va">Oki</span>, <span class="va">roi</span> <span class="op">+</span> <span class="fl">100</span><span class="op">)</span>,
                  <span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/getSeq.html" class="external-link">getSeq</a></span><span class="op">(</span><span class="va">genomes</span><span class="op">$</span><span class="va">Osa</span>, <span class="va">roi</span><span class="op">$</span><span class="va">query</span> <span class="op">+</span> <span class="fl">100</span><span class="op">)</span>,
                  type <span class="op">=</span> <span class="st">"global"</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/PairwiseAlignments-io.html" class="external-link">writePairwiseAlignments</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## ########################################</span>
<span class="co">## # Program: Biostrings (version 2.62.0), a Bioconductor package</span>
<span class="co">## # Rundate: Wed Feb  9 10:25:22 2022</span>
<span class="co">## ########################################</span>
<span class="co">## #=======================================</span>
<span class="co">## #</span>
<span class="co">## # Aligned_sequences: 2</span>
<span class="co">## # 1: P1</span>
<span class="co">## # 2: S1</span>
<span class="co">## # Matrix: NA</span>
<span class="co">## # Gap_penalty: 14.0</span>
<span class="co">## # Extend_penalty: 4.0</span>
<span class="co">## #</span>
<span class="co">## # Length: 441</span>
<span class="co">## # Identity:     261/441 (59.2%)</span>
<span class="co">## # Similarity:    NA/441 (NA%)</span>
<span class="co">## # Gaps:          47/441 (10.7%)</span>
<span class="co">## # Score: -695.3687</span>
<span class="co">## #</span>
<span class="co">## #</span>
<span class="co">## #=======================================</span>
<span class="co">## </span>
<span class="co">## P1                 1 AGATTTCAAGACGCCTGCCCTTTTTCTTTGCCTAAAATGACCTCGAATTT     50</span>
<span class="co">##                       |  |  | || |  |     || || | |   | |||  | | |||| |</span>
<span class="co">## S1                 1 -GCGTAAATGAAGA-TCAAAGTTGTCATAGG--ACAATT-CTTAGAATAT     45</span>
<span class="co">## </span>
<span class="co">## P1                51 --AGTCTTTGAA---CCAATGTTATAAAAAAGCTAGA-AAGACAGAGAAA     94</span>
<span class="co">##                        | | ||||||   |   |||| | | ||| ||| | || | |  | | </span>
<span class="co">## S1                46 CCAATTTTTGAATAGCTCCTGTTTTTATAAATCTATAGAAAAAATGGTAC     95</span>
<span class="co">## </span>
<span class="co">## P1                95 GAGAGCCGGTCTTAAAATGTTTAGAGAAGTGGCTTAAACCCGGAGTATCA    144</span>
<span class="co">##                      ||  || ||||||||| | ||||||  | | |  ||| || |||||||||</span>
<span class="co">## S1                96 GATCGC-GGTCTTAAATTTTTTAGA--ACTCGACTAATCCTGGAGTATCA    142</span>
<span class="co">## </span>
<span class="co">## P1               145 CGTTGAACATTCTACACCGGTAGAGATAACTTTGAGAAATGATTTGAACA    194</span>
<span class="co">##                      |||| || || || |    || ||||||| ||||  |||||||| |||| </span>
<span class="co">## S1               143 CGTTAAAAATCCTGC----GT-GAGATAAGTTTGCAAAATGATTGGAACC    187</span>
<span class="co">## </span>
<span class="co">## P1               195 CACGAACACCTTGAAACTACTCCGGTTTCAGCTGTGAGCTAATTTCCGTT    244</span>
<span class="co">##                       |    |||||||   |||||  ||||||||||| ||||| |||| ||||</span>
<span class="co">## S1               188 TAGTTGCACCTTGTTGCTACTTTGGTTTCAGCTG-GAGCTTATTTGCGTT    236</span>
<span class="co">## </span>
<span class="co">## P1               245 TTTTCTCACGCCACTTTTTT---CAAGATCTAAAAAACAAAAATTAAAAA    291</span>
<span class="co">##                      ||     |    | |||      ||||||  | || |  ||  || |  |</span>
<span class="co">## S1               237 TTA----AAATTAATTTCCCAGACAAGATGCAGAACAGCAATTTTCAGTA    282</span>
<span class="co">## </span>
<span class="co">## P1               292 GCTGTTTTTCGAGCGCTTGAGATAATCAAGTCCCTGTTCCAGCGACTTTC    341</span>
<span class="co">##                       ||| ||||   || ||| ||||||||||||   | ||| ||  |||  |</span>
<span class="co">## S1               283 ACTGCTTTTTCTGCACTTCAGATAATCAAGTA--TTTTCTAGT-ACT--C    327</span>
<span class="co">## </span>
<span class="co">## P1               342 AAATC-AAG-CTGAAAT--TGAACGAAAAAAAACGAAAAAAGAGCCTTTG    387</span>
<span class="co">##                      |  || ||| || | ||  |||  ||  || |  |||  |   || |  |</span>
<span class="co">## S1               328 AGTTCTAAGACTCACATCTTGATAGAGCAACACTGAAGCA---GCATCAG    374</span>
<span class="co">## </span>
<span class="co">## P1               388 TCCGAAAAATAAAGA-TCAAAAAGAAAATCCTCGCT-----    422</span>
<span class="co">##                      |  || |   || || |  |      |||  |   |     </span>
<span class="co">## S1               375 T--GATATTGAACGACTTGAGTTTTCAATGTTTTGTGAAAA    413</span>
<span class="co">## </span>
<span class="co">## </span>
<span class="co">## #---------------------------------------</span>
<span class="co">## #---------------------------------------</span></code></pre>
<div class="sourceCode" id="cb117"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">roi</span> <span class="op">&lt;-</span> <span class="va">x</span><span class="op">[</span><span class="fl">10</span><span class="op">]</span>
<span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/getSeq.html" class="external-link">getSeq</a></span><span class="op">(</span><span class="va">genomes</span><span class="op">$</span><span class="va">Oki</span>, <span class="va">roi</span><span class="op">)</span>       |&gt; <span class="fu"><a href="https://rdrr.io/r/base/character.html" class="external-link">as.character</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## [1] "ATTGTTTAGTCTGGGGTTTACAGAAGAGTTGTTTCTTCATGGAACCGGGCTGCTGTAAATCAGAGGACTCGGTCTCGGCCGAGCGCGCGCTGACCTCTAAAGCGGCGGCCCTAGCGCGCATCGTCAAGCGCAAATTTTTCAGAATTGAGCTTCCAAAATCAAACACCAGAAACGCCTTTCGCCGAAAAATCAAGGTTCCATTGAGT"</span></code></pre>
<div class="sourceCode" id="cb119"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/getSeq.html" class="external-link">getSeq</a></span><span class="op">(</span><span class="va">genomes</span><span class="op">$</span><span class="va">Osa</span>, <span class="va">roi</span><span class="op">$</span><span class="va">query</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/r/base/character.html" class="external-link">as.character</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## [1] "ATTGTTTAGTCTGCTGTCACAGTAGAGTTGTTGCTTCATGGAACCAGCCCGCAGTAAATCAGAAGAGTCAGTCGCGGACGAGTGCGCGCTGACCTCTAAAGCGGCGGCCCCAACGCGCATCGTCAAGCGCAACATTTTCAGAATTGAGCTTCTGAAATCAAAAAGCAGAAACAATTTCGCCGAATAATCACTCAATTGAAAGACGGTTCCATTGAGT"</span></code></pre>
<div class="sourceCode" id="cb121"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">seqs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>
  <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/findOverlaps-methods.html" class="external-link">subsetByOverlaps</a></span><span class="op">(</span><span class="va">coa</span><span class="op">$</span><span class="va">Oki_Osa</span>, <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/genomic-range-squeezers.html" class="external-link">granges</a></span><span class="op">(</span><span class="va">roi</span><span class="op">)</span><span class="op">)</span>                             |&gt; <span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/getSeq.html" class="external-link">getSeq</a></span><span class="op">(</span>x<span class="op">=</span><span class="va">genomes</span><span class="op">$</span><span class="va">Oki</span><span class="op">)</span>,
  <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/findOverlaps-methods.html" class="external-link">subsetByOverlaps</a></span><span class="op">(</span><span class="va">coa</span><span class="op">$</span><span class="va">Oki_Osa</span>, <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/genomic-range-squeezers.html" class="external-link">granges</a></span><span class="op">(</span><span class="va">roi</span><span class="op">)</span><span class="op">)</span><span class="op">$</span><span class="va">query</span>                       |&gt; <span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/getSeq.html" class="external-link">getSeq</a></span><span class="op">(</span>x<span class="op">=</span><span class="va">genomes</span><span class="op">$</span><span class="va">Osa</span><span class="op">)</span>,
  <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/findOverlaps-methods.html" class="external-link">subsetByOverlaps</a></span><span class="op">(</span><span class="va">coa</span><span class="op">$</span><span class="va">Oki_Aom</span>, <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/genomic-range-squeezers.html" class="external-link">granges</a></span><span class="op">(</span><span class="va">roi</span><span class="op">)</span>, ignore.strand <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">$</span><span class="va">query</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/getSeq.html" class="external-link">getSeq</a></span><span class="op">(</span>x<span class="op">=</span><span class="va">genomes</span><span class="op">$</span><span class="va">Aom</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/reverseComplement.html" class="external-link">reverseComplement</a></span><span class="op">(</span><span class="op">)</span>,
  <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/findOverlaps-methods.html" class="external-link">subsetByOverlaps</a></span><span class="op">(</span><span class="va">coa</span><span class="op">$</span><span class="va">Oki_Bar</span>, <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/genomic-range-squeezers.html" class="external-link">granges</a></span><span class="op">(</span><span class="va">roi</span><span class="op">)</span><span class="op">)</span><span class="op">$</span><span class="va">query</span>                       |&gt; <span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/getSeq.html" class="external-link">getSeq</a></span><span class="op">(</span>x<span class="op">=</span><span class="va">genomes</span><span class="op">$</span><span class="va">Bar</span><span class="op">)</span>,
  <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/findOverlaps-methods.html" class="external-link">subsetByOverlaps</a></span><span class="op">(</span><span class="va">coa</span><span class="op">$</span><span class="va">Oki_Nor</span>, <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/genomic-range-squeezers.html" class="external-link">granges</a></span><span class="op">(</span><span class="va">roi</span><span class="op">)</span><span class="op">)</span><span class="op">$</span><span class="va">query</span>                       |&gt; <span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/getSeq.html" class="external-link">getSeq</a></span><span class="op">(</span>x<span class="op">=</span><span class="va">genomes</span><span class="op">$</span><span class="va">Nor</span><span class="op">)</span>
<span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">seqs</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Oki"</span>, <span class="st">"Osa"</span>, <span class="st">"Aom"</span>, <span class="st">"Bar"</span>, <span class="st">"Nor"</span><span class="op">)</span>

<span class="op">(</span><span class="va">seqs_aln</span> <span class="op">&lt;-</span> <span class="fu">msa</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/msa/man/msaMuscle.html" class="external-link">msaMuscle</a></span><span class="op">(</span><span class="va">seqs</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/r/methods/as.html" class="external-link">as</a></span><span class="op">(</span><span class="st">"DNAMultipleAlignment"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## DNAMultipleAlignment with 5 rows and 230 columns</span>
<span class="co">##      aln                                                    names               </span>
<span class="co">## [1] -----<span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">TTT</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">GGGG</span><span style="color: #000000; background-color: #FFD7AF;">TTT</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">C</span>...------------<span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #AFFFAF;">CC</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">T</span>- Oki</span>
<span class="co">## [2] -----<span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">TTT</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">G</span>-<span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">C</span>...<span style="color: #000000; background-color: #FFAFAF;">AAA</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #AFFFFF;">G</span>-----<span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #AFFFAF;">CC</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">T</span>- Osa</span>
<span class="co">## [3] -----<span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">TTT</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">G</span>-<span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">C</span>...<span style="color: #000000; background-color: #FFAFAF;">AAA</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #AFFFAF;">CC</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">T</span>- Aom</span>
<span class="co">## [4] ---<span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">TTT</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span>-<span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">C</span>...<span style="color: #000000; background-color: #AFFFAF;">C</span>-----------<span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #AFFFAF;">CC</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">G</span> Bar</span>
<span class="co">## [5] <span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">TTT</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span>-<span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">C</span>...<span style="color: #000000; background-color: #AFFFFF;">G</span>-----------<span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #AFFFAF;">CC</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">G</span> Nor</span></code></pre>
<div class="sourceCode" id="cb123"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">ggmsa</span><span class="fu">::</span><span class="fu"><a href="http://yulab-smu.top/ggmsa/reference/ggmsa.html" class="external-link">ggmsa</a></span><span class="op">(</span><span class="va">seqs_aln</span>,   <span class="fl">1</span>,  <span class="fl">50</span>, seq_name <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> <span class="op">/</span>
<span class="fu">ggmsa</span><span class="fu">::</span><span class="fu"><a href="http://yulab-smu.top/ggmsa/reference/ggmsa.html" class="external-link">ggmsa</a></span><span class="op">(</span><span class="va">seqs_aln</span>,  <span class="fl">51</span>, <span class="fl">100</span>, seq_name <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> <span class="op">/</span>
<span class="fu">ggmsa</span><span class="fu">::</span><span class="fu"><a href="http://yulab-smu.top/ggmsa/reference/ggmsa.html" class="external-link">ggmsa</a></span><span class="op">(</span><span class="va">seqs_aln</span>, <span class="fl">101</span>, <span class="fl">150</span>, seq_name <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> <span class="op">/</span>
<span class="fu">ggmsa</span><span class="fu">::</span><span class="fu"><a href="http://yulab-smu.top/ggmsa/reference/ggmsa.html" class="external-link">ggmsa</a></span><span class="op">(</span><span class="va">seqs_aln</span>, <span class="fl">151</span>, <span class="fl">200</span>, seq_name <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> <span class="op">/</span>
<span class="fu">ggmsa</span><span class="fu">::</span><span class="fu"><a href="http://yulab-smu.top/ggmsa/reference/ggmsa.html" class="external-link">ggmsa</a></span><span class="op">(</span><span class="va">seqs_aln</span>, <span class="fl">201</span>, <span class="fl">250</span>, seq_name <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<p><img src="RegionsOfInterest_files/figure-html/ROI11_1-1.png" width="864"></p>
<div class="sourceCode" id="cb124"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">roi</span> <span class="op">&lt;-</span> <span class="va">x</span><span class="op">[</span><span class="fl">11</span><span class="op">]</span>
<span class="va">seqs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>
  <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/findOverlaps-methods.html" class="external-link">subsetByOverlaps</a></span><span class="op">(</span><span class="va">coa</span><span class="op">$</span><span class="va">Oki_Osa</span>, <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/genomic-range-squeezers.html" class="external-link">granges</a></span><span class="op">(</span><span class="va">roi</span><span class="op">)</span><span class="op">)</span>                             |&gt; <span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/getSeq.html" class="external-link">getSeq</a></span><span class="op">(</span>x<span class="op">=</span><span class="va">genomes</span><span class="op">$</span><span class="va">Oki</span><span class="op">)</span>,
  <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/findOverlaps-methods.html" class="external-link">subsetByOverlaps</a></span><span class="op">(</span><span class="va">coa</span><span class="op">$</span><span class="va">Oki_Osa</span>, <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/genomic-range-squeezers.html" class="external-link">granges</a></span><span class="op">(</span><span class="va">roi</span><span class="op">)</span><span class="op">)</span><span class="op">$</span><span class="va">query</span>                       |&gt; <span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/getSeq.html" class="external-link">getSeq</a></span><span class="op">(</span>x<span class="op">=</span><span class="va">genomes</span><span class="op">$</span><span class="va">Osa</span><span class="op">)</span>,
  <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/findOverlaps-methods.html" class="external-link">subsetByOverlaps</a></span><span class="op">(</span><span class="va">coa</span><span class="op">$</span><span class="va">Oki_Aom</span>, <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/genomic-range-squeezers.html" class="external-link">granges</a></span><span class="op">(</span><span class="va">roi</span><span class="op">)</span>, ignore.strand <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">$</span><span class="va">query</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/getSeq.html" class="external-link">getSeq</a></span><span class="op">(</span>x<span class="op">=</span><span class="va">genomes</span><span class="op">$</span><span class="va">Aom</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/reverseComplement.html" class="external-link">reverseComplement</a></span><span class="op">(</span><span class="op">)</span>,
  <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/findOverlaps-methods.html" class="external-link">subsetByOverlaps</a></span><span class="op">(</span><span class="va">coa</span><span class="op">$</span><span class="va">Oki_Bar</span>, <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/genomic-range-squeezers.html" class="external-link">granges</a></span><span class="op">(</span><span class="va">roi</span><span class="op">)</span><span class="op">)</span><span class="op">$</span><span class="va">query</span>                       |&gt; <span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/getSeq.html" class="external-link">getSeq</a></span><span class="op">(</span>x<span class="op">=</span><span class="va">genomes</span><span class="op">$</span><span class="va">Bar</span><span class="op">)</span>,
  <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/findOverlaps-methods.html" class="external-link">subsetByOverlaps</a></span><span class="op">(</span><span class="va">coa</span><span class="op">$</span><span class="va">Oki_Nor</span>, <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/genomic-range-squeezers.html" class="external-link">granges</a></span><span class="op">(</span><span class="va">roi</span><span class="op">)</span><span class="op">)</span><span class="op">$</span><span class="va">query</span>                       |&gt; <span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/getSeq.html" class="external-link">getSeq</a></span><span class="op">(</span>x<span class="op">=</span><span class="va">genomes</span><span class="op">$</span><span class="va">Nor</span><span class="op">)</span>
<span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">seqs</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Oki"</span>, <span class="st">"Osa"</span>, <span class="st">"Aom"</span>, <span class="st">"Bar"</span>, <span class="st">"Nor"</span><span class="op">)</span>

<span class="op">(</span><span class="va">seqs_aln</span> <span class="op">&lt;-</span> <span class="fu">msa</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/msa/man/msaMuscle.html" class="external-link">msaMuscle</a></span><span class="op">(</span><span class="va">seqs</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/r/methods/as.html" class="external-link">as</a></span><span class="op">(</span><span class="st">"DNAMultipleAlignment"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## DNAMultipleAlignment with 5 rows and 2474 columns</span>
<span class="co">##      aln                                                    names               </span>
<span class="co">## [1] <span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #AFFFFF;">GGGG</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #AFFFFF;">GG</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">G</span>...------------------------- Oki</span>
<span class="co">## [2] --------<span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #AFFFFF;">GGGG</span>...<span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">CC</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">CC</span> Bar</span>
<span class="co">## [3] --------<span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #AFFFFF;">GGGG</span>...------------------------- Nor</span>
<span class="co">## [4] <span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #AFFFFF;">GG</span>-<span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #AFFFFF;">GG</span>-<span style="color: #000000; background-color: #AFFFFF;">G</span>...------------------------- Osa</span>
<span class="co">## [5] <span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #AFFFFF;">GG</span>-<span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #AFFFFF;">GGGG</span>...------------------------- Aom</span></code></pre>
<div class="sourceCode" id="cb126"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">ggmsa</span><span class="fu">::</span><span class="fu"><a href="http://yulab-smu.top/ggmsa/reference/ggmsa.html" class="external-link">ggmsa</a></span><span class="op">(</span><span class="va">seqs_aln</span>,   <span class="fl">1</span>,  <span class="fl">50</span>, seq_name <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> <span class="op">/</span>
<span class="fu">ggmsa</span><span class="fu">::</span><span class="fu"><a href="http://yulab-smu.top/ggmsa/reference/ggmsa.html" class="external-link">ggmsa</a></span><span class="op">(</span><span class="va">seqs_aln</span>,  <span class="fl">51</span>, <span class="fl">100</span>, seq_name <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> <span class="op">/</span>
<span class="fu">ggmsa</span><span class="fu">::</span><span class="fu"><a href="http://yulab-smu.top/ggmsa/reference/ggmsa.html" class="external-link">ggmsa</a></span><span class="op">(</span><span class="va">seqs_aln</span>, <span class="fl">101</span>, <span class="fl">150</span>, seq_name <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> <span class="op">/</span>
<span class="fu">ggmsa</span><span class="fu">::</span><span class="fu"><a href="http://yulab-smu.top/ggmsa/reference/ggmsa.html" class="external-link">ggmsa</a></span><span class="op">(</span><span class="va">seqs_aln</span>, <span class="fl">151</span>, <span class="fl">200</span>, seq_name <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> <span class="op">/</span>
<span class="fu">ggmsa</span><span class="fu">::</span><span class="fu"><a href="http://yulab-smu.top/ggmsa/reference/ggmsa.html" class="external-link">ggmsa</a></span><span class="op">(</span><span class="va">seqs_aln</span>, <span class="fl">201</span>, <span class="fl">250</span>, seq_name <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> <span class="op">/</span>
<span class="fu">ggmsa</span><span class="fu">::</span><span class="fu"><a href="http://yulab-smu.top/ggmsa/reference/ggmsa.html" class="external-link">ggmsa</a></span><span class="op">(</span><span class="va">seqs_aln</span>, <span class="fl">251</span>, <span class="fl">300</span>, seq_name <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<p><img src="RegionsOfInterest_files/figure-html/ROI11_2-1.png" width="864"></p>
<div class="section level2">
<h2 id="autocorrelations">Autocorrelations<a class="anchor" aria-label="anchor" href="#autocorrelations"></a>
</h2>
<p>There appears to be a small tendency that short width regions are followed by short width regions, but I am not sure if I understand the statistics well.</p>
<div class="sourceCode" id="cb127"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="http://yulab-smu.top/ggmsa/reference/plot-methods.html" class="external-link">plot</a></span><span class="op">(</span><span class="fu">width</span><span class="op">(</span><span class="va">coa</span><span class="op">$</span><span class="va">Oki_Osa</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">tail</a></span><span class="op">(</span><span class="fu">width</span><span class="op">(</span><span class="va">coa</span><span class="op">$</span><span class="va">Oki_Osa</span><span class="op">)</span>, <span class="op">-</span><span class="fl">1</span><span class="op">)</span>, <span class="cn">NA</span><span class="op">)</span>, log <span class="op">=</span> <span class="st">'xy'</span>, pch<span class="op">=</span><span class="st">'.'</span><span class="op">)</span></code></pre></div>
<p><img src="RegionsOfInterest_files/figure-html/autocorrelations-1.png" width="700"></p>
<div class="sourceCode" id="cb128"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/stats/acf.html" class="external-link">acf</a></span><span class="op">(</span><span class="fu">width</span><span class="op">(</span><span class="va">coa</span><span class="op">$</span><span class="va">Oki_Osa</span><span class="op">)</span>, lag.max <span class="op">=</span> <span class="fl">1000</span><span class="op">)</span></code></pre></div>
<p><img src="RegionsOfInterest_files/figure-html/autocorrelations-2.png" width="700"></p>
<div class="sourceCode" id="cb129"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/stats/acf.html" class="external-link">acf</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/sample.html" class="external-link">sample</a></span><span class="op">(</span><span class="fu">width</span><span class="op">(</span><span class="va">coa</span><span class="op">$</span><span class="va">Oki_Osa</span><span class="op">)</span><span class="op">)</span>, lag.max <span class="op">=</span> <span class="fl">1000</span><span class="op">)</span></code></pre></div>
<p><img src="RegionsOfInterest_files/figure-html/autocorrelations-3.png" width="700"></p>
<div class="sourceCode" id="cb130"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/stats/acf.html" class="external-link">pacf</a></span><span class="op">(</span><span class="fu">width</span><span class="op">(</span><span class="va">coa</span><span class="op">$</span><span class="va">Oki_Osa</span><span class="op">)</span>, lag.max <span class="op">=</span> <span class="fl">100</span><span class="op">)</span></code></pre></div>
<p><img src="RegionsOfInterest_files/figure-html/autocorrelations-4.png" width="700"></p>
<div class="sourceCode" id="cb131"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/stats/acf.html" class="external-link">pacf</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/sample.html" class="external-link">sample</a></span><span class="op">(</span><span class="fu">width</span><span class="op">(</span><span class="va">coa</span><span class="op">$</span><span class="va">Oki_Osa</span><span class="op">)</span><span class="op">)</span>, lag.max <span class="op">=</span> <span class="fl">100</span><span class="op">)</span></code></pre></div>
<p><img src="RegionsOfInterest_files/figure-html/autocorrelations-5.png" width="700"></p>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Aleksandra Bliznina, Michael Mansfield, Charles Plessy.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.1.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
