<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Segment movement distances • OikScrambling</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Segment movement distances">
<meta property="og:description" content="OikScrambling">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">OikScrambling</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">5.0.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/OikScrambling.html">Get started</a>
</li>
<li>
  <a href="../articles/LoadGenomicBreaks.html">Data load</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Supplemental material
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/OxfordPlots.html">Figure 1A DNA line plots</a>
    </li>
    <li>
      <a href="../articles/ParallelPlots.html">Figure 1B chromosome parallel plots</a>
    </li>
    <li>
      <a href="../articles/DistanceMeasures.html">Table 1 scrambling index</a>
    </li>
    <li>
      <a href="../articles/ChromosomePlots.html">Figure 5 genome-wide feature patterns</a>
    </li>
    <li>
      <a href="../articles/SquareWindowPlots.html">Figure 7 DNA line plots</a>
    </li>
    <li>
      <a href="../articles/ScalingBreakpoints.html">Figure 7 breakpoint number barplots</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/CNEs.html">Conserved non-coding elements</a>
    </li>
    <li>
      <a href="../articles/ChromosomePlots.html">Chromosome-scale feature density plots</a>
    </li>
    <li>
      <a href="../articles/CoalescingCutoffs.html">Modifying coalescent cutoffs</a>
    </li>
    <li>
      <a href="../articles/ColinearityInterruptors.html">Regions interrupting colinearity</a>
    </li>
    <li>
      <a href="../articles/DistanceMeasures.html">Distance Measures</a>
    </li>
    <li>
      <a href="../articles/GenomicFeatures.html">Genomic features of the breakpoints</a>
    </li>
    <li>
      <a href="../articles/GenomicHeatmaps.html">Genomic features of the breakpoints (heatmaps)</a>
    </li>
    <li>
      <a href="../articles/Inversions.html">Inversions</a>
    </li>
    <li>
      <a href="../articles/LoadGenomicBreaks.html">Load genomic break data</a>
    </li>
    <li>
      <a href="../articles/OperonConservation.html">Operon conservation analysis</a>
    </li>
    <li>
      <a href="../articles/OrthogroupSyntenies.html">Orthogroup syntenies</a>
    </li>
    <li>
      <a href="../articles/OxfordPlots.html">Alignment line plots, whole genomes</a>
    </li>
    <li>
      <a href="../articles/PAC3.html">Region of interest: PAC3</a>
    </li>
    <li>
      <a href="../articles/PWM.html">Motif search and analysis</a>
    </li>
    <li>
      <a href="../articles/ParallelPlots.html">Whole-chromosome parallel plots</a>
    </li>
    <li>
      <a href="../articles/RegionWidths.html">Width of the aligned and mapped regions.</a>
    </li>
    <li>
      <a href="../articles/RegionsOfInterest.html">Regions of interest</a>
    </li>
    <li>
      <a href="../articles/RepeatRegions.html">Repeat regions</a>
    </li>
    <li>
      <a href="../articles/ScalingBreakpoints.html">Scaling breakpoints</a>
    </li>
    <li>
      <a href="../articles/SegmentMovements.html">Segment movement distances</a>
    </li>
    <li>
      <a href="../articles/SquareWindowPlots.html">Alignment line plots, 10×10 Mbp windows</a>
    </li>
    <li>
      <a href="../articles/TFBS.html">Transcription factor binding sites</a>
    </li>
    <li>
      <a href="../articles/TandemSiteDuplications.html">Tandem site duplications</a>
    </li>
    <li>
      <a href="../articles/Translocations.html">Translocations</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/oist/LuscombeU_OikScrambling/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Segment movement distances</h1>
                        <h4 data-toc-skip class="author">Michael
Mansfield</h4>
                        <h4 data-toc-skip class="author">Charles
Plessy</h4>
            
            <h4 data-toc-skip class="date">13 October, 2023</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/oist/LuscombeU_OikScrambling/blob/HEAD/vignettes/SegmentMovements.Rmd" class="external-link"><code>vignettes/SegmentMovements.Rmd</code></a></small>
      <div class="hidden name"><code>SegmentMovements.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="load-packages-and-data">Load packages and data<a class="anchor" aria-label="anchor" href="#load-packages-and-data"></a>
</h2>
<p>See the
<code><a href="../articles/OikScrambling.html">vignette("OikScrambling", package = "OikScrambling")</a></code> for
general details on package and data load.</p>
<p>See
<code><a href="../articles/LoadGenomicBreaks.html">vignette("LoadGenomicBreaks", package = "OikScrambling")</a></code>
for how the different <code>GBreaks</code> objects are prepared.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="op">{</span></span>
<span>  <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://oist.github.io/GenomicBreaks/" class="external-link">'GenomicBreaks'</a></span><span class="op">)</span></span>
<span>  <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://ggplot2.tidyverse.org" class="external-link">'ggplot2'</a></span><span class="op">)</span></span>
<span>  <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st">"BreakpointsData"</span><span class="op">)</span></span>
<span><span class="op">}</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/load.html" class="external-link">load</a></span><span class="op">(</span><span class="st">"BreakPoints.Rdata"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="segment-movement-distances">Segment movement distances<a class="anchor" aria-label="anchor" href="#segment-movement-distances"></a>
</h2>
<p>Whenever a collinear segment moves between species, if the homologous
segment is located on an equivalent chromosome, a distance can be
calculated.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Find distance between two equivalent GRanges.</span></span>
<span><span class="co"># Input: a GBreaks object</span></span>
<span><span class="co"># Output: Absolute distance between the target and query ranges based on some metric ("start", "end", or "midpoint")</span></span>
<span><span class="va">GBreaksDist</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">gb</span>, <span class="va">distMetric</span><span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"midpoint"</span>, <span class="st">"start"</span>, <span class="st">"end"</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="va">distMetric</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/match.arg.html" class="external-link">match.arg</a></span><span class="op">(</span><span class="va">distMetric</span><span class="op">)</span></span>
<span>  <span class="va">distFun</span> <span class="op">&lt;-</span> <span class="kw"><a href="https://rdrr.io/r/base/switch.html" class="external-link">switch</a></span><span class="op">(</span><span class="va">distMetric</span>,</span>
<span>                    start    <span class="op">=</span> <span class="va">start</span>,</span>
<span>                    end      <span class="op">=</span> <span class="va">end</span>,</span>
<span>                    midpoint <span class="op">=</span> \<span class="op">(</span><span class="va">gr</span><span class="op">)</span> <span class="va">gr</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/intra-range-methods.html" class="external-link">resize</a></span><span class="op">(</span><span class="fl">1</span>, fix <span class="op">=</span> <span class="st">"center"</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/stats/start.html" class="external-link">start</a></span><span class="op">(</span><span class="op">)</span> <span class="op">)</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/MathFun.html" class="external-link">abs</a></span><span class="op">(</span><span class="fu">distFun</span><span class="op">(</span><span class="va">gb</span><span class="op">)</span> <span class="op">-</span> <span class="fu">distFun</span><span class="op">(</span><span class="va">gb</span><span class="op">$</span><span class="va">query</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="va">filterSyntenicPairs</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">gb</span>, <span class="va">chromosomePairing</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="va">syntenic_pairs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="va">chromosomePairing</span><span class="op">[[</span><span class="st">"sp1"</span><span class="op">]</span><span class="op">]</span>, <span class="va">chromosomePairing</span><span class="op">[[</span><span class="st">"sp2"</span><span class="op">]</span><span class="op">]</span>, sep <span class="op">=</span> <span class="st">"_"</span><span class="op">)</span></span>
<span>  <span class="va">is_syntenic</span>    <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqnames</a></span><span class="op">(</span><span class="va">gb</span><span class="op">)</span>,               <span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqnames</a></span><span class="op">(</span><span class="va">gb</span><span class="op">$</span><span class="va">query</span><span class="op">)</span>,         sep <span class="op">=</span> <span class="st">"_"</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html" class="external-link">%in%</a></span> <span class="va">syntenic_pairs</span></span>
<span>  <span class="va">gb</span><span class="op">[</span><span class="va">is_syntenic</span><span class="op">]</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="va">chromDist</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">gb</span>, <span class="va">chromosomePairing</span><span class="op">=</span><span class="cn">NULL</span>, <span class="va">distMetric</span><span class="op">=</span><span class="st">"midpoint"</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="co"># Subset to include only ranges in which all chrom in sp1 are paired to their equivalent in sp2</span></span>
<span>  <span class="co"># Note that a similar function could be used to compare the identity of chromosome arms</span></span>
<span>  <span class="va">gb</span> <span class="op">&lt;-</span> <span class="fu">filterSyntenicPairs</span><span class="op">(</span><span class="va">gb</span>, <span class="va">chromosomePairing</span><span class="op">)</span></span>
<span>  </span>
<span>  <span class="co"># Now that the GRanges are only on equivalent chromosomes, calculate the distances between them.</span></span>
<span>  <span class="co"># This can be "start", "end", or "midpoint" distance.</span></span>
<span>  <span class="va">gb</span><span class="op">$</span><span class="va">distance</span> <span class="op">&lt;-</span> <span class="fu">GBreaksDist</span><span class="op">(</span><span class="va">gb</span>, distMetric<span class="op">=</span><span class="va">distMetric</span><span class="op">)</span></span>
<span>  <span class="va">gb</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="co"># Unfortunately, the use of upper and lower case in chromosome names was not</span></span>
<span><span class="co"># standardised when assembling the genomes.</span></span>
<span><span class="va">synPairs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/SimpleList-class.html" class="external-link">SimpleList</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">synPairs</span><span class="op">$</span><span class="va">chr_Chr</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="st">"sp1"</span><span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"chr1"</span>, <span class="st">"chr2"</span>, <span class="st">"PAR"</span>, <span class="st">"XSR"</span>, <span class="st">"YSR"</span><span class="op">)</span>,</span>
<span>                         <span class="st">"sp2"</span><span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Chr1"</span>, <span class="st">"Chr2"</span>, <span class="st">"PAR"</span>, <span class="st">"XSR"</span>, <span class="st">"YSR"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">synPairs</span><span class="op">$</span><span class="va">Chr_Chr</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="st">"sp1"</span><span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Chr1"</span>, <span class="st">"Chr2"</span>, <span class="st">"PAR"</span>, <span class="st">"XSR"</span>, <span class="st">"YSR"</span><span class="op">)</span>,</span>
<span>                         <span class="st">"sp2"</span><span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Chr1"</span>, <span class="st">"Chr2"</span>, <span class="st">"PAR"</span>, <span class="st">"XSR"</span>, <span class="st">"YSR"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">segDists</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/SimpleList-class.html" class="external-link">SimpleList</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">segDists</span><span class="op">$</span><span class="va">Oki_Osa</span> <span class="op">&lt;-</span> <span class="fu">chromDist</span><span class="op">(</span><span class="va">gbs</span><span class="op">$</span><span class="va">Oki_Osa</span>, <span class="va">synPairs</span><span class="op">$</span><span class="va">chr_Chr</span><span class="op">)</span></span>
<span><span class="va">segDists</span><span class="op">$</span><span class="va">Oki_Bar</span> <span class="op">&lt;-</span> <span class="fu">chromDist</span><span class="op">(</span><span class="va">gbs</span><span class="op">$</span><span class="va">Oki_Bar</span>, <span class="va">synPairs</span><span class="op">$</span><span class="va">chr_Chr</span><span class="op">)</span></span>
<span><span class="va">segDists</span><span class="op">$</span><span class="va">Osa_Bar</span> <span class="op">&lt;-</span> <span class="fu">chromDist</span><span class="op">(</span><span class="va">gbs</span><span class="op">$</span><span class="va">Osa_Bar</span>, <span class="va">synPairs</span><span class="op">$</span><span class="va">Chr_Chr</span><span class="op">)</span></span></code></pre></div>
<div class="section level3">
<h3 id="plotting-segment-movement-distances">Plotting segment movement distances<a class="anchor" aria-label="anchor" href="#plotting-segment-movement-distances"></a>
</h3>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">plotSegDists</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">segDist</span>, <span class="va">facet.by</span><span class="op">=</span><span class="st">'seqnames~Arm'</span>, <span class="va">binWidth</span><span class="op">=</span><span class="fl">100000</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="va">segDist</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x<span class="op">=</span><span class="va">distance</span>, fill <span class="op">=</span> <span class="va">Arm</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_histogram.html" class="external-link">geom_histogram</a></span><span class="op">(</span>binwidth <span class="op">=</span> <span class="va">binWidth</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html" class="external-link">facet_wrap</a></span><span class="op">(</span><span class="va">facet.by</span>, scales<span class="op">=</span><span class="st">"free_x"</span>, ncol<span class="op">=</span><span class="fl">2</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span><span class="fu">plotSegDists</span><span class="op">(</span><span class="va">segDists</span><span class="op">$</span><span class="va">Oki_Osa</span>, facet.by<span class="op">=</span><span class="st">'seqnames~Arm'</span>, binWidth<span class="op">=</span><span class="fl">100000</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Oki-Osa: segment movement distances (uncoalesced)"</span><span class="op">)</span></span></code></pre></div>
<p><img src="SegmentMovements_files/figure-html/plot_movement_distances_oki_osa-1.svg" width="1440"></p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">plotSegDists</span><span class="op">(</span><span class="va">segDists</span><span class="op">$</span><span class="va">Oki_Bar</span>, facet.by<span class="op">=</span><span class="st">'seqnames~Arm'</span>, binWidth<span class="op">=</span><span class="fl">100000</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Oki-Bar: segment movement distances (uncoalesced)"</span><span class="op">)</span></span></code></pre></div>
<p><img src="SegmentMovements_files/figure-html/plot_movement_distances_oki_bar-1.svg" width="1440"></p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">plotSegDists</span><span class="op">(</span><span class="va">segDists</span><span class="op">$</span><span class="va">Osa_Bar</span>, facet.by<span class="op">=</span><span class="st">'seqnames~Arm'</span>, binWidth<span class="op">=</span><span class="fl">100000</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Osa-Bar: segment movement distances (uncoalesced)"</span><span class="op">)</span></span></code></pre></div>
<p><img src="SegmentMovements_files/figure-html/plot_movement_distances_osa_bar-1.svg" width="1440"></p>
</div>
<div class="section level3">
<h3 id="segment-movement-distances-coalesced">Segment movement distances (coalesced)<a class="anchor" aria-label="anchor" href="#segment-movement-distances-coalesced"></a>
</h3>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">segDistsCoa</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/SimpleList-class.html" class="external-link">SimpleList</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">segDistsCoa</span><span class="op">$</span><span class="va">Oki_Osa</span> <span class="op">&lt;-</span> <span class="fu">chromDist</span><span class="op">(</span><span class="va">coa</span><span class="op">$</span><span class="va">Oki_Osa</span>, <span class="va">synPairs</span><span class="op">$</span><span class="va">chr_Chr</span><span class="op">)</span></span>
<span><span class="va">segDistsCoa</span><span class="op">$</span><span class="va">Oki_Bar</span> <span class="op">&lt;-</span> <span class="fu">chromDist</span><span class="op">(</span><span class="va">coa</span><span class="op">$</span><span class="va">Oki_Bar</span>, <span class="va">synPairs</span><span class="op">$</span><span class="va">chr_Chr</span><span class="op">)</span></span>
<span><span class="va">segDistsCoa</span><span class="op">$</span><span class="va">Osa_Bar</span> <span class="op">&lt;-</span> <span class="fu">chromDist</span><span class="op">(</span><span class="va">coa</span><span class="op">$</span><span class="va">Osa_Bar</span>, <span class="va">synPairs</span><span class="op">$</span><span class="va">Chr_Chr</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">plotSegDists</span><span class="op">(</span><span class="va">segDistsCoa</span><span class="op">$</span><span class="va">Oki_Osa</span>, facet.by<span class="op">=</span><span class="st">'seqnames~Arm'</span>, binWidth<span class="op">=</span><span class="fl">100000</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Oki-Osa: segment movement distances (coalesced)"</span><span class="op">)</span></span></code></pre></div>
<p><img src="SegmentMovements_files/figure-html/plot_movement_distances_oki_osa_coa-1.svg" width="1440"></p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">plotSegDists</span><span class="op">(</span><span class="va">segDistsCoa</span><span class="op">$</span><span class="va">Oki_Bar</span>, facet.by<span class="op">=</span><span class="st">'seqnames~Arm'</span>, binWidth<span class="op">=</span><span class="fl">100000</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Oki-Bar: segment movement distances (coalesced)"</span><span class="op">)</span></span></code></pre></div>
<p><img src="SegmentMovements_files/figure-html/plot_movement_distances_oki_bar_coa-1.svg" width="1440"></p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">plotSegDists</span><span class="op">(</span><span class="va">segDistsCoa</span><span class="op">$</span><span class="va">Osa_Bar</span>, facet.by<span class="op">=</span><span class="st">'seqnames~Arm'</span>, binWidth<span class="op">=</span><span class="fl">100000</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Osa-Bar: segment movement distances (coalesced)"</span><span class="op">)</span></span></code></pre></div>
<p><img src="SegmentMovements_files/figure-html/plot_movement_distances_osa_bar_coa-1.svg" width="1440"></p>
</div>
<div class="section level3">
<h3 id="plotting-segment-movements-relative-to-coordinates">Plotting segment movements relative to coordinates<a class="anchor" aria-label="anchor" href="#plotting-segment-movements-relative-to-coordinates"></a>
</h3>
<p>Uses, start coordinate as a proxy for midpoint, but the difference
should be hardly perceptible.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">plot_movement_distance_on_coords</span> <span class="op">&lt;-</span> \<span class="op">(</span><span class="va">gb</span><span class="op">)</span> <span class="va">gb</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span><span class="va">start</span>, <span class="va">distance</span>, col<span class="op">=</span><span class="va">Arm</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">xlab</a></span><span class="op">(</span><span class="st">"Start on target"</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ylab</a></span><span class="op">(</span><span class="st">"Distance between midpoints on query and target"</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html" class="external-link">facet_wrap</a></span><span class="op">(</span><span class="op">~</span><span class="va">seqnames</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">segDists</span><span class="op">$</span><span class="va">Oki_Osa</span> <span class="op">|&gt;</span> <span class="fu">plot_movement_distance_on_coords</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Oki - Osa"</span><span class="op">)</span></span></code></pre></div>
<p><img src="SegmentMovements_files/figure-html/plot_movement_distance_on_coords_oki_osa_gb-1.svg" width="1440"></p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">segDists</span><span class="op">$</span><span class="va">Oki_Bar</span> <span class="op">|&gt;</span> <span class="fu">plot_movement_distance_on_coords</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Oki - Bar"</span><span class="op">)</span></span></code></pre></div>
<p><img src="SegmentMovements_files/figure-html/plot_movement_distance_on_coords_oki_bar_gb-1.svg" width="1440"></p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">segDists</span><span class="op">$</span><span class="va">Osa_Bar</span> <span class="op">|&gt;</span> <span class="fu">plot_movement_distance_on_coords</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Osa - Bar"</span><span class="op">)</span></span></code></pre></div>
<p><img src="SegmentMovements_files/figure-html/plot_movement_distance_on_coords_osa_bar_gb-1.svg" width="1440"></p>
</div>
<div class="section level3">
<h3 id="plotting-distance-between-query-ranges-relative-to-target-coordinates">Plotting distance between query ranges relative to target
coordinates<a class="anchor" aria-label="anchor" href="#plotting-distance-between-query-ranges-relative-to-target-coordinates"></a>
</h3>
<p>This assumes that the center an unaligned region would be plotted on
the same pixel as the center of its flanking aligned regions.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">plot_distance_to_next_query_on_coords</span> <span class="op">&lt;-</span> \<span class="op">(</span><span class="va">gb</span><span class="op">)</span> <span class="va">gb</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/dist2next.html" class="external-link">dist2next</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span><span class="va">end</span> <span class="op">+</span> <span class="fl">1</span>, <span class="va">qdist</span>, col<span class="op">=</span><span class="va">Arm</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">xlab</a></span><span class="op">(</span><span class="st">"Start unaligned region on target"</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ylab</a></span><span class="op">(</span><span class="st">"Distance between flanking regions on query"</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html" class="external-link">facet_wrap</a></span><span class="op">(</span><span class="op">~</span><span class="va">seqnames</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">segDists</span><span class="op">$</span><span class="va">Oki_Osa</span> <span class="op">|&gt;</span> <span class="fu">plot_distance_to_next_query_on_coords</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Oki - Osa"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Warning: Removed 4 rows containing missing values (`geom_point()`).</span></span></code></pre>
<p><img src="SegmentMovements_files/figure-html/plot_distance_to_next_query_on_coords_oki_osa_gb-1.svg" width="1440"></p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">segDists</span><span class="op">$</span><span class="va">Oki_Bar</span> <span class="op">|&gt;</span> <span class="fu">plot_distance_to_next_query_on_coords</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Oki - Bar"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Warning: Removed 4 rows containing missing values (`geom_point()`).</span></span></code></pre>
<p><img src="SegmentMovements_files/figure-html/plot_distance_to_next_query_on_coords_oki_bar_gb-1.svg" width="1440"></p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">segDists</span><span class="op">$</span><span class="va">Osa_Bar</span> <span class="op">|&gt;</span> <span class="fu">plot_distance_to_next_query_on_coords</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Osa - Bar"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Warning: Removed 4 rows containing missing values (`geom_point()`).</span></span></code></pre>
<p><img src="SegmentMovements_files/figure-html/plot_distance_to_next_query_on_coords_osa_bar_gb-1.svg" width="1440"></p>
</div>
</div>
<div class="section level2">
<h2 id="session-information">Session information<a class="anchor" aria-label="anchor" href="#session-information"></a>
</h2>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## R version 4.3.1 (2023-06-16)</span></span>
<span><span class="co">## Platform: x86_64-pc-linux-gnu (64-bit)</span></span>
<span><span class="co">## Running under: Debian GNU/Linux 12 (bookworm)</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Matrix products: default</span></span>
<span><span class="co">## BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.11.0 </span></span>
<span><span class="co">## LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.11.0</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## locale:</span></span>
<span><span class="co">##  [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       </span></span>
<span><span class="co">##  [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   </span></span>
<span><span class="co">##  [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          </span></span>
<span><span class="co">## [10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## time zone: Etc/UTC</span></span>
<span><span class="co">## tzcode source: system (glibc)</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## attached base packages:</span></span>
<span><span class="co">## [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span></span>
<span><span class="co">## [8] base     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## other attached packages:</span></span>
<span><span class="co">## [1] BreakpointsData_3.10.1 ggplot2_3.4.3          GenomicBreaks_0.14.2  </span></span>
<span><span class="co">## [4] GenomicRanges_1.52.0   GenomeInfoDb_1.36.1    IRanges_2.34.1        </span></span>
<span><span class="co">## [7] S4Vectors_0.38.1       BiocGenerics_0.46.0   </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## loaded via a namespace (and not attached):</span></span>
<span><span class="co">##   [1] splines_4.3.1               BiocIO_1.10.0              </span></span>
<span><span class="co">##   [3] bitops_1.0-7                tibble_3.2.1               </span></span>
<span><span class="co">##   [5] R.oo_1.25.0                 XML_3.99-0.14              </span></span>
<span><span class="co">##   [7] rpart_4.1.19                lifecycle_1.0.3            </span></span>
<span><span class="co">##   [9] rprojroot_2.0.3             lattice_0.20-45            </span></span>
<span><span class="co">##  [11] MASS_7.3-58.2               backports_1.4.1            </span></span>
<span><span class="co">##  [13] magrittr_2.0.3              Hmisc_5.1-0                </span></span>
<span><span class="co">##  [15] sass_0.4.7                  rmarkdown_2.23             </span></span>
<span><span class="co">##  [17] jquerylib_0.1.4             yaml_2.3.7                 </span></span>
<span><span class="co">##  [19] plotrix_3.8-2               DBI_1.1.3                  </span></span>
<span><span class="co">##  [21] CNEr_1.36.0                 minqa_1.2.5                </span></span>
<span><span class="co">##  [23] RColorBrewer_1.1-3          ade4_1.7-22                </span></span>
<span><span class="co">##  [25] abind_1.4-5                 zlibbioc_1.46.0            </span></span>
<span><span class="co">##  [27] purrr_1.0.2                 R.utils_2.12.2             </span></span>
<span><span class="co">##  [29] RCurl_1.98-1.12             nnet_7.3-18                </span></span>
<span><span class="co">##  [31] pracma_2.4.2                GenomeInfoDbData_1.2.10    </span></span>
<span><span class="co">##  [33] gdata_2.19.0                annotate_1.78.0            </span></span>
<span><span class="co">##  [35] pkgdown_2.0.7               codetools_0.2-19           </span></span>
<span><span class="co">##  [37] DelayedArray_0.26.7         tidyselect_1.2.0           </span></span>
<span><span class="co">##  [39] shape_1.4.6                 farver_2.1.1               </span></span>
<span><span class="co">##  [41] lme4_1.1-34                 matrixStats_1.0.0          </span></span>
<span><span class="co">##  [43] base64enc_0.1-3             GenomicAlignments_1.36.0   </span></span>
<span><span class="co">##  [45] jsonlite_1.8.7              mitml_0.4-5                </span></span>
<span><span class="co">##  [47] Formula_1.2-5               survival_3.5-3             </span></span>
<span><span class="co">##  [49] iterators_1.0.14            systemfonts_1.0.5          </span></span>
<span><span class="co">##  [51] foreach_1.5.2               tools_4.3.1                </span></span>
<span><span class="co">##  [53] ragg_1.2.5                  Rcpp_1.0.11                </span></span>
<span><span class="co">##  [55] glue_1.6.2                  gridExtra_2.3              </span></span>
<span><span class="co">##  [57] pan_1.8                     xfun_0.40                  </span></span>
<span><span class="co">##  [59] MatrixGenerics_1.12.2       EBImage_4.42.0             </span></span>
<span><span class="co">##  [61] dplyr_1.1.3                 withr_2.5.1                </span></span>
<span><span class="co">##  [63] fastmap_1.1.1               boot_1.3-28.1              </span></span>
<span><span class="co">##  [65] fansi_1.0.5                 digest_0.6.33              </span></span>
<span><span class="co">##  [67] R6_2.5.1                    mice_3.16.0                </span></span>
<span><span class="co">##  [69] textshaping_0.3.7           colorspace_2.1-0           </span></span>
<span><span class="co">##  [71] GO.db_3.17.0                gtools_3.9.4               </span></span>
<span><span class="co">##  [73] poweRlaw_0.70.6             jpeg_0.1-10                </span></span>
<span><span class="co">##  [75] RSQLite_2.3.1               weights_1.0.4              </span></span>
<span><span class="co">##  [77] R.methodsS3_1.8.2           utf8_1.2.3                 </span></span>
<span><span class="co">##  [79] tidyr_1.3.0                 generics_0.1.3             </span></span>
<span><span class="co">##  [81] data.table_1.14.8           rtracklayer_1.60.0         </span></span>
<span><span class="co">##  [83] httr_1.4.7                  htmlwidgets_1.6.2          </span></span>
<span><span class="co">##  [85] S4Arrays_1.0.5              pkgconfig_2.0.3            </span></span>
<span><span class="co">##  [87] gtable_0.3.4                blob_1.2.4                 </span></span>
<span><span class="co">##  [89] XVector_0.40.0              htmltools_0.5.6.1          </span></span>
<span><span class="co">##  [91] fftwtools_0.9-11            scales_1.2.1               </span></span>
<span><span class="co">##  [93] Biobase_2.60.0              png_0.1-8                  </span></span>
<span><span class="co">##  [95] knitr_1.44                  heatmaps_1.24.0            </span></span>
<span><span class="co">##  [97] rstudioapi_0.15.0           tzdb_0.4.0                 </span></span>
<span><span class="co">##  [99] reshape2_1.4.4              rjson_0.2.21               </span></span>
<span><span class="co">## [101] checkmate_2.2.0             nlme_3.1-162               </span></span>
<span><span class="co">## [103] nloptr_2.0.3                cachem_1.0.8               </span></span>
<span><span class="co">## [105] stringr_1.5.0               KernSmooth_2.23-20         </span></span>
<span><span class="co">## [107] parallel_4.3.1              genoPlotR_0.8.11           </span></span>
<span><span class="co">## [109] foreign_0.8-84              AnnotationDbi_1.62.2       </span></span>
<span><span class="co">## [111] restfulr_0.0.15             desc_1.4.2                 </span></span>
<span><span class="co">## [113] pillar_1.9.0                grid_4.3.1                 </span></span>
<span><span class="co">## [115] vctrs_0.6.3                 jomo_2.7-6                 </span></span>
<span><span class="co">## [117] xtable_1.8-4                cluster_2.1.4              </span></span>
<span><span class="co">## [119] htmlTable_2.4.1             evaluate_0.22              </span></span>
<span><span class="co">## [121] readr_2.1.4                 cli_3.6.1                  </span></span>
<span><span class="co">## [123] locfit_1.5-9.8              compiler_4.3.1             </span></span>
<span><span class="co">## [125] Rsamtools_2.16.0            rlang_1.1.1                </span></span>
<span><span class="co">## [127] crayon_1.5.2                labeling_0.4.3             </span></span>
<span><span class="co">## [129] plyr_1.8.8                  fs_1.6.3                   </span></span>
<span><span class="co">## [131] stringi_1.7.12              BiocParallel_1.34.2        </span></span>
<span><span class="co">## [133] munsell_0.5.0               Biostrings_2.68.1          </span></span>
<span><span class="co">## [135] tiff_0.1-11                 glmnet_4.1-7               </span></span>
<span><span class="co">## [137] Matrix_1.5-3                BSgenome_1.68.0            </span></span>
<span><span class="co">## [139] hms_1.1.3                   bit64_4.0.5                </span></span>
<span><span class="co">## [141] KEGGREST_1.40.0             SummarizedExperiment_1.30.2</span></span>
<span><span class="co">## [143] broom_1.0.5                 memoise_2.0.1              </span></span>
<span><span class="co">## [145] bslib_0.5.1                 bit_4.0.5</span></span></code></pre>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Aleksandra Bliznina, Michael Mansfield, Charles Plessy.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
