<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Segment movement distances â€¢ OikScrambling</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Segment movement distances">
<meta property="og:description" content="OikScrambling">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">OikScrambling</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.8.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/OikScrambling.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/CNEs.html">Conserved non-coding elements</a>
    </li>
    <li>
      <a href="../articles/CoalescingCutoffs.html">Modifying coalescent cutoffs</a>
    </li>
    <li>
      <a href="../articles/ColinearityInterruptors.html">Regions interrupting colinearity</a>
    </li>
    <li>
      <a href="../articles/DistanceMeasures.html">Distance Measures</a>
    </li>
    <li>
      <a href="../articles/GenomicFeatures.html">Genomic features of the breakpoints</a>
    </li>
    <li>
      <a href="../articles/Inversions.html">Inversions</a>
    </li>
    <li>
      <a href="../articles/LoadGenomicBreaks.html">Load genomic break data</a>
    </li>
    <li>
      <a href="../articles/OrthogroupSyntenies.html">Orthogroup syntenies</a>
    </li>
    <li>
      <a href="../articles/OxfordPlots.html">Whole-genome dot-plots</a>
    </li>
    <li>
      <a href="../articles/PWM.html">Motif finding and analysis</a>
    </li>
    <li>
      <a href="../articles/ParallelPlots.html">Whole-chromosome parallel plots</a>
    </li>
    <li>
      <a href="../articles/RegionWidths.html">Width of the aligned and mapped regions.</a>
    </li>
    <li>
      <a href="../articles/RegionsOfInterest.html">Regions interrupting colinearity</a>
    </li>
    <li>
      <a href="../articles/RepeatRegions.html">Repeat regions</a>
    </li>
    <li>
      <a href="../articles/SegmentMovements.html">Segment movement distances</a>
    </li>
    <li>
      <a href="../articles/TFBS.html">Transcription factor binding sites</a>
    </li>
    <li>
      <a href="../articles/Translocations.html">Translocations</a>
    </li>
    <li>
      <a href="../articles/test.html">Regions interrupting colinearity</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/oist/LuscombeU_OikScrambling/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="SegmentMovements_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Segment movement distances</h1>
                        <h4 data-toc-skip class="author">Michael Mansfield</h4>
                        <h4 data-toc-skip class="author">Charles Plessy</h4>
            
            <h4 data-toc-skip class="date">09 February, 2022</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/oist/LuscombeU_OikScrambling/blob/HEAD/vignettes/SegmentMovements.Rmd" class="external-link"><code>vignettes/SegmentMovements.Rmd</code></a></small>
      <div class="hidden name"><code>SegmentMovements.Rmd</code></div>

    </div>

    
    
<div class="section level1">
<h1 id="load-packages-and-data">Load packages and data<a class="anchor" aria-label="anchor" href="#load-packages-and-data"></a>
</h1>
<p>See the <code><a href="../articles/OikScrambling.html">vignette("OikScrambling", package = "OikScrambling")</a></code> for general details on package and data load.</p>
<p>See <code><a href="../articles/LoadGenomicBreaks.html">vignette("LoadGenomicBreaks", package = "OikScrambling")</a></code> for how the different <code>GBreaks</code> objects are prepared.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="op">{</span>
  <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://oist.github.io/GenomicBreaks" class="external-link">'GenomicBreaks'</a></span><span class="op">)</span>
  <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://ggplot2.tidyverse.org" class="external-link">'ggplot2'</a></span><span class="op">)</span>
  <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st">"BreakpointsData"</span><span class="op">)</span>
<span class="op">}</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/load.html" class="external-link">load</a></span><span class="op">(</span><span class="st">"BreakPoints.Rdata"</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level1">
<h1 id="segment-movement-distances">Segment movement distances<a class="anchor" aria-label="anchor" href="#segment-movement-distances"></a>
</h1>
<p>Whenever a collinear segment moves between species, if the homologous segment is located on an equivalent chromosome, a distance can be calculated.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Find distance between two equivalent GRanges.</span>
<span class="co"># Input: a GBreaks object</span>
<span class="co"># Output: Absolute distance between the target and query ranges based on some metric ("start", "end", or "midpoint")</span>
<span class="va">GBreaksDist</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">gb</span>, <span class="va">distMetric</span><span class="op">=</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"midpoint"</span>, <span class="st">"start"</span>, <span class="st">"end"</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span>
  <span class="va">distMetric</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/match.arg.html" class="external-link">match.arg</a></span><span class="op">(</span><span class="va">distMetric</span><span class="op">)</span>
  <span class="va">distFun</span> <span class="op">&lt;-</span> <span class="kw"><a href="https://rdrr.io/r/base/switch.html" class="external-link">switch</a></span><span class="op">(</span><span class="va">distMetric</span>,
                    start    <span class="op">=</span> <span class="va">start</span>,
                    end      <span class="op">=</span> <span class="va">end</span>,
                    midpoint <span class="op">=</span> \<span class="op">(</span><span class="va">gr</span><span class="op">)</span> <span class="va">gr</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/intra-range-methods.html" class="external-link">resize</a></span><span class="op">(</span><span class="fl">1</span>, fix <span class="op">=</span> <span class="st">"center"</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/start.html" class="external-link">start</a></span><span class="op">(</span><span class="op">)</span> <span class="op">)</span>
  <span class="fu"><a href="https://rdrr.io/r/base/MathFun.html" class="external-link">abs</a></span><span class="op">(</span><span class="fu">distFun</span><span class="op">(</span><span class="va">gb</span><span class="op">)</span> <span class="op">-</span> <span class="fu">distFun</span><span class="op">(</span><span class="va">gb</span><span class="op">$</span><span class="va">query</span><span class="op">)</span><span class="op">)</span>
<span class="op">}</span>

<span class="va">filterSyntenicPairs</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">gb</span>, <span class="va">chromosomePairing</span><span class="op">)</span> <span class="op">{</span>
  <span class="va">syntenic_pairs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="va">chromosomePairing</span><span class="op">[[</span><span class="st">"sp1"</span><span class="op">]</span><span class="op">]</span>, <span class="va">chromosomePairing</span><span class="op">[[</span><span class="st">"sp2"</span><span class="op">]</span><span class="op">]</span>, sep <span class="op">=</span> <span class="st">"_"</span><span class="op">)</span>
  <span class="va">is_syntenic</span>    <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqnames</a></span><span class="op">(</span><span class="va">gb</span><span class="op">)</span>,               <span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqnames</a></span><span class="op">(</span><span class="va">gb</span><span class="op">$</span><span class="va">query</span><span class="op">)</span>,         sep <span class="op">=</span> <span class="st">"_"</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html" class="external-link">%in%</a></span> <span class="va">syntenic_pairs</span>
  <span class="va">gb</span><span class="op">[</span><span class="va">is_syntenic</span><span class="op">]</span>
<span class="op">}</span>

<span class="va">chromDist</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">gb</span>, <span class="va">chromosomePairing</span><span class="op">=</span><span class="cn">NULL</span>, <span class="va">distMetric</span><span class="op">=</span><span class="st">"midpoint"</span><span class="op">)</span><span class="op">{</span>
  <span class="co"># Subset to include only ranges in which all chrom in sp1 are paired to their equivalent in sp2</span>
  <span class="co"># Note that a similar function could be used to compare the identity of chromosome arms</span>
  <span class="va">gb</span> <span class="op">&lt;-</span> <span class="fu">filterSyntenicPairs</span><span class="op">(</span><span class="va">gb</span>, <span class="va">chromosomePairing</span><span class="op">)</span>
  
  <span class="co"># Now that the GRanges are only on equivalent chromosomes, calculate the distances between them.</span>
  <span class="co"># This can be "start", "end", or "midpoint" distance.</span>
  <span class="va">gb</span><span class="op">$</span><span class="va">distance</span> <span class="op">&lt;-</span> <span class="fu">GBreaksDist</span><span class="op">(</span><span class="va">gb</span>, distMetric<span class="op">=</span><span class="va">distMetric</span><span class="op">)</span>
  <span class="va">gb</span>
<span class="op">}</span>

<span class="co"># Unfortunately, the use of upper and lower case in chromosome names was not</span>
<span class="co"># standardised when assembling the genomes.</span>
<span class="va">synPairs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/SimpleList-class.html" class="external-link">SimpleList</a></span><span class="op">(</span><span class="op">)</span>
<span class="va">synPairs</span><span class="op">$</span><span class="va">chr_Chr</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="st">"sp1"</span><span class="op">=</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"chr1"</span>, <span class="st">"chr2"</span>, <span class="st">"PAR"</span>, <span class="st">"XSR"</span>, <span class="st">"YSR"</span><span class="op">)</span>,
                         <span class="st">"sp2"</span><span class="op">=</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Chr1"</span>, <span class="st">"Chr2"</span>, <span class="st">"PAR"</span>, <span class="st">"XSR"</span>, <span class="st">"YSR"</span><span class="op">)</span><span class="op">)</span>
<span class="va">synPairs</span><span class="op">$</span><span class="va">Chr_Chr</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="st">"sp1"</span><span class="op">=</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Chr1"</span>, <span class="st">"Chr2"</span>, <span class="st">"PAR"</span>, <span class="st">"XSR"</span>, <span class="st">"YSR"</span><span class="op">)</span>,
                         <span class="st">"sp2"</span><span class="op">=</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Chr1"</span>, <span class="st">"Chr2"</span>, <span class="st">"PAR"</span>, <span class="st">"XSR"</span>, <span class="st">"YSR"</span><span class="op">)</span><span class="op">)</span>

<span class="va">segDists</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/SimpleList-class.html" class="external-link">SimpleList</a></span><span class="op">(</span><span class="op">)</span>
<span class="va">segDists</span><span class="op">$</span><span class="va">Oki_Osa</span> <span class="op">&lt;-</span> <span class="fu">chromDist</span><span class="op">(</span><span class="va">gbs</span><span class="op">$</span><span class="va">Oki_Osa</span>, <span class="va">synPairs</span><span class="op">$</span><span class="va">chr_Chr</span><span class="op">)</span>
<span class="va">segDists</span><span class="op">$</span><span class="va">Oki_Bar</span> <span class="op">&lt;-</span> <span class="fu">chromDist</span><span class="op">(</span><span class="va">gbs</span><span class="op">$</span><span class="va">Oki_Bar</span>, <span class="va">synPairs</span><span class="op">$</span><span class="va">chr_Chr</span><span class="op">)</span>
<span class="va">segDists</span><span class="op">$</span><span class="va">Osa_Bar</span> <span class="op">&lt;-</span> <span class="fu">chromDist</span><span class="op">(</span><span class="va">gbs</span><span class="op">$</span><span class="va">Osa_Bar</span>, <span class="va">synPairs</span><span class="op">$</span><span class="va">Chr_Chr</span><span class="op">)</span></code></pre></div>
<div class="section level2">
<h2 id="plotting-segment-movement-distances">Plotting segment movement distances<a class="anchor" aria-label="anchor" href="#plotting-segment-movement-distances"></a>
</h2>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">plotSegDists</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">segDist</span>, <span class="va">facet.by</span><span class="op">=</span><span class="st">'seqnames~Arm'</span>, <span class="va">binWidth</span><span class="op">=</span><span class="fl">100000</span><span class="op">)</span> <span class="op">{</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="va">segDist</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x<span class="op">=</span><span class="va">distance</span>, fill <span class="op">=</span> <span class="va">Arm</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_histogram.html" class="external-link">geom_histogram</a></span><span class="op">(</span>binwidth <span class="op">=</span> <span class="va">binWidth</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html" class="external-link">facet_wrap</a></span><span class="op">(</span><span class="va">facet.by</span>, scales<span class="op">=</span><span class="st">"free_x"</span>, ncol<span class="op">=</span><span class="fl">2</span><span class="op">)</span>
<span class="op">}</span>
<span class="fu">plotSegDists</span><span class="op">(</span><span class="va">segDists</span><span class="op">$</span><span class="va">Oki_Osa</span>, facet.by<span class="op">=</span><span class="st">'seqnames~Arm'</span>, binWidth<span class="op">=</span><span class="fl">100000</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Oki-Osa: segment movement distances (uncoalesced)"</span><span class="op">)</span></code></pre></div>
<p><img src="SegmentMovements_files/figure-html/plot_movement_distances_oki_osa-1.svg" width="1440"></p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">plotSegDists</span><span class="op">(</span><span class="va">segDists</span><span class="op">$</span><span class="va">Oki_Bar</span>, facet.by<span class="op">=</span><span class="st">'seqnames~Arm'</span>, binWidth<span class="op">=</span><span class="fl">100000</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Oki-Bar: segment movement distances (uncoalesced)"</span><span class="op">)</span></code></pre></div>
<p><img src="SegmentMovements_files/figure-html/plot_movement_distances_oki_bar-1.svg" width="1440"></p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">plotSegDists</span><span class="op">(</span><span class="va">segDists</span><span class="op">$</span><span class="va">Osa_Bar</span>, facet.by<span class="op">=</span><span class="st">'seqnames~Arm'</span>, binWidth<span class="op">=</span><span class="fl">100000</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Osa-Bar: segment movement distances (uncoalesced)"</span><span class="op">)</span></code></pre></div>
<p><img src="SegmentMovements_files/figure-html/plot_movement_distances_osa_bar-1.svg" width="1440"></p>
</div>
<div class="section level2">
<h2 id="segment-movement-distances-coalesced">Segment movement distances (coalesced)<a class="anchor" aria-label="anchor" href="#segment-movement-distances-coalesced"></a>
</h2>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">segDistsCoa</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/SimpleList-class.html" class="external-link">SimpleList</a></span><span class="op">(</span><span class="op">)</span>
<span class="va">segDistsCoa</span><span class="op">$</span><span class="va">Oki_Osa</span> <span class="op">&lt;-</span> <span class="fu">chromDist</span><span class="op">(</span><span class="va">coa</span><span class="op">$</span><span class="va">Oki_Osa</span>, <span class="va">synPairs</span><span class="op">$</span><span class="va">chr_Chr</span><span class="op">)</span>
<span class="va">segDistsCoa</span><span class="op">$</span><span class="va">Oki_Bar</span> <span class="op">&lt;-</span> <span class="fu">chromDist</span><span class="op">(</span><span class="va">coa</span><span class="op">$</span><span class="va">Oki_Bar</span>, <span class="va">synPairs</span><span class="op">$</span><span class="va">chr_Chr</span><span class="op">)</span>
<span class="va">segDistsCoa</span><span class="op">$</span><span class="va">Osa_Bar</span> <span class="op">&lt;-</span> <span class="fu">chromDist</span><span class="op">(</span><span class="va">coa</span><span class="op">$</span><span class="va">Osa_Bar</span>, <span class="va">synPairs</span><span class="op">$</span><span class="va">Chr_Chr</span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">plotSegDists</span><span class="op">(</span><span class="va">segDistsCoa</span><span class="op">$</span><span class="va">Oki_Osa</span>, facet.by<span class="op">=</span><span class="st">'seqnames~Arm'</span>, binWidth<span class="op">=</span><span class="fl">100000</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Oki-Osa: segment movement distances (coalesced)"</span><span class="op">)</span></code></pre></div>
<p><img src="SegmentMovements_files/figure-html/plot_movement_distances_oki_osa_coa-1.svg" width="1440"></p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">plotSegDists</span><span class="op">(</span><span class="va">segDistsCoa</span><span class="op">$</span><span class="va">Oki_Bar</span>, facet.by<span class="op">=</span><span class="st">'seqnames~Arm'</span>, binWidth<span class="op">=</span><span class="fl">100000</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Oki-Bar: segment movement distances (coalesced)"</span><span class="op">)</span></code></pre></div>
<p><img src="SegmentMovements_files/figure-html/plot_movement_distances_oki_bar_coa-1.svg" width="1440"></p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">plotSegDists</span><span class="op">(</span><span class="va">segDistsCoa</span><span class="op">$</span><span class="va">Osa_Bar</span>, facet.by<span class="op">=</span><span class="st">'seqnames~Arm'</span>, binWidth<span class="op">=</span><span class="fl">100000</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Osa-Bar: segment movement distances (coalesced)"</span><span class="op">)</span></code></pre></div>
<p><img src="SegmentMovements_files/figure-html/plot_movement_distances_osa_bar_coa-1.svg" width="1440"></p>
</div>
<div class="section level2">
<h2 id="plotting-segment-movements-relative-to-coordinates">Plotting segment movements relative to coordinates<a class="anchor" aria-label="anchor" href="#plotting-segment-movements-relative-to-coordinates"></a>
</h2>
<p>Uses, start coordinate as a proxy for midpoint, but the difference should be hardly perceptible.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">plot_movement_distance_on_coords</span> <span class="op">&lt;-</span> \<span class="op">(</span><span class="va">gb</span><span class="op">)</span> <span class="va">gb</span> |&gt;
  <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="op">)</span> |&gt;
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span><span class="va">start</span>, <span class="va">distance</span>, col<span class="op">=</span><span class="va">Arm</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">xlab</a></span><span class="op">(</span><span class="st">"Start on target"</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ylab</a></span><span class="op">(</span><span class="st">"Distance between midpoints on query and target"</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html" class="external-link">facet_wrap</a></span><span class="op">(</span><span class="op">~</span><span class="va">seqnames</span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">segDists</span><span class="op">$</span><span class="va">Oki_Osa</span> |&gt; <span class="fu">plot_movement_distance_on_coords</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Oki - Osa"</span><span class="op">)</span></code></pre></div>
<p><img src="SegmentMovements_files/figure-html/plot_movement_distance_on_coords_oki_osa_gb-1.svg" width="1440"></p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">segDists</span><span class="op">$</span><span class="va">Oki_Bar</span> |&gt; <span class="fu">plot_movement_distance_on_coords</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Oki - Bar"</span><span class="op">)</span></code></pre></div>
<p><img src="SegmentMovements_files/figure-html/plot_movement_distance_on_coords_oki_bar_gb-1.svg" width="1440"></p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">segDists</span><span class="op">$</span><span class="va">Osa_Bar</span> |&gt; <span class="fu">plot_movement_distance_on_coords</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Osa - Bar"</span><span class="op">)</span></code></pre></div>
<p><img src="SegmentMovements_files/figure-html/plot_movement_distance_on_coords_osa_bar_gb-1.svg" width="1440"></p>
</div>
<div class="section level2">
<h2 id="plotting-distance-between-query-ranges-relative-to-target-coordinates">Plotting distance between query ranges relative to target coordinates<a class="anchor" aria-label="anchor" href="#plotting-distance-between-query-ranges-relative-to-target-coordinates"></a>
</h2>
<p>This assumes that the center an unaligned region would be plotted on the same pixel as the center of its flanking aligned regions.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">plot_distance_to_next_query_on_coords</span> <span class="op">&lt;-</span> \<span class="op">(</span><span class="va">gb</span><span class="op">)</span> <span class="va">gb</span> |&gt;
  <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/dist2next.html" class="external-link">dist2next</a></span><span class="op">(</span><span class="op">)</span> |&gt;
  <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="op">)</span> |&gt;
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span><span class="va">end</span> <span class="op">+</span> <span class="fl">1</span>, <span class="va">qdist</span>, col<span class="op">=</span><span class="va">Arm</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">xlab</a></span><span class="op">(</span><span class="st">"Start unaligned region on target"</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ylab</a></span><span class="op">(</span><span class="st">"Distance between flanking regions on query"</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html" class="external-link">facet_wrap</a></span><span class="op">(</span><span class="op">~</span><span class="va">seqnames</span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">segDists</span><span class="op">$</span><span class="va">Oki_Osa</span> |&gt; <span class="fu">plot_distance_to_next_query_on_coords</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Oki - Osa"</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Warning: Removed 4 rows containing missing values (geom_point).</span></code></pre>
<p><img src="SegmentMovements_files/figure-html/plot_distance_to_next_query_on_coords_oki_osa_gb-1.svg" width="1440"></p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">segDists</span><span class="op">$</span><span class="va">Oki_Bar</span> |&gt; <span class="fu">plot_distance_to_next_query_on_coords</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Oki - Bar"</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Warning: Removed 4 rows containing missing values (geom_point).</span></code></pre>
<p><img src="SegmentMovements_files/figure-html/plot_distance_to_next_query_on_coords_oki_bar_gb-1.svg" width="1440"></p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">segDists</span><span class="op">$</span><span class="va">Osa_Bar</span> |&gt; <span class="fu">plot_distance_to_next_query_on_coords</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Osa - Bar"</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Warning: Removed 4 rows containing missing values (geom_point).</span></code></pre>
<p><img src="SegmentMovements_files/figure-html/plot_distance_to_next_query_on_coords_osa_bar_gb-1.svg" width="1440"></p>
</div>
</div>
<div class="section level1">
<h1 id="session-information">Session information<a class="anchor" aria-label="anchor" href="#session-information"></a>
</h1>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## R version 4.1.2 (2021-11-01)</span>
<span class="co">## Platform: x86_64-pc-linux-gnu (64-bit)</span>
<span class="co">## Running under: Debian GNU/Linux bookworm/sid</span>
<span class="co">## </span>
<span class="co">## Matrix products: default</span>
<span class="co">## BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.10.0</span>
<span class="co">## LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.10.0</span>
<span class="co">## </span>
<span class="co">## locale:</span>
<span class="co">##  [1] LC_CTYPE=en_GB.UTF-8       LC_NUMERIC=C              </span>
<span class="co">##  [3] LC_TIME=en_GB.UTF-8        LC_COLLATE=en_GB.UTF-8    </span>
<span class="co">##  [5] LC_MONETARY=en_GB.UTF-8    LC_MESSAGES=en_GB.UTF-8   </span>
<span class="co">##  [7] LC_PAPER=en_GB.UTF-8       LC_NAME=C                 </span>
<span class="co">##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span>
<span class="co">## [11] LC_MEASUREMENT=en_GB.UTF-8 LC_IDENTIFICATION=C       </span>
<span class="co">## </span>
<span class="co">## attached base packages:</span>
<span class="co">## [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span>
<span class="co">## [8] base     </span>
<span class="co">## </span>
<span class="co">## other attached packages:</span>
<span class="co">## [1] BreakpointsData_3.8.0 ggplot2_3.3.5         GenomicBreaks_0.12.2 </span>
<span class="co">## [4] GenomicRanges_1.46.1  GenomeInfoDb_1.30.0   IRanges_2.28.0       </span>
<span class="co">## [7] S4Vectors_0.32.3      BiocGenerics_0.40.0  </span>
<span class="co">## </span>
<span class="co">## loaded via a namespace (and not attached):</span>
<span class="co">##   [1] backports_1.4.1             Hmisc_4.6-0                </span>
<span class="co">##   [3] systemfonts_1.0.3           plyr_1.8.6                 </span>
<span class="co">##   [5] splines_4.1.2               BiocParallel_1.28.3        </span>
<span class="co">##   [7] digest_0.6.29               htmltools_0.5.2            </span>
<span class="co">##   [9] tiff_0.1-10                 GO.db_3.14.0               </span>
<span class="co">##  [11] gdata_2.18.0                fansi_0.5.0                </span>
<span class="co">##  [13] magrittr_2.0.1              checkmate_2.0.0            </span>
<span class="co">##  [15] memoise_2.0.1               cluster_2.1.2              </span>
<span class="co">##  [17] tzdb_0.2.0                  Biostrings_2.62.0          </span>
<span class="co">##  [19] readr_2.1.1                 annotate_1.72.0            </span>
<span class="co">##  [21] matrixStats_0.61.0          R.utils_2.11.0             </span>
<span class="co">##  [23] pkgdown_2.0.1               jpeg_0.1-9                 </span>
<span class="co">##  [25] colorspace_2.0-2            blob_1.2.2                 </span>
<span class="co">##  [27] textshaping_0.3.6           xfun_0.29                  </span>
<span class="co">##  [29] dplyr_1.0.7                 crayon_1.4.2               </span>
<span class="co">##  [31] RCurl_1.98-1.5              jsonlite_1.7.2             </span>
<span class="co">##  [33] lme4_1.1-27.1               survival_3.2-13            </span>
<span class="co">##  [35] glue_1.6.0                  gtable_0.3.0               </span>
<span class="co">##  [37] zlibbioc_1.40.0             XVector_0.34.0             </span>
<span class="co">##  [39] DelayedArray_0.20.0         weights_1.0.4              </span>
<span class="co">##  [41] abind_1.4-5                 scales_1.1.1               </span>
<span class="co">##  [43] DBI_1.1.2                   Rcpp_1.0.7                 </span>
<span class="co">##  [45] plotrix_3.8-2               genoPlotR_0.8.11           </span>
<span class="co">##  [47] xtable_1.8-4                htmlTable_2.3.0            </span>
<span class="co">##  [49] foreign_0.8-81              bit_4.0.4                  </span>
<span class="co">##  [51] Formula_1.2-4               htmlwidgets_1.5.4          </span>
<span class="co">##  [53] httr_1.4.2                  RColorBrewer_1.1-2         </span>
<span class="co">##  [55] ellipsis_0.3.2              mice_3.14.0                </span>
<span class="co">##  [57] farver_2.1.0                pkgconfig_2.0.3            </span>
<span class="co">##  [59] XML_3.99-0.8                R.methodsS3_1.8.1          </span>
<span class="co">##  [61] nnet_7.3-16                 sass_0.4.0                 </span>
<span class="co">##  [63] locfit_1.5-9.4              utf8_1.2.2                 </span>
<span class="co">##  [65] labeling_0.4.2              tidyselect_1.1.1           </span>
<span class="co">##  [67] rlang_0.4.12                reshape2_1.4.4             </span>
<span class="co">##  [69] AnnotationDbi_1.56.2        munsell_0.5.0              </span>
<span class="co">##  [71] tools_4.1.2                 cachem_1.0.6               </span>
<span class="co">##  [73] generics_0.1.1              RSQLite_2.2.9              </span>
<span class="co">##  [75] ade4_1.7-18                 broom_0.7.10               </span>
<span class="co">##  [77] evaluate_0.14               stringr_1.4.0              </span>
<span class="co">##  [79] fastmap_1.1.0               fftwtools_0.9-11           </span>
<span class="co">##  [81] yaml_2.2.1                  ragg_1.2.1                 </span>
<span class="co">##  [83] knitr_1.37                  bit64_4.0.5                </span>
<span class="co">##  [85] fs_1.5.2                    purrr_0.3.4                </span>
<span class="co">##  [87] heatmaps_1.18.0             KEGGREST_1.34.0            </span>
<span class="co">##  [89] nlme_3.1-153                R.oo_1.24.0                </span>
<span class="co">##  [91] poweRlaw_0.70.6             pracma_2.3.6               </span>
<span class="co">##  [93] compiler_4.1.2              rstudioapi_0.13            </span>
<span class="co">##  [95] png_0.1-7                   tibble_3.1.6               </span>
<span class="co">##  [97] bslib_0.3.1                 stringi_1.7.6              </span>
<span class="co">##  [99] highr_0.9                   desc_1.4.0                 </span>
<span class="co">## [101] lattice_0.20-45             CNEr_1.30.0                </span>
<span class="co">## [103] Matrix_1.4-0                nloptr_1.2.2.3             </span>
<span class="co">## [105] vctrs_0.3.8                 pillar_1.6.4               </span>
<span class="co">## [107] lifecycle_1.0.1             jquerylib_0.1.4            </span>
<span class="co">## [109] data.table_1.14.2           bitops_1.0-7               </span>
<span class="co">## [111] rtracklayer_1.54.0          R6_2.5.1                   </span>
<span class="co">## [113] BiocIO_1.4.0                latticeExtra_0.6-29        </span>
<span class="co">## [115] KernSmooth_2.23-20          gridExtra_2.3              </span>
<span class="co">## [117] boot_1.3-28                 MASS_7.3-54                </span>
<span class="co">## [119] gtools_3.9.2                assertthat_0.2.1           </span>
<span class="co">## [121] SummarizedExperiment_1.24.0 rprojroot_2.0.2            </span>
<span class="co">## [123] rjson_0.2.20                withr_2.4.3                </span>
<span class="co">## [125] GenomicAlignments_1.30.0    Rsamtools_2.10.0           </span>
<span class="co">## [127] GenomeInfoDbData_1.2.7      parallel_4.1.2             </span>
<span class="co">## [129] hms_1.1.1                   EBImage_4.36.0             </span>
<span class="co">## [131] grid_4.1.2                  rpart_4.1-15               </span>
<span class="co">## [133] minqa_1.2.4                 tidyr_1.1.4                </span>
<span class="co">## [135] rmarkdown_2.11              MatrixGenerics_1.6.0       </span>
<span class="co">## [137] Biobase_2.54.0              base64enc_0.1-3            </span>
<span class="co">## [139] restfulr_0.0.13</span></code></pre>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Aleksandra Bliznina, Michael Mansfield, Charles Plessy.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.1.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
