<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Alignment line plots, 10×10 Mbp windows • OikScrambling</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Alignment line plots, 10×10 Mbp windows">
<meta property="og:description" content="OikScrambling">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">OikScrambling</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">5.0.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/OikScrambling.html">Get started</a>
</li>
<li>
  <a href="../articles/LoadGenomicBreaks.html">Data load</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Supplemental material
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/OxfordPlots.html">Figure 1A DNA line plots</a>
    </li>
    <li>
      <a href="../articles/ParallelPlots.html">Figure 1B chromosome parallel plots</a>
    </li>
    <li>
      <a href="../articles/DistanceMeasures.html">Table 1 scrambling index</a>
    </li>
    <li>
      <a href="../articles/ChromosomePlots.html">Figure 5 genome-wide feature patterns</a>
    </li>
    <li>
      <a href="../articles/SquareWindowPlots.html">Figure 7 DNA line plots</a>
    </li>
    <li>
      <a href="../articles/ScalingBreakpoints.html">Figure 7 breakpoint number barplots</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/CNEs.html">Conserved non-coding elements</a>
    </li>
    <li>
      <a href="../articles/ChromosomePlots.html">Chromosome-scale feature density plots</a>
    </li>
    <li>
      <a href="../articles/CoalescingCutoffs.html">Modifying coalescent cutoffs</a>
    </li>
    <li>
      <a href="../articles/ColinearityInterruptors.html">Regions interrupting colinearity</a>
    </li>
    <li>
      <a href="../articles/DistanceMeasures.html">Distance Measures</a>
    </li>
    <li>
      <a href="../articles/GenomicFeatures.html">Genomic features of the breakpoints</a>
    </li>
    <li>
      <a href="../articles/GenomicHeatmaps.html">Genomic features of the breakpoints (heatmaps)</a>
    </li>
    <li>
      <a href="../articles/Inversions.html">Inversions</a>
    </li>
    <li>
      <a href="../articles/LoadGenomicBreaks.html">Load genomic break data</a>
    </li>
    <li>
      <a href="../articles/OperonConservation.html">Operon conservation analysis</a>
    </li>
    <li>
      <a href="../articles/OrthogroupSyntenies.html">Orthogroup syntenies</a>
    </li>
    <li>
      <a href="../articles/OxfordPlots.html">Alignment line plots, whole genomes</a>
    </li>
    <li>
      <a href="../articles/PAC3.html">Region of interest: PAC3</a>
    </li>
    <li>
      <a href="../articles/PWM.html">Motif search and analysis</a>
    </li>
    <li>
      <a href="../articles/ParallelPlots.html">Whole-chromosome parallel plots</a>
    </li>
    <li>
      <a href="../articles/RegionWidths.html">Width of the aligned and mapped regions.</a>
    </li>
    <li>
      <a href="../articles/RegionsOfInterest.html">Regions of interest</a>
    </li>
    <li>
      <a href="../articles/RepeatRegions.html">Repeat regions</a>
    </li>
    <li>
      <a href="../articles/ScalingBreakpoints.html">Scaling breakpoints</a>
    </li>
    <li>
      <a href="../articles/SegmentMovements.html">Segment movement distances</a>
    </li>
    <li>
      <a href="../articles/SquareWindowPlots.html">Alignment line plots, 10×10 Mbp windows</a>
    </li>
    <li>
      <a href="../articles/TFBS.html">Transcription factor binding sites</a>
    </li>
    <li>
      <a href="../articles/TandemSiteDuplications.html">Tandem site duplications</a>
    </li>
    <li>
      <a href="../articles/Translocations.html">Translocations</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/oist/LuscombeU_OikScrambling/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Alignment line plots, 10×10 Mbp windows</h1>
                        <h4 data-toc-skip class="author">Charles
Plessy</h4>
            
            <h4 data-toc-skip class="date">13 October, 2023</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/oist/LuscombeU_OikScrambling/blob/HEAD/vignettes/SquareWindowPlots.Rmd" class="external-link"><code>vignettes/SquareWindowPlots.Rmd</code></a></small>
      <div class="hidden name"><code>SquareWindowPlots.Rmd</code></div>

    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">knitr</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/knitr/man/opts_knit.html" class="external-link">opts_knit</a></span><span class="op">$</span><span class="fu">set</span><span class="op">(</span>cache <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>This vignette produces alignment line plots of 10 × 10 Mbp windows in
pairs of genomes to illustrate differences in the extent of scrambling
in <em>Oikopleura</em> compared to equivalent pairs in other genera. The
size of the windows is fixed to allow comparison at the same scale. To
avoid arbitrary choices, the windows are placed at the centre of the
largest pair of homologous chromosomes (or arms, in case of
<em>Drosophila</em>).</p>
<p><strong>Figure 7 alignment panels were generated in this
vignette.</strong> They are presented on white background.</p>
<p>Extra alignments on grey background show how the non-chromosomal
assemblies were scaffolded by hand in the region selected for the window
plot.</p>
</div>
<div class="section level2">
<h2 id="setup">Setup<a class="anchor" aria-label="anchor" href="#setup"></a>
</h2>
<div class="section level3">
<h3 id="load-r-pacakges-and-data">Load R pacakges and data<a class="anchor" aria-label="anchor" href="#load-r-pacakges-and-data"></a>
</h3>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/oist/LuscombeU_OikScrambling" class="external-link">'OikScrambling'</a></span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">genomes</span> <span class="op">&lt;-</span> <span class="fu">OikScrambling</span><span class="fu">:::</span><span class="fu"><a href="../reference/loadAllGenomes.html">loadAllGenomes</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Warning in runHook(".onLoad", env, package.lib, package): input string</span></span>
<span><span class="co">## 'Génoscope' cannot be translated from 'ANSI_X3.4-1968' to UTF-8, but is valid</span></span>
<span><span class="co">## UTF-8</span></span>
<span></span>
<span><span class="co">## Warning in runHook(".onLoad", env, package.lib, package): input string</span></span>
<span><span class="co">## 'Génoscope' cannot be translated from 'ANSI_X3.4-1968' to UTF-8, but is valid</span></span>
<span><span class="co">## UTF-8</span></span></code></pre>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/load.html" class="external-link">load</a></span><span class="op">(</span><span class="st">"BreakPoints.Rdata"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="define-custom-function">Define custom function<a class="anchor" aria-label="anchor" href="#define-custom-function"></a>
</h3>
<p>The <code>mkOxPlotWindow</code> function searches for the longest
<em>sequence features</em> (chromosomes, scaffolds, …) in each genome
and selects a 10 × 10 window centered at their midpoints.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mkOxPlotWindow</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">gb</span>, <span class="va">win</span> <span class="op">=</span> <span class="fl">5e6</span>, <span class="va">size</span> <span class="op">=</span> <span class="fl">2</span>, <span class="va">centre</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'midpoint'</span>, <span class="st">'maxwidth'</span><span class="op">)</span>, <span class="va">...</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="va">centre</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/match.arg.html" class="external-link">match.arg</a></span><span class="op">(</span><span class="va">centre</span><span class="op">)</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqlevels</a></span><span class="op">(</span><span class="va">gb</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqlevelsInUse</a></span><span class="op">(</span><span class="va">gb</span><span class="op">)</span></span>
<span>  <span class="va">gb</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/forceSeqLengths.html" class="external-link">forceSeqLengths</a></span><span class="op">(</span><span class="va">gb</span><span class="op">)</span></span>
<span>  <span class="va">longest_target_seq</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqlengths</a></span><span class="op">(</span><span class="va">gb</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/sort.html" class="external-link">sort</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">tail</a></span><span class="op">(</span><span class="fl">1</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="op">)</span></span>
<span>  <span class="va">best_query_match</span>   <span class="op">&lt;-</span> <span class="va">gb</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqnames</a></span><span class="op">(</span><span class="va">gb</span><span class="op">)</span> <span class="op">==</span> <span class="va">longest_target_seq</span><span class="op">]</span><span class="op">$</span><span class="va">query</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqnames</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/table.html" class="external-link">table</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/sort.html" class="external-link">sort</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">tail</a></span><span class="op">(</span><span class="fl">1</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="op">)</span></span>
<span>  <span class="kw">if</span> <span class="op">(</span><span class="va">centre</span> <span class="op">==</span> <span class="st">'midpoint'</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="va">target_centre</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Round.html" class="external-link">round</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqlengths</a></span><span class="op">(</span><span class="va">gb</span><span class="op">)</span><span class="op">[</span><span class="va">longest_target_seq</span><span class="op">]</span>     <span class="op">/</span> <span class="fl">2</span><span class="op">)</span></span>
<span>    <span class="va">query_centre</span>  <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Round.html" class="external-link">round</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqlengths</a></span><span class="op">(</span><span class="va">gb</span><span class="op">$</span><span class="va">query</span><span class="op">)</span><span class="op">[</span><span class="va">best_query_match</span><span class="op">]</span> <span class="op">/</span> <span class="fl">2</span><span class="op">)</span></span>
<span>  <span class="op">}</span> <span class="kw">else</span> <span class="op">{</span></span>
<span>    <span class="va">gb_coa</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/coalesce_contigs.html" class="external-link">coalesce_contigs</a></span><span class="op">(</span><span class="va">gb</span><span class="op">)</span></span>
<span>    <span class="va">maxwidth</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/which.min.html" class="external-link">which.max</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/start.html" class="external-link">width</a></span><span class="op">(</span><span class="va">gb_coa</span><span class="op">)</span><span class="op">)</span></span>
<span>    <span class="va">target_centre</span> <span class="op">&lt;-</span> <span class="va">gb_coa</span><span class="op">[</span><span class="va">maxwidth</span><span class="op">]</span>       <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/intra-range-methods.html" class="external-link">resize</a></span><span class="op">(</span><span class="fl">1</span>, fix <span class="op">=</span> <span class="st">"center"</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/stats/start.html" class="external-link">start</a></span><span class="op">(</span><span class="op">)</span></span>
<span>    <span class="va">query_centre</span>  <span class="op">&lt;-</span> <span class="va">gb_coa</span><span class="op">$</span><span class="va">query</span><span class="op">[</span><span class="va">maxwidth</span><span class="op">]</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/intra-range-methods.html" class="external-link">resize</a></span><span class="op">(</span><span class="fl">1</span>, fix <span class="op">=</span> <span class="st">"center"</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/stats/start.html" class="external-link">start</a></span><span class="op">(</span><span class="op">)</span></span>
<span>  <span class="op">}</span></span>
<span>  <span class="va">gb</span> <span class="op">&lt;-</span> <span class="va">gb</span> <span class="op">|&gt;</span> <span class="fu">plyranges</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span></span>
<span>    <span class="va">seqnames</span> <span class="op">==</span> <span class="va">longest_target_seq</span>,</span>
<span>    <span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqnames</a></span><span class="op">(</span><span class="va">query</span><span class="op">)</span> <span class="op">==</span> <span class="va">best_query_match</span>,</span>
<span>    <span class="va">start</span> <span class="op">&gt;</span> <span class="va">target_centre</span> <span class="op">-</span> <span class="va">win</span>,</span>
<span>    <span class="va">end</span> <span class="op">&lt;</span> <span class="va">target_centre</span> <span class="op">+</span> <span class="va">win</span>,</span>
<span>    <span class="fu"><a href="https://rdrr.io/r/stats/start.html" class="external-link">start</a></span><span class="op">(</span><span class="va">query</span><span class="op">)</span> <span class="op">&gt;</span> <span class="va">query_centre</span> <span class="op">-</span> <span class="va">win</span>,</span>
<span>    <span class="fu"><a href="https://rdrr.io/r/stats/start.html" class="external-link">end</a></span><span class="op">(</span><span class="va">query</span><span class="op">)</span> <span class="op">&lt;</span> <span class="va">query_centre</span> <span class="op">+</span> <span class="va">win</span></span>
<span>  <span class="op">)</span></span>
<span>  <span class="co"># score(gb) &lt;- stringDist_GBreaks(gb) / width(gb)</span></span>
<span>  <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/makeOxfordPlots.html" class="external-link">makeOxfordPlots</a></span><span class="op">(</span><span class="va">gb</span>, col <span class="op">=</span> <span class="st">"strand"</span>, size <span class="op">=</span> <span class="va">size</span>, <span class="va">...</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_bw</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>legend.position<span class="op">=</span><span class="st">"none"</span><span class="op">)</span></span>
<span><span class="op">}</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="oikopleura">Oikopleura<a class="anchor" aria-label="anchor" href="#oikopleura"></a>
</h2>
<div class="section level3">
<h3 id="okinawa-vs-osaka">Okinawa vs Osaka<a class="anchor" aria-label="anchor" href="#okinawa-vs-osaka"></a>
</h3>
<p>66.12972 % (matches / alignment widths)</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">mkOxPlotWindow</span><span class="op">(</span><span class="va">gbs</span><span class="op">$</span><span class="va">Oki_Osa</span><span class="op">)</span></span></code></pre></div>
<p><img src="SquareWindowPlots_files/figure-html/Oki_Osa_win-1.png" width="700"></p>
</div>
<div class="section level3">
<h3 id="okinawa-vs-barcelona">Okinawa vs Barcelona<a class="anchor" aria-label="anchor" href="#okinawa-vs-barcelona"></a>
</h3>
<p>66.99999 %</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">mkOxPlotWindow</span><span class="op">(</span><span class="va">gbs</span><span class="op">$</span><span class="va">Oki_Bar</span><span class="op">)</span></span></code></pre></div>
<p><img src="SquareWindowPlots_files/figure-html/Oki_Bar_win-1.svg" width="700"></p>
</div>
<div class="section level3">
<h3 id="osaka-vs-barcelona">Osaka vs Barcelona<a class="anchor" aria-label="anchor" href="#osaka-vs-barcelona"></a>
</h3>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">mkOxPlotWindow</span><span class="op">(</span><span class="va">gbs</span><span class="op">$</span><span class="va">Osa_Bar</span><span class="op">)</span></span></code></pre></div>
<p><img src="SquareWindowPlots_files/figure-html/Osa_Bar_win-1.svg" width="700"></p>
</div>
<div class="section level3">
<h3 id="osaka-vs-okinawa">Osaka vs Okinawa<a class="anchor" aria-label="anchor" href="#osaka-vs-okinawa"></a>
</h3>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">mkOxPlotWindow</span><span class="op">(</span><span class="va">gbs</span><span class="op">$</span><span class="va">Osa_Oki</span><span class="op">)</span></span></code></pre></div>
<p><img src="SquareWindowPlots_files/figure-html/Osa_Oki_win-1.png" width="700"></p>
</div>
<div class="section level3">
<h3 id="osaka-vs-aomori">Osaka vs Aomori<a class="anchor" aria-label="anchor" href="#osaka-vs-aomori"></a>
</h3>
<p>94.45642 % matches / alignment length</p>
<p>The Aomori genome needs some custom scaffolding to make a contiguous
10 × 10 region.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">Osa_Aom_PAR</span> <span class="op">&lt;-</span> <span class="va">gbs</span><span class="op">$</span><span class="va">Osa_Aom</span> <span class="op">|&gt;</span> <span class="fu">plyranges</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">seqnames</span> <span class="op">==</span> <span class="st">"PAR"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqlevels</a></span><span class="op">(</span><span class="va">Osa_Aom_PAR</span><span class="op">)</span>       <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqlevelsInUse</a></span><span class="op">(</span><span class="va">Osa_Aom_PAR</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqlevels</a></span><span class="op">(</span><span class="va">Osa_Aom_PAR</span><span class="op">$</span><span class="va">query</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqlevelsInUse</a></span><span class="op">(</span><span class="va">Osa_Aom_PAR</span><span class="op">$</span><span class="va">query</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/makeOxfordPlots.html" class="external-link">makeOxfordPlots</a></span><span class="op">(</span><span class="va">Osa_Aom_PAR</span><span class="op">)</span></span></code></pre></div>
<p><img src="SquareWindowPlots_files/figure-html/Osa_Aom_win_prepare-1.png" width="700"></p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Let's remove the contigs that have their main match elsewhere.</span></span>
<span><span class="va">QTcoverage</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">gb</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/stopifnot.html" class="external-link">stopifnot</a></span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqlevels</a></span><span class="op">(</span><span class="va">gb</span><span class="op">)</span><span class="op">)</span> <span class="op">==</span> <span class="fl">1</span><span class="op">)</span> <span class="co"># Not ready for full objects</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/stopifnot.html" class="external-link">stopifnot</a></span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/any.html" class="external-link">any</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqlengths</a></span><span class="op">(</span><span class="va">gb</span><span class="op">$</span><span class="va">query</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="va">grl</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/splitAsList.html" class="external-link">split</a></span><span class="op">(</span><span class="va">gb</span>, <span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqnames</a></span><span class="op">(</span><span class="va">gb</span><span class="op">$</span><span class="va">query</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">grl</span>, \<span class="op">(</span><span class="va">gb</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/start.html" class="external-link">width</a></span><span class="op">(</span><span class="va">gb</span><span class="op">$</span><span class="va">query</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/unlist.html" class="external-link">unlist</a></span><span class="op">(</span><span class="op">)</span> <span class="op">/</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqlengths</a></span><span class="op">(</span><span class="va">gb</span><span class="op">$</span><span class="va">query</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span><span class="va">Osa_Aom_PAR</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqnames</a></span><span class="op">(</span><span class="va">Osa_Aom_PAR</span><span class="op">$</span><span class="va">query</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqlevels</a></span><span class="op">(</span><span class="va">Osa_Aom_PAR</span><span class="op">$</span><span class="va">query</span><span class="op">)</span><span class="op">[</span><span class="fu">QTcoverage</span><span class="op">(</span><span class="va">Osa_Aom_PAR</span><span class="op">)</span> <span class="op">&lt;</span> <span class="fl">0.5</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="cn">NULL</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqlevels</a></span><span class="op">(</span><span class="va">Osa_Aom_PAR</span><span class="op">$</span><span class="va">query</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqlevelsInUse</a></span><span class="op">(</span><span class="va">Osa_Aom_PAR</span><span class="op">$</span><span class="va">query</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/makeOxfordPlots.html" class="external-link">makeOxfordPlots</a></span><span class="op">(</span><span class="va">Osa_Aom_PAR</span><span class="op">)</span></span></code></pre></div>
<p><img src="SquareWindowPlots_files/figure-html/Osa_Aom_win_prepare-2.png" width="700"></p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Let's remove the smallest ones that barely display on one pixel</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqlengths</a></span><span class="op">(</span><span class="va">Osa_Aom_PAR</span><span class="op">$</span><span class="va">query</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## contig_12_1 contig_15_1 contig_34_1 contig_38_1 contig_49_1 </span></span>
<span><span class="co">##     7097394     6419763     1209792      203373     1316631</span></span></code></pre>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">Osa_Aom_PAR</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqnames</a></span><span class="op">(</span><span class="va">Osa_Aom_PAR</span><span class="op">$</span><span class="va">query</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqlevels</a></span><span class="op">(</span><span class="va">Osa_Aom_PAR</span><span class="op">$</span><span class="va">query</span><span class="op">)</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqlengths</a></span><span class="op">(</span><span class="va">Osa_Aom_PAR</span><span class="op">$</span><span class="va">query</span><span class="op">)</span> <span class="op">&lt;</span> <span class="fl">2e4</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="cn">NULL</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqlevels</a></span><span class="op">(</span><span class="va">Osa_Aom_PAR</span><span class="op">$</span><span class="va">query</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqlevelsInUse</a></span><span class="op">(</span><span class="va">Osa_Aom_PAR</span><span class="op">$</span><span class="va">query</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/makeOxfordPlots.html" class="external-link">makeOxfordPlots</a></span><span class="op">(</span><span class="va">Osa_Aom_PAR</span>, col <span class="op">=</span> <span class="st">"strand"</span><span class="op">)</span></span></code></pre></div>
<p><img src="SquareWindowPlots_files/figure-html/Osa_Aom_win_prepare-3.png" width="700"></p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># And now let's flip by hand the ones that need</span></span>
<span><span class="va">grl</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/splitAsList.html" class="external-link">split</a></span><span class="op">(</span><span class="va">Osa_Aom_PAR</span>, <span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqnames</a></span><span class="op">(</span><span class="va">Osa_Aom_PAR</span><span class="op">$</span><span class="va">query</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">grl</span><span class="op">[[</span><span class="st">"contig_38_1"</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/reverse-methods.html" class="external-link">reverse</a></span><span class="op">(</span><span class="va">grl</span><span class="op">[[</span><span class="st">"contig_38_1"</span><span class="op">]</span><span class="op">]</span>, query <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">grl</span><span class="op">[[</span><span class="st">"contig_49_1"</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/reverse-methods.html" class="external-link">reverse</a></span><span class="op">(</span><span class="va">grl</span><span class="op">[[</span><span class="st">"contig_49_1"</span><span class="op">]</span><span class="op">]</span>, query <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">Osa_Aom_PAR</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/unlist.html" class="external-link">unlist</a></span><span class="op">(</span><span class="va">grl</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/makeOxfordPlots.html" class="external-link">makeOxfordPlots</a></span><span class="op">(</span><span class="va">Osa_Aom_PAR</span>, col <span class="op">=</span> <span class="st">"strand"</span><span class="op">)</span></span></code></pre></div>
<p><img src="SquareWindowPlots_files/figure-html/Osa_Aom_win_prepare-4.png" width="700"></p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Finally, let's reorder and merge remaining seqlevels</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqlevels</a></span><span class="op">(</span><span class="va">Osa_Aom_PAR</span><span class="op">$</span><span class="va">query</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqlevels</a></span><span class="op">(</span><span class="va">Osa_Aom_PAR</span><span class="op">$</span><span class="va">query</span><span class="op">)</span><span class="op">[</span><span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/orderQuerySeqLevels.html" class="external-link">orderQuerySeqLevels</a></span><span class="op">(</span><span class="va">Osa_Aom_PAR</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="va">Osa_Aom_PAR</span><span class="op">$</span><span class="va">query</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/mergeSeqLevels.html" class="external-link">mergeSeqLevels</a></span><span class="op">(</span><span class="va">Osa_Aom_PAR</span><span class="op">$</span><span class="va">query</span>, <span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqlevels</a></span><span class="op">(</span><span class="va">Osa_Aom_PAR</span><span class="op">$</span><span class="va">query</span><span class="op">)</span>, <span class="st">"PAR"</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">mkOxPlotWindow</span><span class="op">(</span><span class="va">Osa_Aom_PAR</span><span class="op">)</span></span></code></pre></div>
<p><img src="SquareWindowPlots_files/figure-html/Osa_Aom_win-1.png" width="700"></p>
</div>
<div class="section level3">
<h3 id="okinawa-vs-kume">Okinawa vs Kume<a class="anchor" aria-label="anchor" href="#okinawa-vs-kume"></a>
</h3>
<p>94.23956 % matches / alignment length</p>
<p>Same for the Kume genome…</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">Oki_Kum_PAR</span> <span class="op">&lt;-</span> <span class="va">gbs</span><span class="op">$</span><span class="va">Oki_Kum</span> <span class="op">|&gt;</span> <span class="fu">plyranges</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">seqnames</span> <span class="op">==</span> <span class="st">"PAR"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqlevels</a></span><span class="op">(</span><span class="va">Oki_Kum_PAR</span><span class="op">)</span>       <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqlevelsInUse</a></span><span class="op">(</span><span class="va">Oki_Kum_PAR</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqlevels</a></span><span class="op">(</span><span class="va">Oki_Kum_PAR</span><span class="op">$</span><span class="va">query</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqlevelsInUse</a></span><span class="op">(</span><span class="va">Oki_Kum_PAR</span><span class="op">$</span><span class="va">query</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/makeOxfordPlots.html" class="external-link">makeOxfordPlots</a></span><span class="op">(</span><span class="va">Oki_Kum_PAR</span><span class="op">)</span></span></code></pre></div>
<p><img src="SquareWindowPlots_files/figure-html/Oki_Kum_win_prepare-1.png" width="700"></p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Let's remove the contigs that have their main match elsewhere.</span></span>
<span><span class="va">QTcoverage</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">gb</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/stopifnot.html" class="external-link">stopifnot</a></span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqlevels</a></span><span class="op">(</span><span class="va">gb</span><span class="op">)</span><span class="op">)</span> <span class="op">==</span> <span class="fl">1</span><span class="op">)</span> <span class="co"># Not ready for full objects</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/stopifnot.html" class="external-link">stopifnot</a></span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/any.html" class="external-link">any</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqlengths</a></span><span class="op">(</span><span class="va">gb</span><span class="op">$</span><span class="va">query</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="va">grl</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/splitAsList.html" class="external-link">split</a></span><span class="op">(</span><span class="va">gb</span>, <span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqnames</a></span><span class="op">(</span><span class="va">gb</span><span class="op">$</span><span class="va">query</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">grl</span>, \<span class="op">(</span><span class="va">gb</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/start.html" class="external-link">width</a></span><span class="op">(</span><span class="va">gb</span><span class="op">$</span><span class="va">query</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/unlist.html" class="external-link">unlist</a></span><span class="op">(</span><span class="op">)</span> <span class="op">/</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqlengths</a></span><span class="op">(</span><span class="va">gb</span><span class="op">$</span><span class="va">query</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span><span class="va">Oki_Kum_PAR</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqnames</a></span><span class="op">(</span><span class="va">Oki_Kum_PAR</span><span class="op">$</span><span class="va">query</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqlevels</a></span><span class="op">(</span><span class="va">Oki_Kum_PAR</span><span class="op">$</span><span class="va">query</span><span class="op">)</span><span class="op">[</span><span class="fu">QTcoverage</span><span class="op">(</span><span class="va">Oki_Kum_PAR</span><span class="op">)</span> <span class="op">&lt;</span> <span class="fl">0.5</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="cn">NULL</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqlevels</a></span><span class="op">(</span><span class="va">Oki_Kum_PAR</span><span class="op">$</span><span class="va">query</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqlevelsInUse</a></span><span class="op">(</span><span class="va">Oki_Kum_PAR</span><span class="op">$</span><span class="va">query</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/makeOxfordPlots.html" class="external-link">makeOxfordPlots</a></span><span class="op">(</span><span class="va">Oki_Kum_PAR</span><span class="op">)</span></span></code></pre></div>
<p><img src="SquareWindowPlots_files/figure-html/Oki_Kum_win_prepare-2.png" width="700"></p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Let's remove the smallest ones that barely display on one pixel</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqlengths</a></span><span class="op">(</span><span class="va">Oki_Kum_PAR</span><span class="op">$</span><span class="va">query</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## contig_12_1 contig_13_1 contig_18_1 contig_28_1 contig_38_1 contig_41_1 </span></span>
<span><span class="co">##     2357322      793615         832     6415228       61242     2611992 </span></span>
<span><span class="co">## contig_43_1 contig_61_1 contig_72_1 contig_82_1 contig_85_1 contig_89_1 </span></span>
<span><span class="co">##      841236       17591       84743     2322619      428051      241503 </span></span>
<span><span class="co">## contig_91_1 </span></span>
<span><span class="co">##      126775</span></span></code></pre>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">Oki_Kum_PAR</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqnames</a></span><span class="op">(</span><span class="va">Oki_Kum_PAR</span><span class="op">$</span><span class="va">query</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqlevels</a></span><span class="op">(</span><span class="va">Oki_Kum_PAR</span><span class="op">$</span><span class="va">query</span><span class="op">)</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqlengths</a></span><span class="op">(</span><span class="va">Oki_Kum_PAR</span><span class="op">$</span><span class="va">query</span><span class="op">)</span> <span class="op">&lt;</span> <span class="fl">2e4</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="cn">NULL</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqlevels</a></span><span class="op">(</span><span class="va">Oki_Kum_PAR</span><span class="op">$</span><span class="va">query</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqlevelsInUse</a></span><span class="op">(</span><span class="va">Oki_Kum_PAR</span><span class="op">$</span><span class="va">query</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/makeOxfordPlots.html" class="external-link">makeOxfordPlots</a></span><span class="op">(</span><span class="va">Oki_Kum_PAR</span><span class="op">)</span></span></code></pre></div>
<p><img src="SquareWindowPlots_files/figure-html/Oki_Kum_win_prepare-3.png" width="700"></p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># And now let's flip by hand the ones that need</span></span>
<span><span class="va">grl</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/splitAsList.html" class="external-link">split</a></span><span class="op">(</span><span class="va">Oki_Kum_PAR</span>, <span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqnames</a></span><span class="op">(</span><span class="va">Oki_Kum_PAR</span><span class="op">$</span><span class="va">query</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">grl</span><span class="op">[[</span><span class="st">"contig_12_1"</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/reverse-methods.html" class="external-link">reverse</a></span><span class="op">(</span><span class="va">grl</span><span class="op">[[</span><span class="st">"contig_12_1"</span><span class="op">]</span><span class="op">]</span>, query <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">grl</span><span class="op">[[</span><span class="st">"contig_43_1"</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/reverse-methods.html" class="external-link">reverse</a></span><span class="op">(</span><span class="va">grl</span><span class="op">[[</span><span class="st">"contig_43_1"</span><span class="op">]</span><span class="op">]</span>, query <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">Oki_Kum_PAR</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/unlist.html" class="external-link">unlist</a></span><span class="op">(</span><span class="va">grl</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/makeOxfordPlots.html" class="external-link">makeOxfordPlots</a></span><span class="op">(</span><span class="va">Oki_Kum_PAR</span><span class="op">)</span></span></code></pre></div>
<p><img src="SquareWindowPlots_files/figure-html/Oki_Kum_win_prepare-4.png" width="700"></p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Finally, let's reorder and merge remaining seqlevels</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqlevels</a></span><span class="op">(</span><span class="va">Oki_Kum_PAR</span><span class="op">$</span><span class="va">query</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqlevels</a></span><span class="op">(</span><span class="va">Oki_Kum_PAR</span><span class="op">$</span><span class="va">query</span><span class="op">)</span><span class="op">[</span><span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/orderQuerySeqLevels.html" class="external-link">orderQuerySeqLevels</a></span><span class="op">(</span><span class="va">Oki_Kum_PAR</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="va">Oki_Kum_PAR</span><span class="op">$</span><span class="va">query</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/mergeSeqLevels.html" class="external-link">mergeSeqLevels</a></span><span class="op">(</span><span class="va">Oki_Kum_PAR</span><span class="op">$</span><span class="va">query</span>, <span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqlevels</a></span><span class="op">(</span><span class="va">Oki_Kum_PAR</span><span class="op">$</span><span class="va">query</span><span class="op">)</span>, <span class="st">"PAR"</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">mkOxPlotWindow</span><span class="op">(</span><span class="va">Oki_Kum_PAR</span><span class="op">)</span></span></code></pre></div>
<p><img src="SquareWindowPlots_files/figure-html/Oki_Kum_win-1.svg" width="700"></p>
</div>
<div class="section level3">
<h3 id="barcelona-vs-norway">Barcelona vs Norway<a class="anchor" aria-label="anchor" href="#barcelona-vs-norway"></a>
</h3>
<p>No plot, but 95.91132 % matches / alignment length</p>
</div>
</div>
<div class="section level2">
<h2 id="ciona">Ciona<a class="anchor" aria-label="anchor" href="#ciona"></a>
</h2>
<div class="section level3">
<h3 id="ciona-intestinalis-plymouth-vs-roscoff">
<em>Ciona intestinalis</em> (Plymouth vs Roscoff)<a class="anchor" aria-label="anchor" href="#ciona-intestinalis-plymouth-vs-roscoff"></a>
</h3>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">mkOxPlotWindow</span><span class="op">(</span><span class="va">gbs</span><span class="op">$</span><span class="va">Ply_Ros</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Ciona intestinalis Plymouth vs Roscoff"</span><span class="op">)</span></span></code></pre></div>
<p><img src="SquareWindowPlots_files/figure-html/Ply_Ros_win-1.svg" width="700"></p>
</div>
<div class="section level3">
<h3 id="ciona-intestinalis-plymouth-vs-ciona-robusta">
<em>Ciona intestinalis</em> (Plymouth) vs <em>Ciona
robusta</em><a class="anchor" aria-label="anchor" href="#ciona-intestinalis-plymouth-vs-ciona-robusta"></a>
</h3>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">mkOxPlotWindow</span><span class="op">(</span><span class="va">gbs</span><span class="op">$</span><span class="va">Ply_Rob</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Ciona intestinalis Plymouth vs Ciona robusta"</span><span class="op">)</span></span></code></pre></div>
<p><img src="SquareWindowPlots_files/figure-html/Ply_Rob_win-1.svg" width="700"></p>
</div>
<div class="section level3">
<h3 id="ciona-intestinalis-plymouth-vs-ciona-savignyi">
<em>Ciona intestinalis</em> (Plymouth) vs <em>Ciona
savignyi</em><a class="anchor" aria-label="anchor" href="#ciona-intestinalis-plymouth-vs-ciona-savignyi"></a>
</h3>
<p>Here again we need some custom scaffolding.</p>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">Ply_Sav_chr1</span> <span class="op">&lt;-</span> <span class="va">gbs</span><span class="op">$</span><span class="va">Ply_Sav</span> <span class="op">|&gt;</span> <span class="fu">plyranges</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">seqnames</span> <span class="op">==</span> <span class="st">"BNJZ01000001.1"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqlevels</a></span><span class="op">(</span><span class="va">Ply_Sav_chr1</span><span class="op">)</span>       <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqlevelsInUse</a></span><span class="op">(</span><span class="va">Ply_Sav_chr1</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqlevels</a></span><span class="op">(</span><span class="va">Ply_Sav_chr1</span><span class="op">$</span><span class="va">query</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqlevelsInUse</a></span><span class="op">(</span><span class="va">Ply_Sav_chr1</span><span class="op">$</span><span class="va">query</span><span class="op">)</span></span>
<span><span class="va">Ply_Sav_chr1</span><span class="op">$</span><span class="va">query</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/forceSeqLengths.html" class="external-link">forceSeqLengths</a></span><span class="op">(</span><span class="va">Ply_Sav_chr1</span><span class="op">$</span><span class="va">query</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/makeOxfordPlots.html" class="external-link">makeOxfordPlots</a></span><span class="op">(</span><span class="va">Ply_Sav_chr1</span><span class="op">)</span></span></code></pre></div>
<p><img src="SquareWindowPlots_files/figure-html/Ply_Sav_win_prepare-1.png" width="700"></p>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Let's remove the contigs that have their main match elsewhere.</span></span>
<span><span class="va">QTcoverage</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">gb</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/stopifnot.html" class="external-link">stopifnot</a></span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqlevels</a></span><span class="op">(</span><span class="va">gb</span><span class="op">)</span><span class="op">)</span> <span class="op">==</span> <span class="fl">1</span><span class="op">)</span> <span class="co"># Not ready for full objects</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/stopifnot.html" class="external-link">stopifnot</a></span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/any.html" class="external-link">any</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqlengths</a></span><span class="op">(</span><span class="va">gb</span><span class="op">$</span><span class="va">query</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="va">grl</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/splitAsList.html" class="external-link">split</a></span><span class="op">(</span><span class="va">gb</span>, <span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqnames</a></span><span class="op">(</span><span class="va">gb</span><span class="op">$</span><span class="va">query</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">grl</span>, \<span class="op">(</span><span class="va">gb</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/start.html" class="external-link">width</a></span><span class="op">(</span><span class="va">gb</span><span class="op">$</span><span class="va">query</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/unlist.html" class="external-link">unlist</a></span><span class="op">(</span><span class="op">)</span> <span class="op">/</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqlengths</a></span><span class="op">(</span><span class="va">gb</span><span class="op">$</span><span class="va">query</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span><span class="va">Ply_Sav_chr1</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqnames</a></span><span class="op">(</span><span class="va">Ply_Sav_chr1</span><span class="op">$</span><span class="va">query</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqlevels</a></span><span class="op">(</span><span class="va">Ply_Sav_chr1</span><span class="op">$</span><span class="va">query</span><span class="op">)</span><span class="op">[</span><span class="fu">QTcoverage</span><span class="op">(</span><span class="va">Ply_Sav_chr1</span><span class="op">)</span> <span class="op">&lt;</span> <span class="fl">0.05</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="cn">NULL</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqlevels</a></span><span class="op">(</span><span class="va">Ply_Sav_chr1</span><span class="op">$</span><span class="va">query</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqlevelsInUse</a></span><span class="op">(</span><span class="va">Ply_Sav_chr1</span><span class="op">$</span><span class="va">query</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/makeOxfordPlots.html" class="external-link">makeOxfordPlots</a></span><span class="op">(</span><span class="va">Ply_Sav_chr1</span><span class="op">)</span></span></code></pre></div>
<p><img src="SquareWindowPlots_files/figure-html/Ply_Sav_win_prepare-2.png" width="700"></p>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Let's remove the smallest ones that barely display on one pixel</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqlengths</a></span><span class="op">(</span><span class="va">Ply_Sav_chr1</span><span class="op">$</span><span class="va">query</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##   R1063    R113   R1167   R1383     R14   R1490   R1596    R162    R173   R1779 </span></span>
<span><span class="co">##   24606  277504   10138    7422 1702960    3223    8495  276690  455223   32296 </span></span>
<span><span class="co">##   R1942   R2368   R2407   R2517    R268     R27    R281   R2872     R30     R31 </span></span>
<span><span class="co">##   10471    6428    9904    5720  550239 2366314  114725    8506 3493700  766944 </span></span>
<span><span class="co">##    R347    R352    R417    R489    R491      R5     R51     R62     R64    R650 </span></span>
<span><span class="co">##  180004   46220   61968  180871   28116 2728683 1639093  608627  540266   23432 </span></span>
<span><span class="co">##    R767     R98 </span></span>
<span><span class="co">##   42085  434095</span></span></code></pre>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">Ply_Sav_chr1</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqnames</a></span><span class="op">(</span><span class="va">Ply_Sav_chr1</span><span class="op">$</span><span class="va">query</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqlevels</a></span><span class="op">(</span><span class="va">Ply_Sav_chr1</span><span class="op">$</span><span class="va">query</span><span class="op">)</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqlengths</a></span><span class="op">(</span><span class="va">Ply_Sav_chr1</span><span class="op">$</span><span class="va">query</span><span class="op">)</span> <span class="op">&lt;</span> <span class="fl">2e4</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="cn">NULL</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqlevels</a></span><span class="op">(</span><span class="va">Ply_Sav_chr1</span><span class="op">$</span><span class="va">query</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqlevelsInUse</a></span><span class="op">(</span><span class="va">Ply_Sav_chr1</span><span class="op">$</span><span class="va">query</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/makeOxfordPlots.html" class="external-link">makeOxfordPlots</a></span><span class="op">(</span><span class="va">Ply_Sav_chr1</span><span class="op">)</span></span></code></pre></div>
<p><img src="SquareWindowPlots_files/figure-html/Ply_Sav_win_prepare-3.png" width="700"></p>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Finally, let's reorder and merge remaining seqlevels</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqlevels</a></span><span class="op">(</span><span class="va">Ply_Sav_chr1</span><span class="op">$</span><span class="va">query</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqlevels</a></span><span class="op">(</span><span class="va">Ply_Sav_chr1</span><span class="op">$</span><span class="va">query</span><span class="op">)</span><span class="op">[</span><span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/orderQuerySeqLevels.html" class="external-link">orderQuerySeqLevels</a></span><span class="op">(</span><span class="va">Ply_Sav_chr1</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="va">Ply_Sav_chr1</span><span class="op">$</span><span class="va">query</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/mergeSeqLevels.html" class="external-link">mergeSeqLevels</a></span><span class="op">(</span><span class="va">Ply_Sav_chr1</span><span class="op">$</span><span class="va">query</span>, <span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqlevels</a></span><span class="op">(</span><span class="va">Ply_Sav_chr1</span><span class="op">$</span><span class="va">query</span><span class="op">)</span>, <span class="st">"Merged matching contigs"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/makeOxfordPlots.html" class="external-link">makeOxfordPlots</a></span><span class="op">(</span><span class="va">Ply_Sav_chr1</span><span class="op">)</span></span></code></pre></div>
<p><img src="SquareWindowPlots_files/figure-html/Ply_Sav_win_prepare-4.png" width="700"></p>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">mkOxPlotWindow</span><span class="op">(</span><span class="va">Ply_Sav_chr1</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Ciona intestinalis Plymouth vs Ciona savignyi"</span><span class="op">)</span></span></code></pre></div>
<p><img src="SquareWindowPlots_files/figure-html/Ply_Sav_win-1.svg" width="700"></p>
</div>
</div>
<div class="section level2">
<h2 id="drosophila">
<em>Drosophila</em><a class="anchor" aria-label="anchor" href="#drosophila"></a>
</h2>
<p><em>Note that in some genomes, each chromosome arm is represented in
a separate sequence</em>.</p>
<div class="section level3">
<h3 id="d--melanogaster-vs-d--mauritania">
<em>D. melanogaster</em> vs <em>D. mauritania</em><a class="anchor" aria-label="anchor" href="#d--melanogaster-vs-d--mauritania"></a>
</h3>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Force strand flipping for aesthetic purpose</span></span>
<span><span class="fu">mkOxPlotWindow</span><span class="op">(</span><span class="va">gbs</span><span class="op">$</span><span class="va">Dme_Dma</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/forceSeqLengths.html" class="external-link">forceSeqLengths</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/reverse-methods.html" class="external-link">reverse</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Drosophila melanogaster vs Drosophila mauritania"</span><span class="op">)</span></span></code></pre></div>
<p><img src="SquareWindowPlots_files/figure-html/Dmel_Dmau_win-1.svg" width="700"></p>
</div>
<div class="section level3">
<h3 id="d--melanogaster-vs-d--yakuba">
<em>D. melanogaster</em> vs <em>D. yakuba</em><a class="anchor" aria-label="anchor" href="#d--melanogaster-vs-d--yakuba"></a>
</h3>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">mkOxPlotWindow</span><span class="op">(</span><span class="va">gbs</span><span class="op">$</span><span class="va">Dme_Dya</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Drosophila melanogaster vs Drosophila yakuba"</span><span class="op">)</span></span></code></pre></div>
<p><img src="SquareWindowPlots_files/figure-html/Dmel_Dyak_win-1.svg" width="700"></p>
</div>
<div class="section level3">
<h3 id="d--melanogaster-vs-d--subpulchrella">
<em>D. melanogaster</em> vs <em>D. subpulchrella</em><a class="anchor" aria-label="anchor" href="#d--melanogaster-vs-d--subpulchrella"></a>
</h3>
<p>On arm 3R there is a large unaligned region that causes the plot to
not be squared. I think that the problem may be related to the presence
of large centromere regions in the best assemblies.</p>
<div class="sourceCode" id="cb37"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">mkOxPlotWindow</span><span class="op">(</span><span class="va">gbs</span><span class="op">$</span><span class="va">Dme_Dsu</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Drosophila melanogaster vs Drosophila subpulchrella"</span><span class="op">)</span></span></code></pre></div>
<p><img src="SquareWindowPlots_files/figure-html/Dmel_Dsub_win_3R-1.svg" width="700"></p>
<p>Blacklist 3R as the algorithm does not find an nicely square window
on it.</p>
<div class="sourceCode" id="cb38"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">mkOxPlotWindow</span><span class="op">(</span><span class="va">gbs</span><span class="op">$</span><span class="va">Dme_Dsu</span> <span class="op">|&gt;</span> <span class="fu">plyranges</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">seqnames</span> <span class="op">!=</span> <span class="st">"3R"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Drosophila melanogaster vs Drosophila subpulchrella"</span><span class="op">)</span></span></code></pre></div>
<p><img src="SquareWindowPlots_files/figure-html/Dmel_Dsub_win-1.svg" width="700"></p>
</div>
<div class="section level3">
<h3 id="d--melanogaster-vs-d--buskii">
<em>D. melanogaster</em> vs <em>D. buskii</em><a class="anchor" aria-label="anchor" href="#d--melanogaster-vs-d--buskii"></a>
</h3>
<div class="sourceCode" id="cb39"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">mkOxPlotWindow</span><span class="op">(</span><span class="va">gbs</span><span class="op">$</span><span class="va">Dme_Dbu</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Drosophila melanogaster vs Drosophila buskii"</span><span class="op">)</span></span></code></pre></div>
<p><img src="SquareWindowPlots_files/figure-html/Dmel_Dbus_win-1.svg" width="700"></p>
</div>
</div>
<div class="section level2">
<h2 id="mammals">Mammals<a class="anchor" aria-label="anchor" href="#mammals"></a>
</h2>
<div class="section level3">
<h3 id="human-versus-mouse">Human versus mouse<a class="anchor" aria-label="anchor" href="#human-versus-mouse"></a>
</h3>
<p>Chromosome 18 was chosen because it is small and it has the same name
in both species. Region was centered on the alignment with the longest
width, as otherwise the window is out of target.</p>
<p>Still, most of the region is not aligned. The syntenic regions only
look contiguous because the dots need to be thick to be easy to see.</p>
<div class="sourceCode" id="cb40"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hgmm</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/load_genomic_breaks.html" class="external-link">load_genomic_breaks</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata/hg38_mm10/hg38_chr18__mm10_chr18.gff3"</span>, package <span class="op">=</span> <span class="st">"BreakpointsData"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu">mkOxPlotWindow</span><span class="op">(</span><span class="va">hgmm</span>, centre <span class="op">=</span> <span class="st">'maxwidth'</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Homo sapiens vs Mus musculus"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Warning in min(x): no non-missing arguments to min; returning Inf</span></span></code></pre>
<pre><code><span><span class="co">## Warning in max(x): no non-missing arguments to max; returning -Inf</span></span></code></pre>
<pre><code><span><span class="co">## Warning in min(x): no non-missing arguments to min; returning Inf</span></span></code></pre>
<pre><code><span><span class="co">## Warning in max(x): no non-missing arguments to max; returning -Inf</span></span></code></pre>
<p><img src="SquareWindowPlots_files/figure-html/hg38_mm10-1.svg" width="700"></p>
</div>
<div class="section level3">
<h3 id="muntjaks">Muntjaks<a class="anchor" aria-label="anchor" href="#muntjaks"></a>
</h3>
<div class="sourceCode" id="cb45"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">Mmun_Mree</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/load_genomic_breaks.html" class="external-link">load_genomic_breaks</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata/Muntjaks/Rrev_Rmun.07.postmasked.gff.gz"</span>, package <span class="op">=</span> <span class="st">"BreakpointsData"</span><span class="op">)</span>, type <span class="op">=</span> <span class="st">"match"</span><span class="op">)</span></span>
<span><span class="co"># Given the enormous differences of karyotype, the M. reevesi chromosomes need to be trimmed for # </span></span>
<span><span class="co"># the window centering alorithm to function.</span></span>
<span><span class="va">Mmun_Mree.trim</span> <span class="op">&lt;-</span> <span class="va">Mmun_Mree</span> <span class="op">|&gt;</span> <span class="fu">plyranges</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">seqnames</span> <span class="op">==</span> <span class="st">"CM018478.1"</span>, <span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqnames</a></span><span class="op">(</span><span class="va">query</span><span class="op">)</span> <span class="op">==</span> <span class="st">"CM018501.1"</span>, <span class="fu"><a href="https://rdrr.io/r/stats/start.html" class="external-link">end</a></span><span class="op">(</span><span class="va">query</span><span class="op">)</span> <span class="op">&lt;</span> <span class="fl">273507479</span><span class="op">)</span></span>
<span><span class="fu">mkOxPlotWindow</span><span class="op">(</span><span class="va">Mmun_Mree.trim</span>, centre <span class="op">=</span> <span class="st">'maxwidth'</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Muntiacus muntjak vs M. reevesi (95% aligned, 98% identical)."</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Warning in min(x): no non-missing arguments to min; returning Inf</span></span></code></pre>
<pre><code><span><span class="co">## Warning in max(x): no non-missing arguments to max; returning -Inf</span></span></code></pre>
<pre><code><span><span class="co">## Warning in min(x): no non-missing arguments to min; returning Inf</span></span></code></pre>
<pre><code><span><span class="co">## Warning in max(x): no non-missing arguments to max; returning -Inf</span></span></code></pre>
<p><img src="SquareWindowPlots_files/figure-html/muntjaks-1.svg" width="700"></p>
</div>
</div>
<div class="section level2">
<h2 id="nematodes">Nematodes<a class="anchor" aria-label="anchor" href="#nematodes"></a>
</h2>
<p><em>Caenorhabditis briggsae</em> and <em>C. elegans</em> are
hermaphrodites, so let’s avoid using them as target genomes.</p>
<p>But <em>C. briggsae</em> is closest to <em>C. nigoni</em> so we need
it once as a query genome.</p>
<div class="section level3">
<h3 id="c--nigoni-vs-c--briggsae">
<em>C. nigoni</em> vs <em>C. briggsae</em><a class="anchor" aria-label="anchor" href="#c--nigoni-vs-c--briggsae"></a>
</h3>
<p><em>C. nigoni</em> genome assembly <code>GCA_027920645.1</code>
length: 128,508,886. It has no unplaced contigs nor mitochondrial
genome.</p>
<p><em>C. briggsae</em>’s genome assmebly <code>GCA_022453885.1</code>
length is 106,953,292, with mitochondrial genome as I forgot to remove
it.</p>
<div class="sourceCode" id="cb50"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">mkOxPlotWindow</span><span class="op">(</span><span class="va">gbs</span><span class="op">$</span><span class="va">Cni_Cbr</span>, centre <span class="op">=</span> <span class="st">'maxwidth'</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Caenorhabditis nigoni vs C. briggsae (61% aligned, 85% identity)"</span><span class="op">)</span></span></code></pre></div>
<p><img src="SquareWindowPlots_files/figure-html/Cni_Cbr_win-1.svg" width="700"></p>
<div class="sourceCode" id="cb51"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">mkOxPlotWindow</span><span class="op">(</span><span class="va">gbs</span><span class="op">$</span><span class="va">Cbr_Cni</span>, centre <span class="op">=</span> <span class="st">'maxwidth'</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Caenorhabditis briggsae vs C. nigoni (69% aligned, 85% identity)"</span><span class="op">)</span></span></code></pre></div>
<p><img src="SquareWindowPlots_files/figure-html/Cbr_Cni_win-1.svg" width="700"></p>
</div>
<div class="section level3">
<h3 id="c--nigoni-or-briggsae-vs-c--remanei">
<em>C. nigoni</em> or <em>briggsae</em> vs <em>C. remanei</em><a class="anchor" aria-label="anchor" href="#c--nigoni-or-briggsae-vs-c--remanei"></a>
</h3>
<p>Unplaced contigs and mitochondrial genome were removed from <em>C.
remanei</em>’s assembly <code>GCA_001643735.4</code>.</p>
<div class="sourceCode" id="cb52"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">mkOxPlotWindow</span><span class="op">(</span><span class="va">gbs</span><span class="op">$</span><span class="va">Cni_Cre</span>, centre <span class="op">=</span> <span class="st">'maxwidth'</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Caenorhabditis nigoni vs C. remanei (25% aligned, 73% identity)"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Warning in min(x): no non-missing arguments to min; returning Inf</span></span></code></pre>
<pre><code><span><span class="co">## Warning in max(x): no non-missing arguments to max; returning -Inf</span></span></code></pre>
<p><img src="SquareWindowPlots_files/figure-html/Cni_Cre_win-1.svg" width="700"></p>
<div class="sourceCode" id="cb55"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">mkOxPlotWindow</span><span class="op">(</span><span class="va">gbs</span><span class="op">$</span><span class="va">Cbr_Cre</span>, centre <span class="op">=</span> <span class="st">'maxwidth'</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Caenorhabditis briggsae vs C. remanei (30% aligned, 72% identity)"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Warning in min(x): no non-missing arguments to min; returning Inf</span></span></code></pre>
<pre><code><span><span class="co">## Warning in max(x): no non-missing arguments to max; returning -Inf</span></span></code></pre>
<p><img src="SquareWindowPlots_files/figure-html/Cbr_Cre_win-1.svg" width="700"></p>
</div>
<div class="section level3">
<h3 id="c--nigoni-vs-c--inopinata">
<em>C. nigoni</em> vs <em>C. inopinata</em><a class="anchor" aria-label="anchor" href="#c--nigoni-vs-c--inopinata"></a>
</h3>
<div class="sourceCode" id="cb58"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">mkOxPlotWindow</span><span class="op">(</span><span class="va">gbs</span><span class="op">$</span><span class="va">Cni_Cin</span>, centre <span class="op">=</span> <span class="st">'maxwidth'</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Caenorhabditis nigoni vs C. inopinata (16% aligned, 72% identity)"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Warning in min(x): no non-missing arguments to min; returning Inf</span></span></code></pre>
<pre><code><span><span class="co">## Warning in max(x): no non-missing arguments to max; returning -Inf</span></span></code></pre>
<p><img src="SquareWindowPlots_files/figure-html/Cni_Cin_win-1.svg" width="700"></p>
</div>
<div class="section level3">
<h3 id="c--briggsae-vs-c--elegans">_C. briggsae vs <em>C. elegans</em><a class="anchor" aria-label="anchor" href="#c--briggsae-vs-c--elegans"></a>
</h3>
<div class="sourceCode" id="cb61"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">mkOxPlotWindow</span><span class="op">(</span><span class="va">gbs</span><span class="op">$</span><span class="va">Cbr_Cel</span>, centre <span class="op">=</span> <span class="st">'maxwidth'</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Caenorhabditis briggsae vs C. elegans (23% aligned, 72% identity)"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Warning in min(x): no non-missing arguments to min; returning Inf</span></span></code></pre>
<pre><code><span><span class="co">## Warning in max(x): no non-missing arguments to max; returning -Inf</span></span></code></pre>
<pre><code><span><span class="co">## Warning in min(x): no non-missing arguments to min; returning Inf</span></span></code></pre>
<pre><code><span><span class="co">## Warning in max(x): no non-missing arguments to max; returning -Inf</span></span></code></pre>
<p><img src="SquareWindowPlots_files/figure-html/Cbr_Cel_win-1.svg" width="700"></p>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Aleksandra Bliznina, Michael Mansfield, Charles Plessy.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
