<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Tandem site duplications • OikScrambling</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Tandem site duplications">
<meta property="og:description" content="OikScrambling">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">OikScrambling</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">2.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/OikScrambling.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/CNEs.html">Conserved non-coding elements</a>
    </li>
    <li>
      <a href="../articles/CoalescingCutoffs.html">Modifying coalescent cutoffs</a>
    </li>
    <li>
      <a href="../articles/ColinearityInterruptors.html">Regions interrupting colinearity</a>
    </li>
    <li>
      <a href="../articles/DistanceMeasures.html">Distance Measures</a>
    </li>
    <li>
      <a href="../articles/GenomicFeatures.html">Genomic features of the breakpoints</a>
    </li>
    <li>
      <a href="../articles/Inversions.html">Inversions</a>
    </li>
    <li>
      <a href="../articles/LoadGenomicBreaks.html">Load genomic break data</a>
    </li>
    <li>
      <a href="../articles/OrthogroupSyntenies.html">Orthogroup syntenies</a>
    </li>
    <li>
      <a href="../articles/OxfordPlots.html">Whole-genome dot-plots</a>
    </li>
    <li>
      <a href="../articles/PAC3.html">Region of interest: PAC3</a>
    </li>
    <li>
      <a href="../articles/PWM.html">Motif search and analysis</a>
    </li>
    <li>
      <a href="../articles/ParallelPlots.html">Whole-chromosome parallel plots</a>
    </li>
    <li>
      <a href="../articles/RegionWidths.html">Width of the aligned and mapped regions.</a>
    </li>
    <li>
      <a href="../articles/RegionsOfInterest.html">Regions of interest</a>
    </li>
    <li>
      <a href="../articles/RepeatRegions.html">Repeat regions</a>
    </li>
    <li>
      <a href="../articles/SegmentMovements.html">Segment movement distances</a>
    </li>
    <li>
      <a href="../articles/TFBS.html">Transcription factor binding sites</a>
    </li>
    <li>
      <a href="../articles/TandemSiteDuplications.html">Tandem site duplications</a>
    </li>
    <li>
      <a href="../articles/Translocations.html">Translocations</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/oist/LuscombeU_OikScrambling/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="TandemSiteDuplications_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Tandem site duplications</h1>
                        <h4 data-toc-skip class="author">Charles Plessy</h4>
            
            <h4 data-toc-skip class="date">08 March, 2022</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/oist/LuscombeU_OikScrambling/blob/HEAD/vignettes/TandemSiteDuplications.Rmd" class="external-link"><code>vignettes/TandemSiteDuplications.Rmd</code></a></small>
      <div class="hidden name"><code>TandemSiteDuplications.Rmd</code></div>

    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">knitr</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/knitr/man/opts_chunk.html" class="external-link">opts_chunk</a></span><span class="op">$</span><span class="fu">set</span><span class="op">(</span>cache <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<div class="section level2">
<h2 id="load-pacakges-and-data">Load pacakges and data<a class="anchor" aria-label="anchor" href="#load-pacakges-and-data"></a>
</h2>
<p>See the <code><a href="../articles/OikScrambling.html">vignette("OikScrambling", package = "OikScrambling")</a></code> for general details on package and data load.</p>
<p>See <code><a href="../articles/LoadGenomicBreaks.html">vignette("LoadGenomicBreaks", package = "OikScrambling")</a></code> for how the different <code>GBreaks</code> objects are prepared.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="op">{</span>
  <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://oist.github.io/GenomicBreaks" class="external-link">'GenomicBreaks'</a></span><span class="op">)</span>
  <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://ggplot2.tidyverse.org" class="external-link">'ggplot2'</a></span><span class="op">)</span>
  <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st">"BreakpointsData"</span><span class="op">)</span>
  <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://bioconductor.org/packages/Biostrings" class="external-link">"Biostrings"</a></span><span class="op">)</span>
<span class="op">}</span><span class="op">)</span>
<span class="va">genomes</span> <span class="op">&lt;-</span> <span class="fu">OikScrambling</span><span class="fu">:::</span><span class="fu"><a href="../reference/loadAllGenomes.html">loadAllGenomes</a></span><span class="op">(</span>compat <span class="op">=</span> <span class="cn">F</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/load.html" class="external-link">load</a></span><span class="op">(</span><span class="st">"BreakPoints.Rdata"</span><span class="op">)</span>
<span class="va">reps</span> <span class="op">&lt;-</span> <span class="fu">OikScrambling</span><span class="fu">:::</span><span class="fu"><a href="../reference/loadAllRepeats.html">loadAllRepeats</a></span><span class="op">(</span>compat <span class="op">=</span> <span class="cn">F</span><span class="op">)</span>
<span class="va">transcripts</span> <span class="op">&lt;-</span> <span class="fu">OikScrambling</span><span class="fu">:::</span><span class="fu"><a href="../reference/loadAllTranscriptsGR.html">loadAllTranscriptsGR</a></span><span class="op">(</span>compat <span class="op">=</span> <span class="cn">F</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="purpose">Purpose<a class="anchor" aria-label="anchor" href="#purpose"></a>
</h2>
<p>Here we explore regions of interest that feature an insertion-deletion site flanked by a <em>tandem site duplication</em>.</p>
<p>In our <em>genomic breaks</em> objects, representing <em>one-to-one</em> alignments, one copy of the tandem-duplicated site is part of the alignment, and the other is not.</p>
<pre><code>                   TSD
query       …──────────━━━━━─────────━━━━━─────────────…
alignment    ...............              .............
target      …──────────━━━━━              ─────────────…
</code></pre>
</div>
<div class="section level2">
<h2 id="service-functions">Service functions<a class="anchor" aria-label="anchor" href="#service-functions"></a>
</h2>
<div class="section level3">
<h3 id="align-breakpoints">Align breakpoints<a class="anchor" aria-label="anchor" href="#align-breakpoints"></a>
</h3>
<pre><code>pat          ++++
range  --------==========----------
sub                    ++++             (or ---- if rev = F)
</code></pre>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">alignBreakPoints</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">gr</span>, <span class="va">n</span> <span class="op">=</span> <span class="fl">20</span>, <span class="va">reverseComplement</span> <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">{</span>
  <span class="kw">if</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Logic.html" class="external-link">isFALSE</a></span><span class="op">(</span><span class="va">reverseComplement</span><span class="op">)</span><span class="op">)</span> <span class="va">reverseComplement</span> <span class="op">&lt;-</span> <span class="va">identity</span>
  <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/pairwiseAlignment.html" class="external-link">pairwiseAlignment</a></span><span class="op">(</span><span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/getSeq.html" class="external-link">getSeq</a></span><span class="op">(</span><span class="va">gr</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/intra-range-methods.html" class="external-link">flank</a></span><span class="op">(</span><span class="va">n</span>, both <span class="op">=</span> <span class="cn">T</span>, start <span class="op">=</span> <span class="cn">T</span><span class="op">)</span><span class="op">)</span>,
                    <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/getSeq.html" class="external-link">getSeq</a></span><span class="op">(</span><span class="va">gr</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/intra-range-methods.html" class="external-link">flank</a></span><span class="op">(</span><span class="va">n</span>, both <span class="op">=</span> <span class="cn">T</span>, start <span class="op">=</span> <span class="cn">F</span><span class="op">)</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/reverseComplement.html" class="external-link">reverseComplement</a></span><span class="op">(</span><span class="op">)</span>,
                    type<span class="op">=</span><span class="st">"global"</span><span class="op">)</span>
<span class="op">}</span></code></pre></div>
</div>
<div class="section level3">
<h3 id="align-ends">Align ends<a class="anchor" aria-label="anchor" href="#align-ends"></a>
</h3>
<pre><code>pat            ++++
range  --------==========----------
sub                  ++++             (or ---- if rev = F)
</code></pre>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">alignEnds</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">gr</span>, <span class="va">n</span> <span class="op">=</span> <span class="fl">20</span>, <span class="va">reverseComplement</span> <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">{</span>
  <span class="va">resize_</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">...</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/intra-range-methods.html" class="external-link">resize</a></span><span class="op">(</span><span class="va">...</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/intra-range-methods.html" class="external-link">trim</a></span><span class="op">(</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/r/base/warning.html" class="external-link">suppressWarnings</a></span><span class="op">(</span><span class="op">)</span>
  <span class="kw">if</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Logic.html" class="external-link">isFALSE</a></span><span class="op">(</span><span class="va">reverseComplement</span><span class="op">)</span><span class="op">)</span> <span class="va">reverseComplement</span> <span class="op">&lt;-</span> <span class="va">identity</span>
  <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/pairwiseAlignment.html" class="external-link">pairwiseAlignment</a></span><span class="op">(</span><span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/getSeq.html" class="external-link">getSeq</a></span><span class="op">(</span><span class="va">gr</span> |&gt; <span class="fu">resize_</span><span class="op">(</span><span class="va">n</span>, fix <span class="op">=</span> <span class="st">"start"</span><span class="op">)</span><span class="op">)</span>,
                    <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/getSeq.html" class="external-link">getSeq</a></span><span class="op">(</span><span class="va">gr</span> |&gt; <span class="fu">resize_</span><span class="op">(</span><span class="va">n</span>, fix <span class="op">=</span> <span class="st">"end"</span><span class="op">)</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/reverseComplement.html" class="external-link">reverseComplement</a></span><span class="op">(</span><span class="op">)</span>,
                    type<span class="op">=</span><span class="st">"global"</span><span class="op">)</span>
<span class="op">}</span></code></pre></div>
</div>
<div class="section level3">
<h3 id="search-for-tsds">Search for TSDs<a class="anchor" aria-label="anchor" href="#search-for-tsds"></a>
</h3>
<pre><code>pat            +
range  --------==========----------
sub                      +         </code></pre>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">searchForTSDs</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">gr</span>, <span class="va">n</span> <span class="op">=</span> <span class="fl">8</span><span class="op">)</span> <span class="op">{</span>
  <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/pairwiseAlignment.html" class="external-link">pairwiseAlignment</a></span><span class="op">(</span><span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/getSeq.html" class="external-link">getSeq</a></span><span class="op">(</span><span class="va">gr</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/intra-range-methods.html" class="external-link">resize</a></span><span class="op">(</span><span class="va">n</span>, fix <span class="op">=</span> <span class="st">"start"</span><span class="op">)</span><span class="op">)</span>,
                    <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/getSeq.html" class="external-link">getSeq</a></span><span class="op">(</span><span class="va">gr</span> |&gt;  <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/intra-range-methods.html" class="external-link">flank</a></span><span class="op">(</span><span class="va">n</span>, start <span class="op">=</span> <span class="cn">F</span><span class="op">)</span><span class="op">)</span>,
                    type<span class="op">=</span><span class="st">"global"</span><span class="op">)</span>
<span class="op">}</span></code></pre></div>
</div>
<div class="section level3">
<h3 id="align-indels">Align indels<a class="anchor" aria-label="anchor" href="#align-indels"></a>
</h3>
<pre><code>                          ++++          ++++
query       …──────────────────────────────────────────…
target      …───────────────              ─────────────…
                          ++..............++</code></pre>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">alignInDel</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">gb</span><span class="op">)</span> <span class="op">{</span>
  <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>
    q_l <span class="op">=</span> <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/getSeq.html" class="external-link">getSeq</a></span><span class="op">(</span><span class="va">gb</span><span class="op">$</span><span class="va">query</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/intra-range-methods.html" class="external-link">flank</a></span><span class="op">(</span><span class="fl">20</span>, both <span class="op">=</span> <span class="cn">T</span>, start <span class="op">=</span> <span class="cn">T</span><span class="op">)</span><span class="op">)</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span>,
    trg <span class="op">=</span> <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/getSeq.html" class="external-link">getSeq</a></span><span class="op">(</span><span class="va">gb</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>       |&gt; <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/intra-range-methods.html" class="external-link">flank</a></span><span class="op">(</span><span class="fl">20</span>, both <span class="op">=</span> <span class="cn">T</span>, start <span class="op">=</span> <span class="cn">F</span><span class="op">)</span><span class="op">)</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span>,
    q_r <span class="op">=</span> <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/getSeq.html" class="external-link">getSeq</a></span><span class="op">(</span><span class="va">gb</span><span class="op">$</span><span class="va">query</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/intra-range-methods.html" class="external-link">flank</a></span><span class="op">(</span><span class="fl">20</span>, both <span class="op">=</span> <span class="cn">T</span>, start <span class="op">=</span> <span class="cn">F</span><span class="op">)</span><span class="op">)</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span>
  <span class="op">)</span> |&gt; <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/as.html" class="external-link">as</a></span><span class="op">(</span><span class="st">"DNAStringSet"</span><span class="op">)</span> |&gt; <span class="fu">msa</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/msa/man/msaClustalW.html" class="external-link">msaClustalW</a></span><span class="op">(</span>order <span class="op">=</span> <span class="st">"input"</span><span class="op">)</span>
<span class="op">}</span></code></pre></div>
</div>
<div class="section level3">
<h3 id="align-to-self">Align to self<a class="anchor" aria-label="anchor" href="#align-to-self"></a>
</h3>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">alignToSelf</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">gr</span><span class="op">)</span> <span class="op">{</span>
  <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/pairwiseAlignment.html" class="external-link">pairwiseAlignment</a></span><span class="op">(</span><span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/getSeq.html" class="external-link">getSeq</a></span><span class="op">(</span><span class="va">gr</span><span class="op">)</span>,
                    <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/getSeq.html" class="external-link">getSeq</a></span><span class="op">(</span><span class="va">gr</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/reverseComplement.html" class="external-link">reverseComplement</a></span><span class="op">(</span><span class="op">)</span>,
                    type<span class="op">=</span><span class="st">"global"</span><span class="op">)</span>
<span class="op">}</span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="roi1-composite-transposon-and-tandem-site-duplication-">ROI1: composite transposon and tandem site duplication.<a class="anchor" aria-label="anchor" href="#roi1-composite-transposon-and-tandem-site-duplication-"></a>
</h2>
<div class="section level3">
<h3 id="details-on-the-region">Details on the region<a class="anchor" aria-label="anchor" href="#details-on-the-region"></a>
</h3>
<p><em>ROI1</em> identified and analysed by Martin Frith. Here I am just transposing the finding in Bioconductor’s context.</p>
<p>In <code>contig_27_1</code> (which is most of Kume genome’s chr1’s short arm), there is an indel region where the aligned regions in the Okinawa genome are directly in contact, but on the Kume genome the aligned regions are separated by a 5146-bp gap.</p>
<p>(The regions align on opposite strands, so the plot looks strange if you are not familiar with it.)</p>
<p>ROI1 is represented by a <code>GBreaks</code> object of lenght 2. The Kume-specific sequence can be obtained with the <code>cleanGaps</code> function.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="op">(</span><span class="va">ROI1</span> <span class="op">&lt;-</span> <span class="va">gbs</span><span class="op">$</span><span class="va">Oki_Kum</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/findOverlaps-methods.html" class="external-link">subsetByOverlaps</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a></span><span class="op">(</span><span class="st">"chr1:3720291-3734643"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## GBreaks object with 2 ranges and 8 metadata columns:</span>
<span class="co">##       seqnames          ranges strand |     score                       query</span>
<span class="co">##          &lt;Rle&gt;       &lt;IRanges&gt;  &lt;Rle&gt; | &lt;numeric&gt;                   &lt;GRanges&gt;</span>
<span class="co">##   [1]     chr1 3720291-3726692      - |     36253 contig_27_1:1233370-1239790</span>
<span class="co">##   [2]     chr1 3726693-3734643      - |     44991 contig_27_1:1220273-1228223</span>
<span class="co">##            Arm                   rep   repOvlp transcripts        flag</span>
<span class="co">##       &lt;factor&gt;       &lt;CharacterList&gt; &lt;integer&gt;       &lt;Rle&gt; &lt;character&gt;</span>
<span class="co">##   [1]    short           unknown,rnd       674        &lt;NA&gt;         Col</span>
<span class="co">##   [2]    short rnd,ltr-1,unknown,...       886        &lt;NA&gt;        &lt;NA&gt;</span>
<span class="co">##          nonCoa</span>
<span class="co">##       &lt;logical&gt;</span>
<span class="co">##   [1]     FALSE</span>
<span class="co">##   [2]     FALSE</span>
<span class="co">##   -------</span>
<span class="co">##   seqinfo: 19 sequences from OKI2018.I69 genome</span></code></pre>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/plotApairOfChrs.html" class="external-link">plotApairOfChrs</a></span><span class="op">(</span><span class="va">ROI1</span><span class="op">)</span></code></pre></div>
<p><img src="TandemSiteDuplications_files/figure-html/ROI1-1.png" width="700"></p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/cleanGaps.html" class="external-link">cleanGaps</a></span><span class="op">(</span><span class="va">ROI1</span><span class="op">$</span><span class="va">query</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">##      seqnames   start     end width strand</span>
<span class="co">## 1 contig_27_1 1228224 1233369  5146      *</span></code></pre>
<p>Interestingly, the unaligned region in Kume contains two repeat elements in inverted tandem. You can see them in the alignment below, of the left-side boundary of the unaligned region to the reverse-complement of its right side.</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ROI1</span><span class="op">$</span><span class="va">query</span> |&gt; <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/cleanGaps.html" class="external-link">cleanGaps</a></span><span class="op">(</span><span class="op">)</span> |&gt; <span class="fu">alignBreakPoints</span><span class="op">(</span><span class="fl">30</span>, rev<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Global PairwiseAlignmentsSingleSubject (1 of 1)</span>
<span class="co">## pattern: <span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">TTTTT</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">CC</span><span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">GG</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">CC</span><span style="color: #000000; background-color: #AFFFFF;">GG</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">AAAA</span><span style="color: #000000; background-color: #FFD7AF;">TTT</span><span style="color: #000000; background-color: #AFFFFF;">GG</span>----</span>
<span class="co">## subject: <span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">AAA</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">TTT</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #AFFFFF;">G</span>--<span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">TT</span>--<span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">GG</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">CC</span><span style="color: #000000; background-color: #AFFFFF;">GG</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">AAAA</span><span style="color: #000000; background-color: #FFD7AF;">TTT</span><span style="color: #000000; background-color: #AFFFFF;">GG</span><span style="color: #000000; background-color: #AFFFAF;">CC</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span></span>
<span class="co">## score: -100.7614</span></code></pre>
<details><p><summary>Click here to see the full pairwise alignment between the Kume insert and its own reverse-complement.</summary></p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ROI1</span><span class="op">$</span><span class="va">query</span> |&gt; <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/cleanGaps.html" class="external-link">cleanGaps</a></span><span class="op">(</span><span class="op">)</span> |&gt; <span class="fu">alignToSelf</span><span class="op">(</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/PairwiseAlignments-io.html" class="external-link">writePairwiseAlignments</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## ########################################</span>
<span class="co">## # Program: Biostrings (version 2.62.0), a Bioconductor package</span>
<span class="co">## # Rundate: Tue Mar  8 17:03:15 2022</span>
<span class="co">## ########################################</span>
<span class="co">## #=======================================</span>
<span class="co">## #</span>
<span class="co">## # Aligned_sequences: 2</span>
<span class="co">## # 1: P1</span>
<span class="co">## # 2: S1</span>
<span class="co">## # Matrix: NA</span>
<span class="co">## # Gap_penalty: 14.0</span>
<span class="co">## # Extend_penalty: 4.0</span>
<span class="co">## #</span>
<span class="co">## # Length: 5518</span>
<span class="co">## # Identity:    3398/5518 (61.6%)</span>
<span class="co">## # Similarity:    NA/5518 (NA%)</span>
<span class="co">## # Gaps:         744/5518 (13.5%)</span>
<span class="co">## # Score: -6519.19</span>
<span class="co">## #</span>
<span class="co">## #</span>
<span class="co">## #=======================================</span>
<span class="co">## </span>
<span class="co">## P1                 1 GATACGTAGGCTTACCGGACTAAAATTTGGCCGATTTTTCTAGCTTAGTG     50</span>
<span class="co">##                        ||    ||||||||||||||||||||||||||||||||||||||||||</span>
<span class="co">## S1                 1 TTTA----GGCTTACCGGACTAAAATTTGGCCGATTTTTCTAGCTTAGTG     46</span>
<span class="co">## </span>
<span class="co">## P1                51 GGGGGGTCATATGATACATTTTTAGAAAGCTCAGAAGCTGGGCTTTCATT    100</span>
<span class="co">##                      ||||||||||||||||||||||||||||||||||||||||||||||||||</span>
<span class="co">## S1                47 GGGGGGTCATATGATACATTTTTAGAAAGCTCAGAAGCTGGGCTTTCATT     96</span>
<span class="co">## </span>
<span class="co">## P1               101 TGAGCCTAAAATCAGGGGGTCTGGGTCATGCCCGCAAGGCAAATTATGGG    150</span>
<span class="co">##                      ||||||||||||||||||||||||||||||||||||||||||||||||||</span>
<span class="co">## S1                97 TGAGCCTAAAATCAGGGGGTCTGGGTCATGCCCGCAAGGCAAATTATGGG    146</span>
<span class="co">## </span>
<span class="co">## P1               151 ACCTGAAGTCCCAAAATAGGCACTAGAACTCTTCAAACTATTTTTTTGGA    200</span>
<span class="co">##                      ||||||||||||||||||||||||||||||||||||||||||||||||||</span>
<span class="co">## S1               147 ACCTGAAGTCCCAAAATAGGCACTAGAACTCTTCAAACTATTTTTTTGGA    196</span>
<span class="co">## </span>
<span class="co">## P1               201 GGCGGGCCGGCCTCCAAAACCGTCGAGACTTATATATTTAGAAAGCTCAG    250</span>
<span class="co">##                      ||||||||||||||||||||||||||||||||||||||||||||||||||</span>
<span class="co">## S1               197 GGCGGGCCGGCCTCCAAAACCGTCGAGACTTATATATTTAGAAAGCTCAG    246</span>
<span class="co">## </span>
<span class="co">## P1               251 GCTCTGGGCTTTCAGATGAGCCCAAGAAATCTTAGTGCAAATGTCTCTGC    300</span>
<span class="co">##                      ||||||||||||||||||||||||||||||||||||||||||||||||||</span>
<span class="co">## S1               247 GCTCTGGGCTTTCAGATGAGCCCAAGAAATCTTAGTGCAAATGTCTCTGC    296</span>
<span class="co">## </span>
<span class="co">## P1               301 GGCCACCTGGGGGCGCTAGAAGTTGGGTGAAAAACCGGAAGTTCCGGAAA    350</span>
<span class="co">##                      ||||||||||||||||||||||||||||||||||||||||||||||||||</span>
<span class="co">## S1               297 GGCCACCTGGGGGCGCTAGAAGTTGGGTGAAAAACCGGAAGTTCCGGAAA    346</span>
<span class="co">## </span>
<span class="co">## P1               351 ACTCGGATTTAGAATTCTAAACTCTTATATTATGTTTGTAAAACATAAAC    400</span>
<span class="co">##                      ||||||||||||||||||||||||||||||||||||||||||||||||||</span>
<span class="co">## S1               347 ACTCGGATTTAGAATTCTAAACTCTTATATTATGTTTGTAAAACATAAAC    396</span>
<span class="co">## </span>
<span class="co">## P1               401 AAACAACAATTGACGCTCAAAAATGATGAACCGCGGTCGAACCGGGTGGT    450</span>
<span class="co">##                      ||||||||||||||||||||||||||||||||||||||||||||||||||</span>
<span class="co">## S1               397 AAACAACAATTGACGCTCAAAAATGATGAACCGCGGTCGAACCGGGTGGT    446</span>
<span class="co">## </span>
<span class="co">## P1               451 CAACCAGGTGGTTCATCGCGGTTCAACTCTAGAAACCTTCTAAAACATGG    500</span>
<span class="co">##                      ||||||||||||||||||||||||||||||||||||| ||||||||||||</span>
<span class="co">## S1               447 CAACCAGGTGGTTCATCGCGGTTCAACTCTAGAAACCGTCTAAAACATGG    496</span>
<span class="co">## </span>
<span class="co">## P1               501 GAATATGGGTATTTTTCGACGGGAAATTGAATGAAAATGAAAATAAAAGT    550</span>
<span class="co">##                      ||||||||||||||||||||||||||||||||||||||||||||||||||</span>
<span class="co">## S1               497 GAATATGGGTATTTTTCGACGGGAAATTGAATGAAAATGAAAATAAAAGT    546</span>
<span class="co">## </span>
<span class="co">## P1               551 TGAACCACGGTCGAACCACGTGGTCAACCACCCGGTTAACCGCGGTTCAA    600</span>
<span class="co">##                      ||||||||||||||||||||||||||||||   |||| | ||||||||||</span>
<span class="co">## S1               547 TGAACCACGGTCGAACCACGTGGTCAACCAGGTGGTTCATCGCGGTTCAA    596</span>
<span class="co">## </span>
<span class="co">## P1               601 CTCTAGAAACCTTCTAAAACATGGGAATATGGGTATTTTTCGACGGGAAA    650</span>
<span class="co">##                      ||||||||  |||||||||| | | |||||||||||||||||||||||||</span>
<span class="co">## S1               597 CTCTAGAATTCTTCTAAAACCTTGCAATATGGGTATTTTTCGACGGGAAA    646</span>
<span class="co">## </span>
<span class="co">## P1               651 TTGAATGAAAATGAAAATAAAAGTTGAACCGCGGTCAAACCGGGTGGTCA    700</span>
<span class="co">##                      |||||||||| |||||||||||||| |||||||||| ||| |||||||||</span>
<span class="co">## S1               647 TTGAATGAAATTGAAAATAAAAGTTAAACCGCGGTCGAACTGGGTGGTCA    696</span>
<span class="co">## </span>
<span class="co">## P1               701 ACCAGGTGGTTCATCGCGGTTCAACTCTAGAAACCTTCTTAAACATGGCA    750</span>
<span class="co">##                      |||||||||||| | |||||||||||||||||||||||| |||||||  |</span>
<span class="co">## S1               697 ACCAGGTGGTTCCTTGCGGTTCAACTCTAGAAACCTTCTAAAACATGATA    746</span>
<span class="co">## </span>
<span class="co">## P1               751 ATATGGGTATTTTTCGACGGGAAATTTGATTTTTATGTTTTTTTTATAAG    800</span>
<span class="co">##                      |||| ||||||||||||||| ||| |  ||  | |  |    ||| | | </span>
<span class="co">## S1               747 ATATTGGTATTTTTCGACGGCAAAGTCTATAATAAGCTAAAATTTCTTAA    796</span>
<span class="co">## </span>
<span class="co">## P1               801 ATCAAAACAACAAAACAAGATAATTTCCAAAAAACCATTTTATTCATTCT    850</span>
<span class="co">##                       |  ||  |         | | |||    |||    |||| ||| |||  </span>
<span class="co">## S1               797 TTTGAATGA---------GTTTATTG---AAATTTTATTTCATTAATTGA    834</span>
<span class="co">## </span>
<span class="co">## P1               851 GTGGTTCAGA--AGGTTCAGAAGAAATTCCTTCAAAATCCATCTCGTC-G    897</span>
<span class="co">##                        ||||  |   |   |||       ||| | |   || |||||  |  |</span>
<span class="co">## S1               835 AAGGTTTCGTCCAAAATCACTTATTTTTCTTCCTTGATTCATCTTCTGAG    884</span>
<span class="co">## </span>
<span class="co">## P1               898 ACATTCTGTTCTTCTTCCTCGTTCATTTCGTCCAATTCCTTTTCGACTAT    947</span>
<span class="co">##                       || ||| | |  ||| | |  || | ||   |    |   ||||||   </span>
<span class="co">## S1               885 TCAATCT-TGCCACTTGCGCAGTCTTATCAGTCGGCGCTGATTCGACGCG    933</span>
<span class="co">## </span>
<span class="co">## P1               948 TTCTTCCACTTCTTCAACAATCTCTTCCTCTCGTTCTTCCTCGTCATCTT    997</span>
<span class="co">##                        || || |||  |||   ||| |||  |||| | |||| || ||| | |</span>
<span class="co">## S1               934 CGCT-CCGCTT--TCACTTATC-CTTTTTCTCTTCCTTC-TC-TCACCGT    977</span>
<span class="co">## </span>
<span class="co">## P1               998 CGTTGATTTCGATTTCAGCCTCTCGCTCGTTCTTGTCTTTGTATTTCATT   1047</span>
<span class="co">##                      | || ||||   |||     |||  ||  |||| | |  || ||  || |</span>
<span class="co">## S1               978 CATTTATTTTTCTTTT----TCTTTCTTTTTCTCG-CAATGGATG-CAAT   1021</span>
<span class="co">## </span>
<span class="co">## P1              1048 TGGTCCTTCTTTTCGAGTTCCTTTGCTTTCTTCTCATTTTTCGCTCGTTC   1097</span>
<span class="co">##                       | |  | ||      || ||  |  |||   |||| ||| | | ||  |</span>
<span class="co">## S1              1022 CGATGATCCTC-----GTACCACT--TTTAAGCTCAATTT-CACCCGAGC   1063</span>
<span class="co">## </span>
<span class="co">## P1              1098 AACCATCGCTTTTGTAATCTCTACAAGACATTCATCATCCCGAAGGTCTT   1147</span>
<span class="co">##                      ||  | |  | ||| |   |||  |||| |||| ||              </span>
<span class="co">## S1              1064 AATGACCATTCTTG-AGCATCTTGAAGAGATTCCTCC-------------   1099</span>
<span class="co">## </span>
<span class="co">## P1              1148 CAAGCCAGAGACTTGTTTTATTGTACTGAATGAGACATTGTTAATATAAT   1197</span>
<span class="co">##                              |||||  |             |||||| ||  | || | |||</span>
<span class="co">## S1              1100 --------AGACTCAT-------------ATGAGAAATC-TCAACAAAAT   1127</span>
<span class="co">## </span>
<span class="co">## P1              1198 AA--AAATAGGCCAAAATTACCATGGAAACAGCCACGACAAATGTTCTTC   1245</span>
<span class="co">##                       |  ||  ||||     ||||    || | ||  |  |    | | || |</span>
<span class="co">## S1              1128 GACGAAGAAGGC-----TTACGCGAGAGAAAGAAAGTA----TCTGCTCC   1168</span>
<span class="co">## </span>
<span class="co">## P1              1246 GAAGTGAAAGCCTCG-ATTGTGTTCTTTCCAGCTCCTTTAATGTCCCAAA   1294</span>
<span class="co">##                       || |    | |  | |   | || ||   ||||| |   || ||  || </span>
<span class="co">## S1              1169 TAATTCTGCGGCAAGGAGACTTTTGTTAGAAGCTCTT---ATTTCTGAAG   1215</span>
<span class="co">## </span>
<span class="co">## P1              1295 CAGGGCTTCCACGCGT-CGAGAAGTCGCTGTCAAAAAGCTGTTTTCGTTC   1343</span>
<span class="co">##                          | ||| | |  | |     || | | |||||        ||  || </span>
<span class="co">## S1              1216 GTAAGATTCAAAGGTTTCTCTTCGTTGATTTCAAA--------TTAATT-   1256</span>
<span class="co">## </span>
<span class="co">## P1              1344 TGGACTCGAGTTGCGATTGCGTCCCATTTTTCTTTCTCCTTCGT-GAACT   1392</span>
<span class="co">##                      |  |   ||   ||||  || |    ||||||    |||  ||| ||| |</span>
<span class="co">## S1              1257 TAAA---GAAAAGCGACCGCAT----TTTTTCGAAATCC--CGTCGAATT   1297</span>
<span class="co">## </span>
<span class="co">## P1              1393 TCTCAAAGCTGCGACCGACAGACTGCCTAGTAACGCTGTCCAAGTGCTTC   1442</span>
<span class="co">##                      ||     | |  |  ||| || | |  | | || ||    |||||     </span>
<span class="co">## S1              1298 TCAAGTGGGTCGGGACGAGAGGCCGTGT-GAAAAGC----CAAGTCGACA   1342</span>
<span class="co">## </span>
<span class="co">## P1              1443 TTTTGTTCACCATCAAGAGCTCGGTACTGGATGAGTAGTCGCTGCTTGAC   1492</span>
<span class="co">##                         | ||| | |  |||   |||| | || |  |  |  |||   |   |</span>
<span class="co">## S1              1343 AGATCTTCTCGA--AAG---TCGGAATTGAAAAA--AACCGCCAGTATCC   1385</span>
<span class="co">## </span>
<span class="co">## P1              1493 TGTT------TTTGTCTCATCGTTGTCGTCGTTGGCCAGTTCTTGAAGC-   1535</span>
<span class="co">##                      | |       ||||| |||      ||   |   ||||   | | |||| </span>
<span class="co">## S1              1386 TTTCGGGAAGTTTGT-TCAAGAACTTCCAAGGCAGCCACCACGTCAAGCG   1434</span>
<span class="co">## </span>
<span class="co">## P1              1536 -ATTCG----CCAGTAAATATTCTTCTCAGATGCCATCATTCGCTTGTAG   1580</span>
<span class="co">##                       | |||    | || ||    ||||| || ||  |  | ||| |  ||  </span>
<span class="co">## S1              1435 AAATCGGAATCAAGGAAGAGATCTTCACAAATCTCC-CTTTCTCAGGT--   1481</span>
<span class="co">## </span>
<span class="co">## P1              1581 TAGTC-TTCCAAATTCGCGATAGCTTCTACAATCTTTTTCAAGGAGCTTG   1629</span>
<span class="co">##                         || || |   ||| |  |   |||   ||| ||||      |||| |</span>
<span class="co">## S1              1482 ---TCATTACCTTTTCCC--TCCTTTCGTTAATATTTTAACTCAAGCTCG   1526</span>
<span class="co">## </span>
<span class="co">## P1              1630 --ATGTTTCACTGACAGCTGC--ATGATATTTCTTCA-AGGAGCCATAGA   1674</span>
<span class="co">##                        ||||   | |||  | ||   ||||  | | |||  | || ||||  |</span>
<span class="co">## S1              1527 CCATGTCCGA-TGAT-GATGACGATGACGTCTTTTCTGAAGAACCATCAA   1574</span>
<span class="co">## </span>
<span class="co">## P1              1675 AAATGGCAAATGGGATGCAAGCAAGGAGGAGAACGAGTCTTTTGTGGGAT   1724</span>
<span class="co">##                            ||  |      || || | ||   || ||| |||     || | </span>
<span class="co">## S1              1575 ------CATCT------CACGCGATGATCCGA-CGA-TCTCAGCCGGAAG   1610</span>
<span class="co">## </span>
<span class="co">## P1              1725 CAACATTCGATCAACAACAACAACAACAACAACAACAACTACAACTA-CA   1773</span>
<span class="co">##                       || |   ||  || || || || || || |  |    | | |  |  ||</span>
<span class="co">## S1              1611 AAAAAGAAGAAGAAAAAGAAGAAGAAGAAGACGAGTCGCGAAAGTTGTCA   1660</span>
<span class="co">## </span>
<span class="co">## P1              1774 ACTAC----AACTACAACTACAACTA-CAACTAGTGATAGATGAACCTTT   1818</span>
<span class="co">##                      ||  |    | | | |  | |||  | || |   |||    |||  | ||</span>
<span class="co">## S1              1661 ACATCGGCGAGCGAGACATTCAAAAAACAGCGGCTGAGCCTTGACTCATT   1710</span>
<span class="co">## </span>
<span class="co">## P1              1819 -----GCCT------------ATTTTCCTTTG-CAGCTTTCTTCATTTCC   1850</span>
<span class="co">##                           || |            ||||||    | |  ||||| ||    ||</span>
<span class="co">## S1              1711 AGACAGCGTCGATGGCCGTGGATTTTCGAAAGTCTCCTTTCATCCGAACC   1760</span>
<span class="co">## </span>
<span class="co">## P1              1851 TCATCATTTTCTTCGTGAGGCCAGA--ATGACTCCAGAAAATCGCGAAGA   1898</span>
<span class="co">##                      | ||||   |  || ||  |||     |||     |||| || || ||| </span>
<span class="co">## S1              1761 TGATCAAGATTCTC-TGTTGCCTTCTCATGCGCGGAGAACATGGC-AAGT   1808</span>
<span class="co">## </span>
<span class="co">## P1              1899 AATTTCGCCGTGCTACTAGAATCCCCTGCGTAGCCCTGTAGAAGCGTTC-   1947</span>
<span class="co">##                        ||  | |   || || |||   | || | |      | || | |||| </span>
<span class="co">## S1              1809 CGTT--GTCAAACTTCTTGAACGTCTTGTGCAAGTGCATCGATGAGTTCA   1856</span>
<span class="co">## </span>
<span class="co">## P1              1948 -GGAT-AGTAG-------------GCATGATCAA----AACC---AGGCT   1975</span>
<span class="co">##                       |||| ||  |             ||| ||||||    || |   ||| |</span>
<span class="co">## S1              1857 AGGATCAGACGCCCCCTTCGATCAGCACGATCAATGAGAAACGGTAGGGT   1906</span>
<span class="co">## </span>
<span class="co">## P1              1976 TCTCATG-TTGCTTGGGAT-GATCATGAATCTTG-AGCCCATGTCTT---   2019</span>
<span class="co">##                      |||| |  ||| ||   || ||   | ||||||  |||   ||  |    </span>
<span class="co">## S1              1907 TCTCTTCCTTGATTTCAATTGAAGCT-AATCTTTTAGCTACTGCATCGAT   1955</span>
<span class="co">## </span>
<span class="co">## P1              2020 --TTTTAAGATTTTTAATATTG------AAATCAAGGAAGAGTTTGACAC   2061</span>
<span class="co">##                        ||| | |   |   || | |      ||||  ||| || |   |||  </span>
<span class="co">## S1              1956 ACTTTGACGGAATCCCATCTGGCAAAAGAAATTGAGGCAGCG---GACGA   2002</span>
<span class="co">## </span>
<span class="co">## P1              2062 ACGATGATGATTTCGATGACACAAACGCATTCAGCACCATCGAAAAC-CC   2110</span>
<span class="co">##                        |  | || || |  |   || | |||   || |  | || | | | ||</span>
<span class="co">## S1              2003 GGGTGGCTGTTTGCTTTTGGACGA-CGC---CACCCTCGTCCATAGCGCC   2048</span>
<span class="co">## </span>
<span class="co">## P1              2111 AAGTCTGGCACCCAAAATGAATCAAACAAG--GACTTTTCAAAAAAAAAG   2158</span>
<span class="co">##                      ||| |   ||||  || ||  |  |  | |  | ||             |</span>
<span class="co">## S1              2049 AAGGC---CACCGGAATTGTCTTTATTACGCCGGCT-------------G   2082</span>
<span class="co">## </span>
<span class="co">## P1              2159 GGTTGAAAATTTTTCTAAGTTTTTCAAAGAAATAGTGCCGCCTGATAAGG   2208</span>
<span class="co">##                      ||  |    |||   |  |  | |   |||  | | |||||   | || |</span>
<span class="co">## S1              2083 GGAAGTCCCTTTGCATCGGCCTGTATGAGACTTTG-GCCGC---AAAAAG   2128</span>
<span class="co">## </span>
<span class="co">## P1              2209 CTAAGCCAGTTATCGGCAAAAATTTCTAGCGTCCTAAACGGCTGGTACTA   2258</span>
<span class="co">##                        ||| ||    || ||| |   |||| |          |||   |||  </span>
<span class="co">## S1              2129 TGAAGACATGGCTCCGCAGA---TTCTTG----------GGCATCTAC--   2163</span>
<span class="co">## </span>
<span class="co">## P1              2259 GGGAGTTTTCCACCTTATTTTTC-GCGCTGAATTC--AAAT----TTTAT   2301</span>
<span class="co">##                          ||| ||  |||| ||  || | ||| || ||  ||||    ||| |</span>
<span class="co">## S1              2164 ----GTTCTCTTCCTTTTTGGTCTGTGCTCAAGTCGAAAATCGCCTTTCT   2209</span>
<span class="co">## </span>
<span class="co">## P1              2302 GATAAGGAATCGCTAAAGTCTCCGACTC---TGCTATTACTTTTCATACT   2348</span>
<span class="co">##                        |   | || |||        | | ||   ||| || |  |  |||   </span>
<span class="co">## S1              2210 TGTGTCGGATAGCTGTGCAGCTCAAATCAAGTGCAATGAGCT--CAT---   2254</span>
<span class="co">## </span>
<span class="co">## P1              2349 CCCTCTAGAATTTCTCACTAGAATATTCAATAATTTCTAACATTTGAAAA   2398</span>
<span class="co">##                       | || | | | ||  |  |||| ||  |||     | | |         </span>
<span class="co">## S1              2255 -CATCAAAAGTCTCCAAGAAGAAGATGGAATCGACGCCATC---------   2294</span>
<span class="co">## </span>
<span class="co">## P1              2399 AGTGCTGGTTTGTTCCATTTTTGATTCTAGCCTGAGGTTTTTGAAGATGC   2448</span>
<span class="co">##                       |   |  ||||| | | |    |    ||||||        || |  | </span>
<span class="co">## S1              2295 -GCCATCCTTTGT-CAAATGCACAGCACAGCCTG--------GATG--GG   2332</span>
<span class="co">## </span>
<span class="co">## P1              2449 TGAATGCGTTTTTTACATCAAAATCATCATTGGGTGCTAAACTCTTCCTT   2498</span>
<span class="co">##                      | ||| | ||||               ||||          |||||||||</span>
<span class="co">## S1              2333 TAAATTCATTTT---------------CATT----------CTCTTCCTT   2357</span>
<span class="co">## </span>
<span class="co">## P1              2499 GATTTCAATAAT------AAAAATCCGTTCCTTGGCCAAACATTCTGCTT   2542</span>
<span class="co">##                      ||||||||| |       |||||| |   |||    |||| | |  ||| </span>
<span class="co">## S1              2358 GATTTCAATTACCTTTAGAAAAATACA--CCTA---CAAAGACTGGGCT-   2401</span>
<span class="co">## </span>
<span class="co">## P1              2543 CTGCATCCGAACCAGTCCTCAAGATAAGCTTCGAGGATGTTTCGCAGGTT   2592</span>
<span class="co">##                      | || | | ||  || || ||       |||| ||     | |  | | |</span>
<span class="co">## S1              2402 CAGCCTGCAAAAAAGACC-CA-------CTTCCAGCCGAATCCAGACGAT   2443</span>
<span class="co">## </span>
<span class="co">## P1              2593 GTCTCGATTGTGGCCTGTCAAGTTCAGCGTTCGCGAGCCGAAGAGTTTCA   2642</span>
<span class="co">##                      |  || ||   || |   |  | ||  | |   | |    |||  || ||</span>
<span class="co">## S1              2444 GCTTCAAT---GGACCAGCCTGATCCTCTTCTTCAA----AAGGCTTCCA   2486</span>
<span class="co">## </span>
<span class="co">## P1              2643 TGGTAGCCATGTGGACCTTGGAAGCCTT----TTGAAGAAGAGGATCAGG   2688</span>
<span class="co">##                       |||   ||||   ||| || ||  |||    | |   | |  || |  |</span>
<span class="co">## S1              2487 AGGTCCACATGGCTACCATGAAACTCTTCGGCTCGCGAACGCTGAACTTG   2536</span>
<span class="co">## </span>
<span class="co">## P1              2689 CTGGTCCA---TTGAAGCATCGTCTGGATTCGGCTGGAAG-------TG-   2727</span>
<span class="co">##                         | |||   | ||  || | |  | |     || ||||       || </span>
<span class="co">## S1              2537 ACAGGCCACAATCGAGACAACCTGCGAAACATCCTCGAAGCTTATCTTGA   2586</span>
<span class="co">## </span>
<span class="co">## P1              2728 GGTCTTTTTTGCAGGCTGA-GCCCAGTCTTTGT---AGGT--GTATTTTT   2771</span>
<span class="co">##                      || ||  || | | || || ||  | | ||||    |||   | ||||||</span>
<span class="co">## S1              2587 GGACTGGTTCGGATGCAGAAGCAGAATGTTTGGCCAAGGAACGGATTTTT   2636</span>
<span class="co">## </span>
<span class="co">## P1              2772 CTAAAGGTAATTGAAATCAAGGAAGAG----------AATGA--------   2803</span>
<span class="co">##                       | |      |||||||||||||||||          |||||        </span>
<span class="co">## S1              2637 ATTA------TTGAAATCAAGGAAGAGTTTAGCACCCAATGATGATTTTG   2680</span>
<span class="co">## </span>
<span class="co">## P1              2804 -------AAATGAATTTACC--CATC--------CAGGCTGTGCTGTGCA   2836</span>
<span class="co">##                             ||| | ||| | |  | ||        ||||||    |    |</span>
<span class="co">## S1              2681 ATGTAAAAAACGCATTCAGCATCTTCAAAAACCTCAGGCTAGAATCAAAA   2730</span>
<span class="co">## </span>
<span class="co">## P1              2837 TTTGA-CAAAGGATGGC----------GATGGCGTCGATTCCATCTTCTT   2875</span>
<span class="co">##                       | || ||||  |   |          | | |     |||  || |||| </span>
<span class="co">## S1              2731 ATGGAACAAACCAGCACTTTTTCAAATGTTAGAAATTATTGAATATTCTA   2780</span>
<span class="co">## </span>
<span class="co">## P1              2876 CTTGGAGACTTTTGATG----ATGAGC--TCATTGCACTTGATTTGAGCT   2919</span>
<span class="co">##                       |  || | | | || |    ||||    | || |||   || | |    </span>
<span class="co">## S1              2781 GTGAGAAATTCTAGAGGGAGTATGAAAAGTAATAGCA---GAGTCGGAGA   2827</span>
<span class="co">## </span>
<span class="co">## P1              2920 GCACAGCTATCCGACACAAGAAAGGCGATTTTCGACTTGAGCACAGACCA   2969</span>
<span class="co">##                          ||| || |   |  | |||    ||||  || || ||| | ||  |</span>
<span class="co">## S1              2828 CTTTAGCGATTCCTTATCATAAA----ATTT--GAATTCAGCGC-GAAAA   2870</span>
<span class="co">## </span>
<span class="co">## P1              2970 AAAAGGAAGAGAAC------GTAGATGCC----------CAAGAA---TC   3000</span>
<span class="co">##                      | ||||  || |||      |||   |||          | ||||   | </span>
<span class="co">## S1              2871 ATAAGGTGGAAAACTCCCTAGTACCAGCCGTTTAGGACGCTAGAAATTTT   2920</span>
<span class="co">## </span>
<span class="co">## P1              3001 TGCGGAGCCATGTCTTCACTTTTT---GCGGC-CAAAGTCTCATACAGGC   3046</span>
<span class="co">##                      ||| ||    || |||  | || |   ||||| | |  |||   | |  |</span>
<span class="co">## S1              2921 TGCCGATAACTGGCTTAGCCTTATCAGGCGGCACTATTTCTTTGAAAAAC   2970</span>
<span class="co">## </span>
<span class="co">## P1              3047 CGATGCAAAGGGACTTCCC-------------AGCCGGCGTAATAAAGAC   3083</span>
<span class="co">##                        |   |||    |  |||             || |   ||  |  |  |</span>
<span class="co">## S1              2971 TTAGAAAAATTTTCAACCCTTTTTTTTTGAAAAGTCCTTGT--TTGATTC   3018</span>
<span class="co">## </span>
<span class="co">## P1              3084 AATTCCGGTGGC---CTTGGCGCTATGGACGAGGGTG---GCGTC-GTCC   3126</span>
<span class="co">##                      | ||  |||| |   ||||| | | | || |  | ||   ||||  ||  </span>
<span class="co">## S1              3019 ATTTTGGGTGCCAGACTTGG-GTTTTCGATGGTGCTGAATGCGTTTGTGT   3067</span>
<span class="co">## </span>
<span class="co">## P1              3127 AAAAGCAAACAGCCACCCTCGTC---CGCTGCCTCAATTTCTTTTGCCAG   3173</span>
<span class="co">##                       |  | || || |  |    |||   | || |||  |||||  |      </span>
<span class="co">## S1              3068 CATCGAAATCATCATCGTGTGTCAAACTCTTCCTTGATTTCAAT------   3111</span>
<span class="co">## </span>
<span class="co">## P1              3174 ATGGGATTCCGTCAAAGTATCGATGCAGTAGCTAAAAGATT-AGCTTCAA   3222</span>
<span class="co">##                      ||   |   | | |||      |  ||   ||| || |||| |   ||| </span>
<span class="co">## S1              3112 ATTAAAAATCTTAAAAA-----AGACATGGGCTCAA-GATTCATGATCA-   3154</span>
<span class="co">## </span>
<span class="co">## P1              3223 TTGAAATCAAGGAAGAGAACCCTACCGTTTCTCATTGATCGTGCTGATCG   3272</span>
<span class="co">##                      |   || |||  | ||||| |||   ||||     ||||| |||  |   </span>
<span class="co">## S1              3155 TCCCAAGCAAC-ATGAGAAGCCTG--GTTT-----TGATCATGCCTA---   3193</span>
<span class="co">## </span>
<span class="co">## P1              3273 AAGGGGGCGTCTGATCCTTGAACTCATCGATGCACTTGCACAAGACGTTC   3322</span>
<span class="co">##                                || ||||  |||| | || |      | | || |   |||</span>
<span class="co">## S1              3194 ----------CT-ATCC--GAACGCTTCTACAGGGCTACGCAGGGGATTC   3230</span>
<span class="co">## </span>
<span class="co">## P1              3323 AAGAAGTTTGAC--AACGACTT-GCCATGTTCTCCGCGCATGAGAAGGCA   3369</span>
<span class="co">##                       || ||   | |  ||   ||| || || ||||     |||     ||| </span>
<span class="co">## S1              3231 TAGTAGCACGGCGAAATTTCTTCGCGATTTTCTGGAGTCATTCT--GGCC   3278</span>
<span class="co">## </span>
<span class="co">## P1              3370 ACA-GAGAATCTTGATCAGGTTCGGATGAAAGGAGACTTTCGAAAATCCA   3418</span>
<span class="co">##                       || ||  |   |||| |||    || |||||  | |    ||||||   </span>
<span class="co">## S1              3279 TCACGAAGAAAATGATGAGGAAATGAAGAAAGCTG-CAAAGGAAAATA--   3325</span>
<span class="co">## </span>
<span class="co">## P1              3419 CGGCCATCGACGCTGTCTAATGAGTCAAGGCTCAGCCGCTGTTTTTTGAA   3468</span>
<span class="co">##                       ||| |  |    | | | |  |||    | |  |  | |||  ||  | </span>
<span class="co">## S1              3326 -GGCAAAGGTTCATCTATCACTAGTTGTAGTT--GTAGTTGTAGTTGTAG   3372</span>
<span class="co">## </span>
<span class="co">## P1              3469 TGTCTCGCTCGCCGATGTTGACAACTTTCGCGACTCGTCTTCTTCTTCTT   3518</span>
<span class="co">##                      | | | | | |  | |||||                 | || || || ||</span>
<span class="co">## S1              3373 T-TGTAGTT-GTAGTTGTTG----------------TTGTTGTTGTTGTT   3404</span>
<span class="co">## </span>
<span class="co">## P1              3519 CTTTTTCTTCTTCTTTTTCTTCCGGCTGAGA-TCGT-CGGATCATCGCGT   3566</span>
<span class="co">##                       || ||  ||   | ||  | ||     ||| |||| |   || | || |</span>
<span class="co">## S1              3405 GTTGTTGATCGAATGTTGATCCCACAAAAGACTCGTTCTCCTCCTTGCTT   3454</span>
<span class="co">## </span>
<span class="co">## P1              3567 GAG-----------ATGTTG-ATGGTTCTTCAGAAAAGACGTCATCGTCA   3604</span>
<span class="co">##                      |             || ||  |||| || |  ||| | |  ||||   ||</span>
<span class="co">## S1              3455 GCATCCCATTTGCCATTTTCTATGGCTCCTT-GAAGAAATATCATG--CA   3501</span>
<span class="co">## </span>
<span class="co">## P1              3605 TCA-TCATCGGA-CATGGCGAGCTTGAGTTAAAATATTAACGAAAGGA--   3650</span>
<span class="co">##                       |  |||  | | |||  | ||||      |||| |||   |||   |  </span>
<span class="co">## S1              3502 GCTGTCAGTGAAACAT--CAAGCTCCTTGAAAAAGATTGTAGAAGCTATC   3549</span>
<span class="co">## </span>
<span class="co">## P1              3651 GGGAAAAGGTAATGA-----ACCTGAGAAAGGGAG-ATTTGTGAAGATCT   3694</span>
<span class="co">##                      | |||   | || ||     ||  | ||| |   | || || |||||   </span>
<span class="co">## S1              3550 GCGAATTTGGAA-GACTACTACAAGCGAATGATGGCATCTGAGAAGAATA   3598</span>
<span class="co">## </span>
<span class="co">## P1              3695 CTTCCTTGATTCCGATTTCGCTTGACGTGGTGGCTGCCTTGGAAGTTCTT   3744</span>
<span class="co">##                       || || |    ||| |  |||| | |   ||||   |   ||      |</span>
<span class="co">## S1              3599 TTTACTGG----CGAAT--GCTTCAAGAACTGGCCAACGACGACAACGAT   3642</span>
<span class="co">## </span>
<span class="co">## P1              3745 GA-ACAAACTTCCCGAAAGGATACTGGCGGTTTTT--TTCAATTCCGA-C   3790</span>
<span class="co">##                      || |||||       | ||   |   |||  |  |  | || | |||| |</span>
<span class="co">## S1              3643 GAGACAAAA------ACAGTCAAGCAGCGACTACTCATCCAGTACCGAGC   3686</span>
<span class="co">## </span>
<span class="co">## P1              3791 TTTCGA----GAAGATCTTGTCGACTTGG----CTTTTC-ACACGGCCTC   3831</span>
<span class="co">##                      | | ||    ||| |        ||||||    | || | |  | | || </span>
<span class="co">## S1              3687 TCTTGATGGTGAACAAAAGAAGCACTTGGACAGCGTTACTAGGCAGTCTG   3736</span>
<span class="co">## </span>
<span class="co">## P1              3832 TCGTCCCGACCCACTTGAAATTCGACG--GGATTTCGAAAAA----ATGC   3875</span>
<span class="co">##                      |||  |  | |     ||| ||| |||  |||    ||||||    | ||</span>
<span class="co">## S1              3737 TCGGTCGCAGCTTTGAGAAGTTC-ACGAAGGAGAAAGAAAAATGGGACGC   3785</span>
<span class="co">## </span>
<span class="co">## P1              3876 GGTCGCTTTTC---TTTA-AATTAA--------TTTGAAATCAACGAAGA   3913</span>
<span class="co">##                        ||||   ||   |  | ||  ||        ||||| | | ||     </span>
<span class="co">## S1              3786 AATCGCAACTCGAGTCCAGAACGAAAACAGCTTTTTGACAGCGACTTCTC   3835</span>
<span class="co">## </span>
<span class="co">## P1              3914 GAAACCTTTGAATCTTACCTTCAGAAATAA---GAGCTTCTAACAAAAGT   3960</span>
<span class="co">##                      ||  | |  ||| |     ||  || || |   |||||   || || |  </span>
<span class="co">## S1              3836 GACGCGTG-GAAGCCCTGTTTGGGACATTAAAGGAGCTGGAAAGAACACA   3884</span>
<span class="co">## </span>
<span class="co">## P1              3961 CTCCTTGCCGCAGAATTAGGAGCAGATACTTTCTTTCTCTCGC----GTA   4006</span>
<span class="co">##                       ||   ||     | || | || | ||   |  |  || |  |    |||</span>
<span class="co">## S1              3885 ATCGAGGCTTTC-ACTTCGAAGAACATTTGTCGTGGCTGTTTCCATGGTA   3933</span>
<span class="co">## </span>
<span class="co">## P1              4007 A-----GCCTTCTTCGTCATTTTGTTGAGATT-TCTCAT-----------   4039</span>
<span class="co">##                      |     ||||  ||  | ||| | || | | | ||||||           </span>
<span class="co">## S1              3934 ATTTTGGCCTATTT--TTATTATATTAACAATGTCTCATTCAGTACAATA   3981</span>
<span class="co">## </span>
<span class="co">## P1              4040 --ATGAGTCTG---------------------GAGGAATCTCTTCAAGAT   4066</span>
<span class="co">##                        |  |||||                      || |||| ||||  ||| </span>
<span class="co">## S1              3982 AAACAAGTCTCTGGCTTGAAGACCTTCGGGATGATGAATGTCTTGTAGAG   4031</span>
<span class="co">## </span>
<span class="co">## P1              4067 GCT-CAAGAATGGTCATTGCTCGGGTGAAAT-TGAGCTTAAA--AGTGGT   4112</span>
<span class="co">##                        | ||| |  | |  |||  || | ||||  ||||   |||  |  || </span>
<span class="co">## S1              4032 ATTACAAAAGCGATGGTTGAACGAGCGAAAAATGAGAAGAAAGCAAAGGA   4081</span>
<span class="co">## </span>
<span class="co">## P1              4113 ACG-----AGGATCATCGATTGCA-TCCATTG-CGAGAAAAAGAAAGAAA   4155</span>
<span class="co">##                      ||      || |  | | | || | | ||  | | ||||  ||  |||  </span>
<span class="co">## S1              4082 ACTCGAAAAGAAGGACCAAATGAAATACAAAGACAAGAACGAGCGAGAGG   4131</span>
<span class="co">## </span>
<span class="co">## P1              4156 AAGAAA----AATAAATGACGGTGA-GAGAAGGAA-GAGAAAAAG-GATA   4198</span>
<span class="co">##                        ||||    ||| || || | ||| ||| | ||| ||||  ||| ||| </span>
<span class="co">## S1              4132 CTGAAATCGAAATCAACGAAGATGACGAGGAAGAACGAGAGGAAGAGATT   4181</span>
<span class="co">## </span>
<span class="co">## P1              4199 AGTGAA--AGCGGA-GCGCGCGTCGAATCAGCGCCGACTGATAAGACTGC   4245</span>
<span class="co">##                        ||||  || ||| |     ||||||   |    |   || | ||  | </span>
<span class="co">## S1              4182 GTTGAAGAAGTGGAAGAAATAGTCGAAAAGGAATTGGACGAAATGAACGA   4231</span>
<span class="co">## </span>
<span class="co">## P1              4246 GCAAGTGGCA-AGATTGACTCAGAAGATGAATCAAGGAAGAAAAATAAGT   4294</span>
<span class="co">##                      | |||  | | ||| || |   | ||||| ||   | | |||       |</span>
<span class="co">## S1              4232 GGAAGAAGAACAGAATGTCGACG-AGATGGATTTTGAAGGAATTTCTTCT   4280</span>
<span class="co">## </span>
<span class="co">## P1              4295 GATTTTGGACGAAACCTTTCAATTAATGAAATAAAATTTC---AATAAAC   4341</span>
<span class="co">##                      ||   |   |  ||||    ||| ||| ||||    |||    ||| | |</span>
<span class="co">## S1              4281 GAACCTT--CTGAACCACAGAATGAATAAAATGGTTTTTTGGAAATTATC   4328</span>
<span class="co">## </span>
<span class="co">## P1              4342 TCATTCAAAT---------TAAGAAATTTTAGCTTATTATAGACTTTGCC   4382</span>
<span class="co">##                      |  ||    |         | | |||    |  | |  ||  | ||| ||</span>
<span class="co">## S1              4329 TTGTTTTGTTGTTTTGATCTTATAAAAAAAACATAAAAATCAAATTTCCC   4378</span>
<span class="co">## </span>
<span class="co">## P1              4383 GTCGAAAAATACCAATATTATCATGTTTTAGAAGGTTTCTAGAGTTGAAC   4432</span>
<span class="co">##                      ||||||||||||| |||||  ||||||| |||||||||||||||||||||</span>
<span class="co">## S1              4379 GTCGAAAAATACCCATATTGCCATGTTTAAGAAGGTTTCTAGAGTTGAAC   4428</span>
<span class="co">## </span>
<span class="co">## P1              4433 CGCAAGGAACCACCTGGTTGACCACCCAGTTCGACCGCGGTTTAACTTTT   4482</span>
<span class="co">##                      ||| | ||||||||||||||||||||| ||| |||||||||| |||||||</span>
<span class="co">## S1              4429 CGCGATGAACCACCTGGTTGACCACCCGGTTTGACCGCGGTTCAACTTTT   4478</span>
<span class="co">## </span>
<span class="co">## P1              4483 ATTTTCAATTTCATTCAATTTCCCGTCGAAAAATACCCATATTGCAAGGT   4532</span>
<span class="co">##                      ||||||| ||||||||||||||||||||||||||||||||||| | | ||</span>
<span class="co">## S1              4479 ATTTTCATTTTCATTCAATTTCCCGTCGAAAAATACCCATATTCCCATGT   4528</span>
<span class="co">## </span>
<span class="co">## P1              4533 TTTAGAAGAATTCTAGAGTTGAACCGCGATGAACCACCTGGTTGACCACG   4582</span>
<span class="co">##                      ||||||||  |||||||||||||||||| | ||||   ||||||||||||</span>
<span class="co">## S1              4529 TTTAGAAGGTTTCTAGAGTTGAACCGCGGTTAACCGGGTGGTTGACCACG   4578</span>
<span class="co">## </span>
<span class="co">## P1              4583 TGGTTCGACCGTGGTTCAACTTTTATTTTCATTTTCATTCAATTTCCCGT   4632</span>
<span class="co">##                      ||||||||||||||||||||||||||||||||||||||||||||||||||</span>
<span class="co">## S1              4579 TGGTTCGACCGTGGTTCAACTTTTATTTTCATTTTCATTCAATTTCCCGT   4628</span>
<span class="co">## </span>
<span class="co">## P1              4633 CGAAAAATACCCATATTCCCATGTTTTAGACGGTTTCTAGAGTTGAACCG   4682</span>
<span class="co">##                      |||||||||||||||||||||||||||||| |||||||||||||||||||</span>
<span class="co">## S1              4629 CGAAAAATACCCATATTCCCATGTTTTAGAAGGTTTCTAGAGTTGAACCG   4678</span>
<span class="co">## </span>
<span class="co">## P1              4683 CGATGAACCACCTGGTTGACCACCCGGTTCGACCGCGGTTCATCATTTTT   4732</span>
<span class="co">##                      ||||||||||||||||||||||||||||||||||||||||||||||||||</span>
<span class="co">## S1              4679 CGATGAACCACCTGGTTGACCACCCGGTTCGACCGCGGTTCATCATTTTT   4728</span>
<span class="co">## </span>
<span class="co">## P1              4733 GAGCGTCAATTGTTGTTTGTTTATGTTTTACAAACATAATATAAGAGTTT   4782</span>
<span class="co">##                      ||||||||||||||||||||||||||||||||||||||||||||||||||</span>
<span class="co">## S1              4729 GAGCGTCAATTGTTGTTTGTTTATGTTTTACAAACATAATATAAGAGTTT   4778</span>
<span class="co">## </span>
<span class="co">## P1              4783 AGAATTCTAAATCCGAGTTTTCCGGAACTTCCGGTTTTTCACCCAACTTC   4832</span>
<span class="co">##                      ||||||||||||||||||||||||||||||||||||||||||||||||||</span>
<span class="co">## S1              4779 AGAATTCTAAATCCGAGTTTTCCGGAACTTCCGGTTTTTCACCCAACTTC   4828</span>
<span class="co">## </span>
<span class="co">## P1              4833 TAGCGCCCCCAGGTGGCCGCAGAGACATTTGCACTAAGATTTCTTGGGCT   4882</span>
<span class="co">##                      ||||||||||||||||||||||||||||||||||||||||||||||||||</span>
<span class="co">## S1              4829 TAGCGCCCCCAGGTGGCCGCAGAGACATTTGCACTAAGATTTCTTGGGCT   4878</span>
<span class="co">## </span>
<span class="co">## P1              4883 CATCTGAAAGCCCAGAGCCTGAGCTTTCTAAATATATAAGTCTCGACGGT   4932</span>
<span class="co">##                      ||||||||||||||||||||||||||||||||||||||||||||||||||</span>
<span class="co">## S1              4879 CATCTGAAAGCCCAGAGCCTGAGCTTTCTAAATATATAAGTCTCGACGGT   4928</span>
<span class="co">## </span>
<span class="co">## P1              4933 TTTGGAGGCCGGCCCGCCTCCAAAAAAATAGTTTGAAGAGTTCTAGTGCC   4982</span>
<span class="co">##                      ||||||||||||||||||||||||||||||||||||||||||||||||||</span>
<span class="co">## S1              4929 TTTGGAGGCCGGCCCGCCTCCAAAAAAATAGTTTGAAGAGTTCTAGTGCC   4978</span>
<span class="co">## </span>
<span class="co">## P1              4983 TATTTTGGGACTTCAGGTCCCATAATTTGCCTTGCGGGCATGACCCAGAC   5032</span>
<span class="co">##                      ||||||||||||||||||||||||||||||||||||||||||||||||||</span>
<span class="co">## S1              4979 TATTTTGGGACTTCAGGTCCCATAATTTGCCTTGCGGGCATGACCCAGAC   5028</span>
<span class="co">## </span>
<span class="co">## P1              5033 CCCCTGATTTTAGGCTCAAATGAAAGCCCAGCTTCTGAGCTTTCTAAAAA   5082</span>
<span class="co">##                      ||||||||||||||||||||||||||||||||||||||||||||||||||</span>
<span class="co">## S1              5029 CCCCTGATTTTAGGCTCAAATGAAAGCCCAGCTTCTGAGCTTTCTAAAAA   5078</span>
<span class="co">## </span>
<span class="co">## P1              5083 TGTATCATATGACCCCCCCACTAAGCTAGAAAAATCGGCCAAATTTTAGT   5132</span>
<span class="co">##                      ||||||||||||||||||||||||||||||||||||||||||||||||||</span>
<span class="co">## S1              5079 TGTATCATATGACCCCCCCACTAAGCTAGAAAAATCGGCCAAATTTTAGT   5128</span>
<span class="co">## </span>
<span class="co">## P1              5133 CCGGTAAGCCT----AAA   5146</span>
<span class="co">##                      |||||||||||    |  </span>
<span class="co">## S1              5129 CCGGTAAGCCTACGTATC   5146</span>
<span class="co">## </span>
<span class="co">## </span>
<span class="co">## #---------------------------------------</span>
<span class="co">## #---------------------------------------</span></code></pre>
</details><p>Immediately upstream these repeats, there a tandem octamer <code>AAGATACG</code>. In the Okinawan genome, it is present as a single copy. You can see it in the alignment below, where the right-side boundary was not reverse-complemented.</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ROI1</span><span class="op">$</span><span class="va">query</span> |&gt; <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/cleanGaps.html" class="external-link">cleanGaps</a></span><span class="op">(</span><span class="op">)</span> |&gt; <span class="fu">alignBreakPoints</span><span class="op">(</span><span class="fl">30</span>, rev<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Global PairwiseAlignmentsSingleSubject (1 of 1)</span>
<span class="co">## pattern: <span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span>--<span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">TTTTT</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">CC</span><span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">GG</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">CC</span><span style="color: #000000; background-color: #AFFFFF;">GG</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">AAAA</span><span style="color: #000000; background-color: #FFD7AF;">TTT</span><span style="color: #000000; background-color: #AFFFFF;">G</span>-<span style="color: #000000; background-color: #AFFFFF;">G</span></span>
<span class="co">## subject: <span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #AFFFFF;">GG</span><span style="color: #000000; background-color: #AFFFAF;">CC</span><span style="color: #000000; background-color: #FFAFAF;">AAA</span><span style="color: #000000; background-color: #FFD7AF;">TTTT</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">CC</span><span style="color: #000000; background-color: #AFFFFF;">GG</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">A</span>--<span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">CC</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">AAA</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #AFFFFF;">G</span>-<span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">AAA</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">TTT</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">G</span></span>
<span class="co">## score: -148.066</span></code></pre>
<p>The same can be seen in the multiple alignment below comparising the left and right arms of the Kume alignments and their match to the Okinawa genome.</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Note that genome alignment is +/- therefore the order below is counter-intuitive.</span>
<span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>
  kum_right <span class="op">=</span> <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/getSeq.html" class="external-link">getSeq</a></span><span class="op">(</span><span class="va">ROI1</span><span class="op">$</span><span class="va">query</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/intra-range-methods.html" class="external-link">flank</a></span><span class="op">(</span><span class="fl">20</span>, both <span class="op">=</span> <span class="cn">T</span>, start <span class="op">=</span> <span class="cn">T</span><span class="op">)</span><span class="op">)</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span>,
  kum_left  <span class="op">=</span> <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/getSeq.html" class="external-link">getSeq</a></span><span class="op">(</span><span class="va">ROI1</span><span class="op">$</span><span class="va">query</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/intra-range-methods.html" class="external-link">flank</a></span><span class="op">(</span><span class="fl">20</span>, both <span class="op">=</span> <span class="cn">T</span>, start <span class="op">=</span> <span class="cn">F</span><span class="op">)</span><span class="op">)</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span>,
  oki       <span class="op">=</span> <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/getSeq.html" class="external-link">getSeq</a></span><span class="op">(</span><span class="va">ROI1</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span>       |&gt; <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/intra-range-methods.html" class="external-link">flank</a></span><span class="op">(</span><span class="fl">20</span>, both <span class="op">=</span> <span class="cn">T</span>, start <span class="op">=</span> <span class="cn">F</span><span class="op">)</span><span class="op">)</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span>
<span class="op">)</span> |&gt; <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/as.html" class="external-link">as</a></span><span class="op">(</span><span class="st">"DNAStringSet"</span><span class="op">)</span> |&gt; <span class="fu">msa</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/msa/man/msaClustalW.html" class="external-link">msaClustalW</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## use default substitution matrix</span></code></pre>
<pre><code><span class="co">## CLUSTAL 2.1  </span>
<span class="co">## </span>
<span class="co">## Call:</span>
<span class="co">##    msa::msaClustalW(as(list(kum_right = getSeq(flank(ROI1$query[1],     20, both = T, start = T))[[1]], kum_left = getSeq(flank(ROI1$query[2],     20, both = T, start = F))[[1]], oki = getSeq(flank(ROI1[2],     20, both = T, start = F))[[1]]), "DNAStringSet"))</span>
<span class="co">## </span>
<span class="co">## MsaDNAMultipleAlignment with 3 rows and 41 columns</span>
<span class="co">##     aln                                       names</span>
<span class="co">## [1] CAGAGCTTTTTCAGATCCAAGATACGTAGGCTTACCGGAC- kum_left</span>
<span class="co">## [2] CAGAGCTTTTTCAGATCCAAGATACG-ACGCATTCTCTTCG oki</span>
<span class="co">## [3] TTTAGTCCGGTAAGCCTAAAGATACG-ACGCATTCTCTTCG kum_right</span>
<span class="co">## Con CAGAGCTTTTTCAGATCCAAGATACG-ACGCATTCTCTTCG Consensus</span></code></pre>
<p>The microhomology does not appear to be more frequent in the Okinawan genome compared to the others.</p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">genomes</span>, <span class="va">vmatchPattern</span>, pat <span class="op">=</span> <span class="st">"AAGATACG"</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">length</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">##  OKI2018.I69 OSKA2016v1.9      Bar2.p4    KUM.M3.7f     AOM.5.5f         OdB3 </span>
<span class="co">##         1478         1334         1485         1493         1324         1730</span></code></pre>
</div>
<div class="section level3">
<h3 id="how-often-do-we-see-tandem-site-duplication-in-similar-indel-configurations">How often do we see tandem site duplication in similar indel configurations ?<a class="anchor" aria-label="anchor" href="#how-often-do-we-see-tandem-site-duplication-in-similar-indel-configurations"></a>
</h3>
<p>The <code>unalMap</code> objects contain unaligned regions flanked by colinear regions. Let’s search for other situations where one of the regions has a width of zero.</p>
<p>Note: <em>in order to represent zero-width intervals in this object, which would be impossible with <code>GRanges</code> objects, the unaligned regions were arbitrarly added one base of each flanking aligned regions</em>.</p>
<p>These regions are rare in between-species alignments (at most a few percents), and common in within-population alignments (14–28 percents).</p>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">isIndel</span> <span class="op">&lt;-</span> \<span class="op">(</span><span class="va">gb</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/start.html" class="external-link">width</a></span><span class="op">(</span><span class="va">gb</span><span class="op">)</span> <span class="op">&lt;=</span> <span class="fl">2</span> <span class="op">|</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/start.html" class="external-link">width</a></span><span class="op">(</span><span class="va">gb</span><span class="op">$</span><span class="va">query</span><span class="op">)</span> <span class="op">&lt;=</span> <span class="fl">2</span>
<span class="va">unalMap</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">sapply</a></span><span class="op">(</span>\<span class="op">(</span><span class="va">gb</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">gb</span><span class="op">[</span><span class="fu">isIndel</span><span class="op">(</span><span class="va">gb</span><span class="op">)</span><span class="op">]</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Oki_Osa Oki_Bar Oki_Kum Oki_Aom Oki_Nor Osa_Oki Osa_Bar Osa_Kum Osa_Aom Osa_Nor </span>
<span class="co">##    1206     997    3848    1111     942    1206    2013    1179    3524    2093 </span>
<span class="co">## Bar_Oki Bar_Osa Bar_Kum Bar_Aom Bar_Nor Ply_Ros Ply_Rob Ply_Sav Ply_Oki Rob_Ros </span>
<span class="co">##     964    1989     952    1948    3929    8896    9537     142     149    8796 </span>
<span class="co">## Rob_Ply Rob_Sav Rob_Oki Dme_Dbu Dme_Dsu Dme_Dya Dme_Dma </span>
<span class="co">##    8645     140     186     124     421     897    1574</span></code></pre>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/Round.html" class="external-link">round</a></span><span class="op">(</span><span class="va">unalMap</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">sapply</a></span><span class="op">(</span>\<span class="op">(</span><span class="va">gb</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">gb</span><span class="op">[</span><span class="fu">isIndel</span><span class="op">(</span><span class="va">gb</span><span class="op">)</span><span class="op">]</span><span class="op">)</span><span class="op">)</span> <span class="op">/</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">unalMap</span>, <span class="va">length</span><span class="op">)</span> <span class="op">*</span> <span class="fl">100</span> <span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Oki_Osa Oki_Bar Oki_Kum Oki_Aom Oki_Nor Osa_Oki Osa_Bar Osa_Kum Osa_Aom Osa_Nor </span>
<span class="co">##       5       4      40       5       4       5      10       5      34      11 </span>
<span class="co">## Bar_Oki Bar_Osa Bar_Kum Bar_Aom Bar_Nor Ply_Ros Ply_Rob Ply_Sav Ply_Oki Rob_Ros </span>
<span class="co">##       4      10       4       9      44      27      19       0      18      16 </span>
<span class="co">## Rob_Ply Rob_Sav Rob_Oki Dme_Dbu Dme_Dsu Dme_Dya Dme_Dma </span>
<span class="co">##      16       0      20       0       1       4      22</span></code></pre>
<p>Let’s inspect within-population alignments first, considering only cases where the indel region is on the <em>query</em> genome. Here is an example with a <code>CATCTTTG</code> TSD.</p>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">gb</span> <span class="op">&lt;-</span> <span class="va">unalMap</span><span class="op">$</span><span class="va">Oki_Kum</span> |&gt; <span class="fu">plyranges</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/plyranges/man/tidyverse-reexports.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">width</span> <span class="op">==</span> <span class="fl">2</span><span class="op">)</span>
<span class="va">gb</span><span class="op">$</span><span class="va">query</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/start.html" class="external-link">width</a></span><span class="op">(</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/CNEr/man/Axt-util.html" class="external-link">summary</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">##    Min. 1st Qu.  Median    Mean 3rd Qu.    Max. </span>
<span class="co">##    58.0   254.2   441.0  1006.1   757.2 19273.0</span></code></pre>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">alignBreakPoints</span><span class="op">(</span><span class="va">gb</span><span class="op">$</span><span class="va">query</span><span class="op">[</span><span class="fl">4</span><span class="op">]</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Global PairwiseAlignmentsSingleSubject (1 of 1)</span>
<span class="co">## pattern: <span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">CC</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">TTT</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #AFFFFF;">GG</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #AFFFFF;">GG</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">CCCCCCCCCCC</span></span>
<span class="co">## subject: <span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">AAAAAA</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">GGGGG</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">CC</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #FFAFAF;">AAA</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">GG</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">TTT</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #AFFFFF;">G</span></span>
<span class="co">## score: -149.28</span></code></pre>
<div class="sourceCode" id="cb37"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">searchForTSDs</span><span class="op">(</span><span class="va">gb</span><span class="op">$</span><span class="va">query</span><span class="op">[</span><span class="fl">4</span><span class="op">]</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/intra-range-methods.html" class="external-link">shift</a></span><span class="op">(</span><span class="fl">1</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Global PairwiseAlignmentsSingleSubject (1 of 1)</span>
<span class="co">## pattern: <span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #AFFFFF;">GG</span><span style="color: #000000; background-color: #FFAFAF;">A</span></span>
<span class="co">## subject: <span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">AAA</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">GG</span><span style="color: #000000; background-color: #FFD7AF;">T</span></span>
<span class="co">## score: -23.55116</span></code></pre>
<div class="sourceCode" id="cb39"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">alignEnds</span><span class="op">(</span><span class="va">gb</span><span class="op">$</span><span class="va">query</span><span class="op">[</span><span class="fl">4</span><span class="op">]</span>, <span class="fl">40</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Global PairwiseAlignmentsSingleSubject (1 of 1)</span>
<span class="co">## pattern: ---<span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #AFFFFF;">GG</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">CCCCCCCCCCC</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">TTTTTTTT</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #AFFFFF;">GG</span><span style="color: #000000; background-color: #FFAFAF;">AAAA</span>--------------</span>
<span class="co">## subject: <span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">AAA</span><span style="color: #000000; background-color: #FFD7AF;">TT</span>-----------------<span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">CC</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">TTTTTT</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">AAAAAA</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">GGGGG</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">CC</span></span>
<span class="co">## score: -167.7059</span></code></pre>
<div class="sourceCode" id="cb41"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">msAlignFlank</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">gb</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>
  query_left  <span class="op">=</span> <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/getSeq.html" class="external-link">getSeq</a></span><span class="op">(</span><span class="va">gb</span><span class="op">$</span><span class="va">query</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/intra-range-methods.html" class="external-link">flank</a></span><span class="op">(</span><span class="fl">20</span>, both <span class="op">=</span> <span class="cn">T</span>, start <span class="op">=</span> <span class="cn">T</span><span class="op">)</span><span class="op">)</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span>,
  target      <span class="op">=</span> <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/getSeq.html" class="external-link">getSeq</a></span><span class="op">(</span><span class="va">gb</span>       |&gt; <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/intra-range-methods.html" class="external-link">flank</a></span><span class="op">(</span><span class="fl">20</span>, both <span class="op">=</span> <span class="cn">T</span>, start <span class="op">=</span> <span class="cn">F</span><span class="op">)</span><span class="op">)</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span>,
  query_right <span class="op">=</span> <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/getSeq.html" class="external-link">getSeq</a></span><span class="op">(</span><span class="va">gb</span><span class="op">$</span><span class="va">query</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/intra-range-methods.html" class="external-link">flank</a></span><span class="op">(</span><span class="fl">20</span>, both <span class="op">=</span> <span class="cn">T</span>, start <span class="op">=</span> <span class="cn">F</span><span class="op">)</span><span class="op">)</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span>
<span class="op">)</span> |&gt; <span class="fu"><a href="https://oist.github.io/GenomicBreaks/reference/as.html" class="external-link">as</a></span><span class="op">(</span><span class="st">"DNAStringSet"</span><span class="op">)</span> |&gt; <span class="fu">msa</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/msa/man/msaClustalW.html" class="external-link">msaClustalW</a></span><span class="op">(</span>order <span class="op">=</span> <span class="st">"input"</span><span class="op">)</span>

<span class="fu">msAlignFlank</span><span class="op">(</span><span class="va">gb</span><span class="op">[</span><span class="fl">4</span><span class="op">]</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## use default substitution matrix</span></code></pre>
<pre><code><span class="co">## CLUSTAL 2.1  </span>
<span class="co">## </span>
<span class="co">## Call:</span>
<span class="co">##    msa::msaClustalW(as(list(query_left = getSeq(flank(gb$query,     20, both = T, start = T))[[1]], target = getSeq(flank(gb,     20, both = T, start = F))[[1]], query_right = getSeq(flank(gb$query,     20, both = T, start = F))[[1]]), "DNAStringSet"), order = "input")</span>
<span class="co">## </span>
<span class="co">## MsaDNAMultipleAlignment with 3 rows and 42 columns</span>
<span class="co">##     aln                                        names</span>
<span class="co">## [1] TTCGCCGCATTTCGGAGTTGAATTAAG-GACCCCCCCCCCC- query_left</span>
<span class="co">## [2] --GGCGGGTTTTCGGAGTTGGATTAAATGGTCTTTCAGATCG target</span>
<span class="co">## [3] --GAAAAAAGCTGAGGGGGTCCTTAAATGGTCTTTCAGATCG query_right</span>
<span class="co">## Con --GGC?G?ATTTCGGAGTTG?ATTAAATGGTCTTTCAGATCG Consensus</span></code></pre>
<p>Let’s count the number of perfect 8-nt matches. See the variety in their sequences: none of them is found more than twice!</p>
<div class="sourceCode" id="cb44"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">searchForTSDs</span><span class="op">(</span><span class="va">gb</span><span class="op">$</span><span class="va">query</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/align-utils.html" class="external-link">nedit</a></span><span class="op">(</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/table.html" class="external-link">table</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## </span>
<span class="co">##  0  1  2  3  4  5  6  7  8 </span>
<span class="co">## 15  6  6 11 16 33 45 21  9</span></code></pre>
<div class="sourceCode" id="cb46"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">gb</span><span class="op">$</span><span class="va">nedit</span> <span class="op">&lt;-</span> <span class="fu">searchForTSDs</span><span class="op">(</span><span class="va">gb</span><span class="op">$</span><span class="va">query</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/align-utils.html" class="external-link">nedit</a></span><span class="op">(</span><span class="op">)</span> 
<span class="va">gb</span><span class="op">$</span><span class="va">consensus</span> <span class="op">&lt;-</span> <span class="fu">searchForTSDs</span><span class="op">(</span><span class="va">gb</span><span class="op">$</span><span class="va">query</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/align-utils.html" class="external-link">compareStrings</a></span><span class="op">(</span><span class="op">)</span>

<span class="va">gb</span><span class="op">[</span><span class="va">gb</span><span class="op">$</span><span class="va">nedit</span><span class="op">==</span><span class="fl">0</span><span class="op">]</span><span class="op">$</span><span class="va">consensus</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/table.html" class="external-link">table</a></span><span class="op">(</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/sort.html" class="external-link">sort</a></span><span class="op">(</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">tail</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## </span>
<span class="co">## GCGTCG TCTAAA TGGTCA TTAACT  TTAAG TTTGAA </span>
<span class="co">##      1      1      1      1      1      1</span></code></pre>
<div class="sourceCode" id="cb48"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">gb</span><span class="op">[</span><span class="va">gb</span><span class="op">$</span><span class="va">nedit</span><span class="op">==</span><span class="fl">0</span><span class="op">]</span><span class="op">$</span><span class="va">consensus</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/table.html" class="external-link">table</a></span><span class="op">(</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## [1] 15</span></code></pre>
</div>
<div class="section level3">
<h3 id="how-frequent-is-the-inverted-pair-of-repeats-that-we-see">How frequent is the inverted pair of repeats that we see?<a class="anchor" aria-label="anchor" href="#how-frequent-is-the-inverted-pair-of-repeats-that-we-see"></a>
</h3>
<div class="sourceCode" id="cb50"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">checkTIR</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">gr</span>, <span class="va">reps</span><span class="op">)</span> <span class="op">{</span>
  <span class="va">genome</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">genome</a></span><span class="op">(</span><span class="va">gr</span><span class="op">)</span><span class="op">)</span>
  <span class="va">r</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/findOverlaps-methods.html" class="external-link">subsetByOverlaps</a></span><span class="op">(</span><span class="va">reps</span><span class="op">[[</span><span class="va">genome</span><span class="op">]</span><span class="op">]</span>, <span class="va">gr</span><span class="op">)</span>
  <span class="kw">if</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">r</span><span class="op">)</span> <span class="op">&lt;</span> <span class="fl">2</span><span class="op">)</span> <span class="kw"><a href="https://rdrr.io/r/base/function.html" class="external-link">return</a></span><span class="op">(</span><span class="cn">Inf</span><span class="op">)</span>
  <span class="kw">if</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/identical.html" class="external-link">identical</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/strand.html" class="external-link">strand</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">head</a></span><span class="op">(</span><span class="va">r</span>,<span class="fl">1</span><span class="op">)</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/strand.html" class="external-link">strand</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">tail</a></span><span class="op">(</span><span class="va">r</span>,<span class="fl">1</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="kw"><a href="https://rdrr.io/r/base/function.html" class="external-link">return</a></span><span class="op">(</span><span class="cn">Inf</span><span class="op">)</span>
  <span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/stringDist.html" class="external-link">stringDist</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">head</a></span><span class="op">(</span><span class="va">r</span>, <span class="fl">1</span><span class="op">)</span><span class="op">$</span><span class="va">Target</span>, <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">tail</a></span><span class="op">(</span><span class="va">r</span>, <span class="fl">1</span><span class="op">)</span><span class="op">$</span><span class="va">Target</span><span class="op">)</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/r/base/numeric.html" class="external-link">as.numeric</a></span><span class="op">(</span><span class="op">)</span>
<span class="op">}</span>

<span class="va">nameTIR</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">gr</span>, <span class="va">reps</span><span class="op">)</span> <span class="op">{</span>
  <span class="va">genome</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">genome</a></span><span class="op">(</span><span class="va">gr</span><span class="op">)</span><span class="op">)</span>
  <span class="va">r</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/findOverlaps-methods.html" class="external-link">subsetByOverlaps</a></span><span class="op">(</span><span class="va">reps</span><span class="op">[[</span><span class="va">genome</span><span class="op">]</span><span class="op">]</span>, <span class="va">gr</span><span class="op">)</span>
  <span class="kw">if</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">r</span><span class="op">)</span> <span class="op">&lt;</span> <span class="fl">2</span><span class="op">)</span> <span class="kw"><a href="https://rdrr.io/r/base/function.html" class="external-link">return</a></span><span class="op">(</span><span class="cn">NA</span><span class="op">)</span>
  <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">head</a></span><span class="op">(</span><span class="va">r</span>, <span class="fl">1</span><span class="op">)</span><span class="op">$</span><span class="va">Target</span>, <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">tail</a></span><span class="op">(</span><span class="va">r</span>, <span class="fl">1</span><span class="op">)</span><span class="op">$</span><span class="va">Target</span><span class="op">)</span>
<span class="op">}</span>

<span class="va">gb</span><span class="op">$</span><span class="va">hasTIR</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq_along</a></span><span class="op">(</span><span class="va">gb</span><span class="op">)</span>, \<span class="op">(</span><span class="va">n</span><span class="op">)</span> <span class="fu">checkTIR</span><span class="op">(</span><span class="va">gb</span><span class="op">$</span><span class="va">query</span><span class="op">[</span><span class="va">n</span><span class="op">]</span>, <span class="va">reps</span><span class="op">)</span><span class="op">)</span>
<span class="co">#gb$TIRname &lt;- sapply(seq_along(gb), \(n) nameTIR(gb$query[n], reps)) #crashes</span>
<span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/which.html" class="external-link">which</a></span><span class="op">(</span><span class="va">gb</span><span class="op">$</span><span class="va">hasTIR</span> <span class="op">&lt;=</span> <span class="fl">4</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">##  [1]  36  43  45  52  71  80  83  87  90  95  97 104 113 118 128 131 150 154</span></code></pre>
<div class="sourceCode" id="cb52"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">gb</span><span class="op">$</span><span class="va">query</span><span class="op">[</span><span class="fl">28</span><span class="op">]</span> |&gt; <span class="fu">alignToSelf</span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Global PairwiseAlignmentsSingleSubject (1 of 1)</span>
<span class="co">## pattern: <span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">TTTT</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">AA</span>---------<span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #AFFFFF;">GG</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">TTT</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">AAA</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">CC</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">CCCCCC</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">TTTT</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #AFFFFF;">GG</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #AFFFFF;">G</span></span>
<span class="co">## subject: <span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">CC</span><span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">AAAA</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">GGGGGG</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #AFFFFF;">GG</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">TTT</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">AAA</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">CC</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #FFAFAF;">A</span>---------<span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">AAAA</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span></span>
<span class="co">## score: -144.6337</span></code></pre>
<div class="sourceCode" id="cb54"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">gb</span><span class="op">$</span><span class="va">query</span><span class="op">[</span><span class="fl">28</span><span class="op">]</span> |&gt; <span class="fu">searchForTSDs</span><span class="op">(</span><span class="fl">20</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Global PairwiseAlignmentsSingleSubject (1 of 1)</span>
<span class="co">## pattern: <span style="color: #000000; background-color: #AFFFFF;">G</span>--<span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">TTTT</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #AFFFFF;">GG</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">TT</span></span>
<span class="co">## subject: <span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #AFFFFF;">GGG</span>--</span>
<span class="co">## score: -71.25777</span></code></pre>
<div class="sourceCode" id="cb56"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># But also in regions not flagged with hasTIR:</span>
<span class="va">gb</span><span class="op">$</span><span class="va">query</span><span class="op">[</span><span class="fl">20</span><span class="op">]</span> |&gt; <span class="fu">alignToSelf</span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Global PairwiseAlignmentsSingleSubject (1 of 1)</span>
<span class="co">## pattern: <span style="color: #000000; background-color: #AFFFFF;">GG</span>-<span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">A</span>--<span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">GGGGG</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">TTT</span><span style="color: #000000; background-color: #AFFFAF;">CC</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">CCCC</span>...<span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">CCCCCC</span><span style="color: #000000; background-color: #AFFFFF;">G</span>-----<span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">CCCCCC</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">CC</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">GG</span><span style="color: #000000; background-color: #AFFFAF;">C</span></span>
<span class="co">## subject: <span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">CC</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #AFFFFF;">GG</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">GGGGGG</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #AFFFFF;">GG</span>-----<span style="color: #000000; background-color: #AFFFFF;">GGGG</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">A</span>...<span style="color: #000000; background-color: #AFFFFF;">GGGG</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">GG</span><span style="color: #000000; background-color: #FFAFAF;">AAA</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #AFFFAF;">CCCCC</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">C</span>--<span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">C</span>-<span style="color: #000000; background-color: #AFFFAF;">C</span></span>
<span class="co">## score: -231.4793</span></code></pre>
<div class="sourceCode" id="cb58"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">gb</span><span class="op">$</span><span class="va">query</span><span class="op">[</span><span class="fl">20</span><span class="op">]</span> |&gt; <span class="fu">searchForTSDs</span><span class="op">(</span><span class="fl">20</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Global PairwiseAlignmentsSingleSubject (1 of 1)</span>
<span class="co">## pattern: --<span style="color: #000000; background-color: #AFFFFF;">GG</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">GGGGG</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">TTT</span></span>
<span class="co">## subject: <span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">AAA</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #AFFFFF;">GGG</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">C</span>--<span style="color: #000000; background-color: #FFD7AF;">TT</span></span>
<span class="co">## score: -47.61464</span></code></pre>
<div class="sourceCode" id="cb60"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># TSDs with short TIR on Oki_Osa</span>
<span class="va">unalMap</span><span class="op">$</span><span class="va">Oki_Osa</span><span class="op">[</span><span class="fu">isIndel</span><span class="op">(</span><span class="va">unalMap</span><span class="op">$</span><span class="va">Oki_Osa</span><span class="op">)</span><span class="op">]</span><span class="op">[</span><span class="fl">30</span><span class="op">]</span><span class="op">$</span><span class="va">query</span> |&gt; <span class="fu">searchForTSDs</span><span class="op">(</span><span class="fl">20</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Global PairwiseAlignmentsSingleSubject (1 of 1)</span>
<span class="co">## pattern: <span style="color: #000000; background-color: #AFFFAF;">CC</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">GG</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span>--<span style="color: #000000; background-color: #AFFFFF;">GG</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">A</span></span>
<span class="co">## subject: <span style="color: #000000; background-color: #FFAFAF;">A</span>-<span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">A</span>-<span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #AFFFFF;">G</span></span>
<span class="co">## score: -73.37673</span></code></pre>
<div class="sourceCode" id="cb62"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">unalMap</span><span class="op">$</span><span class="va">Oki_Osa</span><span class="op">[</span><span class="fu">isIndel</span><span class="op">(</span><span class="va">unalMap</span><span class="op">$</span><span class="va">Oki_Osa</span><span class="op">)</span><span class="op">]</span><span class="op">[</span><span class="fl">30</span><span class="op">]</span><span class="op">$</span><span class="va">query</span> |&gt; <span class="fu">alignToSelf</span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Global PairwiseAlignmentsSingleSubject (1 of 1)</span>
<span class="co">## pattern: <span style="color: #000000; background-color: #AFFFAF;">CC</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">GG</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #AFFFFF;">GG</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">CC</span><span style="color: #000000; background-color: #AFFFFF;">GG</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">AA</span>...------<span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">CC</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span>------------<span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span>-----</span>
<span class="co">## subject: -----<span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span>------------<span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">GG</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #AFFFAF;">C</span>-----...<span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">CC</span><span style="color: #000000; background-color: #AFFFFF;">GG</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">CC</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">CC</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #AFFFFF;">GG</span></span>
<span class="co">## score: -473.3667</span></code></pre>
<div class="sourceCode" id="cb64"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">unalMap</span><span class="op">$</span><span class="va">Oki_Osa</span><span class="op">[</span><span class="fu">isIndel</span><span class="op">(</span><span class="va">unalMap</span><span class="op">$</span><span class="va">Oki_Osa</span><span class="op">)</span><span class="op">]</span><span class="op">[</span><span class="fl">3</span><span class="op">]</span><span class="op">$</span><span class="va">query</span> |&gt; <span class="fu">searchForTSDs</span><span class="op">(</span><span class="fl">20</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Global PairwiseAlignmentsSingleSubject (1 of 1)</span>
<span class="co">## pattern: <span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">TTTTT</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">CC</span><span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #FFD7AF;">TTT</span><span style="color: #000000; background-color: #AFFFFF;">GG</span>---</span>
<span class="co">## subject: <span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">TTTTT</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">TT</span>---<span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #FFD7AF;">TTT</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">T</span></span>
<span class="co">## score: -41.83432</span></code></pre>
<div class="sourceCode" id="cb66"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">unalMap</span><span class="op">$</span><span class="va">Oki_Osa</span><span class="op">[</span><span class="fu">isIndel</span><span class="op">(</span><span class="va">unalMap</span><span class="op">$</span><span class="va">Oki_Osa</span><span class="op">)</span><span class="op">]</span><span class="op">[</span><span class="fl">3</span><span class="op">]</span><span class="op">$</span><span class="va">query</span> |&gt; <span class="fu">alignToSelf</span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Global PairwiseAlignmentsSingleSubject (1 of 1)</span>
<span class="co">## pattern: <span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">TTTTT</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">CC</span><span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #FFD7AF;">TTT</span><span style="color: #000000; background-color: #AFFFFF;">GG</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">TTT</span><span style="color: #000000; background-color: #FFAFAF;">A</span>---<span style="color: #000000; background-color: #FFD7AF;">TT</span>...<span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">GG</span><span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">TTT</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">TTT</span><span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">C</span>---<span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">TTT</span><span style="color: #000000; background-color: #AFFFAF;">C</span></span>
<span class="co">## subject: <span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">AAA</span><span style="color: #000000; background-color: #FFD7AF;">T</span>---<span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #FFAFAF;">AAA</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">AAA</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #AFFFAF;">CC</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">TT</span>...<span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">AAA</span><span style="color: #000000; background-color: #AFFFFF;">G</span>-----<span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">CC</span><span style="color: #000000; background-color: #FFAFAF;">AAA</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #AFFFFF;">GG</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">AAAAA</span><span style="color: #000000; background-color: #AFFFFF;">G</span></span>
<span class="co">## score: -387.543</span></code></pre>
<div class="sourceCode" id="cb68"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">kindOfLogScale</span> <span class="op">&lt;-</span> <span class="fl">2</span><span class="op">^</span><span class="op">(</span><span class="fl">0</span><span class="op">:</span><span class="fl">100</span> <span class="op">/</span> <span class="fl">8</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/r/base/Round.html" class="external-link">round</a></span><span class="op">(</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="op">)</span>
<span class="va">m</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">kindOfLogScale</span>, \<span class="op">(</span><span class="va">n</span><span class="op">)</span> <span class="va">gb</span><span class="op">$</span><span class="va">query</span> |&gt; <span class="fu">alignEnds</span><span class="op">(</span><span class="va">n</span>, r<span class="op">=</span><span class="cn">T</span><span class="op">)</span> |&gt; <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/score.html" class="external-link">score</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="http://yulab-smu.top/ggmsa/reference/plot-methods.html" class="external-link">plot</a></span><span class="op">(</span> <span class="fu"><a href="https://rdrr.io/r/base/apply.html" class="external-link">apply</a></span><span class="op">(</span><span class="va">m</span>,<span class="fl">1</span>, \<span class="op">(</span><span class="va">v</span><span class="op">)</span> <span class="va">kindOfLogScale</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/which.min.html" class="external-link">which.max</a></span><span class="op">(</span><span class="va">v</span><span class="op">)</span><span class="op">]</span><span class="op">)</span>
    , <span class="fu"><a href="https://rdrr.io/r/base/apply.html" class="external-link">apply</a></span><span class="op">(</span><span class="va">m</span>,<span class="fl">1</span>, \<span class="op">(</span><span class="va">v</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html" class="external-link">max</a></span><span class="op">(</span><span class="va">v</span><span class="op">)</span><span class="op">)</span>
    , xlab <span class="op">=</span> <span class="st">"At which length did the alignment score peak?"</span>
    , ylab <span class="op">=</span> <span class="st">"What was the maximum score?"</span><span class="op">)</span></code></pre></div>
<p><img src="TandemSiteDuplications_files/figure-html/compute_and_plot_score_stats-1.png" width="700"></p>
<div class="sourceCode" id="cb69"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="http://yulab-smu.top/ggmsa/reference/plot-methods.html" class="external-link">plot</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/table.html" class="external-link">table</a></span><span class="op">(</span> <span class="fu"><a href="https://rdrr.io/r/base/apply.html" class="external-link">apply</a></span><span class="op">(</span><span class="va">m</span>,<span class="fl">1</span>, \<span class="op">(</span><span class="va">v</span><span class="op">)</span> <span class="va">kindOfLogScale</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/which.min.html" class="external-link">which.max</a></span><span class="op">(</span><span class="va">v</span><span class="op">)</span><span class="op">]</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="TandemSiteDuplications_files/figure-html/compute_and_plot_score_stats-2.png" width="700"></p>
<div class="sourceCode" id="cb70"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">mm</span> <span class="op">&lt;-</span> <span class="va">m</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/apply.html" class="external-link">apply</a></span><span class="op">(</span><span class="va">m</span>,<span class="fl">1</span>, \<span class="op">(</span><span class="va">v</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html" class="external-link">max</a></span><span class="op">(</span><span class="va">v</span><span class="op">)</span><span class="op">)</span> <span class="op">&gt;</span> <span class="fl">6</span>,<span class="op">]</span>
<span class="fu"><a href="http://yulab-smu.top/ggmsa/reference/plot-methods.html" class="external-link">plot</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/table.html" class="external-link">table</a></span><span class="op">(</span> <span class="fu"><a href="https://rdrr.io/r/base/apply.html" class="external-link">apply</a></span><span class="op">(</span><span class="va">mm</span>,<span class="fl">1</span>, \<span class="op">(</span><span class="va">v</span><span class="op">)</span> <span class="va">kindOfLogScale</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/which.min.html" class="external-link">which.max</a></span><span class="op">(</span><span class="va">v</span><span class="op">)</span><span class="op">]</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="TandemSiteDuplications_files/figure-html/compute_and_plot_score_stats-3.png" width="700"></p>
<div class="sourceCode" id="cb71"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/apply.html" class="external-link">apply</a></span><span class="op">(</span><span class="va">m</span>,<span class="fl">1</span>, \<span class="op">(</span><span class="va">v</span><span class="op">)</span> <span class="va">kindOfLogScale</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/which.min.html" class="external-link">which.max</a></span><span class="op">(</span><span class="va">v</span><span class="op">)</span><span class="op">]</span><span class="op">)</span> |&gt;
  <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/funprog.html" class="external-link">Filter</a></span><span class="op">(</span>f<span class="op">=</span>\<span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="va">x</span><span class="op">&gt;</span><span class="fl">7</span><span class="op">)</span> |&gt;
  <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/table.html" class="external-link">table</a></span><span class="op">(</span><span class="op">)</span> |&gt; <span class="fu"><a href="http://yulab-smu.top/ggmsa/reference/plot-methods.html" class="external-link">plot</a></span><span class="op">(</span>log<span class="op">=</span><span class="st">"x"</span>, ylab <span class="op">=</span> <span class="st">"Frequency"</span>,
                  xlab <span class="op">=</span> <span class="st">"Alignment length where score peaks (starting from n=8)"</span>,
                  main <span class="op">=</span> <span class="st">"Aproximate length of TIRs"</span><span class="op">)</span></code></pre></div>
<p><img src="TandemSiteDuplications_files/figure-html/compute_and_plot_score_stats-4.png" width="700"></p>
<div class="sourceCode" id="cb72"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">gb</span><span class="op">$</span><span class="va">hasScorePeak</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/apply.html" class="external-link">apply</a></span><span class="op">(</span><span class="va">m</span>,<span class="fl">1</span>, \<span class="op">(</span><span class="va">v</span><span class="op">)</span> <span class="va">kindOfLogScale</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/which.min.html" class="external-link">which.max</a></span><span class="op">(</span><span class="va">v</span><span class="op">)</span><span class="op">]</span><span class="op">)</span> <span class="op">&gt;</span> <span class="fl">8</span></code></pre></div>
</div>
<div class="section level3">
<h3 id="remaining-questions">Remaining questions<a class="anchor" aria-label="anchor" href="#remaining-questions"></a>
</h3>
<p>Questions:</p>
<ul>
<li>Is there any hint of functionality for the sequence in the gap?</li>
<li>are the indels fixed in their respective populations?</li>
<li>Am I missing half of the microhomologies because of alignments to negative strands ?</li>
</ul>
</div>
</div>
<div class="section level2">
<h2 id="session-information">Session information<a class="anchor" aria-label="anchor" href="#session-information"></a>
</h2>
<div class="sourceCode" id="cb73"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## R version 4.1.2 (2021-11-01)</span>
<span class="co">## Platform: x86_64-pc-linux-gnu (64-bit)</span>
<span class="co">## Running under: Debian GNU/Linux bookworm/sid</span>
<span class="co">## </span>
<span class="co">## Matrix products: default</span>
<span class="co">## BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.10.0</span>
<span class="co">## LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.10.0</span>
<span class="co">## </span>
<span class="co">## locale:</span>
<span class="co">##  [1] LC_CTYPE=en_GB.UTF-8       LC_NUMERIC=C              </span>
<span class="co">##  [3] LC_TIME=en_GB.UTF-8        LC_COLLATE=en_GB.UTF-8    </span>
<span class="co">##  [5] LC_MONETARY=en_GB.UTF-8    LC_MESSAGES=en_GB.UTF-8   </span>
<span class="co">##  [7] LC_PAPER=en_GB.UTF-8       LC_NAME=C                 </span>
<span class="co">##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span>
<span class="co">## [11] LC_MEASUREMENT=en_GB.UTF-8 LC_IDENTIFICATION=C       </span>
<span class="co">## </span>
<span class="co">## attached base packages:</span>
<span class="co">## [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span>
<span class="co">## [8] base     </span>
<span class="co">## </span>
<span class="co">## other attached packages:</span>
<span class="co">##  [1] BSgenome.Oidioi.genoscope.OdB3_1.0.0   </span>
<span class="co">##  [2] BSgenome.Oidioi.OIST.AOM.5.5f_1.0.1    </span>
<span class="co">##  [3] BSgenome.Oidioi.OIST.KUM.M3.7f_1.0.1   </span>
<span class="co">##  [4] BSgenome.Oidioi.OIST.Bar2.p4_1.0.1     </span>
<span class="co">##  [5] BSgenome.Oidioi.OIST.OSKA2016v1.9_1.0.0</span>
<span class="co">##  [6] BSgenome.Oidioi.OIST.OKI2018.I69_1.0.1 </span>
<span class="co">##  [7] BSgenome_1.62.0                        </span>
<span class="co">##  [8] rtracklayer_1.54.0                     </span>
<span class="co">##  [9] Biostrings_2.62.0                      </span>
<span class="co">## [10] XVector_0.34.0                         </span>
<span class="co">## [11] BreakpointsData_3.8.0                  </span>
<span class="co">## [12] ggplot2_3.3.5                          </span>
<span class="co">## [13] GenomicBreaks_0.13.1                   </span>
<span class="co">## [14] GenomicRanges_1.46.1                   </span>
<span class="co">## [15] GenomeInfoDb_1.30.1                    </span>
<span class="co">## [16] IRanges_2.28.0                         </span>
<span class="co">## [17] S4Vectors_0.32.3                       </span>
<span class="co">## [18] BiocGenerics_0.40.0                    </span>
<span class="co">## </span>
<span class="co">## loaded via a namespace (and not attached):</span>
<span class="co">##   [1] backports_1.4.1             Hmisc_4.6-0                </span>
<span class="co">##   [3] BiocFileCache_2.2.1         systemfonts_1.0.3          </span>
<span class="co">##   [5] plyr_1.8.6                  splines_4.1.2              </span>
<span class="co">##   [7] BiocParallel_1.28.3         digest_0.6.29              </span>
<span class="co">##   [9] htmltools_0.5.2             tiff_0.1-11                </span>
<span class="co">##  [11] GO.db_3.14.0                gdata_2.18.0               </span>
<span class="co">##  [13] fansi_1.0.2                 magrittr_2.0.2             </span>
<span class="co">##  [15] checkmate_2.0.0             memoise_2.0.1              </span>
<span class="co">##  [17] cluster_2.1.2               tzdb_0.2.0                 </span>
<span class="co">##  [19] readr_2.1.2                 annotate_1.72.0            </span>
<span class="co">##  [21] matrixStats_0.61.0          R.utils_2.11.0             </span>
<span class="co">##  [23] pkgdown_2.0.2               prettyunits_1.1.1          </span>
<span class="co">##  [25] jpeg_0.1-9                  colorspace_2.0-3           </span>
<span class="co">##  [27] rappdirs_0.3.3              blob_1.2.2                 </span>
<span class="co">##  [29] textshaping_0.3.6           xfun_0.30                  </span>
<span class="co">##  [31] dplyr_1.0.8                 crayon_1.5.0               </span>
<span class="co">##  [33] RCurl_1.98-1.6              jsonlite_1.8.0             </span>
<span class="co">##  [35] lme4_1.1-28                 survival_3.2-13            </span>
<span class="co">##  [37] glue_1.6.2                  gtable_0.3.0               </span>
<span class="co">##  [39] zlibbioc_1.40.0             DelayedArray_0.20.0        </span>
<span class="co">##  [41] plyranges_1.14.0            weights_1.0.4              </span>
<span class="co">##  [43] abind_1.4-5                 scales_1.1.1               </span>
<span class="co">##  [45] DBI_1.1.2                   Rcpp_1.0.8                 </span>
<span class="co">##  [47] plotrix_3.8-2               progress_1.2.2             </span>
<span class="co">##  [49] genoPlotR_0.8.11            xtable_1.8-4               </span>
<span class="co">##  [51] htmlTable_2.4.0             foreign_0.8-81             </span>
<span class="co">##  [53] bit_4.0.4                   Formula_1.2-4              </span>
<span class="co">##  [55] htmlwidgets_1.5.4           httr_1.4.2                 </span>
<span class="co">##  [57] RColorBrewer_1.1-2          ellipsis_0.3.2             </span>
<span class="co">##  [59] mice_3.14.0                 pkgconfig_2.0.3            </span>
<span class="co">##  [61] XML_3.99-0.9                R.methodsS3_1.8.1          </span>
<span class="co">##  [63] dbplyr_2.1.1                nnet_7.3-16                </span>
<span class="co">##  [65] sass_0.4.0                  locfit_1.5-9.5             </span>
<span class="co">##  [67] utf8_1.2.2                  tidyselect_1.1.2           </span>
<span class="co">##  [69] rlang_1.0.2                 reshape2_1.4.4             </span>
<span class="co">##  [71] AnnotationDbi_1.56.2        munsell_0.5.0              </span>
<span class="co">##  [73] tools_4.1.2                 cachem_1.0.6               </span>
<span class="co">##  [75] cli_3.2.0                   generics_0.1.2             </span>
<span class="co">##  [77] RSQLite_2.2.10              ade4_1.7-18                </span>
<span class="co">##  [79] broom_0.7.12                OikScrambling_2.1.0        </span>
<span class="co">##  [81] evaluate_0.15               stringr_1.4.0              </span>
<span class="co">##  [83] fastmap_1.1.0               fftwtools_0.9-11           </span>
<span class="co">##  [85] yaml_2.3.5                  ragg_1.2.1                 </span>
<span class="co">##  [87] knitr_1.37                  bit64_4.0.5                </span>
<span class="co">##  [89] fs_1.5.2                    purrr_0.3.4                </span>
<span class="co">##  [91] heatmaps_1.18.0             KEGGREST_1.34.0            </span>
<span class="co">##  [93] nlme_3.1-153                R.oo_1.24.0                </span>
<span class="co">##  [95] poweRlaw_0.70.6             xml2_1.3.3                 </span>
<span class="co">##  [97] pracma_2.3.8                biomaRt_2.50.3             </span>
<span class="co">##  [99] compiler_4.1.2              rstudioapi_0.13            </span>
<span class="co">## [101] filelock_1.0.2              curl_4.3.2                 </span>
<span class="co">## [103] png_0.1-7                   msa_1.26.0                 </span>
<span class="co">## [105] tibble_3.1.6                bslib_0.3.1                </span>
<span class="co">## [107] stringi_1.7.6               highr_0.9                  </span>
<span class="co">## [109] GenomicFeatures_1.46.5      desc_1.4.1                 </span>
<span class="co">## [111] lattice_0.20-45             CNEr_1.30.0                </span>
<span class="co">## [113] Matrix_1.4-0                nloptr_1.2.2.3             </span>
<span class="co">## [115] vctrs_0.3.8                 pillar_1.7.0               </span>
<span class="co">## [117] lifecycle_1.0.1             jquerylib_0.1.4            </span>
<span class="co">## [119] data.table_1.14.2           bitops_1.0-7               </span>
<span class="co">## [121] R6_2.5.1                    BiocIO_1.4.0               </span>
<span class="co">## [123] latticeExtra_0.6-29         KernSmooth_2.23-20         </span>
<span class="co">## [125] gridExtra_2.3               codetools_0.2-18           </span>
<span class="co">## [127] boot_1.3-28                 gtools_3.9.2               </span>
<span class="co">## [129] MASS_7.3-54                 assertthat_0.2.1           </span>
<span class="co">## [131] SummarizedExperiment_1.24.0 rprojroot_2.0.2            </span>
<span class="co">## [133] rjson_0.2.21                withr_2.5.0                </span>
<span class="co">## [135] GenomicAlignments_1.30.0    Rsamtools_2.10.0           </span>
<span class="co">## [137] GenomeInfoDbData_1.2.7      parallel_4.1.2             </span>
<span class="co">## [139] hms_1.1.1                   EBImage_4.36.0             </span>
<span class="co">## [141] grid_4.1.2                  rpart_4.1-15               </span>
<span class="co">## [143] minqa_1.2.4                 tidyr_1.2.0                </span>
<span class="co">## [145] rmarkdown_2.11              MatrixGenerics_1.6.0       </span>
<span class="co">## [147] Biobase_2.54.0              base64enc_0.1-3            </span>
<span class="co">## [149] restfulr_0.0.13</span></code></pre>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Aleksandra Bliznina, Michael Mansfield, Charles Plessy.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.2.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
